{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from dotenv import load_dotenv\n",
    "\n",
    "load_dotenv('/workspaces/ml-learning/.env')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import logging\n",
    "import sys\n",
    "\n",
    "logging.basicConfig(stream=sys.stdout, level=logging.WARNING)\n",
    "logging.getLogger().addHandler(logging.StreamHandler(stream=sys.stdout))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import nest_asyncio\n",
    "\n",
    "nest_asyncio.apply()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index import (\n",
    "    SimpleDirectoryReader,\n",
    "    ServiceContext,\n",
    "    get_response_synthesizer,\n",
    ")\n",
    "from llama_index.indices.document_summary import DocumentSummaryIndex\n",
    "from llama_index.llms import OpenAI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: arxiv in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (2.1.0)\n",
      "Requirement already satisfied: feedparser==6.0.10 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from arxiv) (6.0.10)\n",
      "Requirement already satisfied: requests==2.31.0 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from arxiv) (2.31.0)\n",
      "Requirement already satisfied: sgmllib3k in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from feedparser==6.0.10->arxiv) (1.0.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from requests==2.31.0->arxiv) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from requests==2.31.0->arxiv) (3.6)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from requests==2.31.0->arxiv) (2.1.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from requests==2.31.0->arxiv) (2023.11.17)\n"
     ]
    }
   ],
   "source": [
    "!pip install arxiv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<itertools.islice object at 0x7fbb41c48d10>\n"
     ]
    },
    {
     "ename": "SSLError",
     "evalue": "HTTPSConnectionPool(host='export.arxiv.org', port=443): Max retries exceeded with url: /api/query?search_query=all%3A%22phages%22+AND+all%3A%22machine+learning%22&id_list=&sortBy=submittedDate&sortOrder=descending&start=0&max_results=100 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1002)')))",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mSSLEOFError\u001b[0m                               Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:467\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    466\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 467\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_validate_conn\u001b[49m\u001b[43m(\u001b[49m\u001b[43mconn\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    468\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (SocketTimeout, BaseSSLError) \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:1096\u001b[0m, in \u001b[0;36mHTTPSConnectionPool._validate_conn\u001b[0;34m(self, conn)\u001b[0m\n\u001b[1;32m   1095\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_closed:\n\u001b[0;32m-> 1096\u001b[0m     \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1098\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_verified:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connection.py:642\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    634\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[1;32m    635\u001b[0m         (\n\u001b[1;32m    636\u001b[0m             \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSystem time is way off (before \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mRECENT_DATE\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m). This will probably \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    639\u001b[0m         SystemTimeWarning,\n\u001b[1;32m    640\u001b[0m     )\n\u001b[0;32m--> 642\u001b[0m sock_and_verified \u001b[38;5;241m=\u001b[39m \u001b[43m_ssl_wrap_socket_and_match_hostname\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    643\u001b[0m \u001b[43m    \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    644\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcert_reqs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcert_reqs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    645\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    646\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_minimum_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_minimum_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    647\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_maximum_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_maximum_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    648\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_certs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_certs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    649\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_cert_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    650\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_cert_data\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    651\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcert_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcert_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    652\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkey_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    653\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_password\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkey_password\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    654\u001b[0m \u001b[43m    \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    655\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_context\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    656\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    657\u001b[0m \u001b[43m    \u001b[49m\u001b[43massert_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43massert_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    658\u001b[0m \u001b[43m    \u001b[49m\u001b[43massert_fingerprint\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43massert_fingerprint\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    659\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    660\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock \u001b[38;5;241m=\u001b[39m sock_and_verified\u001b[38;5;241m.\u001b[39msocket\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connection.py:782\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_and_match_hostname\u001b[0;34m(sock, cert_reqs, ssl_version, ssl_minimum_version, ssl_maximum_version, cert_file, key_file, key_password, ca_certs, ca_cert_dir, ca_cert_data, assert_hostname, assert_fingerprint, server_hostname, ssl_context, tls_in_tls)\u001b[0m\n\u001b[1;32m    780\u001b[0m         server_hostname \u001b[38;5;241m=\u001b[39m normalized\n\u001b[0;32m--> 782\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m \u001b[43mssl_wrap_socket\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    783\u001b[0m \u001b[43m    \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    784\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkeyfile\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkey_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    785\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcertfile\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcert_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    786\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_password\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkey_password\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    787\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_certs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_certs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    788\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_cert_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    789\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_cert_data\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    790\u001b[0m \u001b[43m    \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontext\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/ssl_.py:470\u001b[0m, in \u001b[0;36mssl_wrap_socket\u001b[0;34m(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data, tls_in_tls)\u001b[0m\n\u001b[1;32m    468\u001b[0m     \u001b[38;5;28;01mpass\u001b[39;00m\n\u001b[0;32m--> 470\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m \u001b[43m_ssl_wrap_socket_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcontext\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    471\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ssl_sock\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/ssl_.py:514\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_impl\u001b[0;34m(sock, ssl_context, tls_in_tls, server_hostname)\u001b[0m\n\u001b[1;32m    512\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m SSLTransport(sock, ssl_context, server_hostname)\n\u001b[0;32m--> 514\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwrap_socket\u001b[49m\u001b[43m(\u001b[49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:517\u001b[0m, in \u001b[0;36mSSLContext.wrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[1;32m    511\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrap_socket\u001b[39m(\u001b[38;5;28mself\u001b[39m, sock, server_side\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[1;32m    512\u001b[0m                 do_handshake_on_connect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    513\u001b[0m                 suppress_ragged_eofs\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    514\u001b[0m                 server_hostname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, session\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    515\u001b[0m     \u001b[38;5;66;03m# SSLSocket class handles server_hostname encoding before it calls\u001b[39;00m\n\u001b[1;32m    516\u001b[0m     \u001b[38;5;66;03m# ctx._wrap_socket()\u001b[39;00m\n\u001b[0;32m--> 517\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msslsocket_class\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_create\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    518\u001b[0m \u001b[43m        \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    519\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_side\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_side\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    520\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdo_handshake_on_connect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdo_handshake_on_connect\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    521\u001b[0m \u001b[43m        \u001b[49m\u001b[43msuppress_ragged_eofs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msuppress_ragged_eofs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    522\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    523\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcontext\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    524\u001b[0m \u001b[43m        \u001b[49m\u001b[43msession\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msession\u001b[49m\n\u001b[1;32m    525\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1075\u001b[0m, in \u001b[0;36mSSLSocket._create\u001b[0;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[1;32m   1074\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdo_handshake_on_connect should not be specified for non-blocking sockets\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m-> 1075\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdo_handshake\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1076\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\u001b[38;5;167;01mOSError\u001b[39;00m, \u001b[38;5;167;01mValueError\u001b[39;00m):\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1346\u001b[0m, in \u001b[0;36mSSLSocket.do_handshake\u001b[0;34m(self, block)\u001b[0m\n\u001b[1;32m   1345\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msettimeout(\u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[0;32m-> 1346\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mdo_handshake()\n\u001b[1;32m   1347\u001b[0m \u001b[38;5;28;01mfinally\u001b[39;00m:\n",
      "\u001b[0;31mSSLEOFError\u001b[0m: EOF occurred in violation of protocol (_ssl.c:1002)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:790\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    789\u001b[0m \u001b[38;5;66;03m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[0;32m--> 790\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    794\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout_obj\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    796\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    797\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    798\u001b[0m \u001b[43m    \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mretries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    799\u001b[0m \u001b[43m    \u001b[49m\u001b[43mresponse_conn\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresponse_conn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    800\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpreload_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    801\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdecode_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    802\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kw\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    803\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    805\u001b[0m \u001b[38;5;66;03m# Everything went great!\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:491\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    490\u001b[0m         new_e \u001b[38;5;241m=\u001b[39m _wrap_proxy_error(new_e, conn\u001b[38;5;241m.\u001b[39mproxy\u001b[38;5;241m.\u001b[39mscheme)\n\u001b[0;32m--> 491\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m new_e\n\u001b[1;32m    493\u001b[0m \u001b[38;5;66;03m# conn.request() calls http.client.*.request, not the method in\u001b[39;00m\n\u001b[1;32m    494\u001b[0m \u001b[38;5;66;03m# urllib3.request. It also calls makefile (recv) on the socket.\u001b[39;00m\n",
      "\u001b[0;31mSSLError\u001b[0m: EOF occurred in violation of protocol (_ssl.c:1002)",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/adapters.py:486\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    485\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 486\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murlopen\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    487\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    488\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    489\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    490\u001b[0m \u001b[43m        \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    491\u001b[0m \u001b[43m        \u001b[49m\u001b[43mredirect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    492\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_same_host\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    493\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    494\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    495\u001b[0m \u001b[43m        \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    496\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    497\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    498\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    500\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:844\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    842\u001b[0m     new_e \u001b[38;5;241m=\u001b[39m ProtocolError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConnection aborted.\u001b[39m\u001b[38;5;124m\"\u001b[39m, new_e)\n\u001b[0;32m--> 844\u001b[0m retries \u001b[38;5;241m=\u001b[39m \u001b[43mretries\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mincrement\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    845\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43merror\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnew_e\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_pool\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_stacktrace\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msys\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mexc_info\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m2\u001b[39;49m\u001b[43m]\u001b[49m\n\u001b[1;32m    846\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    847\u001b[0m retries\u001b[38;5;241m.\u001b[39msleep()\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/retry.py:515\u001b[0m, in \u001b[0;36mRetry.increment\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    514\u001b[0m     reason \u001b[38;5;241m=\u001b[39m error \u001b[38;5;129;01mor\u001b[39;00m ResponseError(cause)\n\u001b[0;32m--> 515\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m MaxRetryError(_pool, url, reason) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mreason\u001b[39;00m  \u001b[38;5;66;03m# type: ignore[arg-type]\u001b[39;00m\n\u001b[1;32m    517\u001b[0m log\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIncremented Retry for (url=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, url, new_retry)\n",
      "\u001b[0;31mMaxRetryError\u001b[0m: HTTPSConnectionPool(host='export.arxiv.org', port=443): Max retries exceeded with url: /api/query?search_query=all%3A%22phages%22+AND+all%3A%22machine+learning%22&id_list=&sortBy=submittedDate&sortOrder=descending&start=0&max_results=100 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1002)')))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:634\u001b[0m, in \u001b[0;36mClient._parse_feed\u001b[0;34m(self, url, first_page, _try_index)\u001b[0m\n\u001b[1;32m    633\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 634\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m__try_parse_feed\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    635\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfirst_page\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfirst_page\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtry_index\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m_try_index\u001b[49m\n\u001b[1;32m    636\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    637\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\n\u001b[1;32m    638\u001b[0m     HTTPError,\n\u001b[1;32m    639\u001b[0m     UnexpectedEmptyPageError,\n\u001b[1;32m    640\u001b[0m     requests\u001b[38;5;241m.\u001b[39mexceptions\u001b[38;5;241m.\u001b[39mConnectionError,\n\u001b[1;32m    641\u001b[0m ) \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:674\u001b[0m, in \u001b[0;36mClient.__try_parse_feed\u001b[0;34m(self, url, first_page, try_index)\u001b[0m\n\u001b[1;32m    670\u001b[0m logger\u001b[38;5;241m.\u001b[39minfo(\n\u001b[1;32m    671\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRequesting page (first: \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m, try: \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, first_page, try_index, url\n\u001b[1;32m    672\u001b[0m )\n\u001b[0;32m--> 674\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_session\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m{\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43muser-agent\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43marxiv.py/2.1.0\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m}\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    675\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_last_request_dt \u001b[38;5;241m=\u001b[39m datetime\u001b[38;5;241m.\u001b[39mnow()\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:602\u001b[0m, in \u001b[0;36mSession.get\u001b[0;34m(self, url, **kwargs)\u001b[0m\n\u001b[1;32m    601\u001b[0m kwargs\u001b[38;5;241m.\u001b[39msetdefault(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m--> 602\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mGET\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprep\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43msend_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m \u001b[43madapter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/adapters.py:517\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    515\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(e\u001b[38;5;241m.\u001b[39mreason, _SSLError):\n\u001b[1;32m    516\u001b[0m     \u001b[38;5;66;03m# This branch is for urllib3 v1.22 and later.\u001b[39;00m\n\u001b[0;32m--> 517\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m SSLError(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[1;32m    519\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(e, request\u001b[38;5;241m=\u001b[39mrequest)\n",
      "\u001b[0;31mSSLError\u001b[0m: HTTPSConnectionPool(host='export.arxiv.org', port=443): Max retries exceeded with url: /api/query?search_query=all%3A%22phages%22+AND+all%3A%22machine+learning%22&id_list=&sortBy=submittedDate&sortOrder=descending&start=0&max_results=100 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1002)')))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLEOFError\u001b[0m                               Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:467\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    466\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 467\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_validate_conn\u001b[49m\u001b[43m(\u001b[49m\u001b[43mconn\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    468\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (SocketTimeout, BaseSSLError) \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:1096\u001b[0m, in \u001b[0;36mHTTPSConnectionPool._validate_conn\u001b[0;34m(self, conn)\u001b[0m\n\u001b[1;32m   1095\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_closed:\n\u001b[0;32m-> 1096\u001b[0m     \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1098\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_verified:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connection.py:642\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    634\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[1;32m    635\u001b[0m         (\n\u001b[1;32m    636\u001b[0m             \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSystem time is way off (before \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mRECENT_DATE\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m). This will probably \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    639\u001b[0m         SystemTimeWarning,\n\u001b[1;32m    640\u001b[0m     )\n\u001b[0;32m--> 642\u001b[0m sock_and_verified \u001b[38;5;241m=\u001b[39m \u001b[43m_ssl_wrap_socket_and_match_hostname\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    643\u001b[0m \u001b[43m    \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    644\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcert_reqs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcert_reqs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    645\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    646\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_minimum_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_minimum_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    647\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_maximum_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_maximum_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    648\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_certs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_certs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    649\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_cert_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    650\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_cert_data\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    651\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcert_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcert_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    652\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkey_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    653\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_password\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkey_password\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    654\u001b[0m \u001b[43m    \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    655\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_context\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    656\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    657\u001b[0m \u001b[43m    \u001b[49m\u001b[43massert_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43massert_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    658\u001b[0m \u001b[43m    \u001b[49m\u001b[43massert_fingerprint\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43massert_fingerprint\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    659\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    660\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock \u001b[38;5;241m=\u001b[39m sock_and_verified\u001b[38;5;241m.\u001b[39msocket\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connection.py:782\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_and_match_hostname\u001b[0;34m(sock, cert_reqs, ssl_version, ssl_minimum_version, ssl_maximum_version, cert_file, key_file, key_password, ca_certs, ca_cert_dir, ca_cert_data, assert_hostname, assert_fingerprint, server_hostname, ssl_context, tls_in_tls)\u001b[0m\n\u001b[1;32m    780\u001b[0m         server_hostname \u001b[38;5;241m=\u001b[39m normalized\n\u001b[0;32m--> 782\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m \u001b[43mssl_wrap_socket\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    783\u001b[0m \u001b[43m    \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    784\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkeyfile\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkey_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    785\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcertfile\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcert_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    786\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_password\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkey_password\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    787\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_certs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_certs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    788\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_cert_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    789\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_cert_data\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    790\u001b[0m \u001b[43m    \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontext\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/ssl_.py:470\u001b[0m, in \u001b[0;36mssl_wrap_socket\u001b[0;34m(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data, tls_in_tls)\u001b[0m\n\u001b[1;32m    468\u001b[0m     \u001b[38;5;28;01mpass\u001b[39;00m\n\u001b[0;32m--> 470\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m \u001b[43m_ssl_wrap_socket_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcontext\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    471\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ssl_sock\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/ssl_.py:514\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_impl\u001b[0;34m(sock, ssl_context, tls_in_tls, server_hostname)\u001b[0m\n\u001b[1;32m    512\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m SSLTransport(sock, ssl_context, server_hostname)\n\u001b[0;32m--> 514\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwrap_socket\u001b[49m\u001b[43m(\u001b[49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:517\u001b[0m, in \u001b[0;36mSSLContext.wrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[1;32m    511\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrap_socket\u001b[39m(\u001b[38;5;28mself\u001b[39m, sock, server_side\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[1;32m    512\u001b[0m                 do_handshake_on_connect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    513\u001b[0m                 suppress_ragged_eofs\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    514\u001b[0m                 server_hostname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, session\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    515\u001b[0m     \u001b[38;5;66;03m# SSLSocket class handles server_hostname encoding before it calls\u001b[39;00m\n\u001b[1;32m    516\u001b[0m     \u001b[38;5;66;03m# ctx._wrap_socket()\u001b[39;00m\n\u001b[0;32m--> 517\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msslsocket_class\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_create\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    518\u001b[0m \u001b[43m        \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    519\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_side\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_side\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    520\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdo_handshake_on_connect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdo_handshake_on_connect\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    521\u001b[0m \u001b[43m        \u001b[49m\u001b[43msuppress_ragged_eofs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msuppress_ragged_eofs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    522\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    523\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcontext\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    524\u001b[0m \u001b[43m        \u001b[49m\u001b[43msession\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msession\u001b[49m\n\u001b[1;32m    525\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1075\u001b[0m, in \u001b[0;36mSSLSocket._create\u001b[0;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[1;32m   1074\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdo_handshake_on_connect should not be specified for non-blocking sockets\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m-> 1075\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdo_handshake\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1076\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\u001b[38;5;167;01mOSError\u001b[39;00m, \u001b[38;5;167;01mValueError\u001b[39;00m):\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1346\u001b[0m, in \u001b[0;36mSSLSocket.do_handshake\u001b[0;34m(self, block)\u001b[0m\n\u001b[1;32m   1345\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msettimeout(\u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[0;32m-> 1346\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mdo_handshake()\n\u001b[1;32m   1347\u001b[0m \u001b[38;5;28;01mfinally\u001b[39;00m:\n",
      "\u001b[0;31mSSLEOFError\u001b[0m: EOF occurred in violation of protocol (_ssl.c:1002)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:790\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    789\u001b[0m \u001b[38;5;66;03m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[0;32m--> 790\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    794\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout_obj\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    796\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    797\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    798\u001b[0m \u001b[43m    \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mretries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    799\u001b[0m \u001b[43m    \u001b[49m\u001b[43mresponse_conn\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresponse_conn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    800\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpreload_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    801\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdecode_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    802\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kw\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    803\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    805\u001b[0m \u001b[38;5;66;03m# Everything went great!\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:491\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    490\u001b[0m         new_e \u001b[38;5;241m=\u001b[39m _wrap_proxy_error(new_e, conn\u001b[38;5;241m.\u001b[39mproxy\u001b[38;5;241m.\u001b[39mscheme)\n\u001b[0;32m--> 491\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m new_e\n\u001b[1;32m    493\u001b[0m \u001b[38;5;66;03m# conn.request() calls http.client.*.request, not the method in\u001b[39;00m\n\u001b[1;32m    494\u001b[0m \u001b[38;5;66;03m# urllib3.request. It also calls makefile (recv) on the socket.\u001b[39;00m\n",
      "\u001b[0;31mSSLError\u001b[0m: EOF occurred in violation of protocol (_ssl.c:1002)",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/adapters.py:486\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    485\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 486\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murlopen\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    487\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    488\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    489\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    490\u001b[0m \u001b[43m        \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    491\u001b[0m \u001b[43m        \u001b[49m\u001b[43mredirect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    492\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_same_host\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    493\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    494\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    495\u001b[0m \u001b[43m        \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    496\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    497\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    498\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    500\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:844\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    842\u001b[0m     new_e \u001b[38;5;241m=\u001b[39m ProtocolError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConnection aborted.\u001b[39m\u001b[38;5;124m\"\u001b[39m, new_e)\n\u001b[0;32m--> 844\u001b[0m retries \u001b[38;5;241m=\u001b[39m \u001b[43mretries\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mincrement\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    845\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43merror\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnew_e\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_pool\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_stacktrace\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msys\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mexc_info\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m2\u001b[39;49m\u001b[43m]\u001b[49m\n\u001b[1;32m    846\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    847\u001b[0m retries\u001b[38;5;241m.\u001b[39msleep()\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/retry.py:515\u001b[0m, in \u001b[0;36mRetry.increment\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    514\u001b[0m     reason \u001b[38;5;241m=\u001b[39m error \u001b[38;5;129;01mor\u001b[39;00m ResponseError(cause)\n\u001b[0;32m--> 515\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m MaxRetryError(_pool, url, reason) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mreason\u001b[39;00m  \u001b[38;5;66;03m# type: ignore[arg-type]\u001b[39;00m\n\u001b[1;32m    517\u001b[0m log\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIncremented Retry for (url=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, url, new_retry)\n",
      "\u001b[0;31mMaxRetryError\u001b[0m: HTTPSConnectionPool(host='export.arxiv.org', port=443): Max retries exceeded with url: /api/query?search_query=all%3A%22phages%22+AND+all%3A%22machine+learning%22&id_list=&sortBy=submittedDate&sortOrder=descending&start=0&max_results=100 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1002)')))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:634\u001b[0m, in \u001b[0;36mClient._parse_feed\u001b[0;34m(self, url, first_page, _try_index)\u001b[0m\n\u001b[1;32m    633\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 634\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m__try_parse_feed\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    635\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfirst_page\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfirst_page\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtry_index\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m_try_index\u001b[49m\n\u001b[1;32m    636\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    637\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\n\u001b[1;32m    638\u001b[0m     HTTPError,\n\u001b[1;32m    639\u001b[0m     UnexpectedEmptyPageError,\n\u001b[1;32m    640\u001b[0m     requests\u001b[38;5;241m.\u001b[39mexceptions\u001b[38;5;241m.\u001b[39mConnectionError,\n\u001b[1;32m    641\u001b[0m ) \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:674\u001b[0m, in \u001b[0;36mClient.__try_parse_feed\u001b[0;34m(self, url, first_page, try_index)\u001b[0m\n\u001b[1;32m    670\u001b[0m logger\u001b[38;5;241m.\u001b[39minfo(\n\u001b[1;32m    671\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRequesting page (first: \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m, try: \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, first_page, try_index, url\n\u001b[1;32m    672\u001b[0m )\n\u001b[0;32m--> 674\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_session\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m{\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43muser-agent\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43marxiv.py/2.1.0\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m}\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    675\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_last_request_dt \u001b[38;5;241m=\u001b[39m datetime\u001b[38;5;241m.\u001b[39mnow()\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:602\u001b[0m, in \u001b[0;36mSession.get\u001b[0;34m(self, url, **kwargs)\u001b[0m\n\u001b[1;32m    601\u001b[0m kwargs\u001b[38;5;241m.\u001b[39msetdefault(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m--> 602\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mGET\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprep\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43msend_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m \u001b[43madapter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/adapters.py:517\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    515\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(e\u001b[38;5;241m.\u001b[39mreason, _SSLError):\n\u001b[1;32m    516\u001b[0m     \u001b[38;5;66;03m# This branch is for urllib3 v1.22 and later.\u001b[39;00m\n\u001b[0;32m--> 517\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m SSLError(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[1;32m    519\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(e, request\u001b[38;5;241m=\u001b[39mrequest)\n",
      "\u001b[0;31mSSLError\u001b[0m: HTTPSConnectionPool(host='export.arxiv.org', port=443): Max retries exceeded with url: /api/query?search_query=all%3A%22phages%22+AND+all%3A%22machine+learning%22&id_list=&sortBy=submittedDate&sortOrder=descending&start=0&max_results=100 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1002)')))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLEOFError\u001b[0m                               Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:467\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    466\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 467\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_validate_conn\u001b[49m\u001b[43m(\u001b[49m\u001b[43mconn\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    468\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (SocketTimeout, BaseSSLError) \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:1096\u001b[0m, in \u001b[0;36mHTTPSConnectionPool._validate_conn\u001b[0;34m(self, conn)\u001b[0m\n\u001b[1;32m   1095\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_closed:\n\u001b[0;32m-> 1096\u001b[0m     \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1098\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_verified:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connection.py:642\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    634\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[1;32m    635\u001b[0m         (\n\u001b[1;32m    636\u001b[0m             \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSystem time is way off (before \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mRECENT_DATE\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m). This will probably \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    639\u001b[0m         SystemTimeWarning,\n\u001b[1;32m    640\u001b[0m     )\n\u001b[0;32m--> 642\u001b[0m sock_and_verified \u001b[38;5;241m=\u001b[39m \u001b[43m_ssl_wrap_socket_and_match_hostname\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    643\u001b[0m \u001b[43m    \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    644\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcert_reqs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcert_reqs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    645\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    646\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_minimum_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_minimum_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    647\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_maximum_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_maximum_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    648\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_certs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_certs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    649\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_cert_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    650\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_cert_data\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    651\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcert_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcert_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    652\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkey_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    653\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_password\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkey_password\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    654\u001b[0m \u001b[43m    \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    655\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_context\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    656\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    657\u001b[0m \u001b[43m    \u001b[49m\u001b[43massert_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43massert_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    658\u001b[0m \u001b[43m    \u001b[49m\u001b[43massert_fingerprint\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43massert_fingerprint\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    659\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    660\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock \u001b[38;5;241m=\u001b[39m sock_and_verified\u001b[38;5;241m.\u001b[39msocket\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connection.py:782\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_and_match_hostname\u001b[0;34m(sock, cert_reqs, ssl_version, ssl_minimum_version, ssl_maximum_version, cert_file, key_file, key_password, ca_certs, ca_cert_dir, ca_cert_data, assert_hostname, assert_fingerprint, server_hostname, ssl_context, tls_in_tls)\u001b[0m\n\u001b[1;32m    780\u001b[0m         server_hostname \u001b[38;5;241m=\u001b[39m normalized\n\u001b[0;32m--> 782\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m \u001b[43mssl_wrap_socket\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    783\u001b[0m \u001b[43m    \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    784\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkeyfile\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkey_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    785\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcertfile\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcert_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    786\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_password\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkey_password\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    787\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_certs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_certs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    788\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_cert_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    789\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_cert_data\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    790\u001b[0m \u001b[43m    \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontext\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/ssl_.py:470\u001b[0m, in \u001b[0;36mssl_wrap_socket\u001b[0;34m(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data, tls_in_tls)\u001b[0m\n\u001b[1;32m    468\u001b[0m     \u001b[38;5;28;01mpass\u001b[39;00m\n\u001b[0;32m--> 470\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m \u001b[43m_ssl_wrap_socket_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcontext\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    471\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ssl_sock\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/ssl_.py:514\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_impl\u001b[0;34m(sock, ssl_context, tls_in_tls, server_hostname)\u001b[0m\n\u001b[1;32m    512\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m SSLTransport(sock, ssl_context, server_hostname)\n\u001b[0;32m--> 514\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwrap_socket\u001b[49m\u001b[43m(\u001b[49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:517\u001b[0m, in \u001b[0;36mSSLContext.wrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[1;32m    511\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrap_socket\u001b[39m(\u001b[38;5;28mself\u001b[39m, sock, server_side\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[1;32m    512\u001b[0m                 do_handshake_on_connect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    513\u001b[0m                 suppress_ragged_eofs\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    514\u001b[0m                 server_hostname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, session\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    515\u001b[0m     \u001b[38;5;66;03m# SSLSocket class handles server_hostname encoding before it calls\u001b[39;00m\n\u001b[1;32m    516\u001b[0m     \u001b[38;5;66;03m# ctx._wrap_socket()\u001b[39;00m\n\u001b[0;32m--> 517\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msslsocket_class\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_create\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    518\u001b[0m \u001b[43m        \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    519\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_side\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_side\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    520\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdo_handshake_on_connect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdo_handshake_on_connect\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    521\u001b[0m \u001b[43m        \u001b[49m\u001b[43msuppress_ragged_eofs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msuppress_ragged_eofs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    522\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    523\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcontext\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    524\u001b[0m \u001b[43m        \u001b[49m\u001b[43msession\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msession\u001b[49m\n\u001b[1;32m    525\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1075\u001b[0m, in \u001b[0;36mSSLSocket._create\u001b[0;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[1;32m   1074\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdo_handshake_on_connect should not be specified for non-blocking sockets\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m-> 1075\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdo_handshake\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1076\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\u001b[38;5;167;01mOSError\u001b[39;00m, \u001b[38;5;167;01mValueError\u001b[39;00m):\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1346\u001b[0m, in \u001b[0;36mSSLSocket.do_handshake\u001b[0;34m(self, block)\u001b[0m\n\u001b[1;32m   1345\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msettimeout(\u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[0;32m-> 1346\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mdo_handshake()\n\u001b[1;32m   1347\u001b[0m \u001b[38;5;28;01mfinally\u001b[39;00m:\n",
      "\u001b[0;31mSSLEOFError\u001b[0m: EOF occurred in violation of protocol (_ssl.c:1002)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:790\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    789\u001b[0m \u001b[38;5;66;03m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[0;32m--> 790\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    794\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout_obj\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    796\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    797\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    798\u001b[0m \u001b[43m    \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mretries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    799\u001b[0m \u001b[43m    \u001b[49m\u001b[43mresponse_conn\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresponse_conn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    800\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpreload_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    801\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdecode_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    802\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kw\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    803\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    805\u001b[0m \u001b[38;5;66;03m# Everything went great!\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:491\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    490\u001b[0m         new_e \u001b[38;5;241m=\u001b[39m _wrap_proxy_error(new_e, conn\u001b[38;5;241m.\u001b[39mproxy\u001b[38;5;241m.\u001b[39mscheme)\n\u001b[0;32m--> 491\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m new_e\n\u001b[1;32m    493\u001b[0m \u001b[38;5;66;03m# conn.request() calls http.client.*.request, not the method in\u001b[39;00m\n\u001b[1;32m    494\u001b[0m \u001b[38;5;66;03m# urllib3.request. It also calls makefile (recv) on the socket.\u001b[39;00m\n",
      "\u001b[0;31mSSLError\u001b[0m: EOF occurred in violation of protocol (_ssl.c:1002)",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/adapters.py:486\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    485\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 486\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murlopen\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    487\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    488\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    489\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    490\u001b[0m \u001b[43m        \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    491\u001b[0m \u001b[43m        \u001b[49m\u001b[43mredirect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    492\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_same_host\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    493\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    494\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    495\u001b[0m \u001b[43m        \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    496\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    497\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    498\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    500\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:844\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    842\u001b[0m     new_e \u001b[38;5;241m=\u001b[39m ProtocolError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConnection aborted.\u001b[39m\u001b[38;5;124m\"\u001b[39m, new_e)\n\u001b[0;32m--> 844\u001b[0m retries \u001b[38;5;241m=\u001b[39m \u001b[43mretries\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mincrement\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    845\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43merror\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnew_e\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_pool\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_stacktrace\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msys\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mexc_info\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m2\u001b[39;49m\u001b[43m]\u001b[49m\n\u001b[1;32m    846\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    847\u001b[0m retries\u001b[38;5;241m.\u001b[39msleep()\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/retry.py:515\u001b[0m, in \u001b[0;36mRetry.increment\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    514\u001b[0m     reason \u001b[38;5;241m=\u001b[39m error \u001b[38;5;129;01mor\u001b[39;00m ResponseError(cause)\n\u001b[0;32m--> 515\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m MaxRetryError(_pool, url, reason) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mreason\u001b[39;00m  \u001b[38;5;66;03m# type: ignore[arg-type]\u001b[39;00m\n\u001b[1;32m    517\u001b[0m log\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIncremented Retry for (url=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, url, new_retry)\n",
      "\u001b[0;31mMaxRetryError\u001b[0m: HTTPSConnectionPool(host='export.arxiv.org', port=443): Max retries exceeded with url: /api/query?search_query=all%3A%22phages%22+AND+all%3A%22machine+learning%22&id_list=&sortBy=submittedDate&sortOrder=descending&start=0&max_results=100 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1002)')))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:634\u001b[0m, in \u001b[0;36mClient._parse_feed\u001b[0;34m(self, url, first_page, _try_index)\u001b[0m\n\u001b[1;32m    633\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 634\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m__try_parse_feed\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    635\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfirst_page\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfirst_page\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtry_index\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m_try_index\u001b[49m\n\u001b[1;32m    636\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    637\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\n\u001b[1;32m    638\u001b[0m     HTTPError,\n\u001b[1;32m    639\u001b[0m     UnexpectedEmptyPageError,\n\u001b[1;32m    640\u001b[0m     requests\u001b[38;5;241m.\u001b[39mexceptions\u001b[38;5;241m.\u001b[39mConnectionError,\n\u001b[1;32m    641\u001b[0m ) \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:674\u001b[0m, in \u001b[0;36mClient.__try_parse_feed\u001b[0;34m(self, url, first_page, try_index)\u001b[0m\n\u001b[1;32m    670\u001b[0m logger\u001b[38;5;241m.\u001b[39minfo(\n\u001b[1;32m    671\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRequesting page (first: \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m, try: \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, first_page, try_index, url\n\u001b[1;32m    672\u001b[0m )\n\u001b[0;32m--> 674\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_session\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m{\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43muser-agent\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43marxiv.py/2.1.0\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m}\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    675\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_last_request_dt \u001b[38;5;241m=\u001b[39m datetime\u001b[38;5;241m.\u001b[39mnow()\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:602\u001b[0m, in \u001b[0;36mSession.get\u001b[0;34m(self, url, **kwargs)\u001b[0m\n\u001b[1;32m    601\u001b[0m kwargs\u001b[38;5;241m.\u001b[39msetdefault(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m--> 602\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mGET\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprep\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43msend_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m \u001b[43madapter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/adapters.py:517\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    515\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(e\u001b[38;5;241m.\u001b[39mreason, _SSLError):\n\u001b[1;32m    516\u001b[0m     \u001b[38;5;66;03m# This branch is for urllib3 v1.22 and later.\u001b[39;00m\n\u001b[0;32m--> 517\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m SSLError(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[1;32m    519\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(e, request\u001b[38;5;241m=\u001b[39mrequest)\n",
      "\u001b[0;31mSSLError\u001b[0m: HTTPSConnectionPool(host='export.arxiv.org', port=443): Max retries exceeded with url: /api/query?search_query=all%3A%22phages%22+AND+all%3A%22machine+learning%22&id_list=&sortBy=submittedDate&sortOrder=descending&start=0&max_results=100 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1002)')))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLEOFError\u001b[0m                               Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:467\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    466\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 467\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_validate_conn\u001b[49m\u001b[43m(\u001b[49m\u001b[43mconn\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    468\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (SocketTimeout, BaseSSLError) \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:1096\u001b[0m, in \u001b[0;36mHTTPSConnectionPool._validate_conn\u001b[0;34m(self, conn)\u001b[0m\n\u001b[1;32m   1095\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_closed:\n\u001b[0;32m-> 1096\u001b[0m     \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1098\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_verified:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connection.py:642\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    634\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[1;32m    635\u001b[0m         (\n\u001b[1;32m    636\u001b[0m             \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSystem time is way off (before \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mRECENT_DATE\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m). This will probably \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    639\u001b[0m         SystemTimeWarning,\n\u001b[1;32m    640\u001b[0m     )\n\u001b[0;32m--> 642\u001b[0m sock_and_verified \u001b[38;5;241m=\u001b[39m \u001b[43m_ssl_wrap_socket_and_match_hostname\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    643\u001b[0m \u001b[43m    \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    644\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcert_reqs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcert_reqs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    645\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    646\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_minimum_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_minimum_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    647\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_maximum_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_maximum_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    648\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_certs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_certs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    649\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_cert_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    650\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_cert_data\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    651\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcert_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcert_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    652\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkey_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    653\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_password\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkey_password\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    654\u001b[0m \u001b[43m    \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    655\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_context\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    656\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    657\u001b[0m \u001b[43m    \u001b[49m\u001b[43massert_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43massert_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    658\u001b[0m \u001b[43m    \u001b[49m\u001b[43massert_fingerprint\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43massert_fingerprint\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    659\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    660\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock \u001b[38;5;241m=\u001b[39m sock_and_verified\u001b[38;5;241m.\u001b[39msocket\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connection.py:782\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_and_match_hostname\u001b[0;34m(sock, cert_reqs, ssl_version, ssl_minimum_version, ssl_maximum_version, cert_file, key_file, key_password, ca_certs, ca_cert_dir, ca_cert_data, assert_hostname, assert_fingerprint, server_hostname, ssl_context, tls_in_tls)\u001b[0m\n\u001b[1;32m    780\u001b[0m         server_hostname \u001b[38;5;241m=\u001b[39m normalized\n\u001b[0;32m--> 782\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m \u001b[43mssl_wrap_socket\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    783\u001b[0m \u001b[43m    \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    784\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkeyfile\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkey_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    785\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcertfile\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcert_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    786\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_password\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkey_password\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    787\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_certs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_certs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    788\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_cert_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    789\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_cert_data\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    790\u001b[0m \u001b[43m    \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontext\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/ssl_.py:470\u001b[0m, in \u001b[0;36mssl_wrap_socket\u001b[0;34m(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data, tls_in_tls)\u001b[0m\n\u001b[1;32m    468\u001b[0m     \u001b[38;5;28;01mpass\u001b[39;00m\n\u001b[0;32m--> 470\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m \u001b[43m_ssl_wrap_socket_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcontext\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    471\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ssl_sock\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/ssl_.py:514\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_impl\u001b[0;34m(sock, ssl_context, tls_in_tls, server_hostname)\u001b[0m\n\u001b[1;32m    512\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m SSLTransport(sock, ssl_context, server_hostname)\n\u001b[0;32m--> 514\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwrap_socket\u001b[49m\u001b[43m(\u001b[49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:517\u001b[0m, in \u001b[0;36mSSLContext.wrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[1;32m    511\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrap_socket\u001b[39m(\u001b[38;5;28mself\u001b[39m, sock, server_side\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[1;32m    512\u001b[0m                 do_handshake_on_connect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    513\u001b[0m                 suppress_ragged_eofs\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    514\u001b[0m                 server_hostname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, session\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    515\u001b[0m     \u001b[38;5;66;03m# SSLSocket class handles server_hostname encoding before it calls\u001b[39;00m\n\u001b[1;32m    516\u001b[0m     \u001b[38;5;66;03m# ctx._wrap_socket()\u001b[39;00m\n\u001b[0;32m--> 517\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msslsocket_class\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_create\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    518\u001b[0m \u001b[43m        \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    519\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_side\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_side\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    520\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdo_handshake_on_connect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdo_handshake_on_connect\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    521\u001b[0m \u001b[43m        \u001b[49m\u001b[43msuppress_ragged_eofs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msuppress_ragged_eofs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    522\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    523\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcontext\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    524\u001b[0m \u001b[43m        \u001b[49m\u001b[43msession\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msession\u001b[49m\n\u001b[1;32m    525\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1075\u001b[0m, in \u001b[0;36mSSLSocket._create\u001b[0;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[1;32m   1074\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdo_handshake_on_connect should not be specified for non-blocking sockets\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m-> 1075\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdo_handshake\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1076\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\u001b[38;5;167;01mOSError\u001b[39;00m, \u001b[38;5;167;01mValueError\u001b[39;00m):\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1346\u001b[0m, in \u001b[0;36mSSLSocket.do_handshake\u001b[0;34m(self, block)\u001b[0m\n\u001b[1;32m   1345\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msettimeout(\u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[0;32m-> 1346\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mdo_handshake()\n\u001b[1;32m   1347\u001b[0m \u001b[38;5;28;01mfinally\u001b[39;00m:\n",
      "\u001b[0;31mSSLEOFError\u001b[0m: EOF occurred in violation of protocol (_ssl.c:1002)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:790\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    789\u001b[0m \u001b[38;5;66;03m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[0;32m--> 790\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    794\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout_obj\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    796\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    797\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    798\u001b[0m \u001b[43m    \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mretries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    799\u001b[0m \u001b[43m    \u001b[49m\u001b[43mresponse_conn\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresponse_conn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    800\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpreload_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    801\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdecode_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    802\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kw\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    803\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    805\u001b[0m \u001b[38;5;66;03m# Everything went great!\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:491\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    490\u001b[0m         new_e \u001b[38;5;241m=\u001b[39m _wrap_proxy_error(new_e, conn\u001b[38;5;241m.\u001b[39mproxy\u001b[38;5;241m.\u001b[39mscheme)\n\u001b[0;32m--> 491\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m new_e\n\u001b[1;32m    493\u001b[0m \u001b[38;5;66;03m# conn.request() calls http.client.*.request, not the method in\u001b[39;00m\n\u001b[1;32m    494\u001b[0m \u001b[38;5;66;03m# urllib3.request. It also calls makefile (recv) on the socket.\u001b[39;00m\n",
      "\u001b[0;31mSSLError\u001b[0m: EOF occurred in violation of protocol (_ssl.c:1002)",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/adapters.py:486\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    485\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 486\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murlopen\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    487\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    488\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    489\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    490\u001b[0m \u001b[43m        \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    491\u001b[0m \u001b[43m        \u001b[49m\u001b[43mredirect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    492\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_same_host\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    493\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    494\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    495\u001b[0m \u001b[43m        \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    496\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    497\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    498\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    500\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:844\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    842\u001b[0m     new_e \u001b[38;5;241m=\u001b[39m ProtocolError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConnection aborted.\u001b[39m\u001b[38;5;124m\"\u001b[39m, new_e)\n\u001b[0;32m--> 844\u001b[0m retries \u001b[38;5;241m=\u001b[39m \u001b[43mretries\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mincrement\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    845\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43merror\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnew_e\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_pool\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_stacktrace\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msys\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mexc_info\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m2\u001b[39;49m\u001b[43m]\u001b[49m\n\u001b[1;32m    846\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    847\u001b[0m retries\u001b[38;5;241m.\u001b[39msleep()\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/util/retry.py:515\u001b[0m, in \u001b[0;36mRetry.increment\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    514\u001b[0m     reason \u001b[38;5;241m=\u001b[39m error \u001b[38;5;129;01mor\u001b[39;00m ResponseError(cause)\n\u001b[0;32m--> 515\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m MaxRetryError(_pool, url, reason) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mreason\u001b[39;00m  \u001b[38;5;66;03m# type: ignore[arg-type]\u001b[39;00m\n\u001b[1;32m    517\u001b[0m log\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIncremented Retry for (url=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, url, new_retry)\n",
      "\u001b[0;31mMaxRetryError\u001b[0m: HTTPSConnectionPool(host='export.arxiv.org', port=443): Max retries exceeded with url: /api/query?search_query=all%3A%22phages%22+AND+all%3A%22machine+learning%22&id_list=&sortBy=submittedDate&sortOrder=descending&start=0&max_results=100 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1002)')))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mSSLError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[21], line 47\u001b[0m\n\u001b[1;32m     45\u001b[0m \u001b[38;5;66;03m# Example usage\u001b[39;00m\n\u001b[1;32m     46\u001b[0m keywords \u001b[38;5;241m=\u001b[39m [\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mphages\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmachine learning\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[0;32m---> 47\u001b[0m papers\u001b[38;5;241m=\u001b[39m \u001b[43msearch_arxiv\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkeywords\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmax_age_years\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m10\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m     48\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m p \u001b[38;5;129;01min\u001b[39;00m papers:\n\u001b[1;32m     49\u001b[0m     \u001b[38;5;28mprint\u001b[39m(p)\n",
      "Cell \u001b[0;32mIn[21], line 38\u001b[0m, in \u001b[0;36msearch_arxiv\u001b[0;34m(keywords, max_results, max_age_years)\u001b[0m\n\u001b[1;32m     36\u001b[0m \u001b[38;5;66;03m# Fetch results and filter by age\u001b[39;00m\n\u001b[1;32m     37\u001b[0m papers \u001b[38;5;241m=\u001b[39m []\n\u001b[0;32m---> 38\u001b[0m \u001b[43m\u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mresult\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mresults\u001b[49m\u001b[43m:\u001b[49m\n\u001b[1;32m     39\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43;01mif\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mresult\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpublished\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdate\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m>\u001b[39;49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mdate_threshold\u001b[49m\u001b[43m:\u001b[49m\n\u001b[1;32m     40\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43mprint\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mresult\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:587\u001b[0m, in \u001b[0;36mClient._results\u001b[0;34m(self, search, offset)\u001b[0m\n\u001b[1;32m    583\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_results\u001b[39m(\n\u001b[1;32m    584\u001b[0m     \u001b[38;5;28mself\u001b[39m, search: Search, offset: \u001b[38;5;28mint\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m\n\u001b[1;32m    585\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m Generator[Result, \u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;28;01mNone\u001b[39;00m]:\n\u001b[1;32m    586\u001b[0m     page_url \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_format_url(search, offset, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpage_size)\n\u001b[0;32m--> 587\u001b[0m     feed \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_parse_feed\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpage_url\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfirst_page\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[1;32m    588\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m feed\u001b[38;5;241m.\u001b[39mentries:\n\u001b[1;32m    589\u001b[0m         logger\u001b[38;5;241m.\u001b[39minfo(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGot empty first page; stopping generation\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:644\u001b[0m, in \u001b[0;36mClient._parse_feed\u001b[0;34m(self, url, first_page, _try_index)\u001b[0m\n\u001b[1;32m    642\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m _try_index \u001b[38;5;241m<\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mnum_retries:\n\u001b[1;32m    643\u001b[0m     logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGot error (try \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, _try_index, err)\n\u001b[0;32m--> 644\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_parse_feed\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    645\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfirst_page\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfirst_page\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_try_index\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m_try_index\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\n\u001b[1;32m    646\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    647\u001b[0m logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGiving up (try \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, _try_index, err)\n\u001b[1;32m    648\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m err\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:644\u001b[0m, in \u001b[0;36mClient._parse_feed\u001b[0;34m(self, url, first_page, _try_index)\u001b[0m\n\u001b[1;32m    642\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m _try_index \u001b[38;5;241m<\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mnum_retries:\n\u001b[1;32m    643\u001b[0m     logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGot error (try \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, _try_index, err)\n\u001b[0;32m--> 644\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_parse_feed\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    645\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfirst_page\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfirst_page\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_try_index\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m_try_index\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\n\u001b[1;32m    646\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    647\u001b[0m logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGiving up (try \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, _try_index, err)\n\u001b[1;32m    648\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m err\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:644\u001b[0m, in \u001b[0;36mClient._parse_feed\u001b[0;34m(self, url, first_page, _try_index)\u001b[0m\n\u001b[1;32m    642\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m _try_index \u001b[38;5;241m<\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mnum_retries:\n\u001b[1;32m    643\u001b[0m     logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGot error (try \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, _try_index, err)\n\u001b[0;32m--> 644\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_parse_feed\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    645\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfirst_page\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfirst_page\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_try_index\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m_try_index\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\n\u001b[1;32m    646\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    647\u001b[0m logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGiving up (try \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, _try_index, err)\n\u001b[1;32m    648\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m err\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:648\u001b[0m, in \u001b[0;36mClient._parse_feed\u001b[0;34m(self, url, first_page, _try_index)\u001b[0m\n\u001b[1;32m    644\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_parse_feed(\n\u001b[1;32m    645\u001b[0m         url, first_page\u001b[38;5;241m=\u001b[39mfirst_page, _try_index\u001b[38;5;241m=\u001b[39m_try_index \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m1\u001b[39m\n\u001b[1;32m    646\u001b[0m     )\n\u001b[1;32m    647\u001b[0m logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGiving up (try \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, _try_index, err)\n\u001b[0;32m--> 648\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m err\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:634\u001b[0m, in \u001b[0;36mClient._parse_feed\u001b[0;34m(self, url, first_page, _try_index)\u001b[0m\n\u001b[1;32m    627\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    628\u001b[0m \u001b[38;5;124;03mFetches the specified URL and parses it with feedparser.\u001b[39;00m\n\u001b[1;32m    629\u001b[0m \n\u001b[1;32m    630\u001b[0m \u001b[38;5;124;03mIf a request fails or is unexpectedly empty, retries the request up to\u001b[39;00m\n\u001b[1;32m    631\u001b[0m \u001b[38;5;124;03m`self.num_retries` times.\u001b[39;00m\n\u001b[1;32m    632\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    633\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 634\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m__try_parse_feed\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    635\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfirst_page\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfirst_page\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtry_index\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m_try_index\u001b[49m\n\u001b[1;32m    636\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    637\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\n\u001b[1;32m    638\u001b[0m     HTTPError,\n\u001b[1;32m    639\u001b[0m     UnexpectedEmptyPageError,\n\u001b[1;32m    640\u001b[0m     requests\u001b[38;5;241m.\u001b[39mexceptions\u001b[38;5;241m.\u001b[39mConnectionError,\n\u001b[1;32m    641\u001b[0m ) \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m    642\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m _try_index \u001b[38;5;241m<\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mnum_retries:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/arxiv/__init__.py:674\u001b[0m, in \u001b[0;36mClient.__try_parse_feed\u001b[0;34m(self, url, first_page, try_index)\u001b[0m\n\u001b[1;32m    668\u001b[0m         time\u001b[38;5;241m.\u001b[39msleep(to_sleep)\n\u001b[1;32m    670\u001b[0m logger\u001b[38;5;241m.\u001b[39minfo(\n\u001b[1;32m    671\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRequesting page (first: \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m, try: \u001b[39m\u001b[38;5;132;01m%d\u001b[39;00m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, first_page, try_index, url\n\u001b[1;32m    672\u001b[0m )\n\u001b[0;32m--> 674\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_session\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m{\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43muser-agent\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43marxiv.py/2.1.0\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m}\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    675\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_last_request_dt \u001b[38;5;241m=\u001b[39m datetime\u001b[38;5;241m.\u001b[39mnow()\n\u001b[1;32m    676\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m resp\u001b[38;5;241m.\u001b[39mstatus_code \u001b[38;5;241m!=\u001b[39m requests\u001b[38;5;241m.\u001b[39mcodes\u001b[38;5;241m.\u001b[39mOK:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:602\u001b[0m, in \u001b[0;36mSession.get\u001b[0;34m(self, url, **kwargs)\u001b[0m\n\u001b[1;32m    594\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a GET request. Returns :class:`Response` object.\u001b[39;00m\n\u001b[1;32m    595\u001b[0m \n\u001b[1;32m    596\u001b[0m \u001b[38;5;124;03m:param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[1;32m    597\u001b[0m \u001b[38;5;124;03m:param \\*\\*kwargs: Optional arguments that ``request`` takes.\u001b[39;00m\n\u001b[1;32m    598\u001b[0m \u001b[38;5;124;03m:rtype: requests.Response\u001b[39;00m\n\u001b[1;32m    599\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    601\u001b[0m kwargs\u001b[38;5;241m.\u001b[39msetdefault(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m--> 602\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mGET\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    584\u001b[0m send_kwargs \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m    585\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m\"\u001b[39m: timeout,\n\u001b[1;32m    586\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m: allow_redirects,\n\u001b[1;32m    587\u001b[0m }\n\u001b[1;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprep\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43msend_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    700\u001b[0m start \u001b[38;5;241m=\u001b[39m preferred_clock()\n\u001b[1;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m \u001b[43madapter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[1;32m    706\u001b[0m elapsed \u001b[38;5;241m=\u001b[39m preferred_clock() \u001b[38;5;241m-\u001b[39m start\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/adapters.py:517\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    513\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m ProxyError(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[1;32m    515\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(e\u001b[38;5;241m.\u001b[39mreason, _SSLError):\n\u001b[1;32m    516\u001b[0m         \u001b[38;5;66;03m# This branch is for urllib3 v1.22 and later.\u001b[39;00m\n\u001b[0;32m--> 517\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m SSLError(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[1;32m    519\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[1;32m    521\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m ClosedPoolError \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "\u001b[0;31mSSLError\u001b[0m: HTTPSConnectionPool(host='export.arxiv.org', port=443): Max retries exceeded with url: /api/query?search_query=all%3A%22phages%22+AND+all%3A%22machine+learning%22&id_list=&sortBy=submittedDate&sortOrder=descending&start=0&max_results=100 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1002)')))"
     ]
    }
   ],
   "source": [
    "import arxiv\n",
    "import datetime\n",
    "\n",
    "def search_arxiv(keywords, max_results=10, max_age_years=5):\n",
    "    \"\"\"\n",
    "    Search for papers on arXiv.org matching the given keywords, no older than a specified number of years.\n",
    "\n",
    "    Args:\n",
    "    keywords (str): Keywords to search for.\n",
    "    max_results (int): Maximum number of results to return.\n",
    "    max_age_years (int): Maximum age of papers in years.\n",
    "\n",
    "    Returns:\n",
    "    list: Result list of the resulting papers.\n",
    "    \"\"\"\n",
    "    # Define the search query\n",
    "    query = ' AND '.join(f'all:\"{keyword}\"' for keyword in keywords)\n",
    "\n",
    "    # Calculate the date threshold\n",
    "    date_threshold = datetime.date.today() - datetime.timedelta(days=365 * max_age_years)\n",
    "\n",
    "    # Construct the default API client.\n",
    "    client = arxiv.Client()\n",
    "\n",
    "    # Search arXiv\n",
    "    search = arxiv.Search(\n",
    "        query=query,\n",
    "        max_results=max_results,\n",
    "        sort_by=arxiv.SortCriterion.SubmittedDate,\n",
    "        sort_order=arxiv.SortOrder.Descending\n",
    "    )\n",
    "\n",
    "    results = client.results(search)\n",
    "    print(results)\n",
    "\n",
    "    # Fetch results and filter by age\n",
    "    papers = []\n",
    "    for result in results:\n",
    "        if result.published.date() >= date_threshold:\n",
    "            print(result)\n",
    "            papers.append(result)\n",
    "\n",
    "    return papers\n",
    "\n",
    "# Example usage\n",
    "keywords = [\"phages\", \"machine learning\"]\n",
    "papers= search_arxiv(keywords, max_age_years=10)\n",
    "for p in papers:\n",
    "    print(p)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting wikipedia\n",
      "  Downloading wikipedia-1.4.0.tar.gz (27 kB)\n",
      "  Installing build dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n",
      "\u001b[?25h  Preparing metadata (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: beautifulsoup4 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from wikipedia) (4.12.2)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.0.0 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from wikipedia) (2.31.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from requests<3.0.0,>=2.0.0->wikipedia) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from requests<3.0.0,>=2.0.0->wikipedia) (3.6)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from requests<3.0.0,>=2.0.0->wikipedia) (2.1.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from requests<3.0.0,>=2.0.0->wikipedia) (2023.11.17)\n",
      "Requirement already satisfied: soupsieve>1.2 in /workspaces/ml-learning/.venv/lib/python3.11/site-packages (from beautifulsoup4->wikipedia) (2.5)\n",
      "Building wheels for collected packages: wikipedia\n",
      "  Building wheel for wikipedia (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for wikipedia: filename=wikipedia-1.4.0-py3-none-any.whl size=11678 sha256=24365f4bbf3605ead7845d478010b4b999a315d15a4242234bbd048d3467eb8a\n",
      "  Stored in directory: /home/vscode/.cache/pip/wheels/8f/ab/cb/45ccc40522d3a1c41e1d2ad53b8f33a62f394011ec38cd71c6\n",
      "Successfully built wikipedia\n",
      "Installing collected packages: wikipedia\n",
      "Successfully installed wikipedia-1.4.0\n"
     ]
    }
   ],
   "source": [
    "!pip install wikipedia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['P1 phage',\n",
       " 'Bacteriophage',\n",
       " 'P1',\n",
       " 'Cre-Lox recombination',\n",
       " 'Nat Sternberg',\n",
       " 'Cloning vector',\n",
       " 'Site-specific recombination',\n",
       " 'P1-derived artificial chromosome',\n",
       " 'Pseudomonas virus phi6',\n",
       " 'Ff phages']"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import wikipedia\n",
    "\n",
    "title='P1 Phage'\n",
    "wikipedia.search('P1 Phage')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['http://www.expasy.org/viralzone/all_by_species/505.html',\n",
       " 'https://zenodo.org/record/1235061',\n",
       " 'https://ui.adsabs.harvard.edu/abs/1990PNAS...87.8070S',\n",
       " 'https://api.semanticscholar.org/CorpusID:28785596',\n",
       " 'https://www.irmng.org/aphia.php?p=taxdetails&id=11459714',\n",
       " 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC523184',\n",
       " 'https://doi.org/10.1128%2FJB.186.21.7032-7068.2004',\n",
       " 'https://www.worldcat.org/issn/0021-9193',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/15489417',\n",
       " 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC256520',\n",
       " 'https://doi.org/10.1128%2FJVI.45.3.1118-1139.1983',\n",
       " 'https://www.worldcat.org/issn/0022-538X',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/6834479',\n",
       " 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC206102',\n",
       " 'https://doi.org/10.1128%2Fjb.174.12.3936-3944.1992',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/1534556',\n",
       " 'https://doi.org/10.1016%2F0022-2836(92)90623-R',\n",
       " 'https://www.worldcat.org/issn/0022-2836',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/1324323',\n",
       " 'https://doi.org/10.1016%2F0092-8674(81)90180-X',\n",
       " 'https://www.worldcat.org/issn/0092-8674',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/7026049',\n",
       " 'https://doi.org/10.1016%2F0378-1119(96)00141-2',\n",
       " 'https://www.worldcat.org/issn/0378-1119',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/8917092',\n",
       " 'https://doi.org/10.1016%2F0042-6822(83)90542-1',\n",
       " 'https://www.worldcat.org/issn/0042-6822',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/6359666',\n",
       " 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC54894',\n",
       " 'https://doi.org/10.1073%2Fpnas.87.20.8070',\n",
       " 'https://www.worldcat.org/issn/0027-8424',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/2236019',\n",
       " 'https://doi.org/10.1146%2Fannurev-virology-100114-054930',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/26958905',\n",
       " 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8546580',\n",
       " 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9025707',\n",
       " 'https://doi.org/10.1016%2Fj.jmb.2022.167829',\n",
       " 'https://doi.org/10.1046%2Fj.1365-2443.1997.1450343.x',\n",
       " 'https://doi.org/10.1128%2FmBio.01013-21',\n",
       " 'https://doi.org/10.1006%2Fjmbi.1993.1521',\n",
       " 'https://doi.org/10.1074%2Fjbc.274.24.16813',\n",
       " 'https://doi.org/10.1016%2F0022-2836(86)90228-7',\n",
       " 'https://doi.org/10.3390%2Fijms23084231',\n",
       " 'https://www.worldcat.org/issn/1356-9597',\n",
       " 'https://www.worldcat.org/issn/2150-7511',\n",
       " 'https://www.worldcat.org/issn/0021-9258',\n",
       " 'https://www.worldcat.org/issn/1422-0067',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/34517752',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/35457047',\n",
       " 'https://www.sciencedirect.com/science/article/pii/S0022283683715214',\n",
       " 'https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/36116540',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/9413995',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/8411153',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/10358024',\n",
       " 'https://pubmed.ncbi.nlm.nih.gov/3486297',\n",
       " 'https://api.semanticscholar.org/CorpusID:252325855',\n",
       " 'https://api.semanticscholar.org/CorpusID:26780439',\n",
       " 'https://dx.doi.org/10.1016/0022-2836(86)90228-7']"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "page = wikipedia.page(title)\n",
    "page.references"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting pypdf\n",
      "  Downloading pypdf-3.17.4-py3-none-any.whl.metadata (7.5 kB)\n",
      "Downloading pypdf-3.17.4-py3-none-any.whl (278 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m278.2/278.2 kB\u001b[0m \u001b[31m3.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: pypdf\n",
      "Successfully installed pypdf-3.17.4\n"
     ]
    }
   ],
   "source": [
    "!pip install pypdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# https://docs.llamaindex.ai/en/stable/module_guides/loading/simpledirectoryreader.html\n",
    "\n",
    "from llama_index import SimpleDirectoryReader\n",
    "\n",
    "reader = SimpleDirectoryReader(input_dir=\"/workspaces/ml-learning/src/phages/data/P1\")\n",
    "documents = reader.load_data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# extracting metadata\n",
    "\n",
    "# to extract reference as well? or maybe do it after the nodes and documents are created?\n",
    "import mimetypes\n",
    "import os\n",
    "from datetime import datetime\n",
    "from pathlib import Path\n",
    "from typing import Any, Callable, Dict, Generator, List, Optional, Type\n",
    "\n",
    "from pathlib import Path\n",
    "def default_file_metadata_func(file_path: str) -> Dict:\n",
    "    \"\"\"Get some handy metadate from filesystem.\n",
    "\n",
    "    Args:\n",
    "        file_path: str: file path in str\n",
    "    \"\"\"\n",
    "    return {\n",
    "        \"file_path\": file_path,\n",
    "        \"file_name\": os.path.basename(file_path),\n",
    "        \"file_type\": mimetypes.guess_type(file_path)[0],\n",
    "        \"file_size\": os.path.getsize(file_path),\n",
    "        \"creation_date\": datetime.fromtimestamp(\n",
    "            Path(file_path).stat().st_ctime\n",
    "        ).strftime(\"%Y-%m-%d\"),\n",
    "        \"last_modified_date\": datetime.fromtimestamp(\n",
    "            Path(file_path).stat().st_mtime\n",
    "        ).strftime(\"%Y-%m-%d\"),\n",
    "        \"last_accessed_date\": datetime.fromtimestamp(\n",
    "            Path(file_path).stat().st_atime\n",
    "        ).strftime(\"%Y-%m-%d\"),\n",
    "    }\n",
    "\n",
    "def get_meta(file_path):\n",
    "    return {\"foo\": \"bar\", \"file_path\": file_path}\n",
    "\n",
    "\n",
    "# e.g. SimpleDirectoryReader(input_dir=\"path/to/directory\", file_metadata=get_meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# LLM\n",
    "llm = OpenAI(temperature=0, model=\"gpt-4\")\n",
    "service_context = ServiceContext.from_defaults(llm=llm, chunk_size=1024)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import tqdm as notebook_tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "one = [d for d in documents if d.metadata['file_name'] == '0343-04.pdf']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "37"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(one)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–70...\n",
      "> Adding chunk: JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–70...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Whereas /H9261prophage leads an essentially pas...\n",
      "> Adding chunk: Whereas /H9261prophage leads an essentially pas...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: out in vitro, P1-based vectors are now in commo...\n",
      "> Adding chunk: out in vitro, P1-based vectors are now in commo...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MATERIALS AND METHODS\n",
      "Sources of DNA and sequen...\n",
      "> Adding chunk: MATERIALS AND METHODS\n",
      "Sources of DNA and sequen...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The assignment of previously identi ﬁed genetic...\n",
      "> Adding chunk: The assignment of previously identi ﬁed genetic...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: numbers AF234172 (phage P1 mod749 ::IS5c1-100) ...\n",
      "> Adding chunk: numbers AF234172 (phage P1 mod749 ::IS5c1-100) ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Five of those mutations arescattered in the var...\n",
      "> Adding chunk: Five of those mutations arescattered in the var...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: According to the pattern of C-G skew (the relat...\n",
      "> Adding chunk: According to the pattern of C-G skew (the relat...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: FIG. 1. Genetic and physical organization of th...\n",
      "> Adding chunk: FIG. 1. Genetic and physical organization of th...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: quence that in E. coli ,Salmonella enterica ser...\n",
      "> Adding chunk: quence that in E. coli ,Salmonella enterica ser...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: For orientation, a simpli ﬁed genome map is sho...\n",
      "> Adding chunk: For orientation, a simpli ﬁed genome map is sho...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: RecBCD in P1 cyclization could explain the 10-f...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> Adding chunk: RecBCD in P1 cyclization could explain the 10-f...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The two octamers that are the most frequent in ...\n",
      "> Adding chunk: The two octamers that are the most frequent in ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: It, iteron.\n",
      "bThe ﬁrst number is the coordinate ...\n",
      "> Adding chunk: It, iteron.\n",
      "bThe ﬁrst number is the coordinate ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: as the most infrequent tetramer in E. coli (35)...\n",
      "> Adding chunk: as the most infrequent tetramer in E. coli (35)...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: hot 85.3385.6 ppfA 62.3462.7\n",
      "12 SeeR? hrdC 64.1...\n",
      "> Adding chunk: hot 85.3385.6 ppfA 62.3462.7\n",
      "12 SeeR? hrdC 64.1...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: pacA 89.3390.5 trnT 68.2368.3\n",
      "c6 See lxc pacB 9...\n",
      "> Adding chunk: pacA 89.3390.5 trnT 68.2368.3\n",
      "c6 See lxc pacB 9...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: TABLE 3. Gene products (listed by gene position...\n",
      "> Adding chunk: TABLE 3. Gene products (listed by gene position...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: EcoB,\n",
      "EcoK; putative DNA methyltransferase and ...\n",
      "> Adding chunk: EcoB,\n",
      "EcoK; putative DNA methyltransferase and ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 26343 –26708 /H11002 13,013 Uncertain; product ...\n",
      "> Adding chunk: 26343 –26708 /H11002 13,013 Uncertain; product ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: gp12?)36654 –37088 /H11002 15,973 Tail ﬁber str...\n",
      "> Adding chunk: gp12?)36654 –37088 /H11002 15,973 Tail ﬁber str...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: TABLE 3 —Continued\n",
      "Nameb(former\n",
      "name[s])Gene\n",
      "co...\n",
      "> Adding chunk: TABLE 3 —Continued\n",
      "Nameb(former\n",
      "name[s])Gene\n",
      "co...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: lysis control twS\n",
      "PpfA 62321 –62770 /H11002 15,...\n",
      "> Adding chunk: lysis control twS\n",
      "PpfA 62321 –62770 /H11002 15,...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 81033 –81251 /H11001 8,318 Putative morphogenet...\n",
      "> Adding chunk: 81033 –81251 /H11001 8,318 Putative morphogenet...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: [86]/H11022/H11022GUG[16] /H11022UUG[10]). Tran...\n",
      "> Adding chunk: [86]/H11022/H11022GUG[16] /H11022UUG[10]). Tran...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Whether their distribution results from the hor...\n",
      "> Adding chunk: Whether their distribution results from the hor...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: bGene products marked by an asterisk were eithe...\n",
      "> Adding chunk: bGene products marked by an asterisk were eithe...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: TABLE 4. Known and putative /H926870promoters o...\n",
      "> Adding chunk: TABLE 4. Known and putative /H926870promoters o...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 5 285\n",
      "PdarB 16093 /H11002CGTGATTTTA TTGTCT GCCA...\n",
      "> Adding chunk: 5 285\n",
      "PdarB 16093 /H11002CGTGATTTTA TTGTCT GCCA...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 7 125\n",
      "PrepL 49267 /H11001AAACTATCTC TTCACC TGGT...\n",
      "> Adding chunk: 7 125\n",
      "PrepL 49267 /H11001AAACTATCTC TTCACC TGGT...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 7\n",
      "P1ban 67469 /H11001TTTGGGTATA TTGCTC TAATAAAT...\n",
      "> Adding chunk: 7\n",
      "P1ban 67469 /H11001TTTGGGTATA TTGCTC TAATAAAT...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: promoters are associated with genes expressed d...\n",
      "> Adding chunk: promoters are associated with genes expressed d...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: As many as 12 predicted Rho-independent termina...\n",
      "> Adding chunk: As many as 12 predicted Rho-independent termina...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Coordinates refer to positions of promoters in ...\n",
      "> Adding chunk: Coordinates refer to positions of promoters in ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: regulate translation of the 23gene from an alte...\n",
      "> Adding chunk: regulate translation of the 23gene from an alte...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: TR\n",
      "Taslyz 19655 /H11001 GATAGCGCAAATTGCACCGCCTC...\n",
      "> Adding chunk: TR\n",
      "Taslyz 19655 /H11001 GATAGCGCAAATTGCACCGCCTC...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: TR\n",
      "TrepA 60474 /H11002 TTTGATGTGCGCTGGAGGGGGACG...\n",
      "> Adding chunk: TR\n",
      "TrepA 60474 /H11002 TTTGATGTGCGCTGGAGGGGGACG...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: TR\n",
      "TaspmgT 83144 /H11002 GATTGGAAACATGCCGCGCCTT...\n",
      "> Adding chunk: TR\n",
      "TaspmgT 83144 /H11002 GATTGGAAACATGCCGCGCCTT...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: (i) Immunity. Whether P1 is maintained in cells...\n",
      "> Adding chunk: (i) Immunity. Whether P1 is maintained in cells...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: IfCoi synthesis prevails, C1 becomes inactivate...\n",
      "> Adding chunk: IfCoi synthesis prevails, C1 becomes inactivate...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Bivalent operators names are underlined; consti...\n",
      "> Adding chunk: Bivalent operators names are underlined; consti...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: TABLE 7. Known and potential binding sites in P...\n",
      "> Adding chunk: TABLE 7. Known and potential binding sites in P...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 3 Internal to icd\n",
      "47945 –47959 /H11001 GAATTTTA...\n",
      "> Adding chunk: 3 Internal to icd\n",
      "47945 –47959 /H11001 GAATTTTA...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: An additional component of ImmC or a separate i...\n",
      "> Adding chunk: An additional component of ImmC or a separate i...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Participation of Lxc in the C1-operator complex...\n",
      "> Adding chunk: Participation of Lxc in the C1-operator complex...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: One differ-ence is that the ADP form of DnaA su...\n",
      "> Adding chunk: One differ-ence is that the ADP form of DnaA su...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: incA region (Table 5), and thus occupation of i...\n",
      "> Adding chunk: incA region (Table 5), and thus occupation of i...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: These proteins (ArgR, for whicha putative bindi...\n",
      "> Adding chunk: These proteins (ArgR, for whicha putative bindi...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Phd and Doccan form a complex. This interaction...\n",
      "> Adding chunk: Phd and Doccan form a complex. This interaction...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: ﬁcation enzyme, EcoPI. This bifunctional enzyme...\n",
      "> Adding chunk: ﬁcation enzyme, EcoPI. This bifunctional enzyme...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The superimmunity (or superinfection exclusion)...\n",
      "> Adding chunk: The superimmunity (or superinfection exclusion)...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: cascade has been subdivided into early, middle,...\n",
      "> Adding chunk: cascade has been subdivided into early, middle,...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Some of these tRNAs supplement the host pool of...\n",
      "> Adding chunk: Some of these tRNAs supplement the host pool of...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: All of them are pres-ent in P1 tRNA3. Certain o...\n",
      "> Adding chunk: All of them are pres-ent in P1 tRNA3. Certain o...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: both single ATAs and tandem ATAs. In eight, ATA...\n",
      "> Adding chunk: both single ATAs and tandem ATAs. In eight, ATA...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: It contains multiple sets of GATC se-\n",
      "quences a...\n",
      "> Adding chunk: It contains multiple sets of GATC se-\n",
      "quences a...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: host helicase-loading factor (DnaC), and the Dn...\n",
      "> Adding chunk: host helicase-loading factor (DnaC), and the Dn...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Its homologies to Dam are contained solelywithi...\n",
      "> Adding chunk: Its homologies to Dam are contained solelywithi...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: It\n",
      "encodes a 455-residue protein (195) identica...\n",
      "> Adding chunk: It\n",
      "encodes a 455-residue protein (195) identica...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The ban gene is the penultimate gene in a 5-kb ...\n",
      "> Adding chunk: The ban gene is the penultimate gene in a 5-kb ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The sites recognized by the Lpa protein\n",
      "and req...\n",
      "> Adding chunk: The sites recognized by the Lpa protein\n",
      "and req...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Coordinates refer to positions of promoters in ...\n",
      "> Adding chunk: Coordinates refer to positions of promoters in ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: in control regions of certain operons are commo...\n",
      "> Adding chunk: in control regions of certain operons are commo...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The 11 late promoters control at least 12 andas...\n",
      "> Adding chunk: The 11 late promoters control at least 12 andas...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: phages, a scaffold of the head is assembled on ...\n",
      "> Adding chunk: phages, a scaffold of the head is assembled on ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: It appearsto represent a truncated product of g...\n",
      "> Adding chunk: It appearsto represent a truncated product of g...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The tenth gene in the array, pap, like ppp, can...\n",
      "> Adding chunk: The tenth gene in the array, pap, like ppp, can...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thus, a likely explanation for similar pleio-tr...\n",
      "> Adding chunk: Thus, a likely explanation for similar pleio-tr...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: factor to do so (287). The requirement can be s...\n",
      "> Adding chunk: factor to do so (287). The requirement can be s...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thetubgene is ﬂanked by genes whose products re...\n",
      "> Adding chunk: Thetubgene is ﬂanked by genes whose products re...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Whether bplA and 3are identical re-\n",
      "mains to be...\n",
      "> Adding chunk: Whether bplA and 3are identical re-\n",
      "mains to be...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: larities to that of T4 gp5, a 575-residue prote...\n",
      "> Adding chunk: larities to that of T4 gp5, a 575-residue prote...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The tail ﬁber genes of P1, p15B, and Mu appear ...\n",
      "> Adding chunk: The tail ﬁber genes of P1, p15B, and Mu appear ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The cinsegment, ﬂanked by\n",
      "cixPp and cixL, does ...\n",
      "> Adding chunk: The cinsegment, ﬂanked by\n",
      "cixPp and cixL, does ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: inversion will be especially infrequent during ...\n",
      "> Adding chunk: inversion will be especially infrequent during ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: unclear whether the Hdf protein contributes to ...\n",
      "> Adding chunk: unclear whether the Hdf protein contributes to ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: A loop region in /H9258was identi ﬁed that dire...\n",
      "> Adding chunk: A loop region in /H9258was identi ﬁed that dire...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: two similar residues), suggesting that Hot can ...\n",
      "> Adding chunk: two similar residues), suggesting that Hot can ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Then the production of phagesceases and even ch...\n",
      "> Adding chunk: Then the production of phagesceases and even ch...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: moter of lydA and lydB (LPdar), there is no pre...\n",
      "> Adding chunk: moter of lydA and lydB (LPdar), there is no pre...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: As a general rule, a phage holin gene is associ...\n",
      "> Adding chunk: As a general rule, a phage holin gene is associ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: J. Mol. Biol. 173:307–324.\n",
      "7.Abremski, K., and ...\n",
      "> Adding chunk: J. Mol. Biol. 173:307–324.\n",
      "7.Abremski, K., and ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1994. Escherichia coli Fis\n",
      "and DnaA proteins bi...\n",
      "> Adding chunk: 1994. Escherichia coli Fis\n",
      "and DnaA proteins bi...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: quences of the EcoP1 restriction-modi ﬁcation e...\n",
      "> Adding chunk: quences of the EcoP1 restriction-modi ﬁcation e...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Biosystems 30:161–171.\n",
      "37.Bouet, J. Y., and B. ...\n",
      "> Adding chunk: Biosystems 30:161–171.\n",
      "37.Bouet, J. Y., and B. ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Mol. Microbiol. 37:467–476.\n",
      "50.Chattoraj, D. K....\n",
      "> Adding chunk: Mol. Microbiol. 37:467–476.\n",
      "50.Chattoraj, D. K....\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Discovery of a protein phosphatase\n",
      "activity enc...\n",
      "> Adding chunk: Discovery of a protein phosphatase\n",
      "activity enc...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1990. An attempt\n",
      "to unify the structure of poly...\n",
      "> Adding chunk: 1990. An attempt\n",
      "to unify the structure of poly...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 83.Edgar, R., D. K. Chattoraj, and M. Yarmolins...\n",
      "> Adding chunk: 83.Edgar, R., D. K. Chattoraj, and M. Yarmolins...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1999. Stability and DNA binding of the phd\n",
      "prot...\n",
      "> Adding chunk: 1999. Stability and DNA binding of the phd\n",
      "prot...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1983. DNA restriction-\n",
      "modi ﬁcation enzymes of ...\n",
      "> Adding chunk: 1983. DNA restriction-\n",
      "modi ﬁcation enzymes of ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 178:6258 –6265.\n",
      "118. Hawley, D. K., and W. R. M...\n",
      "> Adding chunk: 178:6258 –6265.\n",
      "118. Hawley, D. K., and W. R. M...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1994. C1\n",
      "repressor-mediated looping is involved...\n",
      "> Adding chunk: 1994. C1\n",
      "repressor-mediated looping is involved...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: clusion gene ( sieA) of bacteriophage P22: iden...\n",
      "> Adding chunk: clusion gene ( sieA) of bacteriophage P22: iden...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 150. Iida, S., H. Huber, R. Hiestand-Nauer, J. ...\n",
      "> Adding chunk: 150. Iida, S., H. Huber, R. Hiestand-Nauer, J. ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Nucleic Acids Res. 18:3903 –3911.\n",
      "160. Ito, J.,...\n",
      "> Adding chunk: Nucleic Acids Res. 18:3903 –3911.\n",
      "160. Ito, J.,...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: IV.\n",
      "Active transducing bacteriophage P1CM produ...\n",
      "> Adding chunk: IV.\n",
      "Active transducing bacteriophage P1CM produ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 183. Lanka, E., B. Geschke, and H. Schuster. 19...\n",
      "> Adding chunk: 183. Lanka, E., B. Geschke, and H. Schuster. 19...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 193. Lehnherr, H., M. Velleman, A. Guidolin, an...\n",
      "> Adding chunk: 193. Lehnherr, H., M. Velleman, A. Guidolin, an...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: J. Mol. Biol. 259:366–382.\n",
      "204. Lowe, T. M., an...\n",
      "> Adding chunk: J. Mol. Biol. 259:366–382.\n",
      "204. Lowe, T. M., an...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1998. Base composition\n",
      "skews, replication orien...\n",
      "> Adding chunk: 1998. Base composition\n",
      "skews, replication orien...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1987. An open reading frame in the Esche-\n",
      "richi...\n",
      "> Adding chunk: 1987. An open reading frame in the Esche-\n",
      "richi...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: J. Mol. Biol.\n",
      "236:710–724.\n",
      "237. Ogawa, T. 1975....\n",
      "> Adding chunk: J. Mol. Biol.\n",
      "236:710–724.\n",
      "237. Ogawa, T. 1975....\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 246. Pierce, J. C., and N. L. Sternberg. 1992. ...\n",
      "> Adding chunk: 246. Pierce, J. C., and N. L. Sternberg. 1992. ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: J. Bacteriol. 175:2833 –2838.\n",
      "258. Rodionov, O....\n",
      "> Adding chunk: J. Bacteriol. 175:2833 –2838.\n",
      "258. Rodionov, O....\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Proc. Natl. Acad. Sci. USA 87:1109 –\n",
      "1113.\n",
      "271....\n",
      "> Adding chunk: Proc. Natl. Acad. Sci. USA 87:1109 –\n",
      "1113.\n",
      "271....\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Mol. Microbiol. 21:501–509.\n",
      "285. Sharrocks, A. ...\n",
      "> Adding chunk: Mol. Microbiol. 21:501–509.\n",
      "285. Sharrocks, A. ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: II. Organization of replicon elements. J. Mol. ...\n",
      "> Adding chunk: II. Organization of replicon elements. J. Mol. ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: accessory proteins. V. Theta encoded by holE . ...\n",
      "> Adding chunk: accessory proteins. V. Theta encoded by holE . ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Velleman, M., B. Dreisekelmann, and H. Schuster...\n",
      "> Adding chunk: Velleman, M., B. Dreisekelmann, and H. Schuster...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1990. Structure of the bacteriophage T4\n",
      "basepla...\n",
      "> Adding chunk: 1990. Structure of the bacteriophage T4\n",
      "basepla...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Biol. 274:1–7.\n",
      "340. Xu, G. L., W. Kapfer, J. Wa...\n",
      "> Adding chunk: Biol. 274:1–7.\n",
      "340. Xu, G. L., W. Kapfer, J. Wa...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Biochem. Biophys.\n",
      "Res. Commun. 203:798–804.\n",
      "351...\n",
      "> Adding chunk: Biochem. Biophys.\n",
      "Res. Commun. 203:798–804.\n",
      "351...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: J. Mol. Biol. (1986) 187, 197-212 \n",
      "Bacteriophag...\n",
      "> Adding chunk: J. Mol. Biol. (1986) 187, 197-212 \n",
      "Bacteriophag...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1985). In the Pl life-cycle, the lox-Cre system...\n",
      "> Adding chunk: 1985). In the Pl life-cycle, the lox-Cre system...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 198 N. Sternberg et al. \n",
      "daughter molecules are...\n",
      "> Adding chunk: 198 N. Sternberg et al. \n",
      "daughter molecules are...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: l(a)). The \n",
      "construction and structure of cre+ ...\n",
      "> Adding chunk: l(a)). The \n",
      "construction and structure of cre+ ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Pl cre Gene and its Regulatory Signals 199 \n",
      "Pi ...\n",
      "> Adding chunk: Pl cre Gene and its Regulatory Signals 199 \n",
      "Pi ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 200 N. Sternberg et al. \n",
      "lo-fold in L-broth wit...\n",
      "> Adding chunk: 200 N. Sternberg et al. \n",
      "lo-fold in L-broth wit...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Analysis \n",
      "of the 1 phage from the Plcre+ lysoge...\n",
      "> Adding chunk: Analysis \n",
      "of the 1 phage from the Plcre+ lysoge...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Deleted DNA was characterized by agarose and po...\n",
      "> Adding chunk: Deleted DNA was characterized by agarose and po...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: PI we Gene and its Regulatory Signals 201 \n",
      "plas...\n",
      "> Adding chunk: PI we Gene and its Regulatory Signals 201 \n",
      "plas...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: At Asso = 0.3, 3-ml of cells were \n",
      "centrifuged ...\n",
      "> Adding chunk: At Asso = 0.3, 3-ml of cells were \n",
      "centrifuged ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 2). \n",
      "The results with pRHlO3 deletions A6 and A...\n",
      "> Adding chunk: 2). \n",
      "The results with pRHlO3 deletions A6 and A...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 202 N. Sternberg et al. \n",
      "Promoter \n",
      "activity \n",
      "(g...\n",
      "> Adding chunk: 202 N. Sternberg et al. \n",
      "Promoter \n",
      "activity \n",
      "(g...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: PI we Gene and its Regulatory Signals 203 \n",
      "Tabl...\n",
      "> Adding chunk: PI we Gene and its Regulatory Signals 203 \n",
      "Tabl...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Plasmids \n",
      "pNS2353 and pNS2355 are described in ...\n",
      "> Adding chunk: Plasmids \n",
      "pNS2353 and pNS2355 are described in ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 204 N. Sternberg et al. \n",
      "4-l 2366-3 \n",
      "-2366-6 \n",
      "-...\n",
      "> Adding chunk: 204 N. Sternberg et al. \n",
      "4-l 2366-3 \n",
      "-2366-6 \n",
      "-...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: NS2300 \n",
      "NS2300 \n",
      "NS23oO \n",
      "NS2300 \n",
      "NS23OO Infectin...\n",
      "> Adding chunk: NS2300 \n",
      "NS2300 \n",
      "NS23oO \n",
      "NS2300 \n",
      "NS23OO Infectin...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: +: i’ TGCGCAGCTG GACGTAAACT CCTCTTCAGA CCTAATAA...\n",
      "> Adding chunk: +: i’ TGCGCAGCTG GACGTAAACT CCTCTTCAGA CCTAATAA...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: ..- ~-or‘, -~ 4 + - -or12 (cm- \n",
      "p-pfi-- + \n",
      "-550...\n",
      "> Adding chunk: ..- ~-or‘, -~ 4 + - -or12 (cm- \n",
      "p-pfi-- + \n",
      "-550...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: thr arq ala leu glu gly lie phe glu ala thr his...\n",
      "> Adding chunk: thr arq ala leu glu gly lie phe glu ala thr his...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 206 N. Sternberg et al. \n",
      "lo ) Promotsr seq”*ncr...\n",
      "> Adding chunk: 206 N. Sternberg et al. \n",
      "lo ) Promotsr seq”*ncr...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: T change at l)osition - 668. \n",
      "(iii) orfl \n",
      "An an...\n",
      "> Adding chunk: T change at l)osition - 668. \n",
      "(iii) orfl \n",
      "An an...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Pl we Gene and its Regulatory Signale 207 \n",
      "orf ...\n",
      "> Adding chunk: Pl we Gene and its Regulatory Signale 207 \n",
      "orf ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: A presumptive pR1 promoter is \n",
      "present at posit...\n",
      "> Adding chunk: A presumptive pR1 promoter is \n",
      "present at posit...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 208 N. Sternberg et al. \n",
      "Table 5 \n",
      "The eSfect of...\n",
      "> Adding chunk: 208 N. Sternberg et al. \n",
      "Table 5 \n",
      "The eSfect of...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: pR3) 96 \n",
      "Acre-103 (pR2, pR3) 88 \n",
      "Acre-103A6 (pR...\n",
      "> Adding chunk: pR3) 96 \n",
      "Acre-103 (pR2, pR3) 88 \n",
      "Acre-103A6 (pR...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: PI me Gene and its Regulatory Signals 209 \n",
      "Tabl...\n",
      "> Adding chunk: PI me Gene and its Regulatory Signals 209 \n",
      "Tabl...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: concentrated IO-fold in L-broth conta:ning 5 mM...\n",
      "> Adding chunk: concentrated IO-fold in L-broth conta:ning 5 mM...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 210 N. Sternberg et al. \n",
      "growth in a recA- host...\n",
      "> Adding chunk: 210 N. Sternberg et al. \n",
      "growth in a recA- host...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: but’ is dispensable for both of those functions...\n",
      "> Adding chunk: but’ is dispensable for both of those functions...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: PI cre Gene ad its Regulatory Signals 211 \n",
      "stra...\n",
      "> Adding chunk: PI cre Gene ad its Regulatory Signals 211 \n",
      "stra...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: We thank Drs Robert Zagursky and Susan Bear for...\n",
      "> Adding chunk: We thank Drs Robert Zagursky and Susan Bear for...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Jorgensen. R. A., Rothstein, S. J. & Reznikoff....\n",
      "> Adding chunk: Jorgensen. R. A., Rothstein, S. J. & Reznikoff....\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 212 N. Sternberg et al. \n",
      "Segev. N. & Cohen, G. ...\n",
      "> Adding chunk: 212 N. Sternberg et al. \n",
      "Segev. N. & Cohen, G. ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: tJ. Md. Bid. (1992) 226, 661-673 \n",
      "Cre-lox Recom...\n",
      "> Adding chunk: tJ. Md. Bid. (1992) 226, 661-673 \n",
      "Cre-lox Recom...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int system (Landy. 1989). have \n",
      "more diverse ta...\n",
      "> Adding chunk: Int system (Landy. 1989). have \n",
      "more diverse ta...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Ordered \n",
      "synapsts Random \n",
      "Sy”apSlS \n",
      "Hi ~-.- \n",
      "Fr...\n",
      "> Adding chunk: Ordered \n",
      "synapsts Random \n",
      "Sy”apSlS \n",
      "Hi ~-.- \n",
      "Fr...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The Mechnn.ism of C’w Rrcornhination in \\‘ivo 6...\n",
      "> Adding chunk: The Mechnn.ism of C’w Rrcornhination in \\‘ivo 6...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: E. X2-c.rr l)hapr \\vas a gift from I<. Saucr at...\n",
      "> Adding chunk: E. X2-c.rr l)hapr \\vas a gift from I<. Saucr at...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: wrw c*hosen to alloy direct c*omparison with re...\n",
      "> Adding chunk: wrw c*hosen to alloy direct c*omparison with re...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: samples were electro- \n",
      "l)hortJstvl in agarostl ...\n",
      "> Adding chunk: samples were electro- \n",
      "l)hortJstvl in agarostl ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: wcv)tnl)i- \n",
      "nases. To tleterminc~ wtrivtr prope...\n",
      "> Adding chunk: wcv)tnl)i- \n",
      "nases. To tleterminc~ wtrivtr prope...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: In paraltrl. WIIS harhoriny only pKS-5 WNY~ inf...\n",
      "> Adding chunk: In paraltrl. WIIS harhoriny only pKS-5 WNY~ inf...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Inducl,an,ln,sslron RE.7 atE I I \n",
      "“73 - \n",
      "p,*RES...\n",
      "> Adding chunk: Inducl,an,ln,sslron RE.7 atE I I \n",
      "“73 - \n",
      "p,*RES...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: results from inc.otnpletc nicktng by UNase I. n...\n",
      "> Adding chunk: results from inc.otnpletc nicktng by UNase I. n...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \\z’cl used t hr rwalts shown in Figure 3 to cBo...\n",
      "> Adding chunk: \\z’cl used t hr rwalts shown in Figure 3 to cBo...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Wi I). I<. :lflfrttln t,t al \n",
      "tl I I I I I II \n",
      "...\n",
      "> Adding chunk: Wi I). I<. :lflfrttln t,t al \n",
      "tl I I I I I II \n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: I+, exc.rpt that thus ittc.ttl)atiott tc~trrt)r...\n",
      "> Adding chunk: I+, exc.rpt that thus ittc.ttl)atiott tc~trrt)r...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The Mechanism of Ore Recombination in Viva 667 ...\n",
      "> Adding chunk: The Mechanism of Ore Recombination in Viva 667 ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 6(a) lane 2). However, in t’he presence \n",
      "of the...\n",
      "> Adding chunk: 6(a) lane 2). However, in t’he presence \n",
      "of the...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 0 2 6 120 \n",
      "4 Time (mln) 4 \n",
      "Phage Cre Time (mln)...\n",
      "> Adding chunk: 0 2 6 120 \n",
      "4 Time (mln) 4 \n",
      "Phage Cre Time (mln)...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Discussion \n",
      "(a) In vitro versus in viva strdiw ...\n",
      "> Adding chunk: Discussion \n",
      "(a) In vitro versus in viva strdiw ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The Mechanism of Cre Recombination in Vivo 669 ...\n",
      "> Adding chunk: The Mechanism of Cre Recombination in Vivo 669 ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: We SW no evittenee for significant nicking of o...\n",
      "> Adding chunk: We SW no evittenee for significant nicking of o...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: But it is also clear from the residual recom- \n",
      "...\n",
      "> Adding chunk: But it is also clear from the residual recom- \n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: lii0 f). K. .-lfffLnl.s r1 al. \n",
      "The hia,s (Ire ...\n",
      "> Adding chunk: lii0 f). K. .-lfffLnl.s r1 al. \n",
      "The hia,s (Ire ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Tt is extraordinary that thtb unit c~)p~ \n",
      "J’l p...\n",
      "> Adding chunk: Tt is extraordinary that thtb unit c~)p~ \n",
      "J’l p...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: cation. After completion of DKA synthesis. \n",
      "(Ir...\n",
      "> Adding chunk: cation. After completion of DKA synthesis. \n",
      "(Ir...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: topoisomerase that maintains I)NA suprrcoiIing ...\n",
      "> Adding chunk: topoisomerase that maintains I)NA suprrcoiIing ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 29. pp. 107-126. Houst.on, TX. V.S.L\\. \n",
      "Bliska,...\n",
      "> Adding chunk: 29. pp. 107-126. Houst.on, TX. V.S.L\\. \n",
      "Bliska,...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 4. 27 -IO. \n",
      "(:lasgov. .\\. (‘.. Hughes. Ii. ‘I 8...\n",
      "> Adding chunk: 4. 27 -IO. \n",
      "(:lasgov. .\\. (‘.. Hughes. Ii. ‘I 8...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: ‘I’. &Y Sltistto. Ii. (l!,!rl). (:yra~ inbib- \n",
      "...\n",
      "> Adding chunk: ‘I’. &Y Sltistto. Ii. (l!,!rl). (:yra~ inbib- \n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The Mechanism qf Cre Recombkation in Viva 67.3 ...\n",
      "> Adding chunk: The Mechanism qf Cre Recombkation in Viva 67.3 ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: ft. E. \n",
      "( I!)x~). (iet1t.t i(. analysis of muta...\n",
      "> Adding chunk: ft. E. \n",
      "( I!)x~). (iet1t.t i(. analysis of muta...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: (‘dl. 46. 511 350. \n",
      "FVattg. ,I. (‘. (1985). DK‘...\n",
      "> Adding chunk: (‘dl. 46. 511 350. \n",
      "FVattg. ,I. (‘. (1985). DK‘...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The Doc Toxin and Phd Antidote Proteins of the ...\n",
      "> Adding chunk: The Doc Toxin and Phd Antidote Proteins of the ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Addiction systems involving a stable toxin and ...\n",
      "> Adding chunk: Addiction systems involving a stable toxin and ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The costs of publication of this article were d...\n",
      "> Adding chunk: The costs of publication of this article were d...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Hewlett Packard 8452A diode array spectrophotom...\n",
      "> Adding chunk: Hewlett Packard 8452A diode array spectrophotom...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: !14Cb,Docexp~A DocMDoc~r22rb2!!\n",
      "16~1/K d!Cb,Doc...\n",
      "> Adding chunk: !14Cb,Docexp~A DocMDoc~r22rb2!!\n",
      "16~1/K d!Cb,Doc...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: within 1 or 2 Da of those expected from the pro...\n",
      "> Adding chunk: within 1 or 2 Da of those expected from the pro...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Fitting of this\n",
      "data to a 2P 1DNP2D model gave ...\n",
      "> Adding chunk: Fitting of this\n",
      "data to a 2P 1DNP2D model gave ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: (the donor molecule). Because energy transfer d...\n",
      "> Adding chunk: (the donor molecule). Because energy transfer d...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: are not common in biological systems and must b...\n",
      "> Adding chunk: are not common in biological systems and must b...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Other\n",
      "conditions identical to panel A. C,increa...\n",
      "> Adding chunk: Other\n",
      "conditions identical to panel A. C,increa...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: the cell. If the assumptions that underlie thes...\n",
      "> Adding chunk: the cell. If the assumptions that underlie thes...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 180, 6342–6351\n",
      "6. Gazit, E. & Sauer, R. T. (199...\n",
      "> Adding chunk: 180, 6342–6351\n",
      "6. Gazit, E. & Sauer, R. T. (199...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The Role of O-antigen in P1\n",
      "Transduction of Shi...\n",
      "> Adding chunk: The Role of O-antigen in P1\n",
      "Transduction of Shi...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: A higher transduction efﬁciency of P1(S’) impro...\n",
      "> Adding chunk: A higher transduction efﬁciency of P1(S’) impro...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Y.W. Huan, J. Fa-arun and B. Wang Journal of Mo...\n",
      "> Adding chunk: Y.W. Huan, J. Fa-arun and B. Wang Journal of Mo...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: sites is mediated by the Cin recombinase, which...\n",
      "> Adding chunk: sites is mediated by the Cin recombinase, which...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Furthermore, mutant strains of S. ﬂexneri 2a\n",
      "24...\n",
      "> Adding chunk: Furthermore, mutant strains of S. ﬂexneri 2a\n",
      "24...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: coli\n",
      "C, while the host range of P1(S’) (with bl...\n",
      "> Adding chunk: coli\n",
      "C, while the host range of P1(S’) (with bl...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Results\n",
      "S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T...\n",
      "> Adding chunk: Results\n",
      "S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Infection of E. coli K12\n",
      "NCM3722 and E. coli B ...\n",
      "> Adding chunk: Infection of E. coli K12\n",
      "NCM3722 and E. coli B ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The genes\n",
      "selected for generating the O-antigen...\n",
      "> Adding chunk: The genes\n",
      "selected for generating the O-antigen...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: well as E. coli O351were shown to give a semi-\n",
      "...\n",
      "> Adding chunk: well as E. coli O351were shown to give a semi-\n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Y.W. Huan, J. Fa-arun and B. Wang Journal of Mo...\n",
      "> Adding chunk: Y.W. Huan, J. Fa-arun and B. Wang Journal of Mo...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: visualised via silver staining. LPS extracted f...\n",
      "> Adding chunk: visualised via silver staining. LPS extracted f...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: While no phagemid transductant\n",
      "was recovered af...\n",
      "> Adding chunk: While no phagemid transductant\n",
      "was recovered af...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The genes involved in the acetylation and gluco...\n",
      "> Adding chunk: The genes involved in the acetylation and gluco...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 8 mutant strains of S. ﬂexneri 2a 2457O and S. ...\n",
      "> Adding chunk: 8 mutant strains of S. ﬂexneri 2a 2457O and S. ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a\n",
      "M...\n",
      "> Adding chunk: The LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a\n",
      "M...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Contrastingly, treatment of P1(S’)\n",
      "lysate with ...\n",
      "> Adding chunk: Contrastingly, treatment of P1(S’)\n",
      "lysate with ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Figure 4. The LPS of S. ﬂexneri 2a 2457O, S. ﬂe...\n",
      "> Adding chunk: Figure 4. The LPS of S. ﬂexneri 2a 2457O, S. ﬂe...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: phagemid transductants after P1(S) infection of...\n",
      "> Adding chunk: phagemid transductants after P1(S) infection of...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 4 , n= 6, p < 0.0005) and 50 % ( s= 3.4, n=6 ,\n",
      "...\n",
      "> Adding chunk: 4 , n= 6, p < 0.0005) and 50 % ( s= 3.4, n=6 ,\n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: We therefore hypothesised that an\n",
      "increase in t...\n",
      "> Adding chunk: We therefore hypothesised that an\n",
      "increase in t...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Figure 5. Low concentration of E. coli O111:B4 ...\n",
      "> Adding chunk: Figure 5. Low concentration of E. coli O111:B4 ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Y.W. Huan, J. Fa-arun and B. Wang Journal of Mo...\n",
      "> Adding chunk: Y.W. Huan, J. Fa-arun and B. Wang Journal of Mo...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: with varying volumes of the cas9-shiA orcas9 -N...\n",
      "> Adding chunk: with varying volumes of the cas9-shiA orcas9 -N...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The effect(s) of O-antigen on P1 transduction\n",
      "T...\n",
      "> Adding chunk: The effect(s) of O-antigen on P1 transduction\n",
      "T...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: p < 0.0005 are shown as ***, for\n",
      "comparisons of...\n",
      "> Adding chunk: p < 0.0005 are shown as ***, for\n",
      "comparisons of...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Cas9 antimicrobial system, it remained unknown ...\n",
      "> Adding chunk: Cas9 antimicrobial system, it remained unknown ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Comparisons between\n",
      "the C-terminus of P1 S’ tai...\n",
      "> Adding chunk: Comparisons between\n",
      "the C-terminus of P1 S’ tai...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: sistent with the fact that P1(S) infected most ...\n",
      "> Adding chunk: sistent with the fact that P1(S) infected most ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The appli-\n",
      "cation of P1(S’) to target other ser...\n",
      "> Adding chunk: The appli-\n",
      "cation of P1(S’) to target other ser...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: accession number JALKIH000000000, Bioproject:\n",
      "P...\n",
      "> Adding chunk: accession number JALKIH000000000, Bioproject:\n",
      "P...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: P1 phagemid lysate preparation\n",
      "Preparation of p...\n",
      "> Adding chunk: P1 phagemid lysate preparation\n",
      "Preparation of p...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Ten microlitres of crude\n",
      "P1 phage lysates, or 1...\n",
      "> Adding chunk: Ten microlitres of crude\n",
      "P1 phage lysates, or 1...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: tion were platted onto LB agar supplemented wit...\n",
      "> Adding chunk: tion were platted onto LB agar supplemented wit...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Extraction and visualisation of LPS\n",
      "LPS of S. ﬂ...\n",
      "> Adding chunk: Extraction and visualisation of LPS\n",
      "LPS of S. ﬂ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Laemmli buffer was added to samples. Ten\n",
      "microl...\n",
      "> Adding chunk: Laemmli buffer was added to samples. Ten\n",
      "microl...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Fifty microlitres of indicator cells were added...\n",
      "> Adding chunk: Fifty microlitres of indicator cells were added...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: plemented with 5 mM of Ca2+and 10 mM Mg2+)\n",
      "at 3...\n",
      "> Adding chunk: plemented with 5 mM of Ca2+and 10 mM Mg2+)\n",
      "at 3...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: DECLARATION OF COMPETING INTEREST\n",
      "The authors d...\n",
      "> Adding chunk: DECLARATION OF COMPETING INTEREST\n",
      "The authors d...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 2.Bengtsson, R.J., Simpkin, A.J., Pulford, C.V....\n",
      "> Adding chunk: 2.Bengtsson, R.J., Simpkin, A.J., Pulford, C.V....\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 97,\n",
      "6640–6645 .\n",
      "13. Davis, M.R., Goldberg, J.B....\n",
      "> Adding chunk: 97,\n",
      "6640–6645 .\n",
      "13. Davis, M.R., Goldberg, J.B....\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Programmable\n",
      "removal of bacterial strains by us...\n",
      "> Adding chunk: Programmable\n",
      "removal of bacterial strains by us...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: the cross-over sites cixfor the inversion of th...\n",
      "> Adding chunk: the cross-over sites cixfor the inversion of th...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Structure and genetics of Escherichia coli O an...\n",
      "> Adding chunk: Structure and genetics of Escherichia coli O an...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Extraction, puriﬁcation and\n",
      "characterization of...\n",
      "> Adding chunk: Extraction, puriﬁcation and\n",
      "characterization of...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 61.Toussaint, A., Lefebvre, N., Scott, J.R., Co...\n",
      "> Adding chunk: 61.Toussaint, A., Lefebvre, N., Scott, J.R., Co...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: REVIEW\n",
      "Phage DNA packaging\n",
      "Hisao Fujisawa1,a,xa...\n",
      "> Adding chunk: REVIEW\n",
      "Phage DNA packaging\n",
      "Hisao Fujisawa1,a,xa...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: hydrolysis ( l, Becker et al. 1977; f-29, Guo e...\n",
      "> Adding chunk: hydrolysis ( l, Becker et al. 1977; f-29, Guo e...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1990; P22, Casjens et al. 1982; SPP1, Travares ...\n",
      "> Adding chunk: 1990; P22, Casjens et al. 1982; SPP1, Travares ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: mutants in prohead assembly and therefore requi...\n",
      "> Adding chunk: mutants in prohead assembly and therefore requi...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: l-terminase intro-\n",
      "duces nicks staggered by 12 ...\n",
      "> Adding chunk: l-terminase intro-\n",
      "duces nicks staggered by 12 ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: polymerase and packaging protein gp19 (Zhang &\n",
      "...\n",
      "> Adding chunk: polymerase and packaging protein gp19 (Zhang &\n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: the packaging machinery recognizes a transcript...\n",
      "> Adding chunk: the packaging machinery recognizes a transcript...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: and the C-terminal domain accessible to proteas...\n",
      "> Adding chunk: and the C-terminal domain accessible to proteas...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1993). This value is very close to\n",
      "the value (2...\n",
      "> Adding chunk: 1993). This value is very close to\n",
      "the value (2...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: mechanism (Serwer 1988). However, it is difﬁcul...\n",
      "> Adding chunk: mechanism (Serwer 1988). However, it is difﬁcul...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Black, L.W . (1989) DNA packaging in dsDNA bact...\n",
      "> Adding chunk: Black, L.W . (1989) DNA packaging in dsDNA bact...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Nature 292, 175–176.\n",
      "Guo, P ., Erickson, S. & A...\n",
      "> Adding chunk: Nature 292, 175–176.\n",
      "Guo, P ., Erickson, S. & A...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: (1994) Analysis of\n",
      "functional domains of the pa...\n",
      "> Adding chunk: (1994) Analysis of\n",
      "functional domains of the pa...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: T omka, M. & Catalano, C.E. (1993) Kinetic char...\n",
      "> Adding chunk: T omka, M. & Catalano, C.E. (1993) Kinetic char...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: \n",
      "> Adding chunk: \n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "Th...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "Th...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "IN...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "IN...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "Wh...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "Wh...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: One of the several biological insights in Jacob...\n",
      "> Adding chunk: One of the several biological insights in Jacob...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "co...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "co...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: About\n",
      "that time I learned that Robert Weisberg,...\n",
      "> Adding chunk: About\n",
      "that time I learned that Robert Weisberg,...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "In...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "In...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "Th...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "Th...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: His subsequent studies revealed that the locus ...\n",
      "> Adding chunk: His subsequent studies revealed that the locus ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "My...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "My...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "Wh...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "Wh...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thus, unlike λ’s attachment\n",
      "site of about 240 b...\n",
      "> Adding chunk: Thus, unlike λ’s attachment\n",
      "site of about 240 b...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "th...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "th...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "On...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "On...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The\n",
      "problem for segregation created by circular...\n",
      "> Adding chunk: The\n",
      "problem for segregation created by circular...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "Th...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "Th...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: It soon became apparent, when he discussed the ...\n",
      "> Adding chunk: It soon became apparent, when he discussed the ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "in...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "in...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: It efﬁciently displaces bound proteins in the p...\n",
      "> Adding chunk: It efﬁciently displaces bound proteins in the p...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "13...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "13...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "DI...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "DI...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: J. Mol. Biol. 150:467–86\n",
      "13. Takano T. 1971. Ba...\n",
      "> Adding chunk: J. Mol. Biol. 150:467–86\n",
      "13. Takano T. 1971. Ba...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "21...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "21...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Protein Eng. 5:87–91\n",
      "34. Guo F, Gopaul DN, Van ...\n",
      "> Adding chunk: Protein Eng. 5:87–91\n",
      "34. Guo F, Gopaul DN, Van ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "48...\n",
      "> Adding chunk: VI02CH02-Yarmolinsky ARI 1 October 2015 7:19\n",
      "48...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02-FrontMatter ARI 15 October 2015 16:33\n",
      "Annu...\n",
      "> Adding chunk: VI02-FrontMatter ARI 15 October 2015 16:33\n",
      "Annu...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02-FrontMatter ARI 15 October 2015 16:33\n",
      "Atta...\n",
      "> Adding chunk: VI02-FrontMatter ARI 15 October 2015 16:33\n",
      "Atta...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VI02-FrontMatter ARI 15 October 2015 16:33\n",
      "Immu...\n",
      "> Adding chunk: VI02-FrontMatter ARI 15 October 2015 16:33\n",
      "Immu...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: /gid00030/gid00035/gid00032/gid00030/gid00038/g...\n",
      "> Adding chunk: /gid00030/gid00035/gid00032/gid00030/gid00038/g...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: We postulate that the complex lytic system\n",
      "faci...\n",
      "> Adding chunk: We postulate that the complex lytic system\n",
      "faci...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 2 of 20\n",
      "small ...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 2 of 20\n",
      "small ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 3 of 20\n",
      "are pr...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 3 of 20\n",
      "are pr...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thus, LydA has been con-\n",
      "sidered a primary holi...\n",
      "> Adding chunk: Thus, LydA has been con-\n",
      "sidered a primary holi...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Known and predicted holin genes are in gray. Ad...\n",
      "> Adding chunk: Known and predicted holin genes are in gray. Ad...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 4 of 20\n",
      "induct...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 4 of 20\n",
      "induct...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thus, LydA has been described as the primary Ly...\n",
      "> Adding chunk: Thus, LydA has been described as the primary Ly...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 5 of 20\n",
      "a few ...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 5 of 20\n",
      "a few ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 2.4. Influence of the P1 lyz Gene Replacement w...\n",
      "> Adding chunk: 2.4. Influence of the P1 lyz Gene Replacement w...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 6 of 20\n",
      "The in...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 6 of 20\n",
      "The in...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: This conforms to the function of LydA as a cano...\n",
      "> Adding chunk: This conforms to the function of LydA as a cano...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 7 of 20\n",
      "Int. J...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 7 of 20\n",
      "Int. J...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 8 of 20\n",
      "Int. J...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 8 of 20\n",
      "Int. J...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: In the case of ʎ, this rupture is caused by hol...\n",
      "> Adding chunk: In the case of ʎ, this rupture is caused by hol...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: In the\n",
      "case of L, this rupture is caused by hol...\n",
      "> Adding chunk: In the\n",
      "case of L, this rupture is caused by hol...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 9 of 20\n",
      "Int. J...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 9 of 20\n",
      "Int. J...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: E. coli\n",
      "N99 lysogens were thermally induced and...\n",
      "> Adding chunk: E. coli\n",
      "N99 lysogens were thermally induced and...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 10 of 20\n",
      "was o...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 10 of 20\n",
      "was o...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: E. coli and E. cloacae lysogens were ther-\n",
      "mall...\n",
      "> Adding chunk: E. coli and E. cloacae lysogens were ther-\n",
      "mall...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The results of our studies on  P1 presented in ...\n",
      "> Adding chunk: The results of our studies on  P1 presented in ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 11 of 20\n",
      "3. Di...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 11 of 20\n",
      "3. Di...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Typically, in the lytic modules of phage genome...\n",
      "> Adding chunk: Typically, in the lytic modules of phage genome...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 12 of 20\n",
      "indic...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 12 of 20\n",
      "indic...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thus, the ability of P1 deprived of lydA to ret...\n",
      "> Adding chunk: Thus, the ability of P1 deprived of lydA to ret...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 13 of 20\n",
      "prote...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 13 of 20\n",
      "prote...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Unlike λ, P1 is a bacteriophage with\n",
      "a wide hos...\n",
      "> Adding chunk: Unlike λ, P1 is a bacteriophage with\n",
      "a wide hos...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 14 of 20\n",
      "cytop...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 14 of 20\n",
      "cytop...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The thermosensitivity of phage repressor allows...\n",
      "> Adding chunk: The thermosensitivity of phage repressor allows...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 15 of 20\n",
      "Table...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 15 of 20\n",
      "Table...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: lydA ,lydDRecombinational replacement with the ...\n",
      "> Adding chunk: lydA ,lydDRecombinational replacement with the ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 16 of 20\n",
      "4.2. ...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 16 of 20\n",
      "4.2. ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Adsorption was stopped by the addition of 1 M s...\n",
      "> Adding chunk: Adsorption was stopped by the addition of 1 M s...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 17 of 20\n",
      "their...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 17 of 20\n",
      "their...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Author Contributions: Conceptualization, A.B. a...\n",
      "> Adding chunk: Author Contributions: Conceptualization, A.B. a...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 18 of 20\n",
      "Ackno...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 18 of 20\n",
      "Ackno...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: ; Briers, Y.; Brøndsted, L.; Sørensen, M.C.H. T...\n",
      "> Adding chunk: ; Briers, Y.; Brøndsted, L.; Sørensen, M.C.H. T...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: [CrossRef]\n",
      "17. Savva, C.G.; Dewey, J.S.; Moussa...\n",
      "> Adding chunk: [CrossRef]\n",
      "17. Savva, C.G.; Dewey, J.S.; Moussa...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 19 of 20\n",
      "25. W...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 19 of 20\n",
      "25. W...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Microbiology 2005 ,151, 2331–2342. [CrossRef] [...\n",
      "> Adding chunk: Microbiology 2005 ,151, 2331–2342. [CrossRef] [...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Virology 1998 ,251,\n",
      "49–58. [CrossRef] [PubMed]\n",
      "...\n",
      "> Adding chunk: Virology 1998 ,251,\n",
      "49–58. [CrossRef] [PubMed]\n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 20 of 20\n",
      "55. W...\n",
      "> Adding chunk: Int. J. Mol. Sci. 2022 ,23, 4231 20 of 20\n",
      "55. W...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: ACS Infect.\n",
      "Dis. 2018 ,4, 1487–1498. [CrossRef]...\n",
      "> Adding chunk: ACS Infect.\n",
      "Dis. 2018 ,4, 1487–1498. [CrossRef]...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: J. Mol. Biol. 1996 ,259, 366–382. [CrossRef]\n",
      "[P...\n",
      "> Adding chunk: J. Mol. Biol. 1996 ,259, 366–382. [CrossRef]\n",
      "[P...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Vol.174,No.12\n",
      "DNAInversion Regions MinofPlasmid...\n",
      "> Adding chunk: Vol.174,No.12\n",
      "DNAInversion Regions MinofPlasmid...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thecomplex Minsystemiscloselyrelated tosimpleDN...\n",
      "> Adding chunk: Thecomplex Minsystemiscloselyrelated tosimpleDN...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VOL.174,1992 ~~~~EVOLUTION OFpl5B3ANDPHAGE P1TA...\n",
      "> Adding chunk: VOL.174,1992 ~~~~EVOLUTION OFpl5B3ANDPHAGE P1TA...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Plasmid\n",
      "pRM2O6 carriesthewild-type P1tailfibero...\n",
      "> Adding chunk: Plasmid\n",
      "pRM2O6 carriesthewild-type P1tailfibero...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 3938 SANDMEIER ETAL.\n",
      "mix\n",
      "aOala2a3a4a5a6a7cmix'I...\n",
      "> Adding chunk: 3938 SANDMEIER ETAL.\n",
      "mix\n",
      "aOala2a3a4a5a6a7cmix'I...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Substitution oftheinvertible DNAandtherecombina...\n",
      "> Adding chunk: Substitution oftheinvertible DNAandtherecombina...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: EVOLUTION OFplSBANDPHAGE P1TAILFIBER GENES 3939...\n",
      "> Adding chunk: EVOLUTION OFplSBANDPHAGE P1TAILFIBER GENES 3939...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: OnlytheSvMrandP(-)segments arehomolo-\n",
      "gousalsob...\n",
      "> Adding chunk: OnlytheSvMrandP(-)segments arehomolo-\n",
      "gousalsob...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 3940 SANDMEIER ETAL.\n",
      "consensus\n",
      "pl5BSvM1\n",
      "plSBSvM...\n",
      "> Adding chunk: 3940 SANDMEIER ETAL.\n",
      "consensus\n",
      "pl5BSvM1\n",
      "plSBSvM...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Asthediagram ofFig.4shows,severalofthebreakpoin...\n",
      "> Adding chunk: Asthediagram ofFig.4shows,severalofthebreakpoin...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thetandemrepeated elements blandb2oftheP1genome...\n",
      "> Adding chunk: Thetandemrepeated elements blandb2oftheP1genome...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: EVOLUTION OFpl5BANDPHAGE P1TAILFIBER GENES 3941...\n",
      "> Adding chunk: EVOLUTION OFpl5BANDPHAGE P1TAILFIBER GENES 3941...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: bT4protein38andXTfahaveaslightlyhigherhomology ...\n",
      "> Adding chunk: bT4protein38andXTfahaveaslightlyhigherhomology ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: QATQNSVLD IDENAMLT LNKQLSTAS*\n",
      "KLQGRRL. QATRNAVL...\n",
      "> Adding chunk: QATQNSVLD IDENAMLT LNKQLSTAS*\n",
      "KLQGRRL. QATRNAVL...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 3942 SANDMEIER ETAL.\n",
      "A B CFIG.7.Phageparticles ...\n",
      "> Adding chunk: 3942 SANDMEIER ETAL.\n",
      "A B CFIG.7.Phageparticles ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: SincetheMinregionappears toencodetailfibergenes...\n",
      "> Adding chunk: SincetheMinregionappears toencodetailfibergenes...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: EVOLUTION OFpl5BANDPHAGE P1TAILFIBERGENES 3943\n",
      "...\n",
      "> Adding chunk: EVOLUTION OFpl5BANDPHAGE P1TAILFIBERGENES 3943\n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Theintermediate region,withitsmosaiclike struc-...\n",
      "> Adding chunk: Theintermediate region,withitsmosaiclike struc-...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 3944 SANDMEIER ETAL.\n",
      "Biol.208:615-622.\n",
      "11.Hagga...\n",
      "> Adding chunk: 3944 SANDMEIER ETAL.\n",
      "Biol.208:615-622.\n",
      "11.Hagga...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 198...\n",
      "> Adding chunk: 22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 198...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 35.Sanger, F.,S.Nicklen, andA.R.Coulson. 1977.D...\n",
      "> Adding chunk: 35.Sanger, F.,S.Nicklen, andA.R.Coulson. 1977.D...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Vol.45,No.3 JOURNAL OFVIROLOGY, Mar.1983,p.1118...\n",
      "> Adding chunk: Vol.45,No.3 JOURNAL OFVIROLOGY, Mar.1983,p.1118...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: AgeneticmapforPl(Fig.2),whichincludes\n",
      "these103a...\n",
      "> Adding chunk: AgeneticmapforPl(Fig.2),whichincludes\n",
      "these103a...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MORPHOLOGY OFP1MUTANTS 1119\n",
      "B\n",
      "FIG.1.Micrographs...\n",
      "> Adding chunk: MORPHOLOGY OFP1MUTANTS 1119\n",
      "B\n",
      "FIG.1.Micrographs...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1120 WALKER ANDWALKER\n",
      "areknown, wedetermined th...\n",
      "> Adding chunk: 1120 WALKER ANDWALKER\n",
      "areknown, wedetermined th...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Theremaining lysate waschilled, and5mlwas\n",
      "centr...\n",
      "> Adding chunk: Theremaining lysate waschilled, and5mlwas\n",
      "centr...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MORPHOLOGY OFP1MUTANTS 1121\n",
      "SBAwasamodification...\n",
      "> Adding chunk: MORPHOLOGY OFP1MUTANTS 1121\n",
      "SBAwasamodification...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: RESULTS\n",
      "Weexamined, byelectron microscopy, ly-\n",
      "...\n",
      "> Adding chunk: RESULTS\n",
      "Weexamined, byelectron microscopy, ly-\n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1122 WALKER ANDWALKER\n",
      "cells,theselysatesweretes...\n",
      "> Adding chunk: 1122 WALKER ANDWALKER\n",
      "cells,theselysatesweretes...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thisdecrease isprobably caused\n",
      "bythesheathsubun...\n",
      "> Adding chunk: Thisdecrease isprobably caused\n",
      "bythesheathsubun...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MORPHOLOGY OFP1MUTANTS 1123\n",
      "TABLE 1.Phenotypic ...\n",
      "> Adding chunk: MORPHOLOGY OFP1MUTANTS 1123\n",
      "TABLE 1.Phenotypic ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 17\n",
      "17\n",
      "17\n",
      "17\n",
      "2\n",
      "12\n",
      "19?\n",
      "19\n",
      "19?\n",
      "20\n",
      "3\n",
      "16\n",
      "3\n",
      "3\n",
      "135 V\n",
      "1...\n",
      "> Adding chunk: 17\n",
      "17\n",
      "17\n",
      "17\n",
      "2\n",
      "12\n",
      "19?\n",
      "19\n",
      "19?\n",
      "20\n",
      "3\n",
      "16\n",
      "3\n",
      "3\n",
      "135 V\n",
      "1...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1124 WALKER ANDWALKER\n",
      "TABLE 1-Continued\n",
      "Mutant ...\n",
      "> Adding chunk: 1124 WALKER ANDWALKER\n",
      "TABLE 1-Continued\n",
      "Mutant ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MORPHOLOGY OFP1MUTANTS 1125\n",
      "FIG.4.Micrograph of...\n",
      "> Adding chunk: MORPHOLOGY OFP1MUTANTS 1125\n",
      "FIG.4.Micrograph of...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1126 WALKER ANDWALKER\n",
      "Mutant180(inclusterV;Fig....\n",
      "> Adding chunk: 1126 WALKER ANDWALKER\n",
      "Mutant180(inclusterV;Fig....\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MORPHOLOGY OFP1MUTANTS 1127\n",
      "FIG.7.Micrographs o...\n",
      "> Adding chunk: MORPHOLOGY OFP1MUTANTS 1127\n",
      "FIG.7.Micrographs o...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Othermutants inclusterVIII\n",
      "(am139, aml40,andam7...\n",
      "> Adding chunk: Othermutants inclusterVIII\n",
      "(am139, aml40,andam7...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1128 WALKER ANDWALKER\n",
      "t~~~~~ s P 1~~~~~~~~~~~~~...\n",
      "> Adding chunk: 1128 WALKER ANDWALKER\n",
      "t~~~~~ s P 1~~~~~~~~~~~~~...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MORPHOLOGY OFP1MUTANTS 1129\n",
      "PS\n",
      ",\n",
      "* .'-4\n",
      ".4\n",
      "IN\n",
      "....\n",
      "> Adding chunk: MORPHOLOGY OFP1MUTANTS 1129\n",
      "PS\n",
      ",\n",
      "* .'-4\n",
      ".4\n",
      "IN\n",
      "....\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1130 WALKER ANDWALKER\n",
      "FIG.11.Micrograph ofP1aml...\n",
      "> Adding chunk: 1130 WALKER ANDWALKER\n",
      "FIG.11.Micrograph ofP1aml...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MORPHOLOGY OFP1MUTANTS 1131\n",
      "FIG.13.Micrograph o...\n",
      "> Adding chunk: MORPHOLOGY OFP1MUTANTS 1131\n",
      "FIG.13.Micrograph o...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1132 WALKER ANDWALKER\n",
      "FIG.15.Micrograph ofP1am4...\n",
      "> Adding chunk: 1132 WALKER ANDWALKER\n",
      "FIG.15.Micrograph ofP1am4...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MORPHOLOGY OFP1MUTANTS 1133\n",
      "FIG.16.Micrograph o...\n",
      "> Adding chunk: MORPHOLOGY OFP1MUTANTS 1133\n",
      "FIG.16.Micrograph o...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1134 WALKER ANDWALKER\n",
      "and14)thatarelocatednearg...\n",
      "> Adding chunk: 1134 WALKER ANDWALKER\n",
      "and14)thatarelocatednearg...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Auniqueproperty ofPlis\n",
      "that,inaddition tonormal...\n",
      "> Adding chunk: Auniqueproperty ofPlis\n",
      "that,inaddition tonormal...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 3~J-c\n",
      "FIG.17.Mapofthebacteriophage P1genome, mo...\n",
      "> Adding chunk: 3~J-c\n",
      "FIG.17.Mapofthebacteriophage P1genome, mo...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1136 WALKER ANDWALKER\n",
      "fiber.Mutant72ismissinga1...\n",
      "> Adding chunk: 1136 WALKER ANDWALKER\n",
      "fiber.Mutant72ismissinga1...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Assembly ofP1particles. Byexamining the\n",
      "structu...\n",
      "> Adding chunk: Assembly ofP1particles. Byexamining the\n",
      "structu...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MORPHOLOGY OFP1MUTANTS 1137\n",
      "between thesegeneso...\n",
      "> Adding chunk: MORPHOLOGY OFP1MUTANTS 1137\n",
      "between thesegeneso...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Inaddition tolysisgene17,P1probably hasa\n",
      "second...\n",
      "> Adding chunk: Inaddition tolysisgene17,P1probably hasa\n",
      "second...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1138 WALKER ANDWALKER\n",
      "ACKNOWLEDGMENTS\n",
      "WethankJe...\n",
      "> Adding chunk: 1138 WALKER ANDWALKER\n",
      "ACKNOWLEDGMENTS\n",
      "WethankJe...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1963.Physiological\n",
      "studiesofconditional lethalm...\n",
      "> Adding chunk: 1963.Physiological\n",
      "studiesofconditional lethalm...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: J.Mol.\n",
      "Biol.99:695-716.\n",
      "26.King,J.1968.Assembly...\n",
      "> Adding chunk: J.Mol.\n",
      "Biol.99:695-716.\n",
      "26.King,J.1968.Assembly...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, ...\n",
      "> Adding chunk: MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Interactions between Viral Regulatory Proteins ...\n",
      "> Adding chunk: Interactions between Viral Regulatory Proteins ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: KEYWORDS bacteriophage P1, cell fate, gene regu...\n",
      "> Adding chunk: KEYWORDS bacteriophage P1, cell fate, gene regu...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Bacteriophages and phage-like plasmids play a p...\n",
      "> Adding chunk: Bacteriophages and phage-like plasmids play a p...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Regrettably,\n",
      "studies of P1 regulatory circuitry...\n",
      "> Adding chunk: Regrettably,\n",
      "studies of P1 regulatory circuitry...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: infecting a cell) and approaches 100% when the ...\n",
      "> Adding chunk: infecting a cell) and approaches 100% when the ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Moreover, during the time-lapse movies, the lyt...\n",
      "> Adding chunk: Moreover, during the time-lapse movies, the lyt...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: was followed through a reporter plasmid express...\n",
      "> Adding chunk: was followed through a reporter plasmid express...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: expression levels exhibit higher probabilities ...\n",
      "> Adding chunk: expression levels exhibit higher probabilities ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Lxc promotes C1\n",
      "binding af ﬁnity to its operato...\n",
      "> Adding chunk: Lxc promotes C1\n",
      "binding af ﬁnity to its operato...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: promoters that drive the transcription of gene ...\n",
      "> Adding chunk: promoters that drive the transcription of gene ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: To explore this, we examined P1 DNA interaction...\n",
      "> Adding chunk: To explore this, we examined P1 DNA interaction...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The nonindividual behavior could be due to phag...\n",
      "> Adding chunk: The nonindividual behavior could be due to phag...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: semble decision and tend toward MOI-independent...\n",
      "> Adding chunk: semble decision and tend toward MOI-independent...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: (B) The lysi s-\n",
      "lysogeny decision-making of P1 ...\n",
      "> Adding chunk: (B) The lysi s-\n",
      "lysogeny decision-making of P1 ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: cell outcome. Furthermore, we found that a larg...\n",
      "> Adding chunk: cell outcome. Furthermore, we found that a larg...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: These suggest a\n",
      "distinct evolutionary lineage o...\n",
      "> Adding chunk: These suggest a\n",
      "distinct evolutionary lineage o...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: K2HPO4(Teknova no. M2102), 1 /C2ACGU solution (...\n",
      "> Adding chunk: K2HPO4(Teknova no. M2102), 1 /C2ACGU solution (...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The genomic construct was then veri ﬁed by sequ...\n",
      "> Adding chunk: The genomic construct was then veri ﬁed by sequ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The phage band was extracted with 18-gauge need...\n",
      "> Adding chunk: The phage band was extracted with 18-gauge need...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: lkilwas induced with ʟ-arabinose when the OD600...\n",
      "> Adding chunk: lkilwas induced with ʟ-arabinose when the OD600...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: (iii) DNA injection on a TetR-mNeonGreen strain...\n",
      "> Adding chunk: (iii) DNA injection on a TetR-mNeonGreen strain...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The rest of the cell culture was mixed with the...\n",
      "> Adding chunk: The rest of the cell culture was mixed with the...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: plus 10 mgm l21DAPI (4 9,6-diamidino-2-phenylin...\n",
      "> Adding chunk: plus 10 mgm l21DAPI (4 9,6-diamidino-2-phenylin...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The lysozyme treatment time\n",
      "yielding ;90 to 95%...\n",
      "> Adding chunk: The lysozyme treatment time\n",
      "yielding ;90 to 95%...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: The rest of the cellculture was mixed with puri...\n",
      "> Adding chunk: The rest of the cellculture was mixed with puri...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: YFP (539 nm, 21; 556 nm; 576 nm, 31; Chroma no....\n",
      "> Adding chunk: YFP (539 nm, 21; 556 nm; 576 nm, 31; Chroma no....\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: (iv) Analysis of DNA number and RNA number. Mic...\n",
      "> Adding chunk: (iv) Analysis of DNA number and RNA number. Mic...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: to 5, respectively. The predictions of DNA numb...\n",
      "> Adding chunk: to 5, respectively. The predictions of DNA numb...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 2014. Characterization of a P1-like bacteriopha...\n",
      "> Adding chunk: 2014. Characterization of a P1-like bacteriopha...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 11. Hendrickx APA, Landman F, de Haan A, Borst ...\n",
      "> Adding chunk: 11. Hendrickx APA, Landman F, de Haan A, Borst ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Science 187:653 –654. https://doi.org/10.1126/\n",
      "...\n",
      "> Adding chunk: Science 187:653 –654. https://doi.org/10.1126/\n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 1990. The c1 repressor inactivator pro-\n",
      "tein co...\n",
      "> Adding chunk: 1990. The c1 repressor inactivator pro-\n",
      "tein co...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Lysogenization by bacteriophage lambda. I. Mult...\n",
      "> Adding chunk: Lysogenization by bacteriophage lambda. I. Mult...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 49. Iida S, Meyer J, Bachi B, Stalhammar-Carlem...\n",
      "> Adding chunk: 49. Iida S, Meyer J, Bachi B, Stalhammar-Carlem...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: complex. J Biol Chem 274:16813 –16818. https://...\n",
      "> Adding chunk: complex. J Biol Chem 274:16813 –16818. https://...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 65. Greer H. 1975. The kil gene of bacteriophag...\n",
      "> Adding chunk: 65. Greer H. 1975. The kil gene of bacteriophag...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Shao Q, Cortes MG, Trinh JT, Guan J, Balazsi G,...\n",
      "> Adding chunk: Shao Q, Cortes MG, Trinh JT, Guan J, Balazsi G,...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Intro |Search taxa  |Taxon tree  |Search litera...\n",
      "> Adding chunk: Intro |Search taxa  |Taxon tree  |Search litera...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: IRMNG taxon details\n",
      "Enterobacteria phage P1\n",
      "bas...\n",
      "> Adding chunk: IRMNG taxon details\n",
      "Enterobacteria phage P1\n",
      "bas...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Website and databases developed and hosted by V...\n",
      "> Adding chunk: Website and databases developed and hosted by V...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Vol.174,No.12\n",
      "DNAInversion Regions MinofPlasmid...\n",
      "> Adding chunk: Vol.174,No.12\n",
      "DNAInversion Regions MinofPlasmid...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thecomplex Minsystemiscloselyrelated tosimpleDN...\n",
      "> Adding chunk: Thecomplex Minsystemiscloselyrelated tosimpleDN...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: VOL.174,1992 ~~~~EVOLUTION OFpl5B3ANDPHAGE P1TA...\n",
      "> Adding chunk: VOL.174,1992 ~~~~EVOLUTION OFpl5B3ANDPHAGE P1TA...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Tailfibercomplementation assay.WA921 cellsharbo...\n",
      "> Adding chunk: Tailfibercomplementation assay.WA921 cellsharbo...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: RESULTS ANDDISCUSSION\n",
      "ScgenesofP1andpl5Bcontain...\n",
      "> Adding chunk: RESULTS ANDDISCUSSION\n",
      "ScgenesofP1andpl5Bcontain...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 3938 SANDMEIER ETAL.\n",
      "mix\n",
      "aOala2a3a4a5a6a7cmix'I...\n",
      "> Adding chunk: 3938 SANDMEIER ETAL.\n",
      "mix\n",
      "aOala2a3a4a5a6a7cmix'I...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Substitution oftheinvertible DNAandtherecombina...\n",
      "> Adding chunk: Substitution oftheinvertible DNAandtherecombina...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: EVOLUTION OFplSBANDPHAGE P1TAILFIBER GENES 3939...\n",
      "> Adding chunk: EVOLUTION OFplSBANDPHAGE P1TAILFIBER GENES 3939...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Sincethebreakpoints fromhomology tononho-\n",
      "molog...\n",
      "> Adding chunk: Sincethebreakpoints fromhomology tononho-\n",
      "molog...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: OnlytheSvMrandP(-)segments arehomolo-\n",
      "gousalsob...\n",
      "> Adding chunk: OnlytheSvMrandP(-)segments arehomolo-\n",
      "gousalsob...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 3940 SANDMEIER ETAL.\n",
      "consensus\n",
      "pl5BSvM1\n",
      "plSBSvM...\n",
      "> Adding chunk: 3940 SANDMEIER ETAL.\n",
      "consensus\n",
      "pl5BSvM1\n",
      "plSBSvM...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Forothersymbols, seethelegend toFig.3.\n",
      "segments...\n",
      "> Adding chunk: Forothersymbols, seethelegend toFig.3.\n",
      "segments...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Thetandemrepeated elements blandb2oftheP1genome...\n",
      "> Adding chunk: Thetandemrepeated elements blandb2oftheP1genome...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: EVOLUTION OFpl5BANDPHAGE P1TAILFIBER GENES 3941...\n",
      "> Adding chunk: EVOLUTION OFpl5BANDPHAGE P1TAILFIBER GENES 3941...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: bT4protein38andXTfahaveaslightlyhigherhomology ...\n",
      "> Adding chunk: bT4protein38andXTfahaveaslightlyhigherhomology ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: QATQNSVLD IDENAMLT LNKQLSTAS*\n",
      "KLQGRRL. QATRNAVL...\n",
      "> Adding chunk: QATQNSVLD IDENAMLT LNKQLSTAS*\n",
      "KLQGRRL. QATRNAVL...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 3942 SANDMEIER ETAL.\n",
      "A B CFIG.7.Phageparticles ...\n",
      "> Adding chunk: 3942 SANDMEIER ETAL.\n",
      "A B CFIG.7.Phageparticles ...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: SincetheMinregionappears toencodetailfibergenes...\n",
      "> Adding chunk: SincetheMinregionappears toencodetailfibergenes...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: EVOLUTION OFpl5BANDPHAGE P1TAILFIBERGENES 3943\n",
      "...\n",
      "> Adding chunk: EVOLUTION OFpl5BANDPHAGE P1TAILFIBERGENES 3943\n",
      "...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Theintermediate region,withitsmosaiclike struc-...\n",
      "> Adding chunk: Theintermediate region,withitsmosaiclike struc-...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 7.Glasgow, A.C.,K.T.Hughes, andM.I.Simon.1989.B...\n",
      "> Adding chunk: 7.Glasgow, A.C.,K.T.Hughes, andM.I.Simon.1989.B...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 3944 SANDMEIER ETAL.\n",
      "Biol.208:615-622.\n",
      "11.Hagga...\n",
      "> Adding chunk: 3944 SANDMEIER ETAL.\n",
      "Biol.208:615-622.\n",
      "11.Hagga...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 198...\n",
      "> Adding chunk: 22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 198...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 34.Sandmeier, H.,S.lida,J.Meyer, R.Hiestand-Nau...\n",
      "> Adding chunk: 34.Sandmeier, H.,S.lida,J.Meyer, R.Hiestand-Nau...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Proc.Natl.Acad.Sci.USAVol.87,pp.8070-8074, Octo...\n",
      "> Adding chunk: Proc.Natl.Acad.Sci.USAVol.87,pp.8070-8074, Octo...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: MATERIAL ANDMETHODS\n",
      "Bacterial andPhageStrains.E...\n",
      "> Adding chunk: MATERIAL ANDMETHODS\n",
      "Bacterial andPhageStrains.E...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: TheDNAsaredigested withBamHI,\n",
      "fractionated byag...\n",
      "> Adding chunk: TheDNAsaredigested withBamHI,\n",
      "fractionated byag...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Proc.Natl.Acad.Sci.USA87(1990) 8071\n",
      "5'CATTGATCA...\n",
      "> Adding chunk: Proc.Natl.Acad.Sci.USA87(1990) 8071\n",
      "5'CATTGATCA...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 2B),thedelayinpaccleavage seeninthedam-hostprob...\n",
      "> Adding chunk: 2B),thedelayinpaccleavage seeninthedam-hostprob...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Positions ofP1BamHIfragments 7-10\n",
      "areshownbyarr...\n",
      "> Adding chunk: Positions ofP1BamHIfragments 7-10\n",
      "areshownbyarr...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 8072Biochemistry: Sternberg andCoulby\n",
      "pacCleava...\n",
      "> Adding chunk: 8072Biochemistry: Sternberg andCoulby\n",
      "pacCleava...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Failuretocleavethe\n",
      "infecting, unmethylated phag...\n",
      "> Adding chunk: Failuretocleavethe\n",
      "infecting, unmethylated phag...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: (C)DNAfromdam'strain\n",
      "N99wasinfectedeitherwithP1...\n",
      "> Adding chunk: (C)DNAfromdam'strain\n",
      "N99wasinfectedeitherwithP1...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Proc.Natl.Acad.Sci.USA87(1990) 8073\n",
      "A - EcoRI-2...\n",
      "> Adding chunk: Proc.Natl.Acad.Sci.USA87(1990) 8073\n",
      "A - EcoRI-2...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 5legend). Incontrast,\n",
      "yieldofclass2cleavage pro...\n",
      "> Adding chunk: 5legend). Incontrast,\n",
      "yieldofclass2cleavage pro...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Lanes:10and11,similar\n",
      "tolanes8and9,exceptthatth...\n",
      "> Adding chunk: Lanes:10and11,similar\n",
      "tolanes8and9,exceptthatth...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: 8074 Biochemistry: Sternberg andCoulby\n",
      "isaninit...\n",
      "> Adding chunk: 8074 Biochemistry: Sternberg andCoulby\n",
      "isaninit...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Alternatively, themethylation function isenhanc...\n",
      "> Adding chunk: Alternatively, themethylation function isenhanc...\n",
      "DEBUG:llama_index.node_parser.node_utils:> Adding chunk: Mol.Biol.187,197-212.\n",
      "13.Shimada, K.,Weisberg, ...\n",
      "> Adding chunk: Mol.Biol.187,197-212.\n",
      "13.Shimada, K.,Weisberg, ...\n",
      "DEBUG:httpx:load_ssl_context verify=True cert=None trust_env=True http2=False\n",
      "load_ssl_context verify=True cert=None trust_env=True http2=False\n",
      "DEBUG:httpx:load_verify_locations cafile='/workspaces/ml-learning/.venv/lib/python3.11/site-packages/certifi/cacert.pem'\n",
      "load_verify_locations cafile='/workspaces/ml-learning/.venv/lib/python3.11/site-packages/certifi/cacert.pem'\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21 0021-9193/04/$08.00 /H110010 DOI: 10.1128/JB.186.21.7032–7068.2004 Copyright © 2004, American Society for Microbiology. All Rights Reserved. Genome of Bacteriophage P1† Małgorzata B. Łobocka,1* Debra J. Rose,2Guy Plunkett, III,2Marek Rusin,3Arkadiusz Samojedny,3 Hansjo ¨rg Lehnherr,4Michael B. Yarmolinsky,5and Frederick R. Blattner2 Department of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, Warsaw,1 and Department of Tumor Biology, Centre of Oncology, M. Skłodowska-Curie Memorial Institute, Gliwice,3Poland; Laboratory of Genetics, University of Wisconsin, Madison, Wisconsin2; Department of Genetics and Biochemistry, Institute of Microbiology, Ernst-Moritz-Arndt-University Greifswald, Greifswald, Germany4; and Laboratory of Biochemistry, National Cancer Institute, National Institutes of Health, Bethesda, Maryland5 Received 11 March 2004/Accepted 9 July 2004 P1 is a bacteriophage of Escherichia coli and other enteric bacteria. It lysogenizes its hosts as a circular, low-copy-number plasmid. We have determined the complete nucleotide sequences of two strains of a P1thermoinducible mutant, P1 c1-100. The P1 genome (93,601 bp) contains at least 117 genes, of which almost two-thirds had not been sequenced previously and 49 have no homologs in other organisms. Protein-codinggenes occupy 92% of the genome and are organized in 45 operons, of which four are decisive for the choicebetween lysis and lysogeny. Four others ensure plasmid maintenance. The majority of the remaining 37operons are involved in lytic development. Seventeen operons are transcribed from /H9268 70promoters directly controlled by the master phage repressor C1. Late operons are transcribed from promoters recognized by theE. coli RNA polymerase holoenzyme in the presence of the Lpa protein, the product of a C1-controlled P1 gene. Three species of P1-encoded tRNAs provide differential controls of translation, and a P1-encoded DNAmethyltransferase with putative bifunctionality inﬂuences transcription, replication, and DNA packaging. Thegenome is particularly rich in Chi recombinogenic sites. The base content and distribution in P1 DNA indicatethat replication of P1 from its plasmid origin had more impact on the base compositional asymmetries of theP1 genome than replication from the lytic origin of replication. Three bacteriophages of Escherichia coli with different life styles have been particularly instrumental in the developmentof the concepts and tools of molecular biology. They are thevirulent phage T4 and the two temperate phages, /H9261and P1. The two temperate phages were ﬁrst described in close suc-cession in the middle of the last century, namely in 1951 (phageP1 [31]) and 1953 ( /H9261[185] and informally even earlier, in the ﬁrst issue of Evelyn Witkin’s Microbial Genetics Bulletin , 1950 [184a]). Whereas /H9261prophage leads an essentially passive exis- tence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia colihave been reported. We present here the complete nucle- otide sequence of P1 and summarize its salient features. P1 infects and lysogenizes E. coli and several other enteric bacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Whereas /H9261prophage leads an essentially passive exis- tence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia colihave been reported. We present here the complete nucle- otide sequence of P1 and summarize its salient features. P1 infects and lysogenizes E. coli and several other enteric bacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath. A variable part of the tail ﬁbers determines the spec-iﬁcity of P1 adsorption on different hosts. The variable part isencoded by an invertible segment of P1 DNA, similar to that ofphage Mu (reviewed in reference 268). Infective particles of P1contain cyclically permuted, linear, double-stranded moleculeswith a terminal redundancy of 10 to15 kb of DNA (293). Priorto this work, about 70 genes had been identiﬁed in P1 by genetic and physiological studies (346). After injection into a host cell, viral DNA circularizes by recombination between redundant sequences to enter a lytic orlysogenic path. The choice between the two paths is dictated bythe interplay of a number of environmental factors with thecomplex immunity circuitry that controls synthesis and activityof a master repressor, C1 (reviewed in reference 126). As aprophage, P1 is a stable plasmid maintained at about onecopy per bacterial chromosome. Several P1 genes scatteredthroughout the genome are expressed in the lysogenic state.Those of primary importance are involved in plasmid mainte-nance functions and in inhibition of lytic development. P1prophage replicates as a circular plasmid from an origin ( oriR) resembling that of several other plasmids and, to a lesser ex-tent, oriC ofE. coli . P1 genes expressed in the lytic pathway are those involved in the timing of phage development, in replication (from a “lytic”origin different from oriR), in the formation of phage particles, including headful packaging, and ﬁnally in cell lysis to releasethe phage progeny. Genes of the lytic pathway have been di-vided into early and late. Transcription of the late genes re-quires, in addition to bacterial RNA polymerase, a P1-encodedactivator protein, Lpa (188), and an E. coli RNA polymerase- associated stringent starvation protein, SspA (111). P1, like /H9261, made its mark early in molecular biology. The signiﬁcant capacity of P1 for mediating generalized transduc-tion (196) led promptly to P1 becoming a workhorse of geneticexchange among strains of E. coli , a role it is still playing today. Moreover, because P1 can package slightly more than twice asmuch DNA as can /H9261, and packaging can be efﬁciently carried * Corresponding author. Mailing address: Department of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the PolishAcademy of Sciences, Ul. Pawinskiego 5A, 02-106 Warsaw, Poland.Phone: 011 4822 8237188. Fax: 011 4822 6584636. E-mail: lobocka@ibb.waw.pl. † Supplemental material for this article may be found at http://jb .asm.org/ 7032', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  out in vitro, P1-based vectors are now in common use for cloning and in vitro packaging of eukaryotic DNA (246, 293).P1 also made its mark early because of the restriction-modi ﬁ- cation genes that it carries, the analysis of which in the 1960sheralded the age of genetic engineering (16). The recognitionthat P1 is maintained as a plasmid prophage led to the iden-tiﬁcation of its plasmid maintenance functions. A site-speci ﬁc recombinase, Cre, which appears promptly after infection, isinvolved in phage DNA cyclization (137), assists plasmid main-tenance by resolving plasmid multimers (21), and perhaps, bya separate mechanism, may stabilize plasmid copy number(10). Ever since the introduction of the enzyme and its recom-bination site, lox, into Saccharomyces cerevisiae (271), the Cre- loxsystem has been a major tool of genetic engineering in eukaryotic cells. The active partitioning of P1 plasmids to daughter cells requires two P1-encoded proteins and a cis-acting site (the centromere) (20). A surprising homology between the struc-tural genes of plasmid partitioning and the sojand spoOJ sporulation genes of Bacillus subtilis provided the motivation for examining the effect of sojand spoOJ mutations on the production of anucleate progeny. Consequently, a connectionwas established between plasmid and nucleoid partitioning(157, 284). The ﬁnding that a gene of P1 encodes a toxin that is activated on P1 loss (192), among other ﬁndings of such “plasmid addiction ”genes, helped to establish the concept that programmed cell death has its place in the life of bacterialpopulations, as in metazoan development (348). Finally, theanalysis of P1 immunity maintenance and establishment (re-viewed in reference 126) enlarges our view of controls onphage development. The master repressor, C1, unlike that oflambda, is monomeric, and the operators to which it binds aremany and dispersed. It is modulated by at least two antirepres-sors (Ant1/2 and Coi) and a corepressor protein (Lxc), thelatter capable of inducing certain C1-bound operators to loop.Atrans -acting RNA (the c4 product) regulates Ant1/2 synthe- sis in a novel way, being excised from a transcript on which liesits site of action. Regrettably, the study of P1 immunity haltedprematurely with the death of its prime mover, Heinz Schuster. The most recent effort to compile a P1 map (made when no more than a total of 40% of the genome had been sequenced)was reported solely as a computer ﬁle (346). A recent short review of P1 biology has been prepared (186). Reviews of theolder P1 literature may be found in references 29, 301, 345, and347 and successive editions of Genetic Maps , one of which (344) was reprinted in reference 225. A thoughtful perspectiveon early studies of lysogeny by one of the pioneers in this ﬁeld and the discoverer of P1 has recently appeared (30). The overview of P1 genome organization and developmental regulation that we present here emerged from results of pre-vious studies combined with the analysis of the entire genomicsequences of two strains of P1 described below. MATERIALS AND METHODS Sources of DNA and sequencing strategy. DNA sequences described here were derived from two sources. One was DNA of P1 phage induced from N1706,anE. coli K-12 P1 c1-100 Tn 9lysogen preserved frozen since the early 1970s (kindly provided by J. L. Rosner). The other was DNA of P1 prophage carriedby an MG1655-derived strain that was used in the E. coli genome sequencing project (35). The prophage had been acquired in the course of a P1-mediatedtransduction with the aid of P1 c1-100 Tn 9r /H11002m/H11002dam (now dmt)/H9004MB rev-6, a high-frequency transducing derivative of the P1 c1-100 Tn 9lysogen (303).', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  MATERIALS AND METHODS Sources of DNA and sequencing strategy. DNA sequences described here were derived from two sources. One was DNA of P1 phage induced from N1706,anE. coli K-12 P1 c1-100 Tn 9lysogen preserved frozen since the early 1970s (kindly provided by J. L. Rosner). The other was DNA of P1 prophage carriedby an MG1655-derived strain that was used in the E. coli genome sequencing project (35). The prophage had been acquired in the course of a P1-mediatedtransduction with the aid of P1 c1-100 Tn 9r /H11002m/H11002dam (now dmt)/H9004MB rev-6, a high-frequency transducing derivative of the P1 c1-100 Tn 9lysogen (303). Prophage induction in N1706 and isolation of phage particles were performed as described previously (112). DNA was isolated from phage particles by usingthe QIAGEN Lambda Midi kit (QIAGEN, Inc.) according to the supplier ’s protocol. DNA of the prophage was isolated from a gel as described in reference35. The shotgun sequencing strategy was used for acquiring most of the sequenc- ing data. Fragments of P1 phage or prophage DNA obtained by sonication werecloned in the pUC18 vector (266) or in the M13 Janus vector (45) and served astemplates for sequencing. The ABI PRISM Dye Terminator Cycle SequencingReady Reaction kit with AmpliTaq DNA polymerase FS (Perkin-Elmer), wasused to perform sequencing reactions. Separation of products of sequencingreactions and reading of results was done with ABI 377 automated sequencers(Perkin-Elmer). After collection of initial sets of sequencing data at a four- toﬁvefold redundancy from readings in one direction, selected templates were resequenced in the opposite direction. The remaining gaps in the sequences wereﬁlled in by primer walking using oligonucleotides complementary to the ends of the sequenced regions and cloned P1 fragments or the entire DNA of therespective P1 strain as a template. Sequences were assembled using the programSeqman II (DNAStar Inc.). Sequence analysis and annotation. The majority of analyses were carried out using programs from either the GCG package version 9.0 (Genetics ComputerGroup, 1996) or the DNAStar package (DNAStar Inc.). Open reading frame (ORF) identi ﬁcation was performed using programs based on a Markov model: the Internet version of GeneMark (http://genemark.biology.gatech.edu/GeneMark) (36, 208) and TIGR Glimmer 2.1 (75, 264).GeneMark was trained on templates for genes of E. coli , and its temperate bacteriophages and transposons, and bacteriophage T4. The model for Glimmer2.1 was prepared with the use of Build-icm trained on P1 ORFs longer than 600nucleotides. Additionally, the entire sequence was divided into 400-bp overlap-ping fragments and the predicted products of their translation were searchedagainst the database for possible similarities with known protein sequences. Identi ﬁcation of a few genes, whose predicted codon usage pattern did not match that of any model organism, was based on homologies of their putative productsto known proteins and on the presence in their upstream regions of sequencesresembling promoters or likely to encode ribosome binding sites. The assignment of previously identi ﬁed genetic loci to newly sequenced ORFs was assisted by alignments to existing physical and genetic maps (230, 252, 280,281, 294, 323, 346). Predicted sizes of putative P1 structural proteins werecompared with sizes of head and tail components of P1 determined by polyacryl-amide gel electrophoresis (PAGE) (325) to verify the identity of certain struc-tural genes. Searching for similarities between P1 proteins and known proteins in data- bases was performed using the Internet versions of programs PSI-BLAST andPHI-BLAST (http://www.ncbi.nlm.nih.gov/BLAST/) (12, 354). Macaw (278)served to create multiple protein alignments.', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The assignment of previously identi ﬁed genetic loci to newly sequenced ORFs was assisted by alignments to existing physical and genetic maps (230, 252, 280,281, 294, 323, 346). Predicted sizes of putative P1 structural proteins werecompared with sizes of head and tail components of P1 determined by polyacryl-amide gel electrophoresis (PAGE) (325) to verify the identity of certain struc-tural genes. Searching for similarities between P1 proteins and known proteins in data- bases was performed using the Internet versions of programs PSI-BLAST andPHI-BLAST (http://www.ncbi.nlm.nih.gov/BLAST/) (12, 354). Macaw (278)served to create multiple protein alignments. The putative helix-turn-helix motifsin protein sequences were identi ﬁed by the HTH program at its website (http: //npsa-pbil.ibcp.fr/cgi-bin/npsa_automat.pl?page /H11005/NPSA/npsa_server.html) (78). Putative signal peptides in proteins were predicted by using the SignalP program(235) at its website (http://www.cbs.dtu.dk/services/SignalP/#submission). Puta-tive transmembrane helices in proteins were predicted by using the programTMHMM (175) at its website (http://www.cbs.dtu.dk/services/TMHMM). Identi ﬁcation of putative /H9268 70promoters was carried out using Targsearch (118). Putative Rho-independent transcriptional terminators were identi ﬁed with the GCG program Terminator (40) and TIGR program TransTerm (85). Toavoid false positives, only terminators that met the following additional criteria(69, 198) were taken into consideration: a 4- to 18-bp stem and a 3- to 10nucleotide (nt) loop of the terminator hairpin; a thymidine-rich region down-stream of the terminator hairpin and separated by less than 3 nt; more than threeGC/CG or GT/TG bp in the hairpin stem; at least three T residues, no more thanone G, and no 5 /H11032-TVVTT stretches (V is A, C, or G) in the 5-nt-long proximal part of the thymidine-rich region; absence of four purine or cytosine residues inthe 4-nt-long distal part of the thymidine-rich region and at least four T residuestogether in the proximal and distal parts of the thymidine-rich region. Putativeintegration host factor (IHF)-binding sites were identi ﬁed with MacTargsearch at SEQSCAN (http://www.bmb.psu.edu/seqscan/seqform1.htm). Oligonucleotide frequencies were determined with the program OCTAMER (L.Łobocka, unpublished). The DNA was scanned for putative tRNA genes by DNA homology searches and by using tRNAscan-SE (81, 204) at its website(http://www.genetics.wustl.edu/eddy/tRNAscan-SE/). Nucleotide sequence accession numbers and strain availability. The nucleo- tide sequences described here have been deposited in GenBank under accessionVOL. 186, 2004 BACTERIOPHAGE P1 7033', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  numbers AF234172 (phage P1 mod749 ::IS5c1-100) and AF234173 (prophage P1 mod1902 ::IS5c1-100 rev-6 dmt /H9004MB). A lysogen (N1706) from which the ﬁrst of these phages was induced has been deposited with the American Type CultureCollection (Manassas, Va.) as ATCC BAA-1001. RESULTS AND DISCUSSION Comparison of the sequenced P1 isolates. Current P1 phages in circulation are derived from P1 kc, a mutant selected in two steps by Lennox (196) as able to plate on E. coli K-12 with plaques of increased clarity. The induction of lysogens wasfacilitated by the introduction of mutations that render P1thermoinducible (279), the most widely used being the c1-100 mutation, originally designated clr100 (261). It is present in both of the P1 isolates that we used for sequencing. Forsimplicity in selecting P1 lysogens as drug-resistant bacteria,Rosner (261) made use of P1 CM (171), a strain that harborsthe unstable Tn 9transposon (encoding chloramphenicol acetyltransferase) acquired from an R factor. However, P1c1-100 Tn 9has been widely used and hence was chosen for sequencing. The phage induced from the supposed P1 c1-100 Tn9lysogen was found to have lost Tn 9and to have acquired IS5inserted in the mod gene, which renders it r /H11002m/H11002. The sequence of this strain, P1 c1-100 mod749 ::IS5(94,800 bp), was determined with an average 7.2-fold redundancy. The assemblywas veri ﬁed by comparison of the restriction map obtained by digestion with EcoRI, HindIII, HincII, BamHI, SacI, and PstIwith that deduced from the entire P1 sequence (without theIS5) (346). The other strain of P1 used here, P1 c1-100 mod1902 ::IS5 dmt/H9004MB rev-6 (94,481 bp), is a prophage carried by an MG1655-derived strain of E. coli K-12 that served in sequenc- ing the E. coli genome (35). Its nucleotide sequence was de- termined with an average 7.8-fold redundancy. The prophageappears to have been acquired in the course of a P1-mediatedtransduction. In addition to a 319-bp deletion in dmt that impairs the function of the viral DNA methyltransferase gene,it carries a mutation, rev-6 , that improves growth of the mutant in adam mutant host and augments transduction (303). In the course of its history, the phage lost Tn 9, acquired IS 5, and accumulated other mutations such that, relative to the ﬁrst strain, there are 13 additional mutations (four of them silentand one in a noncoding region). Five of those mutations arescattered in the variable part of the Sgene ( Sv), which is essential for P1 adsorption to E. coli . Which alleles are paren- tal remains uncertain. The rev-6 mutation has not been corre- lated with any of the sequence differences between the twostrains. In the two sequenced strains, the IS 5insertions occurred at different TTAG sequences, known to belong to the most com-monly used targets of IS 5(C/TTAA/G; reviewed in reference 212). In each case the IS 5insert, accompanied by a 4-bp du- plication of its target, appears responsible for the r /H11002m/H11002phe- notype since, in other respects, the res mod sequences are identical to those of r/H11001m/H11001P1 (145). Genomewide features.', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Five of those mutations arescattered in the variable part of the Sgene ( Sv), which is essential for P1 adsorption to E. coli . Which alleles are paren- tal remains uncertain. The rev-6 mutation has not been corre- lated with any of the sequence differences between the twostrains. In the two sequenced strains, the IS 5insertions occurred at different TTAG sequences, known to belong to the most com-monly used targets of IS 5(C/TTAA/G; reviewed in reference 212). In each case the IS 5insert, accompanied by a 4-bp du- plication of its target, appears responsible for the r /H11002m/H11002phe- notype since, in other respects, the res mod sequences are identical to those of r/H11001m/H11001P1 (145). Genomewide features. The genome of P1, without IS 5and its duplicated target, consists of 93,601 bp of double-strandedDNA, which in the prophage can be represented as a circle orline with the center of the site-speci ﬁc recombination site lox assigned as the zero point such that the ﬁrst nucleotide to itsright on the strand written 5 /H11032to 3/H11032in the direction of creis assigned position 1 and numbering proceeds rightwards (346)(Fig. 1). The genome contains one insertion sequence, IS 1,a s an integral part of P1 (151). In both sequenced strains, IS 1has a base substitution mutation (IS 1G757T) in its right inverted repeat (IRR). (i) Base content and distribution. The GC content of P1 DNA is 47.3%, slightly less than that of its E. coli host (50.8%) (35). The distribution of purines and pyrimidines between thetwo complementary strands is similar, with the upper strand(/H11001; coding strand of genes transcribed clockwise) containing 49.5% (26.0% adenosine and 23.5% guanosine), and the lowerstrand ( /H11002) containing 50.5% (26.7% adenosine and 23.8% guanosine) purine nucleotides. The location of extensive AT-rich regions along the genome map con ﬁrms the denaturation mapping evidence (222, 347) for the relatively recent acquisition by P1 of its restriction-modi ﬁcation genes ( res,mod) and suggests that genes of the simand rlfoperons, as well as hotand isaB, were also incor- porated in the genome of P1 late in evolution. Sharp borders oflong AT-rich regions within certain genes (e.g., the 3 /H11032-moiety ofparB ) may be indicative of mosaic structure. Unique restriction sites in P1 DNA are located preferentially in AT-rich regions or within the IS 1sequence (Fig. 2), suggesting their recent acquisition by P1 and selection against these sites ina P1 ancestor. A few 6-bp palindromes, the targets of knownrestriction enzymes, ApaI (GGGCCC), NarI (GGCGCC), NaeI(GCCGGC), SacI (GAGCTC), SalI (GTCGAC), and AvrII(CCTAGG), are absent from the P1 genome. Short DNA palin-dromes rare in or absent from P1 presumably identify sequencesthat could confer vulnerability in hosts that P1 otherwise ﬁnds particularly congenial. According to the pattern of C-G skew (the relative excess of cytosines versus guanosines in the top strand), the genome ofP1 can be divided into two polarized arms with the averageC-G skew below the mean predominating in one part andabove the mean in the other part (Fig. 2). The region where theaverage sign of C-G skew below the mean predominates coin-cides with the region where transcription in the clockwise di-rection predominates, and vice versa. The C-G skew switchespolarity in two regions: around kilobase 64 of the genome,close to the origin of plasmid replication ( oriR), and around kilobase 3 of the genome. This pattern is consistent with theskew observed for bacterial and double-stranded DNA viralgenomes whose replication proceeds bidirectionally from anorigin (reviewed in references 101 and 216).', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  According to the pattern of C-G skew (the relative excess of cytosines versus guanosines in the top strand), the genome ofP1 can be divided into two polarized arms with the averageC-G skew below the mean predominating in one part andabove the mean in the other part (Fig. 2). The region where theaverage sign of C-G skew below the mean predominates coin-cides with the region where transcription in the clockwise di-rection predominates, and vice versa. The C-G skew switchespolarity in two regions: around kilobase 64 of the genome,close to the origin of plasmid replication ( oriR), and around kilobase 3 of the genome. This pattern is consistent with theskew observed for bacterial and double-stranded DNA viralgenomes whose replication proceeds bidirectionally from anorigin (reviewed in references 101 and 216). Typically, theregions of polarity switch in the C-G skew are close to an originand to the terminus of replication. P1 possesses two separateorigins, oriR, and the origin of lytic replication, oriL, which are about 9 kb apart. Replication from both origins can be bidi-rectional (61, 241). As seen in Fig. 2, P1 prophage replicationappears to have had more impact on the base compositionalasymmetries and transcriptional organization of the genomethan the theta ( /H9258) or rolling-circle ( /H9268) replication that occurs during lytic development. Although oriL is further away than oriR from the region of the C-G skew polarity switch at 64 kb, it is located diametrically opposite to the region of the secondswitch in the C-G skew polarity, which may be indicative of theterminus of P1 replication. We did not ﬁnd in this region (or anywhere else in the P1 DNA) the TGTTGTAACTA se-7034 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  FIG. 1. Genetic and physical organization of the P1 genome. Boxes with internal triangles show positions and orientations of genes, color-coded by function: yellow, plasmid maintenance; red, repression of early functions; pink, immunity control, not c1 itself; magenta, source of tRNAs; brown, DNA methylation; deep blue, transcriptional activation of late genes; grey, defective IS 1; green, all other. Black boxes are intergenic regions of de ﬁned function: recombination sites, iterons to which RepA binds, plasmid centromere, and origin of DNA packaging, the direction of packaging being indicated by an arrowhead at the pacsite. Bidirectional replication determined at the phage (lytic) origin, oriL, and at the plasmid origin, oriR, are indicated by black arrowheads above the genome map. C1 operator sites are marked with red ﬂags pointing to the left or right (see also Table 6). Thin lines with terminal deep blue half arrowheads indicate the start sites and directions of the transcripts from particular latepromoters. GATC sequences that overlap transcriptional promoters and clustered 5 /H11032-GATC sequences (two or more sites with pairwise separation of not more than 50 bp), substrates for Dmt or Dam methylation, are marked above the gene map by brown lollipops that are ﬁlled in the case of sites shown to alter function upon methylation. Hooks indicate Rho-independent transcriptional terminators. They face the starts of transcript s that they terminate. The map refers to the genome of P1 c1-100 mod749 ::IS5without its nonintegral part, IS 5.V OL. 186, 2004 BACTERIOPHAGE P1 7035', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  quence that in E. coli ,Salmonella enterica serovar Typhi- murium, and several plasmids constitutes a conserved core ofTersites (136) at which replication terminates. However, this region is within 3 kb of the P1 site-speci ﬁc recombination site lox, which is involved in resolution of plasmid multimers into monomers, playing a role analogous to that of the difsite in the E. coli chromosome. The loxsite of P1 was shown to suppress the phenotype of an E. coli dif deletion strain in the presence of the lox-speci ﬁc recombinase, Cre, when loxwas inserted in the region of the E. coli terminus of replication (197). This suppression requires the DNA translocase activity of the es-sential host protein FtsK (47), as does dimer resolution at dif (292). The major family of strongly skewed sequences in E. coli with the motif 5 /H11032-RGNAGGGS (R /H11005purine, N /H11005any base, S/H11005G or C) (265) represents putative sites of binding to FtsK, a protein involved in positioning of bacterial DNA within thecell (64). The strongly biased distribution of these so-calledRAG sites in P1 DNA (Fig. 2) possibly argues for a compara-ble role for FtsK in the life of P1 prophage, although a con-nection between RAG sites and FtsK is still uncertain.The most abundant octamer in the genome of P1 corre- sponds to the recombinational hot spot Chi (5 /H11032-GCTGGTGG ; 31 and 19 sites in the upper and lower strands [31/19]). Threeother octamers in descending order of abundance (5 /H11032-TGCTG GTG , 18/14; 5 /H11032-CTGGTGG A, 14/8; and 5 /H11032-CTGGTGG C, 14/6) contain heptamers (boldface) identical to a Chi heptamer. Thecanonical Chi sites are two and one-half times more frequentin the P1 genome (one site per 1,872 bp) than in the genomeof its E. coli host (one site per 4,598 bp) (35, 291), where they are the third most abundant octamers and sevenfold moreabundant than a random octamer would be. P1 could bene ﬁti n several ways from the accumulation of Chi sites in its genome.P1 DNA encapsidated in a virion is a terminally redundantlinear molecule, which upon entering cells at infection, canserve as a substrate for loading of RecBCD nuclease at its endsand for RecBCD-mediated degradation. Each encounter ofRecBCD with Chi modi ﬁes the enzyme from its destructive 3/H11032-to-5/H11032nucleolytic form to its recombination-promoting form (reviewed in reference 173) and thus should both protect in-coming P1 DNA from further degradation and facilitate, byhomologous recombination, its cyclization. The involvement of FIG. 2. Compositional organization of the P1 genome. The G /H11001C content and (C /H11002G)/(C /H11001G) ratio (C-G skew) for one strand of P1 DNA are plotted as deviations from the mean. The plots were made with the GeneQuest program of DNAStar with a window size of 1,000 bp and a shiftincrement of 1 bp. For orientation, a simpli ﬁed genome map is shown centrally. Upper-row genes are transcribed clockwise (left to right), and lower-row genes are transcribed counterclockwise (right to left). Two vertical black lines through the plots show the locations of the lytic ( oriL) and plasmid ( oriR) origins of replication. Two vertical grey bars show regions of polarity switch in the plot of the C-G skew. Unique recognition sites for restriction enzymes are indicated directly above the plot of G /H11001C content; those below the line are for 8-bp cutters. See the text for absent 6-bp restriction sites. Chi sites and RAG sites (64) in the upper and lower strands are indicated by hash marks above and below the horizontal linesat the top of the ﬁgure.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21 0021-9193/04/$08.00 /H110010 DOI: 10.1128/JB.186.21.7032–7068.2004 Copyright © 2004, American Society for Microbiology. All Rights Reserved. Genome of Bacteriophage P1† Małgorzata B. Łobocka,1* Debra J. Rose,2Guy Plunkett, III,2Marek Rusin,3Arkadiusz Samojedny,3 Hansjo ¨rg Lehnherr,4Michael B. Yarmolinsky,5and Frederick R. Blattner2 Department of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, Warsaw,1 and Department of Tumor Biology, Centre of Oncology, M. Skłodowska-Curie Memorial Institute, Gliwice,3Poland; Laboratory of Genetics, University of Wisconsin, Madison, Wisconsin2; Department of Genetics and Biochemistry, Institute of Microbiology, Ernst-Moritz-Arndt-University Greifswald, Greifswald, Germany4; and Laboratory of Biochemistry, National Cancer Institute, National Institutes of Health, Bethesda, Maryland5 Received 11 March 2004/Accepted 9 July 2004 P1 is a bacteriophage of Escherichia coli and other enteric bacteria. It lysogenizes its hosts as a circular, low-copy-number plasmid. We have determined the complete nucleotide sequences of two strains of a P1thermoinducible mutant, P1 c1-100. The P1 genome (93,601 bp) contains at least 117 genes, of which almost two-thirds had not been sequenced previously and 49 have no homologs in other organisms. Protein-codinggenes occupy 92% of the genome and are organized in 45 operons, of which four are decisive for the choicebetween lysis and lysogeny. Four others ensure plasmid maintenance. The majority of the remaining 37operons are involved in lytic development. Seventeen operons are transcribed from /H9268 70promoters directly controlled by the master phage repressor C1. Late operons are transcribed from promoters recognized by theE. coli RNA polymerase holoenzyme in the presence of the Lpa protein, the product of a C1-controlled P1 gene. Three species of P1-encoded tRNAs provide differential controls of translation, and a P1-encoded DNAmethyltransferase with putative bifunctionality inﬂuences transcription, replication, and DNA packaging. Thegenome is particularly rich in Chi recombinogenic sites. The base content and distribution in P1 DNA indicatethat replication of P1 from its plasmid origin had more impact on the base compositional asymmetries of theP1 genome than replication from the lytic origin of replication. Three bacteriophages of Escherichia coli with different life styles have been particularly instrumental in the developmentof the concepts and tools of molecular biology. They are thevirulent phage T4 and the two temperate phages, /H9261and P1. The two temperate phages were ﬁrst described in close suc-cession in the middle of the last century, namely in 1951 (phageP1 [31]) and 1953 ( /H9261[185] and informally even earlier, in the ﬁrst issue of Evelyn Witkin’s Microbial Genetics Bulletin , 1950 [184a]). Whereas /H9261prophage leads an essentially passive exis- tence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia colihave been reported. We present here the complete nucle- otide sequence of P1 and summarize its salient features. P1 infects and lysogenizes E. coli and several other enteric bacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Whereas /H9261prophage leads an essentially passive exis- tence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia colihave been reported. We present here the complete nucle- otide sequence of P1 and summarize its salient features. P1 infects and lysogenizes E. coli and several other enteric bacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath. A variable part of the tail ﬁbers determines the spec-iﬁcity of P1 adsorption on different hosts. The variable part isencoded by an invertible segment of P1 DNA, similar to that ofphage Mu (reviewed in reference 268). Infective particles of P1contain cyclically permuted, linear, double-stranded moleculeswith a terminal redundancy of 10 to15 kb of DNA (293). Priorto this work, about 70 genes had been identiﬁed in P1 by genetic and physiological studies (346). After injection into a host cell, viral DNA circularizes by recombination between redundant sequences to enter a lytic orlysogenic path. The choice between the two paths is dictated bythe interplay of a number of environmental factors with thecomplex immunity circuitry that controls synthesis and activityof a master repressor, C1 (reviewed in reference 126). As aprophage, P1 is a stable plasmid maintained at about onecopy per bacterial chromosome. Several P1 genes scatteredthroughout the genome are expressed in the lysogenic state.Those of primary importance are involved in plasmid mainte-nance functions and in inhibition of lytic development. P1prophage replicates as a circular plasmid from an origin ( oriR) resembling that of several other plasmids and, to a lesser ex-tent, oriC ofE. coli . P1 genes expressed in the lytic pathway are those involved in the timing of phage development, in replication (from a “lytic”origin different from oriR), in the formation of phage particles, including headful packaging, and ﬁnally in cell lysis to releasethe phage progeny. Genes of the lytic pathway have been di-vided into early and late. Transcription of the late genes re-quires, in addition to bacterial RNA polymerase, a P1-encodedactivator protein, Lpa (188), and an E. coli RNA polymerase- associated stringent starvation protein, SspA (111). P1, like /H9261, made its mark early in molecular biology. The signiﬁcant capacity of P1 for mediating generalized transduc-tion (196) led promptly to P1 becoming a workhorse of geneticexchange among strains of E. coli , a role it is still playing today. Moreover, because P1 can package slightly more than twice asmuch DNA as can /H9261, and packaging can be efﬁciently carried * Corresponding author. Mailing address: Department of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the PolishAcademy of Sciences, Ul. Pawinskiego 5A, 02-106 Warsaw, Poland.Phone: 011 4822 8237188. Fax: 011 4822 6584636. E-mail: lobocka@ibb.waw.pl. † Supplemental material for this article may be found at http://jb .asm.org/ 7032', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  out in vitro, P1-based vectors are now in common use for cloning and in vitro packaging of eukaryotic DNA (246, 293).P1 also made its mark early because of the restriction-modi ﬁ- cation genes that it carries, the analysis of which in the 1960sheralded the age of genetic engineering (16). The recognitionthat P1 is maintained as a plasmid prophage led to the iden-tiﬁcation of its plasmid maintenance functions. A site-speci ﬁc recombinase, Cre, which appears promptly after infection, isinvolved in phage DNA cyclization (137), assists plasmid main-tenance by resolving plasmid multimers (21), and perhaps, bya separate mechanism, may stabilize plasmid copy number(10). Ever since the introduction of the enzyme and its recom-bination site, lox, into Saccharomyces cerevisiae (271), the Cre- loxsystem has been a major tool of genetic engineering in eukaryotic cells. The active partitioning of P1 plasmids to daughter cells requires two P1-encoded proteins and a cis-acting site (the centromere) (20). A surprising homology between the struc-tural genes of plasmid partitioning and the sojand spoOJ sporulation genes of Bacillus subtilis provided the motivation for examining the effect of sojand spoOJ mutations on the production of anucleate progeny. Consequently, a connectionwas established between plasmid and nucleoid partitioning(157, 284). The ﬁnding that a gene of P1 encodes a toxin that is activated on P1 loss (192), among other ﬁndings of such “plasmid addiction ”genes, helped to establish the concept that programmed cell death has its place in the life of bacterialpopulations, as in metazoan development (348). Finally, theanalysis of P1 immunity maintenance and establishment (re-viewed in reference 126) enlarges our view of controls onphage development. The master repressor, C1, unlike that oflambda, is monomeric, and the operators to which it binds aremany and dispersed. It is modulated by at least two antirepres-sors (Ant1/2 and Coi) and a corepressor protein (Lxc), thelatter capable of inducing certain C1-bound operators to loop.Atrans -acting RNA (the c4 product) regulates Ant1/2 synthe- sis in a novel way, being excised from a transcript on which liesits site of action. Regrettably, the study of P1 immunity haltedprematurely with the death of its prime mover, Heinz Schuster. The most recent effort to compile a P1 map (made when no more than a total of 40% of the genome had been sequenced)was reported solely as a computer ﬁle (346). A recent short review of P1 biology has been prepared (186). Reviews of theolder P1 literature may be found in references 29, 301, 345, and347 and successive editions of Genetic Maps , one of which (344) was reprinted in reference 225. A thoughtful perspectiveon early studies of lysogeny by one of the pioneers in this ﬁeld and the discoverer of P1 has recently appeared (30). The overview of P1 genome organization and developmental regulation that we present here emerged from results of pre-vious studies combined with the analysis of the entire genomicsequences of two strains of P1 described below. MATERIALS AND METHODS Sources of DNA and sequencing strategy. DNA sequences described here were derived from two sources. One was DNA of P1 phage induced from N1706,anE. coli K-12 P1 c1-100 Tn 9lysogen preserved frozen since the early 1970s (kindly provided by J. L. Rosner). The other was DNA of P1 prophage carriedby an MG1655-derived strain that was used in the E. coli genome sequencing project (35). The prophage had been acquired in the course of a P1-mediatedtransduction with the aid of P1 c1-100 Tn 9r /H11002m/H11002dam (now dmt)/H9004MB rev-6, a high-frequency transducing derivative of the P1 c1-100 Tn 9lysogen (303).', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  MATERIALS AND METHODS Sources of DNA and sequencing strategy. DNA sequences described here were derived from two sources. One was DNA of P1 phage induced from N1706,anE. coli K-12 P1 c1-100 Tn 9lysogen preserved frozen since the early 1970s (kindly provided by J. L. Rosner). The other was DNA of P1 prophage carriedby an MG1655-derived strain that was used in the E. coli genome sequencing project (35). The prophage had been acquired in the course of a P1-mediatedtransduction with the aid of P1 c1-100 Tn 9r /H11002m/H11002dam (now dmt)/H9004MB rev-6, a high-frequency transducing derivative of the P1 c1-100 Tn 9lysogen (303). Prophage induction in N1706 and isolation of phage particles were performed as described previously (112). DNA was isolated from phage particles by usingthe QIAGEN Lambda Midi kit (QIAGEN, Inc.) according to the supplier ’s protocol. DNA of the prophage was isolated from a gel as described in reference35. The shotgun sequencing strategy was used for acquiring most of the sequenc- ing data. Fragments of P1 phage or prophage DNA obtained by sonication werecloned in the pUC18 vector (266) or in the M13 Janus vector (45) and served astemplates for sequencing. The ABI PRISM Dye Terminator Cycle SequencingReady Reaction kit with AmpliTaq DNA polymerase FS (Perkin-Elmer), wasused to perform sequencing reactions. Separation of products of sequencingreactions and reading of results was done with ABI 377 automated sequencers(Perkin-Elmer). After collection of initial sets of sequencing data at a four- toﬁvefold redundancy from readings in one direction, selected templates were resequenced in the opposite direction. The remaining gaps in the sequences wereﬁlled in by primer walking using oligonucleotides complementary to the ends of the sequenced regions and cloned P1 fragments or the entire DNA of therespective P1 strain as a template. Sequences were assembled using the programSeqman II (DNAStar Inc.). Sequence analysis and annotation. The majority of analyses were carried out using programs from either the GCG package version 9.0 (Genetics ComputerGroup, 1996) or the DNAStar package (DNAStar Inc.). Open reading frame (ORF) identi ﬁcation was performed using programs based on a Markov model: the Internet version of GeneMark (http://genemark.biology.gatech.edu/GeneMark) (36, 208) and TIGR Glimmer 2.1 (75, 264).GeneMark was trained on templates for genes of E. coli , and its temperate bacteriophages and transposons, and bacteriophage T4. The model for Glimmer2.1 was prepared with the use of Build-icm trained on P1 ORFs longer than 600nucleotides. Additionally, the entire sequence was divided into 400-bp overlap-ping fragments and the predicted products of their translation were searchedagainst the database for possible similarities with known protein sequences. Identi ﬁcation of a few genes, whose predicted codon usage pattern did not match that of any model organism, was based on homologies of their putative productsto known proteins and on the presence in their upstream regions of sequencesresembling promoters or likely to encode ribosome binding sites. The assignment of previously identi ﬁed genetic loci to newly sequenced ORFs was assisted by alignments to existing physical and genetic maps (230, 252, 280,281, 294, 323, 346). Predicted sizes of putative P1 structural proteins werecompared with sizes of head and tail components of P1 determined by polyacryl-amide gel electrophoresis (PAGE) (325) to verify the identity of certain struc-tural genes. Searching for similarities between P1 proteins and known proteins in data- bases was performed using the Internet versions of programs PSI-BLAST andPHI-BLAST (http://www.ncbi.nlm.nih.gov/BLAST/) (12, 354). Macaw (278)served to create multiple protein alignments.', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The assignment of previously identi ﬁed genetic loci to newly sequenced ORFs was assisted by alignments to existing physical and genetic maps (230, 252, 280,281, 294, 323, 346). Predicted sizes of putative P1 structural proteins werecompared with sizes of head and tail components of P1 determined by polyacryl-amide gel electrophoresis (PAGE) (325) to verify the identity of certain struc-tural genes. Searching for similarities between P1 proteins and known proteins in data- bases was performed using the Internet versions of programs PSI-BLAST andPHI-BLAST (http://www.ncbi.nlm.nih.gov/BLAST/) (12, 354). Macaw (278)served to create multiple protein alignments. The putative helix-turn-helix motifsin protein sequences were identi ﬁed by the HTH program at its website (http: //npsa-pbil.ibcp.fr/cgi-bin/npsa_automat.pl?page /H11005/NPSA/npsa_server.html) (78). Putative signal peptides in proteins were predicted by using the SignalP program(235) at its website (http://www.cbs.dtu.dk/services/SignalP/#submission). Puta-tive transmembrane helices in proteins were predicted by using the programTMHMM (175) at its website (http://www.cbs.dtu.dk/services/TMHMM). Identi ﬁcation of putative /H9268 70promoters was carried out using Targsearch (118). Putative Rho-independent transcriptional terminators were identi ﬁed with the GCG program Terminator (40) and TIGR program TransTerm (85). Toavoid false positives, only terminators that met the following additional criteria(69, 198) were taken into consideration: a 4- to 18-bp stem and a 3- to 10nucleotide (nt) loop of the terminator hairpin; a thymidine-rich region down-stream of the terminator hairpin and separated by less than 3 nt; more than threeGC/CG or GT/TG bp in the hairpin stem; at least three T residues, no more thanone G, and no 5 /H11032-TVVTT stretches (V is A, C, or G) in the 5-nt-long proximal part of the thymidine-rich region; absence of four purine or cytosine residues inthe 4-nt-long distal part of the thymidine-rich region and at least four T residuestogether in the proximal and distal parts of the thymidine-rich region. Putativeintegration host factor (IHF)-binding sites were identi ﬁed with MacTargsearch at SEQSCAN (http://www.bmb.psu.edu/seqscan/seqform1.htm). Oligonucleotide frequencies were determined with the program OCTAMER (L.Łobocka, unpublished). The DNA was scanned for putative tRNA genes by DNA homology searches and by using tRNAscan-SE (81, 204) at its website(http://www.genetics.wustl.edu/eddy/tRNAscan-SE/). Nucleotide sequence accession numbers and strain availability. The nucleo- tide sequences described here have been deposited in GenBank under accessionVOL. 186, 2004 BACTERIOPHAGE P1 7033', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  numbers AF234172 (phage P1 mod749 ::IS5c1-100) and AF234173 (prophage P1 mod1902 ::IS5c1-100 rev-6 dmt /H9004MB). A lysogen (N1706) from which the ﬁrst of these phages was induced has been deposited with the American Type CultureCollection (Manassas, Va.) as ATCC BAA-1001. RESULTS AND DISCUSSION Comparison of the sequenced P1 isolates. Current P1 phages in circulation are derived from P1 kc, a mutant selected in two steps by Lennox (196) as able to plate on E. coli K-12 with plaques of increased clarity. The induction of lysogens wasfacilitated by the introduction of mutations that render P1thermoinducible (279), the most widely used being the c1-100 mutation, originally designated clr100 (261). It is present in both of the P1 isolates that we used for sequencing. Forsimplicity in selecting P1 lysogens as drug-resistant bacteria,Rosner (261) made use of P1 CM (171), a strain that harborsthe unstable Tn 9transposon (encoding chloramphenicol acetyltransferase) acquired from an R factor. However, P1c1-100 Tn 9has been widely used and hence was chosen for sequencing. The phage induced from the supposed P1 c1-100 Tn9lysogen was found to have lost Tn 9and to have acquired IS5inserted in the mod gene, which renders it r /H11002m/H11002. The sequence of this strain, P1 c1-100 mod749 ::IS5(94,800 bp), was determined with an average 7.2-fold redundancy. The assemblywas veri ﬁed by comparison of the restriction map obtained by digestion with EcoRI, HindIII, HincII, BamHI, SacI, and PstIwith that deduced from the entire P1 sequence (without theIS5) (346). The other strain of P1 used here, P1 c1-100 mod1902 ::IS5 dmt/H9004MB rev-6 (94,481 bp), is a prophage carried by an MG1655-derived strain of E. coli K-12 that served in sequenc- ing the E. coli genome (35). Its nucleotide sequence was de- termined with an average 7.8-fold redundancy. The prophageappears to have been acquired in the course of a P1-mediatedtransduction. In addition to a 319-bp deletion in dmt that impairs the function of the viral DNA methyltransferase gene,it carries a mutation, rev-6 , that improves growth of the mutant in adam mutant host and augments transduction (303). In the course of its history, the phage lost Tn 9, acquired IS 5, and accumulated other mutations such that, relative to the ﬁrst strain, there are 13 additional mutations (four of them silentand one in a noncoding region). Five of those mutations arescattered in the variable part of the Sgene ( Sv), which is essential for P1 adsorption to E. coli . Which alleles are paren- tal remains uncertain. The rev-6 mutation has not been corre- lated with any of the sequence differences between the twostrains. In the two sequenced strains, the IS 5insertions occurred at different TTAG sequences, known to belong to the most com-monly used targets of IS 5(C/TTAA/G; reviewed in reference 212). In each case the IS 5insert, accompanied by a 4-bp du- plication of its target, appears responsible for the r /H11002m/H11002phe- notype since, in other respects, the res mod sequences are identical to those of r/H11001m/H11001P1 (145). Genomewide features.', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Five of those mutations arescattered in the variable part of the Sgene ( Sv), which is essential for P1 adsorption to E. coli . Which alleles are paren- tal remains uncertain. The rev-6 mutation has not been corre- lated with any of the sequence differences between the twostrains. In the two sequenced strains, the IS 5insertions occurred at different TTAG sequences, known to belong to the most com-monly used targets of IS 5(C/TTAA/G; reviewed in reference 212). In each case the IS 5insert, accompanied by a 4-bp du- plication of its target, appears responsible for the r /H11002m/H11002phe- notype since, in other respects, the res mod sequences are identical to those of r/H11001m/H11001P1 (145). Genomewide features. The genome of P1, without IS 5and its duplicated target, consists of 93,601 bp of double-strandedDNA, which in the prophage can be represented as a circle orline with the center of the site-speci ﬁc recombination site lox assigned as the zero point such that the ﬁrst nucleotide to itsright on the strand written 5 /H11032to 3/H11032in the direction of creis assigned position 1 and numbering proceeds rightwards (346)(Fig. 1). The genome contains one insertion sequence, IS 1,a s an integral part of P1 (151). In both sequenced strains, IS 1has a base substitution mutation (IS 1G757T) in its right inverted repeat (IRR). (i) Base content and distribution. The GC content of P1 DNA is 47.3%, slightly less than that of its E. coli host (50.8%) (35). The distribution of purines and pyrimidines between thetwo complementary strands is similar, with the upper strand(/H11001; coding strand of genes transcribed clockwise) containing 49.5% (26.0% adenosine and 23.5% guanosine), and the lowerstrand ( /H11002) containing 50.5% (26.7% adenosine and 23.8% guanosine) purine nucleotides. The location of extensive AT-rich regions along the genome map con ﬁrms the denaturation mapping evidence (222, 347) for the relatively recent acquisition by P1 of its restriction-modi ﬁcation genes ( res,mod) and suggests that genes of the simand rlfoperons, as well as hotand isaB, were also incor- porated in the genome of P1 late in evolution. Sharp borders oflong AT-rich regions within certain genes (e.g., the 3 /H11032-moiety ofparB ) may be indicative of mosaic structure. Unique restriction sites in P1 DNA are located preferentially in AT-rich regions or within the IS 1sequence (Fig. 2), suggesting their recent acquisition by P1 and selection against these sites ina P1 ancestor. A few 6-bp palindromes, the targets of knownrestriction enzymes, ApaI (GGGCCC), NarI (GGCGCC), NaeI(GCCGGC), SacI (GAGCTC), SalI (GTCGAC), and AvrII(CCTAGG), are absent from the P1 genome. Short DNA palin-dromes rare in or absent from P1 presumably identify sequencesthat could confer vulnerability in hosts that P1 otherwise ﬁnds particularly congenial. According to the pattern of C-G skew (the relative excess of cytosines versus guanosines in the top strand), the genome ofP1 can be divided into two polarized arms with the averageC-G skew below the mean predominating in one part andabove the mean in the other part (Fig. 2). The region where theaverage sign of C-G skew below the mean predominates coin-cides with the region where transcription in the clockwise di-rection predominates, and vice versa. The C-G skew switchespolarity in two regions: around kilobase 64 of the genome,close to the origin of plasmid replication ( oriR), and around kilobase 3 of the genome. This pattern is consistent with theskew observed for bacterial and double-stranded DNA viralgenomes whose replication proceeds bidirectionally from anorigin (reviewed in references 101 and 216).', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  According to the pattern of C-G skew (the relative excess of cytosines versus guanosines in the top strand), the genome ofP1 can be divided into two polarized arms with the averageC-G skew below the mean predominating in one part andabove the mean in the other part (Fig. 2). The region where theaverage sign of C-G skew below the mean predominates coin-cides with the region where transcription in the clockwise di-rection predominates, and vice versa. The C-G skew switchespolarity in two regions: around kilobase 64 of the genome,close to the origin of plasmid replication ( oriR), and around kilobase 3 of the genome. This pattern is consistent with theskew observed for bacterial and double-stranded DNA viralgenomes whose replication proceeds bidirectionally from anorigin (reviewed in references 101 and 216). Typically, theregions of polarity switch in the C-G skew are close to an originand to the terminus of replication. P1 possesses two separateorigins, oriR, and the origin of lytic replication, oriL, which are about 9 kb apart. Replication from both origins can be bidi-rectional (61, 241). As seen in Fig. 2, P1 prophage replicationappears to have had more impact on the base compositionalasymmetries and transcriptional organization of the genomethan the theta ( /H9258) or rolling-circle ( /H9268) replication that occurs during lytic development. Although oriL is further away than oriR from the region of the C-G skew polarity switch at 64 kb, it is located diametrically opposite to the region of the secondswitch in the C-G skew polarity, which may be indicative of theterminus of P1 replication. We did not ﬁnd in this region (or anywhere else in the P1 DNA) the TGTTGTAACTA se-7034 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  FIG. 1. Genetic and physical organization of the P1 genome. Boxes with internal triangles show positions and orientations of genes, color-coded by function: yellow, plasmid maintenance; red, repression of early functions; pink, immunity control, not c1 itself; magenta, source of tRNAs; brown, DNA methylation; deep blue, transcriptional activation of late genes; grey, defective IS 1; green, all other. Black boxes are intergenic regions of de ﬁned function: recombination sites, iterons to which RepA binds, plasmid centromere, and origin of DNA packaging, the direction of packaging being indicated by an arrowhead at the pacsite. Bidirectional replication determined at the phage (lytic) origin, oriL, and at the plasmid origin, oriR, are indicated by black arrowheads above the genome map. C1 operator sites are marked with red ﬂags pointing to the left or right (see also Table 6). Thin lines with terminal deep blue half arrowheads indicate the start sites and directions of the transcripts from particular latepromoters. GATC sequences that overlap transcriptional promoters and clustered 5 /H11032-GATC sequences (two or more sites with pairwise separation of not more than 50 bp), substrates for Dmt or Dam methylation, are marked above the gene map by brown lollipops that are ﬁlled in the case of sites shown to alter function upon methylation. Hooks indicate Rho-independent transcriptional terminators. They face the starts of transcript s that they terminate. The map refers to the genome of P1 c1-100 mod749 ::IS5without its nonintegral part, IS 5.V OL. 186, 2004 BACTERIOPHAGE P1 7035', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  quence that in E. coli ,Salmonella enterica serovar Typhi- murium, and several plasmids constitutes a conserved core ofTersites (136) at which replication terminates. However, this region is within 3 kb of the P1 site-speci ﬁc recombination site lox, which is involved in resolution of plasmid multimers into monomers, playing a role analogous to that of the difsite in the E. coli chromosome. The loxsite of P1 was shown to suppress the phenotype of an E. coli dif deletion strain in the presence of the lox-speci ﬁc recombinase, Cre, when loxwas inserted in the region of the E. coli terminus of replication (197). This suppression requires the DNA translocase activity of the es-sential host protein FtsK (47), as does dimer resolution at dif (292). The major family of strongly skewed sequences in E. coli with the motif 5 /H11032-RGNAGGGS (R /H11005purine, N /H11005any base, S/H11005G or C) (265) represents putative sites of binding to FtsK, a protein involved in positioning of bacterial DNA within thecell (64). The strongly biased distribution of these so-calledRAG sites in P1 DNA (Fig. 2) possibly argues for a compara-ble role for FtsK in the life of P1 prophage, although a con-nection between RAG sites and FtsK is still uncertain.The most abundant octamer in the genome of P1 corre- sponds to the recombinational hot spot Chi (5 /H11032-GCTGGTGG ; 31 and 19 sites in the upper and lower strands [31/19]). Threeother octamers in descending order of abundance (5 /H11032-TGCTG GTG , 18/14; 5 /H11032-CTGGTGG A, 14/8; and 5 /H11032-CTGGTGG C, 14/6) contain heptamers (boldface) identical to a Chi heptamer. Thecanonical Chi sites are two and one-half times more frequentin the P1 genome (one site per 1,872 bp) than in the genomeof its E. coli host (one site per 4,598 bp) (35, 291), where they are the third most abundant octamers and sevenfold moreabundant than a random octamer would be. P1 could bene ﬁti n several ways from the accumulation of Chi sites in its genome.P1 DNA encapsidated in a virion is a terminally redundantlinear molecule, which upon entering cells at infection, canserve as a substrate for loading of RecBCD nuclease at its endsand for RecBCD-mediated degradation. Each encounter ofRecBCD with Chi modi ﬁes the enzyme from its destructive 3/H11032-to-5/H11032nucleolytic form to its recombination-promoting form (reviewed in reference 173) and thus should both protect in-coming P1 DNA from further degradation and facilitate, byhomologous recombination, its cyclization. The involvement of FIG. 2. Compositional organization of the P1 genome. The G /H11001C content and (C /H11002G)/(C /H11001G) ratio (C-G skew) for one strand of P1 DNA are plotted as deviations from the mean. The plots were made with the GeneQuest program of DNAStar with a window size of 1,000 bp and a shiftincrement of 1 bp. For orientation, a simpli ﬁed genome map is shown centrally. Upper-row genes are transcribed clockwise (left to right), and lower-row genes are transcribed counterclockwise (right to left). Two vertical black lines through the plots show the locations of the lytic ( oriL) and plasmid ( oriR) origins of replication. Two vertical grey bars show regions of polarity switch in the plot of the C-G skew. Unique recognition sites for restriction enzymes are indicated directly above the plot of G /H11001C content; those below the line are for 8-bp cutters. See the text for absent 6-bp restriction sites. Chi sites and RAG sites (64) in the upper and lower strands are indicated by hash marks above and below the horizontal linesat the top of the ﬁgure.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.connection:connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "DEBUG:httpcore.connection:connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde755f6950>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde755f6950>\n",
      "DEBUG:httpcore.connection:start_tls.started ssl_context=<ssl.SSLContext object at 0x7fdebfff7a40> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fdebfff7a40> server_hostname='api.openai.com' timeout=60.0\n",
      "DEBUG:httpcore.connection:start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde75907690>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde75907690>\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:35 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'96'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992107'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'473ms'), (b'x-request-id', b'5a15d1a85a5d6e6827592c086461f809'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Set-Cookie', b'__cf_bm=1O_1S6FkNQITbLb1fXnLmn8vRrmSiSW9d1UPdO7qdj8-1704526535-1-ARZt34VeQ7eWd05kagWs32Ln1SbKJcnOxKs3vDTS8GWWrrABeplhRtc3QC3LmT8Y3MxGKXrXwaFUnzBE13KiLeY=; path=/; expires=Sat, 06-Jan-24 08:05:35 GMT; domain=.api.openai.com; HttpOnly; Secure; SameSite=None'), (b'Set-Cookie', b'_cfuvid=q2kLgbA914Jk6lhG_B73gYTEj5w6WvieSQipkhQVreg-1704526535451-0-604800000; path=/; domain=.api.openai.com; HttpOnly; Secure; SameSite=None'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841248fbfc445d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:35 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'96'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992107'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'473ms'), (b'x-request-id', b'5a15d1a85a5d6e6827592c086461f809'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Set-Cookie', b'__cf_bm=1O_1S6FkNQITbLb1fXnLmn8vRrmSiSW9d1UPdO7qdj8-1704526535-1-ARZt34VeQ7eWd05kagWs32Ln1SbKJcnOxKs3vDTS8GWWrrABeplhRtc3QC3LmT8Y3MxGKXrXwaFUnzBE13KiLeY=; path=/; expires=Sat, 06-Jan-24 08:05:35 GMT; domain=.api.openai.com; HttpOnly; Secure; SameSite=None'), (b'Set-Cookie', b'_cfuvid=q2kLgbA914Jk6lhG_B73gYTEj5w6WvieSQipkhQVreg-1704526535451-0-604800000; path=/; domain=.api.openai.com; HttpOnly; Secure; SameSite=None'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841248fbfc445d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  For orientation, a simpli ﬁed genome map is shown centrally. Upper-row genes are transcribed clockwise (left to right), and lower-row genes are transcribed counterclockwise (right to left). Two vertical black lines through the plots show the locations of the lytic ( oriL) and plasmid ( oriR) origins of replication. Two vertical grey bars show regions of polarity switch in the plot of the C-G skew. Unique recognition sites for restriction enzymes are indicated directly above the plot of G /H11001C content; those below the line are for 8-bp cutters. See the text for absent 6-bp restriction sites. Chi sites and RAG sites (64) in the upper and lower strands are indicated by hash marks above and below the horizontal linesat the top of the ﬁgure. Where two hash marks are so close as to appear superimposed, a line at an angle to the vertical is added. The map refers to the genome of P1 c1-100 mod749 ::IS5without its nonintegral parts, IS 5, and the associated 4-bp duplication.7036 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  RecBCD in P1 cyclization could explain the 10-fold-reduced frequency of lysogenization by P1 of E. coli recB mutant cells observed by Rosner (261). Presumably, as in E. coli (181), Chi sites in P1, by their interaction with the RecBCD enzyme,could also facilitate the reassembly of collapsed replicationforks. The burst size of P1 is reduced 20-fold in a recBC mutant ofE. coli as compared to that in the recB /H11001C/H11001strain (352). The distribution of Chi sites in the P1 genome is unequal; the bias is away from the region that includes the invertible segment encoding tail ﬁber genes and at the right end favors the upper strand, where seven sites are within a 5-kb regionaround the pacsite (Fig. 2). Of these, ﬁve are on the lpaside ofpac. Their orientation is such that they offer protectionagainst the nucleolytic activity of RecBCD loading onto DNA cut at the pacsite and unprotected by packaging. The two octamers that are the most frequent in the genome ofE. coli (5/H11032-CGCTGGCG and 5 /H11032-GGCGCTGG) (35) are not among the most frequent octamers in P1. However, their fre-quencies in both genomes are similar (approximately 1 per6,000 nucleotides), suggesting that they have related functions.The majority of octamers that are frequent in P1 DNA, otherthan Chi sites and its variants, correspond to either sequences that are speci ﬁc for P1 and represent fragments of 17-mer C1 binding sites or 19-mer RepA-binding iterons (Table 1) orfragments of AT-rich regions containing A or T tracts. Themost infrequent tetramer in P1 DNA (5 /H11032-CTAG) is the sameTABLE 1.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The two octamers that are the most frequent in the genome ofE. coli (5/H11032-CGCTGGCG and 5 /H11032-GGCGCTGG) (35) are not among the most frequent octamers in P1. However, their fre-quencies in both genomes are similar (approximately 1 per6,000 nucleotides), suggesting that they have related functions.The majority of octamers that are frequent in P1 DNA, otherthan Chi sites and its variants, correspond to either sequences that are speci ﬁc for P1 and represent fragments of 17-mer C1 binding sites or 19-mer RepA-binding iterons (Table 1) orfragments of AT-rich regions containing A or T tracts. Themost infrequent tetramer in P1 DNA (5 /H11032-CTAG) is the sameTABLE 1. Binding sites in P1 DNA of P1 proteins other than C1 and Lpa NameaCoordinatesbStrand Sequencec Function (protein bound)Reference(s) lox 93587 –93601 and 1 –15 /H11001 AACTTCGTATAGCATACATTATACGAAGTT Site of Cre-mediated recombination (Cre) 138 cixPp 30379 –30419 /H11002 TTAACAAAAACCAGTATGTGTGGAAA cinautoregulation (Cin) 150 cixL 31016 –31035 /H11001 CAATCCTAAACCTTGGTTTAAGAGAA Site of Cin-mediated C-segment inversion 150 cixR 31016 –31035 /H11002 TTATCCAATACCTTGGTTTAAGAGAA Site of Cin-mediated C-segment inversion 150 parS Plasmid centromere 4, 92 boxA4 57713 –57719 /H11001 GTGAAAT (ParB) boxB2 57722 –57727 /H11002 TCGCCA (ParB) boxA3 57722 –57733 /H11002 GTGAAAT (ParB) boxA2 57735 –57741 /H11001 TTGAAAT (ParB) boxA1 57778 –57784 /H11001 GTGAAAT (ParB) boxB1 57790 –57795 /H11002 TCGCCA (ParB) parO 60027 –60068 /H11002 TTTATGCAGCATTTTTAATTAAATTCAAAA ATACAGCATAAAparoperon autoregulation (ParA) 73 incA Plasmid replication control Reviewed in reference 49 It1 60288 –60306 /H11001 GATGTGTGCTGGAGGGAGA (RepA) It2 60319 –60337 /H11001 GATGTGCGATAGAGGGAAG (RepA) It3 60349 –60367 /H11001 TATGTGCTGTGGAGG GATC (RepA) It4 60380 –60398 /H11001 TATGTGTGCTGGAGGGAAA (RepA) It5 60411 –60419 /H11001 CATGTGCACTGGAGGGAAA (RepA) It6 60442 –60460 /H11001 GATGTGTGCTGGAGGGAAA (RepA) It7 60482 –60500 /H11002 GATGTGCGCTGGAGGGGGA (RepA) It8 60523 –60541 /H11002 GATGTGTGCTGCCGGGAAG (RepA) It9 60554 –60572 /H11002 AGTGTGTGCTGGAGGGAAA (RepA) incC Part of plasmid replication initiation region Reviewed in reference 51 It10 61489 –61507 /H11002 GATGTGTGTTGACGGGAAA (RepA) It11 61511 –61529 /H11002 AATGTGTGCTGGCGGGATA (RepA) It12 61532 –61550 /H11002 AATGTGTGCTGACGGGTTG (RepA) It13 61553 –61571 /H11002 GATGTGTGCTGGAGGGATA (RepA) It14 61574 –61592 /H11002 GATGTGTGCTGGAGGGAAA (RepA) pacS Sites required to initiate packaging 288 89523 –89528 /H11001 TGATC A (PacA) 89530 –89535 /H11001 TGATC A (PacA) 89542 –89547 /H11001 TGATC A (PacA) 89557 –89562 /H11001 TGATC A (PacA) 89653 –89658 /H11001 TGATC G (PacA) 89665 –89570 /H11001 TGATC A (PacA) 89676 –89681 /H11001 TGATC A (PacA) 89704 –89709 /H11001 TGATC G (PacA)d aNames of functionally linked clusters of binding sites are followed by a list of particular sites within each cluster.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  It, iteron. bThe ﬁrst number is the coordinate of the ﬁrst nucleotide of upper-strand ( /H11001) binding sites or of the last nucleotide of the lower-strand ( /H11002) binding sites, based on arbitrary convention. Coordinates refer to positions of sites in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. cThe 5 /H11032-GATC sequences that are targets for Dam- or Dmt-mediated N-6 adenine methylation are in boldface. dDeletion of this site reduces the ef ﬁciency of cutting.VOL. 186, 2004 BACTERIOPHAGE P1 7037', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  as the most infrequent tetramer in E. coli (35). Seven of its 27 copies are in intergenic regions. Of the remaining 20 that arewithin 15 protein-coding genes, 9 are within ﬁve genes of especially low GC content (three in res, two in mod, one in isaB, two in rlfA, and one in pmgU ), which may represent the most recent acquisitions in the genome of P1. (ii)Coding sequences. Analysis of the complete sequence of P1 has revealed 112 protein-coding genes and ﬁve genes en- coding untranslated RNA (Tables 2 and 3 see also Table S1 inthe supplemental material). Of these, complete sequences of42 protein-coding genes and two RNA genes had been deter- mined previously (149, 187, 190, 346). The remaining 73 geneswere identi ﬁed as described in Materials and Methods. Of the previously published sequences of P1 genes, some differ fromthose presented here by single-nucleotide changes or theiroverall length, due either to mutations or to previous sequenc-ing errors. Protein-coding genes account for 92% of the P1 genome. Translation of most of the ORFs is initiated independentlyfrom their own ribosome binding and translation start sites (AUGTABLE 2. Alphabetical index of P1 genes GeneLocation and direction or commenta GeneLocation and direction or commentGeneLocation and direction or commentb 1 Seemat gta Seehdf pmgM 77.6378.2 2 SeelydB icd 47.2347.4 pmgN 78.2378.4 3 SeebplA iddB 25.3325.7 pmgO 78.3379.0 4 Seepro imcA 93.2493.3 pmgP 78.9379.5 5 72.5473.1 imcB 93.1493.2 pmgQ 80.2380.9 6 73.1474.1 insA 21.8422.1 pmgR 81.0381.2 7 74.9475.6 insB 21.4421.9 pmgS 81.2382.5 8 SeepmgR ,pmgS isaA 20.3321.3 pmgT 83.3384.2 9 SeepacA isaB 22.3322.9 pmgU 84.0384.3 10 Seelpa hdf 28.6429.2 pmgV 84.3384.7 11 SeeR? hot 85.3385.6 ppfA 62.3462.7 12 SeeR?', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  hot 85.3385.6 ppfA 62.3462.7 12 SeeR? hrdC 64.1365.2 pro 17.8318.8 13 Seeprt humD 86.3386.7 ppp 79.5380.2 14 Seeprt, pro hxr 23.0423.3 prt 16.1317.8 15 See22, 23 kilA 48.6349.4 R 36.6437.0 16 37.1438.0 lpa 88.7389.2 rebA Seeant1 17 Seelyz lxc 8.548.7 rebB Seeant2 18 Seemat lxr 85.6486.2 ref 2.032.6 19 SeeS lydA 29.6429.9 repA 60.0461.4 20 SeeU lydB 29.2429.6 repL 49.5350.3 21 53.3454.2 lydC 30.0430.3 res 3.546.3 22 54.2455.8 lydD 18.8419.1 rlfA 50.5351.2 23 55.8457.5 lydE 18.8419.0 rlfB 51.2352.4 24 74.1474.9 lyz 19.1419.6 sim SeesimC 25 75.6476.8 mat 2.733.4 simA 46.7446.8 26 76.8477.2 mlp 62.1462.8 simB 46.3446.6 ant1 47.4348.5 mod 6.448.3 simC 45.5446.3 ant2 47.7348.5 ORF0 See cre sit 39.7443.2 ban 70.0371.4 ORF1 See cra ssb 19.8320.3 bof Seelxc ORF2 See imcA S (Sc/H11001Sv) 33.6436.6 bplA 38.0439.4 ORF3 See imcB Sv /H11032 31.0332.5 bplB 52.2452.7 ORF4 See coi tciA 69.0369.4 c1 92.0492.8 ORF50 See simB tciB 69.4369.6 c2 See mod ORF93 See pdcB tciC 69.6369.7 c3 See mod ORFVIII See hxr trnA 68.1368.2 c4 47.1347.2 ORFX See icd trnI 69.7369.8 c5 See 6?,24? pacA 89.3390.5 trnT 68.2368.3 c6 See lxc pacB 90.5392.0 tsu SeedarA c7 See 22?,23?', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  pacA 89.3390.5 trnT 68.2368.3 c6 See lxc pacB 90.5392.0 tsu SeedarA c7 See 22?,23? pacC SeepacB tub 44.1444.8 c8 1.531.8 pap 82.5383.0 tsu SeedarA cin 30.4331.0 parA 58.8460.0 U 33.1433.6 cra 0.230.4 parB 57.8458.8 U/H11032 32.5333.1 cre 0.431.4 pdcA 87.4387.6 uhr 64.0364.3 coi 92.9493.2 pdcB 87.6388.7 ulx 8.749.2 dam Seedmt phd 86.8387.0 upfA 61.7462.1 darA 26.7428.6 plp 68.2468.6 upfB 62.7463.6 darB 9.2416.0 pmgA 39.4439.7 upfC 63.5463.8 dbn 71.4372.4 pmgB 43.9444.1 upfM 84.9385.0 ddrA 26.3426.7 pmgC 44.8445.3 upfN 85.0385.2 ddrB 23.3426.3 pmgF 51.8452.1 upfO 85.2385.3 dmt 65.2367.4 pmgG 52.7453.3 upl 68.5468.8 doc 87.0387.4 pmgL 77.4377.6 vad SeeddrA ,darA aReferences to previously mapped genes whose assignment to ORFs in the P1 genome is unclear are summarized in reference 346. bDirections of the tail ﬁber genes are for C( /H11001) phage.7038 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TABLE 3. Gene products (listed by gene position)a Nameb(former name[s])Gene coordinatescCoding strandMol mass (Da) (nucleotides in RNA)Basis for name; known or proposed functionReference(s)d Cra (gpORF0) 207 –428 /H11001 8,704 Putative cre-associated function 302S Cre/H11569 436–1467 /H11001 38,539 Cyclization recombinase acting on P1 DNA at redundant loxsites on infection; resolves dimers, etc.; possibly modulates copy no.7,/H1156910, 21, 106, 182, 302S C8 1518 –1829 /H11001 11,313 Establishment of lysogeny? 280, 281, twS Ref 2077 –2637 /H11001 21,328 Recombination enhancement function; aids microhomology recognition, maximal function requires recBCD activity184, 207, 337, 338S Mat (gp1; gp18?) 2716 –3468 /H11001 28,933 Particle maturation 191,S346 Res/H11569(EcoP1; R) 3544 –6456 /H11002 111,457 Restriction component (EcoP1) of type III restriction-modi ﬁcation (ResMod) system68, 107, /H11569145,S 152 Mod /H11569(M, C2, C3)6459 –8399 /H11002 73,485 Modi ﬁcation and site recognition component of type III ResMod system; modi ﬁes pairs of 5 /H11032-AGACC sequences41, 42, 107, /H11569 145,S152, 262 Lxc/H11569(Bof, C6) 8551 –8799 /H11002 9,645 Lowers expression of c1, enhances binding of C1 to its operators 164, 272, 311, 312, 318 /H11569S Ulx 8796 –9236 /H11002 17,079 Unknown; product of gene upstream of lxc twS DarB /H11569 9270 –16037 /H11002 251,542 Defense (in cis) against a subset of type I Res enzymes, e.g. EcoB, EcoK; putative DNA methyltransferase and DNA helicase; maymethylate DNA during injection155,/H11569tw S Prt (gp13, gp14?) 16113 –17822 /H11001 62,731 Portal protein (by similarity) 252, twS Pro (gp4, gp14?) 17815 –18834 /H11001 36,882 Putative head processing protease and ?', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  EcoB, EcoK; putative DNA methyltransferase and DNA helicase; maymethylate DNA during injection155,/H11569tw S Prt (gp13, gp14?) 16113 –17822 /H11001 62,731 Portal protein (by similarity) 252, twS Pro (gp4, gp14?) 17815 –18834 /H11001 36,882 Putative head processing protease and ? kinase; required for head morphogenesis and maturation of DarA precursor protein321, 324, twS LydE 18817 –19029 /H11002 7,914 Lysis determinant, putative antiholin twS LydD 18879 –19133 /H11002 9,936 Lysis determinant, putative holin twS Lyz (gp17) 19126 –19683 /H11002 20,255 Lysozyme 277S Ssb 19852 –20340 /H11001 18,023 Single-stranded DNA binding protein; can suppress E. coli ssb 187S IsaA 20370 –21333 /H11001 35,969 Unknown; product of IS1associated gene twS InsB 21442 –21945 /H11002 19,742 Transposition protein InsB of integral P1 IS 1 twS InsA 21864 –22139 /H11002 9,901 Transposition protein InsA of integral P1 IS 1 twS IsaB 22328 –22933 /H11001 23,633 Unknown; product of IS 1-associated gene 149S Hxr (ORFVIII) 23045 –23353 /H11002 11,033 Homolog of xre;possible repressor protein 149S DdrB 23343 –26330 /H11002 108,752 Unknown; product of the second gene downstream of darA; possible protease or response regulator149SD IddB 25378 –25776 /H11001 13,540 Unknown; product of gene internal to ddrB tw DdrA (Vad?) 26343 –26708 /H11002 13,013 Uncertain; product of gene downstream of darA; defect possibly alters P1B:P1S:P1M ratios; vad(viral architecture determinant)149S DarA /H11569(Vad?, Tsu?)26705 –28624 /H11002 69,479 Defends against restriction by type I Res endonuclease and enables DarB to function; internal head protein processed by Pro149,S155,/H11569 304/H11569 Hdf (Gta?)', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  26343 –26708 /H11002 13,013 Uncertain; product of gene downstream of darA; defect possibly alters P1B:P1S:P1M ratios; vad(viral architecture determinant)149S DarA /H11569(Vad?, Tsu?)26705 –28624 /H11002 69,479 Defends against restriction by type I Res endonuclease and enables DarB to function; internal head protein processed by Pro149,S155,/H11569 304/H11569 Hdf (Gta?) 28626 –29228 /H11002 22,150 Uncertain; homolog to DarA fragment; defect possibly causes gta (generalized transduction affected) phenotype149S LydB (gp2) 29215 –29658 /H11002 17,097 Lysis determinant; prevents premature lysis, LydA antagonist 149,S277S LydA 29655 –29984 /H11002 11,433 Lysis determinant; holin; promotes cell lysis 149,S277S LydC 30052 –30333 /H11002 10,449 Lysis determinant; putative holin (by homology) twS Cin/H11569 30461 –31021 /H11001 21,230 C-segment inversion; cixsite-speci ﬁc recombinase providing for alternate ﬁber gene expression and hence host range enlargement109, 135,S142, 150/H11569 gp Sv/H11032 31016 –32583 /H11001 55,254 Variable part of S /H11032tailﬁber; gene expressed in ( /H11002) orientation twS gpU/H11032 32586 –33119 /H11001 20,676 Tail ﬁber assembly chaperone (by similarity to Mu homolog); variable gene of tail ﬁber operon, expressed in ( /H11002) orientation146, twS gpU/H11569(gp20) 33148 –33675 /H11002 20,258 As gpU /H11032, but expressed in ( /H11001) orientation; both gpU /H11032and gpU may be virion proteins142,/H11569146, twS gpS:Sc /H11001Sv (gp19)33679 –36642 /H11002 104,837 Tail ﬁber with a constant (Sc) and variable (Sv) segment as in Mu; Sv is exchanged for shorter Sv /H11032upon C-segment inversion104, 146, twS gpR (gp11? gp12?)36654 –37088 /H11002 15,973 Tail ﬁber structure or assembly 104S gp16 37167 –38003 /H11002 31,357 Baseplate or tail tube 104,SP190,SP twS BplA (gp3?)'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  For orientation, a simpli ﬁed genome map is shown centrally. Upper-row genes are transcribed clockwise (left to right), and lower-row genes are transcribed counterclockwise (right to left). Two vertical black lines through the plots show the locations of the lytic ( oriL) and plasmid ( oriR) origins of replication. Two vertical grey bars show regions of polarity switch in the plot of the C-G skew. Unique recognition sites for restriction enzymes are indicated directly above the plot of G /H11001C content; those below the line are for 8-bp cutters. See the text for absent 6-bp restriction sites. Chi sites and RAG sites (64) in the upper and lower strands are indicated by hash marks above and below the horizontal linesat the top of the ﬁgure. Where two hash marks are so close as to appear superimposed, a line at an angle to the vertical is added. The map refers to the genome of P1 c1-100 mod749 ::IS5without its nonintegral parts, IS 5, and the associated 4-bp duplication.7036 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  RecBCD in P1 cyclization could explain the 10-fold-reduced frequency of lysogenization by P1 of E. coli recB mutant cells observed by Rosner (261). Presumably, as in E. coli (181), Chi sites in P1, by their interaction with the RecBCD enzyme,could also facilitate the reassembly of collapsed replicationforks. The burst size of P1 is reduced 20-fold in a recBC mutant ofE. coli as compared to that in the recB /H11001C/H11001strain (352). The distribution of Chi sites in the P1 genome is unequal; the bias is away from the region that includes the invertible segment encoding tail ﬁber genes and at the right end favors the upper strand, where seven sites are within a 5-kb regionaround the pacsite (Fig. 2). Of these, ﬁve are on the lpaside ofpac. Their orientation is such that they offer protectionagainst the nucleolytic activity of RecBCD loading onto DNA cut at the pacsite and unprotected by packaging. The two octamers that are the most frequent in the genome ofE. coli (5/H11032-CGCTGGCG and 5 /H11032-GGCGCTGG) (35) are not among the most frequent octamers in P1. However, their fre-quencies in both genomes are similar (approximately 1 per6,000 nucleotides), suggesting that they have related functions.The majority of octamers that are frequent in P1 DNA, otherthan Chi sites and its variants, correspond to either sequences that are speci ﬁc for P1 and represent fragments of 17-mer C1 binding sites or 19-mer RepA-binding iterons (Table 1) orfragments of AT-rich regions containing A or T tracts. Themost infrequent tetramer in P1 DNA (5 /H11032-CTAG) is the sameTABLE 1.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The two octamers that are the most frequent in the genome ofE. coli (5/H11032-CGCTGGCG and 5 /H11032-GGCGCTGG) (35) are not among the most frequent octamers in P1. However, their fre-quencies in both genomes are similar (approximately 1 per6,000 nucleotides), suggesting that they have related functions.The majority of octamers that are frequent in P1 DNA, otherthan Chi sites and its variants, correspond to either sequences that are speci ﬁc for P1 and represent fragments of 17-mer C1 binding sites or 19-mer RepA-binding iterons (Table 1) orfragments of AT-rich regions containing A or T tracts. Themost infrequent tetramer in P1 DNA (5 /H11032-CTAG) is the sameTABLE 1. Binding sites in P1 DNA of P1 proteins other than C1 and Lpa NameaCoordinatesbStrand Sequencec Function (protein bound)Reference(s) lox 93587 –93601 and 1 –15 /H11001 AACTTCGTATAGCATACATTATACGAAGTT Site of Cre-mediated recombination (Cre) 138 cixPp 30379 –30419 /H11002 TTAACAAAAACCAGTATGTGTGGAAA cinautoregulation (Cin) 150 cixL 31016 –31035 /H11001 CAATCCTAAACCTTGGTTTAAGAGAA Site of Cin-mediated C-segment inversion 150 cixR 31016 –31035 /H11002 TTATCCAATACCTTGGTTTAAGAGAA Site of Cin-mediated C-segment inversion 150 parS Plasmid centromere 4, 92 boxA4 57713 –57719 /H11001 GTGAAAT (ParB) boxB2 57722 –57727 /H11002 TCGCCA (ParB) boxA3 57722 –57733 /H11002 GTGAAAT (ParB) boxA2 57735 –57741 /H11001 TTGAAAT (ParB) boxA1 57778 –57784 /H11001 GTGAAAT (ParB) boxB1 57790 –57795 /H11002 TCGCCA (ParB) parO 60027 –60068 /H11002 TTTATGCAGCATTTTTAATTAAATTCAAAA ATACAGCATAAAparoperon autoregulation (ParA) 73 incA Plasmid replication control Reviewed in reference 49 It1 60288 –60306 /H11001 GATGTGTGCTGGAGGGAGA (RepA) It2 60319 –60337 /H11001 GATGTGCGATAGAGGGAAG (RepA) It3 60349 –60367 /H11001 TATGTGCTGTGGAGG GATC (RepA) It4 60380 –60398 /H11001 TATGTGTGCTGGAGGGAAA (RepA) It5 60411 –60419 /H11001 CATGTGCACTGGAGGGAAA (RepA) It6 60442 –60460 /H11001 GATGTGTGCTGGAGGGAAA (RepA) It7 60482 –60500 /H11002 GATGTGCGCTGGAGGGGGA (RepA) It8 60523 –60541 /H11002 GATGTGTGCTGCCGGGAAG (RepA) It9 60554 –60572 /H11002 AGTGTGTGCTGGAGGGAAA (RepA) incC Part of plasmid replication initiation region Reviewed in reference 51 It10 61489 –61507 /H11002 GATGTGTGTTGACGGGAAA (RepA) It11 61511 –61529 /H11002 AATGTGTGCTGGCGGGATA (RepA) It12 61532 –61550 /H11002 AATGTGTGCTGACGGGTTG (RepA) It13 61553 –61571 /H11002 GATGTGTGCTGGAGGGATA (RepA) It14 61574 –61592 /H11002 GATGTGTGCTGGAGGGAAA (RepA) pacS Sites required to initiate packaging 288 89523 –89528 /H11001 TGATC A (PacA) 89530 –89535 /H11001 TGATC A (PacA) 89542 –89547 /H11001 TGATC A (PacA) 89557 –89562 /H11001 TGATC A (PacA) 89653 –89658 /H11001 TGATC G (PacA) 89665 –89570 /H11001 TGATC A (PacA) 89676 –89681 /H11001 TGATC A (PacA) 89704 –89709 /H11001 TGATC G (PacA)d aNames of functionally linked clusters of binding sites are followed by a list of particular sites within each cluster.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  It, iteron. bThe ﬁrst number is the coordinate of the ﬁrst nucleotide of upper-strand ( /H11001) binding sites or of the last nucleotide of the lower-strand ( /H11002) binding sites, based on arbitrary convention. Coordinates refer to positions of sites in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. cThe 5 /H11032-GATC sequences that are targets for Dam- or Dmt-mediated N-6 adenine methylation are in boldface. dDeletion of this site reduces the ef ﬁciency of cutting.VOL. 186, 2004 BACTERIOPHAGE P1 7037', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  as the most infrequent tetramer in E. coli (35). Seven of its 27 copies are in intergenic regions. Of the remaining 20 that arewithin 15 protein-coding genes, 9 are within ﬁve genes of especially low GC content (three in res, two in mod, one in isaB, two in rlfA, and one in pmgU ), which may represent the most recent acquisitions in the genome of P1. (ii)Coding sequences. Analysis of the complete sequence of P1 has revealed 112 protein-coding genes and ﬁve genes en- coding untranslated RNA (Tables 2 and 3 see also Table S1 inthe supplemental material). Of these, complete sequences of42 protein-coding genes and two RNA genes had been deter- mined previously (149, 187, 190, 346). The remaining 73 geneswere identi ﬁed as described in Materials and Methods. Of the previously published sequences of P1 genes, some differ fromthose presented here by single-nucleotide changes or theiroverall length, due either to mutations or to previous sequenc-ing errors. Protein-coding genes account for 92% of the P1 genome. Translation of most of the ORFs is initiated independentlyfrom their own ribosome binding and translation start sites (AUGTABLE 2. Alphabetical index of P1 genes GeneLocation and direction or commenta GeneLocation and direction or commentGeneLocation and direction or commentb 1 Seemat gta Seehdf pmgM 77.6378.2 2 SeelydB icd 47.2347.4 pmgN 78.2378.4 3 SeebplA iddB 25.3325.7 pmgO 78.3379.0 4 Seepro imcA 93.2493.3 pmgP 78.9379.5 5 72.5473.1 imcB 93.1493.2 pmgQ 80.2380.9 6 73.1474.1 insA 21.8422.1 pmgR 81.0381.2 7 74.9475.6 insB 21.4421.9 pmgS 81.2382.5 8 SeepmgR ,pmgS isaA 20.3321.3 pmgT 83.3384.2 9 SeepacA isaB 22.3322.9 pmgU 84.0384.3 10 Seelpa hdf 28.6429.2 pmgV 84.3384.7 11 SeeR? hot 85.3385.6 ppfA 62.3462.7 12 SeeR?', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  hot 85.3385.6 ppfA 62.3462.7 12 SeeR? hrdC 64.1365.2 pro 17.8318.8 13 Seeprt humD 86.3386.7 ppp 79.5380.2 14 Seeprt, pro hxr 23.0423.3 prt 16.1317.8 15 See22, 23 kilA 48.6349.4 R 36.6437.0 16 37.1438.0 lpa 88.7389.2 rebA Seeant1 17 Seelyz lxc 8.548.7 rebB Seeant2 18 Seemat lxr 85.6486.2 ref 2.032.6 19 SeeS lydA 29.6429.9 repA 60.0461.4 20 SeeU lydB 29.2429.6 repL 49.5350.3 21 53.3454.2 lydC 30.0430.3 res 3.546.3 22 54.2455.8 lydD 18.8419.1 rlfA 50.5351.2 23 55.8457.5 lydE 18.8419.0 rlfB 51.2352.4 24 74.1474.9 lyz 19.1419.6 sim SeesimC 25 75.6476.8 mat 2.733.4 simA 46.7446.8 26 76.8477.2 mlp 62.1462.8 simB 46.3446.6 ant1 47.4348.5 mod 6.448.3 simC 45.5446.3 ant2 47.7348.5 ORF0 See cre sit 39.7443.2 ban 70.0371.4 ORF1 See cra ssb 19.8320.3 bof Seelxc ORF2 See imcA S (Sc/H11001Sv) 33.6436.6 bplA 38.0439.4 ORF3 See imcB Sv /H11032 31.0332.5 bplB 52.2452.7 ORF4 See coi tciA 69.0369.4 c1 92.0492.8 ORF50 See simB tciB 69.4369.6 c2 See mod ORF93 See pdcB tciC 69.6369.7 c3 See mod ORFVIII See hxr trnA 68.1368.2 c4 47.1347.2 ORFX See icd trnI 69.7369.8 c5 See 6?,24? pacA 89.3390.5 trnT 68.2368.3 c6 See lxc pacB 90.5392.0 tsu SeedarA c7 See 22?,23?', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  pacA 89.3390.5 trnT 68.2368.3 c6 See lxc pacB 90.5392.0 tsu SeedarA c7 See 22?,23? pacC SeepacB tub 44.1444.8 c8 1.531.8 pap 82.5383.0 tsu SeedarA cin 30.4331.0 parA 58.8460.0 U 33.1433.6 cra 0.230.4 parB 57.8458.8 U/H11032 32.5333.1 cre 0.431.4 pdcA 87.4387.6 uhr 64.0364.3 coi 92.9493.2 pdcB 87.6388.7 ulx 8.749.2 dam Seedmt phd 86.8387.0 upfA 61.7462.1 darA 26.7428.6 plp 68.2468.6 upfB 62.7463.6 darB 9.2416.0 pmgA 39.4439.7 upfC 63.5463.8 dbn 71.4372.4 pmgB 43.9444.1 upfM 84.9385.0 ddrA 26.3426.7 pmgC 44.8445.3 upfN 85.0385.2 ddrB 23.3426.3 pmgF 51.8452.1 upfO 85.2385.3 dmt 65.2367.4 pmgG 52.7453.3 upl 68.5468.8 doc 87.0387.4 pmgL 77.4377.6 vad SeeddrA ,darA aReferences to previously mapped genes whose assignment to ORFs in the P1 genome is unclear are summarized in reference 346. bDirections of the tail ﬁber genes are for C( /H11001) phage.7038 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TABLE 3. Gene products (listed by gene position)a Nameb(former name[s])Gene coordinatescCoding strandMol mass (Da) (nucleotides in RNA)Basis for name; known or proposed functionReference(s)d Cra (gpORF0) 207 –428 /H11001 8,704 Putative cre-associated function 302S Cre/H11569 436–1467 /H11001 38,539 Cyclization recombinase acting on P1 DNA at redundant loxsites on infection; resolves dimers, etc.; possibly modulates copy no.7,/H1156910, 21, 106, 182, 302S C8 1518 –1829 /H11001 11,313 Establishment of lysogeny? 280, 281, twS Ref 2077 –2637 /H11001 21,328 Recombination enhancement function; aids microhomology recognition, maximal function requires recBCD activity184, 207, 337, 338S Mat (gp1; gp18?) 2716 –3468 /H11001 28,933 Particle maturation 191,S346 Res/H11569(EcoP1; R) 3544 –6456 /H11002 111,457 Restriction component (EcoP1) of type III restriction-modi ﬁcation (ResMod) system68, 107, /H11569145,S 152 Mod /H11569(M, C2, C3)6459 –8399 /H11002 73,485 Modi ﬁcation and site recognition component of type III ResMod system; modi ﬁes pairs of 5 /H11032-AGACC sequences41, 42, 107, /H11569 145,S152, 262 Lxc/H11569(Bof, C6) 8551 –8799 /H11002 9,645 Lowers expression of c1, enhances binding of C1 to its operators 164, 272, 311, 312, 318 /H11569S Ulx 8796 –9236 /H11002 17,079 Unknown; product of gene upstream of lxc twS DarB /H11569 9270 –16037 /H11002 251,542 Defense (in cis) against a subset of type I Res enzymes, e.g. EcoB, EcoK; putative DNA methyltransferase and DNA helicase; maymethylate DNA during injection155,/H11569tw S Prt (gp13, gp14?) 16113 –17822 /H11001 62,731 Portal protein (by similarity) 252, twS Pro (gp4, gp14?) 17815 –18834 /H11001 36,882 Putative head processing protease and ?', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  EcoB, EcoK; putative DNA methyltransferase and DNA helicase; maymethylate DNA during injection155,/H11569tw S Prt (gp13, gp14?) 16113 –17822 /H11001 62,731 Portal protein (by similarity) 252, twS Pro (gp4, gp14?) 17815 –18834 /H11001 36,882 Putative head processing protease and ? kinase; required for head morphogenesis and maturation of DarA precursor protein321, 324, twS LydE 18817 –19029 /H11002 7,914 Lysis determinant, putative antiholin twS LydD 18879 –19133 /H11002 9,936 Lysis determinant, putative holin twS Lyz (gp17) 19126 –19683 /H11002 20,255 Lysozyme 277S Ssb 19852 –20340 /H11001 18,023 Single-stranded DNA binding protein; can suppress E. coli ssb 187S IsaA 20370 –21333 /H11001 35,969 Unknown; product of IS1associated gene twS InsB 21442 –21945 /H11002 19,742 Transposition protein InsB of integral P1 IS 1 twS InsA 21864 –22139 /H11002 9,901 Transposition protein InsA of integral P1 IS 1 twS IsaB 22328 –22933 /H11001 23,633 Unknown; product of IS 1-associated gene 149S Hxr (ORFVIII) 23045 –23353 /H11002 11,033 Homolog of xre;possible repressor protein 149S DdrB 23343 –26330 /H11002 108,752 Unknown; product of the second gene downstream of darA; possible protease or response regulator149SD IddB 25378 –25776 /H11001 13,540 Unknown; product of gene internal to ddrB tw DdrA (Vad?) 26343 –26708 /H11002 13,013 Uncertain; product of gene downstream of darA; defect possibly alters P1B:P1S:P1M ratios; vad(viral architecture determinant)149S DarA /H11569(Vad?, Tsu?)26705 –28624 /H11002 69,479 Defends against restriction by type I Res endonuclease and enables DarB to function; internal head protein processed by Pro149,S155,/H11569 304/H11569 Hdf (Gta?)', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  26343 –26708 /H11002 13,013 Uncertain; product of gene downstream of darA; defect possibly alters P1B:P1S:P1M ratios; vad(viral architecture determinant)149S DarA /H11569(Vad?, Tsu?)26705 –28624 /H11002 69,479 Defends against restriction by type I Res endonuclease and enables DarB to function; internal head protein processed by Pro149,S155,/H11569 304/H11569 Hdf (Gta?) 28626 –29228 /H11002 22,150 Uncertain; homolog to DarA fragment; defect possibly causes gta (generalized transduction affected) phenotype149S LydB (gp2) 29215 –29658 /H11002 17,097 Lysis determinant; prevents premature lysis, LydA antagonist 149,S277S LydA 29655 –29984 /H11002 11,433 Lysis determinant; holin; promotes cell lysis 149,S277S LydC 30052 –30333 /H11002 10,449 Lysis determinant; putative holin (by homology) twS Cin/H11569 30461 –31021 /H11001 21,230 C-segment inversion; cixsite-speci ﬁc recombinase providing for alternate ﬁber gene expression and hence host range enlargement109, 135,S142, 150/H11569 gp Sv/H11032 31016 –32583 /H11001 55,254 Variable part of S /H11032tailﬁber; gene expressed in ( /H11002) orientation twS gpU/H11032 32586 –33119 /H11001 20,676 Tail ﬁber assembly chaperone (by similarity to Mu homolog); variable gene of tail ﬁber operon, expressed in ( /H11002) orientation146, twS gpU/H11569(gp20) 33148 –33675 /H11002 20,258 As gpU /H11032, but expressed in ( /H11001) orientation; both gpU /H11032and gpU may be virion proteins142,/H11569146, twS gpS:Sc /H11001Sv (gp19)33679 –36642 /H11002 104,837 Tail ﬁber with a constant (Sc) and variable (Sv) segment as in Mu; Sv is exchanged for shorter Sv /H11032upon C-segment inversion104, 146, twS gpR (gp11? gp12?)36654 –37088 /H11002 15,973 Tail ﬁber structure or assembly 104S gp16 37167 –38003 /H11002 31,357 Baseplate or tail tube 104,SP190,SP twS BplA (gp3?)'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:36 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'58'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'995978'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'241ms'), (b'x-request-id', b'b3c2d00f30c94e7740887cf39d762c10'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124900f8f25d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:36 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'58'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'995978'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'241ms'), (b'x-request-id', b'b3c2d00f30c94e7740887cf39d762c10'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124900f8f25d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5da0>, 'json_data': {'input': ['page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  gp12?)36654 –37088 /H11002 15,973 Tail ﬁber structure or assembly 104S gp16 37167 –38003 /H11002 31,357 Baseplate or tail tube 104,SP190,SP twS BplA (gp3?) 38003 –39436 /H11002 53,577 Putative baseplate structural protein (by homology), may mediate contact between hub and wedges, as does its homolog in T4twS PmgA 39433 –39789 /H11002 13,239 Putative morphogenetic function twS Sit 39789 –43211 /H11002 120,710 Structural injection transglycosylase; putative tail tube “ruler ” 190S PmgB 43293 –44174 /H11002 34,138 Putative morphogenetic function twS Tub 44189 –44800 /H11002 22,316 Major tail tube protein twS PmgC 44811 –45377 /H11002 21,017 Putative morphogenetic function twS SimC /H11569(Sim) 45587 –46399 /H11002 30,800 Confers superimmunity when in high copy number by blocking P1 at entry; requires removal of leader sequence77, 170, 208 /H11569S SimB (gpORF50) 46374 –46691 /H11002 12,033 Unknown; superimmunity-linked function 213,SDtwS SimA 46713 –46844 /H11002 4,842 Unknown; superimmunity-linked function twS C4/H11569 47144 –47220 /H11001 (77) Antisense RNA that inhibits Icd and Ant1/Ant2 synthesis by acting on target antRNA; processed to active form by RNase P26,S33,S57,/H11569 58, 112,S 115, 130S Icd/H11569(gpORFX) 47271 –47492 /H11001 8,397 Reversible inhibition of cell division, apparently required for ant expression123, 131, 257S/H11569 Continued on following pageVOL. 186, 2004 BACTERIOPHAGE P1 7039', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TABLE 3 —Continued Nameb(former name[s])Gene coordinatescCoding strandMol mass (Da) (nucleotides in RNA)Basis for name; known or proposed functionReference(s)d Ant1 /H11569(RebA) 47489 –48532 /H11001 38,667 Antagonism of C1 repression; forms complex with Ant2 33,S112,SD 122, 131,S 256/H11569 Ant2 /H11569(RebB) 47738 –48532 /H11001 29,039 Antagonism of C1 repression, forms complex with Ant1 33,S112,SD 131,S256/H11569 KilA 48697 –49497 /H11001 29,575 Unknown, expression can kill host 112,S295S RepL 49527 –50372 /H11001 30,857 Lytic replication, initiates replication at oriL (within repL ) 112,S295S RlfA 50541 –51245 /H11001 27,716 Unknown; possibly replication-linked function twS RlfB 51245 –52444 /H11001 18,632 Unknown; possibly replication-linked function twS PmgF 51875 –52171 /H11002 11,423 Putative morphogenetic function twS BplB 52255 –52764 /H11002 18,804 Baseplate twS PmgG 52776 –53357 /H11002 20,696 Putative morphogenetic function twS gp21 53393 –54208 /H11002 29,329 Baseplate or tail tube 323, twS gp22 (gp15?, C7?)54218 –55807 /H11002 56,935 Sheath protein 323, 325, /H11569twS gp23 (gp15?, C7?)55868 –57574 /H11002 62,248 Major head protein, present in P1 heads in full-length (62 kDa) and truncated (44 kDa) forms323, 325, /H11569twS ParB /H11569 57800 –58801 /H11002 37,440 Plasmid partitioning; binds to parS , enhances ParA-mediated repression of paroperon and ATPase of ParA; can spread over DNA ﬂanking parS , silencing gene expression; pairs parS loci4,SD70, 73, 83, 90,/H11569114, 203,SP258, 343 ParA /H11569 58818 –60014 /H11002 44,269 Plasmid partitioning; weak ATPase, binds to parO repressing transcription, binds to ParB- parS partition complex4,S37, 70, 73, /H11569 343 RepA /H11569 60572 –61432 /H11002 32,219 Plasmid replication initiated by chaperone-activated monomers at oriR; represses own synthesis; binds at iterated sites ( incC and incA ) and can handcuff them, controling plasmid copy number2,/H115693, 6,S49, 52 UpfA 61751 –62143 /H11002 14,747 Unknown protein function twS Mlp 62155 –62487 /H11002 12,597 Membrane lipoprotein precursor (by homology); ? lysis control twS PpfA 62321 –62770 /H11002 15,744 Possible periplasmic function twS UpfB 62783 –63628 /H11002 29,310 Unknown protein function twS UpfC 63580 –63864 /H11002 9,478 Unknown protein function twS Uhr 64034 –64318 /H11001 10,781 Unknown; product of gene upstream of hrdC twS HrdC 64311 –65216 /H11001 33,638 Homolog of RdgC ofE.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  lysis control twS PpfA 62321 –62770 /H11002 15,744 Possible periplasmic function twS UpfB 62783 –63628 /H11002 29,310 Unknown protein function twS UpfC 63580 –63864 /H11002 9,478 Unknown protein function twS Uhr 64034 –64318 /H11001 10,781 Unknown; product of gene upstream of hrdC twS HrdC 64311 –65216 /H11001 33,638 Homolog of RdgC ofE. coli; possibly involved in replication, recombination, and/or Dmt functiontwS Dmt (Dam) 65216 –67477 /H11001 83,595 Predicted bifunctional DNA methyltransferase; methylates A ’si n GATC sequences, and (by homology) probably C ’s at unknown sites; affects replication control, late gene expression, packaging59, 66, twS tRNA1 68149 –68224 /H11001 (76) Proposed tRNA-Asn speci ﬁc for the AAC codon twS tRNA2 68227 –68302 /H11001 (76) Proposed tRNA-Thr speci ﬁc for the ACA codon twS Plp 68273 –68650 /H11002 13,800 Unknown; putative lipoprotein twS Upl 68651 –68842 /H11002 7,092 Unknown; product of gene upstream of plp twS TciA 69034 –69459 /H11001 15,187 Tellurite or colicin resistance or inhibition of cell division (by homology)twS TciB 69459 –69623 /H11001 5,752 Tci accessory protein (by homology) twS TciC 69623 –69730 /H11001 3,719 Tci accessory protein twS tRNA3 69734 –69809 /H11001 (76) Proposed tRNA-Ile speci ﬁc for ATA codon twS Ban/H11569 70096 –71460 /H11001 50,479 DnaB (of E. coli ) analog and homolog; replicative DNA helicase 67, 195 /H11569S237, 254/H11569 Dbn 71460 –72458 /H11001 37,379 Unknown; product of gene downstream of ban twS gp5 72505 –73137 /H11002 21,669 Baseplate 323, twS gp6 (C5?) 73130 –74146 /H11002 37,222 Tail length determination 323, twS gp24 (C5?) 74148 –74933 /H11002 28,880 Baseplate or tail stability 323, twS gp7 74920 –75648 /H11002 27,145 Tail stability 323, twS gp25 75652 –76869 /H11002 45,834 Tail stability 323, twS gp26 76879 –77256 /H11002 14,456 Baseplate 323, twS PmgL 77436 –77648 /H11001 9,258 Putative morphogenetic function twS PmgM 77651 –78229 /H11001 21,888 Putative morphogenetic function twS PmgN 78296 –78451 /H11001 5,593 Putative morphogenetic function twS PmgO 78393 –79055 /H11001 24,217 Putative morphogenetic function twS PmgP 78953 –79579 /H11001 23,198 Putative morphogenetic function twS Ppp 79579 –80253 /H11001 25,568 P1 protein phosphatase, possible role in head morphogenesis twS PmgQ 80250 –80951 /H11001 26,912 Putative morphogenetic function twS PmgR (gp8?) 81033 –81251 /H11001 8,318 Putative morphogenetic function twS PmgS (gp8?) 81253 –82515 /H11001 48,345 Putative morphogenetic function; ? prohead scaffolding protein twS Pap 82588 –83094 /H11001 19,200 P1 acid phosphatase (by homology); ?', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  81033 –81251 /H11001 8,318 Putative morphogenetic function twS PmgS (gp8?) 81253 –82515 /H11001 48,345 Putative morphogenetic function; ? prohead scaffolding protein twS Pap 82588 –83094 /H11001 19,200 P1 acid phosphatase (by homology); ? head morphogenesis twS PmgT 83289 –84218 /H11001 35,155 Putative morphogenetic function twS PmgU 84023 –84358 /H11001 12,163 Putative morphogenetic function twS PmgV 84355 –84717 /H11001 13,975 Putative morphogenetic function twS Continued on following page7040 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  [86]/H11022/H11022GUG[16] /H11022UUG[10]). Translation stops predominantly at UAA and less frequently at UGA and UAG. Utilization of UUG as the translation initiation codon ap- pears to be more frequent in P1 than in other coliphages (224)and in E. coli itself (35). It was con ﬁrmed experimentally that translation of the P1 lxcgene, which encodes a modulator of C1-mediated repression, initiates at UUG (272). Whereas the translation initiation codons of P1 are generally 6 to 10 nt downstream of typical E. coli ribosome binding sites (RBS), genes plp,pmgO , and pdcA lack nearby RBSs. They are probably translationally coupled to genes immediately up-stream ( plp) or overlapping ( pmgO and pdcA ). The gene ref (recombination enhancement function) in a monocistronic op-eron is more unusual. The sequence that most closely resem-bles a canonical RBS is separated by 16 bp, including a run ofeight T ’s, from a presumptive refinitiation codon (207, 338). What purpose the extra RNA might serve or how it is accom-modated remains to be determined. Promoter-proximal regions of four genes, pro,hrdC ,pmfA , and pmgL , contain, in addition to ATG codons preceded by sequences with a good match to RBSs in mRNA, TTG codonssimilarly preceded by canonical RBS sequences and furtherupstream. Possibly, each of these genes encodes two proteinsthat differ in the length of their N-terminal domains. Alterna-tive translational starts in mRNA at UUG or AUG couldcontrol expression of these genes by modulating the intracel-lular concentration or activity of their protein products. Although genes of P1 are tightly packed within the genome, overlaps are rare and occur mostly at coding termini. Spacesbetween genes of adjacent operons are in most cases limited toshort regions containing promoter and regulatory sequences.Of those that exceed 250 bp, one contains oriR and another incA , both involved in P1 prophage replication (Table 1). A 272-bp region between the ﬁrst gene of the simoperon, simA , andc4, was shown to encode the 5 /H11032region of the c4 transcript, which is cleaved to yield C4 RNA (115). Two regions withintheban operon (531 bp preceding the trnA gene and 359 bp between the trnIgene and ban, described in detail later in this paper) may also encode species of RNA that are processed. Of previously identi ﬁed or predicted products of P1 genes, 65 exhibit signi ﬁcant homology to known proteins of other organisms (Table 3; see also Table S1 in the supplementalmaterial). Only 29 are homologous to proteins encoded exclu-sively by other phages. Although they include 11 homologs ofproteins of T4 or T4-like phages, there is no clear preferencein these homologies for proteins of a particular phage, suggest-ing that P1 has had a long, separate history. Homologs ofknown genes of different provenance are scattered throughoutthe P1 DNA. Whether their distribution results from the hor-izontal transfer of modules during the evolution of the P1genome or from convergent evolution remains to be deter-mined. (iii)Organization of P1 genes and transcription units. P1 genes can be grouped into 45 operons, of which 15 appear tobe monocistronic. Their organization resembles that of T4 (224) in that only some genes of related function are in clusters. Regulatory regions of 38 operons contain one or more se- quences that resemble strong /H9268 70promoters (Table 4). Of 26 /H926870promoters identi ﬁed previously by transcriptional fusion or primer extension experiments, only 9 had a match to the con- sensus sequence below 50% and required relaxing the strin-gency of the prediction program for their detection.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Whether their distribution results from the hor-izontal transfer of modules during the evolution of the P1genome or from convergent evolution remains to be deter-mined. (iii)Organization of P1 genes and transcription units. P1 genes can be grouped into 45 operons, of which 15 appear tobe monocistronic. Their organization resembles that of T4 (224) in that only some genes of related function are in clusters. Regulatory regions of 38 operons contain one or more se- quences that resemble strong /H9268 70promoters (Table 4). Of 26 /H926870promoters identi ﬁed previously by transcriptional fusion or primer extension experiments, only 9 had a match to the con- sensus sequence below 50% and required relaxing the strin-gency of the prediction program for their detection. All nineTABLE 3 —Continued Nameb(former name[s])Gene coordinatescCoding strandMol mass (Da) (nucleotides in RNA)Basis for name; known or proposed functionReference(s)d UpfM 84900 –85013 /H11001 4,343 Unknown protein function twS UpfN 85029 –85226 /H11001 7,153 Unknown protein function twS UpfO 85204 –85344 /H11001 5,238 Unknown protein function twS Hot 85368 –85631 /H11001 9,694 Homolog of theta subunit of DNA polymerase III; ? replication twS Lxr 85638 –86210 /H11002 22,042 LexA-regulated function twS HumD /H11569 86385 –86774 /H11001 14,313 Homolog of UmuD /H11032subunit of E. coli repair protein DNA polymerase V (UmuD /H11032-UmuC); complements mutant UmuD /H11032200, 217, /H11569276S Phd/H11569 86847 –87068 /H11001 8,132 Prevention of host death by Doc toxicity; confers, with Doc, addiction to P1; represses transcription of addiction operon95,/H11569192,S194, 210 Doc/H11569 87068 –87448 /H11001 13,587 Death on curing; toxin of P1 addiction system; reversible inhibitor of protein synthesis; corepressor of addiction operon95, 121, 192,S 210, 211 /H11569 PdcA 87432 –87632 /H11001 7,630 Unknown; post-doc 192,Stw PdcB (gpORF93) 87660 –88703 /H11001 39,877 Unknown; post-doc 192,SDtw Lpa/H11569(gp10) 88792 –89244 /H11001 18,117 Late promoter activator 111, /H11569188S PacA /H11569(gp9) 89330 –90523 /H11001 45,246 DNA packaging; cuts at pactogether with PacB 286,S288/H11569 PacB /H11569 90523 –92007 /H11001 55,604 DNA packaging; cuts at pactogether with PacA 286,S287/H11569 C1/H11569 92032 –92883 /H11002 32,544 Controls plaque clarity; primary repressor of lytic functions 79, /H1156984, 124, 238,S316, 319 Coi/H11569(gpORF4) 92994 –93203 /H11002 7,741 C one (reversible) inactivator; forms 1:1 complex with C1 25,S27, 84,S 127, 130 /H11569 ImcB (gpORF3) 93169 –93264 /H11002 3,659 Immunity C function; may regulate c1 27, 84S ImcA (gpORF2) 93283 –93396 /H11002 4,340 Immunity C function; may regulate c1 27, 84S aA more complete table, which includes pI, number of amino acid residues (or nucleotides in RNA), and features of the amino acid sequence (closest homol ogs and amino acid sequence motifs) is to be found as Table S1 in the supplemental material.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  bGene products marked by an asterisk were either puri ﬁed or identi ﬁed in PAGE gels. cCoordinates of genes refer to their positions in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. Coordinates of protein-coding genes are from the initial codon through the ﬁrst stop codon. dReferences marked by an asterisk concern identi ﬁcation of a protein or RNA product of a given gene; those marked by a superscript S concern previously published sequences. Superscript SD, published sequence differs internally from sequences presented here; superscript SP, published sequence is partial; t w, this work.VOL. 186, 2004 BACTERIOPHAGE P1 7041', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TABLE 4. Known and putative /H926870promoters of P1 genes Name (former name)PositionaStrandSequenceb Basis for identi ﬁcationc Reference(s) /H1100235 /H1100210 tsp pp cp P2coi 41 /H11002TGCTAAGGTA TTGAAC TGTATGGATTTACAGG TAAATTGATCATATTC 56.8 P3cre 104 /H11001ACTTGATCAT TTGATCAAGGTTGCGCTACG TAAAAT CTGTGAAAAA /H12011/H12011 42.6 302 P2cre 283 /H11001GCAAAAGGGT TTGATCGTGATAGTTGCCAACTGT CATTAT CGCGCGGTGA /H12011/H12011 45.6 302 P1cre 400 /H11001GAGCCGATCC TGTACA CTTTACTTAAAAC CATTAT CTGAGTGTTA /H12011/H12011 /H1102140.0 302 Pc8 1394 /H11001AATGTAAATA TTGTCA TGAACTATATCCGTAACCTG GATAGT GAAACAGGGG 54.4 P2ref (Pref-2)1957 /H11001AACAAAATAT CTATCA TTGCTCTAATTGATTGC TATAAT TGAGCCGCAG 59.8 338 P1ref 2017 /H11001TACGAAGACG TTGCCA TTACTTCACTCCTTGA CATCAT TGGCGGCCAT 52.7 Pmat 2666 /H11001TGATAAATGA TTGACA ACTGACAAGTGACTTCAGT CAGAAT CATCACACGC 56.8 P1res 6558 /H11002GATGAAGATT TTGCCC CAAACAAAGTCGTGTTT TATGGT TGTAACTTTG 60.4 P2res (PR1)6891 /H11002CAGGAGGTAT TTAAAA AAAGCACAACCACAAGT GATAAT GCGGCTAAAA S1/H12011 52.1 285 P3res 6926 /H11002AGATCTATTT TTGAAA TAACAAAAAAAAGGA TACAGG AGGTATTTAA S1/H12011 47.3 285 P1mod (PM1)8450 /H11002CATATAGATA GTGATA GTGCCACAACTTCTGGC TCTAAC GGGCTGGGGA /H12011 45.6 285 P2mod (PM2)8548 /H11002ATCATAGATA ATGCAT TGCTTCTCAATGCGGAT TAAACT TCCCCAAATT /H12011/H12011 50.9 285 P1lxc 8844 /H11002CTGCCAGTCC CTGAAG AAGTTGACGACCAGGAC TATGAG TTTGAGTCTT S1 46.7 285, 318 P2lxc 9022 /H11002AATTAAAAAA TTATTA CTGGAGCGCGGTTATGG TAATTT CGGTCGTCTC S1 51.5 285 P3lxc 9167 /H11002AAAACATATC TTGAGG CGGTACTGCGTACGG CATTGT TAGGAAAGAC S1 51.5 285 PdarB 16093 /H11002CGTGATTTTA TTGTCT GCCAAGATTGCACCTTAAT TAGAAT AATTCACATC 56.8 P2ssb (Pr21)19787 /H11001TGCAAATCAC TTGTTA GCTACGTTTCAAAGATA TACATT ATTGCTCTAA /H12011 52.7 59 P1ssb 19821 /H11001ACATTATTGC TCTAAT TAATTTATTTTATTAGG TAAGAT AAGTGGCACA 55.6 Plyz 19845 /H11002GATGACTTTG TTTACA CCGCGTTGTGCCACT TATCTT ACCTAATAAA 53.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  5 285 PdarB 16093 /H11002CGTGATTTTA TTGTCT GCCAAGATTGCACCTTAAT TAGAAT AATTCACATC 56.8 P2ssb (Pr21)19787 /H11001TGCAAATCAC TTGTTA GCTACGTTTCAAAGATA TACATT ATTGCTCTAA /H12011 52.7 59 P1ssb 19821 /H11001ACATTATTGC TCTAAT TAATTTATTTTATTAGG TAAGAT AAGTGGCACA 55.6 Plyz 19845 /H11002GATGACTTTG TTTACA CCGCGTTGTGCCACT TATCTT ACCTAATAAA 53.3 PisaB 22220 /H11001CAATGATGGT TAGACA AGCAAGATAGGGCGAAT TAAAAT TAATCAAGAG 61.5 PinsA 22229 /H11002AAAGTGACTC TTGATT AATTTTAATTCGCCC TATCTT GCTTGTCTAA 57.4 P2cin 30339 /H11001CACATAACTA TTTCCT TAATAGTGAATTAAA TATCAT TGGGAAACGG 55.6 P1cin 30390 /H11001TACTTTGTGA TTTCCA CACATACTGGTTTTTGTTAAT TAAAAT CCGCAGCTTG 53.3 PlydC 30461 /H11002TGTTCATTTG TTGATA CGCGTACATAGCCTAT TAGCAT ATTTTCTGCT 59.2 P1pmgC 45555 /H11002AAAATAGTGT TTGAAA GGATAGTCAATTTAAGT TTTAAC ACAACGCCCA 59.8 P2pmgC 45636 /H11002CAACAATTCG TTGATG TAATGGAAGGATTTGA TAGCAT CATGTATAAA 59.8 P3pmgC 45771 /H11002CAAATATATA TTCACA AATCAA GATCCTGTTCA GATGAT GAATTTTATA 58.0 PsimB 46843 /H11002TGTACCCTAA ATGAGT TATAAGGCAGGTGAGGT TATAAT GAGAAAACTA 56.8 PsimA 46909 /H11002GCATTATTTG TTTAAT AAATACACAGTTG GATCTAAT AACCTCTTTT 51.5 P2c4 (P51a)46956 /H11001TCTAATAAAT TTGTAT TTTTAAGTCGCGAATGC TATCTT TTCGCATCAT /H12011/H11569 56.8 131 P1c4 (P51b)46996 /H11001TCGCATCATA TTGACC TTTTAATCGTTCAGGCT TATAGT TCCACCGTCG /H12011/H11569 60.4 131 PkilA (P53)48593 /H11001CAACCCGATCTGGATCGGGTCAGAAAAATTTGC TCTAAT AAATTTCGTT /H12011/H12011 50.3 295 PaskilA (P53as)48724 /H11002GGTTGAAGGA TCGACA TTTTGATGAAGGTTTGA TATATT CATATCCGCA /H12011/H12011 62.7 125 PrepL 49267 /H11001AAACTATCTC TTCACC TGGTTGCGTGATAACGG AATTCTGATCGCAACC 52.1 PrlfA 50486 /H11001CGCCCCATTA TTGCAA TTAATAAACAACTAACG GACAAT TCTACCTAAC 52.7 P1rlfB 50663 /H11001AAAAAATTGC TAGAGA CATCAAGGAATTAAAAT TAAAAC ACAATCTAAA 55.0 P2rlfB 50704 /H11001AATCTAAAAC CTGATT TCGAAATCAAGTGGAC TAAAGT ATCTGCGTCC 56.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  7 125 PrepL 49267 /H11001AAACTATCTC TTCACC TGGTTGCGTGATAACGG AATTCTGATCGCAACC 52.1 PrlfA 50486 /H11001CGCCCCATTA TTGCAA TTAATAAACAACTAACG GACAAT TCTACCTAAC 52.7 P1rlfB 50663 /H11001AAAAAATTGC TAGAGA CATCAAGGAATTAAAAT TAAAAC ACAATCTAAA 55.0 P2rlfB 50704 /H11001AATCTAAAAC CTGATT TCGAAATCAAGTGGAC TAAAGT ATCTGCGTCC 56.2 P3rlfB 50746 /H11001GCGTCCAAAG TTGAAT TTTACTTAGATGTCG TAGATT ACTTCTTCAG 55.0 P4rlfB 50971 /H11001ATTGAAAAAA TTGAGA AATTAAGAGGGGTTCTTCA CAATGATCGCTATGAT 57.4 Pasrlf 51784 /H11002TAGCTAGCTA TTGAAA AGCTAAAGTA GATCGGT TAGATT TGCATTACTC 69.2 P1pmgF 52236 /H11002TACAACTGGA TTGAGT GGGATTAAGAGTCATCCCTTG TATTTT AAAGCTCCTT 53.3 P2pmgF 52342 /H11002GCTACAACAA TTGAAA TGTTGGACTGCAAAATC TACAGT GATGCAATCG 60.9 P123 57631 /H11002AAAAAATTGC TTGTCA CGAGAAAGTCAACAAGTGACT TTCAAT AAAATCTCTT 57.4 P223 57740 /H11002TATTAACTGA CTGTTT TTAAAGTAAATTACTC TAAAAT TTCAAGGTGA 56.2 PparB 59274 /H11002ATTGCCGCGG AAGATT TTGCTAAAGCAGTTTTT GACCGT ATTGAATTTA /H12011/H12011 /H1102140.0 A. Dobruk, unpublished Ppar 60069 /H11002GAGAATGCTA TGTACA AGCATCTACGCATACAT TATTAT TTTATGCAGC /H12011/H12011 54.4 73, 120 PrepA 61505 /H11002TGCTGACGGG TTGCTA ATGTGTGCTGGCGGGATA TAGGAT GTGTGTTGAC /H12011/H12011 /H1102140.0 52 P1ppfA 62785 /H11002GATTCTGTTC TTGATG AATTAAACAAAGCGG CACAGT AACAAAGGAC 50.3 P2ppfA 63001 /H11002GAACAACAAG TTGAAG CACTGAAGGAAGCTACTG AACAAT TTGGTGCTTT 52.7 Pdmt 63962 /H11001CTAATAAATA TTGTTT TTTATGTCGTGTTTTCGG TACCAT TCAGCCATCG /H12011 63.3 59 P2ban 67452 /H11001GGTGTAGCTC ATGACG TTTTGGGTATATTGCTCTA TAAATT TATTAGTGTA 52.7 P1ban 67469 /H11001TTTGGGTATA TTGCTC TAATAAATTTATTAGTG TAATAT CGCCTCAATG /H12011/H11569 50.3 209 P1upl 69024 /H11002GAATTTCTTT TTGAGA ATGCCGAGCATTTAT TAACCT CATTACTGGT 56.2 P2upl 69055 /H11002TTCCATCTTC TTGACT CCGCCAGCCGCTTTGCG GAATTT CTTTTTGAGA 51.5 P2dbn 70823 /H11001GTTTTTAGCC TTGAGA TGCCGAGCCACCAGCT GATGAT GCGCTCACTG 53.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5da0>, 'json_data': {'input': ['page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  gp12?)36654 –37088 /H11002 15,973 Tail ﬁber structure or assembly 104S gp16 37167 –38003 /H11002 31,357 Baseplate or tail tube 104,SP190,SP twS BplA (gp3?) 38003 –39436 /H11002 53,577 Putative baseplate structural protein (by homology), may mediate contact between hub and wedges, as does its homolog in T4twS PmgA 39433 –39789 /H11002 13,239 Putative morphogenetic function twS Sit 39789 –43211 /H11002 120,710 Structural injection transglycosylase; putative tail tube “ruler ” 190S PmgB 43293 –44174 /H11002 34,138 Putative morphogenetic function twS Tub 44189 –44800 /H11002 22,316 Major tail tube protein twS PmgC 44811 –45377 /H11002 21,017 Putative morphogenetic function twS SimC /H11569(Sim) 45587 –46399 /H11002 30,800 Confers superimmunity when in high copy number by blocking P1 at entry; requires removal of leader sequence77, 170, 208 /H11569S SimB (gpORF50) 46374 –46691 /H11002 12,033 Unknown; superimmunity-linked function 213,SDtwS SimA 46713 –46844 /H11002 4,842 Unknown; superimmunity-linked function twS C4/H11569 47144 –47220 /H11001 (77) Antisense RNA that inhibits Icd and Ant1/Ant2 synthesis by acting on target antRNA; processed to active form by RNase P26,S33,S57,/H11569 58, 112,S 115, 130S Icd/H11569(gpORFX) 47271 –47492 /H11001 8,397 Reversible inhibition of cell division, apparently required for ant expression123, 131, 257S/H11569 Continued on following pageVOL. 186, 2004 BACTERIOPHAGE P1 7039', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TABLE 3 —Continued Nameb(former name[s])Gene coordinatescCoding strandMol mass (Da) (nucleotides in RNA)Basis for name; known or proposed functionReference(s)d Ant1 /H11569(RebA) 47489 –48532 /H11001 38,667 Antagonism of C1 repression; forms complex with Ant2 33,S112,SD 122, 131,S 256/H11569 Ant2 /H11569(RebB) 47738 –48532 /H11001 29,039 Antagonism of C1 repression, forms complex with Ant1 33,S112,SD 131,S256/H11569 KilA 48697 –49497 /H11001 29,575 Unknown, expression can kill host 112,S295S RepL 49527 –50372 /H11001 30,857 Lytic replication, initiates replication at oriL (within repL ) 112,S295S RlfA 50541 –51245 /H11001 27,716 Unknown; possibly replication-linked function twS RlfB 51245 –52444 /H11001 18,632 Unknown; possibly replication-linked function twS PmgF 51875 –52171 /H11002 11,423 Putative morphogenetic function twS BplB 52255 –52764 /H11002 18,804 Baseplate twS PmgG 52776 –53357 /H11002 20,696 Putative morphogenetic function twS gp21 53393 –54208 /H11002 29,329 Baseplate or tail tube 323, twS gp22 (gp15?, C7?)54218 –55807 /H11002 56,935 Sheath protein 323, 325, /H11569twS gp23 (gp15?, C7?)55868 –57574 /H11002 62,248 Major head protein, present in P1 heads in full-length (62 kDa) and truncated (44 kDa) forms323, 325, /H11569twS ParB /H11569 57800 –58801 /H11002 37,440 Plasmid partitioning; binds to parS , enhances ParA-mediated repression of paroperon and ATPase of ParA; can spread over DNA ﬂanking parS , silencing gene expression; pairs parS loci4,SD70, 73, 83, 90,/H11569114, 203,SP258, 343 ParA /H11569 58818 –60014 /H11002 44,269 Plasmid partitioning; weak ATPase, binds to parO repressing transcription, binds to ParB- parS partition complex4,S37, 70, 73, /H11569 343 RepA /H11569 60572 –61432 /H11002 32,219 Plasmid replication initiated by chaperone-activated monomers at oriR; represses own synthesis; binds at iterated sites ( incC and incA ) and can handcuff them, controling plasmid copy number2,/H115693, 6,S49, 52 UpfA 61751 –62143 /H11002 14,747 Unknown protein function twS Mlp 62155 –62487 /H11002 12,597 Membrane lipoprotein precursor (by homology); ? lysis control twS PpfA 62321 –62770 /H11002 15,744 Possible periplasmic function twS UpfB 62783 –63628 /H11002 29,310 Unknown protein function twS UpfC 63580 –63864 /H11002 9,478 Unknown protein function twS Uhr 64034 –64318 /H11001 10,781 Unknown; product of gene upstream of hrdC twS HrdC 64311 –65216 /H11001 33,638 Homolog of RdgC ofE.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  lysis control twS PpfA 62321 –62770 /H11002 15,744 Possible periplasmic function twS UpfB 62783 –63628 /H11002 29,310 Unknown protein function twS UpfC 63580 –63864 /H11002 9,478 Unknown protein function twS Uhr 64034 –64318 /H11001 10,781 Unknown; product of gene upstream of hrdC twS HrdC 64311 –65216 /H11001 33,638 Homolog of RdgC ofE. coli; possibly involved in replication, recombination, and/or Dmt functiontwS Dmt (Dam) 65216 –67477 /H11001 83,595 Predicted bifunctional DNA methyltransferase; methylates A ’si n GATC sequences, and (by homology) probably C ’s at unknown sites; affects replication control, late gene expression, packaging59, 66, twS tRNA1 68149 –68224 /H11001 (76) Proposed tRNA-Asn speci ﬁc for the AAC codon twS tRNA2 68227 –68302 /H11001 (76) Proposed tRNA-Thr speci ﬁc for the ACA codon twS Plp 68273 –68650 /H11002 13,800 Unknown; putative lipoprotein twS Upl 68651 –68842 /H11002 7,092 Unknown; product of gene upstream of plp twS TciA 69034 –69459 /H11001 15,187 Tellurite or colicin resistance or inhibition of cell division (by homology)twS TciB 69459 –69623 /H11001 5,752 Tci accessory protein (by homology) twS TciC 69623 –69730 /H11001 3,719 Tci accessory protein twS tRNA3 69734 –69809 /H11001 (76) Proposed tRNA-Ile speci ﬁc for ATA codon twS Ban/H11569 70096 –71460 /H11001 50,479 DnaB (of E. coli ) analog and homolog; replicative DNA helicase 67, 195 /H11569S237, 254/H11569 Dbn 71460 –72458 /H11001 37,379 Unknown; product of gene downstream of ban twS gp5 72505 –73137 /H11002 21,669 Baseplate 323, twS gp6 (C5?) 73130 –74146 /H11002 37,222 Tail length determination 323, twS gp24 (C5?) 74148 –74933 /H11002 28,880 Baseplate or tail stability 323, twS gp7 74920 –75648 /H11002 27,145 Tail stability 323, twS gp25 75652 –76869 /H11002 45,834 Tail stability 323, twS gp26 76879 –77256 /H11002 14,456 Baseplate 323, twS PmgL 77436 –77648 /H11001 9,258 Putative morphogenetic function twS PmgM 77651 –78229 /H11001 21,888 Putative morphogenetic function twS PmgN 78296 –78451 /H11001 5,593 Putative morphogenetic function twS PmgO 78393 –79055 /H11001 24,217 Putative morphogenetic function twS PmgP 78953 –79579 /H11001 23,198 Putative morphogenetic function twS Ppp 79579 –80253 /H11001 25,568 P1 protein phosphatase, possible role in head morphogenesis twS PmgQ 80250 –80951 /H11001 26,912 Putative morphogenetic function twS PmgR (gp8?) 81033 –81251 /H11001 8,318 Putative morphogenetic function twS PmgS (gp8?) 81253 –82515 /H11001 48,345 Putative morphogenetic function; ? prohead scaffolding protein twS Pap 82588 –83094 /H11001 19,200 P1 acid phosphatase (by homology); ?', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  81033 –81251 /H11001 8,318 Putative morphogenetic function twS PmgS (gp8?) 81253 –82515 /H11001 48,345 Putative morphogenetic function; ? prohead scaffolding protein twS Pap 82588 –83094 /H11001 19,200 P1 acid phosphatase (by homology); ? head morphogenesis twS PmgT 83289 –84218 /H11001 35,155 Putative morphogenetic function twS PmgU 84023 –84358 /H11001 12,163 Putative morphogenetic function twS PmgV 84355 –84717 /H11001 13,975 Putative morphogenetic function twS Continued on following page7040 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  [86]/H11022/H11022GUG[16] /H11022UUG[10]). Translation stops predominantly at UAA and less frequently at UGA and UAG. Utilization of UUG as the translation initiation codon ap- pears to be more frequent in P1 than in other coliphages (224)and in E. coli itself (35). It was con ﬁrmed experimentally that translation of the P1 lxcgene, which encodes a modulator of C1-mediated repression, initiates at UUG (272). Whereas the translation initiation codons of P1 are generally 6 to 10 nt downstream of typical E. coli ribosome binding sites (RBS), genes plp,pmgO , and pdcA lack nearby RBSs. They are probably translationally coupled to genes immediately up-stream ( plp) or overlapping ( pmgO and pdcA ). The gene ref (recombination enhancement function) in a monocistronic op-eron is more unusual. The sequence that most closely resem-bles a canonical RBS is separated by 16 bp, including a run ofeight T ’s, from a presumptive refinitiation codon (207, 338). What purpose the extra RNA might serve or how it is accom-modated remains to be determined. Promoter-proximal regions of four genes, pro,hrdC ,pmfA , and pmgL , contain, in addition to ATG codons preceded by sequences with a good match to RBSs in mRNA, TTG codonssimilarly preceded by canonical RBS sequences and furtherupstream. Possibly, each of these genes encodes two proteinsthat differ in the length of their N-terminal domains. Alterna-tive translational starts in mRNA at UUG or AUG couldcontrol expression of these genes by modulating the intracel-lular concentration or activity of their protein products. Although genes of P1 are tightly packed within the genome, overlaps are rare and occur mostly at coding termini. Spacesbetween genes of adjacent operons are in most cases limited toshort regions containing promoter and regulatory sequences.Of those that exceed 250 bp, one contains oriR and another incA , both involved in P1 prophage replication (Table 1). A 272-bp region between the ﬁrst gene of the simoperon, simA , andc4, was shown to encode the 5 /H11032region of the c4 transcript, which is cleaved to yield C4 RNA (115). Two regions withintheban operon (531 bp preceding the trnA gene and 359 bp between the trnIgene and ban, described in detail later in this paper) may also encode species of RNA that are processed. Of previously identi ﬁed or predicted products of P1 genes, 65 exhibit signi ﬁcant homology to known proteins of other organisms (Table 3; see also Table S1 in the supplementalmaterial). Only 29 are homologous to proteins encoded exclu-sively by other phages. Although they include 11 homologs ofproteins of T4 or T4-like phages, there is no clear preferencein these homologies for proteins of a particular phage, suggest-ing that P1 has had a long, separate history. Homologs ofknown genes of different provenance are scattered throughoutthe P1 DNA. Whether their distribution results from the hor-izontal transfer of modules during the evolution of the P1genome or from convergent evolution remains to be deter-mined. (iii)Organization of P1 genes and transcription units. P1 genes can be grouped into 45 operons, of which 15 appear tobe monocistronic. Their organization resembles that of T4 (224) in that only some genes of related function are in clusters. Regulatory regions of 38 operons contain one or more se- quences that resemble strong /H9268 70promoters (Table 4). Of 26 /H926870promoters identi ﬁed previously by transcriptional fusion or primer extension experiments, only 9 had a match to the con- sensus sequence below 50% and required relaxing the strin-gency of the prediction program for their detection.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Whether their distribution results from the hor-izontal transfer of modules during the evolution of the P1genome or from convergent evolution remains to be deter-mined. (iii)Organization of P1 genes and transcription units. P1 genes can be grouped into 45 operons, of which 15 appear tobe monocistronic. Their organization resembles that of T4 (224) in that only some genes of related function are in clusters. Regulatory regions of 38 operons contain one or more se- quences that resemble strong /H9268 70promoters (Table 4). Of 26 /H926870promoters identi ﬁed previously by transcriptional fusion or primer extension experiments, only 9 had a match to the con- sensus sequence below 50% and required relaxing the strin-gency of the prediction program for their detection. All nineTABLE 3 —Continued Nameb(former name[s])Gene coordinatescCoding strandMol mass (Da) (nucleotides in RNA)Basis for name; known or proposed functionReference(s)d UpfM 84900 –85013 /H11001 4,343 Unknown protein function twS UpfN 85029 –85226 /H11001 7,153 Unknown protein function twS UpfO 85204 –85344 /H11001 5,238 Unknown protein function twS Hot 85368 –85631 /H11001 9,694 Homolog of theta subunit of DNA polymerase III; ? replication twS Lxr 85638 –86210 /H11002 22,042 LexA-regulated function twS HumD /H11569 86385 –86774 /H11001 14,313 Homolog of UmuD /H11032subunit of E. coli repair protein DNA polymerase V (UmuD /H11032-UmuC); complements mutant UmuD /H11032200, 217, /H11569276S Phd/H11569 86847 –87068 /H11001 8,132 Prevention of host death by Doc toxicity; confers, with Doc, addiction to P1; represses transcription of addiction operon95,/H11569192,S194, 210 Doc/H11569 87068 –87448 /H11001 13,587 Death on curing; toxin of P1 addiction system; reversible inhibitor of protein synthesis; corepressor of addiction operon95, 121, 192,S 210, 211 /H11569 PdcA 87432 –87632 /H11001 7,630 Unknown; post-doc 192,Stw PdcB (gpORF93) 87660 –88703 /H11001 39,877 Unknown; post-doc 192,SDtw Lpa/H11569(gp10) 88792 –89244 /H11001 18,117 Late promoter activator 111, /H11569188S PacA /H11569(gp9) 89330 –90523 /H11001 45,246 DNA packaging; cuts at pactogether with PacB 286,S288/H11569 PacB /H11569 90523 –92007 /H11001 55,604 DNA packaging; cuts at pactogether with PacA 286,S287/H11569 C1/H11569 92032 –92883 /H11002 32,544 Controls plaque clarity; primary repressor of lytic functions 79, /H1156984, 124, 238,S316, 319 Coi/H11569(gpORF4) 92994 –93203 /H11002 7,741 C one (reversible) inactivator; forms 1:1 complex with C1 25,S27, 84,S 127, 130 /H11569 ImcB (gpORF3) 93169 –93264 /H11002 3,659 Immunity C function; may regulate c1 27, 84S ImcA (gpORF2) 93283 –93396 /H11002 4,340 Immunity C function; may regulate c1 27, 84S aA more complete table, which includes pI, number of amino acid residues (or nucleotides in RNA), and features of the amino acid sequence (closest homol ogs and amino acid sequence motifs) is to be found as Table S1 in the supplemental material.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  bGene products marked by an asterisk were either puri ﬁed or identi ﬁed in PAGE gels. cCoordinates of genes refer to their positions in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. Coordinates of protein-coding genes are from the initial codon through the ﬁrst stop codon. dReferences marked by an asterisk concern identi ﬁcation of a protein or RNA product of a given gene; those marked by a superscript S concern previously published sequences. Superscript SD, published sequence differs internally from sequences presented here; superscript SP, published sequence is partial; t w, this work.VOL. 186, 2004 BACTERIOPHAGE P1 7041', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TABLE 4. Known and putative /H926870promoters of P1 genes Name (former name)PositionaStrandSequenceb Basis for identi ﬁcationc Reference(s) /H1100235 /H1100210 tsp pp cp P2coi 41 /H11002TGCTAAGGTA TTGAAC TGTATGGATTTACAGG TAAATTGATCATATTC 56.8 P3cre 104 /H11001ACTTGATCAT TTGATCAAGGTTGCGCTACG TAAAAT CTGTGAAAAA /H12011/H12011 42.6 302 P2cre 283 /H11001GCAAAAGGGT TTGATCGTGATAGTTGCCAACTGT CATTAT CGCGCGGTGA /H12011/H12011 45.6 302 P1cre 400 /H11001GAGCCGATCC TGTACA CTTTACTTAAAAC CATTAT CTGAGTGTTA /H12011/H12011 /H1102140.0 302 Pc8 1394 /H11001AATGTAAATA TTGTCA TGAACTATATCCGTAACCTG GATAGT GAAACAGGGG 54.4 P2ref (Pref-2)1957 /H11001AACAAAATAT CTATCA TTGCTCTAATTGATTGC TATAAT TGAGCCGCAG 59.8 338 P1ref 2017 /H11001TACGAAGACG TTGCCA TTACTTCACTCCTTGA CATCAT TGGCGGCCAT 52.7 Pmat 2666 /H11001TGATAAATGA TTGACA ACTGACAAGTGACTTCAGT CAGAAT CATCACACGC 56.8 P1res 6558 /H11002GATGAAGATT TTGCCC CAAACAAAGTCGTGTTT TATGGT TGTAACTTTG 60.4 P2res (PR1)6891 /H11002CAGGAGGTAT TTAAAA AAAGCACAACCACAAGT GATAAT GCGGCTAAAA S1/H12011 52.1 285 P3res 6926 /H11002AGATCTATTT TTGAAA TAACAAAAAAAAGGA TACAGG AGGTATTTAA S1/H12011 47.3 285 P1mod (PM1)8450 /H11002CATATAGATA GTGATA GTGCCACAACTTCTGGC TCTAAC GGGCTGGGGA /H12011 45.6 285 P2mod (PM2)8548 /H11002ATCATAGATA ATGCAT TGCTTCTCAATGCGGAT TAAACT TCCCCAAATT /H12011/H12011 50.9 285 P1lxc 8844 /H11002CTGCCAGTCC CTGAAG AAGTTGACGACCAGGAC TATGAG TTTGAGTCTT S1 46.7 285, 318 P2lxc 9022 /H11002AATTAAAAAA TTATTA CTGGAGCGCGGTTATGG TAATTT CGGTCGTCTC S1 51.5 285 P3lxc 9167 /H11002AAAACATATC TTGAGG CGGTACTGCGTACGG CATTGT TAGGAAAGAC S1 51.5 285 PdarB 16093 /H11002CGTGATTTTA TTGTCT GCCAAGATTGCACCTTAAT TAGAAT AATTCACATC 56.8 P2ssb (Pr21)19787 /H11001TGCAAATCAC TTGTTA GCTACGTTTCAAAGATA TACATT ATTGCTCTAA /H12011 52.7 59 P1ssb 19821 /H11001ACATTATTGC TCTAAT TAATTTATTTTATTAGG TAAGAT AAGTGGCACA 55.6 Plyz 19845 /H11002GATGACTTTG TTTACA CCGCGTTGTGCCACT TATCTT ACCTAATAAA 53.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  5 285 PdarB 16093 /H11002CGTGATTTTA TTGTCT GCCAAGATTGCACCTTAAT TAGAAT AATTCACATC 56.8 P2ssb (Pr21)19787 /H11001TGCAAATCAC TTGTTA GCTACGTTTCAAAGATA TACATT ATTGCTCTAA /H12011 52.7 59 P1ssb 19821 /H11001ACATTATTGC TCTAAT TAATTTATTTTATTAGG TAAGAT AAGTGGCACA 55.6 Plyz 19845 /H11002GATGACTTTG TTTACA CCGCGTTGTGCCACT TATCTT ACCTAATAAA 53.3 PisaB 22220 /H11001CAATGATGGT TAGACA AGCAAGATAGGGCGAAT TAAAAT TAATCAAGAG 61.5 PinsA 22229 /H11002AAAGTGACTC TTGATT AATTTTAATTCGCCC TATCTT GCTTGTCTAA 57.4 P2cin 30339 /H11001CACATAACTA TTTCCT TAATAGTGAATTAAA TATCAT TGGGAAACGG 55.6 P1cin 30390 /H11001TACTTTGTGA TTTCCA CACATACTGGTTTTTGTTAAT TAAAAT CCGCAGCTTG 53.3 PlydC 30461 /H11002TGTTCATTTG TTGATA CGCGTACATAGCCTAT TAGCAT ATTTTCTGCT 59.2 P1pmgC 45555 /H11002AAAATAGTGT TTGAAA GGATAGTCAATTTAAGT TTTAAC ACAACGCCCA 59.8 P2pmgC 45636 /H11002CAACAATTCG TTGATG TAATGGAAGGATTTGA TAGCAT CATGTATAAA 59.8 P3pmgC 45771 /H11002CAAATATATA TTCACA AATCAA GATCCTGTTCA GATGAT GAATTTTATA 58.0 PsimB 46843 /H11002TGTACCCTAA ATGAGT TATAAGGCAGGTGAGGT TATAAT GAGAAAACTA 56.8 PsimA 46909 /H11002GCATTATTTG TTTAAT AAATACACAGTTG GATCTAAT AACCTCTTTT 51.5 P2c4 (P51a)46956 /H11001TCTAATAAAT TTGTAT TTTTAAGTCGCGAATGC TATCTT TTCGCATCAT /H12011/H11569 56.8 131 P1c4 (P51b)46996 /H11001TCGCATCATA TTGACC TTTTAATCGTTCAGGCT TATAGT TCCACCGTCG /H12011/H11569 60.4 131 PkilA (P53)48593 /H11001CAACCCGATCTGGATCGGGTCAGAAAAATTTGC TCTAAT AAATTTCGTT /H12011/H12011 50.3 295 PaskilA (P53as)48724 /H11002GGTTGAAGGA TCGACA TTTTGATGAAGGTTTGA TATATT CATATCCGCA /H12011/H12011 62.7 125 PrepL 49267 /H11001AAACTATCTC TTCACC TGGTTGCGTGATAACGG AATTCTGATCGCAACC 52.1 PrlfA 50486 /H11001CGCCCCATTA TTGCAA TTAATAAACAACTAACG GACAAT TCTACCTAAC 52.7 P1rlfB 50663 /H11001AAAAAATTGC TAGAGA CATCAAGGAATTAAAAT TAAAAC ACAATCTAAA 55.0 P2rlfB 50704 /H11001AATCTAAAAC CTGATT TCGAAATCAAGTGGAC TAAAGT ATCTGCGTCC 56.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  7 125 PrepL 49267 /H11001AAACTATCTC TTCACC TGGTTGCGTGATAACGG AATTCTGATCGCAACC 52.1 PrlfA 50486 /H11001CGCCCCATTA TTGCAA TTAATAAACAACTAACG GACAAT TCTACCTAAC 52.7 P1rlfB 50663 /H11001AAAAAATTGC TAGAGA CATCAAGGAATTAAAAT TAAAAC ACAATCTAAA 55.0 P2rlfB 50704 /H11001AATCTAAAAC CTGATT TCGAAATCAAGTGGAC TAAAGT ATCTGCGTCC 56.2 P3rlfB 50746 /H11001GCGTCCAAAG TTGAAT TTTACTTAGATGTCG TAGATT ACTTCTTCAG 55.0 P4rlfB 50971 /H11001ATTGAAAAAA TTGAGA AATTAAGAGGGGTTCTTCA CAATGATCGCTATGAT 57.4 Pasrlf 51784 /H11002TAGCTAGCTA TTGAAA AGCTAAAGTA GATCGGT TAGATT TGCATTACTC 69.2 P1pmgF 52236 /H11002TACAACTGGA TTGAGT GGGATTAAGAGTCATCCCTTG TATTTT AAAGCTCCTT 53.3 P2pmgF 52342 /H11002GCTACAACAA TTGAAA TGTTGGACTGCAAAATC TACAGT GATGCAATCG 60.9 P123 57631 /H11002AAAAAATTGC TTGTCA CGAGAAAGTCAACAAGTGACT TTCAAT AAAATCTCTT 57.4 P223 57740 /H11002TATTAACTGA CTGTTT TTAAAGTAAATTACTC TAAAAT TTCAAGGTGA 56.2 PparB 59274 /H11002ATTGCCGCGG AAGATT TTGCTAAAGCAGTTTTT GACCGT ATTGAATTTA /H12011/H12011 /H1102140.0 A. Dobruk, unpublished Ppar 60069 /H11002GAGAATGCTA TGTACA AGCATCTACGCATACAT TATTAT TTTATGCAGC /H12011/H12011 54.4 73, 120 PrepA 61505 /H11002TGCTGACGGG TTGCTA ATGTGTGCTGGCGGGATA TAGGAT GTGTGTTGAC /H12011/H12011 /H1102140.0 52 P1ppfA 62785 /H11002GATTCTGTTC TTGATG AATTAAACAAAGCGG CACAGT AACAAAGGAC 50.3 P2ppfA 63001 /H11002GAACAACAAG TTGAAG CACTGAAGGAAGCTACTG AACAAT TTGGTGCTTT 52.7 Pdmt 63962 /H11001CTAATAAATA TTGTTT TTTATGTCGTGTTTTCGG TACCAT TCAGCCATCG /H12011 63.3 59 P2ban 67452 /H11001GGTGTAGCTC ATGACG TTTTGGGTATATTGCTCTA TAAATT TATTAGTGTA 52.7 P1ban 67469 /H11001TTTGGGTATA TTGCTC TAATAAATTTATTAGTG TAATAT CGCCTCAATG /H12011/H11569 50.3 209 P1upl 69024 /H11002GAATTTCTTT TTGAGA ATGCCGAGCATTTAT TAACCT CATTACTGGT 56.2 P2upl 69055 /H11002TTCCATCTTC TTGACT CCGCCAGCCGCTTTGCG GAATTT CTTTTTGAGA 51.5 P2dbn 70823 /H11001GTTTTTAGCC TTGAGA TGCCGAGCCACCAGCT GATGAT GCGCTCACTG 53.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:36 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'96'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994773'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'313ms'), (b'x-request-id', b'aa3ab0ce450faa35a27fa2f16effbe47'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249057d515d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:36 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'96'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994773'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'313ms'), (b'x-request-id', b'aa3ab0ce450faa35a27fa2f16effbe47'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249057d515d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  7 P1ban 67469 /H11001TTTGGGTATA TTGCTC TAATAAATTTATTAGTG TAATAT CGCCTCAATG /H12011/H11569 50.3 209 P1upl 69024 /H11002GAATTTCTTT TTGAGA ATGCCGAGCATTTAT TAACCT CATTACTGGT 56.2 P2upl 69055 /H11002TTCCATCTTC TTGACT CCGCCAGCCGCTTTGCG GAATTT CTTTTTGAGA 51.5 P2dbn 70823 /H11001GTTTTTAGCC TTGAGA TGCCGAGCCACCAGCT GATGAT GCGCTCACTG 53.8 P1dbn 71207 /H11001AATCGATCGC TTGAAC AGCGTGCGGACAAACGACCGG TAAATT CAGATTTACG 52.1 P2pmgL 77306 /H11001TAGTCATTTG TTGAAT ATTTAACTCAATAAAA GAAAAT TATTAGTGCA 56.8 P1pmgL 77339 /H11001AAAATTATTA GTGCAA TTTTGATTGTGAAATG TATCAT TCTGCCCTTA 56.8 P26 77617 /H11002AACATAATCA GTCAAA TCCTGCCGGTCGCATGC CATATT TACGCTGTTT 50.3 PpmgN 78197 /H11001TGAGCATGTA TTGAAC GGCGATAAGTGGTTG TATGCT GCTTAAAATA 49.1 Continued on following page7042 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  promoters are associated with genes expressed during lysog- eny: three with cre, three with resormod (including the P1 lxc promoter, which can also drive transcription of lxc), one with parB , one with repA , and one with c1. A relatively high pre- dicted strength of known and putative promoters of genesexpressed during P1 lytic development implies that as soon asthese promoters become accessible to the host RNA polymer-ase, they can effectively compete with E. coli promoters for the enzyme. Multiple promoters within regulatory regions of sev-eral operons indicate a requirement for different controls onthe expression of genes at different stages of phage develop-ment or plasmid maintenance or a requirement to adapt toalternative hosts or environmental conditions. Ten promoters, including a promoter of the essential phage replication gene repL , contain the so-called “extended /H1100210 region ”characterized by the presence of T and G at positions /H1100215 and /H1100214, respectively. In E. coli , transcription from ex- tended /H1100210 promoters can initiate in the absence of typical /H1100235 hexamers (178). Sixteen operons end in sequences characteristic of typical Rho-independent transcriptional terminators (Fig. 1 and Table5). Either termination of some operons of P1 occurs by aRho-dependent mechanism or certain Rho-independent tran-scription termination sites are too weak to be detected. Per-haps in the latter case a terminator further downstream isalternatively used. In the case of certain P1 operons expressedduring lysogeny (for instance, resand phd doc pdcA pdcB ), C1repressor-binding sites of downstream operons may possibly be used as roadblocks to transcript elongation. Five predicted Rho-independent transcription terminators (Tc8, T ref,TddrA ,T p lydC , and T dmt) follow closely upon apparently functional promoters. The locations of those termi-nators (in a leader sequence or early in a gene) suggest controlby antitermination. Indeed, the T c8 terminator was shown to participate in the regulation of transcription of a gene down-stream of c8,ref, by prematurely terminating transcription from one of the refpromoters (338). Of the ﬁve terminators, four are positioned in such a way that, in addition to prematuretermination of ref,mat,ddrB , and dmt genes, they can function to terminate transcription of operons immediately upstream.It is likely that P1 encodes an antitermination mechanism.Whereas no P1 protein has been implicated in the process,antitermination might not require such a protein (330). As many as 12 predicted Rho-independent terminators (Tasref, Tas pro, Tas lyz, Tas ddrB , Tas ddrA , Tas kilA, Tas 23, Tas tciA, Tas7, Tas pmgM , Tas pmgT , and Tas doc) are at the beginning of operons but opposite in orientation to the operon. This ar-rangement suggests that these terminators could terminatetranscription of regulatory antisense RNAs. Antisense RNAhas been shown previously to regulate the expression of icd ant1/2 and kilA repL (125, 126). The location of the predicted TaskilA terminator is consistent with that expected from the length of the antisense RNA transcribed from the Pas kilA promoter (125).', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  As many as 12 predicted Rho-independent terminators (Tasref, Tas pro, Tas lyz, Tas ddrB , Tas ddrA , Tas kilA, Tas 23, Tas tciA, Tas7, Tas pmgM , Tas pmgT , and Tas doc) are at the beginning of operons but opposite in orientation to the operon. This ar-rangement suggests that these terminators could terminatetranscription of regulatory antisense RNAs. Antisense RNAhas been shown previously to regulate the expression of icd ant1/2 and kilA repL (125, 126). The location of the predicted TaskilA terminator is consistent with that expected from the length of the antisense RNA transcribed from the Pas kilA promoter (125). The antisense RNA terminated at Tas 23mayTABLE 4 —Continued Name (former name)PositionaStrandSequenceb Basis for identi ﬁcationc Reference(s) /H1100235 /H1100210 tsp pp cp PpmgR 80954 /H11001CAGAATAATA TTGCTC TAATAATTCCATATTTT TAAAAC GTGATGTACA 52.7 P1pmgT 82778 /H11001ACTCCAGTTA TTGATA TGGTGCGCCAGTTATT TAACGT TTACACGGTC 56.8 P2pmgT 83095 /H11001TCAACAATAA TTGCTC TAATAAATCTTGATTTT TAAAAC AGAGAAAGTG 52.7 P3pmgT 83110 /H11001CTAATAAATC TTGATT TTTAAAACAGAGAAAGT GAAAAT AAAAACATGC /H12011 66.3 59 P1upfM 84752 /H11001CAGAAAATGA TTGTTT CCACATCAAGGAGATTT TAATGT TTCACTGAAA 55.6 P2upfM 84785 /H11001TAATGTTTCA CTGAAA CATTAAGTAAGCCAGTG CATAAT TCCATTTTTT 55.6 PupfN 84973 /H11001CTTGCTCACG TTGACG TAGAAACCCAACCCTTA TATAGT TGGATTCGGT 59.8 P1lxr 86267 /H11002TCAACAACCG CTGATG ATTTTGTGCGCTTTGC TACTAT TCATCACCAA 49.7 P2lxr 86347 /H11002CCATAGCGTT TTCACC TCAATAATACTGTTCATTTA TACAGT ACACATTAAA 48.5 P1humD 86318 /H11001ACGTAACGCG TTTGCA CCAAAGGTGTCTCTT TAATGT GTACTGTATA /H1102140.0 200 Pphd 86787 /H11001TGCAAAGTGC TGGTGC TTTATGCCTGTGAAGTT TATAAT TGTGTACACA /H1201152.1 210 Plpa (Pr94)88719 /H11001CTCTAATGTA TTGCTA TTTCTTTAATCGAGGG TATTAT ATTCCACGTT /H12011/H12011 57.4/H11569193 PpacB 90384 /H11001GAAGAAATTG CCGATA TCGTCGATACAGGTGGT TATGGT GATGTCGATG 50.3 Pc1(P99a) 92951 /H11002CACTAATAAA TCTATT ATTTTCGTTG GATCCTTC TATAAT GGTGGCCAAC /H12011/H12011 46.7/H1156984 P1coi (P99d)93438 /H11002TCATAGTTGT TTGACA ATTGCTCTAATAAATTAT AGTTTT GCCGCCGTTT /H12011/H12011 60.9/H11569127 P4cre (P99e)93541 /H11001CTAATAATTC TTGATT TTTATGCGCAGCTGGACG TAAACT CCTCTTCAGA /H12011/H12011 63.9/H1156984 aCoordinate of the ﬁrst nucleotide of the /H1100235 sequence for promoters from which transcription proceeds on the /H11001strand (clockwise) or of the last nucleotide of the/H1100210 sequence for promoters from which transcription proceeds on the /H11002strand (counterclockwise).', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Coordinates refer to positions of promoters in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5and the associated 4-bp duplication. The locations of P3 res,P 1mod,P 1lxc,P 2lxc, and P3 lxcpromoters proposed in reference 285 are corrected. bConserved /H1100210 and /H1100235 promoter hexamers as well as the “extended /H1100210”motifs (T and G at positions /H1100215 and /H1100214, respectively) and known transcription start sites are underlined. The 5 /H11032-GATC sequences that are targets for Dam- or Dmt-mediated N-6 adenine methylation are in boldface. cIdenti ﬁcation of promoters is based on results of transcription start point determination ( tsp) or RNA polymerase binding assay ( tsp/H11569), detection with the promoter probe vector ( pp), or similarity of the sequence to the consensus sequence for /H926870promoters calculated with the use of the program Targsearch ( cp) (229). References that indicate existence of a promoter in a given region based on S1 nuclease mapping of 5 /H11032ends of transcripts are marked with an S1 in the tspcolumn. Putative promoters which have not been veri ﬁed experimentally are shown only if their homology to the consensus exceeds 50%, as calculated by the method of Hawley and McClure (118) and if their location suggests a function in initiation of transcription of genes or regulatory RNA. Exceptionally, putative promoter s that have a homology score to the consensus below 50% but overlap putative regulatory sites are also shown. For comparison, homology scores are shown for both known and put ative promoters, the former marked with an asterisk.VOL. 186, 2004 BACTERIOPHAGE P1 7043', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  regulate translation of the 23gene from an alternative, up- stream start site. Genes expressed during establishment or maintenance of P1 prophage. The alternative life-styles of P1 require two sets of genes with little functional overlap. In general, the genesassociated with lysogeny can be distinguished by the absence ofregulatory regions resembling those known to control P1 lyticfunctions, the C1-controlled operators or the Lpa-activatedpromoters, which will be described later in this paper. Of 45 P1 operons, 14 are not directly controlled by C1 or Lpa (Fig. 1). Their expression is probably not up-regulatedduring lytic growth, other than by gene dosage, an effect that isdamped in those operons that are subject to autoregulation(ImmC and ImmI genes, repA ,parAB ,phd doc pdcAB ,cin, and probably others). As many as 20 genes of operons that appear to be indepen- dent of C1 and Lpa are either known or predicted to beassociated with lysogeny (Fig. 1 and Table 3; see also Table S1in the supplemental material). We are unable to predict the function of 12 other genes in this class ( isaB,upfA ,mlp,ppfA , upfB ,upfC ,plp,upl,upfM, upfN ,upfO , and upfQ ). However, three of them, mlp,ppfA and plp, encode products that have putative periplasmic transport or lipoprotein attachment sig-nals likely to contribute to lysogenic conversion. Only six structural genes are essential for stable inheritance of P1 as a prophage. Other genes associated with lysogeny mayprotect the lysogen from entry of foreign DNA ( resand mod; reviewed in references 32 and 38), from infection by anotherP1 or homologous phages ( sim, superinfection exclusion) (170), and from DNA damage ( humD , homolog of umuD ) (217). Possibly for the simproduct(s) to be effective, the gene(s) must be ampli ﬁed, as superimmunity was observed only with high- copy-number clones of sim. The C-segment inversion recom- binase cinis expressed by the prophage, but the known bene ﬁt it confers is to produce phages that differ in host speci ﬁcity from their parent.TABLE 5. Known and putative Rho-independent transcriptional terminators in P1 DNA NameaPositionbStrand SequencecRelative locationd Basis for identi ﬁcatione Tasref 1836 /H11002 TTTGTGCAGCCTGGCTCCTTGCCAGGCTTTTTTTTATTTCATCATGG 5/H11032-c8-3/H11032\\uf8ee5/H11032-ref-3/H11032 TT, TR Tc8 2046 /H11001 TCCTTGACATCATTGGCGGCCATTAGGCCGCCTTTTTTTTGCCATAT 5/H11032-c8-3/H11032\\uf8f95/H11032-ref-3/H11032 TT, TR Tref 2788 /H11001 CTTCAGTTGTATTGCTAAGCCGCCGCTGGTGGCTTTTCTTTTTTGTA 5/H11032-mat\\uf8f9mat-3/H11032 TR (338) Taspro 17761 /H11002 CCGATTCAAGGATTTGCGCCAGTTCCTGTGGCGGTGTTTTGATGATG 5/H11032-prt\\uf8eeprt-3/H11032 TR Tpro 18848 /H11001 CCACCAAAAATAACCCCGGCAGCTGCCGGGGTTCTCGTTAACTATTA 5/H11032-pro-3/H11032\\uf8f93/H11032-lyz-5/H11032 TR Tlyz 18848 /H11002 ATAGTTAACGAGAACCCCGGCAGCTGCCGGGGTTATTTTTGGTGGTT 5/H11032-pro-3/H11032\\uf8ee3/H11032-lyz-5/H11032 TT,', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TR Taslyz 19655 /H11001 GATAGCGCAAATTGCACCGCCTCCTGCGGCTGTTTTTCCCTTCATAA 3/H11032-lyz\\uf8f9lyz-5/H11032 TR TinsB 21351 /H11002 TTTCGTATTATCAGGGTGCTCTCTGGCACCCCTTTTTATTCTAGTAA 5/H11032-isaA-3/H11032\\uf8ee3/H11032-insB-5/H11032 TR TasddrB 26246 /H11001 CACCACATACGAAACCCGCCCGAAATCGGGCAGGTTCTTTAGCAGCT 3/H11032-ddrB\\uf8f9ddrB -5/H11032 TR TddrA 26246 /H11002 TGCTAAAGAACCTGCCCGATTTCGGGCGGGTTTCGTATGTGGTGACA 3/H11032-ddrB\\uf8eeddrB -5/H11032 TT, TR TasddrA 26508 /H11001 ACCAGTTTTGGCACACCATCATGTTCCGAAGGTTGTTTCTCCTCGCC 3/H11032-ddrA\\uf8f9ddrA -5/H11032 TR TplydC 30287 /H11002 ACAATAATGCTCCTCTCCATCCTTGGCGGGGTGCATTCGTTTCTGAA 3/H11032-lydC\\uf8eelydC -5/H11032 TR TU/H11032 33125 /H11001 AACCTGAATGAGACAAGGCCCGATAGCGGGCCTTAATTTTTATTCAG 5/H11032-U/H11032-3/H11032\\uf8f93/H11032-U-5/H11032 TT, TR T2simC 45384 /H11002 ACTCGTAGAATCGGTTAACACACCAGATTCTACGAGGTTTCAATGAC 3/H11032-pmgC -5/H11032\\uf8ee3/H11032-simC -5/H11032 TT T1simC 45535 /H11002 GTTTTAACACAACGCCCATTAAAGGGCGTTTTATTGTTTTACTCAAA 3/H11032-pmgC -5/H11032\\uf8ee3/H11032-simC -5/H11032 TT, TR TaskilA 48546 /H11002 CTGCTCAGAAACCTGCCAGTTTGCTGGCAGGTTTTTTTCTTTTGTTA 5/H11032-ant-3/H11032\\uf8ee5/H11032-kilA-3/H11032 TR TpmgG 52219 /H11002 CATCCCTTGTATTTTAAAGCTCCTTCGGGAGCTTTTTTGTGCTTAAA 3/H11032-pmgG -5/H11032\\uf8ee3/H11032-pmgF -5/H11032 TT, TR T22 53368 /H11002 GGCTAAAGGAGGAGGCCGGGGAACTCCGGCCTTTAACTTGAATGGCT 3/H11032-21-5/H11032\\uf8ee3/H11032-22-5/H11032 TT, TR Tas23 57399 /H11001 CATTATCAGAGATACCCGCAAAAACCGGGTCTTTACGTGCAGCTTCA 3/H11032-23\\uf8f923-5/H11032 TR TparB 57707 /H11002 TTTCAAGGTGAAATCGCCACGATTTCACCTTGGATTTTACCTTCCTC 3/H11032-23-5/H11032\\uf8ee3/H11032-parB -5/H11032 TT, TR TrepA 60474 /H11002 TTTGATGTGCGCTGGAGGGGGACGCCCCTCAGTTTGCCCAGACTTTC 3/H11032-parA -5/H11032\\uf8ee3/H11032-repA -5/H11032 TT TupfA 61718 /H11002 AATACGACTCCCTTCCAACCGGCTACGTTGGCCGGTTTTTCACTTAT 3/H11032-repA -5/H11032\\uf8ee3/H11032-upfA -5/H11032 TT,', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TR TrepA 60474 /H11002 TTTGATGTGCGCTGGAGGGGGACGCCCCTCAGTTTGCCCAGACTTTC 3/H11032-parA -5/H11032\\uf8ee3/H11032-repA -5/H11032 TT TupfA 61718 /H11002 AATACGACTCCCTTCCAACCGGCTACGTTGGCCGGTTTTTCACTTAT 3/H11032-repA -5/H11032\\uf8ee3/H11032-upfA -5/H11032 TT, TR ThrdC 67115 /H11001 TGGAAGCCTTCCTCGGTGCGGAAGGGCGGTCTGAGTTTGTATGCGGT 5/H11032-dmt\\uf8f9dmt-3/H11032 TT Tdmt 67659 /H11001 GAGCTTCCAGTTTGCCCATCTTCGGGTGGGCGTTTTTTCAGGGTTTT 5/H11032-dmt-3/H11032\\uf8f95/H11032-trnA-3/H11032 TR TastciA 69059 /H11002 TCTTCTTGACTCCGCCAGCCGCTTTGCGGAATTTCTTTTTGAGAATG 5/H11032-tciA\\uf8f9tciA-3/H11032 TR Tdbn 72485 /H11001 CCGTCATAACAGAAAAGCCCGAACGCCGGGCTTTTCTTAAGCCTTGT 5/H11032-dbn-3/H11032\\uf8f93/H11032-5-5/H11032 TT, TR T5 72485 /H11002 TGACAAGGCTTAAGAAAAGCCCGGCGTTCGGGCTTTTCTGTTATGAC 5/H11032-dbn-3/H11032\\uf8ee3/H11032-5-5/H11032 TT, TR Tas7 75768 /H11001 ACTTTATGAGGTTGTTCGCCTGCGTTTGCAGGGCTTTTTTGAGAGCG 3/H11032-25\\uf8f925-5/H11032 TR TaspmgM 77765 /H11002 GTGCTGCAAGTGCCTTCGCATGTGGAGAAACTTCTTTTACCGGCGCA 5/H11032-pmgM\\uf8eepmgM -3/H11032 TR TpmgQ 80982 /H11001 TTTTAAAACGTGATGTACACTCATCACGTTTTTTATTAGAGCAATCT 5/H11032-pmgQ -3/H11032\\uf8f95/H11032-pmgR -3/H11032 TT TpmgS 82537 /H11001 GTCACTTTTAATGCTGGTGGAGTGCGCCCACCAGCATTTTTTTCGTC 5/H11032-pmgS -3/H11032\\uf8f95/H11032-pap-3/H11032 TT Tpap 83146 /H11001 AGTGAAAATAAAAACATGCCGCAAGGCGCGGCATGTTTCCAATCAAT 5/H11032-pap-3/H11032\\uf8f95/H11032-pmgT -3/H11032 TT,', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TR TaspmgT 83144 /H11002 GATTGGAAACATGCCGCGCCTTGCGGCATGTTTTTATTTTCACTTTC 5/H11032-pap-3/H11032\\uf8ee5/H11032-pmgT -3/H11032 TR Tasdoc 87120 /H11002 TCAGACATTCCCGGCAGGCCGCCGTAGCGGCTTATATTCGCATCATG 5/H11032-doc\\uf8eedoc-3/H11032 TR Tlpa 89255 /H11001 ATTGATTGCTTGCCCGTTCCGGGCCTTTTGACATGTGACTTTCGTTA 5/H11032-lpa-3/H11032\\uf8f95/H11032-pacA -3/H11032 TT TpacB 92399 /H11001 TTGTTTCGCTAAAGCCGTGTACGCGCATACGGAATTTTTCATCCTCC 3/H11032-c1\\uf8f9c1-5/H11032 TR aNames of the terminators (T) that could terminate transcription of RNA antisense to mRNA of a given gene contain “as”and the name of the gene. The terminator inlydC is marked Tp lydC to indicate that it could serve primarily to terminate lydC transcription prematurely. bCoordinate of the ﬁrst nucleotide of the terminator hairpin for terminators that terminate transcription on the /H11001strand (clockwise) or the last nucleotide of the terminator hairpin for terminators that terminate transcription on the /H11002strand (counterclockwise). Coordinates refer to positions of terminators in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. cNucleotides that form the stem of the terminator hairpin are underlined. dTerminators are represented schematically by symbols \\uf8f9and\\uf8ee, which face toward the DNA region whose transcription they terminate. eIdenti ﬁcation of terminators is based on their similarity to known transcription terminators as determined by the program TransTerm (TT) or Terminator (TR) and veri ﬁed by additional criteria (see Materials and Methods).7044 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  (i) Immunity. Whether P1 is maintained in cells as a plasmid or enters the lytic pathway is dictated by the interplay of en-vironmental factors with the components of the immunity cir-cuit encoded at loci designated ImmC, ImmI, and ImmT (Fig.1 of the accompanying guest commentary [343a]; reviewed inreference 126). Genes of the immunity loci were sequencedand characterized prior to the present work. Analysis of theentire P1 genome allows us to extend this characterization. (a) ImmC. The key repressor of P1 lytic functions is the C1 protein, encoded by c1 within ImmC (84, 238). Inactivation of the C1 protein or a decrease in C1 synthesis triggers lyticgrowth of the phage. A commonly used allele of c1,c1-100, encodes a thermosensitive C1 protein (261). In both sequencedstrains, c1-100 contained two mutations, T569C and G577T, as compared to what has been taken as wild-type c1 (238). The second of these mutations, which causes the amino acid sub-stitution G193C in C1, probably confers thermosensitivity to the C1-100 protein. The thermolabile C1-100 protein of P1 andthe almost identical thermostable C1 protein of P7 do notdiffer at the locus of the T569C substitution (238). C1 protein exists as a monomer in solution and binds to a score of widely dispersed operators (27, 59, 84, 193). The 17-bpoperator consensus sequence is asymmetric and hence hasdirectionality. Monovalent operators have a single repressor-binding site, whereas bivalent operators consist of two over-lapping repressor-binding sites oriented in opposite directions and forming an incomplete palindrome (129) (Table 6). Transcription of the c1 gene itself is autorepressed at oper- ators that precede c1 and are part of ImmC (317, 128). One of these operators, O coi, additionally represses transcription of three small ORFs preceding c1 (27, 127). The product of the longest ORF, the 7.7-kDa Coi protein, forms a 1:1 complexwith C1 and blocks its ability to bind to operators (25, 127,130). A combination of negative effects of C1 on the synthesisof Coi and of Coi on the activity of C1 creates a sensitiveregulatory system that is crucial for a choice between lysis andlysogeny. If C1 synthesis prevails in P1-infected cells, Coi syn-thesis is shut down, leading to the establishment of lysogeny. IfCoi synthesis prevails, C1 becomes inactivated, leading to lyticgrowth. The E. coli SOS response regulator LexA may be a major factor in this epigenetic switch. A LexA binding site identi ﬁed previously in the P1 genome in the region upstream of theimcAB and coigenes (200) overlaps a predicted strong pro- moter that we designated P2 coi(Tables 4 and 7; reference 343a, Fig. 1). Most likely, this promoter can drive transcriptionofimcAB andcoifollowing inactivation of LexA. In support of this view, inducibility of P1 lysogens by UV light has beenrepeatedly reported, although not always reproduced (re-viewed in reference 347).TABLE 6.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  IfCoi synthesis prevails, C1 becomes inactivated, leading to lyticgrowth. The E. coli SOS response regulator LexA may be a major factor in this epigenetic switch. A LexA binding site identi ﬁed previously in the P1 genome in the region upstream of theimcAB and coigenes (200) overlaps a predicted strong pro- moter that we designated P2 coi(Tables 4 and 7; reference 343a, Fig. 1). Most likely, this promoter can drive transcriptionofimcAB andcoifollowing inactivation of LexA. In support of this view, inducibility of P1 lysogens by UV light has beenrepeatedly reported, although not always reproduced (re-viewed in reference 347).TABLE 6. C1-controlled operators of P1 Operatora (former name[s])PositionbStrand DNA sequencecGenes controlledEvidence for: Reference(s)C1 bindingRegulatory function Oref(Op1, Op2a) 1962 /H11001 ATTGCTCTAATTGATTC ref /H12011/H12011 84, 337 Omat (Op2, Op2b) 2628 /H11001 CATGCACTAATAAATAT mat /H12011/H12011 84 Ossb(Op21) 19816 /H11001 ATTGCTCTAATTAATTT ssb,isaA /H12011/H12011 59 Oc4(Op51) 46941 /H11001 AATGCTCTAATAAATTT c4,icd,ant /H12011/H12011 26, 84, 124, 316 OkilA (Op53) 48611 /H11001 TTTGCTCTAATAAATTT kilA,repL /H12011 27, 84, 316 Orlf(Op55) 50376 /H11001 ATTGCTCTAATTATAAC rlfA,rlfB /H12011 112; reviewed in reference 346 Odmt (Op68) 63946 /H11001 ATTGCTCTAATAAATAT uhr,hrdC ,dmt /H12011/H12011 59 Obanab (Op72, Op72ab) 67468 /H11001//H11002 trnA,trnT,tciA,tciB,tciC, trnI,ban,dbn/H12011/H12011 84, 129, 315, 317 a 67468 /H11001 ATTGCTCTAATAAATTT /H12011 132, 316 b 67479 /H11002 ATTACACTAATAAATTT /H12011 O26(Op82) 77329 /H11002 ATTGCACTAATAATTTT 26,25,7,24,6,5 /H12011 Reviewed in 346 Oppp(Op83ab) 78236 /H11002 CATGCACTAATTAATTT pmgN ,pmgO ,pmgP ,ppp, pmgQ/H12011 Reviewed in reference 346 OpmgR a (Op86a) 80953 /H11001 ATTGCTCTAATAATTCC pmgR ,pmgS /H12011 315, 346 OpmgR b (Op86, Op86b) 81002 /H11002 ATTGCTCTAATAAAAAA pmgR ,pmgS /H12011 84, 316 OpmgT (Op88) 83094 /H11001 ATTGCTCTAATAAATCT pmgT ,pmgU ,pmgV /H12011/H12011 59 Olpa(Op94) 88705 /H11001 ATTGCTCTAATGTATTG lpa /H12011/H12011 193 Oclab/H9011(Op99ab, Op99a) 92967 /H11002//H11001 c1 /H12011/H12011 27, 84, 128, 316, 317 a 92978 /H11002 AATGCACTAATAAATCT /H12011 84, 316 bd92967 /H11001 ACGAAAATAATAGATTT /H12011 317 Oc1c/H9011(Op99c) 93150 /H11002 ATTGCTCTAACGCTTTA c1 /H12011/H12011 84, 128 Ocoi/H9011(Op99d) 93442 /H11002 ATTGCTCTAATAAATTA c1,coi /H12011/H12011 27, 128, 316 Ocre/H9011(Op99e) 93525 /H11001 ATTGCTCTAATAATTCT cra,cre /H12011 27, 128, 316 Consensus ATTGCTCTAATAAATTT aThe name of each operator refers to the name of a selected gene under its control.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  7 P1ban 67469 /H11001TTTGGGTATA TTGCTC TAATAAATTTATTAGTG TAATAT CGCCTCAATG /H12011/H11569 50.3 209 P1upl 69024 /H11002GAATTTCTTT TTGAGA ATGCCGAGCATTTAT TAACCT CATTACTGGT 56.2 P2upl 69055 /H11002TTCCATCTTC TTGACT CCGCCAGCCGCTTTGCG GAATTT CTTTTTGAGA 51.5 P2dbn 70823 /H11001GTTTTTAGCC TTGAGA TGCCGAGCCACCAGCT GATGAT GCGCTCACTG 53.8 P1dbn 71207 /H11001AATCGATCGC TTGAAC AGCGTGCGGACAAACGACCGG TAAATT CAGATTTACG 52.1 P2pmgL 77306 /H11001TAGTCATTTG TTGAAT ATTTAACTCAATAAAA GAAAAT TATTAGTGCA 56.8 P1pmgL 77339 /H11001AAAATTATTA GTGCAA TTTTGATTGTGAAATG TATCAT TCTGCCCTTA 56.8 P26 77617 /H11002AACATAATCA GTCAAA TCCTGCCGGTCGCATGC CATATT TACGCTGTTT 50.3 PpmgN 78197 /H11001TGAGCATGTA TTGAAC GGCGATAAGTGGTTG TATGCT GCTTAAAATA 49.1 Continued on following page7042 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  promoters are associated with genes expressed during lysog- eny: three with cre, three with resormod (including the P1 lxc promoter, which can also drive transcription of lxc), one with parB , one with repA , and one with c1. A relatively high pre- dicted strength of known and putative promoters of genesexpressed during P1 lytic development implies that as soon asthese promoters become accessible to the host RNA polymer-ase, they can effectively compete with E. coli promoters for the enzyme. Multiple promoters within regulatory regions of sev-eral operons indicate a requirement for different controls onthe expression of genes at different stages of phage develop-ment or plasmid maintenance or a requirement to adapt toalternative hosts or environmental conditions. Ten promoters, including a promoter of the essential phage replication gene repL , contain the so-called “extended /H1100210 region ”characterized by the presence of T and G at positions /H1100215 and /H1100214, respectively. In E. coli , transcription from ex- tended /H1100210 promoters can initiate in the absence of typical /H1100235 hexamers (178). Sixteen operons end in sequences characteristic of typical Rho-independent transcriptional terminators (Fig. 1 and Table5). Either termination of some operons of P1 occurs by aRho-dependent mechanism or certain Rho-independent tran-scription termination sites are too weak to be detected. Per-haps in the latter case a terminator further downstream isalternatively used. In the case of certain P1 operons expressedduring lysogeny (for instance, resand phd doc pdcA pdcB ), C1repressor-binding sites of downstream operons may possibly be used as roadblocks to transcript elongation. Five predicted Rho-independent transcription terminators (Tc8, T ref,TddrA ,T p lydC , and T dmt) follow closely upon apparently functional promoters. The locations of those termi-nators (in a leader sequence or early in a gene) suggest controlby antitermination. Indeed, the T c8 terminator was shown to participate in the regulation of transcription of a gene down-stream of c8,ref, by prematurely terminating transcription from one of the refpromoters (338). Of the ﬁve terminators, four are positioned in such a way that, in addition to prematuretermination of ref,mat,ddrB , and dmt genes, they can function to terminate transcription of operons immediately upstream.It is likely that P1 encodes an antitermination mechanism.Whereas no P1 protein has been implicated in the process,antitermination might not require such a protein (330). As many as 12 predicted Rho-independent terminators (Tasref, Tas pro, Tas lyz, Tas ddrB , Tas ddrA , Tas kilA, Tas 23, Tas tciA, Tas7, Tas pmgM , Tas pmgT , and Tas doc) are at the beginning of operons but opposite in orientation to the operon. This ar-rangement suggests that these terminators could terminatetranscription of regulatory antisense RNAs. Antisense RNAhas been shown previously to regulate the expression of icd ant1/2 and kilA repL (125, 126). The location of the predicted TaskilA terminator is consistent with that expected from the length of the antisense RNA transcribed from the Pas kilA promoter (125).', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  As many as 12 predicted Rho-independent terminators (Tasref, Tas pro, Tas lyz, Tas ddrB , Tas ddrA , Tas kilA, Tas 23, Tas tciA, Tas7, Tas pmgM , Tas pmgT , and Tas doc) are at the beginning of operons but opposite in orientation to the operon. This ar-rangement suggests that these terminators could terminatetranscription of regulatory antisense RNAs. Antisense RNAhas been shown previously to regulate the expression of icd ant1/2 and kilA repL (125, 126). The location of the predicted TaskilA terminator is consistent with that expected from the length of the antisense RNA transcribed from the Pas kilA promoter (125). The antisense RNA terminated at Tas 23mayTABLE 4 —Continued Name (former name)PositionaStrandSequenceb Basis for identi ﬁcationc Reference(s) /H1100235 /H1100210 tsp pp cp PpmgR 80954 /H11001CAGAATAATA TTGCTC TAATAATTCCATATTTT TAAAAC GTGATGTACA 52.7 P1pmgT 82778 /H11001ACTCCAGTTA TTGATA TGGTGCGCCAGTTATT TAACGT TTACACGGTC 56.8 P2pmgT 83095 /H11001TCAACAATAA TTGCTC TAATAAATCTTGATTTT TAAAAC AGAGAAAGTG 52.7 P3pmgT 83110 /H11001CTAATAAATC TTGATT TTTAAAACAGAGAAAGT GAAAAT AAAAACATGC /H12011 66.3 59 P1upfM 84752 /H11001CAGAAAATGA TTGTTT CCACATCAAGGAGATTT TAATGT TTCACTGAAA 55.6 P2upfM 84785 /H11001TAATGTTTCA CTGAAA CATTAAGTAAGCCAGTG CATAAT TCCATTTTTT 55.6 PupfN 84973 /H11001CTTGCTCACG TTGACG TAGAAACCCAACCCTTA TATAGT TGGATTCGGT 59.8 P1lxr 86267 /H11002TCAACAACCG CTGATG ATTTTGTGCGCTTTGC TACTAT TCATCACCAA 49.7 P2lxr 86347 /H11002CCATAGCGTT TTCACC TCAATAATACTGTTCATTTA TACAGT ACACATTAAA 48.5 P1humD 86318 /H11001ACGTAACGCG TTTGCA CCAAAGGTGTCTCTT TAATGT GTACTGTATA /H1102140.0 200 Pphd 86787 /H11001TGCAAAGTGC TGGTGC TTTATGCCTGTGAAGTT TATAAT TGTGTACACA /H1201152.1 210 Plpa (Pr94)88719 /H11001CTCTAATGTA TTGCTA TTTCTTTAATCGAGGG TATTAT ATTCCACGTT /H12011/H12011 57.4/H11569193 PpacB 90384 /H11001GAAGAAATTG CCGATA TCGTCGATACAGGTGGT TATGGT GATGTCGATG 50.3 Pc1(P99a) 92951 /H11002CACTAATAAA TCTATT ATTTTCGTTG GATCCTTC TATAAT GGTGGCCAAC /H12011/H12011 46.7/H1156984 P1coi (P99d)93438 /H11002TCATAGTTGT TTGACA ATTGCTCTAATAAATTAT AGTTTT GCCGCCGTTT /H12011/H12011 60.9/H11569127 P4cre (P99e)93541 /H11001CTAATAATTC TTGATT TTTATGCGCAGCTGGACG TAAACT CCTCTTCAGA /H12011/H12011 63.9/H1156984 aCoordinate of the ﬁrst nucleotide of the /H1100235 sequence for promoters from which transcription proceeds on the /H11001strand (clockwise) or of the last nucleotide of the/H1100210 sequence for promoters from which transcription proceeds on the /H11002strand (counterclockwise).', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Coordinates refer to positions of promoters in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5and the associated 4-bp duplication. The locations of P3 res,P 1mod,P 1lxc,P 2lxc, and P3 lxcpromoters proposed in reference 285 are corrected. bConserved /H1100210 and /H1100235 promoter hexamers as well as the “extended /H1100210”motifs (T and G at positions /H1100215 and /H1100214, respectively) and known transcription start sites are underlined. The 5 /H11032-GATC sequences that are targets for Dam- or Dmt-mediated N-6 adenine methylation are in boldface. cIdenti ﬁcation of promoters is based on results of transcription start point determination ( tsp) or RNA polymerase binding assay ( tsp/H11569), detection with the promoter probe vector ( pp), or similarity of the sequence to the consensus sequence for /H926870promoters calculated with the use of the program Targsearch ( cp) (229). References that indicate existence of a promoter in a given region based on S1 nuclease mapping of 5 /H11032ends of transcripts are marked with an S1 in the tspcolumn. Putative promoters which have not been veri ﬁed experimentally are shown only if their homology to the consensus exceeds 50%, as calculated by the method of Hawley and McClure (118) and if their location suggests a function in initiation of transcription of genes or regulatory RNA. Exceptionally, putative promoter s that have a homology score to the consensus below 50% but overlap putative regulatory sites are also shown. For comparison, homology scores are shown for both known and put ative promoters, the former marked with an asterisk.VOL. 186, 2004 BACTERIOPHAGE P1 7043', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  regulate translation of the 23gene from an alternative, up- stream start site. Genes expressed during establishment or maintenance of P1 prophage. The alternative life-styles of P1 require two sets of genes with little functional overlap. In general, the genesassociated with lysogeny can be distinguished by the absence ofregulatory regions resembling those known to control P1 lyticfunctions, the C1-controlled operators or the Lpa-activatedpromoters, which will be described later in this paper. Of 45 P1 operons, 14 are not directly controlled by C1 or Lpa (Fig. 1). Their expression is probably not up-regulatedduring lytic growth, other than by gene dosage, an effect that isdamped in those operons that are subject to autoregulation(ImmC and ImmI genes, repA ,parAB ,phd doc pdcAB ,cin, and probably others). As many as 20 genes of operons that appear to be indepen- dent of C1 and Lpa are either known or predicted to beassociated with lysogeny (Fig. 1 and Table 3; see also Table S1in the supplemental material). We are unable to predict the function of 12 other genes in this class ( isaB,upfA ,mlp,ppfA , upfB ,upfC ,plp,upl,upfM, upfN ,upfO , and upfQ ). However, three of them, mlp,ppfA and plp, encode products that have putative periplasmic transport or lipoprotein attachment sig-nals likely to contribute to lysogenic conversion. Only six structural genes are essential for stable inheritance of P1 as a prophage. Other genes associated with lysogeny mayprotect the lysogen from entry of foreign DNA ( resand mod; reviewed in references 32 and 38), from infection by anotherP1 or homologous phages ( sim, superinfection exclusion) (170), and from DNA damage ( humD , homolog of umuD ) (217). Possibly for the simproduct(s) to be effective, the gene(s) must be ampli ﬁed, as superimmunity was observed only with high- copy-number clones of sim. The C-segment inversion recom- binase cinis expressed by the prophage, but the known bene ﬁt it confers is to produce phages that differ in host speci ﬁcity from their parent.TABLE 5. Known and putative Rho-independent transcriptional terminators in P1 DNA NameaPositionbStrand SequencecRelative locationd Basis for identi ﬁcatione Tasref 1836 /H11002 TTTGTGCAGCCTGGCTCCTTGCCAGGCTTTTTTTTATTTCATCATGG 5/H11032-c8-3/H11032\\uf8ee5/H11032-ref-3/H11032 TT, TR Tc8 2046 /H11001 TCCTTGACATCATTGGCGGCCATTAGGCCGCCTTTTTTTTGCCATAT 5/H11032-c8-3/H11032\\uf8f95/H11032-ref-3/H11032 TT, TR Tref 2788 /H11001 CTTCAGTTGTATTGCTAAGCCGCCGCTGGTGGCTTTTCTTTTTTGTA 5/H11032-mat\\uf8f9mat-3/H11032 TR (338) Taspro 17761 /H11002 CCGATTCAAGGATTTGCGCCAGTTCCTGTGGCGGTGTTTTGATGATG 5/H11032-prt\\uf8eeprt-3/H11032 TR Tpro 18848 /H11001 CCACCAAAAATAACCCCGGCAGCTGCCGGGGTTCTCGTTAACTATTA 5/H11032-pro-3/H11032\\uf8f93/H11032-lyz-5/H11032 TR Tlyz 18848 /H11002 ATAGTTAACGAGAACCCCGGCAGCTGCCGGGGTTATTTTTGGTGGTT 5/H11032-pro-3/H11032\\uf8ee3/H11032-lyz-5/H11032 TT,', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TR Taslyz 19655 /H11001 GATAGCGCAAATTGCACCGCCTCCTGCGGCTGTTTTTCCCTTCATAA 3/H11032-lyz\\uf8f9lyz-5/H11032 TR TinsB 21351 /H11002 TTTCGTATTATCAGGGTGCTCTCTGGCACCCCTTTTTATTCTAGTAA 5/H11032-isaA-3/H11032\\uf8ee3/H11032-insB-5/H11032 TR TasddrB 26246 /H11001 CACCACATACGAAACCCGCCCGAAATCGGGCAGGTTCTTTAGCAGCT 3/H11032-ddrB\\uf8f9ddrB -5/H11032 TR TddrA 26246 /H11002 TGCTAAAGAACCTGCCCGATTTCGGGCGGGTTTCGTATGTGGTGACA 3/H11032-ddrB\\uf8eeddrB -5/H11032 TT, TR TasddrA 26508 /H11001 ACCAGTTTTGGCACACCATCATGTTCCGAAGGTTGTTTCTCCTCGCC 3/H11032-ddrA\\uf8f9ddrA -5/H11032 TR TplydC 30287 /H11002 ACAATAATGCTCCTCTCCATCCTTGGCGGGGTGCATTCGTTTCTGAA 3/H11032-lydC\\uf8eelydC -5/H11032 TR TU/H11032 33125 /H11001 AACCTGAATGAGACAAGGCCCGATAGCGGGCCTTAATTTTTATTCAG 5/H11032-U/H11032-3/H11032\\uf8f93/H11032-U-5/H11032 TT, TR T2simC 45384 /H11002 ACTCGTAGAATCGGTTAACACACCAGATTCTACGAGGTTTCAATGAC 3/H11032-pmgC -5/H11032\\uf8ee3/H11032-simC -5/H11032 TT T1simC 45535 /H11002 GTTTTAACACAACGCCCATTAAAGGGCGTTTTATTGTTTTACTCAAA 3/H11032-pmgC -5/H11032\\uf8ee3/H11032-simC -5/H11032 TT, TR TaskilA 48546 /H11002 CTGCTCAGAAACCTGCCAGTTTGCTGGCAGGTTTTTTTCTTTTGTTA 5/H11032-ant-3/H11032\\uf8ee5/H11032-kilA-3/H11032 TR TpmgG 52219 /H11002 CATCCCTTGTATTTTAAAGCTCCTTCGGGAGCTTTTTTGTGCTTAAA 3/H11032-pmgG -5/H11032\\uf8ee3/H11032-pmgF -5/H11032 TT, TR T22 53368 /H11002 GGCTAAAGGAGGAGGCCGGGGAACTCCGGCCTTTAACTTGAATGGCT 3/H11032-21-5/H11032\\uf8ee3/H11032-22-5/H11032 TT, TR Tas23 57399 /H11001 CATTATCAGAGATACCCGCAAAAACCGGGTCTTTACGTGCAGCTTCA 3/H11032-23\\uf8f923-5/H11032 TR TparB 57707 /H11002 TTTCAAGGTGAAATCGCCACGATTTCACCTTGGATTTTACCTTCCTC 3/H11032-23-5/H11032\\uf8ee3/H11032-parB -5/H11032 TT, TR TrepA 60474 /H11002 TTTGATGTGCGCTGGAGGGGGACGCCCCTCAGTTTGCCCAGACTTTC 3/H11032-parA -5/H11032\\uf8ee3/H11032-repA -5/H11032 TT TupfA 61718 /H11002 AATACGACTCCCTTCCAACCGGCTACGTTGGCCGGTTTTTCACTTAT 3/H11032-repA -5/H11032\\uf8ee3/H11032-upfA -5/H11032 TT,', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TR TrepA 60474 /H11002 TTTGATGTGCGCTGGAGGGGGACGCCCCTCAGTTTGCCCAGACTTTC 3/H11032-parA -5/H11032\\uf8ee3/H11032-repA -5/H11032 TT TupfA 61718 /H11002 AATACGACTCCCTTCCAACCGGCTACGTTGGCCGGTTTTTCACTTAT 3/H11032-repA -5/H11032\\uf8ee3/H11032-upfA -5/H11032 TT, TR ThrdC 67115 /H11001 TGGAAGCCTTCCTCGGTGCGGAAGGGCGGTCTGAGTTTGTATGCGGT 5/H11032-dmt\\uf8f9dmt-3/H11032 TT Tdmt 67659 /H11001 GAGCTTCCAGTTTGCCCATCTTCGGGTGGGCGTTTTTTCAGGGTTTT 5/H11032-dmt-3/H11032\\uf8f95/H11032-trnA-3/H11032 TR TastciA 69059 /H11002 TCTTCTTGACTCCGCCAGCCGCTTTGCGGAATTTCTTTTTGAGAATG 5/H11032-tciA\\uf8f9tciA-3/H11032 TR Tdbn 72485 /H11001 CCGTCATAACAGAAAAGCCCGAACGCCGGGCTTTTCTTAAGCCTTGT 5/H11032-dbn-3/H11032\\uf8f93/H11032-5-5/H11032 TT, TR T5 72485 /H11002 TGACAAGGCTTAAGAAAAGCCCGGCGTTCGGGCTTTTCTGTTATGAC 5/H11032-dbn-3/H11032\\uf8ee3/H11032-5-5/H11032 TT, TR Tas7 75768 /H11001 ACTTTATGAGGTTGTTCGCCTGCGTTTGCAGGGCTTTTTTGAGAGCG 3/H11032-25\\uf8f925-5/H11032 TR TaspmgM 77765 /H11002 GTGCTGCAAGTGCCTTCGCATGTGGAGAAACTTCTTTTACCGGCGCA 5/H11032-pmgM\\uf8eepmgM -3/H11032 TR TpmgQ 80982 /H11001 TTTTAAAACGTGATGTACACTCATCACGTTTTTTATTAGAGCAATCT 5/H11032-pmgQ -3/H11032\\uf8f95/H11032-pmgR -3/H11032 TT TpmgS 82537 /H11001 GTCACTTTTAATGCTGGTGGAGTGCGCCCACCAGCATTTTTTTCGTC 5/H11032-pmgS -3/H11032\\uf8f95/H11032-pap-3/H11032 TT Tpap 83146 /H11001 AGTGAAAATAAAAACATGCCGCAAGGCGCGGCATGTTTCCAATCAAT 5/H11032-pap-3/H11032\\uf8f95/H11032-pmgT -3/H11032 TT,', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TR TaspmgT 83144 /H11002 GATTGGAAACATGCCGCGCCTTGCGGCATGTTTTTATTTTCACTTTC 5/H11032-pap-3/H11032\\uf8ee5/H11032-pmgT -3/H11032 TR Tasdoc 87120 /H11002 TCAGACATTCCCGGCAGGCCGCCGTAGCGGCTTATATTCGCATCATG 5/H11032-doc\\uf8eedoc-3/H11032 TR Tlpa 89255 /H11001 ATTGATTGCTTGCCCGTTCCGGGCCTTTTGACATGTGACTTTCGTTA 5/H11032-lpa-3/H11032\\uf8f95/H11032-pacA -3/H11032 TT TpacB 92399 /H11001 TTGTTTCGCTAAAGCCGTGTACGCGCATACGGAATTTTTCATCCTCC 3/H11032-c1\\uf8f9c1-5/H11032 TR aNames of the terminators (T) that could terminate transcription of RNA antisense to mRNA of a given gene contain “as”and the name of the gene. The terminator inlydC is marked Tp lydC to indicate that it could serve primarily to terminate lydC transcription prematurely. bCoordinate of the ﬁrst nucleotide of the terminator hairpin for terminators that terminate transcription on the /H11001strand (clockwise) or the last nucleotide of the terminator hairpin for terminators that terminate transcription on the /H11002strand (counterclockwise). Coordinates refer to positions of terminators in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. cNucleotides that form the stem of the terminator hairpin are underlined. dTerminators are represented schematically by symbols \\uf8f9and\\uf8ee, which face toward the DNA region whose transcription they terminate. eIdenti ﬁcation of terminators is based on their similarity to known transcription terminators as determined by the program TransTerm (TT) or Terminator (TR) and veri ﬁed by additional criteria (see Materials and Methods).7044 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  (i) Immunity. Whether P1 is maintained in cells as a plasmid or enters the lytic pathway is dictated by the interplay of en-vironmental factors with the components of the immunity cir-cuit encoded at loci designated ImmC, ImmI, and ImmT (Fig.1 of the accompanying guest commentary [343a]; reviewed inreference 126). Genes of the immunity loci were sequencedand characterized prior to the present work. Analysis of theentire P1 genome allows us to extend this characterization. (a) ImmC. The key repressor of P1 lytic functions is the C1 protein, encoded by c1 within ImmC (84, 238). Inactivation of the C1 protein or a decrease in C1 synthesis triggers lyticgrowth of the phage. A commonly used allele of c1,c1-100, encodes a thermosensitive C1 protein (261). In both sequencedstrains, c1-100 contained two mutations, T569C and G577T, as compared to what has been taken as wild-type c1 (238). The second of these mutations, which causes the amino acid sub-stitution G193C in C1, probably confers thermosensitivity to the C1-100 protein. The thermolabile C1-100 protein of P1 andthe almost identical thermostable C1 protein of P7 do notdiffer at the locus of the T569C substitution (238). C1 protein exists as a monomer in solution and binds to a score of widely dispersed operators (27, 59, 84, 193). The 17-bpoperator consensus sequence is asymmetric and hence hasdirectionality. Monovalent operators have a single repressor-binding site, whereas bivalent operators consist of two over-lapping repressor-binding sites oriented in opposite directions and forming an incomplete palindrome (129) (Table 6). Transcription of the c1 gene itself is autorepressed at oper- ators that precede c1 and are part of ImmC (317, 128). One of these operators, O coi, additionally represses transcription of three small ORFs preceding c1 (27, 127). The product of the longest ORF, the 7.7-kDa Coi protein, forms a 1:1 complexwith C1 and blocks its ability to bind to operators (25, 127,130). A combination of negative effects of C1 on the synthesisof Coi and of Coi on the activity of C1 creates a sensitiveregulatory system that is crucial for a choice between lysis andlysogeny. If C1 synthesis prevails in P1-infected cells, Coi syn-thesis is shut down, leading to the establishment of lysogeny. IfCoi synthesis prevails, C1 becomes inactivated, leading to lyticgrowth. The E. coli SOS response regulator LexA may be a major factor in this epigenetic switch. A LexA binding site identi ﬁed previously in the P1 genome in the region upstream of theimcAB and coigenes (200) overlaps a predicted strong pro- moter that we designated P2 coi(Tables 4 and 7; reference 343a, Fig. 1). Most likely, this promoter can drive transcriptionofimcAB andcoifollowing inactivation of LexA. In support of this view, inducibility of P1 lysogens by UV light has beenrepeatedly reported, although not always reproduced (re-viewed in reference 347).TABLE 6.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  IfCoi synthesis prevails, C1 becomes inactivated, leading to lyticgrowth. The E. coli SOS response regulator LexA may be a major factor in this epigenetic switch. A LexA binding site identi ﬁed previously in the P1 genome in the region upstream of theimcAB and coigenes (200) overlaps a predicted strong pro- moter that we designated P2 coi(Tables 4 and 7; reference 343a, Fig. 1). Most likely, this promoter can drive transcriptionofimcAB andcoifollowing inactivation of LexA. In support of this view, inducibility of P1 lysogens by UV light has beenrepeatedly reported, although not always reproduced (re-viewed in reference 347).TABLE 6. C1-controlled operators of P1 Operatora (former name[s])PositionbStrand DNA sequencecGenes controlledEvidence for: Reference(s)C1 bindingRegulatory function Oref(Op1, Op2a) 1962 /H11001 ATTGCTCTAATTGATTC ref /H12011/H12011 84, 337 Omat (Op2, Op2b) 2628 /H11001 CATGCACTAATAAATAT mat /H12011/H12011 84 Ossb(Op21) 19816 /H11001 ATTGCTCTAATTAATTT ssb,isaA /H12011/H12011 59 Oc4(Op51) 46941 /H11001 AATGCTCTAATAAATTT c4,icd,ant /H12011/H12011 26, 84, 124, 316 OkilA (Op53) 48611 /H11001 TTTGCTCTAATAAATTT kilA,repL /H12011 27, 84, 316 Orlf(Op55) 50376 /H11001 ATTGCTCTAATTATAAC rlfA,rlfB /H12011 112; reviewed in reference 346 Odmt (Op68) 63946 /H11001 ATTGCTCTAATAAATAT uhr,hrdC ,dmt /H12011/H12011 59 Obanab (Op72, Op72ab) 67468 /H11001//H11002 trnA,trnT,tciA,tciB,tciC, trnI,ban,dbn/H12011/H12011 84, 129, 315, 317 a 67468 /H11001 ATTGCTCTAATAAATTT /H12011 132, 316 b 67479 /H11002 ATTACACTAATAAATTT /H12011 O26(Op82) 77329 /H11002 ATTGCACTAATAATTTT 26,25,7,24,6,5 /H12011 Reviewed in 346 Oppp(Op83ab) 78236 /H11002 CATGCACTAATTAATTT pmgN ,pmgO ,pmgP ,ppp, pmgQ/H12011 Reviewed in reference 346 OpmgR a (Op86a) 80953 /H11001 ATTGCTCTAATAATTCC pmgR ,pmgS /H12011 315, 346 OpmgR b (Op86, Op86b) 81002 /H11002 ATTGCTCTAATAAAAAA pmgR ,pmgS /H12011 84, 316 OpmgT (Op88) 83094 /H11001 ATTGCTCTAATAAATCT pmgT ,pmgU ,pmgV /H12011/H12011 59 Olpa(Op94) 88705 /H11001 ATTGCTCTAATGTATTG lpa /H12011/H12011 193 Oclab/H9011(Op99ab, Op99a) 92967 /H11002//H11001 c1 /H12011/H12011 27, 84, 128, 316, 317 a 92978 /H11002 AATGCACTAATAAATCT /H12011 84, 316 bd92967 /H11001 ACGAAAATAATAGATTT /H12011 317 Oc1c/H9011(Op99c) 93150 /H11002 ATTGCTCTAACGCTTTA c1 /H12011/H12011 84, 128 Ocoi/H9011(Op99d) 93442 /H11002 ATTGCTCTAATAAATTA c1,coi /H12011/H12011 27, 128, 316 Ocre/H9011(Op99e) 93525 /H11001 ATTGCTCTAATAATTCT cra,cre /H12011 27, 128, 316 Consensus ATTGCTCTAATAAATTT aThe name of each operator refers to the name of a selected gene under its control.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:37 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'83'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994467'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'331ms'), (b'x-request-id', b'807aa89b9b80c5d7f05958dfb01eee8f'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412490a49ae5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:37 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'83'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994467'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'331ms'), (b'x-request-id', b'807aa89b9b80c5d7f05958dfb01eee8f'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412490a49ae5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Bivalent operators names are underlined; constituent monovalent ope rators are listed below the name of each bivalent operator. Operators with a superscript /H9011have been shown to engage in looping to nearby operators in vitro. bPosition of the ﬁrst nucleotide if the operator is assigned to the /H11001strand or to the ﬁrst nucleotide of its complement if the operator is assigned to the /H11002strand. Coordinates refer to positions of operators in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. cThe underlined nucleotides are conserved in monovalent operators and in at least one constituent operator of each bivalent operator. dThe sequence of O c1b corresponds to that identi ﬁed by DNase protection experiments (317) and differs from the sequence proposed earlier (84).VOL. 186, 2004 BACTERIOPHAGE P1 7045', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TABLE 7. Known and potential binding sites in P1 DNA for E. coli proteins IHF, DnaA, and LexA CoordinatesaStrand SequenceSimilarity scorebKnown or potential function or relative location (reference[s]) IHF 1478 –1526 /H11002 TATGTCACCATAAATATCAAATAATTATA 52.0 c8 regulation 6985 –7132 /H11002 TTATATGCATACAAAAAGATGAAGTTATA 53.9 Internal to mod 19809 –19857 /H11002 ATCTTACCTAATAAAATAAATTAATTAGA 55.7 ssbregulation, lyzregulation 21424 –21472 /H11002 TGAACATAAAACACTATCAATAAGTTTGA 50.1 Overlaps terminal inverted repeat of defective IS 1 (93, 94) 30319 –30367 /H11002 TTTAATTCACTATTAAGGAAATAGTTATG 53.7 cinregulation 38415 –38463 /H11002 TATAATGTTCAGAATATCAATAAGATATT 51.8 16regulation 38419 –38466 /H11001 AATATCTTATTGATATTCTGAACATTATA 51.4 16regulation 46372 –46419 /H11001 ATTAATCGGTTTAAACTCACTAATTAATT 51.4 simregulation 46498 –46545 /H11002 TGGTTGATTCTTATAATCAAAAAACTATT 50.3 simregulation 49930 –49977 /H11002 AAAAAATTAAATAAAGCCAATGTCTTAGC 54.2 Internal to repL 50647 –50694 /H11001 AAAATTGCTAGAGACATCAAGGAATTAAA 57.0 rlfBregulation or part of alternative replication origin 50672 –50719 /H11001 TAAAATTAAAACACAATCTAAAACCTGAT 50.5 rlfBregulation or part of alternative replication origin 51151 –51198 /H11001 ATAAAATCACATACAAACAGGGAGTTACT 58.7 rlfBregulation or part of alternative replication origin 57740 –57787 /H11001 GTAATTTACTTTAAAAACAGTCAGTTAAT 57.2 parS -ParB-IHF centromeric complex formation (91) 61404 –61451 /H11002 GAAAAAGTAATATGAATCAATCATTTATC 57.3 repA regulation (reviewed in reference 49) 89562 –89610 /H11001 CTGAAATAGCGGAAAAA CAAAGAGTTAATc40.8 Involved in PacA- and PacB-mediated cleavage at pacsite (287) DnaA 1076 –1090 /H11001 TAGTTACCCCCAGGC 54.3 Internal to cre 5894 –5908 /H11001 GTGTTAATAACAAGA 56.0 Internal to res 5892 –5906 /H11002 TTGTTATTAACACGG 55.1 Internal to res 6716 –6730 /H11001 GTGTTGTTAACAAGC 54.3 Internal to mod 9564 –9578 /H11002 GAGTTATCATCAATA 57.7 Internal to darB 41563 –41577 /H11001 AGGTTATCGACAGTA 55.2 Internal to sit 47322 –47336 /H11002 GTGTTTTTTACAGGA 54.3 Internal to icd 47945 –47959 /H11001 GAATTTTACACAAAA 54.3 Internal to ant1/2 49965 –49979 /H11001 TTTTTATCCACACCC 56.9 Internal to repL , presumably part of oriL 50148 –50162 /H11001 CGGTTATCCAAAAAA 56.0 Internal to repL , presumably part of oriL 50313 –50327 /H11001 GTTTTAGCCAAAAAT 54.4 Internal to repL , presumably part of oriL 50986 –51000 /H11001 GGGTTCTTCACAATG 54.3 Internal to rlfA, overlaps predicted promoter of rlfB, presumably regulates expression of rlfB 52202 –52216 /H11001 GATTTAAGCACAAAA 56.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  3 Internal to icd 47945 –47959 /H11001 GAATTTTACACAAAA 54.3 Internal to ant1/2 49965 –49979 /H11001 TTTTTATCCACACCC 56.9 Internal to repL , presumably part of oriL 50148 –50162 /H11001 CGGTTATCCAAAAAA 56.0 Internal to repL , presumably part of oriL 50313 –50327 /H11001 GTTTTAGCCAAAAAT 54.4 Internal to repL , presumably part of oriL 50986 –51000 /H11001 GGGTTCTTCACAATG 54.3 Internal to rlfA, overlaps predicted promoter of rlfB, presumably regulates expression of rlfB 52202 –52216 /H11001 GATTTAAGCACAAAA 56.9 Internal to pmgF 61448 –61462 /H11002 AAACTATCCACACAA 51.8 Replication initiator complex formation at oriR (reviewed in reference 51) 61457 –61471 /H11002 CACTTATAAAAACTA /H1102150.0 Replication initiator complex formation at oriR (reviewed in reference 51) 61467 –61481 /H11002 TAGTTATCACCATTA 51.9 Replication initiator complex formation at oriR (reviewed in reference 51) 61690 –61703 /H11002 ACATTATCCACTGGA 51.8 Replication initiator complex formation at oriR (reviewed in reference 51) 61698 –61712 /H11002 CACTTATCCACATTA 56.0 Replication initiator complex formation at oriR (reviewed in reference 51) 68953 –68967 /H11002 TGGTTATCCACCGTG 54.3 Between upland its predicted promoter; regulation of upl 72890 –72904 /H11002 CGGTTATGCACAGGA 56.9 Internal to 5 73755 –73769 /H11001 GGCTTATCAACAAAA 56.9 Internal to 6 80593 –80607 /H11002 TTGGTATCCACAGGG 57.7 Internal to pmgQ 85215 –85229 /H11002 AAGTTATGCACTTGC /H1102150.0 hotregulation 85227 –85241 /H11002 GATTTATCCACAAAG 63.5 hotregulation 85851 –85865 /H11002 GAATTTTCCTCAAGG 55.1 Vicinity of P2 humD promoter internal to upfQ; probable humD regulation 89848 –89862 /H11002 GACTTTTCCAGAAGA 56.0 Vicinity of pacsite internal to pacA , packaging regulation LexA 46–65 /H11001//H11002 ACCTGTAAATCCATACAGTT 62.7/59.8 Regulation of ImmC genes 86346 –86365 /H11001//H11002 TACTGTATAAATGAACAGTA 67.7/66.1 humD regulation and possibly lxrregulation aThe ﬁrst number is the coordinate of the ﬁrst nucleotide of upper-strand ( /H11001) binding sites or of the last nucleotide of the lower-strand ( /H11002) binding sites, based on arbitrary convention. Coordinates refer to positions of sites in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. bOf predicted IHF, DnaA, or LexA binding sites, only those whose similarity score to the consensus exceeded 50.0, 54.0, or 60.0, respectively (as calcu lated by the GCG Fitconsensus program) are shown. Exceptions are made for tandem sites. cFor PP1PAC in GenBank, boldface C is a G.7046 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  An additional component of ImmC or a separate immunity locus may be located downstream of cre. This gene, tentatively named c8, is probably the locus ( c8) of a clear plaque mutation (280, 281). Temperature shift experiments with a thermosen-sitive allele of c8 suggested that the gene is probably not ex- pressed in a lysogen and hence is most likely involved uniquelyin the establishment of lysogeny. (b) ImmI. The ImmI region (antipodal to immC in the circular representation of the plasmid genome) contains a C1-controlled operon of three genes: c4,icd, and ant1/2 , which confer a separate speci ﬁcity to immunity. It permits P1 to plate on lysogens of the closely related phage P7, and vice versa,despite functional interchangeability of their C1 repressors(54, 131, 280, 326). The ant1/2 and icdgenes determine an antirepressor protein and an inhibitor of cell division, respec-tively, whereas the c4 gene determines an antisense RNA that acts as a secondary repressor of Icd and antirepressor synthesis(58, 131, 257). Maintenance of lysogeny requires, in addition tothe expression of c1, the expression of c4 to prevent synthesis of the antirepressor protein (reviewed in reference 126). A77-bp antisense RNA encoded by c4 regulates expression of downstream genes in its own message (57; diagrammed in Fig.1 of reference 343a). Two short single-stranded regions, b /H11032and a/H11032, exposed in its cloverleaf-like structure, block translation of icd(and of translationally coupled ant1) by interaction with the complementary regions, a2 and b2, in the c4icd ant1/2 mRNA. The interaction occludes the RBS in front of the icdgene (58). The resulting translational block additionally permits prema-ture termination of anttranscription via a Rho-dependent ter- minator (33). Slight differences between P1 and P7 in theseshort regions are responsible for the heteroimmunity of thetwo phages. An additional speci ﬁcity determinant, sas(site of Ant speci ﬁcity), is involved in determining the capacity of Ant proteins to induce prophage on heteroimmune superinfection.The site resides centrally within the antgenes of P1 and P7 (J. Heinrich and H. Schuster, unpublished results). It was sug-gested that Ant proteins must normally load onto their phage-speci ﬁcsassites to perform their antirepressor function (294). The action of C4 appears limited to the mRNA of its own operon, as we did not ﬁnd any pair of sequences complemen- tary to b /H11032and a/H11032in the P1 genome, other than those designated a1 and b1, located upstream of c4, and presumed to participate in unmasking of a2 and b2 by competitive binding of b /H11032and a/H11032. (c) ImmT. The ImmT region, distant from ImmC and ImmI, contains a small gene, lxc, whose product modulates the func- tion of C1 by formation of a ternary complex with C1 andoperator DNA (272, 317, 318). Participation of Lxc in the C1-operator complexes that control c1 expression increases the afﬁnity of the repressor for the operator and down-regulates the synthesis of C1 itself (318). Lxc can thus relieve repressionat weak operators by decreasing the intracellular concentrationof C1 below that critical for their binding and simultaneouslyenhance repression at strong operators by increasing their af-ﬁnity for C1. Consequently, lxcmutations can have divergent effects on transcription from different C1-controlled promoters(272, 273, 311). Part of the modulating effect of Lxc may bemediated by looping between nearby operators complexedwith C1, as looping is dramatically increased by Lxc (128,343a). As Lxc has been reported to strongly inhibit the abilityof Coi to dissociate operator-C1 complexes (317), it may assist in establishing P1 as a prophage.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Participation of Lxc in the C1-operator complexes that control c1 expression increases the afﬁnity of the repressor for the operator and down-regulates the synthesis of C1 itself (318). Lxc can thus relieve repressionat weak operators by decreasing the intracellular concentrationof C1 below that critical for their binding and simultaneouslyenhance repression at strong operators by increasing their af-ﬁnity for C1. Consequently, lxcmutations can have divergent effects on transcription from different C1-controlled promoters(272, 273, 311). Part of the modulating effect of Lxc may bemediated by looping between nearby operators complexedwith C1, as looping is dramatically increased by Lxc (128,343a). As Lxc has been reported to strongly inhibit the abilityof Coi to dissociate operator-C1 complexes (317), it may assist in establishing P1 as a prophage. The initiation codon of lxcappears to overlap with the ter- mination codon of the preceding ulxgene of unknown func- tion, indicating transcriptional and translational coupling ofthe two genes (Table 3). The coupling of lxcand ulximplies that lxc, in addition to being transcribed from its own promot- ers, may be transcribed from P darB , which presumably drives transcription of ulxand the preceding gene, darB (Table 4). None of the promoters that might drive transcription of lxcis under the control of C1, suggesting that Lxc-mediated modu-lation of C1 function is independent of ImmC and ImmI. (ii) Plasmid maintenance. Despite its low copy number per bacterial origin (0.7 to 1.4) (248), the P1 plasmid is lost with afrequency of only about 10 /H110025per cell per generation (261). Like other plasmids that have accepted the tradeoff of large size for low copy number, P1 must counteract the increasedrisk of loss at cell division. Four functions have been identi ﬁed that allow it to accomplish this task: a regulated plasmid rep-licon, a partitioning mechanism, a site-speci ﬁc recombination system, and a plasmid addiction (postsegregational killing orgrowth inhibition) mechanism. These are discussed below.Analysis of the genome of P1 indicates that whereas the lyticorigin may contribute to plasmid replication (349), there is nosecond addiction or partition module in P1 as there are incertain other plasmids (80, 140). (a) Plasmid replication and partition. The P1 plasmid rep- lication and partition genes are encoded within one region ofP1 DNA, where they form two operons transcribed in the samedirection. The P1 replicon has been intensively studied (re-viewed in references 49 and 51). It belongs to the large familyof plasmid replicons that encode an initiator protein, calledRepA in P1, and contain multiple binding sites for that initi-ator (19-bp iterons in P1). One set of these sites, incC , pre- cedes the repA gene and is part of the replication origin ( oriR). A second set, incA , follows repA and is a regulatory locus (Table 1). Replication from the P1 plasmid origin, oriR, pro- ceeds in both directions (241). P1 plasmid and E. coli chromo- some replication have much in common, including sequestra tion of the origin when its 5 /H11032-GATC sequences are hemimethylated (1) and a requirement for DnaA (Table 7) (113). One differ-ence is that the ADP form of DnaA suf ﬁces for P1 replication, whereas the ATP form is essential for bacterial replication(333). The P1 repA promoter is nested among the incC iterons and consequently is subject to autorepression by RepA (5). Repli-cation control is exerted by incC and the additional iterons of incA in more than one way. By sequestering RepA, the incA iterons can prevent the concentration of RepA at the originfrom attaining the threshold value required for ﬁring (reviewed in reference 49).', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  One differ-ence is that the ADP form of DnaA suf ﬁces for P1 replication, whereas the ATP form is essential for bacterial replication(333). The P1 repA promoter is nested among the incC iterons and consequently is subject to autorepression by RepA (5). Repli-cation control is exerted by incC and the additional iterons of incA in more than one way. By sequestering RepA, the incA iterons can prevent the concentration of RepA at the originfrom attaining the threshold value required for ﬁring (reviewed in reference 49). By becoming “handcuffed ”to the iterons of incC via bound RepA, the additional iterons can interfere with both replisome assembly (239) and the release of autorepres-sion that would otherwise replenish sequestered RepA (50).The availability of RepA for replication is limited not only bysequestration and handcuf ﬁng but also by the formation of inactive RepA dimers which require bacterial chaperones forconversion to active protein monomers (334). We suggest anadditional role of incA in the control of repA expression. The putative transcriptional terminator of repA is located within theVOL. 186, 2004 BACTERIOPHAGE P1 7047', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  incA region (Table 5), and thus occupation of incA iterons by RepA may act as a roadblock to the completion of repA tran- scripts. Abbreviated repA transcripts might undergo more rapid degradation than intact transcripts. The partition module of P1, which is situated downstream of therepA gene and the incA iterons, is a major contributor to stability of the plasmid. It ensures partition of plasmid mole-cules to daughter cells and has been, with its homologs, thesubject of numerous detailed studies and reviews (reviewedmost recently in references 97 and 306). The module consistsof an operon of two genes, parA andparB , which ends at a site, parS , a centromere analog (4). ParA is an ATPase whose ac- tivity is stimulated by ParB (73). Both ATP and ADP forms ofParA act as autorepressors, the ADP form being the moreeffective (70, 71). A stimulation of repression by ParB (87) isassisted by parS (114). The parS site contains two kinds of recognition sequences for ParB, four heptameric and two hex-americ boxes, and a binding site for the host protein IHF(Tables 1 and 7) (249). ParB and IHF form a complex at parS (72, 89) to which ParA, in the ATP form, can bind via ParB (37,90). Binding of ParB to parS can permit parS sites to pair (83) and can nucleate spreading of ParB to the ﬂanking DNA and silence transcription of genes as far as several kilobases away(258). Whether ParA and ParB translocate plasmid moleculesto their target positions within a cell during partitioning bythemselves or attach them to unknown host components ofpartition machinery is unclear. We did not ﬁnd in P1 DNA any additional sequences similar toparS that could serve as potential binding sites for ParB. This suggests that the function of ParB is limited to partition-ing in P1, whereas in certain other plasmids, e.g., RK2 andN15, the homologous proteins appear to have additional func-tions (102, 251, 335). Two putative /H9268 70promoters of P1 gene 23, immediately downstream of parB , of which one overlaps the IHF binding region of parS , appeared weak when tested in vivo in the absence of ParB (259; A. Dobruk and M. Łobocka, unpublished results). The possibility that conditions might existleading to their signi ﬁcant expression and regulation by ParB has not been studied. (b) DNA cyclization and multimer resolution. Faithful par- titioning of newly replicated plasmid molecules is assisted byprior resolution to monomers of plasmid multimers that areinevitably formed in recA /H11001bacteria. The resolution is usually accomplished by a site-speci ﬁc recombination. In P1 it is en- sured by the recombinase Cre, encoded near its site of action,lox(8, 299, 300), far from the genes of replication and partition (Fig. 1). The resolution of plasmid multimers into monomersincreases the number of partitionable P1 molecules and henceplasmid stability (21). The same accessory proteins that arerequired for the resolution of ColE1 dimers by the bacterialrecombinase XerCD are also required for the stable mainte-nance of P1 prophage (244). These proteins (ArgR, for whicha putative binding site to the left of loxhas been located, and PepA) are proposed to constrain the directionality of the re-combination in vivo. The possibility that the lox-cre module might assist plasmid stability in another way has also beensuggested (10). Interplasmid recombination involving a repli-cating plasmid could generate a concatemeric replication sub-strate that, following ampli ﬁcation as a rolling circle, could be resolved into plasmid monomers by Cre-mediated recombina-tion. The plasmid yield per replication initiation event would be increased, providing protection against RepA insuf ﬁciency. The loxsite is the unique target for Cre action in P1 DNA, as we did not ﬁnd any other similar sites.', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  These proteins (ArgR, for whicha putative binding site to the left of loxhas been located, and PepA) are proposed to constrain the directionality of the re-combination in vivo. The possibility that the lox-cre module might assist plasmid stability in another way has also beensuggested (10). Interplasmid recombination involving a repli-cating plasmid could generate a concatemeric replication sub-strate that, following ampli ﬁcation as a rolling circle, could be resolved into plasmid monomers by Cre-mediated recombina-tion. The plasmid yield per replication initiation event would be increased, providing protection against RepA insuf ﬁciency. The loxsite is the unique target for Cre action in P1 DNA, as we did not ﬁnd any other similar sites. In the P1 genome, the cregene is downstream of cra, a gene of unknown function. In the prophage, creis weakly expressed, either from promoters P1creand P2 cre, which lie within cra, or together with crafrom a third promoter, P3 cre(302). The /H1100235 regions of P3 creand P2crepromoters overlap GATC sequences (Table 4), and P3crewas shown to be down-regulated in E. coli by Dam methylation, as P2 creprobably also is. Whether methylation of one strand is suf ﬁcient for the repression is unknown. If this was not so, the expression of crein a prophage would increase immediately following each round of replication and be atten-uated later as a result of methylation of the newly replicatedstrand in the region of P3 creand P2 crepromoters. A putative fourth crepromoter, P4 cre, predicted to be the strongest, lies almost 500 bp upstream of creand on the op- posite side of lox(Table 4). Its /H1100235 region is partially over- lapped by the C1 operator, O cre(Table 6), indicating that the promoter is inactive in the P1 prophage. Apparently, the cre gene is expressed most ef ﬁciently at the time of infection and during P1 lytic development. Following infection, the Cre re-combinase can cyclize the DNA of phages that carry a loxsite on each redundant end (302). The ﬁrst phages to be packaged belong to this privileged minority and hence are to be found inevery productive burst. The other phage DNAs must rely onhomologous recombination to cyclize. (c) Plasmid addiction. A fail-safe P1 stabilization mecha- nism is encoded by the addiction operon, phd doc , located in a region far from other plasmid maintenance genes (Fig. 1). Thephd doc operon programs the inhibition of growth (and the eventual death) of any daughter cells emerging plasmid free.Two small genes of this operon encode a stable protein toxin,Doc, and an unstable cognate antidote, Phd (95, 96, 192, 211).Doc appears to block protein synthesis reversibly (R. Magnu-son and M. Yarmolinsky, unpublished results). Phd and Doccan form a complex. This interaction prevents Doc from killingplasmid-containing cells and enhances the Phd-mediated re-pression of the autoregulated phd doc promoter (210, 211). Phd is slowly degraded by the host protease ClpXP (194),allowing for the release of the toxin protein upon plasmid loss.Recently documented connections among prokaryotic toxin-antitoxin systems, Phd-Doc among them, and their relationshipto the eukaryotic nonsense-mediated RNA decay system places Phd-Doc in a grander evolutionary context than anticipated(13). The nucleotide sequence downstream of doc contains two ORFs that could encode proteins of 66 and 347 residues,designated here pdcA andpdcB (post-doc). The 5 /H11032end of pdcA overlaps 17 terminal nucleotides of the docgene.', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Phd and Doccan form a complex. This interaction prevents Doc from killingplasmid-containing cells and enhances the Phd-mediated re-pression of the autoregulated phd doc promoter (210, 211). Phd is slowly degraded by the host protease ClpXP (194),allowing for the release of the toxin protein upon plasmid loss.Recently documented connections among prokaryotic toxin-antitoxin systems, Phd-Doc among them, and their relationshipto the eukaryotic nonsense-mediated RNA decay system places Phd-Doc in a grander evolutionary context than anticipated(13). The nucleotide sequence downstream of doc contains two ORFs that could encode proteins of 66 and 347 residues,designated here pdcA andpdcB (post-doc). The 5 /H11032end of pdcA overlaps 17 terminal nucleotides of the docgene. The phd-doc transcript appears to include the transcripts of pdcA andpdcB , as suggested by the lack of predicted Rho-independent termi-nators of the phd doc operon within pdcA orpdcB , and the lack of a separate recognizable /H9268 70promoter sequence that could drive their transcription. Whether the functions of pdcA and pdcB are accessory or unrelated to the function of phd doc remains to be seen. (iii) Restriction-modi ﬁcation. P1mod and resare tandem genes that encode the subunits of a type III restriction-modi-7048 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  ﬁcation enzyme, EcoPI. This bifunctional enzyme has been the subject of extensive studies (reviewed in references 32 and 38;see also reference 161). The P1 Mod subunit recognizes the DNA sequence 5 /H11032-A GACC and catalyzes methylation of the central adenine resi-due at the N-6 position (23), using adenosyl methionine as themethyl donor. The Res subunit catalyzes the double-strandcleavage of DNA about 25 bp to the 3 /H11032side of a recognition sequence, but it does so only when bound to the Mod subunitin a Res 2Mod2complex (161). Whereas methylation by Mod can occur at any recognition sequence, each double-strand scission by Res-Mod requires apair of unmethylated recognition sequences, in head-to-headconﬁguration (32, 219). The cut appears to be triggered by collision between two converging Res-Mod enzymes that, pow-ered by ATP hydrolysis, have been translocated along theDNA (220). The requirement for collision, in combination withthe asymmetry of enzyme recognition sites, provides an ef ﬁ- cient protection from cleavage of newly replicated DNA inwhich only one strand is unmethylated, because the unmodi-ﬁed sites in one orientation are always paired with modi ﬁed sites in the opposite orientation. Host killing by restriction is avoided during the process of establishing a P1 prophage in a bacterial cell with an unmod-iﬁed chromosome. This is achieved by complex regulatory mechanisms that delay restriction long enough to allow com-plete methylation of host DNA (253). In P1 lysogens, tran-scripts containing either mod orresmessages are detected. This indicates that mod and resform separate operons, al- though the beginning of resimmediately follows the end of mod (285). The /H9268 70promoters for res, separate from those for mod, have been identi ﬁed previously or are predicted here (Table 4). Additional regulation occurs at the translational andposttranslational levels (253). (iv) Superinfection exclusion. During attempts to clone the c1 gene of P1, Devlin et al. isolated a fragment of DNA that, although not carrying c1, protected cells from superinfection by wild-type P1, its c1 and vir smutants, and the heteroimmune P7 phage (77). This extended immunity phenotype, designated superimmunity and subsequently attributed to superinfectionexclusion, was associated with a gene upstream of the c4icd ant operon and transcribed in the opposite direction (170, 213).We ﬁnd that this gene, designated previously sim and here simC , is the last of three genes of the simABC operon. The sim genes appear to encode precursors of periplasmic proteins,suggesting that their functions are related. The location ofputative promoter sequences in the region preceding simC indicates that simC is cotranscribed with simB , or with simA and simB (Table 4). A plasmid carrying the simoperon was found by minicell analysis to specify a processed, as well asfull-length, form of the SimC protein (213). Two additionalproteins apparently expressed from this clone were seen fol-lowing sodium dodecyl sulfate (SDS)-PAGE (170). One isprobably SimB (12.0 kDa). The other migrated as a very smallprotein and, although not mentioned by the authors, is prob-ably SimA (predicted molecular weight, 4.8 kDa). The superimmunity (or superinfection exclusion) phenotype was observed in the presence of SimC alone (213). The SimCprotein appears to be an analog of superinfection exclusionproteins that act to prevent injection of superinfecting phageDNA into the cytoplasm of the infected cell. Many phages, both temperate and virulent, encode such proteins. SimC ap-pears to act in the periplasmic space, where the processed formis found, or in the cytoplasmic membrane, similarly to the SieAprotein of P22 (139) and the Imm protein of T4 (205), eitherby helping to destroy the injected DNA or by preventing itsentry into the cytoplasm (139, 170, 206, 213).'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Bivalent operators names are underlined; constituent monovalent ope rators are listed below the name of each bivalent operator. Operators with a superscript /H9011have been shown to engage in looping to nearby operators in vitro. bPosition of the ﬁrst nucleotide if the operator is assigned to the /H11001strand or to the ﬁrst nucleotide of its complement if the operator is assigned to the /H11002strand. Coordinates refer to positions of operators in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. cThe underlined nucleotides are conserved in monovalent operators and in at least one constituent operator of each bivalent operator. dThe sequence of O c1b corresponds to that identi ﬁed by DNase protection experiments (317) and differs from the sequence proposed earlier (84).VOL. 186, 2004 BACTERIOPHAGE P1 7045', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  TABLE 7. Known and potential binding sites in P1 DNA for E. coli proteins IHF, DnaA, and LexA CoordinatesaStrand SequenceSimilarity scorebKnown or potential function or relative location (reference[s]) IHF 1478 –1526 /H11002 TATGTCACCATAAATATCAAATAATTATA 52.0 c8 regulation 6985 –7132 /H11002 TTATATGCATACAAAAAGATGAAGTTATA 53.9 Internal to mod 19809 –19857 /H11002 ATCTTACCTAATAAAATAAATTAATTAGA 55.7 ssbregulation, lyzregulation 21424 –21472 /H11002 TGAACATAAAACACTATCAATAAGTTTGA 50.1 Overlaps terminal inverted repeat of defective IS 1 (93, 94) 30319 –30367 /H11002 TTTAATTCACTATTAAGGAAATAGTTATG 53.7 cinregulation 38415 –38463 /H11002 TATAATGTTCAGAATATCAATAAGATATT 51.8 16regulation 38419 –38466 /H11001 AATATCTTATTGATATTCTGAACATTATA 51.4 16regulation 46372 –46419 /H11001 ATTAATCGGTTTAAACTCACTAATTAATT 51.4 simregulation 46498 –46545 /H11002 TGGTTGATTCTTATAATCAAAAAACTATT 50.3 simregulation 49930 –49977 /H11002 AAAAAATTAAATAAAGCCAATGTCTTAGC 54.2 Internal to repL 50647 –50694 /H11001 AAAATTGCTAGAGACATCAAGGAATTAAA 57.0 rlfBregulation or part of alternative replication origin 50672 –50719 /H11001 TAAAATTAAAACACAATCTAAAACCTGAT 50.5 rlfBregulation or part of alternative replication origin 51151 –51198 /H11001 ATAAAATCACATACAAACAGGGAGTTACT 58.7 rlfBregulation or part of alternative replication origin 57740 –57787 /H11001 GTAATTTACTTTAAAAACAGTCAGTTAAT 57.2 parS -ParB-IHF centromeric complex formation (91) 61404 –61451 /H11002 GAAAAAGTAATATGAATCAATCATTTATC 57.3 repA regulation (reviewed in reference 49) 89562 –89610 /H11001 CTGAAATAGCGGAAAAA CAAAGAGTTAATc40.8 Involved in PacA- and PacB-mediated cleavage at pacsite (287) DnaA 1076 –1090 /H11001 TAGTTACCCCCAGGC 54.3 Internal to cre 5894 –5908 /H11001 GTGTTAATAACAAGA 56.0 Internal to res 5892 –5906 /H11002 TTGTTATTAACACGG 55.1 Internal to res 6716 –6730 /H11001 GTGTTGTTAACAAGC 54.3 Internal to mod 9564 –9578 /H11002 GAGTTATCATCAATA 57.7 Internal to darB 41563 –41577 /H11001 AGGTTATCGACAGTA 55.2 Internal to sit 47322 –47336 /H11002 GTGTTTTTTACAGGA 54.3 Internal to icd 47945 –47959 /H11001 GAATTTTACACAAAA 54.3 Internal to ant1/2 49965 –49979 /H11001 TTTTTATCCACACCC 56.9 Internal to repL , presumably part of oriL 50148 –50162 /H11001 CGGTTATCCAAAAAA 56.0 Internal to repL , presumably part of oriL 50313 –50327 /H11001 GTTTTAGCCAAAAAT 54.4 Internal to repL , presumably part of oriL 50986 –51000 /H11001 GGGTTCTTCACAATG 54.3 Internal to rlfA, overlaps predicted promoter of rlfB, presumably regulates expression of rlfB 52202 –52216 /H11001 GATTTAAGCACAAAA 56.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  3 Internal to icd 47945 –47959 /H11001 GAATTTTACACAAAA 54.3 Internal to ant1/2 49965 –49979 /H11001 TTTTTATCCACACCC 56.9 Internal to repL , presumably part of oriL 50148 –50162 /H11001 CGGTTATCCAAAAAA 56.0 Internal to repL , presumably part of oriL 50313 –50327 /H11001 GTTTTAGCCAAAAAT 54.4 Internal to repL , presumably part of oriL 50986 –51000 /H11001 GGGTTCTTCACAATG 54.3 Internal to rlfA, overlaps predicted promoter of rlfB, presumably regulates expression of rlfB 52202 –52216 /H11001 GATTTAAGCACAAAA 56.9 Internal to pmgF 61448 –61462 /H11002 AAACTATCCACACAA 51.8 Replication initiator complex formation at oriR (reviewed in reference 51) 61457 –61471 /H11002 CACTTATAAAAACTA /H1102150.0 Replication initiator complex formation at oriR (reviewed in reference 51) 61467 –61481 /H11002 TAGTTATCACCATTA 51.9 Replication initiator complex formation at oriR (reviewed in reference 51) 61690 –61703 /H11002 ACATTATCCACTGGA 51.8 Replication initiator complex formation at oriR (reviewed in reference 51) 61698 –61712 /H11002 CACTTATCCACATTA 56.0 Replication initiator complex formation at oriR (reviewed in reference 51) 68953 –68967 /H11002 TGGTTATCCACCGTG 54.3 Between upland its predicted promoter; regulation of upl 72890 –72904 /H11002 CGGTTATGCACAGGA 56.9 Internal to 5 73755 –73769 /H11001 GGCTTATCAACAAAA 56.9 Internal to 6 80593 –80607 /H11002 TTGGTATCCACAGGG 57.7 Internal to pmgQ 85215 –85229 /H11002 AAGTTATGCACTTGC /H1102150.0 hotregulation 85227 –85241 /H11002 GATTTATCCACAAAG 63.5 hotregulation 85851 –85865 /H11002 GAATTTTCCTCAAGG 55.1 Vicinity of P2 humD promoter internal to upfQ; probable humD regulation 89848 –89862 /H11002 GACTTTTCCAGAAGA 56.0 Vicinity of pacsite internal to pacA , packaging regulation LexA 46–65 /H11001//H11002 ACCTGTAAATCCATACAGTT 62.7/59.8 Regulation of ImmC genes 86346 –86365 /H11001//H11002 TACTGTATAAATGAACAGTA 67.7/66.1 humD regulation and possibly lxrregulation aThe ﬁrst number is the coordinate of the ﬁrst nucleotide of upper-strand ( /H11001) binding sites or of the last nucleotide of the lower-strand ( /H11002) binding sites, based on arbitrary convention. Coordinates refer to positions of sites in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. bOf predicted IHF, DnaA, or LexA binding sites, only those whose similarity score to the consensus exceeded 50.0, 54.0, or 60.0, respectively (as calcu lated by the GCG Fitconsensus program) are shown. Exceptions are made for tandem sites. cFor PP1PAC in GenBank, boldface C is a G.7046 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  An additional component of ImmC or a separate immunity locus may be located downstream of cre. This gene, tentatively named c8, is probably the locus ( c8) of a clear plaque mutation (280, 281). Temperature shift experiments with a thermosen-sitive allele of c8 suggested that the gene is probably not ex- pressed in a lysogen and hence is most likely involved uniquelyin the establishment of lysogeny. (b) ImmI. The ImmI region (antipodal to immC in the circular representation of the plasmid genome) contains a C1-controlled operon of three genes: c4,icd, and ant1/2 , which confer a separate speci ﬁcity to immunity. It permits P1 to plate on lysogens of the closely related phage P7, and vice versa,despite functional interchangeability of their C1 repressors(54, 131, 280, 326). The ant1/2 and icdgenes determine an antirepressor protein and an inhibitor of cell division, respec-tively, whereas the c4 gene determines an antisense RNA that acts as a secondary repressor of Icd and antirepressor synthesis(58, 131, 257). Maintenance of lysogeny requires, in addition tothe expression of c1, the expression of c4 to prevent synthesis of the antirepressor protein (reviewed in reference 126). A77-bp antisense RNA encoded by c4 regulates expression of downstream genes in its own message (57; diagrammed in Fig.1 of reference 343a). Two short single-stranded regions, b /H11032and a/H11032, exposed in its cloverleaf-like structure, block translation of icd(and of translationally coupled ant1) by interaction with the complementary regions, a2 and b2, in the c4icd ant1/2 mRNA. The interaction occludes the RBS in front of the icdgene (58). The resulting translational block additionally permits prema-ture termination of anttranscription via a Rho-dependent ter- minator (33). Slight differences between P1 and P7 in theseshort regions are responsible for the heteroimmunity of thetwo phages. An additional speci ﬁcity determinant, sas(site of Ant speci ﬁcity), is involved in determining the capacity of Ant proteins to induce prophage on heteroimmune superinfection.The site resides centrally within the antgenes of P1 and P7 (J. Heinrich and H. Schuster, unpublished results). It was sug-gested that Ant proteins must normally load onto their phage-speci ﬁcsassites to perform their antirepressor function (294). The action of C4 appears limited to the mRNA of its own operon, as we did not ﬁnd any pair of sequences complemen- tary to b /H11032and a/H11032in the P1 genome, other than those designated a1 and b1, located upstream of c4, and presumed to participate in unmasking of a2 and b2 by competitive binding of b /H11032and a/H11032. (c) ImmT. The ImmT region, distant from ImmC and ImmI, contains a small gene, lxc, whose product modulates the func- tion of C1 by formation of a ternary complex with C1 andoperator DNA (272, 317, 318). Participation of Lxc in the C1-operator complexes that control c1 expression increases the afﬁnity of the repressor for the operator and down-regulates the synthesis of C1 itself (318). Lxc can thus relieve repressionat weak operators by decreasing the intracellular concentrationof C1 below that critical for their binding and simultaneouslyenhance repression at strong operators by increasing their af-ﬁnity for C1. Consequently, lxcmutations can have divergent effects on transcription from different C1-controlled promoters(272, 273, 311). Part of the modulating effect of Lxc may bemediated by looping between nearby operators complexedwith C1, as looping is dramatically increased by Lxc (128,343a). As Lxc has been reported to strongly inhibit the abilityof Coi to dissociate operator-C1 complexes (317), it may assist in establishing P1 as a prophage.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Participation of Lxc in the C1-operator complexes that control c1 expression increases the afﬁnity of the repressor for the operator and down-regulates the synthesis of C1 itself (318). Lxc can thus relieve repressionat weak operators by decreasing the intracellular concentrationof C1 below that critical for their binding and simultaneouslyenhance repression at strong operators by increasing their af-ﬁnity for C1. Consequently, lxcmutations can have divergent effects on transcription from different C1-controlled promoters(272, 273, 311). Part of the modulating effect of Lxc may bemediated by looping between nearby operators complexedwith C1, as looping is dramatically increased by Lxc (128,343a). As Lxc has been reported to strongly inhibit the abilityof Coi to dissociate operator-C1 complexes (317), it may assist in establishing P1 as a prophage. The initiation codon of lxcappears to overlap with the ter- mination codon of the preceding ulxgene of unknown func- tion, indicating transcriptional and translational coupling ofthe two genes (Table 3). The coupling of lxcand ulximplies that lxc, in addition to being transcribed from its own promot- ers, may be transcribed from P darB , which presumably drives transcription of ulxand the preceding gene, darB (Table 4). None of the promoters that might drive transcription of lxcis under the control of C1, suggesting that Lxc-mediated modu-lation of C1 function is independent of ImmC and ImmI. (ii) Plasmid maintenance. Despite its low copy number per bacterial origin (0.7 to 1.4) (248), the P1 plasmid is lost with afrequency of only about 10 /H110025per cell per generation (261). Like other plasmids that have accepted the tradeoff of large size for low copy number, P1 must counteract the increasedrisk of loss at cell division. Four functions have been identi ﬁed that allow it to accomplish this task: a regulated plasmid rep-licon, a partitioning mechanism, a site-speci ﬁc recombination system, and a plasmid addiction (postsegregational killing orgrowth inhibition) mechanism. These are discussed below.Analysis of the genome of P1 indicates that whereas the lyticorigin may contribute to plasmid replication (349), there is nosecond addiction or partition module in P1 as there are incertain other plasmids (80, 140). (a) Plasmid replication and partition. The P1 plasmid rep- lication and partition genes are encoded within one region ofP1 DNA, where they form two operons transcribed in the samedirection. The P1 replicon has been intensively studied (re-viewed in references 49 and 51). It belongs to the large familyof plasmid replicons that encode an initiator protein, calledRepA in P1, and contain multiple binding sites for that initi-ator (19-bp iterons in P1). One set of these sites, incC , pre- cedes the repA gene and is part of the replication origin ( oriR). A second set, incA , follows repA and is a regulatory locus (Table 1). Replication from the P1 plasmid origin, oriR, pro- ceeds in both directions (241). P1 plasmid and E. coli chromo- some replication have much in common, including sequestra tion of the origin when its 5 /H11032-GATC sequences are hemimethylated (1) and a requirement for DnaA (Table 7) (113). One differ-ence is that the ADP form of DnaA suf ﬁces for P1 replication, whereas the ATP form is essential for bacterial replication(333). The P1 repA promoter is nested among the incC iterons and consequently is subject to autorepression by RepA (5). Repli-cation control is exerted by incC and the additional iterons of incA in more than one way. By sequestering RepA, the incA iterons can prevent the concentration of RepA at the originfrom attaining the threshold value required for ﬁring (reviewed in reference 49).', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  One differ-ence is that the ADP form of DnaA suf ﬁces for P1 replication, whereas the ATP form is essential for bacterial replication(333). The P1 repA promoter is nested among the incC iterons and consequently is subject to autorepression by RepA (5). Repli-cation control is exerted by incC and the additional iterons of incA in more than one way. By sequestering RepA, the incA iterons can prevent the concentration of RepA at the originfrom attaining the threshold value required for ﬁring (reviewed in reference 49). By becoming “handcuffed ”to the iterons of incC via bound RepA, the additional iterons can interfere with both replisome assembly (239) and the release of autorepres-sion that would otherwise replenish sequestered RepA (50).The availability of RepA for replication is limited not only bysequestration and handcuf ﬁng but also by the formation of inactive RepA dimers which require bacterial chaperones forconversion to active protein monomers (334). We suggest anadditional role of incA in the control of repA expression. The putative transcriptional terminator of repA is located within theVOL. 186, 2004 BACTERIOPHAGE P1 7047', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  incA region (Table 5), and thus occupation of incA iterons by RepA may act as a roadblock to the completion of repA tran- scripts. Abbreviated repA transcripts might undergo more rapid degradation than intact transcripts. The partition module of P1, which is situated downstream of therepA gene and the incA iterons, is a major contributor to stability of the plasmid. It ensures partition of plasmid mole-cules to daughter cells and has been, with its homologs, thesubject of numerous detailed studies and reviews (reviewedmost recently in references 97 and 306). The module consistsof an operon of two genes, parA andparB , which ends at a site, parS , a centromere analog (4). ParA is an ATPase whose ac- tivity is stimulated by ParB (73). Both ATP and ADP forms ofParA act as autorepressors, the ADP form being the moreeffective (70, 71). A stimulation of repression by ParB (87) isassisted by parS (114). The parS site contains two kinds of recognition sequences for ParB, four heptameric and two hex-americ boxes, and a binding site for the host protein IHF(Tables 1 and 7) (249). ParB and IHF form a complex at parS (72, 89) to which ParA, in the ATP form, can bind via ParB (37,90). Binding of ParB to parS can permit parS sites to pair (83) and can nucleate spreading of ParB to the ﬂanking DNA and silence transcription of genes as far as several kilobases away(258). Whether ParA and ParB translocate plasmid moleculesto their target positions within a cell during partitioning bythemselves or attach them to unknown host components ofpartition machinery is unclear. We did not ﬁnd in P1 DNA any additional sequences similar toparS that could serve as potential binding sites for ParB. This suggests that the function of ParB is limited to partition-ing in P1, whereas in certain other plasmids, e.g., RK2 andN15, the homologous proteins appear to have additional func-tions (102, 251, 335). Two putative /H9268 70promoters of P1 gene 23, immediately downstream of parB , of which one overlaps the IHF binding region of parS , appeared weak when tested in vivo in the absence of ParB (259; A. Dobruk and M. Łobocka, unpublished results). The possibility that conditions might existleading to their signi ﬁcant expression and regulation by ParB has not been studied. (b) DNA cyclization and multimer resolution. Faithful par- titioning of newly replicated plasmid molecules is assisted byprior resolution to monomers of plasmid multimers that areinevitably formed in recA /H11001bacteria. The resolution is usually accomplished by a site-speci ﬁc recombination. In P1 it is en- sured by the recombinase Cre, encoded near its site of action,lox(8, 299, 300), far from the genes of replication and partition (Fig. 1). The resolution of plasmid multimers into monomersincreases the number of partitionable P1 molecules and henceplasmid stability (21). The same accessory proteins that arerequired for the resolution of ColE1 dimers by the bacterialrecombinase XerCD are also required for the stable mainte-nance of P1 prophage (244). These proteins (ArgR, for whicha putative binding site to the left of loxhas been located, and PepA) are proposed to constrain the directionality of the re-combination in vivo. The possibility that the lox-cre module might assist plasmid stability in another way has also beensuggested (10). Interplasmid recombination involving a repli-cating plasmid could generate a concatemeric replication sub-strate that, following ampli ﬁcation as a rolling circle, could be resolved into plasmid monomers by Cre-mediated recombina-tion. The plasmid yield per replication initiation event would be increased, providing protection against RepA insuf ﬁciency. The loxsite is the unique target for Cre action in P1 DNA, as we did not ﬁnd any other similar sites.', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  These proteins (ArgR, for whicha putative binding site to the left of loxhas been located, and PepA) are proposed to constrain the directionality of the re-combination in vivo. The possibility that the lox-cre module might assist plasmid stability in another way has also beensuggested (10). Interplasmid recombination involving a repli-cating plasmid could generate a concatemeric replication sub-strate that, following ampli ﬁcation as a rolling circle, could be resolved into plasmid monomers by Cre-mediated recombina-tion. The plasmid yield per replication initiation event would be increased, providing protection against RepA insuf ﬁciency. The loxsite is the unique target for Cre action in P1 DNA, as we did not ﬁnd any other similar sites. In the P1 genome, the cregene is downstream of cra, a gene of unknown function. In the prophage, creis weakly expressed, either from promoters P1creand P2 cre, which lie within cra, or together with crafrom a third promoter, P3 cre(302). The /H1100235 regions of P3 creand P2crepromoters overlap GATC sequences (Table 4), and P3crewas shown to be down-regulated in E. coli by Dam methylation, as P2 creprobably also is. Whether methylation of one strand is suf ﬁcient for the repression is unknown. If this was not so, the expression of crein a prophage would increase immediately following each round of replication and be atten-uated later as a result of methylation of the newly replicatedstrand in the region of P3 creand P2 crepromoters. A putative fourth crepromoter, P4 cre, predicted to be the strongest, lies almost 500 bp upstream of creand on the op- posite side of lox(Table 4). Its /H1100235 region is partially over- lapped by the C1 operator, O cre(Table 6), indicating that the promoter is inactive in the P1 prophage. Apparently, the cre gene is expressed most ef ﬁciently at the time of infection and during P1 lytic development. Following infection, the Cre re-combinase can cyclize the DNA of phages that carry a loxsite on each redundant end (302). The ﬁrst phages to be packaged belong to this privileged minority and hence are to be found inevery productive burst. The other phage DNAs must rely onhomologous recombination to cyclize. (c) Plasmid addiction. A fail-safe P1 stabilization mecha- nism is encoded by the addiction operon, phd doc , located in a region far from other plasmid maintenance genes (Fig. 1). Thephd doc operon programs the inhibition of growth (and the eventual death) of any daughter cells emerging plasmid free.Two small genes of this operon encode a stable protein toxin,Doc, and an unstable cognate antidote, Phd (95, 96, 192, 211).Doc appears to block protein synthesis reversibly (R. Magnu-son and M. Yarmolinsky, unpublished results). Phd and Doccan form a complex. This interaction prevents Doc from killingplasmid-containing cells and enhances the Phd-mediated re-pression of the autoregulated phd doc promoter (210, 211). Phd is slowly degraded by the host protease ClpXP (194),allowing for the release of the toxin protein upon plasmid loss.Recently documented connections among prokaryotic toxin-antitoxin systems, Phd-Doc among them, and their relationshipto the eukaryotic nonsense-mediated RNA decay system places Phd-Doc in a grander evolutionary context than anticipated(13). The nucleotide sequence downstream of doc contains two ORFs that could encode proteins of 66 and 347 residues,designated here pdcA andpdcB (post-doc). The 5 /H11032end of pdcA overlaps 17 terminal nucleotides of the docgene.', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Phd and Doccan form a complex. This interaction prevents Doc from killingplasmid-containing cells and enhances the Phd-mediated re-pression of the autoregulated phd doc promoter (210, 211). Phd is slowly degraded by the host protease ClpXP (194),allowing for the release of the toxin protein upon plasmid loss.Recently documented connections among prokaryotic toxin-antitoxin systems, Phd-Doc among them, and their relationshipto the eukaryotic nonsense-mediated RNA decay system places Phd-Doc in a grander evolutionary context than anticipated(13). The nucleotide sequence downstream of doc contains two ORFs that could encode proteins of 66 and 347 residues,designated here pdcA andpdcB (post-doc). The 5 /H11032end of pdcA overlaps 17 terminal nucleotides of the docgene. The phd-doc transcript appears to include the transcripts of pdcA andpdcB , as suggested by the lack of predicted Rho-independent termi-nators of the phd doc operon within pdcA orpdcB , and the lack of a separate recognizable /H9268 70promoter sequence that could drive their transcription. Whether the functions of pdcA and pdcB are accessory or unrelated to the function of phd doc remains to be seen. (iii) Restriction-modi ﬁcation. P1mod and resare tandem genes that encode the subunits of a type III restriction-modi-7048 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  ﬁcation enzyme, EcoPI. This bifunctional enzyme has been the subject of extensive studies (reviewed in references 32 and 38;see also reference 161). The P1 Mod subunit recognizes the DNA sequence 5 /H11032-A GACC and catalyzes methylation of the central adenine resi-due at the N-6 position (23), using adenosyl methionine as themethyl donor. The Res subunit catalyzes the double-strandcleavage of DNA about 25 bp to the 3 /H11032side of a recognition sequence, but it does so only when bound to the Mod subunitin a Res 2Mod2complex (161). Whereas methylation by Mod can occur at any recognition sequence, each double-strand scission by Res-Mod requires apair of unmethylated recognition sequences, in head-to-headconﬁguration (32, 219). The cut appears to be triggered by collision between two converging Res-Mod enzymes that, pow-ered by ATP hydrolysis, have been translocated along theDNA (220). The requirement for collision, in combination withthe asymmetry of enzyme recognition sites, provides an ef ﬁ- cient protection from cleavage of newly replicated DNA inwhich only one strand is unmethylated, because the unmodi-ﬁed sites in one orientation are always paired with modi ﬁed sites in the opposite orientation. Host killing by restriction is avoided during the process of establishing a P1 prophage in a bacterial cell with an unmod-iﬁed chromosome. This is achieved by complex regulatory mechanisms that delay restriction long enough to allow com-plete methylation of host DNA (253). In P1 lysogens, tran-scripts containing either mod orresmessages are detected. This indicates that mod and resform separate operons, al- though the beginning of resimmediately follows the end of mod (285). The /H9268 70promoters for res, separate from those for mod, have been identi ﬁed previously or are predicted here (Table 4). Additional regulation occurs at the translational andposttranslational levels (253). (iv) Superinfection exclusion. During attempts to clone the c1 gene of P1, Devlin et al. isolated a fragment of DNA that, although not carrying c1, protected cells from superinfection by wild-type P1, its c1 and vir smutants, and the heteroimmune P7 phage (77). This extended immunity phenotype, designated superimmunity and subsequently attributed to superinfectionexclusion, was associated with a gene upstream of the c4icd ant operon and transcribed in the opposite direction (170, 213).We ﬁnd that this gene, designated previously sim and here simC , is the last of three genes of the simABC operon. The sim genes appear to encode precursors of periplasmic proteins,suggesting that their functions are related. The location ofputative promoter sequences in the region preceding simC indicates that simC is cotranscribed with simB , or with simA and simB (Table 4). A plasmid carrying the simoperon was found by minicell analysis to specify a processed, as well asfull-length, form of the SimC protein (213). Two additionalproteins apparently expressed from this clone were seen fol-lowing sodium dodecyl sulfate (SDS)-PAGE (170). One isprobably SimB (12.0 kDa). The other migrated as a very smallprotein and, although not mentioned by the authors, is prob-ably SimA (predicted molecular weight, 4.8 kDa). The superimmunity (or superinfection exclusion) phenotype was observed in the presence of SimC alone (213). The SimCprotein appears to be an analog of superinfection exclusionproteins that act to prevent injection of superinfecting phageDNA into the cytoplasm of the infected cell. Many phages, both temperate and virulent, encode such proteins. SimC ap-pears to act in the periplasmic space, where the processed formis found, or in the cytoplasmic membrane, similarly to the SieAprotein of P22 (139) and the Imm protein of T4 (205), eitherby helping to destroy the injected DNA or by preventing itsentry into the cytoplasm (139, 170, 206, 213).'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:38 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'108'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992976'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'421ms'), (b'x-request-id', b'6aeb44930204b4d5c01f221252d1e724'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412490f0e655d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:38 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'108'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992976'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'421ms'), (b'x-request-id', b'6aeb44930204b4d5c01f221252d1e724'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412490f0e655d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The superimmunity (or superinfection exclusion) phenotype was observed in the presence of SimC alone (213). The SimCprotein appears to be an analog of superinfection exclusionproteins that act to prevent injection of superinfecting phageDNA into the cytoplasm of the infected cell. Many phages, both temperate and virulent, encode such proteins. SimC ap-pears to act in the periplasmic space, where the processed formis found, or in the cytoplasmic membrane, similarly to the SieAprotein of P22 (139) and the Imm protein of T4 (205), eitherby helping to destroy the injected DNA or by preventing itsentry into the cytoplasm (139, 170, 206, 213). The processing ofSimC requires SecA (213), as probably does the processing ofSimA and SimB, which, like SimC, have putative signal peptides. Products of the simoperon, like the product of sieA of P22 (139, 307), exclude both phage and transducing particle DNA.However, exclusion of the latter by Sim is much less ef ﬁcient, indicating that the Sim system can discriminate between P1and foreign DNA. Conceivably, SimC or another Sim proteincan interact with a P1 protein that is bound speci ﬁcally with phage DNA as it enters a cell during infection. Cells carrying low-copy-number sim /H11001plasmids display a Sim/H11002phenotype (77, 170), which appears inconsistent with a role of the simgenes in lysogeny. It is possible that in P1 ly- sogens, the simfunctions are induced only under certain cir- cumstances or that the main role of Sim proteins is to protectfrom superinfection those cells in which lytic development hasbeen initiated and the simgene dosage is high. The physiolog- ical role of simfunctions could be analogous to that of super- infection exclusion functions encoded by the immT and sp genes of the lytic phage T4. It has been proposed that super-infection exclusion mechanisms may preserve genetic diversityamong phages, protecting phage populations from dominanceby a phage that would otherwise outcompete any related phagewhen given a possibility to propagate in a superinfected cell(205). (v) HumD and Lxr. The monocistronic operon immediately upstream of the P1 addiction module encodes HumD protein,a homolog of E. coli UmuD /H11032(200). The P1 humD gene, like E. coli umuD , is transcribed from a LexA-regulated promoter (Tables 4 and 7). P1 lacks a homolog of umuC , which in E. coli is functionally associated and cotranscribed with umuD . Aside from recA ,umuD and umuC ofE. coli are the only LexA-regulated genes of the SOS response that are requiredfor DNA damage-induced mutagenesis (290). They encode alow-ﬁdelity and low-processivity DNA polymerase, PolV, whose main function is to bypass DNA lesions (309). A RecA-mediated autocleavage of the umuD gene product (215) is required to form active PolV, which is a complex of UmuD /H11032 and UmuC in a 2:1 ratio (43). P1 HumD corresponds to theprocessed form of UmuD, UmuD /H11032, and can functionally re- place it (217). The regulatory region of humD overlaps the regulatory re- gion of a divergently transcribed gene of unknown function.This gene could encode a 190-residue, slightly acidic proteinthat has no homologs in databases. We surmise that its tran-scription, like that of humD , is negatively regulated by LexA. One of two putative /H9268 70promoters of this gene overlaps the LexA binding site that controls humD (Tables 4 and 7), and thus we name the gene lxr(LexA regulated). Genes expressed in lytic development. Expression of the majority of P1 genes during lytic growth follows a strict tem-poral pattern that can lead to the production of mature phageparticles within less than 1 h. For many bacteriophages such asT4 (180), Mu (18, 55), P2 (339), and P4 (98), the regulatoryVOL. 186, 2004 BACTERIOPHAGE P1 7049', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  cascade has been subdivided into early, middle, and late stages. In P1, the cascade appears to be simpler; early transcriptionswitches directly to late transcription, without a well-de ﬁned intermediate stage (188, 193). C1 operator sequences, whichbind the primary phage repressor C1, act as a switch of earlygenes. Lpa (late promoter activator) binding sequences, whichare in the /H1100222 region of late promoters and enable RNA polymerase to initiate transcription upon binding of Lpa, act asa switch of late genes. (i) C1-controlled operons. Analysis of the entire P1 genome has revealed 17 operons that have C1 operators in the regionof their /H9268 70promoter sequences or, in some cases, between promoters and proximal genes (Table 6). One of these operons (pmgR ) is preceded by two nonoverlapping operators, and two others ( ban and c1) are preceded by operators that are biva- lent, bringing the total number of monovalent C1-binding sitesto 20, all of which were identi ﬁed previously. Although we did not ﬁnd any additional sequences with strong similarity to known C1-binding sites, sequences that resemble the C1-bind-ing sites but are missing the highly conserved C at position 7,are present in the parS site and in the promoter region close to thedarB gene. The sequence in parS did not interact with C1 in vitro (N. Sternberg, personal communication). Whetherthese sequences can act cooperatively with other C1 operatorsequences to bind C1 in vivo, as does the O c1b operator, which also has the C at position 7 replaced by another base, remainsto be seen. Looping between C1-bound operators located closeto each other has been demonstrated (128), but looping be-tween well-separated operator sequences has not. The 17 C1-controlled operons, transcribed from /H9268 70promot - ers, contain 49 genes that are derepressed early in lytic devel-opment. Transcription does not start simultaneously, since theoperators exhibit different af ﬁnities for C1 and are differently inﬂuenced by the Lxc protein, which modulates the C1-oper- ator interaction. Lytic replication genes start to be transcribedwithin 5 min of infection, allowing a prompt initiation of phageDNA replication. Increased amounts of P1 DNA can be de-tected about 15 min after infection, at about the same time ascleavage of the packaging site, pac, in P1 DNA (297). Detec- tion of DNA synthesis within 5 min has also been reported(282). Expression of other early functions was reported to start10 to 15 min after induction of P1 lysogens (187). The intensetranscription of P1 early functions is attenuated later in lyticdevelopment, prior to transcriptional activation of late genes(111). The attenuation depends on the E. coli RNA poly- merase-associated protein SspA. Whether SspA acts directlyon complexes of RNA polymerase with P1 /H9268 70promoter se - quences or requires the action of a P1 protein is unknown. Wedid not ﬁnd any conserved sequences, other than promoter sequences and C1-binding sites, in the regulatory regions of P1early genes. (a) P1-encoded tRNAs. Several phages encode their own tRNAs (e.g., see references 88, 177, and 247); the T4-likevibriophage KVP40 encodes at least 25 tRNAs at a single locus(223). Some of these tRNAs supplement the host pool of raretRNAs to facilitate ef ﬁcient expression of selected phage genes during lysogeny or at certain stages of lytic development (179,247). Additionally, it has been proposed that phage tRNAs canprovide selective pressure to keep base composition and codonusage of phage DNA signi ﬁcantly different from those of itshost (summarized in reference 227). We ﬁnd that the P1 ge- nome contains three sequences characteristic of tRNA genes.All three are located between the ban gene and its promoter, in the apparently untranslated regions of the banoperon. Two are adjacent; one is further downstream, separated from thetwo by three protein-coding genes.', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Some of these tRNAs supplement the host pool of raretRNAs to facilitate ef ﬁcient expression of selected phage genes during lysogeny or at certain stages of lytic development (179,247). Additionally, it has been proposed that phage tRNAs canprovide selective pressure to keep base composition and codonusage of phage DNA signi ﬁcantly different from those of itshost (summarized in reference 227). We ﬁnd that the P1 ge- nome contains three sequences characteristic of tRNA genes.All three are located between the ban gene and its promoter, in the apparently untranslated regions of the banoperon. Two are adjacent; one is further downstream, separated from thetwo by three protein-coding genes. Control of the P1 tRNAgenes from a promoter repressed by C1 implies that they arenot expressed during lysogeny. The proposed cloverleaf structures of all predicted P1 tRNAs are shown in Fig. 3. The ﬁrst tRNA, tRNA1, has the anticodon GUU, characteristic of E. coli tRNA Asn.I ti s9 0 % identical to the predicted tRNAAsnof the Salmonella enterica serovar Typhi plasmid pHCM2 (242) (GenBank accession no. AL513384) and contains all bases essential for the aminoacy-lation of tRNA AsninE. coli : the anticodon bases G34, U35, and U36 and the discriminator base G73 (201). The second tRNA, tRNA2, has the anticodon UGU, char- acteristic of E. coli tRNAThr. Although it is highly homologous to alanine tRNA-UGC of a cyanobacterium, Synechocystis sp. (SYCSLRB; 83% identity), and to tRNAAlaof numerous other organisms, it contains neither the G3-U70 wobble pair, which is the primary determinant of the acceptor identity of theE. coli tRNA Ala, nor the discriminator base G20, which is also an important characteristic of this tRNA (250, 308). Instead, in addition to the anticodon bases G35 and U36, which are iden-tity determinants of all E. coli tRNA Thrs, it contains base pairs G1-C72 and C2-G71, which in E. coli tRNAThr, are crucial for threonine charging activity (116). The third tRNA, tRNA3, has the anticodon CAU, which should correspond to the Met codon AUG. However, in itssequence this tRNA appears to be 89% identical to the E. coli tRNA Ile-2, encoded by the ileX gene. None of the differences (outside the anticodon) between tRNA3 and tRNAIle-2 affect residues known to be essential for the function of E. coli tRNAIle-2. In the tRNAIle-2 of E. coli the wobble position C34 is modi ﬁed by lysinylation to lysidine (i.e., 2-lysylcytidine). The modi ﬁed anticodon acquires the AUA (Ile)-decoding capacity and is required for the recognition of this tRNA by isoleucyl-tRNA synthetase (231, 232, 283). The other identifying char-acteristics of E. coli tRNA Ileare the anticodon loop bases A37 and A38, the discriminator base A73, and the base pairs C4- G69, U12-A23, and C29-G41 (236, 240). All of them are pres-ent in P1 tRNA3. Certain other phages, including 933W (247),T4 (224), and the T4-like phages RB69 (see http://phage.bioc.tulane.edu/) and KVP40 (223) also encode homologs of E. coli tRNA Ile-2, predicted to contain the lysylcytidine modi ﬁcation at C34. The T4 homolog of E. coli tRNAIle-2 was con ﬁrmed to have the capacity to decode ATA (274). Of the codons that are apparently recognized by P1 tRNAs, two, ACA and ATA, are rare in E. coli genes but overrepre- sented in certain P1 genes (Fig. 3).', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  All of them are pres-ent in P1 tRNA3. Certain other phages, including 933W (247),T4 (224), and the T4-like phages RB69 (see http://phage.bioc.tulane.edu/) and KVP40 (223) also encode homologs of E. coli tRNA Ile-2, predicted to contain the lysylcytidine modi ﬁcation at C34. The T4 homolog of E. coli tRNAIle-2 was con ﬁrmed to have the capacity to decode ATA (274). Of the codons that are apparently recognized by P1 tRNAs, two, ACA and ATA, are rare in E. coli genes but overrepre- sented in certain P1 genes (Fig. 3). The ACA codon is therarest threonine codon in E. coli , but its effect on translation has not been studied. The ATA codon, which is the ﬁfth rarest codon in E. coli , is known to dramatically decrease translation of those E. coli mRNAs that contain it, especially when present in multiple copies, or in tandem, or in a single copy in theanterior part of a gene (99, 166, 353). The ATA codons areoverrepresented more than twofold in 59 of 113 protein-codinggenes of P1 as compared to their representation in E. coli genes. Four of these 59 genes ( isaB,rlfA,7, and pacB ) contain7050 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  both single ATAs and tandem ATAs. In eight, ATAs are among the ﬁrstﬁve codons of the proximal region (at position 5o f lydC ,4o f 21,2o f ppfA ,3o f tciB,2o f ppp,4o f doc,4o f pdcB ,a n d2o f c1). At least two of these genes, docandc1, are known to be expressed, albeit not ef ﬁciently, during lysogeny, indicating that the low levels of ileX tRNA that are normally present in E. coli cells are suf ﬁcient for their translation, since tRNA3 is not expressed. However, the known translation-lim-iting role of ATA codons in E. coli implies that the ef ﬁciency of translation of doc,c1, and other genes rich in the ATA codons increases under conditions of abundance of P1 tRNA3. In thecase of doc and c1, the physiological signi ﬁcance of such an increase, if it does occur, is unclear. In the case of certain othergenes, the requirement for abundant tRNA3 to permit theirefﬁcient expression may provide an additional control on the timing of P1 lytic development at the translational level. All P1tRNA genes appear to be transcribed from the C1-controlledpromoter of the ban operon and so become available early during lytic development. It is noteworthy that lpa, whose product activates transcrip- tion of P1 late genes, has in its proximal region three rareE. coli codons, recognized by P1 tRNAs, namely ATA at po- sition 13 and an ATA ACA pair at positions 18 and 19.Whether the abundance of P1 tRNA3, and perhaps tRNA2,early in P1 lytic development normally contributes to ef ﬁcient translation of lpamRNA, which in turn ensures ef ﬁcient tran- scription of late genes, remains to be tested. Insertions of numerous phages into bacterial chromosomes occur at or near tRNA genes (reviewed in reference 46). Con-ceivably, tRNA genes could have been acquired by a P1 an-cestor either through its integration and then imperfect exci-sion from the DNA of its host or via recombination from otherphages whose prophages can integrate into a bacterial chro- mosome. (b) C1-controlled replication functions. P1 DNA synthesis has been reported to start as early as 5 min after infection of E. coli(282). Lytic replication initiates bidirectionally in the theta mode from an origin, oriL, distinct from the plasmid origin and located within the essential replication gene, repL (61, 112, 295). Later, lytic replication switches into the rolling-circlemode (60). The shift is accompanied by a gradual decrease inaccumulation of /H9258-shaped replication intermediates as /H9268- shaped intermediates accumulate. Most likely oriL is within the second half of the repL gene, where we ﬁnd putative binding sites for DnaA and IHF pro- teins and two 5 /H11032-GATC sequences (Table 7). P1 phage repli- cation in the rolling-circle ( /H9268) mode, predominant in the later stages of phage development, might initiate from an originother than oriL. One possible location for an additional P1 origin of lytic replication is within the rlfoperon, which directly follows the repL gene. It contains multiple sets of GATC se- quences and several IHF binding sites, which are internal to itstwo protein-coding regions (Table 7). An alternative proposalis that rolling-circle replication starts at pacfrom a nick intro- duced in DNA by pacase (296). In addition to the essential replication protein, RepL, P1 encodes homologs of certain E. coli replication-associated pro- teins: DnaB helicase (67, 237), single-stranded DNA-bindingprotein, Ssb (164, 187), Dam methyltransferase (66), and thetheta subunit of DNA polymerase III. Genes encoding thesehomologs ( ban,ssb,dmt, and hot, respectively), are in separate operons and, with the exception of the last mentioned, arecontrolled by C1.', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  It contains multiple sets of GATC se- quences and several IHF binding sites, which are internal to itstwo protein-coding regions (Table 7). An alternative proposalis that rolling-circle replication starts at pacfrom a nick intro- duced in DNA by pacase (296). In addition to the essential replication protein, RepL, P1 encodes homologs of certain E. coli replication-associated pro- teins: DnaB helicase (67, 237), single-stranded DNA-bindingprotein, Ssb (164, 187), Dam methyltransferase (66), and thetheta subunit of DNA polymerase III. Genes encoding thesehomologs ( ban,ssb,dmt, and hot, respectively), are in separate operons and, with the exception of the last mentioned, arecontrolled by C1. Four host proteins have been found to beessential for P1 lytic replication: DNA primase (DnaG), the FIG. 3. P1-encoded tRNAs and codons presumably recognized by them. The identity determinants of tRNAs are marked by asterisks. The putative modi ﬁcation of tRNA3 by lysinylation at the 2 position of cytosine alters the recognition speci ﬁcity of the anticodon. The bar graphs compare the usage frequencies of codons recognized by P1 tRNAs in protein-coding genes of P1 (black bars) and E. coli K-12 (grey bars) per 1,000 codons. The usage frequency of a particular codon in P1 is its relative abundance among codons in genes that encode proteins. The usage frequencyof a particular codon in E. coli K-12 is according to the March 2004 edition of the Codon Usage Database found at its website (http: //www.kazusa.or.jp/codon/).V OL. 186, 2004 BACTERIOPHAGE P1 7051', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  host helicase-loading factor (DnaC), and the DnaE and DnaX subunits of the DNA polymerase III holoenzyme (119, 234).This list of essential host proteins is probably incomplete. The repL gene is in an operon with the preceding gene, kilA, whose function is unclear (112, 295). When expressed, kilA is lethal to E. coli . Its product is dispensable for phage replica- tion. The 180-nt antisense RNA transcribed from the consti-tutive Pas kilA promoter internal to kilA interferes with tran- scription of the kilA repL operon initiated from P kilA (125). It may counteract any leakiness of the C1-mediated repression ofPkilA in P1 lysogens. The activity of Pas kilA is much stronger on linear than on circular templates. Thus, it was proposed thattranscription from Pas kilA during lytic development serves to down-regulate (the initiation of) P1 DNA replication at twostages: immediately after infection before the P1 DNA is cir-cularized and late after infection when rolling-circle replicationpredominates and linear P1 DNA concatemers accumulate(125). Presumably the origin region is transcriptionally activated (61, 295). A candidate promoter providing the activation, aswell as the initiator message, is the putative P repL promoter, which we ﬁnd directly upstream of repL (Table 4). The ssbgene of P1 is in an operon with a downstream gene of unknown function, isaA, which does not have a separate promoter sequence (Fig. 1). The isaA gene probably has no essential function as it is absent from the ssboperon of the closely related P7 phage (M. Łobocka, unpublished results). The cloned P1 ssbgene can substitute for E. coli ssb (28). A large P1 gene able to complement the defect in dam mutants of E. coli (originally dam, renamed here dmt) was shown to be directly under C1 control (59). We ﬁnd it to be the last gene in a three-gene operon. The ﬁrst gene of this operon (uhr) encodes a small protein without homology to proteins of known function. The second gene encodes a 34-kDa proteinthat appears to be a homolog of the E. coli nucleoid-associated protein RdgC (233), and thus we designate it hrdC . Proteins of the RdgC family are conserved in several species of bacteria.RdgC was proposed to be an exonuclease that is involved inthe removal of stalled replication forks, based on its beingrequired for proper replication in cells de ﬁcient in the recom- bination enzymes RecABC and SbcCD (263). The gonococcalhomolog of RdgC plays a role in pilin antigenic variation,suggesting that it may have a role in site-speci ﬁc DNA recom- bination (218). The amino acid sequence of the P1 RdgChomolog contains a motif characteristic of the active site of E. coliDNA polymerase I and related polymerases (see Table S1 in the supplemental material) (39, 74, 160; reviewed in refer-ence 243). The function of HrdC, which is one-third the size ofDNA polymerase I (PolI), remains to be elucidated. The P1 homolog of E. coli Dam methyltransferase is a 754- residue protein. Its homologies to Dam are contained solelywithin the C-terminal region of 266 residues. The major part ofthe protein is highly homologous to C-5 cytosine methyltrans-ferases of various bacteria and bacteriophages. Most likely theprotein is able to methylate DNA at both the N-6 positions ofadenines and the C-5 positions of cytosines. Accordingly, wedesignate it Dmt (DNA methyltransferase). InE. coli , the Dam methyltransferase has multiple functions. It is involved in postreplicative methyl-directed mismatch re-pair, contributes to the control of initiation of DNA replica-tion, and regulates the transcription of certain genes (reviewed in reference 255). Functions of the adenine methyltransferaseactivity of the P1-encoded Dmt protein are apparent only in adam host.', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Its homologies to Dam are contained solelywithin the C-terminal region of 266 residues. The major part ofthe protein is highly homologous to C-5 cytosine methyltrans-ferases of various bacteria and bacteriophages. Most likely theprotein is able to methylate DNA at both the N-6 positions ofadenines and the C-5 positions of cytosines. Accordingly, wedesignate it Dmt (DNA methyltransferase). InE. coli , the Dam methyltransferase has multiple functions. It is involved in postreplicative methyl-directed mismatch re-pair, contributes to the control of initiation of DNA replica-tion, and regulates the transcription of certain genes (reviewed in reference 255). Functions of the adenine methyltransferaseactivity of the P1-encoded Dmt protein are apparent only in adam host. In such a host, P1 dmt mutants that are defective in adenine methylation of 5 /H11032-GATC sequences show a dimin- ished burst size of 5 to 10 viable phage per cell (303). One ofthe two sequenced P1 strains contains a 319-bp deletion (for-merly designated dam/H9004MB) in the region of dmtthat codes for the N-6 methyltransferase domain. Decreased speci ﬁcity of the packaging endonuclease, resulting in impaired cleavage of P1DNA at the pac site and increased cleavage of host DNA, could account for the low phage yield as well as the increasedyield of transducing particles that has led to widespread use ofthe mutant phage. In order to ascertain possible targets of Dmt-mediated ad- enine methylation, we searched for 5 /H11032-GATC sequences in the entire sequence of P1. About 20 clusters of two or three 5 /H11032- GATC sites were found, including the few that had been de-scribed previously (Fig. 1). One cluster is in the region of thelytic origin (30 nt from a predicted DnaA box). The location ofanother suggests that it regulates the kilA repL genes and controls transcriptional activation of the origin region. Twoother clusters are internal to genes of the rlfoperon, down- stream of repL , where an additional origin of P1 lytic replica- tion may be located. One that is internal to rlfA isﬂanked by putative IHF binding sites. The predicted strongest promoterin the genome of P1 (Pas rlf), which is located immediately downstream of the rlfAB genes and could drive transcription of RNA antisense to the mRNA of the rlfAB operon, overlaps another cluster of 5 /H11032-GATC sites and thus also appears to be controlled by DNA methylation. Several additional clusterswithin coding sequences, such as the 5 /H11032-GATC sequences that are internal to pacA , may be associated with particular stages of phage development. Of single 5 /H11032-GATC sites, 10 overlap known or putative promoters of P1 genes (Table 4) and pre-sumably affect their function, as was shown for the 5 /H11032-GATC sites that overlap certain promoters of cre(302). The list of genes that are most likely controlled by methylation of 5 /H11032- GATC sites may thus be extended to include c1,coi,rlfB, and genes of the simand sitoperons. The bangene (67, 237), which can complement the replica- tive helicase defect of E. coli dnaB mutants, was mapped to a 1.6-kb region located 2.5 kb downstream of the bivalent C1operator O banab (formerly Op72ab; Table 6) (132, 316). It encodes a 455-residue protein (195) identical in 78% of itsamino acid sequence to E. coli DnaB. The Ban protein cross- reacts with anti-DnaB serum (129) and forms active thermallystabilized heteromultimers with the DnaB protein of E. coli dnaB (Ts) mutants (82, 183, 310). Although ban suppresses a number of conditionally lethal dnaB mutations in E. coli (67), the functional replacement of DnaB by Ban is somewhat in-complete, as indicated by the thermosensitivity of a dnaB de- letion strain that carries the ban gene (195).', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  It encodes a 455-residue protein (195) identical in 78% of itsamino acid sequence to E. coli DnaB. The Ban protein cross- reacts with anti-DnaB serum (129) and forms active thermallystabilized heteromultimers with the DnaB protein of E. coli dnaB (Ts) mutants (82, 183, 310). Although ban suppresses a number of conditionally lethal dnaB mutations in E. coli (67), the functional replacement of DnaB by Ban is somewhat in-complete, as indicated by the thermosensitivity of a dnaB de- letion strain that carries the ban gene (195). Mutations exist which permit growth of this strain at the nonpermissive tem-perature (E. Lanka, personal communication). Whether theyreside in the chromosome or in the ban gene remains to be determined. No phenotype was associated with a banmutation of P1 in a dnaB /H11001E. coli strain. Presumably bancan provide a selective advantage to P1 in other hosts or under other condi- tions.7052 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 22 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The ban gene is the penultimate gene in a 5-kb operon of unusual structure (Fig. 1). Immediately downstream of thepromoter (designated P1 ban) is a 1.5-kb region that apparently speci ﬁes untranslated RNA, including tRNA1 and tRNA2. This region is followed by two clusters of ORFs separated byanother region (0.3 kb) that speci ﬁes untranslated RNA, in- cluding tRNA3. The ban gene is the ﬁrst of two ORFs in the second ORF cluster. The ﬁrst ORF cluster upstream of ban, named by us tciABC , and the ORF downstream of ban, named by us dbn, encode proteins that have no homologies with known replication-as-sociated proteins. The predicted products of tciA andtciB have homologs of unknown function in various prophages (see Ta-ble S1 in the supplemental material). P1 TciA has signi ﬁcant similarities to the TerB proteins of R478 and other plasmids ofthe IncHI2 incompatibility group. The terB genes in those plasmids are internal genes of multigenic operons that conferupon E. coli resistance to tellurite and to channel-forming colicins and cause inhibition of cell division and of the propa-gation of certain phages (163, 260, 331, 332). (c) Recombination enhancement. The product of the P1 gene ref(recombination enhancement function) can stimulate RecA-dependent recombination between regions of DNA ho-mology shorter than the minimum needed for recombinationmediated by RecA alone (207, 337, 338). The proximity of ref tocremight suggest that Ref plays a role in the cyclization of P1 DNA molecules that, lacking two copies of lox, are not substrates of the cyclization recombinase, Cre. However, with-in the limits of the assays, no enhancement of P1 cre /H11002lysog - enization by Ref could be detected. The substrates of Refappear con ﬁned to regions of microhomology (337, 338). Regulation of refis complex (338). One presumptive pro- moter, 91 bp upstream of the ﬁrst ATG of ref, is overlapped by a C1-recognition sequence. A second presumptive promoter(which we locate closer to refthan originally proposed) lies within a region that could fold into a long stem-loop andprevent formation of a terminator stem-loop further down-stream. It is likely that transcripts from the C1-controlled pro-moter form the antiterminator structure, whereas transcripts from the C1-independent promoter would be blocked by theterminator (338). In addition, translation of refmessage can be from either of two AUG codons 46 amino acid residues apart,although the shorter product has low activity. The bene ﬁtt oP 1 that refconfers remains to be determined. (d) Late transcription activation. With a few exceptions, genes involved in late functions are preceded by late promotersequences (Table 8) and are partly or entirely dependent onthe product of lpa(late promoter activator) for their transcrip- tion (188, 193). The lpagene is under the control of a C1- controlled promoter, P lpa. Early in lytic development, it is cotranscribed with two genes, pacA and pacB , that directly follow lpaand that are additionally transcribed from the late promoter LP pac(111). The sites recognized by the Lpa protein and requirements for its function are considered in the section“Lpa-controlled operons. ” (e) Morphogenetic functions that start to be expressed early in lytic development. Among 17 P1 operons controlled by C1, ﬁve (mat,26,ppp,pap, and pmgT ) encode morphogenetic func- tions (Fig. 1 and Table 3; see also Table S1 in the supplementalmaterial). Although early expression of these operons mayseem surprising, it is noteworthy that, in addition to beingcontrolled by C1, they all appear to be activated by Lpa. Twoof them, mat and 26, contain in their immediate upstream regulatory regions, in addition to a C1-controlled /H9268 70pro- moter, an Lpa-activated late promoter.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The superimmunity (or superinfection exclusion) phenotype was observed in the presence of SimC alone (213). The SimCprotein appears to be an analog of superinfection exclusionproteins that act to prevent injection of superinfecting phageDNA into the cytoplasm of the infected cell. Many phages, both temperate and virulent, encode such proteins. SimC ap-pears to act in the periplasmic space, where the processed formis found, or in the cytoplasmic membrane, similarly to the SieAprotein of P22 (139) and the Imm protein of T4 (205), eitherby helping to destroy the injected DNA or by preventing itsentry into the cytoplasm (139, 170, 206, 213). The processing ofSimC requires SecA (213), as probably does the processing ofSimA and SimB, which, like SimC, have putative signal peptides. Products of the simoperon, like the product of sieA of P22 (139, 307), exclude both phage and transducing particle DNA.However, exclusion of the latter by Sim is much less ef ﬁcient, indicating that the Sim system can discriminate between P1and foreign DNA. Conceivably, SimC or another Sim proteincan interact with a P1 protein that is bound speci ﬁcally with phage DNA as it enters a cell during infection. Cells carrying low-copy-number sim /H11001plasmids display a Sim/H11002phenotype (77, 170), which appears inconsistent with a role of the simgenes in lysogeny. It is possible that in P1 ly- sogens, the simfunctions are induced only under certain cir- cumstances or that the main role of Sim proteins is to protectfrom superinfection those cells in which lytic development hasbeen initiated and the simgene dosage is high. The physiolog- ical role of simfunctions could be analogous to that of super- infection exclusion functions encoded by the immT and sp genes of the lytic phage T4. It has been proposed that super-infection exclusion mechanisms may preserve genetic diversityamong phages, protecting phage populations from dominanceby a phage that would otherwise outcompete any related phagewhen given a possibility to propagate in a superinfected cell(205). (v) HumD and Lxr. The monocistronic operon immediately upstream of the P1 addiction module encodes HumD protein,a homolog of E. coli UmuD /H11032(200). The P1 humD gene, like E. coli umuD , is transcribed from a LexA-regulated promoter (Tables 4 and 7). P1 lacks a homolog of umuC , which in E. coli is functionally associated and cotranscribed with umuD . Aside from recA ,umuD and umuC ofE. coli are the only LexA-regulated genes of the SOS response that are requiredfor DNA damage-induced mutagenesis (290). They encode alow-ﬁdelity and low-processivity DNA polymerase, PolV, whose main function is to bypass DNA lesions (309). A RecA-mediated autocleavage of the umuD gene product (215) is required to form active PolV, which is a complex of UmuD /H11032 and UmuC in a 2:1 ratio (43). P1 HumD corresponds to theprocessed form of UmuD, UmuD /H11032, and can functionally re- place it (217). The regulatory region of humD overlaps the regulatory re- gion of a divergently transcribed gene of unknown function.This gene could encode a 190-residue, slightly acidic proteinthat has no homologs in databases. We surmise that its tran-scription, like that of humD , is negatively regulated by LexA. One of two putative /H9268 70promoters of this gene overlaps the LexA binding site that controls humD (Tables 4 and 7), and thus we name the gene lxr(LexA regulated). Genes expressed in lytic development. Expression of the majority of P1 genes during lytic growth follows a strict tem-poral pattern that can lead to the production of mature phageparticles within less than 1 h. For many bacteriophages such asT4 (180), Mu (18, 55), P2 (339), and P4 (98), the regulatoryVOL. 186, 2004 BACTERIOPHAGE P1 7049', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  cascade has been subdivided into early, middle, and late stages. In P1, the cascade appears to be simpler; early transcriptionswitches directly to late transcription, without a well-de ﬁned intermediate stage (188, 193). C1 operator sequences, whichbind the primary phage repressor C1, act as a switch of earlygenes. Lpa (late promoter activator) binding sequences, whichare in the /H1100222 region of late promoters and enable RNA polymerase to initiate transcription upon binding of Lpa, act asa switch of late genes. (i) C1-controlled operons. Analysis of the entire P1 genome has revealed 17 operons that have C1 operators in the regionof their /H9268 70promoter sequences or, in some cases, between promoters and proximal genes (Table 6). One of these operons (pmgR ) is preceded by two nonoverlapping operators, and two others ( ban and c1) are preceded by operators that are biva- lent, bringing the total number of monovalent C1-binding sitesto 20, all of which were identi ﬁed previously. Although we did not ﬁnd any additional sequences with strong similarity to known C1-binding sites, sequences that resemble the C1-bind-ing sites but are missing the highly conserved C at position 7,are present in the parS site and in the promoter region close to thedarB gene. The sequence in parS did not interact with C1 in vitro (N. Sternberg, personal communication). Whetherthese sequences can act cooperatively with other C1 operatorsequences to bind C1 in vivo, as does the O c1b operator, which also has the C at position 7 replaced by another base, remainsto be seen. Looping between C1-bound operators located closeto each other has been demonstrated (128), but looping be-tween well-separated operator sequences has not. The 17 C1-controlled operons, transcribed from /H9268 70promot - ers, contain 49 genes that are derepressed early in lytic devel-opment. Transcription does not start simultaneously, since theoperators exhibit different af ﬁnities for C1 and are differently inﬂuenced by the Lxc protein, which modulates the C1-oper- ator interaction. Lytic replication genes start to be transcribedwithin 5 min of infection, allowing a prompt initiation of phageDNA replication. Increased amounts of P1 DNA can be de-tected about 15 min after infection, at about the same time ascleavage of the packaging site, pac, in P1 DNA (297). Detec- tion of DNA synthesis within 5 min has also been reported(282). Expression of other early functions was reported to start10 to 15 min after induction of P1 lysogens (187). The intensetranscription of P1 early functions is attenuated later in lyticdevelopment, prior to transcriptional activation of late genes(111). The attenuation depends on the E. coli RNA poly- merase-associated protein SspA. Whether SspA acts directlyon complexes of RNA polymerase with P1 /H9268 70promoter se - quences or requires the action of a P1 protein is unknown. Wedid not ﬁnd any conserved sequences, other than promoter sequences and C1-binding sites, in the regulatory regions of P1early genes. (a) P1-encoded tRNAs. Several phages encode their own tRNAs (e.g., see references 88, 177, and 247); the T4-likevibriophage KVP40 encodes at least 25 tRNAs at a single locus(223). Some of these tRNAs supplement the host pool of raretRNAs to facilitate ef ﬁcient expression of selected phage genes during lysogeny or at certain stages of lytic development (179,247). Additionally, it has been proposed that phage tRNAs canprovide selective pressure to keep base composition and codonusage of phage DNA signi ﬁcantly different from those of itshost (summarized in reference 227). We ﬁnd that the P1 ge- nome contains three sequences characteristic of tRNA genes.All three are located between the ban gene and its promoter, in the apparently untranslated regions of the banoperon. Two are adjacent; one is further downstream, separated from thetwo by three protein-coding genes.', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Some of these tRNAs supplement the host pool of raretRNAs to facilitate ef ﬁcient expression of selected phage genes during lysogeny or at certain stages of lytic development (179,247). Additionally, it has been proposed that phage tRNAs canprovide selective pressure to keep base composition and codonusage of phage DNA signi ﬁcantly different from those of itshost (summarized in reference 227). We ﬁnd that the P1 ge- nome contains three sequences characteristic of tRNA genes.All three are located between the ban gene and its promoter, in the apparently untranslated regions of the banoperon. Two are adjacent; one is further downstream, separated from thetwo by three protein-coding genes. Control of the P1 tRNAgenes from a promoter repressed by C1 implies that they arenot expressed during lysogeny. The proposed cloverleaf structures of all predicted P1 tRNAs are shown in Fig. 3. The ﬁrst tRNA, tRNA1, has the anticodon GUU, characteristic of E. coli tRNA Asn.I ti s9 0 % identical to the predicted tRNAAsnof the Salmonella enterica serovar Typhi plasmid pHCM2 (242) (GenBank accession no. AL513384) and contains all bases essential for the aminoacy-lation of tRNA AsninE. coli : the anticodon bases G34, U35, and U36 and the discriminator base G73 (201). The second tRNA, tRNA2, has the anticodon UGU, char- acteristic of E. coli tRNAThr. Although it is highly homologous to alanine tRNA-UGC of a cyanobacterium, Synechocystis sp. (SYCSLRB; 83% identity), and to tRNAAlaof numerous other organisms, it contains neither the G3-U70 wobble pair, which is the primary determinant of the acceptor identity of theE. coli tRNA Ala, nor the discriminator base G20, which is also an important characteristic of this tRNA (250, 308). Instead, in addition to the anticodon bases G35 and U36, which are iden-tity determinants of all E. coli tRNA Thrs, it contains base pairs G1-C72 and C2-G71, which in E. coli tRNAThr, are crucial for threonine charging activity (116). The third tRNA, tRNA3, has the anticodon CAU, which should correspond to the Met codon AUG. However, in itssequence this tRNA appears to be 89% identical to the E. coli tRNA Ile-2, encoded by the ileX gene. None of the differences (outside the anticodon) between tRNA3 and tRNAIle-2 affect residues known to be essential for the function of E. coli tRNAIle-2. In the tRNAIle-2 of E. coli the wobble position C34 is modi ﬁed by lysinylation to lysidine (i.e., 2-lysylcytidine). The modi ﬁed anticodon acquires the AUA (Ile)-decoding capacity and is required for the recognition of this tRNA by isoleucyl-tRNA synthetase (231, 232, 283). The other identifying char-acteristics of E. coli tRNA Ileare the anticodon loop bases A37 and A38, the discriminator base A73, and the base pairs C4- G69, U12-A23, and C29-G41 (236, 240). All of them are pres-ent in P1 tRNA3. Certain other phages, including 933W (247),T4 (224), and the T4-like phages RB69 (see http://phage.bioc.tulane.edu/) and KVP40 (223) also encode homologs of E. coli tRNA Ile-2, predicted to contain the lysylcytidine modi ﬁcation at C34. The T4 homolog of E. coli tRNAIle-2 was con ﬁrmed to have the capacity to decode ATA (274). Of the codons that are apparently recognized by P1 tRNAs, two, ACA and ATA, are rare in E. coli genes but overrepre- sented in certain P1 genes (Fig. 3).', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  All of them are pres-ent in P1 tRNA3. Certain other phages, including 933W (247),T4 (224), and the T4-like phages RB69 (see http://phage.bioc.tulane.edu/) and KVP40 (223) also encode homologs of E. coli tRNA Ile-2, predicted to contain the lysylcytidine modi ﬁcation at C34. The T4 homolog of E. coli tRNAIle-2 was con ﬁrmed to have the capacity to decode ATA (274). Of the codons that are apparently recognized by P1 tRNAs, two, ACA and ATA, are rare in E. coli genes but overrepre- sented in certain P1 genes (Fig. 3). The ACA codon is therarest threonine codon in E. coli , but its effect on translation has not been studied. The ATA codon, which is the ﬁfth rarest codon in E. coli , is known to dramatically decrease translation of those E. coli mRNAs that contain it, especially when present in multiple copies, or in tandem, or in a single copy in theanterior part of a gene (99, 166, 353). The ATA codons areoverrepresented more than twofold in 59 of 113 protein-codinggenes of P1 as compared to their representation in E. coli genes. Four of these 59 genes ( isaB,rlfA,7, and pacB ) contain7050 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  both single ATAs and tandem ATAs. In eight, ATAs are among the ﬁrstﬁve codons of the proximal region (at position 5o f lydC ,4o f 21,2o f ppfA ,3o f tciB,2o f ppp,4o f doc,4o f pdcB ,a n d2o f c1). At least two of these genes, docandc1, are known to be expressed, albeit not ef ﬁciently, during lysogeny, indicating that the low levels of ileX tRNA that are normally present in E. coli cells are suf ﬁcient for their translation, since tRNA3 is not expressed. However, the known translation-lim-iting role of ATA codons in E. coli implies that the ef ﬁciency of translation of doc,c1, and other genes rich in the ATA codons increases under conditions of abundance of P1 tRNA3. In thecase of doc and c1, the physiological signi ﬁcance of such an increase, if it does occur, is unclear. In the case of certain othergenes, the requirement for abundant tRNA3 to permit theirefﬁcient expression may provide an additional control on the timing of P1 lytic development at the translational level. All P1tRNA genes appear to be transcribed from the C1-controlledpromoter of the ban operon and so become available early during lytic development. It is noteworthy that lpa, whose product activates transcrip- tion of P1 late genes, has in its proximal region three rareE. coli codons, recognized by P1 tRNAs, namely ATA at po- sition 13 and an ATA ACA pair at positions 18 and 19.Whether the abundance of P1 tRNA3, and perhaps tRNA2,early in P1 lytic development normally contributes to ef ﬁcient translation of lpamRNA, which in turn ensures ef ﬁcient tran- scription of late genes, remains to be tested. Insertions of numerous phages into bacterial chromosomes occur at or near tRNA genes (reviewed in reference 46). Con-ceivably, tRNA genes could have been acquired by a P1 an-cestor either through its integration and then imperfect exci-sion from the DNA of its host or via recombination from otherphages whose prophages can integrate into a bacterial chro- mosome. (b) C1-controlled replication functions. P1 DNA synthesis has been reported to start as early as 5 min after infection of E. coli(282). Lytic replication initiates bidirectionally in the theta mode from an origin, oriL, distinct from the plasmid origin and located within the essential replication gene, repL (61, 112, 295). Later, lytic replication switches into the rolling-circlemode (60). The shift is accompanied by a gradual decrease inaccumulation of /H9258-shaped replication intermediates as /H9268- shaped intermediates accumulate. Most likely oriL is within the second half of the repL gene, where we ﬁnd putative binding sites for DnaA and IHF pro- teins and two 5 /H11032-GATC sequences (Table 7). P1 phage repli- cation in the rolling-circle ( /H9268) mode, predominant in the later stages of phage development, might initiate from an originother than oriL. One possible location for an additional P1 origin of lytic replication is within the rlfoperon, which directly follows the repL gene. It contains multiple sets of GATC se- quences and several IHF binding sites, which are internal to itstwo protein-coding regions (Table 7). An alternative proposalis that rolling-circle replication starts at pacfrom a nick intro- duced in DNA by pacase (296). In addition to the essential replication protein, RepL, P1 encodes homologs of certain E. coli replication-associated pro- teins: DnaB helicase (67, 237), single-stranded DNA-bindingprotein, Ssb (164, 187), Dam methyltransferase (66), and thetheta subunit of DNA polymerase III. Genes encoding thesehomologs ( ban,ssb,dmt, and hot, respectively), are in separate operons and, with the exception of the last mentioned, arecontrolled by C1.', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  It contains multiple sets of GATC se- quences and several IHF binding sites, which are internal to itstwo protein-coding regions (Table 7). An alternative proposalis that rolling-circle replication starts at pacfrom a nick intro- duced in DNA by pacase (296). In addition to the essential replication protein, RepL, P1 encodes homologs of certain E. coli replication-associated pro- teins: DnaB helicase (67, 237), single-stranded DNA-bindingprotein, Ssb (164, 187), Dam methyltransferase (66), and thetheta subunit of DNA polymerase III. Genes encoding thesehomologs ( ban,ssb,dmt, and hot, respectively), are in separate operons and, with the exception of the last mentioned, arecontrolled by C1. Four host proteins have been found to beessential for P1 lytic replication: DNA primase (DnaG), the FIG. 3. P1-encoded tRNAs and codons presumably recognized by them. The identity determinants of tRNAs are marked by asterisks. The putative modi ﬁcation of tRNA3 by lysinylation at the 2 position of cytosine alters the recognition speci ﬁcity of the anticodon. The bar graphs compare the usage frequencies of codons recognized by P1 tRNAs in protein-coding genes of P1 (black bars) and E. coli K-12 (grey bars) per 1,000 codons. The usage frequency of a particular codon in P1 is its relative abundance among codons in genes that encode proteins. The usage frequencyof a particular codon in E. coli K-12 is according to the March 2004 edition of the Codon Usage Database found at its website (http: //www.kazusa.or.jp/codon/).V OL. 186, 2004 BACTERIOPHAGE P1 7051', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  host helicase-loading factor (DnaC), and the DnaE and DnaX subunits of the DNA polymerase III holoenzyme (119, 234).This list of essential host proteins is probably incomplete. The repL gene is in an operon with the preceding gene, kilA, whose function is unclear (112, 295). When expressed, kilA is lethal to E. coli . Its product is dispensable for phage replica- tion. The 180-nt antisense RNA transcribed from the consti-tutive Pas kilA promoter internal to kilA interferes with tran- scription of the kilA repL operon initiated from P kilA (125). It may counteract any leakiness of the C1-mediated repression ofPkilA in P1 lysogens. The activity of Pas kilA is much stronger on linear than on circular templates. Thus, it was proposed thattranscription from Pas kilA during lytic development serves to down-regulate (the initiation of) P1 DNA replication at twostages: immediately after infection before the P1 DNA is cir-cularized and late after infection when rolling-circle replicationpredominates and linear P1 DNA concatemers accumulate(125). Presumably the origin region is transcriptionally activated (61, 295). A candidate promoter providing the activation, aswell as the initiator message, is the putative P repL promoter, which we ﬁnd directly upstream of repL (Table 4). The ssbgene of P1 is in an operon with a downstream gene of unknown function, isaA, which does not have a separate promoter sequence (Fig. 1). The isaA gene probably has no essential function as it is absent from the ssboperon of the closely related P7 phage (M. Łobocka, unpublished results). The cloned P1 ssbgene can substitute for E. coli ssb (28). A large P1 gene able to complement the defect in dam mutants of E. coli (originally dam, renamed here dmt) was shown to be directly under C1 control (59). We ﬁnd it to be the last gene in a three-gene operon. The ﬁrst gene of this operon (uhr) encodes a small protein without homology to proteins of known function. The second gene encodes a 34-kDa proteinthat appears to be a homolog of the E. coli nucleoid-associated protein RdgC (233), and thus we designate it hrdC . Proteins of the RdgC family are conserved in several species of bacteria.RdgC was proposed to be an exonuclease that is involved inthe removal of stalled replication forks, based on its beingrequired for proper replication in cells de ﬁcient in the recom- bination enzymes RecABC and SbcCD (263). The gonococcalhomolog of RdgC plays a role in pilin antigenic variation,suggesting that it may have a role in site-speci ﬁc DNA recom- bination (218). The amino acid sequence of the P1 RdgChomolog contains a motif characteristic of the active site of E. coliDNA polymerase I and related polymerases (see Table S1 in the supplemental material) (39, 74, 160; reviewed in refer-ence 243). The function of HrdC, which is one-third the size ofDNA polymerase I (PolI), remains to be elucidated. The P1 homolog of E. coli Dam methyltransferase is a 754- residue protein. Its homologies to Dam are contained solelywithin the C-terminal region of 266 residues. The major part ofthe protein is highly homologous to C-5 cytosine methyltrans-ferases of various bacteria and bacteriophages. Most likely theprotein is able to methylate DNA at both the N-6 positions ofadenines and the C-5 positions of cytosines. Accordingly, wedesignate it Dmt (DNA methyltransferase). InE. coli , the Dam methyltransferase has multiple functions. It is involved in postreplicative methyl-directed mismatch re-pair, contributes to the control of initiation of DNA replica-tion, and regulates the transcription of certain genes (reviewed in reference 255). Functions of the adenine methyltransferaseactivity of the P1-encoded Dmt protein are apparent only in adam host.', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Its homologies to Dam are contained solelywithin the C-terminal region of 266 residues. The major part ofthe protein is highly homologous to C-5 cytosine methyltrans-ferases of various bacteria and bacteriophages. Most likely theprotein is able to methylate DNA at both the N-6 positions ofadenines and the C-5 positions of cytosines. Accordingly, wedesignate it Dmt (DNA methyltransferase). InE. coli , the Dam methyltransferase has multiple functions. It is involved in postreplicative methyl-directed mismatch re-pair, contributes to the control of initiation of DNA replica-tion, and regulates the transcription of certain genes (reviewed in reference 255). Functions of the adenine methyltransferaseactivity of the P1-encoded Dmt protein are apparent only in adam host. In such a host, P1 dmt mutants that are defective in adenine methylation of 5 /H11032-GATC sequences show a dimin- ished burst size of 5 to 10 viable phage per cell (303). One ofthe two sequenced P1 strains contains a 319-bp deletion (for-merly designated dam/H9004MB) in the region of dmtthat codes for the N-6 methyltransferase domain. Decreased speci ﬁcity of the packaging endonuclease, resulting in impaired cleavage of P1DNA at the pac site and increased cleavage of host DNA, could account for the low phage yield as well as the increasedyield of transducing particles that has led to widespread use ofthe mutant phage. In order to ascertain possible targets of Dmt-mediated ad- enine methylation, we searched for 5 /H11032-GATC sequences in the entire sequence of P1. About 20 clusters of two or three 5 /H11032- GATC sites were found, including the few that had been de-scribed previously (Fig. 1). One cluster is in the region of thelytic origin (30 nt from a predicted DnaA box). The location ofanother suggests that it regulates the kilA repL genes and controls transcriptional activation of the origin region. Twoother clusters are internal to genes of the rlfoperon, down- stream of repL , where an additional origin of P1 lytic replica- tion may be located. One that is internal to rlfA isﬂanked by putative IHF binding sites. The predicted strongest promoterin the genome of P1 (Pas rlf), which is located immediately downstream of the rlfAB genes and could drive transcription of RNA antisense to the mRNA of the rlfAB operon, overlaps another cluster of 5 /H11032-GATC sites and thus also appears to be controlled by DNA methylation. Several additional clusterswithin coding sequences, such as the 5 /H11032-GATC sequences that are internal to pacA , may be associated with particular stages of phage development. Of single 5 /H11032-GATC sites, 10 overlap known or putative promoters of P1 genes (Table 4) and pre-sumably affect their function, as was shown for the 5 /H11032-GATC sites that overlap certain promoters of cre(302). The list of genes that are most likely controlled by methylation of 5 /H11032- GATC sites may thus be extended to include c1,coi,rlfB, and genes of the simand sitoperons. The bangene (67, 237), which can complement the replica- tive helicase defect of E. coli dnaB mutants, was mapped to a 1.6-kb region located 2.5 kb downstream of the bivalent C1operator O banab (formerly Op72ab; Table 6) (132, 316). It encodes a 455-residue protein (195) identical in 78% of itsamino acid sequence to E. coli DnaB. The Ban protein cross- reacts with anti-DnaB serum (129) and forms active thermallystabilized heteromultimers with the DnaB protein of E. coli dnaB (Ts) mutants (82, 183, 310). Although ban suppresses a number of conditionally lethal dnaB mutations in E. coli (67), the functional replacement of DnaB by Ban is somewhat in-complete, as indicated by the thermosensitivity of a dnaB de- letion strain that carries the ban gene (195).', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  It encodes a 455-residue protein (195) identical in 78% of itsamino acid sequence to E. coli DnaB. The Ban protein cross- reacts with anti-DnaB serum (129) and forms active thermallystabilized heteromultimers with the DnaB protein of E. coli dnaB (Ts) mutants (82, 183, 310). Although ban suppresses a number of conditionally lethal dnaB mutations in E. coli (67), the functional replacement of DnaB by Ban is somewhat in-complete, as indicated by the thermosensitivity of a dnaB de- letion strain that carries the ban gene (195). Mutations exist which permit growth of this strain at the nonpermissive tem-perature (E. Lanka, personal communication). Whether theyreside in the chromosome or in the ban gene remains to be determined. No phenotype was associated with a banmutation of P1 in a dnaB /H11001E. coli strain. Presumably bancan provide a selective advantage to P1 in other hosts or under other condi- tions.7052 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 22 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The ban gene is the penultimate gene in a 5-kb operon of unusual structure (Fig. 1). Immediately downstream of thepromoter (designated P1 ban) is a 1.5-kb region that apparently speci ﬁes untranslated RNA, including tRNA1 and tRNA2. This region is followed by two clusters of ORFs separated byanother region (0.3 kb) that speci ﬁes untranslated RNA, in- cluding tRNA3. The ban gene is the ﬁrst of two ORFs in the second ORF cluster. The ﬁrst ORF cluster upstream of ban, named by us tciABC , and the ORF downstream of ban, named by us dbn, encode proteins that have no homologies with known replication-as-sociated proteins. The predicted products of tciA andtciB have homologs of unknown function in various prophages (see Ta-ble S1 in the supplemental material). P1 TciA has signi ﬁcant similarities to the TerB proteins of R478 and other plasmids ofthe IncHI2 incompatibility group. The terB genes in those plasmids are internal genes of multigenic operons that conferupon E. coli resistance to tellurite and to channel-forming colicins and cause inhibition of cell division and of the propa-gation of certain phages (163, 260, 331, 332). (c) Recombination enhancement. The product of the P1 gene ref(recombination enhancement function) can stimulate RecA-dependent recombination between regions of DNA ho-mology shorter than the minimum needed for recombinationmediated by RecA alone (207, 337, 338). The proximity of ref tocremight suggest that Ref plays a role in the cyclization of P1 DNA molecules that, lacking two copies of lox, are not substrates of the cyclization recombinase, Cre. However, with-in the limits of the assays, no enhancement of P1 cre /H11002lysog - enization by Ref could be detected. The substrates of Refappear con ﬁned to regions of microhomology (337, 338). Regulation of refis complex (338). One presumptive pro- moter, 91 bp upstream of the ﬁrst ATG of ref, is overlapped by a C1-recognition sequence. A second presumptive promoter(which we locate closer to refthan originally proposed) lies within a region that could fold into a long stem-loop andprevent formation of a terminator stem-loop further down-stream. It is likely that transcripts from the C1-controlled pro-moter form the antiterminator structure, whereas transcripts from the C1-independent promoter would be blocked by theterminator (338). In addition, translation of refmessage can be from either of two AUG codons 46 amino acid residues apart,although the shorter product has low activity. The bene ﬁtt oP 1 that refconfers remains to be determined. (d) Late transcription activation. With a few exceptions, genes involved in late functions are preceded by late promotersequences (Table 8) and are partly or entirely dependent onthe product of lpa(late promoter activator) for their transcrip- tion (188, 193). The lpagene is under the control of a C1- controlled promoter, P lpa. Early in lytic development, it is cotranscribed with two genes, pacA and pacB , that directly follow lpaand that are additionally transcribed from the late promoter LP pac(111). The sites recognized by the Lpa protein and requirements for its function are considered in the section“Lpa-controlled operons. ” (e) Morphogenetic functions that start to be expressed early in lytic development. Among 17 P1 operons controlled by C1, ﬁve (mat,26,ppp,pap, and pmgT ) encode morphogenetic func- tions (Fig. 1 and Table 3; see also Table S1 in the supplementalmaterial). Although early expression of these operons mayseem surprising, it is noteworthy that, in addition to beingcontrolled by C1, they all appear to be activated by Lpa. Twoof them, mat and 26, contain in their immediate upstream regulatory regions, in addition to a C1-controlled /H9268 70pro- moter, an Lpa-activated late promoter.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:39 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'86'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992373'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'457ms'), (b'x-request-id', b'7a12deed27a527c296a07b05bda5e8f7'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412491329f45d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:39 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'86'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992373'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'457ms'), (b'x-request-id', b'7a12deed27a527c296a07b05bda5e8f7'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412491329f45d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 22 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The sites recognized by the Lpa protein and requirements for its function are considered in the section“Lpa-controlled operons. ” (e) Morphogenetic functions that start to be expressed early in lytic development. Among 17 P1 operons controlled by C1, ﬁve (mat,26,ppp,pap, and pmgT ) encode morphogenetic func- tions (Fig. 1 and Table 3; see also Table S1 in the supplementalmaterial). Although early expression of these operons mayseem surprising, it is noteworthy that, in addition to beingcontrolled by C1, they all appear to be activated by Lpa. Twoof them, mat and 26, contain in their immediate upstream regulatory regions, in addition to a C1-controlled /H9268 70pro- moter, an Lpa-activated late promoter. Three others, pmgN, pmgR , and pmgT , are likely to be transcribed from an Lpa- activated promoter of the upstream pmgN gene by read- through transcription. An Ssp-dependent mechanism was found to turn down tran- scription from C1-controlled promoters before the Lpa-con-trolled promoters are activated (28). This ﬁnding indicates that, for those genes transcribed from a C1-controlled pro-moter, additional control from a late promoter is required toensure the continuity of their expression throughout lytic de-velopment and to satisfy the requirement late in developmentfor large amounts of certain proteins. Dual regulatory signalsTABLE 8. Known and putative late promoters of P1 Name PositionaStrandSequence Identi ﬁcationb Reference /H1100222 /H1100210 tsp pp cmp LPmat 2678 /H11001TTGACAACTGAC AAGTGACTT CAGT CAGAAT CATCACACGC /H12011 189 LPpro 16052 /H11001GTAATCCTTAAC AAGTGACTA GTGT TAAATT CCGTTCAAAC /H12011This work LPlyz 19770 /H11002AACGTAGCTAAC AAGTGATTT GCGT TATCCT GTGTCTTCTA /H12011/H12011 277 LPdarA 30013 /H11002TGACCGGTTCAA AAGTTACTT TGCA TACCAT TACCTCCTGA /H12011/H12011 105 LPS 37119 /H11002GGGCGGGGTGAC AAGTTACTT ATCT TACAAT GAGGCTTCAC /H12011/H12011 105 LPsit 45409 /H11002TACTCAATGAAC AAATGACTA CTCG TAGAAT CGGTTAACAC /H12011 H. Lehnherr, unpublished LP23 57626 /H11002GAGAAAGTCAAC AAGTGACTT TCAA TAAAAT CTCTTCCGAA /H12011This work LP26 77286 /H11002TAAATATTCAAC AAATGACTA GCGG TAGAAT CACCATCATC /H12011 275 LPpmgL 77296 /H11001TCATTCTACCGC TAGTCATTT GTTG AATATT TAACTCAATA /H12011 H. Lehnherr, unpublished LPhot 85322 /H11001AAAAAAAATTAA AAGTTACTT TGCTGGTTA AATAAT AGTCGTTACT /H12011/H12011 275 LPpac 89276 /H11001GGGCCTTTTGAC ATGTGACTT TCGT TACCCT CGCGTCAAAA /H12011 H. Lehnherr, unpublished aThe coordinate is that of the ﬁrst nucleotide of the promoter, that is, the ﬁrst nucleotide of the /H1100222 sequence for promoters read from the /H11001strand (clockwise) or the last nucleotide of the /H1100210 sequence for promoters read from the /H11002strand (counterclockwise). Coordinates refer to positions of promoters in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. Note that LP mat shares its /H1100210 sequence with P mat. bPromoter identi ﬁcation is based on a transcription start point determination ( tsp), detection with the promoter probe vector ( pp), or computer analysis of the sequence ( cmp).', 'page_label: 22 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Coordinates refer to positions of promoters in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. Note that LP mat shares its /H1100210 sequence with P mat. bPromoter identi ﬁcation is based on a transcription start point determination ( tsp), detection with the promoter probe vector ( pp), or computer analysis of the sequence ( cmp). The promoters distinguished by /H12011in the cmp column were identi ﬁed on the basis of the similarity of their sequences to the sequences of the other nine late promoters, using the GCG Fitconsensus program and the consensus table of /H1100222 sequences of the known late promoters made by the GCG program Consensus. Only the sequences with homology scores similar to the known promoters and located 4 to 14 nt (189) upstream from sequences similar to -10 he xamers of RNA polymerase promoters of E. coli (as estimated by the program Targsearch [229]) were considered.VOL. 186, 2004 BACTERIOPHAGE P1 7053', 'page_label: 23 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  in control regions of certain operons are common in T4 and other phages, where they can serve to extend expression of agiven gene to more than one developmental stage and to mod-ulate it differently at different stages (224, 228). P1matmutants produce viral particles with empty heads and unstable tails with tail sheaths in various stages of contraction(323). The pattern of mat expression following induction of a P1 lysogen is consistent with its being subject to dual control(191). Expression was seen to start within 10 to15 min ofinception of lytic development, to continue at a low level forabout 10 min, and then to increase sharply, exactly when theexpression of other genes activated by Lpa begins. Expressionproﬁles of the remaining genes that are controlled directly by both C1 and Lpa may be similar. As some products of theremaining C1- and Lpa-regulated operons are structural virionproteins, they are described later. Mat was proposed to have aregulatory role in morphogenesis (323), although with scantjustiﬁcation. The pacAB operon, which encodes two subunits of the P1 DNA packaging enzyme, is also subject to dual control. Al-though it is directly preceded only by the late promoter LP pac (Table 8), it is expressed early from the C1-controlled pro-moter of the upstream lpagene (105). Moreover, some PacAB- mediated cleavage of pacsites occurred in the absence of Lpa, indicating that Lpa is not essential for pacAB operon expres- sion during P1 lytic development (297). (ii) Lpa-controlled operons. P1 late promoters resemble typ- icalE. coli /H9268 70promoters at their /H1100210 regions but lack the /H1100235 hexamer (188). Instead, they have a conserved 9-bp inverted repeat (previously termed late operator) that is centered aboutposition /H1100222 of the transcription start site and interacts with Lpa (111, 189). Results of in vivo footprinting experimentsconﬁrm that RNA polymerase binds to late promoter se- quences in the presence, but not in the absence, of Lpa (111).In addition to Lpa, the E. coli RNA polymerase-associated protein, SspA (158), is required both in vivo and in vitro forlate promoter activation (111, 336). The activation requires the/H9268 70RNA polymerase holoenzyme and does not occur when /H926870is replaced by /H9268S. How Lpa and SspA cooperate in redi - recting the host RNA polymerase towards P1 late promotersequences is unknown. Nine P1 late promoters were identi ﬁed previously by empir- ical studies (Table 8). The optimal spacing between /H1100210 and /H1100222 regions of late promoters is 4 bp, and it is conserved in all but one of these promoters, LP hot. Although in the LP hot promoter the spacing (9 bp) is suboptimal, a promoter probeassay con ﬁrmed in vivo LP hottranscriptional activity to be dependent upon Lpa (275). Here we identi ﬁed two additional putative late promoters, based on the strong similarity of their predicted /H1100222 regions to the/H1100222 regions of known late promoters and on the presence of appropriately positioned putative /H1100210 hexamers (Table 8). Both promoters have the optimal spacing between their pre-dicted /H1100210 and /H1100222 regions, and both are located in the regulatory regions of genes known to encode phage morpho-genetic functions. The 11 late promoters control at least 12 andas many as 14 P1 operons. We suspect that transcription fromLPpmgL , before encountering the closest terminator, T pmgQ (Table 5), can read through two genes that are immediatelydownstream, pmgL and pmgM , and a C1-controlled operonfurther downstream that encodes morphogenetic functions (Fig. 1). It is likely that this transcript includes an additionaltwo operons downstream from T pmgQ that also encode mor- phogenetic functions. Expression of these operons late duringlytic development implies that P1 uses an ef ﬁcient antitermi- nation mechanism to attenuate the function of T pmgQ ,TpmgS , and T pap terminators.', 'page_label: 23 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The 11 late promoters control at least 12 andas many as 14 P1 operons. We suspect that transcription fromLPpmgL , before encountering the closest terminator, T pmgQ (Table 5), can read through two genes that are immediatelydownstream, pmgL and pmgM , and a C1-controlled operonfurther downstream that encodes morphogenetic functions (Fig. 1). It is likely that this transcript includes an additionaltwo operons downstream from T pmgQ that also encode mor- phogenetic functions. Expression of these operons late duringlytic development implies that P1 uses an ef ﬁcient antitermi- nation mechanism to attenuate the function of T pmgQ ,TpmgS , and T pap terminators. Attenuation of terminator functions may also occur at T 22,TpmgG , and T ddrA terminators, since these would otherwise prevent late expression of genes 22, pmgG ,pmgF ,ddrB , and hxr, encoding known or predicted late functions. Alternatively, these terminators may be inef ﬁcient and serve to decrease the expression of certain late genesdistant from the LP pmgL ,L P 23, and LP darlate promoters. One other late promoter, LP sit, could drive transcription of both the sitoperon and the following RSU operon. However, theRSU operon is also transcribed from its own late pro- moter, LP S. The remaining nine late promoters drive tran- scription of single operons that adjoin operons transcribed inthe opposite direction. In total, 52 genes of P1 can be tran-scribed from late promoter sequences. InE. coli at 37 °C, late gene products start to be synthesized 20 to 30 min after either transient thermal induction of aP1c1ts lysogen or infection of sensitive cells by P1. The com- pletion of phage morphogenesis at about min 40 is accompa-nied by the onset of cell lysis. The late genes of P1 are involvedin DNA replication, phage tail and head morphogenesis, pack-aging of DNA, synthesis and incorporation of injectable pro-teins, and cell lysis. Some of them, as explained above, areexpressed both early and late in lytic development. Some thatare apparently expressed in late developmental stages cannotbe associated with any late promoter sequences. The darB gene, encoding an antirestriction protein and a P1 head com-ponent, is an example. Transcription of these genes may becontrolled either indirectly by Lpa through another regulatoryprotein or by an as-yet-unidenti ﬁed mechanism. (a) Head morphogenesis. The morphogenesis of P1 resem- bles that of other tailed phages in that heads and tails haveindependent assembly pathways (323). At least some of thehead proteins appear to be proteolytically processed (304).DNA is packaged into proheads from concatemers (60) by aheadful mechanism, as in T4, but the packaging of each con-catemer, unlike that in T4, starts from an initial cut at a speci ﬁc pacsite (286, 297, 298). Infective particles of P1 contain cycli- cally permuted, linear, double-stranded molecules with a ter-minal redundancy of about 10 kb of DNA (296). P1 tail ﬁbers, unlike those of T4, can attach to the tail before head-tailjoining occurs (323). P1 virions resemble in many respects those of T4, although they are less complex, especially in the structures of the headand the baseplate (323). Mature P1 virions contain at least 28proteins (325), as compared to at least 49 proteins in maturevirions of T4 (summarized in reference 224). In T4, a furtherﬁve proteins assist in morphogenesis and are removed during virion maturation. Proteins of similar function in P1 awaitidenti ﬁcation. Analysis of P1 virion proteins by SDS-PAGE indicates that P1 heads are composed of 15 to 19 protein components, pos-sibly an underestimate as small proteins may have escapeddetection (325). Although little is known about the assemblypathway of the P1 head, it is likely that, as in many other tailed7054 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 24 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  phages, a scaffold of the head is assembled on an initiator complex, which is then degraded after being surrounded bystructural head proteins, allowing the resultant prohead toexpand to a mature head in a process coupled to entry of DNA(34). Mutations affecting head morphogenesis were mapped to ﬁve unlinked regions of P1 DNA (323; reviewed in references 346 and 347). Each of these regions contains an operon that isunder the control of a late promoter, LP pro,L Pdar, LP23, and LPpmgL ,o rL P pac(Fig. 1). Together these operons contain 21 genes. Two genes of the pacoperon encode subunits of the P1 DNA packaging enzyme and are described here separately. The assignment of the majority of head components to genes that encode them cannot be made based on simple size corre-lations. Certain P1 virion polypeptides are truncated by pro-teolytic processing; one or more are increased in size by theaddition of multimers of an amino acid chain of 1 to 2 kDa,most likely through disul ﬁde bonds (325). With two exceptions, the lack of signi ﬁcant homologies of P1 head proteins to char- acterized head proteins of other phages excludes sequence-based predictions of their function. At least four head proteins (with molecular masses of 220, 76, 47.5, and 10 kDa) are dispensable for morphogenesis, asthey were absent from heads of phages missing the 9-kb DNAsegment containing the darA operon (325). The missing 220- kDa protein must correspond to DarB, the only P1-encodedprotein larger than 120 kDa (predicted molecular mass, 251kDa). Although the darB gene itself is outside the missing segment, it may require a protein encoded by this segment forits expression or the incorporation of its product into phageheads. The other three dispensable proteins are probably products of the darA operon. The 47.5-kDa protein may represent DarA, which is proteolytically cleaved from a 69.6-kDa pre-cursor. The 76-kDa polypeptide may be a processed form ofthe 108-kDa product of the second large gene of the darA operon, ddrB . The 10-kDa polypeptide could represent the product of hdf, a protein that is homologous to a fragment of DarA and may be a head component, as is DarA itself. An unusual feature of P1 is the variety of head sizes. In addition to normal, infectious particles with big heads (P1B),P1 also produces small (P1S) and, more infrequently, minute(P1M) particles. S and M particles lack the capacity to hold thecomplete P1 genome and are nonproductive in single infec-tions (14, 156, 322). A gene that regulates the proportion ofphages of different head sizes in the progeny, designated vad (viral architecture determinant) was mapped to the late oper-on, under LP dar, but it is unclear whether it corresponds to the dargene of this operon or to a gene close to dar(147, 149, 155). The major head component is a 44-kDa protein that was present in different stoichiometric amounts in heads of big-headed (P1B) and small-headed (P1S) phages (325). It appearsto represent a truncated product of gene 23(H. Lehnherr, unpublished results). The full-length product, a 62.2-kDa pro-tein, is also one of the major bands of P1 head proteins sepa-rated by SDS-PAGE. A mutation in gene 23resulted in the production of complete tails and only a few heads followinginduction of the mutant lysogen (323). Whereas the neck andhead connector are normally associated with the head, the tailsof the mutant contained structures resembling head-neck con-nectors or aberrant tail extensions. The product of gene 23has no homologies to known capsid proteins of other phages. Ei-ther its evolutionary distance from these proteins is so greatthat only similarities at the structural level have been retainedor gp23 is representative of a separate class of phage capsidproteins characteristic of P1.', 'page_label: 24 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  It appearsto represent a truncated product of gene 23(H. Lehnherr, unpublished results). The full-length product, a 62.2-kDa pro-tein, is also one of the major bands of P1 head proteins sepa-rated by SDS-PAGE. A mutation in gene 23resulted in the production of complete tails and only a few heads followinginduction of the mutant lysogen (323). Whereas the neck andhead connector are normally associated with the head, the tailsof the mutant contained structures resembling head-neck con-nectors or aberrant tail extensions. The product of gene 23has no homologies to known capsid proteins of other phages. Ei-ther its evolutionary distance from these proteins is so greatthat only similarities at the structural level have been retainedor gp23 is representative of a separate class of phage capsidproteins characteristic of P1. Gene 23is the ﬁrst of six genes transcribed from the LP 23 late promoter and perhaps the only head morphogenetic geneof the 23operon, since other mutations mapped to this operon cause phenotypes not associated with head morphogenesis(281, 323). Homologies of the predicted gene products of thegenes downstream of 23to known phage proteins suggest that they have roles in sheath, tail tube, and baseplate morphogen-esis. The major set of head structure and assembly genes must be contained within the array of 13 genes transcribed from theLPpmgL promoter, as other late operons of P1 could not encode all the unassigned head proteins. An amber mutationin a gene of this cluster, previously designated 8, conferred a phenotype similar to that of a 23mutation, except that there were half as many empty heads as tails (323). Apparently, gene8encodes an essential head structure or assembly protein. Which of the 13 LP pmgL -controlled genes corresponds to 8is unclear. One candidate is the largest gene of this cluster, pmgS , whose predicted product is a 48-kDa protein of almost entirelyalpha-helical structure, characteristic of prohead scaffoldingproteins of other tailed phages (221, 314). The gene that di-rectly precedes pmgS ,pmgR , is another candidate, as its pre- dicted product contains a putative transmembrane domain. InT4, the assembly of the major head protein into proheads isinitiated on membrane-bound connectors built of other T4-encoded proteins (34). Only one protein encoded by genes transcribed from LPpmgL , the product of the fourth gene in this array, pmgO , appears to have homologies to putative phage head morpho-genesis proteins, namely to the products of two directly linkedORFs of similar sequence (ORF7 and ORF8) of the P. aerugi- nosa D3 phage, which in the D3 genome directly follow a gene for the major capsid protein (176). The predicted product of the sixth gene of this array, desig- nated by us ppp(P1 protein phosphatase) appears to be 64% identical to phage /H9261serine/threonine protein phosphatase (lambdaPP), which is a representative of a group of phage-encoded protein phosphatases and has structural similarity tomammalian Ser/Thr phosphoprotein phosphatases, includingcalcineurin (62, 172, 320). LambdaPP is a product of the 221-codon ORF (ORF221) in the region of eight nonessentialdelayed-early genes (called the ninR region) of the /H9261genome. An ORF221 mutation, known as byp, which abolishes the phos- phatase activity of lambdaPP, caused a defect in the establish-ment of lysogeny, unlike a nindeletion (62, 65). The defect was proposed to depend either on the inability of alteredlambdaPP to dephosphorylate a host protein that normallyparticipates in termination of transcription of /H9261early genes or on constitutive transcription, from a promoter created by themutation, of the /H9261Qgene, encoding the antiterminator of late gene transcription (24, 53). Whether the product of P1 ppp serves P1 by dephosphorylating P1-encoded proteins involvedin virion morphogenesis or a host protein or both remains to bedetermined.VOL. 186, 2004 BACTERIOPHAGE P1 7055', 'page_label: 25 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The tenth gene in the array, pap, like ppp, can be transcribed both early and late in P1 lytic development. The gene encodesa protein that is homologous to the C-terminal moiety of thepolynucleotide kinase (Pnk) of the T4 bacteriophage. By itself,this acid phosphatase domain of Pnk can remove 3 /H11032-PO 4ter- mini from DNA or RNA polynucleotides (328), hence ourassignment of the name pap (P1 acid phosphatase). The as- partate residue at position 167 of T4 Pnk, which is an essentialand presumably catalytic constituent of the Pnk 3 /H11032-phospha- tase domain, is conserved in P1 Pap, as are the surroundingamino acid residues. A gene whose product would be a ho-molog of the N-terminal kinase domain of T4 Pnk could not befound in the genome of P1. Thus P1 Pap appears to have adifferent role from that of Pnk in the lytic development of T4;determination of that role will require further study. As noted above, some structural head proteins of P1 are synthesized as precursors that are proteolytically cleaved to theform found in mature phage particles (304). One such cleav-age, of the product of late gene darA to 9-kDa and 68-kDa proteins, was associated with a locus initially designated 4and later pro(summarized in reference 346). The profunction is essential for P1 development as indicated by the conditionallynonproductive phenotype of all the mutants isolated. It wasproposed that the product of promust be involved in process- ing other structural proteins in addition to DarA, as darA is not an essential gene (304). The progene belongs to an operon of two genes of previ- ously unknown sequence which appear to be controlled by theputative late promoter, LP pro. They were originally de ﬁned as two complementation groups to which mutations with similarpleiotropic phenotypes could be assigned (230, 252, 294, 321,324). In addition to being defective in the proteolytic pro- cessing of DarA, amber mutations in both complementationgroups were found to cause production of complete phage tailsand unattached empty heads, suggesting a defect in a proteininvolved in head-to-tail connection (304, 323). In support ofthis conclusion, the predicted product of the ﬁrst gene of the prooperon is a protein that has similar length and some se- quence similarities to the portal protein of bacteriophage T4,gp20, and its homologs in several T4-like cyanophages (110).We conclude that this gene encodes the P1 portal protein anddesignate it prt. The predicted product of the progene is a protein of com- plex structure. Its N-terminal region is homologous to theN-terminal moiety of the ATP-dependent Hpr kinase/phos-phatase of Enterococcus faecalis , HprK (174). Its central part has homologies to the C-terminal region of the ClpX proteinofE. coli , which is an alternative ATP-binding subunit of the Clp protease (100, 350). Although it seems likely that Proactivates, and acts in a complex with a host protease, ClpP isexcluded from that role since P1 is able to plate on a clpP null mutant of E. coli and transduce the mutation. The Prt protein may be a substrate for Pro-mediated pro- teolytic processing. Thus, a likely explanation for similar pleio-tropic phenotypes of amber mutations in either of the genes oftheprt pro operon, each of which encodes a different function, is that mutations in prthave a polar effect on proexpression, whereas mutations in pro may lead to accumulation of an unprocessed product of prtthat can not be incorporated into P1 virions.(b) DNA packaging into proheads. The processive headful packaging from a speci ﬁc cut in a concatemer that produces cyclically permuted, terminally redundant DNA in T4 phagepopulations was originally proposed to start from a speci ﬁc cut in a concatemeric molecule. T4 was later shown not to bespeci ﬁcally cut for packaging, but certain other phages were, including P1 (22).', 'page_label: 25 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Thus, a likely explanation for similar pleio-tropic phenotypes of amber mutations in either of the genes oftheprt pro operon, each of which encodes a different function, is that mutations in prthave a polar effect on proexpression, whereas mutations in pro may lead to accumulation of an unprocessed product of prtthat can not be incorporated into P1 virions.(b) DNA packaging into proheads. The processive headful packaging from a speci ﬁc cut in a concatemer that produces cyclically permuted, terminally redundant DNA in T4 phagepopulations was originally proposed to start from a speci ﬁc cut in a concatemeric molecule. T4 was later shown not to bespeci ﬁcally cut for packaging, but certain other phages were, including P1 (22). The site, pac, at which each concatemeric P1 DNA is cut to be packaged into a prohead lies just within the pacA gene, encoding one of the two subunits of the pacase enzyme (286).Packaging into the ﬁrst prohead proceeds from pac towards the site-speci ﬁc recombination site lox(22), 4 kb away (Fig. 1). It continues through the next pac and loxand roughly 4 kb further until the prohead is ﬁlled and packaging into a second prohead can begin. Consequently, the ﬁrst DNA that is pack- aged from each concatemer contains two loxsites, one at the beginning and the second at its redundant end. Upon enteringa cell, this DNA has the advantage of being able to cyclizeeither by homologous or site-speci ﬁc recombination. The pro- cessivity of packaging appears limited to three or four headfuls,as determined by the proportion of nonstoichiometric frag-ments (generated by cuts at pac) in restriction digests of DNA from phage particles (22). The limiting factor may not be theprocessivity of pacase itself but the extent to which the sub-strate of packaging is subject to RecBCD-mediated degrada-tion from the end unprotected by bound pacase (297). Cleavages within the pac site are distributed over 13 bp, which are ﬂanked on each side by four similarly oriented hex- anucleotide elements (5 /H11032-TGATCA/G) (298). These sites are substrates for adenine methylation by the bacterial and viralmethyltransferases Dam and Dmt, respectively. If the sites areunmethylated or hemimethylated, cleavage does not occur,although hemimethylated (as well as fully methylated) DNAforms complexes with the pacase (296). These observations have suggested that pacase complexed with hemimethylated pacsites prevents their full methylation and hence their cleavage. The late expression of dmt could switch pacase function from nonselective sequestration of pac sites to selective cutting of those pacsites that are newly gen- erated and able to become fully methylated. Presumably onlylate in lytic development, when Lpa has activated the transcrip-tion of dmtand accelerated the transcription of pacA andpacB , does cleavage at pacsites occur and then only at some nascent sites as they are generated by rolling-circle replication. Thisproposal implies that DNA packaging proceeds from the roll-ing circle in the direction of the ﬁrst DNA to roll out. As sug- gested earlier, the nick responsible for initiating rolling-circlereplication may be made by the same enzyme that cleaves bothDNA strands to initiate packaging (296). The proposed rolesof pacase in successively preventing and promoting pac site cleavage could explain the ﬁnding that pacA and pacB start to be transcribed from an early promoter well before packagingbegins. Cleavage of the P1 packaging site requires the product of both proteins PacA and PacB, presumably as heterooligomers.An additional protein (PacC) encoded within the distal third ofpacB and translated in the same reading frame as PacB has been identi ﬁed, but its role in packaging, if any, remains un- determined (286). The PacA subunit recognizes and binds tomethylated pacsites independently of PacB but requires a host7056 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 26 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  factor to do so (287). The requirement can be satis ﬁed by IHF, for which a binding site lies immediately adjacent to the cleav-age region. HU can also satisfy this requirement, and these twohost proteins work in synergy, apparently cooperating in bend-ing the DNA at pacto bring together the two sets of hexameric repeats (287). PacA is likely to interact with the P1 portal protein, since an analogous interaction of a DNA pacase complex with a portalprotein in T4 has been demonstrated (34, 202). Homologiesbetween P1 and T4 portal proteins strongly support this hy-pothesis. P1 PacB bears an ATP binding motif (see Table S1 inthe supplemental material) and presumably transduces the en-ergy of ATP hydrolysis into the energy of DNA translocationinto the capsid, as does the analogous T4 gp17 protein. (c) Tail and tail ﬁber morphogenesis. The P1 tail consists of a long cylindrical tube surrounded by a contractile sheath andattached at one end to a baseplate containing six tail ﬁbers. Nine virion proteins (with apparent molecular masses of 117,105, 72, 33.5, 32.7, 27.2, 24.0, 21.4, and 15.4 kDa) were iden-tiﬁed as P1 tail structural components on the basis of their absence in preparations of P1 heads (325). The assignment offour additional proteins to a tail structure was uncertain. Smallproteins that could not be detected may extend this list. Mutations affecting tail assembly were assigned to regions where four P1 late operons, transcribed from the LP S,L Psit, LP23, and LP 26promoters, are located. Head and tail assem- bly pathways appear independent, as each tail-defective mu-tant produced complete heads (323). Homologies of putativeproducts of certain P1 tail genes to tail proteins of T4 orT4-like phages indicate a common ancestry of P1 and T4 tailmorphogenetic functions. The clearest difference is in the ap-pearance of the baseplate, which in P1 is thinner and smallerthan in T4 (323). A major tube component is a product of the second gene in thesitoperon (designated here tub). Its C-terminal moiety is homologous to C-terminal moieties of tail tube proteins, prod-ucts of gene 19of T4-related phages RB49 and nt-1, and, to a lesser extent, to gp19 of T4 itself. In addition, the predictedmolecular mass of the P1 Tub protein (22.3 kDa) roughlycorresponds to the estimated molecular mass of one of the twomost abundant tail components (21.4 kDa) (325), which byanalogy to other phages, must correspond to either the tail orsheath subunits. In T4, the tail tube consists of gp19 subunits,arranged in stacked hexameric rings, which form a tunnelthrough which the phage DNA can pass (reviewed in reference63). The organization of Tub protein molecules in the tail tubeof P1 seems likely to be similar. In the genomes of T4 and related phages, a gene encoding the tube protein is typically located between a gene encodingthe sheath protein and a gene encoding a portal vertex (110).In the genome of P1, this organization is not conserved. Thetubgene is ﬂanked by genes whose products resemble an es- sential baseplate hub subunit of T4-related phage RB49 and aprotein of unknown function of another T4-related phage,Aeh1. A single P1 amber mutation that caused production of com- plete heads and tail tubes without sheaths was mapped to agene designated 22(323). This gene is contained within the late operon 23and distant from the sitoperon, in which the tub gene is located. Gene 22was proposed to encode the sheathprotein. This assignment is consistent with the presence in gp22 of a motif characteristic of conserved domains of bacte-riophage tail sheath proteins (see Table S1 in the supplementalmaterial) and regions similar to a nucleotide-binding motif(data not shown) that is present in the sheath protein of T4(63).', 'page_label: 26 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Thetubgene is ﬂanked by genes whose products resemble an es- sential baseplate hub subunit of T4-related phage RB49 and aprotein of unknown function of another T4-related phage,Aeh1. A single P1 amber mutation that caused production of com- plete heads and tail tubes without sheaths was mapped to agene designated 22(323). This gene is contained within the late operon 23and distant from the sitoperon, in which the tub gene is located. Gene 22was proposed to encode the sheathprotein. This assignment is consistent with the presence in gp22 of a motif characteristic of conserved domains of bacte-riophage tail sheath proteins (see Table S1 in the supplementalmaterial) and regions similar to a nucleotide-binding motif(data not shown) that is present in the sheath protein of T4(63). Although the calculated molecular mass of gp22 is only 57kDa, smaller by 15 kDa than that of the P1 sheath proteinestimated by SDS-PAGE, there is no other unassigned P1ORF transcribed from a late promoter that could encode anessential protein of about 72 kDa. It is possible that the maturesheath protein of 72 kDa found in virions of P1 is formed bycross-linking of gp22 with another protein. Although transla-tional frameshifting was found to occur during synthesis of /H9261 and P2 phage tail components (56, 199), it is unlikely to beresponsible for the extension of P1 gp 22. No single frameshift within the 22sequence could generate a signi ﬁcantly longer protein. All four operons that encode P1 tail proteins contain genes for baseplate components. Products of ﬁve of these genes have homologies to baseplate proteins of T4 or T4-related phages.In T4, the baseplate consists of a hub surrounded by sixwedges. Baseplate morphogenesis begins with assembly ofwedges that polymerize around the independently assembledhub (reviewed in references 63 and 86). Two P1 genes, belonging to well-separated operons, encode proteins that are likely to participate early in the assembly ofbaseplate wedges. One of them, which we name bplA , is iden- tiﬁed by the extensive homology of its product to gp6, the largest baseplate protein and major wedge component of T4(329). The bplA gene is located downstream of the tubA gene in the sitoperon. The other gene, 26,i nt h e 26operon, encodes a protein with homology to gp53 of T4 and related phagesAeh1 and RB49. The products of both P1 genes are somewhatsmaller than their T4 homologs. In T4, the order of addition isgp6 before gp53 and the two proteins may interact (reviewed inreference 224). T4 phages mutant in either gp6 or gp53 pro-duce polysheaths in addition to complete heads, but no visibletail structures, consistent with the requirement of gp6 and gp53for the assembly of the T4 tail. P1 proteins BplA and gp26 mayplay analogous roles in baseplate and tail assembly. In supportof this view, gene 26mutant phages produced complete heads and no tail structures other than polysheaths (323). A similarphenotype was attributed to P1 mutants in a gene previouslynamed 3, which was mapped to the pmgC tub pmgB sit pmgA bplA 16 operon region. Whether bplA and 3are identical re- mains to be seen. Mutations in three other P1 genes, R, 16 , and 21, cause phenotypes identical to those of gene 26mutants (323). The products of these genes lack homology to proteinsof other phages, indicating that they form those parts of the P1baseplate whose evolutionary history is different from base-plate components of T4, and they are perhaps unique to P1. The largest tail component is the product of the sitgene (121 kDa). The Sit protein has a mosaic structure. Between aminoacid residues 757 and 885, close to the C-terminal end of this1,140-amino-acid protein, are motifs characteristic of solublelytic transglycosylases (190).'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 22 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The sites recognized by the Lpa protein and requirements for its function are considered in the section“Lpa-controlled operons. ” (e) Morphogenetic functions that start to be expressed early in lytic development. Among 17 P1 operons controlled by C1, ﬁve (mat,26,ppp,pap, and pmgT ) encode morphogenetic func- tions (Fig. 1 and Table 3; see also Table S1 in the supplementalmaterial). Although early expression of these operons mayseem surprising, it is noteworthy that, in addition to beingcontrolled by C1, they all appear to be activated by Lpa. Twoof them, mat and 26, contain in their immediate upstream regulatory regions, in addition to a C1-controlled /H9268 70pro- moter, an Lpa-activated late promoter. Three others, pmgN, pmgR , and pmgT , are likely to be transcribed from an Lpa- activated promoter of the upstream pmgN gene by read- through transcription. An Ssp-dependent mechanism was found to turn down tran- scription from C1-controlled promoters before the Lpa-con-trolled promoters are activated (28). This ﬁnding indicates that, for those genes transcribed from a C1-controlled pro-moter, additional control from a late promoter is required toensure the continuity of their expression throughout lytic de-velopment and to satisfy the requirement late in developmentfor large amounts of certain proteins. Dual regulatory signalsTABLE 8. Known and putative late promoters of P1 Name PositionaStrandSequence Identi ﬁcationb Reference /H1100222 /H1100210 tsp pp cmp LPmat 2678 /H11001TTGACAACTGAC AAGTGACTT CAGT CAGAAT CATCACACGC /H12011 189 LPpro 16052 /H11001GTAATCCTTAAC AAGTGACTA GTGT TAAATT CCGTTCAAAC /H12011This work LPlyz 19770 /H11002AACGTAGCTAAC AAGTGATTT GCGT TATCCT GTGTCTTCTA /H12011/H12011 277 LPdarA 30013 /H11002TGACCGGTTCAA AAGTTACTT TGCA TACCAT TACCTCCTGA /H12011/H12011 105 LPS 37119 /H11002GGGCGGGGTGAC AAGTTACTT ATCT TACAAT GAGGCTTCAC /H12011/H12011 105 LPsit 45409 /H11002TACTCAATGAAC AAATGACTA CTCG TAGAAT CGGTTAACAC /H12011 H. Lehnherr, unpublished LP23 57626 /H11002GAGAAAGTCAAC AAGTGACTT TCAA TAAAAT CTCTTCCGAA /H12011This work LP26 77286 /H11002TAAATATTCAAC AAATGACTA GCGG TAGAAT CACCATCATC /H12011 275 LPpmgL 77296 /H11001TCATTCTACCGC TAGTCATTT GTTG AATATT TAACTCAATA /H12011 H. Lehnherr, unpublished LPhot 85322 /H11001AAAAAAAATTAA AAGTTACTT TGCTGGTTA AATAAT AGTCGTTACT /H12011/H12011 275 LPpac 89276 /H11001GGGCCTTTTGAC ATGTGACTT TCGT TACCCT CGCGTCAAAA /H12011 H. Lehnherr, unpublished aThe coordinate is that of the ﬁrst nucleotide of the promoter, that is, the ﬁrst nucleotide of the /H1100222 sequence for promoters read from the /H11001strand (clockwise) or the last nucleotide of the /H1100210 sequence for promoters read from the /H11002strand (counterclockwise). Coordinates refer to positions of promoters in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. Note that LP mat shares its /H1100210 sequence with P mat. bPromoter identi ﬁcation is based on a transcription start point determination ( tsp), detection with the promoter probe vector ( pp), or computer analysis of the sequence ( cmp).', 'page_label: 22 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Coordinates refer to positions of promoters in the P1 c1-100 mod749 ::IS5genome without its nonintegral parts, IS 5, and the associated 4-bp duplication. Note that LP mat shares its /H1100210 sequence with P mat. bPromoter identi ﬁcation is based on a transcription start point determination ( tsp), detection with the promoter probe vector ( pp), or computer analysis of the sequence ( cmp). The promoters distinguished by /H12011in the cmp column were identi ﬁed on the basis of the similarity of their sequences to the sequences of the other nine late promoters, using the GCG Fitconsensus program and the consensus table of /H1100222 sequences of the known late promoters made by the GCG program Consensus. Only the sequences with homology scores similar to the known promoters and located 4 to 14 nt (189) upstream from sequences similar to -10 he xamers of RNA polymerase promoters of E. coli (as estimated by the program Targsearch [229]) were considered.VOL. 186, 2004 BACTERIOPHAGE P1 7053', 'page_label: 23 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  in control regions of certain operons are common in T4 and other phages, where they can serve to extend expression of agiven gene to more than one developmental stage and to mod-ulate it differently at different stages (224, 228). P1matmutants produce viral particles with empty heads and unstable tails with tail sheaths in various stages of contraction(323). The pattern of mat expression following induction of a P1 lysogen is consistent with its being subject to dual control(191). Expression was seen to start within 10 to15 min ofinception of lytic development, to continue at a low level forabout 10 min, and then to increase sharply, exactly when theexpression of other genes activated by Lpa begins. Expressionproﬁles of the remaining genes that are controlled directly by both C1 and Lpa may be similar. As some products of theremaining C1- and Lpa-regulated operons are structural virionproteins, they are described later. Mat was proposed to have aregulatory role in morphogenesis (323), although with scantjustiﬁcation. The pacAB operon, which encodes two subunits of the P1 DNA packaging enzyme, is also subject to dual control. Al-though it is directly preceded only by the late promoter LP pac (Table 8), it is expressed early from the C1-controlled pro-moter of the upstream lpagene (105). Moreover, some PacAB- mediated cleavage of pacsites occurred in the absence of Lpa, indicating that Lpa is not essential for pacAB operon expres- sion during P1 lytic development (297). (ii) Lpa-controlled operons. P1 late promoters resemble typ- icalE. coli /H9268 70promoters at their /H1100210 regions but lack the /H1100235 hexamer (188). Instead, they have a conserved 9-bp inverted repeat (previously termed late operator) that is centered aboutposition /H1100222 of the transcription start site and interacts with Lpa (111, 189). Results of in vivo footprinting experimentsconﬁrm that RNA polymerase binds to late promoter se- quences in the presence, but not in the absence, of Lpa (111).In addition to Lpa, the E. coli RNA polymerase-associated protein, SspA (158), is required both in vivo and in vitro forlate promoter activation (111, 336). The activation requires the/H9268 70RNA polymerase holoenzyme and does not occur when /H926870is replaced by /H9268S. How Lpa and SspA cooperate in redi - recting the host RNA polymerase towards P1 late promotersequences is unknown. Nine P1 late promoters were identi ﬁed previously by empir- ical studies (Table 8). The optimal spacing between /H1100210 and /H1100222 regions of late promoters is 4 bp, and it is conserved in all but one of these promoters, LP hot. Although in the LP hot promoter the spacing (9 bp) is suboptimal, a promoter probeassay con ﬁrmed in vivo LP hottranscriptional activity to be dependent upon Lpa (275). Here we identi ﬁed two additional putative late promoters, based on the strong similarity of their predicted /H1100222 regions to the/H1100222 regions of known late promoters and on the presence of appropriately positioned putative /H1100210 hexamers (Table 8). Both promoters have the optimal spacing between their pre-dicted /H1100210 and /H1100222 regions, and both are located in the regulatory regions of genes known to encode phage morpho-genetic functions. The 11 late promoters control at least 12 andas many as 14 P1 operons. We suspect that transcription fromLPpmgL , before encountering the closest terminator, T pmgQ (Table 5), can read through two genes that are immediatelydownstream, pmgL and pmgM , and a C1-controlled operonfurther downstream that encodes morphogenetic functions (Fig. 1). It is likely that this transcript includes an additionaltwo operons downstream from T pmgQ that also encode mor- phogenetic functions. Expression of these operons late duringlytic development implies that P1 uses an ef ﬁcient antitermi- nation mechanism to attenuate the function of T pmgQ ,TpmgS , and T pap terminators.', 'page_label: 23 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The 11 late promoters control at least 12 andas many as 14 P1 operons. We suspect that transcription fromLPpmgL , before encountering the closest terminator, T pmgQ (Table 5), can read through two genes that are immediatelydownstream, pmgL and pmgM , and a C1-controlled operonfurther downstream that encodes morphogenetic functions (Fig. 1). It is likely that this transcript includes an additionaltwo operons downstream from T pmgQ that also encode mor- phogenetic functions. Expression of these operons late duringlytic development implies that P1 uses an ef ﬁcient antitermi- nation mechanism to attenuate the function of T pmgQ ,TpmgS , and T pap terminators. Attenuation of terminator functions may also occur at T 22,TpmgG , and T ddrA terminators, since these would otherwise prevent late expression of genes 22, pmgG ,pmgF ,ddrB , and hxr, encoding known or predicted late functions. Alternatively, these terminators may be inef ﬁcient and serve to decrease the expression of certain late genesdistant from the LP pmgL ,L P 23, and LP darlate promoters. One other late promoter, LP sit, could drive transcription of both the sitoperon and the following RSU operon. However, theRSU operon is also transcribed from its own late pro- moter, LP S. The remaining nine late promoters drive tran- scription of single operons that adjoin operons transcribed inthe opposite direction. In total, 52 genes of P1 can be tran-scribed from late promoter sequences. InE. coli at 37 °C, late gene products start to be synthesized 20 to 30 min after either transient thermal induction of aP1c1ts lysogen or infection of sensitive cells by P1. The com- pletion of phage morphogenesis at about min 40 is accompa-nied by the onset of cell lysis. The late genes of P1 are involvedin DNA replication, phage tail and head morphogenesis, pack-aging of DNA, synthesis and incorporation of injectable pro-teins, and cell lysis. Some of them, as explained above, areexpressed both early and late in lytic development. Some thatare apparently expressed in late developmental stages cannotbe associated with any late promoter sequences. The darB gene, encoding an antirestriction protein and a P1 head com-ponent, is an example. Transcription of these genes may becontrolled either indirectly by Lpa through another regulatoryprotein or by an as-yet-unidenti ﬁed mechanism. (a) Head morphogenesis. The morphogenesis of P1 resem- bles that of other tailed phages in that heads and tails haveindependent assembly pathways (323). At least some of thehead proteins appear to be proteolytically processed (304).DNA is packaged into proheads from concatemers (60) by aheadful mechanism, as in T4, but the packaging of each con-catemer, unlike that in T4, starts from an initial cut at a speci ﬁc pacsite (286, 297, 298). Infective particles of P1 contain cycli- cally permuted, linear, double-stranded molecules with a ter-minal redundancy of about 10 kb of DNA (296). P1 tail ﬁbers, unlike those of T4, can attach to the tail before head-tailjoining occurs (323). P1 virions resemble in many respects those of T4, although they are less complex, especially in the structures of the headand the baseplate (323). Mature P1 virions contain at least 28proteins (325), as compared to at least 49 proteins in maturevirions of T4 (summarized in reference 224). In T4, a furtherﬁve proteins assist in morphogenesis and are removed during virion maturation. Proteins of similar function in P1 awaitidenti ﬁcation. Analysis of P1 virion proteins by SDS-PAGE indicates that P1 heads are composed of 15 to 19 protein components, pos-sibly an underestimate as small proteins may have escapeddetection (325). Although little is known about the assemblypathway of the P1 head, it is likely that, as in many other tailed7054 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 24 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  phages, a scaffold of the head is assembled on an initiator complex, which is then degraded after being surrounded bystructural head proteins, allowing the resultant prohead toexpand to a mature head in a process coupled to entry of DNA(34). Mutations affecting head morphogenesis were mapped to ﬁve unlinked regions of P1 DNA (323; reviewed in references 346 and 347). Each of these regions contains an operon that isunder the control of a late promoter, LP pro,L Pdar, LP23, and LPpmgL ,o rL P pac(Fig. 1). Together these operons contain 21 genes. Two genes of the pacoperon encode subunits of the P1 DNA packaging enzyme and are described here separately. The assignment of the majority of head components to genes that encode them cannot be made based on simple size corre-lations. Certain P1 virion polypeptides are truncated by pro-teolytic processing; one or more are increased in size by theaddition of multimers of an amino acid chain of 1 to 2 kDa,most likely through disul ﬁde bonds (325). With two exceptions, the lack of signi ﬁcant homologies of P1 head proteins to char- acterized head proteins of other phages excludes sequence-based predictions of their function. At least four head proteins (with molecular masses of 220, 76, 47.5, and 10 kDa) are dispensable for morphogenesis, asthey were absent from heads of phages missing the 9-kb DNAsegment containing the darA operon (325). The missing 220- kDa protein must correspond to DarB, the only P1-encodedprotein larger than 120 kDa (predicted molecular mass, 251kDa). Although the darB gene itself is outside the missing segment, it may require a protein encoded by this segment forits expression or the incorporation of its product into phageheads. The other three dispensable proteins are probably products of the darA operon. The 47.5-kDa protein may represent DarA, which is proteolytically cleaved from a 69.6-kDa pre-cursor. The 76-kDa polypeptide may be a processed form ofthe 108-kDa product of the second large gene of the darA operon, ddrB . The 10-kDa polypeptide could represent the product of hdf, a protein that is homologous to a fragment of DarA and may be a head component, as is DarA itself. An unusual feature of P1 is the variety of head sizes. In addition to normal, infectious particles with big heads (P1B),P1 also produces small (P1S) and, more infrequently, minute(P1M) particles. S and M particles lack the capacity to hold thecomplete P1 genome and are nonproductive in single infec-tions (14, 156, 322). A gene that regulates the proportion ofphages of different head sizes in the progeny, designated vad (viral architecture determinant) was mapped to the late oper-on, under LP dar, but it is unclear whether it corresponds to the dargene of this operon or to a gene close to dar(147, 149, 155). The major head component is a 44-kDa protein that was present in different stoichiometric amounts in heads of big-headed (P1B) and small-headed (P1S) phages (325). It appearsto represent a truncated product of gene 23(H. Lehnherr, unpublished results). The full-length product, a 62.2-kDa pro-tein, is also one of the major bands of P1 head proteins sepa-rated by SDS-PAGE. A mutation in gene 23resulted in the production of complete tails and only a few heads followinginduction of the mutant lysogen (323). Whereas the neck andhead connector are normally associated with the head, the tailsof the mutant contained structures resembling head-neck con-nectors or aberrant tail extensions. The product of gene 23has no homologies to known capsid proteins of other phages. Ei-ther its evolutionary distance from these proteins is so greatthat only similarities at the structural level have been retainedor gp23 is representative of a separate class of phage capsidproteins characteristic of P1.', 'page_label: 24 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  It appearsto represent a truncated product of gene 23(H. Lehnherr, unpublished results). The full-length product, a 62.2-kDa pro-tein, is also one of the major bands of P1 head proteins sepa-rated by SDS-PAGE. A mutation in gene 23resulted in the production of complete tails and only a few heads followinginduction of the mutant lysogen (323). Whereas the neck andhead connector are normally associated with the head, the tailsof the mutant contained structures resembling head-neck con-nectors or aberrant tail extensions. The product of gene 23has no homologies to known capsid proteins of other phages. Ei-ther its evolutionary distance from these proteins is so greatthat only similarities at the structural level have been retainedor gp23 is representative of a separate class of phage capsidproteins characteristic of P1. Gene 23is the ﬁrst of six genes transcribed from the LP 23 late promoter and perhaps the only head morphogenetic geneof the 23operon, since other mutations mapped to this operon cause phenotypes not associated with head morphogenesis(281, 323). Homologies of the predicted gene products of thegenes downstream of 23to known phage proteins suggest that they have roles in sheath, tail tube, and baseplate morphogen-esis. The major set of head structure and assembly genes must be contained within the array of 13 genes transcribed from theLPpmgL promoter, as other late operons of P1 could not encode all the unassigned head proteins. An amber mutationin a gene of this cluster, previously designated 8, conferred a phenotype similar to that of a 23mutation, except that there were half as many empty heads as tails (323). Apparently, gene8encodes an essential head structure or assembly protein. Which of the 13 LP pmgL -controlled genes corresponds to 8is unclear. One candidate is the largest gene of this cluster, pmgS , whose predicted product is a 48-kDa protein of almost entirelyalpha-helical structure, characteristic of prohead scaffoldingproteins of other tailed phages (221, 314). The gene that di-rectly precedes pmgS ,pmgR , is another candidate, as its pre- dicted product contains a putative transmembrane domain. InT4, the assembly of the major head protein into proheads isinitiated on membrane-bound connectors built of other T4-encoded proteins (34). Only one protein encoded by genes transcribed from LPpmgL , the product of the fourth gene in this array, pmgO , appears to have homologies to putative phage head morpho-genesis proteins, namely to the products of two directly linkedORFs of similar sequence (ORF7 and ORF8) of the P. aerugi- nosa D3 phage, which in the D3 genome directly follow a gene for the major capsid protein (176). The predicted product of the sixth gene of this array, desig- nated by us ppp(P1 protein phosphatase) appears to be 64% identical to phage /H9261serine/threonine protein phosphatase (lambdaPP), which is a representative of a group of phage-encoded protein phosphatases and has structural similarity tomammalian Ser/Thr phosphoprotein phosphatases, includingcalcineurin (62, 172, 320). LambdaPP is a product of the 221-codon ORF (ORF221) in the region of eight nonessentialdelayed-early genes (called the ninR region) of the /H9261genome. An ORF221 mutation, known as byp, which abolishes the phos- phatase activity of lambdaPP, caused a defect in the establish-ment of lysogeny, unlike a nindeletion (62, 65). The defect was proposed to depend either on the inability of alteredlambdaPP to dephosphorylate a host protein that normallyparticipates in termination of transcription of /H9261early genes or on constitutive transcription, from a promoter created by themutation, of the /H9261Qgene, encoding the antiterminator of late gene transcription (24, 53). Whether the product of P1 ppp serves P1 by dephosphorylating P1-encoded proteins involvedin virion morphogenesis or a host protein or both remains to bedetermined.VOL. 186, 2004 BACTERIOPHAGE P1 7055', 'page_label: 25 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The tenth gene in the array, pap, like ppp, can be transcribed both early and late in P1 lytic development. The gene encodesa protein that is homologous to the C-terminal moiety of thepolynucleotide kinase (Pnk) of the T4 bacteriophage. By itself,this acid phosphatase domain of Pnk can remove 3 /H11032-PO 4ter- mini from DNA or RNA polynucleotides (328), hence ourassignment of the name pap (P1 acid phosphatase). The as- partate residue at position 167 of T4 Pnk, which is an essentialand presumably catalytic constituent of the Pnk 3 /H11032-phospha- tase domain, is conserved in P1 Pap, as are the surroundingamino acid residues. A gene whose product would be a ho-molog of the N-terminal kinase domain of T4 Pnk could not befound in the genome of P1. Thus P1 Pap appears to have adifferent role from that of Pnk in the lytic development of T4;determination of that role will require further study. As noted above, some structural head proteins of P1 are synthesized as precursors that are proteolytically cleaved to theform found in mature phage particles (304). One such cleav-age, of the product of late gene darA to 9-kDa and 68-kDa proteins, was associated with a locus initially designated 4and later pro(summarized in reference 346). The profunction is essential for P1 development as indicated by the conditionallynonproductive phenotype of all the mutants isolated. It wasproposed that the product of promust be involved in process- ing other structural proteins in addition to DarA, as darA is not an essential gene (304). The progene belongs to an operon of two genes of previ- ously unknown sequence which appear to be controlled by theputative late promoter, LP pro. They were originally de ﬁned as two complementation groups to which mutations with similarpleiotropic phenotypes could be assigned (230, 252, 294, 321,324). In addition to being defective in the proteolytic pro- cessing of DarA, amber mutations in both complementationgroups were found to cause production of complete phage tailsand unattached empty heads, suggesting a defect in a proteininvolved in head-to-tail connection (304, 323). In support ofthis conclusion, the predicted product of the ﬁrst gene of the prooperon is a protein that has similar length and some se- quence similarities to the portal protein of bacteriophage T4,gp20, and its homologs in several T4-like cyanophages (110).We conclude that this gene encodes the P1 portal protein anddesignate it prt. The predicted product of the progene is a protein of com- plex structure. Its N-terminal region is homologous to theN-terminal moiety of the ATP-dependent Hpr kinase/phos-phatase of Enterococcus faecalis , HprK (174). Its central part has homologies to the C-terminal region of the ClpX proteinofE. coli , which is an alternative ATP-binding subunit of the Clp protease (100, 350). Although it seems likely that Proactivates, and acts in a complex with a host protease, ClpP isexcluded from that role since P1 is able to plate on a clpP null mutant of E. coli and transduce the mutation. The Prt protein may be a substrate for Pro-mediated pro- teolytic processing. Thus, a likely explanation for similar pleio-tropic phenotypes of amber mutations in either of the genes oftheprt pro operon, each of which encodes a different function, is that mutations in prthave a polar effect on proexpression, whereas mutations in pro may lead to accumulation of an unprocessed product of prtthat can not be incorporated into P1 virions.(b) DNA packaging into proheads. The processive headful packaging from a speci ﬁc cut in a concatemer that produces cyclically permuted, terminally redundant DNA in T4 phagepopulations was originally proposed to start from a speci ﬁc cut in a concatemeric molecule. T4 was later shown not to bespeci ﬁcally cut for packaging, but certain other phages were, including P1 (22).', 'page_label: 25 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Thus, a likely explanation for similar pleio-tropic phenotypes of amber mutations in either of the genes oftheprt pro operon, each of which encodes a different function, is that mutations in prthave a polar effect on proexpression, whereas mutations in pro may lead to accumulation of an unprocessed product of prtthat can not be incorporated into P1 virions.(b) DNA packaging into proheads. The processive headful packaging from a speci ﬁc cut in a concatemer that produces cyclically permuted, terminally redundant DNA in T4 phagepopulations was originally proposed to start from a speci ﬁc cut in a concatemeric molecule. T4 was later shown not to bespeci ﬁcally cut for packaging, but certain other phages were, including P1 (22). The site, pac, at which each concatemeric P1 DNA is cut to be packaged into a prohead lies just within the pacA gene, encoding one of the two subunits of the pacase enzyme (286).Packaging into the ﬁrst prohead proceeds from pac towards the site-speci ﬁc recombination site lox(22), 4 kb away (Fig. 1). It continues through the next pac and loxand roughly 4 kb further until the prohead is ﬁlled and packaging into a second prohead can begin. Consequently, the ﬁrst DNA that is pack- aged from each concatemer contains two loxsites, one at the beginning and the second at its redundant end. Upon enteringa cell, this DNA has the advantage of being able to cyclizeeither by homologous or site-speci ﬁc recombination. The pro- cessivity of packaging appears limited to three or four headfuls,as determined by the proportion of nonstoichiometric frag-ments (generated by cuts at pac) in restriction digests of DNA from phage particles (22). The limiting factor may not be theprocessivity of pacase itself but the extent to which the sub-strate of packaging is subject to RecBCD-mediated degrada-tion from the end unprotected by bound pacase (297). Cleavages within the pac site are distributed over 13 bp, which are ﬂanked on each side by four similarly oriented hex- anucleotide elements (5 /H11032-TGATCA/G) (298). These sites are substrates for adenine methylation by the bacterial and viralmethyltransferases Dam and Dmt, respectively. If the sites areunmethylated or hemimethylated, cleavage does not occur,although hemimethylated (as well as fully methylated) DNAforms complexes with the pacase (296). These observations have suggested that pacase complexed with hemimethylated pacsites prevents their full methylation and hence their cleavage. The late expression of dmt could switch pacase function from nonselective sequestration of pac sites to selective cutting of those pacsites that are newly gen- erated and able to become fully methylated. Presumably onlylate in lytic development, when Lpa has activated the transcrip-tion of dmtand accelerated the transcription of pacA andpacB , does cleavage at pacsites occur and then only at some nascent sites as they are generated by rolling-circle replication. Thisproposal implies that DNA packaging proceeds from the roll-ing circle in the direction of the ﬁrst DNA to roll out. As sug- gested earlier, the nick responsible for initiating rolling-circlereplication may be made by the same enzyme that cleaves bothDNA strands to initiate packaging (296). The proposed rolesof pacase in successively preventing and promoting pac site cleavage could explain the ﬁnding that pacA and pacB start to be transcribed from an early promoter well before packagingbegins. Cleavage of the P1 packaging site requires the product of both proteins PacA and PacB, presumably as heterooligomers.An additional protein (PacC) encoded within the distal third ofpacB and translated in the same reading frame as PacB has been identi ﬁed, but its role in packaging, if any, remains un- determined (286). The PacA subunit recognizes and binds tomethylated pacsites independently of PacB but requires a host7056 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 26 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  factor to do so (287). The requirement can be satis ﬁed by IHF, for which a binding site lies immediately adjacent to the cleav-age region. HU can also satisfy this requirement, and these twohost proteins work in synergy, apparently cooperating in bend-ing the DNA at pacto bring together the two sets of hexameric repeats (287). PacA is likely to interact with the P1 portal protein, since an analogous interaction of a DNA pacase complex with a portalprotein in T4 has been demonstrated (34, 202). Homologiesbetween P1 and T4 portal proteins strongly support this hy-pothesis. P1 PacB bears an ATP binding motif (see Table S1 inthe supplemental material) and presumably transduces the en-ergy of ATP hydrolysis into the energy of DNA translocationinto the capsid, as does the analogous T4 gp17 protein. (c) Tail and tail ﬁber morphogenesis. The P1 tail consists of a long cylindrical tube surrounded by a contractile sheath andattached at one end to a baseplate containing six tail ﬁbers. Nine virion proteins (with apparent molecular masses of 117,105, 72, 33.5, 32.7, 27.2, 24.0, 21.4, and 15.4 kDa) were iden-tiﬁed as P1 tail structural components on the basis of their absence in preparations of P1 heads (325). The assignment offour additional proteins to a tail structure was uncertain. Smallproteins that could not be detected may extend this list. Mutations affecting tail assembly were assigned to regions where four P1 late operons, transcribed from the LP S,L Psit, LP23, and LP 26promoters, are located. Head and tail assem- bly pathways appear independent, as each tail-defective mu-tant produced complete heads (323). Homologies of putativeproducts of certain P1 tail genes to tail proteins of T4 orT4-like phages indicate a common ancestry of P1 and T4 tailmorphogenetic functions. The clearest difference is in the ap-pearance of the baseplate, which in P1 is thinner and smallerthan in T4 (323). A major tube component is a product of the second gene in thesitoperon (designated here tub). Its C-terminal moiety is homologous to C-terminal moieties of tail tube proteins, prod-ucts of gene 19of T4-related phages RB49 and nt-1, and, to a lesser extent, to gp19 of T4 itself. In addition, the predictedmolecular mass of the P1 Tub protein (22.3 kDa) roughlycorresponds to the estimated molecular mass of one of the twomost abundant tail components (21.4 kDa) (325), which byanalogy to other phages, must correspond to either the tail orsheath subunits. In T4, the tail tube consists of gp19 subunits,arranged in stacked hexameric rings, which form a tunnelthrough which the phage DNA can pass (reviewed in reference63). The organization of Tub protein molecules in the tail tubeof P1 seems likely to be similar. In the genomes of T4 and related phages, a gene encoding the tube protein is typically located between a gene encodingthe sheath protein and a gene encoding a portal vertex (110).In the genome of P1, this organization is not conserved. Thetubgene is ﬂanked by genes whose products resemble an es- sential baseplate hub subunit of T4-related phage RB49 and aprotein of unknown function of another T4-related phage,Aeh1. A single P1 amber mutation that caused production of com- plete heads and tail tubes without sheaths was mapped to agene designated 22(323). This gene is contained within the late operon 23and distant from the sitoperon, in which the tub gene is located. Gene 22was proposed to encode the sheathprotein. This assignment is consistent with the presence in gp22 of a motif characteristic of conserved domains of bacte-riophage tail sheath proteins (see Table S1 in the supplementalmaterial) and regions similar to a nucleotide-binding motif(data not shown) that is present in the sheath protein of T4(63).', 'page_label: 26 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Thetubgene is ﬂanked by genes whose products resemble an es- sential baseplate hub subunit of T4-related phage RB49 and aprotein of unknown function of another T4-related phage,Aeh1. A single P1 amber mutation that caused production of com- plete heads and tail tubes without sheaths was mapped to agene designated 22(323). This gene is contained within the late operon 23and distant from the sitoperon, in which the tub gene is located. Gene 22was proposed to encode the sheathprotein. This assignment is consistent with the presence in gp22 of a motif characteristic of conserved domains of bacte-riophage tail sheath proteins (see Table S1 in the supplementalmaterial) and regions similar to a nucleotide-binding motif(data not shown) that is present in the sheath protein of T4(63). Although the calculated molecular mass of gp22 is only 57kDa, smaller by 15 kDa than that of the P1 sheath proteinestimated by SDS-PAGE, there is no other unassigned P1ORF transcribed from a late promoter that could encode anessential protein of about 72 kDa. It is possible that the maturesheath protein of 72 kDa found in virions of P1 is formed bycross-linking of gp22 with another protein. Although transla-tional frameshifting was found to occur during synthesis of /H9261 and P2 phage tail components (56, 199), it is unlikely to beresponsible for the extension of P1 gp 22. No single frameshift within the 22sequence could generate a signi ﬁcantly longer protein. All four operons that encode P1 tail proteins contain genes for baseplate components. Products of ﬁve of these genes have homologies to baseplate proteins of T4 or T4-related phages.In T4, the baseplate consists of a hub surrounded by sixwedges. Baseplate morphogenesis begins with assembly ofwedges that polymerize around the independently assembledhub (reviewed in references 63 and 86). Two P1 genes, belonging to well-separated operons, encode proteins that are likely to participate early in the assembly ofbaseplate wedges. One of them, which we name bplA , is iden- tiﬁed by the extensive homology of its product to gp6, the largest baseplate protein and major wedge component of T4(329). The bplA gene is located downstream of the tubA gene in the sitoperon. The other gene, 26,i nt h e 26operon, encodes a protein with homology to gp53 of T4 and related phagesAeh1 and RB49. The products of both P1 genes are somewhatsmaller than their T4 homologs. In T4, the order of addition isgp6 before gp53 and the two proteins may interact (reviewed inreference 224). T4 phages mutant in either gp6 or gp53 pro-duce polysheaths in addition to complete heads, but no visibletail structures, consistent with the requirement of gp6 and gp53for the assembly of the T4 tail. P1 proteins BplA and gp26 mayplay analogous roles in baseplate and tail assembly. In supportof this view, gene 26mutant phages produced complete heads and no tail structures other than polysheaths (323). A similarphenotype was attributed to P1 mutants in a gene previouslynamed 3, which was mapped to the pmgC tub pmgB sit pmgA bplA 16 operon region. Whether bplA and 3are identical re- mains to be seen. Mutations in three other P1 genes, R, 16 , and 21, cause phenotypes identical to those of gene 26mutants (323). The products of these genes lack homology to proteinsof other phages, indicating that they form those parts of the P1baseplate whose evolutionary history is different from base-plate components of T4, and they are perhaps unique to P1. The largest tail component is the product of the sitgene (121 kDa). The Sit protein has a mosaic structure. Between aminoacid residues 757 and 885, close to the C-terminal end of this1,140-amino-acid protein, are motifs characteristic of solublelytic transglycosylases (190).'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:39 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'92'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'991055'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'536ms'), (b'x-request-id', b'fc3ca411d9937932cc712655db0cff20'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124917fe245d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:39 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'92'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'991055'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'536ms'), (b'x-request-id', b'fc3ca411d9937932cc712655db0cff20'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124917fe245d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 26 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Whether bplA and 3are identical re- mains to be seen. Mutations in three other P1 genes, R, 16 , and 21, cause phenotypes identical to those of gene 26mutants (323). The products of these genes lack homology to proteinsof other phages, indicating that they form those parts of the P1baseplate whose evolutionary history is different from base-plate components of T4, and they are perhaps unique to P1. The largest tail component is the product of the sitgene (121 kDa). The Sit protein has a mosaic structure. Between aminoacid residues 757 and 885, close to the C-terminal end of this1,140-amino-acid protein, are motifs characteristic of solublelytic transglycosylases (190). Sit was proposed to contact theouter surface of the bacterial cell wall at infection and facilitatepassage of the viral genome through the cell wall by introduc-ing small gaps into its peptidoglycan layer, as does the solublelytic transglycosylase of E. coli . The structure of Sit has simi- VOL. 186, 2004 BACTERIOPHAGE P1 7057', 'page_label: 27 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  larities to that of T4 gp5, a 575-residue protein that contains a separable domain with lysozyme activity (17, 165). Posttrans-lational cleavage of gp5 allows this domain to join in the as-sembly of the baseplate and form a cell-puncturing device thatpenetrates the outer membrane and locally dissolves the cellwall at infection. Sit does not undergo major proteolytic pro-cessing; a protein of Sit ’s length can be isolated from virions of P1 (323). The maintenance of the integrity of Sit protein maybe necessary to ful ﬁll a putative second Sit function, that of a “ruler. ”Sit is the only protein long enough to be a candidate for the P1 tail length determinant. Moreover, its central region(residues 450 to 691) has homologies to the C-terminal moietyof gp29 protein of T4 and the T4-related phage RB69. In T4,gp29 is the largest baseplate hub protein (63). Additionally,gp29 serves as a tail tube ruler which ﬁlls the core of the tube during assembly and determines the tail length (9). The mod-ular organization of P1 Sit is not the same as T4 gp29 butresembles that of tail tube rulers of some mycobacteriophages,which have additional domains embedded near their C-termi-nal ends (245). Genes of P1 that determine tail ﬁbers and the speci ﬁcity of P1 adsorption to different hosts were previously localized totwo operons. One of them, the tail ﬁber operon, consists of three genes, R,S, and U, transcribed from the late promoter LPS(Tables 3 and 8) (104, 105). Sspeci ﬁes a structural protein that extends along the tail ﬁber with its constant N-terminal part at the baseplate, while its variable C-terminal part isavailable to bind a speci ﬁc receptor on the host cell surface (reviewed in reference 347). The host range speci ﬁcity of P1 is subject to alteration by inversion of the C segment of P1 DNA(Fig. 1). The C segment adjoins the anterior part of the Sgene (Sc) and contains two alternative variants of the posterior part of the Sgene ( Svand Sv/H11032) and two alternative variants of the entire Ugene ( Uand U/H11032). The alternative variants are orga- nized in oppositely oriented Sv/H11032U/H11032and SvU segments so that the inversion connects either of the segments, in frame, to theanterior part of the Sgene (135, 146, 153). In the so-called C /H11001 phage, expression of Svand Uenables infection of E. coli strains K-12 and C (153). Analysis of the predicted amino acid sequence of Sand U gene products and their alternative forms in the C /H11002phage (Sc/H11001Sv/H11032and U/H11032) con ﬁrms extensive homologies of these pro- teins to the products of the corresponding Mu genes (see TableS1 in the supplemental material), consistent with the results ofearly complementation and DNA hybridization experiments. The gpS protein of P1 appears to be extended at its N terminus by 200 residues relative to its Mu-encoded counter-part. This extra region is partly identical to the N-terminalregion of the tail ﬁber structural protein encoded by plasmid p15B, which has extensive regions of homology to P1 (268,269). The tail ﬁber genes of P1, p15B, and Mu appear to have evolved by exchange of segments within a common pool of tailﬁber genes. P1 gpU and gpU /H11032appear to be 95% identical to their Mu- encoded counterparts and have informative homologies to cer-tain tail ﬁber assembly proteins of other phages, including gp38 of T4, /H9261Tfa, and the Tfa protein of a wild strain of /H9261with tail ﬁbers (Ur- /H9261) (133). In T4, gp38 acts as a chaperone essential for the assembly of the distal part of long tail ﬁbers and is absent from virions (48; reviewed in reference 134).', 'page_label: 27 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The tail ﬁber genes of P1, p15B, and Mu appear to have evolved by exchange of segments within a common pool of tailﬁber genes. P1 gpU and gpU /H11032appear to be 95% identical to their Mu- encoded counterparts and have informative homologies to cer-tain tail ﬁber assembly proteins of other phages, including gp38 of T4, /H9261Tfa, and the Tfa protein of a wild strain of /H9261with tail ﬁbers (Ur- /H9261) (133). In T4, gp38 acts as a chaperone essential for the assembly of the distal part of long tail ﬁbers and is absent from virions (48; reviewed in reference 134). In contrastto T4 gp38, the Tfa protein of Ur- /H9261is a virion component (133). Although there is little homology between the two, eachcan replace the chaperone function of the other (117, 226).This function of Tfa does not require virion attachment.Whereas the C-terminal region of /H9261Tfa is not homologous to that of T4 gp38, it is homologous to those of P1 gpU and P1gpU/H11032. It is thus likely that the gpU and gpU /H11032proteins of P1, like/H9261Tfa, can function as both tail ﬁber assembly chaperones and virion components. Comparison of variable parts of alternative P1 gpS proteins and variable P1 gpU and gpU /H11032proteins reveals that the main sequence differences between them reside within the C-termi-nal regions (Fig. 4). Further analysis of these differences mayhelp identify subregions of gpS and gpU proteins essential foradsorption of P1 on its particular hosts. Inversion of the C segment is accomplished by the Cin site- speci ﬁc recombinase, the only product of an operon that ad- joins the C segment on the opposite side from Sc(Fig. 1). Unlike the tail ﬁber operon, which is expressed from the late promoter, the cinoperon is expressed from a /H9268 70promoter, allowing inversions of the C segment to occur in the prophage (Table 4). Cin acts on a pair of inverted 26-bp crossover sites, cixL and cixR, at the C-segment borders (Table 1). The cixL site over- laps with the end of the cingene. Inversions within the analog of the C segment in phages with similar invertible sequenceshave been reported and may well occur in P1 as well (267). Inthe p15B plasmid, whose Sgene is homologous to that of P1 in its posterior region, inversions can occur at six crossover siteswithin the corresponding invertible segment and may result in240 isomeric con ﬁgurations of this region (269, 270). The in- versions depend on the p15B invertase Min, which can func-tionally replace the Cin invertase of P1 (154). A pseudo- cixsite, cixPp, overlaps one of the two predicted cinpromoters, possibly allowing autoregulation of Cin synthe- sis (Tables 1 and 4) (135, 150). The cinsegment, ﬂanked by cixPp and cixL, does not invert at a noticeable frequency, perhaps because the cixPp site does not contain the conserved core sequence, TT, important for ef ﬁcient Cin-mediated re- combination (148). Inversion is an infrequent process and, as a result, the prog- eny of an infecting phage and the infecting phage itself aregenerally of the same host range, whereas an induced lysogenis likely to produce a burst of phages of mixed genotypes, whichare able to plate either on E. coli strains K-12 and C or on other P1 hosts. Control of inversion appears to be mediated ata palindromic recombinational enhancer located 18 bp down-stream of the translational start signal of the cingene (141).', 'page_label: 27 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The cinsegment, ﬂanked by cixPp and cixL, does not invert at a noticeable frequency, perhaps because the cixPp site does not contain the conserved core sequence, TT, important for ef ﬁcient Cin-mediated re- combination (148). Inversion is an infrequent process and, as a result, the prog- eny of an infecting phage and the infecting phage itself aregenerally of the same host range, whereas an induced lysogenis likely to produce a burst of phages of mixed genotypes, whichare able to plate either on E. coli strains K-12 and C or on other P1 hosts. Control of inversion appears to be mediated ata palindromic recombinational enhancer located 18 bp down-stream of the translational start signal of the cingene (141). The host architectural protein Fis (factor for inversion stimu-lation) binds at the enhancer, stimulates (by more than 500-fold) recombination in cisto the recombination sites (109, 142–144), and determines the topological speci ﬁcity of DNA inversion (108). An additional factor that in ﬂuences the rate of C-segment inversion is probably IHF protein. We ﬁnd a puta- tive IHF binding site overlapping one of the two /H9268 70promoters that can drive cintranscription (Tables 4 and 7). IHF protein is three times more abundant in early stationary phase and twotimes more abundant in late stationary phase than in exponen-tial phase (11). Fis levels in turn are drastically reduced instationary phase. These considerations suggest that C-segment7058 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 28 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  inversion will be especially infrequent during the stationary phase due to the predicted IHF-mediated down-regulation ofcinexpression and the lack of Fis-mediated enhancer function. Arber has suggested that the ability of Cin to mediate occa- sional deletions and intermolecular reactions (e.g., under con-ditions of relaxed topological speci ﬁcity) could lead to ex- change of tail ﬁber segments and could have been selected for during evolution (15). Tail ﬁber genes from unrelated phages of enteric bacteria exhibit a mosaic of homologous segmentssuggestive of extensive horizontal transfer (267). On the otherhand, conservation of the order of the various motifs has beentaken to imply that independent divergence from a commonancestor also played a major role in tail ﬁber evolution (342). (d) Antirestriction. Two P1 head components, DarA and DarB, protect infecting P1 DNA against degradation by dif-ferent subsets of enterobacterial type I restriction enzymes(155, 304, 325). Both Dar (defense against restriction) proteinsare synthesized late during lytic development. The correspond- ingdargenes were mapped to two unlinked operons (Fig. 1). P1darA is the fourth gene in a multigene operon, whose nucleotide sequence was determined previously (149) and cor-rected in this study. The operon is transcribed from the latepromoter LP dar (Table 8) (105) and, in addition to darA , contains two lysis-determining genes that will be describedseparately and four other genes (Fig. 1). None of these genesare essential for P1 development, as cells infected with a P1deletion mutant lacking the entire darA operon can produce infective phages. The 5 /H11032region of darA and the entire preceding gene appear to have evolved through internal duplication, as suggested bythe corresponding homology of the predicted products of bothgenes (see Table S1 in the supplemental material). To re ﬂect this property, we give the name hdf(homolog of darA frag- ment) to the gene that is directly upstream of darA .I ti s FIG. 4. Alignment of alternative tail ﬁber proteins and protein segments.VOL. 186, 2004 BACTERIOPHAGE P1 7059', 'page_label: 29 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  unclear whether the Hdf protein contributes to P1 antirestric- tion or has evolved to serve a different function. In addition to antirestriction and cell lysis, the darA operon determines three previously identi ﬁed functions designated Vad, Gta, and Tsu (reviewed in reference 347, see also refer-ence 149). The corresponding mutants have been identi ﬁed on the basis of their variable proportions of heads of differentsizes, altered generalized transduction, and transduction stim-ulation by UV, respectively. Assigning these functions to par-ticular genes of the darA operon will require further study. The DarA protein has no homologies to known proteins, other than Hdf. It is synthesized as a precursor and packed intophage particles following proteolytic processing. DarA andDarB are injected into the host cell with the phage DNA,where they act exclusively in cis. The Dar proteins do not di- rectly inactivate type I restriction-modi ﬁcation enzymes (155). Moreover P1 DNA isolated from phage particles was not pro-tected against type I restriction in vitro, suggesting that theprotection requires entry of phage DNA into cells at infection. DarB appears to be the largest P1-encoded protein (252 kDa). Analysis of the predicted amino acid sequence of DarB(see Table S1 in the supplemental material) reveals that thisgigantic protein consists of at least two domains. The N-ter-minal domain contains nine appropriately ordered amino acidsequence motifs characteristic of DNA N6-adenine methyl-transferases of type “gamma ”(Fig. 5) (reviewed in references 44 and 214) and like the majority of them, may methylateadenine residues in a TNNA motif. DarB contains within motifIV the sequence NPPY/F (with one permitted mismatch)known to be essential for adenine DNA methyltransferasecatalytic activity. The C-terminal moiety of DarB contains se- quence motifs characteristic of DNA or RNA helicases (seeTable S1 in the supplemental material). Most likely DarBmethylates DNA on infection. The DNA methylase function ofDarB is not speci ﬁc for P1 DNA. Any DNA packed into P1 heads is protected from a DarB-counteracted host restrictionsystem following its injection (155). Although expressed late in P1 lytic development, darB is not preceded by any sequence resembling a late promoter. As thefunction of darB depends on functional darA , it is possible that the product of darA (or a downstream gene in the same oper- on) is required for transcriptional activation of darB . (e) Replicative functions expressed late in lytic develop- ment. A late promoter that we designate LP hotcan drive transcription of a gene whose predicted product, an 83-residue,basic polypeptide, appears to be a homolog of the /H9258subunit of the major replicative enzyme of E. coli , DNA polymerase III, whence our gene designation, hot(homolog of theta). In the regions of homology, which include 65 amino acid residues ofboth proteins, 62% of the amino acid residues are identical. E. coli DNA polymerase III is a complex of at least 10 different protein subunits, which cooperate to ensure the rapidsynthesis of DNA with high ﬁdelity (reviewed in reference 168). The /H9258subunit binds tightly to another subunit, /H9280(proof- reading 3 /H11032-5/H11032exonuclease), which, in turn, binds to /H9251(5/H11032-3/H11032 polymerase) in a linear array that forms the catalytic core ofPolIII (76, 305). A loop region in /H9258was identi ﬁed that directly interacts with the /H9280subunit (169). It consists of nine amino acid residues (AAAGVAFKE) at position 29 of /H9280. In the Hot pro- tein of P1 this region is highly conserved (seven identical and FIG. 5. Conserved sequence motifs in the predicted amino acid sequences of the N-terminal domain of DarB and of selected /H9253-type DNA N6-adenine methyltransferases. The motifs are indicated by Roman numerals according to reference 214.', 'page_label: 29 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  A loop region in /H9258was identi ﬁed that directly interacts with the /H9280subunit (169). It consists of nine amino acid residues (AAAGVAFKE) at position 29 of /H9280. In the Hot pro- tein of P1 this region is highly conserved (seven identical and FIG. 5. Conserved sequence motifs in the predicted amino acid sequences of the N-terminal domain of DarB and of selected /H9253-type DNA N6-adenine methyltransferases. The motifs are indicated by Roman numerals according to reference 214. Methylated nucleotides in targetsequences of DarB homologs are underlined (32, 159, 162, 167, 340).7060 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 30 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  two similar residues), suggesting that Hot can replace E. coli /H9258 in the assembly of the E. coli core of PolIII, a conclusion recently con ﬁrmed (R. Schaaper, personal communication). Whereas /H9280is known to play several roles, affecting the pro- cessivity, thermal stability, and ﬁdelity of PolIII, the function of /H9258is elusive. There is some evidence to suggest a role in main- taining ﬁdelity of the enzyme via stabilization of the /H9280subunit (76, 305). No mutant phenotype has been associated with a nullmutation in holE , the structural gene for /H9258(289). Upstream of the hotgene, in addition to the late promoter LPhot, there is a 69-bp region that is 91 to 96% identical to the complement of the regulatory region of the ribonucleotidediphosphate reductase encoded in the nrdoperon of certain E. coli and Shigella ﬂexneri strains. The homologous regions in- clude two sequence motifs characteristic of E. coli DnaA pro- tein recognition sites (Table 7). Transcription of the nrdoper- on was shown to be regulated by the cooperative binding ofDnaA to the upstream sites (19, 313). Whether hotmay be subjected to regulation by DnaA bound to oppositely orientedDnaA boxes is unknown. Expression of an LP hot/H9021(hot/H11032-/H11032lacZ ) in-frame fusion from an insert that, in addition to LP hot, con- tained about 200 bp of upstream P1 DNA, was initiated coor-dinately with expression of other Lpa-controlled operons uponinduction of a P1 lysogen (275). Apparently, Hot synthesisstarts late in P1 lytic development, much later than the syn-thesis of other known P1-encoded replication proteins, whichare expressed from genes controlled directly by C1. Hot maypossibly compensate for a limitation that appears only wellafter phage replication has begun or, as a virion protein that isinjected into host cells together with P1 DNA, may facilitateearly stages of P1 replication. (f) Lysis. Lysis of the host cell to release phage progeny is the last stage of P1 lytic development. Lysis starts 60 to 70 minafter infection under normal laboratory conditions and re-leases a burst of 100 to 200 infective phage particles. Studieswith lysis-defective mutants of P1 have revealed three lyticfunctions typical of tailed phages: a primary cell-wall-degrad-ing enzyme (endolysin), a helper protein (holin), and its an-tagonist (antiholin). The function of holins is to form hole-likelesions in the cytoplasmic membrane that provide endolysinsaccess to the peptidoglycan layer of the cell wall (327, 351).Antiholins regulate the timing of lysis. They block function oftheir cognate holins until a complete set of mature phage particles is ready for release. Genes for the P1 lytic functions were located previously in two unlinked operons (277, 324) (Fig. 1), both transcriptionallyactivated by Lpa. Mutations in only one of these genes, lyz (formerly 17), located separately from the two others, prevent lysis. The lytic development of lyzmutant phages is normal until the expected lysis time. Then the production of phagesceases and even chloroform treatment of the phage-laden cellscannot induce lysis (324). This phenotype is characteristic ofphage mutants with mutations in genes for primary cell-wall-degrading enzymes (reviewed in reference 351). The productof the lyzgene was found to be a 185-residue protein homol- ogous to several proteins of the T4 lysozyme family (277),indicating that it is a “true”lysozyme capable of hydrolyzing peptidoglycan N-acetylmuramyl- /H9252(1,4) linkages. However, un- like the lysozyme of T4, P1 Lyz does not require holin for itspassage to the periplasm, at least in E. coli (341).', 'page_label: 30 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Then the production of phagesceases and even chloroform treatment of the phage-laden cellscannot induce lysis (324). This phenotype is characteristic ofphage mutants with mutations in genes for primary cell-wall-degrading enzymes (reviewed in reference 351). The productof the lyzgene was found to be a 185-residue protein homol- ogous to several proteins of the T4 lysozyme family (277),indicating that it is a “true”lysozyme capable of hydrolyzing peptidoglycan N-acetylmuramyl- /H9252(1,4) linkages. However, un- like the lysozyme of T4, P1 Lyz does not require holin for itspassage to the periplasm, at least in E. coli (341). Its N-termi- nal domain, which resembles a signal sequence, can mediateexport of Lyz to the membrane and its release into the peri-plasm in a sec-dependent manner that is not associated with cleavage of Lyz by SecA. Two other lysis-associated genes, lydA and lydB (formerly gene 2), encode a holin and antiholin, respectively (277, 324). They are the ﬁrst genes in the darA operon (Fig. 1), which also encodes functions not associated with cell lysis. LydA possessestwo putative transmembrane domains characteristic of class IIholins (327). LydB has no homologs among known proteins.Phages carrying an amber mutation in lydB lyse their host cells prematurely, i ndicating that LydB antagonizes a P1 holin (324). Analysis of the entire genome sequence of P1 reveals two additional putative holin genes (Fig. 6). One of them, desig-nated by us lydC , immediately precedes the promoter-operator region of the daroperon (Fig. 1). It can encode a 93-residue protein whose predicted amino acid sequence is 30% identicalto that of LydA and features two putative transmembranehelices, a hallmark of class II holins (327). Recent studiesconﬁrm the holin function of LydC (R. Young, personal com- munication). It is likely that either lydC evolved from a dupli- cation of the neighboring lydA or vice versa. Although the end of lydC is separated from the beginning of lydA by a 67-nt noncoding region that contains the late pro- FIG. 6. Known and putative holins encoded by P1. (Top) Alignment of the amino acid sequences of the LydA holin and the putative holin, LydC. Two putative transmembrane helices (TMH) are highlighted in grey. (Bottom) The amino acid sequence of the predicted holin LydD.Putative transmembrane helices are highlighted.V OL. 186, 2004 BACTERIOPHAGE P1 7061', 'page_label: 31 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  moter of lydA and lydB (LPdar), there is no predicted Rho- independent terminator in this interval. Conceivably lydA and lydB and perhaps other genes of the darA operon, in addition to being transcribed from the late promoter in front of lydA , can also be transcribed from a putative strong promoter thatdrives transcription of lydC . The regulatory region of lydC overlaps the regulatory region of the cingene, which is transcribed in the opposite direction. Some coregulation of the two genes is expected. The /H1100235 and /H1100210 regions of a putative lydC promoter (Table 4) overlap the sequence that acts as an enhancer of Cin-mediated C-segmentinversion and bind the Fis protein. The region between thepredicted lydC promoter and the beginning of lydC also con- tains a binding site for the Cin protein ( cixPp; Table 1) and a putative binding site for the host protein IHF (Table 7). Oc-cupation of these sites by Cin and IHF, respectively, couldpossibly repress the expression of lydC (as well as cin). Another factor that could limit the expression of lydC is a putative strong Rho-independent terminator at the beginning of thelydC coding sequence (Table 5). It is likely that transcription of lydC terminates prematurely in the absence of an ef ﬁcient antitermination mechanism. Since lydC appears subject to mul- tiple levels of repression, it is probably expressed at low levels.Expression of lydA and lydB from the late promoter LP dar, downstream of lydC , is expected to make LydA predominant over LydC late in lytic development. It is likely that LydBantiholin can control the time of lysis mediated by the relatedLydA and LydC proteins. A second predicted holin gene, which we designate lydD ,i s apparently transcribed from the same promoter, P lyz,a st h e upstream gene lyz, which partially overlaps it. It can encode an 84-residue protein, which contains two putative transmem-brane helices, like LydA and LydC. LydD contains a cluster ofﬁve highly basic (arginine) residues at its C-terminal end, an- other hallmark of phage holins (327). LydD may be the pri-mary holin involved in the P1-mediated lysis of E. coli cells, as suggested by the inability of a lyzgene clone to complement fully a lyzamber mutant, unless lydD was part of the clone (59). Polarity of the amber lyzmutation, expected from an overlap of lyzand lydD , could explain this result. It is likely that, in addition to regulation by Lpa, lyzandlydD expression depends on RNA antisense to the proximal region of the lyzgene and is terminated in that region by a predicted Rho-independentterminator (Table 5). In the central region of lydD starts yet another gene that is oriented in the same direction and is read in a different frame(Fig. 1). It can encode a 71-residue protein, which is likely to bean antiholin antagonistic to LydD and which we designatelydE. As a general rule, a phage holin gene is associated with a gene encoding its speci ﬁc antagonist (327). The most frequent organization involves a “dual start motif ”; the holin genes encode two polypeptides, read in the same frame, that differ inlength and have antagonistic functions (holin and antiholin)(103). In P1, genes for known and predicted holins and theirspeci ﬁc antiholins are either closely linked ( lydA and lydB)o r overlapping ( lydD and lydE) but their products differ signi ﬁ- cantly. Thus, the P1 lysis machinery appears atypical in itscomplexity and organization.ACKNOWLEDGMENTS We are deeply indebted to the many members of phage community for their helpful comments on a rough draft of the manuscript and fortheir encouragement.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 26 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Whether bplA and 3are identical re- mains to be seen. Mutations in three other P1 genes, R, 16 , and 21, cause phenotypes identical to those of gene 26mutants (323). The products of these genes lack homology to proteinsof other phages, indicating that they form those parts of the P1baseplate whose evolutionary history is different from base-plate components of T4, and they are perhaps unique to P1. The largest tail component is the product of the sitgene (121 kDa). The Sit protein has a mosaic structure. Between aminoacid residues 757 and 885, close to the C-terminal end of this1,140-amino-acid protein, are motifs characteristic of solublelytic transglycosylases (190). Sit was proposed to contact theouter surface of the bacterial cell wall at infection and facilitatepassage of the viral genome through the cell wall by introduc-ing small gaps into its peptidoglycan layer, as does the solublelytic transglycosylase of E. coli . The structure of Sit has simi- VOL. 186, 2004 BACTERIOPHAGE P1 7057', 'page_label: 27 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  larities to that of T4 gp5, a 575-residue protein that contains a separable domain with lysozyme activity (17, 165). Posttrans-lational cleavage of gp5 allows this domain to join in the as-sembly of the baseplate and form a cell-puncturing device thatpenetrates the outer membrane and locally dissolves the cellwall at infection. Sit does not undergo major proteolytic pro-cessing; a protein of Sit ’s length can be isolated from virions of P1 (323). The maintenance of the integrity of Sit protein maybe necessary to ful ﬁll a putative second Sit function, that of a “ruler. ”Sit is the only protein long enough to be a candidate for the P1 tail length determinant. Moreover, its central region(residues 450 to 691) has homologies to the C-terminal moietyof gp29 protein of T4 and the T4-related phage RB69. In T4,gp29 is the largest baseplate hub protein (63). Additionally,gp29 serves as a tail tube ruler which ﬁlls the core of the tube during assembly and determines the tail length (9). The mod-ular organization of P1 Sit is not the same as T4 gp29 butresembles that of tail tube rulers of some mycobacteriophages,which have additional domains embedded near their C-termi-nal ends (245). Genes of P1 that determine tail ﬁbers and the speci ﬁcity of P1 adsorption to different hosts were previously localized totwo operons. One of them, the tail ﬁber operon, consists of three genes, R,S, and U, transcribed from the late promoter LPS(Tables 3 and 8) (104, 105). Sspeci ﬁes a structural protein that extends along the tail ﬁber with its constant N-terminal part at the baseplate, while its variable C-terminal part isavailable to bind a speci ﬁc receptor on the host cell surface (reviewed in reference 347). The host range speci ﬁcity of P1 is subject to alteration by inversion of the C segment of P1 DNA(Fig. 1). The C segment adjoins the anterior part of the Sgene (Sc) and contains two alternative variants of the posterior part of the Sgene ( Svand Sv/H11032) and two alternative variants of the entire Ugene ( Uand U/H11032). The alternative variants are orga- nized in oppositely oriented Sv/H11032U/H11032and SvU segments so that the inversion connects either of the segments, in frame, to theanterior part of the Sgene (135, 146, 153). In the so-called C /H11001 phage, expression of Svand Uenables infection of E. coli strains K-12 and C (153). Analysis of the predicted amino acid sequence of Sand U gene products and their alternative forms in the C /H11002phage (Sc/H11001Sv/H11032and U/H11032) con ﬁrms extensive homologies of these pro- teins to the products of the corresponding Mu genes (see TableS1 in the supplemental material), consistent with the results ofearly complementation and DNA hybridization experiments. The gpS protein of P1 appears to be extended at its N terminus by 200 residues relative to its Mu-encoded counter-part. This extra region is partly identical to the N-terminalregion of the tail ﬁber structural protein encoded by plasmid p15B, which has extensive regions of homology to P1 (268,269). The tail ﬁber genes of P1, p15B, and Mu appear to have evolved by exchange of segments within a common pool of tailﬁber genes. P1 gpU and gpU /H11032appear to be 95% identical to their Mu- encoded counterparts and have informative homologies to cer-tain tail ﬁber assembly proteins of other phages, including gp38 of T4, /H9261Tfa, and the Tfa protein of a wild strain of /H9261with tail ﬁbers (Ur- /H9261) (133). In T4, gp38 acts as a chaperone essential for the assembly of the distal part of long tail ﬁbers and is absent from virions (48; reviewed in reference 134).', 'page_label: 27 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The tail ﬁber genes of P1, p15B, and Mu appear to have evolved by exchange of segments within a common pool of tailﬁber genes. P1 gpU and gpU /H11032appear to be 95% identical to their Mu- encoded counterparts and have informative homologies to cer-tain tail ﬁber assembly proteins of other phages, including gp38 of T4, /H9261Tfa, and the Tfa protein of a wild strain of /H9261with tail ﬁbers (Ur- /H9261) (133). In T4, gp38 acts as a chaperone essential for the assembly of the distal part of long tail ﬁbers and is absent from virions (48; reviewed in reference 134). In contrastto T4 gp38, the Tfa protein of Ur- /H9261is a virion component (133). Although there is little homology between the two, eachcan replace the chaperone function of the other (117, 226).This function of Tfa does not require virion attachment.Whereas the C-terminal region of /H9261Tfa is not homologous to that of T4 gp38, it is homologous to those of P1 gpU and P1gpU/H11032. It is thus likely that the gpU and gpU /H11032proteins of P1, like/H9261Tfa, can function as both tail ﬁber assembly chaperones and virion components. Comparison of variable parts of alternative P1 gpS proteins and variable P1 gpU and gpU /H11032proteins reveals that the main sequence differences between them reside within the C-termi-nal regions (Fig. 4). Further analysis of these differences mayhelp identify subregions of gpS and gpU proteins essential foradsorption of P1 on its particular hosts. Inversion of the C segment is accomplished by the Cin site- speci ﬁc recombinase, the only product of an operon that ad- joins the C segment on the opposite side from Sc(Fig. 1). Unlike the tail ﬁber operon, which is expressed from the late promoter, the cinoperon is expressed from a /H9268 70promoter, allowing inversions of the C segment to occur in the prophage (Table 4). Cin acts on a pair of inverted 26-bp crossover sites, cixL and cixR, at the C-segment borders (Table 1). The cixL site over- laps with the end of the cingene. Inversions within the analog of the C segment in phages with similar invertible sequenceshave been reported and may well occur in P1 as well (267). Inthe p15B plasmid, whose Sgene is homologous to that of P1 in its posterior region, inversions can occur at six crossover siteswithin the corresponding invertible segment and may result in240 isomeric con ﬁgurations of this region (269, 270). The in- versions depend on the p15B invertase Min, which can func-tionally replace the Cin invertase of P1 (154). A pseudo- cixsite, cixPp, overlaps one of the two predicted cinpromoters, possibly allowing autoregulation of Cin synthe- sis (Tables 1 and 4) (135, 150). The cinsegment, ﬂanked by cixPp and cixL, does not invert at a noticeable frequency, perhaps because the cixPp site does not contain the conserved core sequence, TT, important for ef ﬁcient Cin-mediated re- combination (148). Inversion is an infrequent process and, as a result, the prog- eny of an infecting phage and the infecting phage itself aregenerally of the same host range, whereas an induced lysogenis likely to produce a burst of phages of mixed genotypes, whichare able to plate either on E. coli strains K-12 and C or on other P1 hosts. Control of inversion appears to be mediated ata palindromic recombinational enhancer located 18 bp down-stream of the translational start signal of the cingene (141).', 'page_label: 27 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  The cinsegment, ﬂanked by cixPp and cixL, does not invert at a noticeable frequency, perhaps because the cixPp site does not contain the conserved core sequence, TT, important for ef ﬁcient Cin-mediated re- combination (148). Inversion is an infrequent process and, as a result, the prog- eny of an infecting phage and the infecting phage itself aregenerally of the same host range, whereas an induced lysogenis likely to produce a burst of phages of mixed genotypes, whichare able to plate either on E. coli strains K-12 and C or on other P1 hosts. Control of inversion appears to be mediated ata palindromic recombinational enhancer located 18 bp down-stream of the translational start signal of the cingene (141). The host architectural protein Fis (factor for inversion stimu-lation) binds at the enhancer, stimulates (by more than 500-fold) recombination in cisto the recombination sites (109, 142–144), and determines the topological speci ﬁcity of DNA inversion (108). An additional factor that in ﬂuences the rate of C-segment inversion is probably IHF protein. We ﬁnd a puta- tive IHF binding site overlapping one of the two /H9268 70promoters that can drive cintranscription (Tables 4 and 7). IHF protein is three times more abundant in early stationary phase and twotimes more abundant in late stationary phase than in exponen-tial phase (11). Fis levels in turn are drastically reduced instationary phase. These considerations suggest that C-segment7058 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 28 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  inversion will be especially infrequent during the stationary phase due to the predicted IHF-mediated down-regulation ofcinexpression and the lack of Fis-mediated enhancer function. Arber has suggested that the ability of Cin to mediate occa- sional deletions and intermolecular reactions (e.g., under con-ditions of relaxed topological speci ﬁcity) could lead to ex- change of tail ﬁber segments and could have been selected for during evolution (15). Tail ﬁber genes from unrelated phages of enteric bacteria exhibit a mosaic of homologous segmentssuggestive of extensive horizontal transfer (267). On the otherhand, conservation of the order of the various motifs has beentaken to imply that independent divergence from a commonancestor also played a major role in tail ﬁber evolution (342). (d) Antirestriction. Two P1 head components, DarA and DarB, protect infecting P1 DNA against degradation by dif-ferent subsets of enterobacterial type I restriction enzymes(155, 304, 325). Both Dar (defense against restriction) proteinsare synthesized late during lytic development. The correspond- ingdargenes were mapped to two unlinked operons (Fig. 1). P1darA is the fourth gene in a multigene operon, whose nucleotide sequence was determined previously (149) and cor-rected in this study. The operon is transcribed from the latepromoter LP dar (Table 8) (105) and, in addition to darA , contains two lysis-determining genes that will be describedseparately and four other genes (Fig. 1). None of these genesare essential for P1 development, as cells infected with a P1deletion mutant lacking the entire darA operon can produce infective phages. The 5 /H11032region of darA and the entire preceding gene appear to have evolved through internal duplication, as suggested bythe corresponding homology of the predicted products of bothgenes (see Table S1 in the supplemental material). To re ﬂect this property, we give the name hdf(homolog of darA frag- ment) to the gene that is directly upstream of darA .I ti s FIG. 4. Alignment of alternative tail ﬁber proteins and protein segments.VOL. 186, 2004 BACTERIOPHAGE P1 7059', 'page_label: 29 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  unclear whether the Hdf protein contributes to P1 antirestric- tion or has evolved to serve a different function. In addition to antirestriction and cell lysis, the darA operon determines three previously identi ﬁed functions designated Vad, Gta, and Tsu (reviewed in reference 347, see also refer-ence 149). The corresponding mutants have been identi ﬁed on the basis of their variable proportions of heads of differentsizes, altered generalized transduction, and transduction stim-ulation by UV, respectively. Assigning these functions to par-ticular genes of the darA operon will require further study. The DarA protein has no homologies to known proteins, other than Hdf. It is synthesized as a precursor and packed intophage particles following proteolytic processing. DarA andDarB are injected into the host cell with the phage DNA,where they act exclusively in cis. The Dar proteins do not di- rectly inactivate type I restriction-modi ﬁcation enzymes (155). Moreover P1 DNA isolated from phage particles was not pro-tected against type I restriction in vitro, suggesting that theprotection requires entry of phage DNA into cells at infection. DarB appears to be the largest P1-encoded protein (252 kDa). Analysis of the predicted amino acid sequence of DarB(see Table S1 in the supplemental material) reveals that thisgigantic protein consists of at least two domains. The N-ter-minal domain contains nine appropriately ordered amino acidsequence motifs characteristic of DNA N6-adenine methyl-transferases of type “gamma ”(Fig. 5) (reviewed in references 44 and 214) and like the majority of them, may methylateadenine residues in a TNNA motif. DarB contains within motifIV the sequence NPPY/F (with one permitted mismatch)known to be essential for adenine DNA methyltransferasecatalytic activity. The C-terminal moiety of DarB contains se- quence motifs characteristic of DNA or RNA helicases (seeTable S1 in the supplemental material). Most likely DarBmethylates DNA on infection. The DNA methylase function ofDarB is not speci ﬁc for P1 DNA. Any DNA packed into P1 heads is protected from a DarB-counteracted host restrictionsystem following its injection (155). Although expressed late in P1 lytic development, darB is not preceded by any sequence resembling a late promoter. As thefunction of darB depends on functional darA , it is possible that the product of darA (or a downstream gene in the same oper- on) is required for transcriptional activation of darB . (e) Replicative functions expressed late in lytic develop- ment. A late promoter that we designate LP hotcan drive transcription of a gene whose predicted product, an 83-residue,basic polypeptide, appears to be a homolog of the /H9258subunit of the major replicative enzyme of E. coli , DNA polymerase III, whence our gene designation, hot(homolog of theta). In the regions of homology, which include 65 amino acid residues ofboth proteins, 62% of the amino acid residues are identical. E. coli DNA polymerase III is a complex of at least 10 different protein subunits, which cooperate to ensure the rapidsynthesis of DNA with high ﬁdelity (reviewed in reference 168). The /H9258subunit binds tightly to another subunit, /H9280(proof- reading 3 /H11032-5/H11032exonuclease), which, in turn, binds to /H9251(5/H11032-3/H11032 polymerase) in a linear array that forms the catalytic core ofPolIII (76, 305). A loop region in /H9258was identi ﬁed that directly interacts with the /H9280subunit (169). It consists of nine amino acid residues (AAAGVAFKE) at position 29 of /H9280. In the Hot pro- tein of P1 this region is highly conserved (seven identical and FIG. 5. Conserved sequence motifs in the predicted amino acid sequences of the N-terminal domain of DarB and of selected /H9253-type DNA N6-adenine methyltransferases. The motifs are indicated by Roman numerals according to reference 214.', 'page_label: 29 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  A loop region in /H9258was identi ﬁed that directly interacts with the /H9280subunit (169). It consists of nine amino acid residues (AAAGVAFKE) at position 29 of /H9280. In the Hot pro- tein of P1 this region is highly conserved (seven identical and FIG. 5. Conserved sequence motifs in the predicted amino acid sequences of the N-terminal domain of DarB and of selected /H9253-type DNA N6-adenine methyltransferases. The motifs are indicated by Roman numerals according to reference 214. Methylated nucleotides in targetsequences of DarB homologs are underlined (32, 159, 162, 167, 340).7060 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 30 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  two similar residues), suggesting that Hot can replace E. coli /H9258 in the assembly of the E. coli core of PolIII, a conclusion recently con ﬁrmed (R. Schaaper, personal communication). Whereas /H9280is known to play several roles, affecting the pro- cessivity, thermal stability, and ﬁdelity of PolIII, the function of /H9258is elusive. There is some evidence to suggest a role in main- taining ﬁdelity of the enzyme via stabilization of the /H9280subunit (76, 305). No mutant phenotype has been associated with a nullmutation in holE , the structural gene for /H9258(289). Upstream of the hotgene, in addition to the late promoter LPhot, there is a 69-bp region that is 91 to 96% identical to the complement of the regulatory region of the ribonucleotidediphosphate reductase encoded in the nrdoperon of certain E. coli and Shigella ﬂexneri strains. The homologous regions in- clude two sequence motifs characteristic of E. coli DnaA pro- tein recognition sites (Table 7). Transcription of the nrdoper- on was shown to be regulated by the cooperative binding ofDnaA to the upstream sites (19, 313). Whether hotmay be subjected to regulation by DnaA bound to oppositely orientedDnaA boxes is unknown. Expression of an LP hot/H9021(hot/H11032-/H11032lacZ ) in-frame fusion from an insert that, in addition to LP hot, con- tained about 200 bp of upstream P1 DNA, was initiated coor-dinately with expression of other Lpa-controlled operons uponinduction of a P1 lysogen (275). Apparently, Hot synthesisstarts late in P1 lytic development, much later than the syn-thesis of other known P1-encoded replication proteins, whichare expressed from genes controlled directly by C1. Hot maypossibly compensate for a limitation that appears only wellafter phage replication has begun or, as a virion protein that isinjected into host cells together with P1 DNA, may facilitateearly stages of P1 replication. (f) Lysis. Lysis of the host cell to release phage progeny is the last stage of P1 lytic development. Lysis starts 60 to 70 minafter infection under normal laboratory conditions and re-leases a burst of 100 to 200 infective phage particles. Studieswith lysis-defective mutants of P1 have revealed three lyticfunctions typical of tailed phages: a primary cell-wall-degrad-ing enzyme (endolysin), a helper protein (holin), and its an-tagonist (antiholin). The function of holins is to form hole-likelesions in the cytoplasmic membrane that provide endolysinsaccess to the peptidoglycan layer of the cell wall (327, 351).Antiholins regulate the timing of lysis. They block function oftheir cognate holins until a complete set of mature phage particles is ready for release. Genes for the P1 lytic functions were located previously in two unlinked operons (277, 324) (Fig. 1), both transcriptionallyactivated by Lpa. Mutations in only one of these genes, lyz (formerly 17), located separately from the two others, prevent lysis. The lytic development of lyzmutant phages is normal until the expected lysis time. Then the production of phagesceases and even chloroform treatment of the phage-laden cellscannot induce lysis (324). This phenotype is characteristic ofphage mutants with mutations in genes for primary cell-wall-degrading enzymes (reviewed in reference 351). The productof the lyzgene was found to be a 185-residue protein homol- ogous to several proteins of the T4 lysozyme family (277),indicating that it is a “true”lysozyme capable of hydrolyzing peptidoglycan N-acetylmuramyl- /H9252(1,4) linkages. However, un- like the lysozyme of T4, P1 Lyz does not require holin for itspassage to the periplasm, at least in E. coli (341).', 'page_label: 30 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Then the production of phagesceases and even chloroform treatment of the phage-laden cellscannot induce lysis (324). This phenotype is characteristic ofphage mutants with mutations in genes for primary cell-wall-degrading enzymes (reviewed in reference 351). The productof the lyzgene was found to be a 185-residue protein homol- ogous to several proteins of the T4 lysozyme family (277),indicating that it is a “true”lysozyme capable of hydrolyzing peptidoglycan N-acetylmuramyl- /H9252(1,4) linkages. However, un- like the lysozyme of T4, P1 Lyz does not require holin for itspassage to the periplasm, at least in E. coli (341). Its N-termi- nal domain, which resembles a signal sequence, can mediateexport of Lyz to the membrane and its release into the peri-plasm in a sec-dependent manner that is not associated with cleavage of Lyz by SecA. Two other lysis-associated genes, lydA and lydB (formerly gene 2), encode a holin and antiholin, respectively (277, 324). They are the ﬁrst genes in the darA operon (Fig. 1), which also encodes functions not associated with cell lysis. LydA possessestwo putative transmembrane domains characteristic of class IIholins (327). LydB has no homologs among known proteins.Phages carrying an amber mutation in lydB lyse their host cells prematurely, i ndicating that LydB antagonizes a P1 holin (324). Analysis of the entire genome sequence of P1 reveals two additional putative holin genes (Fig. 6). One of them, desig-nated by us lydC , immediately precedes the promoter-operator region of the daroperon (Fig. 1). It can encode a 93-residue protein whose predicted amino acid sequence is 30% identicalto that of LydA and features two putative transmembranehelices, a hallmark of class II holins (327). Recent studiesconﬁrm the holin function of LydC (R. Young, personal com- munication). It is likely that either lydC evolved from a dupli- cation of the neighboring lydA or vice versa. Although the end of lydC is separated from the beginning of lydA by a 67-nt noncoding region that contains the late pro- FIG. 6. Known and putative holins encoded by P1. (Top) Alignment of the amino acid sequences of the LydA holin and the putative holin, LydC. Two putative transmembrane helices (TMH) are highlighted in grey. (Bottom) The amino acid sequence of the predicted holin LydD.Putative transmembrane helices are highlighted.V OL. 186, 2004 BACTERIOPHAGE P1 7061', 'page_label: 31 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  moter of lydA and lydB (LPdar), there is no predicted Rho- independent terminator in this interval. Conceivably lydA and lydB and perhaps other genes of the darA operon, in addition to being transcribed from the late promoter in front of lydA , can also be transcribed from a putative strong promoter thatdrives transcription of lydC . The regulatory region of lydC overlaps the regulatory region of the cingene, which is transcribed in the opposite direction. Some coregulation of the two genes is expected. The /H1100235 and /H1100210 regions of a putative lydC promoter (Table 4) overlap the sequence that acts as an enhancer of Cin-mediated C-segmentinversion and bind the Fis protein. The region between thepredicted lydC promoter and the beginning of lydC also con- tains a binding site for the Cin protein ( cixPp; Table 1) and a putative binding site for the host protein IHF (Table 7). Oc-cupation of these sites by Cin and IHF, respectively, couldpossibly repress the expression of lydC (as well as cin). Another factor that could limit the expression of lydC is a putative strong Rho-independent terminator at the beginning of thelydC coding sequence (Table 5). It is likely that transcription of lydC terminates prematurely in the absence of an ef ﬁcient antitermination mechanism. Since lydC appears subject to mul- tiple levels of repression, it is probably expressed at low levels.Expression of lydA and lydB from the late promoter LP dar, downstream of lydC , is expected to make LydA predominant over LydC late in lytic development. It is likely that LydBantiholin can control the time of lysis mediated by the relatedLydA and LydC proteins. A second predicted holin gene, which we designate lydD ,i s apparently transcribed from the same promoter, P lyz,a st h e upstream gene lyz, which partially overlaps it. It can encode an 84-residue protein, which contains two putative transmem-brane helices, like LydA and LydC. LydD contains a cluster ofﬁve highly basic (arginine) residues at its C-terminal end, an- other hallmark of phage holins (327). LydD may be the pri-mary holin involved in the P1-mediated lysis of E. coli cells, as suggested by the inability of a lyzgene clone to complement fully a lyzamber mutant, unless lydD was part of the clone (59). Polarity of the amber lyzmutation, expected from an overlap of lyzand lydD , could explain this result. It is likely that, in addition to regulation by Lpa, lyzandlydD expression depends on RNA antisense to the proximal region of the lyzgene and is terminated in that region by a predicted Rho-independentterminator (Table 5). In the central region of lydD starts yet another gene that is oriented in the same direction and is read in a different frame(Fig. 1). It can encode a 71-residue protein, which is likely to bean antiholin antagonistic to LydD and which we designatelydE. As a general rule, a phage holin gene is associated with a gene encoding its speci ﬁc antagonist (327). The most frequent organization involves a “dual start motif ”; the holin genes encode two polypeptides, read in the same frame, that differ inlength and have antagonistic functions (holin and antiholin)(103). In P1, genes for known and predicted holins and theirspeci ﬁc antiholins are either closely linked ( lydA and lydB)o r overlapping ( lydD and lydE) but their products differ signi ﬁ- cantly. Thus, the P1 lysis machinery appears atypical in itscomplexity and organization.ACKNOWLEDGMENTS We are deeply indebted to the many members of phage community for their helpful comments on a rough draft of the manuscript and fortheir encouragement.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:40 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'90'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993323'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'400ms'), (b'x-request-id', b'bdb1d287cc1ff4c72bbee2c660e683f6'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412491cfafe5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:40 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'90'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993323'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'400ms'), (b'x-request-id', b'bdb1d287cc1ff4c72bbee2c660e683f6'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412491cfafe5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 31 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  As a general rule, a phage holin gene is associated with a gene encoding its speci ﬁc antagonist (327). The most frequent organization involves a “dual start motif ”; the holin genes encode two polypeptides, read in the same frame, that differ inlength and have antagonistic functions (holin and antiholin)(103). In P1, genes for known and predicted holins and theirspeci ﬁc antiholins are either closely linked ( lydA and lydB)o r overlapping ( lydD and lydE) but their products differ signi ﬁ- cantly. Thus, the P1 lysis machinery appears atypical in itscomplexity and organization.ACKNOWLEDGMENTS We are deeply indebted to the many members of phage community for their helpful comments on a rough draft of the manuscript and fortheir encouragement. We owe particular thanks to colleagues whohave reviewed the initial draft in its entirety —Hans W. Ackermann, Richard D ’Ari, David Lane, and Erich Miller —and to Sherwood Casjens and Roger Hendrix for suggestions concerning phage morpho-genesis. We thank Judah L. Rosner for providing the N1706 strainfrom which P1 was induced, W łodzimierz Zago ´rski-Ostoja for crucial support of the project at the IBB PAS, and Sankar Adhya and Wac ław Szybalski for their help in initiating this collaborative effort. The main part of this work, including sequencing of P1 c1-100 mod749 ::IS5, was supported by grant Nr6P04B01116 from the Polish State Committee for Scienti ﬁc Research (KBN). Sequencing of the second P1 strain, P1 c1-100 mod1902 ::IS5 dmt /H9004MB rev-6, was sup- ported by NIH grants P01HG01428 and S10RR10379. REFERENCES 1.Abeles, A., T. Brendler, and S. Austin. 1993. Evidence of two levels of control of P1 oriR and host oriC replication origins by DNA adenine methylation. J. Bacteriol. 175:7801 –7807. 2.Abeles, A. L. 1986. P1 plasmid replication. Puri ﬁcation and DNA-binding activity of the replication protein RepA. J. Biol. Chem. 261:3548 –3555. 3.Abeles, A. L., and S. J. Austin. 1991. Antiparallel plasmid-plasmid pairing may control P1 plasmid replication. Proc. Natl. Acad. Sci. USA 88:9011 – 9015. 4.Abeles, A. L., S. A. Friedman, and S. J. Austin. 1985. Partition of unit-copy miniplasmids to daughter cells. III. The DNA sequence and functionalorganization of the P1 partition region. J. Mol. Biol. 185:261–272. 5.Abeles, A. L., L. D. Reaves, and S. J. Austin. 1989. Protein-DNA interac- tions in regulation of P1 plasmid replication. J. Bacteriol. 171:43–52. 6.Abeles, A. L., K. M. Snyder, and D. K. Chattoraj. 1984. P1 plasmid repli- cation: replicon structure. J. Mol. Biol. 173:307–324. 7.Abremski, K., and R. Hoess. 1984. Bacteriophage P1 site-speci ﬁc recombi- nation. Puri ﬁcation and properties of the Cre recombinase protein. J. Biol. Chem. 259:1509 –1514. 8.Abremski, K., R. Hoess, and N. Sternberg. 1983. Studies on the properties of P1 site-speci ﬁc recombination: evidence for topologically unlinked prod- ucts following recombination. Cell 32:1301 –1311. 9.Abuladze, N. K., M. Gingery, J. Tsai, and F. A. Eiserling. 1994. Tail length determination in bacteriophage T4.', 'page_label: 31 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  J. Mol. Biol. 173:307–324. 7.Abremski, K., and R. Hoess. 1984. Bacteriophage P1 site-speci ﬁc recombi- nation. Puri ﬁcation and properties of the Cre recombinase protein. J. Biol. Chem. 259:1509 –1514. 8.Abremski, K., R. Hoess, and N. Sternberg. 1983. Studies on the properties of P1 site-speci ﬁc recombination: evidence for topologically unlinked prod- ucts following recombination. Cell 32:1301 –1311. 9.Abuladze, N. K., M. Gingery, J. Tsai, and F. A. Eiserling. 1994. Tail length determination in bacteriophage T4. Virology 199:301–310. 10.Adams, D. E., J. B. Bliska, and N. Cozzarelli. 1992. Cre- loxrecombination inEscherichia coli cells: mechanistic differences from the in vitro reaction. J. Mol. Biol. 226:661–673. 11.Ali Azam, T., A. Iwata, A. Nishimura, S. Ueda, and A. Ishihama. 1999. Growth phase-dependent variation in protein composition of the Esche- richia coli nucleoid. J. Bacteriol. 181:6361 –6370. 12.Altschul, S. F., T. L. Madden, A. A. Schaffer, J. Zhang, Z. Zhang, W. Miller, and D. J. Lipman. 1997. Gapped BLAST and PSI-BLAST: a new genera- tion of protein database search programs. Nucleic Acids Res. 25:3389 –3402. 13.Anantharaman, V., and L. Aravind. 2003. New connections in the prokary- otic toxin-antitoxin network: relationship with the eukaryotic nonsense-mediated RNA decay system. Genome Biol. 4:R81. 14.Anderson, T. F., and D. H. J. Walker. 1960. Morphological variants of the bacteriophage P1. Science 132:1488. 15.Arber, W. 1991. Elements in microbial evolution. J. Mol. Evol. 33:4–12. 16.Arber, W., and D. Dussoix. 1962. Host speci ﬁcity of DNA produced by Escherichia coli. I. Host controlled modi ﬁcation of bacteriophage /H9261. J. Mol. Biol. 5:18–36. 17.Arisaka, F., S. Kanamaru, P. Leiman, and M. G. Rossmann. 2003. The tail lysozyme complex of bacteriophage T4. Int. J. Biochem. Cell Biol. 35:16–21. 18.Artsimovitch, I., K. Murakami, A. Ishihama, and M. M. Howe. 1996. Tran- scription activation by the bacteriophage Mu Mor protein requires theC-terminal regions of both alpha and sigma70 subunits of Escherichia coli RNA polymerase. J. Biol. Chem. 271:32343 –32348. 19.Augustin, L. B., B. A. Jacobson, and J. A. Fuchs. 1994. Escherichia coli Fis and DnaA proteins bind speci ﬁcally to the nrdpromoter region and affect expression of an nrd-lac fusion. J. Bacteriol. 176:378–387. 20.Austin, S., and A. Abeles. 1983. Partition of unit-copy miniplasmids to daughter cells. II. The partition region of miniplasmid P1 encodes anessential protein and a centromere-like site at which it acts. J. Mol. Biol.169:373–387. 21.Austin, S., M. Ziese, and N. Sternberg. 1981. A novel role for site-speci ﬁc recombination in maintenance of bacterial replicons.', 'page_label: 31 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1994. Escherichia coli Fis and DnaA proteins bind speci ﬁcally to the nrdpromoter region and affect expression of an nrd-lac fusion. J. Bacteriol. 176:378–387. 20.Austin, S., and A. Abeles. 1983. Partition of unit-copy miniplasmids to daughter cells. II. The partition region of miniplasmid P1 encodes anessential protein and a centromere-like site at which it acts. J. Mol. Biol.169:373–387. 21.Austin, S., M. Ziese, and N. Sternberg. 1981. A novel role for site-speci ﬁc recombination in maintenance of bacterial replicons. Cell 25:729–736. 22.Ba¨chi, B., and W. Arber. 1977. Physical mapping of BglII,Bam HI,EcoRI, HindIII and PstI restriction fragments of bacteriophage P1 DNA. Mol. Gen. Genet. 153:311–324. 23.Ba¨chi, B., J. Reiser, and V. Pirrotta. 1979. Methylation and cleavage se-7062 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 32 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  quences of the EcoP1 restriction-modi ﬁcation enzyme. J. Mol. Biol. 128: 143–163. 24.Barik, S. 1993. Expression and biochemical properties of a protein serine/ threonine phosphatase encoded by bacteriophage lambda. Proc. Natl.Acad. Sci. USA 90:10633 –10637. 25.Baumstark, B. R., S. R. Stovall, and P. Bralley. 1990. The ImmC region of phage Pl codes for a gene whose product promotes lytic growth. Virology179:217–227. 26.Baumstark, B. R., and J. R. Scott. 1987. The c4gene of phage P1. Virology 156:197–203. 27.Baumstark, B. R., S. R. Stovall, and S. Ashkar. 1987. Interaction of the Plcl repressor with Pl DNA. Localization of repressor binding sites near the cl gene. Virology 156:404–413. 28.Bendtsen, J. D., A. S. Nilsson, and H. Lehnherr. 2002. Phylogenetic and functional analysis of the bacteriophage P1 single-stranded DNA-bindingprotein. J. Virol. 76:9695 –9701. 29.Bertani, G. 1958. Lysogeny. Adv. Virus Res. 5:151–193. 30.Bertani, G. 2004. Lysogeny at mid-twentieth century: P1, P2, and other experimental systems. J. Bacteriol. 186:595–600. 31.Bertani, G. 1951. Studies on lysogenesis. I. The mode of phage liberation by lysogenic Escherichia coli . J. Bacteriol. 62:293–300. 32.Bickle, T. A., and D. H. Kru ¨ger. 1993. Biology of DNA restriction. Micro- biol. Rev. 57:434–450. 33.Biere, A. L., M. Citron, and H. Schuster. 1992. Transcriptional control via translational repression by c4 antisense RNA of bacteriophages P1 and P7.Genes Dev. 6:2409 –2416. 34.Black, L., M. Showe, and A. Steven. 1994. Morphogenesis of the T4 head., p. 218 –258. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 35.Blattner, F. R., G. Plunkett III, C. A. Bloch, N. T. Perna, V. Burland, M. Riley, J. Collado-Vides, J. D. Glasner, C. K. Rode, G. F. Mayhew, J. Gregor,N. W. Davis, H. A. Kirkpatrick, M. A. Goeden, D. J. Rose, B. Mau, and Y.Shao. 1997. The complete genome sequence of Escherichia coli K-12. Sci- ence 277:1453 –1474. 36.Borodovsky, M., and J. McIninch. 1993. Recognition of genes in DNA sequence with ambiguities. Biosystems 30:161–171. 37.Bouet, J. Y., and B. E. Funnell. 1999. P1 ParA interacts with the P1 partition complex at parS and an ATP-ADP switch controls ParA activities. EMBO J.18:1415 –1424. 38.Bourniquel, A. A., and T. A. Bickle. 2002. Complex restriction enzymes: NTP-driven molecular motors. Biochimie 84:1047 –1059. 39.Braithwaite, D. K., and J. Ito. 1993. Compilation, alignment, and phyloge- netic relationships of DNA polymerases. Nucleic Acids Res. 21:787–802. 40.Brendel, V., and E. N. Trifonov. 1984.', 'page_label: 32 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Biosystems 30:161–171. 37.Bouet, J. Y., and B. E. Funnell. 1999. P1 ParA interacts with the P1 partition complex at parS and an ATP-ADP switch controls ParA activities. EMBO J.18:1415 –1424. 38.Bourniquel, A. A., and T. A. Bickle. 2002. Complex restriction enzymes: NTP-driven molecular motors. Biochimie 84:1047 –1059. 39.Braithwaite, D. K., and J. Ito. 1993. Compilation, alignment, and phyloge- netic relationships of DNA polymerases. Nucleic Acids Res. 21:787–802. 40.Brendel, V., and E. N. Trifonov. 1984. A computer algorithm testing po- tential prokaryotic terminators. Nucleic Acids Res. 12:4411 –4427. 41.Brockes, J. P., P. R. Brown, and K. Murray. 1972. The deoxyribonucleic acid modi ﬁcation enzyme of bacteriophage P1. Biochem. J. 127:1–10. 42.Brockes, J. P., P. R. Brown, and K. Murray. 1974. Nucleotide sequences at the sites of action of the deoxyribonucleic acid modi ﬁcation enzyme of bacteriophage P1. J. Mol. Biol. 88:437–443. 43.Bruck, I., R. Woodgate, K. McEntee, and M. F. Goodman. 1996. Puri ﬁca- tion of a soluble UmuD /H11032C complex from Escherichia coli. Cooperative binding of UmuD /H11032C to single-stranded DNA. J. Biol. Chem. 271:10767 – 10774. 44.Bujnicki, J. M. 2001. Understanding the evolution of restriction-modi ﬁca- tion systems: clues from sequence and structure comparisons. Acta Bio-chim. Pol. 48:935–967. 45.Burland, V., D. L. Daniels, G. Plunkett III, and F. R. Blattner. 1993. Genome sequencing on both strands: the Janus strategy. Nucleic Acids Res.21:3385 –3390. 46.Campbell, A. 2003. Prophage insertion sites. Res. Microbiol. 154:277–282. 47.Capiaux, H., C. Lesterlin, K. Perals, J. M. Louarn, and F. Cornet. 2002. A dual role for the FtsK protein in Escherichia coli chromosome segregation. EMBO Rep. 3:532–536. 48.Cerritelli, M. E., J. S. Wall, M. N. Simon, J. F. Conway, and A. C. Steven. 1996. Stoichiometry and domainal organization of the long tail- ﬁber of bacteriophage T4: a hinged viral adhesin. J. Mol. Biol. 260:767–780. 49.Chattoraj, D. K. 2000. Control of plasmid DNA replication by iterons: no longer paradoxical. Mol. Microbiol. 37:467–476. 50.Chattoraj, D. K., R. J. Mason, and S. H. Wickner. 1988. Mini-P1 plasmid replication: the autoregulation-sequestration paradox. Cell 52:551–557. 51.Chattoraj, D. K., and T. D. Schneider. 1997. Replication control of plasmid P1 and its host chromosome: the common ground. Prog. Nucleic Acid Res.Mol. Biol. 57:145–186. 52.Chattoraj, D. K., K. M. Snyder, and A. L. Abeles. 1985. P1 plasmid repli- cation: multiple functions of RepA protein at the origin. Proc. Natl. Acad.Sci.', 'page_label: 32 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Mol. Microbiol. 37:467–476. 50.Chattoraj, D. K., R. J. Mason, and S. H. Wickner. 1988. Mini-P1 plasmid replication: the autoregulation-sequestration paradox. Cell 52:551–557. 51.Chattoraj, D. K., and T. D. Schneider. 1997. Replication control of plasmid P1 and its host chromosome: the common ground. Prog. Nucleic Acid Res.Mol. Biol. 57:145–186. 52.Chattoraj, D. K., K. M. Snyder, and A. L. Abeles. 1985. P1 plasmid repli- cation: multiple functions of RepA protein at the origin. Proc. Natl. Acad.Sci. USA 82:2588 –2592. 53.Cheng, S. W., D. L. Court, and D. I. Friedman. 1995. Transcription termi- nation signals in the nin region of bacteriophage lambda: identi ﬁcation of Rho-dependent termination regions. Genetics 140:875–887. 54.Chesney, R. H., and J. R. Scott. 1975. Superinfection immunity and pro-phage repression in phage P1. II. Mapping of the immunity-difference and ampicillin-resistance loci of P1 and phi amp. Virology 67:375–384. 55.Chiang, L. W., and M. M. Howe. 1993. Mutational analysis of a C-depen- dent late promoter of bacteriophage Mu. Genetics 135:619–629. 56.Christie, G. E., L. M. Temple, B. A. Bartlett, and T. S. Goodwin. 2002. Programmed translational frameshift in the bacteriophage P2 FETUD tailgene operon. J. Bacteriol. 184:6522 –6531. 57.Citron, M., and H. Schuster. 1992. The c4 repressor of bacteriophage P1 is a processed 77 base antisense RNA. Nucleic Acids Res. 20:3085 –3090. 58.Citron, M., and H. Schuster. 1990. The c4 repressors of bacteriophages Pl and P7 are antisense RNAs. Cell 62:591–598. 59.Citron, M., M. Velleman, and H. Schuster. 1989. Three additional opera- tors, Op21, Op68, and Op88, of bacteriophage P1. Evidence for the controlof the P1 dam methylase by Op68. J. Biol. Chem. 264:3611 –3617. 60.Cohen, G. 1983. Electron microscopy study of early lytic replication forms of bacteriophage P1 DNA. Virology 131:159–170. 61.Cohen, G., E. Or, W. Minas, and N. L. Sternberg. 1996. The bacteriophage P1 lytic replicon: directionality of replication and cis-acting elements. Gene 175:151–155. 62.Cohen, P. T., and P. Cohen. 1989. Discovery of a protein phosphatase activity encoded in the genome of bacteriophage lambda. Probable identitywith open reading frame 221. Biochem. J. 260:931–934. 63.Coombs, D. H., and F. Arisaka. 1994. T4 tail structure and function., p. 259 –281. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 64.Corre, J., and J. M. Louarn. 2002. Evidence from terminal recombination gradients that FtsK uses replichore polarity to control chromosome termi-nus positioning at division in Escherichia coli . J. Bacteriol. 184:3801 –3807. 65.Costantino, N., M. Zuber, and D. Court. 1990.', 'page_label: 32 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Discovery of a protein phosphatase activity encoded in the genome of bacteriophage lambda. Probable identitywith open reading frame 221. Biochem. J. 260:931–934. 63.Coombs, D. H., and F. Arisaka. 1994. T4 tail structure and function., p. 259 –281. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 64.Corre, J., and J. M. Louarn. 2002. Evidence from terminal recombination gradients that FtsK uses replichore polarity to control chromosome termi-nus positioning at division in Escherichia coli . J. Bacteriol. 184:3801 –3807. 65.Costantino, N., M. Zuber, and D. Court. 1990. Analysis of mutations in the ninR region of bacteriophage lambda that bypass a requirement for lambda N antitermination. J. Bacteriol. 172:4610 –4615. 66.Coulby, J. N., and N. L. Sternberg. 1988. Characterization of the phage P1 dam gene. Gene 74:191. 67.D’Ari, R., A. Jaffe ´-Brachet, D. Touati-Schwartz, and M. B. Yarmolinsky. 1975. A dnaB analog speci ﬁed by bacteriophage P1. J. Mol. Biol. 94:341– 366. 68.Dartois, V., O. De Backer, and C. Colson. 1993. Sequence of the Salmonella typhimurium StyLT1 restriction-modi ﬁcation genes: homologies with EcoP1 andEcoP15 type-III R-M systems and presence of helicase domains. Gene 127:105–110. 69.d’Aubenton Carafa, Y., E. Brody, and C. Thermes. 1990. Prediction of rho-independent Escherichia coli transcription terminators. A statistical analysis of their RNA stem-loop structures. J. Mol. Biol. 216:835–858. 70.Davey, M. J., and B. E. Funnell. 1997. Modulation of the P1 plasmid partition protein ParA by ATP, ADP, and P1 ParB. J. Biol. Chem. 272: 15286 –15292. 71.Davey, M. J., and B. E. Funnell. 1994. The P1 plasmid partition protein ParA. A role for ATP in site-speci ﬁc DNA binding. J. Biol. Chem. 269: 29908 –29913. 72.Davis, M. A., and S. J. Austin. 1988. Recognition of the P1 plasmid cen- tromere analog involves binding of the ParB protein and is modi ﬁed by a speci ﬁc host factor. EMBO J. 7:1881 –1888. 73.Davis, M. A., K. A. Martin, and S. J. Austin. 1992. Biochemical activities of the ParA partition protein of the P1 plasmid. Mol. Microbiol. 6:1141 –1147. 74.Delarue, M., O. Poch, N. Tordo, D. Moras, and P. Argos. 1990. An attempt to unify the structure of polymerases. Protein Eng. 3:461–467. 75.Delcher, A. L., D. Harmon, S. Kasif, O. White, and S. L. Salzberg. 1999. Improved microbial gene identi ﬁcation with GLIMMER. Nucleic Acids Res. 27:4636 –4641. 76.DeRose, E. F., T. Darden, S. Harvey, S. Gabel, F. W. Perrino, R. M. Schaaper, and R. E. London. 2003. Elucidation of the epsilon-theta subunit interface of Escherichia coli DNA polymerase III by NMR spectroscopy. Biochemistry 42:3635 –3644.', 'page_label: 32 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1990. An attempt to unify the structure of polymerases. Protein Eng. 3:461–467. 75.Delcher, A. L., D. Harmon, S. Kasif, O. White, and S. L. Salzberg. 1999. Improved microbial gene identi ﬁcation with GLIMMER. Nucleic Acids Res. 27:4636 –4641. 76.DeRose, E. F., T. Darden, S. Harvey, S. Gabel, F. W. Perrino, R. M. Schaaper, and R. E. London. 2003. Elucidation of the epsilon-theta subunit interface of Escherichia coli DNA polymerase III by NMR spectroscopy. Biochemistry 42:3635 –3644. 77.Devlin, B. H., B. R. Baumstark, and J. R. Scott. 1982. Superimmunity: characterization of a new gene in the immunity region of P1. Virology 120: 360–375. 78.Dodd, I. B., and J. B. Egan. 1990. Improved detection of helix-turn-helix DNA-binding motifs in protein sequences. Nucleic Acids Res. 18:5019 – 5026. 79.Dreiseikelmann, B., M. Velleman, and H. Schuster. 1988. The c1 repressor of bacteriophage P1. Isolation and characterization of the repressor pro-tein. J. Biol. Chem. 263:1391 –1397. 80.Ebersbach, G., and K. Gerdes. 2001. The double parlocus of virulence factor pB171: DNA segregation is correlated with oscillation of ParA. Proc.Natl. Acad. Sci. USA 98:15078 –15083. 81.Eddy, S. R., and R. Durbin. 1994. RNA sequence analysis using covariance models. Nucleic Acids Res. 22:2079 –2088. 82.Edelbluth, C., E. Lanka, W. von der Hude, M. Mikolajczyk, and H. Schus- ter.1979. Association of the prophage P1ban protein with the dnaB protein ofEscherichia coli. Overproduction of ban protein by a P1 bac crr mutant. Eur. J. Biochem. 94:427–435.VOL. 186, 2004 BACTERIOPHAGE P1 7063', 'page_label: 33 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  83.Edgar, R., D. K. Chattoraj, and M. Yarmolinsky. 2001. Pairing of P1 plasmid partition sites by ParB. Mol. Microbiol. 42:1363 –1370. 84.Eliason, J. L., and N. Sternberg. 1987. Characterization of the binding sites of cl repressor of bacteriophage Pl: evidence for multiple asymmetric sites.J. Mol. Biol. 198:281–293. 85.Ermolaeva, M. D., H. G. Khalak, O. White, H. O. Smith, and S. L. Salzberg. 2000. Prediction of transcription terminators in bacterial genomes. J. Mol.Biol. 301:27–33. 86.Ferguson, P. L., and D. H. Coombs. 2000. Pulse-chase analysis of the in vivo assembly of the bacteriophage T4 tail. J. Mol. Biol. 297:99–117. 87.Friedman, S. A., and S. J. Austin. 1988. The P1 plasmid-partition system synthesizes two essential proteins from an autoregulated operon. Plasmid19:103–112. 88.Fukada, K., and J. Abelson. 1980. DNA sequence of a T4 transfer RNA gene cluster. J. Mol. Biol. 139:377–391. 89.Funnell, B. E. 1988. Mini-P1 plasmid partitioning: excess ParB protein destabilizes plasmids containing the centromere parS . J. Bacteriol. 170:954– 960. 90.Funnell, B. E. 1991. The P1 plasmid partition complex at parS. The in ﬂu- ence of Escherichia coli integration host factor and of substrate topology. J. Biol. Chem. 266:14328 –14337. 91.Funnell, B. E. 1988. Participation of Escherichia coli integration host factor in the P1 plasmid partition system. Proc. Natl. Acad. Sci. USA 85:6657 – 6661. 92.Funnell, B. E., and L. Gagnier. 1993. The P1 plasmid partition complex at parS. II. Analysis of ParB protein binding activity and speci ﬁcity. J. Biol. Chem. 268:3616 –3624. 93.Gamas, P., M. G. Chandler, P. Prentki, and D. J. Galas. 1987. Escherichia coli integration host factor binds speci ﬁcally to the ends of the insertion sequence IS 1and to its major insertion hot-spot in pBR322. J. Mol. Biol. 195:261–272. 94.Gamas, P., D. Galas, and M. Chandler. 1985. DNA sequence at the end of IS1required for transposition. Nature 317:458–460. 95.Gazit, E., and R. T. Sauer. 1999. The Doc toxin and Phd antidote proteins of the bacteriophage P1 plasmid addiction system form a heterotrimericcomplex. J. Biol. Chem. 274:16813 –16818. 96.Gazit, E., and R. T. Sauer. 1999. Stability and DNA binding of the phd protein of the phage P1 plasmid addiction system. J. Biol. Chem. 274:2652 – 2657. 97.Gerdes, K., J. M øller-Jensen, G. Ebersbach, T. Kruse, and K. Nordstro ¨m. 2004. Bacterial mitotic machineries. Cell 116:359–366. 98.Ghisotti, D., F. Briani, F. Forti, F. Piazza, S. Polo, P. Sabbattini, T. Stur- niolo, S. Terzano, S. Zangrossi, M. Zappone, G. Sironi, and G. Deho `.1995. Multiple regulatory mechanisms controlling phage-plasmid P4 propagation.FEMS Microbiol.', 'page_label: 33 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1999. Stability and DNA binding of the phd protein of the phage P1 plasmid addiction system. J. Biol. Chem. 274:2652 – 2657. 97.Gerdes, K., J. M øller-Jensen, G. Ebersbach, T. Kruse, and K. Nordstro ¨m. 2004. Bacterial mitotic machineries. Cell 116:359–366. 98.Ghisotti, D., F. Briani, F. Forti, F. Piazza, S. Polo, P. Sabbattini, T. Stur- niolo, S. Terzano, S. Zangrossi, M. Zappone, G. Sironi, and G. Deho `.1995. Multiple regulatory mechanisms controlling phage-plasmid P4 propagation.FEMS Microbiol. Rev. 17:127–134. 99.Goldman, E., A. H. Rosenberg, G. Zubay, and F. W. Studier. 1995. Con- secutive low-usage leucine codons block translation only when near the 5 /H11032 end of a message in Escherichia coli . J. Mol. Biol. 245:467–473. 100. Gottesman, S., W. P. Clark, V. de Crecy-Lagard, and M. R. Maurizi. 1993. ClpX, an alternative subunit for the ATP-dependent Clp protease of Esch- erichia coli. Sequence and in vivo activities. J. Biol. Chem. 268:22618 –22626. 101. Grigoriev, A. 1999. Strand-speci ﬁc compositional asymmetries in double- stranded DNA viruses. Virus Res. 60:1–19. 102. Grigoriev, P., and M. B. Łobocka. 2001. Determinants of segregational stability of the linear plasmid prophage N15 of Escherichia coli . Mol. Mi- crobiol. 42:355–368. 103. Gru¨ndling, A., M. D. Manson, and R. Young. 2001. Holins kill without warning. Proc. Natl. Acad. Sci. USA 98:9348 –9352. 104. Guidolin, A., J. M. Zingg, and W. Arber. 1989. Organization of the bacte- riophage P1 tail- ﬁbre operon. Gene 76:239–243. 105. Guidolin, A., J. M. Zingg, H. Lehnherr, and W. Arber. 1989. Bacteriophage P1 tail- ﬁbre and daroperons are expressed from homologous phage-speci ﬁc late promoter sequences. J. Mol. Biol. 208:615–622. 106. Guo, F., D. N. Gopaul, and G. D. van Duyne. 1997. Structure of Cre re- combinase complexed with DNA in a site-speci ﬁc recombination synapse. Nature 389:40–46. 107. Hadi, S. M., B. Ba ¨chi, S. Iida, and T. A. Bickle. 1983. DNA restriction- modi ﬁcation enzymes of phage P1 and plasmid p15B. Subunit functions and structural homologies. J. Mol. Biol. 165:19–34. 108. Haffter, P., and T. A. Bickle. 1988. Enhancer-independent mutants of the Cin recombinase have a relaxed topological speci ﬁcity. EMBO J. 7:3991 – 3996. 109. Haffter, P., and T. A. Bickle. 1987. Puri ﬁcation and DNA-binding properties of FIS and Cin, two proteins required for the bacteriophage P1 site-speci ﬁc recombination system, cin. J. Mol. Biol. 198:579–587. 110.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 31 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  As a general rule, a phage holin gene is associated with a gene encoding its speci ﬁc antagonist (327). The most frequent organization involves a “dual start motif ”; the holin genes encode two polypeptides, read in the same frame, that differ inlength and have antagonistic functions (holin and antiholin)(103). In P1, genes for known and predicted holins and theirspeci ﬁc antiholins are either closely linked ( lydA and lydB)o r overlapping ( lydD and lydE) but their products differ signi ﬁ- cantly. Thus, the P1 lysis machinery appears atypical in itscomplexity and organization.ACKNOWLEDGMENTS We are deeply indebted to the many members of phage community for their helpful comments on a rough draft of the manuscript and fortheir encouragement. We owe particular thanks to colleagues whohave reviewed the initial draft in its entirety —Hans W. Ackermann, Richard D ’Ari, David Lane, and Erich Miller —and to Sherwood Casjens and Roger Hendrix for suggestions concerning phage morpho-genesis. We thank Judah L. Rosner for providing the N1706 strainfrom which P1 was induced, W łodzimierz Zago ´rski-Ostoja for crucial support of the project at the IBB PAS, and Sankar Adhya and Wac ław Szybalski for their help in initiating this collaborative effort. The main part of this work, including sequencing of P1 c1-100 mod749 ::IS5, was supported by grant Nr6P04B01116 from the Polish State Committee for Scienti ﬁc Research (KBN). Sequencing of the second P1 strain, P1 c1-100 mod1902 ::IS5 dmt /H9004MB rev-6, was sup- ported by NIH grants P01HG01428 and S10RR10379. REFERENCES 1.Abeles, A., T. Brendler, and S. Austin. 1993. Evidence of two levels of control of P1 oriR and host oriC replication origins by DNA adenine methylation. J. Bacteriol. 175:7801 –7807. 2.Abeles, A. L. 1986. P1 plasmid replication. Puri ﬁcation and DNA-binding activity of the replication protein RepA. J. Biol. Chem. 261:3548 –3555. 3.Abeles, A. L., and S. J. Austin. 1991. Antiparallel plasmid-plasmid pairing may control P1 plasmid replication. Proc. Natl. Acad. Sci. USA 88:9011 – 9015. 4.Abeles, A. L., S. A. Friedman, and S. J. Austin. 1985. Partition of unit-copy miniplasmids to daughter cells. III. The DNA sequence and functionalorganization of the P1 partition region. J. Mol. Biol. 185:261–272. 5.Abeles, A. L., L. D. Reaves, and S. J. Austin. 1989. Protein-DNA interac- tions in regulation of P1 plasmid replication. J. Bacteriol. 171:43–52. 6.Abeles, A. L., K. M. Snyder, and D. K. Chattoraj. 1984. P1 plasmid repli- cation: replicon structure. J. Mol. Biol. 173:307–324. 7.Abremski, K., and R. Hoess. 1984. Bacteriophage P1 site-speci ﬁc recombi- nation. Puri ﬁcation and properties of the Cre recombinase protein. J. Biol. Chem. 259:1509 –1514. 8.Abremski, K., R. Hoess, and N. Sternberg. 1983. Studies on the properties of P1 site-speci ﬁc recombination: evidence for topologically unlinked prod- ucts following recombination. Cell 32:1301 –1311. 9.Abuladze, N. K., M. Gingery, J. Tsai, and F. A. Eiserling. 1994. Tail length determination in bacteriophage T4.', 'page_label: 31 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  J. Mol. Biol. 173:307–324. 7.Abremski, K., and R. Hoess. 1984. Bacteriophage P1 site-speci ﬁc recombi- nation. Puri ﬁcation and properties of the Cre recombinase protein. J. Biol. Chem. 259:1509 –1514. 8.Abremski, K., R. Hoess, and N. Sternberg. 1983. Studies on the properties of P1 site-speci ﬁc recombination: evidence for topologically unlinked prod- ucts following recombination. Cell 32:1301 –1311. 9.Abuladze, N. K., M. Gingery, J. Tsai, and F. A. Eiserling. 1994. Tail length determination in bacteriophage T4. Virology 199:301–310. 10.Adams, D. E., J. B. Bliska, and N. Cozzarelli. 1992. Cre- loxrecombination inEscherichia coli cells: mechanistic differences from the in vitro reaction. J. Mol. Biol. 226:661–673. 11.Ali Azam, T., A. Iwata, A. Nishimura, S. Ueda, and A. Ishihama. 1999. Growth phase-dependent variation in protein composition of the Esche- richia coli nucleoid. J. Bacteriol. 181:6361 –6370. 12.Altschul, S. F., T. L. Madden, A. A. Schaffer, J. Zhang, Z. Zhang, W. Miller, and D. J. Lipman. 1997. Gapped BLAST and PSI-BLAST: a new genera- tion of protein database search programs. Nucleic Acids Res. 25:3389 –3402. 13.Anantharaman, V., and L. Aravind. 2003. New connections in the prokary- otic toxin-antitoxin network: relationship with the eukaryotic nonsense-mediated RNA decay system. Genome Biol. 4:R81. 14.Anderson, T. F., and D. H. J. Walker. 1960. Morphological variants of the bacteriophage P1. Science 132:1488. 15.Arber, W. 1991. Elements in microbial evolution. J. Mol. Evol. 33:4–12. 16.Arber, W., and D. Dussoix. 1962. Host speci ﬁcity of DNA produced by Escherichia coli. I. Host controlled modi ﬁcation of bacteriophage /H9261. J. Mol. Biol. 5:18–36. 17.Arisaka, F., S. Kanamaru, P. Leiman, and M. G. Rossmann. 2003. The tail lysozyme complex of bacteriophage T4. Int. J. Biochem. Cell Biol. 35:16–21. 18.Artsimovitch, I., K. Murakami, A. Ishihama, and M. M. Howe. 1996. Tran- scription activation by the bacteriophage Mu Mor protein requires theC-terminal regions of both alpha and sigma70 subunits of Escherichia coli RNA polymerase. J. Biol. Chem. 271:32343 –32348. 19.Augustin, L. B., B. A. Jacobson, and J. A. Fuchs. 1994. Escherichia coli Fis and DnaA proteins bind speci ﬁcally to the nrdpromoter region and affect expression of an nrd-lac fusion. J. Bacteriol. 176:378–387. 20.Austin, S., and A. Abeles. 1983. Partition of unit-copy miniplasmids to daughter cells. II. The partition region of miniplasmid P1 encodes anessential protein and a centromere-like site at which it acts. J. Mol. Biol.169:373–387. 21.Austin, S., M. Ziese, and N. Sternberg. 1981. A novel role for site-speci ﬁc recombination in maintenance of bacterial replicons.', 'page_label: 31 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1994. Escherichia coli Fis and DnaA proteins bind speci ﬁcally to the nrdpromoter region and affect expression of an nrd-lac fusion. J. Bacteriol. 176:378–387. 20.Austin, S., and A. Abeles. 1983. Partition of unit-copy miniplasmids to daughter cells. II. The partition region of miniplasmid P1 encodes anessential protein and a centromere-like site at which it acts. J. Mol. Biol.169:373–387. 21.Austin, S., M. Ziese, and N. Sternberg. 1981. A novel role for site-speci ﬁc recombination in maintenance of bacterial replicons. Cell 25:729–736. 22.Ba¨chi, B., and W. Arber. 1977. Physical mapping of BglII,Bam HI,EcoRI, HindIII and PstI restriction fragments of bacteriophage P1 DNA. Mol. Gen. Genet. 153:311–324. 23.Ba¨chi, B., J. Reiser, and V. Pirrotta. 1979. Methylation and cleavage se-7062 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 32 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  quences of the EcoP1 restriction-modi ﬁcation enzyme. J. Mol. Biol. 128: 143–163. 24.Barik, S. 1993. Expression and biochemical properties of a protein serine/ threonine phosphatase encoded by bacteriophage lambda. Proc. Natl.Acad. Sci. USA 90:10633 –10637. 25.Baumstark, B. R., S. R. Stovall, and P. Bralley. 1990. The ImmC region of phage Pl codes for a gene whose product promotes lytic growth. Virology179:217–227. 26.Baumstark, B. R., and J. R. Scott. 1987. The c4gene of phage P1. Virology 156:197–203. 27.Baumstark, B. R., S. R. Stovall, and S. Ashkar. 1987. Interaction of the Plcl repressor with Pl DNA. Localization of repressor binding sites near the cl gene. Virology 156:404–413. 28.Bendtsen, J. D., A. S. Nilsson, and H. Lehnherr. 2002. Phylogenetic and functional analysis of the bacteriophage P1 single-stranded DNA-bindingprotein. J. Virol. 76:9695 –9701. 29.Bertani, G. 1958. Lysogeny. Adv. Virus Res. 5:151–193. 30.Bertani, G. 2004. Lysogeny at mid-twentieth century: P1, P2, and other experimental systems. J. Bacteriol. 186:595–600. 31.Bertani, G. 1951. Studies on lysogenesis. I. The mode of phage liberation by lysogenic Escherichia coli . J. Bacteriol. 62:293–300. 32.Bickle, T. A., and D. H. Kru ¨ger. 1993. Biology of DNA restriction. Micro- biol. Rev. 57:434–450. 33.Biere, A. L., M. Citron, and H. Schuster. 1992. Transcriptional control via translational repression by c4 antisense RNA of bacteriophages P1 and P7.Genes Dev. 6:2409 –2416. 34.Black, L., M. Showe, and A. Steven. 1994. Morphogenesis of the T4 head., p. 218 –258. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 35.Blattner, F. R., G. Plunkett III, C. A. Bloch, N. T. Perna, V. Burland, M. Riley, J. Collado-Vides, J. D. Glasner, C. K. Rode, G. F. Mayhew, J. Gregor,N. W. Davis, H. A. Kirkpatrick, M. A. Goeden, D. J. Rose, B. Mau, and Y.Shao. 1997. The complete genome sequence of Escherichia coli K-12. Sci- ence 277:1453 –1474. 36.Borodovsky, M., and J. McIninch. 1993. Recognition of genes in DNA sequence with ambiguities. Biosystems 30:161–171. 37.Bouet, J. Y., and B. E. Funnell. 1999. P1 ParA interacts with the P1 partition complex at parS and an ATP-ADP switch controls ParA activities. EMBO J.18:1415 –1424. 38.Bourniquel, A. A., and T. A. Bickle. 2002. Complex restriction enzymes: NTP-driven molecular motors. Biochimie 84:1047 –1059. 39.Braithwaite, D. K., and J. Ito. 1993. Compilation, alignment, and phyloge- netic relationships of DNA polymerases. Nucleic Acids Res. 21:787–802. 40.Brendel, V., and E. N. Trifonov. 1984.', 'page_label: 32 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Biosystems 30:161–171. 37.Bouet, J. Y., and B. E. Funnell. 1999. P1 ParA interacts with the P1 partition complex at parS and an ATP-ADP switch controls ParA activities. EMBO J.18:1415 –1424. 38.Bourniquel, A. A., and T. A. Bickle. 2002. Complex restriction enzymes: NTP-driven molecular motors. Biochimie 84:1047 –1059. 39.Braithwaite, D. K., and J. Ito. 1993. Compilation, alignment, and phyloge- netic relationships of DNA polymerases. Nucleic Acids Res. 21:787–802. 40.Brendel, V., and E. N. Trifonov. 1984. A computer algorithm testing po- tential prokaryotic terminators. Nucleic Acids Res. 12:4411 –4427. 41.Brockes, J. P., P. R. Brown, and K. Murray. 1972. The deoxyribonucleic acid modi ﬁcation enzyme of bacteriophage P1. Biochem. J. 127:1–10. 42.Brockes, J. P., P. R. Brown, and K. Murray. 1974. Nucleotide sequences at the sites of action of the deoxyribonucleic acid modi ﬁcation enzyme of bacteriophage P1. J. Mol. Biol. 88:437–443. 43.Bruck, I., R. Woodgate, K. McEntee, and M. F. Goodman. 1996. Puri ﬁca- tion of a soluble UmuD /H11032C complex from Escherichia coli. Cooperative binding of UmuD /H11032C to single-stranded DNA. J. Biol. Chem. 271:10767 – 10774. 44.Bujnicki, J. M. 2001. Understanding the evolution of restriction-modi ﬁca- tion systems: clues from sequence and structure comparisons. Acta Bio-chim. Pol. 48:935–967. 45.Burland, V., D. L. Daniels, G. Plunkett III, and F. R. Blattner. 1993. Genome sequencing on both strands: the Janus strategy. Nucleic Acids Res.21:3385 –3390. 46.Campbell, A. 2003. Prophage insertion sites. Res. Microbiol. 154:277–282. 47.Capiaux, H., C. Lesterlin, K. Perals, J. M. Louarn, and F. Cornet. 2002. A dual role for the FtsK protein in Escherichia coli chromosome segregation. EMBO Rep. 3:532–536. 48.Cerritelli, M. E., J. S. Wall, M. N. Simon, J. F. Conway, and A. C. Steven. 1996. Stoichiometry and domainal organization of the long tail- ﬁber of bacteriophage T4: a hinged viral adhesin. J. Mol. Biol. 260:767–780. 49.Chattoraj, D. K. 2000. Control of plasmid DNA replication by iterons: no longer paradoxical. Mol. Microbiol. 37:467–476. 50.Chattoraj, D. K., R. J. Mason, and S. H. Wickner. 1988. Mini-P1 plasmid replication: the autoregulation-sequestration paradox. Cell 52:551–557. 51.Chattoraj, D. K., and T. D. Schneider. 1997. Replication control of plasmid P1 and its host chromosome: the common ground. Prog. Nucleic Acid Res.Mol. Biol. 57:145–186. 52.Chattoraj, D. K., K. M. Snyder, and A. L. Abeles. 1985. P1 plasmid repli- cation: multiple functions of RepA protein at the origin. Proc. Natl. Acad.Sci.', 'page_label: 32 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Mol. Microbiol. 37:467–476. 50.Chattoraj, D. K., R. J. Mason, and S. H. Wickner. 1988. Mini-P1 plasmid replication: the autoregulation-sequestration paradox. Cell 52:551–557. 51.Chattoraj, D. K., and T. D. Schneider. 1997. Replication control of plasmid P1 and its host chromosome: the common ground. Prog. Nucleic Acid Res.Mol. Biol. 57:145–186. 52.Chattoraj, D. K., K. M. Snyder, and A. L. Abeles. 1985. P1 plasmid repli- cation: multiple functions of RepA protein at the origin. Proc. Natl. Acad.Sci. USA 82:2588 –2592. 53.Cheng, S. W., D. L. Court, and D. I. Friedman. 1995. Transcription termi- nation signals in the nin region of bacteriophage lambda: identi ﬁcation of Rho-dependent termination regions. Genetics 140:875–887. 54.Chesney, R. H., and J. R. Scott. 1975. Superinfection immunity and pro-phage repression in phage P1. II. Mapping of the immunity-difference and ampicillin-resistance loci of P1 and phi amp. Virology 67:375–384. 55.Chiang, L. W., and M. M. Howe. 1993. Mutational analysis of a C-depen- dent late promoter of bacteriophage Mu. Genetics 135:619–629. 56.Christie, G. E., L. M. Temple, B. A. Bartlett, and T. S. Goodwin. 2002. Programmed translational frameshift in the bacteriophage P2 FETUD tailgene operon. J. Bacteriol. 184:6522 –6531. 57.Citron, M., and H. Schuster. 1992. The c4 repressor of bacteriophage P1 is a processed 77 base antisense RNA. Nucleic Acids Res. 20:3085 –3090. 58.Citron, M., and H. Schuster. 1990. The c4 repressors of bacteriophages Pl and P7 are antisense RNAs. Cell 62:591–598. 59.Citron, M., M. Velleman, and H. Schuster. 1989. Three additional opera- tors, Op21, Op68, and Op88, of bacteriophage P1. Evidence for the controlof the P1 dam methylase by Op68. J. Biol. Chem. 264:3611 –3617. 60.Cohen, G. 1983. Electron microscopy study of early lytic replication forms of bacteriophage P1 DNA. Virology 131:159–170. 61.Cohen, G., E. Or, W. Minas, and N. L. Sternberg. 1996. The bacteriophage P1 lytic replicon: directionality of replication and cis-acting elements. Gene 175:151–155. 62.Cohen, P. T., and P. Cohen. 1989. Discovery of a protein phosphatase activity encoded in the genome of bacteriophage lambda. Probable identitywith open reading frame 221. Biochem. J. 260:931–934. 63.Coombs, D. H., and F. Arisaka. 1994. T4 tail structure and function., p. 259 –281. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 64.Corre, J., and J. M. Louarn. 2002. Evidence from terminal recombination gradients that FtsK uses replichore polarity to control chromosome termi-nus positioning at division in Escherichia coli . J. Bacteriol. 184:3801 –3807. 65.Costantino, N., M. Zuber, and D. Court. 1990.', 'page_label: 32 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Discovery of a protein phosphatase activity encoded in the genome of bacteriophage lambda. Probable identitywith open reading frame 221. Biochem. J. 260:931–934. 63.Coombs, D. H., and F. Arisaka. 1994. T4 tail structure and function., p. 259 –281. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 64.Corre, J., and J. M. Louarn. 2002. Evidence from terminal recombination gradients that FtsK uses replichore polarity to control chromosome termi-nus positioning at division in Escherichia coli . J. Bacteriol. 184:3801 –3807. 65.Costantino, N., M. Zuber, and D. Court. 1990. Analysis of mutations in the ninR region of bacteriophage lambda that bypass a requirement for lambda N antitermination. J. Bacteriol. 172:4610 –4615. 66.Coulby, J. N., and N. L. Sternberg. 1988. Characterization of the phage P1 dam gene. Gene 74:191. 67.D’Ari, R., A. Jaffe ´-Brachet, D. Touati-Schwartz, and M. B. Yarmolinsky. 1975. A dnaB analog speci ﬁed by bacteriophage P1. J. Mol. Biol. 94:341– 366. 68.Dartois, V., O. De Backer, and C. Colson. 1993. Sequence of the Salmonella typhimurium StyLT1 restriction-modi ﬁcation genes: homologies with EcoP1 andEcoP15 type-III R-M systems and presence of helicase domains. Gene 127:105–110. 69.d’Aubenton Carafa, Y., E. Brody, and C. Thermes. 1990. Prediction of rho-independent Escherichia coli transcription terminators. A statistical analysis of their RNA stem-loop structures. J. Mol. Biol. 216:835–858. 70.Davey, M. J., and B. E. Funnell. 1997. Modulation of the P1 plasmid partition protein ParA by ATP, ADP, and P1 ParB. J. Biol. Chem. 272: 15286 –15292. 71.Davey, M. J., and B. E. Funnell. 1994. The P1 plasmid partition protein ParA. A role for ATP in site-speci ﬁc DNA binding. J. Biol. Chem. 269: 29908 –29913. 72.Davis, M. A., and S. J. Austin. 1988. Recognition of the P1 plasmid cen- tromere analog involves binding of the ParB protein and is modi ﬁed by a speci ﬁc host factor. EMBO J. 7:1881 –1888. 73.Davis, M. A., K. A. Martin, and S. J. Austin. 1992. Biochemical activities of the ParA partition protein of the P1 plasmid. Mol. Microbiol. 6:1141 –1147. 74.Delarue, M., O. Poch, N. Tordo, D. Moras, and P. Argos. 1990. An attempt to unify the structure of polymerases. Protein Eng. 3:461–467. 75.Delcher, A. L., D. Harmon, S. Kasif, O. White, and S. L. Salzberg. 1999. Improved microbial gene identi ﬁcation with GLIMMER. Nucleic Acids Res. 27:4636 –4641. 76.DeRose, E. F., T. Darden, S. Harvey, S. Gabel, F. W. Perrino, R. M. Schaaper, and R. E. London. 2003. Elucidation of the epsilon-theta subunit interface of Escherichia coli DNA polymerase III by NMR spectroscopy. Biochemistry 42:3635 –3644.', 'page_label: 32 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1990. An attempt to unify the structure of polymerases. Protein Eng. 3:461–467. 75.Delcher, A. L., D. Harmon, S. Kasif, O. White, and S. L. Salzberg. 1999. Improved microbial gene identi ﬁcation with GLIMMER. Nucleic Acids Res. 27:4636 –4641. 76.DeRose, E. F., T. Darden, S. Harvey, S. Gabel, F. W. Perrino, R. M. Schaaper, and R. E. London. 2003. Elucidation of the epsilon-theta subunit interface of Escherichia coli DNA polymerase III by NMR spectroscopy. Biochemistry 42:3635 –3644. 77.Devlin, B. H., B. R. Baumstark, and J. R. Scott. 1982. Superimmunity: characterization of a new gene in the immunity region of P1. Virology 120: 360–375. 78.Dodd, I. B., and J. B. Egan. 1990. Improved detection of helix-turn-helix DNA-binding motifs in protein sequences. Nucleic Acids Res. 18:5019 – 5026. 79.Dreiseikelmann, B., M. Velleman, and H. Schuster. 1988. The c1 repressor of bacteriophage P1. Isolation and characterization of the repressor pro-tein. J. Biol. Chem. 263:1391 –1397. 80.Ebersbach, G., and K. Gerdes. 2001. The double parlocus of virulence factor pB171: DNA segregation is correlated with oscillation of ParA. Proc.Natl. Acad. Sci. USA 98:15078 –15083. 81.Eddy, S. R., and R. Durbin. 1994. RNA sequence analysis using covariance models. Nucleic Acids Res. 22:2079 –2088. 82.Edelbluth, C., E. Lanka, W. von der Hude, M. Mikolajczyk, and H. Schus- ter.1979. Association of the prophage P1ban protein with the dnaB protein ofEscherichia coli. Overproduction of ban protein by a P1 bac crr mutant. Eur. J. Biochem. 94:427–435.VOL. 186, 2004 BACTERIOPHAGE P1 7063', 'page_label: 33 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  83.Edgar, R., D. K. Chattoraj, and M. Yarmolinsky. 2001. Pairing of P1 plasmid partition sites by ParB. Mol. Microbiol. 42:1363 –1370. 84.Eliason, J. L., and N. Sternberg. 1987. Characterization of the binding sites of cl repressor of bacteriophage Pl: evidence for multiple asymmetric sites.J. Mol. Biol. 198:281–293. 85.Ermolaeva, M. D., H. G. Khalak, O. White, H. O. Smith, and S. L. Salzberg. 2000. Prediction of transcription terminators in bacterial genomes. J. Mol.Biol. 301:27–33. 86.Ferguson, P. L., and D. H. Coombs. 2000. Pulse-chase analysis of the in vivo assembly of the bacteriophage T4 tail. J. Mol. Biol. 297:99–117. 87.Friedman, S. A., and S. J. Austin. 1988. The P1 plasmid-partition system synthesizes two essential proteins from an autoregulated operon. Plasmid19:103–112. 88.Fukada, K., and J. Abelson. 1980. DNA sequence of a T4 transfer RNA gene cluster. J. Mol. Biol. 139:377–391. 89.Funnell, B. E. 1988. Mini-P1 plasmid partitioning: excess ParB protein destabilizes plasmids containing the centromere parS . J. Bacteriol. 170:954– 960. 90.Funnell, B. E. 1991. The P1 plasmid partition complex at parS. The in ﬂu- ence of Escherichia coli integration host factor and of substrate topology. J. Biol. Chem. 266:14328 –14337. 91.Funnell, B. E. 1988. Participation of Escherichia coli integration host factor in the P1 plasmid partition system. Proc. Natl. Acad. Sci. USA 85:6657 – 6661. 92.Funnell, B. E., and L. Gagnier. 1993. The P1 plasmid partition complex at parS. II. Analysis of ParB protein binding activity and speci ﬁcity. J. Biol. Chem. 268:3616 –3624. 93.Gamas, P., M. G. Chandler, P. Prentki, and D. J. Galas. 1987. Escherichia coli integration host factor binds speci ﬁcally to the ends of the insertion sequence IS 1and to its major insertion hot-spot in pBR322. J. Mol. Biol. 195:261–272. 94.Gamas, P., D. Galas, and M. Chandler. 1985. DNA sequence at the end of IS1required for transposition. Nature 317:458–460. 95.Gazit, E., and R. T. Sauer. 1999. The Doc toxin and Phd antidote proteins of the bacteriophage P1 plasmid addiction system form a heterotrimericcomplex. J. Biol. Chem. 274:16813 –16818. 96.Gazit, E., and R. T. Sauer. 1999. Stability and DNA binding of the phd protein of the phage P1 plasmid addiction system. J. Biol. Chem. 274:2652 – 2657. 97.Gerdes, K., J. M øller-Jensen, G. Ebersbach, T. Kruse, and K. Nordstro ¨m. 2004. Bacterial mitotic machineries. Cell 116:359–366. 98.Ghisotti, D., F. Briani, F. Forti, F. Piazza, S. Polo, P. Sabbattini, T. Stur- niolo, S. Terzano, S. Zangrossi, M. Zappone, G. Sironi, and G. Deho `.1995. Multiple regulatory mechanisms controlling phage-plasmid P4 propagation.FEMS Microbiol.', 'page_label: 33 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1999. Stability and DNA binding of the phd protein of the phage P1 plasmid addiction system. J. Biol. Chem. 274:2652 – 2657. 97.Gerdes, K., J. M øller-Jensen, G. Ebersbach, T. Kruse, and K. Nordstro ¨m. 2004. Bacterial mitotic machineries. Cell 116:359–366. 98.Ghisotti, D., F. Briani, F. Forti, F. Piazza, S. Polo, P. Sabbattini, T. Stur- niolo, S. Terzano, S. Zangrossi, M. Zappone, G. Sironi, and G. Deho `.1995. Multiple regulatory mechanisms controlling phage-plasmid P4 propagation.FEMS Microbiol. Rev. 17:127–134. 99.Goldman, E., A. H. Rosenberg, G. Zubay, and F. W. Studier. 1995. Con- secutive low-usage leucine codons block translation only when near the 5 /H11032 end of a message in Escherichia coli . J. Mol. Biol. 245:467–473. 100. Gottesman, S., W. P. Clark, V. de Crecy-Lagard, and M. R. Maurizi. 1993. ClpX, an alternative subunit for the ATP-dependent Clp protease of Esch- erichia coli. Sequence and in vivo activities. J. Biol. Chem. 268:22618 –22626. 101. Grigoriev, A. 1999. Strand-speci ﬁc compositional asymmetries in double- stranded DNA viruses. Virus Res. 60:1–19. 102. Grigoriev, P., and M. B. Łobocka. 2001. Determinants of segregational stability of the linear plasmid prophage N15 of Escherichia coli . Mol. Mi- crobiol. 42:355–368. 103. Gru¨ndling, A., M. D. Manson, and R. Young. 2001. Holins kill without warning. Proc. Natl. Acad. Sci. USA 98:9348 –9352. 104. Guidolin, A., J. M. Zingg, and W. Arber. 1989. Organization of the bacte- riophage P1 tail- ﬁbre operon. Gene 76:239–243. 105. Guidolin, A., J. M. Zingg, H. Lehnherr, and W. Arber. 1989. Bacteriophage P1 tail- ﬁbre and daroperons are expressed from homologous phage-speci ﬁc late promoter sequences. J. Mol. Biol. 208:615–622. 106. Guo, F., D. N. Gopaul, and G. D. van Duyne. 1997. Structure of Cre re- combinase complexed with DNA in a site-speci ﬁc recombination synapse. Nature 389:40–46. 107. Hadi, S. M., B. Ba ¨chi, S. Iida, and T. A. Bickle. 1983. DNA restriction- modi ﬁcation enzymes of phage P1 and plasmid p15B. Subunit functions and structural homologies. J. Mol. Biol. 165:19–34. 108. Haffter, P., and T. A. Bickle. 1988. Enhancer-independent mutants of the Cin recombinase have a relaxed topological speci ﬁcity. EMBO J. 7:3991 – 3996. 109. Haffter, P., and T. A. Bickle. 1987. Puri ﬁcation and DNA-binding properties of FIS and Cin, two proteins required for the bacteriophage P1 site-speci ﬁc recombination system, cin. J. Mol. Biol. 198:579–587. 110.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:41 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'73'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993752'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'374ms'), (b'x-request-id', b'a59299f3d67da6232ce5df61ad0246e0'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124921ffec5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:41 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'73'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993752'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'374ms'), (b'x-request-id', b'a59299f3d67da6232ce5df61ad0246e0'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124921ffec5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 33 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1983. DNA restriction- modi ﬁcation enzymes of phage P1 and plasmid p15B. Subunit functions and structural homologies. J. Mol. Biol. 165:19–34. 108. Haffter, P., and T. A. Bickle. 1988. Enhancer-independent mutants of the Cin recombinase have a relaxed topological speci ﬁcity. EMBO J. 7:3991 – 3996. 109. Haffter, P., and T. A. Bickle. 1987. Puri ﬁcation and DNA-binding properties of FIS and Cin, two proteins required for the bacteriophage P1 site-speci ﬁc recombination system, cin. J. Mol. Biol. 198:579–587. 110. Hambly, E., F. Tetart, C. Desplats, W. H. Wilson, H. M. Krisch, and N. H. Mann. 2001. A conserved genetic module that encodes the major virion components in both the coliphage T4 and the marine cyanophage S-PM2.Proc. Natl. Acad. Sci. USA 98:11411 –11416. 111. Hansen, A.-M., H. Lehnherr, X. Wang, V. Mobley, and D. J. Jin. 2003. Escherichia coli SspA is a transcription activator for bacteriophage P1 late genes. Mol. Microbiol. 48:1621 –1631.112. Hansen, E. B. 1989. Structure and regulation of the lytic replicon of phage Pl. J. Mol. Biol. 207:135–149. 113. Hansen, E. B., and M. B. Yarmolinsky. 1986. Host participation in plasmid maintenance: dependence upon DnaA of replicons derived from P1 and F.Proc. Natl. Acad. Sci. USA 83:4423 –4427. 114. Hao, J. J., and M. Yarmolinsky. 2002. Effects of the P1 plasmid centromere on expression of P1 partition genes. J. Bacteriol. 184:4857 –4867. 115. Hartmann, R. K., J. Heinrich, J. Schlegl, and H. Schuster. 1995. Precursor of C4 antisense RNA of bacteriophages P1 and P7 is a substrate for RNasePo f Escherichia coli . Proc. Natl. Acad. Sci. USA 92:5822 –5826. 116. Hasegawa, T., M. Miyano, H. Himeno, Y. Sano, K. Kimura, and M. Shimizu. 1992. Identity determinants of E. coli threonine tRNA. Biochem. Biophys. Res. Commun. 184:478–484. 117. Hashemolhosseini, S., Y. D. Stierhof, I. Hindennach, and U. Henning. 1996. Characterization of the helper proteins for the assembly of tail ﬁbers of coliphages T4 and lambda. J. Bacteriol. 178:6258 –6265. 118. Hawley, D. K., and W. R. McClure. 1983. Compilation and analysis of Escherichia coli promoter DNA sequences. Nucleic Acids Res. 11:2237 – 2255. 119. Hay, N., and G. Cohen. 1983. Requirement of E. coli DNA synthesis functions for the lytic replication of bacteriophage P1. Virology 131:193– 206. 120. Hayes, F., L. Radnedge, M. A. Davis, and S. J. Austin. 1994. The homol- ogous operons for P1 and P7 plasmid partition are autoregulated fromdissimilar operator sites. Mol. Microbiol. 11:249–260. 121.', 'page_label: 33 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  178:6258 –6265. 118. Hawley, D. K., and W. R. McClure. 1983. Compilation and analysis of Escherichia coli promoter DNA sequences. Nucleic Acids Res. 11:2237 – 2255. 119. Hay, N., and G. Cohen. 1983. Requirement of E. coli DNA synthesis functions for the lytic replication of bacteriophage P1. Virology 131:193– 206. 120. Hayes, F., L. Radnedge, M. A. Davis, and S. J. Austin. 1994. The homol- ogous operons for P1 and P7 plasmid partition are autoregulated fromdissimilar operator sites. Mol. Microbiol. 11:249–260. 121. Hazan, R., B. Sat, M. Reches, and H. Engelberg-Kulka. 2001. Postsegre- gational killing mediated by the P1 phage “addiction module ”phd-doc requires the Escherichia coli programmed cell death system mazEF . J. Bac- teriol. 183:2046 –2050. 122. Heilmann, H., J. N. Reeve, and A. Pu ¨hler. 1980. Identi ﬁcation of the repressor and repressor bypass (antirepressor) polypeptides of bacterio-phage P1 synthesized in infected minicells. Mol. Gen. Genet. 178:149–154. 123. Heinrich, J., M. Citron, A.Gu ¨nther, and H. Schuster. 1994. Second-site suppressors of the bacteriophage P1 vir smutant reveal the interdependence of the c4,icd, and antgenes in the P1 imm I operon. J. Bacteriol. 176:4931 – 4936. 124. Heinrich, J., H. D. Riedel, B. R. Baumstark, M. Kimura, and H. Schuster. 1989. The c1 repressor of bacteriophage P1 operator-repressor interactionof wild-type and mutant repressor proteins. Nucleic Acids Res. 17:7681 – 7692. 125. Heinrich, J., H. D. Riedel, B. Ru ¨ckert, R. Lurz, and H. Schuster. 1995. The lytic replicon of bacteriophage P1 is controlled by an antisense RNA.Nucleic Acids Res. 23:1468 –1474. 126. Heinrich, J., M. Velleman, and H. Schuster. 1995. The tripartite immunity system of phages P1 and P7. FEMS Microbiol. Rev. 17:121–126. 127. Heinzel, T., M. Velleman, and H. Schuster. 1990. The cl repressor inacti- vator protein coi of bacteriophage Pl. Cloning and expression of coi and itsinterference with cl repressor function. J. Biol. Chem. 265:17928 –17934. 128. Heinzel, T., R. Lurz, B. Dobrinsky, M. Velleman, and H. Schuster. 1994. C1 repressor-mediated looping is involved in C1 autoregulation of bacterio-phage P1. J. Biol. Chem. 269:31885 –31890. 129. Heinzel, T., M. Velleman, and H. Schuster. 1989. ban operon of bacterio- phage P1. Mutational analysis of the c1 repressor-controlled operator.J. Mol. Biol. 205:127–135. 130. Heinzel, T., M. Velleman, and H. Schuster. 1992. Cl repressor of phage Pl is inactivated by non-covalent binding of Pl Coi protein. J. Biol. Chem. 267: 4183 –4188. 131. Heisig, A., H.-D.', 'page_label: 33 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1994. C1 repressor-mediated looping is involved in C1 autoregulation of bacterio-phage P1. J. Biol. Chem. 269:31885 –31890. 129. Heinzel, T., M. Velleman, and H. Schuster. 1989. ban operon of bacterio- phage P1. Mutational analysis of the c1 repressor-controlled operator.J. Mol. Biol. 205:127–135. 130. Heinzel, T., M. Velleman, and H. Schuster. 1992. Cl repressor of phage Pl is inactivated by non-covalent binding of Pl Coi protein. J. Biol. Chem. 267: 4183 –4188. 131. Heisig, A., H.-D. Riedel, B. Dobrinski, R. Lurz, and H. Schuster. 1989. Organization of the immunity region immI of bacteriophage Pl and syn-thesis of the Pl antirepressor. J. Mol. Biol. 209:525–538. 132. Heisig, A., I. Severin, A. K. Seeﬂuth, and H. Schuster. 1987. Regulation of theban gene containing operon of prophage P1. Mol. Gen. Genet. 206: 368–376. 133. Hendrix, R. W., and R. L. Duda. 1992. Bacteriophage lambda PaPa: not the mother of all lambda phages. Science 258:1145 –1148. 134. Henning, U., and S. Hashemolhosseini. 1994. Receptor recognition by T-even-type coliphages, p. 182 –185. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 135. Hiestand-Nauer, R., and S. Iida. 1983. Sequence of the site-speci ﬁc recom- binase gene cinand of its substrates serving in the inversion of the C seg- ment of bacteriophage P1. EMBO J. 2:1733 –1740. 136. Hill, T. M. 1996. Features of the chromosomal terminus region., p. 1602 – 1614. InF. C. Neidhardt et al. (ed.), Escherichia coli and Salmonella : cellular and molecular biology, 2nd ed., vol. 2. ASM Press, Washington,D.C. 137. Hochman, L., N. Segev, N. Sternberg, and G. Cohen. 1983. Site-speci ﬁc recombinational circularization of bacteriophage P1 DNA. Virology 131: 11–17. 138. Hoess, R. H., M. Ziese, and N. Sternberg. 1982. P1 site-speci ﬁc recombi- nation: nucleotide sequence of the recombining sites. Proc. Natl. Acad. Sci.USA 79:3398 –3402. 139. Hofer, B., M. Ruge, and B. Dreiseikelmann. 1995. The superinfection ex-7064 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 34 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  clusion gene ( sieA) of bacteriophage P22: identi ﬁcation and overexpression of the gene and localization of the gene product. J. Bacteriol. 177:3080 – 3086. 140. Holcik, M., and V. N. Iyer. 1997. Conditionally lethal genes associated with bacterial plasmids. Microbiology 143:3403 –3416. 141. Huber, H. E., S. Iida, W. Arber, and T. A. Bickle. 1985. Site speci ﬁc DNA inversion is enhanced by a speci ﬁc DNA element in cis. Proc. Natl. Acad. Sci. USA. 82:3776 –3780. 142. Huber, H. E., S. Iida, and T. A. Bickle. 1985. Expression of the bacterio- phage P1 cinrecombinase gene from its own and heterologous promoters. Gene 34:63–72. 143. Hu¨bner, P., and W. Arber. 1989. Mutational analysis of a prokaryotic recombinational enhancer element with two functions. EMBO J. 8:577–585. 144. Hu¨bner, P., P. Haffter, S. Iida, and W. Arber. 1989. Bent DNA is needed for recombinational enhancer activity in the site-speci ﬁc recombination system Cin of bacteriophage P1. The role of FIS protein. J. Mol. Biol. 205:493–500. 145. Hu¨mbelin, M., B. Suri, D. N. Rao, D. P. Hornby, H. Eberle, T. Prip ﬂ,S . Kenel, and T. A. Bickle. 1988. Type III DNA restriction and modi ﬁcation systems EcoP1 and EcoP15. Nucleotide sequence of the EcoP1 operon, the EcoP15 mod gene and some EcoP1mod mutants. J. Mol. Biol. 200:23–29. 146. Iida, S. 1984. Bacteriophage P1 carries two related sets of genes determin- ing its host range in the invertible C segment of its genome. Virology 134: 421–434. 147. Iida, S., and W. Arber. 1977. Plaque forming specialized transducing phage P1: isolation of P1CmSmSu, a precursor of P1Cm. Mol. Gen. Genet. 153: 259–269. 148. Iida, S., and R. Hiestand-Nauer. 1987. Role of the central dinucleotide at the crossover sites for the selection of quasi sites in DNA inversion medi-ated by the site-speci ﬁc Cin recombinase of phage P1. Mol. Gen. Genet. 208:464–468. 149. Iida, S., R. Hiestand-Nauer, H. Sandmeier, H. Lehnherr, and W. Arber. 1998. Accessory genes in the darA operon of bacteriophage P1 affect anti- restriction function, generalized transduction, head morphogenesis, andhost cell lysis. Virology 251:49–58. 150. Iida, S., H. Huber, R. Hiestand-Nauer, J. Meyer, T. A. Bickle, and W. Arber. 1984. The bacteriophage P1 site-speci ﬁc recombinase cin: recombination events and DNA recognition sequences. Cold Spring Harbor Symp. Quant.Biol. 49:769–777. 151. Iida, S., J. Meyer, and W. Arber. 1978. The insertion element IS 1is a natural constituent of coliphage P1 DNA. Plasmid 1:357–365. 152. Iida, S., J. Meyer, B. Ba ¨chi, M. Sta ˚lhammar-Carlemalm, S. Schrickel, T. A. Bickle, and W. Arber. 1983.', 'page_label: 34 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  150. Iida, S., H. Huber, R. Hiestand-Nauer, J. Meyer, T. A. Bickle, and W. Arber. 1984. The bacteriophage P1 site-speci ﬁc recombinase cin: recombination events and DNA recognition sequences. Cold Spring Harbor Symp. Quant.Biol. 49:769–777. 151. Iida, S., J. Meyer, and W. Arber. 1978. The insertion element IS 1is a natural constituent of coliphage P1 DNA. Plasmid 1:357–365. 152. Iida, S., J. Meyer, B. Ba ¨chi, M. Sta ˚lhammar-Carlemalm, S. Schrickel, T. A. Bickle, and W. Arber. 1983. DNA restriction-modi ﬁcation genes of phage P1 and plasmid p15B. Structure and in vitro transcription. J. Mol. Biol. 165: 1–18. 153. Iida, S., J. Meyer, K. E. Kennedy, and W. Arber. 1982. A site-speci ﬁc, conservative recombination system carried by bacteriophage P1. Mappingthe recombinase gene cinand the cross-over sites cixfor the inversion of the C segment. EMBO J. 1:1445 –1453. 154. Iida, S., H. Sandmeier, P. Hu ¨bner, R. Hiestand-Nauer, K. Schneitz, and W. Arber. 1990. The Min DNA inversion enzyme of plasmid p15B of Esche- richia coli 15T-: a new member of the Din family of site-speci ﬁc recombi- nases. Mol. Microbiol. 4:991–997. 155. Iida, S., M. B. Streiff, T. A. Bickle, and W. Arber. 1987. Two DNA anti- restriction systems of bacteriophage P1, darA anddarB : characterization of darA/H11002phages. Virology 157:156–166. 156. Ikeda, H., and J.-I. Tomizawa. 1965. Transducing fragments in generalized transduction by phage P1. III. Studies with small phage particles. J. Mol.Biol. 14:120–129. 157. Ireton, K., N. W. Gu ¨nther, and A. D. Grossman. 1994. Spo0J is required for normal chromosome segregation as well as the initiation of sporulation inBacillus subtilis . J. Bacteriol. 176:5320 –5329. 158. Ishihama, A., and T. Saitoh. 1979. Subunits of RNA polymerase in function and structure. IX. Regulation of RNA polymerase activity by stringentstarvation protein (SSP). J. Mol. Biol. 129:517–530. 159. Ito, H., A. Sadaoka, H. Kotani, N. Hiraoka, and T. Nakamura. 1990. Cloning, nucleotide sequence, and expression of the HincII restriction-modi ﬁcation system. Nucleic Acids Res. 18:3903 –3911. 160. Ito, J., and D. K. Braithwaite. 1991. Compilation and alignment of DNA polymerase sequences. Nucleic Acids Res. 19:4045 –4057. 161. Janscak, P., U. Sandmeier, M. D. Szczelkun, and T. A. Bickle. 2001. Subunit assembly and mode of DNA cleavage of the type III restriction endonucle-ases EcoP1I and EcoP15I. J. Mol. Biol. 306:417–431. 162. Janulaitis, A., M. Petrusyte, Z. Maneliene, S. Klimasauskas, and V. But- kus. 1992.', 'page_label: 34 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Nucleic Acids Res. 18:3903 –3911. 160. Ito, J., and D. K. Braithwaite. 1991. Compilation and alignment of DNA polymerase sequences. Nucleic Acids Res. 19:4045 –4057. 161. Janscak, P., U. Sandmeier, M. D. Szczelkun, and T. A. Bickle. 2001. Subunit assembly and mode of DNA cleavage of the type III restriction endonucle-ases EcoP1I and EcoP15I. J. Mol. Biol. 306:417–431. 162. Janulaitis, A., M. Petrusyte, Z. Maneliene, S. Klimasauskas, and V. But- kus. 1992. Puri ﬁcation and properties of the Eco57I restriction endonucle- ase and methylase-prototypes of a new class (type IV). Nucleic Acids Res.20:6043 –6049. 163. Jobling, M. G., and D. A. Ritchie. 1988. The nucleotide sequence of a plasmid determinant for resistance to tellurium anions. Gene 66:245–258. 164. Johnson, B. F. 1982. Suppression of the lexC (ssbA ) mutation of Escherichia coliby a mutant of bacteriophage P1. Mol. Gen. Genet. 186:122–126. 165. Kanamaru, S., P. G. Leiman, V. A. Kostyuchenko, P. R. Chipman, V. V.Mesyanzhinov, F. Arisaka, and M. G. Rossmann. 2002. Structure of the cell-puncturing device of bacteriophage T4. Nature 415:553–557. 166. Kane, J. F. 1995. Effects of rare codon clusters on high-level expression of heterologous proteins in Escherichia coli . Curr. Opin. Biotechnol. 6:494– 500. 167. Kawakami, B., H. Christophe, M. Nagatomo, and M. Oka. 1991. Cloning and nucleotide sequences of the AccI restriction-modi ﬁcation genes in Acinetobacter calcoaceticus . Agric. Biol. Chem. 55:1553 –1559. 168. Kelman, Z., and M. O ’Donnell. 1995. DNA polymerase III holoenzyme: structure and function of a chromosomal replicating machine. Annu. Rev.Biochem. 64:171–200. 169. Keniry, M. A., H. A. Berthon, J. Y. Yang, C. S. Miles, and N. E. Dixon. 2000. NMR solution structure of the theta subunit of DNA polymerase III fromEscherichia coli . Protein Sci. 9:721–733. 170. Kliem, M., and B. Dreiseikelmann. 1989. The superimmunity gene simof bacteriophage P1 causes superinfection exclusion. Virology 171:350–355. 171. Kondo, E., and S. Mitsuhashi. 1964. Drug resistance of enteric bacteria. IV. Active transducing bacteriophage P1CM produced by the combination ofR-factor with bacteriophage P1. J. Bacteriol. 88:1266 –1276. 172. Koonin, E. V. 1993. Bacterial and bacteriophage protein phosphatases. Mol. Microbiol. 8:785–786. 173. Kowalczykowski, S. C. 2000. Initiation of genetic recombination and re- combination-dependent replication. Trends Biochem. Sci. 25:156–165. 174. Kravanja, M., R. Engelmann, V. Dossonnet, M. Blu ¨ggel, H. E. Meyer, R. Frank, A. Galinier, J. Deutscher, N. Schnell, and W. Hengstenberg. 1999.', 'page_label: 34 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  IV. Active transducing bacteriophage P1CM produced by the combination ofR-factor with bacteriophage P1. J. Bacteriol. 88:1266 –1276. 172. Koonin, E. V. 1993. Bacterial and bacteriophage protein phosphatases. Mol. Microbiol. 8:785–786. 173. Kowalczykowski, S. C. 2000. Initiation of genetic recombination and re- combination-dependent replication. Trends Biochem. Sci. 25:156–165. 174. Kravanja, M., R. Engelmann, V. Dossonnet, M. Blu ¨ggel, H. E. Meyer, R. Frank, A. Galinier, J. Deutscher, N. Schnell, and W. Hengstenberg. 1999. The hprK gene of Enterococcus faecalis encodes a novel bifunctional en- zyme: the HPr kinase/phosphatase. Mol. Microbiol. 31:59–66. 175. Krogh, A., B. Larsson, G. von Heijne, and E. L. Sonnhammer. 2001. Pre- dicting transmembrane protein topology with a hidden Markov model:application to complete genomes. J. Mol. Biol. 305:567–580. 176. Kropinski, A. M. 2000. Sequence of the genome of the temperate, serotype- converting, Pseudomonas aeruginosa bacteriophage D3. J. Bacteriol. 182: 6066 –6074. 177. Kropinski, A. M., and M. J. Sibbald. 1999. Transfer RNA genes and their signi ﬁcance to codon usage in the Pseudomonas aeruginosa lamboid bacte- riophage D3. Can J. Microbiol. 45:791–796. 178. Kumar, A., R. A. Malloch, N. Fujita, D. A. Smillie, A. Ishihama, and R. S. Hayward. 1993. The minus 35-recognition region of Escherichia coli sigma 70 is inessential for initiation of transcription at an “extended minus 10 ” promoter. J. Mol. Biol. 232:406–418. 179. Kunisawa, T. 1992. Synonymous codon preferences in bacteriophage T4: a distinctive use of transfer RNAs from T4 and from its host Escherichia coli . J. Theor. Biol. 159:287–298. 180. Kutter, E., K. Gachechiladze, A. Poglazov, E. Marusich, M. Shneider, P. Aronsson, A. Napuli, D. Porter, and V. Mesyanzhinov. 1995. Evolution of T4-related phages. Virus Genes 11:285–297. 181. Kuzminov, A. 1995. Collapse and repair of replication forks in Escherichia coli. Mol. Microbiol. 16:373–384. 182. Kwon, H. J., R. Tirumalai, A. Landy, and T. Ellenberger. 1997. Flexibility in DNA recombination: structure of the lambda integrase catalytic core.Science 276:126–131. 183. Lanka, E., B. Geschke, and H. Schuster. 1978. Escherichia coli dnaB mutant defective in DNA initiation: isolation and properties of the dnaB protein.Proc. Natl. Acad. Sci. USA 75:799–803. 184. Laufer, C. S., J. B. Hays, B. E. Windle, T. S. Schaefer, E. H. Lee, S. L. Hays, and M. R. McClure. 1989. Enhancement of Escherichia coli plasmid and chromosomal recombination by the Ref function of bacteriophage P1. Ge-netics 123:465–476. 184a. Lederberg, E. M. 1950.', 'page_label: 34 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  183. Lanka, E., B. Geschke, and H. Schuster. 1978. Escherichia coli dnaB mutant defective in DNA initiation: isolation and properties of the dnaB protein.Proc. Natl. Acad. Sci. USA 75:799–803. 184. Laufer, C. S., J. B. Hays, B. E. Windle, T. S. Schaefer, E. H. Lee, S. L. Hays, and M. R. McClure. 1989. Enhancement of Escherichia coli plasmid and chromosomal recombination by the Ref function of bacteriophage P1. Ge-netics 123:465–476. 184a. Lederberg, E. M. 1950. Lysogenicity of Escherichia coli strain K-12. Microb. Genet. Bull. 1:5–7. 185. Lederberg, E. M., and J. Lederberg. 1953. Genetic studies of lysogenicity in E. coli. Genetics 38:51–64. 186. Lehnherr, H. Bacteriophage P1. InR. Calendar (ed.), The bacteriophages, in press. Oxford University Press, New York, N.Y. 187. Lehnherr, H., J. D. Bendtsen, F. Preuss, and T. V. Ilyina. 1999. Identi ﬁca- tion and characterization of the single-stranded DNA-binding protein ofbacteriophage P1. J. Bacteriol. 181:6463 –6468. 188. Lehnherr, H., A. Guidolin, and W. Arber. 1991. Bacteriophage P1 gene 10 encodes a trans -activating factor required for late gene expression. J. Bac- teriol. 173:6438 –6445. 189. Lehnherr, H., A. Guidolin, and W. Arber. 1992. Mutational analysis of the bacteriophage P1 late promoter sequence P S. J. Mol. Biol. 228:101–107. 190. Lehnherr, H., A.-M. Hansen, and T. V. Ilyina. 1998. Penetration of the bacterial cell wall: a family of lytic transglycosylases in bacteriophages andconjugative plasmids. Mol. Microbiol. 30:454–457. 191. Lehnherr, H., C. D. Jensen, A. R. Stenholm, and A. Dueholm. 2001. Dual regulatory control of a particle maturation function of bacteriophage P1.J. Bacteriol. 183:4105 –4109. 192. Lehnherr, H., E. Maguin, S. Jafri, and M. Yarmolinsky. 1993. Plasmid addiction genes of bacteriophage P1: doc, which causes cell death on curing of prophage, and phd, which prevents host death when prophage is re- tained. J. Mol. Biol. 233:414–428.VOL. 186, 2004 BACTERIOPHAGE P1 7065', 'page_label: 35 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  193. Lehnherr, H., M. Velleman, A. Guidolin, and W. Arber. 1992. Bacterio- phage P1 gene 10is expressed from a promoter-operator sequence con- trolled by C1 and Bof proteins. J. Bacteriol. 174:6138 –6144. 194. Lehnherr, H., and M. B. Yarmolinsky. 1995. Addiction protein Phd of plasmid prophage P1 is a substrate of the ClpXP serine protease of Esch- erichia coli . Proc. Natl. Acad. Sci. USA 92:3274 –3277. 195. Lemonnier, M., G. Ziegelin, T. Reick, A. M. Gomez, R. Diaz-Orejas, and E. Lanka. 2003. Bacteriophage P1 Ban protein is a hexameric DNA helicase that interacts with and substitutes for Escherichia coli DnaB. Nucleic Acids Res. 31:3918 –3928. 196. Lennox, E. S. 1955. Transduction of linked genetic characters of the host by bacteriophage P1. Virology 1:190–206. 197. Leslie, N. R., and D. J. Sherratt. 1995. Site-speci ﬁc recombination in the replication terminus region of Escherichia coli : functional replacement of dif. EMBO J. 14:1561 –1570. 198. Lesnik, E. A., R. Sampath, H. B. Levene, T. J. Henderson, J. A. McNeil, and D. J. Ecker. 2001. Prediction of rho-independent transcriptional termina- tors in Escherichia coli . Nucleic Acids Res. 29:3583 –3594. 199. Levin, M. E., R. W. Hendrix, and S. R. Casjens. 1993. A programmed translational frameshift is required for the synthesis of a bacteriophagelambda tail assembly protein. J. Mol. Biol. 234:124–139. 200. Lewis, L. K., G. R. Harlow, L. A. Gregg-Jolly, and D. W. Mount. 1994. Identi ﬁcation of high af ﬁnity binding sites for LexA which de ﬁne new DNA damage-inducible genes in Escherichia coli . J. Mol. Biol. 241:507–523. 201. Li, S., H. Pelka, and L. H. Schulman. 1993. The anticodon and discrimi- nator base are important for aminoacylation of Escherichia coli tRNA (Asn). J. Biol. Chem. 268:18335 –18339. 202. Lin, H., V. B. Rao, and L. W. Black. 1999. Analysis of capsid portal protein and terminase functional domains: interaction sites required for DNApackaging in bacteriophage T4. J. Mol. Biol. 289:249–260. 203. Łobocka, M., and M. Yarmolinsky. 1996. P1 plasmid partition: a mutational analysis of ParB. J. Mol. Biol. 259:366–382. 204. Lowe, T. M., and S. R. Eddy. 1997. tRNAscan-SE: a program for improved detection of transfer RNA genes in genomic sequence. Nucleic Acids Res.25:955–964. 205. Lu, M. J., and U. Henning. 1994. Superinfection exclusion by T-even-type coliphages. Trends Microbiol. 2:137–139. 206. Lu, M. J., Y. D. Stierhof, and U. Henning. 1993. Location and unusual membrane topology of the immunity protein of the Escherichia coli phage T4. J. Virol. 67:4905 –4913. 207.', 'page_label: 35 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  J. Mol. Biol. 259:366–382. 204. Lowe, T. M., and S. R. Eddy. 1997. tRNAscan-SE: a program for improved detection of transfer RNA genes in genomic sequence. Nucleic Acids Res.25:955–964. 205. Lu, M. J., and U. Henning. 1994. Superinfection exclusion by T-even-type coliphages. Trends Microbiol. 2:137–139. 206. Lu, M. J., Y. D. Stierhof, and U. Henning. 1993. Location and unusual membrane topology of the immunity protein of the Escherichia coli phage T4. J. Virol. 67:4905 –4913. 207. Lu, S. D., D. Lu, and M. Gottesman. 1989. Stimulation of IS 1excision by bacteriophage P1 reffunction. J. Bacteriol. 171:3427 –3432. 208. Lukashin, A. V., and M. Borodovsky. 1998. GeneMark.hmm: new solutions for gene ﬁnding. Nucleic Acids Res. 26:1107 –1115. 209. Lurz, R., A. Heisig, M. Velleman, B. Dobrinski, and H. Schuster. 1987. The banoperon of bacteriophage P1. Localization of the promoter controlled by P1 repressor. J. Biol. Chem. 262:16575 –16579. 210. Magnuson, R., H. Lehnherr, G. Mukhopadhyay, and M. B. Yarmolinsky. 1996. Autoregulation of the plasmid addiction operon of bacteriophage P1.J. Biol. Chem. 271:18705 –18710. 211. Magnuson, R., and M. B. Yarmolinsky. 1998. Corepression of the P1 addiction operon by Phd and Doc. J. Bacteriol. 180:6342 –6351. 212. Mahillon, J., and M. Chandler. 1998. Insertion sequences. Microbiol. Mol. Biol. Rev. 62:725–774. 213. Maillou, J., and B. Dreiseikelmann. 1990. The simgene of Escherichia coli phage P1: nucleotide sequence and puri ﬁcation of the processed protein. Virology 175:500–507. 214. Malone, T., R. M. Blumenthal, and X. Cheng. 1995. Structure-guided anal- ysis reveals nine sequence motifs conserved among DNA amino-methyl-transferases, and suggests a catalytic mechanism for these enzymes. J. Mol.Biol. 253:618–632. 215. McDonald, J. P., T. S. Peat, A. S. Levine, and R. Woodgate. 1999. Inter- molecular cleavage by UmuD-like enzymes: identi ﬁcation of residues re- quired for cleavage and substrate speci ﬁcity. J. Mol. Biol. 285:2199 –2209. 216. McLean, M. J., K. H. Wolfe, and K. M. Devine. 1998. Base composition skews, replication orientation, and gene orientation in 12 prokaryote ge-nomes. J. Mol. Evol. 47:691–696. 217. McLenigan, M. P., O. I. Kulaeva, D. G. Ennis, A. S. Levine, and R. Wood- gate. 1999. The bacteriophage P1 HumD protein is a functional homolog of the prokaryotic UmuD /H11032-like proteins and facilitates SOS mutagenesis in Escherichia coli . J. Bacteriol. 181:7005 –7013. 218. Mehr, I. J., C. D. Long, C. D. Serkin, and H. S. Seifert. 2000.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 33 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1983. DNA restriction- modi ﬁcation enzymes of phage P1 and plasmid p15B. Subunit functions and structural homologies. J. Mol. Biol. 165:19–34. 108. Haffter, P., and T. A. Bickle. 1988. Enhancer-independent mutants of the Cin recombinase have a relaxed topological speci ﬁcity. EMBO J. 7:3991 – 3996. 109. Haffter, P., and T. A. Bickle. 1987. Puri ﬁcation and DNA-binding properties of FIS and Cin, two proteins required for the bacteriophage P1 site-speci ﬁc recombination system, cin. J. Mol. Biol. 198:579–587. 110. Hambly, E., F. Tetart, C. Desplats, W. H. Wilson, H. M. Krisch, and N. H. Mann. 2001. A conserved genetic module that encodes the major virion components in both the coliphage T4 and the marine cyanophage S-PM2.Proc. Natl. Acad. Sci. USA 98:11411 –11416. 111. Hansen, A.-M., H. Lehnherr, X. Wang, V. Mobley, and D. J. Jin. 2003. Escherichia coli SspA is a transcription activator for bacteriophage P1 late genes. Mol. Microbiol. 48:1621 –1631.112. Hansen, E. B. 1989. Structure and regulation of the lytic replicon of phage Pl. J. Mol. Biol. 207:135–149. 113. Hansen, E. B., and M. B. Yarmolinsky. 1986. Host participation in plasmid maintenance: dependence upon DnaA of replicons derived from P1 and F.Proc. Natl. Acad. Sci. USA 83:4423 –4427. 114. Hao, J. J., and M. Yarmolinsky. 2002. Effects of the P1 plasmid centromere on expression of P1 partition genes. J. Bacteriol. 184:4857 –4867. 115. Hartmann, R. K., J. Heinrich, J. Schlegl, and H. Schuster. 1995. Precursor of C4 antisense RNA of bacteriophages P1 and P7 is a substrate for RNasePo f Escherichia coli . Proc. Natl. Acad. Sci. USA 92:5822 –5826. 116. Hasegawa, T., M. Miyano, H. Himeno, Y. Sano, K. Kimura, and M. Shimizu. 1992. Identity determinants of E. coli threonine tRNA. Biochem. Biophys. Res. Commun. 184:478–484. 117. Hashemolhosseini, S., Y. D. Stierhof, I. Hindennach, and U. Henning. 1996. Characterization of the helper proteins for the assembly of tail ﬁbers of coliphages T4 and lambda. J. Bacteriol. 178:6258 –6265. 118. Hawley, D. K., and W. R. McClure. 1983. Compilation and analysis of Escherichia coli promoter DNA sequences. Nucleic Acids Res. 11:2237 – 2255. 119. Hay, N., and G. Cohen. 1983. Requirement of E. coli DNA synthesis functions for the lytic replication of bacteriophage P1. Virology 131:193– 206. 120. Hayes, F., L. Radnedge, M. A. Davis, and S. J. Austin. 1994. The homol- ogous operons for P1 and P7 plasmid partition are autoregulated fromdissimilar operator sites. Mol. Microbiol. 11:249–260. 121.', 'page_label: 33 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  178:6258 –6265. 118. Hawley, D. K., and W. R. McClure. 1983. Compilation and analysis of Escherichia coli promoter DNA sequences. Nucleic Acids Res. 11:2237 – 2255. 119. Hay, N., and G. Cohen. 1983. Requirement of E. coli DNA synthesis functions for the lytic replication of bacteriophage P1. Virology 131:193– 206. 120. Hayes, F., L. Radnedge, M. A. Davis, and S. J. Austin. 1994. The homol- ogous operons for P1 and P7 plasmid partition are autoregulated fromdissimilar operator sites. Mol. Microbiol. 11:249–260. 121. Hazan, R., B. Sat, M. Reches, and H. Engelberg-Kulka. 2001. Postsegre- gational killing mediated by the P1 phage “addiction module ”phd-doc requires the Escherichia coli programmed cell death system mazEF . J. Bac- teriol. 183:2046 –2050. 122. Heilmann, H., J. N. Reeve, and A. Pu ¨hler. 1980. Identi ﬁcation of the repressor and repressor bypass (antirepressor) polypeptides of bacterio-phage P1 synthesized in infected minicells. Mol. Gen. Genet. 178:149–154. 123. Heinrich, J., M. Citron, A.Gu ¨nther, and H. Schuster. 1994. Second-site suppressors of the bacteriophage P1 vir smutant reveal the interdependence of the c4,icd, and antgenes in the P1 imm I operon. J. Bacteriol. 176:4931 – 4936. 124. Heinrich, J., H. D. Riedel, B. R. Baumstark, M. Kimura, and H. Schuster. 1989. The c1 repressor of bacteriophage P1 operator-repressor interactionof wild-type and mutant repressor proteins. Nucleic Acids Res. 17:7681 – 7692. 125. Heinrich, J., H. D. Riedel, B. Ru ¨ckert, R. Lurz, and H. Schuster. 1995. The lytic replicon of bacteriophage P1 is controlled by an antisense RNA.Nucleic Acids Res. 23:1468 –1474. 126. Heinrich, J., M. Velleman, and H. Schuster. 1995. The tripartite immunity system of phages P1 and P7. FEMS Microbiol. Rev. 17:121–126. 127. Heinzel, T., M. Velleman, and H. Schuster. 1990. The cl repressor inacti- vator protein coi of bacteriophage Pl. Cloning and expression of coi and itsinterference with cl repressor function. J. Biol. Chem. 265:17928 –17934. 128. Heinzel, T., R. Lurz, B. Dobrinsky, M. Velleman, and H. Schuster. 1994. C1 repressor-mediated looping is involved in C1 autoregulation of bacterio-phage P1. J. Biol. Chem. 269:31885 –31890. 129. Heinzel, T., M. Velleman, and H. Schuster. 1989. ban operon of bacterio- phage P1. Mutational analysis of the c1 repressor-controlled operator.J. Mol. Biol. 205:127–135. 130. Heinzel, T., M. Velleman, and H. Schuster. 1992. Cl repressor of phage Pl is inactivated by non-covalent binding of Pl Coi protein. J. Biol. Chem. 267: 4183 –4188. 131. Heisig, A., H.-D.', 'page_label: 33 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1994. C1 repressor-mediated looping is involved in C1 autoregulation of bacterio-phage P1. J. Biol. Chem. 269:31885 –31890. 129. Heinzel, T., M. Velleman, and H. Schuster. 1989. ban operon of bacterio- phage P1. Mutational analysis of the c1 repressor-controlled operator.J. Mol. Biol. 205:127–135. 130. Heinzel, T., M. Velleman, and H. Schuster. 1992. Cl repressor of phage Pl is inactivated by non-covalent binding of Pl Coi protein. J. Biol. Chem. 267: 4183 –4188. 131. Heisig, A., H.-D. Riedel, B. Dobrinski, R. Lurz, and H. Schuster. 1989. Organization of the immunity region immI of bacteriophage Pl and syn-thesis of the Pl antirepressor. J. Mol. Biol. 209:525–538. 132. Heisig, A., I. Severin, A. K. Seeﬂuth, and H. Schuster. 1987. Regulation of theban gene containing operon of prophage P1. Mol. Gen. Genet. 206: 368–376. 133. Hendrix, R. W., and R. L. Duda. 1992. Bacteriophage lambda PaPa: not the mother of all lambda phages. Science 258:1145 –1148. 134. Henning, U., and S. Hashemolhosseini. 1994. Receptor recognition by T-even-type coliphages, p. 182 –185. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 135. Hiestand-Nauer, R., and S. Iida. 1983. Sequence of the site-speci ﬁc recom- binase gene cinand of its substrates serving in the inversion of the C seg- ment of bacteriophage P1. EMBO J. 2:1733 –1740. 136. Hill, T. M. 1996. Features of the chromosomal terminus region., p. 1602 – 1614. InF. C. Neidhardt et al. (ed.), Escherichia coli and Salmonella : cellular and molecular biology, 2nd ed., vol. 2. ASM Press, Washington,D.C. 137. Hochman, L., N. Segev, N. Sternberg, and G. Cohen. 1983. Site-speci ﬁc recombinational circularization of bacteriophage P1 DNA. Virology 131: 11–17. 138. Hoess, R. H., M. Ziese, and N. Sternberg. 1982. P1 site-speci ﬁc recombi- nation: nucleotide sequence of the recombining sites. Proc. Natl. Acad. Sci.USA 79:3398 –3402. 139. Hofer, B., M. Ruge, and B. Dreiseikelmann. 1995. The superinfection ex-7064 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 34 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  clusion gene ( sieA) of bacteriophage P22: identi ﬁcation and overexpression of the gene and localization of the gene product. J. Bacteriol. 177:3080 – 3086. 140. Holcik, M., and V. N. Iyer. 1997. Conditionally lethal genes associated with bacterial plasmids. Microbiology 143:3403 –3416. 141. Huber, H. E., S. Iida, W. Arber, and T. A. Bickle. 1985. Site speci ﬁc DNA inversion is enhanced by a speci ﬁc DNA element in cis. Proc. Natl. Acad. Sci. USA. 82:3776 –3780. 142. Huber, H. E., S. Iida, and T. A. Bickle. 1985. Expression of the bacterio- phage P1 cinrecombinase gene from its own and heterologous promoters. Gene 34:63–72. 143. Hu¨bner, P., and W. Arber. 1989. Mutational analysis of a prokaryotic recombinational enhancer element with two functions. EMBO J. 8:577–585. 144. Hu¨bner, P., P. Haffter, S. Iida, and W. Arber. 1989. Bent DNA is needed for recombinational enhancer activity in the site-speci ﬁc recombination system Cin of bacteriophage P1. The role of FIS protein. J. Mol. Biol. 205:493–500. 145. Hu¨mbelin, M., B. Suri, D. N. Rao, D. P. Hornby, H. Eberle, T. Prip ﬂ,S . Kenel, and T. A. Bickle. 1988. Type III DNA restriction and modi ﬁcation systems EcoP1 and EcoP15. Nucleotide sequence of the EcoP1 operon, the EcoP15 mod gene and some EcoP1mod mutants. J. Mol. Biol. 200:23–29. 146. Iida, S. 1984. Bacteriophage P1 carries two related sets of genes determin- ing its host range in the invertible C segment of its genome. Virology 134: 421–434. 147. Iida, S., and W. Arber. 1977. Plaque forming specialized transducing phage P1: isolation of P1CmSmSu, a precursor of P1Cm. Mol. Gen. Genet. 153: 259–269. 148. Iida, S., and R. Hiestand-Nauer. 1987. Role of the central dinucleotide at the crossover sites for the selection of quasi sites in DNA inversion medi-ated by the site-speci ﬁc Cin recombinase of phage P1. Mol. Gen. Genet. 208:464–468. 149. Iida, S., R. Hiestand-Nauer, H. Sandmeier, H. Lehnherr, and W. Arber. 1998. Accessory genes in the darA operon of bacteriophage P1 affect anti- restriction function, generalized transduction, head morphogenesis, andhost cell lysis. Virology 251:49–58. 150. Iida, S., H. Huber, R. Hiestand-Nauer, J. Meyer, T. A. Bickle, and W. Arber. 1984. The bacteriophage P1 site-speci ﬁc recombinase cin: recombination events and DNA recognition sequences. Cold Spring Harbor Symp. Quant.Biol. 49:769–777. 151. Iida, S., J. Meyer, and W. Arber. 1978. The insertion element IS 1is a natural constituent of coliphage P1 DNA. Plasmid 1:357–365. 152. Iida, S., J. Meyer, B. Ba ¨chi, M. Sta ˚lhammar-Carlemalm, S. Schrickel, T. A. Bickle, and W. Arber. 1983.', 'page_label: 34 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  150. Iida, S., H. Huber, R. Hiestand-Nauer, J. Meyer, T. A. Bickle, and W. Arber. 1984. The bacteriophage P1 site-speci ﬁc recombinase cin: recombination events and DNA recognition sequences. Cold Spring Harbor Symp. Quant.Biol. 49:769–777. 151. Iida, S., J. Meyer, and W. Arber. 1978. The insertion element IS 1is a natural constituent of coliphage P1 DNA. Plasmid 1:357–365. 152. Iida, S., J. Meyer, B. Ba ¨chi, M. Sta ˚lhammar-Carlemalm, S. Schrickel, T. A. Bickle, and W. Arber. 1983. DNA restriction-modi ﬁcation genes of phage P1 and plasmid p15B. Structure and in vitro transcription. J. Mol. Biol. 165: 1–18. 153. Iida, S., J. Meyer, K. E. Kennedy, and W. Arber. 1982. A site-speci ﬁc, conservative recombination system carried by bacteriophage P1. Mappingthe recombinase gene cinand the cross-over sites cixfor the inversion of the C segment. EMBO J. 1:1445 –1453. 154. Iida, S., H. Sandmeier, P. Hu ¨bner, R. Hiestand-Nauer, K. Schneitz, and W. Arber. 1990. The Min DNA inversion enzyme of plasmid p15B of Esche- richia coli 15T-: a new member of the Din family of site-speci ﬁc recombi- nases. Mol. Microbiol. 4:991–997. 155. Iida, S., M. B. Streiff, T. A. Bickle, and W. Arber. 1987. Two DNA anti- restriction systems of bacteriophage P1, darA anddarB : characterization of darA/H11002phages. Virology 157:156–166. 156. Ikeda, H., and J.-I. Tomizawa. 1965. Transducing fragments in generalized transduction by phage P1. III. Studies with small phage particles. J. Mol.Biol. 14:120–129. 157. Ireton, K., N. W. Gu ¨nther, and A. D. Grossman. 1994. Spo0J is required for normal chromosome segregation as well as the initiation of sporulation inBacillus subtilis . J. Bacteriol. 176:5320 –5329. 158. Ishihama, A., and T. Saitoh. 1979. Subunits of RNA polymerase in function and structure. IX. Regulation of RNA polymerase activity by stringentstarvation protein (SSP). J. Mol. Biol. 129:517–530. 159. Ito, H., A. Sadaoka, H. Kotani, N. Hiraoka, and T. Nakamura. 1990. Cloning, nucleotide sequence, and expression of the HincII restriction-modi ﬁcation system. Nucleic Acids Res. 18:3903 –3911. 160. Ito, J., and D. K. Braithwaite. 1991. Compilation and alignment of DNA polymerase sequences. Nucleic Acids Res. 19:4045 –4057. 161. Janscak, P., U. Sandmeier, M. D. Szczelkun, and T. A. Bickle. 2001. Subunit assembly and mode of DNA cleavage of the type III restriction endonucle-ases EcoP1I and EcoP15I. J. Mol. Biol. 306:417–431. 162. Janulaitis, A., M. Petrusyte, Z. Maneliene, S. Klimasauskas, and V. But- kus. 1992.', 'page_label: 34 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Nucleic Acids Res. 18:3903 –3911. 160. Ito, J., and D. K. Braithwaite. 1991. Compilation and alignment of DNA polymerase sequences. Nucleic Acids Res. 19:4045 –4057. 161. Janscak, P., U. Sandmeier, M. D. Szczelkun, and T. A. Bickle. 2001. Subunit assembly and mode of DNA cleavage of the type III restriction endonucle-ases EcoP1I and EcoP15I. J. Mol. Biol. 306:417–431. 162. Janulaitis, A., M. Petrusyte, Z. Maneliene, S. Klimasauskas, and V. But- kus. 1992. Puri ﬁcation and properties of the Eco57I restriction endonucle- ase and methylase-prototypes of a new class (type IV). Nucleic Acids Res.20:6043 –6049. 163. Jobling, M. G., and D. A. Ritchie. 1988. The nucleotide sequence of a plasmid determinant for resistance to tellurium anions. Gene 66:245–258. 164. Johnson, B. F. 1982. Suppression of the lexC (ssbA ) mutation of Escherichia coliby a mutant of bacteriophage P1. Mol. Gen. Genet. 186:122–126. 165. Kanamaru, S., P. G. Leiman, V. A. Kostyuchenko, P. R. Chipman, V. V.Mesyanzhinov, F. Arisaka, and M. G. Rossmann. 2002. Structure of the cell-puncturing device of bacteriophage T4. Nature 415:553–557. 166. Kane, J. F. 1995. Effects of rare codon clusters on high-level expression of heterologous proteins in Escherichia coli . Curr. Opin. Biotechnol. 6:494– 500. 167. Kawakami, B., H. Christophe, M. Nagatomo, and M. Oka. 1991. Cloning and nucleotide sequences of the AccI restriction-modi ﬁcation genes in Acinetobacter calcoaceticus . Agric. Biol. Chem. 55:1553 –1559. 168. Kelman, Z., and M. O ’Donnell. 1995. DNA polymerase III holoenzyme: structure and function of a chromosomal replicating machine. Annu. Rev.Biochem. 64:171–200. 169. Keniry, M. A., H. A. Berthon, J. Y. Yang, C. S. Miles, and N. E. Dixon. 2000. NMR solution structure of the theta subunit of DNA polymerase III fromEscherichia coli . Protein Sci. 9:721–733. 170. Kliem, M., and B. Dreiseikelmann. 1989. The superimmunity gene simof bacteriophage P1 causes superinfection exclusion. Virology 171:350–355. 171. Kondo, E., and S. Mitsuhashi. 1964. Drug resistance of enteric bacteria. IV. Active transducing bacteriophage P1CM produced by the combination ofR-factor with bacteriophage P1. J. Bacteriol. 88:1266 –1276. 172. Koonin, E. V. 1993. Bacterial and bacteriophage protein phosphatases. Mol. Microbiol. 8:785–786. 173. Kowalczykowski, S. C. 2000. Initiation of genetic recombination and re- combination-dependent replication. Trends Biochem. Sci. 25:156–165. 174. Kravanja, M., R. Engelmann, V. Dossonnet, M. Blu ¨ggel, H. E. Meyer, R. Frank, A. Galinier, J. Deutscher, N. Schnell, and W. Hengstenberg. 1999.', 'page_label: 34 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  IV. Active transducing bacteriophage P1CM produced by the combination ofR-factor with bacteriophage P1. J. Bacteriol. 88:1266 –1276. 172. Koonin, E. V. 1993. Bacterial and bacteriophage protein phosphatases. Mol. Microbiol. 8:785–786. 173. Kowalczykowski, S. C. 2000. Initiation of genetic recombination and re- combination-dependent replication. Trends Biochem. Sci. 25:156–165. 174. Kravanja, M., R. Engelmann, V. Dossonnet, M. Blu ¨ggel, H. E. Meyer, R. Frank, A. Galinier, J. Deutscher, N. Schnell, and W. Hengstenberg. 1999. The hprK gene of Enterococcus faecalis encodes a novel bifunctional en- zyme: the HPr kinase/phosphatase. Mol. Microbiol. 31:59–66. 175. Krogh, A., B. Larsson, G. von Heijne, and E. L. Sonnhammer. 2001. Pre- dicting transmembrane protein topology with a hidden Markov model:application to complete genomes. J. Mol. Biol. 305:567–580. 176. Kropinski, A. M. 2000. Sequence of the genome of the temperate, serotype- converting, Pseudomonas aeruginosa bacteriophage D3. J. Bacteriol. 182: 6066 –6074. 177. Kropinski, A. M., and M. J. Sibbald. 1999. Transfer RNA genes and their signi ﬁcance to codon usage in the Pseudomonas aeruginosa lamboid bacte- riophage D3. Can J. Microbiol. 45:791–796. 178. Kumar, A., R. A. Malloch, N. Fujita, D. A. Smillie, A. Ishihama, and R. S. Hayward. 1993. The minus 35-recognition region of Escherichia coli sigma 70 is inessential for initiation of transcription at an “extended minus 10 ” promoter. J. Mol. Biol. 232:406–418. 179. Kunisawa, T. 1992. Synonymous codon preferences in bacteriophage T4: a distinctive use of transfer RNAs from T4 and from its host Escherichia coli . J. Theor. Biol. 159:287–298. 180. Kutter, E., K. Gachechiladze, A. Poglazov, E. Marusich, M. Shneider, P. Aronsson, A. Napuli, D. Porter, and V. Mesyanzhinov. 1995. Evolution of T4-related phages. Virus Genes 11:285–297. 181. Kuzminov, A. 1995. Collapse and repair of replication forks in Escherichia coli. Mol. Microbiol. 16:373–384. 182. Kwon, H. J., R. Tirumalai, A. Landy, and T. Ellenberger. 1997. Flexibility in DNA recombination: structure of the lambda integrase catalytic core.Science 276:126–131. 183. Lanka, E., B. Geschke, and H. Schuster. 1978. Escherichia coli dnaB mutant defective in DNA initiation: isolation and properties of the dnaB protein.Proc. Natl. Acad. Sci. USA 75:799–803. 184. Laufer, C. S., J. B. Hays, B. E. Windle, T. S. Schaefer, E. H. Lee, S. L. Hays, and M. R. McClure. 1989. Enhancement of Escherichia coli plasmid and chromosomal recombination by the Ref function of bacteriophage P1. Ge-netics 123:465–476. 184a. Lederberg, E. M. 1950.', 'page_label: 34 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  183. Lanka, E., B. Geschke, and H. Schuster. 1978. Escherichia coli dnaB mutant defective in DNA initiation: isolation and properties of the dnaB protein.Proc. Natl. Acad. Sci. USA 75:799–803. 184. Laufer, C. S., J. B. Hays, B. E. Windle, T. S. Schaefer, E. H. Lee, S. L. Hays, and M. R. McClure. 1989. Enhancement of Escherichia coli plasmid and chromosomal recombination by the Ref function of bacteriophage P1. Ge-netics 123:465–476. 184a. Lederberg, E. M. 1950. Lysogenicity of Escherichia coli strain K-12. Microb. Genet. Bull. 1:5–7. 185. Lederberg, E. M., and J. Lederberg. 1953. Genetic studies of lysogenicity in E. coli. Genetics 38:51–64. 186. Lehnherr, H. Bacteriophage P1. InR. Calendar (ed.), The bacteriophages, in press. Oxford University Press, New York, N.Y. 187. Lehnherr, H., J. D. Bendtsen, F. Preuss, and T. V. Ilyina. 1999. Identi ﬁca- tion and characterization of the single-stranded DNA-binding protein ofbacteriophage P1. J. Bacteriol. 181:6463 –6468. 188. Lehnherr, H., A. Guidolin, and W. Arber. 1991. Bacteriophage P1 gene 10 encodes a trans -activating factor required for late gene expression. J. Bac- teriol. 173:6438 –6445. 189. Lehnherr, H., A. Guidolin, and W. Arber. 1992. Mutational analysis of the bacteriophage P1 late promoter sequence P S. J. Mol. Biol. 228:101–107. 190. Lehnherr, H., A.-M. Hansen, and T. V. Ilyina. 1998. Penetration of the bacterial cell wall: a family of lytic transglycosylases in bacteriophages andconjugative plasmids. Mol. Microbiol. 30:454–457. 191. Lehnherr, H., C. D. Jensen, A. R. Stenholm, and A. Dueholm. 2001. Dual regulatory control of a particle maturation function of bacteriophage P1.J. Bacteriol. 183:4105 –4109. 192. Lehnherr, H., E. Maguin, S. Jafri, and M. Yarmolinsky. 1993. Plasmid addiction genes of bacteriophage P1: doc, which causes cell death on curing of prophage, and phd, which prevents host death when prophage is re- tained. J. Mol. Biol. 233:414–428.VOL. 186, 2004 BACTERIOPHAGE P1 7065', 'page_label: 35 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  193. Lehnherr, H., M. Velleman, A. Guidolin, and W. Arber. 1992. Bacterio- phage P1 gene 10is expressed from a promoter-operator sequence con- trolled by C1 and Bof proteins. J. Bacteriol. 174:6138 –6144. 194. Lehnherr, H., and M. B. Yarmolinsky. 1995. Addiction protein Phd of plasmid prophage P1 is a substrate of the ClpXP serine protease of Esch- erichia coli . Proc. Natl. Acad. Sci. USA 92:3274 –3277. 195. Lemonnier, M., G. Ziegelin, T. Reick, A. M. Gomez, R. Diaz-Orejas, and E. Lanka. 2003. Bacteriophage P1 Ban protein is a hexameric DNA helicase that interacts with and substitutes for Escherichia coli DnaB. Nucleic Acids Res. 31:3918 –3928. 196. Lennox, E. S. 1955. Transduction of linked genetic characters of the host by bacteriophage P1. Virology 1:190–206. 197. Leslie, N. R., and D. J. Sherratt. 1995. Site-speci ﬁc recombination in the replication terminus region of Escherichia coli : functional replacement of dif. EMBO J. 14:1561 –1570. 198. Lesnik, E. A., R. Sampath, H. B. Levene, T. J. Henderson, J. A. McNeil, and D. J. Ecker. 2001. Prediction of rho-independent transcriptional termina- tors in Escherichia coli . Nucleic Acids Res. 29:3583 –3594. 199. Levin, M. E., R. W. Hendrix, and S. R. Casjens. 1993. A programmed translational frameshift is required for the synthesis of a bacteriophagelambda tail assembly protein. J. Mol. Biol. 234:124–139. 200. Lewis, L. K., G. R. Harlow, L. A. Gregg-Jolly, and D. W. Mount. 1994. Identi ﬁcation of high af ﬁnity binding sites for LexA which de ﬁne new DNA damage-inducible genes in Escherichia coli . J. Mol. Biol. 241:507–523. 201. Li, S., H. Pelka, and L. H. Schulman. 1993. The anticodon and discrimi- nator base are important for aminoacylation of Escherichia coli tRNA (Asn). J. Biol. Chem. 268:18335 –18339. 202. Lin, H., V. B. Rao, and L. W. Black. 1999. Analysis of capsid portal protein and terminase functional domains: interaction sites required for DNApackaging in bacteriophage T4. J. Mol. Biol. 289:249–260. 203. Łobocka, M., and M. Yarmolinsky. 1996. P1 plasmid partition: a mutational analysis of ParB. J. Mol. Biol. 259:366–382. 204. Lowe, T. M., and S. R. Eddy. 1997. tRNAscan-SE: a program for improved detection of transfer RNA genes in genomic sequence. Nucleic Acids Res.25:955–964. 205. Lu, M. J., and U. Henning. 1994. Superinfection exclusion by T-even-type coliphages. Trends Microbiol. 2:137–139. 206. Lu, M. J., Y. D. Stierhof, and U. Henning. 1993. Location and unusual membrane topology of the immunity protein of the Escherichia coli phage T4. J. Virol. 67:4905 –4913. 207.', 'page_label: 35 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  J. Mol. Biol. 259:366–382. 204. Lowe, T. M., and S. R. Eddy. 1997. tRNAscan-SE: a program for improved detection of transfer RNA genes in genomic sequence. Nucleic Acids Res.25:955–964. 205. Lu, M. J., and U. Henning. 1994. Superinfection exclusion by T-even-type coliphages. Trends Microbiol. 2:137–139. 206. Lu, M. J., Y. D. Stierhof, and U. Henning. 1993. Location and unusual membrane topology of the immunity protein of the Escherichia coli phage T4. J. Virol. 67:4905 –4913. 207. Lu, S. D., D. Lu, and M. Gottesman. 1989. Stimulation of IS 1excision by bacteriophage P1 reffunction. J. Bacteriol. 171:3427 –3432. 208. Lukashin, A. V., and M. Borodovsky. 1998. GeneMark.hmm: new solutions for gene ﬁnding. Nucleic Acids Res. 26:1107 –1115. 209. Lurz, R., A. Heisig, M. Velleman, B. Dobrinski, and H. Schuster. 1987. The banoperon of bacteriophage P1. Localization of the promoter controlled by P1 repressor. J. Biol. Chem. 262:16575 –16579. 210. Magnuson, R., H. Lehnherr, G. Mukhopadhyay, and M. B. Yarmolinsky. 1996. Autoregulation of the plasmid addiction operon of bacteriophage P1.J. Biol. Chem. 271:18705 –18710. 211. Magnuson, R., and M. B. Yarmolinsky. 1998. Corepression of the P1 addiction operon by Phd and Doc. J. Bacteriol. 180:6342 –6351. 212. Mahillon, J., and M. Chandler. 1998. Insertion sequences. Microbiol. Mol. Biol. Rev. 62:725–774. 213. Maillou, J., and B. Dreiseikelmann. 1990. The simgene of Escherichia coli phage P1: nucleotide sequence and puri ﬁcation of the processed protein. Virology 175:500–507. 214. Malone, T., R. M. Blumenthal, and X. Cheng. 1995. Structure-guided anal- ysis reveals nine sequence motifs conserved among DNA amino-methyl-transferases, and suggests a catalytic mechanism for these enzymes. J. Mol.Biol. 253:618–632. 215. McDonald, J. P., T. S. Peat, A. S. Levine, and R. Woodgate. 1999. Inter- molecular cleavage by UmuD-like enzymes: identi ﬁcation of residues re- quired for cleavage and substrate speci ﬁcity. J. Mol. Biol. 285:2199 –2209. 216. McLean, M. J., K. H. Wolfe, and K. M. Devine. 1998. Base composition skews, replication orientation, and gene orientation in 12 prokaryote ge-nomes. J. Mol. Evol. 47:691–696. 217. McLenigan, M. P., O. I. Kulaeva, D. G. Ennis, A. S. Levine, and R. Wood- gate. 1999. The bacteriophage P1 HumD protein is a functional homolog of the prokaryotic UmuD /H11032-like proteins and facilitates SOS mutagenesis in Escherichia coli . J. Bacteriol. 181:7005 –7013. 218. Mehr, I. J., C. D. Long, C. D. Serkin, and H. S. Seifert. 2000.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:42 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'235'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993567'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'385ms'), (b'x-request-id', b'f0295ff6974a18daca3071d8ec7c643e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124926aca55d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:42 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'235'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993567'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'385ms'), (b'x-request-id', b'f0295ff6974a18daca3071d8ec7c643e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124926aca55d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 35 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1998. Base composition skews, replication orientation, and gene orientation in 12 prokaryote ge-nomes. J. Mol. Evol. 47:691–696. 217. McLenigan, M. P., O. I. Kulaeva, D. G. Ennis, A. S. Levine, and R. Wood- gate. 1999. The bacteriophage P1 HumD protein is a functional homolog of the prokaryotic UmuD /H11032-like proteins and facilitates SOS mutagenesis in Escherichia coli . J. Bacteriol. 181:7005 –7013. 218. Mehr, I. J., C. D. Long, C. D. Serkin, and H. S. Seifert. 2000. A homologue of the recombination-dependent growth gene, rdgC, is involved in gono-coccal pilin antigenic variation. Genetics 154:523–532. 219. Meisel, A., T. A. Bickle, D. H. Kru ¨ger, and C. Schroeder. 1992. Type III restriction enzymes need two inversely oriented recognition sites for DNAcleavage. Nature 355:467–469. 220. Meisel, A., P. Mackeldanz, T. A. Bickle, D. H. Kru ¨ger, and C. Schroeder. 1995. Type III restriction endonucleases translocate DNA in a reactiondriven by recognition site-speci ﬁc ATP hydrolysis. EMBO J. 14:2958 –2966. 221. Mesyanzhinov, V. V., B. N. Sobolev, E. I. Marusich, A. G. Prilipov, and V. P.Eﬁmov. 1990. A proposed structure of bacteriophage T4 gene product 22, a major prohead scaffolding core protein. J. Struct. Biol. 104:24–31. 222. Meyer, J., M. Stalhammar-Carlemalm, and S. Iida. 1981. Denaturation map of bacteriophage P1 DNA. Virology 110:167–175. 223. Miller, E. S., J. F. Heidelberg, J. A. Eisen, W. C. Nelson, A. S. Durkin, A. Ciecko, T. V. Feldblyum, O. White, I. T. Paulsen, W. C. Nierman, J. Lee,B. Szczypinski, and C. M. Fraser. 2003. Complete genome sequence of the broad-host-range vibriophage KVP40: comparative genomics of a T4-re-lated bacteriophage. J. Bacteriol. 185:5220 –5233. 224. Miller, E. S., E. Kutter, G. Mosig, F. Arisaka, T. Kunisawa, and W. Ru ¨ger. 2003. Bacteriophage T4 genome. Microbiol. Mol. Biol. Rev. 67:86–156. 225. Miller, J. H. 1992. A short course in bacterial genetics. A laboratory hand- book for Escherichia coli and related bacteria, p. 15.3 –15.10. Cold Spring Harbor Laboratory Press, Cold Spring Harbor, N.Y. 226. Montag, D., and U. Henning. 1987. An open reading frame in the Esche- richia coli bacteriophage lambda genome encodes a protein that functions in assembly of the long tail ﬁbers of bacteriophage T4. J. Bacteriol. 169: 5884 –5886. 227. Mosig, G. 1994. Synthesis and maturation of T4-encoded tRNas, p. 182 – 185. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 228. Mosig, G., and D. H. Hall. 1994. Gene expression: a paradigm of integrated circuits., p. 127 –131. InJ. D. Karam (ed.), Molecular biology of bacterio- phage T4.', 'page_label: 35 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1987. An open reading frame in the Esche- richia coli bacteriophage lambda genome encodes a protein that functions in assembly of the long tail ﬁbers of bacteriophage T4. J. Bacteriol. 169: 5884 –5886. 227. Mosig, G. 1994. Synthesis and maturation of T4-encoded tRNas, p. 182 – 185. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 228. Mosig, G., and D. H. Hall. 1994. Gene expression: a paradigm of integrated circuits., p. 127 –131. InJ. D. Karam (ed.), Molecular biology of bacterio- phage T4. ASM Press, Washington, D.C. 229. Mulligan, M. E., D. K. Hawley, E. Entriken, and W. R. McClure. 1984. Escherichia coli promoter sequences predict in vitro RNA polymerase se- lectivity. Nucleic Acids Res. 12:789–800. 230. Mural, R. J., R. H. Chesney, D. Vapnek, M. M. Kropf, and J. R. Scott. 1979. Isolation and characterization of cloned fragments of bacteriophage P1DNA. Virology 93:387–397. 231. Muramatsu, T., K. Nishikawa, F. Nemoto, Y. Kuchino, S. Nishimura, T. Miyazawa, and S. Yokoyama. 1988. Codon and amino-acid speci ﬁcities of a transfer RNA are both converted by a single post-transcriptional modi ﬁ- cation. Nature 336:179–181. 232. Muramatsu, T., S. Yokoyama, N. Horie, A. Matsuda, T. Ueda, Z. Yamai- zumi, Y. Kuchino, S. Nishimura, and T. Miyazawa. 1988. A novel lysine- substituted nucleoside in the ﬁrst position of the anticodon of minor iso- leucine tRNA from Escherichia coli . J. Biol. Chem. 263:9261 –9267. 233. Murphy, L. D., J. L. Rosner, S. B. Zimmerman, and D. Esposito. 1999. Identi ﬁcation of two new proteins in spermidine nucleoids isolated from Escherichia coli . J. Bacteriol. 181:3842 –3844. 234. Nainen, O. 1975. Ph.D. thesis. University of Georgia, Athens. 235. Nielsen, H., J. Engelbrecht, S. Brunak, and G. von Heijne. 1997. Identi ﬁ- cation of prokaryotic and eukaryotic signal peptides and prediction of theircleavage sites. Protein Eng. 10:1–6. 236. Nureki, O., T. Niimi, T. Muramatsu, H. Kanno, T. Kohno, C. Florentz, R. Giege, and S. Yokoyama. 1994. Molecular recognition of the identity-de- terminant set of isoleucine transfer RNA from Escherichia coli . J. Mol. Biol. 236:710–724. 237. Ogawa, T. 1975. Analysis of dnaB function of Escherichia coli K12 and the dnaB -like function of P1 prophage. J. Mol. Biol. 94:327–340. 238. Osborne, F. A., S. R. Stovall, and B. R. Baumstark. 1989. The c1 genes of P1 and P7. Nucleic Acids Res. 17:7671 –7680. 239. Pal, S. K., and D. K. Chattoraj. 1988. P1 plasmid replication: initiator sequestration is inadequate to explain control by initiator-binding sites.J. Bacteriol. 170:3554 –3560. 240.', 'page_label: 35 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  J. Mol. Biol. 236:710–724. 237. Ogawa, T. 1975. Analysis of dnaB function of Escherichia coli K12 and the dnaB -like function of P1 prophage. J. Mol. Biol. 94:327–340. 238. Osborne, F. A., S. R. Stovall, and B. R. Baumstark. 1989. The c1 genes of P1 and P7. Nucleic Acids Res. 17:7671 –7680. 239. Pal, S. K., and D. K. Chattoraj. 1988. P1 plasmid replication: initiator sequestration is inadequate to explain control by initiator-binding sites.J. Bacteriol. 170:3554 –3560. 240. Pallanck, L., M. Pak, and H. Schulman. 1995. tRNA discrimination in aminoacylation, p. 371 –394. InD. Soll and U. I. RajBhandary (ed.), tRNA: structure, biosynthesis, and function. ASM Press, Washington, D.C. 241. Park, K., and D. K. Chattoraj. 2001. DnaA boxes in the P1 plasmid origin: the effect of their position on the directionality of replication and plasmidcopy number. J. Mol. Biol. 310:69–81. 242. Parkhill, J., G. Dougan, K. D. James, N. R. Thomson, D. Pickard, J. Wain, C. Churcher, K. L. Mungall, S. D. Bentley, M. T. Holden, M. Sebaihia, S.Baker, D. Basham, K. Brooks, T. Chillingworth, P. Connerton, A. Cronin,P. Davis, R. M. Davies, L. Dowd, N. White, J. Farrar, T. Feltwell, N.Hamlin, A. Haque, T. T. Hien, S. Holroyd, K. Jagels, A. Krogh, T. S.Larsen, S. Leather, S. Moule, P. O ’Gaora, C. Parry, M. Quail, K. Ruther- ford, M. Simmonds, J. Skelton, K. Stevens, S. Whitehead, and B. G. Bar-rell. 2001. Complete genome sequence of a multiple drug resistant Salmo- nella enterica serovar Typhi CT18. Nature 413:848–852. 243. Patel, P. H., M. Suzuki, E. Adman, A. Shinkai, and L. A. Loeb. 2001. Prokaryotic DNA polymerase I: evolution, structure, and “base ﬂipping ” mechanism for nucleotide selection. J. Mol. Biol. 308:823–837. 244. Paul, S., and D. Summers. 2004. ArgR and PepA, accessory proteins for XerCD-mediated resolution of ColE1 dimers, are also required for stablemaintenance of the P1 prophage. Plasmid 52:63–68. 245. Pedulla, M. L., M. E. Ford, J. M. Houtz, T. Karthikeyan, C. Wadsworth, J. A. Lewis, D. Jacobs-Sera, J. Falbo, J. Gross, N. R. Pannunzio, W.Brucker, V. Kumar, J. Kandasamy, L. Keenan, S. Bardarov, J. Kriakov,J. G. Lawrence, W. R. Jacobs, Jr., R. W. Hendrix, and G. F. Hatfull. 2003. Origins of highly mosaic mycobacteriophage genomes. Cell 113:171–182.7066 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 36 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  246. Pierce, J. C., and N. L. Sternberg. 1992. Using bacteriophage P1 system to clone high molecular weight genomic DNA. Methods Enzymol. 216:549– 574. 247. Plunkett, G., III, D. J. Rose, T. J. Durfee, and F. R. Blattner. 1999. Se- quence of Shiga toxin 2 phage 933W from Escherichia coli O157:H7: Shiga toxin as a phage late-gene product. J. Bacteriol. 181:1767 –1778. 248. Prentki, P., M. Chandler, and L. Caro. 1977. Replication of prophage P1 during the cell cycle of Escherichia coli . Mol. Gen. Genet. 152:71–76. 249. Radnedge, L., M. A. Davis, and S. J. Austin. 1996. P1 and P7 plasmid partition: ParB protein bound to its partition site makes a separate discrim-inator contact with the DNA that determines species speci ﬁcity. EMBO J. 15:1155 –1162. 250. Ramos, A., and G. Varani. 1997. Structure of the acceptor stem of Esche- richia coli tRNA Ala: role of the G3.U70 base pair in synthetase recogni- tion. Nucleic Acids Res. 25:2083 –2090. 251. Ravin, N. V., J. Rech, and D. Lane. 2003. Mapping of functional domains in F plasmid partition proteins reveals a bipartite SopB-recognition domain inSopA. J. Mol. Biol. 329:875–889. 252. Razza, J. B., C. A. Watkins, and J. R. Scott. 1980. Phage P1 temperature- sensitive mutants with defects in the lytic pathway. Virology 105:52–59. 253. Redaschi, N., and T. A. Bickle. 1996. Posttranscriptional regulation of EcoP1I and EcoP15I restriction activity. J. Mol. Biol. 257:790–803. 254. Reeve, J. N., E. Lanka, and H. Schuster. 1980. Synthesis of P1 ban protein in minicells infected by P1 mutants. Mol. Gen. Genet. 177:193–197. 255. Reisenauer, A., L. S. Kahng, S. McCollum, and L. Shapiro. 1999. Bacterial DNA methylation: a cell cycle regulator? J. Bacteriol. 181:5135 –5139. 256. Riedel, H. D., J. Heinrich, A. Heisig, T. Choli, and H. Schuster. 1993. The antirepressor of phage P1. Isolation and interaction with the C1 repressorof P1 and P7. FEBS Lett. 334:165–169. 257. Riedel, H. D., J. Heinrich, and H. Schuster. 1993. Cloning, expression, and characterization of the icdgene in the immI operon of bacteriophage P1. J. Bacteriol. 175:2833 –2838. 258. Rodionov, O., M. Łobocka, and M. Yarmolinsky. 1999. Silencing of genes ﬂanking the P1 plasmid centromere. Science 283:546–549. 259. Rodionov, O., and M. Yarmolinsky. 2004. Plasmid partitioning and the spreading of P1 partition protein ParB. Mol. Microbiol. 52:1215 –1223. 260. Rodriguez-Lemoine, V. 1992. PacB, a plasmid-encoded property which con- fers to E. coli K12 resistance to channel-forming colicins. Zentralbl. Bak- teriol. 276:374–379. 261.', 'page_label: 36 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  J. Bacteriol. 175:2833 –2838. 258. Rodionov, O., M. Łobocka, and M. Yarmolinsky. 1999. Silencing of genes ﬂanking the P1 plasmid centromere. Science 283:546–549. 259. Rodionov, O., and M. Yarmolinsky. 2004. Plasmid partitioning and the spreading of P1 partition protein ParB. Mol. Microbiol. 52:1215 –1223. 260. Rodriguez-Lemoine, V. 1992. PacB, a plasmid-encoded property which con- fers to E. coli K12 resistance to channel-forming colicins. Zentralbl. Bak- teriol. 276:374–379. 261. Rosner, J. L. 1972. Formation, induction and curing of bacteriophage Pl lysogens. Virology 48:679–689. 262. Rosner, J. L. 1973. Modi ﬁcation-de ﬁcient mutants of bacteriophage P1. I. Restriction by P1 cryptic lysogens. Virology 52:213–222. 263. Ryder, L., G. J. Sharples, and R. G. Lloyd. 1996. Recombination-dependent growth in exonuclease-depleted recBC sbcBC strains of Escherichia coli K-12. Genetics 143:1101 –1114. 264. Salzberg, S. L., A. L. Delcher, S. Kasif, and O. White. 1998. Microbial gene identi ﬁcation using interpolated Markov models. Nucleic Acids Res. 26: 544–548. 265. Salzberg, S. L., A. J. Salzberg, A. R. Kerlavage, and J. F. Tomb. 1998. Skewed oligomers and origins of replication. Gene 217:57–67. 266. Sambrook, J., E. F. Fritsch, and T. Maniatis. 1989. Molecular cloning. a laboratory manual. Cold Spring Harber Laboratory Press, Cold SpringHarbor, N.Y. 267. Sandmeier, H. 1994. Acquisition and rearrangement of sequence motifs in the evolution of bacteriophage tail ﬁbres. Mol. Microbiol. 12:343–350. 268. Sandmeier, H., S. Iida, and W. Arber. 1992. DNA inversion regions Min of plasmid pl5B and Cin of bacteriophage Pl: evolution of bacteriophage tailﬁber genes. J. Bacteriol. 174:3936 –3944. 269. Sandmeier, H., S. Iida, P. Hu ¨bner, R. Hiestand-Nauer, and W. Arber. 1991. Gene organization in the multiple DNA inversion region min of plasmid p15B of E.coli 15T-: assemblage of a variable gene. Nucleic Acids Res. 19: 5831 –5838. 270. Sandmeier, H., S. Iida, J. Meyer, R. Hiestand-Nauer, and W. Arber. 1990. Site-speci ﬁc DNA recombination system Min of plasmid p15B: a cluster of overlapping invertible DNA segments. Proc. Natl. Acad. Sci. USA 87:1109 – 1113. 271. Sauer, B. 1987. Functional expression of the cre-lox site-speci ﬁc recombi- nation system in the yeast Saccharomyces cerevisiae . Mol. Cell. Biol. 7:2087 – 2096. 272. Schaefer, T., and J. Hays. 1990. The bofgene of bacteriophage P1: DNA sequence and evidence for roles in regulation of phage c1 and refgenes. J. Bacteriol. 172:3269 –3277. 273. Schaefer, T. S., and J. B. Hays. 1991.', 'page_label: 36 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Proc. Natl. Acad. Sci. USA 87:1109 – 1113. 271. Sauer, B. 1987. Functional expression of the cre-lox site-speci ﬁc recombi- nation system in the yeast Saccharomyces cerevisiae . Mol. Cell. Biol. 7:2087 – 2096. 272. Schaefer, T., and J. Hays. 1990. The bofgene of bacteriophage P1: DNA sequence and evidence for roles in regulation of phage c1 and refgenes. J. Bacteriol. 172:3269 –3277. 273. Schaefer, T. S., and J. B. Hays. 1991. Bacteriophage Pl Bof protein is an indirect positive effector of transcription of the Pl bac-1 ban gene in some circumstances and a direct negative effector in other circumstances. J. Bac-teriol. 173:6469 –6474. 274. Scherberg, N. H., and S. B. Weiss. 1972. T4 transfer RNAs: codon recog- nition and translational properties. Proc. Natl. Acad. Sci. USA 69:1114 – 1118. 275. Schmidt, C. 1991. Ph.D thesis. University of Basel, Basel, Switzerland.276. Schmidt, C., H. Lehnherr, A. Guidolin, and W. Arber. 1994. GenBank accession no. M95666. 277. Schmidt, C., M. Velleman, and W. Arber. 1996. Three functions of bacte- riophage P1 involved in cell lysis. J. Bacteriol. 178:1099 –1104. 278. Schuler, G. D., S. F. Altschul, and D. J. Lipman. 1991. A workbench for multiple alignment construction and analysis. Proteins 9:180–190. 279. Scott, J. R. 1970. Clear plaque mutants of phage Pl. Virology 41:66–71. 280. Scott, J. R., M. Kropf, and L. Mendelson. 1977. Clear plaque mutants of phage P7. Virology 76:39–46. 281. Scott, J. R., and M. M. Kropf. 1977. Location of new clear plaque genes on the P1 map. Virology 82:362–368. 282. Segev, N., A. Laub, and G. Cohen. 1980. A circular form of bacteriophage P1 DNA made in lytically infected cells of Escherichia coli. Characterization and kinetics of formation. Virology 101:261–271. 283. Senger, B., S. Auxilien, U. Englisch, F. Cramer, and F. Fasiolo. 1997. The modi ﬁed wobble base inosine in yeast tRNA-Ile is a positive determinant for aminoacylation by isoleucyl-tRNA synthetase. Biohemistry 36:8269 – 8275. 284. Sharpe, M. E., and J. Errington. 1996. The Bacillus subtilis soj-spo0J locus is required for a centromere-like function involved in prespore chromo-some partitioning. Mol. Microbiol. 21:501–509. 285. Sharrocks, A. D., and D. P. Hornby. 1991. Transcriptional analysis of the restriction and modi ﬁcation genes of bacteriophage P1. Mol. Microbiol. 5: 685–694. 286. Skorupski, K., J. C. Pierce, B. Sauer, and N. Sternberg. 1992. Bacterio- phage Pl genes involved in the recognition and cleavage of the phagepackaging site ( pac). J. Mol. Biol. 223:977–989. 287. Skorupski, K., B. Sauer, and N. Sternberg.', 'page_label: 36 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Mol. Microbiol. 21:501–509. 285. Sharrocks, A. D., and D. P. Hornby. 1991. Transcriptional analysis of the restriction and modi ﬁcation genes of bacteriophage P1. Mol. Microbiol. 5: 685–694. 286. Skorupski, K., J. C. Pierce, B. Sauer, and N. Sternberg. 1992. Bacterio- phage Pl genes involved in the recognition and cleavage of the phagepackaging site ( pac). J. Mol. Biol. 223:977–989. 287. Skorupski, K., B. Sauer, and N. Sternberg. 1994. Faithful cleavage of the P1 packaging site ( pac) requires two phage proteins, PacA and PacB, and two Escherichia coli proteins, IHF and HU. J. Mol. Biol. 243:268–282. 288. Skorupski, K., N. Sternberg, and B. Sauer. 1994. Puri ﬁcation and DNA- binding activity of the PacA subunit of the bacteriophage P1 pacase en-zyme. J. Mol. Biol. 243:258–267. 289. Slater, S. C., M. R. Lifsics, M. O ’Donnell, and R. Maurer. 1994. holE , the gene coding for the theta subunit of DNA polymerase III of Escherichia coli: characterization of a holE mutant and comparison with a dnaQ (ep- silon-subunit) mutant. J. Bacteriol. 176:815–821. 290. Sommer, S., J. Knezevic, A. Bailone, and R. Devoret. 1993. Induction of only one SOS operon, umuDC, is required for SOS mutagenesis in Esch- erichia coli . Mol. Gen. Genet. 239:137–144. 291. Sourice, S., V. Biaudet, M. El Karoui, S. D. Ehrlich, and A. Gruss. 1998. Identi ﬁcation of the Chi site of Haemophilus in ﬂuenzae as several sequences related to the Escherichia coli Chi site. Mol. Microbiol. 27:1021 –1029. 292. Steiner, W., G. Liu, W. D. Donachie, and P. Kuempel. 1999. The cytoplas- mic domain of FtsK protein is required for resolution of chromosomedimers. Mol. Microbiol. 31:579–583. 293. Sternberg, N. 1990. Bacteriophage P1 cloning system for the isolation, ampli ﬁcation, and recovery of DNA fragments as large as 100 kilobase pairs. Proc. Natl. Acad. Sci. USA 87:103–107. 294. Sternberg, N. 1979. A characterization of bacteriophage P1 DNA fragments cloned in a lambda vector. Virology 96:129–142. 295. Sternberg, N., and G. Cohen. 1989. Genetic analysis of the lytic replicon of bacteriophage Pl. II. Organization of replicon elements. J. Mol. Biol. 207: 111–133. 296. Sternberg, N., and J. Coulby. 1990. Cleavage of the bacteriophage P1 packaging site ( pac) is regulated by adenine methylation. Proc. Natl. Acad. Sci. USA 87:8070 –8074. 297. Sternberg, N., and J. Coulby. 1987. Recognition and cleavage of the bac- teriophage Pl packaging site ( pac) I. Differential processing of the cleaved ends in vivo . J. Mol. Biol. 194:453–468. 298. Sternberg, N., and J. Coulby. 1987. Recognition and cleavage of the bac- teriophage Pl packaging site ( pac) II.', 'page_label: 36 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  II. Organization of replicon elements. J. Mol. Biol. 207: 111–133. 296. Sternberg, N., and J. Coulby. 1990. Cleavage of the bacteriophage P1 packaging site ( pac) is regulated by adenine methylation. Proc. Natl. Acad. Sci. USA 87:8070 –8074. 297. Sternberg, N., and J. Coulby. 1987. Recognition and cleavage of the bac- teriophage Pl packaging site ( pac) I. Differential processing of the cleaved ends in vivo . J. Mol. Biol. 194:453–468. 298. Sternberg, N., and J. Coulby. 1987. Recognition and cleavage of the bac- teriophage Pl packaging site ( pac) II. Functional limits of pacand location ofpaccleavage termini. J. Mol. Biol. 194:469–480. 299. Sternberg, N., D. Hamilton, S. Austin, M. Yarmolinsky, and R. Hoess. 1981. Site-speci ﬁc recombination and its role in the life cycle of bacteriophage P1. Cold Spring Harbor Symp. Quant. Biol. 45:297–309. 300. Sternberg, N., D. Hamilton, and R. Hoess. 1981. Bacteriophage P1 site- speci ﬁc recombination. II. Recombination between loxP and the bacterial chromosome. J. Mol. Biol. 150:487–507. 301. Sternberg, N., and R. Hoess. 1983. The molecular genetics of bacteriophage P1. Annu. Rev. Genet. 17:123–154. 302. Sternberg, N., B. Sauer, R. Hoess, and K. Abremski. 1986. Bacteriophage Pl crestructural gene and its regulatory region. Evidence for multiple promot- ers and for regulation by DNA methylation. J. Mol. Biol. 187:197–212. 303. Sternberg, N. L., and R. Maurer. 1991. Bacteriophage-mediated general- ized transduction in Escherichia coli andSalmonella typhimurium . Methods Enzymol. 204:18–43. 304. Streiff, M. B., S. Iida, and T. A. Bickle. 1987. Expression and proteolytic processing of the darA anti-restriction gene product of bacteriophage P1. Virology 157:167–171. 305. Studwell-Vaughan, P. S., and M. O ’Donnell. 1993. DNA polymerase IIIVOL. 186, 2004 BACTERIOPHAGE P1 7067', 'page_label: 37 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  accessory proteins. V. Theta encoded by holE . J. Biol. Chem. 268:11785 – 11791. 306. Surtees, J. A., and B. E. Funnell. 2003. Plasmid and chromosome traf ﬁc control: how ParA and ParB drive partition. Curr. Top. Dev. Biol. 56: 145–180. 307. Susskind, M. M., A. Wright, and D. Botstein. 1971. Superinfection exclu- sion by P22 prophage in lysogens of Salmonella typhimurium. II. Genetic evidence for two exclusion systems. Virology 45:638–652. 308. Tamura, K., H. Asahara, H. Himeno, T. Hasegawa, and M. Shimizu. 1991. Identity elements of Escherichia coli tRNA(Ala). J. Mol. Recognit. 4:129– 132. 309. Tang, M., X. Shen, E. G. Frank, M. O ’Donnell, R. Woodgate, and M. F. Goodman. 1999. UmuD /H11032(2)C is an error-prone DNA polymerase, Esche- richia coli pol V. Proc. Natl. Acad. Sci. USA 96:8919 –8924. 310. Touati-Schwartz, D. 1979. A dnaB analog ban, speci ﬁed by bacteriophage P1: genetic and physiological evidence for functional analogy and interac-tions between the two products. Mol. Gen. Genet. 174:173–188. 311. Touati-Schwartz, D. 1979. A new pleiotropic bacteriophage P1 mutation, bof, affecting c1 repression activity, the expression of plasmid incompati- bility and the expression of certain constitutive prophage genes. Mol. Gen.Genet. 174:189–202. 312. Touati-Schwartz, D. 1978. Two replication functions in phage P1: ban,a n analog of dnaB , and bof, involved in the control of replication., p. 683 –691. InI. Molineaux, and M. Kohiyama (ed.), DNA synthesis: present and future, vol. 17. Plenum Press, New York, N.Y. 313. Tuggle, C. K., and J. A. Fuchs. 1986. Regulation of the synthesis of ribo- nucleotide reductase in Escherichia coli. ,p .2 2 7 –232. InC.-I. Brandon, A. Holmgren, J.-O. Hoag, H. Jornvall, and B.-M. Sjoberg (ed.), Thioredoxinand glutaredoxin systems: structure and function. Raven Press, New York,N.Y. 314. Tuma, R., and G. J. Thomas, Jr. 1997. Mechanisms of virus assembly probed by Raman spectroscopy: the icosahedral bacteriophage P22. Bio-phys. Chem. 68:17–31. 315. Velleman, M. 1989. PhD thesis. Freie Universitat Berlin, Berlin, Germany. 316. Velleman, M., B. Dreisekelmann, and H. Schuster. 1987. Multiple repressor binding sites in the genome of bacteriophage Pl. Proc. Natl. Acad. Sci. USA84:5570 –5574. 317. Velleman, M., T. Heinzel, and H. Schuster. 1992. The Bof protein of bacteriophage Pl exerts its modulating function by formation of a ternarycomplex with operator DNA and Cl repressor. J. Biol. Chem. 267:12174 – 12181. 318. Velleman, M., M. Heirich, A. Gu ¨nther, and H. Schuster. 1990. A bacterio- phage P1-encoded modulator protein affects the P1 c1 repression system. J. Mol.', 'page_label: 37 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Velleman, M., B. Dreisekelmann, and H. Schuster. 1987. Multiple repressor binding sites in the genome of bacteriophage Pl. Proc. Natl. Acad. Sci. USA84:5570 –5574. 317. Velleman, M., T. Heinzel, and H. Schuster. 1992. The Bof protein of bacteriophage Pl exerts its modulating function by formation of a ternarycomplex with operator DNA and Cl repressor. J. Biol. Chem. 267:12174 – 12181. 318. Velleman, M., M. Heirich, A. Gu ¨nther, and H. Schuster. 1990. A bacterio- phage P1-encoded modulator protein affects the P1 c1 repression system. J. Mol. Biol. 265:18511 –18517. 319. Velleman, M., and S. Parbus. 1992. Puri ﬁcation of the C1 repressor of bacteriophage P1 by fast protein liquid chromatography. J. Chromatogr.625:41–46. 320. Voegtli, W. C., D. J. White, N. J. Reiter, F. Rusnak, and A. C. Rosenzweig. 2000. Structure of the bacteriophage lambda Ser/Thr protein phosphatasewith sulfate ion bound in two coordination modes. Biochemistry 39:15365 – 15374. 321. Walker, D. H., Jr., and J. T. Walker. 1976. Genetic studies of coliphage P1. III. Extended genetic map. J. Virol. 20:177–187. 322. Walker, D. H. J., and T. F. Anderson. 1970. Morphological variants of coliphage P1. J. Virol. 5:765–782. 323. Walker, J. T., and D. H. J. Walker. 1983. Coliphage P1 morphogenesis: analysis of mutants by electron microscopy. J. Bacteriol. 45:1118 –1139. 324. Walker, J. T., and D. H. J. Walker. 1980. Mutations in coliphage P1 affecting host cell lysis. J. Virol. 35:519–530. 325. Walker, J. T., and D. H. J. Walker. 1981. Structural proteins of coliphage P1. Prog. Clin. Biol. Res. 64:69–77. 326. Wandersman, C., and M. Yarmolinsky. 1977. Bipartite control of immunity conferred by the related heteroimmune plasmid prophages, P1 and P7(formerly phi Amp). Virology 77:386–400. 327. Wang, I. N., D. L. Smith, and R. Young. 2000. Holins: the protein clocks of bacteriophage infections. Annu. Rev. Microbiol. 54:799–825. 328. Wang, L. K., and S. Shuman. 2001. Domain structure and mutational analysis of T4 polynucleotide kinase. J. Biol. Chem. 276:26868 –26874. 329. Watts, N. R., and D. H. Coombs. 1990. Structure of the bacteriophage T4 baseplate as determined by chemical cross-linking. J. Virol. 64:143–154. 330. Weisberg, R. A., and M. E. Gottesman. 1999. Processive antitermination. J. Bacteriol. 181:359–367. 331. Whelan, K. F., E. Colleran, and D. E. Taylor. 1995. Phage inhibition, colicin resistance, and tellurite resistance are encoded by a single cluster of geneson the IncHI2 plasmid R478. J. Bacteriol. 177:5016 –5027.332. Whelan, K. F., R. K. Sherburne, and D. E. Taylor. 1997.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 35 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1998. Base composition skews, replication orientation, and gene orientation in 12 prokaryote ge-nomes. J. Mol. Evol. 47:691–696. 217. McLenigan, M. P., O. I. Kulaeva, D. G. Ennis, A. S. Levine, and R. Wood- gate. 1999. The bacteriophage P1 HumD protein is a functional homolog of the prokaryotic UmuD /H11032-like proteins and facilitates SOS mutagenesis in Escherichia coli . J. Bacteriol. 181:7005 –7013. 218. Mehr, I. J., C. D. Long, C. D. Serkin, and H. S. Seifert. 2000. A homologue of the recombination-dependent growth gene, rdgC, is involved in gono-coccal pilin antigenic variation. Genetics 154:523–532. 219. Meisel, A., T. A. Bickle, D. H. Kru ¨ger, and C. Schroeder. 1992. Type III restriction enzymes need two inversely oriented recognition sites for DNAcleavage. Nature 355:467–469. 220. Meisel, A., P. Mackeldanz, T. A. Bickle, D. H. Kru ¨ger, and C. Schroeder. 1995. Type III restriction endonucleases translocate DNA in a reactiondriven by recognition site-speci ﬁc ATP hydrolysis. EMBO J. 14:2958 –2966. 221. Mesyanzhinov, V. V., B. N. Sobolev, E. I. Marusich, A. G. Prilipov, and V. P.Eﬁmov. 1990. A proposed structure of bacteriophage T4 gene product 22, a major prohead scaffolding core protein. J. Struct. Biol. 104:24–31. 222. Meyer, J., M. Stalhammar-Carlemalm, and S. Iida. 1981. Denaturation map of bacteriophage P1 DNA. Virology 110:167–175. 223. Miller, E. S., J. F. Heidelberg, J. A. Eisen, W. C. Nelson, A. S. Durkin, A. Ciecko, T. V. Feldblyum, O. White, I. T. Paulsen, W. C. Nierman, J. Lee,B. Szczypinski, and C. M. Fraser. 2003. Complete genome sequence of the broad-host-range vibriophage KVP40: comparative genomics of a T4-re-lated bacteriophage. J. Bacteriol. 185:5220 –5233. 224. Miller, E. S., E. Kutter, G. Mosig, F. Arisaka, T. Kunisawa, and W. Ru ¨ger. 2003. Bacteriophage T4 genome. Microbiol. Mol. Biol. Rev. 67:86–156. 225. Miller, J. H. 1992. A short course in bacterial genetics. A laboratory hand- book for Escherichia coli and related bacteria, p. 15.3 –15.10. Cold Spring Harbor Laboratory Press, Cold Spring Harbor, N.Y. 226. Montag, D., and U. Henning. 1987. An open reading frame in the Esche- richia coli bacteriophage lambda genome encodes a protein that functions in assembly of the long tail ﬁbers of bacteriophage T4. J. Bacteriol. 169: 5884 –5886. 227. Mosig, G. 1994. Synthesis and maturation of T4-encoded tRNas, p. 182 – 185. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 228. Mosig, G., and D. H. Hall. 1994. Gene expression: a paradigm of integrated circuits., p. 127 –131. InJ. D. Karam (ed.), Molecular biology of bacterio- phage T4.', 'page_label: 35 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1987. An open reading frame in the Esche- richia coli bacteriophage lambda genome encodes a protein that functions in assembly of the long tail ﬁbers of bacteriophage T4. J. Bacteriol. 169: 5884 –5886. 227. Mosig, G. 1994. Synthesis and maturation of T4-encoded tRNas, p. 182 – 185. InJ. D. Karam (ed.), Molecular biology of bacteriophage T4. ASM Press, Washington, D.C. 228. Mosig, G., and D. H. Hall. 1994. Gene expression: a paradigm of integrated circuits., p. 127 –131. InJ. D. Karam (ed.), Molecular biology of bacterio- phage T4. ASM Press, Washington, D.C. 229. Mulligan, M. E., D. K. Hawley, E. Entriken, and W. R. McClure. 1984. Escherichia coli promoter sequences predict in vitro RNA polymerase se- lectivity. Nucleic Acids Res. 12:789–800. 230. Mural, R. J., R. H. Chesney, D. Vapnek, M. M. Kropf, and J. R. Scott. 1979. Isolation and characterization of cloned fragments of bacteriophage P1DNA. Virology 93:387–397. 231. Muramatsu, T., K. Nishikawa, F. Nemoto, Y. Kuchino, S. Nishimura, T. Miyazawa, and S. Yokoyama. 1988. Codon and amino-acid speci ﬁcities of a transfer RNA are both converted by a single post-transcriptional modi ﬁ- cation. Nature 336:179–181. 232. Muramatsu, T., S. Yokoyama, N. Horie, A. Matsuda, T. Ueda, Z. Yamai- zumi, Y. Kuchino, S. Nishimura, and T. Miyazawa. 1988. A novel lysine- substituted nucleoside in the ﬁrst position of the anticodon of minor iso- leucine tRNA from Escherichia coli . J. Biol. Chem. 263:9261 –9267. 233. Murphy, L. D., J. L. Rosner, S. B. Zimmerman, and D. Esposito. 1999. Identi ﬁcation of two new proteins in spermidine nucleoids isolated from Escherichia coli . J. Bacteriol. 181:3842 –3844. 234. Nainen, O. 1975. Ph.D. thesis. University of Georgia, Athens. 235. Nielsen, H., J. Engelbrecht, S. Brunak, and G. von Heijne. 1997. Identi ﬁ- cation of prokaryotic and eukaryotic signal peptides and prediction of theircleavage sites. Protein Eng. 10:1–6. 236. Nureki, O., T. Niimi, T. Muramatsu, H. Kanno, T. Kohno, C. Florentz, R. Giege, and S. Yokoyama. 1994. Molecular recognition of the identity-de- terminant set of isoleucine transfer RNA from Escherichia coli . J. Mol. Biol. 236:710–724. 237. Ogawa, T. 1975. Analysis of dnaB function of Escherichia coli K12 and the dnaB -like function of P1 prophage. J. Mol. Biol. 94:327–340. 238. Osborne, F. A., S. R. Stovall, and B. R. Baumstark. 1989. The c1 genes of P1 and P7. Nucleic Acids Res. 17:7671 –7680. 239. Pal, S. K., and D. K. Chattoraj. 1988. P1 plasmid replication: initiator sequestration is inadequate to explain control by initiator-binding sites.J. Bacteriol. 170:3554 –3560. 240.', 'page_label: 35 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  J. Mol. Biol. 236:710–724. 237. Ogawa, T. 1975. Analysis of dnaB function of Escherichia coli K12 and the dnaB -like function of P1 prophage. J. Mol. Biol. 94:327–340. 238. Osborne, F. A., S. R. Stovall, and B. R. Baumstark. 1989. The c1 genes of P1 and P7. Nucleic Acids Res. 17:7671 –7680. 239. Pal, S. K., and D. K. Chattoraj. 1988. P1 plasmid replication: initiator sequestration is inadequate to explain control by initiator-binding sites.J. Bacteriol. 170:3554 –3560. 240. Pallanck, L., M. Pak, and H. Schulman. 1995. tRNA discrimination in aminoacylation, p. 371 –394. InD. Soll and U. I. RajBhandary (ed.), tRNA: structure, biosynthesis, and function. ASM Press, Washington, D.C. 241. Park, K., and D. K. Chattoraj. 2001. DnaA boxes in the P1 plasmid origin: the effect of their position on the directionality of replication and plasmidcopy number. J. Mol. Biol. 310:69–81. 242. Parkhill, J., G. Dougan, K. D. James, N. R. Thomson, D. Pickard, J. Wain, C. Churcher, K. L. Mungall, S. D. Bentley, M. T. Holden, M. Sebaihia, S.Baker, D. Basham, K. Brooks, T. Chillingworth, P. Connerton, A. Cronin,P. Davis, R. M. Davies, L. Dowd, N. White, J. Farrar, T. Feltwell, N.Hamlin, A. Haque, T. T. Hien, S. Holroyd, K. Jagels, A. Krogh, T. S.Larsen, S. Leather, S. Moule, P. O ’Gaora, C. Parry, M. Quail, K. Ruther- ford, M. Simmonds, J. Skelton, K. Stevens, S. Whitehead, and B. G. Bar-rell. 2001. Complete genome sequence of a multiple drug resistant Salmo- nella enterica serovar Typhi CT18. Nature 413:848–852. 243. Patel, P. H., M. Suzuki, E. Adman, A. Shinkai, and L. A. Loeb. 2001. Prokaryotic DNA polymerase I: evolution, structure, and “base ﬂipping ” mechanism for nucleotide selection. J. Mol. Biol. 308:823–837. 244. Paul, S., and D. Summers. 2004. ArgR and PepA, accessory proteins for XerCD-mediated resolution of ColE1 dimers, are also required for stablemaintenance of the P1 prophage. Plasmid 52:63–68. 245. Pedulla, M. L., M. E. Ford, J. M. Houtz, T. Karthikeyan, C. Wadsworth, J. A. Lewis, D. Jacobs-Sera, J. Falbo, J. Gross, N. R. Pannunzio, W.Brucker, V. Kumar, J. Kandasamy, L. Keenan, S. Bardarov, J. Kriakov,J. G. Lawrence, W. R. Jacobs, Jr., R. W. Hendrix, and G. F. Hatfull. 2003. Origins of highly mosaic mycobacteriophage genomes. Cell 113:171–182.7066 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 36 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  246. Pierce, J. C., and N. L. Sternberg. 1992. Using bacteriophage P1 system to clone high molecular weight genomic DNA. Methods Enzymol. 216:549– 574. 247. Plunkett, G., III, D. J. Rose, T. J. Durfee, and F. R. Blattner. 1999. Se- quence of Shiga toxin 2 phage 933W from Escherichia coli O157:H7: Shiga toxin as a phage late-gene product. J. Bacteriol. 181:1767 –1778. 248. Prentki, P., M. Chandler, and L. Caro. 1977. Replication of prophage P1 during the cell cycle of Escherichia coli . Mol. Gen. Genet. 152:71–76. 249. Radnedge, L., M. A. Davis, and S. J. Austin. 1996. P1 and P7 plasmid partition: ParB protein bound to its partition site makes a separate discrim-inator contact with the DNA that determines species speci ﬁcity. EMBO J. 15:1155 –1162. 250. Ramos, A., and G. Varani. 1997. Structure of the acceptor stem of Esche- richia coli tRNA Ala: role of the G3.U70 base pair in synthetase recogni- tion. Nucleic Acids Res. 25:2083 –2090. 251. Ravin, N. V., J. Rech, and D. Lane. 2003. Mapping of functional domains in F plasmid partition proteins reveals a bipartite SopB-recognition domain inSopA. J. Mol. Biol. 329:875–889. 252. Razza, J. B., C. A. Watkins, and J. R. Scott. 1980. Phage P1 temperature- sensitive mutants with defects in the lytic pathway. Virology 105:52–59. 253. Redaschi, N., and T. A. Bickle. 1996. Posttranscriptional regulation of EcoP1I and EcoP15I restriction activity. J. Mol. Biol. 257:790–803. 254. Reeve, J. N., E. Lanka, and H. Schuster. 1980. Synthesis of P1 ban protein in minicells infected by P1 mutants. Mol. Gen. Genet. 177:193–197. 255. Reisenauer, A., L. S. Kahng, S. McCollum, and L. Shapiro. 1999. Bacterial DNA methylation: a cell cycle regulator? J. Bacteriol. 181:5135 –5139. 256. Riedel, H. D., J. Heinrich, A. Heisig, T. Choli, and H. Schuster. 1993. The antirepressor of phage P1. Isolation and interaction with the C1 repressorof P1 and P7. FEBS Lett. 334:165–169. 257. Riedel, H. D., J. Heinrich, and H. Schuster. 1993. Cloning, expression, and characterization of the icdgene in the immI operon of bacteriophage P1. J. Bacteriol. 175:2833 –2838. 258. Rodionov, O., M. Łobocka, and M. Yarmolinsky. 1999. Silencing of genes ﬂanking the P1 plasmid centromere. Science 283:546–549. 259. Rodionov, O., and M. Yarmolinsky. 2004. Plasmid partitioning and the spreading of P1 partition protein ParB. Mol. Microbiol. 52:1215 –1223. 260. Rodriguez-Lemoine, V. 1992. PacB, a plasmid-encoded property which con- fers to E. coli K12 resistance to channel-forming colicins. Zentralbl. Bak- teriol. 276:374–379. 261.', 'page_label: 36 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  J. Bacteriol. 175:2833 –2838. 258. Rodionov, O., M. Łobocka, and M. Yarmolinsky. 1999. Silencing of genes ﬂanking the P1 plasmid centromere. Science 283:546–549. 259. Rodionov, O., and M. Yarmolinsky. 2004. Plasmid partitioning and the spreading of P1 partition protein ParB. Mol. Microbiol. 52:1215 –1223. 260. Rodriguez-Lemoine, V. 1992. PacB, a plasmid-encoded property which con- fers to E. coli K12 resistance to channel-forming colicins. Zentralbl. Bak- teriol. 276:374–379. 261. Rosner, J. L. 1972. Formation, induction and curing of bacteriophage Pl lysogens. Virology 48:679–689. 262. Rosner, J. L. 1973. Modi ﬁcation-de ﬁcient mutants of bacteriophage P1. I. Restriction by P1 cryptic lysogens. Virology 52:213–222. 263. Ryder, L., G. J. Sharples, and R. G. Lloyd. 1996. Recombination-dependent growth in exonuclease-depleted recBC sbcBC strains of Escherichia coli K-12. Genetics 143:1101 –1114. 264. Salzberg, S. L., A. L. Delcher, S. Kasif, and O. White. 1998. Microbial gene identi ﬁcation using interpolated Markov models. Nucleic Acids Res. 26: 544–548. 265. Salzberg, S. L., A. J. Salzberg, A. R. Kerlavage, and J. F. Tomb. 1998. Skewed oligomers and origins of replication. Gene 217:57–67. 266. Sambrook, J., E. F. Fritsch, and T. Maniatis. 1989. Molecular cloning. a laboratory manual. Cold Spring Harber Laboratory Press, Cold SpringHarbor, N.Y. 267. Sandmeier, H. 1994. Acquisition and rearrangement of sequence motifs in the evolution of bacteriophage tail ﬁbres. Mol. Microbiol. 12:343–350. 268. Sandmeier, H., S. Iida, and W. Arber. 1992. DNA inversion regions Min of plasmid pl5B and Cin of bacteriophage Pl: evolution of bacteriophage tailﬁber genes. J. Bacteriol. 174:3936 –3944. 269. Sandmeier, H., S. Iida, P. Hu ¨bner, R. Hiestand-Nauer, and W. Arber. 1991. Gene organization in the multiple DNA inversion region min of plasmid p15B of E.coli 15T-: assemblage of a variable gene. Nucleic Acids Res. 19: 5831 –5838. 270. Sandmeier, H., S. Iida, J. Meyer, R. Hiestand-Nauer, and W. Arber. 1990. Site-speci ﬁc DNA recombination system Min of plasmid p15B: a cluster of overlapping invertible DNA segments. Proc. Natl. Acad. Sci. USA 87:1109 – 1113. 271. Sauer, B. 1987. Functional expression of the cre-lox site-speci ﬁc recombi- nation system in the yeast Saccharomyces cerevisiae . Mol. Cell. Biol. 7:2087 – 2096. 272. Schaefer, T., and J. Hays. 1990. The bofgene of bacteriophage P1: DNA sequence and evidence for roles in regulation of phage c1 and refgenes. J. Bacteriol. 172:3269 –3277. 273. Schaefer, T. S., and J. B. Hays. 1991.', 'page_label: 36 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Proc. Natl. Acad. Sci. USA 87:1109 – 1113. 271. Sauer, B. 1987. Functional expression of the cre-lox site-speci ﬁc recombi- nation system in the yeast Saccharomyces cerevisiae . Mol. Cell. Biol. 7:2087 – 2096. 272. Schaefer, T., and J. Hays. 1990. The bofgene of bacteriophage P1: DNA sequence and evidence for roles in regulation of phage c1 and refgenes. J. Bacteriol. 172:3269 –3277. 273. Schaefer, T. S., and J. B. Hays. 1991. Bacteriophage Pl Bof protein is an indirect positive effector of transcription of the Pl bac-1 ban gene in some circumstances and a direct negative effector in other circumstances. J. Bac-teriol. 173:6469 –6474. 274. Scherberg, N. H., and S. B. Weiss. 1972. T4 transfer RNAs: codon recog- nition and translational properties. Proc. Natl. Acad. Sci. USA 69:1114 – 1118. 275. Schmidt, C. 1991. Ph.D thesis. University of Basel, Basel, Switzerland.276. Schmidt, C., H. Lehnherr, A. Guidolin, and W. Arber. 1994. GenBank accession no. M95666. 277. Schmidt, C., M. Velleman, and W. Arber. 1996. Three functions of bacte- riophage P1 involved in cell lysis. J. Bacteriol. 178:1099 –1104. 278. Schuler, G. D., S. F. Altschul, and D. J. Lipman. 1991. A workbench for multiple alignment construction and analysis. Proteins 9:180–190. 279. Scott, J. R. 1970. Clear plaque mutants of phage Pl. Virology 41:66–71. 280. Scott, J. R., M. Kropf, and L. Mendelson. 1977. Clear plaque mutants of phage P7. Virology 76:39–46. 281. Scott, J. R., and M. M. Kropf. 1977. Location of new clear plaque genes on the P1 map. Virology 82:362–368. 282. Segev, N., A. Laub, and G. Cohen. 1980. A circular form of bacteriophage P1 DNA made in lytically infected cells of Escherichia coli. Characterization and kinetics of formation. Virology 101:261–271. 283. Senger, B., S. Auxilien, U. Englisch, F. Cramer, and F. Fasiolo. 1997. The modi ﬁed wobble base inosine in yeast tRNA-Ile is a positive determinant for aminoacylation by isoleucyl-tRNA synthetase. Biohemistry 36:8269 – 8275. 284. Sharpe, M. E., and J. Errington. 1996. The Bacillus subtilis soj-spo0J locus is required for a centromere-like function involved in prespore chromo-some partitioning. Mol. Microbiol. 21:501–509. 285. Sharrocks, A. D., and D. P. Hornby. 1991. Transcriptional analysis of the restriction and modi ﬁcation genes of bacteriophage P1. Mol. Microbiol. 5: 685–694. 286. Skorupski, K., J. C. Pierce, B. Sauer, and N. Sternberg. 1992. Bacterio- phage Pl genes involved in the recognition and cleavage of the phagepackaging site ( pac). J. Mol. Biol. 223:977–989. 287. Skorupski, K., B. Sauer, and N. Sternberg.', 'page_label: 36 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Mol. Microbiol. 21:501–509. 285. Sharrocks, A. D., and D. P. Hornby. 1991. Transcriptional analysis of the restriction and modi ﬁcation genes of bacteriophage P1. Mol. Microbiol. 5: 685–694. 286. Skorupski, K., J. C. Pierce, B. Sauer, and N. Sternberg. 1992. Bacterio- phage Pl genes involved in the recognition and cleavage of the phagepackaging site ( pac). J. Mol. Biol. 223:977–989. 287. Skorupski, K., B. Sauer, and N. Sternberg. 1994. Faithful cleavage of the P1 packaging site ( pac) requires two phage proteins, PacA and PacB, and two Escherichia coli proteins, IHF and HU. J. Mol. Biol. 243:268–282. 288. Skorupski, K., N. Sternberg, and B. Sauer. 1994. Puri ﬁcation and DNA- binding activity of the PacA subunit of the bacteriophage P1 pacase en-zyme. J. Mol. Biol. 243:258–267. 289. Slater, S. C., M. R. Lifsics, M. O ’Donnell, and R. Maurer. 1994. holE , the gene coding for the theta subunit of DNA polymerase III of Escherichia coli: characterization of a holE mutant and comparison with a dnaQ (ep- silon-subunit) mutant. J. Bacteriol. 176:815–821. 290. Sommer, S., J. Knezevic, A. Bailone, and R. Devoret. 1993. Induction of only one SOS operon, umuDC, is required for SOS mutagenesis in Esch- erichia coli . Mol. Gen. Genet. 239:137–144. 291. Sourice, S., V. Biaudet, M. El Karoui, S. D. Ehrlich, and A. Gruss. 1998. Identi ﬁcation of the Chi site of Haemophilus in ﬂuenzae as several sequences related to the Escherichia coli Chi site. Mol. Microbiol. 27:1021 –1029. 292. Steiner, W., G. Liu, W. D. Donachie, and P. Kuempel. 1999. The cytoplas- mic domain of FtsK protein is required for resolution of chromosomedimers. Mol. Microbiol. 31:579–583. 293. Sternberg, N. 1990. Bacteriophage P1 cloning system for the isolation, ampli ﬁcation, and recovery of DNA fragments as large as 100 kilobase pairs. Proc. Natl. Acad. Sci. USA 87:103–107. 294. Sternberg, N. 1979. A characterization of bacteriophage P1 DNA fragments cloned in a lambda vector. Virology 96:129–142. 295. Sternberg, N., and G. Cohen. 1989. Genetic analysis of the lytic replicon of bacteriophage Pl. II. Organization of replicon elements. J. Mol. Biol. 207: 111–133. 296. Sternberg, N., and J. Coulby. 1990. Cleavage of the bacteriophage P1 packaging site ( pac) is regulated by adenine methylation. Proc. Natl. Acad. Sci. USA 87:8070 –8074. 297. Sternberg, N., and J. Coulby. 1987. Recognition and cleavage of the bac- teriophage Pl packaging site ( pac) I. Differential processing of the cleaved ends in vivo . J. Mol. Biol. 194:453–468. 298. Sternberg, N., and J. Coulby. 1987. Recognition and cleavage of the bac- teriophage Pl packaging site ( pac) II.', 'page_label: 36 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  II. Organization of replicon elements. J. Mol. Biol. 207: 111–133. 296. Sternberg, N., and J. Coulby. 1990. Cleavage of the bacteriophage P1 packaging site ( pac) is regulated by adenine methylation. Proc. Natl. Acad. Sci. USA 87:8070 –8074. 297. Sternberg, N., and J. Coulby. 1987. Recognition and cleavage of the bac- teriophage Pl packaging site ( pac) I. Differential processing of the cleaved ends in vivo . J. Mol. Biol. 194:453–468. 298. Sternberg, N., and J. Coulby. 1987. Recognition and cleavage of the bac- teriophage Pl packaging site ( pac) II. Functional limits of pacand location ofpaccleavage termini. J. Mol. Biol. 194:469–480. 299. Sternberg, N., D. Hamilton, S. Austin, M. Yarmolinsky, and R. Hoess. 1981. Site-speci ﬁc recombination and its role in the life cycle of bacteriophage P1. Cold Spring Harbor Symp. Quant. Biol. 45:297–309. 300. Sternberg, N., D. Hamilton, and R. Hoess. 1981. Bacteriophage P1 site- speci ﬁc recombination. II. Recombination between loxP and the bacterial chromosome. J. Mol. Biol. 150:487–507. 301. Sternberg, N., and R. Hoess. 1983. The molecular genetics of bacteriophage P1. Annu. Rev. Genet. 17:123–154. 302. Sternberg, N., B. Sauer, R. Hoess, and K. Abremski. 1986. Bacteriophage Pl crestructural gene and its regulatory region. Evidence for multiple promot- ers and for regulation by DNA methylation. J. Mol. Biol. 187:197–212. 303. Sternberg, N. L., and R. Maurer. 1991. Bacteriophage-mediated general- ized transduction in Escherichia coli andSalmonella typhimurium . Methods Enzymol. 204:18–43. 304. Streiff, M. B., S. Iida, and T. A. Bickle. 1987. Expression and proteolytic processing of the darA anti-restriction gene product of bacteriophage P1. Virology 157:167–171. 305. Studwell-Vaughan, P. S., and M. O ’Donnell. 1993. DNA polymerase IIIVOL. 186, 2004 BACTERIOPHAGE P1 7067', 'page_label: 37 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  accessory proteins. V. Theta encoded by holE . J. Biol. Chem. 268:11785 – 11791. 306. Surtees, J. A., and B. E. Funnell. 2003. Plasmid and chromosome traf ﬁc control: how ParA and ParB drive partition. Curr. Top. Dev. Biol. 56: 145–180. 307. Susskind, M. M., A. Wright, and D. Botstein. 1971. Superinfection exclu- sion by P22 prophage in lysogens of Salmonella typhimurium. II. Genetic evidence for two exclusion systems. Virology 45:638–652. 308. Tamura, K., H. Asahara, H. Himeno, T. Hasegawa, and M. Shimizu. 1991. Identity elements of Escherichia coli tRNA(Ala). J. Mol. Recognit. 4:129– 132. 309. Tang, M., X. Shen, E. G. Frank, M. O ’Donnell, R. Woodgate, and M. F. Goodman. 1999. UmuD /H11032(2)C is an error-prone DNA polymerase, Esche- richia coli pol V. Proc. Natl. Acad. Sci. USA 96:8919 –8924. 310. Touati-Schwartz, D. 1979. A dnaB analog ban, speci ﬁed by bacteriophage P1: genetic and physiological evidence for functional analogy and interac-tions between the two products. Mol. Gen. Genet. 174:173–188. 311. Touati-Schwartz, D. 1979. A new pleiotropic bacteriophage P1 mutation, bof, affecting c1 repression activity, the expression of plasmid incompati- bility and the expression of certain constitutive prophage genes. Mol. Gen.Genet. 174:189–202. 312. Touati-Schwartz, D. 1978. Two replication functions in phage P1: ban,a n analog of dnaB , and bof, involved in the control of replication., p. 683 –691. InI. Molineaux, and M. Kohiyama (ed.), DNA synthesis: present and future, vol. 17. Plenum Press, New York, N.Y. 313. Tuggle, C. K., and J. A. Fuchs. 1986. Regulation of the synthesis of ribo- nucleotide reductase in Escherichia coli. ,p .2 2 7 –232. InC.-I. Brandon, A. Holmgren, J.-O. Hoag, H. Jornvall, and B.-M. Sjoberg (ed.), Thioredoxinand glutaredoxin systems: structure and function. Raven Press, New York,N.Y. 314. Tuma, R., and G. J. Thomas, Jr. 1997. Mechanisms of virus assembly probed by Raman spectroscopy: the icosahedral bacteriophage P22. Bio-phys. Chem. 68:17–31. 315. Velleman, M. 1989. PhD thesis. Freie Universitat Berlin, Berlin, Germany. 316. Velleman, M., B. Dreisekelmann, and H. Schuster. 1987. Multiple repressor binding sites in the genome of bacteriophage Pl. Proc. Natl. Acad. Sci. USA84:5570 –5574. 317. Velleman, M., T. Heinzel, and H. Schuster. 1992. The Bof protein of bacteriophage Pl exerts its modulating function by formation of a ternarycomplex with operator DNA and Cl repressor. J. Biol. Chem. 267:12174 – 12181. 318. Velleman, M., M. Heirich, A. Gu ¨nther, and H. Schuster. 1990. A bacterio- phage P1-encoded modulator protein affects the P1 c1 repression system. J. Mol.', 'page_label: 37 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Velleman, M., B. Dreisekelmann, and H. Schuster. 1987. Multiple repressor binding sites in the genome of bacteriophage Pl. Proc. Natl. Acad. Sci. USA84:5570 –5574. 317. Velleman, M., T. Heinzel, and H. Schuster. 1992. The Bof protein of bacteriophage Pl exerts its modulating function by formation of a ternarycomplex with operator DNA and Cl repressor. J. Biol. Chem. 267:12174 – 12181. 318. Velleman, M., M. Heirich, A. Gu ¨nther, and H. Schuster. 1990. A bacterio- phage P1-encoded modulator protein affects the P1 c1 repression system. J. Mol. Biol. 265:18511 –18517. 319. Velleman, M., and S. Parbus. 1992. Puri ﬁcation of the C1 repressor of bacteriophage P1 by fast protein liquid chromatography. J. Chromatogr.625:41–46. 320. Voegtli, W. C., D. J. White, N. J. Reiter, F. Rusnak, and A. C. Rosenzweig. 2000. Structure of the bacteriophage lambda Ser/Thr protein phosphatasewith sulfate ion bound in two coordination modes. Biochemistry 39:15365 – 15374. 321. Walker, D. H., Jr., and J. T. Walker. 1976. Genetic studies of coliphage P1. III. Extended genetic map. J. Virol. 20:177–187. 322. Walker, D. H. J., and T. F. Anderson. 1970. Morphological variants of coliphage P1. J. Virol. 5:765–782. 323. Walker, J. T., and D. H. J. Walker. 1983. Coliphage P1 morphogenesis: analysis of mutants by electron microscopy. J. Bacteriol. 45:1118 –1139. 324. Walker, J. T., and D. H. J. Walker. 1980. Mutations in coliphage P1 affecting host cell lysis. J. Virol. 35:519–530. 325. Walker, J. T., and D. H. J. Walker. 1981. Structural proteins of coliphage P1. Prog. Clin. Biol. Res. 64:69–77. 326. Wandersman, C., and M. Yarmolinsky. 1977. Bipartite control of immunity conferred by the related heteroimmune plasmid prophages, P1 and P7(formerly phi Amp). Virology 77:386–400. 327. Wang, I. N., D. L. Smith, and R. Young. 2000. Holins: the protein clocks of bacteriophage infections. Annu. Rev. Microbiol. 54:799–825. 328. Wang, L. K., and S. Shuman. 2001. Domain structure and mutational analysis of T4 polynucleotide kinase. J. Biol. Chem. 276:26868 –26874. 329. Watts, N. R., and D. H. Coombs. 1990. Structure of the bacteriophage T4 baseplate as determined by chemical cross-linking. J. Virol. 64:143–154. 330. Weisberg, R. A., and M. E. Gottesman. 1999. Processive antitermination. J. Bacteriol. 181:359–367. 331. Whelan, K. F., E. Colleran, and D. E. Taylor. 1995. Phage inhibition, colicin resistance, and tellurite resistance are encoded by a single cluster of geneson the IncHI2 plasmid R478. J. Bacteriol. 177:5016 –5027.332. Whelan, K. F., R. K. Sherburne, and D. E. Taylor. 1997.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:43 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'494'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993459'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'392ms'), (b'x-request-id', b'14039fe46dc11ece72dafa3f86a9408a'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412492c79595d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:43 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'494'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993459'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'392ms'), (b'x-request-id', b'14039fe46dc11ece72dafa3f86a9408a'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412492c79595d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 37 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1990. Structure of the bacteriophage T4 baseplate as determined by chemical cross-linking. J. Virol. 64:143–154. 330. Weisberg, R. A., and M. E. Gottesman. 1999. Processive antitermination. J. Bacteriol. 181:359–367. 331. Whelan, K. F., E. Colleran, and D. E. Taylor. 1995. Phage inhibition, colicin resistance, and tellurite resistance are encoded by a single cluster of geneson the IncHI2 plasmid R478. J. Bacteriol. 177:5016 –5027.332. Whelan, K. F., R. K. Sherburne, and D. E. Taylor. 1997. Characterization of a region of the IncHI2 plasmid R478 which protects Escherichia coli from toxic effects speci ﬁed by components of the tellurite, phage, and colicin resistance cluster. J. Bacteriol. 179:63–71. 333. Wickner, S., J. Hoskins, D. Chattoraj, and K. McKenney. 1990. Deletion analysis of the mini-P1 plasmid origin of replication and the role of Esch- erichia coli DnaA protein. J. Biol. Chem. 265:11622 –11627. 334. Wickner, S., J. Hoskins, and K. McKenney. 1991. Monomerization of RepA dimers by heat shock proteins activates binding to DNA replicationorigin. Proc Natl Acad Sci USA 88:7903 –7907. 335. Williams, D. R., D. P. Macartney, and C. M. Thomas. 1998. The partition- ing activity of the RK2 central control region requires only incC ,korB and KorB-binding site O(B)3 but other KorB-binding sites form destabilizingcomplexes in the absence of O(B)3. Microbiology 144:3369 –3378. 336. Williams, M. D., J. A. Fuchs, and M. C. Flickinger. 1991. Null mutation in the stringent starvation protein of Escherichia coli disrupts lytic develop- ment of bacteriophage P1. Gene 109:21–30. 337. Windle, B. E., and J. B. Hays. 1986. A phage P1 function that stimulates homologous recombination of the Escherichia coli chromosome. Proc. Natl. Acad. Sci. USA 83:3885 –3889. 338. Windle, B. E., C. S. Laufer, and J. B. Hays. 1988. Sequence and deletion analysis of the recombination enhancement gene ( ref) of bacteriophage P1: evidence for promoter-operator and attenuator-antiterminator control. J. Bac- teriol. 170:4881 –4889. 339. Wood, L. F., N. Y. Tszine, and G. E. Christie. 1997. Activation of P2 late transcription by P2 Ogr protein requires a discrete contact site on the Cterminus of the alpha subunit of Escherichia coli RNA polymerase. J. Mol. Biol. 274:1–7. 340. Xu, G. L., W. Kapfer, J. Walter, and T. A. Trautner. 1992. BsuBI-an isospeci ﬁc restriction and modi ﬁcation system of PstI: characterization of theBsuBI genes and enzymes. Nucleic Acids Res. 20:6517 –6523. 341. Xu, M., D. K. Struck, J. Deaton, I. N. Wang, and R. Young. 2004. A signal-arrest-release sequence mediates export and control of the phage P1endolysin. Proc. Natl. Acad. Sci. USA 101:6415 –6420. 342. Xue, Q., and J. B. Egan. 1995.', 'page_label: 37 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Biol. 274:1–7. 340. Xu, G. L., W. Kapfer, J. Walter, and T. A. Trautner. 1992. BsuBI-an isospeci ﬁc restriction and modi ﬁcation system of PstI: characterization of theBsuBI genes and enzymes. Nucleic Acids Res. 20:6517 –6523. 341. Xu, M., D. K. Struck, J. Deaton, I. N. Wang, and R. Young. 2004. A signal-arrest-release sequence mediates export and control of the phage P1endolysin. Proc. Natl. Acad. Sci. USA 101:6415 –6420. 342. Xue, Q., and J. B. Egan. 1995. DNA sequence of tail ﬁber genes of coli- phage 186 and evidence for a common ancestor shared by dsDNA phageﬁber genes. Virology 212:128–133. 343. Yamaichi, Y., and H. Niki. 2000. Active segregation by the Bacillus subtilis partitioning system in Escherichia coli . Proc. Natl. Acad. Sci. USA 97:14656 – 14661. 343a. Yarmolinsky, M. B. 2004. Bacteriophage P1 in retrospect and in prospect. J. Bacteriol. 186:7032 –7068. 344. Yarmolinsky, M. 1990. Bacteriophage P1, p. 1.52 –1.58. InS. J. O ’Brien (ed.), Genetic maps. Cold Spring Harbor Laboratory Press, Cold SpringHarbor, N.Y. 345. Yarmolinsky, M. 1977. Genetic and physical structure of bacteriophage P1 DNA, p. 721 –732. InI. A. Bukhari, J. A. Shapiro, and S. L. Adhya (ed.), DNA insertion elements, plasmids, and episomes. Cold Spring HarborLaboratory, Cold Spring Harbor, N.Y. 346. Yarmolinsky, M., and M. £obocka. 1993. Bacteriophage Pl, p. 42 –54.InS. J. O’Brien (ed.), Locus maps of complex genomes., vol. 3. Cold Spring Harbor Laboratory, Cold Spring Harbor, N.Y. 347. Yarmolinsky, M., and N. Sternberg. 1988. Bacteriophage Pl, p. 291 –438. InR. Calendar (ed.), The Bacteriophages, vol. 1. Plenum Press, New York, N.Y. 348. Yarmolinsky, M. B. 1995. Programmed cell death in bacterial populations. Science 267:836–837. 349. Yarmolinsky, M. B., E. B. Hansen, S. Jafri, and D. K. Chattoraj. 1989. Participation of the lytic replicon in bacteriophage P1 plasmid maintenance.J. Bacteriol. 171:4785 –4791. 350. Yoo, S. J., J. H. Seol, M. S. Kang, D. B. Ha, and C. H. Chung. 1994. clpX encoding an alternative ATP-binding subunit of protease Ti (Clp) can beexpressed independently from clpP inEscherichia coli . Biochem. Biophys. Res. Commun. 203:798–804. 351. Young, R. 1992. Bacteriophage lysis: mechanism and regulation. Microbiol. Rev. 56:430–481. 352. Zabrovitz, S., N. Segev, and G. Cohen. 1977. Growth of bacteriophage P1 in recombination-de ﬁcient hosts of Escherichia coli . Virology 80:233–248. 353. Zdanovsky, A. G., and M. V. Zdanovskaia. 2000. Simple and ef ﬁcient method for heterologous expression of clostridial proteins.', 'page_label: 37 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Biochem. Biophys. Res. Commun. 203:798–804. 351. Young, R. 1992. Bacteriophage lysis: mechanism and regulation. Microbiol. Rev. 56:430–481. 352. Zabrovitz, S., N. Segev, and G. Cohen. 1977. Growth of bacteriophage P1 in recombination-de ﬁcient hosts of Escherichia coli . Virology 80:233–248. 353. Zdanovsky, A. G., and M. V. Zdanovskaia. 2000. Simple and ef ﬁcient method for heterologous expression of clostridial proteins. Appl. Environ.Microbiol. 66:3166 –3173. 354. Zhang, Z., A. A. Schaffer, W. Miller, T. L. Madden, D. J. Lipman, E. V. Koonin, and S. F. Altschul. 1998. Protein sequence similarity searches using patterns as seeds. Nucleic Acids Res. 26:3986 –3990.7068 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  J. Mol. Biol. (1986) 187, 197-212  Bacteriophage Pl Ere Gene and its Regulatory Region  Evidence for Multiple Promoters and for Regulation by  DNA Methylation  N. Sternberg?, B. Sauer?, R. Hoessf and K. Abremski?  Frederick Cancer Research Facility  Litton Bionetics Incorporated-Basic Research Program  National Cancer Institute  Frederick, MD 21701, U.S.A.  (Received 1 February 1985, and in revised form 1 August 1985)  The bacteriophage Pl site-specific recombination system consists of two components, a site,  ZoxP, at which recombination occurs, and a recombinase protein, Cre. In this paper, we  present the DNA sequence of the cre structural gene and its upstream regulatory region.  Analysis of the sequence indicates: (1) that cre encodes a protein of 343 amino acids;  (2) that cre and ZoxP are separated by a 434 base-pair region that contains a 73 amino acid  open reading frame, orfl; and (3) that cre and orfl are oriented with their amino-terminal  ends proximal to ZoxP. We have identified three promoters that are located upstream of the  cre structural gene. Their activities range from 7 to 10% of the activity of the galactose  operon promoter. The promoter furthest from cre, pR1, contains two Dam methylation sites  (S’-G-A-T-C-3’) in its -35 region, and is sensitive to Dam methylation. Its transcription is  three- to fourfold higher in a dam- host than it is in a dam+ host. The promoter closest to  cre, pR3, signals the production of an RNA transcript that functions inefficiently for Cre  protein synthesis because it lacks a ribosome recognition site. None of the three cre  promoters is sensitive to proteins expressed by the Pl prophage, including the cl repressor  protein.  To assess the role of cre in the Pl life-cycle, we isolated cre mutants and studied their  behavior in recA+ and recA- hosts. Those studies indicate that Cre is dispensable for viral  vegetative growth and lysogeny in a recA+ host, but is required for both processes in a  recA- host. The cre requirement for lysogeny suggests that the protein is essential for the  cyclization of newly injected terminally redundant virion DNA. The requirement for  vegetative growth suggests that Cre also has a role to play in the viral lytic cycle after the  viral DNA has been cyclized.  1. Introduction  Bacteriophage Pl encodes a site-specific  recombination system that consists of a 34-bp$  DNA site, ZoxP, at which recombination occurs and  a recombinase protein, Cre (Sternberg t Hamilton,  1981; Abremski et al., 1983; Hoess & Abremski,  7 Present address: E. I. Du Pont de Nemours and  Company, Central Research & Development  Department, Experimental Station, E328/145,  Wilmington, DE 19898, U.S.A.  $ Abbreviations used: bp, base-pairs; kb, lo3 bases or  base-pairs; we, we gene; Cre, cre protein; neo, the gene  for aminoglycoside 3’ phosphotransferase from  transposon Tn5; Cm’, chloramphenicol resistance;  kan’, kanamycin resistance; amp’, ampicillin resistance;  aup’, nonsense suppressor negative. 1985). In the Pl life-cycle, the lox-Cre system  performs at least two functions: it cyclizes the viral  DNA after its injection into bacteria (Hochman et  al., 1983), and it facilitates the stable segregation of  the unit-copy Pl plasmid prophage (Austin et al.,  1981). In the cyclization reaction, the DNA  substrate is a linear molecule containing ZoxP sites  in the terminally redundant region of the viral  DNA (Sternberg et al., 1981b; Hochman et al.,  1983). Intramolecular recombination between those  sites produces Pl-sized circles ( -90 kb), which lack  the redundancy.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  1985). In the Pl life-cycle, the lox-Cre system  performs at least two functions: it cyclizes the viral  DNA after its injection into bacteria (Hochman et  al., 1983), and it facilitates the stable segregation of  the unit-copy Pl plasmid prophage (Austin et al.,  1981). In the cyclization reaction, the DNA  substrate is a linear molecule containing ZoxP sites  in the terminally redundant region of the viral  DNA (Sternberg et al., 1981b; Hochman et al.,  1983). Intramolecular recombination between those  sites produces Pl-sized circles ( -90 kb), which lack  the redundancy. For the segregation function, the  substrate for lox-Cre recombination is a dimer  plasmid composed of two Pl monomers oriented  head to tail. The dimer is formed by Pl plasmid  replication followed by homologous recombination  between the two daughter molecules. Because the  0022-2836/86/020197-16 $03.00/O 197 0 1986 Academic Press Inc. (London) Ltd.', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  198 N. Sternberg et al.  daughter molecules are physically linked in the  dimer, they cannot be segregated to separate cells  at cell division. Recombination between the two  1oxP sites present in the dimer efficiently  regenerates the daughter monomer circles so that  they can be faithfully segregated.  In this paper, we present the DNA sequence of  the cre structural gene and its upstream regulatory  region. We show also that there are three potential  promoters upstream from cre and that the activity  of one of those promoters is regulated by DNA  methylation. Finally, we demonstrate that cre  function is necessary for both lysogeny and lytic  growth in a host deficient for general  recombination.  2. Materials and Methods  (a) Bacteria and phage strains  The Escherichia coli strains used in this paper are  shown in Table 1. The Pl wild-type phage used here is  Plr-m-Cm cl.106 (Glover et al, 1963; Rosner, 1972). It  contains mutations that inactivate the viral restriction-  modification system and render the viral repressor gene  (cl) temperature-sensitive, and it carries a Tn9 (Cmr)  transposon. cre- derivatives of Pl wild-type are  described below. I-P1 : 7 is a derivative of i vector Dam15  b538 srIk3 nin5 containing Pl EcoRI fragment 7  (Sternberg et al., 1977; Fig. l(a)). Deletion derivatives of  this phage that are ZoxP-cre+ (AlOl) or ZoxP-cre- (A105) or loxP+cre- (A201) have been described (Sternberg &  Hamilton, 1981). Additional deletion mutations (A301 to  A306) are described here. lloxP2 contains 2 loxP sites in  direct orientation flanking the lred and gam genes  (Abremski et al., 1983). 12oxP’neo is the same phage with  the neo (kan’) gene from Tn5 (Jorgensen et al, 1979)  cloned at the single XhoI site in the phage between the  1oxP sites. 1 cre-103 and I cre-103A6 contain the Pl cre  gene cloned into limm434nin5 at the single X7wI site of  that phage, and are described below.  (b) Plasmids  pRH43 (Abremski et al., 1983) is a 4.3 kb pBR322-  derived plasmid containing 2 ZoxP sites in direct  orientation separated from each other on one side by the  bla (amp’) gene and the pBR322 origin of replication and  on the other side by the neo gene from Tn5, pNS2368 is a  9.5 kb plasmid constructed by cloning the adjacent Pl  fragments BamHI-9 and BamHI-EcoRI-7 upstream of  the gaZK gene of plasmid pK04 (Fig. l(a)). The  construction and structure of cre+ plasmids pRHlO0 and  pRH103 and their deletion derivatives are described in  Figs l(b) and 2.  (c) Media and phuge growth  L-broth and L-agar were described by Sternberg &  Hamilton (1981). When supplemented with antibiotics,  the media contained ampicillin (100 pg/ml), chlor-  amphenicol (50 pg/ml) or kanamycin (50 pg/ml).', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  l(a)). The  construction and structure of cre+ plasmids pRHlO0 and  pRH103 and their deletion derivatives are described in  Figs l(b) and 2.  (c) Media and phuge growth  L-broth and L-agar were described by Sternberg &  Hamilton (1981). When supplemented with antibiotics,  the media contained ampicillin (100 pg/ml), chlor-  amphenicol (50 pg/ml) or kanamycin (50 pg/ml). M9  minimal medium and MacConkey-galactose-ampicillin  Table 1  Bacterial strains  Strains Genotype Source or reference  N99 W3102 sup’ galkf str’  NlOO N99 recA  C600 supE  NS616 C600 recA56  YMC stLpF  NS418 YMC recA56  K175 supD  RW842 SUP0  NS1721 RW842 thyA  NS2300  NS2301  NS2302  NS2114  NS2118  NS1616  N1206  KA1333  NS2626 NS1721 recA56 thy+ (limm434-lozP2 neo)  NS616 recA56 (limm434-lozP* neo)  NS418 recA56 (limm434-1osP’ neo)  NS1721 recA56 (limm21-P1:7AlOl)  NS172 reeA56 (Limmtl-P1:76105)  YMC (PlCmr-m-)  HfrH recA56  OR1265 (pRK13)  N99 daml3::Tn9  NS1027  NS1028  NS1029  NS1030  NS1031 N99 thyA  NS1027 (PlCmr-m-c1.100)  NS1027 (PlCmr-m-cl.100 0x2)  NS1027 (PlCmr-m-cl.100 cre3)  NS1028 thyA+ recA56  NS1032 NS1029 thyA+ recA56  NS1033 NS1030 thyA+ recA56 NIH collection  NIH collection  Appleyard (1954)  This work  Dennert & Henning (1968)  This work  West & Scott (1977)  Enquist & Weisberg (1976)  Trimethoprim-resistant mutant of  RW842  Abremski et al. (1983)  This work  This work  we+ A lysogen  we- I lysogen  This work  NIH collection  Abremski & Hoess (1984)  Pl transduction of Tn9 from strain  GM2199 (Marinus et al., 1983) to N99  Trimethoprim-resistant mutant of N99  This work  This work  This work  Mating between NS1028 and N1200  selection for thyA+ and ultraviolet  light sensitivity  Mating between NS1029 and N1200  selection for thyA+ and ultraviolet  light sensitivity  Mating between NS1030 and NS1200,  selection for thyA+ and ultraviolet  light sensitivity', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Pl cre Gene and its Regulatory Signals 199  Pi ECORI-7  c-8*--w-g--77’-  lb)  ECORI  \\\\ BarnHI  Hpal. EcoRl  EcoRl linker  BM  Figure 1. Construction of Pl chimeric plasmids to assay cre gene functions and promoter activity. (m) Pl DNA;  ( -) plasmid vector DNA; (--- ) galactokinase gene DNA; R, EcoRI; HpI, HpaI; C, CZuI; B, BarnHI; P, PvuII;  Bg, BgZII; (m) ZOXP. The PI segment of DNA shown in (a) is fragment EcoRI-7. It contains ZoxP and cre (Sternberg &  Hamilton, 1981). Cre is located between ZozP and the left EcoRI site of the fragment. A partial BamHI-EcoRI digestion  of that fragment was used to clone the adjacent 7* and 9 subfragments together into the guZK promoter vector pK04,  generating pNS2368. (b) The construction of various plasmids used in this paper that contain either all, or part, of cre.  The starting plasmid pRH31 was described by Hoess et al. (1982). It was constructed by first cloning the Pl BumHI-  fragment into the BumHI site of pBR322 with the cre gene closest to the unique EcoRI site of the plasmid. To generate  pRH31, the plasmid was digested with PvuII and religated, deleting the non-cre PuuII-BumHI portion of BumHI-  and the pBR322 sequences between the unique BumHI and PvuII sites.  agar were described by Miller (1972). Lambda lysogens that before phage adsorption the cells were concentrated  were constructed as described by Sternberg & Hamilton in L-broth with 5 mM-CaCl, rather than L-broth with  (1981). Pl single-cycle growth experiments were carried 10 mM-MgCl,. To measure the frequency of Pl lysogeny,  out as described for phage 1 (Sternberg, 1976) except cells were grown in L-broth to 2 x lOs/ml, concentrated', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  200 N. Sternberg et al.  lo-fold in L-broth with 5 mM-C&l,, and then infected  with PlCm phage at a multiplicity of 02 phage/cell.  After adsorption for 10 min at 32°C the cells were spread  onto LlCm-agar plates with 10 mm-sodium citrate and  chloramphenicol and incubated overnight at 30°C to  measure the number of Cm’ colonies. The total cell count  was determined by spreading uninfected cells on L-agar  plates.  (d) Isolation of PI cre mutants  To isolate cre- mutants of phage Pl, we infected the  Pl lysogenic strain NS1616 (Table 1) with phage  limm21-P1:7A201 and isolated 1imm21 lysogens on  EMBO plates containing clear selector phage (Sternberg,  1979). Since the Aimmel-P1:7A201 phage lacks a  functional phage integration system, it cannot produce  stable lysogens unless the phage DNA integrates into  the resident Pl plasmid using the host’s general  recombination system (Austin et al., 1981). The resulting  Pl-l cointegrate is then maintained by the Pl plasmid  replication-partition system. However, as both the Pl  and 1 DNAs contain ZoxP sites, and the Pl contains a  functional cre gene, the cointegrate is dissociated rapidly  by recombination between the 2 ZoxP sites. Once the L  DNA is separated from the Pl plasmid, it will be rapidly  lost from the expanding cell population because it lacks a  functional origin of replication. Thus, I-P1 phages  containing a losP site lysogenize strains with a Pl  prophage much less efficiently than do I-P1 phages  lacking ZoxP (Sternberg & Hamilton, 1981). Among the  rare stable 1 lysogens obtained when a I-ZoxP phage  infects a Pl lysogen, some should contain Pl-1  cointegrate plasmids that do not dissociate rapidly  because their Pl cre gene is inactive. To determine if this  is the case, 6 I-ZoxP, Pl lysogens were isolated and the Pl  phage therein recovered. The lysogenization frequency  (colonies on selection plate/cells infected) was about 10m6.  To recover the Pl from each of the lysogens, we first grew  the cells for 20 generations in L-broth, spread portions on  L-agar plates and, after overnight incubation at 32”C,  screened 100 colonies for loss of the I prophage. When  challenged with clear selector phage, about 2 to 5% of  those colonies could be shown to have lost the 1  prophage, presumably by homologous recombination  between the 1 and Pl sequences of the composite. The  resulting Pl lysogens were then induced by raising the  temperature of the culture from 32 to 42”C, and the  phage produced were assayed for cre function (see below).  Phage from 3 of the lysogens were shown to be cre-.  Those phage are designated Plcrel. Plcre2 and Plcre3.  Phage from t,he remaining 3 lysogens were cre+. Analysis  of the 1 phage from the Plcre+ lysogens indicated that  those phage contained a deletion mutation that had  removed their ZoxP sites (data not shown).  (e) Enzymes  Restriction enzymes, BaZ31, and phage T4 DNA ligase  were purchased from Bethesda Research Laboratories  and New England Biolabs. E. coli DNA polymerase I  Klenow fragment was obtained from Boehringer-  Mannheim. S, nuclease was obtained from Sigma  Chemical Co.  (f) DNA procedures  Restriction enzymes were used under conditions  specified by the vendor. S1 nuclease reactions, use of E. coli DNA polymerase Klenow fragment to convert  single-stranded ends to blunt ends, and DNA ligation  reactions using T4 DNA ligase were performed as  described by Abremski & Hoess (1983).', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Analysis  of the 1 phage from the Plcre+ lysogens indicated that  those phage contained a deletion mutation that had  removed their ZoxP sites (data not shown).  (e) Enzymes  Restriction enzymes, BaZ31, and phage T4 DNA ligase  were purchased from Bethesda Research Laboratories  and New England Biolabs. E. coli DNA polymerase I  Klenow fragment was obtained from Boehringer-  Mannheim. S, nuclease was obtained from Sigma  Chemical Co.  (f) DNA procedures  Restriction enzymes were used under conditions  specified by the vendor. S1 nuclease reactions, use of E. coli DNA polymerase Klenow fragment to convert  single-stranded ends to blunt ends, and DNA ligation  reactions using T4 DNA ligase were performed as  described by Abremski & Hoess (1983). DNA was  sequenced after cloning relevant fragments into Ml3  vectors (Vieira & Messing, 1982) using the chain-  termination method of Sanger & Coulson (1975).  Transformation of E. coli with DNA was according to the  method of Dagert & Ehrlich (1979). Plasmid DNA from  more than 250 ml of cells was prepared by the cleared  lysate method (Clewell & Helinski, 1969) after cells were  grown in chloramphenicol (100 pg/ml) for 12 h to amplify  the plasmid. With volumes of less than 20 ml of cells, the  boiling method of Holmes & Quigley (1981) was used.  (g) Isolation of DNA deletion mutants  (i) I deEetions in vivo  The method used to isolate deletions in phage 1 has  been described (Sternberg et al., 1977, 1979). Briefly, the  rationale for the method is based on the fact that the I  head gene D is dispensable for phage growth if the  packaged DNA (the DNA between phage 1 cohesive-end  sites) is less than 82% of 1 wild-type. Phage whose  size is less than 75% of 1 wild-type fail to grow  under any circumstances (Feiss et al., 1977). The L-PI :7  phage used for this analysis is about 83% of 1 wild-  type @Dam15 5221 imm434 n&5-P1:7). Most of the few  plaques produced when this phage is grown on a lawn of a  sup’ host contain deletions that range in size from ly/” to  8% (0.5 to 4 kb) of 1 wild-type. To analyze those  deletions, we isolated and amplified the phage and  extracted their DNAs as described (Sternberg et al.,  1977). The amount of DNA deleted was assessed by  agarose and polyacrylamide gel electrophoresis of  restriction digests of phage DNA or purified fragments  (Sternberg et aZ., 1981a).  (ii) Ba131 deletions in vitro  Plasmid DNA was digested with the enzyme Ha/31 and  processed as described by Lin et al. (1984). An  oligonucleotide linker (Collaborative Research) was  ligated onto the ends of the deleted DNA to facilitate  subsequent cloning into Ml3 vectors for DNA sequencing.  Deleted DNA was characterized by agarose and poly-  acrylamide gel electrophoresis of restriction digests and  by DNA sequencing.  (h) cre assays in vivo  (i) Method 1  This method involves 2 steps. In the first step, cells  were lysogenized with a 1 phage containing cre. If that  phage is defective for integrative recombination, strain  NS1721 was used for the assay. This strain contains  cryptic 1 sequences in its gaZT gene, which permit  integration of the phage DNA into the host chromosome  by reciprocal homologous recombination. Stable lysogens  of NS1721 were constructed and then made recA- by  mating with strain N1200, followed by selection for cells  that were thyA+ and sensitive to ultraviolet light  (Sternberg t Weisberg, 1977).'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 37 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  1990. Structure of the bacteriophage T4 baseplate as determined by chemical cross-linking. J. Virol. 64:143–154. 330. Weisberg, R. A., and M. E. Gottesman. 1999. Processive antitermination. J. Bacteriol. 181:359–367. 331. Whelan, K. F., E. Colleran, and D. E. Taylor. 1995. Phage inhibition, colicin resistance, and tellurite resistance are encoded by a single cluster of geneson the IncHI2 plasmid R478. J. Bacteriol. 177:5016 –5027.332. Whelan, K. F., R. K. Sherburne, and D. E. Taylor. 1997. Characterization of a region of the IncHI2 plasmid R478 which protects Escherichia coli from toxic effects speci ﬁed by components of the tellurite, phage, and colicin resistance cluster. J. Bacteriol. 179:63–71. 333. Wickner, S., J. Hoskins, D. Chattoraj, and K. McKenney. 1990. Deletion analysis of the mini-P1 plasmid origin of replication and the role of Esch- erichia coli DnaA protein. J. Biol. Chem. 265:11622 –11627. 334. Wickner, S., J. Hoskins, and K. McKenney. 1991. Monomerization of RepA dimers by heat shock proteins activates binding to DNA replicationorigin. Proc Natl Acad Sci USA 88:7903 –7907. 335. Williams, D. R., D. P. Macartney, and C. M. Thomas. 1998. The partition- ing activity of the RK2 central control region requires only incC ,korB and KorB-binding site O(B)3 but other KorB-binding sites form destabilizingcomplexes in the absence of O(B)3. Microbiology 144:3369 –3378. 336. Williams, M. D., J. A. Fuchs, and M. C. Flickinger. 1991. Null mutation in the stringent starvation protein of Escherichia coli disrupts lytic develop- ment of bacteriophage P1. Gene 109:21–30. 337. Windle, B. E., and J. B. Hays. 1986. A phage P1 function that stimulates homologous recombination of the Escherichia coli chromosome. Proc. Natl. Acad. Sci. USA 83:3885 –3889. 338. Windle, B. E., C. S. Laufer, and J. B. Hays. 1988. Sequence and deletion analysis of the recombination enhancement gene ( ref) of bacteriophage P1: evidence for promoter-operator and attenuator-antiterminator control. J. Bac- teriol. 170:4881 –4889. 339. Wood, L. F., N. Y. Tszine, and G. E. Christie. 1997. Activation of P2 late transcription by P2 Ogr protein requires a discrete contact site on the Cterminus of the alpha subunit of Escherichia coli RNA polymerase. J. Mol. Biol. 274:1–7. 340. Xu, G. L., W. Kapfer, J. Walter, and T. A. Trautner. 1992. BsuBI-an isospeci ﬁc restriction and modi ﬁcation system of PstI: characterization of theBsuBI genes and enzymes. Nucleic Acids Res. 20:6517 –6523. 341. Xu, M., D. K. Struck, J. Deaton, I. N. Wang, and R. Young. 2004. A signal-arrest-release sequence mediates export and control of the phage P1endolysin. Proc. Natl. Acad. Sci. USA 101:6415 –6420. 342. Xue, Q., and J. B. Egan. 1995.', 'page_label: 37 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Biol. 274:1–7. 340. Xu, G. L., W. Kapfer, J. Walter, and T. A. Trautner. 1992. BsuBI-an isospeci ﬁc restriction and modi ﬁcation system of PstI: characterization of theBsuBI genes and enzymes. Nucleic Acids Res. 20:6517 –6523. 341. Xu, M., D. K. Struck, J. Deaton, I. N. Wang, and R. Young. 2004. A signal-arrest-release sequence mediates export and control of the phage P1endolysin. Proc. Natl. Acad. Sci. USA 101:6415 –6420. 342. Xue, Q., and J. B. Egan. 1995. DNA sequence of tail ﬁber genes of coli- phage 186 and evidence for a common ancestor shared by dsDNA phageﬁber genes. Virology 212:128–133. 343. Yamaichi, Y., and H. Niki. 2000. Active segregation by the Bacillus subtilis partitioning system in Escherichia coli . Proc. Natl. Acad. Sci. USA 97:14656 – 14661. 343a. Yarmolinsky, M. B. 2004. Bacteriophage P1 in retrospect and in prospect. J. Bacteriol. 186:7032 –7068. 344. Yarmolinsky, M. 1990. Bacteriophage P1, p. 1.52 –1.58. InS. J. O ’Brien (ed.), Genetic maps. Cold Spring Harbor Laboratory Press, Cold SpringHarbor, N.Y. 345. Yarmolinsky, M. 1977. Genetic and physical structure of bacteriophage P1 DNA, p. 721 –732. InI. A. Bukhari, J. A. Shapiro, and S. L. Adhya (ed.), DNA insertion elements, plasmids, and episomes. Cold Spring HarborLaboratory, Cold Spring Harbor, N.Y. 346. Yarmolinsky, M., and M. £obocka. 1993. Bacteriophage Pl, p. 42 –54.InS. J. O’Brien (ed.), Locus maps of complex genomes., vol. 3. Cold Spring Harbor Laboratory, Cold Spring Harbor, N.Y. 347. Yarmolinsky, M., and N. Sternberg. 1988. Bacteriophage Pl, p. 291 –438. InR. Calendar (ed.), The Bacteriophages, vol. 1. Plenum Press, New York, N.Y. 348. Yarmolinsky, M. B. 1995. Programmed cell death in bacterial populations. Science 267:836–837. 349. Yarmolinsky, M. B., E. B. Hansen, S. Jafri, and D. K. Chattoraj. 1989. Participation of the lytic replicon in bacteriophage P1 plasmid maintenance.J. Bacteriol. 171:4785 –4791. 350. Yoo, S. J., J. H. Seol, M. S. Kang, D. B. Ha, and C. H. Chung. 1994. clpX encoding an alternative ATP-binding subunit of protease Ti (Clp) can beexpressed independently from clpP inEscherichia coli . Biochem. Biophys. Res. Commun. 203:798–804. 351. Young, R. 1992. Bacteriophage lysis: mechanism and regulation. Microbiol. Rev. 56:430–481. 352. Zabrovitz, S., N. Segev, and G. Cohen. 1977. Growth of bacteriophage P1 in recombination-de ﬁcient hosts of Escherichia coli . Virology 80:233–248. 353. Zdanovsky, A. G., and M. V. Zdanovskaia. 2000. Simple and ef ﬁcient method for heterologous expression of clostridial proteins.', 'page_label: 37 file_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf  Biochem. Biophys. Res. Commun. 203:798–804. 351. Young, R. 1992. Bacteriophage lysis: mechanism and regulation. Microbiol. Rev. 56:430–481. 352. Zabrovitz, S., N. Segev, and G. Cohen. 1977. Growth of bacteriophage P1 in recombination-de ﬁcient hosts of Escherichia coli . Virology 80:233–248. 353. Zdanovsky, A. G., and M. V. Zdanovskaia. 2000. Simple and ef ﬁcient method for heterologous expression of clostridial proteins. Appl. Environ.Microbiol. 66:3166 –3173. 354. Zhang, Z., A. A. Schaffer, W. Miller, T. L. Madden, D. J. Lipman, E. V. Koonin, and S. F. Altschul. 1998. Protein sequence similarity searches using patterns as seeds. Nucleic Acids Res. 26:3986 –3990.7068 ŁOBOCKA ET AL. J. B ACTERIOL .', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  J. Mol. Biol. (1986) 187, 197-212  Bacteriophage Pl Ere Gene and its Regulatory Region  Evidence for Multiple Promoters and for Regulation by  DNA Methylation  N. Sternberg?, B. Sauer?, R. Hoessf and K. Abremski?  Frederick Cancer Research Facility  Litton Bionetics Incorporated-Basic Research Program  National Cancer Institute  Frederick, MD 21701, U.S.A.  (Received 1 February 1985, and in revised form 1 August 1985)  The bacteriophage Pl site-specific recombination system consists of two components, a site,  ZoxP, at which recombination occurs, and a recombinase protein, Cre. In this paper, we  present the DNA sequence of the cre structural gene and its upstream regulatory region.  Analysis of the sequence indicates: (1) that cre encodes a protein of 343 amino acids;  (2) that cre and ZoxP are separated by a 434 base-pair region that contains a 73 amino acid  open reading frame, orfl; and (3) that cre and orfl are oriented with their amino-terminal  ends proximal to ZoxP. We have identified three promoters that are located upstream of the  cre structural gene. Their activities range from 7 to 10% of the activity of the galactose  operon promoter. The promoter furthest from cre, pR1, contains two Dam methylation sites  (S’-G-A-T-C-3’) in its -35 region, and is sensitive to Dam methylation. Its transcription is  three- to fourfold higher in a dam- host than it is in a dam+ host. The promoter closest to  cre, pR3, signals the production of an RNA transcript that functions inefficiently for Cre  protein synthesis because it lacks a ribosome recognition site. None of the three cre  promoters is sensitive to proteins expressed by the Pl prophage, including the cl repressor  protein.  To assess the role of cre in the Pl life-cycle, we isolated cre mutants and studied their  behavior in recA+ and recA- hosts. Those studies indicate that Cre is dispensable for viral  vegetative growth and lysogeny in a recA+ host, but is required for both processes in a  recA- host. The cre requirement for lysogeny suggests that the protein is essential for the  cyclization of newly injected terminally redundant virion DNA. The requirement for  vegetative growth suggests that Cre also has a role to play in the viral lytic cycle after the  viral DNA has been cyclized.  1. Introduction  Bacteriophage Pl encodes a site-specific  recombination system that consists of a 34-bp$  DNA site, ZoxP, at which recombination occurs and  a recombinase protein, Cre (Sternberg t Hamilton,  1981; Abremski et al., 1983; Hoess & Abremski,  7 Present address: E. I. Du Pont de Nemours and  Company, Central Research & Development  Department, Experimental Station, E328/145,  Wilmington, DE 19898, U.S.A.  $ Abbreviations used: bp, base-pairs; kb, lo3 bases or  base-pairs; we, we gene; Cre, cre protein; neo, the gene  for aminoglycoside 3’ phosphotransferase from  transposon Tn5; Cm’, chloramphenicol resistance;  kan’, kanamycin resistance; amp’, ampicillin resistance;  aup’, nonsense suppressor negative. 1985). In the Pl life-cycle, the lox-Cre system  performs at least two functions: it cyclizes the viral  DNA after its injection into bacteria (Hochman et  al., 1983), and it facilitates the stable segregation of  the unit-copy Pl plasmid prophage (Austin et al.,  1981). In the cyclization reaction, the DNA  substrate is a linear molecule containing ZoxP sites  in the terminally redundant region of the viral  DNA (Sternberg et al., 1981b; Hochman et al.,  1983). Intramolecular recombination between those  sites produces Pl-sized circles ( -90 kb), which lack  the redundancy.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  1985). In the Pl life-cycle, the lox-Cre system  performs at least two functions: it cyclizes the viral  DNA after its injection into bacteria (Hochman et  al., 1983), and it facilitates the stable segregation of  the unit-copy Pl plasmid prophage (Austin et al.,  1981). In the cyclization reaction, the DNA  substrate is a linear molecule containing ZoxP sites  in the terminally redundant region of the viral  DNA (Sternberg et al., 1981b; Hochman et al.,  1983). Intramolecular recombination between those  sites produces Pl-sized circles ( -90 kb), which lack  the redundancy. For the segregation function, the  substrate for lox-Cre recombination is a dimer  plasmid composed of two Pl monomers oriented  head to tail. The dimer is formed by Pl plasmid  replication followed by homologous recombination  between the two daughter molecules. Because the  0022-2836/86/020197-16 $03.00/O 197 0 1986 Academic Press Inc. (London) Ltd.', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  198 N. Sternberg et al.  daughter molecules are physically linked in the  dimer, they cannot be segregated to separate cells  at cell division. Recombination between the two  1oxP sites present in the dimer efficiently  regenerates the daughter monomer circles so that  they can be faithfully segregated.  In this paper, we present the DNA sequence of  the cre structural gene and its upstream regulatory  region. We show also that there are three potential  promoters upstream from cre and that the activity  of one of those promoters is regulated by DNA  methylation. Finally, we demonstrate that cre  function is necessary for both lysogeny and lytic  growth in a host deficient for general  recombination.  2. Materials and Methods  (a) Bacteria and phage strains  The Escherichia coli strains used in this paper are  shown in Table 1. The Pl wild-type phage used here is  Plr-m-Cm cl.106 (Glover et al, 1963; Rosner, 1972). It  contains mutations that inactivate the viral restriction-  modification system and render the viral repressor gene  (cl) temperature-sensitive, and it carries a Tn9 (Cmr)  transposon. cre- derivatives of Pl wild-type are  described below. I-P1 : 7 is a derivative of i vector Dam15  b538 srIk3 nin5 containing Pl EcoRI fragment 7  (Sternberg et al., 1977; Fig. l(a)). Deletion derivatives of  this phage that are ZoxP-cre+ (AlOl) or ZoxP-cre- (A105) or loxP+cre- (A201) have been described (Sternberg &  Hamilton, 1981). Additional deletion mutations (A301 to  A306) are described here. lloxP2 contains 2 loxP sites in  direct orientation flanking the lred and gam genes  (Abremski et al., 1983). 12oxP’neo is the same phage with  the neo (kan’) gene from Tn5 (Jorgensen et al, 1979)  cloned at the single XhoI site in the phage between the  1oxP sites. 1 cre-103 and I cre-103A6 contain the Pl cre  gene cloned into limm434nin5 at the single X7wI site of  that phage, and are described below.  (b) Plasmids  pRH43 (Abremski et al., 1983) is a 4.3 kb pBR322-  derived plasmid containing 2 ZoxP sites in direct  orientation separated from each other on one side by the  bla (amp’) gene and the pBR322 origin of replication and  on the other side by the neo gene from Tn5, pNS2368 is a  9.5 kb plasmid constructed by cloning the adjacent Pl  fragments BamHI-9 and BamHI-EcoRI-7 upstream of  the gaZK gene of plasmid pK04 (Fig. l(a)). The  construction and structure of cre+ plasmids pRHlO0 and  pRH103 and their deletion derivatives are described in  Figs l(b) and 2.  (c) Media and phuge growth  L-broth and L-agar were described by Sternberg &  Hamilton (1981). When supplemented with antibiotics,  the media contained ampicillin (100 pg/ml), chlor-  amphenicol (50 pg/ml) or kanamycin (50 pg/ml).', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  l(a)). The  construction and structure of cre+ plasmids pRHlO0 and  pRH103 and their deletion derivatives are described in  Figs l(b) and 2.  (c) Media and phuge growth  L-broth and L-agar were described by Sternberg &  Hamilton (1981). When supplemented with antibiotics,  the media contained ampicillin (100 pg/ml), chlor-  amphenicol (50 pg/ml) or kanamycin (50 pg/ml). M9  minimal medium and MacConkey-galactose-ampicillin  Table 1  Bacterial strains  Strains Genotype Source or reference  N99 W3102 sup’ galkf str’  NlOO N99 recA  C600 supE  NS616 C600 recA56  YMC stLpF  NS418 YMC recA56  K175 supD  RW842 SUP0  NS1721 RW842 thyA  NS2300  NS2301  NS2302  NS2114  NS2118  NS1616  N1206  KA1333  NS2626 NS1721 recA56 thy+ (limm434-lozP2 neo)  NS616 recA56 (limm434-lozP* neo)  NS418 recA56 (limm434-1osP’ neo)  NS1721 recA56 (limm21-P1:7AlOl)  NS172 reeA56 (Limmtl-P1:76105)  YMC (PlCmr-m-)  HfrH recA56  OR1265 (pRK13)  N99 daml3::Tn9  NS1027  NS1028  NS1029  NS1030  NS1031 N99 thyA  NS1027 (PlCmr-m-c1.100)  NS1027 (PlCmr-m-cl.100 0x2)  NS1027 (PlCmr-m-cl.100 cre3)  NS1028 thyA+ recA56  NS1032 NS1029 thyA+ recA56  NS1033 NS1030 thyA+ recA56 NIH collection  NIH collection  Appleyard (1954)  This work  Dennert & Henning (1968)  This work  West & Scott (1977)  Enquist & Weisberg (1976)  Trimethoprim-resistant mutant of  RW842  Abremski et al. (1983)  This work  This work  we+ A lysogen  we- I lysogen  This work  NIH collection  Abremski & Hoess (1984)  Pl transduction of Tn9 from strain  GM2199 (Marinus et al., 1983) to N99  Trimethoprim-resistant mutant of N99  This work  This work  This work  Mating between NS1028 and N1200  selection for thyA+ and ultraviolet  light sensitivity  Mating between NS1029 and N1200  selection for thyA+ and ultraviolet  light sensitivity  Mating between NS1030 and NS1200,  selection for thyA+ and ultraviolet  light sensitivity', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Pl cre Gene and its Regulatory Signals 199  Pi ECORI-7  c-8*--w-g--77’-  lb)  ECORI  \\\\ BarnHI  Hpal. EcoRl  EcoRl linker  BM  Figure 1. Construction of Pl chimeric plasmids to assay cre gene functions and promoter activity. (m) Pl DNA;  ( -) plasmid vector DNA; (--- ) galactokinase gene DNA; R, EcoRI; HpI, HpaI; C, CZuI; B, BarnHI; P, PvuII;  Bg, BgZII; (m) ZOXP. The PI segment of DNA shown in (a) is fragment EcoRI-7. It contains ZoxP and cre (Sternberg &  Hamilton, 1981). Cre is located between ZozP and the left EcoRI site of the fragment. A partial BamHI-EcoRI digestion  of that fragment was used to clone the adjacent 7* and 9 subfragments together into the guZK promoter vector pK04,  generating pNS2368. (b) The construction of various plasmids used in this paper that contain either all, or part, of cre.  The starting plasmid pRH31 was described by Hoess et al. (1982). It was constructed by first cloning the Pl BumHI-  fragment into the BumHI site of pBR322 with the cre gene closest to the unique EcoRI site of the plasmid. To generate  pRH31, the plasmid was digested with PvuII and religated, deleting the non-cre PuuII-BumHI portion of BumHI-  and the pBR322 sequences between the unique BumHI and PvuII sites.  agar were described by Miller (1972). Lambda lysogens that before phage adsorption the cells were concentrated  were constructed as described by Sternberg & Hamilton in L-broth with 5 mM-CaCl, rather than L-broth with  (1981). Pl single-cycle growth experiments were carried 10 mM-MgCl,. To measure the frequency of Pl lysogeny,  out as described for phage 1 (Sternberg, 1976) except cells were grown in L-broth to 2 x lOs/ml, concentrated', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  200 N. Sternberg et al.  lo-fold in L-broth with 5 mM-C&l,, and then infected  with PlCm phage at a multiplicity of 02 phage/cell.  After adsorption for 10 min at 32°C the cells were spread  onto LlCm-agar plates with 10 mm-sodium citrate and  chloramphenicol and incubated overnight at 30°C to  measure the number of Cm’ colonies. The total cell count  was determined by spreading uninfected cells on L-agar  plates.  (d) Isolation of PI cre mutants  To isolate cre- mutants of phage Pl, we infected the  Pl lysogenic strain NS1616 (Table 1) with phage  limm21-P1:7A201 and isolated 1imm21 lysogens on  EMBO plates containing clear selector phage (Sternberg,  1979). Since the Aimmel-P1:7A201 phage lacks a  functional phage integration system, it cannot produce  stable lysogens unless the phage DNA integrates into  the resident Pl plasmid using the host’s general  recombination system (Austin et al., 1981). The resulting  Pl-l cointegrate is then maintained by the Pl plasmid  replication-partition system. However, as both the Pl  and 1 DNAs contain ZoxP sites, and the Pl contains a  functional cre gene, the cointegrate is dissociated rapidly  by recombination between the 2 ZoxP sites. Once the L  DNA is separated from the Pl plasmid, it will be rapidly  lost from the expanding cell population because it lacks a  functional origin of replication. Thus, I-P1 phages  containing a losP site lysogenize strains with a Pl  prophage much less efficiently than do I-P1 phages  lacking ZoxP (Sternberg & Hamilton, 1981). Among the  rare stable 1 lysogens obtained when a I-ZoxP phage  infects a Pl lysogen, some should contain Pl-1  cointegrate plasmids that do not dissociate rapidly  because their Pl cre gene is inactive. To determine if this  is the case, 6 I-ZoxP, Pl lysogens were isolated and the Pl  phage therein recovered. The lysogenization frequency  (colonies on selection plate/cells infected) was about 10m6.  To recover the Pl from each of the lysogens, we first grew  the cells for 20 generations in L-broth, spread portions on  L-agar plates and, after overnight incubation at 32”C,  screened 100 colonies for loss of the I prophage. When  challenged with clear selector phage, about 2 to 5% of  those colonies could be shown to have lost the 1  prophage, presumably by homologous recombination  between the 1 and Pl sequences of the composite. The  resulting Pl lysogens were then induced by raising the  temperature of the culture from 32 to 42”C, and the  phage produced were assayed for cre function (see below).  Phage from 3 of the lysogens were shown to be cre-.  Those phage are designated Plcrel. Plcre2 and Plcre3.  Phage from t,he remaining 3 lysogens were cre+. Analysis  of the 1 phage from the Plcre+ lysogens indicated that  those phage contained a deletion mutation that had  removed their ZoxP sites (data not shown).  (e) Enzymes  Restriction enzymes, BaZ31, and phage T4 DNA ligase  were purchased from Bethesda Research Laboratories  and New England Biolabs. E. coli DNA polymerase I  Klenow fragment was obtained from Boehringer-  Mannheim. S, nuclease was obtained from Sigma  Chemical Co.  (f) DNA procedures  Restriction enzymes were used under conditions  specified by the vendor. S1 nuclease reactions, use of E. coli DNA polymerase Klenow fragment to convert  single-stranded ends to blunt ends, and DNA ligation  reactions using T4 DNA ligase were performed as  described by Abremski & Hoess (1983).', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Analysis  of the 1 phage from the Plcre+ lysogens indicated that  those phage contained a deletion mutation that had  removed their ZoxP sites (data not shown).  (e) Enzymes  Restriction enzymes, BaZ31, and phage T4 DNA ligase  were purchased from Bethesda Research Laboratories  and New England Biolabs. E. coli DNA polymerase I  Klenow fragment was obtained from Boehringer-  Mannheim. S, nuclease was obtained from Sigma  Chemical Co.  (f) DNA procedures  Restriction enzymes were used under conditions  specified by the vendor. S1 nuclease reactions, use of E. coli DNA polymerase Klenow fragment to convert  single-stranded ends to blunt ends, and DNA ligation  reactions using T4 DNA ligase were performed as  described by Abremski & Hoess (1983). DNA was  sequenced after cloning relevant fragments into Ml3  vectors (Vieira & Messing, 1982) using the chain-  termination method of Sanger & Coulson (1975).  Transformation of E. coli with DNA was according to the  method of Dagert & Ehrlich (1979). Plasmid DNA from  more than 250 ml of cells was prepared by the cleared  lysate method (Clewell & Helinski, 1969) after cells were  grown in chloramphenicol (100 pg/ml) for 12 h to amplify  the plasmid. With volumes of less than 20 ml of cells, the  boiling method of Holmes & Quigley (1981) was used.  (g) Isolation of DNA deletion mutants  (i) I deEetions in vivo  The method used to isolate deletions in phage 1 has  been described (Sternberg et al., 1977, 1979). Briefly, the  rationale for the method is based on the fact that the I  head gene D is dispensable for phage growth if the  packaged DNA (the DNA between phage 1 cohesive-end  sites) is less than 82% of 1 wild-type. Phage whose  size is less than 75% of 1 wild-type fail to grow  under any circumstances (Feiss et al., 1977). The L-PI :7  phage used for this analysis is about 83% of 1 wild-  type @Dam15 5221 imm434 n&5-P1:7). Most of the few  plaques produced when this phage is grown on a lawn of a  sup’ host contain deletions that range in size from ly/” to  8% (0.5 to 4 kb) of 1 wild-type. To analyze those  deletions, we isolated and amplified the phage and  extracted their DNAs as described (Sternberg et al.,  1977). The amount of DNA deleted was assessed by  agarose and polyacrylamide gel electrophoresis of  restriction digests of phage DNA or purified fragments  (Sternberg et aZ., 1981a).  (ii) Ba131 deletions in vitro  Plasmid DNA was digested with the enzyme Ha/31 and  processed as described by Lin et al. (1984). An  oligonucleotide linker (Collaborative Research) was  ligated onto the ends of the deleted DNA to facilitate  subsequent cloning into Ml3 vectors for DNA sequencing.  Deleted DNA was characterized by agarose and poly-  acrylamide gel electrophoresis of restriction digests and  by DNA sequencing.  (h) cre assays in vivo  (i) Method 1  This method involves 2 steps. In the first step, cells  were lysogenized with a 1 phage containing cre. If that  phage is defective for integrative recombination, strain  NS1721 was used for the assay. This strain contains  cryptic 1 sequences in its gaZT gene, which permit  integration of the phage DNA into the host chromosome  by reciprocal homologous recombination. Stable lysogens  of NS1721 were constructed and then made recA- by  mating with strain N1200, followed by selection for cells  that were thyA+ and sensitive to ultraviolet light  (Sternberg t Weisberg, 1977).'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:44 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'76'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993811'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'371ms'), (b'x-request-id', b'bc5f42cf0e1c0bbe13c16f73bd821a61'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124933dfea5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:44 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'76'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993811'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'371ms'), (b'x-request-id', b'bc5f42cf0e1c0bbe13c16f73bd821a61'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124933dfea5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5da0>, 'json_data': {'input': ['page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Deleted DNA was characterized by agarose and poly-  acrylamide gel electrophoresis of restriction digests and  by DNA sequencing.  (h) cre assays in vivo  (i) Method 1  This method involves 2 steps. In the first step, cells  were lysogenized with a 1 phage containing cre. If that  phage is defective for integrative recombination, strain  NS1721 was used for the assay. This strain contains  cryptic 1 sequences in its gaZT gene, which permit  integration of the phage DNA into the host chromosome  by reciprocal homologous recombination. Stable lysogens  of NS1721 were constructed and then made recA- by  mating with strain N1200, followed by selection for cells  that were thyA+ and sensitive to ultraviolet light  (Sternberg t Weisberg, 1977). The latter step was carried  out to prevent excision of the 1 prophage by homologous  recombination during the subsequent cre assay. If the CTY  gene is present in a 1 phage that is proficient for  integrative recombination, lysogens were constructed  directly by infecting recA- cells. The second step in the  cre assay was to transform the recA- (I-cre) lysogens with', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  PI we Gene and its Regulatory Signals 201  plasmid pRH43. If the lysogens contain a functional cre  gene. then recombination between the 2 1oxP sites on this  plasmid should yield 2 circular products, one carrying the  bla (amp’) gene and the pBR322 origin of replication, and  the other the neo (kan’) gene. The latter contains no  origin of replication and is, therefore, rapidly lost from  the cell population. Better than 95% of the cells that can  supply Cre from the resident 1 prophage rapidly become  kans after being transformed to amp’ by pRH43 DNA. In  contrast, less than lo/, of the cre- cells so transformed  becomes kan”.  (ii) iMethod 2  If the cre mutations to be tested are present on a  plasmid, such as pRHlO0 or pRH103 (Fig. l(b)), the  plasmid DNA was transformed into strain NS2300. This  strain contains a 1 prophage with a neo gene flanked by  two ZoxP sites (Abremski et al., 1983). If the transforming  plasmid expresses cre, the lzeo gene will be excised from  the chromosome. and the cell will become sensitive to  kanamycin. In practice. better than 98% of the amp’  transformed colonies become k&n* if the plasmid carries  an intact, cre gene, and less than l”/b becomes kans if the  plasmid is CTY ~.  (i) Galactokinase measurements  (i) Colony assay  To determine whether a particular DNA fragment  contains a promoter, we inserted that fragment upstream  from the gaZK gene in plasmid pKO1, pK04, pK06 or  pKOl1 (McKenney et al., 1981). The resulting DNA was  used to transform strain NlOO and the transformed cells  were then spread on MacConkey-galactose-ampicillin agar  plates. After overnight incubation at 37”C, plates were  scored for red colonies, an indication that a promoter-  containing fragment was inserted upstream from gaZK.  Generally, the pK0 plasmids produce white to light-pink  colonies after incubation for 18 h, plasmids with weak  promoters produce deep red colonies after incubation for  18 h, and plasmids with strong promoters produce small  colonies that usually turn red after about 30 h. Plasmid  pKG1800, which contains the gal operon promoter  upstream from gaZK (McKenney et al., 1981), is an  example of the latter class of plasmids.  (ii) Enzyme assay  Galactokinase activity was measured in extracts of  toluene-treated cells and the units of enzyme activity  calculated as described by McKenney et al. (1981). When  necessary; the units of activity were corrected for the  copy number of the gaZK plasmid. That copy number was  determined by agarose gel electrophoresis of restriction  digests of boiling plasmid preparations derived from  plasmid-bearing cells. The intensity of the plasmid band  following staining of the gel with ethidium bromide was  measured by scanning the gel with a Helena Quick-Scan  apparatus.  (j) Radioactive labeling of proteins in vivo  Bacteria were grown at 28°C in L-broth containing  ampicillin (50pg/ml). At Asso = 0.3, 3-ml of cells were  centrifuged at 5000 revs/min for 10 min and the pellet  resuspended in an equal volume of M9 medium (Miller,  1972) containing 1 mM-MgSO,, 0.2% (w/v) glucose, and  25 pgjml each of histidine, isoleucine and valine. After  incubation at 28°C for an additional 60 min, the culture was induced by shifting it to 42°C and labeled at various  times after induction with [35S]methionine (5 &i/ml;  spec. act. 400 Ci/mmol) for a period of 10 min. Unlabeled  methionine was then added to a final concentration of  106 pg/ml.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  At Asso = 0.3, 3-ml of cells were  centrifuged at 5000 revs/min for 10 min and the pellet  resuspended in an equal volume of M9 medium (Miller,  1972) containing 1 mM-MgSO,, 0.2% (w/v) glucose, and  25 pgjml each of histidine, isoleucine and valine. After  incubation at 28°C for an additional 60 min, the culture was induced by shifting it to 42°C and labeled at various  times after induction with [35S]methionine (5 &i/ml;  spec. act. 400 Ci/mmol) for a period of 10 min. Unlabeled  methionine was then added to a final concentration of  106 pg/ml. The cells were cooled to 4”C, pelleted by  centrifugation (5600 revs/min for 10 min), and then  washed once with 10 mn/r-?L’ris. HCl (pH 8). The final  pellet was resuspended in 100 pl of sample buffer (0.63 M-  Tris. HCI (pH 6.8) 12.5% (v/v) glycerol, 1.25% (w/v)  sodium dodecyl sulfate, O6O25o/o (v/v) bromophenol blue,  and 20 m&f-2-mercaptoethanol). Samples (20 ~1) were  boiled before sodium dodecyl sulfate/polyacrylamide gel  electrophoresis. After electrophoresis, the gels were dried  and analyzed by autoradiography.  3. Results  (a) Physical mapping of the PI cre gene  To localize the Pl ere gene, we isolated a set of  deletion mutations that dissects a 2.5 kb region of  Pl EcoRI-7 DNA (Fig. l(a)) that is known to  contain the gene (Sternberg & Hamilton, 1981). The  location of the deletion boundaries within the  2.5 kb region was determined either by restriction  site mapping or DNA sequencing (Fig. 2). Each of  the deletions was screened for cre activity (see  Materials and Methods, and the legend to Fig. 2).  The results of that screening process localize cre to a  1.1 kb region of DNA flanked on one side by the  right boundary of deletion pRH100611 (-1515)  and on the other side by the left boundary of  deletion pRH103A6 ( - 383: Fig. 2). The minus  number adjacent to each deletion boundary  designates its distance from the center of the loxP  site.  The deletion mutations were isolated in a variety  of different vectors containing cloned Pl fragments.  Mutations A301 to A306 consist of a mutation set  isolated in a 2 phage containing Pl EcoRI-7  (Sternberg & Hamilton, 1981; Fig. l(a)). The left  end of each of those deletions was mapped relative  to known restriction sites in PI DNA (Hoess et al.,  1982; Fig. 2) and the functional state of cre was  measured as described for Method 1 (see Materials  and Methods). A more precise localization of the  ZoxP-proximal end of cre Cas achieved using sets of  BaZ31 -generated deletions of plasmids pRH3 1,  pRH103, and pNS2368 (see the legend to Fig. 2).  The results with pRHlO3 deletions A6 and A100  localize that end of the cre gene between nucleotides  - 383 and -448. The EoxP-distal boundary of cre  was precisely localized using a set of Bal31-  generated deletions of pRHlO0 (Fig. 2). In par-  ticular, the deletions pRHlOO-All and pRHlOO-A4  localize the ZoxP-distal end of the cre gene between  nucleotides - 1515 and - 1335.  (b) The direction of cre transcription  To determine the direction of cre transcription,  we cloned Pl DNA fragments BamHI-9 and  EcoRI-BamHI-8* (Fig.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  2).  The results with pRHlO3 deletions A6 and A100  localize that end of the cre gene between nucleotides  - 383 and -448. The EoxP-distal boundary of cre  was precisely localized using a set of Bal31-  generated deletions of pRHlO0 (Fig. 2). In par-  ticular, the deletions pRHlOO-All and pRHlOO-A4  localize the ZoxP-distal end of the cre gene between  nucleotides - 1515 and - 1335.  (b) The direction of cre transcription  To determine the direction of cre transcription,  we cloned Pl DNA fragments BamHI-9 and  EcoRI-BamHI-8* (Fig. 1) upstream from the  promoterless galactokinase (gaZK) gene in plasmid  pK04 (McKenney et al., 1981). The plasmids', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  202 N. Sternberg et al.  Promoter  activity  (galK) Cre  activity  + +  + +  + +  NT  NT  NT  NT  NT  NT  NT  +  + -  NT  NT  NT  NT +  +  +  +  +  -I-  -  NT  NT  NT  +  +  - 8*--------c-BarnHI-  m-1580  o-1515  35  1258 Deletions  XPl:?  A304  A305  A302  A301  a306  A303  Al01  Ba131 A pRH31  -4ow  -105qggMR  -151m pRH31-39  pRH31-79  pRH31-33  Ba131 A pRH103  pRH103-All  pRH103-A3  pRHl03-A6  pRH103-A100  Ba131 A pNS2368  pNS2368-A4  pNS2368-A15  pNS2368-36  Ba131 A pRHlO0  pRHlOO-31  pRHlOO-All  pRHlOO-s4  pRHlOO-~12  Figure 2. Deletion analysis of Pl DNA containing CM and its upstream regulatory region. The symbols used are the  same as for Fig. 1 with the additions: (m) deleted DNA; HpII, HpaII; N, NruI; M, &&I; S, Sau3al; A. AluI. Pl  sequences to the left of ZoxP are designated with minus numbers and the sequences to the right of ZoxP are designated  with plus numbers. The I-P1 : 7 deletions were generated as described in Materials and Methods. The right boundaries of  the deletions are located within Pl fragment 7*. The left boundaries of the deletions were localized by restriction  enzyme analysis to the region between ZoxP and the unique BarnHI site that separates fragments 9 and 8*. The  uncertainty at those boundaries (designated I---) reflects the spacing between the restriction sites (Hoess et al., 1982).  For example, L-P1:7A304 removes the leftmost HpaII site of BamHI-9 but not the BumHI site, so the deletion must  end between those 2 sites. Cre activity was assayed by Method 1 (see Materials and Methods). The promoter activity of  the mutants was measured by cloning the EcoRI-BumHI fragments containing the deletions into the ga.ZK vector pKO4.  The resulting plasmid constructions were used to transform strain NlOO and the GalK phenotype was assessed by  colony color on MacConkey-galactose agar plates. The results of galactokinase assays are shown in Table 2. The &z/31-  constructed plasmid deletions were generated as described in Materials and Methods. More specifically, pRH31 DNA  was cleaved with PwuII, treated with BaZ31, and the deletion end-point was marked with an XhoI linker. For the  pRH103 deletions, pRH103 DNA was cleaved with XhoI, treated with BaZ31, and the deletion end-points marked  with an XhoI linker. The pNS2368 deletions were generated by cleaving the plasmid DNA with PvuII. treating it with  BaZ31, and then ligating the deletion ends to an EcoRI linker. The DNA was then digested with the enzyme EcoR,T and  ligated. This procedure produces a nested set of deletions with one end at the EcoRI site of pK04 and the other end at a  variable position in the Pl DNA (see Fig. l(a)). The pRHlO0 deletions were generated by cleaving pRHlO0 DNA with  HpuI, treating it with BuZ31, and marking the deletion end-point with an EeoRI linker. The boundaries of each of the  deletions were sequenced by the chain-termination method of Sanger & Coulson (1975) The last nucleotide removed by  the deletion mutant is shown just to the left or right of the deletion bar. The cre activity of the plasmid mutants was  assayed by Method 2 (see Materials and Methods).', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  PI we Gene and its Regulatory Signals 203  Table 2  cre promoter assays  g&K plasmid  pK04  pKG1800  pNS2353  pNS2355  pK04-A301  pKO4-A302  pK04-A306  pK04-A303  pK04-A101  pKOl1  pK04  pKG1800  pK04-236861  pKOl-2365AlOO  pK04-2368A4  pK04-2368A17  ~804.2368A15  pKO4-2368A.6  pKOl1  pKO1 l-103671  pKOll-103877 PI DNA insert  None  None  BamHI-9 (+ 650 --* - 794)  EcoRI-BarnHI-8* (EcoRI -+ -794)  EcoRI(7*) + BarnHI (-794)A301  EcoRI(7*) +BamHI (- 794)A302  EoRI(7*) + BarnHI (- 794)6306  EcoRI(7*) -+ BwzHI (- 794)A303  EcoRI(7*) -t BamHI (-794)AlOl  None  None  None  EcoRI-BarnHI ( + 43 --t - 794)  EcoRI-AZuI ( + 43 + - 243)  EcoRI-BamHI (- 240 --f - 794)  EcoRI-BamHI ( - 314 + - 794)  EcoRI-BanzHI ( - 351+ - 794)  EcoRI-BarnHI ( - 485 + - 794)  None  Xho-BamHI ( - 152 --t - 265)  Xho-BamHI ( - 152 + - 357) Promoter Units of  inserted g&K activity  None 0.12  gal operon 135  $1, pR2, pR3 26.2  None 0.2  None 0.52  None 0.40  pR3, $2 12.1  pR3, pR2 12.5  pR3, pR2, pR1 24.1  None 0.17  None 0.21  gal operon 141  pRl> pR2, pR3 28.4  pR1 12.4  pR2, pR3 18.3  pR3 12.1  pR3 9.3  None 0.75  None 0.24  None 0.21  pR2 14.0  The strain used in these assays is NlOO. The assays were performed with toluene-treated cells. and  the units of activity were calculated as described by McKenney et al. (1981). For each sample, several  dilutions were assayed to ensure that the activity showed a linear response to cell number. The unit  values shown represent the average of 2 duplicate assays, which generally varied no more than loo/,.  The second column in the Table describes the Pl insert containing the presumptive promoter  sequences. Thus, pK04-A306 contains the A306 EcoRI-BumHI (- 794) fragment cloned into pK04  with the BamHI site proximal to g&K, and pKOll-103A77 contains the XhoI-BamHI fragment from  pRH103A77 (- 152 to -357) cloned into pKOl1 with the BarnHI site proximal to g&K. Plasmids  pNS2353 and pNS2355 are described in Results, section (b).  generated in this way are designated pNS2353 and  pNS2355, respectively. In both cases, the BamHI  site that separates fragment 9 from fragment 8*  (Fig. 1) is adjacent to gaZK. This BamHI site is  probably within the cre structural gene, because of  its central location in the 1 kb region containing the  gene (Fig. 2), and because at least 1 kb of DNA  should be necessary to encode the 35,000 ik’, Cre  protein (Abremski & Hoess, 1984).', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Plasmids  pNS2353 and pNS2355 are described in Results, section (b).  generated in this way are designated pNS2353 and  pNS2355, respectively. In both cases, the BamHI  site that separates fragment 9 from fragment 8*  (Fig. 1) is adjacent to gaZK. This BamHI site is  probably within the cre structural gene, because of  its central location in the 1 kb region containing the  gene (Fig. 2), and because at least 1 kb of DNA  should be necessary to encode the 35,000 ik’, Cre  protein (Abremski & Hoess, 1984). This conclusion  is supported by the observation that neither  plasmid pNS2353 nor plasmid pNS2355 is cre+. On  the basis of these considerations, transcription  through the BamHI site should define the direction  of cre transcription. When the two plasmids were  used to transform t’he gaZK- strain NIOO, only the  transformants with pNS2353 DNA expressed any  detectable galactokinase activity, as judged either  by colony color on MacConkey galactose agar plates  (see Materials and Methods) or galactokinase  activit,y in cell extracts (Table 2). We conclude  from t’hese results that crcl is transcribed from right  t,o left’ away from ZorP (Fig. 2). Moreover, on the  basis of the assay in vitro, we can conclude t)hat,  transcription from the cre promoter in plasmid  pNS2353 is about one-fifth as efficient’ as is  transcript,ion from the gal operon promoter in  plasmid pKGl800.  We also cloned the EcoRI-BamHI fragments  containing the A-PI : 7 deletions, AlOl, A301, A302,  A303 and A306 into pK04 and measured galacto- kinase activity produced by the result,ing plasmids.  On the basis of colony color tests. t’he fragments  with deletion mutations that leave cre intact (AlOl,  A303, A306) produced galK+ plasmids, while the  fragments with mutations that inactivate cre (A301,  A302) produced gaEK- plasmids. Galactokinase  assays (Table 2) indicate that pK04-6101 produces  about as much enzyme as pNS2353, while pK04-  A303 and pK04-A306 produce less enzyme.  (c) DNA sequence of the cre gene and its  adjacent control regions  (i) Sequencing strategy  The strategy used to sequence crp and its  upstream promoter region is shown in Figure 3.  First, we used the enzyme Bal31 to generate  deletion mutations from the HpaI and P&I sites  of plasmids pRHlO0 or pNS2368, respectively (see  Figs 1 and 2). An oligonucleotide linker was then  ligated onto the end of each of the mutant DNAs  and the amount of DNA deleted was determined by  restriction site mapping. A set of deletions that  covers the entire region to be sequenced was then  selected and DNA fragments generated from the  inserted linker sites were cloned into Ml3  sequencing vectors (Vieira. & Messing. 19X2). Those  fragments were sequenced by the chain-t,erminatiorl  method of Sanger & Coulson (1975). To complete', \"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  204 N. Sternberg et al.  4-l 2366-3  -2366-6  - 2366-15  -11 d-4 2366-17  -10 - 2366-20  -9 - 2366-4  -a - 31-79  -2 -1  s C s s B N s s S A  s  I I I I I I I I 1 1 I I 1 I I ,  -1600 -1660 -1400 -1300 -1200 -1100 -1000 -900 -660 -700 -600 -500 -400 -300 -290 -100 0 +50  RHlOO-ll- 3-  RHIOO-4 7 4'  RHlOO-12C-------, 5-  RHlOO-91-w 6-  RHIOO-3- 7-  RHlOO-15-  Figure 3. Strategy used to sequence cre and its upstream regulatory region. Appropriate fragments were cloned into  Ml3 vectors MP8 through MPll and sequenced by the chain-termination method. The extent of DNA sequenced is  indicated by the length of the arrows. The 2368, RHlOO and 31-79 sequences initiate at the boundaries of deletions  generated by BuZ31 (see Materials and Methods, and Fig. 2). The arrows designated 1 through 11 represent the sequence  of specific restriction fragments contained within the region.  the sequencing in regions of DNA lacking  appropriate deletion boundaries, we also sequenced  the BamHI-NruI cre fragment and various SauSal-  generated cre fragments (fragments 1 through 11,  Fig. 3). The DNA sequence from the PvuII site  upstream from ZosP (+48) to a position 52 bp  downstream from the cre structural gene (- 1514) is  shown in Figure 4.  (ii) The cre gene  The DNA region shown in Figure 4 contains a  large open reading frame, designated orf2, from  position -434 to position - 1462. Four lines of  evidence strongly support the contention that this  open reading frame encodes the cre structural gene: (1) an amino-terminal analysis of purified Cre  (Abremski & Hoess, 1984) shows complete agree-  ment of the first 20 amino acid residues with those  derived from the DNA sequence of orf2. (2) The  amino acid composition of Cre obtained from an  analysis of the purified protein (Abremski & Hoess,  1984) and from an analysis of the nucleotide  sequence of orf2 are in good agreement (data not  shown). (3) All BaZ31-generated deletion mutations  that delete sequences of orf2 from either the N or  C-terminal ends (Figs 2 and 3) inactivate cre, while  deletions that do not penetrate orf2 retain cre  function. (4) A mutation (cre3) that converts a  glutamine codon to an amber codon at position  -668 of the orj2 reading frame creates a cre amber  Table 3  Suppression and mapping of the Plcredam mutation  Bacterial strains  A. NS2300 (sup”)  NS23OO(su#)  NS2300 (sup')  NS2301 (supE)  NS2301 (supE)  NS2301 (supE)  NS2302 (supF)  NS2302 (aupF)  X52302 (supF)  13. NS2300  NS2300  NS23oO  NS2300  NS23OO Infecting phage  PlCm  PlCm we2  PlCm cre3  PlCm  PlCm we2  PlCm we3  PlCm  PlCm we2  PlCm we3  PlCm cre3am. N99 (pRH31)  Pl Cm cre3am. N99 (pNS2355)  Pl Cm cre3am. N99 (pNS23WA15)  PlCm cre3am. N99 (pNS2368-86)  PlCm cre3am.\", 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  NS2300  NS2300  NS23oO  NS2300  NS23OO Infecting phage  PlCm  PlCm we2  PlCm cre3  PlCm  PlCm we2  PlCm we3  PlCm  PlCm we2  PlCm we3  PlCm cre3am. N99 (pRH31)  Pl Cm cre3am. N99 (pNS2355)  Pl Cm cre3am. N99 (pNS23WA15)  PlCm cre3am. N99 (pNS2368-86)  PlCm cre3am. N99 (pNS2368-A3) Cm’ lysogens that  are kan” ( sb)  98  <l  <l  100  <l  99  100  tl  <I  4  0.1  2  2.2  1.2  cw gene activity in part A was measured as described in Materials and Methods (Method 2). Cells  containing the I.loxPz neo prophage (NS2300, NS2301 and 1352302) were infected with PlCm phage  (multiplicity of infection of 0.2) and portions of the infected cells were spread on LlCm plates and  incubated overnight at 37°C. Colonies were then scored for kan’ by picking 100 Cm’ colonies to Ll-agar  plates with kanamycin. The Plere3 phage lysates used in part B were made by passaging the phage 1  cycle through N99 strains containing 1 of the 5 plasmids shown in the parentheses. These lysates were  then used to infect strain NS2300 to assay cre activity as described in part A. 1000 Cm’ lysogens were  tested for kan’. All of the Pl phages used in this Table carry the cl ,100 mutations.', \"page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  +: i’ TGCGCAGCTG GACGTAAACT CCTCTTCAGA CCTAATAACT TCGTATAGCA TACATTATAC GAAGTTATAT TAAGGGTTAT TGAATATGAT CAATTTACCT  1-w IOXP c___(  -50 -100  GTRAATCUT ACAGTTCAAT ACCTTAGCAG GTCAAATAGT GACCAC~?jl$/f~ AAGGTTGCGC TAC- CTGTGAAAAA TTGGCGGTGT  t pfl- - - -1  -150 -200  ~AGTCCTACA GATTTCGCGT AcCACTTAGC ACCACCAATC AATCAGAGGT ~AAAA ATG GGA TAT TCA ACT GCT AAA GTG TCC ACT CAT CTT  met gly tyr se? thr ala lys val ser thr his leu  ------ Od---..'  -2:o -3yo  GAG CTT GAG AAA AAC CGT GGT TAC TGG CGG GCA AAA GGG ~~FGAT C$T GAT AGT TGC CAR CTG TEA TTA~$G CGC GGT GAA ti~G  ylu leu glu lys asn arg gly tyr trp arg ala lys gly phe asp arg asp ser iys qln leu ser leu ser arg gly glu 91~  I PR2B  -350 -4TO  AAA ATA SAA CGC ACG CGC GGT CGC TAG CGT TTC TAT GAC GAG AAC CAT AAA CAG GTA AAG GCA GAG CCG AK I@-Tnc/r lT4  lys ile 91~ arg thr arg gly arg trp arg phe tyr asp glu asn his lys gln val lys ala glu pro ile leu tyr thr leu  --- -450  CTT AAA AC-$ TGAGTGTTAA ATG TCC AAT TTA CTG Ak GTA CAC CAA AAT TTG CCT GCA TTA CCG GTC GAT GCA ACG AGT  IPU lys thr ile ile met scr asn leu 1~ thr val his gln asn leu pro ala IPU pro va: asp ala thr ser  .\"], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5da0>, 'json_data': {'input': ['page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Deleted DNA was characterized by agarose and poly-  acrylamide gel electrophoresis of restriction digests and  by DNA sequencing.  (h) cre assays in vivo  (i) Method 1  This method involves 2 steps. In the first step, cells  were lysogenized with a 1 phage containing cre. If that  phage is defective for integrative recombination, strain  NS1721 was used for the assay. This strain contains  cryptic 1 sequences in its gaZT gene, which permit  integration of the phage DNA into the host chromosome  by reciprocal homologous recombination. Stable lysogens  of NS1721 were constructed and then made recA- by  mating with strain N1200, followed by selection for cells  that were thyA+ and sensitive to ultraviolet light  (Sternberg t Weisberg, 1977). The latter step was carried  out to prevent excision of the 1 prophage by homologous  recombination during the subsequent cre assay. If the CTY  gene is present in a 1 phage that is proficient for  integrative recombination, lysogens were constructed  directly by infecting recA- cells. The second step in the  cre assay was to transform the recA- (I-cre) lysogens with', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  PI we Gene and its Regulatory Signals 201  plasmid pRH43. If the lysogens contain a functional cre  gene. then recombination between the 2 1oxP sites on this  plasmid should yield 2 circular products, one carrying the  bla (amp’) gene and the pBR322 origin of replication, and  the other the neo (kan’) gene. The latter contains no  origin of replication and is, therefore, rapidly lost from  the cell population. Better than 95% of the cells that can  supply Cre from the resident 1 prophage rapidly become  kans after being transformed to amp’ by pRH43 DNA. In  contrast, less than lo/, of the cre- cells so transformed  becomes kan”.  (ii) iMethod 2  If the cre mutations to be tested are present on a  plasmid, such as pRHlO0 or pRH103 (Fig. l(b)), the  plasmid DNA was transformed into strain NS2300. This  strain contains a 1 prophage with a neo gene flanked by  two ZoxP sites (Abremski et al., 1983). If the transforming  plasmid expresses cre, the lzeo gene will be excised from  the chromosome. and the cell will become sensitive to  kanamycin. In practice. better than 98% of the amp’  transformed colonies become k&n* if the plasmid carries  an intact, cre gene, and less than l”/b becomes kans if the  plasmid is CTY ~.  (i) Galactokinase measurements  (i) Colony assay  To determine whether a particular DNA fragment  contains a promoter, we inserted that fragment upstream  from the gaZK gene in plasmid pKO1, pK04, pK06 or  pKOl1 (McKenney et al., 1981). The resulting DNA was  used to transform strain NlOO and the transformed cells  were then spread on MacConkey-galactose-ampicillin agar  plates. After overnight incubation at 37”C, plates were  scored for red colonies, an indication that a promoter-  containing fragment was inserted upstream from gaZK.  Generally, the pK0 plasmids produce white to light-pink  colonies after incubation for 18 h, plasmids with weak  promoters produce deep red colonies after incubation for  18 h, and plasmids with strong promoters produce small  colonies that usually turn red after about 30 h. Plasmid  pKG1800, which contains the gal operon promoter  upstream from gaZK (McKenney et al., 1981), is an  example of the latter class of plasmids.  (ii) Enzyme assay  Galactokinase activity was measured in extracts of  toluene-treated cells and the units of enzyme activity  calculated as described by McKenney et al. (1981). When  necessary; the units of activity were corrected for the  copy number of the gaZK plasmid. That copy number was  determined by agarose gel electrophoresis of restriction  digests of boiling plasmid preparations derived from  plasmid-bearing cells. The intensity of the plasmid band  following staining of the gel with ethidium bromide was  measured by scanning the gel with a Helena Quick-Scan  apparatus.  (j) Radioactive labeling of proteins in vivo  Bacteria were grown at 28°C in L-broth containing  ampicillin (50pg/ml). At Asso = 0.3, 3-ml of cells were  centrifuged at 5000 revs/min for 10 min and the pellet  resuspended in an equal volume of M9 medium (Miller,  1972) containing 1 mM-MgSO,, 0.2% (w/v) glucose, and  25 pgjml each of histidine, isoleucine and valine. After  incubation at 28°C for an additional 60 min, the culture was induced by shifting it to 42°C and labeled at various  times after induction with [35S]methionine (5 &i/ml;  spec. act. 400 Ci/mmol) for a period of 10 min. Unlabeled  methionine was then added to a final concentration of  106 pg/ml.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  At Asso = 0.3, 3-ml of cells were  centrifuged at 5000 revs/min for 10 min and the pellet  resuspended in an equal volume of M9 medium (Miller,  1972) containing 1 mM-MgSO,, 0.2% (w/v) glucose, and  25 pgjml each of histidine, isoleucine and valine. After  incubation at 28°C for an additional 60 min, the culture was induced by shifting it to 42°C and labeled at various  times after induction with [35S]methionine (5 &i/ml;  spec. act. 400 Ci/mmol) for a period of 10 min. Unlabeled  methionine was then added to a final concentration of  106 pg/ml. The cells were cooled to 4”C, pelleted by  centrifugation (5600 revs/min for 10 min), and then  washed once with 10 mn/r-?L’ris. HCl (pH 8). The final  pellet was resuspended in 100 pl of sample buffer (0.63 M-  Tris. HCI (pH 6.8) 12.5% (v/v) glycerol, 1.25% (w/v)  sodium dodecyl sulfate, O6O25o/o (v/v) bromophenol blue,  and 20 m&f-2-mercaptoethanol). Samples (20 ~1) were  boiled before sodium dodecyl sulfate/polyacrylamide gel  electrophoresis. After electrophoresis, the gels were dried  and analyzed by autoradiography.  3. Results  (a) Physical mapping of the PI cre gene  To localize the Pl ere gene, we isolated a set of  deletion mutations that dissects a 2.5 kb region of  Pl EcoRI-7 DNA (Fig. l(a)) that is known to  contain the gene (Sternberg & Hamilton, 1981). The  location of the deletion boundaries within the  2.5 kb region was determined either by restriction  site mapping or DNA sequencing (Fig. 2). Each of  the deletions was screened for cre activity (see  Materials and Methods, and the legend to Fig. 2).  The results of that screening process localize cre to a  1.1 kb region of DNA flanked on one side by the  right boundary of deletion pRH100611 (-1515)  and on the other side by the left boundary of  deletion pRH103A6 ( - 383: Fig. 2). The minus  number adjacent to each deletion boundary  designates its distance from the center of the loxP  site.  The deletion mutations were isolated in a variety  of different vectors containing cloned Pl fragments.  Mutations A301 to A306 consist of a mutation set  isolated in a 2 phage containing Pl EcoRI-7  (Sternberg & Hamilton, 1981; Fig. l(a)). The left  end of each of those deletions was mapped relative  to known restriction sites in PI DNA (Hoess et al.,  1982; Fig. 2) and the functional state of cre was  measured as described for Method 1 (see Materials  and Methods). A more precise localization of the  ZoxP-proximal end of cre Cas achieved using sets of  BaZ31 -generated deletions of plasmids pRH3 1,  pRH103, and pNS2368 (see the legend to Fig. 2).  The results with pRHlO3 deletions A6 and A100  localize that end of the cre gene between nucleotides  - 383 and -448. The EoxP-distal boundary of cre  was precisely localized using a set of Bal31-  generated deletions of pRHlO0 (Fig. 2). In par-  ticular, the deletions pRHlOO-All and pRHlOO-A4  localize the ZoxP-distal end of the cre gene between  nucleotides - 1515 and - 1335.  (b) The direction of cre transcription  To determine the direction of cre transcription,  we cloned Pl DNA fragments BamHI-9 and  EcoRI-BamHI-8* (Fig.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  2).  The results with pRHlO3 deletions A6 and A100  localize that end of the cre gene between nucleotides  - 383 and -448. The EoxP-distal boundary of cre  was precisely localized using a set of Bal31-  generated deletions of pRHlO0 (Fig. 2). In par-  ticular, the deletions pRHlOO-All and pRHlOO-A4  localize the ZoxP-distal end of the cre gene between  nucleotides - 1515 and - 1335.  (b) The direction of cre transcription  To determine the direction of cre transcription,  we cloned Pl DNA fragments BamHI-9 and  EcoRI-BamHI-8* (Fig. 1) upstream from the  promoterless galactokinase (gaZK) gene in plasmid  pK04 (McKenney et al., 1981). The plasmids', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  202 N. Sternberg et al.  Promoter  activity  (galK) Cre  activity  + +  + +  + +  NT  NT  NT  NT  NT  NT  NT  +  + -  NT  NT  NT  NT +  +  +  +  +  -I-  -  NT  NT  NT  +  +  - 8*--------c-BarnHI-  m-1580  o-1515  35  1258 Deletions  XPl:?  A304  A305  A302  A301  a306  A303  Al01  Ba131 A pRH31  -4ow  -105qggMR  -151m pRH31-39  pRH31-79  pRH31-33  Ba131 A pRH103  pRH103-All  pRH103-A3  pRHl03-A6  pRH103-A100  Ba131 A pNS2368  pNS2368-A4  pNS2368-A15  pNS2368-36  Ba131 A pRHlO0  pRHlOO-31  pRHlOO-All  pRHlOO-s4  pRHlOO-~12  Figure 2. Deletion analysis of Pl DNA containing CM and its upstream regulatory region. The symbols used are the  same as for Fig. 1 with the additions: (m) deleted DNA; HpII, HpaII; N, NruI; M, &&I; S, Sau3al; A. AluI. Pl  sequences to the left of ZoxP are designated with minus numbers and the sequences to the right of ZoxP are designated  with plus numbers. The I-P1 : 7 deletions were generated as described in Materials and Methods. The right boundaries of  the deletions are located within Pl fragment 7*. The left boundaries of the deletions were localized by restriction  enzyme analysis to the region between ZoxP and the unique BarnHI site that separates fragments 9 and 8*. The  uncertainty at those boundaries (designated I---) reflects the spacing between the restriction sites (Hoess et al., 1982).  For example, L-P1:7A304 removes the leftmost HpaII site of BamHI-9 but not the BumHI site, so the deletion must  end between those 2 sites. Cre activity was assayed by Method 1 (see Materials and Methods). The promoter activity of  the mutants was measured by cloning the EcoRI-BumHI fragments containing the deletions into the ga.ZK vector pKO4.  The resulting plasmid constructions were used to transform strain NlOO and the GalK phenotype was assessed by  colony color on MacConkey-galactose agar plates. The results of galactokinase assays are shown in Table 2. The &z/31-  constructed plasmid deletions were generated as described in Materials and Methods. More specifically, pRH31 DNA  was cleaved with PwuII, treated with BaZ31, and the deletion end-point was marked with an XhoI linker. For the  pRH103 deletions, pRH103 DNA was cleaved with XhoI, treated with BaZ31, and the deletion end-points marked  with an XhoI linker. The pNS2368 deletions were generated by cleaving the plasmid DNA with PvuII. treating it with  BaZ31, and then ligating the deletion ends to an EcoRI linker. The DNA was then digested with the enzyme EcoR,T and  ligated. This procedure produces a nested set of deletions with one end at the EcoRI site of pK04 and the other end at a  variable position in the Pl DNA (see Fig. l(a)). The pRHlO0 deletions were generated by cleaving pRHlO0 DNA with  HpuI, treating it with BuZ31, and marking the deletion end-point with an EeoRI linker. The boundaries of each of the  deletions were sequenced by the chain-termination method of Sanger & Coulson (1975) The last nucleotide removed by  the deletion mutant is shown just to the left or right of the deletion bar. The cre activity of the plasmid mutants was  assayed by Method 2 (see Materials and Methods).', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  PI we Gene and its Regulatory Signals 203  Table 2  cre promoter assays  g&K plasmid  pK04  pKG1800  pNS2353  pNS2355  pK04-A301  pKO4-A302  pK04-A306  pK04-A303  pK04-A101  pKOl1  pK04  pKG1800  pK04-236861  pKOl-2365AlOO  pK04-2368A4  pK04-2368A17  ~804.2368A15  pKO4-2368A.6  pKOl1  pKO1 l-103671  pKOll-103877 PI DNA insert  None  None  BamHI-9 (+ 650 --* - 794)  EcoRI-BarnHI-8* (EcoRI -+ -794)  EcoRI(7*) + BarnHI (-794)A301  EcoRI(7*) +BamHI (- 794)A302  EoRI(7*) + BarnHI (- 794)6306  EcoRI(7*) -+ BwzHI (- 794)A303  EcoRI(7*) -t BamHI (-794)AlOl  None  None  None  EcoRI-BarnHI ( + 43 --t - 794)  EcoRI-AZuI ( + 43 + - 243)  EcoRI-BamHI (- 240 --f - 794)  EcoRI-BamHI ( - 314 + - 794)  EcoRI-BanzHI ( - 351+ - 794)  EcoRI-BarnHI ( - 485 + - 794)  None  Xho-BamHI ( - 152 --t - 265)  Xho-BamHI ( - 152 + - 357) Promoter Units of  inserted g&K activity  None 0.12  gal operon 135  $1, pR2, pR3 26.2  None 0.2  None 0.52  None 0.40  pR3, $2 12.1  pR3, pR2 12.5  pR3, pR2, pR1 24.1  None 0.17  None 0.21  gal operon 141  pRl> pR2, pR3 28.4  pR1 12.4  pR2, pR3 18.3  pR3 12.1  pR3 9.3  None 0.75  None 0.24  None 0.21  pR2 14.0  The strain used in these assays is NlOO. The assays were performed with toluene-treated cells. and  the units of activity were calculated as described by McKenney et al. (1981). For each sample, several  dilutions were assayed to ensure that the activity showed a linear response to cell number. The unit  values shown represent the average of 2 duplicate assays, which generally varied no more than loo/,.  The second column in the Table describes the Pl insert containing the presumptive promoter  sequences. Thus, pK04-A306 contains the A306 EcoRI-BumHI (- 794) fragment cloned into pK04  with the BamHI site proximal to g&K, and pKOll-103A77 contains the XhoI-BamHI fragment from  pRH103A77 (- 152 to -357) cloned into pKOl1 with the BarnHI site proximal to g&K. Plasmids  pNS2353 and pNS2355 are described in Results, section (b).  generated in this way are designated pNS2353 and  pNS2355, respectively. In both cases, the BamHI  site that separates fragment 9 from fragment 8*  (Fig. 1) is adjacent to gaZK. This BamHI site is  probably within the cre structural gene, because of  its central location in the 1 kb region containing the  gene (Fig. 2), and because at least 1 kb of DNA  should be necessary to encode the 35,000 ik’, Cre  protein (Abremski & Hoess, 1984).', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Plasmids  pNS2353 and pNS2355 are described in Results, section (b).  generated in this way are designated pNS2353 and  pNS2355, respectively. In both cases, the BamHI  site that separates fragment 9 from fragment 8*  (Fig. 1) is adjacent to gaZK. This BamHI site is  probably within the cre structural gene, because of  its central location in the 1 kb region containing the  gene (Fig. 2), and because at least 1 kb of DNA  should be necessary to encode the 35,000 ik’, Cre  protein (Abremski & Hoess, 1984). This conclusion  is supported by the observation that neither  plasmid pNS2353 nor plasmid pNS2355 is cre+. On  the basis of these considerations, transcription  through the BamHI site should define the direction  of cre transcription. When the two plasmids were  used to transform t’he gaZK- strain NIOO, only the  transformants with pNS2353 DNA expressed any  detectable galactokinase activity, as judged either  by colony color on MacConkey galactose agar plates  (see Materials and Methods) or galactokinase  activit,y in cell extracts (Table 2). We conclude  from t’hese results that crcl is transcribed from right  t,o left’ away from ZorP (Fig. 2). Moreover, on the  basis of the assay in vitro, we can conclude t)hat,  transcription from the cre promoter in plasmid  pNS2353 is about one-fifth as efficient’ as is  transcript,ion from the gal operon promoter in  plasmid pKGl800.  We also cloned the EcoRI-BamHI fragments  containing the A-PI : 7 deletions, AlOl, A301, A302,  A303 and A306 into pK04 and measured galacto- kinase activity produced by the result,ing plasmids.  On the basis of colony color tests. t’he fragments  with deletion mutations that leave cre intact (AlOl,  A303, A306) produced galK+ plasmids, while the  fragments with mutations that inactivate cre (A301,  A302) produced gaEK- plasmids. Galactokinase  assays (Table 2) indicate that pK04-6101 produces  about as much enzyme as pNS2353, while pK04-  A303 and pK04-A306 produce less enzyme.  (c) DNA sequence of the cre gene and its  adjacent control regions  (i) Sequencing strategy  The strategy used to sequence crp and its  upstream promoter region is shown in Figure 3.  First, we used the enzyme Bal31 to generate  deletion mutations from the HpaI and P&I sites  of plasmids pRHlO0 or pNS2368, respectively (see  Figs 1 and 2). An oligonucleotide linker was then  ligated onto the end of each of the mutant DNAs  and the amount of DNA deleted was determined by  restriction site mapping. A set of deletions that  covers the entire region to be sequenced was then  selected and DNA fragments generated from the  inserted linker sites were cloned into Ml3  sequencing vectors (Vieira. & Messing. 19X2). Those  fragments were sequenced by the chain-t,erminatiorl  method of Sanger & Coulson (1975). To complete', \"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  204 N. Sternberg et al.  4-l 2366-3  -2366-6  - 2366-15  -11 d-4 2366-17  -10 - 2366-20  -9 - 2366-4  -a - 31-79  -2 -1  s C s s B N s s S A  s  I I I I I I I I 1 1 I I 1 I I ,  -1600 -1660 -1400 -1300 -1200 -1100 -1000 -900 -660 -700 -600 -500 -400 -300 -290 -100 0 +50  RHlOO-ll- 3-  RHIOO-4 7 4'  RHlOO-12C-------, 5-  RHlOO-91-w 6-  RHIOO-3- 7-  RHlOO-15-  Figure 3. Strategy used to sequence cre and its upstream regulatory region. Appropriate fragments were cloned into  Ml3 vectors MP8 through MPll and sequenced by the chain-termination method. The extent of DNA sequenced is  indicated by the length of the arrows. The 2368, RHlOO and 31-79 sequences initiate at the boundaries of deletions  generated by BuZ31 (see Materials and Methods, and Fig. 2). The arrows designated 1 through 11 represent the sequence  of specific restriction fragments contained within the region.  the sequencing in regions of DNA lacking  appropriate deletion boundaries, we also sequenced  the BamHI-NruI cre fragment and various SauSal-  generated cre fragments (fragments 1 through 11,  Fig. 3). The DNA sequence from the PvuII site  upstream from ZosP (+48) to a position 52 bp  downstream from the cre structural gene (- 1514) is  shown in Figure 4.  (ii) The cre gene  The DNA region shown in Figure 4 contains a  large open reading frame, designated orf2, from  position -434 to position - 1462. Four lines of  evidence strongly support the contention that this  open reading frame encodes the cre structural gene: (1) an amino-terminal analysis of purified Cre  (Abremski & Hoess, 1984) shows complete agree-  ment of the first 20 amino acid residues with those  derived from the DNA sequence of orf2. (2) The  amino acid composition of Cre obtained from an  analysis of the purified protein (Abremski & Hoess,  1984) and from an analysis of the nucleotide  sequence of orf2 are in good agreement (data not  shown). (3) All BaZ31-generated deletion mutations  that delete sequences of orf2 from either the N or  C-terminal ends (Figs 2 and 3) inactivate cre, while  deletions that do not penetrate orf2 retain cre  function. (4) A mutation (cre3) that converts a  glutamine codon to an amber codon at position  -668 of the orj2 reading frame creates a cre amber  Table 3  Suppression and mapping of the Plcredam mutation  Bacterial strains  A. NS2300 (sup”)  NS23OO(su#)  NS2300 (sup')  NS2301 (supE)  NS2301 (supE)  NS2301 (supE)  NS2302 (supF)  NS2302 (aupF)  X52302 (supF)  13. NS2300  NS2300  NS23oO  NS2300  NS23OO Infecting phage  PlCm  PlCm we2  PlCm cre3  PlCm  PlCm we2  PlCm we3  PlCm  PlCm we2  PlCm we3  PlCm cre3am. N99 (pRH31)  Pl Cm cre3am. N99 (pNS2355)  Pl Cm cre3am. N99 (pNS23WA15)  PlCm cre3am. N99 (pNS2368-86)  PlCm cre3am.\", 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  NS2300  NS2300  NS23oO  NS2300  NS23OO Infecting phage  PlCm  PlCm we2  PlCm cre3  PlCm  PlCm we2  PlCm we3  PlCm  PlCm we2  PlCm we3  PlCm cre3am. N99 (pRH31)  Pl Cm cre3am. N99 (pNS2355)  Pl Cm cre3am. N99 (pNS23WA15)  PlCm cre3am. N99 (pNS2368-86)  PlCm cre3am. N99 (pNS2368-A3) Cm’ lysogens that  are kan” ( sb)  98  <l  <l  100  <l  99  100  tl  <I  4  0.1  2  2.2  1.2  cw gene activity in part A was measured as described in Materials and Methods (Method 2). Cells  containing the I.loxPz neo prophage (NS2300, NS2301 and 1352302) were infected with PlCm phage  (multiplicity of infection of 0.2) and portions of the infected cells were spread on LlCm plates and  incubated overnight at 37°C. Colonies were then scored for kan’ by picking 100 Cm’ colonies to Ll-agar  plates with kanamycin. The Plere3 phage lysates used in part B were made by passaging the phage 1  cycle through N99 strains containing 1 of the 5 plasmids shown in the parentheses. These lysates were  then used to infect strain NS2300 to assay cre activity as described in part A. 1000 Cm’ lysogens were  tested for kan’. All of the Pl phages used in this Table carry the cl ,100 mutations.', \"page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  +: i’ TGCGCAGCTG GACGTAAACT CCTCTTCAGA CCTAATAACT TCGTATAGCA TACATTATAC GAAGTTATAT TAAGGGTTAT TGAATATGAT CAATTTACCT  1-w IOXP c___(  -50 -100  GTRAATCUT ACAGTTCAAT ACCTTAGCAG GTCAAATAGT GACCAC~?jl$/f~ AAGGTTGCGC TAC- CTGTGAAAAA TTGGCGGTGT  t pfl- - - -1  -150 -200  ~AGTCCTACA GATTTCGCGT AcCACTTAGC ACCACCAATC AATCAGAGGT ~AAAA ATG GGA TAT TCA ACT GCT AAA GTG TCC ACT CAT CTT  met gly tyr se? thr ala lys val ser thr his leu  ------ Od---..'  -2:o -3yo  GAG CTT GAG AAA AAC CGT GGT TAC TGG CGG GCA AAA GGG ~~FGAT C$T GAT AGT TGC CAR CTG TEA TTA~$G CGC GGT GAA ti~G  ylu leu glu lys asn arg gly tyr trp arg ala lys gly phe asp arg asp ser iys qln leu ser leu ser arg gly glu 91~  I PR2B  -350 -4TO  AAA ATA SAA CGC ACG CGC GGT CGC TAG CGT TTC TAT GAC GAG AAC CAT AAA CAG GTA AAG GCA GAG CCG AK I@-Tnc/r lT4  lys ile 91~ arg thr arg gly arg trp arg phe tyr asp glu asn his lys gln val lys ala glu pro ile leu tyr thr leu  --- -450  CTT AAA AC-$ TGAGTGTTAA ATG TCC AAT TTA CTG Ak GTA CAC CAA AAT TTG CCT GCA TTA CCG GTC GAT GCA ACG AGT  IPU lys thr ile ile met scr asn leu 1~ thr val his gln asn leu pro ala IPU pro va: asp ala thr ser  .\"], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:45 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'85'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994296'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'342ms'), (b'x-request-id', b'cbf954136c9a0a58f98066f6e32b3254'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124938abca5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:45 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'85'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994296'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'342ms'), (b'x-request-id', b'cbf954136c9a0a58f98066f6e32b3254'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124938abca5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6160>, 'json_data': {'input': ['page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  ..- ~-or‘, -~ 4 + - -or12 (cm-  p-pfi-- +  -550  GAT GAG GTT CGC AAG AAC CTG ATG GAC ATG TTC AGG G4T CGC CAG GCG TTT TCT GAC: CAT ACC TGG AAA ATG CTT CTG TCC GTT  asp glu val ar9 lys asn leu met asp met phe ar9 asp arg qln ala phe ser qlu his thr trp lys met leu leu ser val  NW1  -650  TGC CGG TCG TGG GCG GCA TGG TGC AAG TTG AAT AAC CGG AAA TGG TTT CCC GCA GAA CCT G4A GAT Glfm@ TAT CTT CTA  cys arg ser tro ala ala trp cys lys leu asn asn drg lys trp phe pro ala glu pro ylu asp val dry asp tyr leu ‘eu  : (c’r3am) -AT CT1 CAG GCG CGC GGT CTG GCA GTA AAA ACT ATC CA6 CAA CAT TTG GGC CAG CTA AAC ATG CTT CAT CGT CGG ICC GGG CTG  tyr leu gln ala ary gly leu ala val lys thr ale gln gln his leu yly gln leu as\" met leu his arg ary ser yly leu  -750 6amH I  CCA CkA CCA AGT GAC AGC AAT GCT GrT TCA Clti GTT ATG CGG #%%if$A AAA GAA AAC GTT GAT GCC GGT GAA CGT GCA AAA  pro arg pro ser asp ser asn ala val ser leu val met arg arg ile arg lys glu dsn val asp ala gly glu arg ala lyz  CAG GCT CTA GCG TTC GAA CGb ACT GAT TTC GAC CAG GTT CGT TCA CTC ATG GAA AAT AGC GAT CGC TGC CAG GA1 ATA CGT AAT  gln ala leu ala phe glu ar\\'g thr asp phe asp gln val arg ser leb met ylu asn ser asp arg cys gln asp lie dpq as\"  CTG GCA TTT CTG GGG ATT GCT TAT AAC ACC CTG TTA CGT ATA GCC GAA ATT GCC AGG ATC AGG GTT AAA GAT ATC TCA CGT ACT  leu ala phe leu gly ile ala tyr as\" thr leu leu arg ile ala ylu ile ala arg Ile arg val lys asp ile ser ary thr  GAC GGT GGG AGA ATG TTA ATC CAT ATT GGC AGA ACG AAA ACG CTG GTT AGC ACC GCA GGT GTA GAG AAG GCA CTT AGC CTG GGG  asp gly gly arg met leu ile his ile gly arg thr lys thr leu val ser thr ala yly val glu lys ala leu ser leu 9ly  -1150  GTA ACT AAA CTG GTC GAG CGA TGG ATT TCC GTC TCT GGT GTA GCT GAT GAT CCG AAT AAC TAC CTG TTt TGC CGG GTC AGA AAA  val thp lys lw val glu arg trp ile ser val ser gly val ala asp asp pro asn asn tyr 16x1 phe cys arg val arg lys  RAT GGT GTT GCC GCG CCA TCT GCC ACC AGC CAG CTA TCA ACT CGC GCC CTG GAA GGG ATT TTT GAA GCA ACT C@mijTG ATT  asn gly val ala ala pro ser ala thr ser gin leu se?', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  thr arq ala leu glu gly lie phe glu ala thr his arg leu ile  TAC GGC GCT AAG GAT GAC TCT GGT CAG AGA TAC CTG GCC TGG TCT GGA CAC AGT GCC CGT GTC GGA GCC GCG CGA GAT ATG GCC  tyr gly ala lys asp asp ser yly yln drg tyr leu ala trp ser gly his ser ala arg val gly ala ala arg asp met ala  -1350  CGC GCT GGA GTT TCA AtA CCG GAG ATC ATG CAA GCT GGT GGC TGG ACC AAT GTA AAT ATT GTC ATG AAC TAT ATC CGT A4C CTG  arg ala gly val ser ile pro glu ile met gln ala gly gly trp thr as\" val asn ile val met asn tyr ile arg asn lru  GAT AGT GAA ACA GGG GCA ATG GTG CGC CTG CTG GAA GAT GGC GAT TAGCCATTAA CGCGTAAATG ATTGCTATAA TTAGTTGATP  asp SW glL thr gly ala met val arg leu leu glu asp qly asp  .... ~~~ or,2 ,cre, ~ I  TTTAIGGTGA CA  Figure 4. The DNA sequence of cre and its upstream regulatory region (+ 51 t,o - 1514). The Figure shows the 1565  bases that flank ZozP and CTP. The 34 bp lo& site is located between + 17 and - 17 and contains a 13 bp inverttd repeat  (designated by the arrows) separated by an 8 bp region. The 3 presumptive promoter sequences pR1, pR2 and pR3. are  indicated below the DNA sequence, with the boxed-off sequence at the right representing the - 10 region and the  boxed-off sequence at the left representing the -35 region (Rosenberg C Court, 1979; Hawley & McClure. 1963).  Possible ribosome-binding sites (Shine & Dalgarno, 1974; Stormo et al., 1982) just upstream from either orfl or crc are  underlined. The orfl (-205 to -423) and UP (-434 t)o - 1462) open reading frames are shown wit,h their corresponding  amino acaid sequences. Sequence hyphens are omitted for clarity.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  206 N. Sternberg et al.  lo ) Promotsr seq”*ncrs  COnSWlSUS cTTGACAa I - 13--18-TAtAaT  ---35il------ space + t-- -10 -I  PR1 0\\' CTTGATCAT __ 20 - TAAAAT  - 95 -129  b, TTTGATCAA - 12 - TIAAAT  -103 -129  PR2 TTTGATCGT -----W.-CATTAT  -280 -310  PR3 \\') CTGTACACT -II-CATTAT  -397 -422  b) CTGTACACT -13-TTATCT  -397 -424  , b ) Shine and Dalgamo ssquences  Conse\"s\"s GGAGGTG-3-S-ATT  01‘1 AGAGGTG- 4 -ATG  CW TGAGTGT- 3 -ATG  Figure 5. A comparison of the transcription and  translation signals for cre and orfl with the consensus  signals. The consensus promoter and Shine-Dalgarno  sequences are from Hawley & McClure (1983) and Storm0  et al. (1982), respectively. The cre-orfl promoter  sequences are the best fits to the consensus sequence  within the regions shown to contain promoter activity  (see Fig. 4, Table 2). Alternative possible promoters for  either pR1 or pR3 are indicated by a and b. Sequence  hyphens are omitted for clarity.  phenotype (Fig. 4, Table 3A). We conclude from all  of these observations that orf2 is cre. Consistent  with this interpretation is the previously proposed  direction for cre transcription (away from ZoxP), the  large distance between cre and ZoxP (434 bp), the  location of BarnHI ( - 789) and CZaT ( - 1239) sites  within the cre structural gene, and the suppression  pattern of the cre3am mutation. -This mutation is  suppressed by the supE suppressor but not by the  supD or supF suppressors (Table 3A shows the data  for supE and supF). supE inserts the wild-type  amino acid (glutamine) at the cre3 amber codon.  supD (serine) and supF (t)yrosine) insert new amino  acids at this codon, which presumably render the  resulting proteins inactive.  To sequence the cre3am mutation, we first  localized the mutation by passaging Plcre3 phage  through cells containing plasmids with different  segments of Pl DNA. If the segment used contains  the cre3+ allele, some of the input cre- phage  should recombine with the plasmid to produce cre+  phage. The latter are easily scored by their ability  to excise t,he neo gene from the ;1 prophage in strain  NS2300 (Method 2, Materials and Methods). As  shown in Table 3K, a plasmid with PI sequences  from -528 to -794 (pNS236883, Fig. 3)  transferred the cre3+ allele to PI, while one that  contained fragment S* (pNS2355) did not. We then  cloned and subsequently sequenced the SauBal-  Ii’crmHI fragment ( -529 t’o - 794) from Pl cre3am  IINA. The only change from wild-type was the G . (’  to A . T change at l)osition - 668.  (iii) orfl  An analvsis of the DNA sequence between Z&P  and cre indicates t’hat the region between - 205 and  -4424 contains an open reading frame. designated 43 -  26 -  13 -  6-  Figure 6. Protein synthesis following induction of  plasmid pRK13. Plasmid pRKl3 contains a 1439 bp  fragment, (- 151 to - 1580) of Pl DKA containing orfl  and cre cloned downstream of the i pL promoter in  plasmid pRK12 (Abremski & Hoess. 1983). Tn strain  KA1333. that promoter is repressed by a temperaturr-  sensitive 1 cI857 repressor protein. At various times after  temperature induction of the pL promoter.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  T change at l)osition - 668.  (iii) orfl  An analvsis of the DNA sequence between Z&P  and cre indicates t’hat the region between - 205 and  -4424 contains an open reading frame. designated 43 -  26 -  13 -  6-  Figure 6. Protein synthesis following induction of  plasmid pRK13. Plasmid pRKl3 contains a 1439 bp  fragment, (- 151 to - 1580) of Pl DKA containing orfl  and cre cloned downstream of the i pL promoter in  plasmid pRK12 (Abremski & Hoess. 1983). Tn strain  KA1333. that promoter is repressed by a temperaturr-  sensitive 1 cI857 repressor protein. At various times after  temperature induction of the pL promoter. the cells were  labeled for IO-min intervals with [35S]methioninr and  processed as described in Materials and Methods. Samples  of t’he labeled proteins were boiled and then subjected to  electrophoresis in sodium dodecyl sulfate/lO’?, (w/v)  polyacrylamide gels. The gel was dried and analyzed by  autoradiography. Cells containing pRK13 were labeled  just before induction (lane a) or at 40 min (lane b) or  120 min (lane c) after induction at 42°C. Cells containing  plasmid pRK12 were labeled just before induct,ion  (lane d) or at 80 min after induction at 42°C’ (lane e).  Molecular weights ( x 10V3) are indicated at the left.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Pl we Gene and its Regulatory Signale 207  orf 1, which is preceded by a ribosome recognition  site (Figs 4 and 5). To determine whether a protein  of the size of Orfl (73 amino acid residues) can be  synthesized from a plasmid that contains the open  reading frame, we analyzed the proteins produced  after the induction of plasmid pRK13 (Abremski &  Hoess, 1983, 1984). pRKl3 contains Pl sequences  from - 152 to - 1600 cloned downstream from the  bacteriophage 1 pL promoter. That promoter is  repressed in strain KA1333 (Table 1) at 32°C by the  temperature-sensitive ~I857 gene of a resident 1  prophage. When the temperature is raised to 42”C,  the repressor is inactivated, transcription is  initiated at pL, and an RNA containing both orfl  and cre is synthesized. The proteins made before  and after temperature induction were assayed by  [ 35S]methionine-labeling and sodium dodecyl  sulfate/polyacrylamide gel electrophoresis (Fig. 6).  The results indicate that proteins corresponding to  the expected molecular weights of cre (35,000 M,)  and orfl protein (8000 M,) are produced in large  amounts after temperature induction (lanes b  and c). These proteins appear to be made in much  smaller amounts or not at all before induction  (lane a) and are not made at all either before or  after induction of cells containing a plasmid  (pRKl2) that lacks orfl and cre (lanes e and f). We  conclude from these studies that the orfl reading  frame can code for a protein.  In addition to Orfl and Cre, two other proteins  are produced in large amounts after induction  pRKl3. One is an 80,000 M, protein that is also  overproduced after induction of pRKl2 and is  probably an E. coli heat-shock protein. The other  protein is 65,000 M, and is not produced after  induction of pRK12. This protein is too large to be  accommodated by the coding capacity of the Pl  DNA in pRK13. Moreover, its behavior is different  from that of Orfl and Cre, in that its rate of  synthesis does not increase with time at 42°C. The  source of this protein is unclear.  (d) Transcription control signals  To localize the transcription promoter signals in  the region between cre and ZoxP, we cloned various  DNA fragments from that region upstream from a  promoterless galactokinase (g&K) gene in the pK0  series of vectors (McKenney et al., 1981), and then  measured the levels of galactokinase generated by  those plasmids. Three regions with promoter  activity were identified (Fig. 4, Table 2). The  promoters in those regions are designated pR1, pR2  and pR3 (Table 5). pR1 is contained within the  EcoRI-AZuI subfragment ( +43 to -243) of the  EcoRI-BamHI fragment ( + 43 to - 794) present in  pNS2368Al. It is a relatively weak promoter  (pKOl-23686100, Table 2), having less than 10% of  the activity of the gal operon promoter in plasmid  pGK1800 but 40% of the promoter activity of  pNS2368-Al. A presumptive pR1 promoter is  present at position -95 to -129 (Figs 4 and 5),  75 bp upstream from the start of the orf 1 reading frame. pR2 is defined by a fragment of plasmid  pRH103 that retains only Pl sequences between  positions - 152 and -357. That fragment, and one  slightly smaller, were generated by cleaving  pRH103 with BamHI, digesting the DNA with  Bal31, and then ligating BamHI linkers to the ends  of deletions. The resulting XhoI-BamHI fragments  were cloned upstream from galK in plasmid pKOl1  with the BamHI sites proximal to galK.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  A presumptive pR1 promoter is  present at position -95 to -129 (Figs 4 and 5),  75 bp upstream from the start of the orf 1 reading frame. pR2 is defined by a fragment of plasmid  pRH103 that retains only Pl sequences between  positions - 152 and -357. That fragment, and one  slightly smaller, were generated by cleaving  pRH103 with BamHI, digesting the DNA with  Bal31, and then ligating BamHI linkers to the ends  of deletions. The resulting XhoI-BamHI fragments  were cloned upstream from galK in plasmid pKOl1  with the BamHI sites proximal to galK. Plasmid  pKOll-103A71 (-152 to -265) is gaZK- and  plasmid pKOll-103A77 (- 152 to -357) is gaZK+.  The strength of the pR2 promoter is about the same  as that of the pR1 promoter (Table 2). A possible  pR2 promoter sequence is located between positions  -280 and -310 (Figs 4 and 5). The pR3 promoter  is defined by pNS2368 deletions A15 and A6  (Table 2). The former deletion expresses gaZK but  the latter does not; localizing the pR3 promoter  between positions - 351 and - 484 (Figs 4 and 5).  Its activity is about the same as that of pR2 and  pR1. Possible promoter sequences are located  between positions -397 and -424 (Fig. 5). It  should be noted that the activity of plasmids  containing all three promoters (pK04-236861) or  just pR2 and pR3 (pK04-236864) is nearly equal to  the sum of the activities of the individual promoters  (Table 2). The location of pR1 is consistent with its  possible role as a transcription signal for both orfl  and cre. In contrast, pR2 and pR3 could only be  used to transcribe cre, since they are located in the  orfl structural gene (Fig. 4).  To determine whether any of the differences in  enzyme activity among the strains used in Table 2  reflect differences in plasmid copy number, we  prepared DNA from the strains by the boiling  Table 4  The eJfect of Pl prophage repressors on  transcription frm cre promoters  galK plasmid  (promoter)  pKOl-2368A100 (pR1)  pKOll-103A77 (pR2)  pK04-2368Al7 (pR3)  pKO4-EcoRI-14 (~14)  pK04 Resident Units of  prophage galK activity  I-P1 : 7c1+ 12.0  1-Pl:7e1.100 10.2  Plcl+ 13.4  I-P1 :7c1+ 10.6  I-P1 :7cl.l00 13.4  Plcl+ 94  I-P1 :7c1+ 8.8  A-Pl:7cl.100 10.4  Plcl+ 11.0  I-P1 :7c1+ 8.2  1-Pl:7cl.l00 232.0  Plcl+ 10.0  I-P1 :7c1+ 0.4  1-P1:7cl.l00 0.46  PlCl+ Cl.31  The strain used in these assays is NlOO. The ulasmids with ere  promoter are the same as those in Table 2. pi(Ol-EcoRI-14 is  alasmid pKO4 with Pl EcoRI fragment 14 cloned at the EeoRI  site of tie plasmid. EcoRI-14 coitains a promoter, designated  ~14, that can activate the promoterless gaZK gene in pKO4. The  galactokinese assays were performed as described in Table 2.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  208 N. Sternberg et al.  Table 5  The eSfect of Dam methylation on transcription from  cre promoters  g&K plasmid Dam Units of  (promoter) phenotype g&K activity  pKOl-2368A100 (pR1) Dam+ 16.2  Dam- 58.7  pKOll-103A77 (pR2) Dam+ 10.2  Dam- 8.9  pK04-2368817 (pR3) Dam + 11.4  Dam- 9.8  pKGl800 Dam+ 164  Dam- 139  pK04 Dam+ 1.2  Dam- 0.8  The Dam+ strain used is N99 and the Dam- strain is NS2626.  The latter was constructed by transducing the Tn9 insertion  element in the dam gene of strain GM2199 (&m-13) into strain  N99. The plasmids used and the assay method are the same as  those in Table 2.  method of Holmes & Quigley (1981) and then  measured the amount of plasmid DNA in those  preparations by gel electrophoresis (data not  shown). In no case was the variation in plasmid  DNA great enough to warrant a correction factor.  (e) The regulation of cre expression  (i) Repression of cre promoters  One of the questions posed by the presence of  three transcription promoters upstream from cre is  whether cre expression is regulated in any way  through the action of those promoters. A possible  source of that regulation is the cl protein of Pl, the  phage vegetative growth repressor (Scott, 1968;  Rosner, 1972). To determine whether any of the cre  promoters is sensitive to that repressor, we  measured promoter activity (galactokinase  synthesis) in the presence of cl. No effect of the  repressor could be detected (Table 4). The source of  cl in this experiment is a single copy A prophage  containing PI EcoRI-7 (Sternberg, 1979). To rule  out the possibility that the single copy of cl is  insufficient to repress a multicopy promoter, we  measured the expression of gaZK mediated by PI  promoter ~14. This promoter is located on Pl  EcoRI-14 and is sensitive to cl repression (Table 4).  Despite this result, it still could be argued that the  failure to detect repression of the cre promoters by  cl is due t’o the fact that those promoters in high  copy number are less sensitive to the level of cl  protein made from a single copy gene than is ~14.  To rule out this possibility, we inserted either  mutant or wild-type cl genes into the gaEK  plasmids containing the three cre promoters and  measured galactokinase activity from those  plasmids. Again, no effect of the cl repressor could  be detected (data not shown). To determine  whether a PI prophage might encode a protein Table 6  Cre activity assays  Source of Cre amp’ transformants  (cre promoter(s)) that are kan’ (To)  A. pRHlO0 (pR1, pR2, pR3) 92  pRHlO3 (pR2, pR3) 78  pRHl03-A6 (pR3) 72  pBR322 0.5  13. i-PI :7 (pR1, pR2. pR3) 96  Acre-103 (pR2, pR3) 88  Acre-103A6 (pR3) 3  I <0.5  (:re was assayed in part A by Method 2 (Materials and  Methods). Plasmid DNA was transformed into strain NS2300  and amp’ colonies were selected after overnight incubation and  then tested for kanamycin resistance. If the plasmid produces  Cre, then most of the amp’ cells will lose the kan’ marker located  in the I prophage present in NS2300. For each transformation,  200 amp’ colonies were tested for kanamycin sensitivity. Cre was  assayed in part B by Method 1 (Materials and Methods). The  source of Cre in this case is a single copy I prophage and the  1oxP’ substrate is the pRH43 plasmid.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  pR3) 96  Acre-103 (pR2, pR3) 88  Acre-103A6 (pR3) 3  I <0.5  (:re was assayed in part A by Method 2 (Materials and  Methods). Plasmid DNA was transformed into strain NS2300  and amp’ colonies were selected after overnight incubation and  then tested for kanamycin resistance. If the plasmid produces  Cre, then most of the amp’ cells will lose the kan’ marker located  in the I prophage present in NS2300. For each transformation,  200 amp’ colonies were tested for kanamycin sensitivity. Cre was  assayed in part B by Method 1 (Materials and Methods). The  source of Cre in this case is a single copy I prophage and the  1oxP’ substrate is the pRH43 plasmid. That plasmid was  transformed into cells with the I prophages and amp’ colonies  selected and then tested for kanamycin resistance: 200 colonies  were tested in each experiment. The bacterial strain used in this  experiment was NlOO. The I phage was limm434 nin5 and the  I-P1 :7 phage was &am21 b515 b519-PI :7. Phages J.cere-103 and  IZcre-10386 were constructed by cloning the eve gene from  plasmids pRHl03 and pRHl03A6 (Figs 1 and 2), respectively,  into limm434 nin5. This was accomplished by cleaving the  plasmids with EcoRI, converting the EcoRI sites to Sal1 sites,  and then cloning the cre-containing XhoI-Sal1 fragment into the  single XhoI site of the I phage. The cre gene in all of the phages  wax oriented with its N-terminal portion proximal to the I N  gene. The method of Gottesman & Yarmolinsky (1968) was used  to ensure that each lysogen assayed contained only a single I  prophage.  other than cl that could regulate expression of one,  or all, of the three cre promoters, we measured  transcription from those promoters in PI lysogens.  No significant effect of the prophage on galK  transcription could be detected (Table 4).  (ii) Dam methylation  The DNA methylase encoded by the dam gene of  E. coli adds methyl groups to the N-6 position of  adenine residues in the sequence 5’-G-A-T-C-3  (Marinus, 1984). Examination of the pR1 and pR2  promoter sequences (Fig. 5) reveals the presence of  Dam methylation sites in the -35 region of both of  these promoters. Since others have shown that’  methylation of Dam sites in promoters can affect  their activity (Kleckner et al., 1984; M. Marinus:  personal communication), we measured cre  promoter activity in dam- and dam’ strains. pR1 is  sensitive to Dam methylation but pR2 and pR3 are  not (Table 5). The activity of pR1 is elevated three-  to fourfold in the dam- strain. This effect is not due  to a change in plasmid copy, as the copy number of  the pRl-containing plasmid was the same in both  the dam+ and dam- strains (data not shown).  (iii) Utilization of cre transcripts  Since the pR3 promoter is located just 11 bp  upstream from the first cre codon, its transcript', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  PI me Gene and its Regulatory Signals 209  Table 7  Lysogenization frequency of Pl cre- mutants  Lysogenization frequency with  Bacterial strains Plc1.106 Plcl.lOOcre2 Plc1.1oocr~3am  C~JOO @WE) 0.14 0.13 0.08  NS616 (supE, recA) 0.080 0@01 0.05  N99 (supO) 0.25 0.22 0.24  Xl00 (sup’, WA) 0.05 0.002 0.002  Lysogeny was measured as described in Materials and Methods. All of the phage contain the Cm’  marker, and the frequency of lysogeny is the number of Cm’ cells/number of phage-infected cells.  should not contain the ribosome recognition site  that precedes cre (Fig. 5). Accordingly, we might  expect that the level of Cre made from that  transcript would be much less than that made from  transcripts that contain the appropriate ribosome  recognition sites, such as the pR1 and pR2  transcripts. Despite this expectation, we could  detect little difference in cre expression between  plasmids containing just pR3 and those containing  one or both of the other cre promoters (Table 6A).  A possible explanation is that Cre is so active that  even the synthesis of a small amount of protein  from each of the many copies present in the strain  bearing the pR3-plasmid is sufficient to obscure  detection of a large translational defect. To increase  the sensitivity of the assay, we reduced the copy  number of the cre gene. First, we cloned the cre  genes from pRHlO3 (which contains pR2 and pR3)  and pRH103-A6 (which contains only pR3) into the  XhoI site of phage 1 (see the legend to Table 6). The  resulting A-cre phages and I-P1 : 7, which contains all three cre promoters, were then used to lysogenize  E. coli. Lysogens with a single prophage were  selected and used to assay cre expression  (Table 6B). The lysogens containing cre genes with  either pR1 or pR2 had much more activity than did  the lysogen containing the cre gene with just pR3.  Since the three promoters exhibit similar tran-  scriptional activities, the pR3 transcript must be  translated much less efficiently than are the pR1  and pR2 transcripts.  (iv) Properties of Pl cre- phage  We isolated cre- mutants of Pl (see Materials  and Methods) to determine what role that gene  might play in the PI life-cycle. Using the mutants,  we previously showed that cre was necessary for  faithful prophage plasmid segregation (Austin et al.,  1981). Here, we describe the effects of cre  inactivation on prophage establishment and lytic  growth. Our results indicate that Cre is essential for  prophage establishment and viral vegetative  20 30 40 50 60  Time after InfectIon (nun)  (b)  Figure 7. Single-cycle phage burst experiments. The experiments in (a) were carried out as described bv Sternberg  (1976). RecA+ strain X99 was grown to a density of 2 x 10’ cells/ml. concentrated IO-fold in L-broth conta:ning 5 mM-  (WI, and infected with phage at a multiplicity of 0.1 phage/cell. After 5 min absorption at 37°C. the cells were diluted  104-fold into L-broth with 5 mM-CaCI, and incubated at 37°C. Portions were removed at designated intervals, treated  wit,h chloroform, and titrated. The infecting phage used are PlCmcl.lOOr-m- (0). P1Cmc1.100r~n~cre2 (0) and  1’1Cmc1.000r-m-cre3 (m). For the experiments in (b), phage growth was initiated by the induction of Plc1.100  Iysogens.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  concentrated IO-fold in L-broth conta:ning 5 mM-  (WI, and infected with phage at a multiplicity of 0.1 phage/cell. After 5 min absorption at 37°C. the cells were diluted  104-fold into L-broth with 5 mM-CaCI, and incubated at 37°C. Portions were removed at designated intervals, treated  wit,h chloroform, and titrated. The infecting phage used are PlCmcl.lOOr-m- (0). P1Cmc1.100r~n~cre2 (0) and  1’1Cmc1.000r-m-cre3 (m). For the experiments in (b), phage growth was initiated by the induction of Plc1.100  Iysogens. The lysogens were grown to a density of 2 x 10s cells/ml at 32°C and induced by diluting a portion of the  csulturr 104-fold into L-broth with 5 mM-CaCl,, prewarmed to 42°C. The culture was grown at this temperature for  15 min and then shifted to 37°C’ for the remainder of the experiment. Portions were removed at designated intervals.  treated wit,h chloroform. and titrated. The lysogens used (Table 1) are the ret!- strains X81031 (0). K’s1032 (a) and  KS1033 (U).  x'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6160>, 'json_data': {'input': ['page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  ..- ~-or‘, -~ 4 + - -or12 (cm-  p-pfi-- +  -550  GAT GAG GTT CGC AAG AAC CTG ATG GAC ATG TTC AGG G4T CGC CAG GCG TTT TCT GAC: CAT ACC TGG AAA ATG CTT CTG TCC GTT  asp glu val ar9 lys asn leu met asp met phe ar9 asp arg qln ala phe ser qlu his thr trp lys met leu leu ser val  NW1  -650  TGC CGG TCG TGG GCG GCA TGG TGC AAG TTG AAT AAC CGG AAA TGG TTT CCC GCA GAA CCT G4A GAT Glfm@ TAT CTT CTA  cys arg ser tro ala ala trp cys lys leu asn asn drg lys trp phe pro ala glu pro ylu asp val dry asp tyr leu ‘eu  : (c’r3am) -AT CT1 CAG GCG CGC GGT CTG GCA GTA AAA ACT ATC CA6 CAA CAT TTG GGC CAG CTA AAC ATG CTT CAT CGT CGG ICC GGG CTG  tyr leu gln ala ary gly leu ala val lys thr ale gln gln his leu yly gln leu as\" met leu his arg ary ser yly leu  -750 6amH I  CCA CkA CCA AGT GAC AGC AAT GCT GrT TCA Clti GTT ATG CGG #%%if$A AAA GAA AAC GTT GAT GCC GGT GAA CGT GCA AAA  pro arg pro ser asp ser asn ala val ser leu val met arg arg ile arg lys glu dsn val asp ala gly glu arg ala lyz  CAG GCT CTA GCG TTC GAA CGb ACT GAT TTC GAC CAG GTT CGT TCA CTC ATG GAA AAT AGC GAT CGC TGC CAG GA1 ATA CGT AAT  gln ala leu ala phe glu ar\\'g thr asp phe asp gln val arg ser leb met ylu asn ser asp arg cys gln asp lie dpq as\"  CTG GCA TTT CTG GGG ATT GCT TAT AAC ACC CTG TTA CGT ATA GCC GAA ATT GCC AGG ATC AGG GTT AAA GAT ATC TCA CGT ACT  leu ala phe leu gly ile ala tyr as\" thr leu leu arg ile ala ylu ile ala arg Ile arg val lys asp ile ser ary thr  GAC GGT GGG AGA ATG TTA ATC CAT ATT GGC AGA ACG AAA ACG CTG GTT AGC ACC GCA GGT GTA GAG AAG GCA CTT AGC CTG GGG  asp gly gly arg met leu ile his ile gly arg thr lys thr leu val ser thr ala yly val glu lys ala leu ser leu 9ly  -1150  GTA ACT AAA CTG GTC GAG CGA TGG ATT TCC GTC TCT GGT GTA GCT GAT GAT CCG AAT AAC TAC CTG TTt TGC CGG GTC AGA AAA  val thp lys lw val glu arg trp ile ser val ser gly val ala asp asp pro asn asn tyr 16x1 phe cys arg val arg lys  RAT GGT GTT GCC GCG CCA TCT GCC ACC AGC CAG CTA TCA ACT CGC GCC CTG GAA GGG ATT TTT GAA GCA ACT C@mijTG ATT  asn gly val ala ala pro ser ala thr ser gin leu se?', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  thr arq ala leu glu gly lie phe glu ala thr his arg leu ile  TAC GGC GCT AAG GAT GAC TCT GGT CAG AGA TAC CTG GCC TGG TCT GGA CAC AGT GCC CGT GTC GGA GCC GCG CGA GAT ATG GCC  tyr gly ala lys asp asp ser yly yln drg tyr leu ala trp ser gly his ser ala arg val gly ala ala arg asp met ala  -1350  CGC GCT GGA GTT TCA AtA CCG GAG ATC ATG CAA GCT GGT GGC TGG ACC AAT GTA AAT ATT GTC ATG AAC TAT ATC CGT A4C CTG  arg ala gly val ser ile pro glu ile met gln ala gly gly trp thr as\" val asn ile val met asn tyr ile arg asn lru  GAT AGT GAA ACA GGG GCA ATG GTG CGC CTG CTG GAA GAT GGC GAT TAGCCATTAA CGCGTAAATG ATTGCTATAA TTAGTTGATP  asp SW glL thr gly ala met val arg leu leu glu asp qly asp  .... ~~~ or,2 ,cre, ~ I  TTTAIGGTGA CA  Figure 4. The DNA sequence of cre and its upstream regulatory region (+ 51 t,o - 1514). The Figure shows the 1565  bases that flank ZozP and CTP. The 34 bp lo& site is located between + 17 and - 17 and contains a 13 bp inverttd repeat  (designated by the arrows) separated by an 8 bp region. The 3 presumptive promoter sequences pR1, pR2 and pR3. are  indicated below the DNA sequence, with the boxed-off sequence at the right representing the - 10 region and the  boxed-off sequence at the left representing the -35 region (Rosenberg C Court, 1979; Hawley & McClure. 1963).  Possible ribosome-binding sites (Shine & Dalgarno, 1974; Stormo et al., 1982) just upstream from either orfl or crc are  underlined. The orfl (-205 to -423) and UP (-434 t)o - 1462) open reading frames are shown wit,h their corresponding  amino acaid sequences. Sequence hyphens are omitted for clarity.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  206 N. Sternberg et al.  lo ) Promotsr seq”*ncrs  COnSWlSUS cTTGACAa I - 13--18-TAtAaT  ---35il------ space + t-- -10 -I  PR1 0\\' CTTGATCAT __ 20 - TAAAAT  - 95 -129  b, TTTGATCAA - 12 - TIAAAT  -103 -129  PR2 TTTGATCGT -----W.-CATTAT  -280 -310  PR3 \\') CTGTACACT -II-CATTAT  -397 -422  b) CTGTACACT -13-TTATCT  -397 -424  , b ) Shine and Dalgamo ssquences  Conse\"s\"s GGAGGTG-3-S-ATT  01‘1 AGAGGTG- 4 -ATG  CW TGAGTGT- 3 -ATG  Figure 5. A comparison of the transcription and  translation signals for cre and orfl with the consensus  signals. The consensus promoter and Shine-Dalgarno  sequences are from Hawley & McClure (1983) and Storm0  et al. (1982), respectively. The cre-orfl promoter  sequences are the best fits to the consensus sequence  within the regions shown to contain promoter activity  (see Fig. 4, Table 2). Alternative possible promoters for  either pR1 or pR3 are indicated by a and b. Sequence  hyphens are omitted for clarity.  phenotype (Fig. 4, Table 3A). We conclude from all  of these observations that orf2 is cre. Consistent  with this interpretation is the previously proposed  direction for cre transcription (away from ZoxP), the  large distance between cre and ZoxP (434 bp), the  location of BarnHI ( - 789) and CZaT ( - 1239) sites  within the cre structural gene, and the suppression  pattern of the cre3am mutation. -This mutation is  suppressed by the supE suppressor but not by the  supD or supF suppressors (Table 3A shows the data  for supE and supF). supE inserts the wild-type  amino acid (glutamine) at the cre3 amber codon.  supD (serine) and supF (t)yrosine) insert new amino  acids at this codon, which presumably render the  resulting proteins inactive.  To sequence the cre3am mutation, we first  localized the mutation by passaging Plcre3 phage  through cells containing plasmids with different  segments of Pl DNA. If the segment used contains  the cre3+ allele, some of the input cre- phage  should recombine with the plasmid to produce cre+  phage. The latter are easily scored by their ability  to excise t,he neo gene from the ;1 prophage in strain  NS2300 (Method 2, Materials and Methods). As  shown in Table 3K, a plasmid with PI sequences  from -528 to -794 (pNS236883, Fig. 3)  transferred the cre3+ allele to PI, while one that  contained fragment S* (pNS2355) did not. We then  cloned and subsequently sequenced the SauBal-  Ii’crmHI fragment ( -529 t’o - 794) from Pl cre3am  IINA. The only change from wild-type was the G . (’  to A . T change at l)osition - 668.  (iii) orfl  An analvsis of the DNA sequence between Z&P  and cre indicates t’hat the region between - 205 and  -4424 contains an open reading frame. designated 43 -  26 -  13 -  6-  Figure 6. Protein synthesis following induction of  plasmid pRK13. Plasmid pRKl3 contains a 1439 bp  fragment, (- 151 to - 1580) of Pl DKA containing orfl  and cre cloned downstream of the i pL promoter in  plasmid pRK12 (Abremski & Hoess. 1983). Tn strain  KA1333. that promoter is repressed by a temperaturr-  sensitive 1 cI857 repressor protein. At various times after  temperature induction of the pL promoter.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  T change at l)osition - 668.  (iii) orfl  An analvsis of the DNA sequence between Z&P  and cre indicates t’hat the region between - 205 and  -4424 contains an open reading frame. designated 43 -  26 -  13 -  6-  Figure 6. Protein synthesis following induction of  plasmid pRK13. Plasmid pRKl3 contains a 1439 bp  fragment, (- 151 to - 1580) of Pl DKA containing orfl  and cre cloned downstream of the i pL promoter in  plasmid pRK12 (Abremski & Hoess. 1983). Tn strain  KA1333. that promoter is repressed by a temperaturr-  sensitive 1 cI857 repressor protein. At various times after  temperature induction of the pL promoter. the cells were  labeled for IO-min intervals with [35S]methioninr and  processed as described in Materials and Methods. Samples  of t’he labeled proteins were boiled and then subjected to  electrophoresis in sodium dodecyl sulfate/lO’?, (w/v)  polyacrylamide gels. The gel was dried and analyzed by  autoradiography. Cells containing pRK13 were labeled  just before induction (lane a) or at 40 min (lane b) or  120 min (lane c) after induction at 42°C. Cells containing  plasmid pRK12 were labeled just before induct,ion  (lane d) or at 80 min after induction at 42°C’ (lane e).  Molecular weights ( x 10V3) are indicated at the left.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Pl we Gene and its Regulatory Signale 207  orf 1, which is preceded by a ribosome recognition  site (Figs 4 and 5). To determine whether a protein  of the size of Orfl (73 amino acid residues) can be  synthesized from a plasmid that contains the open  reading frame, we analyzed the proteins produced  after the induction of plasmid pRK13 (Abremski &  Hoess, 1983, 1984). pRKl3 contains Pl sequences  from - 152 to - 1600 cloned downstream from the  bacteriophage 1 pL promoter. That promoter is  repressed in strain KA1333 (Table 1) at 32°C by the  temperature-sensitive ~I857 gene of a resident 1  prophage. When the temperature is raised to 42”C,  the repressor is inactivated, transcription is  initiated at pL, and an RNA containing both orfl  and cre is synthesized. The proteins made before  and after temperature induction were assayed by  [ 35S]methionine-labeling and sodium dodecyl  sulfate/polyacrylamide gel electrophoresis (Fig. 6).  The results indicate that proteins corresponding to  the expected molecular weights of cre (35,000 M,)  and orfl protein (8000 M,) are produced in large  amounts after temperature induction (lanes b  and c). These proteins appear to be made in much  smaller amounts or not at all before induction  (lane a) and are not made at all either before or  after induction of cells containing a plasmid  (pRKl2) that lacks orfl and cre (lanes e and f). We  conclude from these studies that the orfl reading  frame can code for a protein.  In addition to Orfl and Cre, two other proteins  are produced in large amounts after induction  pRKl3. One is an 80,000 M, protein that is also  overproduced after induction of pRKl2 and is  probably an E. coli heat-shock protein. The other  protein is 65,000 M, and is not produced after  induction of pRK12. This protein is too large to be  accommodated by the coding capacity of the Pl  DNA in pRK13. Moreover, its behavior is different  from that of Orfl and Cre, in that its rate of  synthesis does not increase with time at 42°C. The  source of this protein is unclear.  (d) Transcription control signals  To localize the transcription promoter signals in  the region between cre and ZoxP, we cloned various  DNA fragments from that region upstream from a  promoterless galactokinase (g&K) gene in the pK0  series of vectors (McKenney et al., 1981), and then  measured the levels of galactokinase generated by  those plasmids. Three regions with promoter  activity were identified (Fig. 4, Table 2). The  promoters in those regions are designated pR1, pR2  and pR3 (Table 5). pR1 is contained within the  EcoRI-AZuI subfragment ( +43 to -243) of the  EcoRI-BamHI fragment ( + 43 to - 794) present in  pNS2368Al. It is a relatively weak promoter  (pKOl-23686100, Table 2), having less than 10% of  the activity of the gal operon promoter in plasmid  pGK1800 but 40% of the promoter activity of  pNS2368-Al. A presumptive pR1 promoter is  present at position -95 to -129 (Figs 4 and 5),  75 bp upstream from the start of the orf 1 reading frame. pR2 is defined by a fragment of plasmid  pRH103 that retains only Pl sequences between  positions - 152 and -357. That fragment, and one  slightly smaller, were generated by cleaving  pRH103 with BamHI, digesting the DNA with  Bal31, and then ligating BamHI linkers to the ends  of deletions. The resulting XhoI-BamHI fragments  were cloned upstream from galK in plasmid pKOl1  with the BamHI sites proximal to galK.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  A presumptive pR1 promoter is  present at position -95 to -129 (Figs 4 and 5),  75 bp upstream from the start of the orf 1 reading frame. pR2 is defined by a fragment of plasmid  pRH103 that retains only Pl sequences between  positions - 152 and -357. That fragment, and one  slightly smaller, were generated by cleaving  pRH103 with BamHI, digesting the DNA with  Bal31, and then ligating BamHI linkers to the ends  of deletions. The resulting XhoI-BamHI fragments  were cloned upstream from galK in plasmid pKOl1  with the BamHI sites proximal to galK. Plasmid  pKOll-103A71 (-152 to -265) is gaZK- and  plasmid pKOll-103A77 (- 152 to -357) is gaZK+.  The strength of the pR2 promoter is about the same  as that of the pR1 promoter (Table 2). A possible  pR2 promoter sequence is located between positions  -280 and -310 (Figs 4 and 5). The pR3 promoter  is defined by pNS2368 deletions A15 and A6  (Table 2). The former deletion expresses gaZK but  the latter does not; localizing the pR3 promoter  between positions - 351 and - 484 (Figs 4 and 5).  Its activity is about the same as that of pR2 and  pR1. Possible promoter sequences are located  between positions -397 and -424 (Fig. 5). It  should be noted that the activity of plasmids  containing all three promoters (pK04-236861) or  just pR2 and pR3 (pK04-236864) is nearly equal to  the sum of the activities of the individual promoters  (Table 2). The location of pR1 is consistent with its  possible role as a transcription signal for both orfl  and cre. In contrast, pR2 and pR3 could only be  used to transcribe cre, since they are located in the  orfl structural gene (Fig. 4).  To determine whether any of the differences in  enzyme activity among the strains used in Table 2  reflect differences in plasmid copy number, we  prepared DNA from the strains by the boiling  Table 4  The eJfect of Pl prophage repressors on  transcription frm cre promoters  galK plasmid  (promoter)  pKOl-2368A100 (pR1)  pKOll-103A77 (pR2)  pK04-2368Al7 (pR3)  pKO4-EcoRI-14 (~14)  pK04 Resident Units of  prophage galK activity  I-P1 : 7c1+ 12.0  1-Pl:7e1.100 10.2  Plcl+ 13.4  I-P1 :7c1+ 10.6  I-P1 :7cl.l00 13.4  Plcl+ 94  I-P1 :7c1+ 8.8  A-Pl:7cl.100 10.4  Plcl+ 11.0  I-P1 :7c1+ 8.2  1-Pl:7cl.l00 232.0  Plcl+ 10.0  I-P1 :7c1+ 0.4  1-P1:7cl.l00 0.46  PlCl+ Cl.31  The strain used in these assays is NlOO. The ulasmids with ere  promoter are the same as those in Table 2. pi(Ol-EcoRI-14 is  alasmid pKO4 with Pl EcoRI fragment 14 cloned at the EeoRI  site of tie plasmid. EcoRI-14 coitains a promoter, designated  ~14, that can activate the promoterless gaZK gene in pKO4. The  galactokinese assays were performed as described in Table 2.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  208 N. Sternberg et al.  Table 5  The eSfect of Dam methylation on transcription from  cre promoters  g&K plasmid Dam Units of  (promoter) phenotype g&K activity  pKOl-2368A100 (pR1) Dam+ 16.2  Dam- 58.7  pKOll-103A77 (pR2) Dam+ 10.2  Dam- 8.9  pK04-2368817 (pR3) Dam + 11.4  Dam- 9.8  pKGl800 Dam+ 164  Dam- 139  pK04 Dam+ 1.2  Dam- 0.8  The Dam+ strain used is N99 and the Dam- strain is NS2626.  The latter was constructed by transducing the Tn9 insertion  element in the dam gene of strain GM2199 (&m-13) into strain  N99. The plasmids used and the assay method are the same as  those in Table 2.  method of Holmes & Quigley (1981) and then  measured the amount of plasmid DNA in those  preparations by gel electrophoresis (data not  shown). In no case was the variation in plasmid  DNA great enough to warrant a correction factor.  (e) The regulation of cre expression  (i) Repression of cre promoters  One of the questions posed by the presence of  three transcription promoters upstream from cre is  whether cre expression is regulated in any way  through the action of those promoters. A possible  source of that regulation is the cl protein of Pl, the  phage vegetative growth repressor (Scott, 1968;  Rosner, 1972). To determine whether any of the cre  promoters is sensitive to that repressor, we  measured promoter activity (galactokinase  synthesis) in the presence of cl. No effect of the  repressor could be detected (Table 4). The source of  cl in this experiment is a single copy A prophage  containing PI EcoRI-7 (Sternberg, 1979). To rule  out the possibility that the single copy of cl is  insufficient to repress a multicopy promoter, we  measured the expression of gaZK mediated by PI  promoter ~14. This promoter is located on Pl  EcoRI-14 and is sensitive to cl repression (Table 4).  Despite this result, it still could be argued that the  failure to detect repression of the cre promoters by  cl is due t’o the fact that those promoters in high  copy number are less sensitive to the level of cl  protein made from a single copy gene than is ~14.  To rule out this possibility, we inserted either  mutant or wild-type cl genes into the gaEK  plasmids containing the three cre promoters and  measured galactokinase activity from those  plasmids. Again, no effect of the cl repressor could  be detected (data not shown). To determine  whether a PI prophage might encode a protein Table 6  Cre activity assays  Source of Cre amp’ transformants  (cre promoter(s)) that are kan’ (To)  A. pRHlO0 (pR1, pR2, pR3) 92  pRHlO3 (pR2, pR3) 78  pRHl03-A6 (pR3) 72  pBR322 0.5  13. i-PI :7 (pR1, pR2. pR3) 96  Acre-103 (pR2, pR3) 88  Acre-103A6 (pR3) 3  I <0.5  (:re was assayed in part A by Method 2 (Materials and  Methods). Plasmid DNA was transformed into strain NS2300  and amp’ colonies were selected after overnight incubation and  then tested for kanamycin resistance. If the plasmid produces  Cre, then most of the amp’ cells will lose the kan’ marker located  in the I prophage present in NS2300. For each transformation,  200 amp’ colonies were tested for kanamycin sensitivity. Cre was  assayed in part B by Method 1 (Materials and Methods). The  source of Cre in this case is a single copy I prophage and the  1oxP’ substrate is the pRH43 plasmid.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  pR3) 96  Acre-103 (pR2, pR3) 88  Acre-103A6 (pR3) 3  I <0.5  (:re was assayed in part A by Method 2 (Materials and  Methods). Plasmid DNA was transformed into strain NS2300  and amp’ colonies were selected after overnight incubation and  then tested for kanamycin resistance. If the plasmid produces  Cre, then most of the amp’ cells will lose the kan’ marker located  in the I prophage present in NS2300. For each transformation,  200 amp’ colonies were tested for kanamycin sensitivity. Cre was  assayed in part B by Method 1 (Materials and Methods). The  source of Cre in this case is a single copy I prophage and the  1oxP’ substrate is the pRH43 plasmid. That plasmid was  transformed into cells with the I prophages and amp’ colonies  selected and then tested for kanamycin resistance: 200 colonies  were tested in each experiment. The bacterial strain used in this  experiment was NlOO. The I phage was limm434 nin5 and the  I-P1 :7 phage was &am21 b515 b519-PI :7. Phages J.cere-103 and  IZcre-10386 were constructed by cloning the eve gene from  plasmids pRHl03 and pRHl03A6 (Figs 1 and 2), respectively,  into limm434 nin5. This was accomplished by cleaving the  plasmids with EcoRI, converting the EcoRI sites to Sal1 sites,  and then cloning the cre-containing XhoI-Sal1 fragment into the  single XhoI site of the I phage. The cre gene in all of the phages  wax oriented with its N-terminal portion proximal to the I N  gene. The method of Gottesman & Yarmolinsky (1968) was used  to ensure that each lysogen assayed contained only a single I  prophage.  other than cl that could regulate expression of one,  or all, of the three cre promoters, we measured  transcription from those promoters in PI lysogens.  No significant effect of the prophage on galK  transcription could be detected (Table 4).  (ii) Dam methylation  The DNA methylase encoded by the dam gene of  E. coli adds methyl groups to the N-6 position of  adenine residues in the sequence 5’-G-A-T-C-3  (Marinus, 1984). Examination of the pR1 and pR2  promoter sequences (Fig. 5) reveals the presence of  Dam methylation sites in the -35 region of both of  these promoters. Since others have shown that’  methylation of Dam sites in promoters can affect  their activity (Kleckner et al., 1984; M. Marinus:  personal communication), we measured cre  promoter activity in dam- and dam’ strains. pR1 is  sensitive to Dam methylation but pR2 and pR3 are  not (Table 5). The activity of pR1 is elevated three-  to fourfold in the dam- strain. This effect is not due  to a change in plasmid copy, as the copy number of  the pRl-containing plasmid was the same in both  the dam+ and dam- strains (data not shown).  (iii) Utilization of cre transcripts  Since the pR3 promoter is located just 11 bp  upstream from the first cre codon, its transcript', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  PI me Gene and its Regulatory Signals 209  Table 7  Lysogenization frequency of Pl cre- mutants  Lysogenization frequency with  Bacterial strains Plc1.106 Plcl.lOOcre2 Plc1.1oocr~3am  C~JOO @WE) 0.14 0.13 0.08  NS616 (supE, recA) 0.080 0@01 0.05  N99 (supO) 0.25 0.22 0.24  Xl00 (sup’, WA) 0.05 0.002 0.002  Lysogeny was measured as described in Materials and Methods. All of the phage contain the Cm’  marker, and the frequency of lysogeny is the number of Cm’ cells/number of phage-infected cells.  should not contain the ribosome recognition site  that precedes cre (Fig. 5). Accordingly, we might  expect that the level of Cre made from that  transcript would be much less than that made from  transcripts that contain the appropriate ribosome  recognition sites, such as the pR1 and pR2  transcripts. Despite this expectation, we could  detect little difference in cre expression between  plasmids containing just pR3 and those containing  one or both of the other cre promoters (Table 6A).  A possible explanation is that Cre is so active that  even the synthesis of a small amount of protein  from each of the many copies present in the strain  bearing the pR3-plasmid is sufficient to obscure  detection of a large translational defect. To increase  the sensitivity of the assay, we reduced the copy  number of the cre gene. First, we cloned the cre  genes from pRHlO3 (which contains pR2 and pR3)  and pRH103-A6 (which contains only pR3) into the  XhoI site of phage 1 (see the legend to Table 6). The  resulting A-cre phages and I-P1 : 7, which contains all three cre promoters, were then used to lysogenize  E. coli. Lysogens with a single prophage were  selected and used to assay cre expression  (Table 6B). The lysogens containing cre genes with  either pR1 or pR2 had much more activity than did  the lysogen containing the cre gene with just pR3.  Since the three promoters exhibit similar tran-  scriptional activities, the pR3 transcript must be  translated much less efficiently than are the pR1  and pR2 transcripts.  (iv) Properties of Pl cre- phage  We isolated cre- mutants of Pl (see Materials  and Methods) to determine what role that gene  might play in the PI life-cycle. Using the mutants,  we previously showed that cre was necessary for  faithful prophage plasmid segregation (Austin et al.,  1981). Here, we describe the effects of cre  inactivation on prophage establishment and lytic  growth. Our results indicate that Cre is essential for  prophage establishment and viral vegetative  20 30 40 50 60  Time after InfectIon (nun)  (b)  Figure 7. Single-cycle phage burst experiments. The experiments in (a) were carried out as described bv Sternberg  (1976). RecA+ strain X99 was grown to a density of 2 x 10’ cells/ml. concentrated IO-fold in L-broth conta:ning 5 mM-  (WI, and infected with phage at a multiplicity of 0.1 phage/cell. After 5 min absorption at 37°C. the cells were diluted  104-fold into L-broth with 5 mM-CaCI, and incubated at 37°C. Portions were removed at designated intervals, treated  wit,h chloroform, and titrated. The infecting phage used are PlCmcl.lOOr-m- (0). P1Cmc1.100r~n~cre2 (0) and  1’1Cmc1.000r-m-cre3 (m). For the experiments in (b), phage growth was initiated by the induction of Plc1.100  Iysogens.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  concentrated IO-fold in L-broth conta:ning 5 mM-  (WI, and infected with phage at a multiplicity of 0.1 phage/cell. After 5 min absorption at 37°C. the cells were diluted  104-fold into L-broth with 5 mM-CaCI, and incubated at 37°C. Portions were removed at designated intervals, treated  wit,h chloroform, and titrated. The infecting phage used are PlCmcl.lOOr-m- (0). P1Cmc1.100r~n~cre2 (0) and  1’1Cmc1.000r-m-cre3 (m). For the experiments in (b), phage growth was initiated by the induction of Plc1.100  Iysogens. The lysogens were grown to a density of 2 x 10s cells/ml at 32°C and induced by diluting a portion of the  csulturr 104-fold into L-broth with 5 mM-CaCl,, prewarmed to 42°C. The culture was grown at this temperature for  15 min and then shifted to 37°C’ for the remainder of the experiment. Portions were removed at designated intervals.  treated wit,h chloroform. and titrated. The lysogens used (Table 1) are the ret!- strains X81031 (0). K’s1032 (a) and  KS1033 (U).  x'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:45 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'80'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993590'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'384ms'), (b'x-request-id', b'8af03ef08d23f5f40a2d065afaa54c9e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412493d7fb65d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:45 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'80'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993590'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'384ms'), (b'x-request-id', b'8af03ef08d23f5f40a2d065afaa54c9e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412493d7fb65d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  210 N. Sternberg et al.  growth in a recA- host, but not in a recA+ host.  We used two cre mutations in these studies: cre2,  a missense mutation, whose location in the cre gene  has not been determined; and cre3, an amber  mutation located in codon 79 of the gene (see  above). We assessed the effect of the cre gene on  prophage establishment by measuring the fraction  of recAt or recA- cells that become resistant to  chloramphenicol after infection with PlCm or  PlCmcre- phages. While the wild-type and mutant  phages were indistinguishable in the recAf host, t’he  cre- mutants lysogenized the recA- hosts with 25  to 80-fold less efficiency than did Pl wild-tvpe  (Table 7). Amber suppression of the cre3 mutation  restored the wild-type phenotype. The results for  virus production were much the same as those for  lysogeny. There was no significant effect of the cre-  mutation in a recA+ host but a large effect in a  recA- host (Fig. 7). In part)icular, the absence of  Cre severely reduced the yield of phage and  significantly delayed their appearance in a recA-  host (Fig. 7(b)). It should be noted that the recA -  experiment in Figure 7 was carried out by prophage  induction rather than phage infection, so that any  effect’ of Cre on the cyclization of the injected linear  viral DNA could be discounted (see below).  4. Discussion  In this paper, we have made several important  observations about the Pl cre gene and its  upstream regulatory region. First, we showed that  cre and its recombination site, ZoxP, are separated  by a region of some 430 nucleotides that contains  an open reading frame (orfl) of 73 amino acid  residues and at least three possible cre promoters.  The latter result suggests that cre expression might  be regulated through the action of those promoters.  To address this possibility, we have explored  several avenues for that regulation. Analysis of  promoter activity in the presence of proteins  produced by the Pl plasmid prophage, including  the cl repressor protein, indicates that none of the  promot’ers is affected by these proteins. However,  one of the promoters, pR1, is affected by DNA  methylation. It is transcribed three to four times  more efficiently in a host, without a functional Dam  methylase (dam-). That promoter contains two  Dam methylation sites (5’.G-A-T-C-3’) in its -35  region. The cre transcripts also appear to be  translated with different efficiencies. More  particularly, the promoter closest to cre. pR3,  produces a transcript that is translated inefficiently,  probably because it lacks a functional ribosomal  recognit,ion site.  We have studied the biological function of  Cre. Previously, we showed that Cre-mediat’ed  recombination is important, for Pl plasmid st’ability  in a recA+ host (Austin et al., 1981). The results  presented here demonstrate that Cre is required  both for the establishment of stable lysogens and  for vegetative viral growth in a recA- host. but’ is dispensable for both of those functions in a recA+  host. We interpret the lysogeny defect as indicating  that Cre is essential for the cyclization of injected  virion DNA in a host lacking a general  recombination system. The cyclization reaction  presumably involves Cre-mediated recombination  between ZoxP sites in the t,erminally redundant  region of the linear viral DNA. Support for the role  of Cre in this reaction comes from t’he direct’  measurements of cyclization by Hochman et ab.  (1983). An alternative interpretation for the  lysogenic defect’ of cre- phage in a recA- host is  that Cre has a direct effect on either t&he  maintenance of prophage immunity or the stability  of the prophage plasmid. This interpretation,  however, is inconsistent with the properties of PI  cre- lysogens.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  but’ is dispensable for both of those functions in a recA+  host. We interpret the lysogeny defect as indicating  that Cre is essential for the cyclization of injected  virion DNA in a host lacking a general  recombination system. The cyclization reaction  presumably involves Cre-mediated recombination  between ZoxP sites in the t,erminally redundant  region of the linear viral DNA. Support for the role  of Cre in this reaction comes from t’he direct’  measurements of cyclization by Hochman et ab.  (1983). An alternative interpretation for the  lysogenic defect’ of cre- phage in a recA- host is  that Cre has a direct effect on either t&he  maintenance of prophage immunity or the stability  of the prophage plasmid. This interpretation,  however, is inconsistent with the properties of PI  cre- lysogens. The immunity of those lysogens to  Pl superinfection is the same as that of Pl cre+  lysogens (data not shown), and the stability of crv’  and cre- plasmids is the same in a recA - host  (Aust,in et al., 1981). Since lysogeny of cre- phage is  normal in a recA+ host, the host’s general  recombination system, 01 that system in  conjunction with other Pl proteins, must also be  able to cyclize PI DNA. The requirement for Pl  proteins in the cyclization reaction even in a recA+  host is consistent with the observation that Pl  DNA cannot be cyclized after being injected into a  recA+ host that has been pre-treated with chlor-  amphenicol (Segev & Cohen, 1981).  With regard to the role of Cre in the lytic cycle of  Pl, we know very lit’tle. Segev & Cohen (1981) have  shown t’hat I’1 DNA appears to be replicated in two  distinct stages, an early and a late stage, and that  the transition from early to late replication is  severely retarded in a recA- host,. This result  suggests t’hat the growth defect of Pl wild-t’ype in a  recA- host (Rosner, 1972) may be due to a failure  to generate late replicative forms by general  recombination. Perhaps the lox-& system can, in  part,, compensate for the hosta’s general  recombination system in carrying out this step. In  t,he absence of a lox-Cre recombination system, the  PI growth defect> in a rscA- host would be more  pronounced.  One of the more intriguing results obtained in  these st)udies is t,he observed increase in  t,ranscription from cre promoter p&l in n dam-  host. This result, is not’ too surprising, in light of the  presence of two adenine residues in the -35 region  of t,his promoter that can be methylat,ed by the  Dam methylase. Similar results have been obtained  with the promoters for the E. coli trpR, and su1A  genes. which contain Dam sites in the -35 region  (Marinus. 1985; S. Gottesman. personal com-  munication), and with the promoters for the Tnl0  and Tn5 transposase genes, which have Dam sites  in the - 10 region (Kleckner et ab., 1984: ,J. Yin &  W’. Reznikoff, personal communication). Since the  DNA of ii:. coli is methylated soon aRer its  replication, one would not expect to find DNA with  both strands unmethylated. However. hemi-  methylat,ed DNA (DNA with the newly replicated', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  PI cre Gene ad its Regulatory Signals 211  strand unmethylated) will exist for a short period  after DNA replication (Marinus, 1976; Lyons &  Schendel, 1984). D. Roberts & N. Kleckner  (personal communication) have shown that the  effect of undermethylation on TnlO promoter  function is detectable when the DNA is hemi-  methylated. These observations suggest a scheme to  explain the biological relevance of cre regulation by  DNA methylation. If one assumes that the pR1  promoter is rapidly replicated soon after Pl DNA is  injected into a sensitive host, then the resulting  hemimethylated DNA might be transcribed more  efficiently. This would increase the amount of Cre  synthesized just at’ the time when that protein is  needed to cyclize t’he linear viral DNA. Once  repression of lytic growth is established, the  replication of pR1 would be inhibited, resulting in a  lower level of cre expression from the fully  methylated promoter. Presumably, that level of cre  transcription would be sufficient for the plasmid  maintenance function of the lox-Cre system.  While the above hypothesis provides a relevant  biological explanation for the properties of t*he pR1  promoter, it does not account for the presence of  two other promoters, pR2 and pR3, upstream from  cre. Since the pR3 promoter appears to signal the  production of a transcript that lacks a ribosome  recognition site and is poorly translated, its  biological significance can, perhaps, be discounted.  However, the transcript’ generated from the pR2  promoter contains a ribosome recognition site  (Fig. 5) and is efficiently translated (Table 6). Since  pR2 is located within the orfl reading frame, its  presence might provide a way for Pl to synthesize  Cre without synthesizing Orfl. However, the  significance of that remains unclear, because we do  not know what’ role, if any, Orfl plays in the Pl  life-cycle. While its location between ZoxP and cre  would suggest that it is involved in Zox-Cre  recombination, all attempts to demonstrate such an  involvement have failed. Purified Cre prot,ein  promotes recombination in vitro between 1oxP sites  efficiently and appears to act on various lox sites  with a specificity indistinguishable from the  reaction in viwo (Sternberg & Hamilton, 1981;  Abremski et al., 1983). Moreover, I-P1 hybrid  phages that contain cre but not an intact orf I gene  (I-P1 : 76306) or that contain cre and orf 1 but not  pR1 (A cre-103) make nearly as much functional Cre  protein as a phage (A-PI :7) containing the ent*ire  region between cre and ZoxP (Table 6 contains the  data for 1. cre-103 and A-PI :7). These results rule  out the possibility that Orfl is an essential  activatIor of Cre or of the cre promoters pRI and  pR2.  In addition to the studies on cre regulation and  fun&ion, we have presented the nucleotide  srquence of the cre structural gene and from it have  generated t,he primary amino acid sequence of the  protein. A detailed computer comparison of tha,t  amino acid sequence with the sequences of other  rccsombination and DNA binding proteins is non  underway. We thank Drs Robert Zagursky and Susan Bear for  their careful reading of the manuscript and Rj. Henderson  for providing the phages 1 cre-103 and 1 cre-103A6. This  work was supported by The Pu’ational Cancer Institute,  DHHF, under contract no. NOl-CO-23909 wit,h Litton  Bionetics Incorporated.  References  Abremski, K. & Hoess, R. (1983). Gene, 25, 49-58.  Abremski. K. & Hoess, R. (1984). J. Biol. Chem. 259,  1509-1514.  Abremski, K., Hoess, R. & Sternberg. N. (1983).', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  We thank Drs Robert Zagursky and Susan Bear for  their careful reading of the manuscript and Rj. Henderson  for providing the phages 1 cre-103 and 1 cre-103A6. This  work was supported by The Pu’ational Cancer Institute,  DHHF, under contract no. NOl-CO-23909 wit,h Litton  Bionetics Incorporated.  References  Abremski, K. & Hoess, R. (1983). Gene, 25, 49-58.  Abremski. K. & Hoess, R. (1984). J. Biol. Chem. 259,  1509-1514.  Abremski, K., Hoess, R. & Sternberg. N. (1983). Cell, 32,  1301-1311.  Appleyard, R. K. (1954). Genetics, 39, 440-452.  Austin, S., Ziese, M. & Sternberg, iX. (1981). C’ell, 25. 729-  736.  Clewell, D. B. & Helinski, D. R. (1969). Proc. ;Vat. A cad.  A)‘&., U.S.A. 62, 1159-1166.  Dagert. M. & Ehrlich, S. D. (1979). Gene, 6, 23-28.  Dennert, G. & Henning, V. (1968). J. &fol. Biol. 33, 322-  329.  Enquist, L. & Weisberg, R. (1976). Firology, 72. 147--153.  Feiss, M., Fisher, R. A., Crayton. M. A. & Enger. C.  (1977). ViroZogy, 77, 281-293.  Glover, J. W., Schell, J.. Symonds. l\\\\j. & Starep. Ii. A.  (1963). Genet. Res. 4. 480-482.  Gottesman, M. & Yarmolinsky, M. (1968). Cold Spring  Harbor Symp. Quark Biol. 33, 735-748.  Hawley, D. K. 8: McClure, W. R. (1983). A&l. Acids Res.  11, 2237-2255.  Hochman, L.. Seger, E., Sternberg. K. & Cohen. G.  (1983). Virology, 131, 11-17.  Hoess, R. 8: Abremski, K. (1985). J. %oZ. Biol. 181. 3hl-  362.  Hoess. R.. Ziese, M. & Sternberg. X;. (1982). t’roc. Nat.  Acud. Sci., U.S.A. 79, 3398-3402.  Holmes, D. S. & Quigley. M. (1981). An,a/. Biochem. 114.  193-197.  Jorgensen. R. A., Rothstein, S. J. & Reznikoff. W. (1979).  Mol. Gen. Genet. 177. 65-74.  Kleckner. X., Morisato, D.. Roberts. D. & Bender, J.  (1984). Cold Bpring Harbor Symp. @rant. Biol. 49,  235-244.  Lin, F.-L., Sperle, K. & Sternberg, N. (1984). ~ilol. (Tell.  Biol. 4. 1020-1034.  Lyons, S. M. 8: Schendel, P. F. (1984). ,J. Boctrriol. 159.  421-423.  Marinus.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Jorgensen. R. A., Rothstein, S. J. & Reznikoff. W. (1979).  Mol. Gen. Genet. 177. 65-74.  Kleckner. X., Morisato, D.. Roberts. D. & Bender, J.  (1984). Cold Bpring Harbor Symp. @rant. Biol. 49,  235-244.  Lin, F.-L., Sperle, K. & Sternberg, N. (1984). ~ilol. (Tell.  Biol. 4. 1020-1034.  Lyons, S. M. 8: Schendel, P. F. (1984). ,J. Boctrriol. 159.  421-423.  Marinus. M. G. (1976). J. Bacterial. 128. 853-854.  Marinus, M. G. (1984). In DNA Methylation (Razin, A..  Cedar, M. & Riggs, A. D., eds), pp. 81-109, Springer-  Verlag. New York.  Marinus, M. G. (1985). ilrlol. Gen. Den&. In the press.  Marinus. M. G.. Carraway, M., Frey. A. Z.. Brown. L. &  Arraj, J. A. (1983). Mol. Gen. Genet. 192. 288-289.  McKenney. K., Shimataka, H., Court, D.. Schmeissner.  LT.. Brady. C. & Rosenberg. M. (1981). In Gene  Ampli$cation and Analysis (Chirikjian. J. G. &  Papas. T. S.. eds), vol. 2. pp. 38&415. Elsevier.  Xorth-Holland, Amsterdam.  Miller, ,J. H. (1972). In Experiments in Moleculur Genetics,  pp. 47-57. Cold Spring Harbor Laboratory Press,  Cold Spring Harbor.  Rosenberg. M. & Court, D. (1979). Ann u. Rev. (kn,rt. 13.  319-353.  Rosnrr. J. 1,. (1972). I’irology, 48. 678-689.  Sanger. F. & Co&on, A. R. (1975). J. Xol. Biol. 94. 441-  448.  Scott. ,I. R. (1968). L’irology, 41. 66-71.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  212 N. Sternberg et al.  Segev. N. & Cohen, G. (1981). Virology, 114, 333-342.  Shine, J. & Dalgarno, L. (1974). hoc. i%t. Acad. Sci.,  U.S.A. 71, 1342-1346.  Sternberg, N. (1976). Virology, 71, 568-582.  Sternberg, N. (1979). Virology, 96, 129-142.  Sternberg, N. & Hamilton, D. (1981). J. ~llol. Biol. 150,  467-486.  Sternberg, N. & Weisberg. R. (1977). J. X02. Biol. 117.  717-731.  Sternberg, N. L., Tiemeier, D. & Enquist. L. (1977). Gen,e,  1. 255-280. Sternberg, N., Hamilton, D., Enquist, L. & Weisberg, R.  (1979). Gene, 8. 35-41.  Sternberg, N., Hamilton, D. & Hoess, R. (1981a). J. Mol.  Biol. 150, 487-507.  Stormo, G.. Schneider, T. D. & Gold. L. M. (1982). Sucl.  Acids Res. 10, 2971-3011. Sternberg. N., Hamilton. D., Austin, S., Yarmolinsky, M.  & Hoess, R. (1981b). Cold &wing Harbor Symp.  C&ant. Biol. 45, 297-309.  Vieira. J. & Messing. J. (1982). Gene, 19, 259-268.  West. R. N. & Scott. J. R. (1977). Virology, 77. 267-276.  Edited by N. Gottesman', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  tJ. Md. Bid. (1992) 226, 661-673  Cre-lox Recombination in Escherichia coli Cells  Mechanistic Differences from the in Vitro Reaction  David E. Adams, James B. Bliskar and Nicholas R. Cozzarelli  Drpartmrnt of Molecular wnd (Ydl Biology  (“nicersity of California, Berkeley, Berkelq. Cd 94720. r,‘.S.A,  (Received XI October 1991: accrpted 27 Narch 1992)  The mechanism of the Cre recombinase of bacteriophage PI in h’scherichia di eelIs was  analyzed by topological methods in order to determine the important features of the in L?(K)  reaction. 3. infection was used t,o introduce the crp gene into cells containing plasmid  substrates. The products of Cre resolution on substrates with directly repeated sites n-ert  predominantly free circles, even though decatenation bv DNA gyrase was blocked by the  drug norfloxacin. Recombination by (:re was greatly stimulated by negat’ive supercoiling,  and inversion occurred inefficiently. These results are strikinglv different from those found  with purified enzyme in vitro. Our data imply that (‘re recombination in bird is much more  tightly controlled than it is in ,&TO, and that Cre acats predominantly as a resolvase in CSW.  We suggest a role for Cre-mediated recombination in PI plasmid amplification that is  consistent with the selectivity of t)he enzyme in ci~.  KP?rtl;ords: bacteriophage I’1 ; DNA supercoiling and topologv; gene targeting:  plasmid amplification: sit’e-specific- recombination  1. Introduction  Site-specific recombination has a variety of roles  in bot*h prokaryotic and eukaryotic organisms,  including the integration of viral DNA into t’he host  chromosome and its excision (Landy, 1989), the  resolution of cointegrated intermediates in trans-  position (Hatfull & Grindley, 1988), the control of  gene activity on invertible DNA segments (Glasgow  fd al., 1989): the maintenance of replicon copy  number (Aust’in et al., 19X1), and the amplification  of plasmids (Futcher, 1988). In several well-  (Lharacterized syst’ems. the specificity of t’he recom-  bination reaction in, ~ifro has been shown to be  precisely suited t)o its cellular role. For example,  DNA invertases, such as the phage Mu Gin system,  exclusively invert the region between two sites  oriented head-to-head (inverse repeats) on a single  DNA molecule to regulate the expression of genes  involved in host range specificity (Glasgow et a,l.,  1989). Conversely. the resolvases of the Tn3/$  family of transposons are limited to the deletion of  DNA between sites oriented head-to-tail (direct  repeats), resulting in the resolution of intermediat,es  in transposition (Hatfull & Grindley, 198X; Kliska rt  t Present address: Department of Microbiology and  Immunology. Stanford rnivchrsity School of Medicsine.  Stanford, (‘A 9430.5. IV.S.r\\\\. al.. 1991). Other recombination systems, such as the  bacteriophage 1. Int system (Landy. 1989). have  more diverse tasks in the cell and are involved in  both intramolecular (viral excision) and inter-  molecular (viral integration) reactions. Accordingly,  the Int syst,em 60 vitro promotes an expanded reper-  t)oire of react’ions consistent wit’h its wider role iy~  riw.  The (‘re-loz system has been studied extensively  in vitro and requires only the phage-encoded recon-  binase. Cre. and two copies of its 34 base-pair target  site. 1oxP (Hoess bit Abremski, 1990). Given plasmid  substrates, purified Cre protein carries out bot’h  intermolecular and intramolecular reactions. Tn  intramolectular reactions.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Int system (Landy. 1989). have  more diverse tasks in the cell and are involved in  both intramolecular (viral excision) and inter-  molecular (viral integration) reactions. Accordingly,  the Int syst,em 60 vitro promotes an expanded reper-  t)oire of react’ions consistent wit’h its wider role iy~  riw.  The (‘re-loz system has been studied extensively  in vitro and requires only the phage-encoded recon-  binase. Cre. and two copies of its 34 base-pair target  site. 1oxP (Hoess bit Abremski, 1990). Given plasmid  substrates, purified Cre protein carries out bot’h  intermolecular and intramolecular reactions. Tn  intramolectular reactions. the sites can be either  headto-tail or head-to-head (Abremski rt a,/., 1983).  Nicked, linear and supercoiled DNA serve equally  well as substrat,es. These properties distinguish Cre  from recombinases such as Tn3/$ resolva,se and the  DNA invertases. which require supercoiling and  carry out, specific intramolecular reactions only.  Recombination reactions are often divided  mechanistically into two classes (Kanaar & Van de  Putte, 1987: Wasserman & Cozzarelli. 1986). In one  class. an ordered synapsis traps a small fixed  number of supercoils so that the products of recom-  binat,ion contain few, if any, knot or catenane cross-  ings or nodes. In the other class, the products are  topologically complex because synapsis traps a vari-', \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Ordered  synapsts Random  Sy”apSlS  Hi ~-.-  Free ctrcles -2 Catenane -4 Catenane  Figure 1. Pathways for (‘rta rrcvmltinat,ion in drr/. 'I'hc-  rrcomhination products made by purified (“rta (*an he  toplogically simple or c.omplrx. This is illustratrd fi)r a  suhstratr with tlirc~c*tly repeatrd lort’ sites (arrobvs).  Simple products. sucxh as unlinkrd frec~ c*irclrs. arise from  an order4 I)athway of’sitr s.vnapsis in which adventitious  substrate superc~~ils are rxc*h~dvtl front thv synaptic,  c.omples. Thv c~~mplt~s products arr c~atrnant~s c,ontainirrg  a variablr numbt~r of intrrlinks and arise from a mc~c.ha~1-  isni in whi(.h substrate suprrcoils arc1 ratdomly trapf)(lcl  hy the synaptic> caomplex: t,his is illustrat,etl for the form;t-  titjn of c,attanancss linked P~CI 2 and 1 negativts 11odt.h.  (‘ornplex knots can also t)ti formed by (‘rtb i?z vifro I)?  r~rc~oni~~inatioir within a vatrnatrd pr.0duc.t (not showil)  able, and of%en large, number of supercoils. Cre uses  both pathways for site synapsis efficiently (Fig:. 1)  and thus yields a mixture of topologically simple  and complex products. With substrat’es containing  directly repeated sites, the complex products are  caterlanes with a variable number of nodes  (Weinberg. 1988). The simple products. accounting  for about half of the total, are unlinked free circles  (Abremski et al.. 1983).  The in Gfro studies of (‘re paint a picturr, of a  promiscuous recombinase. which exhibits no strong  preference for substratP Qology. site cbonnectivity  or pat,hway for synapsis. The ultimate goal of  mechanistic, studies. however. is to understallti  rt~combinat,ion inside cells. t,he environment iti  which proteins and DNA evolved. To do so ~P(YS-  sitates det,ermining which of’ the in cGtro propert itss  are physiologically relevant.  The (Ye-/olr system may be involved in srvrraI  processes in gizmo (Yarmolinsky & St’ernberg, 198X).  1 t)s proposed roles incalude cyclization of newI>  injected linear PI DNA (Hochman et ccl.. 1983). the  occ~asional integration of the cyclic~ prophage into\", 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  The Mechnn.ism of C’w Rrcornhination in \\\\‘ivo 663  XhoI KhoI r PL  af ‘Icre i I I B I  ! fmm434!  (0)  Eco  ottP  (b)  Figure 2. Struc.turcA of the (Ire expression Jjhage and  substrat,e plasmids. (a) The (‘re expression phage. iX2-  CW. Shown UP th(t ~‘1~ promoter (arrow). the irnm434  region (box). thr UP transcriJ)tion unit (box). and the two  ,YhoT sit,c>s that flank the insert of cre seyuences from  phagr PI. (b) Structures of the substrate ptasmids pKS-5.  J~~JN35tl. pl)A-5 and J)5l .loxP. The lox! sites are  depic~trd as filled arrows, thr rcs sit,es as open arrows. and  the c&P and aftB sites as hatched arrows (not t,o scale).  Rest,ri&on &es used to distinguish recombinant and  parental ctonfigurations are indicated.  rxtent of site-sJ)ecGfic recombination for the various  substrates was determined by digestion using BumHI or  EcoR,I restriction enzymes. The positions of the relevant  rcst,riction sit,es are shown in Fig. 2. For the Cre fusion  and resolution reactions in z)i~o, the extent of recombina-  tion was also determined from the ratio of supercoiled  ptasmid monomer to dimer. Analysis of product topology  was carried out, hy nicking the recombination products  with panc*reatic I)Nase I in the presence of 300 pg ethi-  dium bromide/ml and separating the linked and unlinked  species by high-resolution gel electrophoresis. The nicking  reactions contained 20 mM-Tris . KC1 (pH 75).  50 mn-X?(‘l, IO mM-?Yzg(:l,. 10 pg plasmid DNA/ml,  300 pg ethidium bromide/ml, and 1.6 pg I)i%ase I/ml.  pKS-5 topoisomers differing in average linking number  were prepared by treatment of the supercoited plasmid  with wheat germ topoisomerasr T in the presence of 0 to 2 /.q ethidium bromide per ml of Cre reaction buffer. After  60 min at 30°C. SDS was added to 05% and proteinase K  to 0.2 pa/ml. Aft,er an additional 60 min at 30°C. the  reactions were extracted wit,h phenol and the D&A  precipitated with ethanol. Reactions with phage T4 topo-  isomrrase II. a gift from Bruce Alberts. CniversiW of  California. San Francisco. were carried out as described  (Kreuzer $ ,Jongenerl, 1983).  E. ,o/i strains t~setl in this study are listed ill Table I.  Thr roy)<4 genot?pr of R.rd-tl was c.onfirmed b? the  maint,enance of sul)ercoiling of plasmirls Rftrr I)SX g,vrasr  inhibition.  E. X2-c.rr l)hapr \\\\vas a gift from I<. Saucr at I)uJ)ont and  is identical with i, Xl-cre (Sauer & Henderson, 1988)  excrJ,t that the orientation of t.he cw gene is rt>vrrsed so  that it is tra.nsqribrtl from the /lpi, promot.er. I’lasmid  pJK3.Jtf has been drscribed (Hliska & (‘ozzarrlli. 1987)  and is tliagrammed in Fig. 2. I’lasmids l)KS-,5. ~~5l.losl’  (Fig. 2). and pLOXI’2(rev) (not shown) were obtained  from Ken ;-\\\\brrntski at I)uI’ont.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  210 N. Sternberg et al.  growth in a recA- host, but not in a recA+ host.  We used two cre mutations in these studies: cre2,  a missense mutation, whose location in the cre gene  has not been determined; and cre3, an amber  mutation located in codon 79 of the gene (see  above). We assessed the effect of the cre gene on  prophage establishment by measuring the fraction  of recAt or recA- cells that become resistant to  chloramphenicol after infection with PlCm or  PlCmcre- phages. While the wild-type and mutant  phages were indistinguishable in the recAf host, t’he  cre- mutants lysogenized the recA- hosts with 25  to 80-fold less efficiency than did Pl wild-tvpe  (Table 7). Amber suppression of the cre3 mutation  restored the wild-type phenotype. The results for  virus production were much the same as those for  lysogeny. There was no significant effect of the cre-  mutation in a recA+ host but a large effect in a  recA- host (Fig. 7). In part)icular, the absence of  Cre severely reduced the yield of phage and  significantly delayed their appearance in a recA-  host (Fig. 7(b)). It should be noted that the recA -  experiment in Figure 7 was carried out by prophage  induction rather than phage infection, so that any  effect’ of Cre on the cyclization of the injected linear  viral DNA could be discounted (see below).  4. Discussion  In this paper, we have made several important  observations about the Pl cre gene and its  upstream regulatory region. First, we showed that  cre and its recombination site, ZoxP, are separated  by a region of some 430 nucleotides that contains  an open reading frame (orfl) of 73 amino acid  residues and at least three possible cre promoters.  The latter result suggests that cre expression might  be regulated through the action of those promoters.  To address this possibility, we have explored  several avenues for that regulation. Analysis of  promoter activity in the presence of proteins  produced by the Pl plasmid prophage, including  the cl repressor protein, indicates that none of the  promot’ers is affected by these proteins. However,  one of the promoters, pR1, is affected by DNA  methylation. It is transcribed three to four times  more efficiently in a host, without a functional Dam  methylase (dam-). That promoter contains two  Dam methylation sites (5’.G-A-T-C-3’) in its -35  region. The cre transcripts also appear to be  translated with different efficiencies. More  particularly, the promoter closest to cre. pR3,  produces a transcript that is translated inefficiently,  probably because it lacks a functional ribosomal  recognit,ion site.  We have studied the biological function of  Cre. Previously, we showed that Cre-mediat’ed  recombination is important, for Pl plasmid st’ability  in a recA+ host (Austin et al., 1981). The results  presented here demonstrate that Cre is required  both for the establishment of stable lysogens and  for vegetative viral growth in a recA- host. but’ is dispensable for both of those functions in a recA+  host. We interpret the lysogeny defect as indicating  that Cre is essential for the cyclization of injected  virion DNA in a host lacking a general  recombination system. The cyclization reaction  presumably involves Cre-mediated recombination  between ZoxP sites in the t,erminally redundant  region of the linear viral DNA. Support for the role  of Cre in this reaction comes from t’he direct’  measurements of cyclization by Hochman et ab.  (1983). An alternative interpretation for the  lysogenic defect’ of cre- phage in a recA- host is  that Cre has a direct effect on either t&he  maintenance of prophage immunity or the stability  of the prophage plasmid. This interpretation,  however, is inconsistent with the properties of PI  cre- lysogens.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  but’ is dispensable for both of those functions in a recA+  host. We interpret the lysogeny defect as indicating  that Cre is essential for the cyclization of injected  virion DNA in a host lacking a general  recombination system. The cyclization reaction  presumably involves Cre-mediated recombination  between ZoxP sites in the t,erminally redundant  region of the linear viral DNA. Support for the role  of Cre in this reaction comes from t’he direct’  measurements of cyclization by Hochman et ab.  (1983). An alternative interpretation for the  lysogenic defect’ of cre- phage in a recA- host is  that Cre has a direct effect on either t&he  maintenance of prophage immunity or the stability  of the prophage plasmid. This interpretation,  however, is inconsistent with the properties of PI  cre- lysogens. The immunity of those lysogens to  Pl superinfection is the same as that of Pl cre+  lysogens (data not shown), and the stability of crv’  and cre- plasmids is the same in a recA - host  (Aust,in et al., 1981). Since lysogeny of cre- phage is  normal in a recA+ host, the host’s general  recombination system, 01 that system in  conjunction with other Pl proteins, must also be  able to cyclize PI DNA. The requirement for Pl  proteins in the cyclization reaction even in a recA+  host is consistent with the observation that Pl  DNA cannot be cyclized after being injected into a  recA+ host that has been pre-treated with chlor-  amphenicol (Segev & Cohen, 1981).  With regard to the role of Cre in the lytic cycle of  Pl, we know very lit’tle. Segev & Cohen (1981) have  shown t’hat I’1 DNA appears to be replicated in two  distinct stages, an early and a late stage, and that  the transition from early to late replication is  severely retarded in a recA- host,. This result  suggests t’hat the growth defect of Pl wild-t’ype in a  recA- host (Rosner, 1972) may be due to a failure  to generate late replicative forms by general  recombination. Perhaps the lox-& system can, in  part,, compensate for the hosta’s general  recombination system in carrying out this step. In  t,he absence of a lox-Cre recombination system, the  PI growth defect> in a rscA- host would be more  pronounced.  One of the more intriguing results obtained in  these st)udies is t,he observed increase in  t,ranscription from cre promoter p&l in n dam-  host. This result, is not’ too surprising, in light of the  presence of two adenine residues in the -35 region  of t,his promoter that can be methylat,ed by the  Dam methylase. Similar results have been obtained  with the promoters for the E. coli trpR, and su1A  genes. which contain Dam sites in the -35 region  (Marinus. 1985; S. Gottesman. personal com-  munication), and with the promoters for the Tnl0  and Tn5 transposase genes, which have Dam sites  in the - 10 region (Kleckner et ab., 1984: ,J. Yin &  W’. Reznikoff, personal communication). Since the  DNA of ii:. coli is methylated soon aRer its  replication, one would not expect to find DNA with  both strands unmethylated. However. hemi-  methylat,ed DNA (DNA with the newly replicated', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  PI cre Gene ad its Regulatory Signals 211  strand unmethylated) will exist for a short period  after DNA replication (Marinus, 1976; Lyons &  Schendel, 1984). D. Roberts & N. Kleckner  (personal communication) have shown that the  effect of undermethylation on TnlO promoter  function is detectable when the DNA is hemi-  methylated. These observations suggest a scheme to  explain the biological relevance of cre regulation by  DNA methylation. If one assumes that the pR1  promoter is rapidly replicated soon after Pl DNA is  injected into a sensitive host, then the resulting  hemimethylated DNA might be transcribed more  efficiently. This would increase the amount of Cre  synthesized just at’ the time when that protein is  needed to cyclize t’he linear viral DNA. Once  repression of lytic growth is established, the  replication of pR1 would be inhibited, resulting in a  lower level of cre expression from the fully  methylated promoter. Presumably, that level of cre  transcription would be sufficient for the plasmid  maintenance function of the lox-Cre system.  While the above hypothesis provides a relevant  biological explanation for the properties of t*he pR1  promoter, it does not account for the presence of  two other promoters, pR2 and pR3, upstream from  cre. Since the pR3 promoter appears to signal the  production of a transcript that lacks a ribosome  recognition site and is poorly translated, its  biological significance can, perhaps, be discounted.  However, the transcript’ generated from the pR2  promoter contains a ribosome recognition site  (Fig. 5) and is efficiently translated (Table 6). Since  pR2 is located within the orfl reading frame, its  presence might provide a way for Pl to synthesize  Cre without synthesizing Orfl. However, the  significance of that remains unclear, because we do  not know what’ role, if any, Orfl plays in the Pl  life-cycle. While its location between ZoxP and cre  would suggest that it is involved in Zox-Cre  recombination, all attempts to demonstrate such an  involvement have failed. Purified Cre prot,ein  promotes recombination in vitro between 1oxP sites  efficiently and appears to act on various lox sites  with a specificity indistinguishable from the  reaction in viwo (Sternberg & Hamilton, 1981;  Abremski et al., 1983). Moreover, I-P1 hybrid  phages that contain cre but not an intact orf I gene  (I-P1 : 76306) or that contain cre and orf 1 but not  pR1 (A cre-103) make nearly as much functional Cre  protein as a phage (A-PI :7) containing the ent*ire  region between cre and ZoxP (Table 6 contains the  data for 1. cre-103 and A-PI :7). These results rule  out the possibility that Orfl is an essential  activatIor of Cre or of the cre promoters pRI and  pR2.  In addition to the studies on cre regulation and  fun&ion, we have presented the nucleotide  srquence of the cre structural gene and from it have  generated t,he primary amino acid sequence of the  protein. A detailed computer comparison of tha,t  amino acid sequence with the sequences of other  rccsombination and DNA binding proteins is non  underway. We thank Drs Robert Zagursky and Susan Bear for  their careful reading of the manuscript and Rj. Henderson  for providing the phages 1 cre-103 and 1 cre-103A6. This  work was supported by The Pu’ational Cancer Institute,  DHHF, under contract no. NOl-CO-23909 wit,h Litton  Bionetics Incorporated.  References  Abremski, K. & Hoess, R. (1983). Gene, 25, 49-58.  Abremski. K. & Hoess, R. (1984). J. Biol. Chem. 259,  1509-1514.  Abremski, K., Hoess, R. & Sternberg. N. (1983).', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  We thank Drs Robert Zagursky and Susan Bear for  their careful reading of the manuscript and Rj. Henderson  for providing the phages 1 cre-103 and 1 cre-103A6. This  work was supported by The Pu’ational Cancer Institute,  DHHF, under contract no. NOl-CO-23909 wit,h Litton  Bionetics Incorporated.  References  Abremski, K. & Hoess, R. (1983). Gene, 25, 49-58.  Abremski. K. & Hoess, R. (1984). J. Biol. Chem. 259,  1509-1514.  Abremski, K., Hoess, R. & Sternberg. N. (1983). Cell, 32,  1301-1311.  Appleyard, R. K. (1954). Genetics, 39, 440-452.  Austin, S., Ziese, M. & Sternberg, iX. (1981). C’ell, 25. 729-  736.  Clewell, D. B. & Helinski, D. R. (1969). Proc. ;Vat. A cad.  A)‘&., U.S.A. 62, 1159-1166.  Dagert. M. & Ehrlich, S. D. (1979). Gene, 6, 23-28.  Dennert, G. & Henning, V. (1968). J. &fol. Biol. 33, 322-  329.  Enquist, L. & Weisberg, R. (1976). Firology, 72. 147--153.  Feiss, M., Fisher, R. A., Crayton. M. A. & Enger. C.  (1977). ViroZogy, 77, 281-293.  Glover, J. W., Schell, J.. Symonds. l\\\\j. & Starep. Ii. A.  (1963). Genet. Res. 4. 480-482.  Gottesman, M. & Yarmolinsky, M. (1968). Cold Spring  Harbor Symp. Quark Biol. 33, 735-748.  Hawley, D. K. 8: McClure, W. R. (1983). A&l. Acids Res.  11, 2237-2255.  Hochman, L.. Seger, E., Sternberg. K. & Cohen. G.  (1983). Virology, 131, 11-17.  Hoess, R. 8: Abremski, K. (1985). J. %oZ. Biol. 181. 3hl-  362.  Hoess. R.. Ziese, M. & Sternberg. X;. (1982). t’roc. Nat.  Acud. Sci., U.S.A. 79, 3398-3402.  Holmes, D. S. & Quigley. M. (1981). An,a/. Biochem. 114.  193-197.  Jorgensen. R. A., Rothstein, S. J. & Reznikoff. W. (1979).  Mol. Gen. Genet. 177. 65-74.  Kleckner. X., Morisato, D.. Roberts. D. & Bender, J.  (1984). Cold Bpring Harbor Symp. @rant. Biol. 49,  235-244.  Lin, F.-L., Sperle, K. & Sternberg, N. (1984). ~ilol. (Tell.  Biol. 4. 1020-1034.  Lyons, S. M. 8: Schendel, P. F. (1984). ,J. Boctrriol. 159.  421-423.  Marinus.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  Jorgensen. R. A., Rothstein, S. J. & Reznikoff. W. (1979).  Mol. Gen. Genet. 177. 65-74.  Kleckner. X., Morisato, D.. Roberts. D. & Bender, J.  (1984). Cold Bpring Harbor Symp. @rant. Biol. 49,  235-244.  Lin, F.-L., Sperle, K. & Sternberg, N. (1984). ~ilol. (Tell.  Biol. 4. 1020-1034.  Lyons, S. M. 8: Schendel, P. F. (1984). ,J. Boctrriol. 159.  421-423.  Marinus. M. G. (1976). J. Bacterial. 128. 853-854.  Marinus, M. G. (1984). In DNA Methylation (Razin, A..  Cedar, M. & Riggs, A. D., eds), pp. 81-109, Springer-  Verlag. New York.  Marinus, M. G. (1985). ilrlol. Gen. Den&. In the press.  Marinus. M. G.. Carraway, M., Frey. A. Z.. Brown. L. &  Arraj, J. A. (1983). Mol. Gen. Genet. 192. 288-289.  McKenney. K., Shimataka, H., Court, D.. Schmeissner.  LT.. Brady. C. & Rosenberg. M. (1981). In Gene  Ampli$cation and Analysis (Chirikjian. J. G. &  Papas. T. S.. eds), vol. 2. pp. 38&415. Elsevier.  Xorth-Holland, Amsterdam.  Miller, ,J. H. (1972). In Experiments in Moleculur Genetics,  pp. 47-57. Cold Spring Harbor Laboratory Press,  Cold Spring Harbor.  Rosenberg. M. & Court, D. (1979). Ann u. Rev. (kn,rt. 13.  319-353.  Rosnrr. J. 1,. (1972). I’irology, 48. 678-689.  Sanger. F. & Co&on, A. R. (1975). J. Xol. Biol. 94. 441-  448.  Scott. ,I. R. (1968). L’irology, 41. 66-71.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-0022283686902287-main.pdf  212 N. Sternberg et al.  Segev. N. & Cohen, G. (1981). Virology, 114, 333-342.  Shine, J. & Dalgarno, L. (1974). hoc. i%t. Acad. Sci.,  U.S.A. 71, 1342-1346.  Sternberg, N. (1976). Virology, 71, 568-582.  Sternberg, N. (1979). Virology, 96, 129-142.  Sternberg, N. & Hamilton, D. (1981). J. ~llol. Biol. 150,  467-486.  Sternberg, N. & Weisberg. R. (1977). J. X02. Biol. 117.  717-731.  Sternberg, N. L., Tiemeier, D. & Enquist. L. (1977). Gen,e,  1. 255-280. Sternberg, N., Hamilton, D., Enquist, L. & Weisberg, R.  (1979). Gene, 8. 35-41.  Sternberg, N., Hamilton, D. & Hoess, R. (1981a). J. Mol.  Biol. 150, 487-507.  Stormo, G.. Schneider, T. D. & Gold. L. M. (1982). Sucl.  Acids Res. 10, 2971-3011. Sternberg. N., Hamilton. D., Austin, S., Yarmolinsky, M.  & Hoess, R. (1981b). Cold &wing Harbor Symp.  C&ant. Biol. 45, 297-309.  Vieira. J. & Messing. J. (1982). Gene, 19, 259-268.  West. R. N. & Scott. J. R. (1977). Virology, 77. 267-276.  Edited by N. Gottesman', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  tJ. Md. Bid. (1992) 226, 661-673  Cre-lox Recombination in Escherichia coli Cells  Mechanistic Differences from the in Vitro Reaction  David E. Adams, James B. Bliskar and Nicholas R. Cozzarelli  Drpartmrnt of Molecular wnd (Ydl Biology  (“nicersity of California, Berkeley, Berkelq. Cd 94720. r,‘.S.A,  (Received XI October 1991: accrpted 27 Narch 1992)  The mechanism of the Cre recombinase of bacteriophage PI in h’scherichia di eelIs was  analyzed by topological methods in order to determine the important features of the in L?(K)  reaction. 3. infection was used t,o introduce the crp gene into cells containing plasmid  substrates. The products of Cre resolution on substrates with directly repeated sites n-ert  predominantly free circles, even though decatenation bv DNA gyrase was blocked by the  drug norfloxacin. Recombination by (:re was greatly stimulated by negat’ive supercoiling,  and inversion occurred inefficiently. These results are strikinglv different from those found  with purified enzyme in vitro. Our data imply that (‘re recombination in bird is much more  tightly controlled than it is in ,&TO, and that Cre acats predominantly as a resolvase in CSW.  We suggest a role for Cre-mediated recombination in PI plasmid amplification that is  consistent with the selectivity of t)he enzyme in ci~.  KP?rtl;ords: bacteriophage I’1 ; DNA supercoiling and topologv; gene targeting:  plasmid amplification: sit’e-specific- recombination  1. Introduction  Site-specific recombination has a variety of roles  in bot*h prokaryotic and eukaryotic organisms,  including the integration of viral DNA into t’he host  chromosome and its excision (Landy, 1989), the  resolution of cointegrated intermediates in trans-  position (Hatfull & Grindley, 1988), the control of  gene activity on invertible DNA segments (Glasgow  fd al., 1989): the maintenance of replicon copy  number (Aust’in et al., 19X1), and the amplification  of plasmids (Futcher, 1988). In several well-  (Lharacterized syst’ems. the specificity of t’he recom-  bination reaction in, ~ifro has been shown to be  precisely suited t)o its cellular role. For example,  DNA invertases, such as the phage Mu Gin system,  exclusively invert the region between two sites  oriented head-to-head (inverse repeats) on a single  DNA molecule to regulate the expression of genes  involved in host range specificity (Glasgow et a,l.,  1989). Conversely. the resolvases of the Tn3/$  family of transposons are limited to the deletion of  DNA between sites oriented head-to-tail (direct  repeats), resulting in the resolution of intermediat,es  in transposition (Hatfull & Grindley, 198X; Kliska rt  t Present address: Department of Microbiology and  Immunology. Stanford rnivchrsity School of Medicsine.  Stanford, (‘A 9430.5. IV.S.r\\\\. al.. 1991). Other recombination systems, such as the  bacteriophage 1. Int system (Landy. 1989). have  more diverse tasks in the cell and are involved in  both intramolecular (viral excision) and inter-  molecular (viral integration) reactions. Accordingly,  the Int syst,em 60 vitro promotes an expanded reper-  t)oire of react’ions consistent wit’h its wider role iy~  riw.  The (‘re-loz system has been studied extensively  in vitro and requires only the phage-encoded recon-  binase. Cre. and two copies of its 34 base-pair target  site. 1oxP (Hoess bit Abremski, 1990). Given plasmid  substrates, purified Cre protein carries out bot’h  intermolecular and intramolecular reactions. Tn  intramolectular reactions.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Int system (Landy. 1989). have  more diverse tasks in the cell and are involved in  both intramolecular (viral excision) and inter-  molecular (viral integration) reactions. Accordingly,  the Int syst,em 60 vitro promotes an expanded reper-  t)oire of react’ions consistent wit’h its wider role iy~  riw.  The (‘re-loz system has been studied extensively  in vitro and requires only the phage-encoded recon-  binase. Cre. and two copies of its 34 base-pair target  site. 1oxP (Hoess bit Abremski, 1990). Given plasmid  substrates, purified Cre protein carries out bot’h  intermolecular and intramolecular reactions. Tn  intramolectular reactions. the sites can be either  headto-tail or head-to-head (Abremski rt a,/., 1983).  Nicked, linear and supercoiled DNA serve equally  well as substrat,es. These properties distinguish Cre  from recombinases such as Tn3/$ resolva,se and the  DNA invertases. which require supercoiling and  carry out, specific intramolecular reactions only.  Recombination reactions are often divided  mechanistically into two classes (Kanaar & Van de  Putte, 1987: Wasserman & Cozzarelli. 1986). In one  class. an ordered synapsis traps a small fixed  number of supercoils so that the products of recom-  binat,ion contain few, if any, knot or catenane cross-  ings or nodes. In the other class, the products are  topologically complex because synapsis traps a vari-', \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Ordered  synapsts Random  Sy”apSlS  Hi ~-.-  Free ctrcles -2 Catenane -4 Catenane  Figure 1. Pathways for (‘rta rrcvmltinat,ion in drr/. 'I'hc-  rrcomhination products made by purified (“rta (*an he  toplogically simple or c.omplrx. This is illustratrd fi)r a  suhstratr with tlirc~c*tly repeatrd lort’ sites (arrobvs).  Simple products. sucxh as unlinkrd frec~ c*irclrs. arise from  an order4 I)athway of’sitr s.vnapsis in which adventitious  substrate superc~~ils are rxc*h~dvtl front thv synaptic,  c.omples. Thv c~~mplt~s products arr c~atrnant~s c,ontainirrg  a variablr numbt~r of intrrlinks and arise from a mc~c.ha~1-  isni in whi(.h substrate suprrcoils arc1 ratdomly trapf)(lcl  hy the synaptic> caomplex: t,his is illustrat,etl for the form;t-  titjn of c,attanancss linked P~CI 2 and 1 negativts 11odt.h.  (‘ornplex knots can also t)ti formed by (‘rtb i?z vifro I)?  r~rc~oni~~inatioir within a vatrnatrd pr.0duc.t (not showil)  able, and of%en large, number of supercoils. Cre uses  both pathways for site synapsis efficiently (Fig:. 1)  and thus yields a mixture of topologically simple  and complex products. With substrat’es containing  directly repeated sites, the complex products are  caterlanes with a variable number of nodes  (Weinberg. 1988). The simple products. accounting  for about half of the total, are unlinked free circles  (Abremski et al.. 1983).  The in Gfro studies of (‘re paint a picturr, of a  promiscuous recombinase. which exhibits no strong  preference for substratP Qology. site cbonnectivity  or pat,hway for synapsis. The ultimate goal of  mechanistic, studies. however. is to understallti  rt~combinat,ion inside cells. t,he environment iti  which proteins and DNA evolved. To do so ~P(YS-  sitates det,ermining which of’ the in cGtro propert itss  are physiologically relevant.  The (Ye-/olr system may be involved in srvrraI  processes in gizmo (Yarmolinsky & St’ernberg, 198X).  1 t)s proposed roles incalude cyclization of newI>  injected linear PI DNA (Hochman et ccl.. 1983). the  occ~asional integration of the cyclic~ prophage into\", 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  The Mechnn.ism of C’w Rrcornhination in \\\\‘ivo 663  XhoI KhoI r PL  af ‘Icre i I I B I  ! fmm434!  (0)  Eco  ottP  (b)  Figure 2. Struc.turcA of the (Ire expression Jjhage and  substrat,e plasmids. (a) The (‘re expression phage. iX2-  CW. Shown UP th(t ~‘1~ promoter (arrow). the irnm434  region (box). thr UP transcriJ)tion unit (box). and the two  ,YhoT sit,c>s that flank the insert of cre seyuences from  phagr PI. (b) Structures of the substrate ptasmids pKS-5.  J~~JN35tl. pl)A-5 and J)5l .loxP. The lox! sites are  depic~trd as filled arrows, thr rcs sit,es as open arrows. and  the c&P and aftB sites as hatched arrows (not t,o scale).  Rest,ri&on &es used to distinguish recombinant and  parental ctonfigurations are indicated.  rxtent of site-sJ)ecGfic recombination for the various  substrates was determined by digestion using BumHI or  EcoR,I restriction enzymes. The positions of the relevant  rcst,riction sit,es are shown in Fig. 2. For the Cre fusion  and resolution reactions in z)i~o, the extent of recombina-  tion was also determined from the ratio of supercoiled  ptasmid monomer to dimer. Analysis of product topology  was carried out, hy nicking the recombination products  with panc*reatic I)Nase I in the presence of 300 pg ethi-  dium bromide/ml and separating the linked and unlinked  species by high-resolution gel electrophoresis. The nicking  reactions contained 20 mM-Tris . KC1 (pH 75).  50 mn-X?(‘l, IO mM-?Yzg(:l,. 10 pg plasmid DNA/ml,  300 pg ethidium bromide/ml, and 1.6 pg I)i%ase I/ml.  pKS-5 topoisomers differing in average linking number  were prepared by treatment of the supercoited plasmid  with wheat germ topoisomerasr T in the presence of 0 to 2 /.q ethidium bromide per ml of Cre reaction buffer. After  60 min at 30°C. SDS was added to 05% and proteinase K  to 0.2 pa/ml. Aft,er an additional 60 min at 30°C. the  reactions were extracted wit,h phenol and the D&A  precipitated with ethanol. Reactions with phage T4 topo-  isomrrase II. a gift from Bruce Alberts. CniversiW of  California. San Francisco. were carried out as described  (Kreuzer $ ,Jongenerl, 1983).  E. ,o/i strains t~setl in this study are listed ill Table I.  Thr roy)<4 genot?pr of R.rd-tl was c.onfirmed b? the  maint,enance of sul)ercoiling of plasmirls Rftrr I)SX g,vrasr  inhibition.  E. X2-c.rr l)hapr \\\\vas a gift from I<. Saucr at I)uJ)ont and  is identical with i, Xl-cre (Sauer & Henderson, 1988)  excrJ,t that the orientation of t.he cw gene is rt>vrrsed so  that it is tra.nsqribrtl from the /lpi, promot.er. I’lasmid  pJK3.Jtf has been drscribed (Hliska & (‘ozzarrlli. 1987)  and is tliagrammed in Fig. 2. I’lasmids l)KS-,5. ~~5l.losl’  (Fig. 2). and pLOXI’2(rev) (not shown) were obtained  from Ken ;-\\\\brrntski at I)uI’ont.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:46 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'74'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993610'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'383ms'), (b'x-request-id', b'c34a78c8d206fb03518e41245499bf6e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249428c885d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:46 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'74'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993610'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'383ms'), (b'x-request-id', b'c34a78c8d206fb03518e41245499bf6e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249428c885d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  E. X2-c.rr l)hapr \\\\vas a gift from I<. Saucr at I)uJ)ont and  is identical with i, Xl-cre (Sauer & Henderson, 1988)  excrJ,t that the orientation of t.he cw gene is rt>vrrsed so  that it is tra.nsqribrtl from the /lpi, promot.er. I’lasmid  pJK3.Jtf has been drscribed (Hliska & (‘ozzarrlli. 1987)  and is tliagrammed in Fig. 2. I’lasmids l)KS-,5. ~~5l.losl’  (Fig. 2). and pLOXI’2(rev) (not shown) were obtained  from Ken ;-\\\\brrntski at I)uI’ont. F’lasmid pl)A-5 (Fig. 2)  wa,s l.onstruc%rtl hy insertion of a 2.3 k\\\\)t Hind1 I1 frag-  ment c.ontaining thr, s/rpF gene from J)RK(:3 (ohtainrd  from R Kanaar) into the unique NiltdTlT site in the  4-B kh pLOXP%(rt~v) and selecting for rZmpR TrtR colonies  of ~l(‘lO6l;F’3 crlls. pl).4-5 is 69 kb ill size anal ha.s 2 IosT’  sites in inverted orirntation separated 1)~ :I..5 kb. The  resolrasr expression ptasmid p,JHRIWI ha,s been  described (Hliska it 01.. 1991).  All ire /~i/~o rrc,olnl)ination assays M~‘I’P rvtbfwttd sevwat  titiic,s bvitti c~omJ~arat)lr results. E. coli (dells were trans-  fornitld uit,h t.hr va.rious plasmids and grown at 28°C’ in  Lurta broth supJ)l~~mentrd with O.?“,, (~jv) maltosr and  25 pg ampicillin/ml. At a density of 70 Klett units, t,hr  (~11s wvrr c*oncrntratrd by c~rnt.rif’upatiotl and resus-  prnd(q~ in l/lT,th of their original voiutnr in broth  c*ontainitlg 10 ni~l-MgSO, to promote Jlhape adsorption.  (‘onc.rntratrd ~~11s wert’ thtxn rr.turnrd to growth condi-  tions for IO min prior to rrc*omhinastx intluc~ticnl. Par (‘re  indrlc%ion. the c*ulturr was shified tc, 42 ‘(’ and infectrd  uith 0.5 vol. of’ J)rewarmed broth c,tmtaining i S~-CW  phayta to givr a multiplicity of infrc?ion of IO. At t,imrs  tht~rt+~ft~~r. culls we’re diluted to thrir original volurnt, in  WCC brot,h c,ontaining norfloxac*in at 0. IO or 50 pg/mI:  5 mirr was thr optimal timr to wait twfiw adding norftw-  acain in order to maximize (‘re rrcomhination whilr gyrltse  was inhibited. The caulturr trmperaturrs for (Ire induc,tion  --', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  wrw c*hosen to alloy direct c*omparison with resolvasr irlld  Int induction. For rrsolvast. induc*tion. ~41s harboring  plasmitls JAI BRESCT and pKS-5 were trratrtl ah abo\\\\.(a  cbscrll)t that infrc.tion with 2 Mhrr phagc~ was omittrtl.  J;or Tnt induction. c~oncentratetl exponrntial phasti  Ibt~l-ClA cells harhorinp p,JBS.T,d werr i,lcnbatrtl \\\\vith  shaking at 4d”(‘ for 10 min to induce i~t rxJ)ression. To  irrit,i;LttA rt~t~ortlluilratioil hy thr thrrrnosensitivt~ I nt sy&anl.  (.t,lls \\\\verl~ tlillltrd in mrdium at, XI ‘C’. For, c~iic~lr ,I, vi/v)  rr~l~onil)inat~it,n assay. I.5 ml sarriplf~s oi thts intiucvtl  vulturrs were removed at various times and plasmid I)SA  was isolat,etl by an alkaline Iysis method (Samhrook v( trl..  19X9). The wlls wt‘w t~entrifuged and thr pt4r+ \\\\vrre  rrsuspended in 60 /II of ice-cold 50 rnM-gluWse.  IO ~sI-EI)T,\\\\. 25 m11-Tris. HC’I (pH 8.0). .i mg Iyso~ynl~  nil. and norfloxacirr at tht\\\\ sarn~ drug (~l,llc.t,nti,atiorl as  1rst~1 in t,hc r*‘c,omhina.tioi~ assay. (‘ells were keJ)t on i(*tl foi  5 to IO nlirl. after \\\\\\\\ hitah 2.5 111 of IO”,, (\\\\V!V) SIH antI I ItI  of IO mg proteinasr K/ml \\\\vert’ atldrtl. an,1 thp total  nlixttrre was thran incuhatrd at :37’(’ for SO min  Following addition of 80 ~1 of 0.2 ar-NaOH, I O. SI)S arltl  60 /iI of’ potassium at.etat,r (pH 4.X). thv tuhrs \\\\vt‘rt’ left  c,vtJrnight at 4’(‘. ‘fht~ samples hvrr c~r~ntrif’upetl antI the  supernatant ~vas extracted suc~cwsivt~l~- with phrnol.  phrnol/(‘H(‘I,. arlcl (‘H( ‘I,. I’la.smitl 1)X;\\\\ \\\\\\\\as then  prtvail)itatrd aitti f~ttiatiol.  ProtwIurrs usrd for standard and high rrsolution  I -dimensional gel &ctrophorrsis havr hern drscrihrd  (Bliska & (‘ozzarelli. 1987). To analyzr the distribatjion of  linking nunlber (/A) topoisomers. samples were electro-  l)hortJstvl in agarostl gels in thr prf’st’nve of t~hloroqui~lt.  The gels caontainrtl I O. (w/v) agarosr. 50 tn~I-‘~ris-at,t,t~ltt.  (pH X+). IO m.n-EIYI’A. and 0 to 100 ~rll-c~hlorocJui~l~~, To  calculatt~ the average r~ (i.r. (Lk- ZA,)/Lk,). Lk, was sclt  vyual to the equilibrium /,k after wheat. germ topoiso-  merase I treatment of plasniid in Cre reaction buffer at  WY’.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  samples were electro-  l)hortJstvl in agarostl gels in thr prf’st’nve of t~hloroqui~lt.  The gels caontainrtl I O. (w/v) agarosr. 50 tn~I-‘~ris-at,t,t~ltt.  (pH X+). IO m.n-EIYI’A. and 0 to 100 ~rll-c~hlorocJui~l~~, To  calculatt~ the average r~ (i.r. (Lk- ZA,)/Lk,). Lk, was sclt  vyual to the equilibrium /,k after wheat. germ topoiso-  merase I treatment of plasniid in Cre reaction buffer at  WY’. The mean number of nodes of Tnt ratenanes genar-  ated in viva was clrterminrd hv densitometry of auto-  radiograms of nirkrd I)KA sub~wtrtl to high-resolution  gel rlectroJ)horrsis. These nlean ralur,s w?;(~rc then used to  c2lculatf5 t.lw 0 wquid i/2 c:itt-0 fitr Trot to producr an  tquivalent c,atenanr distribution based on tht> published  cxlihration (‘urv? (Boles (4 c/l.. 1!190). Aut~oradiograptl~  ,vas carrird out with probe I)r\\\\‘X laheletl bvith  ~a-32P]tl(‘Tl’ using a mult~iJ)rimr IIN:\\\\ Iaheling kit from  .Attlrrsham.  Thtx genrration of positivetv supercoiled pRR322 l)IiA  (Lockshon It. Jlorris. 1983; L6u rt al.. 198X) was us4 to  monitor norfloxacin inhibition of the supercoiled a&\\\\-it?  by I)?ilI gyrasr. With IO /,~g norfloxacinjml. approx~-  rtlatt~ly half of pKRX% in MOltX:i. 10 cells brvarnr J)osi-  tivvly supereoilrd in SO min. By inrreasing thri  ncn+ioxacG voncvntration to ,50 g$ml. more than 90”,, of  the 1)rV.A became positively supercoiled with a r~ grrat,er  than +0+X% To measure the efficiency of inhibition of  drcatenation by DNA gyrase, the profile of Tn3 resolvasr  and 2 Tnt, c:a.tenanes generated in viva with and without  norfloxacin was examined. We ryuated the percentage of  resolvase or Tnt products trapped as catenanes as equal to  t,hr efficiency of gyrase inhibition: both recomhinases ix  vitro generate onlg linked products by resolution. Tn the  present-r of 50 /*g norfloxacin/ml, 75”4, of the resolvase  and Int, in P’LW resolution products were routinely trapped  as tsatenanrs. 3. Results  (YP carries out a fiurrib~~r of’ r’eat.tiolis itc vifr~~. ;ltJ(l  the topology- of’t)he iJro(luc.ts is untikv that of’ ittly 01’  the othtlr dell-t~hara~tc~ri~t,(l prokaryot i(. wcv)tnl)i-  nases. To tleterminc~ wtrivtr propertiPs of’ t hti i,J vifr,~j  (‘re rf:avt ion are ph~siologit~allv r.t~tf3vallt. we’ (‘lillll~  inetf (:w r.ec,orn~tirlat.iori ;ri ,.Iu,. %I(: 1655. IO WIIS  containing a plasmid auhst rate> for (‘rtb wt’r(s rriilirl-  tained iL1 30°C’.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  wcv)tnl)i-  nases. To tleterminc~ wtrivtr propertiPs of’ t hti i,J vifr,~j  (‘re rf:avt ion are ph~siologit~allv r.t~tf3vallt. we’ (‘lillll~  inetf (:w r.ec,orn~tirlat.iori ;ri ,.Iu,. %I(: 1655. IO WIIS  containing a plasmid auhst rate> for (‘rtb wt’r(s rriilirl-  tained iL1 30°C’. To init iattl (‘rt4w rt~c,orril)ili;ltioii.  the wlls \\\\verv infet+c-l(l \\\\vith a L fjhsgv (~onstrut~t  (Fig. L)(a)) that rxprwws rrr’ f’rorn thtl vtar~. acativts  ~1, promoter. Thts tolls wt’rt (~(,~lt~t~ntr~;~tt~ct antI  infected at a high multiplicGt.\\\\~ to gu;tratrt.rv c~fficGt~n1  phagt, adsorption. After l.ti minutrl. t hcs itrfivttvl  cells were dilutctl in metlium rit tier lacking or  containing ttir IJtJtent I)SA tr~r’ast’ irrtrihitor.  norfloxacain (Shrtll it rtl.. I SHS). %orfioxitc~iri wiih  addfY1 t0 hlO(‘k dwatrnation SO t hilt t tie ttJ~)010gy ot  t.hr protlucls of’ rec:ornt)il,;lticJtl c~oulcl IJO ttt~t.~~rmint~tl.  NOf4OSi~t~ill treatment also rrslllttvl itI ttlcs loss of‘  plasmid riegativth suprhrc~oiling itj this If)/). I ’ st riiill  and wtlnwtt the initial rattl of t~f~t~tJ~rl~)irl;~lio~l tJ\\\\.  about ti)rirti~ltt.  To vxarnint~ t hr tolJolog>- of (‘rv fJroduc*ts in a  c~ont.rollf~d HRV. \\\\I-(’ rrsttct il sUtX+t.IXtc~. pKS-5  (Fig. Y(l))), \\\\vhit*h harbors tlirrct I,L, rrpeatt~tf WA sites  in addition to tlirec+I>~ rrpeatctl If~.rl’ sites.  Ke~omhinatiotr t)y (‘rts t*onverts this tlirnt:rit~  suhstratti into identitAly sizrtl produt~t rings that  may or may not lye linkrd. antI Tn,Y rvsolvast’ penrxr-  atrs singly linkwl tbat,enantts it/ vi~v~ (I{lisk;t it r/l..  l!wl). .  (‘PIIs harboring [JKS-5 ant1 l)Pl HK fWI (a [Jlitsltlitl  with the resolvasr gvrjv under writrc~l 01’ ,>I“  repressor) were induced for rpsotvase r,rt,tJrnt)illatiol~.  In paraltrl. WIIS harhoriny only pKS-5 WNY~ infvctt*tl  with j. SCerr pha.gr to turn on (‘rv synt hvsis. Ontb  minutr after induction (intivt ion). norfloxa(~in wxs  added to 0. IO or 50 pg/ml. The highthr Itivvl of’ clrug  tnaximizt~d gsrase inhibition Lvhile ht ill allowing  tletet%alJlr fvombination (Pig. 3). At t hti 1 imt5  intfitbat4. plasmid l)f\\\\‘A was isolattvl.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  In paraltrl. WIIS harhoriny only pKS-5 WNY~ infvctt*tl  with j. SCerr pha.gr to turn on (‘rv synt hvsis. Ontb  minutr after induction (intivt ion). norfloxa(~in wxs  added to 0. IO or 50 pg/ml. The highthr Itivvl of’ clrug  tnaximizt~d gsrase inhibition Lvhile ht ill allowing  tletet%alJlr fvombination (Pig. 3). At t hti 1 imt5  intfitbat4. plasmid l)f\\\\‘A was isolattvl. nivketl with  1)Nasfa I. and visualiwd hy get t~t~~c~t,rol)~iorf~sis anal  Southrrn blotting.  f,it’tlrl or Ilo rrcombinatiori hid ot~t~urrrtl twti)rce  induction (Fig. 3. lanes I ant1 5). Trn minutw Iatw.  extensivr rrwtion t)y hot h rwomt~inasrs had take11  place in t hc ahsenw‘of. norfoxavin as evidenwd 1)).  the wnversion of’thr 7.3 kh suhstrattx to 3% kt) f’rc:r*  tGwlr produt~t (laws 2 untl 6). III tht, ~Jrvsvntv of’  IO /LLfr rlorffosi~t~itl~rnl. ttirrtb Ivits illI iI1t.rvasf~ in 1 tICI  rwover,v of siriglc liiikc~tl c3trnikrws ~~rc~clric~d t)J  resolvasr (lant~ 3). At 50 pL, (7 norfox;~t~irl:‘riil. 1111 to  7 2 “() of resolvase prtJ(luvts wert‘ t rapptvl ah singI>  linked c-atrnases (law 4). f II ctjnt rast (‘rti tnaclc~  on I\\\\. t.ra.c~t~ iImounts of’ tinkrti protlucts. Ill tllr~  at)srnvf> of’ tlorflosac~ir~ this t~jrrld hc tlufa to t trtx r;i.J)itl  dwafenat ion t)y I)&;\\\\ gyrase. Howt~\\\\-t&r. t’vtw itI t,ht  presence of’ inh’ilJit,or. (Ire formrtl utltinkvti I,roduc.ts  almost v.xc*lusivet~ (Fig. 3. lanes 7 atitl 8). /tt vit,rr,.  (‘rc. formed la.rpr amounts of singi!; linked c~atc~natlvs', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Inducl,an,ln,sslron RE.7 atE I I  “73 -  p,*RESCI -  L73 -  UT.3 -  L3-6 -  Figure 3. Topology of (‘re and resolvase products i?c  c~ioo. The dimrric substrate pKS-5, has pairs of both  W’S (optan arro\\\\rs) and 10x1’ sites (filled arrows) as direct  repeats. In cults c,orttaining l)KS-5. either (Ire or resotvasr  N as induc*rd and the t~atrnanes produced by each wert’  (*omparrti. Th(l c.;lrtoon summarizes the in Gno results.  and the clata gSrtrrratrtl in MC: t6,5.5.10 cells are drpichtrtl  below. Eac,h recombinase uas induced separately. and at  the times indic+atrti. plasmid DNA was isolated, nicked  wit,h I)Nasr 1. ;+tttl displayed by gel electrophoresis and  Southern blott.irtg. TAUIW I to 4. (‘ells harboring both  pKS-5 and ]).I BKEScl. Afifjr growth at 28’(“. cells were  shifted t,o 12 C’ to itrducae rc~tlvasr expression (time 0). At  1 mitt later. ttot,floxacain was added to 0. 10 or 50 pg/mt.  ,\\\\t 2 min. cells w(%rt’ diluted in the same medium at 30°C’  and plasmid I)SA was isolatjrd at, the times indicated.  I,anrs 5 to 8. (‘tall harboring pKS-5 only. At time 0. cells  wrre shift4 to 1P’(’ and inftac%rd with 1 X2-err phage. At  1 min later, drug was addrd. =\\\\t 2 min. cells were diluted  and 1)X.-\\\\ was isolated and l)rocessed as above. The posi-  tions of the substrat,e in its nicked circular (n7.3): linear  (17.3). anal supercoiled (~7.3) forms are indicated on the  left. h’irkrd ~~JKJIESS~I is detected due to cross-hgbridiza-  tion with the, full-lr%ngth pKS-5 probe. Reference ladders  were grnrratrd b?, t,rrating l)KS-5 in z&o with Crr using  superc~oitrd (lane Ml ) and relaxed (lane MM2) substrates.  i,anes M3 and M-1 caont,ain nit>ked Tn3 resolrase products  and T4 topoisomrrase IT knots, respectively. On the right  is indicatrtl thy trumber of nodes in the rat’enated and  knotted ntarktlr q)t4~s: 4 ttoded catenanes run slightI?-  slower than 1 noded knots. The hrmi-nicked catenanr  product (hemi). ~II whicah only 1 of the 2 rings in nicked.  results from inc.otnpletc nicktng by UNase I. n3.6 marks  the position of rtic,krd free (Grcle product. Lanes 9 and 9’.  Sorfloxac,irr was added t,o 50 pg/mt to uninfected cells.  I,anr 9’ is an ovf’t’(‘xposure of lane 9. =\\\\t the bottom is the  measured extent of re(.ombinat,ion and the percentage of  t.t,t,omhinittiotl proctut+s t)hat are krtotted or catenated  af’ter sttbtrac+ion of’ the bac*hground level of knotting. and more complex topological forms in addit,ion to  fiw product Grcles (Fig. 3. lanes JMl and MS).', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  results from inc.otnpletc nicktng by UNase I. n3.6 marks  the position of rtic,krd free (Grcle product. Lanes 9 and 9’.  Sorfloxac,irr was added t,o 50 pg/mt to uninfected cells.  I,anr 9’ is an ovf’t’(‘xposure of lane 9. =\\\\t the bottom is the  measured extent of re(.ombinat,ion and the percentage of  t.t,t,omhinittiotl proctut+s t)hat are krtotted or catenated  af’ter sttbtrac+ion of’ the bac*hground level of knotting. and more complex topological forms in addit,ion to  fiw product Grcles (Fig. 3. lanes JMl and MS). The  discarepancy between the in viva and i?/ vitro results  suggests that in E. coZi cells Cre carries out only a  subset of’ the reactions it performs in. ~itrr~.  The amount of recombination of pKS-5 was  markedly reduced by the addition of 50 pg/rnl  norfloxacin at one minute for both (Ire (Fig. 3. lane  8) and resolvase (Fig. 3, lane 4). Cnder our experi-  mental protocol. topoisomerase I relaxes DNA when  gyrase is inhibit,ed by norfloxacin. Since Tn3 resol-  vase requires a supercoiled substrate (Uliska rt (x,1.,  1991). recombination by this enzyme would  decrease with increasing relaxation of its substrate.  Clre recombination would likewise decrease if it t,oo  requires a supercoiled substrate in l’iV0.  -Alternatively. the addition of norfloxacin only one  minut,e after infection could have limited (!re  expression by reducing phage DNA replicat.ion and/  or transcription. To test this hypothesis we dela.yed  the ad&ion of norfloxacin until five minutes a.fter  infection to obtain a higher copy number of phages  and to allow less time for substrate plasmid relnxa-  tion (Lockshon & Morris, 1983). The result of the  later addition of drugs was a threefold increase in  C’re recombination but no increase in the arnount of  catenated product (data not shown).  Recombination was still about fourfold less than in  t,he absence of norfloxacin. This reduction is prob-  ably due to substrate relaxation, because in a topo-  isomerase 1 deficient strain. almost no inhibition of  recombination was found in the presence of norflox-  acin (see below).  Sorfloxacin addition trapped knotjted DrVA  species that are not resolvase or Cre products, since  they were produced in the absence of recombinase  intluc%ion (Fig. S, lane 9 and its ovt’resposure la,ne  9’). This incrrasts in the Iert>l of plasmid knotting is  similar to that seen in chertain gyr mutant st)rains  (Shishitlo rt 01.. 19X7) and in t.he present’t’ of other  inhibitors of’ gyrase (I&ii of cxl.. 19!jl ).  \\\\z’cl used t hr rwalts shown in Figure 3 to cBompa,rr  the extent of’ topological linkage of (‘re produc*t~s  made iI/ /‘ilW it,nd in vitro from a supercoiled  substrate. After normalizing for the same amount of  rec.otnhirratiorl. lve estimate t,hat (‘rc makes L-fold  fewer linked products irz /GPO than ir/ vitro. This is a  provisional comparison because of the diminution of  negative supereoiling of the substrate ittier gyrase  inhibition and the fact that (‘re in r?tro makes  predominantl) unlinked products u-it h relaxed  substrat.tJ* (lam‘ M2).  To estimate better the bias (“re exhibit*s fhr  forming unlinked products in V~VO.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  \\\\z’cl used t hr rwalts shown in Figure 3 to cBompa,rr  the extent of’ topological linkage of (‘re produc*t~s  made iI/ /‘ilW it,nd in vitro from a supercoiled  substrate. After normalizing for the same amount of  rec.otnhirratiorl. lve estimate t,hat (‘rc makes L-fold  fewer linked products irz /GPO than ir/ vitro. This is a  provisional comparison because of the diminution of  negative supereoiling of the substrate ittier gyrase  inhibition and the fact that (‘re in r?tro makes  predominantl) unlinked products u-it h relaxed  substrat.tJ* (lam‘ M2).  To estimate better the bias (“re exhibit*s fhr  forming unlinked products in V~VO. we examined the  dependence of (Jre product catenation and knotting  on substrate linking number. A series of topoisomers  of pKS-5 was prepared that differed in average  supercoil density. cr, from O*OO to -0V76. These  DX,;\\\\s were recombined by Cre in Gtro. and the', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Wi I). I<. :lflfrttln t,t al  tl I I I I I II  0 -0.02 -0~04-0.06 -0.08  Speclflc ,,nkmg difference (CT)  Cb)  Figure 4. The rtfrc+ of suMrate supercoilrtl density ott  thr t~opologicd complexity of produc*ts tnatlr Iry (‘re.  (a) Topoisomrrs of ~KS-T, w-ith the average supercoiling  clrnsitv (C-J) shown wart’ reactrd with (‘r~ under st~antlartl  c~ondit’ions. Approxitnxtrly Y!S”,, starting LIB,4 was supt.-  c~oiletl. Re~ombittiltion products were proc~rssrtl atttl  visualizrd RS desc~rihrd for Fig. 3. I,;tnes 111 ntttl M:!  c*ontaitt rrferrncv laclders of T1 topoisotnc~rasr II pKS-5  knots and Tn.7 rtdrasr pKS-5 protluc+s. rrsprctivrly.  Thv markers allo\\\\v assignrnt~nt of the ttutnltrr of cat,rnartt~  nodes in the (“re producb. Linear I)ICiA (17.3) results  front stopping the itr Gt,ro (Ye rractiotis with SDS. (Ir) Tlit~  atnount of linked products (catrnanes and knots) made I-J>  (Ire in /G/W uas quantified I)g tlrnsitotriPtr,v and is plottetl  (‘cr.s/1s substrate suytvoil drtisit>,.  reaction products were displayed by gel ele&ro-  phoresis after nicking with DNase I (Fig. 4(a)).  There was a simple linear relat,ionship between the  fraction of products that) are linked (i.e. catenated  or knotted) and the supercoil den&y of the  substrate. The extent of product linkage varied  from 15 “/;, for relaxed DNA to SSqi, for DNA with a  o of -0.076 (Fig. 4(b)). There was also an increase  in the average number of nodes of the products wit,h  increasing supercoiling (Fig. 4(a)). These more  complex species were present in roughly equal  amounts, as would be expect’ed for a substrate such  as pKS-5. whit+ has t,wo symmetrically placed  recombination sites that collide randomly  (Benjamin & Clozzarelli, 1986). At a o value (;f  -0025, which is the approximate effect’ive level of  supercoiling in viva (for a review, see McClellan rt  nl., 1990), 3094 of the products were linked; even  with a relaxed substrate 15% were linked. Both  values are far more than the 2o/0 linkage of the i7r  r~ivo products. We conclude that the lower super- tnorasv I mutant strain. (a) Strain Itrtlll c~c)trl;titt* it ‘I’tt //1  itisrrtion it1 top.4. (‘rv t,~c,c)ttt\\\\)tttitttott L\\\\ its itttluc,tatl its  &scribed for Fig. I+, exc.rpt that thus ittc.ttl)atiott tc~trrt)r~r’;r.  turn was shifted to 12 (’ IO ttritt prior to ittfectiotr to  triitnic thra Int ititiuc+iott c~otitlitiotis (SW Fig. (i). .\\\\t 5 triitl  after infection. portiotts of ttit, c~rtlturc~ wt’rt’ tlilutt~l in  tndiutti containing 0. 10 or 50 pg tiorfloxac~itt tt11. l’lastttitl  1)S.A wits isolatrtl at the tinttls itttiic~at.tvl atitl proc~t~ssccl its  bforr.', \"page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  I+, exc.rpt that thus ittc.ttl)atiott tc~trrt)r~r’;r.  turn was shifted to 12 (’ IO ttritt prior to ittfectiotr to  triitnic thra Int ititiuc+iott c~otitlitiotis (SW Fig. (i). .\\\\t 5 triitl  after infection. portiotts of ttit, c~rtlturc~ wt’rt’ tlilutt~l in  tndiutti containing 0. 10 or 50 pg tiorfloxac~itt tt11. l’lastttitl  1)S.A wits isolatrtl at the tinttls itttiic~at.tvl atitl proc~t~ssccl its  bforr. I,attr~s bl I ant1 .\\\\I? c.c)trt ain. rt~spc~t ii-vl,v. refrrrttc~c  latltlttrs of I)KS-5 (‘rcs l)rodric+h 1’rotti *ttl)er(~oilrtl ;ititl  rrlaxrtl suhstratrs. (II) ‘l’tic~ I)l;rstttitl kttcbttitig (IIW to  rrorfosacitt addition itt thrj ;rhsc~ttc~t~ of itrlivd iott is sho\\\\+tt.  Lanti M4 contains T4 to])oisotrt~~t~;ts~~ I I litlot%. ‘lb 1)X;\\\\ itt  grl (b) rati slightl,v slo\\\\vt~r t tiatr that itt pvl (8 t. tvsriltitig it1  an offssrt of thrs kttot i)ositiotlh  coiling in GVO after gyrase inhibit ion cannot explain  t*he simple t,opology of the (“re reaction produc+s in  IiZW.  ((3) C ‘rfd-tttdicrtfd rrsoltrfifttt in t 0pX E. wli  To bypass the (~xperirnt~ntal c.orrl~)lic,;rtic,tr ot’  substrate relaxation afitbr tiorfloxac*iri ;~tltlitioti. wo  used ittl /I;. roll strain tleficierit iti 1)X,-l topoiso-  rnrrase 1 ac%ivit\\\\-. This strain. Rrd41. (~otitains ii  TnZ11 insertion ’ in to/~.4 arid in c~otri~tt~risiito~~  rnut,ation in ~~yr.4 (Kaji Pt /I/.. l!M). In the prc~serlc~t~  of 10 pg norfloxa4n;ml. plasmid supf~rhelic*ai  density in t,his st,rairi rc%mained at a ('OtlStiillt itLv4  slightly higher t,han t’hat of wild-typt, E. co/i (Bliska  & (‘ozzarelli. 1!)87).  With this fossil strain, 50 pg norfloxa~in~tril  added five minut’es after infection had very lit)tle  effect OII (‘rc recombination (Fig. .?. lanes 3 a1~1 4).  A few minutes after drug addition. rec*ombinat iota  was extrrisivc, and resolution of pKS-5 was 6X”,, 01  that in the a~bsencc of drug. (‘rp rxhibitrtl at1 evetl  stronger bias for forming only unlinked producats in  the top.4 s;t rain compared with the top24 + st.rain  even though the substrate rema,ined highl~~ nt~gn~  tively supercoiled (Fig. 5, lanes 3 and 1).  A large background of’ plasmid knotting w~,s  again observed in the presence of 50 py riorfloxaciti,’  ml (Fig. 5. lanes 1 and i). The average number of\"], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  E. X2-c.rr l)hapr \\\\vas a gift from I<. Saucr at I)uJ)ont and  is identical with i, Xl-cre (Sauer & Henderson, 1988)  excrJ,t that the orientation of t.he cw gene is rt>vrrsed so  that it is tra.nsqribrtl from the /lpi, promot.er. I’lasmid  pJK3.Jtf has been drscribed (Hliska & (‘ozzarrlli. 1987)  and is tliagrammed in Fig. 2. I’lasmids l)KS-,5. ~~5l.losl’  (Fig. 2). and pLOXI’2(rev) (not shown) were obtained  from Ken ;-\\\\brrntski at I)uI’ont. F’lasmid pl)A-5 (Fig. 2)  wa,s l.onstruc%rtl hy insertion of a 2.3 k\\\\)t Hind1 I1 frag-  ment c.ontaining thr, s/rpF gene from J)RK(:3 (ohtainrd  from R Kanaar) into the unique NiltdTlT site in the  4-B kh pLOXP%(rt~v) and selecting for rZmpR TrtR colonies  of ~l(‘lO6l;F’3 crlls. pl).4-5 is 69 kb ill size anal ha.s 2 IosT’  sites in inverted orirntation separated 1)~ :I..5 kb. The  resolrasr expression ptasmid p,JHRIWI ha,s been  described (Hliska it 01.. 1991).  All ire /~i/~o rrc,olnl)ination assays M~‘I’P rvtbfwttd sevwat  titiic,s bvitti c~omJ~arat)lr results. E. coli (dells were trans-  fornitld uit,h t.hr va.rious plasmids and grown at 28°C’ in  Lurta broth supJ)l~~mentrd with O.?“,, (~jv) maltosr and  25 pg ampicillin/ml. At a density of 70 Klett units, t,hr  (~11s wvrr c*oncrntratrd by c~rnt.rif’upatiotl and resus-  prnd(q~ in l/lT,th of their original voiutnr in broth  c*ontainitlg 10 ni~l-MgSO, to promote Jlhape adsorption.  (‘onc.rntratrd ~~11s wert’ thtxn rr.turnrd to growth condi-  tions for IO min prior to rrc*omhinastx intluc~ticnl. Par (‘re  indrlc%ion. the c*ulturr was shified tc, 42 ‘(’ and infectrd  uith 0.5 vol. of’ J)rewarmed broth c,tmtaining i S~-CW  phayta to givr a multiplicity of infrc?ion of IO. At t,imrs  tht~rt+~ft~~r. culls we’re diluted to thrir original volurnt, in  WCC brot,h c,ontaining norfloxac*in at 0. IO or 50 pg/mI:  5 mirr was thr optimal timr to wait twfiw adding norftw-  acain in order to maximize (‘re rrcomhination whilr gyrltse  was inhibited. The caulturr trmperaturrs for (Ire induc,tion  --', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  wrw c*hosen to alloy direct c*omparison with resolvasr irlld  Int induction. For rrsolvast. induc*tion. ~41s harboring  plasmitls JAI BRESCT and pKS-5 were trratrtl ah abo\\\\.(a  cbscrll)t that infrc.tion with 2 Mhrr phagc~ was omittrtl.  J;or Tnt induction. c~oncentratetl exponrntial phasti  Ibt~l-ClA cells harhorinp p,JBS.T,d werr i,lcnbatrtl \\\\vith  shaking at 4d”(‘ for 10 min to induce i~t rxJ)ression. To  irrit,i;LttA rt~t~ortlluilratioil hy thr thrrrnosensitivt~ I nt sy&anl.  (.t,lls \\\\verl~ tlillltrd in mrdium at, XI ‘C’. For, c~iic~lr ,I, vi/v)  rr~l~onil)inat~it,n assay. I.5 ml sarriplf~s oi thts intiucvtl  vulturrs were removed at various times and plasmid I)SA  was isolat,etl by an alkaline Iysis method (Samhrook v( trl..  19X9). The wlls wt‘w t~entrifuged and thr pt4r+ \\\\vrre  rrsuspended in 60 /II of ice-cold 50 rnM-gluWse.  IO ~sI-EI)T,\\\\. 25 m11-Tris. HC’I (pH 8.0). .i mg Iyso~ynl~  nil. and norfloxacirr at tht\\\\ sarn~ drug (~l,llc.t,nti,atiorl as  1rst~1 in t,hc r*‘c,omhina.tioi~ assay. (‘ells were keJ)t on i(*tl foi  5 to IO nlirl. after \\\\\\\\ hitah 2.5 111 of IO”,, (\\\\V!V) SIH antI I ItI  of IO mg proteinasr K/ml \\\\vert’ atldrtl. an,1 thp total  nlixttrre was thran incuhatrd at :37’(’ for SO min  Following addition of 80 ~1 of 0.2 ar-NaOH, I O. SI)S arltl  60 /iI of’ potassium at.etat,r (pH 4.X). thv tuhrs \\\\vt‘rt’ left  c,vtJrnight at 4’(‘. ‘fht~ samples hvrr c~r~ntrif’upetl antI the  supernatant ~vas extracted suc~cwsivt~l~- with phrnol.  phrnol/(‘H(‘I,. arlcl (‘H( ‘I,. I’la.smitl 1)X;\\\\ \\\\\\\\as then  prtvail)itatrd aitti f~ttiatiol.  ProtwIurrs usrd for standard and high rrsolution  I -dimensional gel &ctrophorrsis havr hern drscrihrd  (Bliska & (‘ozzarelli. 1987). To analyzr the distribatjion of  linking nunlber (/A) topoisomers. samples were electro-  l)hortJstvl in agarostl gels in thr prf’st’nve of t~hloroqui~lt.  The gels caontainrtl I O. (w/v) agarosr. 50 tn~I-‘~ris-at,t,t~ltt.  (pH X+). IO m.n-EIYI’A. and 0 to 100 ~rll-c~hlorocJui~l~~, To  calculatt~ the average r~ (i.r. (Lk- ZA,)/Lk,). Lk, was sclt  vyual to the equilibrium /,k after wheat. germ topoiso-  merase I treatment of plasniid in Cre reaction buffer at  WY’.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  samples were electro-  l)hortJstvl in agarostl gels in thr prf’st’nve of t~hloroqui~lt.  The gels caontainrtl I O. (w/v) agarosr. 50 tn~I-‘~ris-at,t,t~ltt.  (pH X+). IO m.n-EIYI’A. and 0 to 100 ~rll-c~hlorocJui~l~~, To  calculatt~ the average r~ (i.r. (Lk- ZA,)/Lk,). Lk, was sclt  vyual to the equilibrium /,k after wheat. germ topoiso-  merase I treatment of plasniid in Cre reaction buffer at  WY’. The mean number of nodes of Tnt ratenanes genar-  ated in viva was clrterminrd hv densitometry of auto-  radiograms of nirkrd I)KA sub~wtrtl to high-resolution  gel rlectroJ)horrsis. These nlean ralur,s w?;(~rc then used to  c2lculatf5 t.lw 0 wquid i/2 c:itt-0 fitr Trot to producr an  tquivalent c,atenanr distribution based on tht> published  cxlihration (‘urv? (Boles (4 c/l.. 1!190). Aut~oradiograptl~  ,vas carrird out with probe I)r\\\\‘X laheletl bvith  ~a-32P]tl(‘Tl’ using a mult~iJ)rimr IIN:\\\\ Iaheling kit from  .Attlrrsham.  Thtx genrration of positivetv supercoiled pRR322 l)IiA  (Lockshon It. Jlorris. 1983; L6u rt al.. 198X) was us4 to  monitor norfloxacin inhibition of the supercoiled a&\\\\-it?  by I)?ilI gyrasr. With IO /,~g norfloxacinjml. approx~-  rtlatt~ly half of pKRX% in MOltX:i. 10 cells brvarnr J)osi-  tivvly supereoilrd in SO min. By inrreasing thri  ncn+ioxacG voncvntration to ,50 g$ml. more than 90”,, of  the 1)rV.A became positively supercoiled with a r~ grrat,er  than +0+X% To measure the efficiency of inhibition of  drcatenation by DNA gyrase, the profile of Tn3 resolvasr  and 2 Tnt, c:a.tenanes generated in viva with and without  norfloxacin was examined. We ryuated the percentage of  resolvase or Tnt products trapped as catenanes as equal to  t,hr efficiency of gyrase inhibition: both recomhinases ix  vitro generate onlg linked products by resolution. Tn the  present-r of 50 /*g norfloxacin/ml, 75”4, of the resolvase  and Int, in P’LW resolution products were routinely trapped  as tsatenanrs. 3. Results  (YP carries out a fiurrib~~r of’ r’eat.tiolis itc vifr~~. ;ltJ(l  the topology- of’t)he iJro(luc.ts is untikv that of’ ittly 01’  the othtlr dell-t~hara~tc~ri~t,(l prokaryot i(. wcv)tnl)i-  nases. To tleterminc~ wtrivtr propertiPs of’ t hti i,J vifr,~j  (‘re rf:avt ion are ph~siologit~allv r.t~tf3vallt. we’ (‘lillll~  inetf (:w r.ec,orn~tirlat.iori ;ri ,.Iu,. %I(: 1655. IO WIIS  containing a plasmid auhst rate> for (‘rtb wt’r(s rriilirl-  tained iL1 30°C’.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  wcv)tnl)i-  nases. To tleterminc~ wtrivtr propertiPs of’ t hti i,J vifr,~j  (‘re rf:avt ion are ph~siologit~allv r.t~tf3vallt. we’ (‘lillll~  inetf (:w r.ec,orn~tirlat.iori ;ri ,.Iu,. %I(: 1655. IO WIIS  containing a plasmid auhst rate> for (‘rtb wt’r(s rriilirl-  tained iL1 30°C’. To init iattl (‘rt4w rt~c,orril)ili;ltioii.  the wlls \\\\verv infet+c-l(l \\\\vith a L fjhsgv (~onstrut~t  (Fig. L)(a)) that rxprwws rrr’ f’rorn thtl vtar~. acativts  ~1, promoter. Thts tolls wt’rt (~(,~lt~t~ntr~;~tt~ct antI  infected at a high multiplicGt.\\\\~ to gu;tratrt.rv c~fficGt~n1  phagt, adsorption. After l.ti minutrl. t hcs itrfivttvl  cells were dilutctl in metlium rit tier lacking or  containing ttir IJtJtent I)SA tr~r’ast’ irrtrihitor.  norfloxacain (Shrtll it rtl.. I SHS). %orfioxitc~iri wiih  addfY1 t0 hlO(‘k dwatrnation SO t hilt t tie ttJ~)010gy ot  t.hr protlucls of’ rec:ornt)il,;lticJtl c~oulcl IJO ttt~t.~~rmint~tl.  NOf4OSi~t~ill treatment also rrslllttvl itI ttlcs loss of‘  plasmid riegativth suprhrc~oiling itj this If)/). I ’ st riiill  and wtlnwtt the initial rattl of t~f~t~tJ~rl~)irl;~lio~l tJ\\\\.  about ti)rirti~ltt.  To vxarnint~ t hr tolJolog>- of (‘rv fJroduc*ts in a  c~ont.rollf~d HRV. \\\\I-(’ rrsttct il sUtX+t.IXtc~. pKS-5  (Fig. Y(l))), \\\\vhit*h harbors tlirrct I,L, rrpeatt~tf WA sites  in addition to tlirec+I>~ rrpeatctl If~.rl’ sites.  Ke~omhinatiotr t)y (‘rts t*onverts this tlirnt:rit~  suhstratti into identitAly sizrtl produt~t rings that  may or may not lye linkrd. antI Tn,Y rvsolvast’ penrxr-  atrs singly linkwl tbat,enantts it/ vi~v~ (I{lisk;t it r/l..  l!wl). .  (‘PIIs harboring [JKS-5 ant1 l)Pl HK fWI (a [Jlitsltlitl  with the resolvasr gvrjv under writrc~l 01’ ,>I“  repressor) were induced for rpsotvase r,rt,tJrnt)illatiol~.  In paraltrl. WIIS harhoriny only pKS-5 WNY~ infvctt*tl  with j. SCerr pha.gr to turn on (‘rv synt hvsis. Ontb  minutr after induction (intivt ion). norfloxa(~in wxs  added to 0. IO or 50 pg/ml. The highthr Itivvl of’ clrug  tnaximizt~d gsrase inhibition Lvhile ht ill allowing  tletet%alJlr fvombination (Pig. 3). At t hti 1 imt5  intfitbat4. plasmid l)f\\\\‘A was isolattvl.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  In paraltrl. WIIS harhoriny only pKS-5 WNY~ infvctt*tl  with j. SCerr pha.gr to turn on (‘rv synt hvsis. Ontb  minutr after induction (intivt ion). norfloxa(~in wxs  added to 0. IO or 50 pg/ml. The highthr Itivvl of’ clrug  tnaximizt~d gsrase inhibition Lvhile ht ill allowing  tletet%alJlr fvombination (Pig. 3). At t hti 1 imt5  intfitbat4. plasmid l)f\\\\‘A was isolattvl. nivketl with  1)Nasfa I. and visualiwd hy get t~t~~c~t,rol)~iorf~sis anal  Southrrn blotting.  f,it’tlrl or Ilo rrcombinatiori hid ot~t~urrrtl twti)rce  induction (Fig. 3. lanes I ant1 5). Trn minutw Iatw.  extensivr rrwtion t)y hot h rwomt~inasrs had take11  place in t hc ahsenw‘of. norfoxavin as evidenwd 1)).  the wnversion of’thr 7.3 kh suhstrattx to 3% kt) f’rc:r*  tGwlr produt~t (laws 2 untl 6). III tht, ~Jrvsvntv of’  IO /LLfr rlorffosi~t~itl~rnl. ttirrtb Ivits illI iI1t.rvasf~ in 1 tICI  rwover,v of siriglc liiikc~tl c3trnikrws ~~rc~clric~d t)J  resolvasr (lant~ 3). At 50 pL, (7 norfox;~t~irl:‘riil. 1111 to  7 2 “() of resolvase prtJ(luvts wert‘ t rapptvl ah singI>  linked c-atrnases (law 4). f II ctjnt rast (‘rti tnaclc~  on I\\\\. t.ra.c~t~ iImounts of’ tinkrti protlucts. Ill tllr~  at)srnvf> of’ tlorflosac~ir~ this t~jrrld hc tlufa to t trtx r;i.J)itl  dwafenat ion t)y I)&;\\\\ gyrase. Howt~\\\\-t&r. t’vtw itI t,ht  presence of’ inh’ilJit,or. (Ire formrtl utltinkvti I,roduc.ts  almost v.xc*lusivet~ (Fig. 3. lanes 7 atitl 8). /tt vit,rr,.  (‘rc. formed la.rpr amounts of singi!; linked c~atc~natlvs', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Inducl,an,ln,sslron RE.7 atE I I  “73 -  p,*RESCI -  L73 -  UT.3 -  L3-6 -  Figure 3. Topology of (‘re and resolvase products i?c  c~ioo. The dimrric substrate pKS-5, has pairs of both  W’S (optan arro\\\\rs) and 10x1’ sites (filled arrows) as direct  repeats. In cults c,orttaining l)KS-5. either (Ire or resotvasr  N as induc*rd and the t~atrnanes produced by each wert’  (*omparrti. Th(l c.;lrtoon summarizes the in Gno results.  and the clata gSrtrrratrtl in MC: t6,5.5.10 cells are drpichtrtl  below. Eac,h recombinase uas induced separately. and at  the times indic+atrti. plasmid DNA was isolated, nicked  wit,h I)Nasr 1. ;+tttl displayed by gel electrophoresis and  Southern blott.irtg. TAUIW I to 4. (‘ells harboring both  pKS-5 and ]).I BKEScl. Afifjr growth at 28’(“. cells were  shifted t,o 12 C’ to itrducae rc~tlvasr expression (time 0). At  1 mitt later. ttot,floxacain was added to 0. 10 or 50 pg/mt.  ,\\\\t 2 min. cells w(%rt’ diluted in the same medium at 30°C’  and plasmid I)SA was isolatjrd at, the times indicated.  I,anrs 5 to 8. (‘tall harboring pKS-5 only. At time 0. cells  wrre shift4 to 1P’(’ and inftac%rd with 1 X2-err phage. At  1 min later, drug was addrd. =\\\\t 2 min. cells were diluted  and 1)X.-\\\\ was isolated and l)rocessed as above. The posi-  tions of the substrat,e in its nicked circular (n7.3): linear  (17.3). anal supercoiled (~7.3) forms are indicated on the  left. h’irkrd ~~JKJIESS~I is detected due to cross-hgbridiza-  tion with the, full-lr%ngth pKS-5 probe. Reference ladders  were grnrratrd b?, t,rrating l)KS-5 in z&o with Crr using  superc~oitrd (lane Ml ) and relaxed (lane MM2) substrates.  i,anes M3 and M-1 caont,ain nit>ked Tn3 resolrase products  and T4 topoisomrrase IT knots, respectively. On the right  is indicatrtl thy trumber of nodes in the rat’enated and  knotted ntarktlr q)t4~s: 4 ttoded catenanes run slightI?-  slower than 1 noded knots. The hrmi-nicked catenanr  product (hemi). ~II whicah only 1 of the 2 rings in nicked.  results from inc.otnpletc nicktng by UNase I. n3.6 marks  the position of rtic,krd free (Grcle product. Lanes 9 and 9’.  Sorfloxac,irr was added t,o 50 pg/mt to uninfected cells.  I,anr 9’ is an ovf’t’(‘xposure of lane 9. =\\\\t the bottom is the  measured extent of re(.ombinat,ion and the percentage of  t.t,t,omhinittiotl proctut+s t)hat are krtotted or catenated  af’ter sttbtrac+ion of’ the bac*hground level of knotting. and more complex topological forms in addit,ion to  fiw product Grcles (Fig. 3. lanes JMl and MS).', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  results from inc.otnpletc nicktng by UNase I. n3.6 marks  the position of rtic,krd free (Grcle product. Lanes 9 and 9’.  Sorfloxac,irr was added t,o 50 pg/mt to uninfected cells.  I,anr 9’ is an ovf’t’(‘xposure of lane 9. =\\\\t the bottom is the  measured extent of re(.ombinat,ion and the percentage of  t.t,t,omhinittiotl proctut+s t)hat are krtotted or catenated  af’ter sttbtrac+ion of’ the bac*hground level of knotting. and more complex topological forms in addit,ion to  fiw product Grcles (Fig. 3. lanes JMl and MS). The  discarepancy between the in viva and i?/ vitro results  suggests that in E. coZi cells Cre carries out only a  subset of’ the reactions it performs in. ~itrr~.  The amount of recombination of pKS-5 was  markedly reduced by the addition of 50 pg/rnl  norfloxacin at one minute for both (Ire (Fig. 3. lane  8) and resolvase (Fig. 3, lane 4). Cnder our experi-  mental protocol. topoisomerase I relaxes DNA when  gyrase is inhibit,ed by norfloxacin. Since Tn3 resol-  vase requires a supercoiled substrate (Uliska rt (x,1.,  1991). recombination by this enzyme would  decrease with increasing relaxation of its substrate.  Clre recombination would likewise decrease if it t,oo  requires a supercoiled substrate in l’iV0.  -Alternatively. the addition of norfloxacin only one  minut,e after infection could have limited (!re  expression by reducing phage DNA replicat.ion and/  or transcription. To test this hypothesis we dela.yed  the ad&ion of norfloxacin until five minutes a.fter  infection to obtain a higher copy number of phages  and to allow less time for substrate plasmid relnxa-  tion (Lockshon & Morris, 1983). The result of the  later addition of drugs was a threefold increase in  C’re recombination but no increase in the arnount of  catenated product (data not shown).  Recombination was still about fourfold less than in  t,he absence of norfloxacin. This reduction is prob-  ably due to substrate relaxation, because in a topo-  isomerase 1 deficient strain. almost no inhibition of  recombination was found in the presence of norflox-  acin (see below).  Sorfloxacin addition trapped knotjted DrVA  species that are not resolvase or Cre products, since  they were produced in the absence of recombinase  intluc%ion (Fig. S, lane 9 and its ovt’resposure la,ne  9’). This incrrasts in the Iert>l of plasmid knotting is  similar to that seen in chertain gyr mutant st)rains  (Shishitlo rt 01.. 19X7) and in t.he present’t’ of other  inhibitors of’ gyrase (I&ii of cxl.. 19!jl ).  \\\\z’cl used t hr rwalts shown in Figure 3 to cBompa,rr  the extent of’ topological linkage of (‘re produc*t~s  made iI/ /‘ilW it,nd in vitro from a supercoiled  substrate. After normalizing for the same amount of  rec.otnhirratiorl. lve estimate t,hat (‘rc makes L-fold  fewer linked products irz /GPO than ir/ vitro. This is a  provisional comparison because of the diminution of  negative supereoiling of the substrate ittier gyrase  inhibition and the fact that (‘re in r?tro makes  predominantl) unlinked products u-it h relaxed  substrat.tJ* (lam‘ M2).  To estimate better the bias (“re exhibit*s fhr  forming unlinked products in V~VO.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  \\\\z’cl used t hr rwalts shown in Figure 3 to cBompa,rr  the extent of’ topological linkage of (‘re produc*t~s  made iI/ /‘ilW it,nd in vitro from a supercoiled  substrate. After normalizing for the same amount of  rec.otnhirratiorl. lve estimate t,hat (‘rc makes L-fold  fewer linked products irz /GPO than ir/ vitro. This is a  provisional comparison because of the diminution of  negative supereoiling of the substrate ittier gyrase  inhibition and the fact that (‘re in r?tro makes  predominantl) unlinked products u-it h relaxed  substrat.tJ* (lam‘ M2).  To estimate better the bias (“re exhibit*s fhr  forming unlinked products in V~VO. we examined the  dependence of (Jre product catenation and knotting  on substrate linking number. A series of topoisomers  of pKS-5 was prepared that differed in average  supercoil density. cr, from O*OO to -0V76. These  DX,;\\\\s were recombined by Cre in Gtro. and the', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Wi I). I<. :lflfrttln t,t al  tl I I I I I II  0 -0.02 -0~04-0.06 -0.08  Speclflc ,,nkmg difference (CT)  Cb)  Figure 4. The rtfrc+ of suMrate supercoilrtl density ott  thr t~opologicd complexity of produc*ts tnatlr Iry (‘re.  (a) Topoisomrrs of ~KS-T, w-ith the average supercoiling  clrnsitv (C-J) shown wart’ reactrd with (‘r~ under st~antlartl  c~ondit’ions. Approxitnxtrly Y!S”,, starting LIB,4 was supt.-  c~oiletl. Re~ombittiltion products were proc~rssrtl atttl  visualizrd RS desc~rihrd for Fig. 3. I,;tnes 111 ntttl M:!  c*ontaitt rrferrncv laclders of T1 topoisotnc~rasr II pKS-5  knots and Tn.7 rtdrasr pKS-5 protluc+s. rrsprctivrly.  Thv markers allo\\\\v assignrnt~nt of the ttutnltrr of cat,rnartt~  nodes in the (“re producb. Linear I)ICiA (17.3) results  front stopping the itr Gt,ro (Ye rractiotis with SDS. (Ir) Tlit~  atnount of linked products (catrnanes and knots) made I-J>  (Ire in /G/W uas quantified I)g tlrnsitotriPtr,v and is plottetl  (‘cr.s/1s substrate suytvoil drtisit>,.  reaction products were displayed by gel ele&ro-  phoresis after nicking with DNase I (Fig. 4(a)).  There was a simple linear relat,ionship between the  fraction of products that) are linked (i.e. catenated  or knotted) and the supercoil den&y of the  substrate. The extent of product linkage varied  from 15 “/;, for relaxed DNA to SSqi, for DNA with a  o of -0.076 (Fig. 4(b)). There was also an increase  in the average number of nodes of the products wit,h  increasing supercoiling (Fig. 4(a)). These more  complex species were present in roughly equal  amounts, as would be expect’ed for a substrate such  as pKS-5. whit+ has t,wo symmetrically placed  recombination sites that collide randomly  (Benjamin & Clozzarelli, 1986). At a o value (;f  -0025, which is the approximate effect’ive level of  supercoiling in viva (for a review, see McClellan rt  nl., 1990), 3094 of the products were linked; even  with a relaxed substrate 15% were linked. Both  values are far more than the 2o/0 linkage of the i7r  r~ivo products. We conclude that the lower super- tnorasv I mutant strain. (a) Strain Itrtlll c~c)trl;titt* it ‘I’tt //1  itisrrtion it1 top.4. (‘rv t,~c,c)ttt\\\\)tttitttott L\\\\ its itttluc,tatl its  &scribed for Fig. I+, exc.rpt that thus ittc.ttl)atiott tc~trrt)r~r’;r.  turn was shifted to 12 (’ IO ttritt prior to ittfectiotr to  triitnic thra Int ititiuc+iott c~otitlitiotis (SW Fig. (i). .\\\\t 5 triitl  after infection. portiotts of ttit, c~rtlturc~ wt’rt’ tlilutt~l in  tndiutti containing 0. 10 or 50 pg tiorfloxac~itt tt11. l’lastttitl  1)S.A wits isolatrtl at the tinttls itttiic~at.tvl atitl proc~t~ssccl its  bforr.', \"page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  I+, exc.rpt that thus ittc.ttl)atiott tc~trrt)r~r’;r.  turn was shifted to 12 (’ IO ttritt prior to ittfectiotr to  triitnic thra Int ititiuc+iott c~otitlitiotis (SW Fig. (i). .\\\\t 5 triitl  after infection. portiotts of ttit, c~rtlturc~ wt’rt’ tlilutt~l in  tndiutti containing 0. 10 or 50 pg tiorfloxac~itt tt11. l’lastttitl  1)S.A wits isolatrtl at the tinttls itttiic~at.tvl atitl proc~t~ssccl its  bforr. I,attr~s bl I ant1 .\\\\I? c.c)trt ain. rt~spc~t ii-vl,v. refrrrttc~c  latltlttrs of I)KS-5 (‘rcs l)rodric+h 1’rotti *ttl)er(~oilrtl ;ititl  rrlaxrtl suhstratrs. (II) ‘l’tic~ I)l;rstttitl kttcbttitig (IIW to  rrorfosacitt addition itt thrj ;rhsc~ttc~t~ of itrlivd iott is sho\\\\+tt.  Lanti M4 contains T4 to])oisotrt~~t~;ts~~ I I litlot%. ‘lb 1)X;\\\\ itt  grl (b) rati slightl,v slo\\\\vt~r t tiatr that itt pvl (8 t. tvsriltitig it1  an offssrt of thrs kttot i)ositiotlh  coiling in GVO after gyrase inhibit ion cannot explain  t*he simple t,opology of the (“re reaction produc+s in  IiZW.  ((3) C ‘rfd-tttdicrtfd rrsoltrfifttt in t 0pX E. wli  To bypass the (~xperirnt~ntal c.orrl~)lic,;rtic,tr ot’  substrate relaxation afitbr tiorfloxac*iri ;~tltlitioti. wo  used ittl /I;. roll strain tleficierit iti 1)X,-l topoiso-  rnrrase 1 ac%ivit\\\\-. This strain. Rrd41. (~otitains ii  TnZ11 insertion ’ in to/~.4 arid in c~otri~tt~risiito~~  rnut,ation in ~~yr.4 (Kaji Pt /I/.. l!M). In the prc~serlc~t~  of 10 pg norfloxa4n;ml. plasmid supf~rhelic*ai  density in t,his st,rairi rc%mained at a ('OtlStiillt itLv4  slightly higher t,han t’hat of wild-typt, E. co/i (Bliska  & (‘ozzarelli. 1!)87).  With this fossil strain, 50 pg norfloxa~in~tril  added five minut’es after infection had very lit)tle  effect OII (‘rc recombination (Fig. .?. lanes 3 a1~1 4).  A few minutes after drug addition. rec*ombinat iota  was extrrisivc, and resolution of pKS-5 was 6X”,, 01  that in the a~bsencc of drug. (‘rp rxhibitrtl at1 evetl  stronger bias for forming only unlinked producats in  the top.4 s;t rain compared with the top24 + st.rain  even though the substrate rema,ined highl~~ nt~gn~  tively supercoiled (Fig. 5, lanes 3 and 1).  A large background of’ plasmid knotting w~,s  again observed in the presence of 50 py riorfloxaciti,’  ml (Fig. 5. lanes 1 and i). The average number of\"], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:47 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'108'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994211'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'347ms'), (b'x-request-id', b'cacb6f46688d38930fd7d5e5e11b09f2'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124947588d5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:47 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'108'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994211'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'347ms'), (b'x-request-id', b'cacb6f46688d38930fd7d5e5e11b09f2'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124947588d5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  The Mechanism of Ore Recombination in Viva 667  *  135  4  6  -0-E 5 ‘5 5 k .A.  i? -0.02 d / E  -11 E -0.01 2 P :: 0.00  10 20 30 40 50  Figure 6. Products of i Int recombination irk ciro and  estimation of the effective level of superroiling in z’i~o.  (a) Effect of norfloxarin concentration in Go on the  number of nodes in Int product catenanes.  Recombinat,ion of pJB3.5d by Int was induced by an  upshift of Red4li. cells from 30°C to 42°C for 10 mm to  inartil-ate the ~1” repressor controlling int expression.  Dilution into 3OY medium initiated recombination.  Piorfloxacin was added to 0. 10 or 50 pg/ml at tempera-  ture downshift (time 0). and product catenane topology  was assayed at the indicated times. Lane Ml contains  p,JBS.Sd catenanrs made by Int, recombination in vitro.  On the left is indicated the positions of the substrate in its  nicked circular (113.5) and supercoiled (~~3.5) forms and of  the large free product circle in its nicked (n2.9) and  supercoiled (~2.9) forms. On the right is shown the  number of nodes in the catenated Int products and the  position of linear (lS.5) substrate. (b) The complexity of  Into product catenanes as a reporter of effective DKA  supercoiling. The mean number of catenane nodes at each  drug concentrat,ion was determined by densitometrg.  These averages were then used to calculate the level of  substrate supercoiling required i,n vitro to produce the  equivalent oatenane distributions (Roles et al.. 1990).  nodes in the knots was greater than that observed in  wild-t,ype cells (compare with lane 9’! Fig. 3). This  increase in knot c.omplexit’y may reflect the greater  plasmid DNA supercoiling in t’he topA- strain.  To control for the efficiency of decatenation  inhibitSion. we initially tried to compare Cre to resol-  vase recombination. but we were unable to maintain  both the resolvase expression vector and the pKS-5  substrate in the top.4 strain. We therefore switched  to t,hr Int system as a control, using a 1 lysogen that  produces a thermolabile repressor (cI857), which  regulat’es int expression from t’he ~1, promoter  (Rliska & (!ozzarelli. 1987). The plasmid substrate pJB3.51~ (Fig. 2(b)) was introduced into Red412  cells. and recombination was induced by a down-  shift of temperature from 42°C to 30°C.  Recombination by Int converts pJB35d into cate-  nated rings of 0.6 kb and 2.9 kb. In the absence of  norfloxacin. products were rapidly decatenated by  gyrase (Fig. 6(a) lane 2). However, in t’he presence  of the inhibit,or. catenated products were readily  trapped (Fig. 6(a), lanes 3 to 5) with an efficiency as  high as 76q&.  To quantify the bias Cre exhibits irl cillo for  producing unlinked products, we measured the  effective level of subst’rate supercoiling in IGO. This  effective level differs from the linking number of  isolated plasmid DXA due t,o the alternative COII-  formations that underwound DSA can adopt in Go  (Bliska & Cozzarelli, 1987). We ca,lculated the  effective level of supercoiling from the number of  nodes in the catenated Int product,s as explained  elsewhere (Bliska & Cozzarelli. 1987).', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  6(a) lane 2). However, in t’he presence  of the inhibit,or. catenated products were readily  trapped (Fig. 6(a), lanes 3 to 5) with an efficiency as  high as 76q&.  To quantify the bias Cre exhibits irl cillo for  producing unlinked products, we measured the  effective level of subst’rate supercoiling in IGO. This  effective level differs from the linking number of  isolated plasmid DXA due t,o the alternative COII-  formations that underwound DSA can adopt in Go  (Bliska & Cozzarelli, 1987). We ca,lculated the  effective level of supercoiling from the number of  nodes in the catenated Int product,s as explained  elsewhere (Bliska & Cozzarelli. 1987). The mean  numljer of Tnt catenane nodes trapped in the pre-  sence of 50 pg norfloxacin/ml was 4.6 in the topk  st.rain (Fig. 6(a)). This number corresponds to an  effect ire supercoiling level in cli~o of -0027  (Fig!. 6(b); Boles et al.: 1990), which is exact)ly the  value obtained previously with slightly less strin-  gent inhibition of gyrase (Bliska $ C?)zzarelli. 1987).  Our experimental system utilized A phage t’o  deliver the cry gene into cells. whereas the controls.  resolvase and Tnt. were produced from stably main-  tained genes e&her on a plasmid or integrated int,o  the E. coli chromosome. To verify that i infection  does not impair our ability to trap catenane inter-  locks. we cxarried out a control experiment) in which  Red411 cells harboring pJK35d were induced for  Tnt recombination and simultaneously infected wit’h  i, XL’-err phage. In the presence of norfloxacin. a  normal amount of Tnt catenancs was trapped: a  similar result was obtained in wild-type c*rlls wit,11  resolvasc> recombination (data not shown). Hence ;1  infect,ion does not stimulate decbatenation.  Taking into account t’he increased amount of C’re  recaornbination in these topA .. experiments, t’he cate-  nane trapping &iciency, and the measured effective  levrl of plasmid supercoiling, we caloulate that Cre  makes 20 to #I-fold fewer linked products in t’he  top,.4 st)rain c*ompared to in z+t~n. This is the same  range as our results in wild-typrb (ells.  Our results imply t’hat Cre follows rnainly an  ordered site synapsis pat’hway in r$i~o (Fig. 1). Tn3  resolvase follows such a pathway exclusively  (Wasserman & Cozzarelli. 1986) and normally  carries out. only intramolecular resolution reactions.  If analogous to Tn3 resolvase, Cre may be able to  sense the orient,ation of the 10x1’ sites on DNA and  carry out resolution in civo prrf(rrntially over  inversion or plasmid fusion.  To test this prediction. we examined the relative  rates elf C’re-mediated resolution. inversion. and', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  0 2 6 120  4 Time (mln) 4  Phage Cre Time (mln)  (Ire-mediated resolution. inversion and fusion. Tnversion  of pDA-5 (head-to-head arrows) was compared to resolu-  tion pKS-,5 (hea,d-to-tail arrows) and fusion of p:il .10x1’  (smaller ring with single arrow). (a) In rsirw rates of recom-  bination. topd * c&ells were separately transformed with  the 3 substrate plasmids and parallel cultures gro\\\\vn.  Prior t,o the phage addition. 2 cultures were mixed. those  with either pKS-5 and pDA-:i or p5l.loxI’ and pI)X-.>.  I’hagr was added at time 0. and plasmid DNA was  isolated at the indicated times. The extents of resolution.  inversion and fusion were quantified by comparison of the  amount of parental and recombinant DSA bands of either  supercoiled or BccmHT restrirted I)NL4. The data repw  sent, the averages of 2 or more independent experiments.  (b) III ,*itro rates of recombination. Supercwilrd substrates  pKS-5. pI).4-5 and p5l.loxl’ were reartrtl with purified  (‘w under standard reaction conditions. Portions of thus  rrac+ion were removed at the indicated times ant1 tht  reactions were quenched rapidly by direct addition to wld  phenol. Rwombination was quantified as desc*ribed  for (a).  fusion of loxP-containing plasmid substrates in r~iw  and in Gtro. Resolution of t)hr dimeric: 7.3 kb  substrate pKS-Tj was compared with fusion of its  3% kb monomeric form. p~l.10~1’ (Fig. 2(b)). To  assay inversion. we constructed a 6.9 kh substrate.  pl)A-5 (Fig. 2(h)); which harbors inverted lo.r/’ sites  separated by 3.~5 kb. For the in I+~YI experiments.  each sub&rate was individually t,ransformrd into  MC 1656.10 Veils. Prior t,o cell cboncent,ration and  phage infection. an cyual volume of pKS-5 cells and  pI>A-.5 cells. or pill .loxP cells and pI>A-5 cxells. were  mixed t.o control for variability in i adsorption and  (‘re expression. The kinetics and extent of pI)=\\\\-S  inversion were essentially identical in the wparatc  experiments, and the results presented are t>hc  average of t,he two assays. The result’s of quantita-  tive measures of resolution, inversion and fusion in  I+VO are illustrated by Figure 7(a).  Resolution of pKS-5 was both rapid and efic*ient  Aft’er an initial delay, recombination proceeded  quickly and by 30 minutes leveled off to approxi-  mat ely X0 ?(). Fusion of pfil.loxP reached only 3 to  lO(:h by ten minutes and fell thereafter: t#he fusion  reaction showed the same initial time delay as 4. Discussion  (a) In vitro versus in viva strdiw of rP/,otn(tin,nfiott  The conditjions for a t,ypical I’c’(‘OmtJillatiOll rrav  tion in vitr0 bear scant rewmhlan~r to t hca intra-  cellular environment of E. w/i. Itl t~iut. IJNX is  present at very high c.onc:~ntratio~Is. and it is  complexed with an array of binding proteins that  ran significantly alter its tertiary structuw. The  intracellular ionic strength is relatively high, and  includes a variety of polyvalent cations not usually  employed in clitro. All of t)hesr factors are known to  influence strongly recombinase activit’y (Pollock 8  Abremski.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Discussion  (a) In vitro versus in viva strdiw of rP/,otn(tin,nfiott  The conditjions for a t,ypical I’c’(‘OmtJillatiOll rrav  tion in vitr0 bear scant rewmhlan~r to t hca intra-  cellular environment of E. w/i. Itl t~iut. IJNX is  present at very high c.onc:~ntratio~Is. and it is  complexed with an array of binding proteins that  ran significantly alter its tertiary structuw. The  intracellular ionic strength is relatively high, and  includes a variety of polyvalent cations not usually  employed in clitro. All of t)hesr factors are known to  influence strongly recombinase activit’y (Pollock 8  Abremski. 1979; Stark ef nl.. 1989: Kanaar et nl..  1990). We have used tht, topologicaal properties of  substrates t’o drducae pat,h\\\\vays of rec,ornt)inatic,n  inside (~11s because the>- rrmain invariant in thra  absencde of subsequent breakage and reunion and are  effective reporters of rracat ion mechanism  (I%‘asserman & C’ozzarelli. 1986).  Tn view of the differencaes betwren caonditions in  the test-tube and in the intracellular milieu, it is  remarkable how similar t’hc, properties are in vitro  and in IV’ZV for i Int and Tnd rrdvasr. t I-w t,\\\\vo  systems previously studird by this (*omparativcJ  topological approach (Ktiska K- (‘ozzarrlli 19X7:  Kliska uf nl., 1991 ). First. Int and resolva,se both  reyuirr I)SA superwiling. Srcontl. the products of’  Tnt are catrnanrs or knots (very likely of th(l same  topological family) drpending. respec*tively. on  whether the sites are direci Iy or inversely rr[Jeiltrcl.  The products of resolvase. t)y (Lontrast. are prc‘-  dominantly singly linked c*atenanes. Evrn t h(x minor  products from processive resolvasr re(YJmhiWhO~l  appear to be the same in vim and in vitro. Third, lnt  readi]) recombines sii bst rat,es wit.h dirrcat 01  inverted sites, whereas rclsolrasch is sprcific for  directly repeated sites.  The results of our topological analysis of the  Cre-lox system in E. coli were decidedly different.  We used a phage-based induction scheme to  promote rapid Cre-mediated recombination of  plasmid substrates, and we isolated products within', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  The Mechanism of Cre Recombination in Vivo 669  minutes of their formation. Cre produced pre-  dominantly free circular products in viwo, unlike the  spectrum of products made in the in vitro reaction.  In uiuo. hut not in vitro, Cre showed a preference for  directly repeated sites and for negatively super-  coiled substra,tcks.  (t)) (‘onsiduration oj potential objections to  the intrrpretation of tire results  We obtained unlinked Cre products irk uico with  several substrates. concentrations of norfloxacin,  and tC:. coli gent&? backgrounds. Several objeet.ions  may be raised. hou-ever. to our interpretation of  thesrb rrsults. First, DP\\\\‘X gyrase or other topoiso-  merases might have rapidly untied Cre-linked  products to give the detected free circles. Second.  the substrate DXA may have been relaxed at the  time of rrcombinat.ion. and (Jre produces pre-  dominantly free circles with relaxed substrates in  ritro. Third. i infection itself, used to introduce the  I’TP gene into cells. may have altered t.he spectrum of  (Ire products generated in GPO. We next refute each  of these objections.  First. we rulrtl out Crr product decatenation b?  DXA gyrase. Il’e used extremely high levels of nor-  floxacin and demonstrated the resultant inhibi-  tion of gyrase by several means. In the topA+ E. coli  strain. we detrctrd positively supercoiled pBR322;  in the pres~~nc~ of 50 pg norfloxacin/ml. nearly all of  the I)SA became positively supercoiled. Positive  supercoils accumulates in Go after the generation of  positiveIT and negatively supercoiled domains due  to tracking ~~ro~sses such as transcription and  blockagr of the removal of positive supercoiling b)  yyrase (Liu & \\\\\\\\‘ang. 1987). We verified that decate-  nation by any rnzyme was inhibited. Singly linked  catenanes generated by Tn3 resolvase and multiply  linked cat)enanrs produced by A Tnt were trapped  with high e%ciclncy (about 750/b) when norfloxacin  was added prior to recombination (Figs 3 and 6(a)).  Also. WP detected an increased background level of  plasmid knotting following norfloxacin addition  (Figs 3 and 5).  The possibility that some other topoisomerase in  E. coli. such as t)opoisomcJrase TV; a putative type 2  topoisomerase (Kate et ril.. 1990; Lutt.inger et al.,  t 99 I). preferentially unties knots and catenanes of  lorP-c,orltainirtg substrattbs was also ruled out. With  a common subs&ate for both recombinases, pKS-5.  we trap a large amount of resolvase-generated cate-  nanes (up t 0 72?,,) but only trace amounts of  (‘re-genrrat.ed caatenanes (Figs 3 and 5). Despite the  use of high levels of norfloxacin, the majority of the  free product circles released by Cre were negatively  supercoiled in the topA- strain (data not shown).  We SW no evittenee for significant nicking of our Cre  substrates in GPO. The other known topoisomerases  in E. coli. topoisomerases I and ITT, cannot contri-  bute significsantly to the observed bias in viva for  free circle product,ion by Cre, since both require a  nick or gap in the DNA for decatenat’ion (Wang.  1985: I)iC;ate R- Jlarians. 1988). The above observations also help rule out the  second potential objection that the preponderance  of free circle products in the irk viva C?re reaction  result simply because the DNA subst.rat,e was  relaxed.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  We SW no evittenee for significant nicking of our Cre  substrates in GPO. The other known topoisomerases  in E. coli. topoisomerases I and ITT, cannot contri-  bute significsantly to the observed bias in viva for  free circle product,ion by Cre, since both require a  nick or gap in the DNA for decatenat’ion (Wang.  1985: I)iC;ate R- Jlarians. 1988). The above observations also help rule out the  second potential objection that the preponderance  of free circle products in the irk viva C?re reaction  result simply because the DNA subst.rat,e was  relaxed. /n L’ifro, unlinked circles are the major  product from a relaxed substrate (Figs 3 and 5: lane  M2). However. t,he substrate and product DNAs in  the topA- strain were still supercoiled after gyrase  inhibition. We calculated that the effective in zivo  supercoiling densit? for pKS-5 was -0.027 based on  the spectrum of Int catenanes generated in GPO  (Fig. 6). At t’his level of supercoiling, Clre in vitro  generat,ed approximat.ely 339i, linked products  (Fig. 4). This value differs sharply from the I “b  value we obtained in the topA- st’rain (Fig. 5).  The third potential objection is that j. infect.ion.  used to introduce t)he cre gene into ~11s. may stimu-  late decatenation or modify the recombination  substrat,es to favor a different spectrum of pr0duct.s.  WC. have excluded these possibilities. The efficiency  of trapping of Int-generated complex cat,rnanes and  resolvase-generated singly linked catenanes was  unaffect’ed by i infection. Yhage infection does tran-  sient,]? reclucr plasmid supercoiling. but only a sub-  population of plasmid was affec%ed (tlata not  shown). and the majoritS- of recombination occurred  onI\\\\ aftcAr negative supercoiling wax restored  (Fig. 5).  ((2) OthPt, indications of enhnncrd ( ‘rv  srlectioity in ciao  In addition to the difference in (Ire product topo-  logy. we discovered two other biases in (‘re behavior  in Go but not in 2itu1. In, &CO. (‘re prefers super-  coiled substrates and carries out resolution three- to  fivefold more efficiently than inversion. This  behavior is analogous to that of Tn3/$ rrsolvase. in  which these t,wo properties are known to be directly  coupled. Recombinat’ion by rrsolvase requires the  formation of a specific synaptic complex with t,hree  hyperwound negative supercoils and a parallel  recombination site alignment (Cl’asserman 8;  Cozzarelli, 19X6). Xegative supercoiling promotes  the formation of this complex if the sites are  directly repeated. but not if they aw in inverse  orientation.  The inhibition of recombinat,ion aft,er norfloxacin  addition in top4 + but not) topd cells implies that,  suprreoiling is important for t,he (‘r4o.z system in  I:~w. But it is also clear from the residual recom-  bination in the presence of norfloxacin that, super-  coiling is not absolutely required. This is in accord  with previous studies that established t,hat Clre can  recombine linear substrates in uic!o as deduced from  its ability to enhance the transformat.ion efficienc~y  of linear lozP-containing replicons in E. roli (Saner  bt Henderson. 1988; Hochman et nl.. 1983). Also. in  yeast and in higher eukarxotic c>ells, Cre can carr?  out genetic exchanges (Sauer. 1987; Sauer &  Henderson. 1990; Dale & Ow.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  But it is also clear from the residual recom-  bination in the presence of norfloxacin that, super-  coiling is not absolutely required. This is in accord  with previous studies that established t,hat Clre can  recombine linear substrates in uic!o as deduced from  its ability to enhance the transformat.ion efficienc~y  of linear lozP-containing replicons in E. roli (Saner  bt Henderson. 1988; Hochman et nl.. 1983). Also. in  yeast and in higher eukarxotic c>ells, Cre can carr?  out genetic exchanges (Sauer. 1987; Sauer &  Henderson. 1990; Dale & Ow. 1990) t’ven though the  bulk of DNA is likely not to be under superhelical  stress (Sinden rt nl., 1980). (Jre can also recombine', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  lii0 f). K. .-lfffLnl.s r1 al.  The hia,s (Ire exhibits against’ irivf7sion of’ plasniitl  suhstrat)es in E. roli (Fig. 7) may IN, grt,at’rr than  assayed here, as inversion could WCUI r>icr an altr>r-  native pathway involving intrrmolec~nla~l~ fusion and  s:utEw]lmt intramolecular resolution (Fig, 8).  t titrrmolec~ular fusion can c*rtaatr iill asvmmtlt rig.  dimrr that contains two sets of direct 1o.r.b rrptiats.  Ensuing resolution at the pair of sites that wercs not  ustkd lo create the dimrr results in two monon1(‘rs  \\\\vith site,-sprc~ific~ inversion (Fig. X). Siltcts  (‘rcx-mediated fusion and rrsolution o(‘curs i,, /*i,~,  (Fig. 7(a)). t)his trnns pathwav for invrrsion ma!’  (*ontrit)ute to the c>xt’ent 2 (‘rc>-mediated hilt’-  sptlcific inversion i,/ 13ir~. This pathway fol, itlvclrsiolr  is not predicted for strict resolvases such as asyn-  metric Tn3 resolvase, which do not efficiently c.arr>  out fusion. Indeed. resotvase-mediat,ed inversion ir/  c:itro and in Go is barely detectable (Kliska d 01..  1991 ).  The irr vitv I)r~~ferPnc~r for (‘rt’ for rrc~omhining  direct-sit,e substrates over inverse-sit<, suljstrates  may he rationalized on funct~ional grountls. The  dirrct repeat orientation is found for lo.,!’ sites in 1’1  c*ircdutar dimers generated by hotnoloyous r~~(*on~  bination. Moreover. PI T)K\\\\‘A is t.erminaIly rrdurl~  dant and cycalically permuted so that ontl-quarter to  onr-third of t.he phage DNA has two /MI’ sites in  direct repeat (\\\\‘armolinsk\\\\- & Strrnhr*rg. I 9X8).  Specniticit3y for resolution is also c*ollsisttsrrt with thr  proposed Alular rote for (‘re of helping to maintain  pla,smid caopv llumhrr 1)~ rrtsolving 1’1 dimers.  However. neither this rok nor the c+vclization of  nrn-I)- injected linear phage t>X.A is carit ical for thr  phage. The ends of PI phapp I)NX c,ontain a I;irgcl  duplication and are readily cayc4izcd in the ahsenc~r  of (‘re function. and 70 to i5”,, of’ th phapc) t)X;\\\\  has only a single /o.zl’ site (Yarrnolinsky Cy-  StcArnherg. 198X).  LVr sought to identify a more active role for sit(l-  specific resolution in the maint,enance of plasmid  copy number. Tt is extraordinary that thtb unit c~)p~  J’l plasmid is lost from fewer than 1 in 105 cells pet’  generation (\\\\;armolinsky bz Sternberg. 198X). Thrl  (‘rr-1o.r system may help stabilize Pl ~op,v number  by mediating limited plasmid amplification. OUI  proposal (Fig. 9) generalizes the Fut’cher model fo1  the role of Flp-mediated I)KA inversion in ,vrast  1 pm plasmid amplification (Futcher. t9XX). The  st,arting point in the Futcher model is a monomer  2 pm czirc+ with two inverted sites for Flp: replica-  tion of one of the sites creates a molec~ulr with threr  rec*ombinat~ion sites.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Tt is extraordinary that thtb unit c~)p~  J’l plasmid is lost from fewer than 1 in 105 cells pet’  generation (\\\\;armolinsky bz Sternberg. 198X). Thrl  (‘rr-1o.r system may help stabilize Pl ~op,v number  by mediating limited plasmid amplification. OUI  proposal (Fig. 9) generalizes the Fut’cher model fo1  the role of Flp-mediated I)KA inversion in ,vrast  1 pm plasmid amplification (Futcher. t9XX). The  st,arting point in the Futcher model is a monomer  2 pm czirc+ with two inverted sites for Flp: replica-  tion of one of the sites creates a molec~ulr with threr  rec*ombinat~ion sites. Flp-mediated inversion gener-  at)es a rolling circle wit,h replication forks cahasing  eacsh ot’her around the plasmid. resuhinp in 1)X=\\\\ B  c) B +  B  0  + B  B  0  (aoncataniers a,nd plasrnitl amplification. A s;c3,olitl  round of b’lp recombination rcdirecats the replicat iotr  forks so that they mrrgp and t~erminatt~ l)Sr\\\\ ampli-  fication. Support’ for this model has been ohtainrtl in  the I pm plasmid system (Volkrrt & broach. 1986:  Fut)cher. 1988).  To creat,tl the analogous starting structurck for 1’1.  a ropticat,ing PI piasmid dimrr rnust hr generated in  which only one of’ the lolp sites is tiuplic~ated  (Fig. 9) yielding a moleclule with three IozI’ repeats  (step 1). This tnole~lule c,ould he formed either I-1)', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  cation. After completion of DKA synthesis.  (Ire-mediated recombination within the amplified  regions would generate complete plasmid monomers  (step 5).  5  -4  It is not known whether the incwaw iir plasmid  knotting (Figs 3 and 5) following norflosacin addi-  tion is arl alwrtatlt. rraction cw+tl out t)y $1  poisotictl I)S&l g>.rasc. a wrtsquf’tiw of thts attJetltl-  ant Jwsitivtx supr~rcoiling of thrt plasmiti. or the  result of hlocbking t hr normal rott~ of g>.rase of  uttt\\\\.irtp kttots made by some ot twr enzyme. A  similar pattern and frequencyc of plasmid knotting  was ohservrd by Shishido rt cl/. (1987) in cbertain  E. cf,/i y!yr mutant strains and by lshii 01 01. (19!)1)  using other gyraw inhibitors.  (‘w or t)v grnt~ral rwombination. (‘rca-rrtediatc~d  resolution hetwwn the unduplicated repeat’ and  ;I nelvlv wplic.atvtl retwat will result in thv fortna-  tiotl ot: a ti~lrttt,t)clt-shal)~cl molecule. \\\\vhicah \\\\vill t)(s  eit hrr it single or tlouble rolling circle. depending on  whtlt her r~~ltlic~ation is uniditwtiona1 or hittirec+iorral  f~roni t ttcs siti+ ac4 i\\\\-tl replication origin (step 2).  I’tticlirrv~t ionit wplicwtion is shown in the diagram.  13aseti on out’ rwults. t hc Iwo rings of tht, dumbbell  should 1~8 wit hotit topologicaal interlinks. The orien-  t,ation of’ t.he replic:ation forks in the dumbbell inter-  mediatr prevents fork collision, and continued DNA  synthrsis and ptasmid amplification results (step 3).  Subsequent (‘t-e- mediated recombinat,ion between  the ring?; a:ill redirect. the replication forks and  complete l,he started round of DKA synthesis (step  4). This is caritic*;tl to limiting the extent of amplifi-'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  The Mechanism of Ore Recombination in Viva 667  *  135  4  6  -0-E 5 ‘5 5 k .A.  i? -0.02 d / E  -11 E -0.01 2 P :: 0.00  10 20 30 40 50  Figure 6. Products of i Int recombination irk ciro and  estimation of the effective level of superroiling in z’i~o.  (a) Effect of norfloxarin concentration in Go on the  number of nodes in Int product catenanes.  Recombinat,ion of pJB3.5d by Int was induced by an  upshift of Red4li. cells from 30°C to 42°C for 10 mm to  inartil-ate the ~1” repressor controlling int expression.  Dilution into 3OY medium initiated recombination.  Piorfloxacin was added to 0. 10 or 50 pg/ml at tempera-  ture downshift (time 0). and product catenane topology  was assayed at the indicated times. Lane Ml contains  p,JBS.Sd catenanrs made by Int, recombination in vitro.  On the left is indicated the positions of the substrate in its  nicked circular (113.5) and supercoiled (~~3.5) forms and of  the large free product circle in its nicked (n2.9) and  supercoiled (~2.9) forms. On the right is shown the  number of nodes in the catenated Int products and the  position of linear (lS.5) substrate. (b) The complexity of  Into product catenanes as a reporter of effective DKA  supercoiling. The mean number of catenane nodes at each  drug concentrat,ion was determined by densitometrg.  These averages were then used to calculate the level of  substrate supercoiling required i,n vitro to produce the  equivalent oatenane distributions (Roles et al.. 1990).  nodes in the knots was greater than that observed in  wild-t,ype cells (compare with lane 9’! Fig. 3). This  increase in knot c.omplexit’y may reflect the greater  plasmid DNA supercoiling in t’he topA- strain.  To control for the efficiency of decatenation  inhibitSion. we initially tried to compare Cre to resol-  vase recombination. but we were unable to maintain  both the resolvase expression vector and the pKS-5  substrate in the top.4 strain. We therefore switched  to t,hr Int system as a control, using a 1 lysogen that  produces a thermolabile repressor (cI857), which  regulat’es int expression from t’he ~1, promoter  (Rliska & (!ozzarelli. 1987). The plasmid substrate pJB3.51~ (Fig. 2(b)) was introduced into Red412  cells. and recombination was induced by a down-  shift of temperature from 42°C to 30°C.  Recombination by Int converts pJB35d into cate-  nated rings of 0.6 kb and 2.9 kb. In the absence of  norfloxacin. products were rapidly decatenated by  gyrase (Fig. 6(a) lane 2). However, in t’he presence  of the inhibit,or. catenated products were readily  trapped (Fig. 6(a), lanes 3 to 5) with an efficiency as  high as 76q&.  To quantify the bias Cre exhibits irl cillo for  producing unlinked products, we measured the  effective level of subst’rate supercoiling in IGO. This  effective level differs from the linking number of  isolated plasmid DXA due t,o the alternative COII-  formations that underwound DSA can adopt in Go  (Bliska & Cozzarelli, 1987). We ca,lculated the  effective level of supercoiling from the number of  nodes in the catenated Int product,s as explained  elsewhere (Bliska & Cozzarelli. 1987).', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  6(a) lane 2). However, in t’he presence  of the inhibit,or. catenated products were readily  trapped (Fig. 6(a), lanes 3 to 5) with an efficiency as  high as 76q&.  To quantify the bias Cre exhibits irl cillo for  producing unlinked products, we measured the  effective level of subst’rate supercoiling in IGO. This  effective level differs from the linking number of  isolated plasmid DXA due t,o the alternative COII-  formations that underwound DSA can adopt in Go  (Bliska & Cozzarelli, 1987). We ca,lculated the  effective level of supercoiling from the number of  nodes in the catenated Int product,s as explained  elsewhere (Bliska & Cozzarelli. 1987). The mean  numljer of Tnt catenane nodes trapped in the pre-  sence of 50 pg norfloxacin/ml was 4.6 in the topk  st.rain (Fig. 6(a)). This number corresponds to an  effect ire supercoiling level in cli~o of -0027  (Fig!. 6(b); Boles et al.: 1990), which is exact)ly the  value obtained previously with slightly less strin-  gent inhibition of gyrase (Bliska $ C?)zzarelli. 1987).  Our experimental system utilized A phage t’o  deliver the cry gene into cells. whereas the controls.  resolvase and Tnt. were produced from stably main-  tained genes e&her on a plasmid or integrated int,o  the E. coli chromosome. To verify that i infection  does not impair our ability to trap catenane inter-  locks. we cxarried out a control experiment) in which  Red411 cells harboring pJK35d were induced for  Tnt recombination and simultaneously infected wit’h  i, XL’-err phage. In the presence of norfloxacin. a  normal amount of Tnt catenancs was trapped: a  similar result was obtained in wild-type c*rlls wit,11  resolvasc> recombination (data not shown). Hence ;1  infect,ion does not stimulate decbatenation.  Taking into account t’he increased amount of C’re  recaornbination in these topA .. experiments, t’he cate-  nane trapping &iciency, and the measured effective  levrl of plasmid supercoiling, we caloulate that Cre  makes 20 to #I-fold fewer linked products in t’he  top,.4 st)rain c*ompared to in z+t~n. This is the same  range as our results in wild-typrb (ells.  Our results imply t’hat Cre follows rnainly an  ordered site synapsis pat’hway in r$i~o (Fig. 1). Tn3  resolvase follows such a pathway exclusively  (Wasserman & Cozzarelli. 1986) and normally  carries out. only intramolecular resolution reactions.  If analogous to Tn3 resolvase, Cre may be able to  sense the orient,ation of the 10x1’ sites on DNA and  carry out resolution in civo prrf(rrntially over  inversion or plasmid fusion.  To test this prediction. we examined the relative  rates elf C’re-mediated resolution. inversion. and', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  0 2 6 120  4 Time (mln) 4  Phage Cre Time (mln)  (Ire-mediated resolution. inversion and fusion. Tnversion  of pDA-5 (head-to-head arrows) was compared to resolu-  tion pKS-,5 (hea,d-to-tail arrows) and fusion of p:il .10x1’  (smaller ring with single arrow). (a) In rsirw rates of recom-  bination. topd * c&ells were separately transformed with  the 3 substrate plasmids and parallel cultures gro\\\\vn.  Prior t,o the phage addition. 2 cultures were mixed. those  with either pKS-5 and pDA-:i or p5l.loxI’ and pI)X-.>.  I’hagr was added at time 0. and plasmid DNA was  isolated at the indicated times. The extents of resolution.  inversion and fusion were quantified by comparison of the  amount of parental and recombinant DSA bands of either  supercoiled or BccmHT restrirted I)NL4. The data repw  sent, the averages of 2 or more independent experiments.  (b) III ,*itro rates of recombination. Supercwilrd substrates  pKS-5. pI).4-5 and p5l.loxl’ were reartrtl with purified  (‘w under standard reaction conditions. Portions of thus  rrac+ion were removed at the indicated times ant1 tht  reactions were quenched rapidly by direct addition to wld  phenol. Rwombination was quantified as desc*ribed  for (a).  fusion of loxP-containing plasmid substrates in r~iw  and in Gtro. Resolution of t)hr dimeric: 7.3 kb  substrate pKS-Tj was compared with fusion of its  3% kb monomeric form. p~l.10~1’ (Fig. 2(b)). To  assay inversion. we constructed a 6.9 kh substrate.  pl)A-5 (Fig. 2(h)); which harbors inverted lo.r/’ sites  separated by 3.~5 kb. For the in I+~YI experiments.  each sub&rate was individually t,ransformrd into  MC 1656.10 Veils. Prior t,o cell cboncent,ration and  phage infection. an cyual volume of pKS-5 cells and  pI>A-.5 cells. or pill .loxP cells and pI>A-5 cxells. were  mixed t.o control for variability in i adsorption and  (‘re expression. The kinetics and extent of pI)=\\\\-S  inversion were essentially identical in the wparatc  experiments, and the results presented are t>hc  average of t,he two assays. The result’s of quantita-  tive measures of resolution, inversion and fusion in  I+VO are illustrated by Figure 7(a).  Resolution of pKS-5 was both rapid and efic*ient  Aft’er an initial delay, recombination proceeded  quickly and by 30 minutes leveled off to approxi-  mat ely X0 ?(). Fusion of pfil.loxP reached only 3 to  lO(:h by ten minutes and fell thereafter: t#he fusion  reaction showed the same initial time delay as 4. Discussion  (a) In vitro versus in viva strdiw of rP/,otn(tin,nfiott  The conditjions for a t,ypical I’c’(‘OmtJillatiOll rrav  tion in vitr0 bear scant rewmhlan~r to t hca intra-  cellular environment of E. w/i. Itl t~iut. IJNX is  present at very high c.onc:~ntratio~Is. and it is  complexed with an array of binding proteins that  ran significantly alter its tertiary structuw. The  intracellular ionic strength is relatively high, and  includes a variety of polyvalent cations not usually  employed in clitro. All of t)hesr factors are known to  influence strongly recombinase activit’y (Pollock 8  Abremski.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Discussion  (a) In vitro versus in viva strdiw of rP/,otn(tin,nfiott  The conditjions for a t,ypical I’c’(‘OmtJillatiOll rrav  tion in vitr0 bear scant rewmhlan~r to t hca intra-  cellular environment of E. w/i. Itl t~iut. IJNX is  present at very high c.onc:~ntratio~Is. and it is  complexed with an array of binding proteins that  ran significantly alter its tertiary structuw. The  intracellular ionic strength is relatively high, and  includes a variety of polyvalent cations not usually  employed in clitro. All of t)hesr factors are known to  influence strongly recombinase activit’y (Pollock 8  Abremski. 1979; Stark ef nl.. 1989: Kanaar et nl..  1990). We have used tht, topologicaal properties of  substrates t’o drducae pat,h\\\\vays of rec,ornt)inatic,n  inside (~11s because the>- rrmain invariant in thra  absencde of subsequent breakage and reunion and are  effective reporters of rracat ion mechanism  (I%‘asserman & C’ozzarelli. 1986).  Tn view of the differencaes betwren caonditions in  the test-tube and in the intracellular milieu, it is  remarkable how similar t’hc, properties are in vitro  and in IV’ZV for i Int and Tnd rrdvasr. t I-w t,\\\\vo  systems previously studird by this (*omparativcJ  topological approach (Ktiska K- (‘ozzarrlli 19X7:  Kliska uf nl., 1991 ). First. Int and resolva,se both  reyuirr I)SA superwiling. Srcontl. the products of’  Tnt are catrnanrs or knots (very likely of th(l same  topological family) drpending. respec*tively. on  whether the sites are direci Iy or inversely rr[Jeiltrcl.  The products of resolvase. t)y (Lontrast. are prc‘-  dominantly singly linked c*atenanes. Evrn t h(x minor  products from processive resolvasr re(YJmhiWhO~l  appear to be the same in vim and in vitro. Third, lnt  readi]) recombines sii bst rat,es wit.h dirrcat 01  inverted sites, whereas rclsolrasch is sprcific for  directly repeated sites.  The results of our topological analysis of the  Cre-lox system in E. coli were decidedly different.  We used a phage-based induction scheme to  promote rapid Cre-mediated recombination of  plasmid substrates, and we isolated products within', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  The Mechanism of Cre Recombination in Vivo 669  minutes of their formation. Cre produced pre-  dominantly free circular products in viwo, unlike the  spectrum of products made in the in vitro reaction.  In uiuo. hut not in vitro, Cre showed a preference for  directly repeated sites and for negatively super-  coiled substra,tcks.  (t)) (‘onsiduration oj potential objections to  the intrrpretation of tire results  We obtained unlinked Cre products irk uico with  several substrates. concentrations of norfloxacin,  and tC:. coli gent&? backgrounds. Several objeet.ions  may be raised. hou-ever. to our interpretation of  thesrb rrsults. First, DP\\\\‘X gyrase or other topoiso-  merases might have rapidly untied Cre-linked  products to give the detected free circles. Second.  the substrate DXA may have been relaxed at the  time of rrcombinat.ion. and (Jre produces pre-  dominantly free circles with relaxed substrates in  ritro. Third. i infection itself, used to introduce the  I’TP gene into cells. may have altered t.he spectrum of  (Ire products generated in GPO. We next refute each  of these objections.  First. we rulrtl out Crr product decatenation b?  DXA gyrase. Il’e used extremely high levels of nor-  floxacin and demonstrated the resultant inhibi-  tion of gyrase by several means. In the topA+ E. coli  strain. we detrctrd positively supercoiled pBR322;  in the pres~~nc~ of 50 pg norfloxacin/ml. nearly all of  the I)SA became positively supercoiled. Positive  supercoils accumulates in Go after the generation of  positiveIT and negatively supercoiled domains due  to tracking ~~ro~sses such as transcription and  blockagr of the removal of positive supercoiling b)  yyrase (Liu & \\\\\\\\‘ang. 1987). We verified that decate-  nation by any rnzyme was inhibited. Singly linked  catenanes generated by Tn3 resolvase and multiply  linked cat)enanrs produced by A Tnt were trapped  with high e%ciclncy (about 750/b) when norfloxacin  was added prior to recombination (Figs 3 and 6(a)).  Also. WP detected an increased background level of  plasmid knotting following norfloxacin addition  (Figs 3 and 5).  The possibility that some other topoisomerase in  E. coli. such as t)opoisomcJrase TV; a putative type 2  topoisomerase (Kate et ril.. 1990; Lutt.inger et al.,  t 99 I). preferentially unties knots and catenanes of  lorP-c,orltainirtg substrattbs was also ruled out. With  a common subs&ate for both recombinases, pKS-5.  we trap a large amount of resolvase-generated cate-  nanes (up t 0 72?,,) but only trace amounts of  (‘re-genrrat.ed caatenanes (Figs 3 and 5). Despite the  use of high levels of norfloxacin, the majority of the  free product circles released by Cre were negatively  supercoiled in the topA- strain (data not shown).  We SW no evittenee for significant nicking of our Cre  substrates in GPO. The other known topoisomerases  in E. coli. topoisomerases I and ITT, cannot contri-  bute significsantly to the observed bias in viva for  free circle product,ion by Cre, since both require a  nick or gap in the DNA for decatenat’ion (Wang.  1985: I)iC;ate R- Jlarians. 1988). The above observations also help rule out the  second potential objection that the preponderance  of free circle products in the irk viva C?re reaction  result simply because the DNA subst.rat,e was  relaxed.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  We SW no evittenee for significant nicking of our Cre  substrates in GPO. The other known topoisomerases  in E. coli. topoisomerases I and ITT, cannot contri-  bute significsantly to the observed bias in viva for  free circle product,ion by Cre, since both require a  nick or gap in the DNA for decatenat’ion (Wang.  1985: I)iC;ate R- Jlarians. 1988). The above observations also help rule out the  second potential objection that the preponderance  of free circle products in the irk viva C?re reaction  result simply because the DNA subst.rat,e was  relaxed. /n L’ifro, unlinked circles are the major  product from a relaxed substrate (Figs 3 and 5: lane  M2). However. t,he substrate and product DNAs in  the topA- strain were still supercoiled after gyrase  inhibition. We calculated that the effective in zivo  supercoiling densit? for pKS-5 was -0.027 based on  the spectrum of Int catenanes generated in GPO  (Fig. 6). At t’his level of supercoiling, Clre in vitro  generat,ed approximat.ely 339i, linked products  (Fig. 4). This value differs sharply from the I “b  value we obtained in the topA- st’rain (Fig. 5).  The third potential objection is that j. infect.ion.  used to introduce t)he cre gene into ~11s. may stimu-  late decatenation or modify the recombination  substrat,es to favor a different spectrum of pr0duct.s.  WC. have excluded these possibilities. The efficiency  of trapping of Int-generated complex cat,rnanes and  resolvase-generated singly linked catenanes was  unaffect’ed by i infection. Yhage infection does tran-  sient,]? reclucr plasmid supercoiling. but only a sub-  population of plasmid was affec%ed (tlata not  shown). and the majoritS- of recombination occurred  onI\\\\ aftcAr negative supercoiling wax restored  (Fig. 5).  ((2) OthPt, indications of enhnncrd ( ‘rv  srlectioity in ciao  In addition to the difference in (Ire product topo-  logy. we discovered two other biases in (‘re behavior  in Go but not in 2itu1. In, &CO. (‘re prefers super-  coiled substrates and carries out resolution three- to  fivefold more efficiently than inversion. This  behavior is analogous to that of Tn3/$ rrsolvase. in  which these t,wo properties are known to be directly  coupled. Recombinat’ion by rrsolvase requires the  formation of a specific synaptic complex with t,hree  hyperwound negative supercoils and a parallel  recombination site alignment (Cl’asserman 8;  Cozzarelli, 19X6). Xegative supercoiling promotes  the formation of this complex if the sites are  directly repeated. but not if they aw in inverse  orientation.  The inhibition of recombinat,ion aft,er norfloxacin  addition in top4 + but not) topd cells implies that,  suprreoiling is important for t,he (‘r4o.z system in  I:~w. But it is also clear from the residual recom-  bination in the presence of norfloxacin that, super-  coiling is not absolutely required. This is in accord  with previous studies that established t,hat Clre can  recombine linear substrates in uic!o as deduced from  its ability to enhance the transformat.ion efficienc~y  of linear lozP-containing replicons in E. roli (Saner  bt Henderson. 1988; Hochman et nl.. 1983). Also. in  yeast and in higher eukarxotic c>ells, Cre can carr?  out genetic exchanges (Sauer. 1987; Sauer &  Henderson. 1990; Dale & Ow.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  But it is also clear from the residual recom-  bination in the presence of norfloxacin that, super-  coiling is not absolutely required. This is in accord  with previous studies that established t,hat Clre can  recombine linear substrates in uic!o as deduced from  its ability to enhance the transformat.ion efficienc~y  of linear lozP-containing replicons in E. roli (Saner  bt Henderson. 1988; Hochman et nl.. 1983). Also. in  yeast and in higher eukarxotic c>ells, Cre can carr?  out genetic exchanges (Sauer. 1987; Sauer &  Henderson. 1990; Dale & Ow. 1990) t’ven though the  bulk of DNA is likely not to be under superhelical  stress (Sinden rt nl., 1980). (Jre can also recombine', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  lii0 f). K. .-lfffLnl.s r1 al.  The hia,s (Ire exhibits against’ irivf7sion of’ plasniitl  suhstrat)es in E. roli (Fig. 7) may IN, grt,at’rr than  assayed here, as inversion could WCUI r>icr an altr>r-  native pathway involving intrrmolec~nla~l~ fusion and  s:utEw]lmt intramolecular resolution (Fig, 8).  t titrrmolec~ular fusion can c*rtaatr iill asvmmtlt rig.  dimrr that contains two sets of direct 1o.r.b rrptiats.  Ensuing resolution at the pair of sites that wercs not  ustkd lo create the dimrr results in two monon1(‘rs  \\\\vith site,-sprc~ific~ inversion (Fig. X). Siltcts  (‘rcx-mediated fusion and rrsolution o(‘curs i,, /*i,~,  (Fig. 7(a)). t)his trnns pathwav for invrrsion ma!’  (*ontrit)ute to the c>xt’ent 2 (‘rc>-mediated hilt’-  sptlcific inversion i,/ 13ir~. This pathway fol, itlvclrsiolr  is not predicted for strict resolvases such as asyn-  metric Tn3 resolvase, which do not efficiently c.arr>  out fusion. Indeed. resotvase-mediat,ed inversion ir/  c:itro and in Go is barely detectable (Kliska d 01..  1991 ).  The irr vitv I)r~~ferPnc~r for (‘rt’ for rrc~omhining  direct-sit,e substrates over inverse-sit<, suljstrates  may he rationalized on funct~ional grountls. The  dirrct repeat orientation is found for lo.,!’ sites in 1’1  c*ircdutar dimers generated by hotnoloyous r~~(*on~  bination. Moreover. PI T)K\\\\‘A is t.erminaIly rrdurl~  dant and cycalically permuted so that ontl-quarter to  onr-third of t.he phage DNA has two /MI’ sites in  direct repeat (\\\\‘armolinsk\\\\- & Strrnhr*rg. I 9X8).  Specniticit3y for resolution is also c*ollsisttsrrt with thr  proposed Alular rote for (‘re of helping to maintain  pla,smid caopv llumhrr 1)~ rrtsolving 1’1 dimers.  However. neither this rok nor the c+vclization of  nrn-I)- injected linear phage t>X.A is carit ical for thr  phage. The ends of PI phapp I)NX c,ontain a I;irgcl  duplication and are readily cayc4izcd in the ahsenc~r  of (‘re function. and 70 to i5”,, of’ th phapc) t)X;\\\\  has only a single /o.zl’ site (Yarrnolinsky Cy-  StcArnherg. 198X).  LVr sought to identify a more active role for sit(l-  specific resolution in the maint,enance of plasmid  copy number. Tt is extraordinary that thtb unit c~)p~  J’l plasmid is lost from fewer than 1 in 105 cells pet’  generation (\\\\;armolinsky bz Sternberg. 198X). Thrl  (‘rr-1o.r system may help stabilize Pl ~op,v number  by mediating limited plasmid amplification. OUI  proposal (Fig. 9) generalizes the Fut’cher model fo1  the role of Flp-mediated I)KA inversion in ,vrast  1 pm plasmid amplification (Futcher. t9XX). The  st,arting point in the Futcher model is a monomer  2 pm czirc+ with two inverted sites for Flp: replica-  tion of one of the sites creates a molec~ulr with threr  rec*ombinat~ion sites.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  Tt is extraordinary that thtb unit c~)p~  J’l plasmid is lost from fewer than 1 in 105 cells pet’  generation (\\\\;armolinsky bz Sternberg. 198X). Thrl  (‘rr-1o.r system may help stabilize Pl ~op,v number  by mediating limited plasmid amplification. OUI  proposal (Fig. 9) generalizes the Fut’cher model fo1  the role of Flp-mediated I)KA inversion in ,vrast  1 pm plasmid amplification (Futcher. t9XX). The  st,arting point in the Futcher model is a monomer  2 pm czirc+ with two inverted sites for Flp: replica-  tion of one of the sites creates a molec~ulr with threr  rec*ombinat~ion sites. Flp-mediated inversion gener-  at)es a rolling circle wit,h replication forks cahasing  eacsh ot’her around the plasmid. resuhinp in 1)X=\\\\ B  c) B +  B  0  + B  B  0  (aoncataniers a,nd plasrnitl amplification. A s;c3,olitl  round of b’lp recombination rcdirecats the replicat iotr  forks so that they mrrgp and t~erminatt~ l)Sr\\\\ ampli-  fication. Support’ for this model has been ohtainrtl in  the I pm plasmid system (Volkrrt & broach. 1986:  Fut)cher. 1988).  To creat,tl the analogous starting structurck for 1’1.  a ropticat,ing PI piasmid dimrr rnust hr generated in  which only one of’ the lolp sites is tiuplic~ated  (Fig. 9) yielding a moleclule with three IozI’ repeats  (step 1). This tnole~lule c,ould he formed either I-1)', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  cation. After completion of DKA synthesis.  (Ire-mediated recombination within the amplified  regions would generate complete plasmid monomers  (step 5).  5  -4  It is not known whether the incwaw iir plasmid  knotting (Figs 3 and 5) following norflosacin addi-  tion is arl alwrtatlt. rraction cw+tl out t)y $1  poisotictl I)S&l g>.rasc. a wrtsquf’tiw of thts attJetltl-  ant Jwsitivtx supr~rcoiling of thrt plasmiti. or the  result of hlocbking t hr normal rott~ of g>.rase of  uttt\\\\.irtp kttots made by some ot twr enzyme. A  similar pattern and frequencyc of plasmid knotting  was ohservrd by Shishido rt cl/. (1987) in cbertain  E. cf,/i y!yr mutant strains and by lshii 01 01. (19!)1)  using other gyraw inhibitors.  (‘w or t)v grnt~ral rwombination. (‘rca-rrtediatc~d  resolution hetwwn the unduplicated repeat’ and  ;I nelvlv wplic.atvtl retwat will result in thv fortna-  tiotl ot: a ti~lrttt,t)clt-shal)~cl molecule. \\\\vhicah \\\\vill t)(s  eit hrr it single or tlouble rolling circle. depending on  whtlt her r~~ltlic~ation is uniditwtiona1 or hittirec+iorral  f~roni t ttcs siti+ ac4 i\\\\-tl replication origin (step 2).  I’tticlirrv~t ionit wplicwtion is shown in the diagram.  13aseti on out’ rwults. t hc Iwo rings of tht, dumbbell  should 1~8 wit hotit topologicaal interlinks. The orien-  t,ation of’ t.he replic:ation forks in the dumbbell inter-  mediatr prevents fork collision, and continued DNA  synthrsis and ptasmid amplification results (step 3).  Subsequent (‘t-e- mediated recombinat,ion between  the ring?; a:ill redirect. the replication forks and  complete l,he started round of DKA synthesis (step  4). This is caritic*;tl to limiting the extent of amplifi-'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:48 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'77'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993680'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'379ms'), (b'x-request-id', b'bb3ec0cb581105f364a26191b602c26d'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412494b5c195d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:48 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'77'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993680'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'379ms'), (b'x-request-id', b'bb3ec0cb581105f364a26191b602c26d'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412494b5c195d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  topoisomerase that maintains I)NA suprrcoiIing i?~  ,$VO (Wang. 1985). A small amount of relaxation  was observed in t’he topA strain when 50 pg  ttorfloxacin/mI was used. suggesting that. anot’her  t.opoisomerase rnt? csontrihute weakly t 0 t hv  halanc~~ of supvrco~hng in E. coli cells.  \\\\V(s thank 13. Saut~l.. H. f+hols. I’. t)riigta. S.  \\\\\\\\‘asst~~~tnan. .J. L2’anp. M. Yarntolinsky anti H. Rrnjami~t  for helpful tlisc~ussions. This work was srtpportf~tl 1)~  grants frottt IV.1.H. to X.R.(‘. t).E.A. \\\\vas strpport~~~l It). a  NaConal ftrsrarc~h Servic,r Awarcl. grant tto. ‘I’XP  (~MOi2X?(‘-l~~. ant1 .l.f3.fS. was supporte(f Iby a ITttivrrsit>  of’ (‘atif’orrlia I%rg,rcsnts Felto~-ship.  References  .\\\\l)retnski. K.. Horss. R. H. h St,rrnbrrg. N. (1983).  Stutlies on the properties of’ f’l sit<,-spe(*ific r‘rcaotn-  ljination: c~vidrttcr for topolopic~all~ uttlinkrtl  produc+s following rrvotnhination. (‘~11. 32. IWI  1311.  ,Argos. I’.. I,antly. A.. Abrrmski. K.. fCgart. .J. I(..  Haggarti-T~lund(luixt,. E.. Hoess. R. H.. Kahn.  11. I,. K.. Kalionis. I%.. Xarayama. S. \\\\‘. I,.. f’irrson.  1,. S.. Strrnbrrp. ?r’. & T,rong. .I. XI. (1986). The itttrr-  grasr family of sit)r-sprt+ic recotnbinasrs: regional  similarities ant1 global tlivrrsity. EMIZO ./. 5. 1XI  140.  Austin. S.. %irsr. M. B Sternberg. X. (l!Ml). A novel role  for sit,-spevifiv recombination in maintenanc*e of  bac*trriaf rrpli(~ons. (‘ell. 25. 729~ 736.  f%rnjamin. H. B’. & Cozzarelti. N. R. (1986).  DXA-directed synapsis in recombination: slithering  and random collision of sites. Tn Proceedings 0j Ihr  Robert Welsh Foundation C’onfermcrs on t?hmmicnl  Research, vol. 29. pp. 107-126. Houst.on, TX. V.S.L\\\\.  Bliska, J. 13. $ C:ozzarelli, PI’. R. (1987). rse f’or site-  specific recombination as a probe of DNA structure  and metabolistn in W%O. J. Mol. Hiol. 194, 205-218.  Bliska. .J. I%.. l<rnjamin, H. 137. B (‘ozzarelfi. i%-. R. (19!1l). f<ofes:. ‘1’. ( ‘.. \\\\Vhitcs. .I H. & (‘cr~z;~t~(~lli. N. I< [ l!l!)O).', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  29. pp. 107-126. Houst.on, TX. V.S.L\\\\.  Bliska, J. 13. $ C:ozzarelli, PI’. R. (1987). rse f’or site-  specific recombination as a probe of DNA structure  and metabolistn in W%O. J. Mol. Hiol. 194, 205-218.  Bliska. .J. I%.. l<rnjamin, H. 137. B (‘ozzarelfi. i%-. R. (19!1l). f<ofes:. ‘1’. ( ‘.. \\\\Vhitcs. .I H. & (‘cr~z;~t~(~lli. N. I< [ l!l!)O).  Stt~llc*tuw 01’ ftlrc~tottc~t~~ic;~II~ sitf)f~t~(~oile~l 1)X.\\\\.  .I .IIo/. 8iol. 213. !)I:1 %:I  (%?st1v,! I: H S(ac,tt .I I<. ?Q \\\\‘afmvti. I). (f!47!4).  fntf~grati0tt of ltlastrtttl fm)phage~ f’f aml 1’7 trtto ttrt,  (.ll).ott~osottIf~ of E’:schr ~,ic,hitr c,r~li. .I .1lc11 I:iol 130.  I(il 17x  (‘raigitl. R. & 1lizttttc.lti. K. (I!M). Rolfs 01’ I)N.\\\\ tol)otog>  itt 1111 tra~~spo>itiotr tttfv.hattistn of’ <c,ttsing thv  rc~lativr~ orirntatiott ot t ho t\\\\vo tjS.1 sf*gttttatlts. (‘r/l.  45. T!Kl x00.  f)afv. E:. (’ k 0~2.. I). \\\\V. (IWO). ftttru- anti illtt’rtliolt’(.ttlill  sitf%-slttsc,ific. t~f~c~otttt~irtatio~t in plant c,rlls tncvliatc~tl 1))  bac.tc~riophapr I’I tv(.ontttinase. I:P~IP, 91. 79 85.  t)i(:atti. I<. .I. k ~li~ri;rtls. I<. .J. (1988). ftlf,rititic,;\\\\tiort ot’a  fjotcmt tlf~catrtiittitip onz>-tttt front E.whc~~ickirr r71fi.  -1. /jirj/. ~‘f/~rr/. 263. I XM I Xj’7:I.  IGttc~h~~t~. .\\\\. I(. (198X). Tftv ~!/tt,i (.ir(.l(b pl;r~trti~l ol’  ,\\\\‘rrc~~h/Iron,~,,~r.~ cr~rrl~isirrc’ ~~rccsl. 4. 27 -IO.  (:lasgov. .\\\\. (‘.. Hughes. Ii. ‘I 8 Simon. >I. I (I!)X!)).  f<ac~tt~ria I)S.\\\\ irr\\\\-c.t,siott s~~stt~ttts. trt .f/ohilf 1)S.l  (livrg. I). R. & Hou(~. \\\\I. M.. P(L). f’f) (X15 &59.  .\\\\tttvri(~att Soc,irty of’ .\\\\lic~t~c)ttiolop~-. \\\\Vasl~ingtotl. IN’.  Hatfrtll. (:. I’. & Crincllr?-. S. I). I’. (l!lXX).', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  4. 27 -IO.  (:lasgov. .\\\\. (‘.. Hughes. Ii. ‘I 8 Simon. >I. I (I!)X!)).  f<ac~tt~ria I)S.\\\\ irr\\\\-c.t,siott s~~stt~ttts. trt .f/ohilf 1)S.l  (livrg. I). R. & Hou(~. \\\\I. M.. P(L). f’f) (X15 &59.  .\\\\tttvri(~att Soc,irty of’ .\\\\lic~t~c)ttiolop~-. \\\\Vasl~ingtotl. IN’.  Hatfrtll. (:. I’. & Crincllr?-. S. I). I’. (l!lXX). Rc~~lvas;r anti  tjN.1 ittvrrtasrs. it fatttily ot tatrzytttc.s ttc,tivtL tn sitr-  sftvc,itic, t,t,c.c),rlt)inatiott. I II (:el/r/iC t~rl~r,tllhitlotior,  (firtc~h~rlaltti. I’, & Snltth. f:. I<.. c~ls). 1’1). :l.ii :l!)(i.  L\\\\tnt~ric~an Socirtj, of .\\\\tic~t~ot~iolo~~ \\\\Vashirigtort. IN’.  Hochntitn. I... Srgtlv. S. Stt-rnbt’rg. S. & (‘ohcvr. (:  (f!Ml). Sitfl-sfxv.itic, rc~c~otnt~in~ttio~~~al c,ir(,ttlat.iziltioli  ot’ t,ac,tc,riclf)ha#(, 1’1 f)S.\\\\. I’i/v/oyy. 131. I I Ii.  Hotash. I:. H. & .\\\\brc~tt~ski. K, (l!Ml). ‘f’ho (‘rcs-lox re(‘on-  ttittatiott hystenl. ttt .\\\\‘/cc/ric ‘ I C~idS ,1//d .IIr,lf~~trItrr  /2io/oy,q (Ec,kstrJirr. I;. & l,illvy. I). 11 rbtls). \\\\-()I. 1.  ,‘I’. < * 0’4 IO!). Sf)rin~,rrt.-\\\\.c,rla(r. Berlin anti H(~iclt~lbt~rg.  Horss. It. H.. \\\\\\\\‘ivrzbic~ki. :\\\\. & .\\\\bretnski. fi. (l!)X7).  lsolatiott antl c~ll;rt.ac~tt~t~iz;ttiott of intrrtttvdiates in  sitfs-sfx*c,itic. t~c~c~otllt,itl;~tiotl I’UM .Y///. .-fU//i 8SCi..  /...\\\\‘..-I. 84. %-Ml (iX14.  Ishii. 1;. 1fttrakattti. ‘I’. &Y Sltistto. Ii. (l!,!rl). (:yra~ inbib-  tot’s tn(‘tvast lh(b c*otrtc,ttt of’ knottrtl I)NA sI)fs(.ir,s 01’  I)lasttritl Iji3ru22 in fkf~iri~~hitr 4;. ./. /~tf~/~ v;o/. 173.  5.55 I :5;i.Y<.  Kanaar. ft. 6 \\\\-an tfr I’utt,r. I’. (l!#i). Topological  aspec’ts of site-sprc.ifiv l)piA inversiolt. NioEsso,~x 7.  195 200.  Kanaar. I<.. Klippel.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  ‘I’. &Y Sltistto. Ii. (l!,!rl). (:yra~ inbib-  tot’s tn(‘tvast lh(b c*otrtc,ttt of’ knottrtl I)NA sI)fs(.ir,s 01’  I)lasttritl Iji3ru22 in fkf~iri~~hitr 4;. ./. /~tf~/~ v;o/. 173.  5.55 I :5;i.Y<.  Kanaar. ft. 6 \\\\-an tfr I’utt,r. I’. (l!#i). Topological  aspec’ts of site-sprc.ifiv l)piA inversiolt. NioEsso,~x 7.  195 200.  Kanaar. I<.. Klippel. ;\\\\.. Shekht,man. P:.. Ihlngan.  .I. 11.. Kahmann. K. & (‘ozzarrlli. N. I<. (1990).  f’rocessivt rec,ontl)in~ttiott f)y the phagc. Mu (iin  system: implications for the mechanism of DNA  strantl esvhange. 1)X-A site alignment. arttl tvihanve-‘l  ac,tiott. (‘e/Z. 62. X54 X6.  Kate. .1.-f.. Sishintrtra. \\\\‘.. Ttriarllttt,a. I:.. Siki. II..  Hiraga. S. S: Sttzttki. H. (19!N). Sew toftoist)tttt‘t,;rst  ossrtttial for (.hrontosotttt~ srgrcxgatiott it, R.whc~ric~hirr  coli. (‘c/l. 63. :%!):3 10-L  Kikuc,hi. 1’. & Sash. H. .I. (I!$‘iX). The t)ac’tt’t.toltf~“#t i it/l  gertr ltrotlrrc*t, .\\\\ filter assay for gettetic, rc~c~otnbin~c-  tion. lturific.ation of tnt. ant1 spfv*itiv bintling to I)Srl.  ./. Bid. (‘hrr,,. 253. i 11!4 i 157.  Krasrro\\\\c. >I. .\\\\. $ (‘ozzarvlli. N. I<. (1983). Sitvsfxvitic~  rc,lasation atttl rf~c~oml~i~~tttio~~ tty tttr TttX r(~soIvaLst~.  tvc~ognitioll of’ tbr. l)S.\\\\ flath bt~tL\\\\.f~Ptt ot,i(vtt(vt sit,t,s.  / ‘,,/I. 32. 13 I :% I X-c.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  The Mechanism qf Cre Recombkation in Viva 67.3  Landy. .j. ( 1989). Dynamic structural and regufator~  aspects of i site-specific, recombination. Anne. Rec..  Hiochen/. 58. 9 13-939.  f,iu. 1,. F. & Wang. .J. (‘. (1987). Supercoiling of the f)SA  trmplatr during tranxription. Proc. Sal. Acad. ~Sci..  f ‘.S.A. 84. ‘iW--701i.  I,oc*kshon. I). R- hforris. 11. R. (f98S). Positivrlg super-  coilptf plasmid r)X.A ih produced bg twatmmt 4  Esrlrvri~hirc voli with I)SA gyrase inhibitors. iVltc/.  A rids RPS. 11 2999- 30 I 7.  I.rtttinger. A. Sf)rittgrr. A. 8r Svhmid. &I. (1991). X dustrt  of genes that affects ttucleotide segregation in  hbincoftP/to t!~phitr~~critln~ Tilv R’rw i~io/fxqi.d, 3. 687  6!17.  hlcClelfan, ,I. A.. I<oubfikora. I’., Palecek. IX. & Liffey,  11. M. (1990). Superhefjcal torsion in cellular DNA  rrsponds dirwt.ly to environmental and genetic%  factors I’roc. Sat. Acad. Ski., U.S.A. 87. 8.773~-8377.  .\\\\leyrr-Leon. I,.. Huang, I,.-( ‘.. Litnlauf. S. LV.. ~‘ox. M. hf.  B Inman. IZ. R. (198X). Hoffiday intermediates and  rewc$ton by-ftr0duct.s in Ffp protein-promoted sittl-  sfwcitic, tw’otn hination. .flol. Cell. Biol. 8. 3784-3596.  Siash. H. A. & f?,c)bertson, (‘. X. (1981). Purification and  ~Jropwtiw 01 thv li:,srhuich,ia roli J’roteitt factor  rcyuitwl for /I intrgrativr recombinaticrtt. ./ Kid  ( ‘hrnt 256. !kM-!kL53.  f’oflocak. ‘1’. .I. & .\\\\hretnski. K. (1979). I)SA without  sitftet,twists (‘iin be an in citro substrate for site-  sf)witic~ rwomhination of hact?riophapc, i.. -1. Nd.  RkJl. 131. fi.51 fi54.  Raji. :\\\\.. Zitbrf. I). .I Laiit’er. (‘. S. & I)eptw. ft. E.  ( I!)x~). (iet1t.t i(. analysis of mutations that c~omprtt-  sate fi)r loss ot’ B.<chrrichin colt’ DSA topoisorneraw I.  .I fkwl~riol. 162. I Ii:<-I 179.  ~amhrook. .I.. Frifst.1~. E. F. & Maniatis. T. (f98!)).  .I/ olvc,,rlrrr ( ‘lot~in!/: .-I Ltrt~f~rnlory Mfrnunl. 2nd edit  ( ‘oltl Spring Harbor Laboratory Press. (‘old SItring  Harbor. XV. (‘re revombinase. Thp ~V’uw Hiolqi.st. 2.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  ft. E.  ( I!)x~). (iet1t.t i(. analysis of mutations that c~omprtt-  sate fi)r loss ot’ B.<chrrichin colt’ DSA topoisorneraw I.  .I fkwl~riol. 162. I Ii:<-I 179.  ~amhrook. .I.. Frifst.1~. E. F. & Maniatis. T. (f98!)).  .I/ olvc,,rlrrr ( ‘lot~in!/: .-I Ltrt~f~rnlory Mfrnunl. 2nd edit  ( ‘oltl Spring Harbor Laboratory Press. (‘old SItring  Harbor. XV. (‘re revombinase. Thp ~V’uw Hiolqi.st. 2. 441 -149.  Shen. I,. I,., Kohfbrenner. IV. IX.. \\\\VeigI, I). &  ISaranowski. .I. (1989). Mechanism of’ quinilone  itrhihition of‘ I)K;A gyrase. .1. Hid. (‘hrm. 264. t?973  297x.  Shishido. K., Komiyama. S. & Ikawa. S. (1987).  1 ncreased production of a knotted fortn of f)fasmid  f)KRSP:! 1154 in Escherichi:rr coli DNA tc~f~oisotnetxsr  mutants. ,/. J/o/. Biol. 195. PI.5 ‘IX.  Sin&n. R. R.. (‘arfson. .I. 0. & I’rttijohtt. I). E. (IYXO).  Torsional tension in t,he 11X.4 double helix measured  with trinrrth?.If)soralen in living A’. ~01, culls: analo-  gous measrrretnc~nts in insrc+ xntf hrtntati wolfs. (‘~11.  21. ii:& 5x3.  Strrnbrtg. S.. Hatnilton. I). & Horss. 1~. (f!JXf). 1’1 sitr-  hftwitic~ rpc.onlhination. IT. R.rc,omt,irt~tic)tt Jwt\\\\r.wn  loxI’ and the Itacteriaf chrotnosotttr. .I .I/rL /Ii/d.  150. 4x7-507.  Sternberg. S.. Sauer. 13.. Hoess, R. & Ahremski, K.  (I 9M). An initial characterization of t,he bacterio-  phage I’1 cm st*ructural gene and it,s rrgnfat.oq  region. ,I. Xo/. Bid. 187. f97-Yf6.  Yofkrrt. F. C’. K: 1Sroac.h. ,J. R. (1986). SitIs-sfw*tfic. r(v~ottl-  t)in;ltiotr ftrorttotc~s ftlaxmitl atnf~fifi~~atioti in ywst.  (‘dl. 46. 511 350.  FVattg. ,I. (‘. (1985). DK‘A topoisomerases. Alnno. Kpt:.  lkch.mr 54. 665-697.  \\\\Z’rt. H.-F-.. Shuy. 8.. \\\\\\\\‘ang. .J. (I. C“ flttt. I.. ( f!)XX)  Transc*rif)t.it,tt pc~nrratrs fwsit.ivkLly ant1 ntyat,ivrl>,  sttfwrcwiltd tlomains in tlw tc~nif)latc~. I ‘P/I. 53. 1X1  440.  lvartttolitwk>~. Jf. IS.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  (‘dl. 46. 511 350.  FVattg. ,I. (‘. (1985). DK‘A topoisomerases. Alnno. Kpt:.  lkch.mr 54. 665-697.  \\\\Z’rt. H.-F-.. Shuy. 8.. \\\\\\\\‘ang. .J. (I. C“ flttt. I.. ( f!)XX)  Transc*rif)t.it,tt pc~nrratrs fwsit.ivkLly ant1 ntyat,ivrl>,  sttfwrcwiltd tlomains in tlw tc~nif)latc~. I ‘P/I. 53. 1X1  440.  lvartttolitwk>~. Jf. IS. & StcJrn hrty s. (l!)XX).  Il;tc~trriof)hagr 1’1 Tn 5”h.p /2rrc.fPl.iof,hlr!f~.~ (( ‘;tfrrtder.  f:.. d). \\\\.Of. i. ,‘I’. “91 J:sf+ I’fWlllN ~‘WSS. SW  \\\\r‘ork atttf f~~tt~lott.', 'page_label: 16813 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  The Doc Toxin and Phd Antidote Proteins of the Bacteriophage P1 Plasmid Addiction System Form a Heterotrimeric Complex* (Received for publication, March 9, 1999, and in revised form, March 30, 1999) Ehud Gazit‡ and Robert T. Sauer§ From the Department of Biology, Massachusetts Institute of Technology, Cambridge, Massachusetts 02139 The toxin (Doc) and antidote (Phd) proteins of the plasmid addiction system of bacteriophage P1 were pu- rified as a complex. Cocrystals of the complex containeda 2:1 molar ratio of Phd:Doc as assayed by dye bindingfollowing SDS-polyacrylamide gel electrophoresis andas determined by amino acid analysis. Gel filtration andanalytical ultracentrifugation revealed that the two ad-diction proteins interact in solution to form a P 2D tri- mer composed of one Doc and two Phd molecules. Theseresults support a model in which Phd inhibits the toxicactivity of Doc by direct binding. Circular dichroismexperiments showed that changes in secondary struc-ture accompany formation of the heterotrimeric com-plex, raising the possibility that Phd may act by an al-losteric mechanism. Studies of Phd and Doc moleculeslabeled with fluorescent energy donor and acceptorgroups gave an equilibrium dissociation constant ofabout 0.8 mM2and a very short, sub second half-life of complex dissociation. As a consequence, low concentra-tions of free Doc toxin are likely to be present bothtransiently and in the steady state in cells containingthe Phd antidote, making mechanisms of single-hit Doctoxicity improbable. The plasmid addiction system of bacteriophage P1 consists of two proteins: a toxin known as Doc (death on cure) and an antidote named Phd (prevent host death) (1). Bacteriophage P1 lysogenizes Escherichia coli as a low copy plasmid that is inherited in a remarkably stable fashion, in part, because the plasmid addiction system kills cells that have lost the P1 ge-nome (1–5). The mechanism of Doc toxicity is unknown. Phd is,however, known to be degraded in a fashion dependent on thehost ClpXP protease system, and continual synthesis of Phd istherefore required to counteract Doc toxicity (3). In daughtercells that have lost the P1 genome, degradation of Phd even-tually results in reduction of antidote activity to a level wherethe host is killed. Phd may exert its antitoxin activity bybinding Doc and physically blocking its interaction with cellu-lar target molecules (1) or by acting in an allosteric fashion toalter Doc structure and toxin activity. Alternatively, Phd mightexert its antidote activity indirectly by activating another pro-tein that neutralizes Doc. Some of the regulatory and biochemical properties of Phdhave been established. For example, Phd is a DNA-bindingprotein that represses transcription of its own gene as well asthat of Doc by binding to an operator DNA site that overlapsthe addiction promoter (4). Four molecules of Phd bind to theintact operator with dimers binding to adjacent 10-base pairsubsites (6). Phd has a predominantly a-helical structure when bound to DNA or at low temperatures but the freemonomeric protein has a T mof 25 °C and is largely denatured at 37 °C (6). Addiction systems involving a stable toxin and proteolyti- cally unstable antidote are also used by other low copy plas-mids and may be involved in the regulation of programmed celldeath in E. coli (7–13). Although direct interaction between the toxin and its antidote is assumed to be the mechanism ofinhibition for each of these systems, experimental evidence forcomplex formation has been presented only for the CcdA anti-dote and CcdB toxin of the F plasmid of E. coli (14). In the study of the phage P1 addiction proteins presented here, we demon-strate that Phd binds directly to Doc, forming a trimeric com-plex (P 2D) with one molecule of Doc and two molecules of Phd. Complex formation appears to be accompanied by changes insecondary structure.', 'page_label: 16813 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  Addiction systems involving a stable toxin and proteolyti- cally unstable antidote are also used by other low copy plas-mids and may be involved in the regulation of programmed celldeath in E. coli (7–13). Although direct interaction between the toxin and its antidote is assumed to be the mechanism ofinhibition for each of these systems, experimental evidence forcomplex formation has been presented only for the CcdA anti-dote and CcdB toxin of the F plasmid of E. coli (14). In the study of the phage P1 addiction proteins presented here, we demon-strate that Phd binds directly to Doc, forming a trimeric com-plex (P 2D) with one molecule of Doc and two molecules of Phd. Complex formation appears to be accompanied by changes insecondary structure. P 2D complexes form at low micromolar concentrations and are very short lived, ensuring a dynamicequilibrium between bound and free forms within the cell. MATERIALS AND METHODS Protein Expression and Purification—E. coli strain X90/pRDM032, which expresses Phd and Doc under Ptaccontrol (4), was grown at 37 °C to an A600of approximately 0.7 in Luria-Bertani broth containing 100 mg/ml ampicillin, and protein expression was induced by the addition of 100 mg/ml isopropyl-1-thio- b-D-galactopyranoside for 30 min. Cells were harvested by centrifugation (3,500 3gfor 30 min at 4 °C), and the pellets were frozen at 280 °C before purification. Cell pellets were thawed at 4 °C, resuspended in buffer A (50 m MNaCl, 50 m MTris-HCl (pH 7.4), 1 m MEDTA), and phenylmethylsulfonyl fluoride was added to a final concentration of 50 mM. Cells were lysed by sonication, and the resulting solution was centrifuged at 4 °C for 10 min at 13,000 3g. The supernatant was applied to a MonoS HR 5/5 FPLC column (Amersham Pharmacia Biotech) equilibrated in buffer A, which was eluted withgradient from 50 to 1,000 m MNaCl in 50 m MTris-HCl, pH 7.4. Elution was monitored by A280. A peak containing Phd and Doc eluted at approximately 600 m MNaCl. This peak was applied to a semi-prepar- ative reverse-phase C18column, and Phd and Doc were separated using a gradient from 0 to 80% acetonitrile in 0.1% trifluoracetic acid. Fol-lowing this step, the proteins were greater than 95% pure as assessedby Coomassie staining of SDS-polyacrylamide gels. Characterization of Purified Protein— The masses of the purified proteins were determined by MALDI-TOF 1mass spectrometry using a Voyager-DE STR Biospectrometry Workstation (PerSeptive BioSys-tems) with sinapinic acid (3,5-dimethoxy-4-hydroxycinnamic acid) asthe matrix, linear mode operation, a grid voltage at 91.0%, and a pulsedelay time of 150 ns. Protein concentrations were determined by meas-uring tyrosine absorbance at both pH 7 and in 0.1 MKOH using a* Supported by National Institutes of Health Grants AI-15706 and AI-16892. The costs of publication of this article were defrayed in part by the payment of page charges. This article must therefore be herebymarked “advertisement” in accordance with 18 U.S.C. Section 1734solely to indicate this fact. ‡ Supported by postdoctoral fellowships from EMBO and the Human Frontiers Science Foundation Program. § To whom correspondence should be addressed: Massachusetts Institute of Technology 68–571, Cambridge, MA 02139. E-mail:bobsauer@mit.edu.1The abbreviations used are: MALDI-TOF, matrix-assisted laser de- sorption ionization-time-of-flight; HPLC, high pressure liquidchromatography.THEJOURNAL OF BIOLOGICAL CHEMISTRY Vol.', 'page_label: 16813 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  The costs of publication of this article were defrayed in part by the payment of page charges. This article must therefore be herebymarked “advertisement” in accordance with 18 U.S.C. Section 1734solely to indicate this fact. ‡ Supported by postdoctoral fellowships from EMBO and the Human Frontiers Science Foundation Program. § To whom correspondence should be addressed: Massachusetts Institute of Technology 68–571, Cambridge, MA 02139. E-mail:bobsauer@mit.edu.1The abbreviations used are: MALDI-TOF, matrix-assisted laser de- sorption ionization-time-of-flight; HPLC, high pressure liquidchromatography.THEJOURNAL OF BIOLOGICAL CHEMISTRY Vol. 274, No. 24, Issue of June 11, pp. 16813–16818, 1999 © 1999 by The American Society for Biochemistry and Molecular Biology, Inc. Printed in U.S.A. This paper is available on line at http://www.jbc.org 16813 This is an Open Access article under the CC BY  license.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  topoisomerase that maintains I)NA suprrcoiIing i?~  ,$VO (Wang. 1985). A small amount of relaxation  was observed in t’he topA strain when 50 pg  ttorfloxacin/mI was used. suggesting that. anot’her  t.opoisomerase rnt? csontrihute weakly t 0 t hv  halanc~~ of supvrco~hng in E. coli cells.  \\\\V(s thank 13. Saut~l.. H. f+hols. I’. t)riigta. S.  \\\\\\\\‘asst~~~tnan. .J. L2’anp. M. Yarntolinsky anti H. Rrnjami~t  for helpful tlisc~ussions. This work was srtpportf~tl 1)~  grants frottt IV.1.H. to X.R.(‘. t).E.A. \\\\vas strpport~~~l It). a  NaConal ftrsrarc~h Servic,r Awarcl. grant tto. ‘I’XP  (~MOi2X?(‘-l~~. ant1 .l.f3.fS. was supporte(f Iby a ITttivrrsit>  of’ (‘atif’orrlia I%rg,rcsnts Felto~-ship.  References  .\\\\l)retnski. K.. Horss. R. H. h St,rrnbrrg. N. (1983).  Stutlies on the properties of’ f’l sit<,-spe(*ific r‘rcaotn-  ljination: c~vidrttcr for topolopic~all~ uttlinkrtl  produc+s following rrvotnhination. (‘~11. 32. IWI  1311.  ,Argos. I’.. I,antly. A.. Abrrmski. K.. fCgart. .J. I(..  Haggarti-T~lund(luixt,. E.. Hoess. R. H.. Kahn.  11. I,. K.. Kalionis. I%.. Xarayama. S. \\\\‘. I,.. f’irrson.  1,. S.. Strrnbrrp. ?r’. & T,rong. .I. XI. (1986). The itttrr-  grasr family of sit)r-sprt+ic recotnbinasrs: regional  similarities ant1 global tlivrrsity. EMIZO ./. 5. 1XI  140.  Austin. S.. %irsr. M. B Sternberg. X. (l!Ml). A novel role  for sit,-spevifiv recombination in maintenanc*e of  bac*trriaf rrpli(~ons. (‘ell. 25. 729~ 736.  f%rnjamin. H. B’. & Cozzarelti. N. R. (1986).  DXA-directed synapsis in recombination: slithering  and random collision of sites. Tn Proceedings 0j Ihr  Robert Welsh Foundation C’onfermcrs on t?hmmicnl  Research, vol. 29. pp. 107-126. Houst.on, TX. V.S.L\\\\.  Bliska, J. 13. $ C:ozzarelli, PI’. R. (1987). rse f’or site-  specific recombination as a probe of DNA structure  and metabolistn in W%O. J. Mol. Hiol. 194, 205-218.  Bliska. .J. I%.. l<rnjamin, H. 137. B (‘ozzarelfi. i%-. R. (19!1l). f<ofes:. ‘1’. ( ‘.. \\\\Vhitcs. .I H. & (‘cr~z;~t~(~lli. N. I< [ l!l!)O).', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  29. pp. 107-126. Houst.on, TX. V.S.L\\\\.  Bliska, J. 13. $ C:ozzarelli, PI’. R. (1987). rse f’or site-  specific recombination as a probe of DNA structure  and metabolistn in W%O. J. Mol. Hiol. 194, 205-218.  Bliska. .J. I%.. l<rnjamin, H. 137. B (‘ozzarelfi. i%-. R. (19!1l). f<ofes:. ‘1’. ( ‘.. \\\\Vhitcs. .I H. & (‘cr~z;~t~(~lli. N. I< [ l!l!)O).  Stt~llc*tuw 01’ ftlrc~tottc~t~~ic;~II~ sitf)f~t~(~oile~l 1)X.\\\\.  .I .IIo/. 8iol. 213. !)I:1 %:I  (%?st1v,! I: H S(ac,tt .I I<. ?Q \\\\‘afmvti. I). (f!47!4).  fntf~grati0tt of ltlastrtttl fm)phage~ f’f aml 1’7 trtto ttrt,  (.ll).ott~osottIf~ of E’:schr ~,ic,hitr c,r~li. .I .1lc11 I:iol 130.  I(il 17x  (‘raigitl. R. & 1lizttttc.lti. K. (I!M). Rolfs 01’ I)N.\\\\ tol)otog>  itt 1111 tra~~spo>itiotr tttfv.hattistn of’ <c,ttsing thv  rc~lativr~ orirntatiott ot t ho t\\\\vo tjS.1 sf*gttttatlts. (‘r/l.  45. T!Kl x00.  f)afv. E:. (’ k 0~2.. I). \\\\V. (IWO). ftttru- anti illtt’rtliolt’(.ttlill  sitf%-slttsc,ific. t~f~c~otttt~irtatio~t in plant c,rlls tncvliatc~tl 1))  bac.tc~riophapr I’I tv(.ontttinase. I:P~IP, 91. 79 85.  t)i(:atti. I<. .I. k ~li~ri;rtls. I<. .J. (1988). ftlf,rititic,;\\\\tiort ot’a  fjotcmt tlf~catrtiittitip onz>-tttt front E.whc~~ickirr r71fi.  -1. /jirj/. ~‘f/~rr/. 263. I XM I Xj’7:I.  IGttc~h~~t~. .\\\\. I(. (198X). Tftv ~!/tt,i (.ir(.l(b pl;r~trti~l ol’  ,\\\\‘rrc~~h/Iron,~,,~r.~ cr~rrl~isirrc’ ~~rccsl. 4. 27 -IO.  (:lasgov. .\\\\. (‘.. Hughes. Ii. ‘I 8 Simon. >I. I (I!)X!)).  f<ac~tt~ria I)S.\\\\ irr\\\\-c.t,siott s~~stt~ttts. trt .f/ohilf 1)S.l  (livrg. I). R. & Hou(~. \\\\I. M.. P(L). f’f) (X15 &59.  .\\\\tttvri(~att Soc,irty of’ .\\\\lic~t~c)ttiolop~-. \\\\Vasl~ingtotl. IN’.  Hatfrtll. (:. I’. & Crincllr?-. S. I). I’. (l!lXX).', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  4. 27 -IO.  (:lasgov. .\\\\. (‘.. Hughes. Ii. ‘I 8 Simon. >I. I (I!)X!)).  f<ac~tt~ria I)S.\\\\ irr\\\\-c.t,siott s~~stt~ttts. trt .f/ohilf 1)S.l  (livrg. I). R. & Hou(~. \\\\I. M.. P(L). f’f) (X15 &59.  .\\\\tttvri(~att Soc,irty of’ .\\\\lic~t~c)ttiolop~-. \\\\Vasl~ingtotl. IN’.  Hatfrtll. (:. I’. & Crincllr?-. S. I). I’. (l!lXX). Rc~~lvas;r anti  tjN.1 ittvrrtasrs. it fatttily ot tatrzytttc.s ttc,tivtL tn sitr-  sftvc,itic, t,t,c.c),rlt)inatiott. I II (:el/r/iC t~rl~r,tllhitlotior,  (firtc~h~rlaltti. I’, & Snltth. f:. I<.. c~ls). 1’1). :l.ii :l!)(i.  L\\\\tnt~ric~an Socirtj, of .\\\\tic~t~ot~iolo~~ \\\\Vashirigtort. IN’.  Hochntitn. I... Srgtlv. S. Stt-rnbt’rg. S. & (‘ohcvr. (:  (f!Ml). Sitfl-sfxv.itic, rc~c~otnt~in~ttio~~~al c,ir(,ttlat.iziltioli  ot’ t,ac,tc,riclf)ha#(, 1’1 f)S.\\\\. I’i/v/oyy. 131. I I Ii.  Hotash. I:. H. & .\\\\brc~tt~ski. K, (l!Ml). ‘f’ho (‘rcs-lox re(‘on-  ttittatiott hystenl. ttt .\\\\‘/cc/ric ‘ I C~idS ,1//d .IIr,lf~~trItrr  /2io/oy,q (Ec,kstrJirr. I;. & l,illvy. I). 11 rbtls). \\\\-()I. 1.  ,‘I’. < * 0’4 IO!). Sf)rin~,rrt.-\\\\.c,rla(r. Berlin anti H(~iclt~lbt~rg.  Horss. It. H.. \\\\\\\\‘ivrzbic~ki. :\\\\. & .\\\\bretnski. fi. (l!)X7).  lsolatiott antl c~ll;rt.ac~tt~t~iz;ttiott of intrrtttvdiates in  sitfs-sfx*c,itic. t~c~c~otllt,itl;~tiotl I’UM .Y///. .-fU//i 8SCi..  /...\\\\‘..-I. 84. %-Ml (iX14.  Ishii. 1;. 1fttrakattti. ‘I’. &Y Sltistto. Ii. (l!,!rl). (:yra~ inbib-  tot’s tn(‘tvast lh(b c*otrtc,ttt of’ knottrtl I)NA sI)fs(.ir,s 01’  I)lasttritl Iji3ru22 in fkf~iri~~hitr 4;. ./. /~tf~/~ v;o/. 173.  5.55 I :5;i.Y<.  Kanaar. ft. 6 \\\\-an tfr I’utt,r. I’. (l!#i). Topological  aspec’ts of site-sprc.ifiv l)piA inversiolt. NioEsso,~x 7.  195 200.  Kanaar. I<.. Klippel.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  ‘I’. &Y Sltistto. Ii. (l!,!rl). (:yra~ inbib-  tot’s tn(‘tvast lh(b c*otrtc,ttt of’ knottrtl I)NA sI)fs(.ir,s 01’  I)lasttritl Iji3ru22 in fkf~iri~~hitr 4;. ./. /~tf~/~ v;o/. 173.  5.55 I :5;i.Y<.  Kanaar. ft. 6 \\\\-an tfr I’utt,r. I’. (l!#i). Topological  aspec’ts of site-sprc.ifiv l)piA inversiolt. NioEsso,~x 7.  195 200.  Kanaar. I<.. Klippel. ;\\\\.. Shekht,man. P:.. Ihlngan.  .I. 11.. Kahmann. K. & (‘ozzarrlli. N. I<. (1990).  f’rocessivt rec,ontl)in~ttiott f)y the phagc. Mu (iin  system: implications for the mechanism of DNA  strantl esvhange. 1)X-A site alignment. arttl tvihanve-‘l  ac,tiott. (‘e/Z. 62. X54 X6.  Kate. .1.-f.. Sishintrtra. \\\\‘.. Ttriarllttt,a. I:.. Siki. II..  Hiraga. S. S: Sttzttki. H. (19!N). Sew toftoist)tttt‘t,;rst  ossrtttial for (.hrontosotttt~ srgrcxgatiott it, R.whc~ric~hirr  coli. (‘c/l. 63. :%!):3 10-L  Kikuc,hi. 1’. & Sash. H. .I. (I!$‘iX). The t)ac’tt’t.toltf~“#t i it/l  gertr ltrotlrrc*t, .\\\\ filter assay for gettetic, rc~c~otnbin~c-  tion. lturific.ation of tnt. ant1 spfv*itiv bintling to I)Srl.  ./. Bid. (‘hrr,,. 253. i 11!4 i 157.  Krasrro\\\\c. >I. .\\\\. $ (‘ozzarvlli. N. I<. (1983). Sitvsfxvitic~  rc,lasation atttl rf~c~oml~i~~tttio~~ tty tttr TttX r(~soIvaLst~.  tvc~ognitioll of’ tbr. l)S.\\\\ flath bt~tL\\\\.f~Ptt ot,i(vtt(vt sit,t,s.  / ‘,,/I. 32. 13 I :% I X-c.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  The Mechanism qf Cre Recombkation in Viva 67.3  Landy. .j. ( 1989). Dynamic structural and regufator~  aspects of i site-specific, recombination. Anne. Rec..  Hiochen/. 58. 9 13-939.  f,iu. 1,. F. & Wang. .J. (‘. (1987). Supercoiling of the f)SA  trmplatr during tranxription. Proc. Sal. Acad. ~Sci..  f ‘.S.A. 84. ‘iW--701i.  I,oc*kshon. I). R- hforris. 11. R. (f98S). Positivrlg super-  coilptf plasmid r)X.A ih produced bg twatmmt 4  Esrlrvri~hirc voli with I)SA gyrase inhibitors. iVltc/.  A rids RPS. 11 2999- 30 I 7.  I.rtttinger. A. Sf)rittgrr. A. 8r Svhmid. &I. (1991). X dustrt  of genes that affects ttucleotide segregation in  hbincoftP/to t!~phitr~~critln~ Tilv R’rw i~io/fxqi.d, 3. 687  6!17.  hlcClelfan, ,I. A.. I<oubfikora. I’., Palecek. IX. & Liffey,  11. M. (1990). Superhefjcal torsion in cellular DNA  rrsponds dirwt.ly to environmental and genetic%  factors I’roc. Sat. Acad. Ski., U.S.A. 87. 8.773~-8377.  .\\\\leyrr-Leon. I,.. Huang, I,.-( ‘.. Litnlauf. S. LV.. ~‘ox. M. hf.  B Inman. IZ. R. (198X). Hoffiday intermediates and  rewc$ton by-ftr0duct.s in Ffp protein-promoted sittl-  sfwcitic, tw’otn hination. .flol. Cell. Biol. 8. 3784-3596.  Siash. H. A. & f?,c)bertson, (‘. X. (1981). Purification and  ~Jropwtiw 01 thv li:,srhuich,ia roli J’roteitt factor  rcyuitwl for /I intrgrativr recombinaticrtt. ./ Kid  ( ‘hrnt 256. !kM-!kL53.  f’oflocak. ‘1’. .I. & .\\\\hretnski. K. (1979). I)SA without  sitftet,twists (‘iin be an in citro substrate for site-  sf)witic~ rwomhination of hact?riophapc, i.. -1. Nd.  RkJl. 131. fi.51 fi54.  Raji. :\\\\.. Zitbrf. I). .I Laiit’er. (‘. S. & I)eptw. ft. E.  ( I!)x~). (iet1t.t i(. analysis of mutations that c~omprtt-  sate fi)r loss ot’ B.<chrrichin colt’ DSA topoisorneraw I.  .I fkwl~riol. 162. I Ii:<-I 179.  ~amhrook. .I.. Frifst.1~. E. F. & Maniatis. T. (f98!)).  .I/ olvc,,rlrrr ( ‘lot~in!/: .-I Ltrt~f~rnlory Mfrnunl. 2nd edit  ( ‘oltl Spring Harbor Laboratory Press. (‘old SItring  Harbor. XV. (‘re revombinase. Thp ~V’uw Hiolqi.st. 2.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  ft. E.  ( I!)x~). (iet1t.t i(. analysis of mutations that c~omprtt-  sate fi)r loss ot’ B.<chrrichin colt’ DSA topoisorneraw I.  .I fkwl~riol. 162. I Ii:<-I 179.  ~amhrook. .I.. Frifst.1~. E. F. & Maniatis. T. (f98!)).  .I/ olvc,,rlrrr ( ‘lot~in!/: .-I Ltrt~f~rnlory Mfrnunl. 2nd edit  ( ‘oltl Spring Harbor Laboratory Press. (‘old SItring  Harbor. XV. (‘re revombinase. Thp ~V’uw Hiolqi.st. 2. 441 -149.  Shen. I,. I,., Kohfbrenner. IV. IX.. \\\\VeigI, I). &  ISaranowski. .I. (1989). Mechanism of’ quinilone  itrhihition of‘ I)K;A gyrase. .1. Hid. (‘hrm. 264. t?973  297x.  Shishido. K., Komiyama. S. & Ikawa. S. (1987).  1 ncreased production of a knotted fortn of f)fasmid  f)KRSP:! 1154 in Escherichi:rr coli DNA tc~f~oisotnetxsr  mutants. ,/. J/o/. Biol. 195. PI.5 ‘IX.  Sin&n. R. R.. (‘arfson. .I. 0. & I’rttijohtt. I). E. (IYXO).  Torsional tension in t,he 11X.4 double helix measured  with trinrrth?.If)soralen in living A’. ~01, culls: analo-  gous measrrretnc~nts in insrc+ xntf hrtntati wolfs. (‘~11.  21. ii:& 5x3.  Strrnbrtg. S.. Hatnilton. I). & Horss. 1~. (f!JXf). 1’1 sitr-  hftwitic~ rpc.onlhination. IT. R.rc,omt,irt~tic)tt Jwt\\\\r.wn  loxI’ and the Itacteriaf chrotnosotttr. .I .I/rL /Ii/d.  150. 4x7-507.  Sternberg. S.. Sauer. 13.. Hoess, R. & Ahremski, K.  (I 9M). An initial characterization of t,he bacterio-  phage I’1 cm st*ructural gene and it,s rrgnfat.oq  region. ,I. Xo/. Bid. 187. f97-Yf6.  Yofkrrt. F. C’. K: 1Sroac.h. ,J. R. (1986). SitIs-sfw*tfic. r(v~ottl-  t)in;ltiotr ftrorttotc~s ftlaxmitl atnf~fifi~~atioti in ywst.  (‘dl. 46. 511 350.  FVattg. ,I. (‘. (1985). DK‘A topoisomerases. Alnno. Kpt:.  lkch.mr 54. 665-697.  \\\\Z’rt. H.-F-.. Shuy. 8.. \\\\\\\\‘ang. .J. (I. C“ flttt. I.. ( f!)XX)  Transc*rif)t.it,tt pc~nrratrs fwsit.ivkLly ant1 ntyat,ivrl>,  sttfwrcwiltd tlomains in tlw tc~nif)latc~. I ‘P/I. 53. 1X1  440.  lvartttolitwk>~. Jf. IS.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-002228369290623R-main.pdf  (‘dl. 46. 511 350.  FVattg. ,I. (‘. (1985). DK‘A topoisomerases. Alnno. Kpt:.  lkch.mr 54. 665-697.  \\\\Z’rt. H.-F-.. Shuy. 8.. \\\\\\\\‘ang. .J. (I. C“ flttt. I.. ( f!)XX)  Transc*rif)t.it,tt pc~nrratrs fwsit.ivkLly ant1 ntyat,ivrl>,  sttfwrcwiltd tlomains in tlw tc~nif)latc~. I ‘P/I. 53. 1X1  440.  lvartttolitwk>~. Jf. IS. & StcJrn hrty s. (l!)XX).  Il;tc~trriof)hagr 1’1 Tn 5”h.p /2rrc.fPl.iof,hlr!f~.~ (( ‘;tfrrtder.  f:.. d). \\\\.Of. i. ,‘I’. “91 J:sf+ I’fWlllN ~‘WSS. SW  \\\\r‘ork atttf f~~tt~lott.', 'page_label: 16813 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  The Doc Toxin and Phd Antidote Proteins of the Bacteriophage P1 Plasmid Addiction System Form a Heterotrimeric Complex* (Received for publication, March 9, 1999, and in revised form, March 30, 1999) Ehud Gazit‡ and Robert T. Sauer§ From the Department of Biology, Massachusetts Institute of Technology, Cambridge, Massachusetts 02139 The toxin (Doc) and antidote (Phd) proteins of the plasmid addiction system of bacteriophage P1 were pu- rified as a complex. Cocrystals of the complex containeda 2:1 molar ratio of Phd:Doc as assayed by dye bindingfollowing SDS-polyacrylamide gel electrophoresis andas determined by amino acid analysis. Gel filtration andanalytical ultracentrifugation revealed that the two ad-diction proteins interact in solution to form a P 2D tri- mer composed of one Doc and two Phd molecules. Theseresults support a model in which Phd inhibits the toxicactivity of Doc by direct binding. Circular dichroismexperiments showed that changes in secondary struc-ture accompany formation of the heterotrimeric com-plex, raising the possibility that Phd may act by an al-losteric mechanism. Studies of Phd and Doc moleculeslabeled with fluorescent energy donor and acceptorgroups gave an equilibrium dissociation constant ofabout 0.8 mM2and a very short, sub second half-life of complex dissociation. As a consequence, low concentra-tions of free Doc toxin are likely to be present bothtransiently and in the steady state in cells containingthe Phd antidote, making mechanisms of single-hit Doctoxicity improbable. The plasmid addiction system of bacteriophage P1 consists of two proteins: a toxin known as Doc (death on cure) and an antidote named Phd (prevent host death) (1). Bacteriophage P1 lysogenizes Escherichia coli as a low copy plasmid that is inherited in a remarkably stable fashion, in part, because the plasmid addiction system kills cells that have lost the P1 ge-nome (1–5). The mechanism of Doc toxicity is unknown. Phd is,however, known to be degraded in a fashion dependent on thehost ClpXP protease system, and continual synthesis of Phd istherefore required to counteract Doc toxicity (3). In daughtercells that have lost the P1 genome, degradation of Phd even-tually results in reduction of antidote activity to a level wherethe host is killed. Phd may exert its antitoxin activity bybinding Doc and physically blocking its interaction with cellu-lar target molecules (1) or by acting in an allosteric fashion toalter Doc structure and toxin activity. Alternatively, Phd mightexert its antidote activity indirectly by activating another pro-tein that neutralizes Doc. Some of the regulatory and biochemical properties of Phdhave been established. For example, Phd is a DNA-bindingprotein that represses transcription of its own gene as well asthat of Doc by binding to an operator DNA site that overlapsthe addiction promoter (4). Four molecules of Phd bind to theintact operator with dimers binding to adjacent 10-base pairsubsites (6). Phd has a predominantly a-helical structure when bound to DNA or at low temperatures but the freemonomeric protein has a T mof 25 °C and is largely denatured at 37 °C (6). Addiction systems involving a stable toxin and proteolyti- cally unstable antidote are also used by other low copy plas-mids and may be involved in the regulation of programmed celldeath in E. coli (7–13). Although direct interaction between the toxin and its antidote is assumed to be the mechanism ofinhibition for each of these systems, experimental evidence forcomplex formation has been presented only for the CcdA anti-dote and CcdB toxin of the F plasmid of E. coli (14). In the study of the phage P1 addiction proteins presented here, we demon-strate that Phd binds directly to Doc, forming a trimeric com-plex (P 2D) with one molecule of Doc and two molecules of Phd. Complex formation appears to be accompanied by changes insecondary structure.', 'page_label: 16813 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  Addiction systems involving a stable toxin and proteolyti- cally unstable antidote are also used by other low copy plas-mids and may be involved in the regulation of programmed celldeath in E. coli (7–13). Although direct interaction between the toxin and its antidote is assumed to be the mechanism ofinhibition for each of these systems, experimental evidence forcomplex formation has been presented only for the CcdA anti-dote and CcdB toxin of the F plasmid of E. coli (14). In the study of the phage P1 addiction proteins presented here, we demon-strate that Phd binds directly to Doc, forming a trimeric com-plex (P 2D) with one molecule of Doc and two molecules of Phd. Complex formation appears to be accompanied by changes insecondary structure. P 2D complexes form at low micromolar concentrations and are very short lived, ensuring a dynamicequilibrium between bound and free forms within the cell. MATERIALS AND METHODS Protein Expression and Purification—E. coli strain X90/pRDM032, which expresses Phd and Doc under Ptaccontrol (4), was grown at 37 °C to an A600of approximately 0.7 in Luria-Bertani broth containing 100 mg/ml ampicillin, and protein expression was induced by the addition of 100 mg/ml isopropyl-1-thio- b-D-galactopyranoside for 30 min. Cells were harvested by centrifugation (3,500 3gfor 30 min at 4 °C), and the pellets were frozen at 280 °C before purification. Cell pellets were thawed at 4 °C, resuspended in buffer A (50 m MNaCl, 50 m MTris-HCl (pH 7.4), 1 m MEDTA), and phenylmethylsulfonyl fluoride was added to a final concentration of 50 mM. Cells were lysed by sonication, and the resulting solution was centrifuged at 4 °C for 10 min at 13,000 3g. The supernatant was applied to a MonoS HR 5/5 FPLC column (Amersham Pharmacia Biotech) equilibrated in buffer A, which was eluted withgradient from 50 to 1,000 m MNaCl in 50 m MTris-HCl, pH 7.4. Elution was monitored by A280. A peak containing Phd and Doc eluted at approximately 600 m MNaCl. This peak was applied to a semi-prepar- ative reverse-phase C18column, and Phd and Doc were separated using a gradient from 0 to 80% acetonitrile in 0.1% trifluoracetic acid. Fol-lowing this step, the proteins were greater than 95% pure as assessedby Coomassie staining of SDS-polyacrylamide gels. Characterization of Purified Protein— The masses of the purified proteins were determined by MALDI-TOF 1mass spectrometry using a Voyager-DE STR Biospectrometry Workstation (PerSeptive BioSys-tems) with sinapinic acid (3,5-dimethoxy-4-hydroxycinnamic acid) asthe matrix, linear mode operation, a grid voltage at 91.0%, and a pulsedelay time of 150 ns. Protein concentrations were determined by meas-uring tyrosine absorbance at both pH 7 and in 0.1 MKOH using a* Supported by National Institutes of Health Grants AI-15706 and AI-16892. The costs of publication of this article were defrayed in part by the payment of page charges. This article must therefore be herebymarked “advertisement” in accordance with 18 U.S.C. Section 1734solely to indicate this fact. ‡ Supported by postdoctoral fellowships from EMBO and the Human Frontiers Science Foundation Program. § To whom correspondence should be addressed: Massachusetts Institute of Technology 68–571, Cambridge, MA 02139. E-mail:bobsauer@mit.edu.1The abbreviations used are: MALDI-TOF, matrix-assisted laser de- sorption ionization-time-of-flight; HPLC, high pressure liquidchromatography.THEJOURNAL OF BIOLOGICAL CHEMISTRY Vol.', 'page_label: 16813 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  The costs of publication of this article were defrayed in part by the payment of page charges. This article must therefore be herebymarked “advertisement” in accordance with 18 U.S.C. Section 1734solely to indicate this fact. ‡ Supported by postdoctoral fellowships from EMBO and the Human Frontiers Science Foundation Program. § To whom correspondence should be addressed: Massachusetts Institute of Technology 68–571, Cambridge, MA 02139. E-mail:bobsauer@mit.edu.1The abbreviations used are: MALDI-TOF, matrix-assisted laser de- sorption ionization-time-of-flight; HPLC, high pressure liquidchromatography.THEJOURNAL OF BIOLOGICAL CHEMISTRY Vol. 274, No. 24, Issue of June 11, pp. 16813–16818, 1999 © 1999 by The American Society for Biochemistry and Molecular Biology, Inc. Printed in U.S.A. This paper is available on line at http://www.jbc.org 16813 This is an Open Access article under the CC BY  license.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:48 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'113'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994811'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'311ms'), (b'x-request-id', b'd152ea2ba0f7738daf13be0a266d3892'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249504fc25d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:48 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'113'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994811'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'311ms'), (b'x-request-id', b'd152ea2ba0f7738daf13be0a266d3892'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249504fc25d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 16814 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  Hewlett Packard 8452A diode array spectrophotometer with extinction coefficients per tyrosine of 1,394 M21cm21(274 nm at pH 7) and 2377 M21cm21(294 nm in 0.1 MKOH) for Phd (1 Tyr) and Doc (4 Tyr). Circular dichroism spectra were obtained using an AVIV 60DS spec-trapolarimeter equipped with a temperature-controlled sample holderand a 10-mm pathlength cuvette. Thermal denaturation experimentswere performed in a buffer containing 50 m MTris-HCl (pH 7.4), 100 m M NaCl, and 0.1 m MEDTA; samples were equilibrated at 1 °C tempera- ture intervals for 1 min, and the ellipticity at 222 nm was averaged for2 min. Fluorescence Labeling and Spectroscopy— Purified Phd protein was reacted with 2 equivalents of carboxytetramethylrhodamine succinimi-dyl ester (Molecular Probes, Eugene OR) in 0.5 MHEPES (pH 7), 100 mMNaCl, for 24 h at room temperature in the dark (15). Doc was labeled in the same way except using carboxyfluorescein succinimidylester. Unreacted dyes were separated from labeled protein by gel fil-tration chromatography on a QuickSpin G-25 Sephadex column (RocheMolecular Biochemicals). Equilibrium binding, assayed by fluorescenceresonance energy transfer, was performed in 50 m MTris-HCl (pH 7.4), 100 m MNaCl, and 1 m MEDTA using a FluoroMax2 fluorescence spec- trophotometer (Jobin Yvon • Spex, Edison, NJ). Kinetic experimentswere performed using an Applied Photophysics DX17.MV stopped-flowinstrument and were monitored by changes in fluorescence using abandpass filter (520 610 nm). Crystallization and Analysis of the Phd zDoc Complex—Crystals of a PhdzDoc complex grew after several days at 4 °C from solutions con-taining purified protein at concentrations of 5 mg/ml or higher in 50 m M TriszCl (pH 8.0), 500 m MNaCl, and 1 m MEDTA. Individual crystals were thoroughly washed, boiled in SDS sample buffer, and analyzed bySDS-polyacrylamide gel electrophoresis, confirming the presence ofboth Phd and Doc in the crystals at 99% or higher purity. Amino acidanalysis of washed and redisolved crystals was performed by the MITBiopolymers laboratory. Ultracentrifugation and Gel Filtration—Analytical ultracentrifuga- tion experiments were performed using a Beckman Optima XL-A cen-trifuge. Solutions containing Phd and Doc from dissolved crystals werecentrifuged overnight at 15,000 rpm. After reaching equilibrium (estab-lished by unchanged A 274readings at 1 h intervals), absorbance profiles were determined at 274 nm, and five scans were averaged for analysis.Data from experiments in which ln( A) versus r 2plots showed curvature were fitted by nonlinear least-squares methods to a 2P 1DNP2D equilbrium model using Eq. 1.Cr5Cb,Phdexp~A PhdMPhd~r22rb2!!14Cb,Docexp~A DocMDoc~r22rb2!! 16~1/K d!Cb,Doc~Cb,Phd!2exp~A Complex MComplex ~r22rb2!!1e(Eq. 1) where Cris the total concentration and the terms Cb,xrepresent the concentrations of each species at a reference radial position. Ax5(12 yXr)v2/2RT, where yXis the partial specific volume of each species (calculated from its sequence), ris the solution density, vis the rotor angular velocity in radians per second, Ris the gas constant, and Tis the absolute temperature. Kdis the equilibrium dissociation constant, Mxis the molecular mass of each species, ris the radial position, rbis a reference radial position, and eis a small baseline correction term.', 'page_label: 16814 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  !14Cb,Docexp~A DocMDoc~r22rb2!! 16~1/K d!Cb,Doc~Cb,Phd!2exp~A Complex MComplex ~r22rb2!!1e(Eq. 1) where Cris the total concentration and the terms Cb,xrepresent the concentrations of each species at a reference radial position. Ax5(12 yXr)v2/2RT, where yXis the partial specific volume of each species (calculated from its sequence), ris the solution density, vis the rotor angular velocity in radians per second, Ris the gas constant, and Tis the absolute temperature. Kdis the equilibrium dissociation constant, Mxis the molecular mass of each species, ris the radial position, rbis a reference radial position, and eis a small baseline correction term. This equation takes the relative extinction coefficients of Phd (one), Doc(four), and the P 2D complex (six) into account. Gel filtration experiments were performed by chromatographing 200- ml samples on a 10 3300 mm Superdex 75 HR 10/30 column (separation range 3–70 kDa) using a flow rate of 0.5 ml/min. Absorb-ance was monitored at 280 nm, and the column was calibrated using gelfiltration molecular weight standards (Bio-Rad). RESULTS Copurification of Phd and Doc— As a first step in studying potential interactions between Phd and Doc, both proteins were purified from E. coli strain X90/pRDM032, a cell expressing the phage P1 addiction proteins under Ptacpromoter control (4). As shown in Fig. 1 A, Doc protein and the major portion of the Phd protein coeluted during ion-exchange chromatography on aMonoS column; a smaller quantity of Phd alone also eluted ata lower salt concentration than the Phd zDoc complex. During purification from cells without Doc, Phd eluted only in the lowsalt position. Further purification by reverse-phase HPLCchromatography under denaturing conditions separated Docfrom Phd and resulted in purification of each protein to greaterthan 95% homogeneity (Fig. 1B). The identity of each proteinwas confirmed by MALDI-TOF mass spectrometry (Fig. 2). Theobserved masses (8,130 Da for Phd; 13,578 Da for Doc) are FIG.1 .Purification. A,Phd and Doc coelute during chromatography on a MonoS ion-exchange column developed with a gradient of NaCl; B, separation of Phd and Doc on a C18HPLC reverse-phase column devel- oped with a gradient of acetonitrile. FIG.2 .MALDI-TOF mass spectrometry. HPLC-purified proteins have masses within 2 Da of those expected from the unmodified Phd (8,128 Da) and Doc (13,579 Da) protein sequences.Doc and Phd Form a Heterotrimeric Complex 16814', 'page_label: 16815 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  within 1 or 2 Da of those expected from the protein sequences, indicating that the formyl group of the N-terminal fMet residueis removed post-translationally for each protein. Phd and Doc Form a 2:1 Complex in Crystals— Solutions containing high concentrations of purified Phd and Doc formedneedle-shaped crystals that grew to a length of several milli-meters after 24–48 h at 4 °C (Fig. 3 A). Following washing, SDS-polyacrylamide gel electrophoresis showed that individualcrystals contained both Phd and Doc (Fig. 3 B). Densitometry of the Coomassie Blue-stained gel indicated that the Phd and Docproteins were present in the crystal in a 2:1 molar ratio (Fig.3C). Amino acid analysis of protein from washed crystals alsosupported a 2:1 molar ratio of Phd:Doc in the crystals (Table I).Hence, in the crystalline state, Phd and Doc appear to form amacromolecular complex containing twice as many Phd mole-cules as Doc molecules. The smallest oligomer consistent withthis stoichiometry is a P 2D trimer, but P4D2hexamers, P6D3 nonamers, et cetera are also consistent with these data. Oligomeric State in Solution—In gel filtration chromatogra- phy on Superdex 75, purified Doc eluted at an approximatevolume expected for a monomer (data not shown); Phd hadbeen previously shown to be monomeric in solution (6). Todetermine the solution form of PhdzDoc complexes, proteinfrom dissolved cocrystals was also chromatographed on Super-dex 75. At high loading concentrations (80 mMin Doc monomer equivalents), a major peak of approximately 30 kDa was ob-served (Fig. 4A). This value is consistent with formation of aP 2D solution trimer (expected molecular mass, 29.8 kDa). Some trailing asymmetry was evident in the Fig. 4 Acolumn profile, suggesting that a small fraction of P2D complexes dissociated during chromatography. Indeed, when the sample was diluted10-fold immediately before chromatography (loading concen-tration 8 mMin Doc monomer equivalents), less of the P2D peak was present and more prominent, lower molecular weightpeaks corresponding to Doc and Phd alone were observed (datanot shown). These experiments suggest that Phd zDoc com- plexes dissociate in the micromolar concentration range withkinetics on the time scale of minutes or faster. Gel filtration ofa mixture of Phd:Doc in a 4:1 ratio revealed no species largerthan 30 kDa but showed significant free Phd (data not shown).These results show that the P 2D complex is the major solution form, even in the presence of excess Phd. As a second assay of solution oligomeric form, protein from dissolved PhdzDoc cocrystals (80 mMin Doc equivalents) was analyzed by analytical ultracentrifugation. These data were fitbest by a 2:1 ratio of Phd:Doc and showed significant deviationsfrom the behavior expected for 1:1 or 2:4 complexes of Phd:Doc(Fig. 4B). Equilibrium centrifugation experiments performedusing lower protein concentrations (loading concentration of 20 mMin Doc equivalents) showed curvature in the ln(A274)versus r2plot, indicative of some complex dissociation. Fitting of this data to a 2P 1DNP2D model gave an equilibrium dissociation constant of 1.0 60.3mM2at 20 °C (Fig. 4C). Changes in Secondary Structure Induced by Complex Forma- tion—The circular dichroism (CD) spectrum of purified Doc hadfeatures expected for a protein containing approximately 50% a-helix (Fig. 5 A). Melting experiments in a neutral buffer at protein concentrations of either 1 or 15 mMhad similar shapes and midpoints (Fig. 5B), as expected if Doc is monomeric. Thenative structure of Doc ( T m60 °C) is quite stable in comparison to Phd ( Tm25 °C) (6). In fact, Phd is almost completely dena- tured at 37 °C, whereas Doc is native at this temperature.', 'page_label: 16815 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  Fitting of this data to a 2P 1DNP2D model gave an equilibrium dissociation constant of 1.0 60.3mM2at 20 °C (Fig. 4C). Changes in Secondary Structure Induced by Complex Forma- tion—The circular dichroism (CD) spectrum of purified Doc hadfeatures expected for a protein containing approximately 50% a-helix (Fig. 5 A). Melting experiments in a neutral buffer at protein concentrations of either 1 or 15 mMhad similar shapes and midpoints (Fig. 5B), as expected if Doc is monomeric. Thenative structure of Doc ( T m60 °C) is quite stable in comparison to Phd ( Tm25 °C) (6). In fact, Phd is almost completely dena- tured at 37 °C, whereas Doc is native at this temperature. By comparing the CD spectra of a mixture of 2 mMPhd and 1 mMDoc with the summed spectra of the uncomplexed proteins, secondary structural changes induced by complex formationwere assessed. At 4 °C, where Phd and Doc by themselves areboth native, the mixture had slightly less ellipticity than theisolated proteins, suggesting that complex formation leads tosome change in structure (Fig. 6A). At 37 °C, where Phd byitself is primarily unfolded, the P 2D complex had greater ellip- ticity than the individual proteins, suggesting an increase insecondary structure upon binding (Fig. 6B). In the latter in-stance, it seems probable that at least a portion of Phd folds asit binds to Doc. Fitting of titration experiments, monitored bychanges in CD ellipticity at 37 °C, gave an equilibrium disso-ciation constant of 0.7 60.1 mM2(Fig. 6C). Affinity and Kinetics of Complex Formation—To study the affinity of complex formation by another method, we performedfluorescence resonance energy transfer experiments by titrat-ing increasing amounts of rhodamine-labeled Doc (the acceptormolecule) against a constant amount of fluorescein-labeled Phd FIG.3 .Crystals of PhdzDoc complex. A, photograph of crystals; B, SDS-polyacrylamide gel electrophoresis of washed and dissolved crys- tals shows presence of both Phd and Doc; C,densitometry of the Coo- massie Blue-stained polyacrylamide gel is most consistent with a 2:1ratio of Phd:Doc.T ABLE I Amino acid analysis of dissolved co-crystals Amino acidNumber of amino acids Doc:PhdTheoretical value 1:1 Doc:Phd ratioTheoretical value 1:2 Doc:Phd ratioExperimental value Asx 12:9 21 30 30.7 Glx 11:10 21 31 33.0 Ser 10:5 15 20 18.5 Gly 10:3 13 16 17.4 Arg 11:6 17 23 25.3 Thr 7:5 12 17 17.8 Ala 17:9 26 35 34.9 Pro 4:1 5 6 6.8 Tyr 4:1 5 6 6.3 Val 9:6 15 21 20.5 Met 2:1 3 4 4.7 Ile 8:3 11 14 14.6 Leu 13:6 19 25 25.8 Phe 4:4 8 12 12.9 Lys 1:4 5 9 10.4 His 3:0 3 3 3.0a aHis value was used to normalize the measurement, because there are 3 histidines in Doc and none in Phd.Doc and Phd Form a Heterotrimeric Complex 16815', 'page_label: 16816 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  (the donor molecule). Because energy transfer depends on the inverse sixth power of the distance between the fluorescentdyes, any change in fluorescence should arise from interactionsbetween the proteins. Fig. 7 Ashows fluorescence spectra, and Fig. 7B shows the change in fluorescence at 25 °C as a function of Phd concentration. These experimental data were fit best bya binding curve calculated for a model 2P1DNP2D Kd5@P]2[D]/[P 2D] (Eq. 2) that is second-order in Phd and first-order in Doc, with anequilibrium dissociation constant of 0.83 60.02 mM2. To determine the half-life of complex dissociation, fluores- cently labeled P2D complexes were mixed with excess unla- beled Phd in a stopped-flow instrument, and the change influorescence was monitored (Fig. 7C). The rate constant of thisexchange process, which should be limited by dissociation was42 (6 7) s 21indicating that complex dissociation must occur very rapidly. Similar results were obtained when the dissocia-tion reaction was initiated by mixing fluorescently labeled P 2D complexes with unlabeled Doc. The rapid dissociation rate isconsistent with the tailing seen in gel filtration. From theequilibrium constant and a dissociation rate constant, a third-order association rate constant of approximately 6 z10 13M22s21 can be calculated. DISCUSSION Efficient inhibition of the toxic activity of Doc by the Phd antidote is essential for maintaining the viability of bacterioph-age P1 lysogens of E. coli (1) but little concrete information concerning the molecular mechanism of this inhibition hasbeen available. The results presented here show that Phd andDoc interact directly to form a heterotrimeric complex in solu-tion. In the absence of more detailed information, it is notpossible to know whether formation of the Phd zDoc complex blocks the toxic activity of Doc by sterically occluding interac-tions with its cellular target or if formation of the complexalters the structure of Doc in a way that prevents this interac-tion. It is intriguing, however, that formation of the PhdzDoccomplex appears to result in a small change in secondarystructure. Were this alteration of structure to involve a portionof Doc, then this could serve as the mechanism of toxinneutralization. The PhdzDoc complex is a P 2D heterotrimer. Heterotrimers FIG.4 .Oligomeric state. A,the PhdzDoc complex chromatographs at an apparent molecular mass of 30 kDa in gel filtration of dissolved cocrystals (80 mMin Doc equivalents) on a Superdex 75 column. The elution positions of ovalbumin (44 kDa) and myoglobin (17 kDa) areindicated. B,equilibrium distribution of Phd zDoc complex absorbance (loading concentration 80 mMin Doc equivalents) during analytical centrifugation at 15,000 rpm, 20 °C. Calculated distributions for P2D complexes (solid line), P4D2complexes (dashed line), and PD complexes (dashed-dotted line) are shown. C,equilibrium distribution of Phd zDoc complex absorbance (loading concentration 20 mMin Doc equivalents) during analytical centrifugation at 15,000 rpm, 20 °C. The solid line iscalculated for the reaction 2P 1DNP 2D with a Kdof 0.96 mM2. FIG.5 .Doc secondary structure and stability. A,far ultraviolet CD spectrum of purified Doc (5 mM) at 4 °C suggests an a-helical content of approximately 50%. B,Doc at concentrations of 1 and 15 mMunder- goes cooperative thermal denaturation with a Tmof 60 °C as monitored by changes in CD ellipticity at 222 nm. The CD signals for the twoprotein concentrations were normalized to allow comparison. The bufferfor the experiments shown in panels A andBwas 50 m MTris-HCl (pH 7.4), 100 m MNaCl, and 0.1 m MEDTA.Doc and Phd Form a Heterotrimeric Complex 16816', \"page_label: 16817 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  are not common in biological systems and must be assembled asymmetrically. A well known example with a 2:1 ratio ofcomponents is the heterotrimer of growth hormone with twoextracellular receptor domains, in which each receptor binds toa distinct surface of growth hormone (16). The oligomeric formof the antidote-toxin complex for the ccdsystem is hexameric with two dimers of the toxin and one dimer of the antidote (14),but this architecture also requires asymmetric constructionand, indeed, can be considered as a heterotrimer of dimers. Wefind it intriguing that there are 2:1 or 1:2 ratios of antidote:toxin in both the P1 addiction system and F ccd system. Whether these and other addiction systems share commonstructural features is not known. Because simultaneous three-body collisions between protein monomers of Phd and Doc is not a statistically viable mecha-nism of assembly, formation of the P 2D complex must proceed either through a P2or a PD intermediate. Neither of these dimeric species has been detected in solution, implying that ifeither species can form then it must be unstable and poorlypopulated at equilibrium. We note, however, that two mono-mers of Phd bind cooperatively to operator subsites, suggestingthat Phd subunits interact with each other in the DNA-boundstate (6). It is also worth noting that the P 2D solution complex may also be an active operator binding species; using a mutantform of Doc, Magnuson and Yarmolinsky (5) have suggestedthat a 2:1 ratio of Phd:Doc night occur in operator complexes. The P 2D complex of Phd and Doc is unstable by comparison with many macromolecular complexes. Kdvalues for complexdissociation of 0.7–1.0 mM2were obtained from analytical ul- tracentrifugation at 20 °C, from fluorescence resonance energytransfer at 25 °C, and from changes in CD ellipticity at 37 °C.Dissociation of unlabeled Phd zDoc complexes to free compo- nents was also observed in the micromolar concentration rangeby gel filtration. P 2D complexes equilibrate with free subunits on a subsecond time scale in vitro, which should ensure rapid equilibration of the bound and free intracellular pools of Phdand Doc. There is an approximate 3-fold ratio of Phd to Doc in over- producing cells. Because the phd anddocgenes are part of the same operon, a similar ratio would be expected in P1 lysogens.Moreover, because Phd represses its own synthesis and that ofDoc (1–5), P1 lysogens are unlikely to contain very high con-centrations of either protein. To ensure 90–99.9% binding ofDoc with a K dof 0.7–1.0 mM2, we calculate that the free Phd concentration would need to be from 2.2 to 36 mM('2,200 to 36,000 molecules/cell). However, under these conditions therewould still be free Doc concentrations of 260–22 n M. Hence, it is important to view the Phd-Doc interaction as buffering thefree concentration of Doc rather than eliminating free Doc from FIG.6 .Changes in CD ellipticity accompany complex forma- tion. A,CD spectra, at 4 °C, of 1 mMDoc and 2 mMPhd before and after mixing in a tandem cuvette; B,mixing experiment at 37 °C. Other conditions identical to panel A. C,increasing quantities of Doc were added to 2 mMPhd protein and the change in CD ellipticity at 222 nm was monitored. The solid line is calculated for the reaction 2P 1DN P2D with a Kdof 0.72 mM2. The buffer for all experiments was 50 m M Tris-HCl (pH 7.4), 100 m MNaCl, and 0.1 m MEDTA. FIG.7 .Resonance energy transfer. A,fluorescence spectra of rho- damine-labeled Phd (0.8 mM), fluorescein-labeled Doc (0.02 mM), and a mixture of both proteins; B,complex formation assayed by resonance energy transfer.\", 'page_label: 16817 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  Other conditions identical to panel A. C,increasing quantities of Doc were added to 2 mMPhd protein and the change in CD ellipticity at 222 nm was monitored. The solid line is calculated for the reaction 2P 1DN P2D with a Kdof 0.72 mM2. The buffer for all experiments was 50 m M Tris-HCl (pH 7.4), 100 m MNaCl, and 0.1 m MEDTA. FIG.7 .Resonance energy transfer. A,fluorescence spectra of rho- damine-labeled Phd (0.8 mM), fluorescein-labeled Doc (0.02 mM), and a mixture of both proteins; B,complex formation assayed by resonance energy transfer. The solid line is calculated for the reaction 2P 1DN P2D with a Kdof 0.81 mM2.C,kinetics of dissociation of complexes containing rhodamine-labeled Phd and fluorescein-labeled Doc moni- tored by fluorescence after addition of a 250-fold excess of unlabeledPhd. The solid line is a single exponential fit with a rate constant of 42 s 21. Experiments in all panels were performed at 25 °C in a buffer containing 50 m MTris-HCl (pH 7.4), 100 m MNaCl, and 1 m MEDTA.Doc and Phd Form a Heterotrimeric Complex 16817', 'page_label: 16818 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  the cell. If the assumptions that underlie these rough calcula- tions are sound, then free Doc at submicromolar levels must berelatively innocuous to cells. Otherwise, P1 lysogens wouldhave a significant growth disadvantage relative to nonlyso-gens. However, higher levels of free Doc (presumably in themicromolar range) must be lethal for the plasmid addictionmechanism to function. These considerations make it ex-tremely unlikely that the mechanism of Doc toxicity involvessingle hit enzymatic lethality and suggest that Doc may revers-ibly inhibit some required cellular activity in a cooperativefashion that results in a steep concentration dependence oftoxicity. In P1 lysogens, Phd binding spares the cell from Doc-medi- ated killing (3). However, Doc binding induces structure indenatured Phd at physiological temperatures, and thus itseems likely that Doc binding would also help protect Phd fromdegradation. Degradation of Phd is mediated directly or indi-rectly by the ClpXP protease (3), which consists of ClpP prote-ase subunits and regulatory ClpX ATPase subunits (17–19).ClpX is responsible for substrate recognition by ClpXP and hasbeen shown to recognize specific peptide sequences at the Ctermini of several substrate proteins (20–22). If ClpXP directlydegrades Phd, then C-terminal substrate recognition is proba-bly also involved. By this model, formation of the P 2D complex with Doc could prevent degradation of Phd by sequesteringC-terminal sequences required for ClpX recognition and per-haps by stabilizing at least some parts of Phd in a nativestructure. We note, however, that the rapid dissociation rateand relatively high equilibrium dissociation constant for P 2D complexes makes it likely that a pool of free Phd, in a largelydenatured form, is present even when Doc is present in P1lysogens. If Phd were too good a substrate for ClpXP, thendegradation of the free pool and mass action to produce morefree Phd could lead to Doc toxicity even in lysogens. Hence, itseems likely that Phd should be a rather poor substrate forClpXP. Indeed, in ClpXP 1strains, Phd is degraded relatively slowly over the course of several cell generations (3) Acknowledgments—We thank Alok Srivastava for help and advice, Roy Magnuson and Michael Yarmolinsky for pRDM032, and Carl Pabo, Harvey Lodish, and Tania Baker for use of equipment. REFERENCES 1. Lehnherr, H., Maguin, E., Jafri, S. & Yarmolinsky, M. B. (1993) J. Mol. Biol. 233, 414–428 2. Rosner, J. L. (1972) Virology 49,679–680 3. Lehnherr, H. & Yarmolinsky, M. B. (1995) Proc. Natl. Acad. Sci. U. S. A. 92, 3274–3277 4. Magnuson, R., Lehnherr, H., Mukhopadhyay, G. & Yarmolinsky, M. B. (1996) J. Biol. Chem. 271, 18705–18710 5. Magnuson, R. & Yarmolinsky, M. B. (1998) J. Bacteriol. 180, 6342–6351 6. Gazit, E. & Sauer, R. T. (1999) J. Biol. Chem. 274, 2652–2657 7. Loh, S. M., Cram, D. S. & Skurray, R. A. (1988) Gene 66,259–268 8. Young, C., Prince, A. S. & Figurski, D. H. (1985) Proc. Natl. Acad. Sci. U. S. A. 82,7374–7378 9. Gerdes, K., Rasmussen, P. B. & Molin, S. (1986) Proc. Natl. Acad. Sci. U. S. A.', 'page_label: 16818 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  180, 6342–6351 6. Gazit, E. & Sauer, R. T. (1999) J. Biol. Chem. 274, 2652–2657 7. Loh, S. M., Cram, D. S. & Skurray, R. A. (1988) Gene 66,259–268 8. Young, C., Prince, A. S. & Figurski, D. H. (1985) Proc. Natl. Acad. Sci. U. S. A. 82,7374–7378 9. Gerdes, K., Rasmussen, P. B. & Molin, S. (1986) Proc. Natl. Acad. Sci. U. S. A. 83,3116–3120 10. Jensen, R. B. & Gerdes, K. (1995) Mol. Microbiol. 17,205–210 11. Aizenman, E., Engelberg-Kulka, H. & Glaser, G. (1996) Proc. Natl. Acad. Sci U. S. A. 93,6059–6063 12. Holc ˇı´k, M. & Iyer, V. N. (1997) Microbiology 143, 3403–3416 13. Engelberg-Kulka, H., Reches, M., Narasimhan, S., Schoulaker-Schwarz, R., Klemes, Y., Aizenman, E. & Glaser, G. (1998) Proc. Natl. Acad. Sci. U. S. A. 95,15481–15486 14. Van Melderen, L., Thi, M. H. D., Lecchi, P., Gottesman, S., Couturier, M. & Maurizi, M. R. (1996) J. Biol. Chem. 271, 27730–27738 15. Gazit, E., Burshtein, N., Ellar, D. J., Saywer, T. & Shai, Y. (1997) Biochemistry 36,15546–15554 16. de Vos, A. M., Ultsch, M. & Kossiakoff, A. A. (1992) Science 255, 306–312 17. Wawrzynow, A., Wojtkowiak, D., Marszalek, J., Banecki, B., Jonsen, M., Graves, B., Georgopoulos, C. & Zylicz, M. (1995) EMBO J. 14,1867–1877 18. Wang, J., Hartling, J. A. & Flanagan, J. M. (1997) Cell 91,447–456 19. Grimaud, R., Kessel, M., Beuron, F., Steven, A. C. & Maurizi, M. R. (1998) J. Biol. Chem. 273, 12476–12481 20. Levchenko, I., Smith, C. K., Walsh, N. P., Sauer, R. T. & Baker, T. A. (1997) Cell 91,939–947 21. Levchenko, I., Yamauchi, M. & Baker, T. A. (1997) Genes Dev. 11,1561–1572 22. Gottesman, S., Roche, E., Zhou, Y. & Sauer, R. T. (1998) Genes Dev. 12, 1338–1347Doc and Phd Form a Heterotrimeric Complex 16818', 'page_label: 167829 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The Role of O-antigen in P1 Transduction of Shigella ﬂexneri and Escherichia coli with its Alternative S’ Tail Fibre Yang W. Huan2, Jidapha Fa-arun2and Baojun Wang1,3⇑ 1-College of Chemical and Biological Engineering & ZJU-Hangzhou Global Scientiﬁc and Technological Innovation Center, Zhejiang University, Hangzhou 310058, China 2-Centre for Synthetic and Systems Biology, School of Biological Sciences, University of Edinburgh, Edinburgh EH9 3FF, United Kingdom 3-Research Centre of Biological Computation, Zhejiang Laboratory, Hangzhou 311100, China Correspondence to Baojun Wang: College of Chemical and Biological Engineering, Zhejiang University, Hangzhou 310058, China. baojun.wang@zju.edu.cn (B. Wang) @SynthSysEd ,@SBSatEd (B. Wang) https://doi.org/10.1016/j.jmb.2022.167829Edited by Tao Liu Abstract Enterobacteria phage P1 expresses two types of tail ﬁbre, S and S’. Despite the wide usage of phage P1 for transduction, the host range and the receptor for its alternative S’ tail ﬁbre was never determined. Here, aDS-cinDpac E. coli P1 lysogenic strain was generated to allow packaging of phagemid DNA into P1 phage having either S or S’ tail ﬁbre. P1(S’) could transduce phagemid DNA into Shigella ﬂexneri 2a 2457O, Shigella ﬂexneri 5a M90T and Escherichia coli O3 efﬁciently. Mutational analysis of the O- antigen assembly genes and LPS inhibition assays indicated that P1(S’) transduction requires at leastone O-antigen unit. E. coli O111:B4 LPS produced a high neutralising effect against P1(S’) transduction, indicating that this E. coli strain could be susceptible to P1(S’)-mediated transduction. Mutations in the O- antigen modiﬁcation genes of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T did not cause signiﬁcant changes to P1(S’) transduction efﬁciency. A higher transduction efﬁciency of P1(S’) improved the delivery of a cas9 antimicrobial phagemid into both S. ﬂexneri 2457O and M90T. These ﬁndings provide novel insights into P1 tropism-switching, by identifying the bacterial strains which are susceptible to P1(S’)-mediated transduction, as well as demonstrating its potential for delivering a DNA sequence-speciﬁcCas9 antimicrobial into clinically relevant S. ﬂexneri . /C2112022 The Author(s). Published by Elsevier Ltd. Introduction Enterobacteria phage P1 has a wide host range of Gram-negative Enterobacteriaceae , including multiple strains of Escherichia coli ,3–4,21,20,47,63Shi- gella ﬂexneri ,63S.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 16814 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  Hewlett Packard 8452A diode array spectrophotometer with extinction coefficients per tyrosine of 1,394 M21cm21(274 nm at pH 7) and 2377 M21cm21(294 nm in 0.1 MKOH) for Phd (1 Tyr) and Doc (4 Tyr). Circular dichroism spectra were obtained using an AVIV 60DS spec-trapolarimeter equipped with a temperature-controlled sample holderand a 10-mm pathlength cuvette. Thermal denaturation experimentswere performed in a buffer containing 50 m MTris-HCl (pH 7.4), 100 m M NaCl, and 0.1 m MEDTA; samples were equilibrated at 1 °C tempera- ture intervals for 1 min, and the ellipticity at 222 nm was averaged for2 min. Fluorescence Labeling and Spectroscopy— Purified Phd protein was reacted with 2 equivalents of carboxytetramethylrhodamine succinimi-dyl ester (Molecular Probes, Eugene OR) in 0.5 MHEPES (pH 7), 100 mMNaCl, for 24 h at room temperature in the dark (15). Doc was labeled in the same way except using carboxyfluorescein succinimidylester. Unreacted dyes were separated from labeled protein by gel fil-tration chromatography on a QuickSpin G-25 Sephadex column (RocheMolecular Biochemicals). Equilibrium binding, assayed by fluorescenceresonance energy transfer, was performed in 50 m MTris-HCl (pH 7.4), 100 m MNaCl, and 1 m MEDTA using a FluoroMax2 fluorescence spec- trophotometer (Jobin Yvon • Spex, Edison, NJ). Kinetic experimentswere performed using an Applied Photophysics DX17.MV stopped-flowinstrument and were monitored by changes in fluorescence using abandpass filter (520 610 nm). Crystallization and Analysis of the Phd zDoc Complex—Crystals of a PhdzDoc complex grew after several days at 4 °C from solutions con-taining purified protein at concentrations of 5 mg/ml or higher in 50 m M TriszCl (pH 8.0), 500 m MNaCl, and 1 m MEDTA. Individual crystals were thoroughly washed, boiled in SDS sample buffer, and analyzed bySDS-polyacrylamide gel electrophoresis, confirming the presence ofboth Phd and Doc in the crystals at 99% or higher purity. Amino acidanalysis of washed and redisolved crystals was performed by the MITBiopolymers laboratory. Ultracentrifugation and Gel Filtration—Analytical ultracentrifuga- tion experiments were performed using a Beckman Optima XL-A cen-trifuge. Solutions containing Phd and Doc from dissolved crystals werecentrifuged overnight at 15,000 rpm. After reaching equilibrium (estab-lished by unchanged A 274readings at 1 h intervals), absorbance profiles were determined at 274 nm, and five scans were averaged for analysis.Data from experiments in which ln( A) versus r 2plots showed curvature were fitted by nonlinear least-squares methods to a 2P 1DNP2D equilbrium model using Eq. 1.Cr5Cb,Phdexp~A PhdMPhd~r22rb2!!14Cb,Docexp~A DocMDoc~r22rb2!! 16~1/K d!Cb,Doc~Cb,Phd!2exp~A Complex MComplex ~r22rb2!!1e(Eq. 1) where Cris the total concentration and the terms Cb,xrepresent the concentrations of each species at a reference radial position. Ax5(12 yXr)v2/2RT, where yXis the partial specific volume of each species (calculated from its sequence), ris the solution density, vis the rotor angular velocity in radians per second, Ris the gas constant, and Tis the absolute temperature. Kdis the equilibrium dissociation constant, Mxis the molecular mass of each species, ris the radial position, rbis a reference radial position, and eis a small baseline correction term.', 'page_label: 16814 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  !14Cb,Docexp~A DocMDoc~r22rb2!! 16~1/K d!Cb,Doc~Cb,Phd!2exp~A Complex MComplex ~r22rb2!!1e(Eq. 1) where Cris the total concentration and the terms Cb,xrepresent the concentrations of each species at a reference radial position. Ax5(12 yXr)v2/2RT, where yXis the partial specific volume of each species (calculated from its sequence), ris the solution density, vis the rotor angular velocity in radians per second, Ris the gas constant, and Tis the absolute temperature. Kdis the equilibrium dissociation constant, Mxis the molecular mass of each species, ris the radial position, rbis a reference radial position, and eis a small baseline correction term. This equation takes the relative extinction coefficients of Phd (one), Doc(four), and the P 2D complex (six) into account. Gel filtration experiments were performed by chromatographing 200- ml samples on a 10 3300 mm Superdex 75 HR 10/30 column (separation range 3–70 kDa) using a flow rate of 0.5 ml/min. Absorb-ance was monitored at 280 nm, and the column was calibrated using gelfiltration molecular weight standards (Bio-Rad). RESULTS Copurification of Phd and Doc— As a first step in studying potential interactions between Phd and Doc, both proteins were purified from E. coli strain X90/pRDM032, a cell expressing the phage P1 addiction proteins under Ptacpromoter control (4). As shown in Fig. 1 A, Doc protein and the major portion of the Phd protein coeluted during ion-exchange chromatography on aMonoS column; a smaller quantity of Phd alone also eluted ata lower salt concentration than the Phd zDoc complex. During purification from cells without Doc, Phd eluted only in the lowsalt position. Further purification by reverse-phase HPLCchromatography under denaturing conditions separated Docfrom Phd and resulted in purification of each protein to greaterthan 95% homogeneity (Fig. 1B). The identity of each proteinwas confirmed by MALDI-TOF mass spectrometry (Fig. 2). Theobserved masses (8,130 Da for Phd; 13,578 Da for Doc) are FIG.1 .Purification. A,Phd and Doc coelute during chromatography on a MonoS ion-exchange column developed with a gradient of NaCl; B, separation of Phd and Doc on a C18HPLC reverse-phase column devel- oped with a gradient of acetonitrile. FIG.2 .MALDI-TOF mass spectrometry. HPLC-purified proteins have masses within 2 Da of those expected from the unmodified Phd (8,128 Da) and Doc (13,579 Da) protein sequences.Doc and Phd Form a Heterotrimeric Complex 16814', 'page_label: 16815 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  within 1 or 2 Da of those expected from the protein sequences, indicating that the formyl group of the N-terminal fMet residueis removed post-translationally for each protein. Phd and Doc Form a 2:1 Complex in Crystals— Solutions containing high concentrations of purified Phd and Doc formedneedle-shaped crystals that grew to a length of several milli-meters after 24–48 h at 4 °C (Fig. 3 A). Following washing, SDS-polyacrylamide gel electrophoresis showed that individualcrystals contained both Phd and Doc (Fig. 3 B). Densitometry of the Coomassie Blue-stained gel indicated that the Phd and Docproteins were present in the crystal in a 2:1 molar ratio (Fig.3C). Amino acid analysis of protein from washed crystals alsosupported a 2:1 molar ratio of Phd:Doc in the crystals (Table I).Hence, in the crystalline state, Phd and Doc appear to form amacromolecular complex containing twice as many Phd mole-cules as Doc molecules. The smallest oligomer consistent withthis stoichiometry is a P 2D trimer, but P4D2hexamers, P6D3 nonamers, et cetera are also consistent with these data. Oligomeric State in Solution—In gel filtration chromatogra- phy on Superdex 75, purified Doc eluted at an approximatevolume expected for a monomer (data not shown); Phd hadbeen previously shown to be monomeric in solution (6). Todetermine the solution form of PhdzDoc complexes, proteinfrom dissolved cocrystals was also chromatographed on Super-dex 75. At high loading concentrations (80 mMin Doc monomer equivalents), a major peak of approximately 30 kDa was ob-served (Fig. 4A). This value is consistent with formation of aP 2D solution trimer (expected molecular mass, 29.8 kDa). Some trailing asymmetry was evident in the Fig. 4 Acolumn profile, suggesting that a small fraction of P2D complexes dissociated during chromatography. Indeed, when the sample was diluted10-fold immediately before chromatography (loading concen-tration 8 mMin Doc monomer equivalents), less of the P2D peak was present and more prominent, lower molecular weightpeaks corresponding to Doc and Phd alone were observed (datanot shown). These experiments suggest that Phd zDoc com- plexes dissociate in the micromolar concentration range withkinetics on the time scale of minutes or faster. Gel filtration ofa mixture of Phd:Doc in a 4:1 ratio revealed no species largerthan 30 kDa but showed significant free Phd (data not shown).These results show that the P 2D complex is the major solution form, even in the presence of excess Phd. As a second assay of solution oligomeric form, protein from dissolved PhdzDoc cocrystals (80 mMin Doc equivalents) was analyzed by analytical ultracentrifugation. These data were fitbest by a 2:1 ratio of Phd:Doc and showed significant deviationsfrom the behavior expected for 1:1 or 2:4 complexes of Phd:Doc(Fig. 4B). Equilibrium centrifugation experiments performedusing lower protein concentrations (loading concentration of 20 mMin Doc equivalents) showed curvature in the ln(A274)versus r2plot, indicative of some complex dissociation. Fitting of this data to a 2P 1DNP2D model gave an equilibrium dissociation constant of 1.0 60.3mM2at 20 °C (Fig. 4C). Changes in Secondary Structure Induced by Complex Forma- tion—The circular dichroism (CD) spectrum of purified Doc hadfeatures expected for a protein containing approximately 50% a-helix (Fig. 5 A). Melting experiments in a neutral buffer at protein concentrations of either 1 or 15 mMhad similar shapes and midpoints (Fig. 5B), as expected if Doc is monomeric. Thenative structure of Doc ( T m60 °C) is quite stable in comparison to Phd ( Tm25 °C) (6). In fact, Phd is almost completely dena- tured at 37 °C, whereas Doc is native at this temperature.', 'page_label: 16815 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  Fitting of this data to a 2P 1DNP2D model gave an equilibrium dissociation constant of 1.0 60.3mM2at 20 °C (Fig. 4C). Changes in Secondary Structure Induced by Complex Forma- tion—The circular dichroism (CD) spectrum of purified Doc hadfeatures expected for a protein containing approximately 50% a-helix (Fig. 5 A). Melting experiments in a neutral buffer at protein concentrations of either 1 or 15 mMhad similar shapes and midpoints (Fig. 5B), as expected if Doc is monomeric. Thenative structure of Doc ( T m60 °C) is quite stable in comparison to Phd ( Tm25 °C) (6). In fact, Phd is almost completely dena- tured at 37 °C, whereas Doc is native at this temperature. By comparing the CD spectra of a mixture of 2 mMPhd and 1 mMDoc with the summed spectra of the uncomplexed proteins, secondary structural changes induced by complex formationwere assessed. At 4 °C, where Phd and Doc by themselves areboth native, the mixture had slightly less ellipticity than theisolated proteins, suggesting that complex formation leads tosome change in structure (Fig. 6A). At 37 °C, where Phd byitself is primarily unfolded, the P 2D complex had greater ellip- ticity than the individual proteins, suggesting an increase insecondary structure upon binding (Fig. 6B). In the latter in-stance, it seems probable that at least a portion of Phd folds asit binds to Doc. Fitting of titration experiments, monitored bychanges in CD ellipticity at 37 °C, gave an equilibrium disso-ciation constant of 0.7 60.1 mM2(Fig. 6C). Affinity and Kinetics of Complex Formation—To study the affinity of complex formation by another method, we performedfluorescence resonance energy transfer experiments by titrat-ing increasing amounts of rhodamine-labeled Doc (the acceptormolecule) against a constant amount of fluorescein-labeled Phd FIG.3 .Crystals of PhdzDoc complex. A, photograph of crystals; B, SDS-polyacrylamide gel electrophoresis of washed and dissolved crys- tals shows presence of both Phd and Doc; C,densitometry of the Coo- massie Blue-stained polyacrylamide gel is most consistent with a 2:1ratio of Phd:Doc.T ABLE I Amino acid analysis of dissolved co-crystals Amino acidNumber of amino acids Doc:PhdTheoretical value 1:1 Doc:Phd ratioTheoretical value 1:2 Doc:Phd ratioExperimental value Asx 12:9 21 30 30.7 Glx 11:10 21 31 33.0 Ser 10:5 15 20 18.5 Gly 10:3 13 16 17.4 Arg 11:6 17 23 25.3 Thr 7:5 12 17 17.8 Ala 17:9 26 35 34.9 Pro 4:1 5 6 6.8 Tyr 4:1 5 6 6.3 Val 9:6 15 21 20.5 Met 2:1 3 4 4.7 Ile 8:3 11 14 14.6 Leu 13:6 19 25 25.8 Phe 4:4 8 12 12.9 Lys 1:4 5 9 10.4 His 3:0 3 3 3.0a aHis value was used to normalize the measurement, because there are 3 histidines in Doc and none in Phd.Doc and Phd Form a Heterotrimeric Complex 16815', 'page_label: 16816 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  (the donor molecule). Because energy transfer depends on the inverse sixth power of the distance between the fluorescentdyes, any change in fluorescence should arise from interactionsbetween the proteins. Fig. 7 Ashows fluorescence spectra, and Fig. 7B shows the change in fluorescence at 25 °C as a function of Phd concentration. These experimental data were fit best bya binding curve calculated for a model 2P1DNP2D Kd5@P]2[D]/[P 2D] (Eq. 2) that is second-order in Phd and first-order in Doc, with anequilibrium dissociation constant of 0.83 60.02 mM2. To determine the half-life of complex dissociation, fluores- cently labeled P2D complexes were mixed with excess unla- beled Phd in a stopped-flow instrument, and the change influorescence was monitored (Fig. 7C). The rate constant of thisexchange process, which should be limited by dissociation was42 (6 7) s 21indicating that complex dissociation must occur very rapidly. Similar results were obtained when the dissocia-tion reaction was initiated by mixing fluorescently labeled P 2D complexes with unlabeled Doc. The rapid dissociation rate isconsistent with the tailing seen in gel filtration. From theequilibrium constant and a dissociation rate constant, a third-order association rate constant of approximately 6 z10 13M22s21 can be calculated. DISCUSSION Efficient inhibition of the toxic activity of Doc by the Phd antidote is essential for maintaining the viability of bacterioph-age P1 lysogens of E. coli (1) but little concrete information concerning the molecular mechanism of this inhibition hasbeen available. The results presented here show that Phd andDoc interact directly to form a heterotrimeric complex in solu-tion. In the absence of more detailed information, it is notpossible to know whether formation of the Phd zDoc complex blocks the toxic activity of Doc by sterically occluding interac-tions with its cellular target or if formation of the complexalters the structure of Doc in a way that prevents this interac-tion. It is intriguing, however, that formation of the PhdzDoccomplex appears to result in a small change in secondarystructure. Were this alteration of structure to involve a portionof Doc, then this could serve as the mechanism of toxinneutralization. The PhdzDoc complex is a P 2D heterotrimer. Heterotrimers FIG.4 .Oligomeric state. A,the PhdzDoc complex chromatographs at an apparent molecular mass of 30 kDa in gel filtration of dissolved cocrystals (80 mMin Doc equivalents) on a Superdex 75 column. The elution positions of ovalbumin (44 kDa) and myoglobin (17 kDa) areindicated. B,equilibrium distribution of Phd zDoc complex absorbance (loading concentration 80 mMin Doc equivalents) during analytical centrifugation at 15,000 rpm, 20 °C. Calculated distributions for P2D complexes (solid line), P4D2complexes (dashed line), and PD complexes (dashed-dotted line) are shown. C,equilibrium distribution of Phd zDoc complex absorbance (loading concentration 20 mMin Doc equivalents) during analytical centrifugation at 15,000 rpm, 20 °C. The solid line iscalculated for the reaction 2P 1DNP 2D with a Kdof 0.96 mM2. FIG.5 .Doc secondary structure and stability. A,far ultraviolet CD spectrum of purified Doc (5 mM) at 4 °C suggests an a-helical content of approximately 50%. B,Doc at concentrations of 1 and 15 mMunder- goes cooperative thermal denaturation with a Tmof 60 °C as monitored by changes in CD ellipticity at 222 nm. The CD signals for the twoprotein concentrations were normalized to allow comparison. The bufferfor the experiments shown in panels A andBwas 50 m MTris-HCl (pH 7.4), 100 m MNaCl, and 0.1 m MEDTA.Doc and Phd Form a Heterotrimeric Complex 16816', \"page_label: 16817 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  are not common in biological systems and must be assembled asymmetrically. A well known example with a 2:1 ratio ofcomponents is the heterotrimer of growth hormone with twoextracellular receptor domains, in which each receptor binds toa distinct surface of growth hormone (16). The oligomeric formof the antidote-toxin complex for the ccdsystem is hexameric with two dimers of the toxin and one dimer of the antidote (14),but this architecture also requires asymmetric constructionand, indeed, can be considered as a heterotrimer of dimers. Wefind it intriguing that there are 2:1 or 1:2 ratios of antidote:toxin in both the P1 addiction system and F ccd system. Whether these and other addiction systems share commonstructural features is not known. Because simultaneous three-body collisions between protein monomers of Phd and Doc is not a statistically viable mecha-nism of assembly, formation of the P 2D complex must proceed either through a P2or a PD intermediate. Neither of these dimeric species has been detected in solution, implying that ifeither species can form then it must be unstable and poorlypopulated at equilibrium. We note, however, that two mono-mers of Phd bind cooperatively to operator subsites, suggestingthat Phd subunits interact with each other in the DNA-boundstate (6). It is also worth noting that the P 2D solution complex may also be an active operator binding species; using a mutantform of Doc, Magnuson and Yarmolinsky (5) have suggestedthat a 2:1 ratio of Phd:Doc night occur in operator complexes. The P 2D complex of Phd and Doc is unstable by comparison with many macromolecular complexes. Kdvalues for complexdissociation of 0.7–1.0 mM2were obtained from analytical ul- tracentrifugation at 20 °C, from fluorescence resonance energytransfer at 25 °C, and from changes in CD ellipticity at 37 °C.Dissociation of unlabeled Phd zDoc complexes to free compo- nents was also observed in the micromolar concentration rangeby gel filtration. P 2D complexes equilibrate with free subunits on a subsecond time scale in vitro, which should ensure rapid equilibration of the bound and free intracellular pools of Phdand Doc. There is an approximate 3-fold ratio of Phd to Doc in over- producing cells. Because the phd anddocgenes are part of the same operon, a similar ratio would be expected in P1 lysogens.Moreover, because Phd represses its own synthesis and that ofDoc (1–5), P1 lysogens are unlikely to contain very high con-centrations of either protein. To ensure 90–99.9% binding ofDoc with a K dof 0.7–1.0 mM2, we calculate that the free Phd concentration would need to be from 2.2 to 36 mM('2,200 to 36,000 molecules/cell). However, under these conditions therewould still be free Doc concentrations of 260–22 n M. Hence, it is important to view the Phd-Doc interaction as buffering thefree concentration of Doc rather than eliminating free Doc from FIG.6 .Changes in CD ellipticity accompany complex forma- tion. A,CD spectra, at 4 °C, of 1 mMDoc and 2 mMPhd before and after mixing in a tandem cuvette; B,mixing experiment at 37 °C. Other conditions identical to panel A. C,increasing quantities of Doc were added to 2 mMPhd protein and the change in CD ellipticity at 222 nm was monitored. The solid line is calculated for the reaction 2P 1DN P2D with a Kdof 0.72 mM2. The buffer for all experiments was 50 m M Tris-HCl (pH 7.4), 100 m MNaCl, and 0.1 m MEDTA. FIG.7 .Resonance energy transfer. A,fluorescence spectra of rho- damine-labeled Phd (0.8 mM), fluorescein-labeled Doc (0.02 mM), and a mixture of both proteins; B,complex formation assayed by resonance energy transfer.\", 'page_label: 16817 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  Other conditions identical to panel A. C,increasing quantities of Doc were added to 2 mMPhd protein and the change in CD ellipticity at 222 nm was monitored. The solid line is calculated for the reaction 2P 1DN P2D with a Kdof 0.72 mM2. The buffer for all experiments was 50 m M Tris-HCl (pH 7.4), 100 m MNaCl, and 0.1 m MEDTA. FIG.7 .Resonance energy transfer. A,fluorescence spectra of rho- damine-labeled Phd (0.8 mM), fluorescein-labeled Doc (0.02 mM), and a mixture of both proteins; B,complex formation assayed by resonance energy transfer. The solid line is calculated for the reaction 2P 1DN P2D with a Kdof 0.81 mM2.C,kinetics of dissociation of complexes containing rhodamine-labeled Phd and fluorescein-labeled Doc moni- tored by fluorescence after addition of a 250-fold excess of unlabeledPhd. The solid line is a single exponential fit with a rate constant of 42 s 21. Experiments in all panels were performed at 25 °C in a buffer containing 50 m MTris-HCl (pH 7.4), 100 m MNaCl, and 1 m MEDTA.Doc and Phd Form a Heterotrimeric Complex 16817', 'page_label: 16818 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  the cell. If the assumptions that underlie these rough calcula- tions are sound, then free Doc at submicromolar levels must berelatively innocuous to cells. Otherwise, P1 lysogens wouldhave a significant growth disadvantage relative to nonlyso-gens. However, higher levels of free Doc (presumably in themicromolar range) must be lethal for the plasmid addictionmechanism to function. These considerations make it ex-tremely unlikely that the mechanism of Doc toxicity involvessingle hit enzymatic lethality and suggest that Doc may revers-ibly inhibit some required cellular activity in a cooperativefashion that results in a steep concentration dependence oftoxicity. In P1 lysogens, Phd binding spares the cell from Doc-medi- ated killing (3). However, Doc binding induces structure indenatured Phd at physiological temperatures, and thus itseems likely that Doc binding would also help protect Phd fromdegradation. Degradation of Phd is mediated directly or indi-rectly by the ClpXP protease (3), which consists of ClpP prote-ase subunits and regulatory ClpX ATPase subunits (17–19).ClpX is responsible for substrate recognition by ClpXP and hasbeen shown to recognize specific peptide sequences at the Ctermini of several substrate proteins (20–22). If ClpXP directlydegrades Phd, then C-terminal substrate recognition is proba-bly also involved. By this model, formation of the P 2D complex with Doc could prevent degradation of Phd by sequesteringC-terminal sequences required for ClpX recognition and per-haps by stabilizing at least some parts of Phd in a nativestructure. We note, however, that the rapid dissociation rateand relatively high equilibrium dissociation constant for P 2D complexes makes it likely that a pool of free Phd, in a largelydenatured form, is present even when Doc is present in P1lysogens. If Phd were too good a substrate for ClpXP, thendegradation of the free pool and mass action to produce morefree Phd could lead to Doc toxicity even in lysogens. Hence, itseems likely that Phd should be a rather poor substrate forClpXP. Indeed, in ClpXP 1strains, Phd is degraded relatively slowly over the course of several cell generations (3) Acknowledgments—We thank Alok Srivastava for help and advice, Roy Magnuson and Michael Yarmolinsky for pRDM032, and Carl Pabo, Harvey Lodish, and Tania Baker for use of equipment. REFERENCES 1. Lehnherr, H., Maguin, E., Jafri, S. & Yarmolinsky, M. B. (1993) J. Mol. Biol. 233, 414–428 2. Rosner, J. L. (1972) Virology 49,679–680 3. Lehnherr, H. & Yarmolinsky, M. B. (1995) Proc. Natl. Acad. Sci. U. S. A. 92, 3274–3277 4. Magnuson, R., Lehnherr, H., Mukhopadhyay, G. & Yarmolinsky, M. B. (1996) J. Biol. Chem. 271, 18705–18710 5. Magnuson, R. & Yarmolinsky, M. B. (1998) J. Bacteriol. 180, 6342–6351 6. Gazit, E. & Sauer, R. T. (1999) J. Biol. Chem. 274, 2652–2657 7. Loh, S. M., Cram, D. S. & Skurray, R. A. (1988) Gene 66,259–268 8. Young, C., Prince, A. S. & Figurski, D. H. (1985) Proc. Natl. Acad. Sci. U. S. A. 82,7374–7378 9. Gerdes, K., Rasmussen, P. B. & Molin, S. (1986) Proc. Natl. Acad. Sci. U. S. A.', 'page_label: 16818 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0021925819728452-main.pdf  180, 6342–6351 6. Gazit, E. & Sauer, R. T. (1999) J. Biol. Chem. 274, 2652–2657 7. Loh, S. M., Cram, D. S. & Skurray, R. A. (1988) Gene 66,259–268 8. Young, C., Prince, A. S. & Figurski, D. H. (1985) Proc. Natl. Acad. Sci. U. S. A. 82,7374–7378 9. Gerdes, K., Rasmussen, P. B. & Molin, S. (1986) Proc. Natl. Acad. Sci. U. S. A. 83,3116–3120 10. Jensen, R. B. & Gerdes, K. (1995) Mol. Microbiol. 17,205–210 11. Aizenman, E., Engelberg-Kulka, H. & Glaser, G. (1996) Proc. Natl. Acad. Sci U. S. A. 93,6059–6063 12. Holc ˇı´k, M. & Iyer, V. N. (1997) Microbiology 143, 3403–3416 13. Engelberg-Kulka, H., Reches, M., Narasimhan, S., Schoulaker-Schwarz, R., Klemes, Y., Aizenman, E. & Glaser, G. (1998) Proc. Natl. Acad. Sci. U. S. A. 95,15481–15486 14. Van Melderen, L., Thi, M. H. D., Lecchi, P., Gottesman, S., Couturier, M. & Maurizi, M. R. (1996) J. Biol. Chem. 271, 27730–27738 15. Gazit, E., Burshtein, N., Ellar, D. J., Saywer, T. & Shai, Y. (1997) Biochemistry 36,15546–15554 16. de Vos, A. M., Ultsch, M. & Kossiakoff, A. A. (1992) Science 255, 306–312 17. Wawrzynow, A., Wojtkowiak, D., Marszalek, J., Banecki, B., Jonsen, M., Graves, B., Georgopoulos, C. & Zylicz, M. (1995) EMBO J. 14,1867–1877 18. Wang, J., Hartling, J. A. & Flanagan, J. M. (1997) Cell 91,447–456 19. Grimaud, R., Kessel, M., Beuron, F., Steven, A. C. & Maurizi, M. R. (1998) J. Biol. Chem. 273, 12476–12481 20. Levchenko, I., Smith, C. K., Walsh, N. P., Sauer, R. T. & Baker, T. A. (1997) Cell 91,939–947 21. Levchenko, I., Yamauchi, M. & Baker, T. A. (1997) Genes Dev. 11,1561–1572 22. Gottesman, S., Roche, E., Zhou, Y. & Sauer, R. T. (1998) Genes Dev. 12, 1338–1347Doc and Phd Form a Heterotrimeric Complex 16818', 'page_label: 167829 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The Role of O-antigen in P1 Transduction of Shigella ﬂexneri and Escherichia coli with its Alternative S’ Tail Fibre Yang W. Huan2, Jidapha Fa-arun2and Baojun Wang1,3⇑ 1-College of Chemical and Biological Engineering & ZJU-Hangzhou Global Scientiﬁc and Technological Innovation Center, Zhejiang University, Hangzhou 310058, China 2-Centre for Synthetic and Systems Biology, School of Biological Sciences, University of Edinburgh, Edinburgh EH9 3FF, United Kingdom 3-Research Centre of Biological Computation, Zhejiang Laboratory, Hangzhou 311100, China Correspondence to Baojun Wang: College of Chemical and Biological Engineering, Zhejiang University, Hangzhou 310058, China. baojun.wang@zju.edu.cn (B. Wang) @SynthSysEd ,@SBSatEd (B. Wang) https://doi.org/10.1016/j.jmb.2022.167829Edited by Tao Liu Abstract Enterobacteria phage P1 expresses two types of tail ﬁbre, S and S’. Despite the wide usage of phage P1 for transduction, the host range and the receptor for its alternative S’ tail ﬁbre was never determined. Here, aDS-cinDpac E. coli P1 lysogenic strain was generated to allow packaging of phagemid DNA into P1 phage having either S or S’ tail ﬁbre. P1(S’) could transduce phagemid DNA into Shigella ﬂexneri 2a 2457O, Shigella ﬂexneri 5a M90T and Escherichia coli O3 efﬁciently. Mutational analysis of the O- antigen assembly genes and LPS inhibition assays indicated that P1(S’) transduction requires at leastone O-antigen unit. E. coli O111:B4 LPS produced a high neutralising effect against P1(S’) transduction, indicating that this E. coli strain could be susceptible to P1(S’)-mediated transduction. Mutations in the O- antigen modiﬁcation genes of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T did not cause signiﬁcant changes to P1(S’) transduction efﬁciency. A higher transduction efﬁciency of P1(S’) improved the delivery of a cas9 antimicrobial phagemid into both S. ﬂexneri 2457O and M90T. These ﬁndings provide novel insights into P1 tropism-switching, by identifying the bacterial strains which are susceptible to P1(S’)-mediated transduction, as well as demonstrating its potential for delivering a DNA sequence-speciﬁcCas9 antimicrobial into clinically relevant S. ﬂexneri . /C2112022 The Author(s). Published by Elsevier Ltd. Introduction Enterobacteria phage P1 has a wide host range of Gram-negative Enterobacteriaceae , including multiple strains of Escherichia coli ,3–4,21,20,47,63Shi- gella ﬂexneri ,63S.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:49 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'74'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992738'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'435ms'), (b'x-request-id', b'746fcdc2c7d2d7aac7c1d416b0233720'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124953cacc5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:49 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'74'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992738'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'435ms'), (b'x-request-id', b'746fcdc2c7d2d7aac7c1d416b0233720'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124953cacc5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 167829 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  A higher transduction efﬁciency of P1(S’) improved the delivery of a cas9 antimicrobial phagemid into both S. ﬂexneri 2457O and M90T. These ﬁndings provide novel insights into P1 tropism-switching, by identifying the bacterial strains which are susceptible to P1(S’)-mediated transduction, as well as demonstrating its potential for delivering a DNA sequence-speciﬁcCas9 antimicrobial into clinically relevant S. ﬂexneri . /C2112022 The Author(s). Published by Elsevier Ltd. Introduction Enterobacteria phage P1 has a wide host range of Gram-negative Enterobacteriaceae , including multiple strains of Escherichia coli ,3–4,21,20,47,63Shi- gella ﬂexneri ,63S. dysenteriae ,3–4,21,63Klebsiella pneumoniae ,20,47,63Citrobacter freundii20,47,63and Pseudomonas aeruginosa .20,47,63Genes that deter- mine P1 phage adsorption and host speciﬁcity areassociated with the cinoperon, as well as the tail ﬁbre operon encoding for three genes, R,S,U under the regulation of late promoter LP Sit.23–24,43S is a structural protein that extends along the tail ﬁbre with a constant N-terminus (Sc) attached to the baseplate while its variable C-terminus (Sv)binds to a speciﬁc receptor on the host bacterium allowing for phage adsorption.65Uwas predicted to encode an essential chaperone protein for theassembly of a distal part of the tail ﬁbre.43The cod- ing sequences of SvandUare arranged in an oppo- site orientation to the alternative variants, namelySv’andU’, which together form the invertible C seg- ment ﬂanked by inverted repeats, cixL and cixR.31,26,30Inversion of the C segment at the cix Research Article 0022-2836/ /C2112022 The Author(s). Published by Elsevier Ltd. Journal of Molecular Biology 434 (2022) 167829', 'page_label: 167830 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 2', 'page_label: 167831 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  sites is mediated by the Cin recombinase, which allows for expression of the alternative S’ tail ﬁbre and its assembly protein U’, changing the host range speciﬁcity of P1.31Inversion of the C seg- ment, however, is thought to be infrequent hence most of the P1 phage progeny generated lytically (i.e., phage progeny from a single plaque or fromcontinuous lysis of cell culture) would have the same tail ﬁbre as the infecting phage whereas the induction of lysogenic host cell would generate amixture of P1 phage with either the S or S’ tailﬁbre. 31,26,30 The host range of P1 with its S tail (termed P1(S) hereafter) include E. coli K12,31,26,30E. coli C31,26,30 andS. dysenteriae strain “Sh”.30Previous studies of E. coli andSalmonella enterica serovar Typhimur- ium lipopolysaccharide (LPS)-defective mutantssuggested that glucose molecule(s) on the outer core of LPS might serve as the receptor for P1 S tail ﬁbre.16,49,56P1 with its S’ tail ﬁbre (termed P1(S’) in this study), however, failed to transduce into either E. coli K12 or E. coli C and its bacterial host range remained unknown.31,30On the contrary, a mutant strain of E. coli strain C that supports transduction and plaque formation with both P1(S) and P1(S’) was identiﬁed in a previous study, although the nat- ure of the mutation(s) was not known.30 Analysis of P1 S,Uand the alternative S’,U’ genes sequences suggested extensive homologies of these proteins to the genes’products of bacteriophage Mu invertible G segment.43Consistently, the P1 Sand Ugenes were shown to complement and rescue transduc-tion of Mu SandUmutants at varying efﬁciencies.61 The genes of Mu G segment, however, failed to complement amber mutations within the P1 C seg-ment, suggesting functional differences between Mu and P1 tail ﬁbres and chaperones despite high DNA sequence homology. This was consistent with discrepancies in the host range of Mu(S) and P1(S),as Mu(S) can only infect E. coli K12 but not E. coli C while P1(S) can infect both strains of E. coli .56 In this study, we have created a mutant strain of anE. coli P1 lysogen in which the entire C segment and the cin recombinase coding sequence has been deleted. The mutations werecomplemented in trans by providing the Sand U or the S’and U’genes from a plasmid, giving phage progeny with homogenous S or S’ tail ﬁbres. Our data demonstrated that P1(S’) phagecould transduce phagemid DNA into S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 at a high efﬁciency. Transduction of phagemid DNA byP1(S’) was inhibited on DwaaL andDrfemutant strains of these bacteria lacking O-antigen. Furthermore, mutant strains of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with the O-antigen polymerase gene wzy/rfcdeleted were susceptible to P1(S’) transduction, indicatingthat P1(S’) transduction requires at least one O-antigen unit. Consistently, LPS extracted from wildtype S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T, E. coli O3 and E. coli O111:B4 can inhibit P1(S’) transduction, which demonstrated that P1 (S’) could adsorb to LPS with these O-antigens.', 'page_label: 167831 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Furthermore, mutant strains of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with the O-antigen polymerase gene wzy/rfcdeleted were susceptible to P1(S’) transduction, indicatingthat P1(S’) transduction requires at least one O-antigen unit. Consistently, LPS extracted from wildtype S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T, E. coli O3 and E. coli O111:B4 can inhibit P1(S’) transduction, which demonstrated that P1 (S’) could adsorb to LPS with these O-antigens. Taken together, this study ﬁlls the knowledge gapof P1 transduction with its alternative S’ tail ﬁbre, with evidence supporting the potential use of P1 (S’) to improve the transduction efﬁciency of aDNA sequence-speciﬁc Cas9 antimicrobial systeminS. ﬂexneri , which is a clinically relevant human pathogen. Figure 1. S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 supported P1(S’) transduction of phagemid DNA. (a) The organisation of P1 tail ﬁbre operon. Black triangles mark the start site for the coding sequences. The expression of R(hypothetical tail assembly protein), S(tail ﬁbre protein) and U(hypothetical chaperone protein) are controlled by the P1 late promoter, LP Sit.The tail ﬁbre is divided into a constant Sc region as well as a variable Sv region. The coding sequence for the alternative variable region, Sv’, and its chaperone, U’are located adjacent to SvandU, in an opposite orientation. Cin recombinase mediates inversion of the C segment at the cixsites (in blue), allowing expression of the S’tail ﬁbre and U’chaperone. An alignment of Sv and Sv’ amino acid sequences was included in Supplementary Figure 3 .(b) Schematic diagram showing the packaging of a phagemid into P1 phage particles using the DS-cin E. coli EMG16 P1 lysogen. The tail ﬁbre operon and the cinrecombinase coding sequence were deleted. The DS-cin mutation was complemented in trans , by providing either a combination of SandU(plasmid termed pP1SU) or S’andU’(plasmid termed pP1S’U’) genes, controlled by the P1 late promoter LPSit. The lysates prepared is expected to contain a homogenous population of P1 phage with either S (P1(S)) or S’ (P1(S’)) tail ﬁbre. P1(S) (with yellow tail ﬁbre terminals) was shown to transduce strains of E. coli K12, E. coli B,E. coli C, while the host range of P1(S’) (with blue tail ﬁbre terminals) was not determined.41,31,30,56A mutant strain of E. coli C could support transduction by both the P1(S) and P1(S’), although the nature of mutation was not identiﬁed.30(c) The number of phagemid transductant recovered, after P1(S) (yellow bars) and P1(S’) (blue bars) phagemid lysates treatment of E. coli K12 NCM3722, E. coli BL21, E. coli E24377A, S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3. The p-values were determined by Welch’s ANOVA. n.s.', 'page_label: 167831 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  coli C, while the host range of P1(S’) (with blue tail ﬁbre terminals) was not determined.41,31,30,56A mutant strain of E. coli C could support transduction by both the P1(S) and P1(S’), although the nature of mutation was not identiﬁed.30(c) The number of phagemid transductant recovered, after P1(S) (yellow bars) and P1(S’) (blue bars) phagemid lysates treatment of E. coli K12 NCM3722, E. coli BL21, E. coli E24377A, S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3. The p-values were determined by Welch’s ANOVA. n.s. represented no signiﬁcant difference(s) in p- values for comparison of the number of S. ﬂexneri 2457O phagemid transductant recovered between P1(S) and P1 (S’) infection, as well as comparison between the number of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T phagemid transductant recovered after P1(S’) infection.3Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 3', 'page_label: 167832 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Results S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 supported P1(S’) transduction ofphagemid DNA We have established a J72114 P1 phagemid system in a Dpac E. coli K12 EMG16 P1 kc lysogenic strain that allows the packaging of the phagemid DNA into infectious P1 phage particlesupon arabinose induction ( Supplementary Figures 1a, 1b, 1c ). The chloramphenicol acetyltransferase ( cat) gene of the phagemid allows quantiﬁcation of transduction events by enumerating the chloramphenicol resistant colony forming units (CmRCFU) recovered after P1 infection. The P1 phagemid contains aconstitutively expressed cas9 system with a DNA sequence-speciﬁc antimicrobial effect in the presence of a chromosomal-targeting crRNAguide. 5,9,22,11Our preliminary results suggested that P1 can transduce the cas9 phagemid into E. coli K12 NCM3722, S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T at a high efﬁciency, giving at least 109phagemid transductants per mL lysate (TU/mL) (data not shown). Since the induction ofa P1 lysogen is expected to produce a mixed popu-lation of P1 progeny with either the S or S’ tail ﬁbre, we sought to compare the transduction efﬁciency of P1(S) and P1(S’) on different strains of E. coli and S. ﬂexneri . The aim was to identify the P1 tail ﬁbre that could give a higher transduction efﬁciency, par- ticularly in targeting different strains of S. ﬂexneri . P1cas9 phagemid with a randomly generated spacer sequence (non-chromosomal targeting, cas9 -NT) was used for the preparation of lysates, as the phagemid can be stably maintained in E. coli andS. ﬂexneri , which allowed quantiﬁcation of phagemid transductants. To obtain a homogeneous population of P1 phage particles with either the S or S’ tail ﬁbre, the coding sequences for Sc, Sv’ ,U,U’,Sv’and cin recombinase were deleted via the lambda Redrecombineering system ( Figures 1 (a),(b)). The DS-cin mutation was then complemented in trans , by transforming the mutant strain with a low-copynumber plasmid containing genes encoding the S tail ﬁbre and U chaperone (plasmid termed pP1SU) or the S’ tail ﬁbre and U’ chaperone(plasmid pP1S’U’) ( Figure 1 (b)).Dpac EMG16 cells co-transformed with a cas9 -NT phagemid and the pP1SU or pP1S’U’ plasmid were used forphagemid lysate, which were then used fortreatment of E. coli K12 NCM3722, E. coli BL21, E. coli 24377A, E. coli O3, S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T. Infection of E. coli K12 NCM3722 and E. coli B BL21 with P1(S) but not P1(S’) yielded phagemid transductants ( Figure 1 (c)). On the contrary, infection of E. coli O3 with P1(S’) but not P1(S) yielded phagemid transductants. The lack of phagemid transductant was not due to cell death after P1 infection, as wecould recover approximately 5 /C2108to 1/C2109 E. coli BL21, E. coli NCM3722 and E. coli O3 CFU (per mL reaction mix) on non-selective LB (data not shown).', 'page_label: 167832 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Infection of E. coli K12 NCM3722 and E. coli B BL21 with P1(S) but not P1(S’) yielded phagemid transductants ( Figure 1 (c)). On the contrary, infection of E. coli O3 with P1(S’) but not P1(S) yielded phagemid transductants. The lack of phagemid transductant was not due to cell death after P1 infection, as wecould recover approximately 5 /C2108to 1/C2109 E. coli BL21, E. coli NCM3722 and E. coli O3 CFU (per mL reaction mix) on non-selective LB (data not shown). P1(S’) infection of S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O yielded a 207-fold (s=3 1 , n= 12, p < 0.005) and a 1.61-fold (s= 0.24, n= 12, p > 0.05) increase in the number of phagemid transductant respectively, when compared to P1(S) infection of these bacteria. P1 (S’) and P1(S) infection of E. coli E24377A yielded a similar number of phagemid transductant, whichwas approximately 1 /C2106transductants per mL lysate (TU/mL) ( s=2 . 6 /C2105TU/mL, n=2 4 , p > 0.05). To verify that differences in the numberof P1(S) and P1(S’) phagemid transductant were not due to variations in the titre of phagemid transducing particles, all phagemid lysates weretreated with DNase I and quantiﬁed for P1- packaged phagemid DNA via quantitative PCR (qPCR). qCPR quantiﬁcation of the cas9 DNA sequence in DNase I-treated lysates indicated a similar concentration of phagemid DNA that were packaged into phage particles ( Supplementary Figures 2a, b ). Since all lysates contained a similar concentration of P1-packaged phagemid DNA, differences in the number of S. ﬂexneri and E. coli phagemid transductants were indicative of variations in the transduction efﬁciency of phagemid DNA. Taken together, P1(S’) transduced phagemid DNA into S. ﬂexneri 5a M90T, S. ﬂexneri 2a 2457O and E. coli O3 at a high efﬁciency thus identiﬁed several bacterial strains susceptible toP1(S’)-mediated transduction, which were notdetermined in previous studies. P1(S’) transduction of phagemid DNA required O-antigen Previous data indicated that only E. coli and S. ﬂexneri strains with O-antigen supported P1(S’) transduction ( Figure 1 (c)). To determine if the O- antigen of S. ﬂexneri andE. coli O3 were required for P1(S) and P1(S’) transduction, the relativetransduction efﬁciency of both phages was assessed on wildtype (WT) and O-antigen deﬁcient mutant strains of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3. The genes selected for generating the O-antigen deﬁcient mutant strains were waaL ,rfe and rfc/wzy (Figure 2 (a)).waaL and rfegenes encode the O- antigen ligase and the WecA integral membrane protein, which are required for the ligation of O-antigen onto the LPS core 64and the biosynthesis ofE. coli as well as S. ﬂexneri O-antigens containing aN-acetylglucosamine residue,1respectively.', 'page_label: 167832 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The genes selected for generating the O-antigen deﬁcient mutant strains were waaL ,rfe and rfc/wzy (Figure 2 (a)).waaL and rfegenes encode the O- antigen ligase and the WecA integral membrane protein, which are required for the ligation of O-antigen onto the LPS core 64and the biosynthesis ofE. coli as well as S. ﬂexneri O-antigens containing aN-acetylglucosamine residue,1respectively. DwaaL andDrfestrains of S. ﬂexneri64,55,62and E. coli64,1were shown to give a rough type LPS without any O-antigen. On the contrary, the gene product of rfc/wzy is required for the polymerisation of O-antigen polysaccharide and a loss-of-function mutation in the rfc/wzy gene of S. ﬂexneri48,38asY.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 4', 'page_label: 167833 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  well as E. coli O351were shown to give a semi- rough type LPS that has only one O-antigen unit. DwaaL ,DrfeandDrfc/wzy strains of S. ﬂexneri 2a2457O, S. ﬂexneri 5a M90T and E. coli O3 were generated via lamda Red recombineering, and the effects of these mutations on the bacterial LPS were Figure 2. P1(S’) transduction of S. ﬂexneri 5a M90T, S. ﬂexneri 2a 2457O and E. coli O3 required O-antigen. (a) Schematic diagram showing the expected LPS structure of wildtype (WT), Drfe,DwaaL andDrfcstrains of S. ﬂexneri 5a M90T, S. ﬂexneri 2a 2457O and E. coli O3. The LPS of DrfeandDwaaL mutants contains Lipid A, inner and outer core oligosaccharide structures without any O-antigen unit(s), while that of Drfcmutant has only 1O-antigen unit compared to the multiple O-antigen repeating units on the LPS of WT cells. Analysis of LPS extracted fromwildtype (WT), DwaaL ,DrfeandDrfcstrains of ( b)S. ﬂexneri 5a M90T ( S. ﬂexneri M90T), S. ﬂexneri 2a 2457O ( S. ﬂexneri 2457O) and ( c)E. coli O3. LPS samples were analysed on 12% precast polyacrylamide gels and silver stained. ( d) The number of phagemid transductant recovered, after P1(S) (yellow bars) and P1(S’) (blue bars) phagemid lysates treatment of wildtype (WT), DwaaL ,DrfeandDrfcstrains of S. ﬂexneri 5a M90T, S. ﬂexneri 2a 2457O and E. coli O3.Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 5', 'page_label: 167834 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 6'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 167829 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  A higher transduction efﬁciency of P1(S’) improved the delivery of a cas9 antimicrobial phagemid into both S. ﬂexneri 2457O and M90T. These ﬁndings provide novel insights into P1 tropism-switching, by identifying the bacterial strains which are susceptible to P1(S’)-mediated transduction, as well as demonstrating its potential for delivering a DNA sequence-speciﬁcCas9 antimicrobial into clinically relevant S. ﬂexneri . /C2112022 The Author(s). Published by Elsevier Ltd. Introduction Enterobacteria phage P1 has a wide host range of Gram-negative Enterobacteriaceae , including multiple strains of Escherichia coli ,3–4,21,20,47,63Shi- gella ﬂexneri ,63S. dysenteriae ,3–4,21,63Klebsiella pneumoniae ,20,47,63Citrobacter freundii20,47,63and Pseudomonas aeruginosa .20,47,63Genes that deter- mine P1 phage adsorption and host speciﬁcity areassociated with the cinoperon, as well as the tail ﬁbre operon encoding for three genes, R,S,U under the regulation of late promoter LP Sit.23–24,43S is a structural protein that extends along the tail ﬁbre with a constant N-terminus (Sc) attached to the baseplate while its variable C-terminus (Sv)binds to a speciﬁc receptor on the host bacterium allowing for phage adsorption.65Uwas predicted to encode an essential chaperone protein for theassembly of a distal part of the tail ﬁbre.43The cod- ing sequences of SvandUare arranged in an oppo- site orientation to the alternative variants, namelySv’andU’, which together form the invertible C seg- ment ﬂanked by inverted repeats, cixL and cixR.31,26,30Inversion of the C segment at the cix Research Article 0022-2836/ /C2112022 The Author(s). Published by Elsevier Ltd. Journal of Molecular Biology 434 (2022) 167829', 'page_label: 167830 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 2', 'page_label: 167831 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  sites is mediated by the Cin recombinase, which allows for expression of the alternative S’ tail ﬁbre and its assembly protein U’, changing the host range speciﬁcity of P1.31Inversion of the C seg- ment, however, is thought to be infrequent hence most of the P1 phage progeny generated lytically (i.e., phage progeny from a single plaque or fromcontinuous lysis of cell culture) would have the same tail ﬁbre as the infecting phage whereas the induction of lysogenic host cell would generate amixture of P1 phage with either the S or S’ tailﬁbre. 31,26,30 The host range of P1 with its S tail (termed P1(S) hereafter) include E. coli K12,31,26,30E. coli C31,26,30 andS. dysenteriae strain “Sh”.30Previous studies of E. coli andSalmonella enterica serovar Typhimur- ium lipopolysaccharide (LPS)-defective mutantssuggested that glucose molecule(s) on the outer core of LPS might serve as the receptor for P1 S tail ﬁbre.16,49,56P1 with its S’ tail ﬁbre (termed P1(S’) in this study), however, failed to transduce into either E. coli K12 or E. coli C and its bacterial host range remained unknown.31,30On the contrary, a mutant strain of E. coli strain C that supports transduction and plaque formation with both P1(S) and P1(S’) was identiﬁed in a previous study, although the nat- ure of the mutation(s) was not known.30 Analysis of P1 S,Uand the alternative S’,U’ genes sequences suggested extensive homologies of these proteins to the genes’products of bacteriophage Mu invertible G segment.43Consistently, the P1 Sand Ugenes were shown to complement and rescue transduc-tion of Mu SandUmutants at varying efﬁciencies.61 The genes of Mu G segment, however, failed to complement amber mutations within the P1 C seg-ment, suggesting functional differences between Mu and P1 tail ﬁbres and chaperones despite high DNA sequence homology. This was consistent with discrepancies in the host range of Mu(S) and P1(S),as Mu(S) can only infect E. coli K12 but not E. coli C while P1(S) can infect both strains of E. coli .56 In this study, we have created a mutant strain of anE. coli P1 lysogen in which the entire C segment and the cin recombinase coding sequence has been deleted. The mutations werecomplemented in trans by providing the Sand U or the S’and U’genes from a plasmid, giving phage progeny with homogenous S or S’ tail ﬁbres. Our data demonstrated that P1(S’) phagecould transduce phagemid DNA into S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 at a high efﬁciency. Transduction of phagemid DNA byP1(S’) was inhibited on DwaaL andDrfemutant strains of these bacteria lacking O-antigen. Furthermore, mutant strains of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with the O-antigen polymerase gene wzy/rfcdeleted were susceptible to P1(S’) transduction, indicatingthat P1(S’) transduction requires at least one O-antigen unit. Consistently, LPS extracted from wildtype S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T, E. coli O3 and E. coli O111:B4 can inhibit P1(S’) transduction, which demonstrated that P1 (S’) could adsorb to LPS with these O-antigens.', 'page_label: 167831 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Furthermore, mutant strains of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with the O-antigen polymerase gene wzy/rfcdeleted were susceptible to P1(S’) transduction, indicatingthat P1(S’) transduction requires at least one O-antigen unit. Consistently, LPS extracted from wildtype S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T, E. coli O3 and E. coli O111:B4 can inhibit P1(S’) transduction, which demonstrated that P1 (S’) could adsorb to LPS with these O-antigens. Taken together, this study ﬁlls the knowledge gapof P1 transduction with its alternative S’ tail ﬁbre, with evidence supporting the potential use of P1 (S’) to improve the transduction efﬁciency of aDNA sequence-speciﬁc Cas9 antimicrobial systeminS. ﬂexneri , which is a clinically relevant human pathogen. Figure 1. S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 supported P1(S’) transduction of phagemid DNA. (a) The organisation of P1 tail ﬁbre operon. Black triangles mark the start site for the coding sequences. The expression of R(hypothetical tail assembly protein), S(tail ﬁbre protein) and U(hypothetical chaperone protein) are controlled by the P1 late promoter, LP Sit.The tail ﬁbre is divided into a constant Sc region as well as a variable Sv region. The coding sequence for the alternative variable region, Sv’, and its chaperone, U’are located adjacent to SvandU, in an opposite orientation. Cin recombinase mediates inversion of the C segment at the cixsites (in blue), allowing expression of the S’tail ﬁbre and U’chaperone. An alignment of Sv and Sv’ amino acid sequences was included in Supplementary Figure 3 .(b) Schematic diagram showing the packaging of a phagemid into P1 phage particles using the DS-cin E. coli EMG16 P1 lysogen. The tail ﬁbre operon and the cinrecombinase coding sequence were deleted. The DS-cin mutation was complemented in trans , by providing either a combination of SandU(plasmid termed pP1SU) or S’andU’(plasmid termed pP1S’U’) genes, controlled by the P1 late promoter LPSit. The lysates prepared is expected to contain a homogenous population of P1 phage with either S (P1(S)) or S’ (P1(S’)) tail ﬁbre. P1(S) (with yellow tail ﬁbre terminals) was shown to transduce strains of E. coli K12, E. coli B,E. coli C, while the host range of P1(S’) (with blue tail ﬁbre terminals) was not determined.41,31,30,56A mutant strain of E. coli C could support transduction by both the P1(S) and P1(S’), although the nature of mutation was not identiﬁed.30(c) The number of phagemid transductant recovered, after P1(S) (yellow bars) and P1(S’) (blue bars) phagemid lysates treatment of E. coli K12 NCM3722, E. coli BL21, E. coli E24377A, S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3. The p-values were determined by Welch’s ANOVA. n.s.', 'page_label: 167831 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  coli C, while the host range of P1(S’) (with blue tail ﬁbre terminals) was not determined.41,31,30,56A mutant strain of E. coli C could support transduction by both the P1(S) and P1(S’), although the nature of mutation was not identiﬁed.30(c) The number of phagemid transductant recovered, after P1(S) (yellow bars) and P1(S’) (blue bars) phagemid lysates treatment of E. coli K12 NCM3722, E. coli BL21, E. coli E24377A, S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3. The p-values were determined by Welch’s ANOVA. n.s. represented no signiﬁcant difference(s) in p- values for comparison of the number of S. ﬂexneri 2457O phagemid transductant recovered between P1(S) and P1 (S’) infection, as well as comparison between the number of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T phagemid transductant recovered after P1(S’) infection.3Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 3', 'page_label: 167832 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Results S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 supported P1(S’) transduction ofphagemid DNA We have established a J72114 P1 phagemid system in a Dpac E. coli K12 EMG16 P1 kc lysogenic strain that allows the packaging of the phagemid DNA into infectious P1 phage particlesupon arabinose induction ( Supplementary Figures 1a, 1b, 1c ). The chloramphenicol acetyltransferase ( cat) gene of the phagemid allows quantiﬁcation of transduction events by enumerating the chloramphenicol resistant colony forming units (CmRCFU) recovered after P1 infection. The P1 phagemid contains aconstitutively expressed cas9 system with a DNA sequence-speciﬁc antimicrobial effect in the presence of a chromosomal-targeting crRNAguide. 5,9,22,11Our preliminary results suggested that P1 can transduce the cas9 phagemid into E. coli K12 NCM3722, S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T at a high efﬁciency, giving at least 109phagemid transductants per mL lysate (TU/mL) (data not shown). Since the induction ofa P1 lysogen is expected to produce a mixed popu-lation of P1 progeny with either the S or S’ tail ﬁbre, we sought to compare the transduction efﬁciency of P1(S) and P1(S’) on different strains of E. coli and S. ﬂexneri . The aim was to identify the P1 tail ﬁbre that could give a higher transduction efﬁciency, par- ticularly in targeting different strains of S. ﬂexneri . P1cas9 phagemid with a randomly generated spacer sequence (non-chromosomal targeting, cas9 -NT) was used for the preparation of lysates, as the phagemid can be stably maintained in E. coli andS. ﬂexneri , which allowed quantiﬁcation of phagemid transductants. To obtain a homogeneous population of P1 phage particles with either the S or S’ tail ﬁbre, the coding sequences for Sc, Sv’ ,U,U’,Sv’and cin recombinase were deleted via the lambda Redrecombineering system ( Figures 1 (a),(b)). The DS-cin mutation was then complemented in trans , by transforming the mutant strain with a low-copynumber plasmid containing genes encoding the S tail ﬁbre and U chaperone (plasmid termed pP1SU) or the S’ tail ﬁbre and U’ chaperone(plasmid pP1S’U’) ( Figure 1 (b)).Dpac EMG16 cells co-transformed with a cas9 -NT phagemid and the pP1SU or pP1S’U’ plasmid were used forphagemid lysate, which were then used fortreatment of E. coli K12 NCM3722, E. coli BL21, E. coli 24377A, E. coli O3, S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T. Infection of E. coli K12 NCM3722 and E. coli B BL21 with P1(S) but not P1(S’) yielded phagemid transductants ( Figure 1 (c)). On the contrary, infection of E. coli O3 with P1(S’) but not P1(S) yielded phagemid transductants. The lack of phagemid transductant was not due to cell death after P1 infection, as wecould recover approximately 5 /C2108to 1/C2109 E. coli BL21, E. coli NCM3722 and E. coli O3 CFU (per mL reaction mix) on non-selective LB (data not shown).', 'page_label: 167832 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Infection of E. coli K12 NCM3722 and E. coli B BL21 with P1(S) but not P1(S’) yielded phagemid transductants ( Figure 1 (c)). On the contrary, infection of E. coli O3 with P1(S’) but not P1(S) yielded phagemid transductants. The lack of phagemid transductant was not due to cell death after P1 infection, as wecould recover approximately 5 /C2108to 1/C2109 E. coli BL21, E. coli NCM3722 and E. coli O3 CFU (per mL reaction mix) on non-selective LB (data not shown). P1(S’) infection of S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O yielded a 207-fold (s=3 1 , n= 12, p < 0.005) and a 1.61-fold (s= 0.24, n= 12, p > 0.05) increase in the number of phagemid transductant respectively, when compared to P1(S) infection of these bacteria. P1 (S’) and P1(S) infection of E. coli E24377A yielded a similar number of phagemid transductant, whichwas approximately 1 /C2106transductants per mL lysate (TU/mL) ( s=2 . 6 /C2105TU/mL, n=2 4 , p > 0.05). To verify that differences in the numberof P1(S) and P1(S’) phagemid transductant were not due to variations in the titre of phagemid transducing particles, all phagemid lysates weretreated with DNase I and quantiﬁed for P1- packaged phagemid DNA via quantitative PCR (qPCR). qCPR quantiﬁcation of the cas9 DNA sequence in DNase I-treated lysates indicated a similar concentration of phagemid DNA that were packaged into phage particles ( Supplementary Figures 2a, b ). Since all lysates contained a similar concentration of P1-packaged phagemid DNA, differences in the number of S. ﬂexneri and E. coli phagemid transductants were indicative of variations in the transduction efﬁciency of phagemid DNA. Taken together, P1(S’) transduced phagemid DNA into S. ﬂexneri 5a M90T, S. ﬂexneri 2a 2457O and E. coli O3 at a high efﬁciency thus identiﬁed several bacterial strains susceptible toP1(S’)-mediated transduction, which were notdetermined in previous studies. P1(S’) transduction of phagemid DNA required O-antigen Previous data indicated that only E. coli and S. ﬂexneri strains with O-antigen supported P1(S’) transduction ( Figure 1 (c)). To determine if the O- antigen of S. ﬂexneri andE. coli O3 were required for P1(S) and P1(S’) transduction, the relativetransduction efﬁciency of both phages was assessed on wildtype (WT) and O-antigen deﬁcient mutant strains of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3. The genes selected for generating the O-antigen deﬁcient mutant strains were waaL ,rfe and rfc/wzy (Figure 2 (a)).waaL and rfegenes encode the O- antigen ligase and the WecA integral membrane protein, which are required for the ligation of O-antigen onto the LPS core 64and the biosynthesis ofE. coli as well as S. ﬂexneri O-antigens containing aN-acetylglucosamine residue,1respectively.', 'page_label: 167832 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The genes selected for generating the O-antigen deﬁcient mutant strains were waaL ,rfe and rfc/wzy (Figure 2 (a)).waaL and rfegenes encode the O- antigen ligase and the WecA integral membrane protein, which are required for the ligation of O-antigen onto the LPS core 64and the biosynthesis ofE. coli as well as S. ﬂexneri O-antigens containing aN-acetylglucosamine residue,1respectively. DwaaL andDrfestrains of S. ﬂexneri64,55,62and E. coli64,1were shown to give a rough type LPS without any O-antigen. On the contrary, the gene product of rfc/wzy is required for the polymerisation of O-antigen polysaccharide and a loss-of-function mutation in the rfc/wzy gene of S. ﬂexneri48,38asY.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 4', 'page_label: 167833 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  well as E. coli O351were shown to give a semi- rough type LPS that has only one O-antigen unit. DwaaL ,DrfeandDrfc/wzy strains of S. ﬂexneri 2a2457O, S. ﬂexneri 5a M90T and E. coli O3 were generated via lamda Red recombineering, and the effects of these mutations on the bacterial LPS were Figure 2. P1(S’) transduction of S. ﬂexneri 5a M90T, S. ﬂexneri 2a 2457O and E. coli O3 required O-antigen. (a) Schematic diagram showing the expected LPS structure of wildtype (WT), Drfe,DwaaL andDrfcstrains of S. ﬂexneri 5a M90T, S. ﬂexneri 2a 2457O and E. coli O3. The LPS of DrfeandDwaaL mutants contains Lipid A, inner and outer core oligosaccharide structures without any O-antigen unit(s), while that of Drfcmutant has only 1O-antigen unit compared to the multiple O-antigen repeating units on the LPS of WT cells. Analysis of LPS extracted fromwildtype (WT), DwaaL ,DrfeandDrfcstrains of ( b)S. ﬂexneri 5a M90T ( S. ﬂexneri M90T), S. ﬂexneri 2a 2457O ( S. ﬂexneri 2457O) and ( c)E. coli O3. LPS samples were analysed on 12% precast polyacrylamide gels and silver stained. ( d) The number of phagemid transductant recovered, after P1(S) (yellow bars) and P1(S’) (blue bars) phagemid lysates treatment of wildtype (WT), DwaaL ,DrfeandDrfcstrains of S. ﬂexneri 5a M90T, S. ﬂexneri 2a 2457O and E. coli O3.Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 5', 'page_label: 167834 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 6'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:50 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'79'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'995530'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'268ms'), (b'x-request-id', b'af4937599a12fa3b2b0a67b8e0b5926e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249586f9c5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:50 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'79'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'995530'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'268ms'), (b'x-request-id', b'af4937599a12fa3b2b0a67b8e0b5926e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249586f9c5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 167835 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  visualised via silver staining. LPS extracted from WT cells gave a ladder-like pattern of multiple bands on the silver-stained gel, indicating the pres- ence of O-antigen repeating units of varying lengths(Figures 2 (b),(c)). LPS extracted from the DwaaL andDrfemutants gave a single band, indicating the absence of O-antigen. The LPS of the Drfc mutants gave two bands on the silver-stained gel, in which one was a unit higher than the bands observed for DwaaL andDrfemutants therefore veriﬁed the semi-rough type LPS of Drfcmutants with a single O-antigen unit. To determine the relative transduction efﬁciency of phagemid in WT, DwaaL ,Drfeand Drfc/wzy strains of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3, the bacteria cells were treated with either P1(S) or P1(S’) phagemidlysates and the number of phagemid transductants was enumerated. No phagemid transductant was recovered after P1(S’) infectionof the DwaaL andDrfemutants ( Figure 2 (d)). The lack of phagemid transductant was not due to cell death after P1(S’) infection, as we recoveredapproximately 1 /C210 8to 5/C2108CFU (per mL reaction mix) on non-selective LB (data not shown). Complementation of the DwaaL andDrfe mutations in trans, by providing a waaL and a rfe gene respectively, restored the smooth type LPS as well as the recovery of phagemid transductants after P1(S’) infection of the DwaaL and Drfe mutants ( Supplementary Figures 4a, 4b, 4c ). On the contrary, P1(S’) infection of Drfc S. ﬂexneri 5a M90T and Drfc S. ﬂexneri 2a 2457O yielded a 216-fold ( s=4 4 , n= 12, p < 0.0005) and 58-fold (s=9 . 8 , n= 12, p < 0.0005) lower number of phagemid transductant respectively, when compared to P1(S’) infection of WT S. ﬂexneri (Figure 2 (d)). P1(S’) infection of Drfc E. coli O3 yielded a 2-fold higher ( s=0 . 1 8 , n=1 2 , p < 0.0005) number of phagemid transductantwhen compared to P1(S’) infection of E. coli O3. There were no signiﬁcant differences in the number of phagemid transductant recoveredbetween P1(S) infection of WT, DwaaL andDrfestrains of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T (p > 0.05) ( Figure 2 (d)). Although P1(S) infection of WT and Drfc S. ﬂexneri 2a 2457O yielded a similar number of phagemid transductant(p > 0.05), P1(S) infection of the Drfc S. ﬂexneri 5a M90T yielded a 321-fold ( s=1 4 8 , n=1 2 , p < 0.005) lower number of phagemidtransductant when compared to infection of WT S. ﬂexneri 5a M90T. While no phagemid transductant was recovered after P1(S) infection of WT E. coli O3, P1(S) infection of its isogenic DwaaL and Drfe mutants yielded approximately 4.3 X108(s=6 . 6 /C2 107,n=1 2 )a n d1 . 3 /C2109(s=1 . 8 /C2108,n= 12) phagemid transductants per mL lysate,respectively.', 'page_label: 167835 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  While no phagemid transductant was recovered after P1(S) infection of WT E. coli O3, P1(S) infection of its isogenic DwaaL and Drfe mutants yielded approximately 4.3 X108(s=6 . 6 /C2 107,n=1 2 )a n d1 . 3 /C2109(s=1 . 8 /C2108,n= 12) phagemid transductants per mL lysate,respectively. Drfc E. coli O3, however, was permissive for P1(S) transduction yet the number of phagemid transductant was approximately1500-fold ( s= 93, n= 12, p < 0.0005) and 5000- fold ( s= 5000, n= 297, p < 0.0005) lower, when compared to that recovered after P1(S) infectionof its isogenic DwaaL and Drfe mutants, respectively. Overall, these data indicated that P1(S’) transduction of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 required at least one O- antigen unit, while the O-antigen deﬁcient mutants were still susceptible to P1(S) transduction. P1(S’) transduction of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T was not signiﬁcantly affected by mutations in the O-antigenmodiﬁcation genes All known serotypes of S. ﬂexneri , except for serotype 6, share the same O-antigen backbone with modiﬁcations such as glucoslylation and acetylation, which give rise to variation inserotypes. 37,15,50Despite the variations in O- antigen modiﬁcations, P1(S’) could transduce pha- gemid DNA into S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T at a similar efﬁciency ( Figure 1 (c)). We sought to determine if changes to the O-antigen modiﬁcations of S. ﬂexneri would affect P1(S’) and P1(S) transduction efﬁciency. We have generated Figure 3. P1(S’) transduction of S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O was not signiﬁcantly affected by mutations in the O-antigen modiﬁcation genes. (a) Schematic diagrams showing isogenic mutant strains of S. ﬂexneri 5a M90T (5a M90T) and S. ﬂexneri 2a 2457O (2a 2457O) with the expected change(s) to their glucosylation and/or acetylation of O-antigen, based on the respective genetic deletion(s).62,60All serotype of S. ﬂexneri , except for serotype 6, share a common O-antigen backbone comprised of repeating units of three L-rhamnose residues (RhaI- RhaII-RhaIII) and a N-acetylglucosamine residue (GlcNac). The genes involved in the acetylation and glucosylation of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T O-antigens were indicated as arrows.62,60The orientation of the glycosidic linkages between the acetyl and glucose molecules and the O-antigen backbone were shown as Arabic numerals. Blue dotted lines represented genetic deletion(s). (b) The number of phagemid transductants recovered after P1(S) (yellow bars) and P1(S’) (blue bars) phagemid lysates treatment of wildtype (WT) and all 8 mutant strainsofS.', 'page_label: 167835 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The genes involved in the acetylation and glucosylation of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T O-antigens were indicated as arrows.62,60The orientation of the glycosidic linkages between the acetyl and glucose molecules and the O-antigen backbone were shown as Arabic numerals. Blue dotted lines represented genetic deletion(s). (b) The number of phagemid transductants recovered after P1(S) (yellow bars) and P1(S’) (blue bars) phagemid lysates treatment of wildtype (WT) and all 8 mutant strainsofS. ﬂexneri 5a M90T (5a M90T) and S. ﬂexneri 2a 2457O (2a 2457O). The p-values were determined by Welch’s ANOVA. p < 0.0005 was indicated as as ***, for comparison of the number of phagemid transductants recovered between P1(S) infection of wildtype and Dgtr S. ﬂexneri 5a M90T, as well as the number of phagemid transductants recovered between P1(S) infection of wildtype and DoacB S. ﬂexneri 2a 2457O.3Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 7', 'page_label: 167836 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  8 mutant strains of S. ﬂexneri 2a 2457O and S. ﬂex- neri5a M90T that were defective in combinations of O-antigen acetylation and glycosylation, by deleting the O-antigen modiﬁcation gene(s) via lamda Redrecombination ( Figure 3 (a)).62,60The effects, if any, of these mutations on the LPS structures of S. ﬂexneri were visualised by silver stain. All the LPS samples showed a ladder-like pattern on the silver-stained gel, indicating the presence of O- antigen repeating units ( Supplementary Figure 5 a). Only the LPS of Dgtrmutants lacking a glucose modiﬁcation showed shift(s) in the band size(s) when compared to the LPS of WT cells.62,60 WT and all 8 mutant strains of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T were treated with P1(S) and P1(S’) phagemid lysate and the number of phagemid transductant wasenumerated. P1(S’) phagemid lysate treatment of WT and all 8 mutant strains of S. ﬂexneri yielded a similar number of phagemid transductant(p > 0.05) ( Figure 3 (b)). Contrastingly, P1(S) infection of Dgtr S. ﬂexneri 5a M90T yielded a 235-fold ( s=4 0 , n= 12, p < 0.0005) higher number of phagemid transductant when comparedto P1(S) infection of WT S. ﬂexneri 5a M90T (Figure 3 (b)).DoacB S. ﬂexneri 2a 2457O mutant yielded a 12-fold ( s= 1.4, n= 12, p < 0.0005) lower number of phagemid transductant after P1 (S) infection, when compared to the infection of WT S . ﬂexneri 2a 2457O. Complementation of the DoacB mutation in trans increased the number of phagemid transductant recovered after P1(S) infection ( Supplementary Figure 5 b). Unexpectedly, P1(S) infection of DoacB gtr, DoacB oacD and DoacB oacD gtr strains of S. ﬂexneri 2a 2457O yielded a similar number of phagemid transductant when compared to P1(S)infection of WT S. ﬂexneri 2a 2457O (p > 0.05) (Figure 3 (b),refer to Supplementary Materials 2 for p-values). Overall, the results indicated that P1(S’) transduction of phagemid DNA was not signiﬁcantly affected by mutations in the O-antigenmodiﬁcations genes of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T, when compared to P1(S) transduction. The LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with at least one O-antigen unit could act as an inhibitor against P1(S’)transduction P1(S’) transduction of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 required the presence of at least one O-antigen unit ( Figure 2 (d)). We therefore hypothesised that the O- antigens of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 might be important for P1(S’) adsorption.', 'page_label: 167836 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with at least one O-antigen unit could act as an inhibitor against P1(S’)transduction P1(S’) transduction of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 required the presence of at least one O-antigen unit ( Figure 2 (d)). We therefore hypothesised that the O- antigens of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 might be important for P1(S’) adsorption. If this is true, an excess S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 LPS inphagemid lysates could sequester P1(S’) phage and this might reduce its transduction efﬁciency. To verify this hypothesis, LPS extracted from WT and O-antigen deﬁcient mutant strains of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 were assessed for their inhibitory effect against P1(S) and P1(S’)-mediated transduction ofE. coli NCM3722 and S. ﬂexneri 2457O, respectively ( Figure 4 (a)). LPS extracted from WT S. ﬂexneri 5a M90T and WTS. ﬂexneri 2a 2457O completely blocked P1 (S’) transduction when applied at concentrations greater or equal to 5 lg/mL (p < 0.0005) and 0.5lg/mL (p < 0.0005), respectively ( Figures 4 (b), (c)). The LPS of Drfc S. ﬂexneri 5a M90T completely blocked P1(S’) transduction at concentrations greater or equal to 50 lg/mL (p < 0.0005), while 5000 lg/mL of Drfc S. ﬂexneri 2a 2457O LPS was required to reduce the transduction efﬁciency of P1(S’) by approximately64 % ( s=7 . 3 , n=6 ,p<0 . 0 0 0 5 )( Figure 4 (e), (f)). 500 lg/mL of WT and Dwzy E. coli O3 LPS reduced P1(S’) transduction efﬁciency byapproximately 94 % (WT E. coli O3,s=2 . 6 , n=6 , p < 0.0005) and 96 % ( Dwzy E. coli O3,s=1 . 8 , n= 6, p < 0.0005), respectively ( Figures 4 (d), (g)). Overall, the inhibitory effect of LPS with at leastone O-antigen unit indicated that P1(S’) might adsorb to the LPS of S. ﬂexneri 5a M90T, S. ﬂexneri 2a 2457O and E. coli O3 with at least one O-antigen unit. Contrastingly, treatment of P1(S’) lysate with the LPS of DwaaL mutants yielded an average transduction efﬁciency of 107 % at allconcentrations ( DwaaL S. ﬂexneri M90T, s=3 . 4 , n=5 ,p>0 . 0 5 ; DwaaL S. ﬂexneri 2457O, s= 4.9, n= 30, p > 0.05; DwaaL E. coli O3, s=5 . 7 , n= 30, p > 0.05) ( Figures 4 (h), (i), (j)).', 'page_label: 167836 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Contrastingly, treatment of P1(S’) lysate with the LPS of DwaaL mutants yielded an average transduction efﬁciency of 107 % at allconcentrations ( DwaaL S. ﬂexneri M90T, s=3 . 4 , n=5 ,p>0 . 0 5 ; DwaaL S. ﬂexneri 2457O, s= 4.9, n= 30, p > 0.05; DwaaL E. coli O3, s=5 . 7 , n= 30, p > 0.05) ( Figures 4 (h), (i), (j)). Similarly, treatment of P1(S’) lysate with E. coli K12 LPS yielded an average transduction efﬁciency of 105 % ( s=6 . 1 , n= 30, p > 0.05) at all concentrations ( Supplementary Figure 6 ). Therefore, the lack of inhibitory effect against P1 (S’) transduction indicated that P1(S’) might notadsorb to the LPS of E. coli K12 as well as the LPS of DwaaL mutant strains without O-antigen, which was consistent with the absence ofphagemid transductant after P1(S’) infection ofthese bacteria ( Figures 1 (c),2(d)). 5lg/mL of E. coli K12 LPS reduced P1(S) transduction efﬁciency by approximately 76 %(s=3 . 3 , n= 6, p < 0.0005), while concentrations greater or equal to 50 lg/mL completely blocked P1(S) transduction Supplementary Figure 6 ). 500lg/mL to 5000 lg/mL of LPS extracted from all strains of S. ﬂexneri reduced the transduction efﬁciency of P1(S) by at least 50 % ( Figures 4 (b), (c), (e), (f), (h), (i)). The inhibitory effect of S. ﬂexneri and E. coli K12 LPS against P1(S) transduction indicated phage adsorption to these LPS, which was consistent with the recovery ofY.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 8', 'page_label: 167837 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Figure 4. The LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with at least one O-antigen unit could act as an inhibitor against P1(S’) transduction. (a) Schematic diagram showing a LPS inhibition assay. LPS that allows the adsorption of P1(S) and/or P1(S’) could potentially sequester the phage, which would reduce its relative transduction efﬁciency on E. coli K12 NCM3722 (yellow) and S. ﬂexneri 2a 2457O (blue), respectively.56A reduced transduction efﬁciency would give a lower number of phagemid transductant when compared to thatrecovered with non-LPS treated phagemid lysates. The transduction efﬁciency of P1(S) (yellow diamonds, blacklines) and P1(S’) (blue squares, blue lines) after pre-incubation with 0.5 lg/mL, 5.0 lg/mL, 50 lg/mL, 500 lg/mL and 5000 lg/mL of LPS extracted from ( b)W T S. ﬂexneri 5a M90T, ( c)W T S. ﬂexneri 2a 2457O, ( d)W T E. coli O3, ( e) Drfc S. ﬂexneri 5a M90T, ( f)Drfc S. ﬂexneri 2a 2457O, ( g)Dwzy E. coli O3 2457O, ( h)DwaaL S. ﬂexneri 5a M90T, ( i) DwaaL S. ﬂexneri 2a 2457O, ( j)DwaaL E. coli O3. All the graphs share the same y-axis for transduction efﬁciency (%) and an x-axis for LPS concentration ( lg/mL). Each data point represented results generated using 1 LPS sample, 1 (polythene glycol-treated)-lysate sample, 3 biological repeats (host cells used) and 2 technical repeats. Data wererepresented as mean ± SEM.Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 9', 'page_label: 167838 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  phagemid transductants after P1(S) infection of WT and O-antigen deﬁcient mutant strains of S. ﬂexneri , as well as E. coli K12 NCM3722 ( Figure 2 (d)). 5lg/mL to 50 lg/mL Drfc S. ﬂexneri 2a 2457O LPS reduced P1(S) transduction efﬁciency by approximately 57 % ( s=2 . 0 , n= 12, p < 0.0005), indicating a greater inhibitory effect of the LPSagainst P1(S) transduction when compared to the LPS of WT S. ﬂexneri 2a 2457O ( Figure 4 (f)). Treatment of P1(S) lysates with WT E. coli O3 LPS yielded an average transduction efﬁciency of110 % at all concentrations tested ( s=7 . 5 , n= 30, p > 0.05), while 5000 lg/mL of Dwzy E. coli O3 LPS was needed to reduce P1(S) transduction efﬁciency by approximately 54 % (s = 6.2, n = 6, p < 0.0005) ( Figure 4 (d), (g)). Contrastingly, 5 lg/mL of DwaaL E. coli O3 LPS reduced P1(S) transduction efﬁciency by 47 % (s = 4.8, n = 6, p < 0.0005), followed by complete inhibition of its transduction at concentrationsgreater or equal to 500 lg/mL ( Figure 4 (j)). Differences in the inhibitory effect of E. coli O3 LPS against P1(S) transduction suggested agreater adsorption efﬁciency of P1(S) to the LPSofDwaaL E. coli O3 when compared to the LPS of WT E. coli O3 and its isogenic Dwzy mutant. A greater adsorption efﬁciency of P1(S) on DwaaL E. coli O3 was consistent with a signiﬁcantly higher number of phagemid transductants recovered after P1(S) infection of the mutantstrain, when compared to that recovered after P1 (S) infection of WT and its isogenic Dwzy strains (Figure 2 (d)). Taken together, the LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with at least one O-antigen unit produced inhibitory effects against P1(S’) transduction, indicating that these LPSmight allow P1(S’) phage adsorption. E. coli O111:B4 LPS blocked P1(S’) transduction completely The LPS inhibition assay could potentially identify other bacterial strains which might be susceptible toP1(S’)- and P1(S)-mediated transduction. Hence, we sought to test if P1(S) and P1(S’) could adsorb to the smooth LPS of other E. coli strains, by repeating the LPS inhibition assay with E. coli O26:B6, E. coli O55:B5, E. coli O111:B4 and E. coli O127:B8 LPS (refer to Supplementary Figures 7 for the chemical structures of LPS). 5000 lg/mL of E. coli O26:B6 and E. coli O55:B5 LPS reduced P1(S’) transduction efﬁciency by 76 %(s=1 . 4 , n= 6, p < 0.0005) and 50 % ( s= 3.4, n=6 , p < 0.0005) respectively ( Figures 5 (a),(b)). 5lg/mL to 5000 lg/mL of E. coli O127:B8 LPS caused an average 23 % ( s= 1.6, n= 24, p < 0.05) reduction in P1(S’) transduction efﬁciency ( Figure 5 (d)). On the contrary, E. coli O111:B4 LPS completely blocked P1(S’) transduction when applied atconcentrations greater or equal to 5 lg/mL (p < 0.0005) ( Figure 5 (c)).', 'page_label: 167838 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  4 , n= 6, p < 0.0005) and 50 % ( s= 3.4, n=6 , p < 0.0005) respectively ( Figures 5 (a),(b)). 5lg/mL to 5000 lg/mL of E. coli O127:B8 LPS caused an average 23 % ( s= 1.6, n= 24, p < 0.05) reduction in P1(S’) transduction efﬁciency ( Figure 5 (d)). On the contrary, E. coli O111:B4 LPS completely blocked P1(S’) transduction when applied atconcentrations greater or equal to 5 lg/mL (p < 0.0005) ( Figure 5 (c)). The high inhibitoryeffect of E. coli O111:B4 LPS was similar to that of WT S. ﬂexneri 2a 2457O and WT S. ﬂexneri 5a M90T LPS, indicating that P1(S’) could adsorb to these LPS at a high efﬁciency ( Figures 4 (b), (c)). Treatment of P1(S) lysates with E. coli O111:B4 and E. coli O127:B8 LPS yielded an average transduction efﬁciency of 105 % at allconcentrations ( E. coli O111:B4, s=4 . 8 , n=3 0 , p > 0.05; E. coli O127:B8, s = 5.6, n=3 0 , p > 0.05) ( Figures 5 (c), (d)). 5000 lg/mL of E. coli O26:B6 and E. coli O55:B5 LPS reduced P1(S) transduction efﬁciency by approximately 93 % (s= 2.0, n= 6, p < 0.0005) and 38 % ( s=1 . 7 , n= 6, p < 0.0005) respectively ( Figures 5 (a),(b)). A higher inhibitory effect of E. coli O26:B6 LPS against P1(S) transduction indicated a greater P1 (S) adsorption efﬁciency to this LPS whencompared to the LPS of E. coli O55:B5, E. coli O111:B4 and E. coli O127:B8. Taken together, a low concentration of E. coli O111:B4 LPS blocked (S’) transduction completely, which indicated a high P1(S’) adsorption efﬁciency on E. coli O111:B4 LPS, and that this bacterial strain might be susceptible to P1(S’)-mediated transduction. P1(S’) lysates produced a higher Cas9- mediated antimicrobial effect on S. ﬂexneri 5aM90T and S. ﬂexneri 2a 2457O P1(S’) infection of S. ﬂexneri 5a M90T and S. ﬂexneri 2457O yielded a higher number of phagemid transductant, indicating that P1(S’) could deliver the phagemid DNA into S. ﬂexneri at a higher efﬁciency when compared to P1(S) (Figure 1 (c)). We therefore hypothesised that an increase in the transduction efﬁciency of a cas9 phagemid into S. ﬂexneri would produce a higher antimicrobial effect on the bacteria. To verify this hypothesis, the cas9 system was transduced into S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O with either P1(S) or P1(S’), and the differences in the Cas9-mediated antimicrobial effect on both S. ﬂexneri strains were assessed. A protospacer sequence targeting the chromosomal shiA gene sequence of S. ﬂexneri was cloned into the cas9 phagemid (phagemid termed cas9 -shiA hereafter) (Figure 6 (a)).', 'page_label: 167838 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  We therefore hypothesised that an increase in the transduction efﬁciency of a cas9 phagemid into S. ﬂexneri would produce a higher antimicrobial effect on the bacteria. To verify this hypothesis, the cas9 system was transduced into S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O with either P1(S) or P1(S’), and the differences in the Cas9-mediated antimicrobial effect on both S. ﬂexneri strains were assessed. A protospacer sequence targeting the chromosomal shiA gene sequence of S. ﬂexneri was cloned into the cas9 phagemid (phagemid termed cas9 -shiA hereafter) (Figure 6 (a)). cas9 -shiA phagemid as well as the non-chromosomal-targeting, cas9 -NT phagemid, were used for preparation of P1(S) and P1(S’)phagemid lysates. To verify if all the lysates yielded a similar titre of phagemid transducing particles, phagemid lysates were treated withDNase I and quantiﬁed for P1-packagedphagemid DNA via qPCR. All the DNase I-treated lysates yielded a similar concentration of P1- packaged phagemid DNA, suggesting thatdifferences in the Cas9 antimicrobial effect on S. ﬂexneri would be attributed to the relative transduction efﬁciency of phagemid DNA(Supplementary Figures 2a, 2b ).S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O were then treatedY.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 10'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 167835 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  visualised via silver staining. LPS extracted from WT cells gave a ladder-like pattern of multiple bands on the silver-stained gel, indicating the pres- ence of O-antigen repeating units of varying lengths(Figures 2 (b),(c)). LPS extracted from the DwaaL andDrfemutants gave a single band, indicating the absence of O-antigen. The LPS of the Drfc mutants gave two bands on the silver-stained gel, in which one was a unit higher than the bands observed for DwaaL andDrfemutants therefore veriﬁed the semi-rough type LPS of Drfcmutants with a single O-antigen unit. To determine the relative transduction efﬁciency of phagemid in WT, DwaaL ,Drfeand Drfc/wzy strains of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3, the bacteria cells were treated with either P1(S) or P1(S’) phagemidlysates and the number of phagemid transductants was enumerated. No phagemid transductant was recovered after P1(S’) infectionof the DwaaL andDrfemutants ( Figure 2 (d)). The lack of phagemid transductant was not due to cell death after P1(S’) infection, as we recoveredapproximately 1 /C210 8to 5/C2108CFU (per mL reaction mix) on non-selective LB (data not shown). Complementation of the DwaaL andDrfe mutations in trans, by providing a waaL and a rfe gene respectively, restored the smooth type LPS as well as the recovery of phagemid transductants after P1(S’) infection of the DwaaL and Drfe mutants ( Supplementary Figures 4a, 4b, 4c ). On the contrary, P1(S’) infection of Drfc S. ﬂexneri 5a M90T and Drfc S. ﬂexneri 2a 2457O yielded a 216-fold ( s=4 4 , n= 12, p < 0.0005) and 58-fold (s=9 . 8 , n= 12, p < 0.0005) lower number of phagemid transductant respectively, when compared to P1(S’) infection of WT S. ﬂexneri (Figure 2 (d)). P1(S’) infection of Drfc E. coli O3 yielded a 2-fold higher ( s=0 . 1 8 , n=1 2 , p < 0.0005) number of phagemid transductantwhen compared to P1(S’) infection of E. coli O3. There were no signiﬁcant differences in the number of phagemid transductant recoveredbetween P1(S) infection of WT, DwaaL andDrfestrains of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T (p > 0.05) ( Figure 2 (d)). Although P1(S) infection of WT and Drfc S. ﬂexneri 2a 2457O yielded a similar number of phagemid transductant(p > 0.05), P1(S) infection of the Drfc S. ﬂexneri 5a M90T yielded a 321-fold ( s=1 4 8 , n=1 2 , p < 0.005) lower number of phagemidtransductant when compared to infection of WT S. ﬂexneri 5a M90T. While no phagemid transductant was recovered after P1(S) infection of WT E. coli O3, P1(S) infection of its isogenic DwaaL and Drfe mutants yielded approximately 4.3 X108(s=6 . 6 /C2 107,n=1 2 )a n d1 . 3 /C2109(s=1 . 8 /C2108,n= 12) phagemid transductants per mL lysate,respectively.', 'page_label: 167835 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  While no phagemid transductant was recovered after P1(S) infection of WT E. coli O3, P1(S) infection of its isogenic DwaaL and Drfe mutants yielded approximately 4.3 X108(s=6 . 6 /C2 107,n=1 2 )a n d1 . 3 /C2109(s=1 . 8 /C2108,n= 12) phagemid transductants per mL lysate,respectively. Drfc E. coli O3, however, was permissive for P1(S) transduction yet the number of phagemid transductant was approximately1500-fold ( s= 93, n= 12, p < 0.0005) and 5000- fold ( s= 5000, n= 297, p < 0.0005) lower, when compared to that recovered after P1(S) infectionof its isogenic DwaaL and Drfe mutants, respectively. Overall, these data indicated that P1(S’) transduction of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 required at least one O- antigen unit, while the O-antigen deﬁcient mutants were still susceptible to P1(S) transduction. P1(S’) transduction of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T was not signiﬁcantly affected by mutations in the O-antigenmodiﬁcation genes All known serotypes of S. ﬂexneri , except for serotype 6, share the same O-antigen backbone with modiﬁcations such as glucoslylation and acetylation, which give rise to variation inserotypes. 37,15,50Despite the variations in O- antigen modiﬁcations, P1(S’) could transduce pha- gemid DNA into S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T at a similar efﬁciency ( Figure 1 (c)). We sought to determine if changes to the O-antigen modiﬁcations of S. ﬂexneri would affect P1(S’) and P1(S) transduction efﬁciency. We have generated Figure 3. P1(S’) transduction of S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O was not signiﬁcantly affected by mutations in the O-antigen modiﬁcation genes. (a) Schematic diagrams showing isogenic mutant strains of S. ﬂexneri 5a M90T (5a M90T) and S. ﬂexneri 2a 2457O (2a 2457O) with the expected change(s) to their glucosylation and/or acetylation of O-antigen, based on the respective genetic deletion(s).62,60All serotype of S. ﬂexneri , except for serotype 6, share a common O-antigen backbone comprised of repeating units of three L-rhamnose residues (RhaI- RhaII-RhaIII) and a N-acetylglucosamine residue (GlcNac). The genes involved in the acetylation and glucosylation of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T O-antigens were indicated as arrows.62,60The orientation of the glycosidic linkages between the acetyl and glucose molecules and the O-antigen backbone were shown as Arabic numerals. Blue dotted lines represented genetic deletion(s). (b) The number of phagemid transductants recovered after P1(S) (yellow bars) and P1(S’) (blue bars) phagemid lysates treatment of wildtype (WT) and all 8 mutant strainsofS.', 'page_label: 167835 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The genes involved in the acetylation and glucosylation of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T O-antigens were indicated as arrows.62,60The orientation of the glycosidic linkages between the acetyl and glucose molecules and the O-antigen backbone were shown as Arabic numerals. Blue dotted lines represented genetic deletion(s). (b) The number of phagemid transductants recovered after P1(S) (yellow bars) and P1(S’) (blue bars) phagemid lysates treatment of wildtype (WT) and all 8 mutant strainsofS. ﬂexneri 5a M90T (5a M90T) and S. ﬂexneri 2a 2457O (2a 2457O). The p-values were determined by Welch’s ANOVA. p < 0.0005 was indicated as as ***, for comparison of the number of phagemid transductants recovered between P1(S) infection of wildtype and Dgtr S. ﬂexneri 5a M90T, as well as the number of phagemid transductants recovered between P1(S) infection of wildtype and DoacB S. ﬂexneri 2a 2457O.3Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 7', 'page_label: 167836 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  8 mutant strains of S. ﬂexneri 2a 2457O and S. ﬂex- neri5a M90T that were defective in combinations of O-antigen acetylation and glycosylation, by deleting the O-antigen modiﬁcation gene(s) via lamda Redrecombination ( Figure 3 (a)).62,60The effects, if any, of these mutations on the LPS structures of S. ﬂexneri were visualised by silver stain. All the LPS samples showed a ladder-like pattern on the silver-stained gel, indicating the presence of O- antigen repeating units ( Supplementary Figure 5 a). Only the LPS of Dgtrmutants lacking a glucose modiﬁcation showed shift(s) in the band size(s) when compared to the LPS of WT cells.62,60 WT and all 8 mutant strains of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T were treated with P1(S) and P1(S’) phagemid lysate and the number of phagemid transductant wasenumerated. P1(S’) phagemid lysate treatment of WT and all 8 mutant strains of S. ﬂexneri yielded a similar number of phagemid transductant(p > 0.05) ( Figure 3 (b)). Contrastingly, P1(S) infection of Dgtr S. ﬂexneri 5a M90T yielded a 235-fold ( s=4 0 , n= 12, p < 0.0005) higher number of phagemid transductant when comparedto P1(S) infection of WT S. ﬂexneri 5a M90T (Figure 3 (b)).DoacB S. ﬂexneri 2a 2457O mutant yielded a 12-fold ( s= 1.4, n= 12, p < 0.0005) lower number of phagemid transductant after P1 (S) infection, when compared to the infection of WT S . ﬂexneri 2a 2457O. Complementation of the DoacB mutation in trans increased the number of phagemid transductant recovered after P1(S) infection ( Supplementary Figure 5 b). Unexpectedly, P1(S) infection of DoacB gtr, DoacB oacD and DoacB oacD gtr strains of S. ﬂexneri 2a 2457O yielded a similar number of phagemid transductant when compared to P1(S)infection of WT S. ﬂexneri 2a 2457O (p > 0.05) (Figure 3 (b),refer to Supplementary Materials 2 for p-values). Overall, the results indicated that P1(S’) transduction of phagemid DNA was not signiﬁcantly affected by mutations in the O-antigenmodiﬁcations genes of S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T, when compared to P1(S) transduction. The LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with at least one O-antigen unit could act as an inhibitor against P1(S’)transduction P1(S’) transduction of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 required the presence of at least one O-antigen unit ( Figure 2 (d)). We therefore hypothesised that the O- antigens of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 might be important for P1(S’) adsorption.', 'page_label: 167836 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with at least one O-antigen unit could act as an inhibitor against P1(S’)transduction P1(S’) transduction of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 required the presence of at least one O-antigen unit ( Figure 2 (d)). We therefore hypothesised that the O- antigens of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 might be important for P1(S’) adsorption. If this is true, an excess S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 LPS inphagemid lysates could sequester P1(S’) phage and this might reduce its transduction efﬁciency. To verify this hypothesis, LPS extracted from WT and O-antigen deﬁcient mutant strains of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 were assessed for their inhibitory effect against P1(S) and P1(S’)-mediated transduction ofE. coli NCM3722 and S. ﬂexneri 2457O, respectively ( Figure 4 (a)). LPS extracted from WT S. ﬂexneri 5a M90T and WTS. ﬂexneri 2a 2457O completely blocked P1 (S’) transduction when applied at concentrations greater or equal to 5 lg/mL (p < 0.0005) and 0.5lg/mL (p < 0.0005), respectively ( Figures 4 (b), (c)). The LPS of Drfc S. ﬂexneri 5a M90T completely blocked P1(S’) transduction at concentrations greater or equal to 50 lg/mL (p < 0.0005), while 5000 lg/mL of Drfc S. ﬂexneri 2a 2457O LPS was required to reduce the transduction efﬁciency of P1(S’) by approximately64 % ( s=7 . 3 , n=6 ,p<0 . 0 0 0 5 )( Figure 4 (e), (f)). 500 lg/mL of WT and Dwzy E. coli O3 LPS reduced P1(S’) transduction efﬁciency byapproximately 94 % (WT E. coli O3,s=2 . 6 , n=6 , p < 0.0005) and 96 % ( Dwzy E. coli O3,s=1 . 8 , n= 6, p < 0.0005), respectively ( Figures 4 (d), (g)). Overall, the inhibitory effect of LPS with at leastone O-antigen unit indicated that P1(S’) might adsorb to the LPS of S. ﬂexneri 5a M90T, S. ﬂexneri 2a 2457O and E. coli O3 with at least one O-antigen unit. Contrastingly, treatment of P1(S’) lysate with the LPS of DwaaL mutants yielded an average transduction efﬁciency of 107 % at allconcentrations ( DwaaL S. ﬂexneri M90T, s=3 . 4 , n=5 ,p>0 . 0 5 ; DwaaL S. ﬂexneri 2457O, s= 4.9, n= 30, p > 0.05; DwaaL E. coli O3, s=5 . 7 , n= 30, p > 0.05) ( Figures 4 (h), (i), (j)).', 'page_label: 167836 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Contrastingly, treatment of P1(S’) lysate with the LPS of DwaaL mutants yielded an average transduction efﬁciency of 107 % at allconcentrations ( DwaaL S. ﬂexneri M90T, s=3 . 4 , n=5 ,p>0 . 0 5 ; DwaaL S. ﬂexneri 2457O, s= 4.9, n= 30, p > 0.05; DwaaL E. coli O3, s=5 . 7 , n= 30, p > 0.05) ( Figures 4 (h), (i), (j)). Similarly, treatment of P1(S’) lysate with E. coli K12 LPS yielded an average transduction efﬁciency of 105 % ( s=6 . 1 , n= 30, p > 0.05) at all concentrations ( Supplementary Figure 6 ). Therefore, the lack of inhibitory effect against P1 (S’) transduction indicated that P1(S’) might notadsorb to the LPS of E. coli K12 as well as the LPS of DwaaL mutant strains without O-antigen, which was consistent with the absence ofphagemid transductant after P1(S’) infection ofthese bacteria ( Figures 1 (c),2(d)). 5lg/mL of E. coli K12 LPS reduced P1(S) transduction efﬁciency by approximately 76 %(s=3 . 3 , n= 6, p < 0.0005), while concentrations greater or equal to 50 lg/mL completely blocked P1(S) transduction Supplementary Figure 6 ). 500lg/mL to 5000 lg/mL of LPS extracted from all strains of S. ﬂexneri reduced the transduction efﬁciency of P1(S) by at least 50 % ( Figures 4 (b), (c), (e), (f), (h), (i)). The inhibitory effect of S. ﬂexneri and E. coli K12 LPS against P1(S) transduction indicated phage adsorption to these LPS, which was consistent with the recovery ofY.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 8', 'page_label: 167837 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Figure 4. The LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with at least one O-antigen unit could act as an inhibitor against P1(S’) transduction. (a) Schematic diagram showing a LPS inhibition assay. LPS that allows the adsorption of P1(S) and/or P1(S’) could potentially sequester the phage, which would reduce its relative transduction efﬁciency on E. coli K12 NCM3722 (yellow) and S. ﬂexneri 2a 2457O (blue), respectively.56A reduced transduction efﬁciency would give a lower number of phagemid transductant when compared to thatrecovered with non-LPS treated phagemid lysates. The transduction efﬁciency of P1(S) (yellow diamonds, blacklines) and P1(S’) (blue squares, blue lines) after pre-incubation with 0.5 lg/mL, 5.0 lg/mL, 50 lg/mL, 500 lg/mL and 5000 lg/mL of LPS extracted from ( b)W T S. ﬂexneri 5a M90T, ( c)W T S. ﬂexneri 2a 2457O, ( d)W T E. coli O3, ( e) Drfc S. ﬂexneri 5a M90T, ( f)Drfc S. ﬂexneri 2a 2457O, ( g)Dwzy E. coli O3 2457O, ( h)DwaaL S. ﬂexneri 5a M90T, ( i) DwaaL S. ﬂexneri 2a 2457O, ( j)DwaaL E. coli O3. All the graphs share the same y-axis for transduction efﬁciency (%) and an x-axis for LPS concentration ( lg/mL). Each data point represented results generated using 1 LPS sample, 1 (polythene glycol-treated)-lysate sample, 3 biological repeats (host cells used) and 2 technical repeats. Data wererepresented as mean ± SEM.Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 9', 'page_label: 167838 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  phagemid transductants after P1(S) infection of WT and O-antigen deﬁcient mutant strains of S. ﬂexneri , as well as E. coli K12 NCM3722 ( Figure 2 (d)). 5lg/mL to 50 lg/mL Drfc S. ﬂexneri 2a 2457O LPS reduced P1(S) transduction efﬁciency by approximately 57 % ( s=2 . 0 , n= 12, p < 0.0005), indicating a greater inhibitory effect of the LPSagainst P1(S) transduction when compared to the LPS of WT S. ﬂexneri 2a 2457O ( Figure 4 (f)). Treatment of P1(S) lysates with WT E. coli O3 LPS yielded an average transduction efﬁciency of110 % at all concentrations tested ( s=7 . 5 , n= 30, p > 0.05), while 5000 lg/mL of Dwzy E. coli O3 LPS was needed to reduce P1(S) transduction efﬁciency by approximately 54 % (s = 6.2, n = 6, p < 0.0005) ( Figure 4 (d), (g)). Contrastingly, 5 lg/mL of DwaaL E. coli O3 LPS reduced P1(S) transduction efﬁciency by 47 % (s = 4.8, n = 6, p < 0.0005), followed by complete inhibition of its transduction at concentrationsgreater or equal to 500 lg/mL ( Figure 4 (j)). Differences in the inhibitory effect of E. coli O3 LPS against P1(S) transduction suggested agreater adsorption efﬁciency of P1(S) to the LPSofDwaaL E. coli O3 when compared to the LPS of WT E. coli O3 and its isogenic Dwzy mutant. A greater adsorption efﬁciency of P1(S) on DwaaL E. coli O3 was consistent with a signiﬁcantly higher number of phagemid transductants recovered after P1(S) infection of the mutantstrain, when compared to that recovered after P1 (S) infection of WT and its isogenic Dwzy strains (Figure 2 (d)). Taken together, the LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3 with at least one O-antigen unit produced inhibitory effects against P1(S’) transduction, indicating that these LPSmight allow P1(S’) phage adsorption. E. coli O111:B4 LPS blocked P1(S’) transduction completely The LPS inhibition assay could potentially identify other bacterial strains which might be susceptible toP1(S’)- and P1(S)-mediated transduction. Hence, we sought to test if P1(S) and P1(S’) could adsorb to the smooth LPS of other E. coli strains, by repeating the LPS inhibition assay with E. coli O26:B6, E. coli O55:B5, E. coli O111:B4 and E. coli O127:B8 LPS (refer to Supplementary Figures 7 for the chemical structures of LPS). 5000 lg/mL of E. coli O26:B6 and E. coli O55:B5 LPS reduced P1(S’) transduction efﬁciency by 76 %(s=1 . 4 , n= 6, p < 0.0005) and 50 % ( s= 3.4, n=6 , p < 0.0005) respectively ( Figures 5 (a),(b)). 5lg/mL to 5000 lg/mL of E. coli O127:B8 LPS caused an average 23 % ( s= 1.6, n= 24, p < 0.05) reduction in P1(S’) transduction efﬁciency ( Figure 5 (d)). On the contrary, E. coli O111:B4 LPS completely blocked P1(S’) transduction when applied atconcentrations greater or equal to 5 lg/mL (p < 0.0005) ( Figure 5 (c)).', 'page_label: 167838 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  4 , n= 6, p < 0.0005) and 50 % ( s= 3.4, n=6 , p < 0.0005) respectively ( Figures 5 (a),(b)). 5lg/mL to 5000 lg/mL of E. coli O127:B8 LPS caused an average 23 % ( s= 1.6, n= 24, p < 0.05) reduction in P1(S’) transduction efﬁciency ( Figure 5 (d)). On the contrary, E. coli O111:B4 LPS completely blocked P1(S’) transduction when applied atconcentrations greater or equal to 5 lg/mL (p < 0.0005) ( Figure 5 (c)). The high inhibitoryeffect of E. coli O111:B4 LPS was similar to that of WT S. ﬂexneri 2a 2457O and WT S. ﬂexneri 5a M90T LPS, indicating that P1(S’) could adsorb to these LPS at a high efﬁciency ( Figures 4 (b), (c)). Treatment of P1(S) lysates with E. coli O111:B4 and E. coli O127:B8 LPS yielded an average transduction efﬁciency of 105 % at allconcentrations ( E. coli O111:B4, s=4 . 8 , n=3 0 , p > 0.05; E. coli O127:B8, s = 5.6, n=3 0 , p > 0.05) ( Figures 5 (c), (d)). 5000 lg/mL of E. coli O26:B6 and E. coli O55:B5 LPS reduced P1(S) transduction efﬁciency by approximately 93 % (s= 2.0, n= 6, p < 0.0005) and 38 % ( s=1 . 7 , n= 6, p < 0.0005) respectively ( Figures 5 (a),(b)). A higher inhibitory effect of E. coli O26:B6 LPS against P1(S) transduction indicated a greater P1 (S) adsorption efﬁciency to this LPS whencompared to the LPS of E. coli O55:B5, E. coli O111:B4 and E. coli O127:B8. Taken together, a low concentration of E. coli O111:B4 LPS blocked (S’) transduction completely, which indicated a high P1(S’) adsorption efﬁciency on E. coli O111:B4 LPS, and that this bacterial strain might be susceptible to P1(S’)-mediated transduction. P1(S’) lysates produced a higher Cas9- mediated antimicrobial effect on S. ﬂexneri 5aM90T and S. ﬂexneri 2a 2457O P1(S’) infection of S. ﬂexneri 5a M90T and S. ﬂexneri 2457O yielded a higher number of phagemid transductant, indicating that P1(S’) could deliver the phagemid DNA into S. ﬂexneri at a higher efﬁciency when compared to P1(S) (Figure 1 (c)). We therefore hypothesised that an increase in the transduction efﬁciency of a cas9 phagemid into S. ﬂexneri would produce a higher antimicrobial effect on the bacteria. To verify this hypothesis, the cas9 system was transduced into S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O with either P1(S) or P1(S’), and the differences in the Cas9-mediated antimicrobial effect on both S. ﬂexneri strains were assessed. A protospacer sequence targeting the chromosomal shiA gene sequence of S. ﬂexneri was cloned into the cas9 phagemid (phagemid termed cas9 -shiA hereafter) (Figure 6 (a)).', 'page_label: 167838 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  We therefore hypothesised that an increase in the transduction efﬁciency of a cas9 phagemid into S. ﬂexneri would produce a higher antimicrobial effect on the bacteria. To verify this hypothesis, the cas9 system was transduced into S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O with either P1(S) or P1(S’), and the differences in the Cas9-mediated antimicrobial effect on both S. ﬂexneri strains were assessed. A protospacer sequence targeting the chromosomal shiA gene sequence of S. ﬂexneri was cloned into the cas9 phagemid (phagemid termed cas9 -shiA hereafter) (Figure 6 (a)). cas9 -shiA phagemid as well as the non-chromosomal-targeting, cas9 -NT phagemid, were used for preparation of P1(S) and P1(S’)phagemid lysates. To verify if all the lysates yielded a similar titre of phagemid transducing particles, phagemid lysates were treated withDNase I and quantiﬁed for P1-packagedphagemid DNA via qPCR. All the DNase I-treated lysates yielded a similar concentration of P1- packaged phagemid DNA, suggesting thatdifferences in the Cas9 antimicrobial effect on S. ﬂexneri would be attributed to the relative transduction efﬁciency of phagemid DNA(Supplementary Figures 2a, 2b ).S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O were then treatedY.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 10'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:50 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'77'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994459'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'332ms'), (b'x-request-id', b'a4000e63f1f461517d01485fe117c14f'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412495d1b735d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:50 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'77'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994459'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'332ms'), (b'x-request-id', b'a4000e63f1f461517d01485fe117c14f'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412495d1b735d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 167839 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Figure 5. Low concentration of E. coli O111:B4 LPS blocked P1(S’) transduction completely. Quantiﬁcation of P1(S) (yellow diamonds, black lines) and P1(S’) (blue squares, blue lines) transduction efﬁciency when pre- incubated with 0.5 lg/mL, 5.0 lg/mL, 50 lg/mL, 500 lg/mL and 5000 lg/mL of commercially available ( a)E. coli O26: B6, (b)E. coli O55:B5, ( c)E. coli O111:B4 and ( d)E. coli O127:B8 LPS samples. Each data point represented results generated from 1 LPS sample, 1 (polythene glycol treated-) lysate sample, 3 biological repeats (host cells used) and 2 technical repeats. Data were represented as mean ± SEM. ( e) Schematic diagrams showing the predicted chemical formulas of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T, E. coli O3 and E. coli O111:B4 LPS, which were shown to give at least one log reduction in the transduction efﬁciency of P1(S’) when applied at concentrations lower or equal to 500lg/mL. The LPS structures were determined in previous studies.37,59,51,29,50Differences between the outer core and O-antigens were highlighted in blue for the LPS of S. ﬂexneri 5a M90T and S. ﬂexneri 2457O, green for the LPS of E. coli O3 and yellow for the LPS of E. coli O111:B4. For simpliﬁcation purpose, the orientation of glycosidic bond linking the O-antigen to the outer oligosaccharide core of LPS, as well as possible modiﬁcations to inner and lipid A core of LPS were not shown. The O-antigens are expected to be linked to the terminal a-D-glucose of both R225and R336type LPS outer cores via a (O-antigen)-1,4-( a-D-glucose of outer core) glycosidic bond. p, phosphorylation; Ac, acetyl group; GlcNac, N-acetylglucosamine; Rha, rhamnose; Glc, glucose, RhaNac; N-acetylrhamnosamine; Gal, galactose; Col, colitose.Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 11', 'page_label: 167840 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 12', 'page_label: 167841 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  with varying volumes of the cas9-shiA orcas9 -NT phagemid lysates and the number of CFU recovered on non-selective agar was enumerated. The Cas9-antimicrobial effect was represented bya reduction in the number of CFU recovered after cas9-shiA phagemid lysates treatment, when compared to the use of cas9 -NT phagemid lysates. 4lLt o1 2 lL of P1(S’) cas9 -shiA lysates reduced the number of S. ﬂexneri 5a M90T CFU by 141-fold to 1000-fold, when compared to the number of CFUrecovered after treatment with P1(S’) cas9 -NT lysates ( Figures 6 (b)–(d)). There were no signiﬁcant differences in the number of S. ﬂexneri 5a M90T CFU recovered between P1(S) cas9 - shiA and P1(S) cas9 -NT lysates treatment (p > 0.05), which indicated a signiﬁcantly lower Cas9-mediated antimicrobial effect of P1(S)lysates on S. ﬂexneri 5a M90T when compared to that of P1(S’) lysates. P1(S’) cas9 -shiA lysates reduced the number of S. ﬂexneri 2a 2457O CFU by 100-fold to 850-fold when compared to the number of CFU recovered after treatment with P1 (S’) cas9 -NT lysates ( Figures 6 (b)–(d)). P1(S) cas9 -shiA treatment caused 7.5-fold to 220-fold reduction in the number of S. ﬂexneri 2a 2457O CFU when compared to the number of CFU recovered P1(S) cas9 -NT treatment, which indicated a lower Cas9-mediated antimicrobial effect of P1(S) lysates on S. ﬂexneri 2a 2457O when compared to P1(S’) lysates. Taken together, P1(S’) cas9 phagemid lysates produced a greater antimicrobial effect on both S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O, indicating that P1(S’) improved the transductionefﬁciency of cas9 phagemids into these bacteria when compared to P1(S). Discussion P1 bacteriophage has an invertible C segment, encoding two sets of tail ﬁbres namely the S andS’.31,26,30Despite the wide usage of P1 for gener- alised transduction, the host range of P1(S’) was never determined. Here, we showed evidence of P1(S’) transduction of S. ﬂexneri 2a 2457O, S. ﬂex- neri5a M90T and E. coli O3, as well as the require- ment of O-antigen for its transduction. Data presented in this study provided novel insights intoP1 tropism-switching and showed promise as a tool for improving the transduction efﬁciency of a cas9 antimicrobial phagemid into S. ﬂexneri . The effect(s) of O-antigen on P1 transduction The presence of O-antigen was reported to reduce the transduction efﬁciency ofbacteriophage P1. Previous studies have shown that restoring the O-antigen expression of E. coli K12 MG1655 inhibited P1 phage transduction, 6 while galmutants of E. coli O157:H7 lacking O- antigen were permissive for P1 infection.27The inhi- bitory effect of O-antigen against P1 transductioncontradicts with an earlier study by Godard et al.,19 which showed that P1kc virinfected smooth strains ofS. ﬂexneri F6S.', 'page_label: 167841 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The effect(s) of O-antigen on P1 transduction The presence of O-antigen was reported to reduce the transduction efﬁciency ofbacteriophage P1. Previous studies have shown that restoring the O-antigen expression of E. coli K12 MG1655 inhibited P1 phage transduction, 6 while galmutants of E. coli O157:H7 lacking O- antigen were permissive for P1 infection.27The inhi- bitory effect of O-antigen against P1 transductioncontradicts with an earlier study by Godard et al.,19 which showed that P1kc virinfected smooth strains ofS. ﬂexneri F6S. Notably, P1 lysates used in all these studies were prepared lytically on E. coli K12, which would have given a majority of P1 pro- geny with the S tail ﬁbre, since only P1(S) phage can infect E. coli K12 and that inversion of the C segment is infrequent. Our results indicated that both the P1(S’) and P1(S) can transduce phagemid DNA into wildtype S. ﬂexneri 5a M90T and S. ﬂex- neri 2a 2457O with intact O-antigen, albeit with varying efﬁciencies. Therefore, the inhibitory effects of O-antigen against P1 transduction might bedependent on the bacterial strains used, discrepan-cies between transduction of phagemid DNA versus P1 genomic DNA, as well as the tail ﬁbre of P1 phage, in which the host range(s) of P1 alternativeS’ tail ﬁbre was not determined from previous stud- ies. 31,30Since a non-replicative P1 phage is desir- able for delivering the DNA sequence-speciﬁc Figure 6. A higher antimicrobial effect of P1(S’) lysates on S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O. a) Schematic diagram showing the cas9 genetic construct of P1 J72114 phagemid, with protospacer sequence targeting chromosomal gene, shiA ofS. ﬂexneri (in light blue). Following transduction and circularisation of the phagemid, the presence of a CRISPR guide RNA (crRNA) with spacer sequence complementary to the chromosomal shiA gene of S. ﬂexneri would cause dsDNA cleavage of the chromosome via Cas9 endonuclease, leading to cell death (shown as dashed lines). cas9 -NT phagemid with a randomly-generated spacer sequence can be stably maintained in transduced cells (host cell survival indicated by ﬁlled lines). Quantiﬁcation of the number of S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O CFU recovered after treatments with b)4lL,c)8lL and d)12lL of P1(S) or P1(S’) non-targeting cas9 -NT (light brown bars) and chromosomal-targeting cas9 -shiA (light blue bars) lysates. Approximately 107CFU was used for each experiment. n.s. represents no signiﬁcant difference in the number of S. ﬂexneri 5a M90T CFU recovered between P1(S) cas9 -NT and P1(S) cas9 -shiA lysates treatments. p < 0.0005 are shown as ***, for comparisons of the antimicrobial effects between P1(S) and P1(S’) cas9 lysates on S. ﬂexneri 5a M90T at all three volumes, as well as that on S. ﬂexneri 2a 2457O at 8 lL of lysates. p < 0.005 is shown as **, for comparisons of the antimicrobial effects between 4 lL of P1(S) and 4 lL of P1(S’) cas9 lysates on S. ﬂexneri 2a 2457O.', 'page_label: 167841 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  p < 0.0005 are shown as ***, for comparisons of the antimicrobial effects between P1(S) and P1(S’) cas9 lysates on S. ﬂexneri 5a M90T at all three volumes, as well as that on S. ﬂexneri 2a 2457O at 8 lL of lysates. p < 0.005 is shown as **, for comparisons of the antimicrobial effects between 4 lL of P1(S) and 4 lL of P1(S’) cas9 lysates on S. ﬂexneri 2a 2457O. p < 0.05 is shown as *, for comparisons of the antimicrobial effects between 12 lL of P1(S) and 12 lL of P1(S’) cas9 lysates on S. ﬂexneri 2a 2457O.3Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 13', 'page_label: 167842 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Cas9 antimicrobial system, it remained unknown if E. coli O3 and S. ﬂexneri could support the produc- tion of P1(S’) progeny. Therefore, the infection of E. coli O3 and S. ﬂexneri strains with a replicative P1 mutant which only expresses the S’ tail ﬁbre, such as the mutant P1-15::IS 2phage described by Iida et al., (1984), would be important to verifyif these bacterial strains could support phage prop- agation after transduction. The requirement of O-antigen for P1(S’)- mediated transduction Our results indicated that a lack of O-antigen inhibited P1(S’) transduction of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3. Indeed, rough LPS extracted from the O-antigendeﬁcient mutant strains lacked P1(S’) neutralising effect, highlighting the possible role of these O- antigens in mediating P1(S’) adsorption. O-antigenchains serve as receptors for the tailspike proteinsof multiple bacteriophages, including bacteriophage P22 that infects Samonella spp. 58 and bacteriophage Sf6 which infects S. ﬂexneri ser- otypes X and Y.17Most of these tail spike proteins promote the degradation and/or the modiﬁcation of O-antigen chains thus orientating the bacterio-phage to a secondary receptor for irreversible bind- ing. Such mechanism, however, is expected to be different to that of the P1(S’) tail ﬁbre, due to a lackof signiﬁcant amino acid sequence homology to anyknown tail spike protein targeting S. ﬂexneri O- antigen, nor to any known enzymatic domain involved in degrading glycosidic bond. Comparative analysis of P1 S’ tail ﬁbre amino acid sequence (NCBI, NC_005856.1, 32225.0.33782) indicated a high sequence similarity ( /C2599 %) to the protein product of multiple phage tail ﬁbre genes that were annotated in E. coli ,S. enterica and E. albertii genomes (data not shown). One such homologue of P1 S’gene is the S’tail ﬁbre gene of Mu bacteriophage (NCBI, NC_000929.1, 34067.0.35053). The Mu S’gene shows a high sequence similarity ( /C2590 %) to the 3 0-end (639th to 2919th DNA sequence) of P1 S’ gene, which encodes the predicted host-range-determining- region (HRDR) of P1 S’ tail ﬁbre (NCBI,NC_005856.1, 32225.0.33782) (data not shown). The Mu S’ tail ﬁbre, however, was shown to recognise a glucose- b1,6-glucose structure of Erwinia chrysanthemi (B37) LPS,57and this struc- ture is not found on the LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T, E. coli O3 and E. coli O111:B4 .Hence, P1 S’ might have a different receptor compared to Mu S’, which is consistent with the fact that neither Mu(S) nor Mu(S’) formed plaques on 12 different serotypes of S. ﬂexneri .32 Interestingly, Jakhetia and Verma32reported a Mu-like phage, SfMu which formed plaques on S. ﬂexneri serotypes 3b and Y. The differences between Mu and SfMu host range is attributed to 18 amino acid changes in the tail ﬁbre of SfMu whencompared to Mu S’ tail ﬁbre.', 'page_label: 167842 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Comparisons between the C-terminus of P1 S’ tail ﬁbre, the tail ﬁbre of SfMu (NCBI, KP010268, ORF49) and the Mu S’ tail ﬁbre indicated that P1 S’ tail ﬁbre contains additionalamino acid changes at multiple sites, which could potentially contribute towards differences in the host range of these phages (data not shown). Acomparative analysis of the amino acid sequence between P1 S’ tail ﬁbre and its homologs could determine how changes in speciﬁc amino acid resi-dues lead to variations in the phages host range. The LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T, E. coli O3 and E. coli O111:B4 with O- antigen (termed O-PS hereafter) produced a highneutralising activity against P1(S’) transduction, which indicated common feature(s) shared among these O-PS might be the receptor for P1(S’). Thepredicted similarity between S. ﬂexneri ,E. coli O3 andE. coli O111 O-PS is a b-D-Glc pNac-1, 4- a-D- Glc residue which links the O-antigens to theterminal glucose of LPS outer core oligosaccharide ( Figure 5 (e)). This residue, however, is also present in the LPS of E. coli O26: B6 29,42which yielded a weaker P1(S’) neutralising effect compared to the LPS of wildtype E. coli O3 andS. ﬂexneri . Therefore, it remained inconclusive if ab-D-Glc pNac-1, 4- a-D-Glc residue serves as the receptor for P1(S’) since LPS can adopt different conformations based on their respective side chains, which could have reduced the accessibilityof this residue on the LPS of E. coli O26:B6.53,18 Furthermore, we could not dismiss the role of other cell surface antigens in mediating P1(S’) transduc-tion and/or adsorption. Genomic analysis indicatedthe presence of a K2ab polysaccharide capsule operon and a H2 ﬂagellar operon on the strain of E. coli O3 used for this study. Hence, the effects, if any, of these surface antigens on P1(S’) adsorp- tion and/or transduction remain to be elucidated. While the bacterial LPS were only assessed for inhi-bitory effect against P1 transduction, a direct mea- surement of the binding afﬁnity of P1 tail ﬁbres to the different bacterial LPS would provide insightsinto discrepancies in the inhibitory effect of different LPS against P1(S) and P1(S’) transduction. Fur- thermore, quantiﬁcation of P1(S’) transduction efﬁ-ciency on various Gram-negativeEnterobactericae , coupled with structural studies of the P1(S’) tail ﬁbre as well as the use of synthetic glycans mimicking different regions of LPSoligosaccharides, will help in identifying its receptor and the mechanisms behind P1(S’) adsorption. The effects of O-antigen and its modiﬁcations on P1(S)-mediated transduction Earlier studies suggested that P1(S) tail ﬁbre recognises the terminal glucose molecule on the outer LPS core without discriminating the orientation of a glycosidic bond. 16,49,56Indeed, glu- cose molecule is present in the LPS of E. coli and S. ﬂexneri strains tested in our study, which is con-Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 14', 'page_label: 167843 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  sistent with the fact that P1(S) infected most of these bacteria. A previous study of S. ﬂexneri 2a O-antigen indicated that loosing speciﬁc acetylation modiﬁcation(s) affected the sensitivity of S. ﬂexneri 2a 2457O towards bacteriophage Sf6c infection.60 Similarly, our results indicated that P1(S) transduc-tion was signiﬁcantly affected by a Dgtrand a Drfc mutation in S. ﬂexneri 5a M90T, as well as a DoacB mutation in S. ﬂexneri 2a 2457O. While P1(S) failed to transduce phagemid DNA into wildtype E. coli O3, its isogenic DwaaL andDrfemutants were per- missive for P1(S) transduction, and the rough LPS of its DwaaL mutant produced a high inhibitory effect against P1(S). Taken together, accessibilityof the P1(S) receptor might be affected by muta- tions in the genes of O-antigen assembly, biosyn- thesis and its modiﬁcation. Further structuralstudies of the P1(S) tail ﬁbre might provide insights into the molecular mechanisms behind P1(S) adsorption to its receptor. The potential use of P1(S’) for delivering an antimicrobial cas9 system into pathogenic strains of Enterobacteriaceae Besides S. ﬂexneri and E. coli O3, the high inhibitory effect of E. coli O111:B4 LPS against P1 (S’) transduction indicated that this E. coli strain might be susceptible to P1(S’)-mediated transduction. E. coli O111:B4, however, was shown to be resistant to P1 phage infection, dueto the presence of a P1 or a P1-like resident phage. 10Subsequently, cells cured from the lyso- genic phage were susceptible to P1 infection usinglysates prepared from a thermoinducible P1 lyso- gen. Since lysates prepared from an induced lyso- gen would yield both P1(S) and P1(S’) phages,the factor(s) required for P1(S’) adsorption might be present in E. coli O111:B4, which is consistent with the high P1(S’) neutralising effect of E. coli O111:B4 LPS. Serotyping analysis showed that E. coli O111 is associated with strains of enterotox- igenic E. coli (ETEC) 7while E. coli O3 is associated with clinical isolates of enteroaggregative (EAEC)E. coli .33,35,14Hence, P1(S’) could be a valuable vector for the delivery of the cas9 antimicrobial pha- gemid into clinical isolates of E. coli O111 and E. coli O3. Furthermore, P1(S’) transduction could also be expanded to target other serotypes of S. ﬂexneri , since most serotypes share a common O- antigen backbone, and that P1(S’) transduction might not be signiﬁcantly affected by changes in the O-antigen modiﬁcations of S. ﬂexneri . The appli- cation of P1(S’) to target other serotypes of S. ﬂex- neri is relevant, as the development of a broadly protective vaccine against S. ﬂexneri is hampered by its serotypic and genomic variations.44,2Notably, our preliminary transduction assay indicated that P1 (S’) infection of S. ﬂexneri serotype 2b ATCC/C210 12022 yielded a higher number of phagemid trans- ductant when compared to that recovered after P1 (S) infection (data not shown) thus highlighting thepotential of P1(S’) for delivering the DNA sequence-speciﬁc Cas9 antimicrobial system into different S. ﬂexneri serotypes.', 'page_label: 167843 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The appli- cation of P1(S’) to target other serotypes of S. ﬂex- neri is relevant, as the development of a broadly protective vaccine against S. ﬂexneri is hampered by its serotypic and genomic variations.44,2Notably, our preliminary transduction assay indicated that P1 (S’) infection of S. ﬂexneri serotype 2b ATCC/C210 12022 yielded a higher number of phagemid trans- ductant when compared to that recovered after P1 (S) infection (data not shown) thus highlighting thepotential of P1(S’) for delivering the DNA sequence-speciﬁc Cas9 antimicrobial system into different S. ﬂexneri serotypes. It is noteworthy that P1 transduction and the propagation of bacterialstrains were performed in nutrient-rich LB medium for this study thus ﬂuctuations in the LPS polysac- charide composition were expected to be minimal,in the absence of mutations in LPS biosynthesis and assembly gene(s). While a wildtype LPS struc- ture is critical for the survival of S. ﬂexneri and pathogenic E. coli in the intestinal milieu, 28,62,45,40,54 variations in nutrients availability, particularly in the external environment, could potentially alter the bacterial LPS polysaccharide composition and con-sequently affecting the rate of P1 adsorption. Therefore, the use of chemically-deﬁned medium with speciﬁc carbon source(s) would provide impor-tant insights into the phage-bacteria interactions under nutrients-limiting conditions. Understanding such interactions could demonstrate the potentialapplication of P1 phagemid system to target S. ﬂex- neriand other susceptible pathogenic bacteria out- side of the human and/or animal host(s). Materials and Methods Bacterial strains, plasmids and media used in this study All the bacterial strains used for this study were listed in Supplementary Table 1 while plasmids used were listed in Supplementary Table 2 .A l l PCR reactions for molecular cloning was carried out using Q5/C210High-Fidelity DNA Polymerase (NEB, M0491) while colony PCR was carried outusing Taq Polymerase (NEB, M0270L). All plasmids were assembled using Gibson Assembly (NEB, E2621). Bacteria of this study wereroutinely cultured in Luria–Bertani (LB) broth andagar. Drfcmutants of S. ﬂexneri exhibited very slow growth after lambda Red recombineering. Hence mutants were cultured on Tryptic SoyBroth (TSB) before subsequent culturing in LB media. SM buffer (50 mM Tris-HCl, 8 mM magnesium sulfate, 100 mM sodium chloride, pH7.5), was routinely used as the buffer for manipulating phage lysates. DNA sequences used in this study were listed in Supplementary Tables 4 . Genomic sequencing of E. coli O3 and agglutination assay Preliminary results suggested that P1(S’) can transduce phagemid DNA into an unknown E. coli strain at a high efﬁciency. Since the identity of this bacterial strain was not determined, genomic sequencing was performed by MicrobesNG(Birmingham, United Kingdom, https:// microbesng.com/ ), with llumina next-generation sequencing data at a coverage of 30X. Cell masswas provided to MicrobesNG .Genomic sequencing data is available at NCBI, under theY.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 15', 'page_label: 167844 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  accession number JALKIH000000000, Bioproject: PRJNA825326, Biosample: SAMN27512004. Artemis (Oxford, United Kingdom) was used to browse the assembly and annotations.8The O- antigen gene cluster of E. coli is commonly located between the galF and gnd genes sequences,59 which was annotated in the assembly data. BLASTn (Nucleotide BLAST, NIH) was used to identify homologous DNA sequence, and the results suggested a 99 % (14043/14049) identitymatch to the E. coli O3 O-antigen gene cluster iden- tiﬁed in Ren et al., study (2008). DNA sequence of the O3 O-antigen gene cluster was deposited onto GenBank, with the accession number ON210125.The expression of E. coli O3 O-antigen was con- ﬁrmed by agglutination assay using anti- E. coli O3 antiserum (Oxford Biosystems, 84999). Overnightculture of E. coli O3 was boiled for 1 h on heat block at 100 /C176C. The boiled culture was left to stand at room temperature for 1 h to allow cell debris to sed-iment. Equal volume of antiserum was added to the boiled cultures in glass tubes, and the reactions were incubated overnight at 52 /C176C in a water bath. Glass tubes were shaken vigorously before readingthe reaction. E. coli O3 culture gave a granular tex- ture indicating positive reaction while a homoge- nous milky turbidity was observed for S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli E24377A cultures, indicating negative reactions. Transformation of E. coli and S. ﬂexneri Transformation of E. coli and S. ﬂexneri was carried out using standard electroporation protocol. Overnight culture of cells was sub-cultured by 1/100 in fresh LB media at 37 /C176Cw i t h shaking until it reached an OD 600of approximately 0.35. Cells were centrifuged at 3000 /C2g for 15 min at 4/C176C. Cell pellet was washed with ice-cold 10 % glycerol (in distilled water), and the washing step was repeated for another 2 times. The ﬁnal cell pellet was concentrated by 200-fold in distilledwater with 10 % glycerol (e.g., 250 lLo f1 0% glycerol for cell pelleted from a 50 mL culture). One lL of DNA (approximately 10–20 ng) was electroporated into 40 lL of cells using Gene Pulser/MicroPulser Electroporation Cuvettes, 0.1 cm gap (Biorad, 1652083), and immediatelytransferred into 1 mL of ice-cold SOC. Cells were incubated at 30 /C176C for temperature sensitive plasmids, pKD46 and pCP20, or 37 /C176C for other DNA substrate, with shaking for 1 h. Cells wereplatted onto LB agar plates with the appropriate antibiotics, followed by overnight incubation at 30/C176Co r3 7 /C176C. P1 phagemid lysate preparation Preparation of phagemid lysates was carried out based on a protocol from Kittleson et al. 39study, with slight modiﬁcations. Overnight cultures of E. coli P1 lysogenic EMG16 cells transformed with phagemidand/or with plasmid, were sub-cultured in fresh PLM media (LB supplemented with 5 mM of Ca2+ and 10 mM Mg2+)a t3 7 /C176C, with shaking until it reached an OD 600between 0.15 to 0.2, which took approximately 2 hours. L-arabinose was added to the culture, giving a ﬁnal concentration of 0.02 % or approximately 1.3 mM.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 167839 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Figure 5. Low concentration of E. coli O111:B4 LPS blocked P1(S’) transduction completely. Quantiﬁcation of P1(S) (yellow diamonds, black lines) and P1(S’) (blue squares, blue lines) transduction efﬁciency when pre- incubated with 0.5 lg/mL, 5.0 lg/mL, 50 lg/mL, 500 lg/mL and 5000 lg/mL of commercially available ( a)E. coli O26: B6, (b)E. coli O55:B5, ( c)E. coli O111:B4 and ( d)E. coli O127:B8 LPS samples. Each data point represented results generated from 1 LPS sample, 1 (polythene glycol treated-) lysate sample, 3 biological repeats (host cells used) and 2 technical repeats. Data were represented as mean ± SEM. ( e) Schematic diagrams showing the predicted chemical formulas of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T, E. coli O3 and E. coli O111:B4 LPS, which were shown to give at least one log reduction in the transduction efﬁciency of P1(S’) when applied at concentrations lower or equal to 500lg/mL. The LPS structures were determined in previous studies.37,59,51,29,50Differences between the outer core and O-antigens were highlighted in blue for the LPS of S. ﬂexneri 5a M90T and S. ﬂexneri 2457O, green for the LPS of E. coli O3 and yellow for the LPS of E. coli O111:B4. For simpliﬁcation purpose, the orientation of glycosidic bond linking the O-antigen to the outer oligosaccharide core of LPS, as well as possible modiﬁcations to inner and lipid A core of LPS were not shown. The O-antigens are expected to be linked to the terminal a-D-glucose of both R225and R336type LPS outer cores via a (O-antigen)-1,4-( a-D-glucose of outer core) glycosidic bond. p, phosphorylation; Ac, acetyl group; GlcNac, N-acetylglucosamine; Rha, rhamnose; Glc, glucose, RhaNac; N-acetylrhamnosamine; Gal, galactose; Col, colitose.Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 11', 'page_label: 167840 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 12', 'page_label: 167841 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  with varying volumes of the cas9-shiA orcas9 -NT phagemid lysates and the number of CFU recovered on non-selective agar was enumerated. The Cas9-antimicrobial effect was represented bya reduction in the number of CFU recovered after cas9-shiA phagemid lysates treatment, when compared to the use of cas9 -NT phagemid lysates. 4lLt o1 2 lL of P1(S’) cas9 -shiA lysates reduced the number of S. ﬂexneri 5a M90T CFU by 141-fold to 1000-fold, when compared to the number of CFUrecovered after treatment with P1(S’) cas9 -NT lysates ( Figures 6 (b)–(d)). There were no signiﬁcant differences in the number of S. ﬂexneri 5a M90T CFU recovered between P1(S) cas9 - shiA and P1(S) cas9 -NT lysates treatment (p > 0.05), which indicated a signiﬁcantly lower Cas9-mediated antimicrobial effect of P1(S)lysates on S. ﬂexneri 5a M90T when compared to that of P1(S’) lysates. P1(S’) cas9 -shiA lysates reduced the number of S. ﬂexneri 2a 2457O CFU by 100-fold to 850-fold when compared to the number of CFU recovered after treatment with P1 (S’) cas9 -NT lysates ( Figures 6 (b)–(d)). P1(S) cas9 -shiA treatment caused 7.5-fold to 220-fold reduction in the number of S. ﬂexneri 2a 2457O CFU when compared to the number of CFU recovered P1(S) cas9 -NT treatment, which indicated a lower Cas9-mediated antimicrobial effect of P1(S) lysates on S. ﬂexneri 2a 2457O when compared to P1(S’) lysates. Taken together, P1(S’) cas9 phagemid lysates produced a greater antimicrobial effect on both S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O, indicating that P1(S’) improved the transductionefﬁciency of cas9 phagemids into these bacteria when compared to P1(S). Discussion P1 bacteriophage has an invertible C segment, encoding two sets of tail ﬁbres namely the S andS’.31,26,30Despite the wide usage of P1 for gener- alised transduction, the host range of P1(S’) was never determined. Here, we showed evidence of P1(S’) transduction of S. ﬂexneri 2a 2457O, S. ﬂex- neri5a M90T and E. coli O3, as well as the require- ment of O-antigen for its transduction. Data presented in this study provided novel insights intoP1 tropism-switching and showed promise as a tool for improving the transduction efﬁciency of a cas9 antimicrobial phagemid into S. ﬂexneri . The effect(s) of O-antigen on P1 transduction The presence of O-antigen was reported to reduce the transduction efﬁciency ofbacteriophage P1. Previous studies have shown that restoring the O-antigen expression of E. coli K12 MG1655 inhibited P1 phage transduction, 6 while galmutants of E. coli O157:H7 lacking O- antigen were permissive for P1 infection.27The inhi- bitory effect of O-antigen against P1 transductioncontradicts with an earlier study by Godard et al.,19 which showed that P1kc virinfected smooth strains ofS. ﬂexneri F6S.', 'page_label: 167841 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The effect(s) of O-antigen on P1 transduction The presence of O-antigen was reported to reduce the transduction efﬁciency ofbacteriophage P1. Previous studies have shown that restoring the O-antigen expression of E. coli K12 MG1655 inhibited P1 phage transduction, 6 while galmutants of E. coli O157:H7 lacking O- antigen were permissive for P1 infection.27The inhi- bitory effect of O-antigen against P1 transductioncontradicts with an earlier study by Godard et al.,19 which showed that P1kc virinfected smooth strains ofS. ﬂexneri F6S. Notably, P1 lysates used in all these studies were prepared lytically on E. coli K12, which would have given a majority of P1 pro- geny with the S tail ﬁbre, since only P1(S) phage can infect E. coli K12 and that inversion of the C segment is infrequent. Our results indicated that both the P1(S’) and P1(S) can transduce phagemid DNA into wildtype S. ﬂexneri 5a M90T and S. ﬂex- neri 2a 2457O with intact O-antigen, albeit with varying efﬁciencies. Therefore, the inhibitory effects of O-antigen against P1 transduction might bedependent on the bacterial strains used, discrepan-cies between transduction of phagemid DNA versus P1 genomic DNA, as well as the tail ﬁbre of P1 phage, in which the host range(s) of P1 alternativeS’ tail ﬁbre was not determined from previous stud- ies. 31,30Since a non-replicative P1 phage is desir- able for delivering the DNA sequence-speciﬁc Figure 6. A higher antimicrobial effect of P1(S’) lysates on S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O. a) Schematic diagram showing the cas9 genetic construct of P1 J72114 phagemid, with protospacer sequence targeting chromosomal gene, shiA ofS. ﬂexneri (in light blue). Following transduction and circularisation of the phagemid, the presence of a CRISPR guide RNA (crRNA) with spacer sequence complementary to the chromosomal shiA gene of S. ﬂexneri would cause dsDNA cleavage of the chromosome via Cas9 endonuclease, leading to cell death (shown as dashed lines). cas9 -NT phagemid with a randomly-generated spacer sequence can be stably maintained in transduced cells (host cell survival indicated by ﬁlled lines). Quantiﬁcation of the number of S. ﬂexneri 5a M90T and S. ﬂexneri 2a 2457O CFU recovered after treatments with b)4lL,c)8lL and d)12lL of P1(S) or P1(S’) non-targeting cas9 -NT (light brown bars) and chromosomal-targeting cas9 -shiA (light blue bars) lysates. Approximately 107CFU was used for each experiment. n.s. represents no signiﬁcant difference in the number of S. ﬂexneri 5a M90T CFU recovered between P1(S) cas9 -NT and P1(S) cas9 -shiA lysates treatments. p < 0.0005 are shown as ***, for comparisons of the antimicrobial effects between P1(S) and P1(S’) cas9 lysates on S. ﬂexneri 5a M90T at all three volumes, as well as that on S. ﬂexneri 2a 2457O at 8 lL of lysates. p < 0.005 is shown as **, for comparisons of the antimicrobial effects between 4 lL of P1(S) and 4 lL of P1(S’) cas9 lysates on S. ﬂexneri 2a 2457O.', 'page_label: 167841 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  p < 0.0005 are shown as ***, for comparisons of the antimicrobial effects between P1(S) and P1(S’) cas9 lysates on S. ﬂexneri 5a M90T at all three volumes, as well as that on S. ﬂexneri 2a 2457O at 8 lL of lysates. p < 0.005 is shown as **, for comparisons of the antimicrobial effects between 4 lL of P1(S) and 4 lL of P1(S’) cas9 lysates on S. ﬂexneri 2a 2457O. p < 0.05 is shown as *, for comparisons of the antimicrobial effects between 12 lL of P1(S) and 12 lL of P1(S’) cas9 lysates on S. ﬂexneri 2a 2457O.3Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 13', 'page_label: 167842 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Cas9 antimicrobial system, it remained unknown if E. coli O3 and S. ﬂexneri could support the produc- tion of P1(S’) progeny. Therefore, the infection of E. coli O3 and S. ﬂexneri strains with a replicative P1 mutant which only expresses the S’ tail ﬁbre, such as the mutant P1-15::IS 2phage described by Iida et al., (1984), would be important to verifyif these bacterial strains could support phage prop- agation after transduction. The requirement of O-antigen for P1(S’)- mediated transduction Our results indicated that a lack of O-antigen inhibited P1(S’) transduction of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli O3. Indeed, rough LPS extracted from the O-antigendeﬁcient mutant strains lacked P1(S’) neutralising effect, highlighting the possible role of these O- antigens in mediating P1(S’) adsorption. O-antigenchains serve as receptors for the tailspike proteinsof multiple bacteriophages, including bacteriophage P22 that infects Samonella spp. 58 and bacteriophage Sf6 which infects S. ﬂexneri ser- otypes X and Y.17Most of these tail spike proteins promote the degradation and/or the modiﬁcation of O-antigen chains thus orientating the bacterio-phage to a secondary receptor for irreversible bind- ing. Such mechanism, however, is expected to be different to that of the P1(S’) tail ﬁbre, due to a lackof signiﬁcant amino acid sequence homology to anyknown tail spike protein targeting S. ﬂexneri O- antigen, nor to any known enzymatic domain involved in degrading glycosidic bond. Comparative analysis of P1 S’ tail ﬁbre amino acid sequence (NCBI, NC_005856.1, 32225.0.33782) indicated a high sequence similarity ( /C2599 %) to the protein product of multiple phage tail ﬁbre genes that were annotated in E. coli ,S. enterica and E. albertii genomes (data not shown). One such homologue of P1 S’gene is the S’tail ﬁbre gene of Mu bacteriophage (NCBI, NC_000929.1, 34067.0.35053). The Mu S’gene shows a high sequence similarity ( /C2590 %) to the 3 0-end (639th to 2919th DNA sequence) of P1 S’ gene, which encodes the predicted host-range-determining- region (HRDR) of P1 S’ tail ﬁbre (NCBI,NC_005856.1, 32225.0.33782) (data not shown). The Mu S’ tail ﬁbre, however, was shown to recognise a glucose- b1,6-glucose structure of Erwinia chrysanthemi (B37) LPS,57and this struc- ture is not found on the LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T, E. coli O3 and E. coli O111:B4 .Hence, P1 S’ might have a different receptor compared to Mu S’, which is consistent with the fact that neither Mu(S) nor Mu(S’) formed plaques on 12 different serotypes of S. ﬂexneri .32 Interestingly, Jakhetia and Verma32reported a Mu-like phage, SfMu which formed plaques on S. ﬂexneri serotypes 3b and Y. The differences between Mu and SfMu host range is attributed to 18 amino acid changes in the tail ﬁbre of SfMu whencompared to Mu S’ tail ﬁbre.', 'page_label: 167842 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Comparisons between the C-terminus of P1 S’ tail ﬁbre, the tail ﬁbre of SfMu (NCBI, KP010268, ORF49) and the Mu S’ tail ﬁbre indicated that P1 S’ tail ﬁbre contains additionalamino acid changes at multiple sites, which could potentially contribute towards differences in the host range of these phages (data not shown). Acomparative analysis of the amino acid sequence between P1 S’ tail ﬁbre and its homologs could determine how changes in speciﬁc amino acid resi-dues lead to variations in the phages host range. The LPS of S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T, E. coli O3 and E. coli O111:B4 with O- antigen (termed O-PS hereafter) produced a highneutralising activity against P1(S’) transduction, which indicated common feature(s) shared among these O-PS might be the receptor for P1(S’). Thepredicted similarity between S. ﬂexneri ,E. coli O3 andE. coli O111 O-PS is a b-D-Glc pNac-1, 4- a-D- Glc residue which links the O-antigens to theterminal glucose of LPS outer core oligosaccharide ( Figure 5 (e)). This residue, however, is also present in the LPS of E. coli O26: B6 29,42which yielded a weaker P1(S’) neutralising effect compared to the LPS of wildtype E. coli O3 andS. ﬂexneri . Therefore, it remained inconclusive if ab-D-Glc pNac-1, 4- a-D-Glc residue serves as the receptor for P1(S’) since LPS can adopt different conformations based on their respective side chains, which could have reduced the accessibilityof this residue on the LPS of E. coli O26:B6.53,18 Furthermore, we could not dismiss the role of other cell surface antigens in mediating P1(S’) transduc-tion and/or adsorption. Genomic analysis indicatedthe presence of a K2ab polysaccharide capsule operon and a H2 ﬂagellar operon on the strain of E. coli O3 used for this study. Hence, the effects, if any, of these surface antigens on P1(S’) adsorp- tion and/or transduction remain to be elucidated. While the bacterial LPS were only assessed for inhi-bitory effect against P1 transduction, a direct mea- surement of the binding afﬁnity of P1 tail ﬁbres to the different bacterial LPS would provide insightsinto discrepancies in the inhibitory effect of different LPS against P1(S) and P1(S’) transduction. Fur- thermore, quantiﬁcation of P1(S’) transduction efﬁ-ciency on various Gram-negativeEnterobactericae , coupled with structural studies of the P1(S’) tail ﬁbre as well as the use of synthetic glycans mimicking different regions of LPSoligosaccharides, will help in identifying its receptor and the mechanisms behind P1(S’) adsorption. The effects of O-antigen and its modiﬁcations on P1(S)-mediated transduction Earlier studies suggested that P1(S) tail ﬁbre recognises the terminal glucose molecule on the outer LPS core without discriminating the orientation of a glycosidic bond. 16,49,56Indeed, glu- cose molecule is present in the LPS of E. coli and S. ﬂexneri strains tested in our study, which is con-Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 14', 'page_label: 167843 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  sistent with the fact that P1(S) infected most of these bacteria. A previous study of S. ﬂexneri 2a O-antigen indicated that loosing speciﬁc acetylation modiﬁcation(s) affected the sensitivity of S. ﬂexneri 2a 2457O towards bacteriophage Sf6c infection.60 Similarly, our results indicated that P1(S) transduc-tion was signiﬁcantly affected by a Dgtrand a Drfc mutation in S. ﬂexneri 5a M90T, as well as a DoacB mutation in S. ﬂexneri 2a 2457O. While P1(S) failed to transduce phagemid DNA into wildtype E. coli O3, its isogenic DwaaL andDrfemutants were per- missive for P1(S) transduction, and the rough LPS of its DwaaL mutant produced a high inhibitory effect against P1(S). Taken together, accessibilityof the P1(S) receptor might be affected by muta- tions in the genes of O-antigen assembly, biosyn- thesis and its modiﬁcation. Further structuralstudies of the P1(S) tail ﬁbre might provide insights into the molecular mechanisms behind P1(S) adsorption to its receptor. The potential use of P1(S’) for delivering an antimicrobial cas9 system into pathogenic strains of Enterobacteriaceae Besides S. ﬂexneri and E. coli O3, the high inhibitory effect of E. coli O111:B4 LPS against P1 (S’) transduction indicated that this E. coli strain might be susceptible to P1(S’)-mediated transduction. E. coli O111:B4, however, was shown to be resistant to P1 phage infection, dueto the presence of a P1 or a P1-like resident phage. 10Subsequently, cells cured from the lyso- genic phage were susceptible to P1 infection usinglysates prepared from a thermoinducible P1 lyso- gen. Since lysates prepared from an induced lyso- gen would yield both P1(S) and P1(S’) phages,the factor(s) required for P1(S’) adsorption might be present in E. coli O111:B4, which is consistent with the high P1(S’) neutralising effect of E. coli O111:B4 LPS. Serotyping analysis showed that E. coli O111 is associated with strains of enterotox- igenic E. coli (ETEC) 7while E. coli O3 is associated with clinical isolates of enteroaggregative (EAEC)E. coli .33,35,14Hence, P1(S’) could be a valuable vector for the delivery of the cas9 antimicrobial pha- gemid into clinical isolates of E. coli O111 and E. coli O3. Furthermore, P1(S’) transduction could also be expanded to target other serotypes of S. ﬂexneri , since most serotypes share a common O- antigen backbone, and that P1(S’) transduction might not be signiﬁcantly affected by changes in the O-antigen modiﬁcations of S. ﬂexneri . The appli- cation of P1(S’) to target other serotypes of S. ﬂex- neri is relevant, as the development of a broadly protective vaccine against S. ﬂexneri is hampered by its serotypic and genomic variations.44,2Notably, our preliminary transduction assay indicated that P1 (S’) infection of S. ﬂexneri serotype 2b ATCC/C210 12022 yielded a higher number of phagemid trans- ductant when compared to that recovered after P1 (S) infection (data not shown) thus highlighting thepotential of P1(S’) for delivering the DNA sequence-speciﬁc Cas9 antimicrobial system into different S. ﬂexneri serotypes.', 'page_label: 167843 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  The appli- cation of P1(S’) to target other serotypes of S. ﬂex- neri is relevant, as the development of a broadly protective vaccine against S. ﬂexneri is hampered by its serotypic and genomic variations.44,2Notably, our preliminary transduction assay indicated that P1 (S’) infection of S. ﬂexneri serotype 2b ATCC/C210 12022 yielded a higher number of phagemid trans- ductant when compared to that recovered after P1 (S) infection (data not shown) thus highlighting thepotential of P1(S’) for delivering the DNA sequence-speciﬁc Cas9 antimicrobial system into different S. ﬂexneri serotypes. It is noteworthy that P1 transduction and the propagation of bacterialstrains were performed in nutrient-rich LB medium for this study thus ﬂuctuations in the LPS polysac- charide composition were expected to be minimal,in the absence of mutations in LPS biosynthesis and assembly gene(s). While a wildtype LPS struc- ture is critical for the survival of S. ﬂexneri and pathogenic E. coli in the intestinal milieu, 28,62,45,40,54 variations in nutrients availability, particularly in the external environment, could potentially alter the bacterial LPS polysaccharide composition and con-sequently affecting the rate of P1 adsorption. Therefore, the use of chemically-deﬁned medium with speciﬁc carbon source(s) would provide impor-tant insights into the phage-bacteria interactions under nutrients-limiting conditions. Understanding such interactions could demonstrate the potentialapplication of P1 phagemid system to target S. ﬂex- neriand other susceptible pathogenic bacteria out- side of the human and/or animal host(s). Materials and Methods Bacterial strains, plasmids and media used in this study All the bacterial strains used for this study were listed in Supplementary Table 1 while plasmids used were listed in Supplementary Table 2 .A l l PCR reactions for molecular cloning was carried out using Q5/C210High-Fidelity DNA Polymerase (NEB, M0491) while colony PCR was carried outusing Taq Polymerase (NEB, M0270L). All plasmids were assembled using Gibson Assembly (NEB, E2621). Bacteria of this study wereroutinely cultured in Luria–Bertani (LB) broth andagar. Drfcmutants of S. ﬂexneri exhibited very slow growth after lambda Red recombineering. Hence mutants were cultured on Tryptic SoyBroth (TSB) before subsequent culturing in LB media. SM buffer (50 mM Tris-HCl, 8 mM magnesium sulfate, 100 mM sodium chloride, pH7.5), was routinely used as the buffer for manipulating phage lysates. DNA sequences used in this study were listed in Supplementary Tables 4 . Genomic sequencing of E. coli O3 and agglutination assay Preliminary results suggested that P1(S’) can transduce phagemid DNA into an unknown E. coli strain at a high efﬁciency. Since the identity of this bacterial strain was not determined, genomic sequencing was performed by MicrobesNG(Birmingham, United Kingdom, https:// microbesng.com/ ), with llumina next-generation sequencing data at a coverage of 30X. Cell masswas provided to MicrobesNG .Genomic sequencing data is available at NCBI, under theY.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 15', 'page_label: 167844 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  accession number JALKIH000000000, Bioproject: PRJNA825326, Biosample: SAMN27512004. Artemis (Oxford, United Kingdom) was used to browse the assembly and annotations.8The O- antigen gene cluster of E. coli is commonly located between the galF and gnd genes sequences,59 which was annotated in the assembly data. BLASTn (Nucleotide BLAST, NIH) was used to identify homologous DNA sequence, and the results suggested a 99 % (14043/14049) identitymatch to the E. coli O3 O-antigen gene cluster iden- tiﬁed in Ren et al., study (2008). DNA sequence of the O3 O-antigen gene cluster was deposited onto GenBank, with the accession number ON210125.The expression of E. coli O3 O-antigen was con- ﬁrmed by agglutination assay using anti- E. coli O3 antiserum (Oxford Biosystems, 84999). Overnightculture of E. coli O3 was boiled for 1 h on heat block at 100 /C176C. The boiled culture was left to stand at room temperature for 1 h to allow cell debris to sed-iment. Equal volume of antiserum was added to the boiled cultures in glass tubes, and the reactions were incubated overnight at 52 /C176C in a water bath. Glass tubes were shaken vigorously before readingthe reaction. E. coli O3 culture gave a granular tex- ture indicating positive reaction while a homoge- nous milky turbidity was observed for S. ﬂexneri 2a 2457O, S. ﬂexneri 5a M90T and E. coli E24377A cultures, indicating negative reactions. Transformation of E. coli and S. ﬂexneri Transformation of E. coli and S. ﬂexneri was carried out using standard electroporation protocol. Overnight culture of cells was sub-cultured by 1/100 in fresh LB media at 37 /C176Cw i t h shaking until it reached an OD 600of approximately 0.35. Cells were centrifuged at 3000 /C2g for 15 min at 4/C176C. Cell pellet was washed with ice-cold 10 % glycerol (in distilled water), and the washing step was repeated for another 2 times. The ﬁnal cell pellet was concentrated by 200-fold in distilledwater with 10 % glycerol (e.g., 250 lLo f1 0% glycerol for cell pelleted from a 50 mL culture). One lL of DNA (approximately 10–20 ng) was electroporated into 40 lL of cells using Gene Pulser/MicroPulser Electroporation Cuvettes, 0.1 cm gap (Biorad, 1652083), and immediatelytransferred into 1 mL of ice-cold SOC. Cells were incubated at 30 /C176C for temperature sensitive plasmids, pKD46 and pCP20, or 37 /C176C for other DNA substrate, with shaking for 1 h. Cells wereplatted onto LB agar plates with the appropriate antibiotics, followed by overnight incubation at 30/C176Co r3 7 /C176C. P1 phagemid lysate preparation Preparation of phagemid lysates was carried out based on a protocol from Kittleson et al. 39study, with slight modiﬁcations. Overnight cultures of E. coli P1 lysogenic EMG16 cells transformed with phagemidand/or with plasmid, were sub-cultured in fresh PLM media (LB supplemented with 5 mM of Ca2+ and 10 mM Mg2+)a t3 7 /C176C, with shaking until it reached an OD 600between 0.15 to 0.2, which took approximately 2 hours. L-arabinose was added to the culture, giving a ﬁnal concentration of 0.02 % or approximately 1.3 mM.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:51 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'87'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993782'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'373ms'), (b'x-request-id', b'369a703b22dddd076630a4699b6a135d'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249620f3d5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:51 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'87'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993782'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'373ms'), (b'x-request-id', b'369a703b22dddd076630a4699b6a135d'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249620f3d5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 167844 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  P1 phagemid lysate preparation Preparation of phagemid lysates was carried out based on a protocol from Kittleson et al. 39study, with slight modiﬁcations. Overnight cultures of E. coli P1 lysogenic EMG16 cells transformed with phagemidand/or with plasmid, were sub-cultured in fresh PLM media (LB supplemented with 5 mM of Ca2+ and 10 mM Mg2+)a t3 7 /C176C, with shaking until it reached an OD 600between 0.15 to 0.2, which took approximately 2 hours. L-arabinose was added to the culture, giving a ﬁnal concentration of 0.02 % or approximately 1.3 mM. Induction was carriedout at 37 /C176C, with shaking, until clear culture was obtained, which took approximately 2 h to 3 h. Lysates were transferred into 1.5 mL polypropenetubes and chloroform was then added to the culture,giving a ﬁnal concentration of 2.5 % (e.g., 25 lL for 1 mL of lysate). Cultures were incubated at 37 /C176C, with shaking for at least 15 min. Samples were cen-trifuged at maximum speed for 3 min. The super- natant was carefully collected and ﬁltered using a 0.22 lm Syringe ﬁlter (Millipore). Lysates can be stored at 4 /C176C. For Polythene Glycol 6000 (PEG 6000) precipitation, sodium chloride and chloroform wereadded to lysates, giving a ﬁnal concentration of 0.3 M and 2.5 % respectively. Cultures were incubated at 37 /C176C with shaking for at least 15 min. Lysates were centrifuged at 5000 rpm for30 min and the supernatant was carefully collected. PEG 6000 was added to a ﬁnal concentration of 4 % (e.g., 4 g in 100 mL oflysate). Lysates were shaken until all the PEG 6000 dissolved, and incubated overnight at 4 /C176C. The PEG precipitate was spun down at 5000 rpmfor 45 min, supernatant was then removed, and SM buffer was added to give 100-fold concentration of phage precipitate (e.g., 500 lL SM buffer added to the pellet obtained from a50 mL lysates). Pellet was dissolved by leaving it in SM buffer for 2 h, followed by gentle vortexing and resuspension using Gilson Pipette. Thelysates were then passed through a 0.22 lm Syringe ﬁlter (Millipore) and stored at 4 /C176C. Quantiﬁcation of phagemid transductant Quantiﬁcation of phagemid transductant was carried out based on a protocol derived fromKittleson et al. 39study with slight modiﬁcations. Overnight cultures of the host cells were sub- cultured in fresh LB media at 37 /C176C, with shaking until it reached an OD 600of 0.5. Cells were concen- trated by 10-fold, and used at 100 lL for transduc- tion, which would give a range of 1 /C2108to 5/C2108cells per reaction. Ten microlitres of crude P1 phage lysates, or 10 lL of 10-fold diluted PEG 6000 puriﬁed lysate, was diluted in 90 lLo fS Mb u f - fer supplemented with 5 mM Ca2+. A hundred microlitres of host cells were then added to 100 lL of phage lysates, incubated at 37 /C176C for 30 min with shaking to maximise phage adsorption. Eight hun-dred microlitres of SOC with 10 mM citrate was added to the mixture, and cells were incubated for 1 h at 37 /C176C with shaking for the expression of chloramphenicol-resistant gene.', 'page_label: 167844 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Ten microlitres of crude P1 phage lysates, or 10 lL of 10-fold diluted PEG 6000 puriﬁed lysate, was diluted in 90 lLo fS Mb u f - fer supplemented with 5 mM Ca2+. A hundred microlitres of host cells were then added to 100 lL of phage lysates, incubated at 37 /C176C for 30 min with shaking to maximise phage adsorption. Eight hun-dred microlitres of SOC with 10 mM citrate was added to the mixture, and cells were incubated for 1 h at 37 /C176C with shaking for the expression of chloramphenicol-resistant gene. Serial dilution of the ﬁnal mixture was made, and 5 lLo fe a c hd i l u -Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 16', 'page_label: 167845 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  tion were platted onto LB agar supplemented with 25lg/mL of chloramphenicol. Plates were incu- bated at 37 /C176C overnight and the number of chlo- ramphenicol resistant CFU were enumerated after14 h to 16 h of incubation. Genetic modiﬁcation of E. coli P1 lysogen and S. ﬂexneri All genetic modiﬁcations of E. coli P1 lysogen, S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T were carried out via lambda Red recombineering technique as described in a previous study by Datsenko and Wanner.12The gene knockout tem- plate was designed to have homology arms of 60 bps complementary to both upstream and down- stream DNA sequence of the target site (refer toSupplementary Table 3 for primer sequence). The marker gene for selection of mutation was a consti- tutively expressed kanamycin resistance gene cas- sette, npt, ﬂanked by Flippase recognition sites (FRT). The dsDNA DNA template was generated via PCR, and agarose gel puriﬁed. The target bacte- ria were transformed with pKD46 plasmid. Station-ary phase culture of the transformed cells was outgrown in fresh LB at a dilution factor of 100, at 30/C176C with shaking, until it reached an OD 600of 0.35. 0.1% or 0.065 M of L-arabinose was added to the culture to induce the expression of lambda Red genes ( exo,bet,gam), and cultured for a maxi- mum of 30 min at 30 /C176C, with shaking. Cells were chilled on ice for 40 min and made electrocompetent, using standard protocols for preparing electrocom- petent cells. Approximately 100 ng of dsDNA sub-strate was electroporated into the cells, followed by outgrowth in SOC at 30 /C176C with shaking for 2 h. Cells were plated onto LB agar supplemented with 50 lg/ mL of kanamycin, incubated at 37 /C176C for at least 16 h. Colony PCR was carried out on colonies recovered, using primer pairs that anneal to the junction of inte-gration to verify correct insertion of the nptcassette. Positive clones were re-streaked onto new LB sup- plemented with 50 lg/mL of kanamycin, incubated at 37 /C176C for at least 16 h, and this re-streaking pro- cess was repeated for at least 3 generations. To remove the nptcassette, cells were transformed with pCP20, platted onto LB agar supplemented with100lg/mL of ampicillin and incubated at 30 /C176C for at least 16 h. Colonies were picked for PCR, to verify the loss of nptcassette. PCR products of the correct sizes were gel-puriﬁed and sent for sequencing. For curing of the pCP20 plasmid, colonies having the correct PCR band size and DNA sequence werediluted in 5 mL of LB broth without ampicillin andincubated overnight at 43 /C176C with shaking. Serial dilution of the culture was made, and 25 lLo ft h e dilution range of 10 3to 106was platted onto LB agar, followed by incubation at 30 /C176C for at least 16 h. Colonies were then platted onto LB agar with or with- out 50 lg/mL kanamycin or 100 lg/mL ampicillin.Agar plates with or without kanamycin were incu- bated at 37 /C176C, while agar plates with ampicillin were incubated at 30 /C176C, for at least 16 h. Colonies were selected for its sensitivity towards both kanamycinand ampicillin. Extraction and visualisation of LPS LPS of S. ﬂexneri andE. coli O3 were extracted using the hot aqueous-phenol method as described in Davis et al.13studies with slight modiﬁ- cations.', 'page_label: 167845 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Extraction and visualisation of LPS LPS of S. ﬂexneri andE. coli O3 were extracted using the hot aqueous-phenol method as described in Davis et al.13studies with slight modiﬁ- cations. Brieﬂy, bacterial suspension of approxi- mately 108CFU/mL were centrifuged at 5000 rpm for 10 min. Bacterial cell pellet was washed twicein PBS (0.15 M, pH = 7.2). For every 108CFU, cell pellet was gently resuspended in 200 lLo f 1/C2Laemmli Buffer (Biorad) supplemented with b- mercaptoethanol at a ﬁnal concentration of 2.5%.Cells were boiled in a water bath or heat block (at 100/C176C) for 15 min and cooled to room temperature. For every 200 lL of sample, 5 lL of 1 unit/ lL DNase I (Sigma, AMPD1) and 5 lL of 10 mg/mL RNase A (Thermo, EN0531) were added and incubated at 37/C176C for 1 h. For every 200 lL of sample, 5 lLo f 20 mg/mL Proteinase K (Thermo, AM2546) was added, and incubated at 59 /C176C for 3 h. To extract the LPS, an equal volume of hot (65 /C176Ct o7 0 /C176C) UltraPure TMBuffer-Saturated Phenol (Thermo, 15513039) was added to the sample, followed by vigorous mixing and incubation at 65 /C176C in a water bath or heat block for 20 min. Samples were vor-texed occasionally during the 20 min incubation time. Samples were transferred to 1.5 mL polypro- pene tubes, cooled to room temperature. For every400 mL of samples, 1 mL diethyl ether (Sigma, 91238) was added, and the mixture was vortexed for 5 s to 10 s. The samples were then centrifugedat 20600 /C2g for 10 min at 4 /C176C. The bottom aque- ous layer was carefully removed. Phenol extraction was repeated, by using the same heating and cen-trifugation steps mentioned previously on the aque- ous layer. The samples were combined, and LPS was precipitated by adding 10 volumes of 90%ethanol and 3 M sodium acetate to a ﬁnal concen-tration of 0.5 M, and left overnight at /C020/C176C, as described by Rezania et al.,. 52The mixture was centrifuged at 4000 rpm for 15 min at 4 /C176C, super- natant was removed, and the pellet is resuspended in distilled water to give a 10 /C2concentrated LPS solution (e.g., 0.5 mL of distilled water for pellet cen-trifuged from 5 mL of sample). To remove contami- nating phenol, samples were extensively dialysed in distilled water at 4 /C176C, using Slide-A-Lyzer TMDialysis Cassettes, 10 K MWCO (Thermo, 66380). The ﬁnalLPS samples can be stored at 4 /C176C for short term usage or at /C020/C176C for long term storage. LPS samples were visualised using Pierce TM Silver Staining kit (Thermo, 24612), following manufacturer’s protocol. For concentrated LPS, samples were diluted by 10-fold in distilled water.Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 17', 'page_label: 167846 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Laemmli buffer was added to samples. Ten microlitres of samples were loaded onto 12% Mini- PROTEAN/C210TGXTMPrecast Protein Gels (Biorad). Silver staining was carried out at roomtemperature for 1 h and gels were developed for 2 to 3 min, until bands were visible. Gel images were scanned using Canon CanoScan LiDE 120and images were processed, by adjusting the brightness and contrast of images using ZEN 2 (blue edition) | V1.0 en. Images of 600 dpiresolution were used for this manuscript.Coomassie Blue staining of LPS samples were routinely carried out to determine protein contamination of samples, if any. Estimation of LPS concentration and LPS inhibition assay LPS of E. coli K12 (Invivogen, tlrl-eklps), E. coli O26:B6 (Sigma, L8274), E. coli O55:B5 (Sigma, L2880), E. coli O111:B4 (Sigma, L2630) and E. coli O127:B8 (Sigma, L3129) were reconstituted in distilled water to give a ﬁnal concentration of5 mg/mL following manufacturers’ protocol. Samples were stored at 4 /C176C for short term usage or at /C020/C176C for long term storage. Silver staining of the LPS was carried out to verify presence of O- antigen repeating units (data not shown). The concentration of LPS samples were estimated using a colorimetric assay forestimating carbohydrate concentration, developed by Masuko et al. 46A range of diluted LPS samples were made in distilled water to give a ﬁnal volume of30lL, followed by the addition of 150 lL of concen- trated sulfuric acid (Sigma, 84727) to each sample. The mixture was vortexed and incubation for 15 minat 90/C176C. Thirty microlitres of 5% phenol solution (in distilled water) was added to the mixture, followed by vigorous mixing, and incubation at room temper-ature with shaking for 5 min. The absorption was then measured at a wavelength of A 490. The con- centrations of smooth LPS with O-antigen repeatingunits were estimated, using standard curves gener-ated from E. coli O26:B6, E. coli O55:B5, E. coli O111:B4 and E. coli O127:B8 LPS. The concentra- tions of rough or semi rough LPS (without O-antigenor with only one O-antigen unit) were estimated using standard curves generated from E. coli K12 LPS. For the LPS inhibition assay, a modiﬁed protocol based on Sandulache et al., 56study was used. Brieﬂy, 0.5 lg, 5 lg, 50 lga n d 500 lg of LPS were resuspended in distilled water, giving a ﬁnal volume of 100 lL. PEG 6000-puriﬁed P1 phage lysates were diluted in SM buffer giving 1 /C2103in 50 lL. Phage lysates and LPS samples were mixed with Ca2+(to a ﬁnal concentration of 5 mM), incu- bated at 37 /C176C for 1 h with shaking. Indicatorcells, E. coli NCM3722 for P1(S) or S. ﬂexneri 2a 2457O for P1(S’) were prepared as described forquantiﬁcation of phagemid transductant . Fifty microlitres of indicator cells were added tothe LPS and phage lysate mixture, followed by incubation at 37 /C176C for 15 min with shaking. Cells were centrifuged for 4000 /C2g for 1 min at room temperature to remove unabsorbed phage. Cell pellet was then washed twice with 500lL of SOC with 10 mM citrate supplement. The cell pellet was resuspended in a ﬁnal vol-ume of 1 mL SOC with 10 mM citrate supple- ment. Cells were incubated at 37 /C176Cf o r1h with shaking.', 'page_label: 167846 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Fifty microlitres of indicator cells were added tothe LPS and phage lysate mixture, followed by incubation at 37 /C176C for 15 min with shaking. Cells were centrifuged for 4000 /C2g for 1 min at room temperature to remove unabsorbed phage. Cell pellet was then washed twice with 500lL of SOC with 10 mM citrate supplement. The cell pellet was resuspended in a ﬁnal vol-ume of 1 mL SOC with 10 mM citrate supple- ment. Cells were incubated at 37 /C176Cf o r1h with shaking. Cells were platted onto LB agarsupplemented with 25 lg/mL of chloramphenicol, incubated at at 37 /C176C for at least 16 h before enumeration of chloramphenicol resistant CFU.The transduction efﬁciency was calculated by comparing the number of phagemid transduc- tants to that recovered after treatment withnon-LPS treated phagemid lysates. Quantiﬁcation of the antimicrobial effect of P1 cas9-shiA phagemid The CRISPR-Cas9 construct of this study was derived from pCas9 plasmid (Addgene plasmid #42876) and assembled onto P1 phagemid via Gibson assembly. The system providesconstitutive expression of cas9 , encoding the endoculease, as well as a trans -activating CRISPR RNA, tracRNA and a crRNA guide. The crRNA coding sequence contains two BsaI sites, allowing the ligation of a protospacer sequence complementary to chromosomal DNA sequence of the targeted bacterial species.Protospacer sequence complementary to the shiA gene sequence of S. ﬂexneri was cloned into the crRNA coding sequence via BsaI digestion of the phagemid and the ligation of an annealed, oligonucleotide pair ( shiA-targeting oligonucleotide pair, forward: 5 0AAACGCATGAC- TTCTCCGGCTCTCG-30; reverse: 50- AAAACC- GAGAGCCGGAGAAGTCATG-30), as described in by Jiang et al.34Phage lysates were prepared as described previously, using the Dpac E. coli EMG16 P1 lysogen transformed with either the targeting cas9 -shiA phagemid or non-targeting cas9 -NT phagemid. Since a shiA-targeting pha- gemid is inherently unstable in S. ﬂexneri ,t h e titre of phagemid transducing particle was ﬁrst assessed using qPCR ( Supplementary Fig- ures 2a, 2b ) as well as estimated on E. coli NCM3722 for P1(S) lysates and E. coli O3 for P1(S’) lysates ( Supplementary Figures 8a, 8b ). The indicator cells were prepared as describedpreviously for the quantiﬁcation of phagemid transductants . Overnight cultures of S. ﬂexneri were sub-cultured in fresh PLM media (LB sup-Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 18', 'page_label: 167847 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  plemented with 5 mM of Ca2+and 10 mM Mg2+) at 37 /C176C with shaking. For S. ﬂexneri 2a 2457O, cells were cultured until an OD 600of approxi- mately 0.3, which was diluted to give an OD 600 of approximately 0.1. S. ﬂexneri 5a M90T exhib- ited a slower growth rate, hence cells were cul- tured for 2 h, giving an OD 600of approximately 0.12 and cultures were used directly for experi- ments. One hundred microlitres of the cultures gave approximately 1 /C2107cells per reaction. The shiA-targeting and non-targeting phagemids lysates were diluted in SM buffer supplemented with 5 mM Ca2to a ﬁnal volume of 100 lL. One hundred microlitres of host cells were thenadded to 100 lL of phage lysates, incubated at 37 /C176C for 30 min with shaking for phage adsorption. Eight hundred microlitres of SOCwas added to the mixture, and cells were incu- bated for 1 h at 37 /C176C with shaking. Serial dilu- tion of the ﬁnal mixture was made, and 5 lLo f each dilution were platted onto LB agar. Plates were incubated at 37 /C176C overnight and the num- ber of CFU were enumerated after 14 h to 16 hof incubation. Differences in the number of S. ﬂexneri CFU recovered between the cas9-shiA and cas9 -NT phagemid lysates treatment repre- sented the Cas9-mediated antimicrobial effect. Statistical analysis Data of this study was generated from at least 2 phage lysate samples, 3 biological repeats (hostcells tested) and 2 technical repeats, unless asstated otherwise. Data of LPS inhibition assays were generated from 1 LPS sample, 1 phage lysate sample, 3 biological repeats (host cellstested) and 2 technical repeats. Statistical analysis of this study was carried out using Real Statistics Resource Pack on Microsoft excel (Microsoft,Redmond, WA, USA). srepresented S.E.M while n represented the number of both biological and technical repeats. Welch’s ANOVA was used todetermine the p-values, unless for comparisons involving 0 values, such as the total inhibition of transduction efﬁciency for LPS inhibition assay. Inthe latter case, a two-tailed, Student’s t-test(assuming unequal variance) was used. Graphpad Prism 6 was used to generate the graphs of this study. p-value of < 0.05 is considered statistically signiﬁcant, for this study. The exact p-values of comparisons made are available in Supplementary Materials 2 . Data availability All relevant data are within the manuscript and supplementary information ﬁles. Data will be shared upon request to the corresponding author.CRediT authorship contribution statement Yang W. Huan: Methodology, Investigation, Formal analysis, Writing – original draft. Jidapha Fa-arun: Methodology. Baojun Wang: Conceptualization, Methodology, Supervision, Writing – review & editing, Project administration,Funding acquisition. DECLARATION OF COMPETING INTEREST The authors declare that they have no known competing ﬁnancial interests or personal relationshipsthat could have appeared to inﬂuence the work reported in this paper. Acknowledgements This work was supported by the Bill and Melinda Gates Foundation under the Grand Challenges Explorations grant (OPP1139488), UK Research and Innovation Future Leaders Fellowship (MR/S018875/1) and the Fundamental Research Funds for the Central Universities (226-2022- 00178, 226-2022-00214). The authors wish tothank Professor Garry Blakely and Dr Tahar Ait-Ali for their advice and support for this study. Appendix A. Supplementary Data Supplementary Data are available online, including 8 Supplementary Figures and 4Supplementary Tables and the p-values of statistical analysis in this study.', 'page_label: 167847 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  DECLARATION OF COMPETING INTEREST The authors declare that they have no known competing ﬁnancial interests or personal relationshipsthat could have appeared to inﬂuence the work reported in this paper. Acknowledgements This work was supported by the Bill and Melinda Gates Foundation under the Grand Challenges Explorations grant (OPP1139488), UK Research and Innovation Future Leaders Fellowship (MR/S018875/1) and the Fundamental Research Funds for the Central Universities (226-2022- 00178, 226-2022-00214). The authors wish tothank Professor Garry Blakely and Dr Tahar Ait-Ali for their advice and support for this study. Appendix A. Supplementary Data Supplementary Data are available online, including 8 Supplementary Figures and 4Supplementary Tables and the p-values of statistical analysis in this study. Supplementary data to this article can be found online athttps://doi.org/10.1016/j.jmb.2022.167829 . Received 25 July 2022; Accepted 12 September 2022; Available online 15 September 2022 Keywords : P1 bacteriophage; S’ tail ﬁbre; Shigella ﬂexneri ; O-antigen; cas9 antimicrobial phagemid References 1.Alexander, D.C., Valvano, M.A., (1994). Role of the rfe gene in the biosynthesis of the Escherichia coli O7- speciﬁc lipopolysaccharide and other O-speciﬁc polysaccharides containing N-acetylglucosamine . J. Bacteriol. 176, 7079– 7084 .Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 19', 'page_label: 167848 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  2.Bengtsson, R.J., Simpkin, A.J., Pulford, C.V., Low, R., Rasko, D.A., Rigden, D.J., Hall, N., Barry, E.M., et al., (2022). Pathogenomic analyses of Shigella isolates inform factors limiting shigellosis prevention and control across LMICs . Nat. Microbiol. 7, 251–261 . 3.Bertani, G., (1951). Studies on lysogenesis I . J. Bacteriol. 62, 293–300 . 4.Bertani, G., (1958). Lysogeny . Adv. Virus Res. 5, 151–193 . 5.Bikard, D., Jiang, W., Samai, P., Hochschild, A., Zhang, F., Marrafﬁni, L.A., (2013). Programmable repression and activation of bacterial gene expression using an engineered CRISPR-Cas system . Nucleic Acids Res. 41, 7429–7437 . 6.Browning, D.F., Wells, T.J., Franc ¸a, F.L.S., Morris, F.C., Sevastsyanovich, Y.R., Bryant, J.A., Johnson, M.D., Lund, P.A., et al., (2013). Laboratory adapted Escherichia coli K- 12 becomes a pathogen of Caenorhabditis elegans upon restoration of O antigen biosynthesis . Mol. Microbiol. 87, 939–950 . 7.Campos, L.C., Whittam, T.S., Gomes, T.A.T., Andrade, J. R.C., Trabulsi, L.R., (1994). Escherichia coli serogroup O111 includes several clones of diarrheagenic strains with different virulence properties . Infect. Immun. 62, 3282– 3288 . 8.Carver, T., Harris, S.R., Berriman, M., Parkhill, J., McQuillan, J.A., (2012). Artemis: An integrated platform for visualization and analysis of high-throughput sequence-based experimental data . Bioinformatics 28, 464–469 . 9.Citorik, R.J., Mimee, M., Lu, T.K., (2014). Sequence- speciﬁc antimicrobials using efﬁciently delivered RNA- guided nucleases . Nat. Biotechnol. 32, 1141–1145 . 10.Coleman, W.G., Goebel, P.J., Leive, L., (1977). Genetic analysis of Escherichia coli O111:B4, a strain of medical and biochemical interest . J. Bacteriol. 130, 656–660 . 11.Cui, L., Bikard, D., (2016). Consequences of Cas9 cleavage in the chromosome of Escherichia coli. Nucleic Acids Res. 44, 4243–4251 . 12.Datsenko, K.A., Wanner, B.L., (2000). One-step inactivation of chromosomal genes in Escherichia coli K- 12 using PCR products . Proc. Natl. Acad. Sci. U. S. A. 97, 6640–6645 . 13. Davis, M.R., Goldberg, J.B., (2012). Puriﬁcation and visualization of lipopolysaccharide from gram-negative bacteria by hot aqueous-phenol extraction . J. Vis. Exp. . https://doi.org/10.3791/3916 . 14.de Lira, D.R.P., Cavalcanti, A.M.F., Pinheiro, S.R.S., Orsi, H., dos Santos, L.F., Hernandes, R.T., (2021). Identiﬁcation of a hybrid atypical enteropathogenic and enteroaggregative Escherichia coli (aEPEC/EAEC) clone of serotype O3:H2 associated with a diarrheal outbreak in Brazil . Brazilian J. Microbiol. 52, 2075–2079 .', 'page_label: 167848 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  97, 6640–6645 . 13. Davis, M.R., Goldberg, J.B., (2012). Puriﬁcation and visualization of lipopolysaccharide from gram-negative bacteria by hot aqueous-phenol extraction . J. Vis. Exp. . https://doi.org/10.3791/3916 . 14.de Lira, D.R.P., Cavalcanti, A.M.F., Pinheiro, S.R.S., Orsi, H., dos Santos, L.F., Hernandes, R.T., (2021). Identiﬁcation of a hybrid atypical enteropathogenic and enteroaggregative Escherichia coli (aEPEC/EAEC) clone of serotype O3:H2 associated with a diarrheal outbreak in Brazil . Brazilian J. Microbiol. 52, 2075–2079 . 15.Foster, R.A., Carlin, N.I.A., Majcher, M., Tabor, H., Ng, L. K., Widmalm, G., (2011). Structural elucidation of the O- antigen of the Shigella ﬂexneri provisional serotype 88– 893: Structural and serological similarities with S. ﬂexneri provisional serotype Y394 (1c) . Carbohydr. Res. 346, 872– 876. 16.Franklin, N.C., (1969). Mutation in galU gene of E. coli blocks phage P1 infection . Virology 38, 189–191 .17.Freiberg, A., Morona, R., Van den Bosch, L., Jung, C., Behlke, J., Carlin, N., Seckler, R., Baxa, U., (2003). The tailspike protein of Shigella phage Sf6: A structural homolog of Salmonella phage P22 tailspike protein without sequence similarity in the b-helix domain . J. Biol. Chem. 278, 1542–1548 . 18.Gao, Y., Lee, J., Widmalm, G., Im, W., (2020). Modeling and Simulation of Bacterial Outer Membranes with Lipopolysaccharides and Enterobacterial Common Antigen . J. Phys. Chem. B 124, 5948–5956 . 19.Godard, C., Beumer-Jochmans, M.P., Beumer, J., (1971). Shigella ﬂexneri F6S with extended patterns of sensitivity to the T phages and to colicins after infection by a Lisbonne phage. 1. Modiﬁed biological properties of the bacterial surface . Ann. Inst. Pasteur 121, 475–489 . 20.Goldberg, R.B., Bender, R.A., Streicher, S.L., (1974). Direct selection for P1 sensitive mutants of enteric bacteria . J. Bacteriol. 118, 810–814 . 21.Goldhar, J., Eylan, E., Goldschmied-Reouven, A., (1973). Lysogeny and lysosensitivity in Shigella dysenteriae group of bacteria - I. Isolation of bacteriophages and receptor substances . Med. Microbiol. Immunol. 159, 63–72 . [22]. Gomaa, A.A., Klumpe, H.E., Luo, M.L., Selle, K., Barrangou, R., Beisel, C.L., (2014). Programmable removal of bacterial strains by use of genome- targeting CRISPR-cas systems . mBio 5, e00928–e1013 . 23.Guidolin, A., Zingg, J.M., Arber, W., (1989). Organization of the bacteriophage P1 tail-ﬁbre operon . Gene 76, 239–243 . 24.Guidolin, A., Zingg, J.M., Lehnherr, H., Arber, W., (1989). Bacteriophage P1 tail-ﬁbre and daroperons are expressed from homologous phage-speciﬁc late promoter sequences . J. Mol. Biol. 208, 615–622 .'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 167844 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  P1 phagemid lysate preparation Preparation of phagemid lysates was carried out based on a protocol from Kittleson et al. 39study, with slight modiﬁcations. Overnight cultures of E. coli P1 lysogenic EMG16 cells transformed with phagemidand/or with plasmid, were sub-cultured in fresh PLM media (LB supplemented with 5 mM of Ca2+ and 10 mM Mg2+)a t3 7 /C176C, with shaking until it reached an OD 600between 0.15 to 0.2, which took approximately 2 hours. L-arabinose was added to the culture, giving a ﬁnal concentration of 0.02 % or approximately 1.3 mM. Induction was carriedout at 37 /C176C, with shaking, until clear culture was obtained, which took approximately 2 h to 3 h. Lysates were transferred into 1.5 mL polypropenetubes and chloroform was then added to the culture,giving a ﬁnal concentration of 2.5 % (e.g., 25 lL for 1 mL of lysate). Cultures were incubated at 37 /C176C, with shaking for at least 15 min. Samples were cen-trifuged at maximum speed for 3 min. The super- natant was carefully collected and ﬁltered using a 0.22 lm Syringe ﬁlter (Millipore). Lysates can be stored at 4 /C176C. For Polythene Glycol 6000 (PEG 6000) precipitation, sodium chloride and chloroform wereadded to lysates, giving a ﬁnal concentration of 0.3 M and 2.5 % respectively. Cultures were incubated at 37 /C176C with shaking for at least 15 min. Lysates were centrifuged at 5000 rpm for30 min and the supernatant was carefully collected. PEG 6000 was added to a ﬁnal concentration of 4 % (e.g., 4 g in 100 mL oflysate). Lysates were shaken until all the PEG 6000 dissolved, and incubated overnight at 4 /C176C. The PEG precipitate was spun down at 5000 rpmfor 45 min, supernatant was then removed, and SM buffer was added to give 100-fold concentration of phage precipitate (e.g., 500 lL SM buffer added to the pellet obtained from a50 mL lysates). Pellet was dissolved by leaving it in SM buffer for 2 h, followed by gentle vortexing and resuspension using Gilson Pipette. Thelysates were then passed through a 0.22 lm Syringe ﬁlter (Millipore) and stored at 4 /C176C. Quantiﬁcation of phagemid transductant Quantiﬁcation of phagemid transductant was carried out based on a protocol derived fromKittleson et al. 39study with slight modiﬁcations. Overnight cultures of the host cells were sub- cultured in fresh LB media at 37 /C176C, with shaking until it reached an OD 600of 0.5. Cells were concen- trated by 10-fold, and used at 100 lL for transduc- tion, which would give a range of 1 /C2108to 5/C2108cells per reaction. Ten microlitres of crude P1 phage lysates, or 10 lL of 10-fold diluted PEG 6000 puriﬁed lysate, was diluted in 90 lLo fS Mb u f - fer supplemented with 5 mM Ca2+. A hundred microlitres of host cells were then added to 100 lL of phage lysates, incubated at 37 /C176C for 30 min with shaking to maximise phage adsorption. Eight hun-dred microlitres of SOC with 10 mM citrate was added to the mixture, and cells were incubated for 1 h at 37 /C176C with shaking for the expression of chloramphenicol-resistant gene.', 'page_label: 167844 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Ten microlitres of crude P1 phage lysates, or 10 lL of 10-fold diluted PEG 6000 puriﬁed lysate, was diluted in 90 lLo fS Mb u f - fer supplemented with 5 mM Ca2+. A hundred microlitres of host cells were then added to 100 lL of phage lysates, incubated at 37 /C176C for 30 min with shaking to maximise phage adsorption. Eight hun-dred microlitres of SOC with 10 mM citrate was added to the mixture, and cells were incubated for 1 h at 37 /C176C with shaking for the expression of chloramphenicol-resistant gene. Serial dilution of the ﬁnal mixture was made, and 5 lLo fe a c hd i l u -Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 16', 'page_label: 167845 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  tion were platted onto LB agar supplemented with 25lg/mL of chloramphenicol. Plates were incu- bated at 37 /C176C overnight and the number of chlo- ramphenicol resistant CFU were enumerated after14 h to 16 h of incubation. Genetic modiﬁcation of E. coli P1 lysogen and S. ﬂexneri All genetic modiﬁcations of E. coli P1 lysogen, S. ﬂexneri 2a 2457O and S. ﬂexneri 5a M90T were carried out via lambda Red recombineering technique as described in a previous study by Datsenko and Wanner.12The gene knockout tem- plate was designed to have homology arms of 60 bps complementary to both upstream and down- stream DNA sequence of the target site (refer toSupplementary Table 3 for primer sequence). The marker gene for selection of mutation was a consti- tutively expressed kanamycin resistance gene cas- sette, npt, ﬂanked by Flippase recognition sites (FRT). The dsDNA DNA template was generated via PCR, and agarose gel puriﬁed. The target bacte- ria were transformed with pKD46 plasmid. Station-ary phase culture of the transformed cells was outgrown in fresh LB at a dilution factor of 100, at 30/C176C with shaking, until it reached an OD 600of 0.35. 0.1% or 0.065 M of L-arabinose was added to the culture to induce the expression of lambda Red genes ( exo,bet,gam), and cultured for a maxi- mum of 30 min at 30 /C176C, with shaking. Cells were chilled on ice for 40 min and made electrocompetent, using standard protocols for preparing electrocom- petent cells. Approximately 100 ng of dsDNA sub-strate was electroporated into the cells, followed by outgrowth in SOC at 30 /C176C with shaking for 2 h. Cells were plated onto LB agar supplemented with 50 lg/ mL of kanamycin, incubated at 37 /C176C for at least 16 h. Colony PCR was carried out on colonies recovered, using primer pairs that anneal to the junction of inte-gration to verify correct insertion of the nptcassette. Positive clones were re-streaked onto new LB sup- plemented with 50 lg/mL of kanamycin, incubated at 37 /C176C for at least 16 h, and this re-streaking pro- cess was repeated for at least 3 generations. To remove the nptcassette, cells were transformed with pCP20, platted onto LB agar supplemented with100lg/mL of ampicillin and incubated at 30 /C176C for at least 16 h. Colonies were picked for PCR, to verify the loss of nptcassette. PCR products of the correct sizes were gel-puriﬁed and sent for sequencing. For curing of the pCP20 plasmid, colonies having the correct PCR band size and DNA sequence werediluted in 5 mL of LB broth without ampicillin andincubated overnight at 43 /C176C with shaking. Serial dilution of the culture was made, and 25 lLo ft h e dilution range of 10 3to 106was platted onto LB agar, followed by incubation at 30 /C176C for at least 16 h. Colonies were then platted onto LB agar with or with- out 50 lg/mL kanamycin or 100 lg/mL ampicillin.Agar plates with or without kanamycin were incu- bated at 37 /C176C, while agar plates with ampicillin were incubated at 30 /C176C, for at least 16 h. Colonies were selected for its sensitivity towards both kanamycinand ampicillin. Extraction and visualisation of LPS LPS of S. ﬂexneri andE. coli O3 were extracted using the hot aqueous-phenol method as described in Davis et al.13studies with slight modiﬁ- cations.', 'page_label: 167845 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Extraction and visualisation of LPS LPS of S. ﬂexneri andE. coli O3 were extracted using the hot aqueous-phenol method as described in Davis et al.13studies with slight modiﬁ- cations. Brieﬂy, bacterial suspension of approxi- mately 108CFU/mL were centrifuged at 5000 rpm for 10 min. Bacterial cell pellet was washed twicein PBS (0.15 M, pH = 7.2). For every 108CFU, cell pellet was gently resuspended in 200 lLo f 1/C2Laemmli Buffer (Biorad) supplemented with b- mercaptoethanol at a ﬁnal concentration of 2.5%.Cells were boiled in a water bath or heat block (at 100/C176C) for 15 min and cooled to room temperature. For every 200 lL of sample, 5 lL of 1 unit/ lL DNase I (Sigma, AMPD1) and 5 lL of 10 mg/mL RNase A (Thermo, EN0531) were added and incubated at 37/C176C for 1 h. For every 200 lL of sample, 5 lLo f 20 mg/mL Proteinase K (Thermo, AM2546) was added, and incubated at 59 /C176C for 3 h. To extract the LPS, an equal volume of hot (65 /C176Ct o7 0 /C176C) UltraPure TMBuffer-Saturated Phenol (Thermo, 15513039) was added to the sample, followed by vigorous mixing and incubation at 65 /C176C in a water bath or heat block for 20 min. Samples were vor-texed occasionally during the 20 min incubation time. Samples were transferred to 1.5 mL polypro- pene tubes, cooled to room temperature. For every400 mL of samples, 1 mL diethyl ether (Sigma, 91238) was added, and the mixture was vortexed for 5 s to 10 s. The samples were then centrifugedat 20600 /C2g for 10 min at 4 /C176C. The bottom aque- ous layer was carefully removed. Phenol extraction was repeated, by using the same heating and cen-trifugation steps mentioned previously on the aque- ous layer. The samples were combined, and LPS was precipitated by adding 10 volumes of 90%ethanol and 3 M sodium acetate to a ﬁnal concen-tration of 0.5 M, and left overnight at /C020/C176C, as described by Rezania et al.,. 52The mixture was centrifuged at 4000 rpm for 15 min at 4 /C176C, super- natant was removed, and the pellet is resuspended in distilled water to give a 10 /C2concentrated LPS solution (e.g., 0.5 mL of distilled water for pellet cen-trifuged from 5 mL of sample). To remove contami- nating phenol, samples were extensively dialysed in distilled water at 4 /C176C, using Slide-A-Lyzer TMDialysis Cassettes, 10 K MWCO (Thermo, 66380). The ﬁnalLPS samples can be stored at 4 /C176C for short term usage or at /C020/C176C for long term storage. LPS samples were visualised using Pierce TM Silver Staining kit (Thermo, 24612), following manufacturer’s protocol. For concentrated LPS, samples were diluted by 10-fold in distilled water.Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 17', 'page_label: 167846 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Laemmli buffer was added to samples. Ten microlitres of samples were loaded onto 12% Mini- PROTEAN/C210TGXTMPrecast Protein Gels (Biorad). Silver staining was carried out at roomtemperature for 1 h and gels were developed for 2 to 3 min, until bands were visible. Gel images were scanned using Canon CanoScan LiDE 120and images were processed, by adjusting the brightness and contrast of images using ZEN 2 (blue edition) | V1.0 en. Images of 600 dpiresolution were used for this manuscript.Coomassie Blue staining of LPS samples were routinely carried out to determine protein contamination of samples, if any. Estimation of LPS concentration and LPS inhibition assay LPS of E. coli K12 (Invivogen, tlrl-eklps), E. coli O26:B6 (Sigma, L8274), E. coli O55:B5 (Sigma, L2880), E. coli O111:B4 (Sigma, L2630) and E. coli O127:B8 (Sigma, L3129) were reconstituted in distilled water to give a ﬁnal concentration of5 mg/mL following manufacturers’ protocol. Samples were stored at 4 /C176C for short term usage or at /C020/C176C for long term storage. Silver staining of the LPS was carried out to verify presence of O- antigen repeating units (data not shown). The concentration of LPS samples were estimated using a colorimetric assay forestimating carbohydrate concentration, developed by Masuko et al. 46A range of diluted LPS samples were made in distilled water to give a ﬁnal volume of30lL, followed by the addition of 150 lL of concen- trated sulfuric acid (Sigma, 84727) to each sample. The mixture was vortexed and incubation for 15 minat 90/C176C. Thirty microlitres of 5% phenol solution (in distilled water) was added to the mixture, followed by vigorous mixing, and incubation at room temper-ature with shaking for 5 min. The absorption was then measured at a wavelength of A 490. The con- centrations of smooth LPS with O-antigen repeatingunits were estimated, using standard curves gener-ated from E. coli O26:B6, E. coli O55:B5, E. coli O111:B4 and E. coli O127:B8 LPS. The concentra- tions of rough or semi rough LPS (without O-antigenor with only one O-antigen unit) were estimated using standard curves generated from E. coli K12 LPS. For the LPS inhibition assay, a modiﬁed protocol based on Sandulache et al., 56study was used. Brieﬂy, 0.5 lg, 5 lg, 50 lga n d 500 lg of LPS were resuspended in distilled water, giving a ﬁnal volume of 100 lL. PEG 6000-puriﬁed P1 phage lysates were diluted in SM buffer giving 1 /C2103in 50 lL. Phage lysates and LPS samples were mixed with Ca2+(to a ﬁnal concentration of 5 mM), incu- bated at 37 /C176C for 1 h with shaking. Indicatorcells, E. coli NCM3722 for P1(S) or S. ﬂexneri 2a 2457O for P1(S’) were prepared as described forquantiﬁcation of phagemid transductant . Fifty microlitres of indicator cells were added tothe LPS and phage lysate mixture, followed by incubation at 37 /C176C for 15 min with shaking. Cells were centrifuged for 4000 /C2g for 1 min at room temperature to remove unabsorbed phage. Cell pellet was then washed twice with 500lL of SOC with 10 mM citrate supplement. The cell pellet was resuspended in a ﬁnal vol-ume of 1 mL SOC with 10 mM citrate supple- ment. Cells were incubated at 37 /C176Cf o r1h with shaking.', 'page_label: 167846 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Fifty microlitres of indicator cells were added tothe LPS and phage lysate mixture, followed by incubation at 37 /C176C for 15 min with shaking. Cells were centrifuged for 4000 /C2g for 1 min at room temperature to remove unabsorbed phage. Cell pellet was then washed twice with 500lL of SOC with 10 mM citrate supplement. The cell pellet was resuspended in a ﬁnal vol-ume of 1 mL SOC with 10 mM citrate supple- ment. Cells were incubated at 37 /C176Cf o r1h with shaking. Cells were platted onto LB agarsupplemented with 25 lg/mL of chloramphenicol, incubated at at 37 /C176C for at least 16 h before enumeration of chloramphenicol resistant CFU.The transduction efﬁciency was calculated by comparing the number of phagemid transduc- tants to that recovered after treatment withnon-LPS treated phagemid lysates. Quantiﬁcation of the antimicrobial effect of P1 cas9-shiA phagemid The CRISPR-Cas9 construct of this study was derived from pCas9 plasmid (Addgene plasmid #42876) and assembled onto P1 phagemid via Gibson assembly. The system providesconstitutive expression of cas9 , encoding the endoculease, as well as a trans -activating CRISPR RNA, tracRNA and a crRNA guide. The crRNA coding sequence contains two BsaI sites, allowing the ligation of a protospacer sequence complementary to chromosomal DNA sequence of the targeted bacterial species.Protospacer sequence complementary to the shiA gene sequence of S. ﬂexneri was cloned into the crRNA coding sequence via BsaI digestion of the phagemid and the ligation of an annealed, oligonucleotide pair ( shiA-targeting oligonucleotide pair, forward: 5 0AAACGCATGAC- TTCTCCGGCTCTCG-30; reverse: 50- AAAACC- GAGAGCCGGAGAAGTCATG-30), as described in by Jiang et al.34Phage lysates were prepared as described previously, using the Dpac E. coli EMG16 P1 lysogen transformed with either the targeting cas9 -shiA phagemid or non-targeting cas9 -NT phagemid. Since a shiA-targeting pha- gemid is inherently unstable in S. ﬂexneri ,t h e titre of phagemid transducing particle was ﬁrst assessed using qPCR ( Supplementary Fig- ures 2a, 2b ) as well as estimated on E. coli NCM3722 for P1(S) lysates and E. coli O3 for P1(S’) lysates ( Supplementary Figures 8a, 8b ). The indicator cells were prepared as describedpreviously for the quantiﬁcation of phagemid transductants . Overnight cultures of S. ﬂexneri were sub-cultured in fresh PLM media (LB sup-Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 18', 'page_label: 167847 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  plemented with 5 mM of Ca2+and 10 mM Mg2+) at 37 /C176C with shaking. For S. ﬂexneri 2a 2457O, cells were cultured until an OD 600of approxi- mately 0.3, which was diluted to give an OD 600 of approximately 0.1. S. ﬂexneri 5a M90T exhib- ited a slower growth rate, hence cells were cul- tured for 2 h, giving an OD 600of approximately 0.12 and cultures were used directly for experi- ments. One hundred microlitres of the cultures gave approximately 1 /C2107cells per reaction. The shiA-targeting and non-targeting phagemids lysates were diluted in SM buffer supplemented with 5 mM Ca2to a ﬁnal volume of 100 lL. One hundred microlitres of host cells were thenadded to 100 lL of phage lysates, incubated at 37 /C176C for 30 min with shaking for phage adsorption. Eight hundred microlitres of SOCwas added to the mixture, and cells were incu- bated for 1 h at 37 /C176C with shaking. Serial dilu- tion of the ﬁnal mixture was made, and 5 lLo f each dilution were platted onto LB agar. Plates were incubated at 37 /C176C overnight and the num- ber of CFU were enumerated after 14 h to 16 hof incubation. Differences in the number of S. ﬂexneri CFU recovered between the cas9-shiA and cas9 -NT phagemid lysates treatment repre- sented the Cas9-mediated antimicrobial effect. Statistical analysis Data of this study was generated from at least 2 phage lysate samples, 3 biological repeats (hostcells tested) and 2 technical repeats, unless asstated otherwise. Data of LPS inhibition assays were generated from 1 LPS sample, 1 phage lysate sample, 3 biological repeats (host cellstested) and 2 technical repeats. Statistical analysis of this study was carried out using Real Statistics Resource Pack on Microsoft excel (Microsoft,Redmond, WA, USA). srepresented S.E.M while n represented the number of both biological and technical repeats. Welch’s ANOVA was used todetermine the p-values, unless for comparisons involving 0 values, such as the total inhibition of transduction efﬁciency for LPS inhibition assay. Inthe latter case, a two-tailed, Student’s t-test(assuming unequal variance) was used. Graphpad Prism 6 was used to generate the graphs of this study. p-value of < 0.05 is considered statistically signiﬁcant, for this study. The exact p-values of comparisons made are available in Supplementary Materials 2 . Data availability All relevant data are within the manuscript and supplementary information ﬁles. Data will be shared upon request to the corresponding author.CRediT authorship contribution statement Yang W. Huan: Methodology, Investigation, Formal analysis, Writing – original draft. Jidapha Fa-arun: Methodology. Baojun Wang: Conceptualization, Methodology, Supervision, Writing – review & editing, Project administration,Funding acquisition. DECLARATION OF COMPETING INTEREST The authors declare that they have no known competing ﬁnancial interests or personal relationshipsthat could have appeared to inﬂuence the work reported in this paper. Acknowledgements This work was supported by the Bill and Melinda Gates Foundation under the Grand Challenges Explorations grant (OPP1139488), UK Research and Innovation Future Leaders Fellowship (MR/S018875/1) and the Fundamental Research Funds for the Central Universities (226-2022- 00178, 226-2022-00214). The authors wish tothank Professor Garry Blakely and Dr Tahar Ait-Ali for their advice and support for this study. Appendix A. Supplementary Data Supplementary Data are available online, including 8 Supplementary Figures and 4Supplementary Tables and the p-values of statistical analysis in this study.', 'page_label: 167847 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  DECLARATION OF COMPETING INTEREST The authors declare that they have no known competing ﬁnancial interests or personal relationshipsthat could have appeared to inﬂuence the work reported in this paper. Acknowledgements This work was supported by the Bill and Melinda Gates Foundation under the Grand Challenges Explorations grant (OPP1139488), UK Research and Innovation Future Leaders Fellowship (MR/S018875/1) and the Fundamental Research Funds for the Central Universities (226-2022- 00178, 226-2022-00214). The authors wish tothank Professor Garry Blakely and Dr Tahar Ait-Ali for their advice and support for this study. Appendix A. Supplementary Data Supplementary Data are available online, including 8 Supplementary Figures and 4Supplementary Tables and the p-values of statistical analysis in this study. Supplementary data to this article can be found online athttps://doi.org/10.1016/j.jmb.2022.167829 . Received 25 July 2022; Accepted 12 September 2022; Available online 15 September 2022 Keywords : P1 bacteriophage; S’ tail ﬁbre; Shigella ﬂexneri ; O-antigen; cas9 antimicrobial phagemid References 1.Alexander, D.C., Valvano, M.A., (1994). Role of the rfe gene in the biosynthesis of the Escherichia coli O7- speciﬁc lipopolysaccharide and other O-speciﬁc polysaccharides containing N-acetylglucosamine . J. Bacteriol. 176, 7079– 7084 .Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 19', 'page_label: 167848 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  2.Bengtsson, R.J., Simpkin, A.J., Pulford, C.V., Low, R., Rasko, D.A., Rigden, D.J., Hall, N., Barry, E.M., et al., (2022). Pathogenomic analyses of Shigella isolates inform factors limiting shigellosis prevention and control across LMICs . Nat. Microbiol. 7, 251–261 . 3.Bertani, G., (1951). Studies on lysogenesis I . J. Bacteriol. 62, 293–300 . 4.Bertani, G., (1958). Lysogeny . Adv. Virus Res. 5, 151–193 . 5.Bikard, D., Jiang, W., Samai, P., Hochschild, A., Zhang, F., Marrafﬁni, L.A., (2013). Programmable repression and activation of bacterial gene expression using an engineered CRISPR-Cas system . Nucleic Acids Res. 41, 7429–7437 . 6.Browning, D.F., Wells, T.J., Franc ¸a, F.L.S., Morris, F.C., Sevastsyanovich, Y.R., Bryant, J.A., Johnson, M.D., Lund, P.A., et al., (2013). Laboratory adapted Escherichia coli K- 12 becomes a pathogen of Caenorhabditis elegans upon restoration of O antigen biosynthesis . Mol. Microbiol. 87, 939–950 . 7.Campos, L.C., Whittam, T.S., Gomes, T.A.T., Andrade, J. R.C., Trabulsi, L.R., (1994). Escherichia coli serogroup O111 includes several clones of diarrheagenic strains with different virulence properties . Infect. Immun. 62, 3282– 3288 . 8.Carver, T., Harris, S.R., Berriman, M., Parkhill, J., McQuillan, J.A., (2012). Artemis: An integrated platform for visualization and analysis of high-throughput sequence-based experimental data . Bioinformatics 28, 464–469 . 9.Citorik, R.J., Mimee, M., Lu, T.K., (2014). Sequence- speciﬁc antimicrobials using efﬁciently delivered RNA- guided nucleases . Nat. Biotechnol. 32, 1141–1145 . 10.Coleman, W.G., Goebel, P.J., Leive, L., (1977). Genetic analysis of Escherichia coli O111:B4, a strain of medical and biochemical interest . J. Bacteriol. 130, 656–660 . 11.Cui, L., Bikard, D., (2016). Consequences of Cas9 cleavage in the chromosome of Escherichia coli. Nucleic Acids Res. 44, 4243–4251 . 12.Datsenko, K.A., Wanner, B.L., (2000). One-step inactivation of chromosomal genes in Escherichia coli K- 12 using PCR products . Proc. Natl. Acad. Sci. U. S. A. 97, 6640–6645 . 13. Davis, M.R., Goldberg, J.B., (2012). Puriﬁcation and visualization of lipopolysaccharide from gram-negative bacteria by hot aqueous-phenol extraction . J. Vis. Exp. . https://doi.org/10.3791/3916 . 14.de Lira, D.R.P., Cavalcanti, A.M.F., Pinheiro, S.R.S., Orsi, H., dos Santos, L.F., Hernandes, R.T., (2021). Identiﬁcation of a hybrid atypical enteropathogenic and enteroaggregative Escherichia coli (aEPEC/EAEC) clone of serotype O3:H2 associated with a diarrheal outbreak in Brazil . Brazilian J. Microbiol. 52, 2075–2079 .', 'page_label: 167848 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  97, 6640–6645 . 13. Davis, M.R., Goldberg, J.B., (2012). Puriﬁcation and visualization of lipopolysaccharide from gram-negative bacteria by hot aqueous-phenol extraction . J. Vis. Exp. . https://doi.org/10.3791/3916 . 14.de Lira, D.R.P., Cavalcanti, A.M.F., Pinheiro, S.R.S., Orsi, H., dos Santos, L.F., Hernandes, R.T., (2021). Identiﬁcation of a hybrid atypical enteropathogenic and enteroaggregative Escherichia coli (aEPEC/EAEC) clone of serotype O3:H2 associated with a diarrheal outbreak in Brazil . Brazilian J. Microbiol. 52, 2075–2079 . 15.Foster, R.A., Carlin, N.I.A., Majcher, M., Tabor, H., Ng, L. K., Widmalm, G., (2011). Structural elucidation of the O- antigen of the Shigella ﬂexneri provisional serotype 88– 893: Structural and serological similarities with S. ﬂexneri provisional serotype Y394 (1c) . Carbohydr. Res. 346, 872– 876. 16.Franklin, N.C., (1969). Mutation in galU gene of E. coli blocks phage P1 infection . Virology 38, 189–191 .17.Freiberg, A., Morona, R., Van den Bosch, L., Jung, C., Behlke, J., Carlin, N., Seckler, R., Baxa, U., (2003). The tailspike protein of Shigella phage Sf6: A structural homolog of Salmonella phage P22 tailspike protein without sequence similarity in the b-helix domain . J. Biol. Chem. 278, 1542–1548 . 18.Gao, Y., Lee, J., Widmalm, G., Im, W., (2020). Modeling and Simulation of Bacterial Outer Membranes with Lipopolysaccharides and Enterobacterial Common Antigen . J. Phys. Chem. B 124, 5948–5956 . 19.Godard, C., Beumer-Jochmans, M.P., Beumer, J., (1971). Shigella ﬂexneri F6S with extended patterns of sensitivity to the T phages and to colicins after infection by a Lisbonne phage. 1. Modiﬁed biological properties of the bacterial surface . Ann. Inst. Pasteur 121, 475–489 . 20.Goldberg, R.B., Bender, R.A., Streicher, S.L., (1974). Direct selection for P1 sensitive mutants of enteric bacteria . J. Bacteriol. 118, 810–814 . 21.Goldhar, J., Eylan, E., Goldschmied-Reouven, A., (1973). Lysogeny and lysosensitivity in Shigella dysenteriae group of bacteria - I. Isolation of bacteriophages and receptor substances . Med. Microbiol. Immunol. 159, 63–72 . [22]. Gomaa, A.A., Klumpe, H.E., Luo, M.L., Selle, K., Barrangou, R., Beisel, C.L., (2014). Programmable removal of bacterial strains by use of genome- targeting CRISPR-cas systems . mBio 5, e00928–e1013 . 23.Guidolin, A., Zingg, J.M., Arber, W., (1989). Organization of the bacteriophage P1 tail-ﬁbre operon . Gene 76, 239–243 . 24.Guidolin, A., Zingg, J.M., Lehnherr, H., Arber, W., (1989). Bacteriophage P1 tail-ﬁbre and daroperons are expressed from homologous phage-speciﬁc late promoter sequences . J. Mol. Biol. 208, 615–622 .'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:52 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'209'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993272'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'403ms'), (b'x-request-id', b'd7fb18429de7cae4ed93bccd2ca80035'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124966ec515d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:52 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'209'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993272'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'403ms'), (b'x-request-id', b'd7fb18429de7cae4ed93bccd2ca80035'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124966ec515d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 167848 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Programmable removal of bacterial strains by use of genome- targeting CRISPR-cas systems . mBio 5, e00928–e1013 . 23.Guidolin, A., Zingg, J.M., Arber, W., (1989). Organization of the bacteriophage P1 tail-ﬁbre operon . Gene 76, 239–243 . 24.Guidolin, A., Zingg, J.M., Lehnherr, H., Arber, W., (1989). Bacteriophage P1 tail-ﬁbre and daroperons are expressed from homologous phage-speciﬁc late promoter sequences . J. Mol. Biol. 208, 615–622 . 25.Heinrichs, D.E., Monteiro, M.A., Perry, M.B., Whitﬁeld, C., (1998). The assembly system for the lipopolysaccharide R2 core-type of Escherichia coli is a hybrid of those found in Escherichia coli K-12 and Salmonella enterica . Structure and function of the R2 WaaK and WaaL homologs . J. Biol. Chem. 273, 8849–8859 . 26.Hiestand-Nauer, R., Iida, S., (1983). Sequence of the site- speciﬁc recombinase gene cinand of its substrates serving in the inversion of the C segment of bacteriophage P1 . EMBO J. 2, 1733–1740 . 27.Ho, T.D., Waldor, M.K., (2007). Enterohemorrhagic Escherichia coli O157:H7 galmutants are sensitive to bacteriophage P1 and defective in intestinal colonization . Infect. Immun. 75, 1661–1666 . 28.Hong, M., Payne, S., (1997). Effect of mutations in Shigella ﬂexneri chromosomal and plasmid-encoded lipopolysaccharide genes on invasion and serum resistance . Mol. Microbiol 24, 779–791 . 29.Huang, J.X., Azad, M.A.K., Yuriev, E., Baker, M.A., Nation, R.L., Li, J., Cooper, M.A., Velkov, T., (2012). Molecular Characterization of Lipopolysaccharide Binding to Human a-1-Acid Glycoprotein . J. Lipids 2012 , 1–15 . 30.Iida, S., (1984). Bacteriophage P1 carries two related sets of genes determining its host range in the invertible C segment of its genome . Virology 134, 421–434 . 31.Iida, S., Meyer, J., Kennedy, K.E., Arber, W., (1982). A site- speciﬁc, conservative recombination system carried by bacteriophage P1. Mapping the recombinase gene cinandY.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 20', 'page_label: 167849 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  the cross-over sites cixfor the inversion of the C segment . EMBO J 1, 1445–1453 . 32.Jakhetia, R., Verma, N.K., (2015). Identiﬁcation and molecular characterisation of a novel Mu-like bacteriophage, SfMu, of Shigella ﬂexneri. PLoS One 10 33.Jenkins, C., Chart, H., Willshaw, G.A., Cheasty, T., Smith, H.R., (2006). Genotyping of enteroaggregative Escherichia coliand identiﬁcation of target genes for the detection of both typical and atypical strains . Diagn. Microbiol. Infect. Dis.55, 13–19 . 34.Jiang, W., Bikard, D., Cox, D., Zhang, F., Marrafﬁni, L.A., (2013). RNA-guided editing of bacterial genomes using CRISPR-Cas systems . Nat. Biotechnol. 31, 233–239 . 35.Kai, A., Konishi, N., Obata, H., (2010). Diarrheagenic Escherichia coli . Nippon rinsho . Japanese J. Clin. Med. 68 (Suppl 6), 203–207 . 36.Kaniuk, N.A., Vinogradov, E., Li, J., Monteiro, M.A., Whitﬁeld, C., (2004). Chromosomal and plasmid-encoded enzymes are required for assembly of the R3-type core oligosaccharide in the lipopolysaccharide of Escherichia coliO157:H7 . J. Biol. Chem. 279, 31237–31250 . 37.Kenne, L., Lindberg, B., Petersson, K., Katzenellenbogen, E., Romanowska, E., (1978). Structural studies of Shigella ﬂexneri O-antigens . Eur. J. Biochem. 91, 279–284 . 38.Kim, M.J., Moon, Y.H., Kim, H.J., Rho, S., Shin, Y.K., Song, M., Walker, R., Czerkinsky, C., et al., (2018). Cross- protective Shigella whole-cell vaccine with a truncated O- polysaccharide chain . Front. Microbiol. 9, 1–11 . 39.Kittleson, J.T., Deloache, W., Cheng, H.Y., Anderson, J.C., (2012). Scalable plasmid transfer using engineered P1- based phagemids . ACS Synth. Biol. 1, 583–589 . 40.Kuo, C., Chen, J., Chiu, H., Teng, C., Hsu, T., Lu, P., Syu, W., (2016). Mutation of the Enterohemorrhagic Escherichia coli core LPS biosynthesis enzyme RfaD confers hypersusceptibility to host intestinal innate immunity in vivo. Front. Cell. Infect. Microbiol. 12,6. 41.Lennox, E.S., (1955). Transduction of linked genetic characters of the host by bacteriophage P1 . Virology 1, 190–206 . 42.Liu, B., Furevi, A., Perepelov, A.V., Guo, X., Cao, H., Wang, Q., Reeves, P.R., Knirel, Y.A., et al., (2020). Structure and genetics of Escherichia coli O antigens . FEMS Microbiol. Rev. 44, 655–683 . 43.Łobocka, M.B., Rose, D.J., Plunkett, G., Rusin, M., Samojedny, A., Lehnherr, H., Yarmolinsky, M.B., Blattner, F.R., (2004). Genome of bacteriophage P1 . J. Bacteriol. 186, 7032–7068 . 44.Mani, S., Wierzba, T., Walker, R.I., (2016). Status of vaccine research and development for Shigella. Vaccine 34, 2887–2894 .', 'page_label: 167849 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Structure and genetics of Escherichia coli O antigens . FEMS Microbiol. Rev. 44, 655–683 . 43.Łobocka, M.B., Rose, D.J., Plunkett, G., Rusin, M., Samojedny, A., Lehnherr, H., Yarmolinsky, M.B., Blattner, F.R., (2004). Genome of bacteriophage P1 . J. Bacteriol. 186, 7032–7068 . 44.Mani, S., Wierzba, T., Walker, R.I., (2016). Status of vaccine research and development for Shigella. Vaccine 34, 2887–2894 . 45.Martinic ´, M., Hoare, A., Contreras, I., A ´lvarez, S., (2011). Contribution of the lipopolysaccharide to resistance of Shigella ﬂexneri 2a to extreme acidity . PLoS ONE 6, e25557 . 46.Masuko, T., Minami, A., Iwasaki, N., Majima, T., Nishimura, S.I., Lee, Y.C., (2005). Carbohydrate analysis by a phenol- sulfuric acid method in microplate format . Anal. Biochem. 339, 69–72 .47.Murooka, Y., Harada, T., (1979). Expansion of the host range of coliphage P1 and gene transfer from enteric bacteria to other Gram-negative bacteria . Appl. Environ. Microbiol. 38, 754–757 . 48.Nath, P., Tran, E.N.H., Morona, R., (2015). Mutational analysis of the Shigella ﬂexneri O-antigen polymerase Wzy: Identiﬁcation of Wzz-dependent Wzy mutants .J . Bacteriol. 197, 108–119 . 49.Ornellas, E.P., Stocker, B.A.D., (1974). Relation of lipopolysaccharide character to P1 sensitivity in Salmonella typhimurium. Virology 60, 491–502 . 50.Perepelov, A.V., Shekht, M.E., Liu, B., Shevelev, S.D., Ledov, V.A., Senchenkova, S.N., L’vov, V.L., Shashkov, A. S., et al., (2012). Shigella ﬂexneri O-antigens revisited: Final elucidation of the O-acetylation proﬁles and a survey of the O-antigen structure diversity . FEMS Immunol. Med. Microbiol. 66, 201–210 . 51.Ren, Y., Liu, B., Cheng, J., Liu, F., Feng, L., Wang, L., (2008). Characterization of Escherichia coli O3 and O21 O antigen gene clusters and development of serogroup- speciﬁc PCR assays . J. Microbiol. Methods 75, 329–334 . 52.Rezania, S., Amirmozaffari, N., Tabarraei, B., Jeddi- Tehrani, M., Zarei, O., Alizadeh, R., Masjedian, F., Zarnani, A.H., (2011). Extraction, puriﬁcation and characterization of lipopolysaccharide from Escherichia coli and Salmonella typhi. Avicenna J. Med. Biotechnol. 3, 3–9 . 53.Rodriguez-Loureiro, I., Latza, V.M., Fragneto, G., Schneck, E., (2018). Conformation of Single and Interacting Lipopolysaccharide Surfaces Bearing O-Side Chains . Biophys. J. 114, 1624–1635 . 54.Salamon, H., Nissim-Eliraz, E., Ardronai, O., Nissan, I., Shpigel, N., (2020). The role of O-polysaccharide chain and complement resistance of Escherichia coli in mammary virulence . Vet. Res.', 'page_label: 167849 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Extraction, puriﬁcation and characterization of lipopolysaccharide from Escherichia coli and Salmonella typhi. Avicenna J. Med. Biotechnol. 3, 3–9 . 53.Rodriguez-Loureiro, I., Latza, V.M., Fragneto, G., Schneck, E., (2018). Conformation of Single and Interacting Lipopolysaccharide Surfaces Bearing O-Side Chains . Biophys. J. 114, 1624–1635 . 54.Salamon, H., Nissim-Eliraz, E., Ardronai, O., Nissan, I., Shpigel, N., (2020). The role of O-polysaccharide chain and complement resistance of Escherichia coli in mammary virulence . Vet. Res. 51 55.Sandlin, R.C., Lampel, K.A., Keasler, S.P., Goldberg, M.B., Stolzer, A.L., Maurelli, A.T., (1995). Avirulence of rough mutants of Shigella ﬂexneri : Requirement of O antigen for correct unipolar localization of IcsA in the bacterial outer membrane . Infect. Immun. 63, 229–237 . 56.Sandulache, R., Prehm, P., Kamp, D., (1984). Cell wall receptor for bacteriophage Mu G(+) . J. Bacteriol. 160, 299– 303. 57.Sandulache, R., Prehm, P., Expert, D., Toussaint, A., Kamp, D., (1985). The cell wall receptor for bacteriophage Mu G(-) in Erwinia andEscherichia coli C. FEMS Microbiol. Lett. 28, 307–310 . 58.Steinbacher, S., Baxa, U., Miller, S., Weintraub, A., Seckler, R., Huber, R., (1996). Crystal structure of phage P22 tailspike protein complexed with Salmonella sp. O- antigen receptors . Proc. Natl. Acad. Sci. U. S. A. 93, 10584–10588 . 59.Stenutz, R., Weintraub, A., Widmalm, G., (2006). The structures of Escherichia coli O-polysaccharide antigens . FEMS Microbiol. Rev. 30, 382–403 . 60.Teh, M.Y., Furevi, A., Widmalm, G., Morona, R., (2020). Inﬂuence of Shigella ﬂexneri 2a O-antigen acetylation on its bacteriophage Sf6 receptor activity and bacterial interaction with human cells . J. Bacteriol. 202, 1–19 .Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 21', 'page_label: 167850 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  61.Toussaint, A., Lefebvre, N., Scott, J.R., Cowan, J.A., De Bruijn, F., Bukhari, A.I., (1978). Relationships between temperate phages Mu and P1 . Virology 89, 146–161 . 62.West, N.P., Sansonetti, P., Mounier, J., Exley, R.M., Parsot, C., Guadagnini, S., Pre ´vost, M.C., Prochnicka- Chalufour, A., et al., (2005). Optimization of virulence functions through glucosylation of Shigella LPS. Science 307, 1313–1317 . 63.Westwater, C., Schoﬁeld, D.A., Schmidt, M.G., Norris, J.S., Dolan, J.W., (2002). Development of a P1 phagemidsystem for the delivery of DNA into Gram-negative bacteria . Microbiology 148, 943–950 . 64.Yao, Z., Valvano, M.A., (1994). Genetic analysis of the O- speciﬁc lipopolysaccharide biosynthesis region ( rfb)o f Escherichia coli K-12 W3110: Identiﬁcation of genes that confer group 6 speciﬁcity to Shigella ﬂexneri serotypes Y and 4a . J. Bacteriol. 176, 4133–4143 . 65.Yarmolinsky, M.B., Sternberg, N., (1988) . The Bacteriophages ,vol. I Plenum Press, New York .Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 22', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f5bc0>, 'json_data': {'input': ['page_label: 167848 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Programmable removal of bacterial strains by use of genome- targeting CRISPR-cas systems . mBio 5, e00928–e1013 . 23.Guidolin, A., Zingg, J.M., Arber, W., (1989). Organization of the bacteriophage P1 tail-ﬁbre operon . Gene 76, 239–243 . 24.Guidolin, A., Zingg, J.M., Lehnherr, H., Arber, W., (1989). Bacteriophage P1 tail-ﬁbre and daroperons are expressed from homologous phage-speciﬁc late promoter sequences . J. Mol. Biol. 208, 615–622 . 25.Heinrichs, D.E., Monteiro, M.A., Perry, M.B., Whitﬁeld, C., (1998). The assembly system for the lipopolysaccharide R2 core-type of Escherichia coli is a hybrid of those found in Escherichia coli K-12 and Salmonella enterica . Structure and function of the R2 WaaK and WaaL homologs . J. Biol. Chem. 273, 8849–8859 . 26.Hiestand-Nauer, R., Iida, S., (1983). Sequence of the site- speciﬁc recombinase gene cinand of its substrates serving in the inversion of the C segment of bacteriophage P1 . EMBO J. 2, 1733–1740 . 27.Ho, T.D., Waldor, M.K., (2007). Enterohemorrhagic Escherichia coli O157:H7 galmutants are sensitive to bacteriophage P1 and defective in intestinal colonization . Infect. Immun. 75, 1661–1666 . 28.Hong, M., Payne, S., (1997). Effect of mutations in Shigella ﬂexneri chromosomal and plasmid-encoded lipopolysaccharide genes on invasion and serum resistance . Mol. Microbiol 24, 779–791 . 29.Huang, J.X., Azad, M.A.K., Yuriev, E., Baker, M.A., Nation, R.L., Li, J., Cooper, M.A., Velkov, T., (2012). Molecular Characterization of Lipopolysaccharide Binding to Human a-1-Acid Glycoprotein . J. Lipids 2012 , 1–15 . 30.Iida, S., (1984). Bacteriophage P1 carries two related sets of genes determining its host range in the invertible C segment of its genome . Virology 134, 421–434 . 31.Iida, S., Meyer, J., Kennedy, K.E., Arber, W., (1982). A site- speciﬁc, conservative recombination system carried by bacteriophage P1. Mapping the recombinase gene cinandY.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 20', 'page_label: 167849 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  the cross-over sites cixfor the inversion of the C segment . EMBO J 1, 1445–1453 . 32.Jakhetia, R., Verma, N.K., (2015). Identiﬁcation and molecular characterisation of a novel Mu-like bacteriophage, SfMu, of Shigella ﬂexneri. PLoS One 10 33.Jenkins, C., Chart, H., Willshaw, G.A., Cheasty, T., Smith, H.R., (2006). Genotyping of enteroaggregative Escherichia coliand identiﬁcation of target genes for the detection of both typical and atypical strains . Diagn. Microbiol. Infect. Dis.55, 13–19 . 34.Jiang, W., Bikard, D., Cox, D., Zhang, F., Marrafﬁni, L.A., (2013). RNA-guided editing of bacterial genomes using CRISPR-Cas systems . Nat. Biotechnol. 31, 233–239 . 35.Kai, A., Konishi, N., Obata, H., (2010). Diarrheagenic Escherichia coli . Nippon rinsho . Japanese J. Clin. Med. 68 (Suppl 6), 203–207 . 36.Kaniuk, N.A., Vinogradov, E., Li, J., Monteiro, M.A., Whitﬁeld, C., (2004). Chromosomal and plasmid-encoded enzymes are required for assembly of the R3-type core oligosaccharide in the lipopolysaccharide of Escherichia coliO157:H7 . J. Biol. Chem. 279, 31237–31250 . 37.Kenne, L., Lindberg, B., Petersson, K., Katzenellenbogen, E., Romanowska, E., (1978). Structural studies of Shigella ﬂexneri O-antigens . Eur. J. Biochem. 91, 279–284 . 38.Kim, M.J., Moon, Y.H., Kim, H.J., Rho, S., Shin, Y.K., Song, M., Walker, R., Czerkinsky, C., et al., (2018). Cross- protective Shigella whole-cell vaccine with a truncated O- polysaccharide chain . Front. Microbiol. 9, 1–11 . 39.Kittleson, J.T., Deloache, W., Cheng, H.Y., Anderson, J.C., (2012). Scalable plasmid transfer using engineered P1- based phagemids . ACS Synth. Biol. 1, 583–589 . 40.Kuo, C., Chen, J., Chiu, H., Teng, C., Hsu, T., Lu, P., Syu, W., (2016). Mutation of the Enterohemorrhagic Escherichia coli core LPS biosynthesis enzyme RfaD confers hypersusceptibility to host intestinal innate immunity in vivo. Front. Cell. Infect. Microbiol. 12,6. 41.Lennox, E.S., (1955). Transduction of linked genetic characters of the host by bacteriophage P1 . Virology 1, 190–206 . 42.Liu, B., Furevi, A., Perepelov, A.V., Guo, X., Cao, H., Wang, Q., Reeves, P.R., Knirel, Y.A., et al., (2020). Structure and genetics of Escherichia coli O antigens . FEMS Microbiol. Rev. 44, 655–683 . 43.Łobocka, M.B., Rose, D.J., Plunkett, G., Rusin, M., Samojedny, A., Lehnherr, H., Yarmolinsky, M.B., Blattner, F.R., (2004). Genome of bacteriophage P1 . J. Bacteriol. 186, 7032–7068 . 44.Mani, S., Wierzba, T., Walker, R.I., (2016). Status of vaccine research and development for Shigella. Vaccine 34, 2887–2894 .', 'page_label: 167849 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Structure and genetics of Escherichia coli O antigens . FEMS Microbiol. Rev. 44, 655–683 . 43.Łobocka, M.B., Rose, D.J., Plunkett, G., Rusin, M., Samojedny, A., Lehnherr, H., Yarmolinsky, M.B., Blattner, F.R., (2004). Genome of bacteriophage P1 . J. Bacteriol. 186, 7032–7068 . 44.Mani, S., Wierzba, T., Walker, R.I., (2016). Status of vaccine research and development for Shigella. Vaccine 34, 2887–2894 . 45.Martinic ´, M., Hoare, A., Contreras, I., A ´lvarez, S., (2011). Contribution of the lipopolysaccharide to resistance of Shigella ﬂexneri 2a to extreme acidity . PLoS ONE 6, e25557 . 46.Masuko, T., Minami, A., Iwasaki, N., Majima, T., Nishimura, S.I., Lee, Y.C., (2005). Carbohydrate analysis by a phenol- sulfuric acid method in microplate format . Anal. Biochem. 339, 69–72 .47.Murooka, Y., Harada, T., (1979). Expansion of the host range of coliphage P1 and gene transfer from enteric bacteria to other Gram-negative bacteria . Appl. Environ. Microbiol. 38, 754–757 . 48.Nath, P., Tran, E.N.H., Morona, R., (2015). Mutational analysis of the Shigella ﬂexneri O-antigen polymerase Wzy: Identiﬁcation of Wzz-dependent Wzy mutants .J . Bacteriol. 197, 108–119 . 49.Ornellas, E.P., Stocker, B.A.D., (1974). Relation of lipopolysaccharide character to P1 sensitivity in Salmonella typhimurium. Virology 60, 491–502 . 50.Perepelov, A.V., Shekht, M.E., Liu, B., Shevelev, S.D., Ledov, V.A., Senchenkova, S.N., L’vov, V.L., Shashkov, A. S., et al., (2012). Shigella ﬂexneri O-antigens revisited: Final elucidation of the O-acetylation proﬁles and a survey of the O-antigen structure diversity . FEMS Immunol. Med. Microbiol. 66, 201–210 . 51.Ren, Y., Liu, B., Cheng, J., Liu, F., Feng, L., Wang, L., (2008). Characterization of Escherichia coli O3 and O21 O antigen gene clusters and development of serogroup- speciﬁc PCR assays . J. Microbiol. Methods 75, 329–334 . 52.Rezania, S., Amirmozaffari, N., Tabarraei, B., Jeddi- Tehrani, M., Zarei, O., Alizadeh, R., Masjedian, F., Zarnani, A.H., (2011). Extraction, puriﬁcation and characterization of lipopolysaccharide from Escherichia coli and Salmonella typhi. Avicenna J. Med. Biotechnol. 3, 3–9 . 53.Rodriguez-Loureiro, I., Latza, V.M., Fragneto, G., Schneck, E., (2018). Conformation of Single and Interacting Lipopolysaccharide Surfaces Bearing O-Side Chains . Biophys. J. 114, 1624–1635 . 54.Salamon, H., Nissim-Eliraz, E., Ardronai, O., Nissan, I., Shpigel, N., (2020). The role of O-polysaccharide chain and complement resistance of Escherichia coli in mammary virulence . Vet. Res.', 'page_label: 167849 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  Extraction, puriﬁcation and characterization of lipopolysaccharide from Escherichia coli and Salmonella typhi. Avicenna J. Med. Biotechnol. 3, 3–9 . 53.Rodriguez-Loureiro, I., Latza, V.M., Fragneto, G., Schneck, E., (2018). Conformation of Single and Interacting Lipopolysaccharide Surfaces Bearing O-Side Chains . Biophys. J. 114, 1624–1635 . 54.Salamon, H., Nissim-Eliraz, E., Ardronai, O., Nissan, I., Shpigel, N., (2020). The role of O-polysaccharide chain and complement resistance of Escherichia coli in mammary virulence . Vet. Res. 51 55.Sandlin, R.C., Lampel, K.A., Keasler, S.P., Goldberg, M.B., Stolzer, A.L., Maurelli, A.T., (1995). Avirulence of rough mutants of Shigella ﬂexneri : Requirement of O antigen for correct unipolar localization of IcsA in the bacterial outer membrane . Infect. Immun. 63, 229–237 . 56.Sandulache, R., Prehm, P., Kamp, D., (1984). Cell wall receptor for bacteriophage Mu G(+) . J. Bacteriol. 160, 299– 303. 57.Sandulache, R., Prehm, P., Expert, D., Toussaint, A., Kamp, D., (1985). The cell wall receptor for bacteriophage Mu G(-) in Erwinia andEscherichia coli C. FEMS Microbiol. Lett. 28, 307–310 . 58.Steinbacher, S., Baxa, U., Miller, S., Weintraub, A., Seckler, R., Huber, R., (1996). Crystal structure of phage P22 tailspike protein complexed with Salmonella sp. O- antigen receptors . Proc. Natl. Acad. Sci. U. S. A. 93, 10584–10588 . 59.Stenutz, R., Weintraub, A., Widmalm, G., (2006). The structures of Escherichia coli O-polysaccharide antigens . FEMS Microbiol. Rev. 30, 382–403 . 60.Teh, M.Y., Furevi, A., Widmalm, G., Morona, R., (2020). Inﬂuence of Shigella ﬂexneri 2a O-antigen acetylation on its bacteriophage Sf6 receptor activity and bacterial interaction with human cells . J. Bacteriol. 202, 1–19 .Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 21', 'page_label: 167850 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283622004491-main.pdf  61.Toussaint, A., Lefebvre, N., Scott, J.R., Cowan, J.A., De Bruijn, F., Bukhari, A.I., (1978). Relationships between temperate phages Mu and P1 . Virology 89, 146–161 . 62.West, N.P., Sansonetti, P., Mounier, J., Exley, R.M., Parsot, C., Guadagnini, S., Pre ´vost, M.C., Prochnicka- Chalufour, A., et al., (2005). Optimization of virulence functions through glucosylation of Shigella LPS. Science 307, 1313–1317 . 63.Westwater, C., Schoﬁeld, D.A., Schmidt, M.G., Norris, J.S., Dolan, J.W., (2002). Development of a P1 phagemidsystem for the delivery of DNA into Gram-negative bacteria . Microbiology 148, 943–950 . 64.Yao, Z., Valvano, M.A., (1994). Genetic analysis of the O- speciﬁc lipopolysaccharide biosynthesis region ( rfb)o f Escherichia coli K-12 W3110: Identiﬁcation of genes that confer group 6 speciﬁcity to Shigella ﬂexneri serotypes Y and 4a . J. Bacteriol. 176, 4133–4143 . 65.Yarmolinsky, M.B., Sternberg, N., (1988) . The Bacteriophages ,vol. I Plenum Press, New York .Y.W. Huan, J. Fa-arun and B. Wang Journal of Molecular Biology 434 (2022) 167829 22', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:53 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'105'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'997197'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'168ms'), (b'x-request-id', b'42a225fa5513d4be7ca2a4990e72bbad'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412496c9aa75d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:53 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'105'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'997197'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'168ms'), (b'x-request-id', b'42a225fa5513d4be7ca2a4990e72bbad'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412496c9aa75d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f71a0>, 'json_data': {'input': ['page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f71a0>, 'json_data': {'input': ['page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/1-s2.0-S0022283683715214-main.pdf'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:54 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'63'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'999746'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'15ms'), (b'x-request-id', b'2073382a51f7ce61a4962bf7b1427f76'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124971bf9a5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:54 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'63'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'999746'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'15ms'), (b'x-request-id', b'2073382a51f7ce61a4962bf7b1427f76'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124971bf9a5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6160>, 'json_data': {'input': ['page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  REVIEW Phage DNA packaging Hisao Fujisawa1,a,xand Miyo Morita2 1Department of Botany, Division of Biological Science, Graduate School of Science, Kyoto University, Kyoto 606, Japan 2HSP Research Institute, INC. Kyoto Research Park, Shimogyou-ku, Kyoto 600, Japan Phage DNA packaging occurs by DNA translocation into a preformed protein shell—a prohead—with the aid of a packaging enzyme or a terminase. The packaging enzyme iscomposed of two subunits: the large subunit has ATP-binding, prohead binding, and DNAcleavage activities, and the small subunit is a DNA binding protein. DNA translocation is drivenby ATP hydrolysis. In general, phage DNA replication mechanisms lead to the accumulation ofconcatemers. Concatemers are processed to mature DNA during and depending upon DNA packaging. This review will focus on the molecular mechanism of concatemer processing and the coupling of ATP hydrolysis to DNA translocation. Introduction The DNA of double-stranded DNA bacteriophages (ds-DNA phages) is contained in a highly condensed state within an outer shell of protein. The nucleotide sequence of the DNA is divided into the followingcategories: (i) linear with a unique nucleotide sequencerepeated at the ends (linear, nonpermuted, terminallyrepetitious: l, P2, P4, T3, T7); (ii) linear with a nonunique nucleotide sequence repeated at the ends(linear, permuted, terminally repetitious: P1, P22, T4);and (iii) miscellaneous, such as linear with a protein covalently attached ( f-29) or with host sequences (Mu). Variations of these types include a single-strandedterminal repetition ( l, P22, P4) and different degrees of permutation (P1, P2, T4). However, in general, theDNA replication mechanisms of these phages lead tothe accumulation of DNA concatemers in whichmature phage DNAs are joined together in a head-to-tail manner through the terminal repetitions. The concatemer is a substrate for DNA encapsidation into preformed proheads. During DNA packaging, viralDNA is recognized and unit-length virion DNA isproduced from the concatemer by a mechanism speciﬁcfor each phage. Exceptions to this are the packaging ofreplicated circular DNA for P2 and P4, protein- terminated mature length DNA for f-29, and chro- mosomally integrated DNA for Mu phage. Themechanism by which the concatemer intermediate isprocessed to generate the mature DNA is vital inallowing the ends of the linear chromosomal DNA to bereplicated completely without a loss of the terminalbases at the 5 0end of each strand. Therefore, phage DNA packaging reactions include two important mechanisms: (i) the mechanism of end-formationfrom the concatemer, details of which are speciﬁc foreach phage, and (ii) the energetic mechanism of DNAtranslocation into the prohead driven by ATP hydro-lysis, which is common among phages. DNA packaging machinery DNA packaging is thought to proceed by a commonmechanism for most ds-DNA phages. DNA is packaged into the cavity of a preformed protein shell, called the prohead, through the connector located at the portalvertex, with the aid of a noncapsid protein called aterminase or packaging enzyme (Earnshaw & Casjens1980; Black 1989). During the packaging of DNA,mature monomers are cleaved from the concatemer.The formation of precursors for DNA translocation,ternary complexes containing the prohead, the packa- ging enzyme, and the DNA, depends upon either ATP as an allosteric effector (T3, Shibata et al. 1987) or ATP qBlackwell Science Limited Genes to Cells (1997) 2, 537–545 537*Correspondence : E-mail: hisao.fujisawa@ma1.seikyou. ne.jp aPresent address : 2-22-12-Hieidaira, Otsu 520, Japan.', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  hydrolysis ( l, Becker et al. 1977; f-29, Guo et al. 1987a). DNA translocation into the prohead is driven by ATPhydrolysis. Packaging enzymes (terminases) and in vitro packaging systems display a DNA-dependent ATPaseactivity. These results indicate that ATP plays multipleroles in the DNA packaging reaction. The connectorsshare some common features: an outer region with 12small domains and an inner region with 6 domains surrounding a channel (Bazinet & King 1986; Valpuesta & Carrascosa 1994). There is genetic evidence for theinvolvement of the connector in DNA packaging forT4 (Hsiao & Black 1977), T3 (Nakasu et al. 1983) and l (Cue & Feiss 1997). In general, the packaging enzyme iscommonly composed of two subunits: the small subunithas DNA binding activity and is involved in recognitionof its own DNA, and the large subunit is responsible for prohead binding, DNA translocation and DNA clea- vage (Catalano et al. 1995). The large subunit binds the connector of the prohead (T4, Hsiao & Black 1977; T3,Fujisawa et al. 1991; l, Y eo & Feiss 1995). The putative prohead binding domain has been identiﬁed toward theC-terminus. The large subunits are ATP-bindingproteins and are thought to carry crucial functions forDNA translocation. In the case of l, the small subunit also has ATP binding motifs (Becker & Gold 1988) and displays a DNA-stimulated ATPase (T omka & Catalano1993; Hwang et al. 1996). DNA processing Although the mechanisms of end-formation of mature DNAs are different among phages, they have common features. The packaging of concatemeric DNA mole- cules occurs by the recognition of a speciﬁc sequencecalled cosorpac, followed by speciﬁc ( l, T3, T7) or non-sequence speciﬁc cleavage (P1, P22, T1, SPP1)(initiation cleavage) and unidirectional DNA transloca-tion from the generated end. When the prohead shell isfull of DNA, a second cleavage (termination cleavage) ismade at a speciﬁc ( l, T3, T7) or nonspeciﬁc (P1, P22, T1, SPP1) site to generate the mature form of phage DNA present in virions (the headful mechanism). WithT4, there are completely non-sequence speciﬁc initia-tion and termination cleavages (T -even). Initiationcleavage is subject to different and more stringentcontrol than termination cleavage. T oo infrequentcleavage would leave unﬁlled virus particles, whereastoo frequent initiation would produce defective parti- cles. In spite of the variety of cleavage speciﬁcities, the strict dependence of the termination cleavage on theheadful mechanism is common among phages, includ-ing even l(Feiss et al. 1977) and T3 (Hashimoto &Fujisawa 1988). Although the headful mechanism is not understood, there is accumulating evidence that theportal proteins are involved in the process (T3, Fujisawaet al. 1990; P22, Casjens et al. 1982; SPP1, Travares et al. 1992; l, Cue & Feiss 1997). Deﬁned in vitro systems developed for T3, T4, P1 and P22 can package anylinear ds-DNA, indicating that the selective packagingof viral DNA into a capsid, resides in the speciﬁcity of the initiation cleavage which generates a DNA end essential for initiation of DNA packaging. There isevidence with some phages that the end generated bythe termination cleavage is preferentially used for theinitiation of the second packaging, and so forth. Sincephages can package and transduce recombinant plasmidscarrying sequences necessary for packaging ( pacsignals), pacsignals have been identiﬁed by transduction assays.', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  1990; P22, Casjens et al. 1982; SPP1, Travares et al. 1992; l, Cue & Feiss 1997). Deﬁned in vitro systems developed for T3, T4, P1 and P22 can package anylinear ds-DNA, indicating that the selective packagingof viral DNA into a capsid, resides in the speciﬁcity of the initiation cleavage which generates a DNA end essential for initiation of DNA packaging. There isevidence with some phages that the end generated bythe termination cleavage is preferentially used for theinitiation of the second packaging, and so forth. Sincephages can package and transduce recombinant plasmidscarrying sequences necessary for packaging ( pacsignals), pacsignals have been identiﬁed by transduction assays. Thepacsignals have a bipartite structure composed of separate recognition ( pacB,cosB,cosQ) and cleavage sites (pacC,cosN) and lie either within or very close to the genes encoding the packaging proteins. For l, the pac signal seems to be composed of two sets of bipartitestructures: one for initiation ( cosB,cosN) and another for termination ( cosQ,cosN) (Catalano et al. 1995). Phage lmature DNA molecules are unique with 12- base pair long, complementary single-stranded 5 0ends, called cohesive ends. In the late stage of linfection, DNA is synthesized as concatemers by a rolling circlemechanism. DNA packaging proceeds rightward fromthe end generated by staggered nicks at the cosN site. After headful packaging, introduction of staggered nicksat the second cosN completes DNA packaging. cosB and cosQ are required for initiation and termination cleavage, respectively (Catalano et al. 1995) (Fig. 1). The genomes of P1 and P22 phages have restricted circular permutation maps (P1, Ikeda & T omizawa1965; P22, T ye et al. 1974). For these phages (Fig. 1), DNA packaging initiates from the end generated byimprecise cleavage near a speciﬁc site, called pac,o n concatemeric DNA molecules. Packaging proceedsrightward until the prohead shell is ﬁlled with DNA (about 104% of the genome sequence), after which a headful cleavage is made in a nonspeciﬁc sequencemanner on the concatemer that releases the packagedDNA. Subsequent packaging events on the concate-meric DNA begin at the end generated by the headfulcleavage of the previous chromosome and proceed inthe same direction as the previous event. On average,such processive series are 3–5 chromosomes in length, yielding terminally redundant and partially circularly permuted mature DNAs (P1, Sternberg & Coulby 1987;P22, Casjens et al. 1987). The paccleavage of these phages occurs without concomitant DNA packaging in 538 Genes to Cells (1997) 2, 537–545 qBlackwell Science LimitedH Fujisawa & M Morita  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  mutants in prohead assembly and therefore requires control over the frequency of pacsite cleavage. In P1, DNA methylation of 50-GATC-30in the pacsignal has been proposed to be a regulatory mechanism for pacsite cleavage (Sternberg & Coulby 1990). Phages P1 and P22 can package host DNA from the ends of DNA fragments produced by phage-induced DNase activity(generalized transduction). T -even phages apparently have randomly permuted DNA. These phages have therefore been concluded topackage their DNA according to a pure headfulmechanism (Streisinger et al. 1967). Although features recognized by the T4 terminase for making the sequence nonspeciﬁc initiation cleavage are unknown, it is proposed that an open DNA structure generated bytranscription or recombination is a potential pacsite for recognition and attachment of T4 terminase (Mosig et al. 1981; Bhattacharyya & Rao 1993). Mu phage recognizes the pacsite located at the left end of the genome DNA and generates cleaved ends inhost DNA to the left end of the genome. DNA packaging proceeds rightward past the right end of the phage DNA and terminates in the host sequence by aheadful mechanism (George & Bukhari 1981). T3 and T7 phages are closely related and have extensive sequences of homology. Their DNAs areunique collections with the terminal repetitions of 230and 160 bp, respectively. The pacsignal of T3 and T7 iscomposed of pacC, the cleavage site for processing of concatemeric DNA, and pacB, the recognition site for the packaging machinery (T7, Chung & Hinkle 1990:T3, Hashimoto & Fujisawa 1992a). DNA packagingproceeds leftward from the end generated by the initiation cleavage at the right end of the terminal repetition ( pacCR), following recognition of pacB. The second termination cleavage can operate by a headfulmechanism at the left end of the terminal repetition(pacCL), or at any sequence if pacCL is deleted (Hashimoto & Fujisawa 1992a). (Fig. 1). DNA packa-ging is speciﬁc between T3 and T7. The packagingspeciﬁcity resides in concatemer processing, but the termination cleavage is not speciﬁc between T3 and T7, indicating that the packaging speciﬁcity resides in theinitiation event(s), including the initiation cleavage.ThepacB site contains a promoter sequence for phage RNA polymerase (the pacB promoter). The speciﬁcity analysis of the pacsignals in T3/T7 in chimeric pac signals indicates that the speciﬁcity is determined bypacB, i.e. the transcriptional speciﬁcity (Tsuchida et al. 1996). Although DNA packaging in a crude in vitro system depends on transcription (Hashimoto & Fujisawa1992b), it is rather stimulated by RNaseA but inhibitedby RNaseH. An RNA polymerase mutant active intranscription but defective in DNA packaging of T7 wasisolated, and this mutation was suppressed by a mutationin gene 19, suggesting direct interaction between RNA qBlackwell Science Limited Genes to Cells (1997) 2, 537–545 539genome DNA replication DNA processingconcatemer λcosNNu1, A cos Q cosB Packaging 1 2 Packaging 1 23 (A), 2(B) P22 (P1) T3, T7 Packaging18, 19 ;pacB ; nicking ; double-strand breakageFigure1 Concatemer processing path- ways of several phages .In general, the packaging of concatemeric DNA mole- cules is initiated from the end generated bythe initiation cleavage, proceeds unidir- ectionally, and is terminated by the termination cleavage. l-terminase intro- duces nicks staggered by 12 bp at cosN and melts the strands by its helicase activity after recognition of cosB ( X) and cosQ ( W) for initiation and termination cleavage, respectively.', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  l-terminase intro- duces nicks staggered by 12 bp at cosN and melts the strands by its helicase activity after recognition of cosB ( X) and cosQ ( W) for initiation and termination cleavage, respectively. With other phages, both cleavages are double-stranded breaksPhage DNA packaging  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  polymerase and packaging protein gp19 (Zhang & Studier 1995). The mutant was defective in pausing ata transcription-pause signal adjacent to the right end ofthe terminal repetitions in the concatemer (Lyakhov et al. 1997). The pacsignal of T3 becomes defective by mutations in the pause signal. For T3 and T7, theterminally repetitive sequence should be duplicatedduring DNA packaging from the concatemer. If there were not such duplication, half the replicated DNAwould be necessarily lost. Watson (1972) proposed a nicking model in which the terminal repetitivesequences are duplicated by DNA synthesis from 3 0 end produced by staggered nicks at the ends ( pacCL and CR) of the terminal repetitions (Fig. 2A). However, themodel is not consistent with evidence that the initiationor termination cleavages are double-strand cleavages(White & Richardson 1987: Fujisawa et al. 1990) and do not require pacCL or CR, respectively. We propose that 540 Genes to Cells (1997) 2, 537–545 qBlackwell Science Limited5‘ 3‘ 5‘3‘ 5‘ 3‘ 1 1 2 2 33 44 55 6 5‘ 3‘ 5‘3‘5‘ 3‘DNA polymerase DNA polymerase DNA polymeraseDNA polymerase DNA polymerase5‘ 5‘ 5‘ 5‘ 5‘ 5‘ DNA polymerasePackaginginitiation cleavage gp6 exonuclease Packaging Termination cleavage ; nascent DNA ; RNA ;pacB ; nicking ; double-strand breakage ; RNA polymerase (gp1)(A) (B) 5‘3‘ Figure2 Models for concatemer proces- sing of T3/T7 phage. (A) A nicking model (Watson 1972): DNA synthesis from 30- end produced by staggered nicks at theends of the terminal repetition (1) dupli- cates the terminal repetitions and separates the genome DNA from concatemericDNA (2, 3, 4, 5). (B) Our double-strandbreakage model: the packaging machinery recognizes a transcribing DNA/RNA hybrid structure and introduces a nick(1). After strand displacement synthesis (2), a double-stranded break is made to free the right end of the genome (3). The gp6exonuclease acts to degrade the newly synthesized DNA which remains annealed at the left end of the genome (4). Theresulting primer is extended to replicatethe left end (5), which leads to a second double-stranded break to complete con- catemer processing (6).H Fujisawa & M Morita  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  the packaging machinery recognizes a transcriptionally active DNA/RNA hybrid structure, generated bypausing at the pause signal, and introduces a nick. Theterminal repetition would be duplicated by displacementsynthesis from the nick (Fig. 2B). These events shouldoccur in adjacent terminal repetitions in a controlledmanner such as the synaptic mechanism proposed forT4 (Black 1995). Inf-29, DNA replication proceeds by a protein- priming mechanism from either end of the linearmolecule by strand displacement to generate maturemolecules with gp3 covalently linked to the 5 0-termini (Mellado et al. 1980). Because f-29 DNA is always linear and unit length, there is no need for a DNAcleavage reaction. The DNA-gp3 complex is thepackaging substrate and DNA must have attached gp3 for efﬁcient DNA packaging (Bjornsti et al. 1982). DNA condensation—DNA translocation In the phage head, the DNA is in a highly condensed state, with a packaging density approximately equal tothat of crystalline DNA. T o elucidate the molecular mechanism of DNA condensation by translocation, deﬁned in vitro DNA packaging systems, composed of puriﬁed proheads,packaging enzyme and DNA, have been developed forf-29 (Guo et al. 1987b) and T3 phages (Hamada et al. 1986). The author will focus on phage T3. The T3phage assembly pathway in the deﬁned in vitro system is shown in Fig. 3. The small subunit, gp18, and the largesubunit, gp19, form separate complexes with DNA andproheads, respectively. These complexes associate toform a direct precursor complex (50S complex) forDNA translocation into the head. The formation of thegp19-prohead complex and the 50S complex requiresATP as an allosteric effector (Shibata et al. 1987). gp19 is the only component which interacts with ATP in the deﬁned in vitro system and reveals multiple activities affected by interactions with ATP , which includespeciﬁc and nonspeciﬁc DNA cleavage, proheadbinding, DNA translocation and ATP hydrolysis. gp19 changes its conformation by binding to ATP , as judged from limited proteolysis experiments (Fujisawaet al. 1991). This conformational change is prerequisite for gp19 to form functional complexes with proheads, because proheads are inactivated by gp19 in the absenceof ATP . gp19 binds to the connector at the portal vertexof a prohead at a molar ratio of 6 or 20–30 in thepresence or absence of ATP , respectively, suggesting thatthe gp19-prohead complex is the main DNA packagingmachine. Based on the information obtained from thelimited proteolysis experiments, we propose three structural domains related to gp19 functions in DNA packaging; the N-terminal half domain accessible toproteases in the absence of ATP (amino acids 1–281),the internal domain inaccessible to proteases both in thepresence and absence of ATP (amino acids 281–500), qBlackwell Science Limited Genes to Cells (1997) 2, 537–545 541 Figure3 In vitro assembly pathway of T3 phage. See text for details.Phage DNA packaging  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  and the C-terminal domain accessible to proteases in the presence of ATP (amino acids 500 to the C-terminus)(Morita et al. 1995). T o deﬁne a region involved in prohead binding more precisely, deletion mutants,DC10, lacking the last 10 amino acids, and DC15, lacking the last 15 amino acids, were constructed bysite-directed mutagenesis. gp19- DC15 is defective in prohead binding, resulting in a defect in DNA packaging. On the other hand, gp19- DC10 showed signiﬁcant packaging activity in the deﬁned in vitro system, but gp19- DC10-prohead complexes were not detected in the presence of ATP . gp19- DC10 might dissociate from the complexes during sedimentation.Forl, Sippy & Feiss (1992) suggested that the last ﬁve amino acid residues are involved in prohead binding ofgpA, the large subunit, based on genetic analyses. The last six amino acid residues of gpA (L-S-G-E-D-E) are similar to L571 to D576 of gp19 (L-Y -W-E-D-D). Bothhave a hydrophobic residue ﬁrst, and acidic residues atthe end. A synthetic hexapeptide corresponding to thesix amino acid residues did not inhibit gp19 (wt) -prohead binding but interfered with their functionalinteraction, resulting in inhibition of DNA packaging(Morita et al. 1996). On the basis of these facts, the last 16 amino acids of gp19 (amino acids 571–586) could be divided into two functional domains, Region I (aminoacids 571–576) is a ‘core’domain, and its binding to theprohead is crucial for DNA packaging, and Region II(amino acids 577–586) is an ‘anchor’ domain stabilizingthe binding by Region I. Strategies employed in assembling the packaging machine are somewhat different among phages. f–29 also utilizes a bipartite packaging enzyme, where the small subunit, gp3, is covalently bound to the 5 0-end of the DNA. Packaging is initiated by binding of gp16, thelarge subunit, to the prohead and this complex thenbinds DNA-gp3, depending on ATP hydrolysis (Guo et al. 1987b). Six copies of the 174 base-long prohead RNA (pRNA) are attached to the connector; pRNA isessential for binding of gp16 to the prohead (Guo et al. 1987a). For lterminase, hetero-oligomers of gpNu1 (the small subunit) and gpA (the large subunit)bind to DNA to form a complex, called complex I,which then binds a prohead to form complex II,followed by DNA translocation (Becker et al. 1977). DNA translocation into the prohead is driven by ATP hydrolysis. The energetics of the DNA transloca-tion process may hold general signiﬁcance. The ability to transduce chemical energy from ATP hydrolysis into movement, ubiquitous to biological systems, is manifestin the translocation of viral DNA into a protein shell.With several phages, packaging proteins and in vitropackaging systems display DNA-dependent ATPase activity. However, the ATPases are stimulated bynonpackageable DNA. By using actinomycin D,which inhibits speciﬁcally the ATPase activity tightlycoupled to DNA translocation ( pac-ATPase), it is estimated that one molecule of ATP is hydrolysedduring translocation of 1.8 bp of DNA with the T3system (Morita et al. 1993). This value is very close to the value (2 bp/ATP) obtained with the f-29 system (Guo et al. 1987b). Since a gp19 mutant, G429R, is defective in coupling of ATP hydrolysis and DNAtranslocation, gp19 plays a central role in coupling ofATP hydrolysis to DNA translocation (Morita et al. 1994). Since the large subunit of every phage packagingenzyme has been shown to have an ATPase activity, it istempting to postulate that the large subunit is the energy transducer for translocation (Catalano et al. 1995).', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  1993). This value is very close to the value (2 bp/ATP) obtained with the f-29 system (Guo et al. 1987b). Since a gp19 mutant, G429R, is defective in coupling of ATP hydrolysis and DNAtranslocation, gp19 plays a central role in coupling ofATP hydrolysis to DNA translocation (Morita et al. 1994). Since the large subunit of every phage packagingenzyme has been shown to have an ATPase activity, it istempting to postulate that the large subunit is the energy transducer for translocation (Catalano et al. 1995). In the deﬁned in vitro system of T3 phage, DNA translocation can be synchronized by adding ATP toprecursor 50S complexes accumulated by incubationwith ATP- g-S. With this system, the average rate of DNA translocation is estimated to be 5.6 and 22 kbp/min at 20 8C and 30 8C, respectively (Shibata et al. 1987). Other important properties of DNA translocation obtained with the T3 system are as follows: (i) thepackaging machinery–DNA interaction is weak ortight in the presence or absence of ATP , respectively; (ii)nicked or crosslinked DNA is packaged, indicating thatDNA packaging does not require DNA superhelicity;and (iii) intercalators inhibit translocation, suggestingthat the machinery recognizes the exterior structure of DNA helix (Fujisawa et al. 1987; Shibata et al. 1987). Many models have been proposed for the molecular mechanism of DNA translocation in packaging drivenby ATP hydrolysis. None of these models has beenshown to be correct. According to the transientsupercoiling model, the packaging machinery translo-cates DNA by introducing a transient supercoiling intoDNA (Black & Silverman 1978). The transient super- coiling model is not consistent with the fact that nicked or cross-linked DNA can be packaged in the T3packaging system. In the rotating structure models,DNA threaded through the connector (Hendrix 1978)or wrapped around the connector (Turnquist et al. 1992) is translocated by rotation of the connector. In theformer model, a role of the packaging enzyme is notidentiﬁed and, in the latter model, the superhelicity of DNA is supposed to be a prerequisite for DNA packaging. The gated osmotic model predicts ATPdriven expansion-contraction cycles of the head shellpower DNA movement by an osmotic pressure 542 Genes to Cells (1997) 2, 537–545 qBlackwell Science LimitedH Fujisawa & M Morita  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  mechanism (Serwer 1988). However, it is difﬁcult to image that expanded protein shells contract. Our ‘ratchet’ model (Fig. 4) is as follows: one molecule of gp19 is bound to each component of the sixdomains of the connector, by the C-terminal region of gp19. When one of the gp19 molecules is bound to the sugar-phosphate backbone of the DNA duplex, the pac- ATPase is activated. When ATP is hydrolysed, aconformational change of gp19 is induced, forcingDNA into the head. When a new ATP binds to thisgp19, gp19 dissociates from the DNA and returns to theoriginal conformation. At this moment, the next gp19binds the neighbouring sugar-phosphate backbone. When this cyclic change proceeds six times, six molecules of ATP are hydrolysed and one turn ofduplex DNA is translocated into the head. Therefore, inthis model, one ATP molecule is utilized for thetranslocation of 1.7 bp DNA. This value is very close tothe observed value (Shibata et al. 1987; Morita et al. 1995). The assumption that gp19 molecules recognizedifferent regions of the sugar-phosphate backbone could explain why the second gp19 recognizes the DNA after translocation of 1.7 bp (not 2 bp) by the ﬁrstgp19. Ordered conformational changes in proteins thatare driven by nucleotide-triphosphate hydrolysis mayaccount for ATP-driven biological machines (Alberts & Miyake-Lye 1992). Acknowledgements We are grateful to Dr Michael Feiss (The University of Iowa) for discussions and invaluable help with the manuscript. References Alberts, B. & Miyake-Lye, R. (1992) Unscrambling the puzzle of biological machines: the importance of the details. Cell68, 415–420. Bazinet, C. & King, J. (1986) The DNA translocating vertex of dsDNA bacteriophage. Annu. Rev. Microbiol. 39, 109–129. Becker, A. & Gold, M. (1988) Prediction of an ATP reactive center in the small subunit, gpNu1, of the phage lambdaterminase enzyme. J. Mol. Biol. 199, 2819–2823. Becker, A., Marko, M. & Gold, M. (1977) Early events in the in vitro packaging of bacteriophage lDNA. Virology 78, 291– 305. Bhattacharyya, S.P . & Rao, V .B. (1993) A novel terminase activity associated with the DNA packaging protein gp17 of bacteriophage T4. Virology 196, 34–44. Bjornsti, M.A., Reilly, B.E. & Anderson, D.L. (1982) Morpho- genesis of bacteriophage f-29 of Bacillus subtilis : DNA-gp3 intermediate in vivo andin vitro assembly. J. Virol. 41, 508–517. qBlackwell Science Limited Genes to Cells (1997) 2, 537–545 5436 5 4 3 216 5 4 3 21 DNA translocation connector gp 19 ;A T P ; ADP1 2 3 4(a) (b) (c) (d) Figure4 A ratchet model for DNA translocation of T3 DNA. The gp19 which binds to the sugar-phosphate backbone is marked with a pale blue colour. In the upper part, the arrangement of gp19 on the connector is shown. See text for details.Phage DNA packaging  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6160>, 'json_data': {'input': ['page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  REVIEW Phage DNA packaging Hisao Fujisawa1,a,xand Miyo Morita2 1Department of Botany, Division of Biological Science, Graduate School of Science, Kyoto University, Kyoto 606, Japan 2HSP Research Institute, INC. Kyoto Research Park, Shimogyou-ku, Kyoto 600, Japan Phage DNA packaging occurs by DNA translocation into a preformed protein shell—a prohead—with the aid of a packaging enzyme or a terminase. The packaging enzyme iscomposed of two subunits: the large subunit has ATP-binding, prohead binding, and DNAcleavage activities, and the small subunit is a DNA binding protein. DNA translocation is drivenby ATP hydrolysis. In general, phage DNA replication mechanisms lead to the accumulation ofconcatemers. Concatemers are processed to mature DNA during and depending upon DNA packaging. This review will focus on the molecular mechanism of concatemer processing and the coupling of ATP hydrolysis to DNA translocation. Introduction The DNA of double-stranded DNA bacteriophages (ds-DNA phages) is contained in a highly condensed state within an outer shell of protein. The nucleotide sequence of the DNA is divided into the followingcategories: (i) linear with a unique nucleotide sequencerepeated at the ends (linear, nonpermuted, terminallyrepetitious: l, P2, P4, T3, T7); (ii) linear with a nonunique nucleotide sequence repeated at the ends(linear, permuted, terminally repetitious: P1, P22, T4);and (iii) miscellaneous, such as linear with a protein covalently attached ( f-29) or with host sequences (Mu). Variations of these types include a single-strandedterminal repetition ( l, P22, P4) and different degrees of permutation (P1, P2, T4). However, in general, theDNA replication mechanisms of these phages lead tothe accumulation of DNA concatemers in whichmature phage DNAs are joined together in a head-to-tail manner through the terminal repetitions. The concatemer is a substrate for DNA encapsidation into preformed proheads. During DNA packaging, viralDNA is recognized and unit-length virion DNA isproduced from the concatemer by a mechanism speciﬁcfor each phage. Exceptions to this are the packaging ofreplicated circular DNA for P2 and P4, protein- terminated mature length DNA for f-29, and chro- mosomally integrated DNA for Mu phage. Themechanism by which the concatemer intermediate isprocessed to generate the mature DNA is vital inallowing the ends of the linear chromosomal DNA to bereplicated completely without a loss of the terminalbases at the 5 0end of each strand. Therefore, phage DNA packaging reactions include two important mechanisms: (i) the mechanism of end-formationfrom the concatemer, details of which are speciﬁc foreach phage, and (ii) the energetic mechanism of DNAtranslocation into the prohead driven by ATP hydro-lysis, which is common among phages. DNA packaging machinery DNA packaging is thought to proceed by a commonmechanism for most ds-DNA phages. DNA is packaged into the cavity of a preformed protein shell, called the prohead, through the connector located at the portalvertex, with the aid of a noncapsid protein called aterminase or packaging enzyme (Earnshaw & Casjens1980; Black 1989). During the packaging of DNA,mature monomers are cleaved from the concatemer.The formation of precursors for DNA translocation,ternary complexes containing the prohead, the packa- ging enzyme, and the DNA, depends upon either ATP as an allosteric effector (T3, Shibata et al. 1987) or ATP qBlackwell Science Limited Genes to Cells (1997) 2, 537–545 537*Correspondence : E-mail: hisao.fujisawa@ma1.seikyou. ne.jp aPresent address : 2-22-12-Hieidaira, Otsu 520, Japan.', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  hydrolysis ( l, Becker et al. 1977; f-29, Guo et al. 1987a). DNA translocation into the prohead is driven by ATPhydrolysis. Packaging enzymes (terminases) and in vitro packaging systems display a DNA-dependent ATPaseactivity. These results indicate that ATP plays multipleroles in the DNA packaging reaction. The connectorsshare some common features: an outer region with 12small domains and an inner region with 6 domains surrounding a channel (Bazinet & King 1986; Valpuesta & Carrascosa 1994). There is genetic evidence for theinvolvement of the connector in DNA packaging forT4 (Hsiao & Black 1977), T3 (Nakasu et al. 1983) and l (Cue & Feiss 1997). In general, the packaging enzyme iscommonly composed of two subunits: the small subunithas DNA binding activity and is involved in recognitionof its own DNA, and the large subunit is responsible for prohead binding, DNA translocation and DNA clea- vage (Catalano et al. 1995). The large subunit binds the connector of the prohead (T4, Hsiao & Black 1977; T3,Fujisawa et al. 1991; l, Y eo & Feiss 1995). The putative prohead binding domain has been identiﬁed toward theC-terminus. The large subunits are ATP-bindingproteins and are thought to carry crucial functions forDNA translocation. In the case of l, the small subunit also has ATP binding motifs (Becker & Gold 1988) and displays a DNA-stimulated ATPase (T omka & Catalano1993; Hwang et al. 1996). DNA processing Although the mechanisms of end-formation of mature DNAs are different among phages, they have common features. The packaging of concatemeric DNA mole- cules occurs by the recognition of a speciﬁc sequencecalled cosorpac, followed by speciﬁc ( l, T3, T7) or non-sequence speciﬁc cleavage (P1, P22, T1, SPP1)(initiation cleavage) and unidirectional DNA transloca-tion from the generated end. When the prohead shell isfull of DNA, a second cleavage (termination cleavage) ismade at a speciﬁc ( l, T3, T7) or nonspeciﬁc (P1, P22, T1, SPP1) site to generate the mature form of phage DNA present in virions (the headful mechanism). WithT4, there are completely non-sequence speciﬁc initia-tion and termination cleavages (T -even). Initiationcleavage is subject to different and more stringentcontrol than termination cleavage. T oo infrequentcleavage would leave unﬁlled virus particles, whereastoo frequent initiation would produce defective parti- cles. In spite of the variety of cleavage speciﬁcities, the strict dependence of the termination cleavage on theheadful mechanism is common among phages, includ-ing even l(Feiss et al. 1977) and T3 (Hashimoto &Fujisawa 1988). Although the headful mechanism is not understood, there is accumulating evidence that theportal proteins are involved in the process (T3, Fujisawaet al. 1990; P22, Casjens et al. 1982; SPP1, Travares et al. 1992; l, Cue & Feiss 1997). Deﬁned in vitro systems developed for T3, T4, P1 and P22 can package anylinear ds-DNA, indicating that the selective packagingof viral DNA into a capsid, resides in the speciﬁcity of the initiation cleavage which generates a DNA end essential for initiation of DNA packaging. There isevidence with some phages that the end generated bythe termination cleavage is preferentially used for theinitiation of the second packaging, and so forth. Sincephages can package and transduce recombinant plasmidscarrying sequences necessary for packaging ( pacsignals), pacsignals have been identiﬁed by transduction assays.', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  1990; P22, Casjens et al. 1982; SPP1, Travares et al. 1992; l, Cue & Feiss 1997). Deﬁned in vitro systems developed for T3, T4, P1 and P22 can package anylinear ds-DNA, indicating that the selective packagingof viral DNA into a capsid, resides in the speciﬁcity of the initiation cleavage which generates a DNA end essential for initiation of DNA packaging. There isevidence with some phages that the end generated bythe termination cleavage is preferentially used for theinitiation of the second packaging, and so forth. Sincephages can package and transduce recombinant plasmidscarrying sequences necessary for packaging ( pacsignals), pacsignals have been identiﬁed by transduction assays. Thepacsignals have a bipartite structure composed of separate recognition ( pacB,cosB,cosQ) and cleavage sites (pacC,cosN) and lie either within or very close to the genes encoding the packaging proteins. For l, the pac signal seems to be composed of two sets of bipartitestructures: one for initiation ( cosB,cosN) and another for termination ( cosQ,cosN) (Catalano et al. 1995). Phage lmature DNA molecules are unique with 12- base pair long, complementary single-stranded 5 0ends, called cohesive ends. In the late stage of linfection, DNA is synthesized as concatemers by a rolling circlemechanism. DNA packaging proceeds rightward fromthe end generated by staggered nicks at the cosN site. After headful packaging, introduction of staggered nicksat the second cosN completes DNA packaging. cosB and cosQ are required for initiation and termination cleavage, respectively (Catalano et al. 1995) (Fig. 1). The genomes of P1 and P22 phages have restricted circular permutation maps (P1, Ikeda & T omizawa1965; P22, T ye et al. 1974). For these phages (Fig. 1), DNA packaging initiates from the end generated byimprecise cleavage near a speciﬁc site, called pac,o n concatemeric DNA molecules. Packaging proceedsrightward until the prohead shell is ﬁlled with DNA (about 104% of the genome sequence), after which a headful cleavage is made in a nonspeciﬁc sequencemanner on the concatemer that releases the packagedDNA. Subsequent packaging events on the concate-meric DNA begin at the end generated by the headfulcleavage of the previous chromosome and proceed inthe same direction as the previous event. On average,such processive series are 3–5 chromosomes in length, yielding terminally redundant and partially circularly permuted mature DNAs (P1, Sternberg & Coulby 1987;P22, Casjens et al. 1987). The paccleavage of these phages occurs without concomitant DNA packaging in 538 Genes to Cells (1997) 2, 537–545 qBlackwell Science LimitedH Fujisawa & M Morita  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  mutants in prohead assembly and therefore requires control over the frequency of pacsite cleavage. In P1, DNA methylation of 50-GATC-30in the pacsignal has been proposed to be a regulatory mechanism for pacsite cleavage (Sternberg & Coulby 1990). Phages P1 and P22 can package host DNA from the ends of DNA fragments produced by phage-induced DNase activity(generalized transduction). T -even phages apparently have randomly permuted DNA. These phages have therefore been concluded topackage their DNA according to a pure headfulmechanism (Streisinger et al. 1967). Although features recognized by the T4 terminase for making the sequence nonspeciﬁc initiation cleavage are unknown, it is proposed that an open DNA structure generated bytranscription or recombination is a potential pacsite for recognition and attachment of T4 terminase (Mosig et al. 1981; Bhattacharyya & Rao 1993). Mu phage recognizes the pacsite located at the left end of the genome DNA and generates cleaved ends inhost DNA to the left end of the genome. DNA packaging proceeds rightward past the right end of the phage DNA and terminates in the host sequence by aheadful mechanism (George & Bukhari 1981). T3 and T7 phages are closely related and have extensive sequences of homology. Their DNAs areunique collections with the terminal repetitions of 230and 160 bp, respectively. The pacsignal of T3 and T7 iscomposed of pacC, the cleavage site for processing of concatemeric DNA, and pacB, the recognition site for the packaging machinery (T7, Chung & Hinkle 1990:T3, Hashimoto & Fujisawa 1992a). DNA packagingproceeds leftward from the end generated by the initiation cleavage at the right end of the terminal repetition ( pacCR), following recognition of pacB. The second termination cleavage can operate by a headfulmechanism at the left end of the terminal repetition(pacCL), or at any sequence if pacCL is deleted (Hashimoto & Fujisawa 1992a). (Fig. 1). DNA packa-ging is speciﬁc between T3 and T7. The packagingspeciﬁcity resides in concatemer processing, but the termination cleavage is not speciﬁc between T3 and T7, indicating that the packaging speciﬁcity resides in theinitiation event(s), including the initiation cleavage.ThepacB site contains a promoter sequence for phage RNA polymerase (the pacB promoter). The speciﬁcity analysis of the pacsignals in T3/T7 in chimeric pac signals indicates that the speciﬁcity is determined bypacB, i.e. the transcriptional speciﬁcity (Tsuchida et al. 1996). Although DNA packaging in a crude in vitro system depends on transcription (Hashimoto & Fujisawa1992b), it is rather stimulated by RNaseA but inhibitedby RNaseH. An RNA polymerase mutant active intranscription but defective in DNA packaging of T7 wasisolated, and this mutation was suppressed by a mutationin gene 19, suggesting direct interaction between RNA qBlackwell Science Limited Genes to Cells (1997) 2, 537–545 539genome DNA replication DNA processingconcatemer λcosNNu1, A cos Q cosB Packaging 1 2 Packaging 1 23 (A), 2(B) P22 (P1) T3, T7 Packaging18, 19 ;pacB ; nicking ; double-strand breakageFigure1 Concatemer processing path- ways of several phages .In general, the packaging of concatemeric DNA mole- cules is initiated from the end generated bythe initiation cleavage, proceeds unidir- ectionally, and is terminated by the termination cleavage. l-terminase intro- duces nicks staggered by 12 bp at cosN and melts the strands by its helicase activity after recognition of cosB ( X) and cosQ ( W) for initiation and termination cleavage, respectively.', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  l-terminase intro- duces nicks staggered by 12 bp at cosN and melts the strands by its helicase activity after recognition of cosB ( X) and cosQ ( W) for initiation and termination cleavage, respectively. With other phages, both cleavages are double-stranded breaksPhage DNA packaging  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  polymerase and packaging protein gp19 (Zhang & Studier 1995). The mutant was defective in pausing ata transcription-pause signal adjacent to the right end ofthe terminal repetitions in the concatemer (Lyakhov et al. 1997). The pacsignal of T3 becomes defective by mutations in the pause signal. For T3 and T7, theterminally repetitive sequence should be duplicatedduring DNA packaging from the concatemer. If there were not such duplication, half the replicated DNAwould be necessarily lost. Watson (1972) proposed a nicking model in which the terminal repetitivesequences are duplicated by DNA synthesis from 3 0 end produced by staggered nicks at the ends ( pacCL and CR) of the terminal repetitions (Fig. 2A). However, themodel is not consistent with evidence that the initiationor termination cleavages are double-strand cleavages(White & Richardson 1987: Fujisawa et al. 1990) and do not require pacCL or CR, respectively. We propose that 540 Genes to Cells (1997) 2, 537–545 qBlackwell Science Limited5‘ 3‘ 5‘3‘ 5‘ 3‘ 1 1 2 2 33 44 55 6 5‘ 3‘ 5‘3‘5‘ 3‘DNA polymerase DNA polymerase DNA polymeraseDNA polymerase DNA polymerase5‘ 5‘ 5‘ 5‘ 5‘ 5‘ DNA polymerasePackaginginitiation cleavage gp6 exonuclease Packaging Termination cleavage ; nascent DNA ; RNA ;pacB ; nicking ; double-strand breakage ; RNA polymerase (gp1)(A) (B) 5‘3‘ Figure2 Models for concatemer proces- sing of T3/T7 phage. (A) A nicking model (Watson 1972): DNA synthesis from 30- end produced by staggered nicks at theends of the terminal repetition (1) dupli- cates the terminal repetitions and separates the genome DNA from concatemericDNA (2, 3, 4, 5). (B) Our double-strandbreakage model: the packaging machinery recognizes a transcribing DNA/RNA hybrid structure and introduces a nick(1). After strand displacement synthesis (2), a double-stranded break is made to free the right end of the genome (3). The gp6exonuclease acts to degrade the newly synthesized DNA which remains annealed at the left end of the genome (4). Theresulting primer is extended to replicatethe left end (5), which leads to a second double-stranded break to complete con- catemer processing (6).H Fujisawa & M Morita  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  the packaging machinery recognizes a transcriptionally active DNA/RNA hybrid structure, generated bypausing at the pause signal, and introduces a nick. Theterminal repetition would be duplicated by displacementsynthesis from the nick (Fig. 2B). These events shouldoccur in adjacent terminal repetitions in a controlledmanner such as the synaptic mechanism proposed forT4 (Black 1995). Inf-29, DNA replication proceeds by a protein- priming mechanism from either end of the linearmolecule by strand displacement to generate maturemolecules with gp3 covalently linked to the 5 0-termini (Mellado et al. 1980). Because f-29 DNA is always linear and unit length, there is no need for a DNAcleavage reaction. The DNA-gp3 complex is thepackaging substrate and DNA must have attached gp3 for efﬁcient DNA packaging (Bjornsti et al. 1982). DNA condensation—DNA translocation In the phage head, the DNA is in a highly condensed state, with a packaging density approximately equal tothat of crystalline DNA. T o elucidate the molecular mechanism of DNA condensation by translocation, deﬁned in vitro DNA packaging systems, composed of puriﬁed proheads,packaging enzyme and DNA, have been developed forf-29 (Guo et al. 1987b) and T3 phages (Hamada et al. 1986). The author will focus on phage T3. The T3phage assembly pathway in the deﬁned in vitro system is shown in Fig. 3. The small subunit, gp18, and the largesubunit, gp19, form separate complexes with DNA andproheads, respectively. These complexes associate toform a direct precursor complex (50S complex) forDNA translocation into the head. The formation of thegp19-prohead complex and the 50S complex requiresATP as an allosteric effector (Shibata et al. 1987). gp19 is the only component which interacts with ATP in the deﬁned in vitro system and reveals multiple activities affected by interactions with ATP , which includespeciﬁc and nonspeciﬁc DNA cleavage, proheadbinding, DNA translocation and ATP hydrolysis. gp19 changes its conformation by binding to ATP , as judged from limited proteolysis experiments (Fujisawaet al. 1991). This conformational change is prerequisite for gp19 to form functional complexes with proheads, because proheads are inactivated by gp19 in the absenceof ATP . gp19 binds to the connector at the portal vertexof a prohead at a molar ratio of 6 or 20–30 in thepresence or absence of ATP , respectively, suggesting thatthe gp19-prohead complex is the main DNA packagingmachine. Based on the information obtained from thelimited proteolysis experiments, we propose three structural domains related to gp19 functions in DNA packaging; the N-terminal half domain accessible toproteases in the absence of ATP (amino acids 1–281),the internal domain inaccessible to proteases both in thepresence and absence of ATP (amino acids 281–500), qBlackwell Science Limited Genes to Cells (1997) 2, 537–545 541 Figure3 In vitro assembly pathway of T3 phage. See text for details.Phage DNA packaging  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  and the C-terminal domain accessible to proteases in the presence of ATP (amino acids 500 to the C-terminus)(Morita et al. 1995). T o deﬁne a region involved in prohead binding more precisely, deletion mutants,DC10, lacking the last 10 amino acids, and DC15, lacking the last 15 amino acids, were constructed bysite-directed mutagenesis. gp19- DC15 is defective in prohead binding, resulting in a defect in DNA packaging. On the other hand, gp19- DC10 showed signiﬁcant packaging activity in the deﬁned in vitro system, but gp19- DC10-prohead complexes were not detected in the presence of ATP . gp19- DC10 might dissociate from the complexes during sedimentation.Forl, Sippy & Feiss (1992) suggested that the last ﬁve amino acid residues are involved in prohead binding ofgpA, the large subunit, based on genetic analyses. The last six amino acid residues of gpA (L-S-G-E-D-E) are similar to L571 to D576 of gp19 (L-Y -W-E-D-D). Bothhave a hydrophobic residue ﬁrst, and acidic residues atthe end. A synthetic hexapeptide corresponding to thesix amino acid residues did not inhibit gp19 (wt) -prohead binding but interfered with their functionalinteraction, resulting in inhibition of DNA packaging(Morita et al. 1996). On the basis of these facts, the last 16 amino acids of gp19 (amino acids 571–586) could be divided into two functional domains, Region I (aminoacids 571–576) is a ‘core’domain, and its binding to theprohead is crucial for DNA packaging, and Region II(amino acids 577–586) is an ‘anchor’ domain stabilizingthe binding by Region I. Strategies employed in assembling the packaging machine are somewhat different among phages. f–29 also utilizes a bipartite packaging enzyme, where the small subunit, gp3, is covalently bound to the 5 0-end of the DNA. Packaging is initiated by binding of gp16, thelarge subunit, to the prohead and this complex thenbinds DNA-gp3, depending on ATP hydrolysis (Guo et al. 1987b). Six copies of the 174 base-long prohead RNA (pRNA) are attached to the connector; pRNA isessential for binding of gp16 to the prohead (Guo et al. 1987a). For lterminase, hetero-oligomers of gpNu1 (the small subunit) and gpA (the large subunit)bind to DNA to form a complex, called complex I,which then binds a prohead to form complex II,followed by DNA translocation (Becker et al. 1977). DNA translocation into the prohead is driven by ATP hydrolysis. The energetics of the DNA transloca-tion process may hold general signiﬁcance. The ability to transduce chemical energy from ATP hydrolysis into movement, ubiquitous to biological systems, is manifestin the translocation of viral DNA into a protein shell.With several phages, packaging proteins and in vitropackaging systems display DNA-dependent ATPase activity. However, the ATPases are stimulated bynonpackageable DNA. By using actinomycin D,which inhibits speciﬁcally the ATPase activity tightlycoupled to DNA translocation ( pac-ATPase), it is estimated that one molecule of ATP is hydrolysedduring translocation of 1.8 bp of DNA with the T3system (Morita et al. 1993). This value is very close to the value (2 bp/ATP) obtained with the f-29 system (Guo et al. 1987b). Since a gp19 mutant, G429R, is defective in coupling of ATP hydrolysis and DNAtranslocation, gp19 plays a central role in coupling ofATP hydrolysis to DNA translocation (Morita et al. 1994). Since the large subunit of every phage packagingenzyme has been shown to have an ATPase activity, it istempting to postulate that the large subunit is the energy transducer for translocation (Catalano et al. 1995).', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  1993). This value is very close to the value (2 bp/ATP) obtained with the f-29 system (Guo et al. 1987b). Since a gp19 mutant, G429R, is defective in coupling of ATP hydrolysis and DNAtranslocation, gp19 plays a central role in coupling ofATP hydrolysis to DNA translocation (Morita et al. 1994). Since the large subunit of every phage packagingenzyme has been shown to have an ATPase activity, it istempting to postulate that the large subunit is the energy transducer for translocation (Catalano et al. 1995). In the deﬁned in vitro system of T3 phage, DNA translocation can be synchronized by adding ATP toprecursor 50S complexes accumulated by incubationwith ATP- g-S. With this system, the average rate of DNA translocation is estimated to be 5.6 and 22 kbp/min at 20 8C and 30 8C, respectively (Shibata et al. 1987). Other important properties of DNA translocation obtained with the T3 system are as follows: (i) thepackaging machinery–DNA interaction is weak ortight in the presence or absence of ATP , respectively; (ii)nicked or crosslinked DNA is packaged, indicating thatDNA packaging does not require DNA superhelicity;and (iii) intercalators inhibit translocation, suggestingthat the machinery recognizes the exterior structure of DNA helix (Fujisawa et al. 1987; Shibata et al. 1987). Many models have been proposed for the molecular mechanism of DNA translocation in packaging drivenby ATP hydrolysis. None of these models has beenshown to be correct. According to the transientsupercoiling model, the packaging machinery translo-cates DNA by introducing a transient supercoiling intoDNA (Black & Silverman 1978). The transient super- coiling model is not consistent with the fact that nicked or cross-linked DNA can be packaged in the T3packaging system. In the rotating structure models,DNA threaded through the connector (Hendrix 1978)or wrapped around the connector (Turnquist et al. 1992) is translocated by rotation of the connector. In theformer model, a role of the packaging enzyme is notidentiﬁed and, in the latter model, the superhelicity of DNA is supposed to be a prerequisite for DNA packaging. The gated osmotic model predicts ATPdriven expansion-contraction cycles of the head shellpower DNA movement by an osmotic pressure 542 Genes to Cells (1997) 2, 537–545 qBlackwell Science LimitedH Fujisawa & M Morita  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  mechanism (Serwer 1988). However, it is difﬁcult to image that expanded protein shells contract. Our ‘ratchet’ model (Fig. 4) is as follows: one molecule of gp19 is bound to each component of the sixdomains of the connector, by the C-terminal region of gp19. When one of the gp19 molecules is bound to the sugar-phosphate backbone of the DNA duplex, the pac- ATPase is activated. When ATP is hydrolysed, aconformational change of gp19 is induced, forcingDNA into the head. When a new ATP binds to thisgp19, gp19 dissociates from the DNA and returns to theoriginal conformation. At this moment, the next gp19binds the neighbouring sugar-phosphate backbone. When this cyclic change proceeds six times, six molecules of ATP are hydrolysed and one turn ofduplex DNA is translocated into the head. Therefore, inthis model, one ATP molecule is utilized for thetranslocation of 1.7 bp DNA. This value is very close tothe observed value (Shibata et al. 1987; Morita et al. 1995). The assumption that gp19 molecules recognizedifferent regions of the sugar-phosphate backbone could explain why the second gp19 recognizes the DNA after translocation of 1.7 bp (not 2 bp) by the ﬁrstgp19. Ordered conformational changes in proteins thatare driven by nucleotide-triphosphate hydrolysis mayaccount for ATP-driven biological machines (Alberts & Miyake-Lye 1992). Acknowledgements We are grateful to Dr Michael Feiss (The University of Iowa) for discussions and invaluable help with the manuscript. References Alberts, B. & Miyake-Lye, R. (1992) Unscrambling the puzzle of biological machines: the importance of the details. Cell68, 415–420. Bazinet, C. & King, J. (1986) The DNA translocating vertex of dsDNA bacteriophage. Annu. Rev. Microbiol. 39, 109–129. Becker, A. & Gold, M. (1988) Prediction of an ATP reactive center in the small subunit, gpNu1, of the phage lambdaterminase enzyme. J. Mol. Biol. 199, 2819–2823. Becker, A., Marko, M. & Gold, M. (1977) Early events in the in vitro packaging of bacteriophage lDNA. Virology 78, 291– 305. Bhattacharyya, S.P . & Rao, V .B. (1993) A novel terminase activity associated with the DNA packaging protein gp17 of bacteriophage T4. Virology 196, 34–44. Bjornsti, M.A., Reilly, B.E. & Anderson, D.L. (1982) Morpho- genesis of bacteriophage f-29 of Bacillus subtilis : DNA-gp3 intermediate in vivo andin vitro assembly. J. Virol. 41, 508–517. qBlackwell Science Limited Genes to Cells (1997) 2, 537–545 5436 5 4 3 216 5 4 3 21 DNA translocation connector gp 19 ;A T P ; ADP1 2 3 4(a) (b) (c) (d) Figure4 A ratchet model for DNA translocation of T3 DNA. The gp19 which binds to the sugar-phosphate backbone is marked with a pale blue colour. In the upper part, the arrangement of gp19 on the connector is shown. See text for details.Phage DNA packaging  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:54 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'85'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992207'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'467ms'), (b'x-request-id', b'6f053c8f58eb9e6606110973e7803ba2'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124975fb515d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:54 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'85'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992207'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'467ms'), (b'x-request-id', b'6f053c8f58eb9e6606110973e7803ba2'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84124975fb515d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6160>, 'json_data': {'input': ['page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  Black, L.W . (1989) DNA packaging in dsDNA bacteriophage. Annu. Rev. Microbiol. 43, 267–292. Black, L.W . (1995) DNA packaging and cutting by phage terminases: control in phage T4 by a synaptic mechanism.BioEssays 17, 1025–1030. Black, L.W . & Silverman, D. (1978) Model for DNA packaging into bacteriophage T4 heads. J. Virol. 28, 643–655. Casjens, S., Huang, W .M., Hayden, M. & Parr, R. (1987) Initiation of bacteriophage P22 DNA packaging series. Analysis of a mutant that alters the DNA target speciﬁcity of the packaging apparatus. J. Mol. Biol. 194, 411–422. Casjens, S., Wyckoff, E., Hayden, M., et al. (1982) Bacteriophage P22 portal protein is part of the gauge that regulates packaging density of intravirion DNA. J. Mol. Biol. 224, 1055–1074. Catalano, C.E., Cue, D. & Feiss, M. (1995) Virus DNA packaging: the strategy used by phage l.Mol. Microbiol. 16, 1075–1086. Chung, Y .B. & Hinkle, D.C. (1990) Bacteriophage T7 DNA packaging. II. Analysis of the DNA sequences required forpackaging using a plasmid transduction assay. J. Mol. Biol. 216, 927–938. Cue, D. & Feiss, M. (1997) Genetic evidence that recognition ofcosQ, the signal for termination of phage lDNA packaging, depends on the extent of head ﬁlling. Genetics ,i n press. Earnshaw, W .C. & Casjens, S.R. (1980) DNA packaging by the double-stranded DNA bacteriophages. Cell21, 319–331. Feiss, M., Fisher, R.A., Crayton, M.A. & Egner, C. (1977) Packaging of bacteriophage lchromosome: effect of chromo- some length. Virology 77, 281–293. Fujisawa, H., Hamada, K., Shibata, H. & Minagawa, T. (1987) On the molecular mechanism of DNA translocation during in vitro packaging of bacteriophage T3 DNA. Virology 161, 228–233. Fujisawa, H., Kimura, M. & Hashimoto, C. (1990) In vitro cleavage of the concatemer joint of bacteriophage T3. Virology 174, 26–34. Fujisawa, H., Shibata, H. & Kato, H. (1991) Analysis of interactions among factors involved in the bacteriophage T3 DNA packaging reaction in a deﬁned in vitro system. Virology 185, 788–794. George, M. & Bukhari, A.I. (1981) Heterologous host DNA attached to the left end of mature bacteriophage Mu DNA. Nature 292, 175–176. Guo, P ., Erickson, S. & Anderson, D. (1987a) A small viral RNA is required for in vitro packaging of bacteriophage f-29 DNA- gp3.Science 236, 690–694. Guo, P ., Peterson, C. & Anderson, D. (1987b) Initiation events in in vitro packaging of bacteriophage f-29 DNA-gp3. J. Mol. Biol.197, 219–228. Hamada, K., Fujisawa, H. & Minagawa, T. (1986) A deﬁned in vitrosystem for packaging of bacteriophage T3 DNA. Virology 151, 119–123. Hashimoto, C. & Fujisawa, H.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  Nature 292, 175–176. Guo, P ., Erickson, S. & Anderson, D. (1987a) A small viral RNA is required for in vitro packaging of bacteriophage f-29 DNA- gp3.Science 236, 690–694. Guo, P ., Peterson, C. & Anderson, D. (1987b) Initiation events in in vitro packaging of bacteriophage f-29 DNA-gp3. J. Mol. Biol.197, 219–228. Hamada, K., Fujisawa, H. & Minagawa, T. (1986) A deﬁned in vitrosystem for packaging of bacteriophage T3 DNA. Virology 151, 119–123. Hashimoto, C. & Fujisawa, H. (1988) Packaging and transduc- tion of non-T3 DNA by bacteriophage T3. Virology 166, 788–796. Hashimoto, C. & Fujisawa, H. (1992a) DNA sequences necessary for packaging bacteriophage T3 DNA. Virology 187, 788–795. Hashimoto, C. & Fujisawa, H. (1992b) Transcription depen- dence of DNA packaging of bacteriophages T3 and T7.Virology 191, 246–250. Hendrix, R.W . (1978) Symmetry mismatch and DNA packagingin large bacteriophges. Proc. Natl. Acad. Sci. USA 75, 4779– 4783. Hsiao, C.L. & Black, L.W . (1977) DNA packaging and the pathway of bacteriophage T4 head assembly. Proc. Natl. Acad. Sci. USA. 74, 3552–3656. Hwang, Y ., Catalano, C. & Feiss, M. (1996) Kinetic and mutational dissection of the two ATP activities of terminase,the DNA packaging enzyme of bacteriophage l.Biochemistry 35, 2796–2803. Ikeda, H. & T omizawa, J.-I. (1965) Transducing fragments in generalized tranduction by P1. I. Molecular origin of thefragments. J. Mol. Biol. 14, 85–109. Lyakhov, D.L., He, B., Zhang, X., Studier, F .W ., Dunn, J.J. & McAllister, W .T. (1997) Mutant Bacteriophage T7 RNA polymerases with altered termination properties. J. Mol. Biol. 269, 28–40. Mellado, R.P ., Penalva, M.A., Inciarte, M.R. & Salas, M. (1980) The protein covalently linked to the 5 0-termini of the DNA of Bacillus subtilis phage f-29 is involved in the initiation of DNA replication. Virology 104, 84–96. Morita, M., Tasaka, M. & Fujisawa, H. (1993) DNA packaging ATPase of bacteriophage T3. Virology 193, 748–752. Morita, M., Tasaka, M. & Fujisawa, H. (1994) Analysis of functional domains of the packaging proteins of bacteriophage T3 by site-directed mutagebesis. J. Mol. Biol. 235, 248–259. Morita, M., Tasaka, M. & Fujisawa, H. (1995) Structural and functional domains of the DNA packaging protein ofbacteriophage T3: importance of the C-terminal region of the large subunit in prohead binding. J. Mol. Biol. 245, 635– 644. Morita, M., Tasaka, M. & Fujisawa, H. (1996) Analysis of the ﬁne structure of the prohead binding domain of the packaging protein of bacteriophage T3 using a hexapeptide, an analog ofa prohead binding site. Virology 211, 516–524.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  (1994) Analysis of functional domains of the packaging proteins of bacteriophage T3 by site-directed mutagebesis. J. Mol. Biol. 235, 248–259. Morita, M., Tasaka, M. & Fujisawa, H. (1995) Structural and functional domains of the DNA packaging protein ofbacteriophage T3: importance of the C-terminal region of the large subunit in prohead binding. J. Mol. Biol. 245, 635– 644. Morita, M., Tasaka, M. & Fujisawa, H. (1996) Analysis of the ﬁne structure of the prohead binding domain of the packaging protein of bacteriophage T3 using a hexapeptide, an analog ofa prohead binding site. Virology 211, 516–524. Mosig, G., Ghosal, D. & Bock, S. (1981) Interactions between the maturation protein gp17 and single stranded DNA binding protein gp32 initiate DNA packaging and compete withinitiation of secondary replication forks in phage T4. In:Bacteriophage Assembly (ed. M.S. Dubow), pp. 139–151, New Y ork: Alan R. Liss. Nakasu, S., Fujisawa, H. & Minagawa, T. (1983) Role of gene 8 product in morphogenesis of bacteriophage T3. Virology 127, 124–133. Serwer, P . (1988) The source of energy for bacteriophage DNA packaging: an osmotic pump explains the data. Biopolymers 27, 165–169. Shibata, H., Fujisawa, H. & Minagawa, T. (1987) Characteriza- tion of the bacteriophage T3 DNA packaging reaction in vitro in a deﬁned system. J. Mol. Biol. 196, 845–851. Sippy, J. & Feiss, M. (1992) Analysis of a mutation affecting the speciﬁcity domain for prohead binding of the bacteriophage l terminase. J. Bacteriol. 174, 850–856. Sternberg, N. & Coulby, J. (1987) Recognition and cleavage of the bacteriophage P1 packaging site ( pac) I. Differential processing of the cleaved ends in vivo .J. Mol. Biol. 194, 453– 468. Sternberg, N. & Coulby, J. (1990) Cleavage of the bacteriophage P1 packaging site ( pac) is regulated by adenine methylation. Proc. Natl. Acad. Sci. USA 87, 8070–8074. Streisinger, G., Emrick, J. & Stahl, M.M. (1967) Chromosome structure in phage T4. III. T erminal redundancy and length determination. Proc. Natl. Acad. Sci. USA 57, 292–295. 544 Genes to Cells (1997) 2, 537–545 qBlackwell Science LimitedH Fujisawa & M Morita  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  T omka, M. & Catalano, C.E. (1993) Kinetic characterization of the ATPase activity of the DNA packaging enzyme from bacteriophage l.Biochemistry 32, 11992–11997. Travares, P ., Santos, M.A., Lurz, R., Morelli, G., de Lencastre, H. & Trautner, T.A. (1992) Identiﬁcation of a gene in Bacillus subtilis bacteriophage SPP1 determining the amount of packaged DNA. J. Mol. Biol. 225, 81–92. Tsuchida, S., Kokubo, H., Tasaka, M. & Fujisawa, H. (1996) DNA sequences responsible for speciﬁcity of DNA packaging and phage growth interference of bacteriophages T3 and T7. Virology 217, 332–337. Turnquist, S., Simon, M., Egelman, E. & Anderson, D. (1992) Supercoiled DNA wraps around the bacteriophage f-29 head-tail connector. Proc. Natl. Acad. Sci. USA 89, 10479– 10483. T ye, B.K., Huberman, J.A. & Botstein, D. (1974) Non-randomcircular permutation of phage P22 DNA. J. Mol. Biol. 85, 501–532. Valpuesta, J.M. & Carrascosa, J.L. (1994) Structure of viral connectors and their function in bactertiophage assembly andDNA packaging. Q. Rev. Biophys. 27, 107–155. Watson, J.D. (1972) Origin of concatemeric T7 DNA. Nature (London) New Biol. 23, 197–201. White, J.H. & Richardson, C.C. (1987) Processing of con- catemers of bacteriophage T7 DNA in vitro .J. Biol. Chem. 262, 8851–8860. Y eo, A. & Feiss, M. (1995) Speciﬁc interaction of terminase, the packaging enzyme of bacteriophage l, with the portal protein of the prohead. J. Mol. Biol. 245, 141–150. Zhang, X. & Studier, F .W . (1995) Isolation of transcriptionally active mutants of T7 RNA polymerase that do not supportphage growth. J. Mol. Biol. 250, 156–168. qBlackwell Science Limited Genes to Cells (1997) 2, 537–545 545Phage DNA packaging  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6160>, 'json_data': {'input': ['page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  Black, L.W . (1989) DNA packaging in dsDNA bacteriophage. Annu. Rev. Microbiol. 43, 267–292. Black, L.W . (1995) DNA packaging and cutting by phage terminases: control in phage T4 by a synaptic mechanism.BioEssays 17, 1025–1030. Black, L.W . & Silverman, D. (1978) Model for DNA packaging into bacteriophage T4 heads. J. Virol. 28, 643–655. Casjens, S., Huang, W .M., Hayden, M. & Parr, R. (1987) Initiation of bacteriophage P22 DNA packaging series. Analysis of a mutant that alters the DNA target speciﬁcity of the packaging apparatus. J. Mol. Biol. 194, 411–422. Casjens, S., Wyckoff, E., Hayden, M., et al. (1982) Bacteriophage P22 portal protein is part of the gauge that regulates packaging density of intravirion DNA. J. Mol. Biol. 224, 1055–1074. Catalano, C.E., Cue, D. & Feiss, M. (1995) Virus DNA packaging: the strategy used by phage l.Mol. Microbiol. 16, 1075–1086. Chung, Y .B. & Hinkle, D.C. (1990) Bacteriophage T7 DNA packaging. II. Analysis of the DNA sequences required forpackaging using a plasmid transduction assay. J. Mol. Biol. 216, 927–938. Cue, D. & Feiss, M. (1997) Genetic evidence that recognition ofcosQ, the signal for termination of phage lDNA packaging, depends on the extent of head ﬁlling. Genetics ,i n press. Earnshaw, W .C. & Casjens, S.R. (1980) DNA packaging by the double-stranded DNA bacteriophages. Cell21, 319–331. Feiss, M., Fisher, R.A., Crayton, M.A. & Egner, C. (1977) Packaging of bacteriophage lchromosome: effect of chromo- some length. Virology 77, 281–293. Fujisawa, H., Hamada, K., Shibata, H. & Minagawa, T. (1987) On the molecular mechanism of DNA translocation during in vitro packaging of bacteriophage T3 DNA. Virology 161, 228–233. Fujisawa, H., Kimura, M. & Hashimoto, C. (1990) In vitro cleavage of the concatemer joint of bacteriophage T3. Virology 174, 26–34. Fujisawa, H., Shibata, H. & Kato, H. (1991) Analysis of interactions among factors involved in the bacteriophage T3 DNA packaging reaction in a deﬁned in vitro system. Virology 185, 788–794. George, M. & Bukhari, A.I. (1981) Heterologous host DNA attached to the left end of mature bacteriophage Mu DNA. Nature 292, 175–176. Guo, P ., Erickson, S. & Anderson, D. (1987a) A small viral RNA is required for in vitro packaging of bacteriophage f-29 DNA- gp3.Science 236, 690–694. Guo, P ., Peterson, C. & Anderson, D. (1987b) Initiation events in in vitro packaging of bacteriophage f-29 DNA-gp3. J. Mol. Biol.197, 219–228. Hamada, K., Fujisawa, H. & Minagawa, T. (1986) A deﬁned in vitrosystem for packaging of bacteriophage T3 DNA. Virology 151, 119–123. Hashimoto, C. & Fujisawa, H.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  Nature 292, 175–176. Guo, P ., Erickson, S. & Anderson, D. (1987a) A small viral RNA is required for in vitro packaging of bacteriophage f-29 DNA- gp3.Science 236, 690–694. Guo, P ., Peterson, C. & Anderson, D. (1987b) Initiation events in in vitro packaging of bacteriophage f-29 DNA-gp3. J. Mol. Biol.197, 219–228. Hamada, K., Fujisawa, H. & Minagawa, T. (1986) A deﬁned in vitrosystem for packaging of bacteriophage T3 DNA. Virology 151, 119–123. Hashimoto, C. & Fujisawa, H. (1988) Packaging and transduc- tion of non-T3 DNA by bacteriophage T3. Virology 166, 788–796. Hashimoto, C. & Fujisawa, H. (1992a) DNA sequences necessary for packaging bacteriophage T3 DNA. Virology 187, 788–795. Hashimoto, C. & Fujisawa, H. (1992b) Transcription depen- dence of DNA packaging of bacteriophages T3 and T7.Virology 191, 246–250. Hendrix, R.W . (1978) Symmetry mismatch and DNA packagingin large bacteriophges. Proc. Natl. Acad. Sci. USA 75, 4779– 4783. Hsiao, C.L. & Black, L.W . (1977) DNA packaging and the pathway of bacteriophage T4 head assembly. Proc. Natl. Acad. Sci. USA. 74, 3552–3656. Hwang, Y ., Catalano, C. & Feiss, M. (1996) Kinetic and mutational dissection of the two ATP activities of terminase,the DNA packaging enzyme of bacteriophage l.Biochemistry 35, 2796–2803. Ikeda, H. & T omizawa, J.-I. (1965) Transducing fragments in generalized tranduction by P1. I. Molecular origin of thefragments. J. Mol. Biol. 14, 85–109. Lyakhov, D.L., He, B., Zhang, X., Studier, F .W ., Dunn, J.J. & McAllister, W .T. (1997) Mutant Bacteriophage T7 RNA polymerases with altered termination properties. J. Mol. Biol. 269, 28–40. Mellado, R.P ., Penalva, M.A., Inciarte, M.R. & Salas, M. (1980) The protein covalently linked to the 5 0-termini of the DNA of Bacillus subtilis phage f-29 is involved in the initiation of DNA replication. Virology 104, 84–96. Morita, M., Tasaka, M. & Fujisawa, H. (1993) DNA packaging ATPase of bacteriophage T3. Virology 193, 748–752. Morita, M., Tasaka, M. & Fujisawa, H. (1994) Analysis of functional domains of the packaging proteins of bacteriophage T3 by site-directed mutagebesis. J. Mol. Biol. 235, 248–259. Morita, M., Tasaka, M. & Fujisawa, H. (1995) Structural and functional domains of the DNA packaging protein ofbacteriophage T3: importance of the C-terminal region of the large subunit in prohead binding. J. Mol. Biol. 245, 635– 644. Morita, M., Tasaka, M. & Fujisawa, H. (1996) Analysis of the ﬁne structure of the prohead binding domain of the packaging protein of bacteriophage T3 using a hexapeptide, an analog ofa prohead binding site. Virology 211, 516–524.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  (1994) Analysis of functional domains of the packaging proteins of bacteriophage T3 by site-directed mutagebesis. J. Mol. Biol. 235, 248–259. Morita, M., Tasaka, M. & Fujisawa, H. (1995) Structural and functional domains of the DNA packaging protein ofbacteriophage T3: importance of the C-terminal region of the large subunit in prohead binding. J. Mol. Biol. 245, 635– 644. Morita, M., Tasaka, M. & Fujisawa, H. (1996) Analysis of the ﬁne structure of the prohead binding domain of the packaging protein of bacteriophage T3 using a hexapeptide, an analog ofa prohead binding site. Virology 211, 516–524. Mosig, G., Ghosal, D. & Bock, S. (1981) Interactions between the maturation protein gp17 and single stranded DNA binding protein gp32 initiate DNA packaging and compete withinitiation of secondary replication forks in phage T4. In:Bacteriophage Assembly (ed. M.S. Dubow), pp. 139–151, New Y ork: Alan R. Liss. Nakasu, S., Fujisawa, H. & Minagawa, T. (1983) Role of gene 8 product in morphogenesis of bacteriophage T3. Virology 127, 124–133. Serwer, P . (1988) The source of energy for bacteriophage DNA packaging: an osmotic pump explains the data. Biopolymers 27, 165–169. Shibata, H., Fujisawa, H. & Minagawa, T. (1987) Characteriza- tion of the bacteriophage T3 DNA packaging reaction in vitro in a deﬁned system. J. Mol. Biol. 196, 845–851. Sippy, J. & Feiss, M. (1992) Analysis of a mutation affecting the speciﬁcity domain for prohead binding of the bacteriophage l terminase. J. Bacteriol. 174, 850–856. Sternberg, N. & Coulby, J. (1987) Recognition and cleavage of the bacteriophage P1 packaging site ( pac) I. Differential processing of the cleaved ends in vivo .J. Mol. Biol. 194, 453– 468. Sternberg, N. & Coulby, J. (1990) Cleavage of the bacteriophage P1 packaging site ( pac) is regulated by adenine methylation. Proc. Natl. Acad. Sci. USA 87, 8070–8074. Streisinger, G., Emrick, J. & Stahl, M.M. (1967) Chromosome structure in phage T4. III. T erminal redundancy and length determination. Proc. Natl. Acad. Sci. USA 57, 292–295. 544 Genes to Cells (1997) 2, 537–545 qBlackwell Science LimitedH Fujisawa & M Morita  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/Genes to Cells - 2003 - Fujisawa - Phage DNA packaging.pdf  T omka, M. & Catalano, C.E. (1993) Kinetic characterization of the ATPase activity of the DNA packaging enzyme from bacteriophage l.Biochemistry 32, 11992–11997. Travares, P ., Santos, M.A., Lurz, R., Morelli, G., de Lencastre, H. & Trautner, T.A. (1992) Identiﬁcation of a gene in Bacillus subtilis bacteriophage SPP1 determining the amount of packaged DNA. J. Mol. Biol. 225, 81–92. Tsuchida, S., Kokubo, H., Tasaka, M. & Fujisawa, H. (1996) DNA sequences responsible for speciﬁcity of DNA packaging and phage growth interference of bacteriophages T3 and T7. Virology 217, 332–337. Turnquist, S., Simon, M., Egelman, E. & Anderson, D. (1992) Supercoiled DNA wraps around the bacteriophage f-29 head-tail connector. Proc. Natl. Acad. Sci. USA 89, 10479– 10483. T ye, B.K., Huberman, J.A. & Botstein, D. (1974) Non-randomcircular permutation of phage P22 DNA. J. Mol. Biol. 85, 501–532. Valpuesta, J.M. & Carrascosa, J.L. (1994) Structure of viral connectors and their function in bactertiophage assembly andDNA packaging. Q. Rev. Biophys. 27, 107–155. Watson, J.D. (1972) Origin of concatemeric T7 DNA. Nature (London) New Biol. 23, 197–201. White, J.H. & Richardson, C.C. (1987) Processing of con- catemers of bacteriophage T7 DNA in vitro .J. Biol. Chem. 262, 8851–8860. Y eo, A. & Feiss, M. (1995) Speciﬁc interaction of terminase, the packaging enzyme of bacteriophage l, with the portal protein of the prohead. J. Mol. Biol. 245, 141–150. Zhang, X. & Studier, F .W . (1995) Isolation of transcriptionally active mutants of T7 RNA polymerase that do not supportphage growth. J. Mol. Biol. 250, 156–168. qBlackwell Science Limited Genes to Cells (1997) 2, 537–545 545Phage DNA packaging  13652443, 1997, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1046/j.1365-2443.1997.1450343.x by National Health And Medical Research Council, Wiley Online Library on [05/01/2024]. See the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:55 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'50'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'997221'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'166ms'), (b'x-request-id', b'2b97c8f57f07e51f2db8da34b009e9c0'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412497a9fd75d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:55 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'50'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'997221'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'166ms'), (b'x-request-id', b'2b97c8f57f07e51f2db8da34b009e9c0'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412497a9fd75d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f7100>, 'json_data': {'input': ['page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 The Legacy of Nat Sternberg: The Genesis of Cre-loxTechnology Michael Yarmolinsky1and Ronald Hoess2 1Laboratory of Biochemistry and Molecular Biology, National Cancer Institute, National Institutes of Health, Bethesda, Maryland 20892; email: myarmo@helix.nih.gov 2Chadds Ford, Pennsylvania 19317; email: hoessr@earthlink.net Annu. Rev. Virol. 2015. 2:25–40 The Annual Review of Virology is online at virology.annualreviews.orgThis article’s doi: 10.1146/annurev-virology-100114-054930 Copyright c⃝2015 by Annual Reviews. All rights reservedKeywords bacteriophage P1, plasmid, site-speciﬁc recombination, chromosome segregation, genetic engineering Abstract Cre- loxof bacteriophage P1 has become one of the most widely used tools for genetic engineering in eukaryotes. The origins of this tool date to more than 30 years ago when Nat L. Sternberg discovered the recombinase, Cre, and its speciﬁc locus of crossover, lox, while studying the maintenance of bac- teriophage P1 as a stable plasmid. Recombinations mediated by Cre assistin cyclization of the DNA of infecting phage and in resolution of prophagemultimers created by generalized recombination. Early in vitro work demon- strated that, although it shares similarities with the well-characterized bac- teriophage λintegration, Cre-lox is in many ways far simpler in its require- ments for carrying out recombination. These features would prove critical for its development as a powerful and versatile tool in genetic engineering. We review the history of the discovery and characterization of Cre- loxand touch upon the present direction of Cre- loxresearch. 25 Click here to view this article's online features:  • Download ﬁgures as PPT slides• Navigate linked references• Download citations• Explore related articles• Search keywords ANNUAL  REVIEWS Further Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.\", 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 INTRODUCTION Biology has entered a new era, an era in which we can manipulate genes at will and do so even during the course of development. In the accomplishment of this technological feat, a major role has been played by site-speciﬁc recombination mediated by the recombinase, Cre, and speciﬁcDNA sites, lox. Here we narrate the story of the discovery and initial characterization of Cre- lox.W em i x science and our personal reminiscences of Nat L. Sternberg (1942–1995), the scientist whose insights and efforts made Cre- loxan invaluable genetic tool ( Figure 1 ). We were associated with Nat in different ways. M.Y. recruited Nat to work on bacteriophage P1, the organism that turned out to be the source of both Cre and lox. R.H. was recruited by Nat to study their interactions. FROM HERESY TO RESPECTABILITY Several historical circumstances delayed the discovery of temperate bacteriophages and made thestudy of the temperate bacteriophage P1 highly unlikely. The scientist-adventurer F ´elix d’Herelle, who, in 1917, gave the name bacteriophage to infectious particles responsible for killing bacteria, assumed that bacteriophages, whatever their nature, are invariably lethal. D’Herelle’s assumption that all phages are virulent (as opposed to temperate) relegated to obscurity a phenomenon dis- covered in 1925 known now as lysogeny. Lysogeny is the hereditary power of certain bacteria to perpetuate a noninfectious entity ( Anlage or prophage) able, on occasion, to produce bacterio- phages and release them by cell lysis (1). For a quarter of a century lysogeny was deemed artifactual, the fantasy of incompetent ex- perimentalists whose sloppy technique resulted in bacterial cultures that carried along a viral contamination that only occasionally resulted in a productive infection, a “carrier state.” Andr ´e Lwoff, in his 1965 Nobel lecture, stated the situation with characteristic wit: “In 1949, the new school of American virologists, to which virology owes so much, condemned lysogeny. ...Like those wisps of cloud that a breath of wind dispels, the problem was blown away from the templeof science and a smell of sulfur was left ﬂoating in the air. Lysogeny had become a heresy” (2). Figure 1 Nat L. Sternberg (1942–1995). The undated photograph was provided by his widow, Ilene Harfenist Sternberg. It was the only satisfactory image of her self-effacing husband that she had. 26 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 When, in 1949, Giuseppe Bertani joined the laboratory of Salvador Luria, a leader of the new school of American virologists, and asked to study lysogeny, Luria was initially reluctant to givehis blessing. Bertani attributed that reluctance to the wording of Luria’s grant (3), but lysogeny’s sulfurous smell must have had something to do with it, too. Whatever his misgivings, Luria did agree to Bertani’s request and obtained from Joshua and Esther Lederberg two lysogenic strains of Escherichia coli :K - 1 2( λ) and the strain Lisbonne-Carr `ere (Li). Bertani began studying both lysogens, but discontinued studying λlysogeny at the urging of the Lederbergs, as Esther Lederberg had published on λonly informally (in 1950). Bertani, along with the Lederbergs, was the ﬁrst to embark on the study of lysogeny on the American continent. E. coli Li had been known to yield phages of various plaque sizes. That these might represent different phages had not been anticipated. In 1951 Bertani showed that the different types of plaques were made by the growth of phages of three immunologically distinct types. He namedthem P1, P2, and P3. Understandably, Bertani concentrated his attention on P2, as it made plaques large enough to select mutant phages differing in plaque morphology. He did not entirely stop working on P1,although he did abandon P3. (Its resurrection in 1976 by Harvard Professor Nancy Kleckner was nipped in the bud by the Cambridge City Council, fearing the risk to her students of studying an organism requiring P3-level containment.) The P1 kcfrom which all laboratory strains of P1 have been derived is a mutant isolated by Bertani that grows on both E. coli Ca n d E. coli K-12. Like the original P1 (which was known to grow only on Shigella dysenteriae ), its plaques on E. coli are tiny. This feature made it an uncongenial phage to study. P1, like P3, might well have faded from view, but was saved from oblivion in 1954 by the initiative of Ed Lennox, a physicist turned biologist who had joined Luria’s lab. He asked whether Bertani’s phages could transduce (transfer genetic markers from a donor to a recipient host). The remarkable capacity of Salmonella phage P22 to transduce closely linked genetic markers had been demonstrated in 1951 by Norton Zinder and Joshua Lederberg. Lennox showed that P2 lacksthat capacity, but P1 has it in spades (4). P1 is capacious enough to transduce prophages such as λ along with associated genetic markers. P1 promptly became a workhorse of genetic manipulation in bacteria and the phage itself an object of interest (5, 6). The DNA extracted from plaque-forming particles, characterized in the 1960s by Hideo Ikeda and Jun-ichi Tomizawa, was shown to be double-stranded, linear, terminally redundant, and cyclically permuted (7). What controls the frequency with which an infecting phage reduces to a prophage and the sporadic entry of the prophage into a lytic cycle? One of the several biological insights in Jacob and Monod’s landmark paper of 1961 was the recognition that the vegetative viral functions are repressed in the prophage at the level of the gene by repressor proteins, and in the sameway that inducible enzymes are repressed by their cognate repressors (8). Phage mutants with thermosensitive repressors soon became the “wild type” in laboratories studying lysogeny. These repressors are responsible for the immunity of lysogens to superinfection by a phage of the identicalkind. Where in the cell are prophages located? Already in her informal 1950 report, Esther Lederberg indicated that λprophage was somehow linked to genetic markers of its host, but the nature of the linkage was unclear. Subsequent studies of bacterial sexuality in the early 1950s clariﬁed this linkage.', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  One of the several biological insights in Jacob and Monod’s landmark paper of 1961 was the recognition that the vegetative viral functions are repressed in the prophage at the level of the gene by repressor proteins, and in the sameway that inducible enzymes are repressed by their cognate repressors (8). Phage mutants with thermosensitive repressors soon became the “wild type” in laboratories studying lysogeny. These repressors are responsible for the immunity of lysogens to superinfection by a phage of the identicalkind. Where in the cell are prophages located? Already in her informal 1950 report, Esther Lederberg indicated that λprophage was somehow linked to genetic markers of its host, but the nature of the linkage was unclear. Subsequent studies of bacterial sexuality in the early 1950s clariﬁed this linkage. Lambda prophage was mapped to a unique chromosomal site and P2 prophage to morethan one, but P1 prophage resisted attempts at assigning it a chromosomal location. How does a prophage without a ﬁxed chromosomal address replicate? The possibility that P1 prophages are passively replicated by transient integration into the host chromosome was www.annualreviews.org •The Legacy of Nat Sternberg 27 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 considered, but was conclusively eliminated in 1968 (7). Unlikely as it seems, passive replication of a short DNA segment that transiently integrates into the E. coli chromosome at the terminus region was found to mimic maintenance as a plasmid (9). The responsible site-speciﬁc recombinase was probably XerC. P1 prophage is an independent replication unit: a plasmid, and, as it turned out, a plasmid of low copy number and high stability. These features imply that P1, as a prophage,possesses the wherewithal to control its copy number within narrow limits and to distribute at least one plasmid copy to each nascent daughter cell. No other bacteriophages were known at the time to possess these capabilities. The study of these fundamental processes is greatly facilitated by the existence of a simple procedure to amplify and purify the plasmid DNA. One has only to induce viral replication andextract the DNA from harvested bacteriophage particles. P1 seemed uniquely suited for studying the basic capabilities of plasmids. The phage, however, was largely unexplored. PARIS Nat Sternberg and I (M.Y.) arrived in Paris in the same year, 1970. Nat came on a postdoctoral fellowship to work in the laboratory of Franc ¸ois Gros in a newly established Centre National de la Recherche Scientiﬁque (CNRS) laboratory of molecular biology, I to set up my own laboratory on the ﬂoor below. The institute was located on the new campus of the Pierre and Marie CurieUniversity. I had expressed an interest in a position in Paris, some six years previously, when a postdoctoral fellow at the Pasteur Institute in the laboratory of Franc ¸ois Jacob; my request had not been forgotten. Nat and I had considerable similarities in our cultural and scientiﬁc backgrounds. Both of us had grown up in New York City and possessed a shared interest in phage-host relationships. Interactions between λand E. coli had been the subject of my own research in the United States both before my fellowship at the Pasteur Institute (escape synthesis, i.e., derepression of host genes mediated by prophage induction) and after (integration/excision). Nat had earned his doctoratein 1969 with Sewell Champe at Purdue, studying the head proteins of the virulent T-even phages, and in Paris began a study that introduced him to the temperate phage λ. Whereas my laboratory in Paris was spacious and tranquil, Nat found himself jockeying for workspace in a noisy, overcrowded room where sudden shortages of media were not uncommon. Undaunted, Nat embarked on an ambitious project to select and characterize E. coli mutants with mutations affecting the capacity to synthesize head proteins of λphage. He periodically reported on his progress in lucid presentations that casually displayed a remarkable productivity. When I asked Nat how he acquired his capacity for work with unﬂagging concentration, he referred to his years as a (reluctant) student in a rabbinical academy (yeshiva). In yeshiva, he said, you learnSitzﬂeisch , the stamina to persevere. Sitzﬂeisch may have been part of it, but Nat’s extraordinary talents also resided elsewhere. When, in 1972, Nat’s fellowship was coming to a close and he was preparing to return to the United States, he told me of his dissatisfaction with the opportunities available to him. About that time I learned that Robert Weisberg, whose thriving laboratory at the National Institutes of Health (NIH) was devoted to λ, had a post to ﬁll and was planning to choose from among candidates at an upcoming phage meeting. Conﬁdent that Weisberg could do no better than offer the post to Nat, I took the initiative to telephone him and persuade him to interview Nat. In 1972a long-distance call from overseas had gravitas. Nat, at his own expense, made the trip for the (successful) interview. For the next four years he held the position of staff fellow in the Weisberg lab, publishing a number of papers, ﬁve of them with a colleague in the lab, Lynn Enquist, thefuture founding editor of this journal. 28 Yarmolinsky ·Hoess Annu. Rev. Virol.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  About that time I learned that Robert Weisberg, whose thriving laboratory at the National Institutes of Health (NIH) was devoted to λ, had a post to ﬁll and was planning to choose from among candidates at an upcoming phage meeting. Conﬁdent that Weisberg could do no better than offer the post to Nat, I took the initiative to telephone him and persuade him to interview Nat. In 1972a long-distance call from overseas had gravitas. Nat, at his own expense, made the trip for the (successful) interview. For the next four years he held the position of staff fellow in the Weisberg lab, publishing a number of papers, ﬁve of them with a colleague in the lab, Lynn Enquist, thefuture founding editor of this journal. 28 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 In the same four-year period my group in Paris pursued genetic studies of P1 replication and immunity. Immunity maintenance being central to prophage stability, its study seemed a naturalplace to begin exploring the biology of P1. In September of 1974 I visited the United States and let it be known to certain colleagues at NIH that I would be interested in returning to the fold. Nat learned of my interest and wrote tome in October to express his interest in becoming a member of my laboratory, should I succeed in establishing one at NIH. A position for Nat in my lab became a top priority. FREDERICK In 1976, I (M.Y.) accepted an offer to direct the Molecular Genetics Section in a new Laboratoryof Molecular Biology in Frederick, Maryland, funded by the National Cancer Institute. The laboratory was located in Fort Detrick in space liberated by the closure of facilities for biologicalwarfare research. Dedicated lobbying by Harvard Professor Matthew Meselson was instrumental in that important decision. Some reminders remained of Fort Detrick’s sinister past, but there were compensations: considerable freedom and adequate funding. Nat Sternberg became the ﬁrst member of my group. He was of immeasurable practical help in setting up the lab prior to my arrival and the arrival of Stuart Austin (from my lab in Paris) and Robert Yuan (from Werner Arber’s lab in Basel). Nat was persuaded that P1 was, as he put it, “just ripe for the picking.” Nat’s ﬁrst contribution to P1 biology, made while still in Weisberg’s laboratory, was to construct a P1 library in a λvector, thus making P1 genes much more amenable to individual study (10). Lambda, Nat recognized,provided possibilities for DNA manipulation that were not obtainable with the small plasmid vectors with which, a few years earlier, the gene cloning revolution began. He took advantage of features of the λDNA packaging process to dissect the P1 DNA fragments. The P1 library proved useful in studying a variety of aspects of P1 biology, including replication, immunity, incompatibility, and site-speciﬁc recombination. It is this last topic that constitutes the subject ofhis ﬁrst publication on P1 from Frederick, in 1978. This article (of which Nat was sole author) is brief, but information-rich, and gives a foretaste of much that followed (11). In it he deﬁned a site-speciﬁc recombination as the explanation for anunexpected linearity of the recombination map of P1. As noted earlier, P1 phage particles contain DNA that is cyclically permuted and terminally redundant. Recombination between P1 phages was expected to produce a circular, not a linear, map, but previous reports by June Scott andby Donald and Jean Walker indicated that the map was, indeed, linear. Nat proposed that the simplest explanation for this result is that P1 DNA contains a hotspot for genetic recombination located in the region corresponding to the ends of the map. Consistent with this hypothesis, he observed efﬁcient recombination within a cloned small (6-kb) fragment of P1 DNA that, by marker rescue experiments, spans the ends of the genetic map. The recombination was detected as recombination between λmarkers ﬂanking the cloned DNA. Moreover, several other cloned P1 fragments, including some much larger, did not support efﬁcient recombination. Thus, the absence of genetic linkage between the ﬂanking markers could not be due to their separation by a large unmarked region. Nor could it be due to a nonspeciﬁc recombinase, because Nat showed the recombination to be independent of bacterial and λgenes essential for recombination. He postulated that the active fragment carried a site or sites at whicha site-speciﬁc recombinase acts, that the recombinase is probably of P1 origin, and, if so, that it must be encoded within the same 6-kb P1 DNA fragment. Nat named the site lox, for locus of crossover (x), simultaneously honoring a favorite traditional food of American Jewry. The choice of a name for the recombinase required a sense of its function. www.annualreviews.org •The Legacy of Nat Sternberg 29 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 The name did not appear in print until 1981, when Nat had isolated an amber mutation in the structural gene (12). The name Cre, on which Nat settled, is an abbreviation of cyclizationrecombinase and an anagram of Rec. It did not escape us that as lox, the smoked salmon, is commonly served over cream cheese, the proximity of the mutant cre amgene to lox, the DNA site, has a certain logic. In his 1978 publication on the P1 genetic map, Nat had already noted that experiments then in progress strongly suggested a role for lox-mediated recombination in cyclization of the phage DNA, an essential step in lysogenization as a plasmid. Host-mediated recombination between the terminally redundant regions (encompassing roughly 10% of the genome) can accomplish the task. If the duplicated regions in a signiﬁcant number of phage particles were to include lox,t h e n the DNA in those phages would be capable of cyclizing in the absence of the host recombination functions; in that case P1 should be able to lysogenize a recA E. coli at a modest but signiﬁcant frequency. That it can do so had been shown in 1971 by Toshiya Takano (13). Although P1 resembles λin possessing a site-speciﬁc recombinase and a cognate recombination site, the prophage is maintained as a plasmid in E. coli , rather than, like λ,a sp a r to ft h e E. coli chromosome. By application of appropriate selection pressure, one can isolate rare individualE. coli strains (including recA mutants deﬁcient in generalized recombination) in which P1 is carried as a stable integrated prophage. At the time that Nat began studying P1, such bacterial strains had recently been described and the recombination site or sites within P1 shown to be situated at the ends of the P1 genetic map (14). Nat suspected that integration had occurred by Cre-mediated recombination, albeit highly inefﬁcient, of cyclized P1 DNA into a bacterial pseudo- loxsite. He showed this to be the case and named the putative pseudo- loxsite loxB. He named the P1 loxsite loxPand the hybrid sites at the ends of the integrated DNA loxL and loxR. Comparisons of these sites for their efﬁcacy in recombination with a loxPsite revealed that P ×La n dP ×R are much more efﬁcient than P×B. Whereas these anomalous recombinations have no apparent signiﬁcance in the physiology of P1 itself, their study has proved to be of signiﬁcant value for understanding the sequence ofevents in the recombination process. The extensive terminal redundancy of P1 DNA in phage particles is a consequence of how the viral DNA is packaged from the linear polymer (concatemer) that is generated by rollingcircle replication. Packaging into phage heads begins at a deﬁned locus and proceeds for about four or ﬁve headfuls, each headful corresponding to about 1.1 genomes of DNA. The resulting redundant DNA ends in each phage particle enable cyclization in the infected cell to occur bybacterial recombination functions and make Cre-mediated recombination of duplicated loxsites largely superﬂuous. But Nat must have been correct in assigning cyclization as a signiﬁcant function of Cre. His subsequent studies revealed that the locus on the DNA from which packaging begins(about 5 kb from lox) and the direction of packaging (toward lox) are such that the ﬁrst DNA to be packaged in a phage head has a loxsite centrally located in the roughly 10-kb region of terminal redundancy (15, 16). Selection pressure for an assist to cyclization must have placed it there. Nat did not ignore other possible functions of Cre-mediated recombination and in his initial publication suggested a role in the formation of concatemers, the requisite substrate of head- ful packaging. Based on evidence that, under sufﬁcient selection pressure, P1 integrated into the E. coli chromosome can be selected (14), Nat further suggested a likely role of P ×B recombination in transducing particle generation.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f7100>, 'json_data': {'input': ['page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/NovelRoleForSiteSPecificRecombination.pdf', \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 The Legacy of Nat Sternberg: The Genesis of Cre-loxTechnology Michael Yarmolinsky1and Ronald Hoess2 1Laboratory of Biochemistry and Molecular Biology, National Cancer Institute, National Institutes of Health, Bethesda, Maryland 20892; email: myarmo@helix.nih.gov 2Chadds Ford, Pennsylvania 19317; email: hoessr@earthlink.net Annu. Rev. Virol. 2015. 2:25–40 The Annual Review of Virology is online at virology.annualreviews.orgThis article’s doi: 10.1146/annurev-virology-100114-054930 Copyright c⃝2015 by Annual Reviews. All rights reservedKeywords bacteriophage P1, plasmid, site-speciﬁc recombination, chromosome segregation, genetic engineering Abstract Cre- loxof bacteriophage P1 has become one of the most widely used tools for genetic engineering in eukaryotes. The origins of this tool date to more than 30 years ago when Nat L. Sternberg discovered the recombinase, Cre, and its speciﬁc locus of crossover, lox, while studying the maintenance of bac- teriophage P1 as a stable plasmid. Recombinations mediated by Cre assistin cyclization of the DNA of infecting phage and in resolution of prophagemultimers created by generalized recombination. Early in vitro work demon- strated that, although it shares similarities with the well-characterized bac- teriophage λintegration, Cre-lox is in many ways far simpler in its require- ments for carrying out recombination. These features would prove critical for its development as a powerful and versatile tool in genetic engineering. We review the history of the discovery and characterization of Cre- loxand touch upon the present direction of Cre- loxresearch. 25 Click here to view this article's online features:  • Download ﬁgures as PPT slides• Navigate linked references• Download citations• Explore related articles• Search keywords ANNUAL  REVIEWS Further Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.\", 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 INTRODUCTION Biology has entered a new era, an era in which we can manipulate genes at will and do so even during the course of development. In the accomplishment of this technological feat, a major role has been played by site-speciﬁc recombination mediated by the recombinase, Cre, and speciﬁcDNA sites, lox. Here we narrate the story of the discovery and initial characterization of Cre- lox.W em i x science and our personal reminiscences of Nat L. Sternberg (1942–1995), the scientist whose insights and efforts made Cre- loxan invaluable genetic tool ( Figure 1 ). We were associated with Nat in different ways. M.Y. recruited Nat to work on bacteriophage P1, the organism that turned out to be the source of both Cre and lox. R.H. was recruited by Nat to study their interactions. FROM HERESY TO RESPECTABILITY Several historical circumstances delayed the discovery of temperate bacteriophages and made thestudy of the temperate bacteriophage P1 highly unlikely. The scientist-adventurer F ´elix d’Herelle, who, in 1917, gave the name bacteriophage to infectious particles responsible for killing bacteria, assumed that bacteriophages, whatever their nature, are invariably lethal. D’Herelle’s assumption that all phages are virulent (as opposed to temperate) relegated to obscurity a phenomenon dis- covered in 1925 known now as lysogeny. Lysogeny is the hereditary power of certain bacteria to perpetuate a noninfectious entity ( Anlage or prophage) able, on occasion, to produce bacterio- phages and release them by cell lysis (1). For a quarter of a century lysogeny was deemed artifactual, the fantasy of incompetent ex- perimentalists whose sloppy technique resulted in bacterial cultures that carried along a viral contamination that only occasionally resulted in a productive infection, a “carrier state.” Andr ´e Lwoff, in his 1965 Nobel lecture, stated the situation with characteristic wit: “In 1949, the new school of American virologists, to which virology owes so much, condemned lysogeny. ...Like those wisps of cloud that a breath of wind dispels, the problem was blown away from the templeof science and a smell of sulfur was left ﬂoating in the air. Lysogeny had become a heresy” (2). Figure 1 Nat L. Sternberg (1942–1995). The undated photograph was provided by his widow, Ilene Harfenist Sternberg. It was the only satisfactory image of her self-effacing husband that she had. 26 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 When, in 1949, Giuseppe Bertani joined the laboratory of Salvador Luria, a leader of the new school of American virologists, and asked to study lysogeny, Luria was initially reluctant to givehis blessing. Bertani attributed that reluctance to the wording of Luria’s grant (3), but lysogeny’s sulfurous smell must have had something to do with it, too. Whatever his misgivings, Luria did agree to Bertani’s request and obtained from Joshua and Esther Lederberg two lysogenic strains of Escherichia coli :K - 1 2( λ) and the strain Lisbonne-Carr `ere (Li). Bertani began studying both lysogens, but discontinued studying λlysogeny at the urging of the Lederbergs, as Esther Lederberg had published on λonly informally (in 1950). Bertani, along with the Lederbergs, was the ﬁrst to embark on the study of lysogeny on the American continent. E. coli Li had been known to yield phages of various plaque sizes. That these might represent different phages had not been anticipated. In 1951 Bertani showed that the different types of plaques were made by the growth of phages of three immunologically distinct types. He namedthem P1, P2, and P3. Understandably, Bertani concentrated his attention on P2, as it made plaques large enough to select mutant phages differing in plaque morphology. He did not entirely stop working on P1,although he did abandon P3. (Its resurrection in 1976 by Harvard Professor Nancy Kleckner was nipped in the bud by the Cambridge City Council, fearing the risk to her students of studying an organism requiring P3-level containment.) The P1 kcfrom which all laboratory strains of P1 have been derived is a mutant isolated by Bertani that grows on both E. coli Ca n d E. coli K-12. Like the original P1 (which was known to grow only on Shigella dysenteriae ), its plaques on E. coli are tiny. This feature made it an uncongenial phage to study. P1, like P3, might well have faded from view, but was saved from oblivion in 1954 by the initiative of Ed Lennox, a physicist turned biologist who had joined Luria’s lab. He asked whether Bertani’s phages could transduce (transfer genetic markers from a donor to a recipient host). The remarkable capacity of Salmonella phage P22 to transduce closely linked genetic markers had been demonstrated in 1951 by Norton Zinder and Joshua Lederberg. Lennox showed that P2 lacksthat capacity, but P1 has it in spades (4). P1 is capacious enough to transduce prophages such as λ along with associated genetic markers. P1 promptly became a workhorse of genetic manipulation in bacteria and the phage itself an object of interest (5, 6). The DNA extracted from plaque-forming particles, characterized in the 1960s by Hideo Ikeda and Jun-ichi Tomizawa, was shown to be double-stranded, linear, terminally redundant, and cyclically permuted (7). What controls the frequency with which an infecting phage reduces to a prophage and the sporadic entry of the prophage into a lytic cycle? One of the several biological insights in Jacob and Monod’s landmark paper of 1961 was the recognition that the vegetative viral functions are repressed in the prophage at the level of the gene by repressor proteins, and in the sameway that inducible enzymes are repressed by their cognate repressors (8). Phage mutants with thermosensitive repressors soon became the “wild type” in laboratories studying lysogeny. These repressors are responsible for the immunity of lysogens to superinfection by a phage of the identicalkind. Where in the cell are prophages located? Already in her informal 1950 report, Esther Lederberg indicated that λprophage was somehow linked to genetic markers of its host, but the nature of the linkage was unclear. Subsequent studies of bacterial sexuality in the early 1950s clariﬁed this linkage.', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  One of the several biological insights in Jacob and Monod’s landmark paper of 1961 was the recognition that the vegetative viral functions are repressed in the prophage at the level of the gene by repressor proteins, and in the sameway that inducible enzymes are repressed by their cognate repressors (8). Phage mutants with thermosensitive repressors soon became the “wild type” in laboratories studying lysogeny. These repressors are responsible for the immunity of lysogens to superinfection by a phage of the identicalkind. Where in the cell are prophages located? Already in her informal 1950 report, Esther Lederberg indicated that λprophage was somehow linked to genetic markers of its host, but the nature of the linkage was unclear. Subsequent studies of bacterial sexuality in the early 1950s clariﬁed this linkage. Lambda prophage was mapped to a unique chromosomal site and P2 prophage to morethan one, but P1 prophage resisted attempts at assigning it a chromosomal location. How does a prophage without a ﬁxed chromosomal address replicate? The possibility that P1 prophages are passively replicated by transient integration into the host chromosome was www.annualreviews.org •The Legacy of Nat Sternberg 27 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 considered, but was conclusively eliminated in 1968 (7). Unlikely as it seems, passive replication of a short DNA segment that transiently integrates into the E. coli chromosome at the terminus region was found to mimic maintenance as a plasmid (9). The responsible site-speciﬁc recombinase was probably XerC. P1 prophage is an independent replication unit: a plasmid, and, as it turned out, a plasmid of low copy number and high stability. These features imply that P1, as a prophage,possesses the wherewithal to control its copy number within narrow limits and to distribute at least one plasmid copy to each nascent daughter cell. No other bacteriophages were known at the time to possess these capabilities. The study of these fundamental processes is greatly facilitated by the existence of a simple procedure to amplify and purify the plasmid DNA. One has only to induce viral replication andextract the DNA from harvested bacteriophage particles. P1 seemed uniquely suited for studying the basic capabilities of plasmids. The phage, however, was largely unexplored. PARIS Nat Sternberg and I (M.Y.) arrived in Paris in the same year, 1970. Nat came on a postdoctoral fellowship to work in the laboratory of Franc ¸ois Gros in a newly established Centre National de la Recherche Scientiﬁque (CNRS) laboratory of molecular biology, I to set up my own laboratory on the ﬂoor below. The institute was located on the new campus of the Pierre and Marie CurieUniversity. I had expressed an interest in a position in Paris, some six years previously, when a postdoctoral fellow at the Pasteur Institute in the laboratory of Franc ¸ois Jacob; my request had not been forgotten. Nat and I had considerable similarities in our cultural and scientiﬁc backgrounds. Both of us had grown up in New York City and possessed a shared interest in phage-host relationships. Interactions between λand E. coli had been the subject of my own research in the United States both before my fellowship at the Pasteur Institute (escape synthesis, i.e., derepression of host genes mediated by prophage induction) and after (integration/excision). Nat had earned his doctoratein 1969 with Sewell Champe at Purdue, studying the head proteins of the virulent T-even phages, and in Paris began a study that introduced him to the temperate phage λ. Whereas my laboratory in Paris was spacious and tranquil, Nat found himself jockeying for workspace in a noisy, overcrowded room where sudden shortages of media were not uncommon. Undaunted, Nat embarked on an ambitious project to select and characterize E. coli mutants with mutations affecting the capacity to synthesize head proteins of λphage. He periodically reported on his progress in lucid presentations that casually displayed a remarkable productivity. When I asked Nat how he acquired his capacity for work with unﬂagging concentration, he referred to his years as a (reluctant) student in a rabbinical academy (yeshiva). In yeshiva, he said, you learnSitzﬂeisch , the stamina to persevere. Sitzﬂeisch may have been part of it, but Nat’s extraordinary talents also resided elsewhere. When, in 1972, Nat’s fellowship was coming to a close and he was preparing to return to the United States, he told me of his dissatisfaction with the opportunities available to him. About that time I learned that Robert Weisberg, whose thriving laboratory at the National Institutes of Health (NIH) was devoted to λ, had a post to ﬁll and was planning to choose from among candidates at an upcoming phage meeting. Conﬁdent that Weisberg could do no better than offer the post to Nat, I took the initiative to telephone him and persuade him to interview Nat. In 1972a long-distance call from overseas had gravitas. Nat, at his own expense, made the trip for the (successful) interview. For the next four years he held the position of staff fellow in the Weisberg lab, publishing a number of papers, ﬁve of them with a colleague in the lab, Lynn Enquist, thefuture founding editor of this journal. 28 Yarmolinsky ·Hoess Annu. Rev. Virol.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  About that time I learned that Robert Weisberg, whose thriving laboratory at the National Institutes of Health (NIH) was devoted to λ, had a post to ﬁll and was planning to choose from among candidates at an upcoming phage meeting. Conﬁdent that Weisberg could do no better than offer the post to Nat, I took the initiative to telephone him and persuade him to interview Nat. In 1972a long-distance call from overseas had gravitas. Nat, at his own expense, made the trip for the (successful) interview. For the next four years he held the position of staff fellow in the Weisberg lab, publishing a number of papers, ﬁve of them with a colleague in the lab, Lynn Enquist, thefuture founding editor of this journal. 28 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 In the same four-year period my group in Paris pursued genetic studies of P1 replication and immunity. Immunity maintenance being central to prophage stability, its study seemed a naturalplace to begin exploring the biology of P1. In September of 1974 I visited the United States and let it be known to certain colleagues at NIH that I would be interested in returning to the fold. Nat learned of my interest and wrote tome in October to express his interest in becoming a member of my laboratory, should I succeed in establishing one at NIH. A position for Nat in my lab became a top priority. FREDERICK In 1976, I (M.Y.) accepted an offer to direct the Molecular Genetics Section in a new Laboratoryof Molecular Biology in Frederick, Maryland, funded by the National Cancer Institute. The laboratory was located in Fort Detrick in space liberated by the closure of facilities for biologicalwarfare research. Dedicated lobbying by Harvard Professor Matthew Meselson was instrumental in that important decision. Some reminders remained of Fort Detrick’s sinister past, but there were compensations: considerable freedom and adequate funding. Nat Sternberg became the ﬁrst member of my group. He was of immeasurable practical help in setting up the lab prior to my arrival and the arrival of Stuart Austin (from my lab in Paris) and Robert Yuan (from Werner Arber’s lab in Basel). Nat was persuaded that P1 was, as he put it, “just ripe for the picking.” Nat’s ﬁrst contribution to P1 biology, made while still in Weisberg’s laboratory, was to construct a P1 library in a λvector, thus making P1 genes much more amenable to individual study (10). Lambda, Nat recognized,provided possibilities for DNA manipulation that were not obtainable with the small plasmid vectors with which, a few years earlier, the gene cloning revolution began. He took advantage of features of the λDNA packaging process to dissect the P1 DNA fragments. The P1 library proved useful in studying a variety of aspects of P1 biology, including replication, immunity, incompatibility, and site-speciﬁc recombination. It is this last topic that constitutes the subject ofhis ﬁrst publication on P1 from Frederick, in 1978. This article (of which Nat was sole author) is brief, but information-rich, and gives a foretaste of much that followed (11). In it he deﬁned a site-speciﬁc recombination as the explanation for anunexpected linearity of the recombination map of P1. As noted earlier, P1 phage particles contain DNA that is cyclically permuted and terminally redundant. Recombination between P1 phages was expected to produce a circular, not a linear, map, but previous reports by June Scott andby Donald and Jean Walker indicated that the map was, indeed, linear. Nat proposed that the simplest explanation for this result is that P1 DNA contains a hotspot for genetic recombination located in the region corresponding to the ends of the map. Consistent with this hypothesis, he observed efﬁcient recombination within a cloned small (6-kb) fragment of P1 DNA that, by marker rescue experiments, spans the ends of the genetic map. The recombination was detected as recombination between λmarkers ﬂanking the cloned DNA. Moreover, several other cloned P1 fragments, including some much larger, did not support efﬁcient recombination. Thus, the absence of genetic linkage between the ﬂanking markers could not be due to their separation by a large unmarked region. Nor could it be due to a nonspeciﬁc recombinase, because Nat showed the recombination to be independent of bacterial and λgenes essential for recombination. He postulated that the active fragment carried a site or sites at whicha site-speciﬁc recombinase acts, that the recombinase is probably of P1 origin, and, if so, that it must be encoded within the same 6-kb P1 DNA fragment. Nat named the site lox, for locus of crossover (x), simultaneously honoring a favorite traditional food of American Jewry. The choice of a name for the recombinase required a sense of its function. www.annualreviews.org •The Legacy of Nat Sternberg 29 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 The name did not appear in print until 1981, when Nat had isolated an amber mutation in the structural gene (12). The name Cre, on which Nat settled, is an abbreviation of cyclizationrecombinase and an anagram of Rec. It did not escape us that as lox, the smoked salmon, is commonly served over cream cheese, the proximity of the mutant cre amgene to lox, the DNA site, has a certain logic. In his 1978 publication on the P1 genetic map, Nat had already noted that experiments then in progress strongly suggested a role for lox-mediated recombination in cyclization of the phage DNA, an essential step in lysogenization as a plasmid. Host-mediated recombination between the terminally redundant regions (encompassing roughly 10% of the genome) can accomplish the task. If the duplicated regions in a signiﬁcant number of phage particles were to include lox,t h e n the DNA in those phages would be capable of cyclizing in the absence of the host recombination functions; in that case P1 should be able to lysogenize a recA E. coli at a modest but signiﬁcant frequency. That it can do so had been shown in 1971 by Toshiya Takano (13). Although P1 resembles λin possessing a site-speciﬁc recombinase and a cognate recombination site, the prophage is maintained as a plasmid in E. coli , rather than, like λ,a sp a r to ft h e E. coli chromosome. By application of appropriate selection pressure, one can isolate rare individualE. coli strains (including recA mutants deﬁcient in generalized recombination) in which P1 is carried as a stable integrated prophage. At the time that Nat began studying P1, such bacterial strains had recently been described and the recombination site or sites within P1 shown to be situated at the ends of the P1 genetic map (14). Nat suspected that integration had occurred by Cre-mediated recombination, albeit highly inefﬁcient, of cyclized P1 DNA into a bacterial pseudo- loxsite. He showed this to be the case and named the putative pseudo- loxsite loxB. He named the P1 loxsite loxPand the hybrid sites at the ends of the integrated DNA loxL and loxR. Comparisons of these sites for their efﬁcacy in recombination with a loxPsite revealed that P ×La n dP ×R are much more efﬁcient than P×B. Whereas these anomalous recombinations have no apparent signiﬁcance in the physiology of P1 itself, their study has proved to be of signiﬁcant value for understanding the sequence ofevents in the recombination process. The extensive terminal redundancy of P1 DNA in phage particles is a consequence of how the viral DNA is packaged from the linear polymer (concatemer) that is generated by rollingcircle replication. Packaging into phage heads begins at a deﬁned locus and proceeds for about four or ﬁve headfuls, each headful corresponding to about 1.1 genomes of DNA. The resulting redundant DNA ends in each phage particle enable cyclization in the infected cell to occur bybacterial recombination functions and make Cre-mediated recombination of duplicated loxsites largely superﬂuous. But Nat must have been correct in assigning cyclization as a signiﬁcant function of Cre. His subsequent studies revealed that the locus on the DNA from which packaging begins(about 5 kb from lox) and the direction of packaging (toward lox) are such that the ﬁrst DNA to be packaged in a phage head has a loxsite centrally located in the roughly 10-kb region of terminal redundancy (15, 16). Selection pressure for an assist to cyclization must have placed it there. Nat did not ignore other possible functions of Cre-mediated recombination and in his initial publication suggested a role in the formation of concatemers, the requisite substrate of head- ful packaging. Based on evidence that, under sufﬁcient selection pressure, P1 integrated into the E. coli chromosome can be selected (14), Nat further suggested a likely role of P ×B recombination in transducing particle generation.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:56 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'88'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993980'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'361ms'), (b'x-request-id', b'3910c0b78365c44e013895be8c9448ca'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412497f6c315d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:56 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'88'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993980'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'361ms'), (b'x-request-id', b'3910c0b78365c44e013895be8c9448ca'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412497f6c315d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  His subsequent studies revealed that the locus on the DNA from which packaging begins(about 5 kb from lox) and the direction of packaging (toward lox) are such that the ﬁrst DNA to be packaged in a phage head has a loxsite centrally located in the roughly 10-kb region of terminal redundancy (15, 16). Selection pressure for an assist to cyclization must have placed it there. Nat did not ignore other possible functions of Cre-mediated recombination and in his initial publication suggested a role in the formation of concatemers, the requisite substrate of head- ful packaging. Based on evidence that, under sufﬁcient selection pressure, P1 integrated into the E. coli chromosome can be selected (14), Nat further suggested a likely role of P ×B recombination in transducing particle generation. Whereas concatemer formation is important in phage produc-tion, concatemer resolution to monomers is important for plasmid maintenance. P1 prophage is stable despite its low copy number because of an active segregation mechanism. If the units to be segregated are covalently joined, plasmid loss at cell division will ensue. In 1981, Stuart Austin,Marcia Ziese, and Nat Sternberg documented this additional role for Cre- lox(17). 30 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 My (R.H.) introduction to Nat Sternberg came on a bleak November day in 1978. I was visiting the Frederick Cancer Research Center to interview for a position. The barrack-like buildingsconnected by elevated steam pipes and the six-story bricked-up building referred to as the anthrax tower added to the gloominess of the place. My last interview that day was with Nat. In the laboratory that he shared with Michael Yarmolinsky were crammed four shaking water baths,buckets bristling with dirty pipettes, and stacks of Petri dishes. Nat pulled up a couple of chairs for us, grabbed a legal pad, and proceeded to describe his projects. As I discovered then, talking and simultaneously drawing was Nat’s expository style. Describing his work on the P1 repressor, he quickly ﬁlled the page, tore it off the pad, and wrote on the other side before using the next page. Near the end of the half hour he began describing the remarkable progress he had made on the site-speciﬁc recombination mediated by Cre- loxthat he had found in P1. Having depleted the pad, Nat continued writing and drawing on both sides of the cardboard backing ( Figure 2 ). It particularly interested me because I was just ﬁnishing a postdoctoral fellowship in the laboratory of ArthurLandy, studying the λattachment site and the genes involved in λsite-speciﬁc recombination. Figure 2 The back of the legal pad on which Nat Sternberg wrote during the interview with R.H., November 16, 1978. www.annualreviews.org •The Legacy of Nat Sternberg 31 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 When I accepted the offer to work at Frederick I had planned to continue working on λ, but not long after I arrived in the spring of 1979, Nat asked me rather casually whether I might beinterested in helping out on the Cre- loxproject. This was an opportunity too good to pass up, and within a few months nearly all my attention was directed at lox. Many of the features that have made Cre- loxa useful tool are best appreciated by comparison with what I will refer to as the λparadigm for site-speciﬁc recombination. Lambda Int- atthad been studied for over a decade prior to the discovery of Cre -lox, ﬁrst by extensive genetic analysis, then, following development of an in vitro reaction, by biochemistry (18). By the late 1970s, many of the protein components had been puriﬁed and the sites of recombination (attachment sites, viral and bacterial) had been sequenced. The picture of recombination that emerged was complex. The recombinase (Int) anda number of accessory proteins ﬁrst had to assemble into a nucleoprotein structure, which would then capture the bacterial attachment site. How recombination then took place was unclear, but a four-stranded, mobile (Holliday) junction seemed a likely intermediate. At that time λInt-att represented the best-characterized site-speciﬁc recombination. Not surprisingly, Nat and I, along with Kenneth Abremski, who would later join us, were inﬂuenced by the λparadigm, because each of us had worked on λrecombination. It inﬂuenced the design of our experiments and our expectations. Thus we were excited, even at the earliest stage, to uncover differences between Cre-loxand Int- att. More importantly, many of these differences would later translate into making Cre- loxthe versatile genetic engineering tool that it is today. By the end of 1979 we had determined the sequences of the various loxsites (19). The most striking feature was the presence of perfect 13-bp inverted repeats separated by an 8-bp spacer encompassing the crossover point. We supposed that the inverted repeats were potential bindingsites for Cre. The minimal size of a functional loxsite remained to be determined, but we assumed, by analogy with the λattachment site, that additional DNA sequences beyond the inverted repeats were likely to be required for recombination. In order to deﬁne the loxP site boundaries more precisely, we moved the putative loxP- containing region to a small plasmid convenient for further manipulation. It carried an ampicillin-resistance marker. Nat devised a simple pickup assay for testing loxfunction. The plasmid carrying theloxsite to be tested was introduced into an E. coli strain expressing a functional cregene. These cells were then infected with a λcontaining a functional loxPsite. If Cre-mediated recombination could occur between the two loxsites, then the resulting lysate should be able to transduce the ampicillin-resistance marker of the plasmid with high frequency (19). In the original assay, the region of several hundred base pairs of P1 DNA ﬂanking the functional loxPsite in one partner of the recombination made it impossible to exclude its role in the process. Further experiments revealed that a plasmid containing two loxP sites, each less than 60 bp in length, could undergo efﬁcient Cre-mediated recombination (20). Thus, unlike λ’s attachment site of about 240 bp, the loxPsite is quite small and therefore easily manipulated. Early in 1981 Nat received permission to form his own group. However, because Yarmolinsky’s group continued to study aspects of P1 biology and prokaryote research was deemed of onlymarginal relevance to cancer research, Nat was informed that he would have to switch his efforts to eukaryotes. For Nat, whose experience was entirely in the study of bacteria and their viruses, this was a daunting challenge. He accepted it, but had no intention of giving up his work on P1. Nat forged ahead on two fronts. He began studying recombination in mouse L cells, and he continued studying various aspects of P1 biology.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  Thus, unlike λ’s attachment site of about 240 bp, the loxPsite is quite small and therefore easily manipulated. Early in 1981 Nat received permission to form his own group. However, because Yarmolinsky’s group continued to study aspects of P1 biology and prokaryote research was deemed of onlymarginal relevance to cancer research, Nat was informed that he would have to switch his efforts to eukaryotes. For Nat, whose experience was entirely in the study of bacteria and their viruses, this was a daunting challenge. He accepted it, but had no intention of giving up his work on P1. Nat forged ahead on two fronts. He began studying recombination in mouse L cells, and he continued studying various aspects of P1 biology. Nat’s ﬁrst hire was Ken Abremski, who had just ﬁnished postdoctoral studies at NIH on the λ protein Xis, which is required in conjunction with Int for excision of λprophage from the bacterial chromosome (18, 21). Within months of his arrival, Ken developed an in vitro Cre-dependent recombination. One of the ﬁrst tests of the in vitro reaction used two loxP- containing λsubstrates 32 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 that Nat had constructed. Each substrate contained two loxPsites, one as direct repeats and the other as inverted repeats. Both substrates recombined efﬁciently in vivo and in vitro (20). Thiswas in marked contrast to λInt, which could recombine only directly repeated sites, and to the resolvases of certain of the small mobile genetic elements, such as that of γδ, which are able to catalyze only inversions (22). Expression vectors were constructed that provided us with ample quantities of Cre protein. Ken made rapid progress in purifying Cre and determining the requirements for the recombination reaction. At almost every turn we found some difference when compared with λInt recombina- tion. Unlike λInt recombination, no external energy source or additional accessory proteins were required for the reaction (23). Additionally, the different forms of the DNA substrate containingloxPsites, whether supercoiled, relaxed circle, or linear molecules, were all equally efﬁcient sub- strates. Given the minimal requirements of the in vitro reaction along with the robustness of the recombination, working with Cre was as simple as doing a restriction enzyme digest. WILMINGTON (DUPONT) During our time in Frederick we were ofﬁcially employed by Litton Industries, which ran our part of Fort Detrick under contract for the National Cancer Institute. This arrangement enabled the National Cancer Institute to respond rapidly to changing trends in cancer research, and aspart of a prokaryotic research group, we were aware of our vulnerability when the contract came up for renewal in 1984. For Nat and the members of his burgeoning laboratory, rescue came via DuPont. A relatively new member of the Frederick Cancer Research Center, Mark Pearson,was offered a position in DuPont’s Central Research and Development Department with the mandate to create a world-class molecular biology laboratory. Building on a well-established group of animal virologists, molecular geneticists, and chemists, Pearson rapidly assembled a department that included Nat’s entire group as well as biophysicists with expertise in molecular structure. Nat’s group moved to DuPont in the spring of 1984. Unlike the dreary environs of Frederick, we found ourselves on a tree-lined campus with spacious, newly renovated labs. Funding for equipment and supplies was never an issue. By this time Ken and I were absorbed in working outthe molecular details of Cre recombination. With puriﬁed Cre we were able to show by DNA footprinting that Cre does not contact DNA outside the 34 bp of the loxP site (24). Because Cre is able to carry out either excision or inversion depending on the orientation of the loxP sites, the footprinting results imply that the 8-bp spacer region imparts the directionality to the site. We demonstrated in vitro that Cre generates a 6-bp staggered cut within the spacer region (25) and made single-base substitutions within the spacer region to determine how theyaffected recombination. The results showed that homology of the spacer regions is required for recombination (26). Later these ﬁndings were exploited in genetic engineering with Cre and lox. Multiple loxsites, each addressable by the sequence of the spacer region, could be introduced into a cell, and only like sites would recombine with each other (27). The products of recombination between directly repeated loxsites on a highly supercoiled substrate were found not to be entangled with each other, as might be expected. Instead, recom- bination yielded supercoiled molecules entirely free of each other or linked as simple catenanes, rings that are topologically, but not chemically, linked (20, 28). This result suggested that beforethe initiation of recombination, the recombination sites come together in a way that physically separates the domains between them, as might occur if the sites are brought together via a slid- ing of DNA upon itself. Note that the rings of a DNA catenane are readily unlinked by cellulartopoisomerases. www.annualreviews.org •The Legacy of Nat Sternberg 33 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 One of Nat’s early papers on Cre- loxlooked at strand exchange during recombination of lox sites (29). A series of λ-P1 hybrid phages carrying loxPorloxRwere crossed and the reassortment of outside markers scored. From these results it was noted that branch migration had frequently occurred following an initial crossover. More direct physical evidence would come later when we began isolating and characterizing various mutants of Cre. One particular mutant appearedpartially deﬁcient for recombination in vitro (30). When the products of the reaction were analyzed it was shown that Cre preferentially exchanged one set of strands but only partially exchanged the second set of strands, thereby generating Holliday structure intermediates among the products. To rule out that the Holliday intermediates were aberrant end products of the reaction, they were puriﬁed and shown to be resolved on further addition of Cre, thus demonstrating they wereintermediates in the reaction. In addition to the biochemical dissection of the Cre- loxinteraction, the sequence of Cre itself was determined (31). Not long thereafter, an effort was made to align the sequence of Cre andother known recombinases, including Int of λ, P2, P22, P4, ϕ80, and ϕ186 (32). Despite the enormous sequence diversity among these recombinases, alignment by their C termini indicated a few small regions of homology with three residues (His, Arg, and Tyr) strictly conserved among allmembers. These residues were hypothesized to be involved in the catalysis of strand cleavage by the recombinases. A short time later a fourth conserved residue (Arg) was also identiﬁed (33). Three of these residues—Arg, His, and Arg—were shown by X-ray crystallography to form hydrogenbonds to the phosphate to which the conserved Tyr residue becomes transiently covalently bonded during cleavage, thus preserving the energy of the phosphodiester bond during the exchange (34). Over the years more than 100 potential recombinases have been identiﬁed by these alignments,indicating that Cre is a member of an enormously divergent recombinase family. Nat was particularly interested in the regulation of Cre expression and how it is modulated in relation to the P1 life cycle. Three promoters of varying strengths were identiﬁed upstream of Cre (31). One of these promoters contains Dam methylation sites, and experiments showed that in the absence of Dam methylation Cre expression increases. This leads to the idea that uponinfection, before newly replicated P1 DNA could be fully methylated, Cre expression would be elevated precisely when needed to cyclize the linear viral DNA. Cre expression is also expected to be elevated during lytic growth, whether following infection or following induction of an established prophage. This elevated expression could, in principle, increase the burst size of phage particles by recruiting those P1 plasmids that have not initiated replication into P1 plasmids that are in the process of replication (35), or it could do so simply byforming concatemers, the substrate for DNA packaging into phage heads (11). P1 concatemers are formed during lytic growth by rolling circle ( σ) replication, which, during the course of infection, takes over from bidirectional ( θ) replication (36). Whereas concatemers are important for producing viral particles, they (as well as circular oligomers) are disastrous for plasmid maintenance, because they are unpartitionable units. The problem for segregation created by circular chromosome dimers generated by interchromosomerecombination was recognized as early as 1932 by Barbara McClintock [cited by Barre et al. (37)]. As Barre et al. observe, the problem appears to have been largely ignored prior to work on P1 segregation (17). During lysogenic growth of P1, plasmid oligomers are formed occasionally by RecA-mediated homologous recombination. Their resolution prior to cell division, a function of Cre- lox, is essential for prophage stability (17, 38). Ironically, problems arising from recombina- tion events that are infrequent with respect to the cell cycle are remedied by raising the frequency of those events. Raising the recombination frequency without introducing new problems is ac- complished by limiting the site of recombination to a highly localized region.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  The problem for segregation created by circular chromosome dimers generated by interchromosomerecombination was recognized as early as 1932 by Barbara McClintock [cited by Barre et al. (37)]. As Barre et al. observe, the problem appears to have been largely ignored prior to work on P1 segregation (17). During lysogenic growth of P1, plasmid oligomers are formed occasionally by RecA-mediated homologous recombination. Their resolution prior to cell division, a function of Cre- lox, is essential for prophage stability (17, 38). Ironically, problems arising from recombina- tion events that are infrequent with respect to the cell cycle are remedied by raising the frequency of those events. Raising the recombination frequency without introducing new problems is ac- complished by limiting the site of recombination to a highly localized region. 34 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 The Int of λand Cre of P1 are similar in having more than a single role in the life of their respective phages, but they differ in the diversity of their substrates. Whereas Int acts on supercoiledcircular DNA, the substrates of Cre include the linear DNA of infecting phages, the rolling circle DNA of phages in the lytic growth phase, and the supercoiled DNA of plasmid oligomers. The striking promiscuity of Cre, which is so seductive a feature of the enzyme for genetic engineers,correlates with the diversity of the substrates with which it has biologically signiﬁcant interactions. Cre- loxCOMMERCIALIZATION AND A FAUSTIAN BARGAIN One member of Nat’s group, Brian Sauer, upon moving to DuPont, began experiments to deter- mine whether Cre- loxcould function in eukaryotic cells. Initially that possibility seemed unlikely. Cre, a prokaryotic protein, would now be made in the cytoplasm and would need to ﬁnd its way to the nucleus. Once in the nucleus, Cre would need to ﬁnd loxsites while surrounded by orders of magnitude more DNA than in an E. coli cell. Despite initial concerns raised by these considera- tions, Brian constructed vectors for the expression of crein yeast as well as loxsubstrates that could easily monitor Cre-mediated recombination. The substrates were cassettes with loxsites on each side of the yeast LEU2 gene. These were then stably inserted into the yeast genome in two differ- ent chromosomal locations by homologous recombination. A separate vector was then introduced containing the cregene under the control of the yeast galactose promoter. Induction with galactose resulted in creexpression and generation of leucine auxotrophs as a result of excision of LEU2 via loxrecombination (39). This was the basis for the ﬁrst patent ﬁling on Cre- loxfunctioning in eu- karyotic cells. Initially the work received a muted response from the scientiﬁc community becausefunctionally expressing a bacterial or phage protein in yeast was not considered that remarkable. Brian, undeterred, went on to show that Cre- loxcould also function efﬁciently in mouse cell lines (40). A stable mouse cell line was established with creexpressed from an inducible metallothionein promoter. Various loxsubstrates were subsequently introduced into this cell line by transforma- tion, and upon induction of the promoter, efﬁcient loxrecombination could be detected. It was the beginning of a revolution in genetic engineering. Late in 1990, Brian was granted a US patent for site-speciﬁc recombination of DNA in eukaryotic cells. It marked the commercialization of what was to become one of the most widely used tools in genetic engineering. Most companies,when issued a patent of this sort, continue working on the technology with the hopes of not only reﬁning the tools but also broadening and strengthening their patent position. This was not to be the case, and by the close of 1990, research on Cre- loxat DuPont had ended. In the fall of 1990 the molecular biology department had an off-site retreat organized by Mark Pearson. An after-dinner talk on the last evening was given by the head of DuPont Central Research and Development. The talk began innocuously enough, lauding our accomplishments asa department. It soon became apparent, when he discussed the DuPont bottom line, that Mark’s vision of continuing a world-class molecular biology department was a mirage. Within a few short months a joint venture was announced between DuPont Medical Products and Merck. Nolonger was there a need for a freestanding molecular biology department at DuPont; we were now incorporated within a pharmaceutical company. We had made our bargain and, after almost seven years, it was time to pay. A few of us managed to escape to the relative shelter of academia to continue basic research, but for most, this was not the case. The real tragedy was that Nat, because of a recurring cancer, was unable toleave. He was entering the last four years of his life, during which he endured almost continuous rounds of chemotherapy which left him bald and emaciated.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  It soon became apparent, when he discussed the DuPont bottom line, that Mark’s vision of continuing a world-class molecular biology department was a mirage. Within a few short months a joint venture was announced between DuPont Medical Products and Merck. Nolonger was there a need for a freestanding molecular biology department at DuPont; we were now incorporated within a pharmaceutical company. We had made our bargain and, after almost seven years, it was time to pay. A few of us managed to escape to the relative shelter of academia to continue basic research, but for most, this was not the case. The real tragedy was that Nat, because of a recurring cancer, was unable toleave. He was entering the last four years of his life, during which he endured almost continuous rounds of chemotherapy which left him bald and emaciated. Barely able to stand at his lab bench, Nat continued working. During this time, with little to no encouragement from anyone www.annualreviews.org •The Legacy of Nat Sternberg 35 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 in the company, he and a few postdoctoral students made signiﬁcant contributions to the human genome project (and other genome projects) by developing P1 in vitro packaging for cloninglarge DNA molecules (41). The vectors included not only P1 packaging sites but also loxsites for cyclizing the large DNA once injected into E. coli . For replication of the clones, a P1 replicon was used to maintain low copy number, but a controllable lytic P1 replicon was also includedto provide for ampliﬁcation of clonal DNA. It was as if Nat was reconstructing P1 from ﬁrst principles. EPILOGUE On June 29, 1995, three months before his death, a celebration of Nat’s career was held at NIH asa gala Lambda Lunch (a long-established NIH interest group). A large contingent of laboratory members and friends from Wilmington as well as former colleagues from NIH attended. The daywas marked by a number of talks relating to Nat’s many contributions over the years. Along with the science there was mention of Nat’s impatience to get on with his work, evident from his teeth- marks on hard-to-open restriction enzyme vials and the abnormal whine of unbalanced Eppendorfcentrifuges. The highlight of the day was Nat’s own talk, in which he described his training in the yeshiva, debating Talmudic questions, and then turned to his current interest in creating vectors for cloning large DNAs. His enthusiasm for science, even in those ﬁnal days, remained unabated.It seems only ﬁtting that his spark has been kept alive through the establishment by his former colleagues of the Nat Sternberg Thesis Prize, awarded annually for the best thesis in prokaryotic research. For those of us who had the good fortune to work alongside Nat, our lives were enrichedby his example as scientist and friend. In the past twenty years, the avenues of research into site-speciﬁc recombination that had been laid out by Nat and his coworkers have been extended in two directions, aided by high- tech advances in visualizing the choreography of proteins and DNA within living cells and in observing the staged performances of individual DNA molecules in vitro. Advances of no lessimportance have come from old-fashioned, low-tech bacterial genetics. One avenue of research leads to a deeper understanding of the cell cycle context of site-speciﬁc recombination. The other avenue, seemingly more relevant to Cre- loxtechnology, leads to an understanding of the successive molecular events in the recombination process. We cannot do justice here to recent developments in either of these areas, but cannot resist adding a few words about each. Nat did not live to become aware of the important bacterial cell division protein FtsK and its involvement in a function of a bacterial recombinase, XerCD. Like Cre and Int, both components of this bacterial recombinase possess an essential Arg-His-Arg triad and the catalytic Tyr residue at their active sites (42). The existence of the ftsKgene (discovered in E. coli but since shown to be widely distributed among eubacteria) was reported only late in the year of Nat’s death (43). The FtsK protein is a DNA translocase that reads the polarity of DNA based on its afﬁnity for 8-bp sequences that are preferentially oriented toward the replication terminus (44). From itslocation in the nascent septum it rapidly reels in DNA, sorting sister terminus regions on either side of the septal ring. It efﬁciently displaces bound proteins in the process until it ﬁnds the region of the replication terminus ( Figure 3 ). Within that region lies dif, the site of action of the XerCD recombinase. When the XerCD- difcomplex reaches FtsK, the translocase activates XerD to generate Holliday junction intermediates that can then be resolved by XerC (45). Inthis way FtsK facilitates completion of chromosome segregation prior to cell division. If Cre and suitably positioned loxsites are substituted for XerCD and difsites, Cre will satisfactorily resolve chromosome dimers (46). Dimer resolution by Cre- loxwas shown to depend on both FtsK and the appropriate location of the loxsites.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  His subsequent studies revealed that the locus on the DNA from which packaging begins(about 5 kb from lox) and the direction of packaging (toward lox) are such that the ﬁrst DNA to be packaged in a phage head has a loxsite centrally located in the roughly 10-kb region of terminal redundancy (15, 16). Selection pressure for an assist to cyclization must have placed it there. Nat did not ignore other possible functions of Cre-mediated recombination and in his initial publication suggested a role in the formation of concatemers, the requisite substrate of head- ful packaging. Based on evidence that, under sufﬁcient selection pressure, P1 integrated into the E. coli chromosome can be selected (14), Nat further suggested a likely role of P ×B recombination in transducing particle generation. Whereas concatemer formation is important in phage produc-tion, concatemer resolution to monomers is important for plasmid maintenance. P1 prophage is stable despite its low copy number because of an active segregation mechanism. If the units to be segregated are covalently joined, plasmid loss at cell division will ensue. In 1981, Stuart Austin,Marcia Ziese, and Nat Sternberg documented this additional role for Cre- lox(17). 30 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 My (R.H.) introduction to Nat Sternberg came on a bleak November day in 1978. I was visiting the Frederick Cancer Research Center to interview for a position. The barrack-like buildingsconnected by elevated steam pipes and the six-story bricked-up building referred to as the anthrax tower added to the gloominess of the place. My last interview that day was with Nat. In the laboratory that he shared with Michael Yarmolinsky were crammed four shaking water baths,buckets bristling with dirty pipettes, and stacks of Petri dishes. Nat pulled up a couple of chairs for us, grabbed a legal pad, and proceeded to describe his projects. As I discovered then, talking and simultaneously drawing was Nat’s expository style. Describing his work on the P1 repressor, he quickly ﬁlled the page, tore it off the pad, and wrote on the other side before using the next page. Near the end of the half hour he began describing the remarkable progress he had made on the site-speciﬁc recombination mediated by Cre- loxthat he had found in P1. Having depleted the pad, Nat continued writing and drawing on both sides of the cardboard backing ( Figure 2 ). It particularly interested me because I was just ﬁnishing a postdoctoral fellowship in the laboratory of ArthurLandy, studying the λattachment site and the genes involved in λsite-speciﬁc recombination. Figure 2 The back of the legal pad on which Nat Sternberg wrote during the interview with R.H., November 16, 1978. www.annualreviews.org •The Legacy of Nat Sternberg 31 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 When I accepted the offer to work at Frederick I had planned to continue working on λ, but not long after I arrived in the spring of 1979, Nat asked me rather casually whether I might beinterested in helping out on the Cre- loxproject. This was an opportunity too good to pass up, and within a few months nearly all my attention was directed at lox. Many of the features that have made Cre- loxa useful tool are best appreciated by comparison with what I will refer to as the λparadigm for site-speciﬁc recombination. Lambda Int- atthad been studied for over a decade prior to the discovery of Cre -lox, ﬁrst by extensive genetic analysis, then, following development of an in vitro reaction, by biochemistry (18). By the late 1970s, many of the protein components had been puriﬁed and the sites of recombination (attachment sites, viral and bacterial) had been sequenced. The picture of recombination that emerged was complex. The recombinase (Int) anda number of accessory proteins ﬁrst had to assemble into a nucleoprotein structure, which would then capture the bacterial attachment site. How recombination then took place was unclear, but a four-stranded, mobile (Holliday) junction seemed a likely intermediate. At that time λInt-att represented the best-characterized site-speciﬁc recombination. Not surprisingly, Nat and I, along with Kenneth Abremski, who would later join us, were inﬂuenced by the λparadigm, because each of us had worked on λrecombination. It inﬂuenced the design of our experiments and our expectations. Thus we were excited, even at the earliest stage, to uncover differences between Cre-loxand Int- att. More importantly, many of these differences would later translate into making Cre- loxthe versatile genetic engineering tool that it is today. By the end of 1979 we had determined the sequences of the various loxsites (19). The most striking feature was the presence of perfect 13-bp inverted repeats separated by an 8-bp spacer encompassing the crossover point. We supposed that the inverted repeats were potential bindingsites for Cre. The minimal size of a functional loxsite remained to be determined, but we assumed, by analogy with the λattachment site, that additional DNA sequences beyond the inverted repeats were likely to be required for recombination. In order to deﬁne the loxP site boundaries more precisely, we moved the putative loxP- containing region to a small plasmid convenient for further manipulation. It carried an ampicillin-resistance marker. Nat devised a simple pickup assay for testing loxfunction. The plasmid carrying theloxsite to be tested was introduced into an E. coli strain expressing a functional cregene. These cells were then infected with a λcontaining a functional loxPsite. If Cre-mediated recombination could occur between the two loxsites, then the resulting lysate should be able to transduce the ampicillin-resistance marker of the plasmid with high frequency (19). In the original assay, the region of several hundred base pairs of P1 DNA ﬂanking the functional loxPsite in one partner of the recombination made it impossible to exclude its role in the process. Further experiments revealed that a plasmid containing two loxP sites, each less than 60 bp in length, could undergo efﬁcient Cre-mediated recombination (20). Thus, unlike λ’s attachment site of about 240 bp, the loxPsite is quite small and therefore easily manipulated. Early in 1981 Nat received permission to form his own group. However, because Yarmolinsky’s group continued to study aspects of P1 biology and prokaryote research was deemed of onlymarginal relevance to cancer research, Nat was informed that he would have to switch his efforts to eukaryotes. For Nat, whose experience was entirely in the study of bacteria and their viruses, this was a daunting challenge. He accepted it, but had no intention of giving up his work on P1. Nat forged ahead on two fronts. He began studying recombination in mouse L cells, and he continued studying various aspects of P1 biology.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  Thus, unlike λ’s attachment site of about 240 bp, the loxPsite is quite small and therefore easily manipulated. Early in 1981 Nat received permission to form his own group. However, because Yarmolinsky’s group continued to study aspects of P1 biology and prokaryote research was deemed of onlymarginal relevance to cancer research, Nat was informed that he would have to switch his efforts to eukaryotes. For Nat, whose experience was entirely in the study of bacteria and their viruses, this was a daunting challenge. He accepted it, but had no intention of giving up his work on P1. Nat forged ahead on two fronts. He began studying recombination in mouse L cells, and he continued studying various aspects of P1 biology. Nat’s ﬁrst hire was Ken Abremski, who had just ﬁnished postdoctoral studies at NIH on the λ protein Xis, which is required in conjunction with Int for excision of λprophage from the bacterial chromosome (18, 21). Within months of his arrival, Ken developed an in vitro Cre-dependent recombination. One of the ﬁrst tests of the in vitro reaction used two loxP- containing λsubstrates 32 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 that Nat had constructed. Each substrate contained two loxPsites, one as direct repeats and the other as inverted repeats. Both substrates recombined efﬁciently in vivo and in vitro (20). Thiswas in marked contrast to λInt, which could recombine only directly repeated sites, and to the resolvases of certain of the small mobile genetic elements, such as that of γδ, which are able to catalyze only inversions (22). Expression vectors were constructed that provided us with ample quantities of Cre protein. Ken made rapid progress in purifying Cre and determining the requirements for the recombination reaction. At almost every turn we found some difference when compared with λInt recombina- tion. Unlike λInt recombination, no external energy source or additional accessory proteins were required for the reaction (23). Additionally, the different forms of the DNA substrate containingloxPsites, whether supercoiled, relaxed circle, or linear molecules, were all equally efﬁcient sub- strates. Given the minimal requirements of the in vitro reaction along with the robustness of the recombination, working with Cre was as simple as doing a restriction enzyme digest. WILMINGTON (DUPONT) During our time in Frederick we were ofﬁcially employed by Litton Industries, which ran our part of Fort Detrick under contract for the National Cancer Institute. This arrangement enabled the National Cancer Institute to respond rapidly to changing trends in cancer research, and aspart of a prokaryotic research group, we were aware of our vulnerability when the contract came up for renewal in 1984. For Nat and the members of his burgeoning laboratory, rescue came via DuPont. A relatively new member of the Frederick Cancer Research Center, Mark Pearson,was offered a position in DuPont’s Central Research and Development Department with the mandate to create a world-class molecular biology laboratory. Building on a well-established group of animal virologists, molecular geneticists, and chemists, Pearson rapidly assembled a department that included Nat’s entire group as well as biophysicists with expertise in molecular structure. Nat’s group moved to DuPont in the spring of 1984. Unlike the dreary environs of Frederick, we found ourselves on a tree-lined campus with spacious, newly renovated labs. Funding for equipment and supplies was never an issue. By this time Ken and I were absorbed in working outthe molecular details of Cre recombination. With puriﬁed Cre we were able to show by DNA footprinting that Cre does not contact DNA outside the 34 bp of the loxP site (24). Because Cre is able to carry out either excision or inversion depending on the orientation of the loxP sites, the footprinting results imply that the 8-bp spacer region imparts the directionality to the site. We demonstrated in vitro that Cre generates a 6-bp staggered cut within the spacer region (25) and made single-base substitutions within the spacer region to determine how theyaffected recombination. The results showed that homology of the spacer regions is required for recombination (26). Later these ﬁndings were exploited in genetic engineering with Cre and lox. Multiple loxsites, each addressable by the sequence of the spacer region, could be introduced into a cell, and only like sites would recombine with each other (27). The products of recombination between directly repeated loxsites on a highly supercoiled substrate were found not to be entangled with each other, as might be expected. Instead, recom- bination yielded supercoiled molecules entirely free of each other or linked as simple catenanes, rings that are topologically, but not chemically, linked (20, 28). This result suggested that beforethe initiation of recombination, the recombination sites come together in a way that physically separates the domains between them, as might occur if the sites are brought together via a slid- ing of DNA upon itself. Note that the rings of a DNA catenane are readily unlinked by cellulartopoisomerases. www.annualreviews.org •The Legacy of Nat Sternberg 33 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 One of Nat’s early papers on Cre- loxlooked at strand exchange during recombination of lox sites (29). A series of λ-P1 hybrid phages carrying loxPorloxRwere crossed and the reassortment of outside markers scored. From these results it was noted that branch migration had frequently occurred following an initial crossover. More direct physical evidence would come later when we began isolating and characterizing various mutants of Cre. One particular mutant appearedpartially deﬁcient for recombination in vitro (30). When the products of the reaction were analyzed it was shown that Cre preferentially exchanged one set of strands but only partially exchanged the second set of strands, thereby generating Holliday structure intermediates among the products. To rule out that the Holliday intermediates were aberrant end products of the reaction, they were puriﬁed and shown to be resolved on further addition of Cre, thus demonstrating they wereintermediates in the reaction. In addition to the biochemical dissection of the Cre- loxinteraction, the sequence of Cre itself was determined (31). Not long thereafter, an effort was made to align the sequence of Cre andother known recombinases, including Int of λ, P2, P22, P4, ϕ80, and ϕ186 (32). Despite the enormous sequence diversity among these recombinases, alignment by their C termini indicated a few small regions of homology with three residues (His, Arg, and Tyr) strictly conserved among allmembers. These residues were hypothesized to be involved in the catalysis of strand cleavage by the recombinases. A short time later a fourth conserved residue (Arg) was also identiﬁed (33). Three of these residues—Arg, His, and Arg—were shown by X-ray crystallography to form hydrogenbonds to the phosphate to which the conserved Tyr residue becomes transiently covalently bonded during cleavage, thus preserving the energy of the phosphodiester bond during the exchange (34). Over the years more than 100 potential recombinases have been identiﬁed by these alignments,indicating that Cre is a member of an enormously divergent recombinase family. Nat was particularly interested in the regulation of Cre expression and how it is modulated in relation to the P1 life cycle. Three promoters of varying strengths were identiﬁed upstream of Cre (31). One of these promoters contains Dam methylation sites, and experiments showed that in the absence of Dam methylation Cre expression increases. This leads to the idea that uponinfection, before newly replicated P1 DNA could be fully methylated, Cre expression would be elevated precisely when needed to cyclize the linear viral DNA. Cre expression is also expected to be elevated during lytic growth, whether following infection or following induction of an established prophage. This elevated expression could, in principle, increase the burst size of phage particles by recruiting those P1 plasmids that have not initiated replication into P1 plasmids that are in the process of replication (35), or it could do so simply byforming concatemers, the substrate for DNA packaging into phage heads (11). P1 concatemers are formed during lytic growth by rolling circle ( σ) replication, which, during the course of infection, takes over from bidirectional ( θ) replication (36). Whereas concatemers are important for producing viral particles, they (as well as circular oligomers) are disastrous for plasmid maintenance, because they are unpartitionable units. The problem for segregation created by circular chromosome dimers generated by interchromosomerecombination was recognized as early as 1932 by Barbara McClintock [cited by Barre et al. (37)]. As Barre et al. observe, the problem appears to have been largely ignored prior to work on P1 segregation (17). During lysogenic growth of P1, plasmid oligomers are formed occasionally by RecA-mediated homologous recombination. Their resolution prior to cell division, a function of Cre- lox, is essential for prophage stability (17, 38). Ironically, problems arising from recombina- tion events that are infrequent with respect to the cell cycle are remedied by raising the frequency of those events. Raising the recombination frequency without introducing new problems is ac- complished by limiting the site of recombination to a highly localized region.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  The problem for segregation created by circular chromosome dimers generated by interchromosomerecombination was recognized as early as 1932 by Barbara McClintock [cited by Barre et al. (37)]. As Barre et al. observe, the problem appears to have been largely ignored prior to work on P1 segregation (17). During lysogenic growth of P1, plasmid oligomers are formed occasionally by RecA-mediated homologous recombination. Their resolution prior to cell division, a function of Cre- lox, is essential for prophage stability (17, 38). Ironically, problems arising from recombina- tion events that are infrequent with respect to the cell cycle are remedied by raising the frequency of those events. Raising the recombination frequency without introducing new problems is ac- complished by limiting the site of recombination to a highly localized region. 34 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 The Int of λand Cre of P1 are similar in having more than a single role in the life of their respective phages, but they differ in the diversity of their substrates. Whereas Int acts on supercoiledcircular DNA, the substrates of Cre include the linear DNA of infecting phages, the rolling circle DNA of phages in the lytic growth phase, and the supercoiled DNA of plasmid oligomers. The striking promiscuity of Cre, which is so seductive a feature of the enzyme for genetic engineers,correlates with the diversity of the substrates with which it has biologically signiﬁcant interactions. Cre- loxCOMMERCIALIZATION AND A FAUSTIAN BARGAIN One member of Nat’s group, Brian Sauer, upon moving to DuPont, began experiments to deter- mine whether Cre- loxcould function in eukaryotic cells. Initially that possibility seemed unlikely. Cre, a prokaryotic protein, would now be made in the cytoplasm and would need to ﬁnd its way to the nucleus. Once in the nucleus, Cre would need to ﬁnd loxsites while surrounded by orders of magnitude more DNA than in an E. coli cell. Despite initial concerns raised by these considera- tions, Brian constructed vectors for the expression of crein yeast as well as loxsubstrates that could easily monitor Cre-mediated recombination. The substrates were cassettes with loxsites on each side of the yeast LEU2 gene. These were then stably inserted into the yeast genome in two differ- ent chromosomal locations by homologous recombination. A separate vector was then introduced containing the cregene under the control of the yeast galactose promoter. Induction with galactose resulted in creexpression and generation of leucine auxotrophs as a result of excision of LEU2 via loxrecombination (39). This was the basis for the ﬁrst patent ﬁling on Cre- loxfunctioning in eu- karyotic cells. Initially the work received a muted response from the scientiﬁc community becausefunctionally expressing a bacterial or phage protein in yeast was not considered that remarkable. Brian, undeterred, went on to show that Cre- loxcould also function efﬁciently in mouse cell lines (40). A stable mouse cell line was established with creexpressed from an inducible metallothionein promoter. Various loxsubstrates were subsequently introduced into this cell line by transforma- tion, and upon induction of the promoter, efﬁcient loxrecombination could be detected. It was the beginning of a revolution in genetic engineering. Late in 1990, Brian was granted a US patent for site-speciﬁc recombination of DNA in eukaryotic cells. It marked the commercialization of what was to become one of the most widely used tools in genetic engineering. Most companies,when issued a patent of this sort, continue working on the technology with the hopes of not only reﬁning the tools but also broadening and strengthening their patent position. This was not to be the case, and by the close of 1990, research on Cre- loxat DuPont had ended. In the fall of 1990 the molecular biology department had an off-site retreat organized by Mark Pearson. An after-dinner talk on the last evening was given by the head of DuPont Central Research and Development. The talk began innocuously enough, lauding our accomplishments asa department. It soon became apparent, when he discussed the DuPont bottom line, that Mark’s vision of continuing a world-class molecular biology department was a mirage. Within a few short months a joint venture was announced between DuPont Medical Products and Merck. Nolonger was there a need for a freestanding molecular biology department at DuPont; we were now incorporated within a pharmaceutical company. We had made our bargain and, after almost seven years, it was time to pay. A few of us managed to escape to the relative shelter of academia to continue basic research, but for most, this was not the case. The real tragedy was that Nat, because of a recurring cancer, was unable toleave. He was entering the last four years of his life, during which he endured almost continuous rounds of chemotherapy which left him bald and emaciated.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  It soon became apparent, when he discussed the DuPont bottom line, that Mark’s vision of continuing a world-class molecular biology department was a mirage. Within a few short months a joint venture was announced between DuPont Medical Products and Merck. Nolonger was there a need for a freestanding molecular biology department at DuPont; we were now incorporated within a pharmaceutical company. We had made our bargain and, after almost seven years, it was time to pay. A few of us managed to escape to the relative shelter of academia to continue basic research, but for most, this was not the case. The real tragedy was that Nat, because of a recurring cancer, was unable toleave. He was entering the last four years of his life, during which he endured almost continuous rounds of chemotherapy which left him bald and emaciated. Barely able to stand at his lab bench, Nat continued working. During this time, with little to no encouragement from anyone www.annualreviews.org •The Legacy of Nat Sternberg 35 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 in the company, he and a few postdoctoral students made signiﬁcant contributions to the human genome project (and other genome projects) by developing P1 in vitro packaging for cloninglarge DNA molecules (41). The vectors included not only P1 packaging sites but also loxsites for cyclizing the large DNA once injected into E. coli . For replication of the clones, a P1 replicon was used to maintain low copy number, but a controllable lytic P1 replicon was also includedto provide for ampliﬁcation of clonal DNA. It was as if Nat was reconstructing P1 from ﬁrst principles. EPILOGUE On June 29, 1995, three months before his death, a celebration of Nat’s career was held at NIH asa gala Lambda Lunch (a long-established NIH interest group). A large contingent of laboratory members and friends from Wilmington as well as former colleagues from NIH attended. The daywas marked by a number of talks relating to Nat’s many contributions over the years. Along with the science there was mention of Nat’s impatience to get on with his work, evident from his teeth- marks on hard-to-open restriction enzyme vials and the abnormal whine of unbalanced Eppendorfcentrifuges. The highlight of the day was Nat’s own talk, in which he described his training in the yeshiva, debating Talmudic questions, and then turned to his current interest in creating vectors for cloning large DNAs. His enthusiasm for science, even in those ﬁnal days, remained unabated.It seems only ﬁtting that his spark has been kept alive through the establishment by his former colleagues of the Nat Sternberg Thesis Prize, awarded annually for the best thesis in prokaryotic research. For those of us who had the good fortune to work alongside Nat, our lives were enrichedby his example as scientist and friend. In the past twenty years, the avenues of research into site-speciﬁc recombination that had been laid out by Nat and his coworkers have been extended in two directions, aided by high- tech advances in visualizing the choreography of proteins and DNA within living cells and in observing the staged performances of individual DNA molecules in vitro. Advances of no lessimportance have come from old-fashioned, low-tech bacterial genetics. One avenue of research leads to a deeper understanding of the cell cycle context of site-speciﬁc recombination. The other avenue, seemingly more relevant to Cre- loxtechnology, leads to an understanding of the successive molecular events in the recombination process. We cannot do justice here to recent developments in either of these areas, but cannot resist adding a few words about each. Nat did not live to become aware of the important bacterial cell division protein FtsK and its involvement in a function of a bacterial recombinase, XerCD. Like Cre and Int, both components of this bacterial recombinase possess an essential Arg-His-Arg triad and the catalytic Tyr residue at their active sites (42). The existence of the ftsKgene (discovered in E. coli but since shown to be widely distributed among eubacteria) was reported only late in the year of Nat’s death (43). The FtsK protein is a DNA translocase that reads the polarity of DNA based on its afﬁnity for 8-bp sequences that are preferentially oriented toward the replication terminus (44). From itslocation in the nascent septum it rapidly reels in DNA, sorting sister terminus regions on either side of the septal ring. It efﬁciently displaces bound proteins in the process until it ﬁnds the region of the replication terminus ( Figure 3 ). Within that region lies dif, the site of action of the XerCD recombinase. When the XerCD- difcomplex reaches FtsK, the translocase activates XerD to generate Holliday junction intermediates that can then be resolved by XerC (45). Inthis way FtsK facilitates completion of chromosome segregation prior to cell division. If Cre and suitably positioned loxsites are substituted for XerCD and difsites, Cre will satisfactorily resolve chromosome dimers (46). Dimer resolution by Cre- loxwas shown to depend on both FtsK and the appropriate location of the loxsites.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:57 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'76'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992880'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'427ms'), (b'x-request-id', b'f77041172cf5511565973b354254d39b'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412498478fc5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:57 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'76'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992880'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'427ms'), (b'x-request-id', b'f77041172cf5511565973b354254d39b'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412498478fc5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  It efﬁciently displaces bound proteins in the process until it ﬁnds the region of the replication terminus ( Figure 3 ). Within that region lies dif, the site of action of the XerCD recombinase. When the XerCD- difcomplex reaches FtsK, the translocase activates XerD to generate Holliday junction intermediates that can then be resolved by XerC (45). Inthis way FtsK facilitates completion of chromosome segregation prior to cell division. If Cre and suitably positioned loxsites are substituted for XerCD and difsites, Cre will satisfactorily resolve chromosome dimers (46). Dimer resolution by Cre- loxwas shown to depend on both FtsK and the appropriate location of the loxsites. This action of FtsK on Cre introduces an intriguing new 36 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 13 bp 13 bpATGTATGC TACATACGP1   loxP 11 bp 11 bpTGTATA ACATATEscherichia coli   dif 9 bp ~12 bp ~12 bp ~12 bp ~12 bp ~12 bp 9 bpTTTATAC AAATATGλ   attP Figure 3 Comparison of the binding sites of recombinases Cre (19), XerCD (48), and Int (49). During recombination each recombinase generates staggered cleavages ( small triangles ) at the periphery of a spacer between elements of dyad symmetry of a core binding site. Int has additional arm recognition elements ( dark red arrows ) distributed over 240 bp. They are related (but not identical) and necessary for the assembly of an active recombination complex in which the DNA is bound to an Int tetramer and to host proteins IHF and Fis. aspect of Cre-mediated recombination in vivo. The intimate molecular details of recombinase activation by FtsK have now been tracked by three single-molecule techniques (47). Not long after Nat’s death, Greg Van Duyne and his group solved the crystal structure of Cre- lox(34). The structure not only revealed the molecular architecture of Cre and the loxPsite but also showed them in the process of recombining via a Holliday structure intermediate. This work was rapidly followed by reports of additional structures where altered loxP sites or mutant Cre proteins were cocrystallized (50). The combination of structural work with traditional biochemicalanalysis has resulted in a detailed picture of recombination. Cre binds to the loxPsite and induces the DNA to bend. Formation of a synaptic structure begins a well-ordered series of steps in which cleavage and strand exchange ﬁrst occur predominantly on the bottom strand to form the Hollidaystructure intermediate. Following isomerization of the Holliday structure, top-strand cleavage and strand exchange take place to complete the reaction. This picture has recently been bolstered by the new technique of single-molecule ﬂuorescence in which individual molecules can be observedduring the recombination process (51). At the beginning of this review we noted the coincidence that λand P1 had parallel roles in the rediscovery of lysogeny and then in the recognition of the important, but differing, roles ofsite-speciﬁc recombination in the life of these phages. Lambda’s Int-mediated recombination is unidirectional and is under control by three accessory proteins, of which one (Xis) can reverse its direction. Int belongs to the class of regulated recombinases. P1’s Cre-mediated recombination is bidirectional, requires no accessory proteins, and is unregulated. It belongs, along with XerCD, to the class of recombinases that are largely unregulated. In the course of this review, as in thecourse of characterizing Cre- lox, the contrast between Int and Cre and between att-λand loxP has been a recurring theme. It therefore seems ﬁtting that we close by considering their union, speciﬁcally the interaction of an Int-Cre chimera with chimeric binding sites. These were createdin the laboratory of Arthur Landy (52). A small (76-residue) N-terminal fragment of Int was fused to Cre. The resulting chimeric enzyme recombines complex binding sites with all the regulated directionality and responses to accessory proteins that characterize Int-promoted recombination.The authors suggest that this construction, in its striking simplicity, may simulate the evolutionary interconversions responsible for the large variety of site-speciﬁc recombinases observed in the living world. www.annualreviews.org •The Legacy of Nat Sternberg 37 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 DISCLOSURE STATEMENT The authors are not aware of any afﬁliations, memberships, funding, or ﬁnancial holdings that might be perceived as affecting the objectivity of this review. ACKNOWLEDGMENTS We thank Richard D’Ari, Robert LaRossa, Greg Van Duyne, and David Sherratt for their critical reading of a draft of this article and Franc ¸ois Cornet for enlightening comments on FtsK Orienting Polar Sequences (KOPS). This essay was supported by the Intramural Research Program of the National Cancer Institute’s Center for Cancer Research at the National Institutes of Health. LITERATURE CITED 1. Lwoff A. 1953. Lysogeny. Bacteriol. Rev. 17:269–337 2. Lwoff A. Nobel Lecture: Interaction Among Virus, Cell, and Organism . Stockholm: Nobel Media AB. http:// www.nobelprize.org/nobel_prizes/medicine/laureates/1965/lwoff-lecture.html 3. Bertani G. 2004. Lysogeny at mid-twentieth century: P1, P2, and other experimental systems. J. Bacteriol. 186:595–600 4. Lennox ES. 1955. Transduction of linked genetic characters of the host by bacteriophage P1. Virology 1:190–206 5. Yarmolinsky MB, Sternberg N. 1988. Bacteriophage P1. In The Bacteriophages , Vol. 1, ed. R Calendar, pp. 291–438. New York: Plenum 6. Łobocka MB, Rose DJ, Plunket G III, Rusin M, Samojedny A, et al. 2004. Genome of bacteriophage P1. J. Bacteriol. 186:7032–68 7. Ikeda H, Tomizawa J. 1968. Prophage P1, an extra-chromosomal replication unit. Cold Spring Harb. Symp. Quant. Biol. 33:791–98 8. Jacob F, Monod J. 1961. Genetic regulatory mechanisms in the synthesis of proteins. J. Mol. Biol. 3:315–56 9. Clerget M. 1991. Site-speciﬁc recombination promoted by a short DNA segment of plasmid R1 and by a homologous segment in the terminus region of the Escherichia coli chromosome. New Biol. 3:780–88 10. Sternberg N, Tiemeier D, Enquist L. 1977. In vitro packaging of a λDamvector containing EcoRI DNA fragments of Escherichia coli and phage P1. Gene 1:255–80 11. Sternberg N. 1978. Demonstration and analysis of P1 site-speciﬁc recombination using λ-P1 hybrid phages constructed in vitro. Cold Spring Harb. Symp. Quant. Biol. 43:1143–46 12. Sternberg N, Hamilton D. 1981. Bacteriophage P1 site-speciﬁc recombination. I. Recombination between loxPsites. J. Mol. Biol. 150:467–86 13. Takano T. 1971. Bacterial mutant defective in plasmid formation: requirement for the lon+allele. PNAS 68:1469–73 14. Chesney RH, Vapnek D, Scott JR. 1978. Recombination between the plasmid prophages P1 and P7 and theE. coli chromosome. Contrib. Microbiol. Immunol. 6:78–88 15. Sternberg N, Coulby J. 1987. Recognition and cleavage of the bacteriophage P1 packaging site ( pac). I. Differential processing of the cleaved ends in vivo. J. Mol. Biol. 194:453–68 16. Sternberg N, Coulby J. 1987.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  J. Mol. Biol. 150:467–86 13. Takano T. 1971. Bacterial mutant defective in plasmid formation: requirement for the lon+allele. PNAS 68:1469–73 14. Chesney RH, Vapnek D, Scott JR. 1978. Recombination between the plasmid prophages P1 and P7 and theE. coli chromosome. Contrib. Microbiol. Immunol. 6:78–88 15. Sternberg N, Coulby J. 1987. Recognition and cleavage of the bacteriophage P1 packaging site ( pac). I. Differential processing of the cleaved ends in vivo. J. Mol. Biol. 194:453–68 16. Sternberg N, Coulby J. 1987. Recognition and cleavage of the bacteriophage P1 packaging site ( pac). II. Functional limits of pacand location of paccleavage termini. J. Mol. Biol. 194:469–83 17. Austin S, Zeise M, Sternberg N. 1981. A novel role for site-speciﬁc recombination in the maintenance of bacterial replicons. Cell25:729–36 18. Nash HA. 1981. Integration and excision of bacteriophage λ: the mechanism of conservative site speciﬁc recombination. Annu. Rev. Genet. 15:143–67 19. Hoess RH, Ziese M, Sternberg N. 1982. P1 site-speciﬁc recombination: nucleotide sequence of the recombining sites. PNAS 79:3398–402 20. Abremski K, Hoess R, Sternberg N. 1983. Studies on the properties of P1 site-speciﬁc recombination: evidence for topologically unlinked productions following recombination. Cell32:1301–11 38 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 21. Abremski K, Gottesman S. 1981. Site-speciﬁc recombination: Xis-independent excisive recombination of bacteriophage lambda. J. Mol. Biol. 153:67–78 22. Reed RR. 1981. Transposon-mediated site-speciﬁc recombination: a deﬁned in vitro system. Cell25:713– 19 23. Abremski K, Hoess R. 1984. Bacteriophage P1 site-speciﬁc recombination: puriﬁcation and properties of the Cre recombinase protein. J. Biol. Chem. 259:1509–14 24. Hoess R, Abremski K. 1984. Interaction of the bacteriophage P1 recombinase Cre with the recombining siteloxP.PNAS 81:1026–29 25. Hoess R, Abremski K. 1985. Mechanism of strand cleavage and exchange in the Cre- loxsite-speciﬁc recombination system. J. Mol. Biol. 181:351–62 26. Hoess R, Wierzbicki A, Abremski K. 1986. The role of the loxPspacer region in P1 site-speciﬁc recom- bination. Nucleic Acids Res. 14:2287–300 27. Sauer B. 1996. Multiplex Cre/ loxrecombination permits selective site-speciﬁc DNA targeting to both a natural and an engineered site in the yeast genome. Nucleic Acids Res. 24:4608–13 28. Abremski K, Hoess R. 1985. Phage P1 Cre- loxsite-speciﬁc recombination: effects of DNA supercoiling on catenation and knotting of recombinant products. J. Mol. Biol. 184:211–20 29. Sternberg N. 1981. Bacteriophage P1 site-speciﬁc recombination. III. Strand exchange during recombi- nation at loxsites. J. Mol. Biol. 150:603–8 30. Hoess R, Wierzbicki A, Abremski K. 1987. Isolation and characterization of potential intermediates in site-speciﬁc recombination. PNAS 84:6840–44 31. Sternberg N, Sauer B, Hoess R, Abremski K. 1986. Bacteriophage P1 cregene and its regulatory region: evidence for multiple promoters and regulation by DNA methylation. J. Mol. Biol. 187:197–212 32. Argos P, Landy A, Abremski K, Egan JB, Haggard-Ljungquist E, et al. 1986. The integrase family of site-speciﬁc recombinases: regional similarities and global diversity. EMBO J. 5:433–40 33. Abremski K, Hoess R. 1992. Evidence for a second conserved arginine residue in the integrase family of recombination proteins. Protein Eng. 5:87–91 34. Guo F, Gopaul DN, Van Duyne GD. 1997. Structure of Cre recombinase complexed with DNA in a site-speciﬁc recombination synapse. Nature 389:40–46 35. Adams DE, Bliska JB, Cozzarelli NR. 1992. Cre- loxrecombination in Escherichia coli cells: mechanistic differences from the in vitro reaction. J. Mol. Biol. 226:661–73 36. Cohen G. 1983. Electron microscopy study of early lytic replication forms of bacteriophage P1 DNA. Virology 131:159–70 37. Barre FX, Soballe B, Michel B, Aroyo M, Robertson M, et al. 2001.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  Protein Eng. 5:87–91 34. Guo F, Gopaul DN, Van Duyne GD. 1997. Structure of Cre recombinase complexed with DNA in a site-speciﬁc recombination synapse. Nature 389:40–46 35. Adams DE, Bliska JB, Cozzarelli NR. 1992. Cre- loxrecombination in Escherichia coli cells: mechanistic differences from the in vitro reaction. J. Mol. Biol. 226:661–73 36. Cohen G. 1983. Electron microscopy study of early lytic replication forms of bacteriophage P1 DNA. Virology 131:159–70 37. Barre FX, Soballe B, Michel B, Aroyo M, Robertson M, et al. 2001. Circles: the replication-recombination- chromosome segregation connection. PNAS 98:8189–95 38. Sternberg N, Hamilton D, Austin S, Yarmolinsky M, Hoess R. 1981. Site-speciﬁc recombination and its role in the life cycle of bacteriophage P1. Cold Spring Harb. Symp. Quant. Biol. 45:297–309 39. Sauer B. 1987. Functional expression of the cre-loxsite-speciﬁc recombination system in the yeast Saccha- romyces cerevisiae .Mol. Cell. Biol. 7:2087–96 40. Sauer B, Henderson N. 1988. Site-speciﬁc recombination in mammalian cells by the Cre recombinase of bacteriophage P1. PNAS 85:5166–70 41. Sternberg N. 1994. The P1 cloning system past and future. Mamm. Genome 5:397–404 42. Esposito D, Scocca JJ. 1997. The integrase family of tyrosine recombinases: evolution of a conserved active site domain. Nucleic Acids Res. 25:3605–14 43. Begg KJ, Dewar SJ, Donachie WD. 1995. A new Escherichia coli cell division gene, ftsK.J. Bacteriol. 177:6211–22 44. Bigot S, Sivanatan V, Possoz C, Barre FX, Cornet F. 2007. FtsK, a literate chromosome segregation machine. Mol. Microbiol. 64:1434–41 45. Grainge I, Lesterlin C, Sherratt DJ. 2011. Activation of XerCD- difrecombination by the FtsK DNA translocase. Nucleic Acids Res. 39:5140–48 46. Capiaux H, Lesterlin C, P ´erals K, Louarn JM, Cornet F. 2002. A dual role for the FtsK protein in Escherichia coli chromosome segregation. EMBO Rep. 3:532–36 47. Zawadzki P, May PF, Baker RA, Pinkney JN, Kapanidis AN, et al. 2013. Conformational transitions during FtsK translocase activation of individual XerCD- difrecombination complexes. PNAS 110:17302–7 www.annualreviews.org •The Legacy of Nat Sternberg 39 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 48. Carnoy C, Roten CA. 2009. The dif/Xer recombination systems in proteobacteria. PLOS ONE 4:e6531 49. Hsu PL, Ross W, Landy A. 1980. The λphage attsite: functional limits and interaction with Int protein. Nature 285:85–91 50. Van Duyne GD. 2001. A structural view of Cre- loxP site-speciﬁc recombination. Annu. Rev. Biophys. Biomol. Struct. 30:87–104 51. Pinkney JN, Zawadzki P, Mazauryic J, Arciszewska LK, Sherratt DJ, Kapanidis AN. 2012. Capturing reaction paths and intermediates in Cre- loxP recombination using single molecule ﬂuorescence. PNAS 109:20871–76 52. Warren D, Laxmikanthan G, Landy A. 2008. A chimeric Cre recombinase with regulated directionality. PNAS 105:18278–83 40 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02-FrontMatter ARI 15 October 2015 16:33 Annual Review of Virology Volume 2, 2015 Contents History The Maturation of a Scientist: An Autobiography Bernard Roizman ppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 1 The Legacy of Nat Sternberg: The Genesis of Cre- loxTechnology Michael Yarmolinsky and Ronald Hoess pppppppppppppppppppppppppppppppppppppppppppppppppppppp 25 Ecology and Evolution Biogeography of Viruses in the Sea Cheryl-Emiliane T. Chow and Curtis A. Suttle ppppppppppppppppppppppppppppppppppppppppppppp 41 Role of the Insect Supervectors Bemisia tabaci and Frankliniella occidentalis in the Emergence and Global Spread of Plant Viruses Robert L. Gilbertson, Ozgur Batuman, Craig G. Webster, and Scott Adkins ppppppppppppp 67 Coronavirus Host Range Expansion and Middle East Respiratory Syndrome Coronavirus Emergence: Biochemical Mechanisms and Evolutionary Perspectives Kayla M. Peck, Christina L. Burch, Mark T. Heise, and Ralph S. Baric pppppppppppppppppp 95 Koala Retroviruses: Evolution and Disease Dynamics Wenqin Xu and Maribeth V. Eiden pppppppppppppppppppppppppppppppppppppppppppppppppppppppp 119 Endogenous Retroviruses in the Genomics Era Welkin E. Johnson pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 135 Matters of Size: Genetic Bottlenecks in Virus Infection and Their Potential Impact on Evolution Mark P. Zwart and Santiago F. Elena pppppppppppppppppppppppppppppppppppppppppppppppppppp 161 The Phage-Inducible Chromosomal Islands: A Family of Highly Evolved Molecular Parasites Jos´e R. Penad´ es and Gail E. Christie ppppppppppppppppppppppppppppppppppppppppppppppppppppppp 181 Deep Recombination: RNA and ssDNA Virus Genes in DNA Virus and Host Genomes Kenneth M. Stedman pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 203 xi Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02-FrontMatter ARI 15 October 2015 16:33 Attachment and Cell Entry Hunting Viral Receptors Using Haploid Cells Sirika Pillay and Jan E. Carette ppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 219 Genome Replication, Regulation of Gene Expression, and Biosynthesis Retroviral Integrase: Then and Now Mark D. Andrake and Anna Marie Skalka ppppppppppppppppppppppppppppppppppppppppppppppp 241 Continuous and Discontinuous RNA Synthesis in Coronaviruses Isabel Sola, Fernando Almaz´ an, Sonia Z ´ u˜niga, and Luis Enjuanes pppppppppppppppppppppp 265 Flaviviridae Replication Organelles: Oh, What a Tangled Web We Weave David Paul and Ralf Bartenschlager ppppppppppppppppppppppppppppppppppppppppppppppppppppppp 289 Modulation of the Translational Landscape During Herpesvirus Infection Britt A. Glaunsinger pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 311 Ribosome Proﬁling as a Tool to Decipher Viral Complexity Noam Stern-Ginossar and Nicholas T. Ingolia pppppppppppppppppppppppppppppppppppppppppppp 335 Mechanisms of DNA Packaging by Large Double-Stranded DNA Viruses Venigalla B. Rao and Michael Feiss pppppppppppppppppppppppppppppppppppppppppppppppppppppppp 351 Assembly and Egress Virus-Based Nanoparticles as Versatile Nanomachines Kristopher J. Koudelka, Andrzej S. Pitek, Marianne Manchester, and Nicole F. Steinmetz pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 379 Transformation and Oncogenesis Novel Functions of the Human Papillomavirus E6 Oncoproteins Nicholas A. Wallace and Denise A. Galloway ppppppppppppppppppppppppppppppppppppppppppppp 403 Pathogenesis Parvovirus Family Conundrum: What Makes a Killer? Shweta Kailasan, Mavis Agbandje-McKenna, and Colin R. Parrish ppppppppppppppppppppp 425 Everything You Always Wanted to Know About Rabies Virus (But Were Afraid to Ask) Benjamin M. Davis, Glenn F. Rall, and Matthias J. Schnell pppppppppppppppppppppppppppp 451 Pathophysiological Consequences of Calcium-Conducting Viroporins Joseph M. Hyser and Mary K. Estes pppppppppppppppppppppppppppppppppppppppppppppppppppppppp 473 Virus-Host Interactions: From Unbiased Genetic Screens to Function Holly Ramage and Sara Cherry pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 497 Viruses in Rodent Colonies: Lessons Learned from Murine Noroviruses Stephanie M. Karst and Christiane E. Wobus ppppppppppppppppppppppppppppppppppppppppppppp 525 xii Contents Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02-FrontMatter ARI 15 October 2015 16:33 Immunity No Love Lost Between Viruses and Interferons Volker Fensterl, Saurabh Chattopadhyay, and Ganes C. Sen ppppppppppppppppppppppppppppp 549 Innate and Adaptive Immune Regulation During Chronic Viral Infections Elina I. Zuniga, Monica Macal, Gavin M. Lewis, and James A. Harker ppppppppppppppp 573 Viral Vectors and Therapeutics Bacteriophage Therapy: Advances in Formulation Strategies and Human Clinical Trials Dieter Vandenheuvel, Rob Lavigne, and Harald Br ¨ ussow pppppppppppppppppppppppppppppppp 599 Errata An online log of corrections to Annual Review of Virology articles may be found at http://www.annualreviews.org/errata/virology Contents xiii Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  It efﬁciently displaces bound proteins in the process until it ﬁnds the region of the replication terminus ( Figure 3 ). Within that region lies dif, the site of action of the XerCD recombinase. When the XerCD- difcomplex reaches FtsK, the translocase activates XerD to generate Holliday junction intermediates that can then be resolved by XerC (45). Inthis way FtsK facilitates completion of chromosome segregation prior to cell division. If Cre and suitably positioned loxsites are substituted for XerCD and difsites, Cre will satisfactorily resolve chromosome dimers (46). Dimer resolution by Cre- loxwas shown to depend on both FtsK and the appropriate location of the loxsites. This action of FtsK on Cre introduces an intriguing new 36 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 13 bp 13 bpATGTATGC TACATACGP1   loxP 11 bp 11 bpTGTATA ACATATEscherichia coli   dif 9 bp ~12 bp ~12 bp ~12 bp ~12 bp ~12 bp 9 bpTTTATAC AAATATGλ   attP Figure 3 Comparison of the binding sites of recombinases Cre (19), XerCD (48), and Int (49). During recombination each recombinase generates staggered cleavages ( small triangles ) at the periphery of a spacer between elements of dyad symmetry of a core binding site. Int has additional arm recognition elements ( dark red arrows ) distributed over 240 bp. They are related (but not identical) and necessary for the assembly of an active recombination complex in which the DNA is bound to an Int tetramer and to host proteins IHF and Fis. aspect of Cre-mediated recombination in vivo. The intimate molecular details of recombinase activation by FtsK have now been tracked by three single-molecule techniques (47). Not long after Nat’s death, Greg Van Duyne and his group solved the crystal structure of Cre- lox(34). The structure not only revealed the molecular architecture of Cre and the loxPsite but also showed them in the process of recombining via a Holliday structure intermediate. This work was rapidly followed by reports of additional structures where altered loxP sites or mutant Cre proteins were cocrystallized (50). The combination of structural work with traditional biochemicalanalysis has resulted in a detailed picture of recombination. Cre binds to the loxPsite and induces the DNA to bend. Formation of a synaptic structure begins a well-ordered series of steps in which cleavage and strand exchange ﬁrst occur predominantly on the bottom strand to form the Hollidaystructure intermediate. Following isomerization of the Holliday structure, top-strand cleavage and strand exchange take place to complete the reaction. This picture has recently been bolstered by the new technique of single-molecule ﬂuorescence in which individual molecules can be observedduring the recombination process (51). At the beginning of this review we noted the coincidence that λand P1 had parallel roles in the rediscovery of lysogeny and then in the recognition of the important, but differing, roles ofsite-speciﬁc recombination in the life of these phages. Lambda’s Int-mediated recombination is unidirectional and is under control by three accessory proteins, of which one (Xis) can reverse its direction. Int belongs to the class of regulated recombinases. P1’s Cre-mediated recombination is bidirectional, requires no accessory proteins, and is unregulated. It belongs, along with XerCD, to the class of recombinases that are largely unregulated. In the course of this review, as in thecourse of characterizing Cre- lox, the contrast between Int and Cre and between att-λand loxP has been a recurring theme. It therefore seems ﬁtting that we close by considering their union, speciﬁcally the interaction of an Int-Cre chimera with chimeric binding sites. These were createdin the laboratory of Arthur Landy (52). A small (76-residue) N-terminal fragment of Int was fused to Cre. The resulting chimeric enzyme recombines complex binding sites with all the regulated directionality and responses to accessory proteins that characterize Int-promoted recombination.The authors suggest that this construction, in its striking simplicity, may simulate the evolutionary interconversions responsible for the large variety of site-speciﬁc recombinases observed in the living world. www.annualreviews.org •The Legacy of Nat Sternberg 37 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 DISCLOSURE STATEMENT The authors are not aware of any afﬁliations, memberships, funding, or ﬁnancial holdings that might be perceived as affecting the objectivity of this review. ACKNOWLEDGMENTS We thank Richard D’Ari, Robert LaRossa, Greg Van Duyne, and David Sherratt for their critical reading of a draft of this article and Franc ¸ois Cornet for enlightening comments on FtsK Orienting Polar Sequences (KOPS). This essay was supported by the Intramural Research Program of the National Cancer Institute’s Center for Cancer Research at the National Institutes of Health. LITERATURE CITED 1. Lwoff A. 1953. Lysogeny. Bacteriol. Rev. 17:269–337 2. Lwoff A. Nobel Lecture: Interaction Among Virus, Cell, and Organism . Stockholm: Nobel Media AB. http:// www.nobelprize.org/nobel_prizes/medicine/laureates/1965/lwoff-lecture.html 3. Bertani G. 2004. Lysogeny at mid-twentieth century: P1, P2, and other experimental systems. J. Bacteriol. 186:595–600 4. Lennox ES. 1955. Transduction of linked genetic characters of the host by bacteriophage P1. Virology 1:190–206 5. Yarmolinsky MB, Sternberg N. 1988. Bacteriophage P1. In The Bacteriophages , Vol. 1, ed. R Calendar, pp. 291–438. New York: Plenum 6. Łobocka MB, Rose DJ, Plunket G III, Rusin M, Samojedny A, et al. 2004. Genome of bacteriophage P1. J. Bacteriol. 186:7032–68 7. Ikeda H, Tomizawa J. 1968. Prophage P1, an extra-chromosomal replication unit. Cold Spring Harb. Symp. Quant. Biol. 33:791–98 8. Jacob F, Monod J. 1961. Genetic regulatory mechanisms in the synthesis of proteins. J. Mol. Biol. 3:315–56 9. Clerget M. 1991. Site-speciﬁc recombination promoted by a short DNA segment of plasmid R1 and by a homologous segment in the terminus region of the Escherichia coli chromosome. New Biol. 3:780–88 10. Sternberg N, Tiemeier D, Enquist L. 1977. In vitro packaging of a λDamvector containing EcoRI DNA fragments of Escherichia coli and phage P1. Gene 1:255–80 11. Sternberg N. 1978. Demonstration and analysis of P1 site-speciﬁc recombination using λ-P1 hybrid phages constructed in vitro. Cold Spring Harb. Symp. Quant. Biol. 43:1143–46 12. Sternberg N, Hamilton D. 1981. Bacteriophage P1 site-speciﬁc recombination. I. Recombination between loxPsites. J. Mol. Biol. 150:467–86 13. Takano T. 1971. Bacterial mutant defective in plasmid formation: requirement for the lon+allele. PNAS 68:1469–73 14. Chesney RH, Vapnek D, Scott JR. 1978. Recombination between the plasmid prophages P1 and P7 and theE. coli chromosome. Contrib. Microbiol. Immunol. 6:78–88 15. Sternberg N, Coulby J. 1987. Recognition and cleavage of the bacteriophage P1 packaging site ( pac). I. Differential processing of the cleaved ends in vivo. J. Mol. Biol. 194:453–68 16. Sternberg N, Coulby J. 1987.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  J. Mol. Biol. 150:467–86 13. Takano T. 1971. Bacterial mutant defective in plasmid formation: requirement for the lon+allele. PNAS 68:1469–73 14. Chesney RH, Vapnek D, Scott JR. 1978. Recombination between the plasmid prophages P1 and P7 and theE. coli chromosome. Contrib. Microbiol. Immunol. 6:78–88 15. Sternberg N, Coulby J. 1987. Recognition and cleavage of the bacteriophage P1 packaging site ( pac). I. Differential processing of the cleaved ends in vivo. J. Mol. Biol. 194:453–68 16. Sternberg N, Coulby J. 1987. Recognition and cleavage of the bacteriophage P1 packaging site ( pac). II. Functional limits of pacand location of paccleavage termini. J. Mol. Biol. 194:469–83 17. Austin S, Zeise M, Sternberg N. 1981. A novel role for site-speciﬁc recombination in the maintenance of bacterial replicons. Cell25:729–36 18. Nash HA. 1981. Integration and excision of bacteriophage λ: the mechanism of conservative site speciﬁc recombination. Annu. Rev. Genet. 15:143–67 19. Hoess RH, Ziese M, Sternberg N. 1982. P1 site-speciﬁc recombination: nucleotide sequence of the recombining sites. PNAS 79:3398–402 20. Abremski K, Hoess R, Sternberg N. 1983. Studies on the properties of P1 site-speciﬁc recombination: evidence for topologically unlinked productions following recombination. Cell32:1301–11 38 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 21. Abremski K, Gottesman S. 1981. Site-speciﬁc recombination: Xis-independent excisive recombination of bacteriophage lambda. J. Mol. Biol. 153:67–78 22. Reed RR. 1981. Transposon-mediated site-speciﬁc recombination: a deﬁned in vitro system. Cell25:713– 19 23. Abremski K, Hoess R. 1984. Bacteriophage P1 site-speciﬁc recombination: puriﬁcation and properties of the Cre recombinase protein. J. Biol. Chem. 259:1509–14 24. Hoess R, Abremski K. 1984. Interaction of the bacteriophage P1 recombinase Cre with the recombining siteloxP.PNAS 81:1026–29 25. Hoess R, Abremski K. 1985. Mechanism of strand cleavage and exchange in the Cre- loxsite-speciﬁc recombination system. J. Mol. Biol. 181:351–62 26. Hoess R, Wierzbicki A, Abremski K. 1986. The role of the loxPspacer region in P1 site-speciﬁc recom- bination. Nucleic Acids Res. 14:2287–300 27. Sauer B. 1996. Multiplex Cre/ loxrecombination permits selective site-speciﬁc DNA targeting to both a natural and an engineered site in the yeast genome. Nucleic Acids Res. 24:4608–13 28. Abremski K, Hoess R. 1985. Phage P1 Cre- loxsite-speciﬁc recombination: effects of DNA supercoiling on catenation and knotting of recombinant products. J. Mol. Biol. 184:211–20 29. Sternberg N. 1981. Bacteriophage P1 site-speciﬁc recombination. III. Strand exchange during recombi- nation at loxsites. J. Mol. Biol. 150:603–8 30. Hoess R, Wierzbicki A, Abremski K. 1987. Isolation and characterization of potential intermediates in site-speciﬁc recombination. PNAS 84:6840–44 31. Sternberg N, Sauer B, Hoess R, Abremski K. 1986. Bacteriophage P1 cregene and its regulatory region: evidence for multiple promoters and regulation by DNA methylation. J. Mol. Biol. 187:197–212 32. Argos P, Landy A, Abremski K, Egan JB, Haggard-Ljungquist E, et al. 1986. The integrase family of site-speciﬁc recombinases: regional similarities and global diversity. EMBO J. 5:433–40 33. Abremski K, Hoess R. 1992. Evidence for a second conserved arginine residue in the integrase family of recombination proteins. Protein Eng. 5:87–91 34. Guo F, Gopaul DN, Van Duyne GD. 1997. Structure of Cre recombinase complexed with DNA in a site-speciﬁc recombination synapse. Nature 389:40–46 35. Adams DE, Bliska JB, Cozzarelli NR. 1992. Cre- loxrecombination in Escherichia coli cells: mechanistic differences from the in vitro reaction. J. Mol. Biol. 226:661–73 36. Cohen G. 1983. Electron microscopy study of early lytic replication forms of bacteriophage P1 DNA. Virology 131:159–70 37. Barre FX, Soballe B, Michel B, Aroyo M, Robertson M, et al. 2001.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  Protein Eng. 5:87–91 34. Guo F, Gopaul DN, Van Duyne GD. 1997. Structure of Cre recombinase complexed with DNA in a site-speciﬁc recombination synapse. Nature 389:40–46 35. Adams DE, Bliska JB, Cozzarelli NR. 1992. Cre- loxrecombination in Escherichia coli cells: mechanistic differences from the in vitro reaction. J. Mol. Biol. 226:661–73 36. Cohen G. 1983. Electron microscopy study of early lytic replication forms of bacteriophage P1 DNA. Virology 131:159–70 37. Barre FX, Soballe B, Michel B, Aroyo M, Robertson M, et al. 2001. Circles: the replication-recombination- chromosome segregation connection. PNAS 98:8189–95 38. Sternberg N, Hamilton D, Austin S, Yarmolinsky M, Hoess R. 1981. Site-speciﬁc recombination and its role in the life cycle of bacteriophage P1. Cold Spring Harb. Symp. Quant. Biol. 45:297–309 39. Sauer B. 1987. Functional expression of the cre-loxsite-speciﬁc recombination system in the yeast Saccha- romyces cerevisiae .Mol. Cell. Biol. 7:2087–96 40. Sauer B, Henderson N. 1988. Site-speciﬁc recombination in mammalian cells by the Cre recombinase of bacteriophage P1. PNAS 85:5166–70 41. Sternberg N. 1994. The P1 cloning system past and future. Mamm. Genome 5:397–404 42. Esposito D, Scocca JJ. 1997. The integrase family of tyrosine recombinases: evolution of a conserved active site domain. Nucleic Acids Res. 25:3605–14 43. Begg KJ, Dewar SJ, Donachie WD. 1995. A new Escherichia coli cell division gene, ftsK.J. Bacteriol. 177:6211–22 44. Bigot S, Sivanatan V, Possoz C, Barre FX, Cornet F. 2007. FtsK, a literate chromosome segregation machine. Mol. Microbiol. 64:1434–41 45. Grainge I, Lesterlin C, Sherratt DJ. 2011. Activation of XerCD- difrecombination by the FtsK DNA translocase. Nucleic Acids Res. 39:5140–48 46. Capiaux H, Lesterlin C, P ´erals K, Louarn JM, Cornet F. 2002. A dual role for the FtsK protein in Escherichia coli chromosome segregation. EMBO Rep. 3:532–36 47. Zawadzki P, May PF, Baker RA, Pinkney JN, Kapanidis AN, et al. 2013. Conformational transitions during FtsK translocase activation of individual XerCD- difrecombination complexes. PNAS 110:17302–7 www.annualreviews.org •The Legacy of Nat Sternberg 39 Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02CH02-Yarmolinsky ARI 1 October 2015 7:19 48. Carnoy C, Roten CA. 2009. The dif/Xer recombination systems in proteobacteria. PLOS ONE 4:e6531 49. Hsu PL, Ross W, Landy A. 1980. The λphage attsite: functional limits and interaction with Int protein. Nature 285:85–91 50. Van Duyne GD. 2001. A structural view of Cre- loxP site-speciﬁc recombination. Annu. Rev. Biophys. Biomol. Struct. 30:87–104 51. Pinkney JN, Zawadzki P, Mazauryic J, Arciszewska LK, Sherratt DJ, Kapanidis AN. 2012. Capturing reaction paths and intermediates in Cre- loxP recombination using single molecule ﬂuorescence. PNAS 109:20871–76 52. Warren D, Laxmikanthan G, Landy A. 2008. A chimeric Cre recombinase with regulated directionality. PNAS 105:18278–83 40 Yarmolinsky ·Hoess Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02-FrontMatter ARI 15 October 2015 16:33 Annual Review of Virology Volume 2, 2015 Contents History The Maturation of a Scientist: An Autobiography Bernard Roizman ppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 1 The Legacy of Nat Sternberg: The Genesis of Cre- loxTechnology Michael Yarmolinsky and Ronald Hoess pppppppppppppppppppppppppppppppppppppppppppppppppppppp 25 Ecology and Evolution Biogeography of Viruses in the Sea Cheryl-Emiliane T. Chow and Curtis A. Suttle ppppppppppppppppppppppppppppppppppppppppppppp 41 Role of the Insect Supervectors Bemisia tabaci and Frankliniella occidentalis in the Emergence and Global Spread of Plant Viruses Robert L. Gilbertson, Ozgur Batuman, Craig G. Webster, and Scott Adkins ppppppppppppp 67 Coronavirus Host Range Expansion and Middle East Respiratory Syndrome Coronavirus Emergence: Biochemical Mechanisms and Evolutionary Perspectives Kayla M. Peck, Christina L. Burch, Mark T. Heise, and Ralph S. Baric pppppppppppppppppp 95 Koala Retroviruses: Evolution and Disease Dynamics Wenqin Xu and Maribeth V. Eiden pppppppppppppppppppppppppppppppppppppppppppppppppppppppp 119 Endogenous Retroviruses in the Genomics Era Welkin E. Johnson pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 135 Matters of Size: Genetic Bottlenecks in Virus Infection and Their Potential Impact on Evolution Mark P. Zwart and Santiago F. Elena pppppppppppppppppppppppppppppppppppppppppppppppppppp 161 The Phage-Inducible Chromosomal Islands: A Family of Highly Evolved Molecular Parasites Jos´e R. Penad´ es and Gail E. Christie ppppppppppppppppppppppppppppppppppppppppppppppppppppppp 181 Deep Recombination: RNA and ssDNA Virus Genes in DNA Virus and Host Genomes Kenneth M. Stedman pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 203 xi Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02-FrontMatter ARI 15 October 2015 16:33 Attachment and Cell Entry Hunting Viral Receptors Using Haploid Cells Sirika Pillay and Jan E. Carette ppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 219 Genome Replication, Regulation of Gene Expression, and Biosynthesis Retroviral Integrase: Then and Now Mark D. Andrake and Anna Marie Skalka ppppppppppppppppppppppppppppppppppppppppppppppp 241 Continuous and Discontinuous RNA Synthesis in Coronaviruses Isabel Sola, Fernando Almaz´ an, Sonia Z ´ u˜niga, and Luis Enjuanes pppppppppppppppppppppp 265 Flaviviridae Replication Organelles: Oh, What a Tangled Web We Weave David Paul and Ralf Bartenschlager ppppppppppppppppppppppppppppppppppppppppppppppppppppppp 289 Modulation of the Translational Landscape During Herpesvirus Infection Britt A. Glaunsinger pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 311 Ribosome Proﬁling as a Tool to Decipher Viral Complexity Noam Stern-Ginossar and Nicholas T. Ingolia pppppppppppppppppppppppppppppppppppppppppppp 335 Mechanisms of DNA Packaging by Large Double-Stranded DNA Viruses Venigalla B. Rao and Michael Feiss pppppppppppppppppppppppppppppppppppppppppppppppppppppppp 351 Assembly and Egress Virus-Based Nanoparticles as Versatile Nanomachines Kristopher J. Koudelka, Andrzej S. Pitek, Marianne Manchester, and Nicole F. Steinmetz pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 379 Transformation and Oncogenesis Novel Functions of the Human Papillomavirus E6 Oncoproteins Nicholas A. Wallace and Denise A. Galloway ppppppppppppppppppppppppppppppppppppppppppppp 403 Pathogenesis Parvovirus Family Conundrum: What Makes a Killer? Shweta Kailasan, Mavis Agbandje-McKenna, and Colin R. Parrish ppppppppppppppppppppp 425 Everything You Always Wanted to Know About Rabies Virus (But Were Afraid to Ask) Benjamin M. Davis, Glenn F. Rall, and Matthias J. Schnell pppppppppppppppppppppppppppp 451 Pathophysiological Consequences of Calcium-Conducting Viroporins Joseph M. Hyser and Mary K. Estes pppppppppppppppppppppppppppppppppppppppppppppppppppppppp 473 Virus-Host Interactions: From Unbiased Genetic Screens to Function Holly Ramage and Sara Cherry pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp 497 Viruses in Rodent Colonies: Lessons Learned from Murine Noroviruses Stephanie M. Karst and Christiane E. Wobus ppppppppppppppppppppppppppppppppppppppppppppp 525 xii Contents Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/article.pdf  VI02-FrontMatter ARI 15 October 2015 16:33 Immunity No Love Lost Between Viruses and Interferons Volker Fensterl, Saurabh Chattopadhyay, and Ganes C. Sen ppppppppppppppppppppppppppppp 549 Innate and Adaptive Immune Regulation During Chronic Viral Infections Elina I. Zuniga, Monica Macal, Gavin M. Lewis, and James A. Harker ppppppppppppppp 573 Viral Vectors and Therapeutics Bacteriophage Therapy: Advances in Formulation Strategies and Human Clinical Trials Dieter Vandenheuvel, Rob Lavigne, and Harald Br ¨ ussow pppppppppppppppppppppppppppppppp 599 Errata An online log of corrections to Annual Review of Virology articles may be found at http://www.annualreviews.org/errata/virology Contents xiii Annu. Rev. Virol. 2015.2:25-40. Downloaded from www.annualreviews.org  Access provided by Johns Hopkins University on 12/12/15. For personal use only.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:58 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'75'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994709'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'317ms'), (b'x-request-id', b'529b568f8343f5e36098ff2cc2cf1925'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249893d375d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:58 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'75'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994709'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'317ms'), (b'x-request-id', b'529b568f8343f5e36098ff2cc2cf1925'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249893d375d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  /gid00030/gid00035/gid00032/gid00030/gid00038/gid00001/gid00033/gid00042/gid00045 /gid00001 /gid00048/gid00043/gid00031/gid00028/gid00047/gid00032/gid00046Citation: Bednarek, A.; Cena, A.; Izak, W.; Bigos, J.; Łobocka, M. Functional Dissection of P1 Bacteriophage Holin-like Proteins Reveals the Biological Sense of P1 Lytic System Complexity. Int. J. Mol. Sci. 2022 ,23, 4231. https:// doi.org/10.3390/ijms23084231 Academic Editor: Alicja Wegrzyn Received: 12 December 2021 Accepted: 7 April 2022 Published: 11 April 2022 Publisher’s Note: MDPI stays neutral with regard to jurisdictional claims in published maps and institutional afﬁl- iations. Copyright: © 2022 by the authors. Licensee MDPI, Basel, Switzerland. This article is an open access article distributed under the terms and conditions of the Creative Commons Attribution (CC BY) license (https:// creativecommons.org/licenses/by/ 4.0/).  International Journal of   Molecular Sciences Article Functional Dissection of P1 Bacteriophage Holin-like Proteins Reveals the Biological Sense of P1 Lytic System Complexity Agnieszka Bednarek1, Agata Cena1, Wioleta Izak1, Joanna Bigos2and Małgorzata Łobocka1,* 1Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, 02-106 Warsaw, Poland; a.kielan@ibb.waw.pl (A.B.); agatacena@yahoo.com (A.C.); wioleta.woznica@gmail.com (W.I.) 2Autonomous Department of Microbiology, Faculty of Agriculture and Biology, Warsaw University of Life Sciences, SGGW, 02-776 Warsaw, Poland; nasierowskajoanna@gmail.com *Correspondence: lobocka@ibb.waw.pl Abstract: P1 is a model temperate myovirus. It infects different Enterobacteriaceae and can develop lytically or form lysogens. Only some P1 adaptation strategies to propagate in different hosts are known. An atypical feature of P1 is the number and organization of cell lysis-associated genes. In addition to SAR-endolysin Lyz, holin LydA, and antiholin LydB, P1 encodes other predicted holins, LydC and LydD. LydD is encoded by the same operon as Lyz, LydA and LydB are encoded by an unlinked operon, and LydC is encoded by an operon preceding the lydA gene. By analyzing the phenotypes of P1 mutants in known or predicted holin genes, we show that all the products of these genes cooperate with the P1 SAR-endolysin in cell lysis and that LydD is a pinholin. The contributions of holins/pinholins to cell lysis by P1 appear to vary depending on the host of P1 and the bacterial growth conditions. The pattern of morphological transitions characteristic of SAR-endolysin–pinholin action dominates during lysis by wild-type P1, but in the case of lydC lydD mutant it changes to that characteristic of classical endolysin-pinholin action. We postulate that the complex lytic system facilitates P1 adaptation to various hosts and their growth conditions. Keywords: bacteriophage P1; bacterial cell lysis; holin; pinholin; endolysin; SAR-endolysin; Escherichia coli ;Enterobacter cloacae 1. Introduction The lytic development of bacteriophages in bacterial cells ends with cell lysis to release the progeny phages. The effectiveness of lysis and its precise control in time is crucial for the evolutionary fate of a given phage [ 1]. Premature lysis leads to the release of immature, noninfective phages. Delayed lysis leads to failure in evolution with competing phages.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  We postulate that the complex lytic system facilitates P1 adaptation to various hosts and their growth conditions. Keywords: bacteriophage P1; bacterial cell lysis; holin; pinholin; endolysin; SAR-endolysin; Escherichia coli ;Enterobacter cloacae 1. Introduction The lytic development of bacteriophages in bacterial cells ends with cell lysis to release the progeny phages. The effectiveness of lysis and its precise control in time is crucial for the evolutionary fate of a given phage [ 1]. Premature lysis leads to the release of immature, noninfective phages. Delayed lysis leads to failure in evolution with competing phages. Typically, double-stranded DNA phages encode two kinds of essential proteins to lyse cells at the release of phage progeny (Figure 1). One of them, endolysin, destroys the cell wall by digesting chemical bonds essential for the integrity of cell wall peptidoglycan (PG). Most phages produce endolysins that are unable to cross the cytoplasmic membrane (CM) and require holins, proteins that form large, nonspeciﬁc membrane pores (holes) [ 2–4]. In the classical paradigm, endolysin is free in the cytoplasm and the holin, dispersed in the cellular membrane, is triggered to assemble into large rafts, which create very large sites of membrane disruption (Figure 1A). This provides fully active endolysins with access to the cell wall peptidoglycan, which results in the local disruption of the cell wall envelope and escape of the cell contents. As an exception to this classical paradigm, some phages were found to produce endolysins that can cross the cytoplasmic membrane without the help of canonical holins [ 5–10]. In the phages of Gram-negative bacteria, such endolysins contain at their N-terminus a so-called signal arrest release (SAR) domain, a non-cleavable type II signal anchor [ 3,9–16]. Thus, in a recently described second paradigm, the inactive SAR-endolysin anchored to the cytoplasmic membrane is released and folded to its active form when triggered by the depolarization of the membrane by so-called pinholin, a Int. J. Mol. Sci. 2022 ,23, 4231. https://doi.org/10.3390/ijms23084231 https://www.mdpi.com/journal/ijms', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 2 of 20 small holin-like protein that can form a multitude of nm scale holes (pinholes) in the CM (Figure 1B) [6,17–21]. Figure 1. Two paradigms of murein degradation: ( A) canonical holin–endolysin and ( B) pinholin– SAR-endolysin pathways of murein degradation. Reproduced with permission from Young, 2014 [ 22] published in J. Microbiol. 2014, Springer. Only the CM (gray rectangle) and PG (grid) are shown. The cartoon series begins early in late gene expression (morphogenesis period) and progresses downwards. Holins (blue ovals in ( A)), pinholins (blue ovals in ( B)), soluble endolysins (green ovals with open “active site cleft”), and SAR-endolysins (green ovals with N-terminal SAR domains) depicted either in TMD conformation (green rectangle in top two panels under ( B)) or extracted, refolded conformation (jack-knifed green rectangles in bottom panel), attached to the green globular (enzymatic and PG binding) domain. Holins accumulate in the CM (top two panels of ( A,B)). Note that the prototype holin, S105 of phage λ, and pinholin, S2168 protein of lambdoid phage 21, accumulate as homodimers or heterodimers with their cognate antiholins (holin antagonists); however, holins are represented as single ovals here, for simplicity. The bottom panels represent the triggered cells, in which the canonical holins form a large (“micron-scale”) hole ( A) or the pinholins form many small heptameric pinholes ( B) in the CM. Proton motive force (pmf) of CM is indicated for each stage. Despite the different mechanisms of action, the triggering of lysis by either holins or pinholins is associated with the collapse of proton motive force (PMF) and cessation of macromolecular synthesis, leading to cell death [ 9,23,24]). Thus, in addition to their role in the liberation of endolysins or SAR-endolysins to the cell wall, holins/pinholins are postulated to be essential for cell lysis by causing cell death. This, in turn, would allow lysis to occur due to the accumulation of endolysin-mediated cell wall damage, which otherwise would be readily repaired in rapidly growing cells [24]. The triggering of cell lysis occurs precisely when mature phages are ready to be re- leased. It is associated with the achievement of a threshold concentration of holin/pinholin in the cytoplasmic membrane, leading to the formation of holes/pinholes and the release or activation of endolysin/SAR-endolysin [ 20,25] (reviewed in [ 26]). In certain phages, the timing of lysis triggering is inﬂuenced by so-called antiholins, proteins that act by forming membrane complexes with holins or regulating holin function in another, unknown way (reviewed in [ 9]). Additionally, the phages of Gram-negative bacteria encode so-called spanins, proteins that destroy the outer membrane (OM) of lysing cells by spanning it with the cytoplasmic membrane through the endolysin/SAR-endolysin-mediated damage of the cell wall [ 27–30]. Most often, spanins are either two-component complexes of OM lipoprotein (o-spanin) and a type II inner membrane (IM) protein (i-spanin), although some', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 3 of 20 are produced in the form of a single polypeptide of each of these functions (reviewed in [ 9]). In laboratory cultures grown with shaking, spannins are not essential for lysis. Typically, phages encode one set of lytic proteins, and typically endolysin, holin, and antiholin are encoded by the same operon. The antiholin gene is often embedded within the holin gene, and spanin genes are immediately downstream, as in the case ofλphage [ 31–38]. P1, a model temperate bacteriophage of known genomic sequence (93.6 kb) and wide host range among enterobacteria, is atypical in that respect [ 39–41]. Three P1 proteins, a prototypical SAR–endolysin Lyz of lysozyme activity, holin LydA, and antiholin LydB, have been shown to cooperate in the lysis of E. coli when produced from their genes cloned in a plasmid under the control of the inducible promoter [ 5,41]. Thus, LydA has been considered a primary holin of P1. However, Lyz is encoded by a late operon, unlinked to the late daroperon, which encodes LydA and LydB and also antirestriction and capsid morphogenesis-associated functions (Figure 2) [ 40,42–48]. Moreover, in addition to LydA, the P1 genome encodes two other proteins with predicted holin features, LydC and LydD [ 40]. LydC (93 aa) is encoded by a monocistronic operon preceded by the putative early promoter and immediately preceding the daroperon (Figure 2). LydD (84 aa) is encoded by the second gene of the lyzoperon, as would be expected for a pinholin. Its amino acid sequence differs from those of LydA and LydC, which are 29% identical to each other. However, like LydA and LydC, LydD contains two predicted transmembrane helices, and its C-terminus with an array of ﬁve arginine residues is highly basic. The beginning of the lydD gene overlaps with the end of lyzand the 3′moiety of lydD overlaps with the beginning of a small gene of unknown function, designated as lydE . Int. J. Mol. Sci.  2022 , 23, 4231 3 of 21     some are produced in the form of a single poly peptide of each of these functions (reviewed  in [9]). In laboratory cultures grown with sh aking, spannins are no t essential for lysis.  Typically, phages encode one set of lytic proteins, and typically endolysin, holin, and  antiholin are encoded by the same operon. The antiholin gene is often embedded within the  holin gene, and spanin genes are immedi ately downstream, as in the case of λ phage [31–38].  P1, a model temperate bacteriophage of know n genomic sequence (93.6 kb) and wide host  range among enterobacteria, is atypical in that  respect [39–41]. Three P1 proteins, a proto- typical SAR–endolysin Lyz of lysozyme acti vity, holin LydA, and antiholin LydB, have  been shown to cooperate in the lysis of E. coli  when produced from their genes cloned in  a plasmid under the control of the inducible promoter [5,41]. Thus, LydA has been con- sidered a primary holin of P1. However, Lyz is encoded by a late operon, unlinked to the  late dar operon, which encodes LydA and LydB and also antirestriction and capsid mor- phogenesis-associated functions (Figure 2) [40, 42–48]. Moreover, in addition to LydA, the  P1 genome encodes two other proteins with  predicted holin features, LydC and LydD   [40].', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Thus, LydA has been con- sidered a primary holin of P1. However, Lyz is encoded by a late operon, unlinked to the  late dar operon, which encodes LydA and LydB and also antirestriction and capsid mor- phogenesis-associated functions (Figure 2) [40, 42–48]. Moreover, in addition to LydA, the  P1 genome encodes two other proteins with  predicted holin features, LydC and LydD   [40]. LydC (93 aa) is encoded by a monocist ronic operon preceded by the putative early  promoter and immediately preceding the dar operon (Figure 2). LydD (84 aa) is encoded  by the second gene of the lyz operon, as would be expected for a pinholin. Its amino acid  sequence differs from those of LydA and Ly dC, which are 29% identical to each other.  However, like LydA and LydC, LydD contai ns two predicted transmembrane helices, and  its C-terminus with an array of five arginine residues is highly basic. The beginning of the  lydD  gene overlaps with the end of lyz and the 3′ moiety of lydD overlaps with the beginning  of a small gene of unknown function, designated as lydE.    Figure 2. Genetic organization of th e SAR-endolysin gene ( lyz) and known or predicted holin genes  in genomes of P1 and P7 phages. Known and pred icted holin genes are in gray. Additional genes  shown encode antiholin ( lydB ) and a putative protein of unknown function ( lydE ). Arrows indicate  promoters. LP, late phage promoter . The ruler at the bottom correspon ds to nucleotide positions as  they appear in the published P1 genome (GenBank acc. no. NC_005856.1).  The spanin genes of P1, ppfA  (o-spanin) and mlp (i-spanin) partially overlap and are  located in yet another unlinked operon. They were identified based on the predicted fea- tures of their products, but their function ha s not been verified experimentally [27]. In  laboratory cultures grown with shaking, it is not essential for lysis.  The organization of the known and putative cell lysis-associated genes of P1 is con- served in a P1-related phage, P7 (GenBank a cc. no. AF503408). The amino acid sequences  of their products, except that of LydC, are iden tical or nearly identical in these two phages.  The complexity of P1/P7 lytic functions poses a question about its biological sense. In  this work, we dissect the functions of known and predicted P1 holins, show the different  impacts of particular holins on the cell lysis of different P1 hosts, and the concerted con- tribution of all of them to efficient lysis in a programmed time. Additionally, we demon- strate that LydD is a pinholin, and that it has a dominant influence on the morphological  transitions of E. coli  and Enterobacter cloacae  cells during lysis mediated by P1.     Figure 2. Genetic organization of the SAR-endolysin gene ( lyz) and known or predicted holin genes in genomes of P1 and P7 phages. Known and predicted holin genes are in gray. Additional genes shown encode antiholin ( lydB ) and a putative protein of unknown function ( lydE ). Arrows indicate promoters. LP , late phage promoter. The ruler at the bottom corresponds to nucleotide positions as they appear in the published P1 genome (GenBank acc. no. NC_005856.1). The spanin genes of P1, ppfA (o-spanin) and mlp(i-spanin) partially overlap and are located in yet another unlinked operon. They were identiﬁed based on the predicted features of their products, but their function has not been veriﬁed experimentally [ 27]. In laboratory cultures grown with shaking, it is not essential for lysis.', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Known and predicted holin genes are in gray. Additional genes shown encode antiholin ( lydB ) and a putative protein of unknown function ( lydE ). Arrows indicate promoters. LP , late phage promoter. The ruler at the bottom corresponds to nucleotide positions as they appear in the published P1 genome (GenBank acc. no. NC_005856.1). The spanin genes of P1, ppfA (o-spanin) and mlp(i-spanin) partially overlap and are located in yet another unlinked operon. They were identiﬁed based on the predicted features of their products, but their function has not been veriﬁed experimentally [ 27]. In laboratory cultures grown with shaking, it is not essential for lysis. The organization of the known and putative cell lysis-associated genes of P1 is con- served in a P1-related phage, P7 (GenBank acc. no. AF503408). The amino acid sequences of their products, except that of LydC, are identical or nearly identical in these two phages. The complexity of P1/P7 lytic functions poses a question about its biological sense. In this work, we dissect the functions of known and predicted P1 holins, show the differ- ent impacts of particular holins on the cell lysis of different P1 hosts, and the concerted contribution of all of them to efﬁcient lysis in a programmed time. Additionally, we demon- strate that LydD is a pinholin, and that it has a dominant inﬂuence on the morphological transitions of E. coli and Enterobacter cloacae cells during lysis mediated by P1. 2. Results 2.1. Inﬂuence of LydD on the Phenotype of Cells Producing the P1 Lyz Endolysin The insertional inactivation of the P1 lyzgene in a P1 variant encoding the thermosen- sitive phage repressor (C1-100) results in the inability of lysogens to lyse upon the thermal', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 4 of 20 induction of P1 lytic development (Figure S1). This is consistent with the inability of the P1 amber mutant in the region of lyzto cause lysis of the host cell [ 43] and indicates that the Lyz endolysin is the only P1 muralytic enzyme required for P1-mediated cellular lysis. As a SAR domain-containing endolysin, Lyz can cause lysis in the absence of holins and can therefore be cloned only under repression control. This was achieved by cloning it under the control of the Serratia marcescens trp promoter with the trprepressor, and by growing cells in the presence of glucose. As shown in Figure 2, P1 encodes at least three holin-like proteins, and we undertook a study to clarify their roles. The induction of the expression of lydA and lyzgenes cloned in a plasmid was reported to cause E. coli cell lysis in the absence of any other P1 protein [ 42]. Thus, LydA has been described as the primary Lyz-associated holin. This seemed odd, since SAR-domain lysins require a pinholin, which is classically encoded adjacent to the lysin. There is a putative pinholin gene, lydD , whose 5′end overlaps the 3′end of the lyz gene [ 40]. Indeed, we observed that cells containing a plasmid with the entire P1/P7 lyz operon ( lyz+lydD+lydE+) under the control of an inducible promoter lysed faster and more efﬁciently upon the induction of this promoter than cells containing only the lyzgene in an isogenic plasmid (Figure 3). The presence of lydE did not inﬂuence the lysis phenotype, indicating that LydD protein alone can cooperate with Lyz in cell lysis in the absence of LydA and other P1 proteins. Int. J. Mol. Sci.  2022 , 23, 4231 4 of 21     2. Results  2.1. Influence of LydD on the Phenotype of Cells Producing the P1 Lyz Endolysin  The insertional inactivation of the P1 lyz gene in a P1 variant encoding the thermo- sensitive phage repressor (C1-100) results in the inability of lysogens to lyse upon the  thermal induction of P1 lytic development (Figure S1). This is consistent with the inability of the P1 amber mutant in the region of lyz to cause lysis of the host cell [43] and indicates  that the Lyz endolysin is the only P1 muraly tic enzyme required for P1-mediated cellular  lysis. As a SAR domain-containing endolysin, Lyz can cause lysis in the absence of holins  and can therefore be cloned only under repression control. This was achieved by cloning  it under the control of the Serratia marcescens  trp promoter with the trp repressor, and by  growing cells in the presence of glucose.  As shown in Figure 2, P1 encodes at least three holin-like proteins, and we undertook  a study to clarify their roles. The induction of the expression of lydA  and lyz genes cloned  in a plasmid was reported to cause E. coli  cell lysis in the absence of any other P1 protein  [42]. Thus, LydA has been described as the primary Lyz-associated holin. This seemed  odd, since SAR-domain lysins require a pinholin , which is classically encoded adjacent to  the lysin. There is a putative pinholin gene, lydD , whose 5 ′ end overlaps the 3 ′ end of the  lyz gene [40]. Indeed, we observed that cells containing a plasmid with the entire P1/P7  lyz operon ( lyz +lydD+lydE+) under the control of an inducible promoter lysed faster and  more efficiently upon the induction of this  promoter than cells containing only the  lyz  gene in an isogenic plasmid (Figure 3).', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Thus, LydA has been described as the primary Lyz-associated holin. This seemed  odd, since SAR-domain lysins require a pinholin , which is classically encoded adjacent to  the lysin. There is a putative pinholin gene, lydD , whose 5 ′ end overlaps the 3 ′ end of the  lyz gene [40]. Indeed, we observed that cells containing a plasmid with the entire P1/P7  lyz operon ( lyz +lydD+lydE+) under the control of an inducible promoter lysed faster and  more efficiently upon the induction of this  promoter than cells containing only the  lyz  gene in an isogenic plasmid (Figure 3). The presence of lydE did not influence the lysis  phenotype, indicating that LydD protein alone can cooperate with Lyz in cell lysis in the  absence of LydA and other P1 proteins.    Figure 3. Changes in the optical density of  E. coli  cells carrying the intact P7 lyz operon or its trun- cated versions in a multicopy plasmid under the control of the S. marcescens trp operon promoter- operator region. Cells containing  the relevant plasmid (pACE3 [ lyz+lydD+lydE+], pACE4 [ lyz+lydD+]  or pACE5 [ lyz+]) and the pRPG18 plasmid as a source of trp repressor (TrpR) were grown overnight  in LB medium supplemented with ampicillin and ch loramphenicol at 37 °C with vigorous shaking,  diluted 100 × in fresh LB with the same antibiot ics, and further grown un der similar conditions.  When the optical density of the cell cultures (OD 600) reached about 0.2, they were supplemented  with β-indole-acrylic acid (20 µg/mL) to induce transcription from the trp promoter. Changes in  culture optical density were further monitored sp ectrophotometrically. Genes whose transcription  from the trp operon promoter was induced in particular cultures are marked with arrows. The graph  Figure 3. Changes in the optical density of E. coli cells carrying the intact P7 lyzoperon or its truncated versions in a multicopy plasmid under the control of the S. marcescens trp operon promoter-operator region. Cells containing the relevant plasmid (pACE3 [ lyz+lydD+lydE+], pACE4 [ lyz+lydD+] or pACE5 [lyz+]) and the pRPG18 plasmid as a source of trprepressor (TrpR) were grown overnight in LB medium supplemented with ampicillin and chloramphenicol at 37◦C with vigorous shaking, diluted 100×in fresh LB with the same antibiotics, and further grown under similar conditions. When the optical density of the cell cultures (OD 600) reached about 0.2, they were supplemented with β-indole-acrylic acid (20 µg/mL) to induce transcription from the trppromoter. Changes in culture optical density were further monitored spectrophotometrically. Genes whose transcription from the trpoperon promoter was induced in particular cultures are marked with arrows. The graph shows the average results of the measurement of changes in the optical density of cell cultures in at least three independent experiments. Bars represent the standard deviation. 2.2. Lethal Effect of Cloned lydC or lydA on E. coli Cells In the P1 genome, the lydA gene is immediately preceded by lydC , whose product has features characteristic of class II holins, as does LydA [ 40]. Our attempts to clone the lydC orlydA gene in the ColE1-derived expression vector (pGBT30) led to the formation of only', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 5 of 20 a few small colonies of transformants. They either did not form colonies upon passaging or could be transferred to a liquid medium but stopped growing upon the induction of cloned gene expression. The growth cessation and lethal phenotypes of cells producing LydC or LydA observed here are consistent with the previously observed phenotypes of bacterial cells expressing cloned holin genes (see, e.g., [ 38,49]). They were attributed to the holin-mediated cell membrane puncturing or disruption of the proton motive force, resulting in cell death. 2.3. Effects of LydA, LydC, or LydD Depletion on P1-Mediated Cell Lysis To test the contributions of LydA, LydC, and LydD to the P1-mediated lysis of E. coli cells, we introduced to the P1 genome, by recombinational replacements, the kanamycin resistance cassette or indels inactivating the lydA ,lydC , orlydD gene. Additionally, P1 vari- ants that carried the combinations of mutations in any two of these genes were constructed. The lydA mutant and the double lydA lydC mutant formed plaques on a layer of indicator strain cells ( E. coli N99) when plates with infected cells were incubated at 42◦C (under conditions non-permissive for the formation of lysogens). In the case of the double lydA lydD mutant, no plaques could be observed. This shows that LydD alone is sufﬁcient to cooperate with Lyz in cell lysis by P1, but LydC alone is not, at least under the conditions of our experiment. In liquid medium phages deprived of functional LydC or LydD, or both these proteins could lyse E. coli cells, although with a delay. The delay differed slightly in the case of two different lydD mutants: lydD ::kanRand lydD∆16-81 (Figure 4A). In contrast to that, phages deprived of LydA did not lyse cells, at least for the duration of the experiment. Taken together, our results demonstrate that LydA, LydC and LydD contribute to the P1-mediated E. coli cell lysis in a programmed time under these conditions. However, while LydA appears to be the main holin involved in P1-mediated cell lysis, in liquid medium, the functions of LydC and LydD are accessory. Int. J. Mol. Sci.  2022 , 23, 4231 6 of 21       Figure 4. Changes in the optical density of cultures of ( A) E. coli  and ( B) E. cloacae  cells containing  wild-type P1 or P1 mutants depleted of particular holin functions, upon the thermal induction of P1  lytic development. The graphs show the average re sults of measurement of changes in the optical  density of cell cultures in at least three independ ent experiments. The legend applies to both charts.  Bars represent the standard deviation.  2.4. Influence of the P1 lyz Gene Replacement with the λ R Gene on P1-Mediated Cell Lysis  In certain phages that encode SAR-endolysins, the onset of lysis is associated with  the action of pinholins, and typically the SA R-endolysin-encoding genes of these phages  are in the same operons as the genes for th e accompanying pinholins [11–13,16,19]. Thus,  we tested whether the P1 lydD  gene could encode a pinholi n. P1 bacteriophage mutants  with the lyz gene replaced with the phage λ R gene (encoding a canonical endolysin) and  with various holin genes inactive were tested for the ability to lyse cells (Figure 5A,B). The  replacement of lyz with R delayed the lysis of E. coli  cells.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  2.4. Influence of the P1 lyz Gene Replacement with the λ R Gene on P1-Mediated Cell Lysis  In certain phages that encode SAR-endolysins, the onset of lysis is associated with  the action of pinholins, and typically the SA R-endolysin-encoding genes of these phages  are in the same operons as the genes for th e accompanying pinholins [11–13,16,19]. Thus,  we tested whether the P1 lydD  gene could encode a pinholi n. P1 bacteriophage mutants  with the lyz gene replaced with the phage λ R gene (encoding a canonical endolysin) and  with various holin genes inactive were tested for the ability to lyse cells (Figure 5A,B). The  replacement of lyz with R delayed the lysis of E. coli  cells. The lack of functional lydD  did  not influence the lysis by the replacement muta nt, suggesting that Ly dD protein does not  contribute to lysis with the participation of R. No lysis could be seen in the case of replace- ment mutant lacking functional lydA , independent of whether lydC  and lydD  genes were  intact or mutated. This conforms to the func tion of LydA as a canonical holin and is con- sistent with the essential role of LydA in lysis of E. coli  cells by P1 in liquid cell cultures.  The lack of functional lydC  in the replacement mutant slig htly accelerated the lysis, sug- gesting a possible role of LydC in the regulation of lysis timing.  Contrary to E. coli , the lyz to R replacement did not have a significant influence on  the onset of the lysis of E. cloacae  cells (Figure 5C,D). However, as in E. coli , the lack of  functional lydD  did not significantly influence the lysis by the replacement mutant, sug- gesting that LydD protein does not contribute to lysis with the participation of R. In sup- port of that, the E. cloacae  lysogen of the replacement mutant deprived of LydA, or LydA  and LydC did not lyse upon the induction of P1 lytic development, even though LydA is  not essential for the P1-mediated lysis of E. cloacae  in liquid cultures , and even though the  lydD  gene was intact in the mutant. Clearly, LydD alone cannot cooperate with R endoly- sin in cell lysis, as if it was not a canonical holin but a pinholin.  The main protein cooperating with λ R in the lysis of E. cloacae  cells grown in liquid  medium appeared to be LydA. The replacement mutant lacking functional lydA  did not  Figure 4. Changes in the optical density of cultures of ( A)E. coli and ( B)E. cloacae cells containing wild-type P1 or P1 mutants depleted of particular holin functions, upon the thermal induction of P1 lytic development. The graphs show the average results of measurement of changes in the optical density of cell cultures in at least three independent experiments. The legend applies to both charts. Bars represent the standard deviation.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  /gid00030/gid00035/gid00032/gid00030/gid00038/gid00001/gid00033/gid00042/gid00045 /gid00001 /gid00048/gid00043/gid00031/gid00028/gid00047/gid00032/gid00046Citation: Bednarek, A.; Cena, A.; Izak, W.; Bigos, J.; Łobocka, M. Functional Dissection of P1 Bacteriophage Holin-like Proteins Reveals the Biological Sense of P1 Lytic System Complexity. Int. J. Mol. Sci. 2022 ,23, 4231. https:// doi.org/10.3390/ijms23084231 Academic Editor: Alicja Wegrzyn Received: 12 December 2021 Accepted: 7 April 2022 Published: 11 April 2022 Publisher’s Note: MDPI stays neutral with regard to jurisdictional claims in published maps and institutional afﬁl- iations. Copyright: © 2022 by the authors. Licensee MDPI, Basel, Switzerland. This article is an open access article distributed under the terms and conditions of the Creative Commons Attribution (CC BY) license (https:// creativecommons.org/licenses/by/ 4.0/).  International Journal of   Molecular Sciences Article Functional Dissection of P1 Bacteriophage Holin-like Proteins Reveals the Biological Sense of P1 Lytic System Complexity Agnieszka Bednarek1, Agata Cena1, Wioleta Izak1, Joanna Bigos2and Małgorzata Łobocka1,* 1Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, 02-106 Warsaw, Poland; a.kielan@ibb.waw.pl (A.B.); agatacena@yahoo.com (A.C.); wioleta.woznica@gmail.com (W.I.) 2Autonomous Department of Microbiology, Faculty of Agriculture and Biology, Warsaw University of Life Sciences, SGGW, 02-776 Warsaw, Poland; nasierowskajoanna@gmail.com *Correspondence: lobocka@ibb.waw.pl Abstract: P1 is a model temperate myovirus. It infects different Enterobacteriaceae and can develop lytically or form lysogens. Only some P1 adaptation strategies to propagate in different hosts are known. An atypical feature of P1 is the number and organization of cell lysis-associated genes. In addition to SAR-endolysin Lyz, holin LydA, and antiholin LydB, P1 encodes other predicted holins, LydC and LydD. LydD is encoded by the same operon as Lyz, LydA and LydB are encoded by an unlinked operon, and LydC is encoded by an operon preceding the lydA gene. By analyzing the phenotypes of P1 mutants in known or predicted holin genes, we show that all the products of these genes cooperate with the P1 SAR-endolysin in cell lysis and that LydD is a pinholin. The contributions of holins/pinholins to cell lysis by P1 appear to vary depending on the host of P1 and the bacterial growth conditions. The pattern of morphological transitions characteristic of SAR-endolysin–pinholin action dominates during lysis by wild-type P1, but in the case of lydC lydD mutant it changes to that characteristic of classical endolysin-pinholin action. We postulate that the complex lytic system facilitates P1 adaptation to various hosts and their growth conditions. Keywords: bacteriophage P1; bacterial cell lysis; holin; pinholin; endolysin; SAR-endolysin; Escherichia coli ;Enterobacter cloacae 1. Introduction The lytic development of bacteriophages in bacterial cells ends with cell lysis to release the progeny phages. The effectiveness of lysis and its precise control in time is crucial for the evolutionary fate of a given phage [ 1]. Premature lysis leads to the release of immature, noninfective phages. Delayed lysis leads to failure in evolution with competing phages.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  We postulate that the complex lytic system facilitates P1 adaptation to various hosts and their growth conditions. Keywords: bacteriophage P1; bacterial cell lysis; holin; pinholin; endolysin; SAR-endolysin; Escherichia coli ;Enterobacter cloacae 1. Introduction The lytic development of bacteriophages in bacterial cells ends with cell lysis to release the progeny phages. The effectiveness of lysis and its precise control in time is crucial for the evolutionary fate of a given phage [ 1]. Premature lysis leads to the release of immature, noninfective phages. Delayed lysis leads to failure in evolution with competing phages. Typically, double-stranded DNA phages encode two kinds of essential proteins to lyse cells at the release of phage progeny (Figure 1). One of them, endolysin, destroys the cell wall by digesting chemical bonds essential for the integrity of cell wall peptidoglycan (PG). Most phages produce endolysins that are unable to cross the cytoplasmic membrane (CM) and require holins, proteins that form large, nonspeciﬁc membrane pores (holes) [ 2–4]. In the classical paradigm, endolysin is free in the cytoplasm and the holin, dispersed in the cellular membrane, is triggered to assemble into large rafts, which create very large sites of membrane disruption (Figure 1A). This provides fully active endolysins with access to the cell wall peptidoglycan, which results in the local disruption of the cell wall envelope and escape of the cell contents. As an exception to this classical paradigm, some phages were found to produce endolysins that can cross the cytoplasmic membrane without the help of canonical holins [ 5–10]. In the phages of Gram-negative bacteria, such endolysins contain at their N-terminus a so-called signal arrest release (SAR) domain, a non-cleavable type II signal anchor [ 3,9–16]. Thus, in a recently described second paradigm, the inactive SAR-endolysin anchored to the cytoplasmic membrane is released and folded to its active form when triggered by the depolarization of the membrane by so-called pinholin, a Int. J. Mol. Sci. 2022 ,23, 4231. https://doi.org/10.3390/ijms23084231 https://www.mdpi.com/journal/ijms', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 2 of 20 small holin-like protein that can form a multitude of nm scale holes (pinholes) in the CM (Figure 1B) [6,17–21]. Figure 1. Two paradigms of murein degradation: ( A) canonical holin–endolysin and ( B) pinholin– SAR-endolysin pathways of murein degradation. Reproduced with permission from Young, 2014 [ 22] published in J. Microbiol. 2014, Springer. Only the CM (gray rectangle) and PG (grid) are shown. The cartoon series begins early in late gene expression (morphogenesis period) and progresses downwards. Holins (blue ovals in ( A)), pinholins (blue ovals in ( B)), soluble endolysins (green ovals with open “active site cleft”), and SAR-endolysins (green ovals with N-terminal SAR domains) depicted either in TMD conformation (green rectangle in top two panels under ( B)) or extracted, refolded conformation (jack-knifed green rectangles in bottom panel), attached to the green globular (enzymatic and PG binding) domain. Holins accumulate in the CM (top two panels of ( A,B)). Note that the prototype holin, S105 of phage λ, and pinholin, S2168 protein of lambdoid phage 21, accumulate as homodimers or heterodimers with their cognate antiholins (holin antagonists); however, holins are represented as single ovals here, for simplicity. The bottom panels represent the triggered cells, in which the canonical holins form a large (“micron-scale”) hole ( A) or the pinholins form many small heptameric pinholes ( B) in the CM. Proton motive force (pmf) of CM is indicated for each stage. Despite the different mechanisms of action, the triggering of lysis by either holins or pinholins is associated with the collapse of proton motive force (PMF) and cessation of macromolecular synthesis, leading to cell death [ 9,23,24]). Thus, in addition to their role in the liberation of endolysins or SAR-endolysins to the cell wall, holins/pinholins are postulated to be essential for cell lysis by causing cell death. This, in turn, would allow lysis to occur due to the accumulation of endolysin-mediated cell wall damage, which otherwise would be readily repaired in rapidly growing cells [24]. The triggering of cell lysis occurs precisely when mature phages are ready to be re- leased. It is associated with the achievement of a threshold concentration of holin/pinholin in the cytoplasmic membrane, leading to the formation of holes/pinholes and the release or activation of endolysin/SAR-endolysin [ 20,25] (reviewed in [ 26]). In certain phages, the timing of lysis triggering is inﬂuenced by so-called antiholins, proteins that act by forming membrane complexes with holins or regulating holin function in another, unknown way (reviewed in [ 9]). Additionally, the phages of Gram-negative bacteria encode so-called spanins, proteins that destroy the outer membrane (OM) of lysing cells by spanning it with the cytoplasmic membrane through the endolysin/SAR-endolysin-mediated damage of the cell wall [ 27–30]. Most often, spanins are either two-component complexes of OM lipoprotein (o-spanin) and a type II inner membrane (IM) protein (i-spanin), although some', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 3 of 20 are produced in the form of a single polypeptide of each of these functions (reviewed in [ 9]). In laboratory cultures grown with shaking, spannins are not essential for lysis. Typically, phages encode one set of lytic proteins, and typically endolysin, holin, and antiholin are encoded by the same operon. The antiholin gene is often embedded within the holin gene, and spanin genes are immediately downstream, as in the case ofλphage [ 31–38]. P1, a model temperate bacteriophage of known genomic sequence (93.6 kb) and wide host range among enterobacteria, is atypical in that respect [ 39–41]. Three P1 proteins, a prototypical SAR–endolysin Lyz of lysozyme activity, holin LydA, and antiholin LydB, have been shown to cooperate in the lysis of E. coli when produced from their genes cloned in a plasmid under the control of the inducible promoter [ 5,41]. Thus, LydA has been considered a primary holin of P1. However, Lyz is encoded by a late operon, unlinked to the late daroperon, which encodes LydA and LydB and also antirestriction and capsid morphogenesis-associated functions (Figure 2) [ 40,42–48]. Moreover, in addition to LydA, the P1 genome encodes two other proteins with predicted holin features, LydC and LydD [ 40]. LydC (93 aa) is encoded by a monocistronic operon preceded by the putative early promoter and immediately preceding the daroperon (Figure 2). LydD (84 aa) is encoded by the second gene of the lyzoperon, as would be expected for a pinholin. Its amino acid sequence differs from those of LydA and LydC, which are 29% identical to each other. However, like LydA and LydC, LydD contains two predicted transmembrane helices, and its C-terminus with an array of ﬁve arginine residues is highly basic. The beginning of the lydD gene overlaps with the end of lyzand the 3′moiety of lydD overlaps with the beginning of a small gene of unknown function, designated as lydE . Int. J. Mol. Sci.  2022 , 23, 4231 3 of 21     some are produced in the form of a single poly peptide of each of these functions (reviewed  in [9]). In laboratory cultures grown with sh aking, spannins are no t essential for lysis.  Typically, phages encode one set of lytic proteins, and typically endolysin, holin, and  antiholin are encoded by the same operon. The antiholin gene is often embedded within the  holin gene, and spanin genes are immedi ately downstream, as in the case of λ phage [31–38].  P1, a model temperate bacteriophage of know n genomic sequence (93.6 kb) and wide host  range among enterobacteria, is atypical in that  respect [39–41]. Three P1 proteins, a proto- typical SAR–endolysin Lyz of lysozyme acti vity, holin LydA, and antiholin LydB, have  been shown to cooperate in the lysis of E. coli  when produced from their genes cloned in  a plasmid under the control of the inducible promoter [5,41]. Thus, LydA has been con- sidered a primary holin of P1. However, Lyz is encoded by a late operon, unlinked to the  late dar operon, which encodes LydA and LydB and also antirestriction and capsid mor- phogenesis-associated functions (Figure 2) [40, 42–48]. Moreover, in addition to LydA, the  P1 genome encodes two other proteins with  predicted holin features, LydC and LydD   [40].', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Thus, LydA has been con- sidered a primary holin of P1. However, Lyz is encoded by a late operon, unlinked to the  late dar operon, which encodes LydA and LydB and also antirestriction and capsid mor- phogenesis-associated functions (Figure 2) [40, 42–48]. Moreover, in addition to LydA, the  P1 genome encodes two other proteins with  predicted holin features, LydC and LydD   [40]. LydC (93 aa) is encoded by a monocist ronic operon preceded by the putative early  promoter and immediately preceding the dar operon (Figure 2). LydD (84 aa) is encoded  by the second gene of the lyz operon, as would be expected for a pinholin. Its amino acid  sequence differs from those of LydA and Ly dC, which are 29% identical to each other.  However, like LydA and LydC, LydD contai ns two predicted transmembrane helices, and  its C-terminus with an array of five arginine residues is highly basic. The beginning of the  lydD  gene overlaps with the end of lyz and the 3′ moiety of lydD overlaps with the beginning  of a small gene of unknown function, designated as lydE.    Figure 2. Genetic organization of th e SAR-endolysin gene ( lyz) and known or predicted holin genes  in genomes of P1 and P7 phages. Known and pred icted holin genes are in gray. Additional genes  shown encode antiholin ( lydB ) and a putative protein of unknown function ( lydE ). Arrows indicate  promoters. LP, late phage promoter . The ruler at the bottom correspon ds to nucleotide positions as  they appear in the published P1 genome (GenBank acc. no. NC_005856.1).  The spanin genes of P1, ppfA  (o-spanin) and mlp (i-spanin) partially overlap and are  located in yet another unlinked operon. They were identified based on the predicted fea- tures of their products, but their function ha s not been verified experimentally [27]. In  laboratory cultures grown with shaking, it is not essential for lysis.  The organization of the known and putative cell lysis-associated genes of P1 is con- served in a P1-related phage, P7 (GenBank a cc. no. AF503408). The amino acid sequences  of their products, except that of LydC, are iden tical or nearly identical in these two phages.  The complexity of P1/P7 lytic functions poses a question about its biological sense. In  this work, we dissect the functions of known and predicted P1 holins, show the different  impacts of particular holins on the cell lysis of different P1 hosts, and the concerted con- tribution of all of them to efficient lysis in a programmed time. Additionally, we demon- strate that LydD is a pinholin, and that it has a dominant influence on the morphological  transitions of E. coli  and Enterobacter cloacae  cells during lysis mediated by P1.     Figure 2. Genetic organization of the SAR-endolysin gene ( lyz) and known or predicted holin genes in genomes of P1 and P7 phages. Known and predicted holin genes are in gray. Additional genes shown encode antiholin ( lydB ) and a putative protein of unknown function ( lydE ). Arrows indicate promoters. LP , late phage promoter. The ruler at the bottom corresponds to nucleotide positions as they appear in the published P1 genome (GenBank acc. no. NC_005856.1). The spanin genes of P1, ppfA (o-spanin) and mlp(i-spanin) partially overlap and are located in yet another unlinked operon. They were identiﬁed based on the predicted features of their products, but their function has not been veriﬁed experimentally [ 27]. In laboratory cultures grown with shaking, it is not essential for lysis.', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Known and predicted holin genes are in gray. Additional genes shown encode antiholin ( lydB ) and a putative protein of unknown function ( lydE ). Arrows indicate promoters. LP , late phage promoter. The ruler at the bottom corresponds to nucleotide positions as they appear in the published P1 genome (GenBank acc. no. NC_005856.1). The spanin genes of P1, ppfA (o-spanin) and mlp(i-spanin) partially overlap and are located in yet another unlinked operon. They were identiﬁed based on the predicted features of their products, but their function has not been veriﬁed experimentally [ 27]. In laboratory cultures grown with shaking, it is not essential for lysis. The organization of the known and putative cell lysis-associated genes of P1 is con- served in a P1-related phage, P7 (GenBank acc. no. AF503408). The amino acid sequences of their products, except that of LydC, are identical or nearly identical in these two phages. The complexity of P1/P7 lytic functions poses a question about its biological sense. In this work, we dissect the functions of known and predicted P1 holins, show the differ- ent impacts of particular holins on the cell lysis of different P1 hosts, and the concerted contribution of all of them to efﬁcient lysis in a programmed time. Additionally, we demon- strate that LydD is a pinholin, and that it has a dominant inﬂuence on the morphological transitions of E. coli and Enterobacter cloacae cells during lysis mediated by P1. 2. Results 2.1. Inﬂuence of LydD on the Phenotype of Cells Producing the P1 Lyz Endolysin The insertional inactivation of the P1 lyzgene in a P1 variant encoding the thermosen- sitive phage repressor (C1-100) results in the inability of lysogens to lyse upon the thermal', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 4 of 20 induction of P1 lytic development (Figure S1). This is consistent with the inability of the P1 amber mutant in the region of lyzto cause lysis of the host cell [ 43] and indicates that the Lyz endolysin is the only P1 muralytic enzyme required for P1-mediated cellular lysis. As a SAR domain-containing endolysin, Lyz can cause lysis in the absence of holins and can therefore be cloned only under repression control. This was achieved by cloning it under the control of the Serratia marcescens trp promoter with the trprepressor, and by growing cells in the presence of glucose. As shown in Figure 2, P1 encodes at least three holin-like proteins, and we undertook a study to clarify their roles. The induction of the expression of lydA and lyzgenes cloned in a plasmid was reported to cause E. coli cell lysis in the absence of any other P1 protein [ 42]. Thus, LydA has been described as the primary Lyz-associated holin. This seemed odd, since SAR-domain lysins require a pinholin, which is classically encoded adjacent to the lysin. There is a putative pinholin gene, lydD , whose 5′end overlaps the 3′end of the lyz gene [ 40]. Indeed, we observed that cells containing a plasmid with the entire P1/P7 lyz operon ( lyz+lydD+lydE+) under the control of an inducible promoter lysed faster and more efﬁciently upon the induction of this promoter than cells containing only the lyzgene in an isogenic plasmid (Figure 3). The presence of lydE did not inﬂuence the lysis phenotype, indicating that LydD protein alone can cooperate with Lyz in cell lysis in the absence of LydA and other P1 proteins. Int. J. Mol. Sci.  2022 , 23, 4231 4 of 21     2. Results  2.1. Influence of LydD on the Phenotype of Cells Producing the P1 Lyz Endolysin  The insertional inactivation of the P1 lyz gene in a P1 variant encoding the thermo- sensitive phage repressor (C1-100) results in the inability of lysogens to lyse upon the  thermal induction of P1 lytic development (Figure S1). This is consistent with the inability of the P1 amber mutant in the region of lyz to cause lysis of the host cell [43] and indicates  that the Lyz endolysin is the only P1 muraly tic enzyme required for P1-mediated cellular  lysis. As a SAR domain-containing endolysin, Lyz can cause lysis in the absence of holins  and can therefore be cloned only under repression control. This was achieved by cloning  it under the control of the Serratia marcescens  trp promoter with the trp repressor, and by  growing cells in the presence of glucose.  As shown in Figure 2, P1 encodes at least three holin-like proteins, and we undertook  a study to clarify their roles. The induction of the expression of lydA  and lyz genes cloned  in a plasmid was reported to cause E. coli  cell lysis in the absence of any other P1 protein  [42]. Thus, LydA has been described as the primary Lyz-associated holin. This seemed  odd, since SAR-domain lysins require a pinholin , which is classically encoded adjacent to  the lysin. There is a putative pinholin gene, lydD , whose 5 ′ end overlaps the 3 ′ end of the  lyz gene [40]. Indeed, we observed that cells containing a plasmid with the entire P1/P7  lyz operon ( lyz +lydD+lydE+) under the control of an inducible promoter lysed faster and  more efficiently upon the induction of this  promoter than cells containing only the  lyz  gene in an isogenic plasmid (Figure 3).', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Thus, LydA has been described as the primary Lyz-associated holin. This seemed  odd, since SAR-domain lysins require a pinholin , which is classically encoded adjacent to  the lysin. There is a putative pinholin gene, lydD , whose 5 ′ end overlaps the 3 ′ end of the  lyz gene [40]. Indeed, we observed that cells containing a plasmid with the entire P1/P7  lyz operon ( lyz +lydD+lydE+) under the control of an inducible promoter lysed faster and  more efficiently upon the induction of this  promoter than cells containing only the  lyz  gene in an isogenic plasmid (Figure 3). The presence of lydE did not influence the lysis  phenotype, indicating that LydD protein alone can cooperate with Lyz in cell lysis in the  absence of LydA and other P1 proteins.    Figure 3. Changes in the optical density of  E. coli  cells carrying the intact P7 lyz operon or its trun- cated versions in a multicopy plasmid under the control of the S. marcescens trp operon promoter- operator region. Cells containing  the relevant plasmid (pACE3 [ lyz+lydD+lydE+], pACE4 [ lyz+lydD+]  or pACE5 [ lyz+]) and the pRPG18 plasmid as a source of trp repressor (TrpR) were grown overnight  in LB medium supplemented with ampicillin and ch loramphenicol at 37 °C with vigorous shaking,  diluted 100 × in fresh LB with the same antibiot ics, and further grown un der similar conditions.  When the optical density of the cell cultures (OD 600) reached about 0.2, they were supplemented  with β-indole-acrylic acid (20 µg/mL) to induce transcription from the trp promoter. Changes in  culture optical density were further monitored sp ectrophotometrically. Genes whose transcription  from the trp operon promoter was induced in particular cultures are marked with arrows. The graph  Figure 3. Changes in the optical density of E. coli cells carrying the intact P7 lyzoperon or its truncated versions in a multicopy plasmid under the control of the S. marcescens trp operon promoter-operator region. Cells containing the relevant plasmid (pACE3 [ lyz+lydD+lydE+], pACE4 [ lyz+lydD+] or pACE5 [lyz+]) and the pRPG18 plasmid as a source of trprepressor (TrpR) were grown overnight in LB medium supplemented with ampicillin and chloramphenicol at 37◦C with vigorous shaking, diluted 100×in fresh LB with the same antibiotics, and further grown under similar conditions. When the optical density of the cell cultures (OD 600) reached about 0.2, they were supplemented with β-indole-acrylic acid (20 µg/mL) to induce transcription from the trppromoter. Changes in culture optical density were further monitored spectrophotometrically. Genes whose transcription from the trpoperon promoter was induced in particular cultures are marked with arrows. The graph shows the average results of the measurement of changes in the optical density of cell cultures in at least three independent experiments. Bars represent the standard deviation. 2.2. Lethal Effect of Cloned lydC or lydA on E. coli Cells In the P1 genome, the lydA gene is immediately preceded by lydC , whose product has features characteristic of class II holins, as does LydA [ 40]. Our attempts to clone the lydC orlydA gene in the ColE1-derived expression vector (pGBT30) led to the formation of only', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 5 of 20 a few small colonies of transformants. They either did not form colonies upon passaging or could be transferred to a liquid medium but stopped growing upon the induction of cloned gene expression. The growth cessation and lethal phenotypes of cells producing LydC or LydA observed here are consistent with the previously observed phenotypes of bacterial cells expressing cloned holin genes (see, e.g., [ 38,49]). They were attributed to the holin-mediated cell membrane puncturing or disruption of the proton motive force, resulting in cell death. 2.3. Effects of LydA, LydC, or LydD Depletion on P1-Mediated Cell Lysis To test the contributions of LydA, LydC, and LydD to the P1-mediated lysis of E. coli cells, we introduced to the P1 genome, by recombinational replacements, the kanamycin resistance cassette or indels inactivating the lydA ,lydC , orlydD gene. Additionally, P1 vari- ants that carried the combinations of mutations in any two of these genes were constructed. The lydA mutant and the double lydA lydC mutant formed plaques on a layer of indicator strain cells ( E. coli N99) when plates with infected cells were incubated at 42◦C (under conditions non-permissive for the formation of lysogens). In the case of the double lydA lydD mutant, no plaques could be observed. This shows that LydD alone is sufﬁcient to cooperate with Lyz in cell lysis by P1, but LydC alone is not, at least under the conditions of our experiment. In liquid medium phages deprived of functional LydC or LydD, or both these proteins could lyse E. coli cells, although with a delay. The delay differed slightly in the case of two different lydD mutants: lydD ::kanRand lydD∆16-81 (Figure 4A). In contrast to that, phages deprived of LydA did not lyse cells, at least for the duration of the experiment. Taken together, our results demonstrate that LydA, LydC and LydD contribute to the P1-mediated E. coli cell lysis in a programmed time under these conditions. However, while LydA appears to be the main holin involved in P1-mediated cell lysis, in liquid medium, the functions of LydC and LydD are accessory. Int. J. Mol. Sci.  2022 , 23, 4231 6 of 21       Figure 4. Changes in the optical density of cultures of ( A) E. coli  and ( B) E. cloacae  cells containing  wild-type P1 or P1 mutants depleted of particular holin functions, upon the thermal induction of P1  lytic development. The graphs show the average re sults of measurement of changes in the optical  density of cell cultures in at least three independ ent experiments. The legend applies to both charts.  Bars represent the standard deviation.  2.4. Influence of the P1 lyz Gene Replacement with the λ R Gene on P1-Mediated Cell Lysis  In certain phages that encode SAR-endolysins, the onset of lysis is associated with  the action of pinholins, and typically the SA R-endolysin-encoding genes of these phages  are in the same operons as the genes for th e accompanying pinholins [11–13,16,19]. Thus,  we tested whether the P1 lydD  gene could encode a pinholi n. P1 bacteriophage mutants  with the lyz gene replaced with the phage λ R gene (encoding a canonical endolysin) and  with various holin genes inactive were tested for the ability to lyse cells (Figure 5A,B). The  replacement of lyz with R delayed the lysis of E. coli  cells.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  2.4. Influence of the P1 lyz Gene Replacement with the λ R Gene on P1-Mediated Cell Lysis  In certain phages that encode SAR-endolysins, the onset of lysis is associated with  the action of pinholins, and typically the SA R-endolysin-encoding genes of these phages  are in the same operons as the genes for th e accompanying pinholins [11–13,16,19]. Thus,  we tested whether the P1 lydD  gene could encode a pinholi n. P1 bacteriophage mutants  with the lyz gene replaced with the phage λ R gene (encoding a canonical endolysin) and  with various holin genes inactive were tested for the ability to lyse cells (Figure 5A,B). The  replacement of lyz with R delayed the lysis of E. coli  cells. The lack of functional lydD  did  not influence the lysis by the replacement muta nt, suggesting that Ly dD protein does not  contribute to lysis with the participation of R. No lysis could be seen in the case of replace- ment mutant lacking functional lydA , independent of whether lydC  and lydD  genes were  intact or mutated. This conforms to the func tion of LydA as a canonical holin and is con- sistent with the essential role of LydA in lysis of E. coli  cells by P1 in liquid cell cultures.  The lack of functional lydC  in the replacement mutant slig htly accelerated the lysis, sug- gesting a possible role of LydC in the regulation of lysis timing.  Contrary to E. coli , the lyz to R replacement did not have a significant influence on  the onset of the lysis of E. cloacae  cells (Figure 5C,D). However, as in E. coli , the lack of  functional lydD  did not significantly influence the lysis by the replacement mutant, sug- gesting that LydD protein does not contribute to lysis with the participation of R. In sup- port of that, the E. cloacae  lysogen of the replacement mutant deprived of LydA, or LydA  and LydC did not lyse upon the induction of P1 lytic development, even though LydA is  not essential for the P1-mediated lysis of E. cloacae  in liquid cultures , and even though the  lydD  gene was intact in the mutant. Clearly, LydD alone cannot cooperate with R endoly- sin in cell lysis, as if it was not a canonical holin but a pinholin.  The main protein cooperating with λ R in the lysis of E. cloacae  cells grown in liquid  medium appeared to be LydA. The replacement mutant lacking functional lydA  did not  Figure 4. Changes in the optical density of cultures of ( A)E. coli and ( B)E. cloacae cells containing wild-type P1 or P1 mutants depleted of particular holin functions, upon the thermal induction of P1 lytic development. The graphs show the average results of measurement of changes in the optical density of cell cultures in at least three independent experiments. The legend applies to both charts. Bars represent the standard deviation.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:59 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'736'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992286'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'462ms'), (b'x-request-id', b'e647d773d2e8b1a28c76eea14834b777'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412498e0ab35d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:35:59 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'736'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992286'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'462ms'), (b'x-request-id', b'e647d773d2e8b1a28c76eea14834b777'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412498e0ab35d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 6 of 20 The inability of LydD to cause lysis by P1 depleted of functional lydA and lydC genes, in the case of cells grown in liquid medium, is surprising in a view of the ability of LydD to cause lysis with the participation of Lyz in the absence of other P1 proteins (see Figure 3). It is also surprising in a view of the ability of P1 lydA lydC mutant to form plaques. Most likely, the contribution of LydD to cell lysis by P1 is inﬂuenced by the presence of other components of the P1 cell lysis system and by cell growth conditions. To see whether the contributions of different holins to the cell lysis of different hosts of P1 vary, we lysogenized, with P1 or its mutants, cells of Enterobacter cloacae (Figure 4B). The onset of E. cloacae lysis upon the induction of P1 lytic development occurred later than that of E. coli , but at a similar optical density of the cell culture. The inactivation of lydC or insertional inactivation of lydD delayed the lysis, as in the case of E. coli . The deletion within the N-terminal moiety of lydD did not delay the onset of lysis, but slightly slowed down the dynamics of lysis. The mutation in lydA delayed the onset of lysis for longer than the mutations in lydC and lydD , but did not prevent lysis for the duration of the experiment, unlike in E. coli . Taken together, these results indicate that, although LydA, LydC, and LydD contribute to P1-mediated E. cloacae and E. coli cell lysis in a programmed time, the contribution of each one to lysis can vary depending on the bacterial growth conditions and the host of P1. 2.4. Inﬂuence of the P1 lyz Gene Replacement with the λR Gene on P1-Mediated Cell Lysis In certain phages that encode SAR-endolysins, the onset of lysis is associated with the action of pinholins, and typically the SAR-endolysin-encoding genes of these phages are in the same operons as the genes for the accompanying pinholins [ 11–13,16,19]. Thus, we tested whether the P1 lydD gene could encode a pinholin. P1 bacteriophage mutants with the lyzgene replaced with the phage λRgene (encoding a canonical endolysin) and with various holin genes inactive were tested for the ability to lyse cells (Figure 5A,B). The replacement of lyzwith Rdelayed the lysis of E. coli cells. The lack of functional lydD did not inﬂuence the lysis by the replacement mutant, suggesting that LydD protein does not contribute to lysis with the participation of R. No lysis could be seen in the case of replacement mutant lacking functional lydA , independent of whether lydC and lydD genes were intact or mutated. This conforms to the function of LydA as a canonical holin and is consistent with the essential role of LydA in lysis of E. coli cells by P1 in liquid cell cultures. The lack of functional lydC in the replacement mutant slightly accelerated the lysis, suggesting a possible role of LydC in the regulation of lysis timing. Contrary to E. coli , the lyztoRreplacement did not have a signiﬁcant inﬂuence on the onset of the lysis of E. cloacae cells (Figure 5C,D).', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  This conforms to the function of LydA as a canonical holin and is consistent with the essential role of LydA in lysis of E. coli cells by P1 in liquid cell cultures. The lack of functional lydC in the replacement mutant slightly accelerated the lysis, suggesting a possible role of LydC in the regulation of lysis timing. Contrary to E. coli , the lyztoRreplacement did not have a signiﬁcant inﬂuence on the onset of the lysis of E. cloacae cells (Figure 5C,D). However, as in E. coli , the lack of functional lydD did not signiﬁcantly inﬂuence the lysis by the replacement mutant, suggesting that LydD protein does not contribute to lysis with the participation of R. In support of that, the E. cloacae lysogen of the replacement mutant deprived of LydA, or LydA and LydC did not lyse upon the induction of P1 lytic development, even though LydA is not essential for the P1-mediated lysis of E. cloacae in liquid cultures, and even though the lydD gene was intact in the mutant. Clearly, LydD alone cannot cooperate with R endolysin in cell lysis, as if it was not a canonical holin but a pinholin. The main protein cooperating with λR in the lysis of E. cloacae cells grown in liquid medium appeared to be LydA. The replacement mutant lacking functional lydA did not lyse cells independent of whether the lydC and lydD genes were present or not. Moreover, the lack of functional lydC , orlydC and lydD in the replacement mutant only delayed the lysis. As the lack of LydD alone did not inﬂuence the lysis by the replacement mutant, this delay indicates that LydC can also contribute to lysis with the participation of R, but its role in lysis under these conditions is accessory. Surprisingly, we observed that the replacement mutant lacking lydA could form plaques on the layer of E. cloacae cells when plates with the infected cells were incubated at 42◦C, as if LydC or LydD, or two of them would be sufﬁcient to support the function of R under these conditions.', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 7 of 20 Int. J. Mol. Sci.  2022 , 23, 4231 7 of 21     lyse cells independent of whether the lydC  and lydD  genes were present or not. Moreover,  the lack of functional lydC,  or lydC and lydD in the replacement mutant only delayed the  lysis. As the lack of LydD alone did not infl uence the lysis by the replacement mutant, this  delay indicates that LydC can also contribute to lysis with the participation of R, but its  role in lysis under these conditions is accessory. Surprisingly, we observed that the re-placement mutant lacking lydA  could form plaques on the layer of E. cloacae  cells when  plates with the infected cells were incubated at  42 °C, as if LydC or LydD, or two of them  would be sufficient to support the fu nction of R under these conditions.    Figure 5. Influence of replacing the P1 lyz gene in the P1 genome with the λ R gene on the lysis of  (A,B) E. coli  and ( C,D) E. cloacae  cells, mediated by P1 or its mutants deprived of certain holin func- tions. The graphs show the averag e results of the measurement of changes in the optical density of  cell cultures in at least three independent experime nts. The legend applies to both charts. Bars rep- resent the standard deviation.  2.5. Influence of LydD on E. coli Cells Producing R Endolysin  In contrast to canonical holins, which form  large non-specific pores in a cytoplasmic  membrane, pinholins can form only nanometer-scale pores, which are too small to allow  for the passage of canonical endolysins. Since λ R is a canonical endolysin, it would not  be expected to pair with LydD, if LydD is a pinholin. To test this, we used a two-plasmid  system in which R and lydD  genes were cloned separately in different plasmids under the  Figure 5. Inﬂuence of replacing the P1 lyzgene in the P1 genome with the λRgene on the lysis of (A,B)E. coli and ( C,D)E. cloacae cells, mediated by P1 or its mutants deprived of certain holin functions. The graphs show the average results of the measurement of changes in the optical density of cell cultures in at least three independent experiments. The legend applies to both charts. Bars represent the standard deviation. 2.5. Inﬂuence of LydD on E. coli Cells Producing R Endolysin In contrast to canonical holins, which form large non-speciﬁc pores in a cytoplasmic membrane, pinholins can form only nanometer-scale pores, which are too small to allow for the passage of canonical endolysins. Since λR is a canonical endolysin, it would not be expected to pair with LydD, if LydD is a pinholin. To test this, we used a two-plasmid system in which Rand lydD genes were cloned separately in different plasmids under the control of different inducible promoters (Figure 6). The induction of the expression of each gene separately or both together caused a signiﬁcant retardation in the increase in optical density of cell cultures, but no decrease in the optical density that would be characteristic of lysis. However, the number of colony-forming units in the cultures of cells with the induced expression of lydD decreased, indicating cell death. Clearly, LydD cannot cooperate with the canonical R endolysin in cell lysis, but can cause cell death, which supports its function as a pinholin.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 8 of 20 Int. J. Mol. Sci.  2022 , 23, 4231 8 of 21     control of different inducible promoters (Figure 6). The induction of the expression of each  gene separately or both together caused a sign ificant retardation in the increase in optical  density of cell cultures, but no decrease in th e optical density that would be characteristic  of lysis. However, the number of colony-forming units in the cultures of cells with the  induced expression of lydD  decreased, indicating cell death. Clearly, LydD cannot coop- erate with the canonical R endolysin in cell ly sis, but can cause cell  death, which supports  its function as a pinholin.    Figure 6. Influence of P1 LydD and λ R co-production on ( A) E. coli cell growth and ( B) ability to  form colonies. Cells containing  relevant plasmids (pAKI6 [ tacPO -lydD+], pACE10 [ trpPO -R+]) or con- trol plasmids (pGBT30,  pMLO30) were grown overnight in B medium supplemented with ampicil- lin and chloramphenicol at 37 °C with vigorous sh aking, diluted 100× in fres h LB with these antibi- otics, and grown further under similar conditions. When the optical density of cell cultures (OD 600)  reached about 0.2, they were supplemented with IPTG (1 mM) to induce  the transcription of lydD   from the tac promoter (time zero), grown for 60 min, and supplemented with β-indole-acrylic acid  (20 µg/mL) to induce the transcription of  R from the trp promoter. Changes in the culture optical  density were monitored spectropho tometrically at 1 h intervals ( A). Parallel changes in the number  of colony-forming units in each culture were measured by plating diluted samples of each culture  on LB agar plates supplemented with ampicillin and chloramphenicol and counting colonies after  overnight incubation of plates at 37 °C ( B). Cultures without one or bo th inducers added or with  control plasmids served for compar ative purposes. Genes, whose transcription in particular cultures  was induced, are marked with arrows. Graphs sh ow the average results of the measurement of  changes in the optical density of cell cultures ( A) or the number of colony-forming units ( B) in at  least three independent experiments. Ba rs represent the standard deviation.  2.6. Morphological Transitions during E. coli an d E. cloacae Cell Lysis Mediated by P1 or Its  Mutants  The differences in the mechanisms of ca nonical and pinholin– SAR-endolysin-medi- ated lysis were shown to be associated with  differences in the morphological transitions  of lysing cells [29]. Thus, we visualized E. coli  cell lysis mediated by P1 or its lydC  or lydD   mutants by time-lapse microscopy, and compared it with that mediated by ʎ phage. A  total of 2 out of 50 individual ʎ-mediated lysis events captured  in samples of 10 independ- ent cultures are shown in Figure 7A,B as time -lapse series (see also Movie S1). Typically,  the morphological changes of E. coli  cells during ʎ-mediated lysis started from cell short- ening, which was associated with the formation of a ball-shaped form at one end (45/50  events, Figure 7A) or in the center of the cell (5/50 events, Figure 7B), and by localized cell rupture. In the case of ʎ, this rupture is caused by holins, which create large holes in the  cell membrane, leading to the formation of a transient non-refractile ghost that retains a  circular shape, and to the expulsion of the cytoplasmic content, as was observed previ- ously [29]. The morphological changes of E. coli  cells during P1-mediated lysis appeared  to be different from those observed in the case of λ; they progressed through the reduction  in cell length and increase in cell diameter (68/70 events) (Figure 7C,D; see also Movie S2),  Figure 6.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  In the case of ʎ, this rupture is caused by holins, which create large holes in the  cell membrane, leading to the formation of a transient non-refractile ghost that retains a  circular shape, and to the expulsion of the cytoplasmic content, as was observed previ- ously [29]. The morphological changes of E. coli  cells during P1-mediated lysis appeared  to be different from those observed in the case of λ; they progressed through the reduction  in cell length and increase in cell diameter (68/70 events) (Figure 7C,D; see also Movie S2),  Figure 6. Inﬂuence of P1 LydD and λR co-production on ( A)E. coli cell growth and ( B) ability to form colonies. Cells containing relevant plasmids (pAKI6 [ tacPO -lydD+], pACE10 [ trpPO -R+]) or control plasmids (pGBT30, pMLO30) were grown overnight in B medium supplemented with ampicillin and chloramphenicol at 37◦C with vigorous shaking, diluted 100 ×in fresh LB with these antibiotics, and grown further under similar conditions. When the optical density of cell cultures (OD 600) reached about 0.2, they were supplemented with IPTG (1 mM) to induce the transcription of lydD from the tacpromoter (time zero), grown for 60 min, and supplemented with β-indole-acrylic acid (20 µg/mL) to induce the transcription of Rfrom the trppromoter. Changes in the culture optical density were monitored spectrophotometrically at 1 h intervals ( A). Parallel changes in the number of colony-forming units in each culture were measured by plating diluted samples of each culture on LB agar plates supplemented with ampicillin and chloramphenicol and counting colonies after overnight incubation of plates at 37◦C (B). Cultures without one or both inducers added or with control plasmids served for comparative purposes. Genes, whose transcription in particular cultures was induced, are marked with arrows. Graphs show the average results of the measurement of changes in the optical density of cell cultures ( A) or the number of colony-forming units ( B) in at least three independent experiments. Bars represent the standard deviation. 2.6. Morphological T ransitions during E. coli and E. cloacae Cell Lysis Mediated by P1 or Its Mutants The differences in the mechanisms of canonical and pinholin–SAR-endolysin-mediated lysis were shown to be associated with differences in the morphological transitions of lysing cells [ 29]. Thus, we visualized E. coli cell lysis mediated by P1 or its lydC orlydD mutants by time-lapse microscopy, and compared it with that mediated by Lphage. A total of 2 out of 50 individual L-mediated lysis events captured in samples of 10 independent cultures are shown in Figure 7A,B as time-lapse series (see also Movie S1). Typically, the morphological changes of E. coli cells during L-mediated lysis started from cell shortening, which was associated with the formation of a ball-shaped form at one end (45/50 events, Figure 7A) or in the center of the cell (5/50 events, Figure 7B), and by localized cell rupture. In the case of L, this rupture is caused by holins, which create large holes in the cell membrane, leading to the formation of a transient non-refractile ghost that retains a circular shape, and to the expulsion of the cytoplasmic content, as was observed previously [ 29]. The morphological changes of E. coli cells during P1-mediated lysis appeared to be different from those observed in the case of λ; they progressed through the reduction in cell length and increase in cell diameter (68/70 events) (Figure 7C,D; see also Movie S2), as was ob- served previously in the case of pinholin–SAR-endolysin-mediated lysis [ 29].', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  In the case of L, this rupture is caused by holins, which create large holes in the cell membrane, leading to the formation of a transient non-refractile ghost that retains a circular shape, and to the expulsion of the cytoplasmic content, as was observed previously [ 29]. The morphological changes of E. coli cells during P1-mediated lysis appeared to be different from those observed in the case of λ; they progressed through the reduction in cell length and increase in cell diameter (68/70 events) (Figure 7C,D; see also Movie S2), as was ob- served previously in the case of pinholin–SAR-endolysin-mediated lysis [ 29]. Surprisingly, the inactivation of lydC and lydD genes (Figure 6E,F) resulted in a switch in morpholog- ical changes during lysis to those characteristic of L-mediated lysis: cell was shortened at one end (56/69 events, Figure 7E) or in the center (13/69 events, Figure 7F; see also Movies S3 and S4 ). Taken together, these results demonstrate that LydA is a canonical holin, and support the pinholin function of LydD. Additionally, they indicate that pinholin– SAR-endolysin mediated morphological transitions are dominant during the lysis of E. coli cells at the end of P1 lytic development.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 9 of 20 Int. J. Mol. Sci.  2022 , 23, 4231 9 of 21     as was observed previously in the case of pinholin–SAR-endolysin-mediated lysis [29].  Surprisingly, the inactivation of lydC  and lydD  genes (Figure 6E,F) resulted in a switch in  morphological changes during lysis to those characteristic of ʎ-mediated lysis: cell was  shortened at one end (56/69 even ts, Figure 7E) or in the center  (13/69 events, Figure 7F; see  also Movies S3 and S4). Taken together, these results demonstrate that LydA is a canonical  holin, and support the pinholin function of LydD. Additionally, they indicate that pinho- lin–SAR-endolysin mediated morphological transitions are dominant during the lysis of  E. coli  cells at the end of P1 lytic development.    Figure 7. Morphological transitions of E. coli cells during lysis mediated by ( A,B) phage λ, (C,D)  wild-type phage P1, and ( E,F) phage P1 mutant deprived of functional lydC  and lydD  genes. E. coli  N99 lysogens were thermally induced and withdrawn from the culture just be fore the start of lysis  to document morphological changes during lysis by time-lapse microscopy. The names of bacterio-phages and their genotypes are indicated above each group of photographs. Time (in seconds) is displayed above each frame, with time zero repres enting the first photogra ph acquired before the  observable morphological change or lysis. Bars = 1 µm.  The P1-mediated lysis of E. cloacae  cells progressed through similar morphological  changes to those that accompanied E. coli  cell lysis: the cells shortened and thickened,  converting into a more ball-shaped form, and then cell envelopes disintegrated more or less evenly (52/52 events, Figure 8A; see also Movies S5 and S6). In contrast, E. cloacae  cell  lysis caused by phage P1 with inactivated lydD and lydC  genes was accompanied by slight  cell shortening, the formation of a small protru sion at one cell pole, and the bursting of  the end of this protrusion, resulting in the abrupt expulsion of cell contents through it  (24/25 events, Figure 8B,C). In this case, as in the case of E. coli  with the relevant mutant,  the lysis occurred from one cell pole. Despite that the changes in E. cloacae  and E. coli  mor- phology at this pole during lysis were significantly different, in both cases the lysis was  initiated from one cell pole, fitting to the canonical holin–endolysin-mediated pattern of  cell lysis. This is consistent with the canonical holin function of LydA. Additionally, as in  the case of E. coli , the comparison of morphological transition patterns during the lysis of  E. cloacae  cells with wild-type P1 and P1 lydC lydD  mutant indicates that lysis mediated  Figure 7. Morphological transitions of E. coli cells during lysis mediated by ( A,B) phage λ, (C,D) wild-type phage P1, and ( E,F) phage P1 mutant deprived of functional lydC and lydD genes. E. coli N99 lysogens were thermally induced and withdrawn from the culture just before the start of lysis to document morphological changes during lysis by time-lapse microscopy. The names of bacteriophages and their genotypes are indicated above each group of photographs. Time (in seconds) is displayed above each frame, with time zero representing the ﬁrst photograph acquired before the observable morphological change or lysis. Bars = 1 µm. The P1-mediated lysis of E. cloacae cells progressed through similar morphological changes to those that accompanied E. coli cell lysis: the cells shortened and thickened, converting into a more ball-shaped form, and then cell envelopes disintegrated more or less evenly (52/52 events, Figure 8A; see also Movie S5).', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  E. coli N99 lysogens were thermally induced and withdrawn from the culture just before the start of lysis to document morphological changes during lysis by time-lapse microscopy. The names of bacteriophages and their genotypes are indicated above each group of photographs. Time (in seconds) is displayed above each frame, with time zero representing the ﬁrst photograph acquired before the observable morphological change or lysis. Bars = 1 µm. The P1-mediated lysis of E. cloacae cells progressed through similar morphological changes to those that accompanied E. coli cell lysis: the cells shortened and thickened, converting into a more ball-shaped form, and then cell envelopes disintegrated more or less evenly (52/52 events, Figure 8A; see also Movie S5). In contrast, E. cloacae cell lysis caused by phage P1 with inactivated lydD and lydC genes was accompanied by slight cell shortening, the formation of a small protrusion at one cell pole, and the bursting of the end of this protrusion, resulting in the abrupt expulsion of cell contents through it (24/25 events, Figure 8B,C; see also Movie S6). In this case, as in the case of E. coli with the relevant mutant, the lysis occurred from one cell pole. Despite that the changes in E. cloacae and E. coli morphology at this pole during lysis were signiﬁcantly different, in both cases the lysis was initiated from one cell pole, ﬁtting to the canonical holin–endolysin-mediated pattern of cell lysis. This is consistent with the canonical holin function of LydA. Additionally, as in the case of E. coli , the comparison of morphological transition patterns during the lysis of E. cloacae cells with wild-type P1 and P1 lydC lydD mutant indicates that lysis mediated by wild-type P1 involves the concerted action of all P1-encoded holins/pinholins, and that pinholin-SAR-endolysin-mediated morphological transitions are dominant during E. cloacae cell lysis. 2.7. Influence of P1 lyz Gene Replacement with λR on Cell Morphological T ransitions during Lysis To see whether the replacement of the lyzgene in the P1 genome with λRgene would switch the pattern of morphological transition of the replacement mutant carrying cells during lysis, we visualized lysing E. coli and E. cloacae cells with the replacement mutant. As expected, in both cases, the pattern of morphological changes during lysis switched to that characteristic of canonical holin–endolysin-mediated lysis (Figure 9). A similar switch', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 10 of 20 was observed independent of whether the mutant could produce LydC, consistent with our previous observations that LydA and LydC act as canonical holins and that the pattern of morphological changes during the lysis of E. cloacae cells by P1 is inﬂuenced by the concerted action of LydA, LydC, and LydD. Int. J. Mol. Sci.  2022 , 23, 4231 10 of 21     by wild-type P1 involves the concerted action  of all P1-encoded holins/pinholins, and that  pinholin-SAR-endolysin-mediated morphologi cal transitions are dominant during E. clo- acae cell lysis.    Figure 8. Morphological transitions of E. cloacae cells during lysis mediated by ( A) wild-type phage  P1 and ( B,C) phage P1 mutant deprived of functional lydC  and lydD  genes. E. cloacae  lysogens were  thermally induced and withdrawn from the culture before the start of lysis to document morpho- logical changes during lysis by time-lapse microscopy. The relevant genotypes of P1 phages are indicated above each group of photographs. Time (in seconds) is displayed above each photograph, with  time zero representing the first photograph acquired  before the observable morphological change or ly- sis. Bars = 1 µm.  2.7. Influence of P1 lyz Gene Replacement with λ R on Cell Morphological Transitions during Lysis  To see whether the replacement of the lyz gene in the P1 genome with λ R gene would  switch the pattern of morphological transiti on of the replacement mutant carrying cells  during lysis, we visualized lysing E. coli and E. cloacae  cells with the replacement mutant.  As expected, in both cases, the pattern of morphological changes during lysis switched to  that characteristic of canonical holin–endolysi n-mediated lysis (Figure 9). A similar switch  was observed independent of whether the mu tant could produce LydC, consistent with  our previous observations that LydA and LydC act as canonical holins and that the pat- tern of morphological changes during the lysis of E. cloacae  cells by P1 is influenced by the  concerted action of LydA, LydC, and LydD.  Figure 8. Morphological transitions of E. cloacae cells during lysis mediated by ( A) wild-type phage P1 and ( B,C) phage P1 mutant deprived of functional lydC and lydD genes. E. cloacae lysogens were thermally induced and withdrawn from the culture before the start of lysis to document morphologi- cal changes during lysis by time-lapse microscopy. The relevant genotypes of P1 phages are indicated above each group of photographs. Time (in seconds) is displayed above each photograph, with time zero representing the ﬁrst photograph acquired before the observable morphological change or lysis. Bars = 1 µm. Int. J. Mol. Sci.  2022 , 23, 4231 11 of 21       Figure 9. Morphological transitions of ( A,B) E. coli  and ( C,D) E. cloacae  cells during lysis mediated  by phage P1 with the replacement of lyz gene with the λ R gene ( A,C) and by the P1 replacement  mutant deprived of functional lydC  and lydD  genes ( C,D). E. coli and E. cloacae lysogens were ther- mally induced and withdrawn from the cultures befo re the start of lysis to document morphological  changes during lysis by time-lapse microscopy. Th e names of bacterial species and relevant geno- types of P1 phages are indicated above each group of photographs. Time (in seconds) is displayed  above each photograph, with time zero representi ng the first photograph acquired before the ob- servable morphological change or lysis. Bars = 1 µm.  3. Discussion  The lysis of a phage-infected cell, as soon as the mature phage progeny is ready to be  released, is crucial for the phage’s evolutionary  fate in the world of phages competing for  the same hosts.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  E. coli and E. cloacae lysogens were ther- mally induced and withdrawn from the cultures befo re the start of lysis to document morphological  changes during lysis by time-lapse microscopy. Th e names of bacterial species and relevant geno- types of P1 phages are indicated above each group of photographs. Time (in seconds) is displayed  above each photograph, with time zero representi ng the first photograph acquired before the ob- servable morphological change or lysis. Bars = 1 µm.  3. Discussion  The lysis of a phage-infected cell, as soon as the mature phage progeny is ready to be  released, is crucial for the phage’s evolutionary  fate in the world of phages competing for  the same hosts. The breakage of the cytoplasmic membrane and cell wall integrity is es- sential for lysis under any bacterial growth conditions. Most phages, such as, e.g., λ, can  relay in these processes on a single pair of proteins, one of which (endolysin or SAR-en- dolysin) can digest cell wall peptidoglycan, while the other (holin or pinholin) can kill the  cell by disrupting the transmembrane proton motive force and enabling or facilitating the  access of endolysin/SAR-endolysin to the cell wa ll in an active form (reviewed in [1,9,22]).  The analysis of the P1 bacteriophage genome indicates that its lytic system is more com- plex. In addition to SAR-endolysin (Lyz), ho lin (LydA), and antiholin (LydB), whose func- tions in lysis were confirmed previously [42], P1 encodes two additional proteins of amino  acid sequence features characteristic of holins (LydC and LydD) [40]. In this paper, we  show that LydC and LydD contribute to P1-m ediated cell lysis, and that, while LydA and  LydC function as canonical holins, LydD fu nctions as a pinholin. Moreover, both holins  and the pinholin are essential for the efficien t lysis of bacteria in a programmed time. We  demonstrate the biological sense of such lyti c system complexity by showing the different  contributions of particular ho lins/pinholins to the lysis of cells grown under different con- ditions and cells of different P1 hosts, E. coli  and E. cloacae . Additionally, we demonstrate  for the first time that, while the morphological transitions of the cells of these various hosts  lysed by P1 are similar, they may differ if lysis is supported by an incomplete set of  holins/pinholins.  The redundancy of the genes that could encode proteins of predicted holin features  is not unique for P1 and its relatives [49–53]. However, whether one of these predicted holins in these cases is not an antiholin (a holin antagonist) was verified only in a few  cases. For instance, the endolysin gene of Actinomyces naeslundii  phage Av-1 is flanked by  two genes that encode two functional holins of unrelated amino acid sequences [54]. In  the case of Bacillus  phage SPP1, which also encodes two holin-like proteins, cell lysis could  only be caused by the concerted action of th e endolysin of this phage with both of these  proteins [49,53]. The results of our studies on  P1 presented in this paper demonstrate for  the first time the concerted action of three pr oteins of holin features, namely two holins  Figure 9. Morphological transitions of ( A,B)E. coli and ( C,D)E. cloacae cells during lysis mediated by phage P1 with the replacement of lyzgene with the λRgene ( A,C) and by the P1 replacement mutant deprived of functional lydC and lydD genes ( C,D).E. coli and E. cloacae lysogens were thermally induced and withdrawn from the cultures before the start of lysis to document morphological changes during lysis by time-lapse microscopy. The names of bacterial species and relevant genotypes of P1 phages are indicated above each group of photographs.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 6 of 20 The inability of LydD to cause lysis by P1 depleted of functional lydA and lydC genes, in the case of cells grown in liquid medium, is surprising in a view of the ability of LydD to cause lysis with the participation of Lyz in the absence of other P1 proteins (see Figure 3). It is also surprising in a view of the ability of P1 lydA lydC mutant to form plaques. Most likely, the contribution of LydD to cell lysis by P1 is inﬂuenced by the presence of other components of the P1 cell lysis system and by cell growth conditions. To see whether the contributions of different holins to the cell lysis of different hosts of P1 vary, we lysogenized, with P1 or its mutants, cells of Enterobacter cloacae (Figure 4B). The onset of E. cloacae lysis upon the induction of P1 lytic development occurred later than that of E. coli , but at a similar optical density of the cell culture. The inactivation of lydC or insertional inactivation of lydD delayed the lysis, as in the case of E. coli . The deletion within the N-terminal moiety of lydD did not delay the onset of lysis, but slightly slowed down the dynamics of lysis. The mutation in lydA delayed the onset of lysis for longer than the mutations in lydC and lydD , but did not prevent lysis for the duration of the experiment, unlike in E. coli . Taken together, these results indicate that, although LydA, LydC, and LydD contribute to P1-mediated E. cloacae and E. coli cell lysis in a programmed time, the contribution of each one to lysis can vary depending on the bacterial growth conditions and the host of P1. 2.4. Inﬂuence of the P1 lyz Gene Replacement with the λR Gene on P1-Mediated Cell Lysis In certain phages that encode SAR-endolysins, the onset of lysis is associated with the action of pinholins, and typically the SAR-endolysin-encoding genes of these phages are in the same operons as the genes for the accompanying pinholins [ 11–13,16,19]. Thus, we tested whether the P1 lydD gene could encode a pinholin. P1 bacteriophage mutants with the lyzgene replaced with the phage λRgene (encoding a canonical endolysin) and with various holin genes inactive were tested for the ability to lyse cells (Figure 5A,B). The replacement of lyzwith Rdelayed the lysis of E. coli cells. The lack of functional lydD did not inﬂuence the lysis by the replacement mutant, suggesting that LydD protein does not contribute to lysis with the participation of R. No lysis could be seen in the case of replacement mutant lacking functional lydA , independent of whether lydC and lydD genes were intact or mutated. This conforms to the function of LydA as a canonical holin and is consistent with the essential role of LydA in lysis of E. coli cells by P1 in liquid cell cultures. The lack of functional lydC in the replacement mutant slightly accelerated the lysis, suggesting a possible role of LydC in the regulation of lysis timing. Contrary to E. coli , the lyztoRreplacement did not have a signiﬁcant inﬂuence on the onset of the lysis of E. cloacae cells (Figure 5C,D).', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  This conforms to the function of LydA as a canonical holin and is consistent with the essential role of LydA in lysis of E. coli cells by P1 in liquid cell cultures. The lack of functional lydC in the replacement mutant slightly accelerated the lysis, suggesting a possible role of LydC in the regulation of lysis timing. Contrary to E. coli , the lyztoRreplacement did not have a signiﬁcant inﬂuence on the onset of the lysis of E. cloacae cells (Figure 5C,D). However, as in E. coli , the lack of functional lydD did not signiﬁcantly inﬂuence the lysis by the replacement mutant, suggesting that LydD protein does not contribute to lysis with the participation of R. In support of that, the E. cloacae lysogen of the replacement mutant deprived of LydA, or LydA and LydC did not lyse upon the induction of P1 lytic development, even though LydA is not essential for the P1-mediated lysis of E. cloacae in liquid cultures, and even though the lydD gene was intact in the mutant. Clearly, LydD alone cannot cooperate with R endolysin in cell lysis, as if it was not a canonical holin but a pinholin. The main protein cooperating with λR in the lysis of E. cloacae cells grown in liquid medium appeared to be LydA. The replacement mutant lacking functional lydA did not lyse cells independent of whether the lydC and lydD genes were present or not. Moreover, the lack of functional lydC , orlydC and lydD in the replacement mutant only delayed the lysis. As the lack of LydD alone did not inﬂuence the lysis by the replacement mutant, this delay indicates that LydC can also contribute to lysis with the participation of R, but its role in lysis under these conditions is accessory. Surprisingly, we observed that the replacement mutant lacking lydA could form plaques on the layer of E. cloacae cells when plates with the infected cells were incubated at 42◦C, as if LydC or LydD, or two of them would be sufﬁcient to support the function of R under these conditions.', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 7 of 20 Int. J. Mol. Sci.  2022 , 23, 4231 7 of 21     lyse cells independent of whether the lydC  and lydD  genes were present or not. Moreover,  the lack of functional lydC,  or lydC and lydD in the replacement mutant only delayed the  lysis. As the lack of LydD alone did not infl uence the lysis by the replacement mutant, this  delay indicates that LydC can also contribute to lysis with the participation of R, but its  role in lysis under these conditions is accessory. Surprisingly, we observed that the re-placement mutant lacking lydA  could form plaques on the layer of E. cloacae  cells when  plates with the infected cells were incubated at  42 °C, as if LydC or LydD, or two of them  would be sufficient to support the fu nction of R under these conditions.    Figure 5. Influence of replacing the P1 lyz gene in the P1 genome with the λ R gene on the lysis of  (A,B) E. coli  and ( C,D) E. cloacae  cells, mediated by P1 or its mutants deprived of certain holin func- tions. The graphs show the averag e results of the measurement of changes in the optical density of  cell cultures in at least three independent experime nts. The legend applies to both charts. Bars rep- resent the standard deviation.  2.5. Influence of LydD on E. coli Cells Producing R Endolysin  In contrast to canonical holins, which form  large non-specific pores in a cytoplasmic  membrane, pinholins can form only nanometer-scale pores, which are too small to allow  for the passage of canonical endolysins. Since λ R is a canonical endolysin, it would not  be expected to pair with LydD, if LydD is a pinholin. To test this, we used a two-plasmid  system in which R and lydD  genes were cloned separately in different plasmids under the  Figure 5. Inﬂuence of replacing the P1 lyzgene in the P1 genome with the λRgene on the lysis of (A,B)E. coli and ( C,D)E. cloacae cells, mediated by P1 or its mutants deprived of certain holin functions. The graphs show the average results of the measurement of changes in the optical density of cell cultures in at least three independent experiments. The legend applies to both charts. Bars represent the standard deviation. 2.5. Inﬂuence of LydD on E. coli Cells Producing R Endolysin In contrast to canonical holins, which form large non-speciﬁc pores in a cytoplasmic membrane, pinholins can form only nanometer-scale pores, which are too small to allow for the passage of canonical endolysins. Since λR is a canonical endolysin, it would not be expected to pair with LydD, if LydD is a pinholin. To test this, we used a two-plasmid system in which Rand lydD genes were cloned separately in different plasmids under the control of different inducible promoters (Figure 6). The induction of the expression of each gene separately or both together caused a signiﬁcant retardation in the increase in optical density of cell cultures, but no decrease in the optical density that would be characteristic of lysis. However, the number of colony-forming units in the cultures of cells with the induced expression of lydD decreased, indicating cell death. Clearly, LydD cannot cooperate with the canonical R endolysin in cell lysis, but can cause cell death, which supports its function as a pinholin.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 8 of 20 Int. J. Mol. Sci.  2022 , 23, 4231 8 of 21     control of different inducible promoters (Figure 6). The induction of the expression of each  gene separately or both together caused a sign ificant retardation in the increase in optical  density of cell cultures, but no decrease in th e optical density that would be characteristic  of lysis. However, the number of colony-forming units in the cultures of cells with the  induced expression of lydD  decreased, indicating cell death. Clearly, LydD cannot coop- erate with the canonical R endolysin in cell ly sis, but can cause cell  death, which supports  its function as a pinholin.    Figure 6. Influence of P1 LydD and λ R co-production on ( A) E. coli cell growth and ( B) ability to  form colonies. Cells containing  relevant plasmids (pAKI6 [ tacPO -lydD+], pACE10 [ trpPO -R+]) or con- trol plasmids (pGBT30,  pMLO30) were grown overnight in B medium supplemented with ampicil- lin and chloramphenicol at 37 °C with vigorous sh aking, diluted 100× in fres h LB with these antibi- otics, and grown further under similar conditions. When the optical density of cell cultures (OD 600)  reached about 0.2, they were supplemented with IPTG (1 mM) to induce  the transcription of lydD   from the tac promoter (time zero), grown for 60 min, and supplemented with β-indole-acrylic acid  (20 µg/mL) to induce the transcription of  R from the trp promoter. Changes in the culture optical  density were monitored spectropho tometrically at 1 h intervals ( A). Parallel changes in the number  of colony-forming units in each culture were measured by plating diluted samples of each culture  on LB agar plates supplemented with ampicillin and chloramphenicol and counting colonies after  overnight incubation of plates at 37 °C ( B). Cultures without one or bo th inducers added or with  control plasmids served for compar ative purposes. Genes, whose transcription in particular cultures  was induced, are marked with arrows. Graphs sh ow the average results of the measurement of  changes in the optical density of cell cultures ( A) or the number of colony-forming units ( B) in at  least three independent experiments. Ba rs represent the standard deviation.  2.6. Morphological Transitions during E. coli an d E. cloacae Cell Lysis Mediated by P1 or Its  Mutants  The differences in the mechanisms of ca nonical and pinholin– SAR-endolysin-medi- ated lysis were shown to be associated with  differences in the morphological transitions  of lysing cells [29]. Thus, we visualized E. coli  cell lysis mediated by P1 or its lydC  or lydD   mutants by time-lapse microscopy, and compared it with that mediated by ʎ phage. A  total of 2 out of 50 individual ʎ-mediated lysis events captured  in samples of 10 independ- ent cultures are shown in Figure 7A,B as time -lapse series (see also Movie S1). Typically,  the morphological changes of E. coli  cells during ʎ-mediated lysis started from cell short- ening, which was associated with the formation of a ball-shaped form at one end (45/50  events, Figure 7A) or in the center of the cell (5/50 events, Figure 7B), and by localized cell rupture. In the case of ʎ, this rupture is caused by holins, which create large holes in the  cell membrane, leading to the formation of a transient non-refractile ghost that retains a  circular shape, and to the expulsion of the cytoplasmic content, as was observed previ- ously [29]. The morphological changes of E. coli  cells during P1-mediated lysis appeared  to be different from those observed in the case of λ; they progressed through the reduction  in cell length and increase in cell diameter (68/70 events) (Figure 7C,D; see also Movie S2),  Figure 6.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  In the case of ʎ, this rupture is caused by holins, which create large holes in the  cell membrane, leading to the formation of a transient non-refractile ghost that retains a  circular shape, and to the expulsion of the cytoplasmic content, as was observed previ- ously [29]. The morphological changes of E. coli  cells during P1-mediated lysis appeared  to be different from those observed in the case of λ; they progressed through the reduction  in cell length and increase in cell diameter (68/70 events) (Figure 7C,D; see also Movie S2),  Figure 6. Inﬂuence of P1 LydD and λR co-production on ( A)E. coli cell growth and ( B) ability to form colonies. Cells containing relevant plasmids (pAKI6 [ tacPO -lydD+], pACE10 [ trpPO -R+]) or control plasmids (pGBT30, pMLO30) were grown overnight in B medium supplemented with ampicillin and chloramphenicol at 37◦C with vigorous shaking, diluted 100 ×in fresh LB with these antibiotics, and grown further under similar conditions. When the optical density of cell cultures (OD 600) reached about 0.2, they were supplemented with IPTG (1 mM) to induce the transcription of lydD from the tacpromoter (time zero), grown for 60 min, and supplemented with β-indole-acrylic acid (20 µg/mL) to induce the transcription of Rfrom the trppromoter. Changes in the culture optical density were monitored spectrophotometrically at 1 h intervals ( A). Parallel changes in the number of colony-forming units in each culture were measured by plating diluted samples of each culture on LB agar plates supplemented with ampicillin and chloramphenicol and counting colonies after overnight incubation of plates at 37◦C (B). Cultures without one or both inducers added or with control plasmids served for comparative purposes. Genes, whose transcription in particular cultures was induced, are marked with arrows. Graphs show the average results of the measurement of changes in the optical density of cell cultures ( A) or the number of colony-forming units ( B) in at least three independent experiments. Bars represent the standard deviation. 2.6. Morphological T ransitions during E. coli and E. cloacae Cell Lysis Mediated by P1 or Its Mutants The differences in the mechanisms of canonical and pinholin–SAR-endolysin-mediated lysis were shown to be associated with differences in the morphological transitions of lysing cells [ 29]. Thus, we visualized E. coli cell lysis mediated by P1 or its lydC orlydD mutants by time-lapse microscopy, and compared it with that mediated by Lphage. A total of 2 out of 50 individual L-mediated lysis events captured in samples of 10 independent cultures are shown in Figure 7A,B as time-lapse series (see also Movie S1). Typically, the morphological changes of E. coli cells during L-mediated lysis started from cell shortening, which was associated with the formation of a ball-shaped form at one end (45/50 events, Figure 7A) or in the center of the cell (5/50 events, Figure 7B), and by localized cell rupture. In the case of L, this rupture is caused by holins, which create large holes in the cell membrane, leading to the formation of a transient non-refractile ghost that retains a circular shape, and to the expulsion of the cytoplasmic content, as was observed previously [ 29]. The morphological changes of E. coli cells during P1-mediated lysis appeared to be different from those observed in the case of λ; they progressed through the reduction in cell length and increase in cell diameter (68/70 events) (Figure 7C,D; see also Movie S2), as was ob- served previously in the case of pinholin–SAR-endolysin-mediated lysis [ 29].', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  In the case of L, this rupture is caused by holins, which create large holes in the cell membrane, leading to the formation of a transient non-refractile ghost that retains a circular shape, and to the expulsion of the cytoplasmic content, as was observed previously [ 29]. The morphological changes of E. coli cells during P1-mediated lysis appeared to be different from those observed in the case of λ; they progressed through the reduction in cell length and increase in cell diameter (68/70 events) (Figure 7C,D; see also Movie S2), as was ob- served previously in the case of pinholin–SAR-endolysin-mediated lysis [ 29]. Surprisingly, the inactivation of lydC and lydD genes (Figure 6E,F) resulted in a switch in morpholog- ical changes during lysis to those characteristic of L-mediated lysis: cell was shortened at one end (56/69 events, Figure 7E) or in the center (13/69 events, Figure 7F; see also Movies S3 and S4 ). Taken together, these results demonstrate that LydA is a canonical holin, and support the pinholin function of LydD. Additionally, they indicate that pinholin– SAR-endolysin mediated morphological transitions are dominant during the lysis of E. coli cells at the end of P1 lytic development.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 9 of 20 Int. J. Mol. Sci.  2022 , 23, 4231 9 of 21     as was observed previously in the case of pinholin–SAR-endolysin-mediated lysis [29].  Surprisingly, the inactivation of lydC  and lydD  genes (Figure 6E,F) resulted in a switch in  morphological changes during lysis to those characteristic of ʎ-mediated lysis: cell was  shortened at one end (56/69 even ts, Figure 7E) or in the center  (13/69 events, Figure 7F; see  also Movies S3 and S4). Taken together, these results demonstrate that LydA is a canonical  holin, and support the pinholin function of LydD. Additionally, they indicate that pinho- lin–SAR-endolysin mediated morphological transitions are dominant during the lysis of  E. coli  cells at the end of P1 lytic development.    Figure 7. Morphological transitions of E. coli cells during lysis mediated by ( A,B) phage λ, (C,D)  wild-type phage P1, and ( E,F) phage P1 mutant deprived of functional lydC  and lydD  genes. E. coli  N99 lysogens were thermally induced and withdrawn from the culture just be fore the start of lysis  to document morphological changes during lysis by time-lapse microscopy. The names of bacterio-phages and their genotypes are indicated above each group of photographs. Time (in seconds) is displayed above each frame, with time zero repres enting the first photogra ph acquired before the  observable morphological change or lysis. Bars = 1 µm.  The P1-mediated lysis of E. cloacae  cells progressed through similar morphological  changes to those that accompanied E. coli  cell lysis: the cells shortened and thickened,  converting into a more ball-shaped form, and then cell envelopes disintegrated more or less evenly (52/52 events, Figure 8A; see also Movies S5 and S6). In contrast, E. cloacae  cell  lysis caused by phage P1 with inactivated lydD and lydC  genes was accompanied by slight  cell shortening, the formation of a small protru sion at one cell pole, and the bursting of  the end of this protrusion, resulting in the abrupt expulsion of cell contents through it  (24/25 events, Figure 8B,C). In this case, as in the case of E. coli  with the relevant mutant,  the lysis occurred from one cell pole. Despite that the changes in E. cloacae  and E. coli  mor- phology at this pole during lysis were significantly different, in both cases the lysis was  initiated from one cell pole, fitting to the canonical holin–endolysin-mediated pattern of  cell lysis. This is consistent with the canonical holin function of LydA. Additionally, as in  the case of E. coli , the comparison of morphological transition patterns during the lysis of  E. cloacae  cells with wild-type P1 and P1 lydC lydD  mutant indicates that lysis mediated  Figure 7. Morphological transitions of E. coli cells during lysis mediated by ( A,B) phage λ, (C,D) wild-type phage P1, and ( E,F) phage P1 mutant deprived of functional lydC and lydD genes. E. coli N99 lysogens were thermally induced and withdrawn from the culture just before the start of lysis to document morphological changes during lysis by time-lapse microscopy. The names of bacteriophages and their genotypes are indicated above each group of photographs. Time (in seconds) is displayed above each frame, with time zero representing the ﬁrst photograph acquired before the observable morphological change or lysis. Bars = 1 µm. The P1-mediated lysis of E. cloacae cells progressed through similar morphological changes to those that accompanied E. coli cell lysis: the cells shortened and thickened, converting into a more ball-shaped form, and then cell envelopes disintegrated more or less evenly (52/52 events, Figure 8A; see also Movie S5).', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  E. coli N99 lysogens were thermally induced and withdrawn from the culture just before the start of lysis to document morphological changes during lysis by time-lapse microscopy. The names of bacteriophages and their genotypes are indicated above each group of photographs. Time (in seconds) is displayed above each frame, with time zero representing the ﬁrst photograph acquired before the observable morphological change or lysis. Bars = 1 µm. The P1-mediated lysis of E. cloacae cells progressed through similar morphological changes to those that accompanied E. coli cell lysis: the cells shortened and thickened, converting into a more ball-shaped form, and then cell envelopes disintegrated more or less evenly (52/52 events, Figure 8A; see also Movie S5). In contrast, E. cloacae cell lysis caused by phage P1 with inactivated lydD and lydC genes was accompanied by slight cell shortening, the formation of a small protrusion at one cell pole, and the bursting of the end of this protrusion, resulting in the abrupt expulsion of cell contents through it (24/25 events, Figure 8B,C; see also Movie S6). In this case, as in the case of E. coli with the relevant mutant, the lysis occurred from one cell pole. Despite that the changes in E. cloacae and E. coli morphology at this pole during lysis were signiﬁcantly different, in both cases the lysis was initiated from one cell pole, ﬁtting to the canonical holin–endolysin-mediated pattern of cell lysis. This is consistent with the canonical holin function of LydA. Additionally, as in the case of E. coli , the comparison of morphological transition patterns during the lysis of E. cloacae cells with wild-type P1 and P1 lydC lydD mutant indicates that lysis mediated by wild-type P1 involves the concerted action of all P1-encoded holins/pinholins, and that pinholin-SAR-endolysin-mediated morphological transitions are dominant during E. cloacae cell lysis. 2.7. Influence of P1 lyz Gene Replacement with λR on Cell Morphological T ransitions during Lysis To see whether the replacement of the lyzgene in the P1 genome with λRgene would switch the pattern of morphological transition of the replacement mutant carrying cells during lysis, we visualized lysing E. coli and E. cloacae cells with the replacement mutant. As expected, in both cases, the pattern of morphological changes during lysis switched to that characteristic of canonical holin–endolysin-mediated lysis (Figure 9). A similar switch', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 10 of 20 was observed independent of whether the mutant could produce LydC, consistent with our previous observations that LydA and LydC act as canonical holins and that the pattern of morphological changes during the lysis of E. cloacae cells by P1 is inﬂuenced by the concerted action of LydA, LydC, and LydD. Int. J. Mol. Sci.  2022 , 23, 4231 10 of 21     by wild-type P1 involves the concerted action  of all P1-encoded holins/pinholins, and that  pinholin-SAR-endolysin-mediated morphologi cal transitions are dominant during E. clo- acae cell lysis.    Figure 8. Morphological transitions of E. cloacae cells during lysis mediated by ( A) wild-type phage  P1 and ( B,C) phage P1 mutant deprived of functional lydC  and lydD  genes. E. cloacae  lysogens were  thermally induced and withdrawn from the culture before the start of lysis to document morpho- logical changes during lysis by time-lapse microscopy. The relevant genotypes of P1 phages are indicated above each group of photographs. Time (in seconds) is displayed above each photograph, with  time zero representing the first photograph acquired  before the observable morphological change or ly- sis. Bars = 1 µm.  2.7. Influence of P1 lyz Gene Replacement with λ R on Cell Morphological Transitions during Lysis  To see whether the replacement of the lyz gene in the P1 genome with λ R gene would  switch the pattern of morphological transiti on of the replacement mutant carrying cells  during lysis, we visualized lysing E. coli and E. cloacae  cells with the replacement mutant.  As expected, in both cases, the pattern of morphological changes during lysis switched to  that characteristic of canonical holin–endolysi n-mediated lysis (Figure 9). A similar switch  was observed independent of whether the mu tant could produce LydC, consistent with  our previous observations that LydA and LydC act as canonical holins and that the pat- tern of morphological changes during the lysis of E. cloacae  cells by P1 is influenced by the  concerted action of LydA, LydC, and LydD.  Figure 8. Morphological transitions of E. cloacae cells during lysis mediated by ( A) wild-type phage P1 and ( B,C) phage P1 mutant deprived of functional lydC and lydD genes. E. cloacae lysogens were thermally induced and withdrawn from the culture before the start of lysis to document morphologi- cal changes during lysis by time-lapse microscopy. The relevant genotypes of P1 phages are indicated above each group of photographs. Time (in seconds) is displayed above each photograph, with time zero representing the ﬁrst photograph acquired before the observable morphological change or lysis. Bars = 1 µm. Int. J. Mol. Sci.  2022 , 23, 4231 11 of 21       Figure 9. Morphological transitions of ( A,B) E. coli  and ( C,D) E. cloacae  cells during lysis mediated  by phage P1 with the replacement of lyz gene with the λ R gene ( A,C) and by the P1 replacement  mutant deprived of functional lydC  and lydD  genes ( C,D). E. coli and E. cloacae lysogens were ther- mally induced and withdrawn from the cultures befo re the start of lysis to document morphological  changes during lysis by time-lapse microscopy. Th e names of bacterial species and relevant geno- types of P1 phages are indicated above each group of photographs. Time (in seconds) is displayed  above each photograph, with time zero representi ng the first photograph acquired before the ob- servable morphological change or lysis. Bars = 1 µm.  3. Discussion  The lysis of a phage-infected cell, as soon as the mature phage progeny is ready to be  released, is crucial for the phage’s evolutionary  fate in the world of phages competing for  the same hosts.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  E. coli and E. cloacae lysogens were ther- mally induced and withdrawn from the cultures befo re the start of lysis to document morphological  changes during lysis by time-lapse microscopy. Th e names of bacterial species and relevant geno- types of P1 phages are indicated above each group of photographs. Time (in seconds) is displayed  above each photograph, with time zero representi ng the first photograph acquired before the ob- servable morphological change or lysis. Bars = 1 µm.  3. Discussion  The lysis of a phage-infected cell, as soon as the mature phage progeny is ready to be  released, is crucial for the phage’s evolutionary  fate in the world of phages competing for  the same hosts. The breakage of the cytoplasmic membrane and cell wall integrity is es- sential for lysis under any bacterial growth conditions. Most phages, such as, e.g., λ, can  relay in these processes on a single pair of proteins, one of which (endolysin or SAR-en- dolysin) can digest cell wall peptidoglycan, while the other (holin or pinholin) can kill the  cell by disrupting the transmembrane proton motive force and enabling or facilitating the  access of endolysin/SAR-endolysin to the cell wa ll in an active form (reviewed in [1,9,22]).  The analysis of the P1 bacteriophage genome indicates that its lytic system is more com- plex. In addition to SAR-endolysin (Lyz), ho lin (LydA), and antiholin (LydB), whose func- tions in lysis were confirmed previously [42], P1 encodes two additional proteins of amino  acid sequence features characteristic of holins (LydC and LydD) [40]. In this paper, we  show that LydC and LydD contribute to P1-m ediated cell lysis, and that, while LydA and  LydC function as canonical holins, LydD fu nctions as a pinholin. Moreover, both holins  and the pinholin are essential for the efficien t lysis of bacteria in a programmed time. We  demonstrate the biological sense of such lyti c system complexity by showing the different  contributions of particular ho lins/pinholins to the lysis of cells grown under different con- ditions and cells of different P1 hosts, E. coli  and E. cloacae . Additionally, we demonstrate  for the first time that, while the morphological transitions of the cells of these various hosts  lysed by P1 are similar, they may differ if lysis is supported by an incomplete set of  holins/pinholins.  The redundancy of the genes that could encode proteins of predicted holin features  is not unique for P1 and its relatives [49–53]. However, whether one of these predicted holins in these cases is not an antiholin (a holin antagonist) was verified only in a few  cases. For instance, the endolysin gene of Actinomyces naeslundii  phage Av-1 is flanked by  two genes that encode two functional holins of unrelated amino acid sequences [54]. In  the case of Bacillus  phage SPP1, which also encodes two holin-like proteins, cell lysis could  only be caused by the concerted action of th e endolysin of this phage with both of these  proteins [49,53]. The results of our studies on  P1 presented in this paper demonstrate for  the first time the concerted action of three pr oteins of holin features, namely two holins  Figure 9. Morphological transitions of ( A,B)E. coli and ( C,D)E. cloacae cells during lysis mediated by phage P1 with the replacement of lyzgene with the λRgene ( A,C) and by the P1 replacement mutant deprived of functional lydC and lydD genes ( C,D).E. coli and E. cloacae lysogens were thermally induced and withdrawn from the cultures before the start of lysis to document morphological changes during lysis by time-lapse microscopy. The names of bacterial species and relevant genotypes of P1 phages are indicated above each group of photographs.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:00 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'77'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992135'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'471ms'), (b'x-request-id', b'7e540a5074e806a08afed60a4aa6ebe3'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249970c175d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:00 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'77'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992135'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'471ms'), (b'x-request-id', b'7e540a5074e806a08afed60a4aa6ebe3'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249970c175d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  The results of our studies on  P1 presented in this paper demonstrate for  the first time the concerted action of three pr oteins of holin features, namely two holins  Figure 9. Morphological transitions of ( A,B)E. coli and ( C,D)E. cloacae cells during lysis mediated by phage P1 with the replacement of lyzgene with the λRgene ( A,C) and by the P1 replacement mutant deprived of functional lydC and lydD genes ( C,D).E. coli and E. cloacae lysogens were thermally induced and withdrawn from the cultures before the start of lysis to document morphological changes during lysis by time-lapse microscopy. The names of bacterial species and relevant genotypes of P1 phages are indicated above each group of photographs. Time (in seconds) is displayed above each photograph, with time zero representing the ﬁrst photograph acquired before the observable morphological change or lysis. Bars = 1 µm.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 11 of 20 3. Discussion The lysis of a phage-infected cell, as soon as the mature phage progeny is ready to be released, is crucial for the phage’s evolutionary fate in the world of phages competing for the same hosts. The breakage of the cytoplasmic membrane and cell wall integrity is essential for lysis under any bacterial growth conditions. Most phages, such as, e.g., λ, can relay in these processes on a single pair of proteins, one of which (endolysin or SAR- endolysin) can digest cell wall peptidoglycan, while the other (holin or pinholin) can kill the cell by disrupting the transmembrane proton motive force and enabling or facilitating the access of endolysin/SAR-endolysin to the cell wall in an active form (reviewed in [ 1,9,22]). The analysis of the P1 bacteriophage genome indicates that its lytic system is more complex. In addition to SAR-endolysin (Lyz), holin (LydA), and antiholin (LydB), whose functions in lysis were conﬁrmed previously [ 42], P1 encodes two additional proteins of amino acid sequence features characteristic of holins (LydC and LydD) [ 40]. In this paper, we show that LydC and LydD contribute to P1-mediated cell lysis, and that, while LydA and LydC function as canonical holins, LydD functions as a pinholin. Moreover, both holins and the pinholin are essential for the efﬁcient lysis of bacteria in a programmed time. We demonstrate the biological sense of such lytic system complexity by showing the different contributions of particular holins/pinholins to the lysis of cells grown under different conditions and cells of different P1 hosts, E. coli and E. cloacae . Additionally, we demonstrate for the ﬁrst time that, while the morphological transitions of the cells of these various hosts lysed by P1 are similar, they may differ if lysis is supported by an incomplete set of holins/pinholins. The redundancy of the genes that could encode proteins of predicted holin features is not unique for P1 and its relatives [ 49–53]. However, whether one of these predicted holins in these cases is not an antiholin (a holin antagonist) was veriﬁed only in a few cases. For instance, the endolysin gene of Actinomyces naeslundii phage Av-1 is ﬂanked by two genes that encode two functional holins of unrelated amino acid sequences [ 54]. In the case of Bacillus phage SPP1, which also encodes two holin-like proteins, cell lysis could only be caused by the concerted action of the endolysin of this phage with both of these proteins [ 49,53]. The results of our studies on P1 presented in this paper demonstrate for the ﬁrst time the concerted action of three proteins of holin features, namely two holins and one pinholin, in phage-mediated cell lysis. Additionally, as P1 Lyz protein is the only endolysin encoded by P1, all these proteins cooperate with Lyz in cell lysis. Typically, in the lytic modules of phage genomes, genes for canonical endolysins adjoin the genes for holins, while genes for SAR-endolysins adjoin the genes for pinholins [10–13] (reviewed in [ 38,53,55]). The amino acid sequences of holins and pinholins are much alike [ 22]. They are both characterized by the presence of transmembrane domain(s) and positively charged amino acid residues at the C-terminal end. However, the differences in the number and size of pores formed by holins (a few near-micrometer-scale pores) and pinholins (a multitude of nanometer-scale pores) are associated with phenotypic differences in lysis with the action of these proteins [ 17–19].', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Typically, in the lytic modules of phage genomes, genes for canonical endolysins adjoin the genes for holins, while genes for SAR-endolysins adjoin the genes for pinholins [10–13] (reviewed in [ 38,53,55]). The amino acid sequences of holins and pinholins are much alike [ 22]. They are both characterized by the presence of transmembrane domain(s) and positively charged amino acid residues at the C-terminal end. However, the differences in the number and size of pores formed by holins (a few near-micrometer-scale pores) and pinholins (a multitude of nanometer-scale pores) are associated with phenotypic differences in lysis with the action of these proteins [ 17–19]. First, while holins can permeate the membrane for the unspeciﬁc passage of even large proteins, hence can cooperate with canonical endolysins in cell lysis, pinholins cannot support the lysis of cells producing canonical endolysins [ 19], as the pores that they form are too small for the unspeciﬁc passage of the latter to the periplasmic space. Second, transitions in cell morphology during lysis caused by a canonical endolysin and holin and by a SAR-endolysin and pinholin differ substantially [ 29]. In this paper, we show that, while LydA and LydC can support the lysis of cells by P1-producing canonical endolysin R instead of Lyz, which is characteristic of holins, LydD cannot. The latter result was conﬁrmed by the lack of lysis of cells with the Rand lydD genes cloned in separate plasmids under the control of different inducible promoters. Clearly, the inability of LydD to support the lysis of cells producing the canonical endolysin is not associated with the action of any P1 protein that could potentially inhibit the function or synthesis of LydD. Taken together, these results', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 12 of 20 indicate that LydD is a pinholin, consistent with the location of the lydD gene in the same operon as lyz. The morphological transitions of E. coli cells containing wild-type P1 or its mutants during lysis conform to the functional assignment of LydD to pinholins, and LydA and LydC to canonical holins, and the concerted action of all of them in P1-mediated cell lysis. While the lysis of E. coli by wild-type P1 progressed through morphological changes charac- teristic of SAR-endolysin–pinholin-mediated lysis, lysis by the P1 mutant deprived of LydD progressed through changes characteristic of canonical endolysin–holin-mediated lysis. Comparisons of the sequence and predicted structure of LydD with the sequence and structure of prototypical pinholin S2168 of lambdoid phage 21 indicate similarities as well as differences. Both, LydD and S2168 are in about 80% alpha-helical. They both contain two TMDs and positively charged C-terminus [ 40,56,57]. The best matching pro- teins with a structure similar to the predicted structure of S2168 and LydD and identiﬁed with HHPred [ 58,59] in the PDB database are involved in ion transport. MerF, an 81-aa bacterial mercury transporter, is the best matching structural homolog of S2168 (2MOZ_A; 90.6% probability, matching region of S2168: pos. 2-65), while the 91-aa MrpF subunit of multisubunit Na+/H+ antiporter Mrp is the best matching structural homolog of LydD (6HWH_X; 59.85% probability, matching region: pos. 15-72). MerF is also among the predicted structural homologs of LydD, but it is only in the fourth position among other homologs (37.63% probability, matching region: pos. 12-76). However, while the 5-aa N-terminal region preceding the TMD1 of S2168 contains only one positively charged amino acid residue and the addition of two positively charged amino acid residues to it converts modiﬁed S2168 to a pinholin antagonist that inhibits lysis triggering [ 18], the 12-aa N-terminal region of LydD preceding TMD1 contains four positively charged amino acid residues [ 40]. Additionally, the glycine zipper motif identiﬁed in the TMD2 of S2168 as involved in interhelical interaction in the pathway of pinhole formation [ 60] is absent from LydD (data not shown). Thus, the exact mechanisms of the pinhole formation pathway by LydD and S2168 may differ as do the mechanisms of activation of LydD and S2168 cognate SAR-endolysins (reviewed in [22,56]). Lyz was shown previously to cooperate with LydA in cell lysis when expressed with LydA from a plasmid [ 42]. However, when the Lyz protein is overproduced in E. coli , it causes cell lysis in the absence of holins, if it can be transferred to the periplasmic side of the CM with the help of SecA [ 5]. Thus, the ability of P1 deprived of lydA to retain some activity in E. coli cell lysis was explained previously as attributed to a slow, spontaneous release of the membrane-tethered Lyz. Our results indicate that it can be attributed, at least in part, to the activity of LydD. LydD alone could cooperate with Lyz in the lysis of E. coli cells with cloned lyzand lydD genes, and in the lysis of immobilized E. coli cells by P1. Apparently, under the latter conditions, the canonical Lyz/LydD pairing is more efﬁcient in P1-mediated cell lysis than Lyz/LydA pairing. In a liquid medium, LydD appears to have only an accessory role in P1-mediated lysis.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Thus, the ability of P1 deprived of lydA to retain some activity in E. coli cell lysis was explained previously as attributed to a slow, spontaneous release of the membrane-tethered Lyz. Our results indicate that it can be attributed, at least in part, to the activity of LydD. LydD alone could cooperate with Lyz in the lysis of E. coli cells with cloned lyzand lydD genes, and in the lysis of immobilized E. coli cells by P1. Apparently, under the latter conditions, the canonical Lyz/LydD pairing is more efﬁcient in P1-mediated cell lysis than Lyz/LydA pairing. In a liquid medium, LydD appears to have only an accessory role in P1-mediated lysis. LydA alone could pair with Lyz in lysis under these conditions, suggesting the sufﬁcient production of Lyz in its active form. In its ability to cooperate with pinholin as well as with canonical holin, Lyz resembles the SAR endolysin of lambdoid phage 21, which normally cooperates with pinholin in cell lysis, but could also complement the lysis defect of an induced lambda lysogen encoding a canonical holin and depleted of its canonical endolysin gene [19]. Our results indicate that whether the Lyz–LydD or Lyz–LydA pairing predominate during lysis mediated by P1 depends on cell growth conditions and a host of P1. The P1-mediated lysis of E. coli cells in a liquid culture with shaking depended mainly on LydA, but LydA was not as important for the lysis of E. cloacae cells under similar conditions. While P1 producing LydD or LydC could not cause the efﬁcient lysis of E. coli cells if it did not produce LydA, it could cause the delayed but efﬁcient lysis of E. cloacae cells. Additionally, while LydA appeared to be a dominant cytoplasmic membrane puncturing', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 13 of 20 protein, which could not be replaced by LydD, when bacteria grew with shaking in liquid culture, LydD was sufﬁcient for the lysis of cells immobilized in a solid medium. Perhaps, whether LydA or LydD predominantly cooperate with Lyz under the given conditions of P1 development depends on the regulation of their activity or the regulation of their gene expression. One possible regulator is the LydB protein, which acts as an antiholin, antagonistic to LydA. Phages carrying an amber mutation in lydB lysed cells prematurely [ 43]. Cells with a plasmid carrying the lyz,lydA and lydB genes cloned under the control of an inducible promoter started to lyse much later upon the induction of this promoter than cells containing the isogenic plasmid without lydB [42]. How exactly LydB acts is unknown, as it has no homologs among proteins of known function, and, unlike the antiholin of lambda phage, it does not contain any predicted transmembrane domains. Whether LydB can interact with LydC or LydD has not been studied. The lyzand lydD genes, and the lydA gene with its cognate antiholin gene lydB , are expressed from late P1 promoters, opposite to lydC , which is preceded by a predicted early promoter [ 40,42]. The role of LydC in E. coli cell lysis by P1 appears to be an accessory one independent of the bacterial growth conditions that were tested in this study. The P1 lydA lydD mutant could not form plaques. Additionally, the lack of LydC did not inﬂuence cell lysis in the absence of LydA. One cannot exclude that the level of early lydC expression may predetermine the ratio of LydA to LydD in P1-mediated lysis. A candidate for an additional regulator might be LydE, whose predicted gene partially overlaps the end of lydD , which could suggest antipinholin function. However, the removal of lydE from the lyz lydD lydE operon cloned under the inducible promoter in a plasmid did not inﬂuence lysis upon the induction of transcription from this promoter (Figure 3). Differences in the onset of lysis upon the induction of wild-type P1 and lydA ,lydC or lydD mutants indicate that each of the P1 holin/pinholin inﬂuences lysis timing, which is a parameter of crucial importance for phage ﬁtness. The optimal lysis time that ensures the highest phage fecundity varies even in the case of the same host strain infection by a phage, depending on the ecological settings [ 22,61,62]. Bacterial growth conditions and host cell density are only some examples. For instance, while delaying the onset of lysis, which results in an increased burst size, beneﬁts a phage in environments with a low host concentration, accelerating the onset of lysis can be advantageous in environments with a high host concentration by increasing the phage population growth through a rapid establishment of new infections by progeny phages. Unlike λ, P1 is a bacteriophage with a wide host range among Enterobacteriaceae [39,41]. The differences between its different hosts that determine adsorption and infection efﬁciency, burst size and the duration of latent period may demand the more ﬂexible regulation of lysis timing than in the case of narrow-host-range phages. Additionally, while membrane composition and permeability may not be constant even in the bacteria of the same species exposed to different environ- mental conditions, in the case of bacteria of different genera it may differ substantially, as implied, e.g., from differences between Enterobacteriaceae strains of various genera in their susceptibility to different antibiotics, antimicrobial peptides, and detergents as well as in the amounts of various compounds released upon osmotic shock (see, e.g., [ 63–68]).', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Unlike λ, P1 is a bacteriophage with a wide host range among Enterobacteriaceae [39,41]. The differences between its different hosts that determine adsorption and infection efﬁciency, burst size and the duration of latent period may demand the more ﬂexible regulation of lysis timing than in the case of narrow-host-range phages. Additionally, while membrane composition and permeability may not be constant even in the bacteria of the same species exposed to different environ- mental conditions, in the case of bacteria of different genera it may differ substantially, as implied, e.g., from differences between Enterobacteriaceae strains of various genera in their susceptibility to different antibiotics, antimicrobial peptides, and detergents as well as in the amounts of various compounds released upon osmotic shock (see, e.g., [ 63–68]). Thus, it is not surprising that the ability to permeabilize or puncture the CM of representatives of different bacterial genera efﬁciently and in a programmed time may require more functions and more ﬂexible regulatory mechanisms than the ability to permeabilize or puncture the CM of more closely related bacteria. Conceivably, the contribution of two holins and one pinholin to cell lysis by P1 helps in the choice of optimal lysis timing as well as in the efﬁcient lysis of cells grown under different conditions and originating from different hosts. Pinholin–SAR-endolysin-encoding gene pairs were suggested to represent an inter- mediate stage in the evolution from the most ancient dedicated lysis systems represented by SAR-endolysins, to the holin–endolysin systems, which are considered the most ad- vanced [ 19]. SAR-endolysins alone ensure slow, gradual lysis. Their cooperation with pinholins enables the lysis to be saltatory and regulated in time, but their spontaneous activation makes this regulation imprecise. In the classical endolysin–holin-mediated lysis,', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 14 of 20 cytoplasmic endolysin can be accumulated in a cell in great excess without consequences, making the lysis triggering dependent only on holin activity. It is unclear whether the lytic system of P1 represents simply an evolutionary intermediate between the SAR-endolysin– pinholin system and the endolysin–holin system. In this scenario, the lydA and lydC genes could have been acquired by a P1 ancestor that initially encoded only Lyz and LydD, but any canonical endolysin gene have not been acquired or evolved yet. However, our database search for close homologs of P1 cell lysis-associated proteins indicates that the redundancy of holin/pinholin functions, while retaining SAR-endolysin, may be evolu- tionarily beneﬁcial for Punavirus genus phages, represented by P1. It is conserved not only in P7 phage (see Figure 1), but also in other P1-related phages isolated from various bacteria (GenBank acc. no.: AF503408.1, MZ188976, MH160767; data not shown) [ 69–72]. Possibly, under conditions of competing with numerous phages for different hosts, e.g., in gut environments of enteric bacteria, redundant holins/pinholins could allow lysis triggering in the optimal time while SAR endolysin could additionally ensure premature lysis triggering and release of even a few phage progeny in response to any temporal membrane depolarization caused by a superinfecting phage, which might otherwise over- whelm the original infection [ 22]. If the premature lysis is too early to allow the primary phage maturation, at least it can eliminate the superinfecting phage and hence prevent the production of its progeny. 4. Materials and Methods 4.1. Bacterial Strains, Bacteriophages, and Plasmids Experiments to test the lytic phenotypes of bacteriophages or their mutant deriva- tives were performed with the use of Escherichia coli N99 ( galK2 ,str) [73] and Enterobacter cloacae subsp. cloacae DSM6234 strains (Deutsche Sammlung von Mikroorganismen und Zellkulturen GmbH) as phage hosts. E. coli strain DJ125 ( supE44 ,hsdR ,thi-1 ,trh-1 ,leuB6 , lacY1 ,tonA2 ,recA54 ), which is a recombination-deﬁcient derivative of C600 [ 74,75], was used for plasmid propagation and cloning experiments. Recombinational replacement experiments to transfer mutations from plasmids to bacteriophages were performed with the use of E. coli N99 strain. The bacteriophage P1 used in this study is a mutant (P1 c1-100 IS1::Tn9 mod749 ::IS5) that carries transposon Tn 9with a chloramphenicol resistance marker and the c1-100mutation conferring thermosensitivity to the phage repressor protein [ 40]. The thermosensitivity of phage repressor allows synchronous induction of P1 c1-100Tn9 lysogens upon transfer from 30 to 42◦C. The derivatives of P1 carrying mutations in genes encoding known or predicted lytic functions and constructed in this study are listed in Table 1. The bacteriophage λcI857, which was used as a control and as the source of Rgene, was kindly provided by Anna B˛ ebenek. Table 1. Bacteriophage P1 c1-100 mod749:: IS5IS1::Tn9derivatives constructed in this study.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  The thermosensitivity of phage repressor allows synchronous induction of P1 c1-100Tn9 lysogens upon transfer from 30 to 42◦C. The derivatives of P1 carrying mutations in genes encoding known or predicted lytic functions and constructed in this study are listed in Table 1. The bacteriophage λcI857, which was used as a control and as the source of Rgene, was kindly provided by Anna B˛ ebenek. Table 1. Bacteriophage P1 c1-100 mod749:: IS5IS1::Tn9derivatives constructed in this study. Mutant of Bacteriophage P1c1-100 mod749:: IS5IS1::Tn9P1 Gene(s) Inactivated [Gene Inserted]Construction; Antibiotic Resistance Marker lyz::kanRlyzRecombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9as the recipient and plasmid pWWO2 as the DNA donor; cmR, kanR lydD ::kanRlydDRecombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9as the recipient and plasmid pAKI1 as the DNA donor; cmR, kanR lydC∆12TG lydCRecombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9as the recipient and plasmid pAKI2 as the DNA donor; cmR', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 15 of 20 Table 1. Cont. Mutant of Bacteriophage P1c1-100 mod749:: IS5IS1::Tn9P1 Gene(s) Inactivated [Gene Inserted]Construction; Antibiotic Resistance Marker lydA::195GATC lydARecombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9as the recipient and plasmid pAKI3 as the DNA donor; cmR lydD:: kanRlydC∆12TG lydD ,lydCRecombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9 lydC∆12_13TG as the recipient and plasmid pAKI1 as the DNA donor; cmR, kanR lydD:: kanRlydA::195GATC lydA ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydA::195_196 GATC as the recipient and pAKI1 plasmid as the DNA donor; cmR, kanR lydA::195GATC lydC ∆12TG lydA ,lydCRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydC∆12_13TG as the recipient and plasmid pAKI3 as the DNA donor; cmR ∆lyz::Rλlyz[R]Recombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9as the recipient and plasmid pAKI13 as the DNA donor; cmR ∆lyz::RλlydC∆12TG lyz [R],lydCRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydC∆12_13TG as the recipient and plasmid pAKI13 as the DNA donor; cmR ∆lyz::Rλ lydA::195GATC lydC ∆12TGlyz[R],lydA ,lydCRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydA::195_196GATC lydC ∆12_13TG as the recipient and plasmid pAKI13 as the DNA donor; cmR ∆lyz::Rλ lydA::195GATClyz[R],lydARecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9∆lyz::Rλas the recipient and plasmid pAKI3 as the DNA donor; cmR ∆lyz::RλlydD∆1-79 * lyz[R],lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9as the recipient and plasmid pAKI23 as the DNA donor; cmR ∆lyz::RλlydA ::195GATC lydD∆1-79 *lyz[R],lydA ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydA::195_196GATC as the recipient and pAKI23 plasmid as the DNA donor; cmR, kanR ∆lyz::RλlydC∆12TG lydD∆1-79 *lyz[R],lydC ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydC∆12_13TG as the recipient and plasmid pAKI23 as the DNA donor; cmR ∆lyz::RλlydA ::195GATC lydC∆12TG lydD∆1-79 *lyz[R],lydA ,'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  The results of our studies on  P1 presented in this paper demonstrate for  the first time the concerted action of three pr oteins of holin features, namely two holins  Figure 9. Morphological transitions of ( A,B)E. coli and ( C,D)E. cloacae cells during lysis mediated by phage P1 with the replacement of lyzgene with the λRgene ( A,C) and by the P1 replacement mutant deprived of functional lydC and lydD genes ( C,D).E. coli and E. cloacae lysogens were thermally induced and withdrawn from the cultures before the start of lysis to document morphological changes during lysis by time-lapse microscopy. The names of bacterial species and relevant genotypes of P1 phages are indicated above each group of photographs. Time (in seconds) is displayed above each photograph, with time zero representing the ﬁrst photograph acquired before the observable morphological change or lysis. Bars = 1 µm.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 11 of 20 3. Discussion The lysis of a phage-infected cell, as soon as the mature phage progeny is ready to be released, is crucial for the phage’s evolutionary fate in the world of phages competing for the same hosts. The breakage of the cytoplasmic membrane and cell wall integrity is essential for lysis under any bacterial growth conditions. Most phages, such as, e.g., λ, can relay in these processes on a single pair of proteins, one of which (endolysin or SAR- endolysin) can digest cell wall peptidoglycan, while the other (holin or pinholin) can kill the cell by disrupting the transmembrane proton motive force and enabling or facilitating the access of endolysin/SAR-endolysin to the cell wall in an active form (reviewed in [ 1,9,22]). The analysis of the P1 bacteriophage genome indicates that its lytic system is more complex. In addition to SAR-endolysin (Lyz), holin (LydA), and antiholin (LydB), whose functions in lysis were conﬁrmed previously [ 42], P1 encodes two additional proteins of amino acid sequence features characteristic of holins (LydC and LydD) [ 40]. In this paper, we show that LydC and LydD contribute to P1-mediated cell lysis, and that, while LydA and LydC function as canonical holins, LydD functions as a pinholin. Moreover, both holins and the pinholin are essential for the efﬁcient lysis of bacteria in a programmed time. We demonstrate the biological sense of such lytic system complexity by showing the different contributions of particular holins/pinholins to the lysis of cells grown under different conditions and cells of different P1 hosts, E. coli and E. cloacae . Additionally, we demonstrate for the ﬁrst time that, while the morphological transitions of the cells of these various hosts lysed by P1 are similar, they may differ if lysis is supported by an incomplete set of holins/pinholins. The redundancy of the genes that could encode proteins of predicted holin features is not unique for P1 and its relatives [ 49–53]. However, whether one of these predicted holins in these cases is not an antiholin (a holin antagonist) was veriﬁed only in a few cases. For instance, the endolysin gene of Actinomyces naeslundii phage Av-1 is ﬂanked by two genes that encode two functional holins of unrelated amino acid sequences [ 54]. In the case of Bacillus phage SPP1, which also encodes two holin-like proteins, cell lysis could only be caused by the concerted action of the endolysin of this phage with both of these proteins [ 49,53]. The results of our studies on P1 presented in this paper demonstrate for the ﬁrst time the concerted action of three proteins of holin features, namely two holins and one pinholin, in phage-mediated cell lysis. Additionally, as P1 Lyz protein is the only endolysin encoded by P1, all these proteins cooperate with Lyz in cell lysis. Typically, in the lytic modules of phage genomes, genes for canonical endolysins adjoin the genes for holins, while genes for SAR-endolysins adjoin the genes for pinholins [10–13] (reviewed in [ 38,53,55]). The amino acid sequences of holins and pinholins are much alike [ 22]. They are both characterized by the presence of transmembrane domain(s) and positively charged amino acid residues at the C-terminal end. However, the differences in the number and size of pores formed by holins (a few near-micrometer-scale pores) and pinholins (a multitude of nanometer-scale pores) are associated with phenotypic differences in lysis with the action of these proteins [ 17–19].', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Typically, in the lytic modules of phage genomes, genes for canonical endolysins adjoin the genes for holins, while genes for SAR-endolysins adjoin the genes for pinholins [10–13] (reviewed in [ 38,53,55]). The amino acid sequences of holins and pinholins are much alike [ 22]. They are both characterized by the presence of transmembrane domain(s) and positively charged amino acid residues at the C-terminal end. However, the differences in the number and size of pores formed by holins (a few near-micrometer-scale pores) and pinholins (a multitude of nanometer-scale pores) are associated with phenotypic differences in lysis with the action of these proteins [ 17–19]. First, while holins can permeate the membrane for the unspeciﬁc passage of even large proteins, hence can cooperate with canonical endolysins in cell lysis, pinholins cannot support the lysis of cells producing canonical endolysins [ 19], as the pores that they form are too small for the unspeciﬁc passage of the latter to the periplasmic space. Second, transitions in cell morphology during lysis caused by a canonical endolysin and holin and by a SAR-endolysin and pinholin differ substantially [ 29]. In this paper, we show that, while LydA and LydC can support the lysis of cells by P1-producing canonical endolysin R instead of Lyz, which is characteristic of holins, LydD cannot. The latter result was conﬁrmed by the lack of lysis of cells with the Rand lydD genes cloned in separate plasmids under the control of different inducible promoters. Clearly, the inability of LydD to support the lysis of cells producing the canonical endolysin is not associated with the action of any P1 protein that could potentially inhibit the function or synthesis of LydD. Taken together, these results', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 12 of 20 indicate that LydD is a pinholin, consistent with the location of the lydD gene in the same operon as lyz. The morphological transitions of E. coli cells containing wild-type P1 or its mutants during lysis conform to the functional assignment of LydD to pinholins, and LydA and LydC to canonical holins, and the concerted action of all of them in P1-mediated cell lysis. While the lysis of E. coli by wild-type P1 progressed through morphological changes charac- teristic of SAR-endolysin–pinholin-mediated lysis, lysis by the P1 mutant deprived of LydD progressed through changes characteristic of canonical endolysin–holin-mediated lysis. Comparisons of the sequence and predicted structure of LydD with the sequence and structure of prototypical pinholin S2168 of lambdoid phage 21 indicate similarities as well as differences. Both, LydD and S2168 are in about 80% alpha-helical. They both contain two TMDs and positively charged C-terminus [ 40,56,57]. The best matching pro- teins with a structure similar to the predicted structure of S2168 and LydD and identiﬁed with HHPred [ 58,59] in the PDB database are involved in ion transport. MerF, an 81-aa bacterial mercury transporter, is the best matching structural homolog of S2168 (2MOZ_A; 90.6% probability, matching region of S2168: pos. 2-65), while the 91-aa MrpF subunit of multisubunit Na+/H+ antiporter Mrp is the best matching structural homolog of LydD (6HWH_X; 59.85% probability, matching region: pos. 15-72). MerF is also among the predicted structural homologs of LydD, but it is only in the fourth position among other homologs (37.63% probability, matching region: pos. 12-76). However, while the 5-aa N-terminal region preceding the TMD1 of S2168 contains only one positively charged amino acid residue and the addition of two positively charged amino acid residues to it converts modiﬁed S2168 to a pinholin antagonist that inhibits lysis triggering [ 18], the 12-aa N-terminal region of LydD preceding TMD1 contains four positively charged amino acid residues [ 40]. Additionally, the glycine zipper motif identiﬁed in the TMD2 of S2168 as involved in interhelical interaction in the pathway of pinhole formation [ 60] is absent from LydD (data not shown). Thus, the exact mechanisms of the pinhole formation pathway by LydD and S2168 may differ as do the mechanisms of activation of LydD and S2168 cognate SAR-endolysins (reviewed in [22,56]). Lyz was shown previously to cooperate with LydA in cell lysis when expressed with LydA from a plasmid [ 42]. However, when the Lyz protein is overproduced in E. coli , it causes cell lysis in the absence of holins, if it can be transferred to the periplasmic side of the CM with the help of SecA [ 5]. Thus, the ability of P1 deprived of lydA to retain some activity in E. coli cell lysis was explained previously as attributed to a slow, spontaneous release of the membrane-tethered Lyz. Our results indicate that it can be attributed, at least in part, to the activity of LydD. LydD alone could cooperate with Lyz in the lysis of E. coli cells with cloned lyzand lydD genes, and in the lysis of immobilized E. coli cells by P1. Apparently, under the latter conditions, the canonical Lyz/LydD pairing is more efﬁcient in P1-mediated cell lysis than Lyz/LydA pairing. In a liquid medium, LydD appears to have only an accessory role in P1-mediated lysis.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Thus, the ability of P1 deprived of lydA to retain some activity in E. coli cell lysis was explained previously as attributed to a slow, spontaneous release of the membrane-tethered Lyz. Our results indicate that it can be attributed, at least in part, to the activity of LydD. LydD alone could cooperate with Lyz in the lysis of E. coli cells with cloned lyzand lydD genes, and in the lysis of immobilized E. coli cells by P1. Apparently, under the latter conditions, the canonical Lyz/LydD pairing is more efﬁcient in P1-mediated cell lysis than Lyz/LydA pairing. In a liquid medium, LydD appears to have only an accessory role in P1-mediated lysis. LydA alone could pair with Lyz in lysis under these conditions, suggesting the sufﬁcient production of Lyz in its active form. In its ability to cooperate with pinholin as well as with canonical holin, Lyz resembles the SAR endolysin of lambdoid phage 21, which normally cooperates with pinholin in cell lysis, but could also complement the lysis defect of an induced lambda lysogen encoding a canonical holin and depleted of its canonical endolysin gene [19]. Our results indicate that whether the Lyz–LydD or Lyz–LydA pairing predominate during lysis mediated by P1 depends on cell growth conditions and a host of P1. The P1-mediated lysis of E. coli cells in a liquid culture with shaking depended mainly on LydA, but LydA was not as important for the lysis of E. cloacae cells under similar conditions. While P1 producing LydD or LydC could not cause the efﬁcient lysis of E. coli cells if it did not produce LydA, it could cause the delayed but efﬁcient lysis of E. cloacae cells. Additionally, while LydA appeared to be a dominant cytoplasmic membrane puncturing', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 13 of 20 protein, which could not be replaced by LydD, when bacteria grew with shaking in liquid culture, LydD was sufﬁcient for the lysis of cells immobilized in a solid medium. Perhaps, whether LydA or LydD predominantly cooperate with Lyz under the given conditions of P1 development depends on the regulation of their activity or the regulation of their gene expression. One possible regulator is the LydB protein, which acts as an antiholin, antagonistic to LydA. Phages carrying an amber mutation in lydB lysed cells prematurely [ 43]. Cells with a plasmid carrying the lyz,lydA and lydB genes cloned under the control of an inducible promoter started to lyse much later upon the induction of this promoter than cells containing the isogenic plasmid without lydB [42]. How exactly LydB acts is unknown, as it has no homologs among proteins of known function, and, unlike the antiholin of lambda phage, it does not contain any predicted transmembrane domains. Whether LydB can interact with LydC or LydD has not been studied. The lyzand lydD genes, and the lydA gene with its cognate antiholin gene lydB , are expressed from late P1 promoters, opposite to lydC , which is preceded by a predicted early promoter [ 40,42]. The role of LydC in E. coli cell lysis by P1 appears to be an accessory one independent of the bacterial growth conditions that were tested in this study. The P1 lydA lydD mutant could not form plaques. Additionally, the lack of LydC did not inﬂuence cell lysis in the absence of LydA. One cannot exclude that the level of early lydC expression may predetermine the ratio of LydA to LydD in P1-mediated lysis. A candidate for an additional regulator might be LydE, whose predicted gene partially overlaps the end of lydD , which could suggest antipinholin function. However, the removal of lydE from the lyz lydD lydE operon cloned under the inducible promoter in a plasmid did not inﬂuence lysis upon the induction of transcription from this promoter (Figure 3). Differences in the onset of lysis upon the induction of wild-type P1 and lydA ,lydC or lydD mutants indicate that each of the P1 holin/pinholin inﬂuences lysis timing, which is a parameter of crucial importance for phage ﬁtness. The optimal lysis time that ensures the highest phage fecundity varies even in the case of the same host strain infection by a phage, depending on the ecological settings [ 22,61,62]. Bacterial growth conditions and host cell density are only some examples. For instance, while delaying the onset of lysis, which results in an increased burst size, beneﬁts a phage in environments with a low host concentration, accelerating the onset of lysis can be advantageous in environments with a high host concentration by increasing the phage population growth through a rapid establishment of new infections by progeny phages. Unlike λ, P1 is a bacteriophage with a wide host range among Enterobacteriaceae [39,41]. The differences between its different hosts that determine adsorption and infection efﬁciency, burst size and the duration of latent period may demand the more ﬂexible regulation of lysis timing than in the case of narrow-host-range phages. Additionally, while membrane composition and permeability may not be constant even in the bacteria of the same species exposed to different environ- mental conditions, in the case of bacteria of different genera it may differ substantially, as implied, e.g., from differences between Enterobacteriaceae strains of various genera in their susceptibility to different antibiotics, antimicrobial peptides, and detergents as well as in the amounts of various compounds released upon osmotic shock (see, e.g., [ 63–68]).', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Unlike λ, P1 is a bacteriophage with a wide host range among Enterobacteriaceae [39,41]. The differences between its different hosts that determine adsorption and infection efﬁciency, burst size and the duration of latent period may demand the more ﬂexible regulation of lysis timing than in the case of narrow-host-range phages. Additionally, while membrane composition and permeability may not be constant even in the bacteria of the same species exposed to different environ- mental conditions, in the case of bacteria of different genera it may differ substantially, as implied, e.g., from differences between Enterobacteriaceae strains of various genera in their susceptibility to different antibiotics, antimicrobial peptides, and detergents as well as in the amounts of various compounds released upon osmotic shock (see, e.g., [ 63–68]). Thus, it is not surprising that the ability to permeabilize or puncture the CM of representatives of different bacterial genera efﬁciently and in a programmed time may require more functions and more ﬂexible regulatory mechanisms than the ability to permeabilize or puncture the CM of more closely related bacteria. Conceivably, the contribution of two holins and one pinholin to cell lysis by P1 helps in the choice of optimal lysis timing as well as in the efﬁcient lysis of cells grown under different conditions and originating from different hosts. Pinholin–SAR-endolysin-encoding gene pairs were suggested to represent an inter- mediate stage in the evolution from the most ancient dedicated lysis systems represented by SAR-endolysins, to the holin–endolysin systems, which are considered the most ad- vanced [ 19]. SAR-endolysins alone ensure slow, gradual lysis. Their cooperation with pinholins enables the lysis to be saltatory and regulated in time, but their spontaneous activation makes this regulation imprecise. In the classical endolysin–holin-mediated lysis,', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 14 of 20 cytoplasmic endolysin can be accumulated in a cell in great excess without consequences, making the lysis triggering dependent only on holin activity. It is unclear whether the lytic system of P1 represents simply an evolutionary intermediate between the SAR-endolysin– pinholin system and the endolysin–holin system. In this scenario, the lydA and lydC genes could have been acquired by a P1 ancestor that initially encoded only Lyz and LydD, but any canonical endolysin gene have not been acquired or evolved yet. However, our database search for close homologs of P1 cell lysis-associated proteins indicates that the redundancy of holin/pinholin functions, while retaining SAR-endolysin, may be evolu- tionarily beneﬁcial for Punavirus genus phages, represented by P1. It is conserved not only in P7 phage (see Figure 1), but also in other P1-related phages isolated from various bacteria (GenBank acc. no.: AF503408.1, MZ188976, MH160767; data not shown) [ 69–72]. Possibly, under conditions of competing with numerous phages for different hosts, e.g., in gut environments of enteric bacteria, redundant holins/pinholins could allow lysis triggering in the optimal time while SAR endolysin could additionally ensure premature lysis triggering and release of even a few phage progeny in response to any temporal membrane depolarization caused by a superinfecting phage, which might otherwise over- whelm the original infection [ 22]. If the premature lysis is too early to allow the primary phage maturation, at least it can eliminate the superinfecting phage and hence prevent the production of its progeny. 4. Materials and Methods 4.1. Bacterial Strains, Bacteriophages, and Plasmids Experiments to test the lytic phenotypes of bacteriophages or their mutant deriva- tives were performed with the use of Escherichia coli N99 ( galK2 ,str) [73] and Enterobacter cloacae subsp. cloacae DSM6234 strains (Deutsche Sammlung von Mikroorganismen und Zellkulturen GmbH) as phage hosts. E. coli strain DJ125 ( supE44 ,hsdR ,thi-1 ,trh-1 ,leuB6 , lacY1 ,tonA2 ,recA54 ), which is a recombination-deﬁcient derivative of C600 [ 74,75], was used for plasmid propagation and cloning experiments. Recombinational replacement experiments to transfer mutations from plasmids to bacteriophages were performed with the use of E. coli N99 strain. The bacteriophage P1 used in this study is a mutant (P1 c1-100 IS1::Tn9 mod749 ::IS5) that carries transposon Tn 9with a chloramphenicol resistance marker and the c1-100mutation conferring thermosensitivity to the phage repressor protein [ 40]. The thermosensitivity of phage repressor allows synchronous induction of P1 c1-100Tn9 lysogens upon transfer from 30 to 42◦C. The derivatives of P1 carrying mutations in genes encoding known or predicted lytic functions and constructed in this study are listed in Table 1. The bacteriophage λcI857, which was used as a control and as the source of Rgene, was kindly provided by Anna B˛ ebenek. Table 1. Bacteriophage P1 c1-100 mod749:: IS5IS1::Tn9derivatives constructed in this study.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  The thermosensitivity of phage repressor allows synchronous induction of P1 c1-100Tn9 lysogens upon transfer from 30 to 42◦C. The derivatives of P1 carrying mutations in genes encoding known or predicted lytic functions and constructed in this study are listed in Table 1. The bacteriophage λcI857, which was used as a control and as the source of Rgene, was kindly provided by Anna B˛ ebenek. Table 1. Bacteriophage P1 c1-100 mod749:: IS5IS1::Tn9derivatives constructed in this study. Mutant of Bacteriophage P1c1-100 mod749:: IS5IS1::Tn9P1 Gene(s) Inactivated [Gene Inserted]Construction; Antibiotic Resistance Marker lyz::kanRlyzRecombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9as the recipient and plasmid pWWO2 as the DNA donor; cmR, kanR lydD ::kanRlydDRecombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9as the recipient and plasmid pAKI1 as the DNA donor; cmR, kanR lydC∆12TG lydCRecombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9as the recipient and plasmid pAKI2 as the DNA donor; cmR', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 15 of 20 Table 1. Cont. Mutant of Bacteriophage P1c1-100 mod749:: IS5IS1::Tn9P1 Gene(s) Inactivated [Gene Inserted]Construction; Antibiotic Resistance Marker lydA::195GATC lydARecombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9as the recipient and plasmid pAKI3 as the DNA donor; cmR lydD:: kanRlydC∆12TG lydD ,lydCRecombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9 lydC∆12_13TG as the recipient and plasmid pAKI1 as the DNA donor; cmR, kanR lydD:: kanRlydA::195GATC lydA ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydA::195_196 GATC as the recipient and pAKI1 plasmid as the DNA donor; cmR, kanR lydA::195GATC lydC ∆12TG lydA ,lydCRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydC∆12_13TG as the recipient and plasmid pAKI3 as the DNA donor; cmR ∆lyz::Rλlyz[R]Recombinational replacement with the use of P1 c1-100 mod749:: IS5IS1::Tn9as the recipient and plasmid pAKI13 as the DNA donor; cmR ∆lyz::RλlydC∆12TG lyz [R],lydCRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydC∆12_13TG as the recipient and plasmid pAKI13 as the DNA donor; cmR ∆lyz::Rλ lydA::195GATC lydC ∆12TGlyz[R],lydA ,lydCRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydA::195_196GATC lydC ∆12_13TG as the recipient and plasmid pAKI13 as the DNA donor; cmR ∆lyz::Rλ lydA::195GATClyz[R],lydARecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9∆lyz::Rλas the recipient and plasmid pAKI3 as the DNA donor; cmR ∆lyz::RλlydD∆1-79 * lyz[R],lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9as the recipient and plasmid pAKI23 as the DNA donor; cmR ∆lyz::RλlydA ::195GATC lydD∆1-79 *lyz[R],lydA ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydA::195_196GATC as the recipient and pAKI23 plasmid as the DNA donor; cmR, kanR ∆lyz::RλlydC∆12TG lydD∆1-79 *lyz[R],lydC ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydC∆12_13TG as the recipient and plasmid pAKI23 as the DNA donor; cmR ∆lyz::RλlydA ::195GATC lydC∆12TG lydD∆1-79 *lyz[R],lydA ,'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:01 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'76'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993888'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'366ms'), (b'x-request-id', b'5ae69dd54ada193e8e4554d6cf236d51'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412499bd85f5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:01 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'76'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993888'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'366ms'), (b'x-request-id', b'5ae69dd54ada193e8e4554d6cf236d51'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412499bd85f5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  lydA ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydA::195_196GATC as the recipient and pAKI23 plasmid as the DNA donor; cmR, kanR ∆lyz::RλlydC∆12TG lydD∆1-79 *lyz[R],lydC ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydC∆12_13TG as the recipient and plasmid pAKI23 as the DNA donor; cmR ∆lyz::RλlydA ::195GATC lydC∆12TG lydD∆1-79 *lyz[R],lydA ,lydC ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydA::195_196GATC lydC ∆12_13TG as the recipient and plasmid pAKI23 as the DNA donor; cmR lydD∆16-81 * lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9as the recipient and plasmid pAKI26 as the DNA donor; cmR *lydE gene and the Shine–Dalgarno sequence of lydE are intact in the P1 lydD deletion mutants. The plasmids used are derivatives of pMB1, pACYC184, or pSC101, and are listed in Table S1. Nucleotide sequences of plasmid inserts that were obtained by ampliﬁcation were veriﬁed by sequencing. Primers that were used for the PCR ampliﬁcation of the desired fragments or for sequencing are listed in Table S2. Oligonucleotide synthesis and sequenc- ing services were provided by the Laboratory of DNA Sequencing and Oligonucleotide Synthesis of IBB PAS.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 16 of 20 4.2. Bacterial Growth Conditions and Bacteriophage Propagation Bacteria were grown in Luria–Bertani liquid medium (LB; Difco) or on LB medium solidiﬁed with 1.5% agar at 30, 37, or 42◦C, as indicated. When required, the medium was supplemented with glucose or maltose to a ﬁnal concentration 0.2%, or with antibiotics: ampicillin (amp) (50, 100, or 500 µg/mL), chloramphenicol (cm) (12.5 or 25 µg/mL), or kanamycin (km) (12.5 or 25 µg/mL). Incubations of liquid cultures were performed with shaking (200 rpm) overnight or until the desired optical density (OD 600). Bacteriophages were propagated upon thermal induction of lysogens. Brieﬂy, cells lysogenized with λcI857or P1 c1-100IS1::Tn9 mod749::IS5 (or its mutant derivatives) were grown in LB or LB supplemented with chloramphenicol (12.5 µg/mL), respectively, overnight at 30◦C. The overnight cultures were diluted 50 ×in fresh LB with 0.2% maltose or 0.2% glucose and chloramphenicol, respectively, and grown at 30◦C until the optical density (OD 600) reached 0.3–0.4. Prophages were thermally induced by rapid heating of cultures to 42◦C in a shaking water bath heated to about 60◦C, then the cultures were returned to 42◦C and grown with shaking for about 45–60 min until signs of lysis were noticed. The crude lysates were harvested when the OD 600fell to 0.15 and centrifuged at 29,030 ×gfor 40 min at 4◦C. The remaining cells and cell debris were removed from the lysates by ﬁltration through 0.22 µm syringe ﬁlter (Filtropur, Sarstedt, Nümbrecht, Germany). The titer of phages in the lysates was assayed using the standard double-layer agar method, according to Adams (1959) [76]. 4.3. DNA Manipulation DNA ampliﬁcation, plasmid isolation, and agarose gel electrophoresis were carried out using standard techniques [ 77].E. coli cells were transformed with plasmids as described previously [ 77]. Restriction enzymes, T4 DNA ligase (Sigma-Aldrich, St. Louis, MO, USA), the Klenow fragment of E. coli DNA polymerase I (Thermo Fisher Scientiﬁc, Waltham, MA, USA), thermostable Ex Taq DNA polymerase (Takara, Kusatsu, Japan), and DreamTaq DNA Polymerase (Thermo Fisher Scientiﬁc, Waltham, MA, USA) were used according to the supplier’s protocols. 4.4. Lysogenization A portion of fresh overnight culture of bacteria (100 µL) was supplemented with 100µLof 20 mM CaCl 2and 100 µL of 20 mM MgSO 4mixed with 100 µL of phage lysate (10−8–10−9PFU/mL) and incubated at room temperature for 20 min to let the phages to adsorb the bacteria. Adsorption was stopped by the addition of 1 M sodium citrate (200 µL) and vigorous mixing, followed by the addition of 1 mL LB and incubation of the mixtures for 60 min at 30◦C to express the antibiotic-resistance phenotype. The phage–host mixture was then plated on solid LB medium supplemented with chloramphenicol and, when required, additional antibiotics, and incubated overnight at 30◦C to obtain colonies of lysogens. 4.5.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Adsorption was stopped by the addition of 1 M sodium citrate (200 µL) and vigorous mixing, followed by the addition of 1 mL LB and incubation of the mixtures for 60 min at 30◦C to express the antibiotic-resistance phenotype. The phage–host mixture was then plated on solid LB medium supplemented with chloramphenicol and, when required, additional antibiotics, and incubated overnight at 30◦C to obtain colonies of lysogens. 4.5. Construction of P1 Phage Mutants All P1 c1-100 mod749 ::IS5IS1::Tn9mutants were constructed by RecA-mediated re- combinational replacement in vivo using plasmids carrying relevant P1 genes with the desired mutations as DNA donors (see Table S1). Cells carrying the relevant plasmid were lysogenized with phage P1 c1-100 mod749 ::IS5IS1::Tn9, and the lysogens were selected on LB solid medium supplemented with chloramphenicol (12.5 µg/mL) and ampicillin (100 µg/mL), and, when required, kanamycin (25 µg/mL in the case of plasmids carrying P1 genes inactivated by the insertion of kanRcassette). They were puriﬁed through single colonies and used to thermally induce prophages, as described above. The obtained phages were used to lysogenize N99 cells. Lysogens expected to contain the insertions of the kanR cassette in the target prophage gene were selected on LB medium supplemented with chloramphenicol and kanamycin, and then screened for resistance to ampicillin by testing', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 17 of 20 their ability to grow on LB medium supplemented with chloramphenicol, kanamycin, and ampicillin (50 µg/mL). Lysogens sensitive to ampicillin were used for further studies. Lysogens expected to contain a mutation without an antibiotic resistance determinant in the target prophage gene were selected on LB medium supplemented with chlorampheni- col (12.5 µg/mL) and ampicillin (50 µg/mL). To increase the probability of recombination between the phage and the donor DNA in the plasmid, each lysogen was used to inoculate 5 mL of LB medium supplemented with chloramphenicol (12.5 µg/mL) and ampicillin (50µg/mL). The lysogen cultures were grown overnight with shaking at 30◦C and plated on LB solid medium supplemented with chloramphenicol (12.5 µg/mL) and ampicillin (500 µg/mL). Single colonies of lysogens able to grow on this medium were used for ther- mal induction of prophages. The obtained phages were used to lysogenize N99 cells, and the lysogens were selected on LB solid medium supplemented with chloramphenicol only (12.5 µg/mL). Lysogens were subsequently screened for the ability to grow on medium supplemented with chloramphenicol and ampicillin (50 µg/mL), to eliminate those that contained a donor plasmid either integrated with the prophage or in a circular form. Lyso- gens sensitive to ampicillin were tested for the presence of the desired mutation in the prophage by PCR ampliﬁcation of the relevant DNA fragment and restriction digestion or sequencing of the obtained amplicon. 4.6. Assays of Lysis Kinetics A fresh overnight culture of cells lysogenized with P1 c1-100 mod749 ::IS5IS1::Tn9or its mutant was used to inoculate LB medium supplemented with glucose and chloram- phenicol to an optical density (OD 600) of about 0.1. Cells were grown with shaking at 30◦C. When the OD 600reached 0.3–0.4, the prophage was induced by thermal induction, as described above, and further changes in the optical density of the culture were measured spectrophotometrically at indicated time intervals. 4.7. Time-Lapse Microscopy of Living Cells Morphological changes accompanying cell lysis upon the thermal induction of prophages in lysogens were monitored by the time-lapse microscopy of living cells. To immobilize cells in one focal plane, samples (10 µL) withdrawn from the cultures of induced lysogens before the expected lysis time were placed on a microscopic slide on the upper surface of an agarose microslab (0.7% agarose in physiological saline) and covered with a coverslip. The microslabs were prepared as described previously [ 78]. Cells were immediately im- aged using a Zeiss Axio M2 microscope with an oil immersion objective (100 ×, numerical aperture = 1.4). A time-lapse video was captured at ﬁve frames per second. The video was edited and scaled using the AxioVision 4 software package (Zeiss, Oberkochen, Germany). All images were stored as .zvi ﬁles and converted to TIFF or JPEG format as necessary. Supplementary Materials: The following are available online at https://www.mdpi.com/article/10 .3390/ijms23084231/s1. Author Contributions: Conceptualization, A.B. and M.Ł.; methodology, A.B., A.C., W.I. and M.Ł.; validation, A.B. and M.Ł.; formal analysis, A.B. and M.Ł.; investigation, A.B., A.C., W.I. and J.B.; resources, A.B. and M.Ł.; data curation, M.Ł.; writing—original draft preparation, A.B. and M.Ł.; writing—review and editing, M.Ł.; visualization, A.B.; supervision, M.Ł.; project administration, A.B. and M.Ł.; funding acquisition, A.B. and M.Ł. All authors have read and agreed to the published version of the manuscript.', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Author Contributions: Conceptualization, A.B. and M.Ł.; methodology, A.B., A.C., W.I. and M.Ł.; validation, A.B. and M.Ł.; formal analysis, A.B. and M.Ł.; investigation, A.B., A.C., W.I. and J.B.; resources, A.B. and M.Ł.; data curation, M.Ł.; writing—original draft preparation, A.B. and M.Ł.; writing—review and editing, M.Ł.; visualization, A.B.; supervision, M.Ł.; project administration, A.B. and M.Ł.; funding acquisition, A.B. and M.Ł. All authors have read and agreed to the published version of the manuscript. Funding: This research was funded by the National Science Centre (PRELUDIUM; Grant No. UMO- 2019/35/N/NZ6/03789) and by statutory funds from the Institute of Biochemistry and Biophysics of the Polish Academy of Sciences. Institutional Review Board Statement: Not applicable. Informed Consent Statement: Not applicable. Data Availability Statement: Not applicable.', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 18 of 20 Acknowledgments: We thank Agnieszka Laudy from the Warsaw Medical University for lending us the E. cloacae subsp. cloacae strain DSM6234, which was used in a part of this study, and Anna B˛ ebenek of the Institute of Biochemistry and Biophysics, PAS, for providing us with the lysogen of phage λcI857. Conﬂicts of Interest: The authors declare no conﬂict of interest. The funders had no role in the design of the study; in the collection, analyses, or interpretation of data; in the writing of the manuscript; or in the decision to publish the results. References 1. Wang, I.N.; Smith, D.L.; Young, R. Holins: The Protein Clocks of Bacteriophage Infections. Annu. Rev. Microbiol. 2000 ,54, 799–825. [CrossRef] [PubMed] 2. Wang, I.-N.; Deaton, J.; Young, R. Sizing the Holin Lesion with an Endolysin- β-Galactosidase Fusion. J. Bacteriol. 2003 ,185, 779–787. [CrossRef] [PubMed] 3. Oliveira, H.; Melo, L.D.R.; Santos, S.B.; Nobrega, F.L.; Ferreira, E.C.; Cerca, N.; Azeredo, J.; Kluskens, L.D. Molecular Aspects and Comparative Genomics of Bacteriophage Endolysins. J. Virol. 2013 ,87, 4558–4570. [CrossRef] [PubMed] 4. Dewey, J.S.; Savva, C.G.; White, R.L.; Vitha, S.; Holzenburg, A.; Young, R. Micron-Scale Holes Terminate the Phage Infection Cycle. Proc. Natl. Acad. Sci. USA 2010 ,107, 2219–2223. [CrossRef] 5. Xu, M.; Struck, D.K.; Deaton, J.; Wang, I.-N.; Young, R. A Signal-Arrest-Release Sequence Mediates Export and Control of the Phage P1 Endolysin. Proc. Natl. Acad. Sci. USA 2004 ,101, 6415–6420. [CrossRef] 6. Kuty, G.F.; Xu, M.; Struck, D.K.; Summer, E.J.; Young, R. Regulation of a Phage Endolysin by Disulﬁde Caging. J. Bacteriol. 2010 , 192, 5682–5687. [CrossRef] 7. São-José, C.; Parreira, R.; Vieira, G.; Santos, M.A. The N-Terminal Region of the Oenococcus Oeni Bacteriophage FOg44 Lysin Behaves as a Bona Fide Signal Peptide in Escherichia Coli and as a Cis-Inhibitory Element, Preventing Lytic Activity on Oenococcal Cells. J. Bacteriol. 2000 ,182, 5823–5831. [CrossRef] 8. Zampara, A.; Ahern, S.J.; Briers, Y.; Brøndsted, L.; Sørensen, M.C.H. Two Distinct Modes of Lysis Regulation in Campylobacter Fletchervirus and Firehammervirus Phages. Viruses 2020 ,12, 1247. [CrossRef] 9. Cahill, J.; Young, R. Phage Lysis: Multiple Genes for Multiple Barriers. In Advances in Virus Research ; Elsevier: Amsterdam, The Netherlands, 2019; Volume 103, pp. 33–70. [CrossRef] 10. Fernandes, S.; S ão-José, C. Enzymes and Mechanisms Employed by Tailed Bacteriophages to Breach the Bacterial Cell Barriers. Viruses 2018 ,10, 396.', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  ; Briers, Y.; Brøndsted, L.; Sørensen, M.C.H. Two Distinct Modes of Lysis Regulation in Campylobacter Fletchervirus and Firehammervirus Phages. Viruses 2020 ,12, 1247. [CrossRef] 9. Cahill, J.; Young, R. Phage Lysis: Multiple Genes for Multiple Barriers. In Advances in Virus Research ; Elsevier: Amsterdam, The Netherlands, 2019; Volume 103, pp. 33–70. [CrossRef] 10. Fernandes, S.; S ão-José, C. Enzymes and Mechanisms Employed by Tailed Bacteriophages to Breach the Bacterial Cell Barriers. Viruses 2018 ,10, 396. [CrossRef] 11. Gontijo, M.T.P .; Vidigal, P .M.P .; Lopez, M.E.S.; Brocchi, M. Bacteriophages That Infect Gram-Negative Bacteria as Source of Signal-Arrest-Release Motif Lysins. Res. Microbiol. 2021 ,172, 103794. [CrossRef] 12. Briers, Y.; Peeters, L.M.; Volckaert, G.; Lavigne, R. The Lysis Cassette of Bacteriophage ΦKMV Encodes a Signal-Arrest-Release Endolysin and a Pinholin. Bacteriophage 2011 ,1, 25–30. [CrossRef] [PubMed] 13. Adamczyk-Popławska, M.; Tracz-Gaszewska, Z.; Lasota, P .; Kwiatek, A.; Piekarowicz, A. Haemophilus Inﬂuenzae HP1 Bacterio- phage Encodes a Lytic Cassette with a Pinholin and a Signal-Arrest-Release Endolysin. Int. J. Mol. Sci. 2020 ,21, 4013. [CrossRef] [PubMed] 14. Buttimer, C.; Born, Y.; Lucid, A.; Loessner, M.J.; Fieseler, L.; Coffey, A. Erwinia Amylovora Phage VB_EamM_Y3 Represents Another Lineage of Hairy Myoviridae. Res. Microbiol. 2018 ,169, 505–514. [CrossRef] [PubMed] 15. Buttimer, C.; Lucid, A.; Neve, H.; Franz, C.; O’Mahony, J.; Turner, D.; Lavigne, R.; Coffey, A. Pectobacterium Atrosepticum Phage VB_PatP_CB5: A Member of the Proposed Genus ‘Phimunavirus’. Viruses 2018 ,10, 394. [CrossRef] 16. Cahil, E.J.; Enderle, C.J.; Ahern, S.J.; Gill, J.J.; Torres, C.P .; Appel, D.N.; Black, M.C.; Young, R.; Gonzalez, C.F. Genomic and Biological Analysis of Phage Xfas53 and Related Prophages of Xylella fastidiosa .J. Bacteriol. 2010 ,192, 179–190. [CrossRef] 17. Savva, C.G.; Dewey, J.S.; Moussa, S.H.; To, K.H.; Holzenburg, A.; Young, R. Stable Micron-Scale Holes Are a General Feature of Canonical Holins: Canonical Holins Form Micron-Scale Holes. Mol. Microbiol. 2014 ,91, 57–65. [CrossRef] 18. Pang, T.; Savva, C.G.; Fleming, K.G.; Struck, D.K.; Young, R. Structure of the Lethal Phage Pinhole. Proc. Natl. Acad. Sci. USA 2009 ,106, 18966–18971. [CrossRef] 19. Park, T.; Struck, D.K.; Dankenbring, C.A.', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  [CrossRef] 17. Savva, C.G.; Dewey, J.S.; Moussa, S.H.; To, K.H.; Holzenburg, A.; Young, R. Stable Micron-Scale Holes Are a General Feature of Canonical Holins: Canonical Holins Form Micron-Scale Holes. Mol. Microbiol. 2014 ,91, 57–65. [CrossRef] 18. Pang, T.; Savva, C.G.; Fleming, K.G.; Struck, D.K.; Young, R. Structure of the Lethal Phage Pinhole. Proc. Natl. Acad. Sci. USA 2009 ,106, 18966–18971. [CrossRef] 19. Park, T.; Struck, D.K.; Dankenbring, C.A.; Young, R. The Pinholin of Lambdoid Phage 21: Control of Lysis by Membrane Depolarization. J. Bacteriol. 2007 ,189, 9135–9139. [CrossRef] 20. Xu, M.; Arulandu, A.; Struck, D.K.; Swanson, S.; Sacchettini, J.C.; Young, R. Disulﬁde Isomerization After Membrane Release of Its SAR Domain Activates P1 Lysozyme. Science 2005 ,307, 113–117. [CrossRef] 21. Sun, Q.; Kuty, G.F.; Arockiasamy, A.; Xu, M.; Young, R.; Sacchettini, J.C. Regulation of a Muralytic Enzyme by Dynamic Membrane Topology. Nat. Struct. Mol. Biol. 2009 ,16, 1192–1194. [CrossRef] 22. Young, R. Phage lysis: Three steps, three choices, one outcome. J. Microbiol. 2014 ,52, 243–258. [CrossRef] [PubMed] 23. Gründling, A.; Manson, M.D.; Young, R. Holins Kill without Warning. Proc. Natl. Acad. Sci. USA 2001 ,98, 9348–9352. [CrossRef] [PubMed] 24. Fernandes, S.; S ão-José, C. More than a Hole: The Holin Lethal Function May Be Required to Fully Sensitize Bacteria to the Lytic Action of Canonical Endolysins: Holin Function and Endolysin Activity. Mol. Microbiol. 2016 ,102, 92–106. [CrossRef]', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 19 of 20 25. White, R.; Chiba, S.; Pang, T.; Dewey, J.S.; Savva, C.G.; Holzenburg, A.; Pogliano, K.; Young, R. Holin Triggering in Real Time. Proc. Natl. Acad. Sci. USA 2011 ,108, 798–803. [CrossRef] 26. Young, R. Phage Lysis: Do We Have the Hole Story Yet? Curr. Opin. Microbiol. 2013 ,16, 790–797. [CrossRef] 27. Summer, E.J.; Berry, J.; Tran, T.A.T.; Niu, L.; Struck, D.K.; Young, R. Rz/Rz1 Lysis Gene Equivalents in Phages of Gram-Negative Hosts. J. Mol. Biol. 2007 ,373, 1098–1112. [CrossRef] [PubMed] 28. Berry, J.; Summer, E.J.; Struck, D.K.; Young, R. The Final Step in the Phage Infection Cycle: The Rz and Rz1 Lysis Proteins Link the Inner and Outer Membranes. Mol. Microbiol. 2008 ,70, 341–351. [CrossRef] 29. Berry, J.; Rajaure, M.; Pang, T.; Young, R. The Spanin Complex Is Essential for Lambda Lysis. J. Bacteriol. 2012 ,194, 5667–5674. [CrossRef] 30. Rajaure, M.; Berry, J.; Kongari, R.; Cahill, J.; Young, R. Membrane fusion during phage lysis. Proc. Natl. Acad. Sci. USA 2015 ,112, 5497–5502. [CrossRef] 31. Guo, T.; Xin, Y.; Zhang, C.; Kong, J. A Cytoplasmic Antiholin Is Embedded in Frame with the Holin in a Lactobacillus Fermentum Bacteriophage. Appl. Environ. Microbiol. 2018 ,84, e02518-17. [CrossRef] 32. Ziermann, R.; Bartlett, B.; Calendar, R.; Christie, G.E. Functions Involved in Bacteriophage P2-Induced Host Cell Lysis and Identiﬁcation of a New Tail Gene. J. Bacteriol. 1994 ,176, 4974–4984. [CrossRef] [PubMed] 33. Chang, C.Y.; Nam, K.; Young, R. S Gene Expression and the Timing of Lysis by Bacteriophage Lambda. J. Bacteriol. 1995 ,177, 3283–3294. [CrossRef] [PubMed] 34. Pontarollo, R.A.; Rioux, C.R.; Potter, A.A. Cloning and Characterization of Bacteriophage-like DNA from Haemophilus Somnus Homologous to Phages P2 and HP1. J. Bacteriol. 1997 ,179, 1872–1879. [CrossRef] [PubMed] 35. Takáˇ c, M.; Witte, A.; Bläsi, U. Functional Analysis of the Lysis Genes of Staphylococcus Aureus Phage P68 in Escherichia coli . Microbiology 2005 ,151, 2331–2342. [CrossRef] [PubMed] 36. Srividhya, K.V .; Krishnaswamy, S. Sub Classiﬁcation and Targeted Characterization of Prophage-Encoded Two-Component Cell Lysis Cassette. J. Biosci. 2007 ,32, 979–990. [CrossRef] [PubMed] 37. Kongari, R.; Rajaure, M.; Cahill, J.; Rasche, E.; Mijalis, E.; Berry, J.; Young, R. Phage spanins: Diversity, topological dynamics and gene co vergence. BMC Bioinform. 2018 ,19, 326. [CrossRef] 38.', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Microbiology 2005 ,151, 2331–2342. [CrossRef] [PubMed] 36. Srividhya, K.V .; Krishnaswamy, S. Sub Classiﬁcation and Targeted Characterization of Prophage-Encoded Two-Component Cell Lysis Cassette. J. Biosci. 2007 ,32, 979–990. [CrossRef] [PubMed] 37. Kongari, R.; Rajaure, M.; Cahill, J.; Rasche, E.; Mijalis, E.; Berry, J.; Young, R. Phage spanins: Diversity, topological dynamics and gene co vergence. BMC Bioinform. 2018 ,19, 326. [CrossRef] 38. Young, R. Bacteriophage Lysis: Mechanism and Regulation. Microbiol. Rev. 1992 ,56, 430–481. [CrossRef] 39. Yarmolinsky, M.B.; Sternberg, N. Bacteriophage P1. In The Bacteriophages ; Calendar, R., Ed.; (The Viruses); Springer: Boston, MA, USA, 1988; pp. 291–438. [CrossRef] 40. Łobocka, M.B.; Rose, D.J.; Plunkett, G.; Rusin, M.; Samojedny, A.; Lehnherr, H.; Yarmolinsky, M.B.; Blattner, F.R. Genome of Bacteriophage P1. J. Bacteriol. 2004 ,186, 7032–7068. [CrossRef] 41. Giermasi ´ nska, K.; Łobocka, M. Interakcje bakteriofaga P1 z kom órkami wybranych patogen ów ro´ slin z rodzaju Erwinia i rodzaj ów pokrewnych. In W: Wybrane Zagadnienia z Zakresu Chemii, Biologii i Fizyki ; Zdunek, B., Olsz ówka, M., Eds.; Wydawnictwo Naukowe Tygiel: Lublin, Poland, 2016; pp. 48–67. 42. Schmidt, C.; Velleman, M.; Arber, W. Three Functions of Bacteriophage P1 Involved in Cell Lysis. J. Bacteriol. 1996 ,178, 1099–1104. [CrossRef] 43. Walker, J.T.; Walker, D.H. Mutations in Coliphage P1 Affecting Host Cell Lysis. J. Virol. 1980 ,35, 519–530. [CrossRef] 44. Walker, J.T.; Walker, D.H. Coliphage P1 Morphogenesis: Analysis of Mutants by Electron Microscopy. J. Virol. 1983 ,45, 1118–1139. [CrossRef] [PubMed] 45. Guidolin, A.; Zingg, J.M.; Lehnherr, H.; Arber, W. Bacteriophage P1 tail-ﬁbre and dar operons are expressed from homologous phage-speciﬁc late promoter sequences. J. Mol. Biol. 1989 ,208, 615–622. [CrossRef] 46. Iida, S.; Hiestand-Nauer, R.; Sandmeier, H.; Lehnherr, H.; Arber, W. Accessory Genes in the darA Operon of Bacteriophage P1 Affect Antirestriction Function, Generalized Transduction, Head Morphogenesis, and Host Cell Lysis. Virology 1998 ,251, 49–58. [CrossRef] [PubMed] 47. Iida, S.; Streiff, M.B.; Bickle, T.A.; Arber, W. Two DNA Antirestriction Systems of Bacteriophage P1, DarA, and DarB: Characteri- zation of DarA−Phages. Virology 1987 ,157, 156–166. [CrossRef] 48. Piya, D.; Vara, L.; Russell, W.K.; Young, R.; Gill, J.J.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  lydA ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydA::195_196GATC as the recipient and pAKI23 plasmid as the DNA donor; cmR, kanR ∆lyz::RλlydC∆12TG lydD∆1-79 *lyz[R],lydC ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydC∆12_13TG as the recipient and plasmid pAKI23 as the DNA donor; cmR ∆lyz::RλlydA ::195GATC lydC∆12TG lydD∆1-79 *lyz[R],lydA ,lydC ,lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9 lydA::195_196GATC lydC ∆12_13TG as the recipient and plasmid pAKI23 as the DNA donor; cmR lydD∆16-81 * lydDRecombinational replacement with the use of P1 c1-100 mod749::IS 5 IS1::Tn9as the recipient and plasmid pAKI26 as the DNA donor; cmR *lydE gene and the Shine–Dalgarno sequence of lydE are intact in the P1 lydD deletion mutants. The plasmids used are derivatives of pMB1, pACYC184, or pSC101, and are listed in Table S1. Nucleotide sequences of plasmid inserts that were obtained by ampliﬁcation were veriﬁed by sequencing. Primers that were used for the PCR ampliﬁcation of the desired fragments or for sequencing are listed in Table S2. Oligonucleotide synthesis and sequenc- ing services were provided by the Laboratory of DNA Sequencing and Oligonucleotide Synthesis of IBB PAS.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 16 of 20 4.2. Bacterial Growth Conditions and Bacteriophage Propagation Bacteria were grown in Luria–Bertani liquid medium (LB; Difco) or on LB medium solidiﬁed with 1.5% agar at 30, 37, or 42◦C, as indicated. When required, the medium was supplemented with glucose or maltose to a ﬁnal concentration 0.2%, or with antibiotics: ampicillin (amp) (50, 100, or 500 µg/mL), chloramphenicol (cm) (12.5 or 25 µg/mL), or kanamycin (km) (12.5 or 25 µg/mL). Incubations of liquid cultures were performed with shaking (200 rpm) overnight or until the desired optical density (OD 600). Bacteriophages were propagated upon thermal induction of lysogens. Brieﬂy, cells lysogenized with λcI857or P1 c1-100IS1::Tn9 mod749::IS5 (or its mutant derivatives) were grown in LB or LB supplemented with chloramphenicol (12.5 µg/mL), respectively, overnight at 30◦C. The overnight cultures were diluted 50 ×in fresh LB with 0.2% maltose or 0.2% glucose and chloramphenicol, respectively, and grown at 30◦C until the optical density (OD 600) reached 0.3–0.4. Prophages were thermally induced by rapid heating of cultures to 42◦C in a shaking water bath heated to about 60◦C, then the cultures were returned to 42◦C and grown with shaking for about 45–60 min until signs of lysis were noticed. The crude lysates were harvested when the OD 600fell to 0.15 and centrifuged at 29,030 ×gfor 40 min at 4◦C. The remaining cells and cell debris were removed from the lysates by ﬁltration through 0.22 µm syringe ﬁlter (Filtropur, Sarstedt, Nümbrecht, Germany). The titer of phages in the lysates was assayed using the standard double-layer agar method, according to Adams (1959) [76]. 4.3. DNA Manipulation DNA ampliﬁcation, plasmid isolation, and agarose gel electrophoresis were carried out using standard techniques [ 77].E. coli cells were transformed with plasmids as described previously [ 77]. Restriction enzymes, T4 DNA ligase (Sigma-Aldrich, St. Louis, MO, USA), the Klenow fragment of E. coli DNA polymerase I (Thermo Fisher Scientiﬁc, Waltham, MA, USA), thermostable Ex Taq DNA polymerase (Takara, Kusatsu, Japan), and DreamTaq DNA Polymerase (Thermo Fisher Scientiﬁc, Waltham, MA, USA) were used according to the supplier’s protocols. 4.4. Lysogenization A portion of fresh overnight culture of bacteria (100 µL) was supplemented with 100µLof 20 mM CaCl 2and 100 µL of 20 mM MgSO 4mixed with 100 µL of phage lysate (10−8–10−9PFU/mL) and incubated at room temperature for 20 min to let the phages to adsorb the bacteria. Adsorption was stopped by the addition of 1 M sodium citrate (200 µL) and vigorous mixing, followed by the addition of 1 mL LB and incubation of the mixtures for 60 min at 30◦C to express the antibiotic-resistance phenotype. The phage–host mixture was then plated on solid LB medium supplemented with chloramphenicol and, when required, additional antibiotics, and incubated overnight at 30◦C to obtain colonies of lysogens. 4.5.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Adsorption was stopped by the addition of 1 M sodium citrate (200 µL) and vigorous mixing, followed by the addition of 1 mL LB and incubation of the mixtures for 60 min at 30◦C to express the antibiotic-resistance phenotype. The phage–host mixture was then plated on solid LB medium supplemented with chloramphenicol and, when required, additional antibiotics, and incubated overnight at 30◦C to obtain colonies of lysogens. 4.5. Construction of P1 Phage Mutants All P1 c1-100 mod749 ::IS5IS1::Tn9mutants were constructed by RecA-mediated re- combinational replacement in vivo using plasmids carrying relevant P1 genes with the desired mutations as DNA donors (see Table S1). Cells carrying the relevant plasmid were lysogenized with phage P1 c1-100 mod749 ::IS5IS1::Tn9, and the lysogens were selected on LB solid medium supplemented with chloramphenicol (12.5 µg/mL) and ampicillin (100 µg/mL), and, when required, kanamycin (25 µg/mL in the case of plasmids carrying P1 genes inactivated by the insertion of kanRcassette). They were puriﬁed through single colonies and used to thermally induce prophages, as described above. The obtained phages were used to lysogenize N99 cells. Lysogens expected to contain the insertions of the kanR cassette in the target prophage gene were selected on LB medium supplemented with chloramphenicol and kanamycin, and then screened for resistance to ampicillin by testing', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 17 of 20 their ability to grow on LB medium supplemented with chloramphenicol, kanamycin, and ampicillin (50 µg/mL). Lysogens sensitive to ampicillin were used for further studies. Lysogens expected to contain a mutation without an antibiotic resistance determinant in the target prophage gene were selected on LB medium supplemented with chlorampheni- col (12.5 µg/mL) and ampicillin (50 µg/mL). To increase the probability of recombination between the phage and the donor DNA in the plasmid, each lysogen was used to inoculate 5 mL of LB medium supplemented with chloramphenicol (12.5 µg/mL) and ampicillin (50µg/mL). The lysogen cultures were grown overnight with shaking at 30◦C and plated on LB solid medium supplemented with chloramphenicol (12.5 µg/mL) and ampicillin (500 µg/mL). Single colonies of lysogens able to grow on this medium were used for ther- mal induction of prophages. The obtained phages were used to lysogenize N99 cells, and the lysogens were selected on LB solid medium supplemented with chloramphenicol only (12.5 µg/mL). Lysogens were subsequently screened for the ability to grow on medium supplemented with chloramphenicol and ampicillin (50 µg/mL), to eliminate those that contained a donor plasmid either integrated with the prophage or in a circular form. Lyso- gens sensitive to ampicillin were tested for the presence of the desired mutation in the prophage by PCR ampliﬁcation of the relevant DNA fragment and restriction digestion or sequencing of the obtained amplicon. 4.6. Assays of Lysis Kinetics A fresh overnight culture of cells lysogenized with P1 c1-100 mod749 ::IS5IS1::Tn9or its mutant was used to inoculate LB medium supplemented with glucose and chloram- phenicol to an optical density (OD 600) of about 0.1. Cells were grown with shaking at 30◦C. When the OD 600reached 0.3–0.4, the prophage was induced by thermal induction, as described above, and further changes in the optical density of the culture were measured spectrophotometrically at indicated time intervals. 4.7. Time-Lapse Microscopy of Living Cells Morphological changes accompanying cell lysis upon the thermal induction of prophages in lysogens were monitored by the time-lapse microscopy of living cells. To immobilize cells in one focal plane, samples (10 µL) withdrawn from the cultures of induced lysogens before the expected lysis time were placed on a microscopic slide on the upper surface of an agarose microslab (0.7% agarose in physiological saline) and covered with a coverslip. The microslabs were prepared as described previously [ 78]. Cells were immediately im- aged using a Zeiss Axio M2 microscope with an oil immersion objective (100 ×, numerical aperture = 1.4). A time-lapse video was captured at ﬁve frames per second. The video was edited and scaled using the AxioVision 4 software package (Zeiss, Oberkochen, Germany). All images were stored as .zvi ﬁles and converted to TIFF or JPEG format as necessary. Supplementary Materials: The following are available online at https://www.mdpi.com/article/10 .3390/ijms23084231/s1. Author Contributions: Conceptualization, A.B. and M.Ł.; methodology, A.B., A.C., W.I. and M.Ł.; validation, A.B. and M.Ł.; formal analysis, A.B. and M.Ł.; investigation, A.B., A.C., W.I. and J.B.; resources, A.B. and M.Ł.; data curation, M.Ł.; writing—original draft preparation, A.B. and M.Ł.; writing—review and editing, M.Ł.; visualization, A.B.; supervision, M.Ł.; project administration, A.B. and M.Ł.; funding acquisition, A.B. and M.Ł. All authors have read and agreed to the published version of the manuscript.', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Author Contributions: Conceptualization, A.B. and M.Ł.; methodology, A.B., A.C., W.I. and M.Ł.; validation, A.B. and M.Ł.; formal analysis, A.B. and M.Ł.; investigation, A.B., A.C., W.I. and J.B.; resources, A.B. and M.Ł.; data curation, M.Ł.; writing—original draft preparation, A.B. and M.Ł.; writing—review and editing, M.Ł.; visualization, A.B.; supervision, M.Ł.; project administration, A.B. and M.Ł.; funding acquisition, A.B. and M.Ł. All authors have read and agreed to the published version of the manuscript. Funding: This research was funded by the National Science Centre (PRELUDIUM; Grant No. UMO- 2019/35/N/NZ6/03789) and by statutory funds from the Institute of Biochemistry and Biophysics of the Polish Academy of Sciences. Institutional Review Board Statement: Not applicable. Informed Consent Statement: Not applicable. Data Availability Statement: Not applicable.', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 18 of 20 Acknowledgments: We thank Agnieszka Laudy from the Warsaw Medical University for lending us the E. cloacae subsp. cloacae strain DSM6234, which was used in a part of this study, and Anna B˛ ebenek of the Institute of Biochemistry and Biophysics, PAS, for providing us with the lysogen of phage λcI857. Conﬂicts of Interest: The authors declare no conﬂict of interest. The funders had no role in the design of the study; in the collection, analyses, or interpretation of data; in the writing of the manuscript; or in the decision to publish the results. References 1. Wang, I.N.; Smith, D.L.; Young, R. Holins: The Protein Clocks of Bacteriophage Infections. Annu. Rev. Microbiol. 2000 ,54, 799–825. [CrossRef] [PubMed] 2. Wang, I.-N.; Deaton, J.; Young, R. Sizing the Holin Lesion with an Endolysin- β-Galactosidase Fusion. J. Bacteriol. 2003 ,185, 779–787. [CrossRef] [PubMed] 3. Oliveira, H.; Melo, L.D.R.; Santos, S.B.; Nobrega, F.L.; Ferreira, E.C.; Cerca, N.; Azeredo, J.; Kluskens, L.D. Molecular Aspects and Comparative Genomics of Bacteriophage Endolysins. J. Virol. 2013 ,87, 4558–4570. [CrossRef] [PubMed] 4. Dewey, J.S.; Savva, C.G.; White, R.L.; Vitha, S.; Holzenburg, A.; Young, R. Micron-Scale Holes Terminate the Phage Infection Cycle. Proc. Natl. Acad. Sci. USA 2010 ,107, 2219–2223. [CrossRef] 5. Xu, M.; Struck, D.K.; Deaton, J.; Wang, I.-N.; Young, R. A Signal-Arrest-Release Sequence Mediates Export and Control of the Phage P1 Endolysin. Proc. Natl. Acad. Sci. USA 2004 ,101, 6415–6420. [CrossRef] 6. Kuty, G.F.; Xu, M.; Struck, D.K.; Summer, E.J.; Young, R. Regulation of a Phage Endolysin by Disulﬁde Caging. J. Bacteriol. 2010 , 192, 5682–5687. [CrossRef] 7. São-José, C.; Parreira, R.; Vieira, G.; Santos, M.A. The N-Terminal Region of the Oenococcus Oeni Bacteriophage FOg44 Lysin Behaves as a Bona Fide Signal Peptide in Escherichia Coli and as a Cis-Inhibitory Element, Preventing Lytic Activity on Oenococcal Cells. J. Bacteriol. 2000 ,182, 5823–5831. [CrossRef] 8. Zampara, A.; Ahern, S.J.; Briers, Y.; Brøndsted, L.; Sørensen, M.C.H. Two Distinct Modes of Lysis Regulation in Campylobacter Fletchervirus and Firehammervirus Phages. Viruses 2020 ,12, 1247. [CrossRef] 9. Cahill, J.; Young, R. Phage Lysis: Multiple Genes for Multiple Barriers. In Advances in Virus Research ; Elsevier: Amsterdam, The Netherlands, 2019; Volume 103, pp. 33–70. [CrossRef] 10. Fernandes, S.; S ão-José, C. Enzymes and Mechanisms Employed by Tailed Bacteriophages to Breach the Bacterial Cell Barriers. Viruses 2018 ,10, 396.', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  ; Briers, Y.; Brøndsted, L.; Sørensen, M.C.H. Two Distinct Modes of Lysis Regulation in Campylobacter Fletchervirus and Firehammervirus Phages. Viruses 2020 ,12, 1247. [CrossRef] 9. Cahill, J.; Young, R. Phage Lysis: Multiple Genes for Multiple Barriers. In Advances in Virus Research ; Elsevier: Amsterdam, The Netherlands, 2019; Volume 103, pp. 33–70. [CrossRef] 10. Fernandes, S.; S ão-José, C. Enzymes and Mechanisms Employed by Tailed Bacteriophages to Breach the Bacterial Cell Barriers. Viruses 2018 ,10, 396. [CrossRef] 11. Gontijo, M.T.P .; Vidigal, P .M.P .; Lopez, M.E.S.; Brocchi, M. Bacteriophages That Infect Gram-Negative Bacteria as Source of Signal-Arrest-Release Motif Lysins. Res. Microbiol. 2021 ,172, 103794. [CrossRef] 12. Briers, Y.; Peeters, L.M.; Volckaert, G.; Lavigne, R. The Lysis Cassette of Bacteriophage ΦKMV Encodes a Signal-Arrest-Release Endolysin and a Pinholin. Bacteriophage 2011 ,1, 25–30. [CrossRef] [PubMed] 13. Adamczyk-Popławska, M.; Tracz-Gaszewska, Z.; Lasota, P .; Kwiatek, A.; Piekarowicz, A. Haemophilus Inﬂuenzae HP1 Bacterio- phage Encodes a Lytic Cassette with a Pinholin and a Signal-Arrest-Release Endolysin. Int. J. Mol. Sci. 2020 ,21, 4013. [CrossRef] [PubMed] 14. Buttimer, C.; Born, Y.; Lucid, A.; Loessner, M.J.; Fieseler, L.; Coffey, A. Erwinia Amylovora Phage VB_EamM_Y3 Represents Another Lineage of Hairy Myoviridae. Res. Microbiol. 2018 ,169, 505–514. [CrossRef] [PubMed] 15. Buttimer, C.; Lucid, A.; Neve, H.; Franz, C.; O’Mahony, J.; Turner, D.; Lavigne, R.; Coffey, A. Pectobacterium Atrosepticum Phage VB_PatP_CB5: A Member of the Proposed Genus ‘Phimunavirus’. Viruses 2018 ,10, 394. [CrossRef] 16. Cahil, E.J.; Enderle, C.J.; Ahern, S.J.; Gill, J.J.; Torres, C.P .; Appel, D.N.; Black, M.C.; Young, R.; Gonzalez, C.F. Genomic and Biological Analysis of Phage Xfas53 and Related Prophages of Xylella fastidiosa .J. Bacteriol. 2010 ,192, 179–190. [CrossRef] 17. Savva, C.G.; Dewey, J.S.; Moussa, S.H.; To, K.H.; Holzenburg, A.; Young, R. Stable Micron-Scale Holes Are a General Feature of Canonical Holins: Canonical Holins Form Micron-Scale Holes. Mol. Microbiol. 2014 ,91, 57–65. [CrossRef] 18. Pang, T.; Savva, C.G.; Fleming, K.G.; Struck, D.K.; Young, R. Structure of the Lethal Phage Pinhole. Proc. Natl. Acad. Sci. USA 2009 ,106, 18966–18971. [CrossRef] 19. Park, T.; Struck, D.K.; Dankenbring, C.A.', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  [CrossRef] 17. Savva, C.G.; Dewey, J.S.; Moussa, S.H.; To, K.H.; Holzenburg, A.; Young, R. Stable Micron-Scale Holes Are a General Feature of Canonical Holins: Canonical Holins Form Micron-Scale Holes. Mol. Microbiol. 2014 ,91, 57–65. [CrossRef] 18. Pang, T.; Savva, C.G.; Fleming, K.G.; Struck, D.K.; Young, R. Structure of the Lethal Phage Pinhole. Proc. Natl. Acad. Sci. USA 2009 ,106, 18966–18971. [CrossRef] 19. Park, T.; Struck, D.K.; Dankenbring, C.A.; Young, R. The Pinholin of Lambdoid Phage 21: Control of Lysis by Membrane Depolarization. J. Bacteriol. 2007 ,189, 9135–9139. [CrossRef] 20. Xu, M.; Arulandu, A.; Struck, D.K.; Swanson, S.; Sacchettini, J.C.; Young, R. Disulﬁde Isomerization After Membrane Release of Its SAR Domain Activates P1 Lysozyme. Science 2005 ,307, 113–117. [CrossRef] 21. Sun, Q.; Kuty, G.F.; Arockiasamy, A.; Xu, M.; Young, R.; Sacchettini, J.C. Regulation of a Muralytic Enzyme by Dynamic Membrane Topology. Nat. Struct. Mol. Biol. 2009 ,16, 1192–1194. [CrossRef] 22. Young, R. Phage lysis: Three steps, three choices, one outcome. J. Microbiol. 2014 ,52, 243–258. [CrossRef] [PubMed] 23. Gründling, A.; Manson, M.D.; Young, R. Holins Kill without Warning. Proc. Natl. Acad. Sci. USA 2001 ,98, 9348–9352. [CrossRef] [PubMed] 24. Fernandes, S.; S ão-José, C. More than a Hole: The Holin Lethal Function May Be Required to Fully Sensitize Bacteria to the Lytic Action of Canonical Endolysins: Holin Function and Endolysin Activity. Mol. Microbiol. 2016 ,102, 92–106. [CrossRef]', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 19 of 20 25. White, R.; Chiba, S.; Pang, T.; Dewey, J.S.; Savva, C.G.; Holzenburg, A.; Pogliano, K.; Young, R. Holin Triggering in Real Time. Proc. Natl. Acad. Sci. USA 2011 ,108, 798–803. [CrossRef] 26. Young, R. Phage Lysis: Do We Have the Hole Story Yet? Curr. Opin. Microbiol. 2013 ,16, 790–797. [CrossRef] 27. Summer, E.J.; Berry, J.; Tran, T.A.T.; Niu, L.; Struck, D.K.; Young, R. Rz/Rz1 Lysis Gene Equivalents in Phages of Gram-Negative Hosts. J. Mol. Biol. 2007 ,373, 1098–1112. [CrossRef] [PubMed] 28. Berry, J.; Summer, E.J.; Struck, D.K.; Young, R. The Final Step in the Phage Infection Cycle: The Rz and Rz1 Lysis Proteins Link the Inner and Outer Membranes. Mol. Microbiol. 2008 ,70, 341–351. [CrossRef] 29. Berry, J.; Rajaure, M.; Pang, T.; Young, R. The Spanin Complex Is Essential for Lambda Lysis. J. Bacteriol. 2012 ,194, 5667–5674. [CrossRef] 30. Rajaure, M.; Berry, J.; Kongari, R.; Cahill, J.; Young, R. Membrane fusion during phage lysis. Proc. Natl. Acad. Sci. USA 2015 ,112, 5497–5502. [CrossRef] 31. Guo, T.; Xin, Y.; Zhang, C.; Kong, J. A Cytoplasmic Antiholin Is Embedded in Frame with the Holin in a Lactobacillus Fermentum Bacteriophage. Appl. Environ. Microbiol. 2018 ,84, e02518-17. [CrossRef] 32. Ziermann, R.; Bartlett, B.; Calendar, R.; Christie, G.E. Functions Involved in Bacteriophage P2-Induced Host Cell Lysis and Identiﬁcation of a New Tail Gene. J. Bacteriol. 1994 ,176, 4974–4984. [CrossRef] [PubMed] 33. Chang, C.Y.; Nam, K.; Young, R. S Gene Expression and the Timing of Lysis by Bacteriophage Lambda. J. Bacteriol. 1995 ,177, 3283–3294. [CrossRef] [PubMed] 34. Pontarollo, R.A.; Rioux, C.R.; Potter, A.A. Cloning and Characterization of Bacteriophage-like DNA from Haemophilus Somnus Homologous to Phages P2 and HP1. J. Bacteriol. 1997 ,179, 1872–1879. [CrossRef] [PubMed] 35. Takáˇ c, M.; Witte, A.; Bläsi, U. Functional Analysis of the Lysis Genes of Staphylococcus Aureus Phage P68 in Escherichia coli . Microbiology 2005 ,151, 2331–2342. [CrossRef] [PubMed] 36. Srividhya, K.V .; Krishnaswamy, S. Sub Classiﬁcation and Targeted Characterization of Prophage-Encoded Two-Component Cell Lysis Cassette. J. Biosci. 2007 ,32, 979–990. [CrossRef] [PubMed] 37. Kongari, R.; Rajaure, M.; Cahill, J.; Rasche, E.; Mijalis, E.; Berry, J.; Young, R. Phage spanins: Diversity, topological dynamics and gene co vergence. BMC Bioinform. 2018 ,19, 326. [CrossRef] 38.', 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Microbiology 2005 ,151, 2331–2342. [CrossRef] [PubMed] 36. Srividhya, K.V .; Krishnaswamy, S. Sub Classiﬁcation and Targeted Characterization of Prophage-Encoded Two-Component Cell Lysis Cassette. J. Biosci. 2007 ,32, 979–990. [CrossRef] [PubMed] 37. Kongari, R.; Rajaure, M.; Cahill, J.; Rasche, E.; Mijalis, E.; Berry, J.; Young, R. Phage spanins: Diversity, topological dynamics and gene co vergence. BMC Bioinform. 2018 ,19, 326. [CrossRef] 38. Young, R. Bacteriophage Lysis: Mechanism and Regulation. Microbiol. Rev. 1992 ,56, 430–481. [CrossRef] 39. Yarmolinsky, M.B.; Sternberg, N. Bacteriophage P1. In The Bacteriophages ; Calendar, R., Ed.; (The Viruses); Springer: Boston, MA, USA, 1988; pp. 291–438. [CrossRef] 40. Łobocka, M.B.; Rose, D.J.; Plunkett, G.; Rusin, M.; Samojedny, A.; Lehnherr, H.; Yarmolinsky, M.B.; Blattner, F.R. Genome of Bacteriophage P1. J. Bacteriol. 2004 ,186, 7032–7068. [CrossRef] 41. Giermasi ´ nska, K.; Łobocka, M. Interakcje bakteriofaga P1 z kom órkami wybranych patogen ów ro´ slin z rodzaju Erwinia i rodzaj ów pokrewnych. In W: Wybrane Zagadnienia z Zakresu Chemii, Biologii i Fizyki ; Zdunek, B., Olsz ówka, M., Eds.; Wydawnictwo Naukowe Tygiel: Lublin, Poland, 2016; pp. 48–67. 42. Schmidt, C.; Velleman, M.; Arber, W. Three Functions of Bacteriophage P1 Involved in Cell Lysis. J. Bacteriol. 1996 ,178, 1099–1104. [CrossRef] 43. Walker, J.T.; Walker, D.H. Mutations in Coliphage P1 Affecting Host Cell Lysis. J. Virol. 1980 ,35, 519–530. [CrossRef] 44. Walker, J.T.; Walker, D.H. Coliphage P1 Morphogenesis: Analysis of Mutants by Electron Microscopy. J. Virol. 1983 ,45, 1118–1139. [CrossRef] [PubMed] 45. Guidolin, A.; Zingg, J.M.; Lehnherr, H.; Arber, W. Bacteriophage P1 tail-ﬁbre and dar operons are expressed from homologous phage-speciﬁc late promoter sequences. J. Mol. Biol. 1989 ,208, 615–622. [CrossRef] 46. Iida, S.; Hiestand-Nauer, R.; Sandmeier, H.; Lehnherr, H.; Arber, W. Accessory Genes in the darA Operon of Bacteriophage P1 Affect Antirestriction Function, Generalized Transduction, Head Morphogenesis, and Host Cell Lysis. Virology 1998 ,251, 49–58. [CrossRef] [PubMed] 47. Iida, S.; Streiff, M.B.; Bickle, T.A.; Arber, W. Two DNA Antirestriction Systems of Bacteriophage P1, DarA, and DarB: Characteri- zation of DarA−Phages. Virology 1987 ,157, 156–166. [CrossRef] 48. Piya, D.; Vara, L.; Russell, W.K.; Young, R.; Gill, J.J.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:01 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'74'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994334'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'339ms'), (b'x-request-id', b'52d0584d221e940f5dd64c8573a23529'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249a0cd305d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:01 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'74'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994334'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'339ms'), (b'x-request-id', b'52d0584d221e940f5dd64c8573a23529'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249a0cd305d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Virology 1998 ,251, 49–58. [CrossRef] [PubMed] 47. Iida, S.; Streiff, M.B.; Bickle, T.A.; Arber, W. Two DNA Antirestriction Systems of Bacteriophage P1, DarA, and DarB: Characteri- zation of DarA−Phages. Virology 1987 ,157, 156–166. [CrossRef] 48. Piya, D.; Vara, L.; Russell, W.K.; Young, R.; Gill, J.J. The Multicomponent Antirestriction System of Phage P1 Is Linked to Capsid Morphogenesis: P1 Antirestriction Is Linked to Morphogenesis. Mol. Microbiol. 2017 ,105, 399–412. [CrossRef] 49. Fernandes, S.; S ão-José, C. Probing the Function of the Two Holin-like Proteins of Bacteriophage SPP1. Virology 2017 ,500, 184–189. [CrossRef] 50. Halgasova, N.; Ugorcakova, J.; Gerova, M.; Timko, J.; Bukovska, G. Isolation and Characterization of Bacteriophage ΦBP from Paenibacillus Polymyxa CCM 7400. FEMS Microbiol. Lett. 2010 ,305, 128–135. [CrossRef] 51. Lu, Z.; Altermann, E.; Breidt, F.; Kozyavkin, S. Sequence Analysis of Leuconostoc Mesenteroides Bacteriophage Φ1-A4 Isolated from an Industrial Vegetable Fermentation. Appl. Environ. Microbiol. 2010 ,76, 1955–1966. [CrossRef] 52. Nakonieczna, A.; Rutyna, P .; Fedorowicz, M.; Kwiatek, M.; Mizak, L.; Łobocka, M. Three Novel Bacteriophages, J5a, F16Ba, and z1a, Speciﬁc for Bacillus anthracis , Deﬁne a New Clade of Historical Wbeta Phage Relatives. Viruses 2022 ,14, 213. [CrossRef] 53. Catal ão, M.J.; Gil, F.; Moniz-Pereira, J.; S ão-José, C.; Pimentel, M. Diversity in Bacterial Lysis Systems: Bacteriophages Show the Way. FEMS Microbiol. Rev. 2013 ,37, 554–571. [CrossRef] 54. Delisle, A.L.; Barcak, G.J.; Guo, M. Isolation and Expression of the Lysis Genes of Actinomyces naeslundii Phage Av-1. Appl. Environ. Microbiol. 2006 ,72, 1110–1117. [CrossRef] [PubMed]', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 20 of 20 55. Wo´ znica, W.M.; Bigos, J.; Łobocka, M.B. Lysis of bacterial cells in the process of bacteriophage release-canonical and newly discovered mechanisms. Postepy Hig. Med. Dosw. 2015 ,69, 114–126. 56. Park, T.; Struck, D.K.; Deaton, J.F.; Young, R. Topological dynamics of holins in programmed bacterial lysis. Proc. Natl. Acad. Sci. USA 2006 ,103, 19713–19718. [CrossRef] [PubMed] 57. Pang, T.; Park, T.; Young, R. Mapping the pinhole formation pathway of S21. Mol. Microbiol. 2010 ,78, 710–719. [CrossRef] [PubMed] 58. Zimmermann, L.; Stephens, A.; Nam, S.Z.; Rau, D.; Kübler, J.; Lozajic, M.; Gabler, F.; Söding, J.; Lupas, A.N.; Alva, V . A Completely Reimplemented MPI Bioinformatics Toolkit with a New HHpred Server at its Core. J. Mol. Biol. 2018 ,430, 2237–2243. [CrossRef] [PubMed] 59. Gabler, F.; Nam, S.Z.; Till, S.; Mirdita, M.; Steinegger, M.; Söding, J.; Lupas, A.N.; Alva, V . Protein Sequence Analysis Using the MPI Bioinformatics Toolkit. Curr. Protoc. Bioinform. 2020 ,72, e108. [CrossRef] 60. Pang, T.; Park, T.; Young, R. Mutational analysis of the S21 pinholin. Mol. Microbiol. 2010 ,76, 68–77. [CrossRef] 61. Wang, I.N. Lysis timing and bacteriophage ﬁtness. Genetics 2010 ,172, 17–26. [CrossRef] 62. Gallet, R.; Kannoly, S.; Wang, I.N. Effects of bacteriophage traits on plaque formation. BMC Microbiol. 2011 ,11, 181. [CrossRef] 63. Cornaglia, G.; Mazzariol, A.; Fontana, R.; Satta, G. Diffusion of Carbapenems Through the Outer Membrane of Enterobacteriaceae and Correlation of Their Activities with Their Periplasmic Concentrations. Microb. Drug Resist. 1996 ,2, 273–276. [CrossRef] 64. Sohlenkamp, C.; Geiger, O. Bacterial Membrane Lipids: Diversity in Structures and Pathways. FEMS Microbiol. Rev. 2016 ,40, 133–159. [CrossRef] [PubMed] 65. Acosta-Guti érrez, S.; Ferrara, L.; Pathania, M.; Masi, M.; Wang, J.; Bodrenko, I.; Zahn, M.; Winterhalter, M.; Stavenger, R.A.; Pagès, J.-M.; et al. Getting Drugs into Gram-Negative Bacteria: Rational Rules for Permeation through General Porins. ACS Infect. Dis. 2018 ,4, 1487–1498. [CrossRef] [PubMed] 66. Pomares, M.F.; Delgado, M.A.; Corbal án, N.S.; Far ías, R.N.; Vincent, P .A. Sensitization of Microcin J25-Resistant Strains by a Membrane-Permeabilizing Peptide. Appl. Environ. Microbiol. 2010 ,76, 6837–6842. [CrossRef] 67. Neu, H.C.; Chou, J. Release of Surface Enzymes in Enterobacteriaceae by Osmotic Shock. J. Bacteriol. 1967 ,94, 1934–1945. [CrossRef] [PubMed] 68.', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  ACS Infect. Dis. 2018 ,4, 1487–1498. [CrossRef] [PubMed] 66. Pomares, M.F.; Delgado, M.A.; Corbal án, N.S.; Far ías, R.N.; Vincent, P .A. Sensitization of Microcin J25-Resistant Strains by a Membrane-Permeabilizing Peptide. Appl. Environ. Microbiol. 2010 ,76, 6837–6842. [CrossRef] 67. Neu, H.C.; Chou, J. Release of Surface Enzymes in Enterobacteriaceae by Osmotic Shock. J. Bacteriol. 1967 ,94, 1934–1945. [CrossRef] [PubMed] 68. Nickerson, K.W.; Aspedon, A. Detergent-Shock Response in Enteric Bacteria. Mol. Microbiol. 1992 ,6, 957–961. [CrossRef] 69. Meyer, J.; Stålhammar-Carlemalm, M.; Streiff, M.; Iida, S.; Arber, W. Sequence Relations among the IncY Plasmid P15B, P1, and P7 Prophages. Plasmid 1986 ,16, 81–89. [CrossRef] 70. Yang, L.; Li, W.; Jiang, G.-Z.; Zhang, W.-H.; Ding, H.-Z.; Liu, Y.-H.; Zeng, Z.-L.; Jiang, H.-X. Characterization of a P1-like Bacteriophage Carrying CTX-M-27 in Salmonella Spp. Resistant to Third Generation Cephalosporins Isolated from Pork in China. Sci. Rep. 2017 ,7, 40710. [CrossRef] 71. Billard-Pomares, T.; Fouteau, S.; Jacquet, M.E.; Roche, D.; Barbe, V .; Castellanos, M.; Bouet, J.Y.; Cruveiller, S.; M édigue, C.; Blanco, J.; et al. Characterization of a P1-Like Bacteriophage Carrying an SHV-2 Extended-Spectrum β-Lactamase from an Escherichia Coli Strain. Antimicrob. Agents Chemother. 2014 ,58, 6550–6557. [CrossRef] 72. Hagbø, M.; Ravi, A.; Angell, I.L.; Sunde, M.; Ludvigsen, J.; Diep, D.B.; Foley , S.L.; V ento, M.; Collado, M.C.; Perez-Martinez, G.; et al. Experimental Support for Multidrug Resistance Transfer Potential in the Preterm Infant Gut Microbiota. Pediatr. Res. 2020 ,88, 57–65. [CrossRef] 73. Gottesman, M.E.; Yarmolinsky, M.B. Integration-Negative Mutants of Bacteriophage Lambda. J. Mol. Biol. 1968 ,31, 487–505. [CrossRef] 74. Łobocka, M.; Yarmolinsky, M. P1 Plasmid Partition: A Mutational Analysis of ParB. J. Mol. Biol. 1996 ,259, 366–382. [CrossRef] [PubMed] 75. Sambrook, J.; Fritsch, E.F.; Maniatis, T. Molecular Cloning: A Laboratory Manual , 2nd. ed.; Cold Spring Harbor Laboratory Press: Cold Spring Harbor, NY, USA, 1989. 76. Adams, M.H. Bacteriophages ; Interscience Publishers Inc.: New York, NY, USA, 1959. 77. Sambrook, J.; Russell, D.W. Molecular Cloning: A Laboratory Manual , 3rd ed.; Cold Spring Harbor Laboratory Press: Cold Spring Harbor, NY, USA, 2001; Volume 1. 78. Helvoort, J.M.L.M.; Woldringh, C.L.', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  J. Mol. Biol. 1996 ,259, 366–382. [CrossRef] [PubMed] 75. Sambrook, J.; Fritsch, E.F.; Maniatis, T. Molecular Cloning: A Laboratory Manual , 2nd. ed.; Cold Spring Harbor Laboratory Press: Cold Spring Harbor, NY, USA, 1989. 76. Adams, M.H. Bacteriophages ; Interscience Publishers Inc.: New York, NY, USA, 1959. 77. Sambrook, J.; Russell, D.W. Molecular Cloning: A Laboratory Manual , 3rd ed.; Cold Spring Harbor Laboratory Press: Cold Spring Harbor, NY, USA, 2001; Volume 1. 78. Helvoort, J.M.L.M.; Woldringh, C.L. Nucleoid Partitioning in Escherichia coli during Steady-State Growth and upon Recovery from Chloramphenicol Treatment. Mol. Microbiol. 1994 ,13, 577–583. [CrossRef] [PubMed]', \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Vol.174,No.12 DNAInversion Regions MinofPlasmid pl5BandCin ofBacteriophage P1:Evolution ofBacteriophage TailFiberGenes HEINRICH SANDMEIER,lt SHIGERU IIDA,1'2 ANDWERNER ARBERl* Abteilung Mikrobiologie, Biozentrum derUniversitat Basel,Klingelbergstrasse 70,CH-4056 Basel, Switzerland, 1andDepartment ofBiological Sciences andTechnology, Science University ofTokyo, Yamazaki, Chiba278,Japan2 Received 21November 1991/Accepted 11April1992 Plasmid pl5Bandthegenome ofbacteriophage P1arecloselyrelated,buttheirsite-specific DNAinversion systems, MinandCin,respectively, donothavestrictstructural homology. Rather, thecomplex Minsystem represents asubstitution ofaCin-like systemintoanancestral pl5Bgenome. Thesubstituting sequences of boththeminrecombinase geneandthemultiple invertible DNAsegments ofpl5B are,respectively, homologous tothepinrecombinase geneandtopartoftheinvertible DNAofthePinsystem onthedefective viralelement e14ofEscherichia coliK-12.Tomapthesitesofthissubstitution, theDNAsequence ofasegment adjacent totheinvertible segment intheP1genome wasdetermined. This,together withalready available sequence data,indicated thatbothP1andpl5Bhadsuffered various sequence acquisitions ordeletions and sequence amplifications givingrisetomosaics ofpartially relatedrepeated elements. Databasesearches revealed segments ofhomology intheDNAinversion regionsofpl5B,e14,andP1andintailfibergenesof phagesMu,T4,P2,andK.Thisresultsuggests thattheevolution ofphagetailfibergenesinvolves horizontal genetransfer andthattheMinandPinregionsencodetailfibergenes.Afunctional testprovedthatthepl5B Minregioncarries atailfiberoperonandsuggests thatthealternative expression ofsixdifferent genevariants byMininversion offersextensive hostrangevariation. Plasmid p15BisaP1prophage-related resident ofEsche- richiacoli15T-(18).Electron microscopy heteroduplex studiesrevealed thatapproximately 82%ofthe94-kbpl5B DNAhybridizes withtheP1genome (23).Since nophage particle withaP1-likemorphology couldbeproduced inE. coli15T-(1,18),pl5BDNAhasbeenconsidered tobea defective prophage. Oneregionofnonhomology between pl5BandP1DNA coverstheinvertible segment oftheP1 DNA(16,23).Itscorresponding region onpl5Bcarries a complex site-specific DNAinversion system calledMin(17, 34).TheMinregioncontains sixcrossover sitescalled mix. Successive DNAinversions between pairsofmixsites mediated byMinrecombinase canproduce 240isomeric formsofa3.5-kbregion(34).Theinvertible DNAsegments contain sixdifferent 3'geneparts,eachofwhich canbecome fusedinframetoanadjacent constant 5'genesegment upon DNAinversion (33)(seeFig.4). Thecomplex Minsystemiscloselyrelated tosimpleDNA inversion systems whichcontain twocrossover sitesand constitute atwo-state switch forgeneexpression: Cinof phageP1,GinofphageMu,Pinofthee14element onE.coli K-12,andHinofSalmonella typhimurium Thecrossover sitesandrecombinase genesofthesesystems arestructur- allyhomologous andcanmutually substitute fortheirfunc- tions(forareviewoftheseDNAinversion systems, called heretheDinfamily, seereference 7;seealsoFig.4).Inthe Hinsystem, twoflagellar geneswithdistinctantigenic spec- ificities arealternatively expressed byinversion ofaDNA *Corresponding author.\", \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Thecomplex Minsystemiscloselyrelated tosimpleDNA inversion systems whichcontain twocrossover sitesand constitute atwo-state switch forgeneexpression: Cinof phageP1,GinofphageMu,Pinofthee14element onE.coli K-12,andHinofSalmonella typhimurium Thecrossover sitesandrecombinase genesofthesesystems arestructur- allyhomologous andcanmutually substitute fortheirfunc- tions(forareviewoftheseDNAinversion systems, called heretheDinfamily, seereference 7;seealsoFig.4).Inthe Hinsystem, twoflagellar geneswithdistinctantigenic spec- ificities arealternatively expressed byinversion ofaDNA *Corresponding author. tPresent address: Abteilung furPraventivzahnmedizin undorale Mikrobiologie, Zahnarztliches InstitutderUniversitat Basel,Peter- splatz14,CH-4051 Basel,Switzerland.segment including apromoter. InallotherDinsystems described sofar,astationary promoter transcribes agene whichisassembled byDNAinversion fromconstant and variable genesegments. SuchgenefusionsbytheCinand Gininversions inphages P1andMuleadtoalternate expression oftwosetsoftailfibergeneswhichconfer different receptor specificities. TheseP1andMuSgenes consistofa5'constant portion, calledSc,andtwoversions ofthe3'portion, calledSvandSv'.Thee14element, which canbeconsidered adefective prophage, isanexcisable element ofsomeE.coliK-12chromosomes (la,21).The biological roleofPininversion one14,aswellasthoseof recently identified related DNAinversion systems onShi- gellachromosomes (39),isnotyetknown. Thecomparison oftheDNA sequences ofthepl5BMin regionwithpreviously determined sequences oftheCin regionofphageP1suggested thattheinvertible segment and partoftherecombinase genehadbeensubstituted by nonhomologous DNA onanancestral pl5B genome (33). While oneofthebreakpoints fromhomologous tononho- mologous sequences hadalready beenlocated withinthe recombinase genes,thesecondbreakpoint remained unde- termined because ofincomplete P1sequence dataofthis region. Inthispaper,theDNA sequence oftheconstant part(Sc) oftheP1geneSwascompleted toelucidate thesecond breakpoint. Theresultindicates thatcomplicated recombi- nations musthavetakenplaceatthissecondbreakpoint on anancestral genome. Inaddition, acomputer-aided compar- isonofthese sequences withavailable DNAandprotein sequences ofotherbacteriophages identified various se- quence domains thatareshared amongthegenesoftheP1 andpl5BDNAinversion regions andtailfibergenesof phages Mu,P2,X,andT4,aswellastheelement e14. 3936JOURNAL OFBACTERIOLOGY, June1992,p.3936-3944 0021-9193/92/123936-09$02.00/0 Copyright ©1992,American SocietyforMicrobiology\", \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  VOL.174,1992 ~~~~EVOLUTION OFpl5B3ANDPHAGE P1TAILFIBERGENES 3937 AccI  eal~~~~~~~~~~~~~~~~~~~~~~~~~~~~~0YTAQDASTTQKGLVQLSSATNSDSETMAATPKA *BglII alN,.,e..bl . .  * ~~~~~StyI bl%,,.~.b2. INPYTTINNNLA PLASPA LTGVPTAP200 300 400 500 TAAQGTNNTQ IATTAY VRAAI SALVGSSPEALD T b%..c SfaNI CCCTGTATGAGCTTGCAGCAGCACTGGGCAATGACCCGAACTTTGCGACAACAATGACAAATGCGCTGGCAGGGAAA,CAGCCACTTIU2gC-AACTTTAAC 600 LYELAAALGNDPNFATTMTNALAG --------- 700 ALAGL ATG ANKL PY FT GT DTVSQ TD LTSVG RD I CTGGCCAAAACAAGCATTCTTGCT ITA TCOTA 760 LAK TSI LAV IQYLGRE LGT cixR FIG.1.Partialsequence oftheconstantportion(Sc)ofP1Stailfibergeneanditspredicted aminoacidsequence. Restriction enzymesites areprintedinboldfacetype,andthecixRcrossover siteisboxed.Bordersofrepeated elements(al,bl,b2,andc)areindicated abovethe sequence. Thesequencepartiallyoverlaps thatofafragmentcontaining the5'endofScuptotheBgllIsite(9,38)aswellasthatofa'noth'erfragmentcontaining thecixRsitewhichstartsfromtheSfaNIsite(12).Accession numberM81956includesthesequence shownhereandthe sequences oftheseflankingfragments. Finally, directevidence thattheinvertible segments ofpl5Bcodeforphagetailfiberswasobtainedbyacomplementationexperiment withaP1mutant defective initstailfiber synthesis. Theevolutionary relevance oftheseresultsis addressed. MATERIALS ANDMETHODS Bacterial andphagestrains.E.coliK-12WA921 ispin hsdKlacleumetthrsupE(40).M47,aderivative ofP1-15 prophage withatemperature-sensitive cirepressor gene, carries apolarmutation inthetailfiberoperonbecause of insertion of152(37).Plasmids. ThepBR322-derived plasmidpRM2O6 contains the7.9-kbBamHI-5 fragment ofthephageP1genome(29),andthepUC19-based plasmidpAW800 carriesthecorre- sponding 8.3-kbBamHI-5 fragment ofpl5B(17).Therele- vantfeatures oftheseBamHIfragments arealsoseeninFig.4.InbacteriacarryingpAW800, all240possibleconfigura-tionsoftheMinsystem arethought tobepresent, and therefore, plasmidpreparations ofpAW800 consist ofa mixture ofisomers. Tailfibercomplementation assay.WA921 cellsharboringthedefective P1-derived prophage M47weretransformed witheitherpRM206 orpAW800. Titersofinfectious parti-clesinlysatesproduced byheat-induced cellswereassayedbyplatingwithWA921(pRM206) indicator cellsonLuria agarplatescontaining ampicillin (200p.gIml). Plasmid pRM2O6 carriesthewild-type P1tailfiberoperon, sothe indicator bacteria canallowmultiplecyclesofreproductionofphageparticlesharboring themutantM47genome. Determination ofthePISctailfibergenesequence. The 1.7-kbIApnlfragment including thecixRcrossover siteanda largepartoftheP1SctailfibergeneofphageP1wascutout frompRM2O6 andligatedintotheKpnlsiteofpEMBL18 to generatepAW81O andintotheKpnlsiteofpEMBL19 to generatepAW811.\", \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Plasmid pRM2O6 carriesthewild-type P1tailfiberoperon, sothe indicator bacteria canallowmultiplecyclesofreproductionofphageparticlesharboring themutantM47genome. Determination ofthePISctailfibergenesequence. The 1.7-kbIApnlfragment including thecixRcrossover siteanda largepartoftheP1SctailfibergeneofphageP1wascutout frompRM2O6 andligatedintotheKpnlsiteofpEMBL18 to generatepAW81O andintotheKpnlsiteofpEMBL19 to generatepAW811. pAW812 andpAW813 areAccldeletion derivatives ofthepAW81O andpAW811plasmids, respec- tively.PlasmidpAW814 resulted fromaStyl-PstI deletion of pAW812, andpAW815 resulted fromaStyl-Kpnl deletion of pAW813. DNAsequencing wasperformed bythechain termination method(35)usingsingle-stranded DNApro-ducedfromplasmids pAW812, pAW813, pAW814, and pAW815.Laboratory procedures. Cloning wascarried outasde- scribedbyManiatis etal.(22).Computer analysis. Homology searches andsequence alignments weremadewiththeprograms oftheGenetics Computer GroupoftheUniversity ofWisconsin (5).EMBL andGenBank DNAsequence databasesandMIPSXproteinsequence databasescontaining updates untilSeptember1991werescreened withtheprograms WORDSEARCH and PROFILESEARCH. Nucleotide sequence accessi'on numbers. Theaccession number ofthe2.5-kbP1sequence whichincludes these- quence oftheAccI-KpnI fragment showninFig.1is M81956. Theaccession numbers ofDNAsequences with homology toP1orpl5BDNAreported inthisstudy(andthe corresponding references) areasfollows: forthepl5BMin region,M62121(33);forphageP1totheleftoftheinvertible segment, M81956(thisworkand9and12);onlytheamino acidsequence oftheentireP1invertible segment isavailable (20);forfragments oftheP1invertible segment, K03173 and X01828(12);forphageP1totherightoftheinvertible segment, X01828(12,15a);forthee14Pinregion,.K03521 (31);noaccession number fortheGinregionofphageMu (19);forphageX,J02459(3);forphageP2,M64677(11);for phageT4gene12,X06729(36);forphageT4gene37,J02509 (30);andforphageT4gene38,X05677(26). RESULTS ANDDISCUSSION ScgenesofP1andpl5Bcontainpartially relatedarraysof repeated elements. TheDNAsequence oftheScgeneofP1 wascompleted bysubcloning andsequencing anAccI-KpnIfragment ofphageP1including thecixRcrossover sitethat separates 739bpoftheconstantportion(Sc)from21bpof onevariableportion(Sv)ofthetailfibergeneS(Fig.1).Thestronghomology of99%identical basesofthecon- stantpart(Sc)ofgenesSofpl5BandP1includes the elements termedaO(Fig.2).Theseshort36-bpsequences share58and64%identical baseswiththe3'endsofthe succeeding 132-bpelements alofP1andp15B,respectively.Sixelements, termeda2toa7,sharebetween 64and100%VOL.174,1992\", \"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  3938 SANDMEIER ETAL. mix aOala2a3a4a5a6a7cmix'Il. p15B aI_,.'.'.'. ......................'...'.'.'.'.'.'.'.'.'.'.'.'.'.'.' I~~~~~~~~~~~................_..... ............................................................... .... ............ .............................  zato .......w o. wocst z :.::.:.::.::.:::: ...:,.::,::,..:,:,:,,::,:.:.:::::,:::::::.. :,::, ::,:::.: ..n::.:.:::.::: :::,::,:...... ....... 1.......--........aCa1b1b2.cix 200bpI Sc SvAcciBgIllStylSfaNIKpni regionsequenced FIG.2.Structure andhomology amongrepeated elements inS genesofpl5BandP1.TheaOsmallelements arehomologous tothe 3'endsofelements alinP1andinp15BDNAs.Elements a2toa7 ofp15Barehomologous toaOandalofP1,exceptforthe5'parts ofa3anda6,whichareshownasopenboxes.Thelongarrows represent aduplication ofelements a2toa4anda5toa7,respec- tively.Adjacent invertible segments areshownasopenboxes,and crossover sites(mixandcixR)areshownastriangles. Thelatter separate theconstant (Sc)fromthevariable parts(SvandSvQ)of genesS.Sequence homologies between thep15BandP1DNAsare indicated inpercentages. Thesegment fromwhichthesequence in Fig.1wasdetermined isindicated. Therestriction enzyme sites usedforcloningandsequencing areindicated.identity withaOandalofpl5Bandfollowalonpl5BDNA asdirectrepeats. Itseemsthatthea5-to-a7 segment isa recentduplication ofa2toa4,sincethesetwo432-bpand 144-amino-acid (aa)segments contain onlyfournucleotide substitutions leading toonlyoneaminoacidchange. The elements a3anda6shareattheir5'endshighlyhomologous 51-bpsequences codingfor17aawhicharenotrelated to anysequences oftheotherrepeated elements onpl5Borto aOandalonP1(Fig.3A). NexttoalofP1,a444-bpand148-aastretchconsists of twodirectly repeated elements andexhibits nosimilarity to pl5BDNA(Fig.1and2).Theseelements aredesignated bl andb2andhave63%identical basesand68%identical aminoacids(Fig.3B). Between b2andthecixRcrossover siteflanking the invertible segment ofP1liesa162-bpand54-aasequence, termedc,whichresumes homology tothecorresponding c elementofp15BDNAbetween a7andthemixcrossover site (Fig.1and2).Thosecelements have59%identical bases and69%identical aminoacids(Fig.3C). Substitution oftheinvertible DNAandtherecombinase geneonanancestral pl5Bgenome. Dataobtained onhomol- ogiesbetween theCinandMinregions aresummarized in thecentralpartofFig.4.Strikingly, theinvertible DNA segments ofP1andpl5Bbetween theircrossover sitesare nothomologous. Theminandcingenesarelocated inboth systems adjacent totheinvertible segments, buttheyare carriedinopposite orientations.\", 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Substitution oftheinvertible DNAandtherecombinase geneonanancestral pl5Bgenome. Dataobtained onhomol- ogiesbetween theCinandMinregions aresummarized in thecentralpartofFig.4.Strikingly, theinvertible DNA segments ofP1andpl5Bbetween theircrossover sitesare nothomologous. Theminandcingenesarelocated inboth systems adjacent totheinvertible segments, buttheyare carriedinopposite orientations. Theirsequences have67% identical basesand74%identical aminoacids.Incontrast, (197)............................................(209) ............ L (253) .. T D (297)APSSHTHPWNQITGVPT LAT V (353)....... TA V (397)............ T D (441)APSSHTHPWNQITVVPT LAT V(497)....... TA V(197)............................................(209) ............ L DM (164) ............ RRHP LAFT (208).. TARL (107) ............ LDIY LNKLTDP A (351).VALE DR21 41 KKS(208) (252) A(296) (352) (396) A(440) (496) (540) KKS(208) SIKDT(252) (207) TVMETK(251) IMNAR(150) VVMETR(394) 1 21 41 61 74 APL-SPALTGTPTAPTA-QGTNNTQIANTAFV-AAI-AL---SP-AL-TL-ELAAALGNDPNFATTMTNALAGK (253) IES AS KTINGAGTDKIINYSINL(326)(327) A V A TYRSVGSEDY (400)(252) N TR MAVDSDNN (325)20834).D LR LADVIDA DNN D (21053) 1 21 41 54 QP-DATLTALAGLATAA-KLPYFTG-D-ASLT-LT-VGRDILAK-S---V--YL (541)AA AD VRAASAGTIQSLD(594)(401) L GN TTVQDS TILAIQ(454)(326) K DRF NVATK STVAAIE(379)(9)KN SKNAENAE0 NVADLE(62) FIG.3.Alignment ofrepeated andothershortelements present intheScprotein sequences ofpl5B,P1,andMu,inthetailfiberprotein HofphageP2,andintailfiberproteins ofphage X.Aconsensus sequence with aplurality ofsix(A)andofthree(BandC)wasderived. Hyphens areintroduced where noconsensus wasderived. Inthespecific sequences, blanksdenoteresidues thatconform totheconsensus sequence anddotsindicate gapsintroduced foroptimalalignment. Numbers inparentheses givethecoordinates oftheaminoacidsequences predicted fromthepublished DNA sequences. (A)Alignment ofelements designated a.Therepeated elements aOtoa7ofpl5Bhaveformerly beencalledreOtore7(33).(B)Alignment ofelements designated b.Theelement bofX401-Stf hasbeendeduced from anintergenic region (seetext),andtherefore, thenumbers inparentheses arecoordinates ofthe Xgenome. (C)Alignment ofelements designated c.56 YTAQDASTTQKGIVQLSSATNSTSETLAATPKAVKAAYDLANGKA consensus pl5BScaO pl5BScal p15BSca2 pl5BSca3 p15BSca4 p15BScaS p15BSca plSBSca7 P1ScaO P1Scal P2Hal P2Ha2 MuSca )401a B cOnsensu P1Scbl P1Scb2 P2Hb X401-Stf b(2 C consensus plSBScc P1Scc P2Hc ).StfcJ.BACTERIOL.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Virology 1998 ,251, 49–58. [CrossRef] [PubMed] 47. Iida, S.; Streiff, M.B.; Bickle, T.A.; Arber, W. Two DNA Antirestriction Systems of Bacteriophage P1, DarA, and DarB: Characteri- zation of DarA−Phages. Virology 1987 ,157, 156–166. [CrossRef] 48. Piya, D.; Vara, L.; Russell, W.K.; Young, R.; Gill, J.J. The Multicomponent Antirestriction System of Phage P1 Is Linked to Capsid Morphogenesis: P1 Antirestriction Is Linked to Morphogenesis. Mol. Microbiol. 2017 ,105, 399–412. [CrossRef] 49. Fernandes, S.; S ão-José, C. Probing the Function of the Two Holin-like Proteins of Bacteriophage SPP1. Virology 2017 ,500, 184–189. [CrossRef] 50. Halgasova, N.; Ugorcakova, J.; Gerova, M.; Timko, J.; Bukovska, G. Isolation and Characterization of Bacteriophage ΦBP from Paenibacillus Polymyxa CCM 7400. FEMS Microbiol. Lett. 2010 ,305, 128–135. [CrossRef] 51. Lu, Z.; Altermann, E.; Breidt, F.; Kozyavkin, S. Sequence Analysis of Leuconostoc Mesenteroides Bacteriophage Φ1-A4 Isolated from an Industrial Vegetable Fermentation. Appl. Environ. Microbiol. 2010 ,76, 1955–1966. [CrossRef] 52. Nakonieczna, A.; Rutyna, P .; Fedorowicz, M.; Kwiatek, M.; Mizak, L.; Łobocka, M. Three Novel Bacteriophages, J5a, F16Ba, and z1a, Speciﬁc for Bacillus anthracis , Deﬁne a New Clade of Historical Wbeta Phage Relatives. Viruses 2022 ,14, 213. [CrossRef] 53. Catal ão, M.J.; Gil, F.; Moniz-Pereira, J.; S ão-José, C.; Pimentel, M. Diversity in Bacterial Lysis Systems: Bacteriophages Show the Way. FEMS Microbiol. Rev. 2013 ,37, 554–571. [CrossRef] 54. Delisle, A.L.; Barcak, G.J.; Guo, M. Isolation and Expression of the Lysis Genes of Actinomyces naeslundii Phage Av-1. Appl. Environ. Microbiol. 2006 ,72, 1110–1117. [CrossRef] [PubMed]', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  Int. J. Mol. Sci. 2022 ,23, 4231 20 of 20 55. Wo´ znica, W.M.; Bigos, J.; Łobocka, M.B. Lysis of bacterial cells in the process of bacteriophage release-canonical and newly discovered mechanisms. Postepy Hig. Med. Dosw. 2015 ,69, 114–126. 56. Park, T.; Struck, D.K.; Deaton, J.F.; Young, R. Topological dynamics of holins in programmed bacterial lysis. Proc. Natl. Acad. Sci. USA 2006 ,103, 19713–19718. [CrossRef] [PubMed] 57. Pang, T.; Park, T.; Young, R. Mapping the pinhole formation pathway of S21. Mol. Microbiol. 2010 ,78, 710–719. [CrossRef] [PubMed] 58. Zimmermann, L.; Stephens, A.; Nam, S.Z.; Rau, D.; Kübler, J.; Lozajic, M.; Gabler, F.; Söding, J.; Lupas, A.N.; Alva, V . A Completely Reimplemented MPI Bioinformatics Toolkit with a New HHpred Server at its Core. J. Mol. Biol. 2018 ,430, 2237–2243. [CrossRef] [PubMed] 59. Gabler, F.; Nam, S.Z.; Till, S.; Mirdita, M.; Steinegger, M.; Söding, J.; Lupas, A.N.; Alva, V . Protein Sequence Analysis Using the MPI Bioinformatics Toolkit. Curr. Protoc. Bioinform. 2020 ,72, e108. [CrossRef] 60. Pang, T.; Park, T.; Young, R. Mutational analysis of the S21 pinholin. Mol. Microbiol. 2010 ,76, 68–77. [CrossRef] 61. Wang, I.N. Lysis timing and bacteriophage ﬁtness. Genetics 2010 ,172, 17–26. [CrossRef] 62. Gallet, R.; Kannoly, S.; Wang, I.N. Effects of bacteriophage traits on plaque formation. BMC Microbiol. 2011 ,11, 181. [CrossRef] 63. Cornaglia, G.; Mazzariol, A.; Fontana, R.; Satta, G. Diffusion of Carbapenems Through the Outer Membrane of Enterobacteriaceae and Correlation of Their Activities with Their Periplasmic Concentrations. Microb. Drug Resist. 1996 ,2, 273–276. [CrossRef] 64. Sohlenkamp, C.; Geiger, O. Bacterial Membrane Lipids: Diversity in Structures and Pathways. FEMS Microbiol. Rev. 2016 ,40, 133–159. [CrossRef] [PubMed] 65. Acosta-Guti érrez, S.; Ferrara, L.; Pathania, M.; Masi, M.; Wang, J.; Bodrenko, I.; Zahn, M.; Winterhalter, M.; Stavenger, R.A.; Pagès, J.-M.; et al. Getting Drugs into Gram-Negative Bacteria: Rational Rules for Permeation through General Porins. ACS Infect. Dis. 2018 ,4, 1487–1498. [CrossRef] [PubMed] 66. Pomares, M.F.; Delgado, M.A.; Corbal án, N.S.; Far ías, R.N.; Vincent, P .A. Sensitization of Microcin J25-Resistant Strains by a Membrane-Permeabilizing Peptide. Appl. Environ. Microbiol. 2010 ,76, 6837–6842. [CrossRef] 67. Neu, H.C.; Chou, J. Release of Surface Enzymes in Enterobacteriaceae by Osmotic Shock. J. Bacteriol. 1967 ,94, 1934–1945. [CrossRef] [PubMed] 68.', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  ACS Infect. Dis. 2018 ,4, 1487–1498. [CrossRef] [PubMed] 66. Pomares, M.F.; Delgado, M.A.; Corbal án, N.S.; Far ías, R.N.; Vincent, P .A. Sensitization of Microcin J25-Resistant Strains by a Membrane-Permeabilizing Peptide. Appl. Environ. Microbiol. 2010 ,76, 6837–6842. [CrossRef] 67. Neu, H.C.; Chou, J. Release of Surface Enzymes in Enterobacteriaceae by Osmotic Shock. J. Bacteriol. 1967 ,94, 1934–1945. [CrossRef] [PubMed] 68. Nickerson, K.W.; Aspedon, A. Detergent-Shock Response in Enteric Bacteria. Mol. Microbiol. 1992 ,6, 957–961. [CrossRef] 69. Meyer, J.; Stålhammar-Carlemalm, M.; Streiff, M.; Iida, S.; Arber, W. Sequence Relations among the IncY Plasmid P15B, P1, and P7 Prophages. Plasmid 1986 ,16, 81–89. [CrossRef] 70. Yang, L.; Li, W.; Jiang, G.-Z.; Zhang, W.-H.; Ding, H.-Z.; Liu, Y.-H.; Zeng, Z.-L.; Jiang, H.-X. Characterization of a P1-like Bacteriophage Carrying CTX-M-27 in Salmonella Spp. Resistant to Third Generation Cephalosporins Isolated from Pork in China. Sci. Rep. 2017 ,7, 40710. [CrossRef] 71. Billard-Pomares, T.; Fouteau, S.; Jacquet, M.E.; Roche, D.; Barbe, V .; Castellanos, M.; Bouet, J.Y.; Cruveiller, S.; M édigue, C.; Blanco, J.; et al. Characterization of a P1-Like Bacteriophage Carrying an SHV-2 Extended-Spectrum β-Lactamase from an Escherichia Coli Strain. Antimicrob. Agents Chemother. 2014 ,58, 6550–6557. [CrossRef] 72. Hagbø, M.; Ravi, A.; Angell, I.L.; Sunde, M.; Ludvigsen, J.; Diep, D.B.; Foley , S.L.; V ento, M.; Collado, M.C.; Perez-Martinez, G.; et al. Experimental Support for Multidrug Resistance Transfer Potential in the Preterm Infant Gut Microbiota. Pediatr. Res. 2020 ,88, 57–65. [CrossRef] 73. Gottesman, M.E.; Yarmolinsky, M.B. Integration-Negative Mutants of Bacteriophage Lambda. J. Mol. Biol. 1968 ,31, 487–505. [CrossRef] 74. Łobocka, M.; Yarmolinsky, M. P1 Plasmid Partition: A Mutational Analysis of ParB. J. Mol. Biol. 1996 ,259, 366–382. [CrossRef] [PubMed] 75. Sambrook, J.; Fritsch, E.F.; Maniatis, T. Molecular Cloning: A Laboratory Manual , 2nd. ed.; Cold Spring Harbor Laboratory Press: Cold Spring Harbor, NY, USA, 1989. 76. Adams, M.H. Bacteriophages ; Interscience Publishers Inc.: New York, NY, USA, 1959. 77. Sambrook, J.; Russell, D.W. Molecular Cloning: A Laboratory Manual , 3rd ed.; Cold Spring Harbor Laboratory Press: Cold Spring Harbor, NY, USA, 2001; Volume 1. 78. Helvoort, J.M.L.M.; Woldringh, C.L.', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf  J. Mol. Biol. 1996 ,259, 366–382. [CrossRef] [PubMed] 75. Sambrook, J.; Fritsch, E.F.; Maniatis, T. Molecular Cloning: A Laboratory Manual , 2nd. ed.; Cold Spring Harbor Laboratory Press: Cold Spring Harbor, NY, USA, 1989. 76. Adams, M.H. Bacteriophages ; Interscience Publishers Inc.: New York, NY, USA, 1959. 77. Sambrook, J.; Russell, D.W. Molecular Cloning: A Laboratory Manual , 3rd ed.; Cold Spring Harbor Laboratory Press: Cold Spring Harbor, NY, USA, 2001; Volume 1. 78. Helvoort, J.M.L.M.; Woldringh, C.L. Nucleoid Partitioning in Escherichia coli during Steady-State Growth and upon Recovery from Chloramphenicol Treatment. Mol. Microbiol. 1994 ,13, 577–583. [CrossRef] [PubMed]', \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Vol.174,No.12 DNAInversion Regions MinofPlasmid pl5BandCin ofBacteriophage P1:Evolution ofBacteriophage TailFiberGenes HEINRICH SANDMEIER,lt SHIGERU IIDA,1'2 ANDWERNER ARBERl* Abteilung Mikrobiologie, Biozentrum derUniversitat Basel,Klingelbergstrasse 70,CH-4056 Basel, Switzerland, 1andDepartment ofBiological Sciences andTechnology, Science University ofTokyo, Yamazaki, Chiba278,Japan2 Received 21November 1991/Accepted 11April1992 Plasmid pl5Bandthegenome ofbacteriophage P1arecloselyrelated,buttheirsite-specific DNAinversion systems, MinandCin,respectively, donothavestrictstructural homology. Rather, thecomplex Minsystem represents asubstitution ofaCin-like systemintoanancestral pl5Bgenome. Thesubstituting sequences of boththeminrecombinase geneandthemultiple invertible DNAsegments ofpl5B are,respectively, homologous tothepinrecombinase geneandtopartoftheinvertible DNAofthePinsystem onthedefective viralelement e14ofEscherichia coliK-12.Tomapthesitesofthissubstitution, theDNAsequence ofasegment adjacent totheinvertible segment intheP1genome wasdetermined. This,together withalready available sequence data,indicated thatbothP1andpl5Bhadsuffered various sequence acquisitions ordeletions and sequence amplifications givingrisetomosaics ofpartially relatedrepeated elements. Databasesearches revealed segments ofhomology intheDNAinversion regionsofpl5B,e14,andP1andintailfibergenesof phagesMu,T4,P2,andK.Thisresultsuggests thattheevolution ofphagetailfibergenesinvolves horizontal genetransfer andthattheMinandPinregionsencodetailfibergenes.Afunctional testprovedthatthepl5B Minregioncarries atailfiberoperonandsuggests thatthealternative expression ofsixdifferent genevariants byMininversion offersextensive hostrangevariation. Plasmid p15BisaP1prophage-related resident ofEsche- richiacoli15T-(18).Electron microscopy heteroduplex studiesrevealed thatapproximately 82%ofthe94-kbpl5B DNAhybridizes withtheP1genome (23).Since nophage particle withaP1-likemorphology couldbeproduced inE. coli15T-(1,18),pl5BDNAhasbeenconsidered tobea defective prophage. Oneregionofnonhomology between pl5BandP1DNA coverstheinvertible segment oftheP1 DNA(16,23).Itscorresponding region onpl5Bcarries a complex site-specific DNAinversion system calledMin(17, 34).TheMinregioncontains sixcrossover sitescalled mix. Successive DNAinversions between pairsofmixsites mediated byMinrecombinase canproduce 240isomeric formsofa3.5-kbregion(34).Theinvertible DNAsegments contain sixdifferent 3'geneparts,eachofwhich canbecome fusedinframetoanadjacent constant 5'genesegment upon DNAinversion (33)(seeFig.4). Thecomplex Minsystemiscloselyrelated tosimpleDNA inversion systems whichcontain twocrossover sitesand constitute atwo-state switch forgeneexpression: Cinof phageP1,GinofphageMu,Pinofthee14element onE.coli K-12,andHinofSalmonella typhimurium Thecrossover sitesandrecombinase genesofthesesystems arestructur- allyhomologous andcanmutually substitute fortheirfunc- tions(forareviewoftheseDNAinversion systems, called heretheDinfamily, seereference 7;seealsoFig.4).Inthe Hinsystem, twoflagellar geneswithdistinctantigenic spec- ificities arealternatively expressed byinversion ofaDNA *Corresponding author.\", \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Thecomplex Minsystemiscloselyrelated tosimpleDNA inversion systems whichcontain twocrossover sitesand constitute atwo-state switch forgeneexpression: Cinof phageP1,GinofphageMu,Pinofthee14element onE.coli K-12,andHinofSalmonella typhimurium Thecrossover sitesandrecombinase genesofthesesystems arestructur- allyhomologous andcanmutually substitute fortheirfunc- tions(forareviewoftheseDNAinversion systems, called heretheDinfamily, seereference 7;seealsoFig.4).Inthe Hinsystem, twoflagellar geneswithdistinctantigenic spec- ificities arealternatively expressed byinversion ofaDNA *Corresponding author. tPresent address: Abteilung furPraventivzahnmedizin undorale Mikrobiologie, Zahnarztliches InstitutderUniversitat Basel,Peter- splatz14,CH-4051 Basel,Switzerland.segment including apromoter. InallotherDinsystems described sofar,astationary promoter transcribes agene whichisassembled byDNAinversion fromconstant and variable genesegments. SuchgenefusionsbytheCinand Gininversions inphages P1andMuleadtoalternate expression oftwosetsoftailfibergeneswhichconfer different receptor specificities. TheseP1andMuSgenes consistofa5'constant portion, calledSc,andtwoversions ofthe3'portion, calledSvandSv'.Thee14element, which canbeconsidered adefective prophage, isanexcisable element ofsomeE.coliK-12chromosomes (la,21).The biological roleofPininversion one14,aswellasthoseof recently identified related DNAinversion systems onShi- gellachromosomes (39),isnotyetknown. Thecomparison oftheDNA sequences ofthepl5BMin regionwithpreviously determined sequences oftheCin regionofphageP1suggested thattheinvertible segment and partoftherecombinase genehadbeensubstituted by nonhomologous DNA onanancestral pl5B genome (33). While oneofthebreakpoints fromhomologous tononho- mologous sequences hadalready beenlocated withinthe recombinase genes,thesecondbreakpoint remained unde- termined because ofincomplete P1sequence dataofthis region. Inthispaper,theDNA sequence oftheconstant part(Sc) oftheP1geneSwascompleted toelucidate thesecond breakpoint. Theresultindicates thatcomplicated recombi- nations musthavetakenplaceatthissecondbreakpoint on anancestral genome. Inaddition, acomputer-aided compar- isonofthese sequences withavailable DNAandprotein sequences ofotherbacteriophages identified various se- quence domains thatareshared amongthegenesoftheP1 andpl5BDNAinversion regions andtailfibergenesof phages Mu,P2,X,andT4,aswellastheelement e14. 3936JOURNAL OFBACTERIOLOGY, June1992,p.3936-3944 0021-9193/92/123936-09$02.00/0 Copyright ©1992,American SocietyforMicrobiology\", \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  VOL.174,1992 ~~~~EVOLUTION OFpl5B3ANDPHAGE P1TAILFIBERGENES 3937 AccI  eal~~~~~~~~~~~~~~~~~~~~~~~~~~~~~0YTAQDASTTQKGLVQLSSATNSDSETMAATPKA *BglII alN,.,e..bl . .  * ~~~~~StyI bl%,,.~.b2. INPYTTINNNLA PLASPA LTGVPTAP200 300 400 500 TAAQGTNNTQ IATTAY VRAAI SALVGSSPEALD T b%..c SfaNI CCCTGTATGAGCTTGCAGCAGCACTGGGCAATGACCCGAACTTTGCGACAACAATGACAAATGCGCTGGCAGGGAAA,CAGCCACTTIU2gC-AACTTTAAC 600 LYELAAALGNDPNFATTMTNALAG --------- 700 ALAGL ATG ANKL PY FT GT DTVSQ TD LTSVG RD I CTGGCCAAAACAAGCATTCTTGCT ITA TCOTA 760 LAK TSI LAV IQYLGRE LGT cixR FIG.1.Partialsequence oftheconstantportion(Sc)ofP1Stailfibergeneanditspredicted aminoacidsequence. Restriction enzymesites areprintedinboldfacetype,andthecixRcrossover siteisboxed.Bordersofrepeated elements(al,bl,b2,andc)areindicated abovethe sequence. Thesequencepartiallyoverlaps thatofafragmentcontaining the5'endofScuptotheBgllIsite(9,38)aswellasthatofa'noth'erfragmentcontaining thecixRsitewhichstartsfromtheSfaNIsite(12).Accession numberM81956includesthesequence shownhereandthe sequences oftheseflankingfragments. Finally, directevidence thattheinvertible segments ofpl5Bcodeforphagetailfiberswasobtainedbyacomplementationexperiment withaP1mutant defective initstailfiber synthesis. Theevolutionary relevance oftheseresultsis addressed. MATERIALS ANDMETHODS Bacterial andphagestrains.E.coliK-12WA921 ispin hsdKlacleumetthrsupE(40).M47,aderivative ofP1-15 prophage withatemperature-sensitive cirepressor gene, carries apolarmutation inthetailfiberoperonbecause of insertion of152(37).Plasmids. ThepBR322-derived plasmidpRM2O6 contains the7.9-kbBamHI-5 fragment ofthephageP1genome(29),andthepUC19-based plasmidpAW800 carriesthecorre- sponding 8.3-kbBamHI-5 fragment ofpl5B(17).Therele- vantfeatures oftheseBamHIfragments arealsoseeninFig.4.InbacteriacarryingpAW800, all240possibleconfigura-tionsoftheMinsystem arethought tobepresent, and therefore, plasmidpreparations ofpAW800 consist ofa mixture ofisomers. Tailfibercomplementation assay.WA921 cellsharboringthedefective P1-derived prophage M47weretransformed witheitherpRM206 orpAW800. Titersofinfectious parti-clesinlysatesproduced byheat-induced cellswereassayedbyplatingwithWA921(pRM206) indicator cellsonLuria agarplatescontaining ampicillin (200p.gIml). Plasmid pRM2O6 carriesthewild-type P1tailfiberoperon, sothe indicator bacteria canallowmultiplecyclesofreproductionofphageparticlesharboring themutantM47genome. Determination ofthePISctailfibergenesequence. The 1.7-kbIApnlfragment including thecixRcrossover siteanda largepartoftheP1SctailfibergeneofphageP1wascutout frompRM2O6 andligatedintotheKpnlsiteofpEMBL18 to generatepAW81O andintotheKpnlsiteofpEMBL19 to generatepAW811.\", \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Plasmid pRM2O6 carriesthewild-type P1tailfiberoperon, sothe indicator bacteria canallowmultiplecyclesofreproductionofphageparticlesharboring themutantM47genome. Determination ofthePISctailfibergenesequence. The 1.7-kbIApnlfragment including thecixRcrossover siteanda largepartoftheP1SctailfibergeneofphageP1wascutout frompRM2O6 andligatedintotheKpnlsiteofpEMBL18 to generatepAW81O andintotheKpnlsiteofpEMBL19 to generatepAW811. pAW812 andpAW813 areAccldeletion derivatives ofthepAW81O andpAW811plasmids, respec- tively.PlasmidpAW814 resulted fromaStyl-PstI deletion of pAW812, andpAW815 resulted fromaStyl-Kpnl deletion of pAW813. DNAsequencing wasperformed bythechain termination method(35)usingsingle-stranded DNApro-ducedfromplasmids pAW812, pAW813, pAW814, and pAW815.Laboratory procedures. Cloning wascarried outasde- scribedbyManiatis etal.(22).Computer analysis. Homology searches andsequence alignments weremadewiththeprograms oftheGenetics Computer GroupoftheUniversity ofWisconsin (5).EMBL andGenBank DNAsequence databasesandMIPSXproteinsequence databasescontaining updates untilSeptember1991werescreened withtheprograms WORDSEARCH and PROFILESEARCH. Nucleotide sequence accessi'on numbers. Theaccession number ofthe2.5-kbP1sequence whichincludes these- quence oftheAccI-KpnI fragment showninFig.1is M81956. Theaccession numbers ofDNAsequences with homology toP1orpl5BDNAreported inthisstudy(andthe corresponding references) areasfollows: forthepl5BMin region,M62121(33);forphageP1totheleftoftheinvertible segment, M81956(thisworkand9and12);onlytheamino acidsequence oftheentireP1invertible segment isavailable (20);forfragments oftheP1invertible segment, K03173 and X01828(12);forphageP1totherightoftheinvertible segment, X01828(12,15a);forthee14Pinregion,.K03521 (31);noaccession number fortheGinregionofphageMu (19);forphageX,J02459(3);forphageP2,M64677(11);for phageT4gene12,X06729(36);forphageT4gene37,J02509 (30);andforphageT4gene38,X05677(26). RESULTS ANDDISCUSSION ScgenesofP1andpl5Bcontainpartially relatedarraysof repeated elements. TheDNAsequence oftheScgeneofP1 wascompleted bysubcloning andsequencing anAccI-KpnIfragment ofphageP1including thecixRcrossover sitethat separates 739bpoftheconstantportion(Sc)from21bpof onevariableportion(Sv)ofthetailfibergeneS(Fig.1).Thestronghomology of99%identical basesofthecon- stantpart(Sc)ofgenesSofpl5BandP1includes the elements termedaO(Fig.2).Theseshort36-bpsequences share58and64%identical baseswiththe3'endsofthe succeeding 132-bpelements alofP1andp15B,respectively.Sixelements, termeda2toa7,sharebetween 64and100%VOL.174,1992\", \"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  3938 SANDMEIER ETAL. mix aOala2a3a4a5a6a7cmix'Il. p15B aI_,.'.'.'. ......................'...'.'.'.'.'.'.'.'.'.'.'.'.'.'.' I~~~~~~~~~~~................_..... ............................................................... .... ............ .............................  zato .......w o. wocst z :.::.:.::.::.:::: ...:,.::,::,..:,:,:,,::,:.:.:::::,:::::::.. :,::, ::,:::.: ..n::.:.:::.::: :::,::,:...... ....... 1.......--........aCa1b1b2.cix 200bpI Sc SvAcciBgIllStylSfaNIKpni regionsequenced FIG.2.Structure andhomology amongrepeated elements inS genesofpl5BandP1.TheaOsmallelements arehomologous tothe 3'endsofelements alinP1andinp15BDNAs.Elements a2toa7 ofp15Barehomologous toaOandalofP1,exceptforthe5'parts ofa3anda6,whichareshownasopenboxes.Thelongarrows represent aduplication ofelements a2toa4anda5toa7,respec- tively.Adjacent invertible segments areshownasopenboxes,and crossover sites(mixandcixR)areshownastriangles. Thelatter separate theconstant (Sc)fromthevariable parts(SvandSvQ)of genesS.Sequence homologies between thep15BandP1DNAsare indicated inpercentages. Thesegment fromwhichthesequence in Fig.1wasdetermined isindicated. Therestriction enzyme sites usedforcloningandsequencing areindicated.identity withaOandalofpl5Bandfollowalonpl5BDNA asdirectrepeats. Itseemsthatthea5-to-a7 segment isa recentduplication ofa2toa4,sincethesetwo432-bpand 144-amino-acid (aa)segments contain onlyfournucleotide substitutions leading toonlyoneaminoacidchange. The elements a3anda6shareattheir5'endshighlyhomologous 51-bpsequences codingfor17aawhicharenotrelated to anysequences oftheotherrepeated elements onpl5Borto aOandalonP1(Fig.3A). NexttoalofP1,a444-bpand148-aastretchconsists of twodirectly repeated elements andexhibits nosimilarity to pl5BDNA(Fig.1and2).Theseelements aredesignated bl andb2andhave63%identical basesand68%identical aminoacids(Fig.3B). Between b2andthecixRcrossover siteflanking the invertible segment ofP1liesa162-bpand54-aasequence, termedc,whichresumes homology tothecorresponding c elementofp15BDNAbetween a7andthemixcrossover site (Fig.1and2).Thosecelements have59%identical bases and69%identical aminoacids(Fig.3C). Substitution oftheinvertible DNAandtherecombinase geneonanancestral pl5Bgenome. Dataobtained onhomol- ogiesbetween theCinandMinregions aresummarized in thecentralpartofFig.4.Strikingly, theinvertible DNA segments ofP1andpl5Bbetween theircrossover sitesare nothomologous. Theminandcingenesarelocated inboth systems adjacent totheinvertible segments, buttheyare carriedinopposite orientations.\", 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Substitution oftheinvertible DNAandtherecombinase geneonanancestral pl5Bgenome. Dataobtained onhomol- ogiesbetween theCinandMinregions aresummarized in thecentralpartofFig.4.Strikingly, theinvertible DNA segments ofP1andpl5Bbetween theircrossover sitesare nothomologous. Theminandcingenesarelocated inboth systems adjacent totheinvertible segments, buttheyare carriedinopposite orientations. Theirsequences have67% identical basesand74%identical aminoacids.Incontrast, (197)............................................(209) ............ L (253) .. T D (297)APSSHTHPWNQITGVPT LAT V (353)....... TA V (397)............ T D (441)APSSHTHPWNQITVVPT LAT V(497)....... TA V(197)............................................(209) ............ L DM (164) ............ RRHP LAFT (208).. TARL (107) ............ LDIY LNKLTDP A (351).VALE DR21 41 KKS(208) (252) A(296) (352) (396) A(440) (496) (540) KKS(208) SIKDT(252) (207) TVMETK(251) IMNAR(150) VVMETR(394) 1 21 41 61 74 APL-SPALTGTPTAPTA-QGTNNTQIANTAFV-AAI-AL---SP-AL-TL-ELAAALGNDPNFATTMTNALAGK (253) IES AS KTINGAGTDKIINYSINL(326)(327) A V A TYRSVGSEDY (400)(252) N TR MAVDSDNN (325)20834).D LR LADVIDA DNN D (21053) 1 21 41 54 QP-DATLTALAGLATAA-KLPYFTG-D-ASLT-LT-VGRDILAK-S---V--YL (541)AA AD VRAASAGTIQSLD(594)(401) L GN TTVQDS TILAIQ(454)(326) K DRF NVATK STVAAIE(379)(9)KN SKNAENAE0 NVADLE(62) FIG.3.Alignment ofrepeated andothershortelements present intheScprotein sequences ofpl5B,P1,andMu,inthetailfiberprotein HofphageP2,andintailfiberproteins ofphage X.Aconsensus sequence with aplurality ofsix(A)andofthree(BandC)wasderived. Hyphens areintroduced where noconsensus wasderived. Inthespecific sequences, blanksdenoteresidues thatconform totheconsensus sequence anddotsindicate gapsintroduced foroptimalalignment. Numbers inparentheses givethecoordinates oftheaminoacidsequences predicted fromthepublished DNA sequences. (A)Alignment ofelements designated a.Therepeated elements aOtoa7ofpl5Bhaveformerly beencalledreOtore7(33).(B)Alignment ofelements designated b.Theelement bofX401-Stf hasbeendeduced from anintergenic region (seetext),andtherefore, thenumbers inparentheses arecoordinates ofthe Xgenome. (C)Alignment ofelements designated c.56 YTAQDASTTQKGIVQLSSATNSTSETLAATPKAVKAAYDLANGKA consensus pl5BScaO pl5BScal p15BSca2 pl5BSca3 p15BSca4 p15BScaS p15BSca plSBSca7 P1ScaO P1Scal P2Hal P2Ha2 MuSca )401a B cOnsensu P1Scbl P1Scb2 P2Hb X401-Stf b(2 C consensus plSBScc P1Scc P2Hc ).StfcJ.BACTERIOL.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:02 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'132'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993918'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'364ms'), (b'x-request-id', b'c3940f087bc1d6997bae7573ec5286a0'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249a579c45d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:02 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'132'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993918'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'364ms'), (b'x-request-id', b'c3940f087bc1d6997bae7573ec5286a0'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249a579c45d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  EVOLUTION OFplSBANDPHAGE P1TAILFIBER GENES 3939 -=DC==>gin P.Sc Sv U U Sv FIG.4.Mapofhomologous DNAsequences amongtheDNA inversion regionsofe14,pl5B,P1,andMu.P1andp1SBregionsare represented bytheirBamHI-5 fragments. Invertible segments are shownasboxesflanked bycrossover sitesdepicted asblack triangles. Genesofoperons thatareaffected byDNAinversion and DNAinvertase genesareshownasopenarrows. TheP1cingeneis opposite inorientation tothatoftherelatedDNAinvertase genes. ThePspromoters oftailfibergenesareshownasshortblack arrows.Homology between pl5BandP1totheleftoftheinvertible segments issimplified fromthatshowninFig.2.Therefore, thea,b, andcelements composing theseregions arenotspecifically shown here.Thesiteofintegration ofIS2intoP1prophage M47ismarked. Theindicated highdegreeofhomology ofCinandGininvertible segments (20)isinterrupted byanapproximately 190-bpsegment of only35%homology aroundtheendofSvandthestartofUgenes. Thesmall63%homologous sequences inScofP1andMucorre- spondtotheaelements whicharealignedinFig.3A.The67-bp sequences between thecrossover pointsandtheDNAinvertase genesofpl5Bande14show31%homology. remarkable homologies inthesameorientation areseenon boththerightandtheleftsidesoftheinvertible segments andtheirrecombinase genes.Totheright,homologous DNA onP1startsstillwithinthecingeneandhence,thepl5B genome harbors aremnant fragment ofcin,thepseudo-cin gene(ilcin).Ithasbeenshownelsewhere thattlcinandthe recombinational enhancer sequence ontjcinarenonfunc- tional(33).Presence ofthetlcingeneindicates thattheDNA inversion regionofanancestral p15Bgenome hadcarrieda site-specific recombinase geneoftheDinfamilyinthesameconfiguration asthatoftheP1genome. Itismostlikelythat thecingeneandtheinvertible DNAofthisancestral pl5B genome hadbeensubstituted bytheMinsystemcomposed ofthemingeneandtheadjacent multiple invertible DNA segments. Sincethebreakpoints fromhomology tononho- mology between p15BandP1DNA aresharpandsomeof themcoincide withthecrossover sitesforDinrecombinases, itistempting tospeculate thataDinsite-specific recombi- nasehadmediated thesubstitution process. Thestructures ofrepeated elements neartheleftbreakpoints onp15Band P1suggest thatsubstitution mighthavebeenassociated with complicated recombinational processes. Alternatively, the presence ofthesestructures andthemultiple invertible DNA segments ofpl5Bcouldhaveresulted fromindependent events. Relation between pl5Bande14invertible DNAsegments. Comparison withthesequences ofotherDNAinversion systems revealed thatthefiveinvertible segments ofpl5B arerelated totheinvertible DNAofthePinsystemofe14.A 1.1-kb partofthe1.8-kbPininvertible segment (31)is between 60and80%homologous tothe3.5-kbMininvertible segments (Fig.4).ThesixSvcarboxy-terminal variable segments ofpl5BsharewiththeP(-)carboxy-terminal segment ofe14ahighlyhomologous domain ofabout70aa (Fig.5).When anucleotide isinserted afterposition 497of thepublished sequence ofthePinregionofe14toproduce a frameshift inP(-),thelast13aaofthepredicted P(-) product areidentical totheSvM1,SvP,andSvQandalmost identical totheSvMr,SvN,andSvOcarboxy-terminal segments (Fig.5).Thissecondhomologous domain issepa- ratedfromthefirstdomainby61to84aaintheSvandP(-) sequences.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  OnlytheSvMrandP(-)segments arehomolo- gousalsobetween the70-and13-aadomains (datanot shown). Thealternative P(+)carboxy-terminal segment of e14exhibits nosequence homology toP(-)ortopl5BSv segments. Thepostulated geneTofpl5B,two-thirds ofe14open reading frame(ORF)293,andtheirputative products have 75%oftheirbasesandaminoacidsincommon. Homology oftheMinandPinregionsincludes notonlythe invertible DNAsegments butalsotheadjacent pinandmin genes,whichhave72%identical basesand80%identical aminoacids.Preliminary sequence datashowed that0.6kb ofe14flanking theothersideofthePininvertible segment arenothomologous totheMinregion(32). Sincethepingeneistheclosest relative ofthemin recombinase gene(17)andthepl5Bande14invertible DNA TABLE 1.Percentages ofidentical (andsimilar) residues ofaelements alignedinFig.3A %ofidentical (%ofsimilar) residues with: Element5Bcal pl5BScpl5BSca3pl5BSc PSa uc 2a 2a 4l EeplSB Scala2andaS anda6b a4anda7PlScalMuScaP2HalP2Ha2i401a pl5BScal 100 pl5BSca2andaS 89(91) 100 pl5BSca3anda6fi 90(92) 82(82) 100 pl5BSca4anda7 93(95) 91(91) 87(87) 100 P1Scal 82(86) 77(82) 69(77) 75(82) 100 MuSca 61(70) 55(66) 64(77) 57(68) 57(68) 100 P2Hal 80(82) 75(77) 86(89) 75(77) 66(73) 61(73) 100 P2Ha2 77(77) 75(77) 59(61) 80(82) 73(80) 59(70 64(66) 100 X401a 68(73) 68(70) 72(72) 68(70) 64(75) 57(77) 66(68) 80(86) 100 Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). hThefirst17aaofpl5Ba3anda6havenotbeenconsidered forthecalculation ofhomology values.VOL.174,1992', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  3940 SANDMEIER ETAL. consensus pl5BSvM1 plSBSvMr P15BSVN pl5BSvO pl5BSvP P15BSVQ e14P(-) XStf T437 T412 consensus pl5BSvMl pl5BSvMr P1SBSVN plSBSvO plSBSvP P15BSVQ e14P(-) XStf T437 T4121 21 41 61 80 GLGEGSALPVGVPVPWPSATPPT. GWLKCNGAAFSSE-YPKLAK ....... AYPTNK. LPDLRGEFIRGWDDGRGVD-GR (1) (1) (1) (1) (1) (1) (1) (63) (806) (324)LE . M ....... KN........ L. LE . M ....... K N........ V L. M ....... KN....... AEE....... A N FAAI DIV S.YVLMQ DKSA .... SGV. KTDIM SYIAIDSVA.FALME QTDKSA V SGV.I QNEIDRTI AIMMAADSL SDARFHGTVASDCLYSRIGTRYGGN SPG 81 101 116 --LS-Q--- ......---H-H-- VI IGWL ...TGS YNI SI EGYA ...TED AGL AL NW PHT ..ILD AYM SL GTAT FIRTALQD ALITGM LEKRIV QL SGDA IRNIEGFA SIIGYA TEDAGL AV QEQDG IKS TSA AV AEADG. VKA SSA VNGNDFGKPRLGVGCTGGYVGEVQIQQMSY KAG(88) (88) (88) (88) (88) (88) (88) (145) (887) (439)(132) (147) (149) (149) (143) (135) (149) (295) (1007) (497)A Is A SIT A A IT NWTKKPAS..... MQTKKP.S..... MLVSGRSHLTNPN 1 21 ----E-RPRNIAFNYIVRAA* SGGNT *(153) TAGLT D*(166) NVGTT V *(168) GAWISM V T*(168) TGGST *(159) PTANN *(145) AAASAX *(165) AGNANTVK L*(314) TGNTNTVK L*(1026) LIGNTWSLIKVKE* (517) FIG.5.Alignment oftwoconserved domains amongtheSvvariable portions ofthepl5BSprotein, oftheP(-)variable partcodedinthe Pininversion systemofe14,oftheStfproteinofphageA,ofprotein37ofphageT4,andofprotein12ofphageT4.Xintheseconddomain ofthee14sequence identifies thesiteatwhich asinglebasepairwasintroduced toallowalignment ofthecarboxy-terminal ends.Asterisks areappended tothecarboxy-terminal endsoftheproteins. Forothersymbols, seethelegend toFig.3. segments arehighlyhomologous, theDNAwhichhasre- placedtheDNAinversion regionoftheancestral p15B genome musthavebeencloselyrelatedtothee14Pinregion. Asthediagram ofFig.4shows,severalofthebreakpoints between homology andnonhomology amongthePinand Minsystems andtheCinandGinsystems, respectively, also coincide withcrossover sitesforDinrecombinases. The breakpoints oftheDNAinversion regionsofShigella boydii anditsclosestrelative, theGinsystemofphageMu,whose invertible segments have62to98%identical bases,also coincide withthecrossover sitesonbothgenomes (39).This supports anearlierpostulate thatDinrecombinases could mediate segmental substitutions amongDNAinversion sys- tems(12,16).Otherbreakpoints whichexhibit noobvious similarity toconsensus crossover sitescouldnevertheless oncehavebeenusedassubstrates forsuchrecombinases.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Asthediagram ofFig.4shows,severalofthebreakpoints between homology andnonhomology amongthePinand Minsystems andtheCinandGinsystems, respectively, also coincide withcrossover sitesforDinrecombinases. The breakpoints oftheDNAinversion regionsofShigella boydii anditsclosestrelative, theGinsystemofphageMu,whose invertible segments have62to98%identical bases,also coincide withthecrossover sitesonbothgenomes (39).This supports anearlierpostulate thatDinrecombinases could mediate segmental substitutions amongDNAinversion sys- tems(12,16).Otherbreakpoints whichexhibit noobvious similarity toconsensus crossover sitescouldnevertheless oncehavebeenusedassubstrates forsuchrecombinases. It hasbeenshownelsewhere thatDNAinvertases canutilize secondary crossover siteswithquitedegenerate sequences (14,15),andsuchsitesmighthavebeenmutated during evolution subsequent toasubstitution event. Homologous domains ofgenescontained inpl5B,P1,Mu, ande14DNAinversion regionsandintailfibergenesofphages T4,P2,andX.Screening ofnucleic acidandaminoacid sequence databasesrevealed unexpected relationships amongthecodingsequences ofDNAinversion systems of TABLE 2.Percentages ofidentical (andsimilar) residues of belements alignedinFig.3B %ofidentical (%ofsimilar)\" residues with: Element X401-StfPlScblPlScb2 P2Hb b P1Scbl 100 P1Scb2 68(81) 100 P2Hb 68(82) 84(89) 100 X401-Stfb 65(81) 77(86) 86(92) 100 \"Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8).pl5B,e14,andphageP1andgenesofotherwise unrelated bacteriophages. Thepartlyrelatedandrepeated elements inthepl5Band P1Scgeneshadcounterparts intailfibergenesofphages Mu,P2,andX.Segments homologous tothepl5BandP1 elements designated awerefoundinduplicate inthephage P2geneHandinsinglecopiesintheconstant part,Sc,ofthe MutailfibergeneS,andinXORF401.These44-aaelements present onfivedistinctgenomes havebetween 55and93% identical aminoacids(Fig.3A;Table1).Theclosestrela- tionships, withatleast80%identical aminoacids,are between thepl5Belements altoa7amongeachother,the element alofP1andalofpl5B,theelement alofP2andal, a3,anda6ofpl5B,andtheelement a2ofP2anda4anda7 ofpl5Bandthecorresponding Xsequence. Thetandemrepeated elements blandb2oftheP1genome andthecelements inP1andpl5Barerelatedtosequences intheP2andXgenomes (Fig.3BandC).Thebelements havebetween 65and86%identical aminoacids(Table2), andthecelements havebetween 61and70%identical amino acids(Table3).Notably, thehighlyconserved a,b,andc elements areimmediately adjacent toeachotherinthepl5B, P1,P2,andXgenomes.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Thetandemrepeated elements blandb2oftheP1genome andthecelements inP1andpl5Barerelatedtosequences intheP2andXgenomes (Fig.3BandC).Thebelements havebetween 65and86%identical aminoacids(Table2), andthecelements havebetween 61and70%identical amino acids(Table3).Notably, thehighlyconserved a,b,andc elements areimmediately adjacent toeachotherinthepl5B, P1,P2,andXgenomes. Mostoftheaminoacidsequence of theXelement bshowninFig.3Bisdeduced froman intergenic regionbetween ORF401andgenestf.However, aninsertion ofasinglebasepairafterelement awithinORF TABLE 3.Percentages ofidentical (andsimilar) residues of celements alignedinFig.3C %ofidentical (%ofsimilar) residues with: Element pl5BScc PlScc P2Hc XStfc pl5BScc 100 P1Scc 69(81) 100 P2Hc 63(78) 69(80) 100 XStfc 61(76) 65(78) 70(81) 100 \"Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8).J.BACTECRIOL.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  EVOLUTION OFpl5BANDPHAGE P1TAILFIBER GENES 3941 TABLE 4.Percentages ofidentical (andsimilar) residues ofthe largeconserved domainofproteinsequences alignedinFig.5 %ofidentical (%ofsimilar)a residues with: Element pl5BSvore14 AStf T437T412 p1SBSvore14P(-)h 100 XStf 45(58) 100 T437 38(56) 74(82)100 T412 28(45) 27(52)28(50)100 aHomology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). \"Aconsensus sequence ofthehomologous Svportions ofpl5BandofP(-) ofe14(Fig.5)isused. 401resultsinacontinuous reading framefor774aa,which includes ORF401,stf,andtheirintergenic region. Since thesecodingsequences arecontained withinanonessential regionoftheAgenome, thisputative longreading frame mighthavebeendisrupted bymutation without lossofphage viability. Itisintriguing thatthe3\\'endsofthecelements inP2and XDNA,whichdonotcontaininvertible DNA,coincide with sequences thatresemble crossover sitesforDinrecombi- nases.IntheP2genome, thiscrossover site-like sequence separates theelement cfromsequences homologous tothose oftailfibergenescontained withintheinvertible segments of MuandP1(datanotshown), andintheXgenome the crossover site-like sequence separates theelement cfrom sequences homologous tothoseofthevariable genesofthe pl5Bande14invertible DNAsegments (seebelow). Itis therefore possible thatDinrecombinases alsomediate seg- mentalsubstitution apartfrominvertible DNAandcontrib- utesignificantly totheevolution oftailfibergenes. Twoconserved domains amongthesixvariable endsof thepl5BSproteinandtheirP(-)homolog inthePinsystem provedtobesimilartotwodomains oftheStfproteinofX, aswellastotwodomains intheproduct ofphageT4gene37 (Fig.5andTable4).(Thestronghomology between proteins Stfand37hasbeennoticedbefore[6,24].)TheproductofT4 gene12sharessomeoftheresidues conserved amongthe formerproteindomains (Fig.5),although overallsequence similarity isquitelow(Table4). 1 21TABLE 5.Percentages ofidentical (andsimilar) residues of proteinsequences alignedinFig.6 %ofidentical (%ofsimilar)a residues with: Elementpl5BT e14293 XTfa T438 pl5BT 100 e14293 75(80) 100 XTfa 37(53) 36(52) 100 T438 21(39) 20(39) 33(51)b 100 aHomology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). bT4protein38andXTfahaveaslightlyhigherhomology value(36%[55%l)iftheyareoptimally alignedtoeachotherinsteadofoptimizing alignment of atleastthreesequences. Thecounterparts oftheproducts ofthepl5BgeneTandof e14ORF293areTfaofphageXandthegene38proteinof phageT4(Fig.6).Thedegreeofconservation ofthose sequences islowbutstillsignificant, especially concerning similarratherthanidentical aminoacids(Table5). Genes37and38ofphageT4codeforproteins necessary forassembly ofthesixlongtailfibersoftheT4virion(28).', \"page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  bT4protein38andXTfahaveaslightlyhigherhomology value(36%[55%l)iftheyareoptimally alignedtoeachotherinsteadofoptimizing alignment of atleastthreesequences. Thecounterparts oftheproducts ofthepl5BgeneTandof e14ORF293areTfaofphageXandthegene38proteinof phageT4(Fig.6).Thedegreeofconservation ofthose sequences islowbutstillsignificant, especially concerning similarratherthanidentical aminoacids(Table5). Genes37and38ofphageT4codeforproteins necessary forassembly ofthesixlongtailfibersoftheT4virion(28). Twomolecules ofprotein37formthedistalportionofthese fibers,whichcontact thehostreceptors, whileprotein38, whichisrequired fordimerization ofprotein37,isnota constituent ofthefibers.ORF401ofphageXandgenesstf(sidetailfiber;formerly calledORF314)andtfa(tailfiber assembly; formerly calledORF194)arepartofitsnones- sentialbregionatthedistalendofthePR'operon(2).Ithas beenshownthatthepresence ofORF401andgenesstfand tfainXprophage DNAcorrelates withtheappearance ofthe nonessential sidetailfibersinmature virions (25).The product oftfapossesses acatalytic function infiberassem- bly,sincetfacancomplement anambermutation inthe homologous T4gene38(25).Theproduct ofstfdetermines receptor specificity, sincereplacement ofa3'endofgene37 ofT4bystfproduces T4virionswithalteredhostspecifici- ties(27). Thesequence homology ofthep15BgeneTandofthe constant andvariable partsofthepl5BgeneStoseveral bacteriophage genomes, together withtheearlierreported highdegreeofhomology ofalargepartofScgenesandthe preceding geneRofpl5Btothecorresponding genesofthe 41 61 80 (1)MDNILNELIAIQAGNIIV DGGNR YSASTE LAG.V (1)MHKILNDLIATKAGDVTV DGERYSTSNE LAG.V (1) ..MFRME....QPRTIKI LLAGNFGEGDA IPPH.T (1) MIHYFD KFYRE.EN KPKGLINCLDGSH .ILREDLS EP ICCLDGTH .ICRADFN EP LNTDIPDIP FVVFNDE.A HLE LHTIKRLEPE .VVFDERTQD IEE 81 101 121 141 160 G-TVY-----G----I---G.--PEN-T---P---YDKW-G--WV-DTEA--..------AE----SL------S-----L ESIDTNPEETVL.DY TTIALTP DEKVQH..SAAVEA TKRQIDTAMD ISLIQ ESTRTESKEKAP.DY TTIALSP DEKTQH..SAAVDA AQRQIDAAMA ISLIQ KDVASDALFSEL.PL FWLSGGEQNTAKEK..LFRIRE ETRK M..QVA EHIAP K.RAWTFNREEIF SDISPVGITFDEP...GEF ITDDG.KEDTYKRVLIRNRKI ELYK.EF..QV....... 161 181 201 217 KLRGRKL. QATQNSVLD IDENAMLT LNKQLSTAS* KLQGRRL.\", 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  QATQNSVLD IDENAMLT LNKQLSTAS* KLQGRRL. QATRNAVLDIDAVTAT TS VILPERRPFNIWRTGCI QDADLEIA RESLEAWKR RVLNRVTS IEAVPVME* NNMIEASVANKK KFYYKNLKRFFAL ERHEHLGGEFPS KEQKPWYRRLFKHYV*(203) (211) (194) (183) FIG.6.Alignment oftheproduct ofpl5BTwiththee14ORF293product, Tfaofphage X,andprotein 38ofphageT4.Forsymbols, see thelegend toFig.5.p15BT e14293 ATfa T438 cornse pl5BT e14293 XTfaT438 conenus p15BT e14293 XTfa T438VOL.174,1992', \"page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  3942 SANDMEIER ETAL. A B CFIG.7.Phageparticles obtained bycomplementation oftheP1tailfibermutantM47withpl5B-encoded tailfibers.Specimens were negatively stainedwith1%uranylacetate.Magnification bythetransmission electron microscope, x40,000; finalmagnification, x190,000. (A)Mutant M47prophage givesrisetoparticles without tailfiberstructures. (B)M47mutantcomplemented withtheP1tailfiberoperon on multicopy plasmid pRM206 isindistinguishable fromawild-type P1phage.(C)Complementation ofM47withthepl5BMinregionpresent onmulticopy plasmidpAW800 produces phageparticles withpl5B-encoded tailfibers. phageP1tailfiberoperon(Fig.4)(33),strongly suggests that theMinregioncontains tailfibergenes.Thebiological role ofthegenetic information contained inthePininvertible segment hasnotbeendetermined, butthehomologies of ORF293andP(-)toT4andKtailfibergenescouldsuggest thatthePinsystemisinvolved intailfibersynthesis. Complementation ofdefective P1prophage withtheMin regionofpl5B.Earlierelectron microscopy studieswiththe closelyrelatedP1andpl5Bgenomes (23)revealed thatthe majorpartsofthegenomic regionsencoding thetailgenes arehomologous. Furthermore, asdescribed above,theP1 amino-terminal sequences whichconfer totailfiberstheir attachment tothebaseplate (41)arealmostidentical to corresponding sequences inpl5B.Toinvestigate thefunc- tionalhomology ofthesegeneproducts, wehavetested whether amulticopy plasmid carrying thepl5BMinregion cancomplement P1mutantM47,whichisdefective fortail fibersynthesis. ThemutantM47carries anIS2insertion in the5'partoftheSgene.Multicopy plasmids pRM206 and pAW800 carry,respectively, theP1BamHI-5 fragment with thewild-type P1tailfiberoperon andthepl5BBamHI-5 fragment containing theputative tailfibergenesunderthe controlofthePspromoter (Fig.4).Ithasbeenshownthat thepl5BPspromoter isastructural andfunctional analogof theP1latepromoter Pstranscribing theP1tailfibergenes andthatitsactivitydepends onP1functions expressed after prophage induction (10).Hence, theputative tailfibergenes onpAW800 areexpected tobeexpressed attheappropriate timeinthepresence ofaninduced M47prophage. E.coliWA921 carrying theM47prophage wastrans- formed withpRM206 orpAW800, andlysates fromthe transformants wereprepared byheatinduction. Asseenin theelectron microscope, themorphologically intactphageparticles fromWA921(M47, pRM206) carriedtailfiberstruc- tures(Fig.7B),whileparticles fromWA921(M47) exhibited notailfibers(Fig.7A).Underthesamecondition, the majority ofmorphologically intactphagesproduced from heat-induced WA921(M47, pAW800) cellscarriedtailfiber structures (Fig.7C).However, theconcentration ofcom- pletephageparticles inbothlysatesofcellswiththetrans- complemented M47wasverylowcompared withthatof lysatesofcellswithM47orawild-type P1,andthemajority ofparticles wereemptyheadsandloosesheathstructures. SincetheMinregionappears toencodetailfibergenes, whoseproducts canattachtoaP1baseplate, wetested whether P1particles withpl5BtailfiberscouldinfectE.coli WA921(seeMaterials andMethods). Asexpected, noinfec- tiousparticles (fewerthan102PFU/ml) couldbedetected on WA921 indicator cellsforlysatesofWA921(M47) cells, whereas 106PFU/ml wasmeasured onWA921(pRM206) indicator cellsforlysatesofWA921(M47, pRM206) cells.\", 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  SincetheMinregionappears toencodetailfibergenes, whoseproducts canattachtoaP1baseplate, wetested whether P1particles withpl5BtailfiberscouldinfectE.coli WA921(seeMaterials andMethods). Asexpected, noinfec- tiousparticles (fewerthan102PFU/ml) couldbedetected on WA921 indicator cellsforlysatesofWA921(M47) cells, whereas 106PFU/ml wasmeasured onWA921(pRM206) indicator cellsforlysatesofWA921(M47, pRM206) cells. Sincearound10i\"PFU/ml isexpected withwild-type P1 prophages, theseresultsandtheelectron microscopy obser- vations confirm thattranscomplementation occurs, al- though quiteinefficiently. Inlysates ofWA921(M47, pAW800) cells,noplaques(fewerthan102PFU/ml) couldbe detected ontheWA921(pRM206) orWA921(pAW800) indi- catorstrain.Itislikelythatthepl5Btailfibergenesconfer hostspecificities otherthanthoseofP1andthatthepl5B geneproducts failedtointeract withE.coliK-12WA921. A searchforotherpotential hoststrainshasnotyetbeenmade. Functional aspectsoftailfibersandoftheirevolution. The pl5Btailfiberoperoncontains genesR,S,andT,underthe control ofthephage-specific latepromoter Ps(33).Most likelytheSvsequences determine thehostrange,andRand Scareamongthesequences responsible forattachment ofJ.BAC-TERIOL.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  EVOLUTION OFplSBANDPHAGE P1TAILFIBER GENES 3939 -=DC==>gin P.Sc Sv U U Sv FIG.4.Mapofhomologous DNAsequences amongtheDNA inversion regionsofe14,pl5B,P1,andMu.P1andp1SBregionsare represented bytheirBamHI-5 fragments. Invertible segments are shownasboxesflanked bycrossover sitesdepicted asblack triangles. Genesofoperons thatareaffected byDNAinversion and DNAinvertase genesareshownasopenarrows. TheP1cingeneis opposite inorientation tothatoftherelatedDNAinvertase genes. ThePspromoters oftailfibergenesareshownasshortblack arrows.Homology between pl5BandP1totheleftoftheinvertible segments issimplified fromthatshowninFig.2.Therefore, thea,b, andcelements composing theseregions arenotspecifically shown here.Thesiteofintegration ofIS2intoP1prophage M47ismarked. Theindicated highdegreeofhomology ofCinandGininvertible segments (20)isinterrupted byanapproximately 190-bpsegment of only35%homology aroundtheendofSvandthestartofUgenes. Thesmall63%homologous sequences inScofP1andMucorre- spondtotheaelements whicharealignedinFig.3A.The67-bp sequences between thecrossover pointsandtheDNAinvertase genesofpl5Bande14show31%homology. remarkable homologies inthesameorientation areseenon boththerightandtheleftsidesoftheinvertible segments andtheirrecombinase genes.Totheright,homologous DNA onP1startsstillwithinthecingeneandhence,thepl5B genome harbors aremnant fragment ofcin,thepseudo-cin gene(ilcin).Ithasbeenshownelsewhere thattlcinandthe recombinational enhancer sequence ontjcinarenonfunc- tional(33).Presence ofthetlcingeneindicates thattheDNA inversion regionofanancestral p15Bgenome hadcarrieda site-specific recombinase geneoftheDinfamilyinthesameconfiguration asthatoftheP1genome. Itismostlikelythat thecingeneandtheinvertible DNAofthisancestral pl5B genome hadbeensubstituted bytheMinsystemcomposed ofthemingeneandtheadjacent multiple invertible DNA segments. Sincethebreakpoints fromhomology tononho- mology between p15BandP1DNA aresharpandsomeof themcoincide withthecrossover sitesforDinrecombinases, itistempting tospeculate thataDinsite-specific recombi- nasehadmediated thesubstitution process. Thestructures ofrepeated elements neartheleftbreakpoints onp15Band P1suggest thatsubstitution mighthavebeenassociated with complicated recombinational processes. Alternatively, the presence ofthesestructures andthemultiple invertible DNA segments ofpl5Bcouldhaveresulted fromindependent events. Relation between pl5Bande14invertible DNAsegments. Comparison withthesequences ofotherDNAinversion systems revealed thatthefiveinvertible segments ofpl5B arerelated totheinvertible DNAofthePinsystemofe14.A 1.1-kb partofthe1.8-kbPininvertible segment (31)is between 60and80%homologous tothe3.5-kbMininvertible segments (Fig.4).ThesixSvcarboxy-terminal variable segments ofpl5BsharewiththeP(-)carboxy-terminal segment ofe14ahighlyhomologous domain ofabout70aa (Fig.5).When anucleotide isinserted afterposition 497of thepublished sequence ofthePinregionofe14toproduce a frameshift inP(-),thelast13aaofthepredicted P(-) product areidentical totheSvM1,SvP,andSvQandalmost identical totheSvMr,SvN,andSvOcarboxy-terminal segments (Fig.5).Thissecondhomologous domain issepa- ratedfromthefirstdomainby61to84aaintheSvandP(-) sequences.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  OnlytheSvMrandP(-)segments arehomolo- gousalsobetween the70-and13-aadomains (datanot shown). Thealternative P(+)carboxy-terminal segment of e14exhibits nosequence homology toP(-)ortopl5BSv segments. Thepostulated geneTofpl5B,two-thirds ofe14open reading frame(ORF)293,andtheirputative products have 75%oftheirbasesandaminoacidsincommon. Homology oftheMinandPinregionsincludes notonlythe invertible DNAsegments butalsotheadjacent pinandmin genes,whichhave72%identical basesand80%identical aminoacids.Preliminary sequence datashowed that0.6kb ofe14flanking theothersideofthePininvertible segment arenothomologous totheMinregion(32). Sincethepingeneistheclosest relative ofthemin recombinase gene(17)andthepl5Bande14invertible DNA TABLE 1.Percentages ofidentical (andsimilar) residues ofaelements alignedinFig.3A %ofidentical (%ofsimilar) residues with: Element5Bcal pl5BScpl5BSca3pl5BSc PSa uc 2a 2a 4l EeplSB Scala2andaS anda6b a4anda7PlScalMuScaP2HalP2Ha2i401a pl5BScal 100 pl5BSca2andaS 89(91) 100 pl5BSca3anda6fi 90(92) 82(82) 100 pl5BSca4anda7 93(95) 91(91) 87(87) 100 P1Scal 82(86) 77(82) 69(77) 75(82) 100 MuSca 61(70) 55(66) 64(77) 57(68) 57(68) 100 P2Hal 80(82) 75(77) 86(89) 75(77) 66(73) 61(73) 100 P2Ha2 77(77) 75(77) 59(61) 80(82) 73(80) 59(70 64(66) 100 X401a 68(73) 68(70) 72(72) 68(70) 64(75) 57(77) 66(68) 80(86) 100 Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). hThefirst17aaofpl5Ba3anda6havenotbeenconsidered forthecalculation ofhomology values.VOL.174,1992', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  3940 SANDMEIER ETAL. consensus pl5BSvM1 plSBSvMr P15BSVN pl5BSvO pl5BSvP P15BSVQ e14P(-) XStf T437 T412 consensus pl5BSvMl pl5BSvMr P1SBSVN plSBSvO plSBSvP P15BSVQ e14P(-) XStf T437 T4121 21 41 61 80 GLGEGSALPVGVPVPWPSATPPT. GWLKCNGAAFSSE-YPKLAK ....... AYPTNK. LPDLRGEFIRGWDDGRGVD-GR (1) (1) (1) (1) (1) (1) (1) (63) (806) (324)LE . M ....... KN........ L. LE . M ....... K N........ V L. M ....... KN....... AEE....... A N FAAI DIV S.YVLMQ DKSA .... SGV. KTDIM SYIAIDSVA.FALME QTDKSA V SGV.I QNEIDRTI AIMMAADSL SDARFHGTVASDCLYSRIGTRYGGN SPG 81 101 116 --LS-Q--- ......---H-H-- VI IGWL ...TGS YNI SI EGYA ...TED AGL AL NW PHT ..ILD AYM SL GTAT FIRTALQD ALITGM LEKRIV QL SGDA IRNIEGFA SIIGYA TEDAGL AV QEQDG IKS TSA AV AEADG. VKA SSA VNGNDFGKPRLGVGCTGGYVGEVQIQQMSY KAG(88) (88) (88) (88) (88) (88) (88) (145) (887) (439)(132) (147) (149) (149) (143) (135) (149) (295) (1007) (497)A Is A SIT A A IT NWTKKPAS..... MQTKKP.S..... MLVSGRSHLTNPN 1 21 ----E-RPRNIAFNYIVRAA* SGGNT *(153) TAGLT D*(166) NVGTT V *(168) GAWISM V T*(168) TGGST *(159) PTANN *(145) AAASAX *(165) AGNANTVK L*(314) TGNTNTVK L*(1026) LIGNTWSLIKVKE* (517) FIG.5.Alignment oftwoconserved domains amongtheSvvariable portions ofthepl5BSprotein, oftheP(-)variable partcodedinthe Pininversion systemofe14,oftheStfproteinofphageA,ofprotein37ofphageT4,andofprotein12ofphageT4.Xintheseconddomain ofthee14sequence identifies thesiteatwhich asinglebasepairwasintroduced toallowalignment ofthecarboxy-terminal ends.Asterisks areappended tothecarboxy-terminal endsoftheproteins. Forothersymbols, seethelegend toFig.3. segments arehighlyhomologous, theDNAwhichhasre- placedtheDNAinversion regionoftheancestral p15B genome musthavebeencloselyrelatedtothee14Pinregion. Asthediagram ofFig.4shows,severalofthebreakpoints between homology andnonhomology amongthePinand Minsystems andtheCinandGinsystems, respectively, also coincide withcrossover sitesforDinrecombinases. The breakpoints oftheDNAinversion regionsofShigella boydii anditsclosestrelative, theGinsystemofphageMu,whose invertible segments have62to98%identical bases,also coincide withthecrossover sitesonbothgenomes (39).This supports anearlierpostulate thatDinrecombinases could mediate segmental substitutions amongDNAinversion sys- tems(12,16).Otherbreakpoints whichexhibit noobvious similarity toconsensus crossover sitescouldnevertheless oncehavebeenusedassubstrates forsuchrecombinases.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Asthediagram ofFig.4shows,severalofthebreakpoints between homology andnonhomology amongthePinand Minsystems andtheCinandGinsystems, respectively, also coincide withcrossover sitesforDinrecombinases. The breakpoints oftheDNAinversion regionsofShigella boydii anditsclosestrelative, theGinsystemofphageMu,whose invertible segments have62to98%identical bases,also coincide withthecrossover sitesonbothgenomes (39).This supports anearlierpostulate thatDinrecombinases could mediate segmental substitutions amongDNAinversion sys- tems(12,16).Otherbreakpoints whichexhibit noobvious similarity toconsensus crossover sitescouldnevertheless oncehavebeenusedassubstrates forsuchrecombinases. It hasbeenshownelsewhere thatDNAinvertases canutilize secondary crossover siteswithquitedegenerate sequences (14,15),andsuchsitesmighthavebeenmutated during evolution subsequent toasubstitution event. Homologous domains ofgenescontained inpl5B,P1,Mu, ande14DNAinversion regionsandintailfibergenesofphages T4,P2,andX.Screening ofnucleic acidandaminoacid sequence databasesrevealed unexpected relationships amongthecodingsequences ofDNAinversion systems of TABLE 2.Percentages ofidentical (andsimilar) residues of belements alignedinFig.3B %ofidentical (%ofsimilar)\" residues with: Element X401-StfPlScblPlScb2 P2Hb b P1Scbl 100 P1Scb2 68(81) 100 P2Hb 68(82) 84(89) 100 X401-Stfb 65(81) 77(86) 86(92) 100 \"Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8).pl5B,e14,andphageP1andgenesofotherwise unrelated bacteriophages. Thepartlyrelatedandrepeated elements inthepl5Band P1Scgeneshadcounterparts intailfibergenesofphages Mu,P2,andX.Segments homologous tothepl5BandP1 elements designated awerefoundinduplicate inthephage P2geneHandinsinglecopiesintheconstant part,Sc,ofthe MutailfibergeneS,andinXORF401.These44-aaelements present onfivedistinctgenomes havebetween 55and93% identical aminoacids(Fig.3A;Table1).Theclosestrela- tionships, withatleast80%identical aminoacids,are between thepl5Belements altoa7amongeachother,the element alofP1andalofpl5B,theelement alofP2andal, a3,anda6ofpl5B,andtheelement a2ofP2anda4anda7 ofpl5Bandthecorresponding Xsequence. Thetandemrepeated elements blandb2oftheP1genome andthecelements inP1andpl5Barerelatedtosequences intheP2andXgenomes (Fig.3BandC).Thebelements havebetween 65and86%identical aminoacids(Table2), andthecelements havebetween 61and70%identical amino acids(Table3).Notably, thehighlyconserved a,b,andc elements areimmediately adjacent toeachotherinthepl5B, P1,P2,andXgenomes.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Thetandemrepeated elements blandb2oftheP1genome andthecelements inP1andpl5Barerelatedtosequences intheP2andXgenomes (Fig.3BandC).Thebelements havebetween 65and86%identical aminoacids(Table2), andthecelements havebetween 61and70%identical amino acids(Table3).Notably, thehighlyconserved a,b,andc elements areimmediately adjacent toeachotherinthepl5B, P1,P2,andXgenomes. Mostoftheaminoacidsequence of theXelement bshowninFig.3Bisdeduced froman intergenic regionbetween ORF401andgenestf.However, aninsertion ofasinglebasepairafterelement awithinORF TABLE 3.Percentages ofidentical (andsimilar) residues of celements alignedinFig.3C %ofidentical (%ofsimilar) residues with: Element pl5BScc PlScc P2Hc XStfc pl5BScc 100 P1Scc 69(81) 100 P2Hc 63(78) 69(80) 100 XStfc 61(76) 65(78) 70(81) 100 \"Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8).J.BACTECRIOL.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  EVOLUTION OFpl5BANDPHAGE P1TAILFIBER GENES 3941 TABLE 4.Percentages ofidentical (andsimilar) residues ofthe largeconserved domainofproteinsequences alignedinFig.5 %ofidentical (%ofsimilar)a residues with: Element pl5BSvore14 AStf T437T412 p1SBSvore14P(-)h 100 XStf 45(58) 100 T437 38(56) 74(82)100 T412 28(45) 27(52)28(50)100 aHomology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). \"Aconsensus sequence ofthehomologous Svportions ofpl5BandofP(-) ofe14(Fig.5)isused. 401resultsinacontinuous reading framefor774aa,which includes ORF401,stf,andtheirintergenic region. Since thesecodingsequences arecontained withinanonessential regionoftheAgenome, thisputative longreading frame mighthavebeendisrupted bymutation without lossofphage viability. Itisintriguing thatthe3\\'endsofthecelements inP2and XDNA,whichdonotcontaininvertible DNA,coincide with sequences thatresemble crossover sitesforDinrecombi- nases.IntheP2genome, thiscrossover site-like sequence separates theelement cfromsequences homologous tothose oftailfibergenescontained withintheinvertible segments of MuandP1(datanotshown), andintheXgenome the crossover site-like sequence separates theelement cfrom sequences homologous tothoseofthevariable genesofthe pl5Bande14invertible DNAsegments (seebelow). Itis therefore possible thatDinrecombinases alsomediate seg- mentalsubstitution apartfrominvertible DNAandcontrib- utesignificantly totheevolution oftailfibergenes. Twoconserved domains amongthesixvariable endsof thepl5BSproteinandtheirP(-)homolog inthePinsystem provedtobesimilartotwodomains oftheStfproteinofX, aswellastotwodomains intheproduct ofphageT4gene37 (Fig.5andTable4).(Thestronghomology between proteins Stfand37hasbeennoticedbefore[6,24].)TheproductofT4 gene12sharessomeoftheresidues conserved amongthe formerproteindomains (Fig.5),although overallsequence similarity isquitelow(Table4). 1 21TABLE 5.Percentages ofidentical (andsimilar) residues of proteinsequences alignedinFig.6 %ofidentical (%ofsimilar)a residues with: Elementpl5BT e14293 XTfa T438 pl5BT 100 e14293 75(80) 100 XTfa 37(53) 36(52) 100 T438 21(39) 20(39) 33(51)b 100 aHomology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). bT4protein38andXTfahaveaslightlyhigherhomology value(36%[55%l)iftheyareoptimally alignedtoeachotherinsteadofoptimizing alignment of atleastthreesequences. Thecounterparts oftheproducts ofthepl5BgeneTandof e14ORF293areTfaofphageXandthegene38proteinof phageT4(Fig.6).Thedegreeofconservation ofthose sequences islowbutstillsignificant, especially concerning similarratherthanidentical aminoacids(Table5). Genes37and38ofphageT4codeforproteins necessary forassembly ofthesixlongtailfibersoftheT4virion(28).', \"page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  bT4protein38andXTfahaveaslightlyhigherhomology value(36%[55%l)iftheyareoptimally alignedtoeachotherinsteadofoptimizing alignment of atleastthreesequences. Thecounterparts oftheproducts ofthepl5BgeneTandof e14ORF293areTfaofphageXandthegene38proteinof phageT4(Fig.6).Thedegreeofconservation ofthose sequences islowbutstillsignificant, especially concerning similarratherthanidentical aminoacids(Table5). Genes37and38ofphageT4codeforproteins necessary forassembly ofthesixlongtailfibersoftheT4virion(28). Twomolecules ofprotein37formthedistalportionofthese fibers,whichcontact thehostreceptors, whileprotein38, whichisrequired fordimerization ofprotein37,isnota constituent ofthefibers.ORF401ofphageXandgenesstf(sidetailfiber;formerly calledORF314)andtfa(tailfiber assembly; formerly calledORF194)arepartofitsnones- sentialbregionatthedistalendofthePR'operon(2).Ithas beenshownthatthepresence ofORF401andgenesstfand tfainXprophage DNAcorrelates withtheappearance ofthe nonessential sidetailfibersinmature virions (25).The product oftfapossesses acatalytic function infiberassem- bly,sincetfacancomplement anambermutation inthe homologous T4gene38(25).Theproduct ofstfdetermines receptor specificity, sincereplacement ofa3'endofgene37 ofT4bystfproduces T4virionswithalteredhostspecifici- ties(27). Thesequence homology ofthep15BgeneTandofthe constant andvariable partsofthepl5BgeneStoseveral bacteriophage genomes, together withtheearlierreported highdegreeofhomology ofalargepartofScgenesandthe preceding geneRofpl5Btothecorresponding genesofthe 41 61 80 (1)MDNILNELIAIQAGNIIV DGGNR YSASTE LAG.V (1)MHKILNDLIATKAGDVTV DGERYSTSNE LAG.V (1) ..MFRME....QPRTIKI LLAGNFGEGDA IPPH.T (1) MIHYFD KFYRE.EN KPKGLINCLDGSH .ILREDLS EP ICCLDGTH .ICRADFN EP LNTDIPDIP FVVFNDE.A HLE LHTIKRLEPE .VVFDERTQD IEE 81 101 121 141 160 G-TVY-----G----I---G.--PEN-T---P---YDKW-G--WV-DTEA--..------AE----SL------S-----L ESIDTNPEETVL.DY TTIALTP DEKVQH..SAAVEA TKRQIDTAMD ISLIQ ESTRTESKEKAP.DY TTIALSP DEKTQH..SAAVDA AQRQIDAAMA ISLIQ KDVASDALFSEL.PL FWLSGGEQNTAKEK..LFRIRE ETRK M..QVA EHIAP K.RAWTFNREEIF SDISPVGITFDEP...GEF ITDDG.KEDTYKRVLIRNRKI ELYK.EF..QV....... 161 181 201 217 KLRGRKL. QATQNSVLD IDENAMLT LNKQLSTAS* KLQGRRL.\", 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  QATQNSVLD IDENAMLT LNKQLSTAS* KLQGRRL. QATRNAVLDIDAVTAT TS VILPERRPFNIWRTGCI QDADLEIA RESLEAWKR RVLNRVTS IEAVPVME* NNMIEASVANKK KFYYKNLKRFFAL ERHEHLGGEFPS KEQKPWYRRLFKHYV*(203) (211) (194) (183) FIG.6.Alignment oftheproduct ofpl5BTwiththee14ORF293product, Tfaofphage X,andprotein 38ofphageT4.Forsymbols, see thelegend toFig.5.p15BT e14293 ATfa T438 cornse pl5BT e14293 XTfaT438 conenus p15BT e14293 XTfa T438VOL.174,1992', \"page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  3942 SANDMEIER ETAL. A B CFIG.7.Phageparticles obtained bycomplementation oftheP1tailfibermutantM47withpl5B-encoded tailfibers.Specimens were negatively stainedwith1%uranylacetate.Magnification bythetransmission electron microscope, x40,000; finalmagnification, x190,000. (A)Mutant M47prophage givesrisetoparticles without tailfiberstructures. (B)M47mutantcomplemented withtheP1tailfiberoperon on multicopy plasmid pRM206 isindistinguishable fromawild-type P1phage.(C)Complementation ofM47withthepl5BMinregionpresent onmulticopy plasmidpAW800 produces phageparticles withpl5B-encoded tailfibers. phageP1tailfiberoperon(Fig.4)(33),strongly suggests that theMinregioncontains tailfibergenes.Thebiological role ofthegenetic information contained inthePininvertible segment hasnotbeendetermined, butthehomologies of ORF293andP(-)toT4andKtailfibergenescouldsuggest thatthePinsystemisinvolved intailfibersynthesis. Complementation ofdefective P1prophage withtheMin regionofpl5B.Earlierelectron microscopy studieswiththe closelyrelatedP1andpl5Bgenomes (23)revealed thatthe majorpartsofthegenomic regionsencoding thetailgenes arehomologous. Furthermore, asdescribed above,theP1 amino-terminal sequences whichconfer totailfiberstheir attachment tothebaseplate (41)arealmostidentical to corresponding sequences inpl5B.Toinvestigate thefunc- tionalhomology ofthesegeneproducts, wehavetested whether amulticopy plasmid carrying thepl5BMinregion cancomplement P1mutantM47,whichisdefective fortail fibersynthesis. ThemutantM47carries anIS2insertion in the5'partoftheSgene.Multicopy plasmids pRM206 and pAW800 carry,respectively, theP1BamHI-5 fragment with thewild-type P1tailfiberoperon andthepl5BBamHI-5 fragment containing theputative tailfibergenesunderthe controlofthePspromoter (Fig.4).Ithasbeenshownthat thepl5BPspromoter isastructural andfunctional analogof theP1latepromoter Pstranscribing theP1tailfibergenes andthatitsactivitydepends onP1functions expressed after prophage induction (10).Hence, theputative tailfibergenes onpAW800 areexpected tobeexpressed attheappropriate timeinthepresence ofaninduced M47prophage. E.coliWA921 carrying theM47prophage wastrans- formed withpRM206 orpAW800, andlysates fromthe transformants wereprepared byheatinduction. Asseenin theelectron microscope, themorphologically intactphageparticles fromWA921(M47, pRM206) carriedtailfiberstruc- tures(Fig.7B),whileparticles fromWA921(M47) exhibited notailfibers(Fig.7A).Underthesamecondition, the majority ofmorphologically intactphagesproduced from heat-induced WA921(M47, pAW800) cellscarriedtailfiber structures (Fig.7C).However, theconcentration ofcom- pletephageparticles inbothlysatesofcellswiththetrans- complemented M47wasverylowcompared withthatof lysatesofcellswithM47orawild-type P1,andthemajority ofparticles wereemptyheadsandloosesheathstructures. SincetheMinregionappears toencodetailfibergenes, whoseproducts canattachtoaP1baseplate, wetested whether P1particles withpl5BtailfiberscouldinfectE.coli WA921(seeMaterials andMethods). Asexpected, noinfec- tiousparticles (fewerthan102PFU/ml) couldbedetected on WA921 indicator cellsforlysatesofWA921(M47) cells, whereas 106PFU/ml wasmeasured onWA921(pRM206) indicator cellsforlysatesofWA921(M47, pRM206) cells.\", 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  SincetheMinregionappears toencodetailfibergenes, whoseproducts canattachtoaP1baseplate, wetested whether P1particles withpl5BtailfiberscouldinfectE.coli WA921(seeMaterials andMethods). Asexpected, noinfec- tiousparticles (fewerthan102PFU/ml) couldbedetected on WA921 indicator cellsforlysatesofWA921(M47) cells, whereas 106PFU/ml wasmeasured onWA921(pRM206) indicator cellsforlysatesofWA921(M47, pRM206) cells. Sincearound10i\"PFU/ml isexpected withwild-type P1 prophages, theseresultsandtheelectron microscopy obser- vations confirm thattranscomplementation occurs, al- though quiteinefficiently. Inlysates ofWA921(M47, pAW800) cells,noplaques(fewerthan102PFU/ml) couldbe detected ontheWA921(pRM206) orWA921(pAW800) indi- catorstrain.Itislikelythatthepl5Btailfibergenesconfer hostspecificities otherthanthoseofP1andthatthepl5B geneproducts failedtointeract withE.coliK-12WA921. A searchforotherpotential hoststrainshasnotyetbeenmade. Functional aspectsoftailfibersandoftheirevolution. The pl5Btailfiberoperoncontains genesR,S,andT,underthe control ofthephage-specific latepromoter Ps(33).Most likelytheSvsequences determine thehostrange,andRand Scareamongthesequences responsible forattachment ofJ.BAC-TERIOL.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:03 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'125'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994672'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'319ms'), (b'x-request-id', b'b7372daa1826fb3a1e8262ec3cd49dbd'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249ab6eec5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:03 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'125'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994672'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'319ms'), (b'x-request-id', b'b7372daa1826fb3a1e8262ec3cd49dbd'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249ab6eec5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': [\"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  EVOLUTION OFpl5BANDPHAGE P1TAILFIBERGENES 3943 D. P2H ARF>ala2bc MuS a PiS ~11 IaOalblb2c p15BS-|aOaa2a3a4a5a6 c X401-stf-| 500bp abc FIG.8.Arrangement ofhomologous elements intheintermedi- atepartoflongtailfibergenes.Genesaredepicted inthe5'to3' direction, andhomologous regions areboxed.Thecorresponding aminoacidsequences ofthea,b,andcelements arealignedinFig. 3.Homologous sequences amongthe3'domains (Sv)ofthepl5B geneSandX401-stfareshowninFig.5,whilethoseofthe3' domains oftheP2geneH,MugeneS,andP1geneSarenotshown. TheXtailfibergeneisrepresented asafusionofORF401andgene stf(seetext).DNAinversion crossover sitesandcrossover site-like sequences areshownasblackandopentriangles, respectively. thetailfiberstothebaseplate ofthevirion.TheproductofT probably possesses acatalytic roleintailfiberassembly of oneofthesixfibervariants. Expression ofTdepends onthe configuration oftheinvertible segments intheMinregion, anditismostlikelytranslationally coupled toSvMJ(33). SinceMininversion alternatively provides expression of eachofthesixvariants ofS(33),sixdifferent tailfibers,each encoding adistincthostrange,couldbesynthesized. How- ever,itisexpected thatindividual phageparticles would usually carrypl5Btailfibersofonlyonehostspecificity corresponding tothegeneticconfiguration oftheparental genome. Thephagesproduced duringlyticgrowthofphage P1conferpredominantly oneofthetwopossible host specificities (13),andDNAinversions byCinaswellasby Minrecombinases arelow-frequency events. Thebiological significance ofamultiple extension ofthe hostphagesuchasthatdisplayed bypl5Bisuncertain. Although itissufficient formostphagestodetermine only onehostrange,itcouldallowaphagetomorereadilyadapt tochanging environments composed ofseveral bacterial species, oraphagecouldevadeadsorption-resistant host mutations bybinding toalternate receptors. Itisstriking thatthelargetailfibergenesofseveral unrelated, well-studied phagesofE.colicarrydomains of highlysimilarsequences. Suchhomologous domains are observed amongthe3'ends(e.g.,amonggeneSofP1,Sof Mu,andHofP2andamongSvgenesofpl5B,gene37ofT4, andstfofX)whichconferhostspecificity, whilenohomol- ogyhassofarbeendetected amongthe5'ends,which specifytheattachment tothebaseplates.Itisshownhere thattheintermediate partsofthetailfibergenesofsome phagesarearrangements ofseveralshortelements (Fig.8). Threetypesofsuchshortelements (a,b,andc),each presentinmorethanonegenome, havebeendetected among theavailable sequences. Theelements maybeamplified(e.g.,altoa7inp15BgeneS,alanda2inP2geneH,andblandb2inP1geneS).Theelements presentinagenome appearinthesameorder,i.e.,abeforebbeforec.The function oftheseelements duringtheinfection process isnot known. Theintermediate region,withitsmosaiclike struc- ture,isexpected tooccupy acorresponding position onthe tailfiberprotein, sincethetertiary structure oftailfiber proteins canbeconsidered largelylinear.Thiscanserveas abasisforstudiesoftheirroleininteractions between tail fibersandmacromolecules onthesurfaceofthehostbacte- rium,including reversible andirreversible adsorption. Thedatapresented abovegiveaninsightintothelong- termevolution ofgenes,whichareunderstrongselective pressure.\", \"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Theintermediate region,withitsmosaiclike struc- ture,isexpected tooccupy acorresponding position onthe tailfiberprotein, sincethetertiary structure oftailfiber proteins canbeconsidered largelylinear.Thiscanserveas abasisforstudiesoftheirroleininteractions between tail fibersandmacromolecules onthesurfaceofthehostbacte- rium,including reversible andirreversible adsorption. Thedatapresented abovegiveaninsightintothelong- termevolution ofgenes,whichareunderstrongselective pressure. Obviously, otherwise unrelated phagesexchanged shortdomains byhorizontal transfer whichresulted in chimeric tailfibergenes.Thepresence ofcrossover sitesand crossover site-like sequences forDNAinvertases between 3' andintermediate domains suggests adecisive roleforsite- specificrecombination inevolutionary reshuffling. ACKNOWLElDGMENTS WethankMichelWurtzforelectron microscopy andReinhard Doelzformaintaining anexcellent computer software environment. Thisworkwassupported bygrant3.624.087 oftheSwissNational Science Foundation. ADDENDUM Inanindependent study,E.Haggard-Ljungquist and coauthors (11)carriedoutacomplementary comparison of tailfibergenes,andtheyreached similarconclusions. REFERENCES 1.Arber,W.Unpublished data. la.Brody, H.,A.Greener, andC.W.Hill.1985.Excision and reintegration oftheEscherichia coliK-12chromosomal element e14.J.Bacteriol. 161:1112-1117. 2.Daniels, D.L.,J.L.Schroeder, W.Szybalski, F.Sanger, and F.R.Blattner. 1983.Amolecular mapofcoliphage lambda, p. 469-517. InR.W.Hendrix, J.W.Roberts, F.W.Stahl,and R.A.Weisberg (ed.),Lambda II.ColdSpringHarborLabora- tory,ColdSpringHarbor, N.Y. 3.Daniels, D.L.,J.L.Schroeder, W.Szybalski, F.Sanger, A.R. Coulson, G.F.Hong,D.F.Hill,G.B.Petersen, andF.R. Blattner. 1983.Complete annotated lambda sequence, p.519- 676.InR.W.Hendrix, J.W.Roberts, F.W.Stahl,andR.A. Weisberg (ed.),Lambda II.ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 4.Dayhoff M.O.,(ed.).1979.Atlasofprotein sequence and structure. National Biomedical Research Foundation, Washing- ton,D.C. 5.Devereux, J.,P.Haeberli, and0.Smithies. 1984.Acomprehen- sivesetofsequence analysis programs fortheVAX.Nucleic AcidsRes.12:387-395. 6.George, D.G.,L.-S.L.Yeh,andW.C.Barker. 1983.Unex- pectedrelationships between bacteriophage lambdahypotheti- calproteins andbacteriophage T4tail-fiber proteins. Biochem. Biophys. Res.Commun. 115:1061-1068. 7.Glasgow, A.C.,K.T.Hughes, andM.I.Simon.1989.Bacterial DNAinversion systems, p.637-659. InD.E.BergandM.M. Howe(ed.),MobileDNA.American Society forMicrobiology, Washington, D.C. 8.Gribskov, M.,andR.R.Burgess. 1986.SigmafactorsfromE. coli,B.subtilis,phageSPOl,andphageT4arehomologousproteins. Nucleic AcidsRes.14:6745-6763.9.Guidolin, A.,J.-M.Zingg,andW.Arber.1989.Organization of thebacteriophage P1tail-fibre operon. Gene76:239-243. 10.Guidolin, A.,J.-M.Zingg,H.Lehnherr, andW.Arber.1989. Bacteriophage P1tail-fibre anddaroperons areexpressed from homologous phage-specific latepromoter sequences. J.Mol.VOL.174,1992\", 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  3944 SANDMEIER ETAL. Biol.208:615-622. 11.Haggard-Ljungquist, E.,C.Hailing, andR.Calendar. 1992. DNAsequence ofthetailfibergenesofbacteriophage P2: evidence forhorizontal transfer oftailfibergenesamongunre- latedbacteriophages. J.Bacteriol. 174:1462-1477. 12.Hiestand-Nauer, R.,andS.lida.1983.Sequence ofthesite- specificrecombinase genecinandofitssubstrates servinginthe inversion oftheCsegment ofbacteriophage P1.EMBO J. 2:1733-1740. 13.lida,S.1984.Bacteriophage P1carriestworelatedsetsofgenes determining itshostrangeintheinvertible Csegment ofits genome. Virology 134:421-434. 14.lida,S.,andR.Hiestand-Nauer. 1986.Localized conversion at thecrossover sequences inthesite-specific DNAinversion systemofbacteriophage P1.Cell45:71-79. 15.Iida,S.,andR.Hiestand-Nauer. 1987.Roleofthecentral dinucleotide atthecrossover sitesfortheselection ofquasisites inDNAinversion mediated bythesite-specific Cinrecombinase ofphageP1.Mol.Gen.Genet.208:464-468. 15a.Iida, S.,andR.Hiestand-Nauer. Unpublished data. 16.lida,S.,H.Huber, R.Hiestand-Nauer, J.Meyer,T.A.Bickle, andW.Arber.1984.Thebacteriophage P1site-specific recom- binaseCin:recombination eventsandDNArecognition se- quences. ColdSpringHarborSymp.Quant.Biol.49:769-777. 17.lida,S.,H.Sandmeier, P.Hubner, R.Hiestand-Nauer, K. Schneitz, andW.Arber.1990.TheMinDNAinversion enzyme ofplasmidpl5BofEscherichia coli15T-:anewmember ofthe Dinfamilyofsite-specific recombinases. Mol.Microbiol. 4:991- 997. 18.Ikeda,H.,M.Inuzuka, andJ.-I.Tomizawa. 1970.P1-like plasmid inEscherichia coli15.J.Mol.Biol.50:457-470. 19.Kahmann, R.,andD.Kamp.1987.Sequence oftherightendof Mu,p.297-308. InN.Symonds, A.Toussaint, P.vandePutte, andM.M.Howe(ed.),PhageMu.ColdSpringHarborLabo- ratory,ColdSpringHarbor, N.Y. 20.Kamp, D.,E.Kardas, W.Ritthaler, R.Sandulache, R. Schmucker, andB.Stern.1984.Comparative analysis ofinvert- ibleDNAinphagegenomes. ColdSpringHarborSymp.Quant. Biol.49:301-311. 21.Kutsukake, K.,T.Nakao,andT.Iino.1985.AgeneforDNA invertase andaninvertible DNAinEscherichia coliK-12.Gene 34:343-350. 22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 1982.Molecular cloning: alaboratory manual. ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 23.Meyer,J.,M.Stalhammar-Carlemalm, M.Streiff,S.lida,and W.Arber.1986.Sequence relations amongtheIncYplasmid plSB,P1,andP7prophages. Plasmid 16:81-89. 24.Michel,C.J.,B.Jaqc,D.G.Arques, andT.A.Bickle.1986.A remarkable aminoacidsequence homology between aphageT4 tailfibreprotein andORF314 ofphageXlocated inthetail operon. Gene44:147-150. 25.Montag, D.,andU.Henning.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 1982.Molecular cloning: alaboratory manual. ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 23.Meyer,J.,M.Stalhammar-Carlemalm, M.Streiff,S.lida,and W.Arber.1986.Sequence relations amongtheIncYplasmid plSB,P1,andP7prophages. Plasmid 16:81-89. 24.Michel,C.J.,B.Jaqc,D.G.Arques, andT.A.Bickle.1986.A remarkable aminoacidsequence homology between aphageT4 tailfibreprotein andORF314 ofphageXlocated inthetail operon. Gene44:147-150. 25.Montag, D.,andU.Henning. 1987.Anopenreadingframeinthe Escherichia colibacteriophage Xgenome encodes aproteinthat functions inassembly ofthelongtailfibersofbacteriophage T4.J.Bacteriol. 169:5884-5886. 26.Montag, D.,I.Riede,M.-L.Eschbach, M.Degen, andU. Henning. 1987.Receptor-recognizing proteins ofT-even type bacteriophages. Constant andhypervariable regions andan unusual caseofevolution. J.Mol.Biol.196:165-174. 27.Montag, D.,H.Schwarz, andU.Henning. 1989.Acomponent of thesidetailfiberofEscherichia colibacteriophage Xcan functionally replace thereceptor-recognizing partofalongtail fiberprotein oftheunrelated bacteriophage T4.J.Bacteriol. 171:4378-4384. 28.Mosig,G.,andF.Eiserling. 1988.PhageT4structure and metabolism, p.521-606. InR.Calendar (ed.),Thebacterio- phages. Plenum Press,NewYork. 29.Mural,R.J.,R.H.Chesney, D.Vapnek, M.M.Kropf,andJ.R. Scott.1979.Isolation andcharacterization ofclonedfragments ofbacteriophage P1DNA.Virology 93:387-397. 30.Oliver,D.B.,andR.A.Crowther. 1981.DNAsequence ofthe tailfibregenes36and37ofbacteriophage T4.J.Mol.Biol. 153:545-568. 31.Plasterk, R.H.A.,andP.vandePutte.1985.Theinvertible P-DNAsegment inthechromosome ofEscherichia coli.EMBO J.4:237-242. 32.Sandmeier, H.1990.Characterization ofacomplex DNAinver- sionsystem. Ph.D.thesis.BaselUniversity, Basel. 33.Sandmeier, H.,S.lida,P.Hubner, R.Hiestand-Nauer, andW. Arber.1991.Geneorganization inthemultiple DNAinversion regionMinofplasmid plSBofE.coli15T-:assemblage ofa variable gene.Nucleic AcidsRes.19:5831-5838. 34.Sandmeier, H.,S.lida,J.Meyer, R.Hiestand-Nauer, andW. Arber.1990.Site-specific DNArecombination systemMinof plasmid pl5B:aclusterofoverlapping invertible DNAseg- ments.Proc.Natl.Acad.Sci.USA87:1109-1113. 35.Sanger, F.,S.Nicklen, andA.R.Coulson. 1977.DNAsequenc- ingwithchain-terminating inhibitors. Proc.Natl.Acad.Sci. USA74:5463-5467. 36.Selivanov, N.A.,A.G.Prilipov, andV.V.Mesyanzhinov. 1988. Nucleotide anddeduced aminoacidsequence ofbacteriophage T4gene12.Nucleic AcidsRes.16:2334. 37.Sengstag, C.,andW.Arber.1983.IS2insertion isamajorcause ofspontaneous mutagenesis ofthebacteriophage P1:non- random distribution oftargetsites.EMBOJ.2:67-71.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  35.Sanger, F.,S.Nicklen, andA.R.Coulson. 1977.DNAsequenc- ingwithchain-terminating inhibitors. Proc.Natl.Acad.Sci. USA74:5463-5467. 36.Selivanov, N.A.,A.G.Prilipov, andV.V.Mesyanzhinov. 1988. Nucleotide anddeduced aminoacidsequence ofbacteriophage T4gene12.Nucleic AcidsRes.16:2334. 37.Sengstag, C.,andW.Arber.1983.IS2insertion isamajorcause ofspontaneous mutagenesis ofthebacteriophage P1:non- random distribution oftargetsites.EMBOJ.2:67-71. 38.Sengstag, C.,J.C.W.Shepherd, andW.Arber.1983.The sequence ofthebacteriophage P1genome regionserving ashot targetforIS2insertion. EMBOJ.2:1777-1781. 39.Tominaga, A.,S.Ikemizu, andM.Enomoto. 1991.Site-specific recombinase genesinthreeShigella subgroups andnucleotide sequences ofapinBgeneandaninvertible Bsegment from Shigella boydii.J.Bacteriol. 173:4079-4087. 40.Wood, W.B.1966.Hostspecificity ofDNAproduced by Escherichia coli:bacterial mutations affecting therestriction andmodification ofDNA.J.Mol.Biol.16:118-133. 41.Yarmolinsky, M.B.,andN.Sternberg. 1988.Bacteriophage P1, p.291-438. InR.Calendar (ed.),Thebacteriophages. Plenum Press,NewYork.J.BACTERIOL.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Vol.45,No.3 JOURNAL OFVIROLOGY, Mar.1983,p.1118-1139 0022-538X/83/031118-22$02.00/0 Copyright C)1983,American SocietyforMicrobiology Coliphage P1Morphogenesis: Analysis ofMutants byElectron Microscopy JEANT.WALKER ANDDONALD H.WALKER, JR.* Department ofMicrobiology, University ofIowa, IowaCity,Iowa52242 Received 1July1982/Accepted 19October 1982 Weusedelectron microscopy andserumblocking powerteststodetermine the phenotypes of47phageP1ambermutants thathavedefectsinparticlemorpho- genesis. Eleven mutants showedheaddefects,30showedtaildefects,and6hada defectinparticlematuration (whichcouldbeeitherintheheadorinthetail). Consideration ofprevious complementation testresults,geneticandphysical positions ofthemutations, andphenotypes ofthemutants allowedassignment of mostofthe47mutations togenes.Thus, aminimum of12tailgenes,4headgenes, and1particlematuration genearenowknownforP1.Ofthe12tailgenes,1(gene 19,locatedwithintheinvertible Cloop)codesfortailfibers,6(genes3,5,16,20, 21,and26)codeforbaseplate components (although oneofthese genescould codeforthetailtube), 1(gene22)codesforthesheath,1(gene6)affectstail length,2(genes7and25)areinvolved intailstability, and1(gene24)eithercodes forabaseplate component orisinvolved intailstability. Ofthefourheadgenes, gene9codesforaproteinrequired forDNApackaging. Thefunction ofheadgene 4isunclear. Head gene8probably codesforaminorheadprotein, whereas head gene23couldcodeforeither aminorheadprotein orthemajorheadprotein. Excluding theparticlematuration gene(gene1),the12tailgenes areclustered in threeregionsoftheP1physical genome. Thefourheadgenes areatfourseparate locations. However, someP1headgeneshavenotyetbeendetected andcouldbe locatedintworegions(forwhichtherearenoknowngenes)adjacent togenes4 and8.TheP1morphogenetic geneclusters areinterrupted bymanygenesthatare expressed intheprophage. Coliphage P1consists ofanicosahedral, DNA-containing headandacomplex tail(47; Fig.1).Thetailhasatubesurrounded bya contractile sheath. Atthedistalendofthe sheathisanill-defined baseplate towhich kinkedtailfibers(probably six)areattached. At theproximal endofthesheathisacap(occa- sionally seenontailsunattached toheads).The headisattached tothetailviaaneckwhich protrudes fromavertexofthehead(47).The neckisattached totheheadviaahead-neck connector (whichisonlyseenwhentheheadis emptyorhasbeendisrupted). Oneoftheuniquecharacteristics ofP1isthat itnormally produces isometric headsofatleast threedifferent sizes(PlB,P1S,andPlM,having diameters of86,65,and47nm,respectively [47]).Aspartofanapproach todetermine how thedifferent headsizesareassembled, weexam- ined103P1ambermutants toidentify those defective inheadstructure. AgeneticmapforPl(Fig.2),whichincludes these103ambermutations, hasbeenestablished (49).Two-factor crossespermitted assignmentofthemutations into10lineararrays,orlinkage clusters, Ithrough X,withmutations incluster Xbehaving astwosubgroups, X-1andX-2. However, cistrondesignations, apartfromthose originally assigned byScott(39),werenotas- signedbecausecomplementation testsinliquid frequently gaveambiguous results. Withotherphages, ambermutants canbe dividedintotwomajorclassesdepending on whether theyhaveagenedefective inafunction whichisrequired earlyoronewhichisrequired lateinphageparticledevelopment.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  AgeneticmapforPl(Fig.2),whichincludes these103ambermutations, hasbeenestablished (49).Two-factor crossespermitted assignmentofthemutations into10lineararrays,orlinkage clusters, Ithrough X,withmutations incluster Xbehaving astwosubgroups, X-1andX-2. However, cistrondesignations, apartfromthose originally assigned byScott(39),werenotas- signedbecausecomplementation testsinliquid frequently gaveambiguous results. Withotherphages, ambermutants canbe dividedintotwomajorclassesdepending on whether theyhaveagenedefective inafunction whichisrequired earlyoronewhichisrequired lateinphageparticledevelopment. Lategenes areinvolved inparticlemorphogenesis (e.g., production ofproteins whichbecomestructural components oftheparticle) orincelllysis.Generally, earlygenemutants(i.e.,thosedefec- tiveinregulation oftranscription, DNAsynthe- sis,orrecombination) donotallowsynthesis of anyofthelategeneproducts, including those causinglysis.Thus,thosemutants thatcause celllysisshouldhavemorphological defects. Therefore, wedetermined whichofthe103P1 ambermutants couldcausecelllysis(50)and examined 55ofthembyelectronmicroscopy for 1118', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1119 B FIG.1.Micrographs showing normalPlparticles. (A)LysateofP1vir*Shhasbigheads(P1B),fullofDNA, andtails,themajority ofwhichhavesheaths. (B)Particles fromaCsCIgradient. Thefullsmallheads(P1S)have tailswithsheaths invarying stagesofcontraction; baseplates andtailfiberscanbeeasilyseen.Insetsshow diagrammatic scaledrawings ofPl;dimensions areinnanometers. Bar,100nm. structural defects(thispaper). Indoing so,we identified thosemutants whosedefects arein earlyfunctions, inlysisfunctions, andinheadandtailmorphology. Inaddition, because the genetic(39,50)andphysical (21,34,38,43,45) locations ofthesemutations ontheP1genomeVOL.45,1983 i... 4i', \"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1120 WALKER ANDWALKER areknown, wedetermined theextentofcluster- ingofthefunctions represented bythese mu- tants. MATERIALS ANDMETHODS Bacterial andphagestrains. Bacterial strains were Shigella dysenteriae sup'strainsSh-15andSh-16 (whichisastreptomycin-resistant Sh-15strain)(5)and thetetracycline-resistant sup+strainSh-16supF con- structed byVanMontagu (50).Phagestrains wereP1 vir(thesupervirulent mutantP1kcvirsofSarkarthat grows onstrainslysogenic forP1[39]),P1amber mutants (Fig.2)13,16,19,and21ofScott(39),and51 P1amber mutants ofWalkerandWalker(49).These P1ambermutants alsohavethevirmutation (i.e.,they areviramberdoublemutants); forsimplicity these mutants willbereferred toasaml3,etc.,without the virdesignation. Media,phage assays,andthepreparation ofphage stocks.Media,phage assays, andthepreparation of phagestocks wereasdescribed previously (50). Preparation ofdefective lysatesofamberphage- infected cells.At0minthedesiredambermutant was added(atamultiplicity ofinfection of4to6)toalog- phaseL-brothcultureofS.dysenteriae sup'strainSh- 15.virambermutants wereusedtoavoidlysogeniza- tion(32),andamber' P1virwasincluded asacontrol. Afterallowing 10minat37°Cforadsorption inthe LINKAGE CLUSTER CISTRONI presence of5mMCaCl2,theculture wasfiltered to remove unadsorbed phage,diluted2.5-fold infresh medium containing 5mMMgSO4, andincubated at 37°Cwithvigorous shaking. Withtheonsetoflysis (usually at60to70minwhenitoccurred), DNaseand RNase wereaddedtofinalconcentrations of1,ug/ml. At110min,whenlysiswascomplete, asampleofthe lysate wasprepared immediately forelectron micros- copyasdescribed below,andthelysate wasassayed, withsupoandsup'strainsSh-16andSh-16supF,for viablephageandrevertants. Undertheseconditions titersformutants onsup'cellswerelowerbyafactor of104thanthoseobtained using aP1virlysate asa control. Thisiscomparable toresultsobtained forT4 (55)andP2(32).Suchtitersformutants aredue primarily totransmission ofambers. Theremaining lysate waschilled, and5mlwas centrifuged (at35,000 rpmfor2sat4°CinanSW50.1 rotor)topelletunlysed cellsandcelldebris. The supernatant fluidswerecentrifuged (at22,000 rpmfor 2hinthesamerotor)topelletanyphagecomponents present. Thepellet wasleftat4°Cfor2daysto resuspend in0.25mlofculturefluid(i.e.,a20-fold concentration ofphagecomponents), andasample wasprepared forelectron microscopy. Serumblocking assays.Thosedefective lysatesused forelectron microscopy werealsoassayed forserum blocking antigen(SBA).Themethod fordetermining m  4 17 2 3 AMBER 471413734 417357129 7.\", \"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Theremaining lysate waschilled, and5mlwas centrifuged (at35,000 rpmfor2sat4°CinanSW50.1 rotor)topelletunlysed cellsandcelldebris. The supernatant fluidswerecentrifuged (at22,000 rpmfor 2hinthesamerotor)topelletanyphagecomponents present. Thepellet wasleftat4°Cfor2daysto resuspend in0.25mlofculturefluid(i.e.,a20-fold concentration ofphagecomponents), andasample wasprepared forelectron microscopy. Serumblocking assays.Thosedefective lysatesused forelectron microscopy werealsoassayed forserum blocking antigen(SBA).Themethod fordetermining m  4 17 2 3 AMBER 471413734 417357129 7. 1244731342017U7266497475 6171 MUTATION 132_ 170 155 LYSIS NORMAL 11NOIl-vJI NORMAL I PHENOTYPE LYSISPREMATURE SERUMBLOCKING I POSITIVE 'L-NEGATIVE'1I POSITIVE POWER Vll VlilX11 -2 I- 1 56 7 6 10 9 ItWI...11_..i..a...1.M.....W1lai(111......._..Si.1(I_t,,,.JmI11.,!i.,1illim(!l,1 I35Io7 4528191622L963214395UI102IQ4029139517113178 36 1615246644843 iBO11513861 1092767 116 4210115168 11104 54 365806 179 110125 160 113 4 140 70 . 131112 175 NORMALJI11JI-NORMAL-I LYSIS POSITIVE POSITIVE NEGATIVE FIG.2.Genetic mapofPlambermutations, modified fromWalkerandWalker(50),showing resultsofhost celllysisandserumblocking powertests.Regionsdemarcating the10linkageclusters arerepresented bysolid blocks.Thecistrondesignations abovethemaplineincludetheoriginal1to10ofScott(39)and17ofWalkerand Walker(50).Groupsofambermutations vertically stackedunderveryclosehashmarksorwithinparentheses do notshowcomplementation (49).Theorientation ofgroupsofambermutations withinparentheses orwithin brackets withrespecttomutations outsidethesegroupsisunknown. Thespacingofthehashmarksisnotmeant toindicateeithergenetic orphysical distance. Underlined mutations indicate themutants thatwereusedinthe experiments reported inthispaper.J.VIROL.\"], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': [\"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  EVOLUTION OFpl5BANDPHAGE P1TAILFIBERGENES 3943 D. P2H ARF>ala2bc MuS a PiS ~11 IaOalblb2c p15BS-|aOaa2a3a4a5a6 c X401-stf-| 500bp abc FIG.8.Arrangement ofhomologous elements intheintermedi- atepartoflongtailfibergenes.Genesaredepicted inthe5'to3' direction, andhomologous regions areboxed.Thecorresponding aminoacidsequences ofthea,b,andcelements arealignedinFig. 3.Homologous sequences amongthe3'domains (Sv)ofthepl5B geneSandX401-stfareshowninFig.5,whilethoseofthe3' domains oftheP2geneH,MugeneS,andP1geneSarenotshown. TheXtailfibergeneisrepresented asafusionofORF401andgene stf(seetext).DNAinversion crossover sitesandcrossover site-like sequences areshownasblackandopentriangles, respectively. thetailfiberstothebaseplate ofthevirion.TheproductofT probably possesses acatalytic roleintailfiberassembly of oneofthesixfibervariants. Expression ofTdepends onthe configuration oftheinvertible segments intheMinregion, anditismostlikelytranslationally coupled toSvMJ(33). SinceMininversion alternatively provides expression of eachofthesixvariants ofS(33),sixdifferent tailfibers,each encoding adistincthostrange,couldbesynthesized. How- ever,itisexpected thatindividual phageparticles would usually carrypl5Btailfibersofonlyonehostspecificity corresponding tothegeneticconfiguration oftheparental genome. Thephagesproduced duringlyticgrowthofphage P1conferpredominantly oneofthetwopossible host specificities (13),andDNAinversions byCinaswellasby Minrecombinases arelow-frequency events. Thebiological significance ofamultiple extension ofthe hostphagesuchasthatdisplayed bypl5Bisuncertain. Although itissufficient formostphagestodetermine only onehostrange,itcouldallowaphagetomorereadilyadapt tochanging environments composed ofseveral bacterial species, oraphagecouldevadeadsorption-resistant host mutations bybinding toalternate receptors. Itisstriking thatthelargetailfibergenesofseveral unrelated, well-studied phagesofE.colicarrydomains of highlysimilarsequences. Suchhomologous domains are observed amongthe3'ends(e.g.,amonggeneSofP1,Sof Mu,andHofP2andamongSvgenesofpl5B,gene37ofT4, andstfofX)whichconferhostspecificity, whilenohomol- ogyhassofarbeendetected amongthe5'ends,which specifytheattachment tothebaseplates.Itisshownhere thattheintermediate partsofthetailfibergenesofsome phagesarearrangements ofseveralshortelements (Fig.8). Threetypesofsuchshortelements (a,b,andc),each presentinmorethanonegenome, havebeendetected among theavailable sequences. Theelements maybeamplified(e.g.,altoa7inp15BgeneS,alanda2inP2geneH,andblandb2inP1geneS).Theelements presentinagenome appearinthesameorder,i.e.,abeforebbeforec.The function oftheseelements duringtheinfection process isnot known. Theintermediate region,withitsmosaiclike struc- ture,isexpected tooccupy acorresponding position onthe tailfiberprotein, sincethetertiary structure oftailfiber proteins canbeconsidered largelylinear.Thiscanserveas abasisforstudiesoftheirroleininteractions between tail fibersandmacromolecules onthesurfaceofthehostbacte- rium,including reversible andirreversible adsorption. Thedatapresented abovegiveaninsightintothelong- termevolution ofgenes,whichareunderstrongselective pressure.\", \"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  Theintermediate region,withitsmosaiclike struc- ture,isexpected tooccupy acorresponding position onthe tailfiberprotein, sincethetertiary structure oftailfiber proteins canbeconsidered largelylinear.Thiscanserveas abasisforstudiesoftheirroleininteractions between tail fibersandmacromolecules onthesurfaceofthehostbacte- rium,including reversible andirreversible adsorption. Thedatapresented abovegiveaninsightintothelong- termevolution ofgenes,whichareunderstrongselective pressure. Obviously, otherwise unrelated phagesexchanged shortdomains byhorizontal transfer whichresulted in chimeric tailfibergenes.Thepresence ofcrossover sitesand crossover site-like sequences forDNAinvertases between 3' andintermediate domains suggests adecisive roleforsite- specificrecombination inevolutionary reshuffling. ACKNOWLElDGMENTS WethankMichelWurtzforelectron microscopy andReinhard Doelzformaintaining anexcellent computer software environment. Thisworkwassupported bygrant3.624.087 oftheSwissNational Science Foundation. ADDENDUM Inanindependent study,E.Haggard-Ljungquist and coauthors (11)carriedoutacomplementary comparison of tailfibergenes,andtheyreached similarconclusions. REFERENCES 1.Arber,W.Unpublished data. la.Brody, H.,A.Greener, andC.W.Hill.1985.Excision and reintegration oftheEscherichia coliK-12chromosomal element e14.J.Bacteriol. 161:1112-1117. 2.Daniels, D.L.,J.L.Schroeder, W.Szybalski, F.Sanger, and F.R.Blattner. 1983.Amolecular mapofcoliphage lambda, p. 469-517. InR.W.Hendrix, J.W.Roberts, F.W.Stahl,and R.A.Weisberg (ed.),Lambda II.ColdSpringHarborLabora- tory,ColdSpringHarbor, N.Y. 3.Daniels, D.L.,J.L.Schroeder, W.Szybalski, F.Sanger, A.R. Coulson, G.F.Hong,D.F.Hill,G.B.Petersen, andF.R. Blattner. 1983.Complete annotated lambda sequence, p.519- 676.InR.W.Hendrix, J.W.Roberts, F.W.Stahl,andR.A. Weisberg (ed.),Lambda II.ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 4.Dayhoff M.O.,(ed.).1979.Atlasofprotein sequence and structure. National Biomedical Research Foundation, Washing- ton,D.C. 5.Devereux, J.,P.Haeberli, and0.Smithies. 1984.Acomprehen- sivesetofsequence analysis programs fortheVAX.Nucleic AcidsRes.12:387-395. 6.George, D.G.,L.-S.L.Yeh,andW.C.Barker. 1983.Unex- pectedrelationships between bacteriophage lambdahypotheti- calproteins andbacteriophage T4tail-fiber proteins. Biochem. Biophys. Res.Commun. 115:1061-1068. 7.Glasgow, A.C.,K.T.Hughes, andM.I.Simon.1989.Bacterial DNAinversion systems, p.637-659. InD.E.BergandM.M. Howe(ed.),MobileDNA.American Society forMicrobiology, Washington, D.C. 8.Gribskov, M.,andR.R.Burgess. 1986.SigmafactorsfromE. coli,B.subtilis,phageSPOl,andphageT4arehomologousproteins. Nucleic AcidsRes.14:6745-6763.9.Guidolin, A.,J.-M.Zingg,andW.Arber.1989.Organization of thebacteriophage P1tail-fibre operon. Gene76:239-243. 10.Guidolin, A.,J.-M.Zingg,H.Lehnherr, andW.Arber.1989. Bacteriophage P1tail-fibre anddaroperons areexpressed from homologous phage-specific latepromoter sequences. J.Mol.VOL.174,1992\", 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  3944 SANDMEIER ETAL. Biol.208:615-622. 11.Haggard-Ljungquist, E.,C.Hailing, andR.Calendar. 1992. DNAsequence ofthetailfibergenesofbacteriophage P2: evidence forhorizontal transfer oftailfibergenesamongunre- latedbacteriophages. J.Bacteriol. 174:1462-1477. 12.Hiestand-Nauer, R.,andS.lida.1983.Sequence ofthesite- specificrecombinase genecinandofitssubstrates servinginthe inversion oftheCsegment ofbacteriophage P1.EMBO J. 2:1733-1740. 13.lida,S.1984.Bacteriophage P1carriestworelatedsetsofgenes determining itshostrangeintheinvertible Csegment ofits genome. Virology 134:421-434. 14.lida,S.,andR.Hiestand-Nauer. 1986.Localized conversion at thecrossover sequences inthesite-specific DNAinversion systemofbacteriophage P1.Cell45:71-79. 15.Iida,S.,andR.Hiestand-Nauer. 1987.Roleofthecentral dinucleotide atthecrossover sitesfortheselection ofquasisites inDNAinversion mediated bythesite-specific Cinrecombinase ofphageP1.Mol.Gen.Genet.208:464-468. 15a.Iida, S.,andR.Hiestand-Nauer. Unpublished data. 16.lida,S.,H.Huber, R.Hiestand-Nauer, J.Meyer,T.A.Bickle, andW.Arber.1984.Thebacteriophage P1site-specific recom- binaseCin:recombination eventsandDNArecognition se- quences. ColdSpringHarborSymp.Quant.Biol.49:769-777. 17.lida,S.,H.Sandmeier, P.Hubner, R.Hiestand-Nauer, K. Schneitz, andW.Arber.1990.TheMinDNAinversion enzyme ofplasmidpl5BofEscherichia coli15T-:anewmember ofthe Dinfamilyofsite-specific recombinases. Mol.Microbiol. 4:991- 997. 18.Ikeda,H.,M.Inuzuka, andJ.-I.Tomizawa. 1970.P1-like plasmid inEscherichia coli15.J.Mol.Biol.50:457-470. 19.Kahmann, R.,andD.Kamp.1987.Sequence oftherightendof Mu,p.297-308. InN.Symonds, A.Toussaint, P.vandePutte, andM.M.Howe(ed.),PhageMu.ColdSpringHarborLabo- ratory,ColdSpringHarbor, N.Y. 20.Kamp, D.,E.Kardas, W.Ritthaler, R.Sandulache, R. Schmucker, andB.Stern.1984.Comparative analysis ofinvert- ibleDNAinphagegenomes. ColdSpringHarborSymp.Quant. Biol.49:301-311. 21.Kutsukake, K.,T.Nakao,andT.Iino.1985.AgeneforDNA invertase andaninvertible DNAinEscherichia coliK-12.Gene 34:343-350. 22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 1982.Molecular cloning: alaboratory manual. ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 23.Meyer,J.,M.Stalhammar-Carlemalm, M.Streiff,S.lida,and W.Arber.1986.Sequence relations amongtheIncYplasmid plSB,P1,andP7prophages. Plasmid 16:81-89. 24.Michel,C.J.,B.Jaqc,D.G.Arques, andT.A.Bickle.1986.A remarkable aminoacidsequence homology between aphageT4 tailfibreprotein andORF314 ofphageXlocated inthetail operon. Gene44:147-150. 25.Montag, D.,andU.Henning.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 1982.Molecular cloning: alaboratory manual. ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 23.Meyer,J.,M.Stalhammar-Carlemalm, M.Streiff,S.lida,and W.Arber.1986.Sequence relations amongtheIncYplasmid plSB,P1,andP7prophages. Plasmid 16:81-89. 24.Michel,C.J.,B.Jaqc,D.G.Arques, andT.A.Bickle.1986.A remarkable aminoacidsequence homology between aphageT4 tailfibreprotein andORF314 ofphageXlocated inthetail operon. Gene44:147-150. 25.Montag, D.,andU.Henning. 1987.Anopenreadingframeinthe Escherichia colibacteriophage Xgenome encodes aproteinthat functions inassembly ofthelongtailfibersofbacteriophage T4.J.Bacteriol. 169:5884-5886. 26.Montag, D.,I.Riede,M.-L.Eschbach, M.Degen, andU. Henning. 1987.Receptor-recognizing proteins ofT-even type bacteriophages. Constant andhypervariable regions andan unusual caseofevolution. J.Mol.Biol.196:165-174. 27.Montag, D.,H.Schwarz, andU.Henning. 1989.Acomponent of thesidetailfiberofEscherichia colibacteriophage Xcan functionally replace thereceptor-recognizing partofalongtail fiberprotein oftheunrelated bacteriophage T4.J.Bacteriol. 171:4378-4384. 28.Mosig,G.,andF.Eiserling. 1988.PhageT4structure and metabolism, p.521-606. InR.Calendar (ed.),Thebacterio- phages. Plenum Press,NewYork. 29.Mural,R.J.,R.H.Chesney, D.Vapnek, M.M.Kropf,andJ.R. Scott.1979.Isolation andcharacterization ofclonedfragments ofbacteriophage P1DNA.Virology 93:387-397. 30.Oliver,D.B.,andR.A.Crowther. 1981.DNAsequence ofthe tailfibregenes36and37ofbacteriophage T4.J.Mol.Biol. 153:545-568. 31.Plasterk, R.H.A.,andP.vandePutte.1985.Theinvertible P-DNAsegment inthechromosome ofEscherichia coli.EMBO J.4:237-242. 32.Sandmeier, H.1990.Characterization ofacomplex DNAinver- sionsystem. Ph.D.thesis.BaselUniversity, Basel. 33.Sandmeier, H.,S.lida,P.Hubner, R.Hiestand-Nauer, andW. Arber.1991.Geneorganization inthemultiple DNAinversion regionMinofplasmid plSBofE.coli15T-:assemblage ofa variable gene.Nucleic AcidsRes.19:5831-5838. 34.Sandmeier, H.,S.lida,J.Meyer, R.Hiestand-Nauer, andW. Arber.1990.Site-specific DNArecombination systemMinof plasmid pl5B:aclusterofoverlapping invertible DNAseg- ments.Proc.Natl.Acad.Sci.USA87:1109-1113. 35.Sanger, F.,S.Nicklen, andA.R.Coulson. 1977.DNAsequenc- ingwithchain-terminating inhibitors. Proc.Natl.Acad.Sci. USA74:5463-5467. 36.Selivanov, N.A.,A.G.Prilipov, andV.V.Mesyanzhinov. 1988. Nucleotide anddeduced aminoacidsequence ofbacteriophage T4gene12.Nucleic AcidsRes.16:2334. 37.Sengstag, C.,andW.Arber.1983.IS2insertion isamajorcause ofspontaneous mutagenesis ofthebacteriophage P1:non- random distribution oftargetsites.EMBOJ.2:67-71.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/jbacter00078-0106.pdf  35.Sanger, F.,S.Nicklen, andA.R.Coulson. 1977.DNAsequenc- ingwithchain-terminating inhibitors. Proc.Natl.Acad.Sci. USA74:5463-5467. 36.Selivanov, N.A.,A.G.Prilipov, andV.V.Mesyanzhinov. 1988. Nucleotide anddeduced aminoacidsequence ofbacteriophage T4gene12.Nucleic AcidsRes.16:2334. 37.Sengstag, C.,andW.Arber.1983.IS2insertion isamajorcause ofspontaneous mutagenesis ofthebacteriophage P1:non- random distribution oftargetsites.EMBOJ.2:67-71. 38.Sengstag, C.,J.C.W.Shepherd, andW.Arber.1983.The sequence ofthebacteriophage P1genome regionserving ashot targetforIS2insertion. EMBOJ.2:1777-1781. 39.Tominaga, A.,S.Ikemizu, andM.Enomoto. 1991.Site-specific recombinase genesinthreeShigella subgroups andnucleotide sequences ofapinBgeneandaninvertible Bsegment from Shigella boydii.J.Bacteriol. 173:4079-4087. 40.Wood, W.B.1966.Hostspecificity ofDNAproduced by Escherichia coli:bacterial mutations affecting therestriction andmodification ofDNA.J.Mol.Biol.16:118-133. 41.Yarmolinsky, M.B.,andN.Sternberg. 1988.Bacteriophage P1, p.291-438. InR.Calendar (ed.),Thebacteriophages. Plenum Press,NewYork.J.BACTERIOL.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Vol.45,No.3 JOURNAL OFVIROLOGY, Mar.1983,p.1118-1139 0022-538X/83/031118-22$02.00/0 Copyright C)1983,American SocietyforMicrobiology Coliphage P1Morphogenesis: Analysis ofMutants byElectron Microscopy JEANT.WALKER ANDDONALD H.WALKER, JR.* Department ofMicrobiology, University ofIowa, IowaCity,Iowa52242 Received 1July1982/Accepted 19October 1982 Weusedelectron microscopy andserumblocking powerteststodetermine the phenotypes of47phageP1ambermutants thathavedefectsinparticlemorpho- genesis. Eleven mutants showedheaddefects,30showedtaildefects,and6hada defectinparticlematuration (whichcouldbeeitherintheheadorinthetail). Consideration ofprevious complementation testresults,geneticandphysical positions ofthemutations, andphenotypes ofthemutants allowedassignment of mostofthe47mutations togenes.Thus, aminimum of12tailgenes,4headgenes, and1particlematuration genearenowknownforP1.Ofthe12tailgenes,1(gene 19,locatedwithintheinvertible Cloop)codesfortailfibers,6(genes3,5,16,20, 21,and26)codeforbaseplate components (although oneofthese genescould codeforthetailtube), 1(gene22)codesforthesheath,1(gene6)affectstail length,2(genes7and25)areinvolved intailstability, and1(gene24)eithercodes forabaseplate component orisinvolved intailstability. Ofthefourheadgenes, gene9codesforaproteinrequired forDNApackaging. Thefunction ofheadgene 4isunclear. Head gene8probably codesforaminorheadprotein, whereas head gene23couldcodeforeither aminorheadprotein orthemajorheadprotein. Excluding theparticlematuration gene(gene1),the12tailgenes areclustered in threeregionsoftheP1physical genome. Thefourheadgenes areatfourseparate locations. However, someP1headgeneshavenotyetbeendetected andcouldbe locatedintworegions(forwhichtherearenoknowngenes)adjacent togenes4 and8.TheP1morphogenetic geneclusters areinterrupted bymanygenesthatare expressed intheprophage. Coliphage P1consists ofanicosahedral, DNA-containing headandacomplex tail(47; Fig.1).Thetailhasatubesurrounded bya contractile sheath. Atthedistalendofthe sheathisanill-defined baseplate towhich kinkedtailfibers(probably six)areattached. At theproximal endofthesheathisacap(occa- sionally seenontailsunattached toheads).The headisattached tothetailviaaneckwhich protrudes fromavertexofthehead(47).The neckisattached totheheadviaahead-neck connector (whichisonlyseenwhentheheadis emptyorhasbeendisrupted). Oneoftheuniquecharacteristics ofP1isthat itnormally produces isometric headsofatleast threedifferent sizes(PlB,P1S,andPlM,having diameters of86,65,and47nm,respectively [47]).Aspartofanapproach todetermine how thedifferent headsizesareassembled, weexam- ined103P1ambermutants toidentify those defective inheadstructure. AgeneticmapforPl(Fig.2),whichincludes these103ambermutations, hasbeenestablished (49).Two-factor crossespermitted assignmentofthemutations into10lineararrays,orlinkage clusters, Ithrough X,withmutations incluster Xbehaving astwosubgroups, X-1andX-2. However, cistrondesignations, apartfromthose originally assigned byScott(39),werenotas- signedbecausecomplementation testsinliquid frequently gaveambiguous results. Withotherphages, ambermutants canbe dividedintotwomajorclassesdepending on whether theyhaveagenedefective inafunction whichisrequired earlyoronewhichisrequired lateinphageparticledevelopment.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  AgeneticmapforPl(Fig.2),whichincludes these103ambermutations, hasbeenestablished (49).Two-factor crossespermitted assignmentofthemutations into10lineararrays,orlinkage clusters, Ithrough X,withmutations incluster Xbehaving astwosubgroups, X-1andX-2. However, cistrondesignations, apartfromthose originally assigned byScott(39),werenotas- signedbecausecomplementation testsinliquid frequently gaveambiguous results. Withotherphages, ambermutants canbe dividedintotwomajorclassesdepending on whether theyhaveagenedefective inafunction whichisrequired earlyoronewhichisrequired lateinphageparticledevelopment. Lategenes areinvolved inparticlemorphogenesis (e.g., production ofproteins whichbecomestructural components oftheparticle) orincelllysis.Generally, earlygenemutants(i.e.,thosedefec- tiveinregulation oftranscription, DNAsynthe- sis,orrecombination) donotallowsynthesis of anyofthelategeneproducts, including those causinglysis.Thus,thosemutants thatcause celllysisshouldhavemorphological defects. Therefore, wedetermined whichofthe103P1 ambermutants couldcausecelllysis(50)and examined 55ofthembyelectronmicroscopy for 1118', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1119 B FIG.1.Micrographs showing normalPlparticles. (A)LysateofP1vir*Shhasbigheads(P1B),fullofDNA, andtails,themajority ofwhichhavesheaths. (B)Particles fromaCsCIgradient. Thefullsmallheads(P1S)have tailswithsheaths invarying stagesofcontraction; baseplates andtailfiberscanbeeasilyseen.Insetsshow diagrammatic scaledrawings ofPl;dimensions areinnanometers. Bar,100nm. structural defects(thispaper). Indoing so,we identified thosemutants whosedefects arein earlyfunctions, inlysisfunctions, andinheadandtailmorphology. Inaddition, because the genetic(39,50)andphysical (21,34,38,43,45) locations ofthesemutations ontheP1genomeVOL.45,1983 i... 4i', \"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1120 WALKER ANDWALKER areknown, wedetermined theextentofcluster- ingofthefunctions represented bythese mu- tants. MATERIALS ANDMETHODS Bacterial andphagestrains. Bacterial strains were Shigella dysenteriae sup'strainsSh-15andSh-16 (whichisastreptomycin-resistant Sh-15strain)(5)and thetetracycline-resistant sup+strainSh-16supF con- structed byVanMontagu (50).Phagestrains wereP1 vir(thesupervirulent mutantP1kcvirsofSarkarthat grows onstrainslysogenic forP1[39]),P1amber mutants (Fig.2)13,16,19,and21ofScott(39),and51 P1amber mutants ofWalkerandWalker(49).These P1ambermutants alsohavethevirmutation (i.e.,they areviramberdoublemutants); forsimplicity these mutants willbereferred toasaml3,etc.,without the virdesignation. Media,phage assays,andthepreparation ofphage stocks.Media,phage assays, andthepreparation of phagestocks wereasdescribed previously (50). Preparation ofdefective lysatesofamberphage- infected cells.At0minthedesiredambermutant was added(atamultiplicity ofinfection of4to6)toalog- phaseL-brothcultureofS.dysenteriae sup'strainSh- 15.virambermutants wereusedtoavoidlysogeniza- tion(32),andamber' P1virwasincluded asacontrol. Afterallowing 10minat37°Cforadsorption inthe LINKAGE CLUSTER CISTRONI presence of5mMCaCl2,theculture wasfiltered to remove unadsorbed phage,diluted2.5-fold infresh medium containing 5mMMgSO4, andincubated at 37°Cwithvigorous shaking. Withtheonsetoflysis (usually at60to70minwhenitoccurred), DNaseand RNase wereaddedtofinalconcentrations of1,ug/ml. At110min,whenlysiswascomplete, asampleofthe lysate wasprepared immediately forelectron micros- copyasdescribed below,andthelysate wasassayed, withsupoandsup'strainsSh-16andSh-16supF,for viablephageandrevertants. Undertheseconditions titersformutants onsup'cellswerelowerbyafactor of104thanthoseobtained using aP1virlysate asa control. Thisiscomparable toresultsobtained forT4 (55)andP2(32).Suchtitersformutants aredue primarily totransmission ofambers. Theremaining lysate waschilled, and5mlwas centrifuged (at35,000 rpmfor2sat4°CinanSW50.1 rotor)topelletunlysed cellsandcelldebris. The supernatant fluidswerecentrifuged (at22,000 rpmfor 2hinthesamerotor)topelletanyphagecomponents present. Thepellet wasleftat4°Cfor2daysto resuspend in0.25mlofculturefluid(i.e.,a20-fold concentration ofphagecomponents), andasample wasprepared forelectron microscopy. Serumblocking assays.Thosedefective lysatesused forelectron microscopy werealsoassayed forserum blocking antigen(SBA).Themethod fordetermining m  4 17 2 3 AMBER 471413734 417357129 7.\", \"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Theremaining lysate waschilled, and5mlwas centrifuged (at35,000 rpmfor2sat4°CinanSW50.1 rotor)topelletunlysed cellsandcelldebris. The supernatant fluidswerecentrifuged (at22,000 rpmfor 2hinthesamerotor)topelletanyphagecomponents present. Thepellet wasleftat4°Cfor2daysto resuspend in0.25mlofculturefluid(i.e.,a20-fold concentration ofphagecomponents), andasample wasprepared forelectron microscopy. Serumblocking assays.Thosedefective lysatesused forelectron microscopy werealsoassayed forserum blocking antigen(SBA).Themethod fordetermining m  4 17 2 3 AMBER 471413734 417357129 7. 1244731342017U7266497475 6171 MUTATION 132_ 170 155 LYSIS NORMAL 11NOIl-vJI NORMAL I PHENOTYPE LYSISPREMATURE SERUMBLOCKING I POSITIVE 'L-NEGATIVE'1I POSITIVE POWER Vll VlilX11 -2 I- 1 56 7 6 10 9 ItWI...11_..i..a...1.M.....W1lai(111......._..Si.1(I_t,,,.JmI11.,!i.,1illim(!l,1 I35Io7 4528191622L963214395UI102IQ4029139517113178 36 1615246644843 iBO11513861 1092767 116 4210115168 11104 54 365806 179 110125 160 113 4 140 70 . 131112 175 NORMALJI11JI-NORMAL-I LYSIS POSITIVE POSITIVE NEGATIVE FIG.2.Genetic mapofPlambermutations, modified fromWalkerandWalker(50),showing resultsofhost celllysisandserumblocking powertests.Regionsdemarcating the10linkageclusters arerepresented bysolid blocks.Thecistrondesignations abovethemaplineincludetheoriginal1to10ofScott(39)and17ofWalkerand Walker(50).Groupsofambermutations vertically stackedunderveryclosehashmarksorwithinparentheses do notshowcomplementation (49).Theorientation ofgroupsofambermutations withinparentheses orwithin brackets withrespecttomutations outsidethesegroupsisunknown. Thespacingofthehashmarksisnotmeant toindicateeithergenetic orphysical distance. Underlined mutations indicate themutants thatwereusedinthe experiments reported inthispaper.J.VIROL.\"], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:04 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'112'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994311'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'341ms'), (b'x-request-id', b'85705c033bf225a00845134b6ddc241e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249b0bb355d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:04 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'112'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994311'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'341ms'), (b'x-request-id', b'85705c033bf225a00845134b6ddc241e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249b0bb355d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': [\"page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1121 SBAwasamodification oftheDeMars (9)method as usedbyIsraeletal.(22)forP22andWardetal.(52)for T4;itcompares ratesratherthanfinallevelsofserum inactivation byblocking antigens. Serialtwofold dilutions weremadeofthelysates aftertheyhadbeentreatedwithchloroform for20min at37°Ctokillanyviablecells.Cellsinfected with mutants (inlinkageclusters IIIandX-2;Fig.2)thatdo notcausecelllysiswerelysedbysonication atthe timewhenlysisoccurred withamber'-infected cells. Cellsinfected withothermutants andwithP1virwere notsonicated, except asdescribed belowforPlvir,to obtain astandard curve.Reaction mixtures contained 0.45mlofalysatedilution and0.1mloftesterphage, towhich wassubsequently added0.45mlofantiphage serum.Thetesterphage wasP1kcatafinalconcentra- tionof3x105PFU/ml; theturbidplaques couldbe distinguished fromtheclearplaquesofanyrevertants presentintheP1v'iramberlysates. Thefirst-order rate constant (k)forphageinactivation oftheundiluted hyperimmune serumat37°Cwas1,500min-'.This serumhadakof0.1min-'inthefinaldilution ofthe reaction mixture. Thereaction mixtures wereassayed (onstrainSh-16)forsurviving testerphageattheend of46min,whentesterphagesurvival wasabout1%in theabsence ofSBA.Thiscontrolwithout SBAwas alwaysincluded ineachgroupoftestsdoneondiffer- entdays.Thisinactivation wasassumed tobeexpo- nentialthroughout the46-minincubation. Toobtainstandard curves, lysatesofamber' Plvir grown onstrainSh-15 wereusedasSBA;thephage wereinactivated byeithersonication orUVtreatment sotheplaques wouldnotobscure thoseofthecompet- ingP1kctesterphage. Theresultsoftheserumblocking assays were plotted aspercent residual k=(1-kfinal/kinitial) X100 =(1-lnSISd/lnSISO) x100,whereS/SO =survival of thetesterphageinthecontrolafter46minwithserum butnoSBA,andSISd =survival ofthetesterphage after46minwithbothserumandSBAatdilution dof thelatter.Ratherthanshowing graphsforthe55 mutants tested,comparisons weremadeusingthe dilution atwhichresidual k=50%. Electron microscopy. A0.4-misampleofconcentrat- edorunconcentrated lysateofthephage-infected cells wasdialyzed for15minbyplacingitonamembrane filter(Millipore Corp.; average poresize,25nm) floating onP1buffer(ratherthanwater,whichcaused headdisruption). Thisdialysis wasdoneatroom temperature todecrease tailcontractions obtained by dialysis at4°C.Adropofsample wasplaced onaglow- discharged, carbon-coated, 400-mesh coppergridand allowed tostandfor2min.Asmalldropofsodium phosphotungstate (1%;pH7.0)wasaddedandleftfor 30sbeforetheexcessliquid wasremoved withfilter paper.AJEOL100Belectron microscope withananti- contamination device wasusedat80kV.Micrographs weretakenatinitialmagnifications of10,000 to50,000 onKodak4463film. RESULTS Weexamined, byelectron microscopy, ly- satesof55P1amber mutants grownunder nonpermissive conditions todetermine their phenotypes. Insuchlysates, mutants defectiveinsomeaspectoftailassembly shouldproduce normalheadsandoneofavarietyoftailcompo- nentcombinations. Similarly, mutants defective insomeaspectofheadassembly shouldproduce normaltailsandeithernoheadsoroneofa varietyofdefective headstructures.\", 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  RESULTS Weexamined, byelectron microscopy, ly- satesof55P1amber mutants grownunder nonpermissive conditions todetermine their phenotypes. Insuchlysates, mutants defectiveinsomeaspectoftailassembly shouldproduce normalheadsandoneofavarietyoftailcompo- nentcombinations. Similarly, mutants defective insomeaspectofheadassembly shouldproduce normaltailsandeithernoheadsoroneofa varietyofdefective headstructures. Inaddition, weusedserumblocking assaystodetectthe presence oftailfibersinthedefective lysates. Of103P1ambermutantspreviously testedfor lysisofnonpermissive cells(50),83causedcell lysisatthenormallysistimeforPl.These83 mutations areinlinkageclusters(Fig.2)1,II,V,VI,VII,VIII,andIXandpartsofIVandX(i.e., X-1).Byanalogy withotherphages,thesemu- tantswouldbedefective inlategenesinvolved inparticlemorphogenesis. Thenumberofcistrons represented bythese 83mutants isunknown. Therefore, 47mutants (underlined inFig.2)werechosenforstudyby electron microscopy fromthemaximum possi- blenumber ofcistrons inthefollowing way. These47mutants represent (i)atleastone mutation fromeachofthegroupings ofmuta- tions(showninFig.2asvertically stackedunder veryclosehashmarksorwithinparentheses) thathadbeenfound(49)nottoshowcomple- mentation and(ii)eachofthesinglemutations outsidethesegroupings. Inaddition, eightmu- tants(alsounderlined inFig.2),representing thosethatcannotcausecelllysis(linkage clus- tersIIIandX-2)orthatcausepremature cell lysis(linkage clusterIV),werealsoincluded in thisstudy. Serumblocking assays.P1tailfibersareonly1 to2nminthickness (47)andsoareonlyseenby electron microscopy whentheyareattached to phageparticles andwhenthestainingconditions aresuchastogiveoptimal contrast between the fibersandthebackground (Fig.1).Therefore, because inotherphageswithcontractile tails, suchasT4,themajority ofSBAcanbeattribut- edtotailfibers,serumblocking assayswere donetoconfirm thepresence ofattached fibers ortodetectthepresence ofunattached fibersin theP1defective lysates.Results wereplottedas percentresidual kversuslysatedilution asde- scribedabove,andthedilutiongiving50%resid- ualkwasdetermined foreachdefective lysate. Thesedilutions weredesignated SBAvalues. Aswouldbeexpected foraphagesuchasP1 thathasacomplex morphology andtherefore wouldcodeforvariousstructures inaddition to tailfibers,themajority ofmutants showedsig- nificant serumblocking power(similar tothatof thecontrol Plvirlysates), indicating thattail fiberswerepresent(Table1;Fig.2). Thosemutants thatgavenegative results(i.e., SBAvaluesoflessthan1)havemutations in linkageclusterX-2andpartofclusterIV.Be- causeclusterX-2mutants didnotlysethehostVOL.45,1983', \"page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1122 WALKER ANDWALKER cells,theselysatesweretestedforSBAboth beforeandaftersonication tocausecelllysis; negative resultswereobtained inbothinstances, indicating thatthesenegative resultswerenot duetoafailureofthecellstolyse.(Mutants in clusterIIIalsodidnotlysethehostcells,but gavepositive SBAresults,indicating thatsoni- cationdoesnotadversely affectSBAactivity.) ClusterX-2mutations havebeenshowntobein thesamecistron(cistron10),andmutants repre- sentingthiscistronareprobably defective inthe expression oflategenes(50).SinceSBAsare lategeneproducts, thiswouldexplainwhynega- tiveresultsareobtained withdefective lysatesof thesemutants. Allofthosemutants (74,75,21,35,and171) locatedintherightpartoflinkageclusterIV gavenormallevelsofSBA.However, mutants (31,56,33,72,66,and49)locatedintheleftpart ofclusterIVgavenegative SBAresults(i.e., SBAvaluesoflessthan1),indicating thatthese mutants produce few,ifany,tailfibers.The significance oftheseresultsobtained withthe lattersixmutants isdiscussed inthenextsec- tion. Serumblocking assaysconfirmed electron mi- croscopic examination (seebelow)ofthemu- tantsinthecasesinwhichnormalamounts of sheathed tailswereproduced. Thatis,tailfibers wereseenbyelectron microscopy onparticles produced byallofthemutants thatproduced normalamounts ofsheathed tails(eitherextend- edorcontracted), exceptmutants128,150,33, 72,66,and138.Thepreparations ofmutants 128,150,and138werenotgoodenough to determine thepresence oftailfibers,butthese mutants allshowed serumblocking power,and foreachthereareallelicmutants(i.e.,mutants thatdidnotshowcomplementation [49])where tailfiberscouldbeseen.Therefore, thesemu- tantsarepresumed tohavetailfibersattached to theirtails.Preparations ofmutants, 33,72,and 66inlinkageclusterIVweregoodenough to detectanytailfibersifpresent, andthesemu- tantsdidnotshowserumblocking power,con- firmingthatthesemutants donotproduce tail fibers. SBAwasproduced byallmutants (except am49;seenextsection) thatdidnotproduce sheathed tails(andwheretailfiberstherefore couldnotbedetected byelectronmicroscopy), indicating thattailfibersweremadeeventhough nosheathed tailwasavailable fortheirattach- ment. Electron microscopy ofP1ambermutants. Defective lysatesofthe55mutants wereexam- inedbyelectron microscopy before,aswellas after,differential centrifugation toconcentrate anyphagestructures present.Amber' P1vir controls(Fig.1)prepared atthesametimeasthedefective lysateswereexamined inparallel. The typesofphagestructures seenbeforeandafter centrifugation weresimilarexceptthatwhere unattached tailshadsheathswhichhadslipped downfromthetopofthetailbeforecentrifuga- tion(mutants inclusterVIIandtherightpartof VI;seebelow), therewasadecrease inthe proportion oftailsrelative topolysheath after centrifugation. Thisdecrease isprobably caused bythesheathsubunits dissociating fromthetail tubesaftercentrifugation andpolymerizing to formpolysheath duringresuspension ofthecen- trifuged pelletat4°Cfor2days,conditions whichallowsheathsubunits topolymerize inT4 (1).Wheretails(eitherattached toorfreefrom heads)hadextended sheathsbeforecentrifuga- tion,themajority hadcontracted orsemicon- tractedsheathsaftertheconcentration steps.\", 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Thisdecrease isprobably caused bythesheathsubunits dissociating fromthetail tubesaftercentrifugation andpolymerizing to formpolysheath duringresuspension ofthecen- trifuged pelletat4°Cfor2days,conditions whichallowsheathsubunits topolymerize inT4 (1).Wheretails(eitherattached toorfreefrom heads)hadextended sheathsbeforecentrifuga- tion,themajority hadcontracted orsemicon- tractedsheathsaftertheconcentration steps. Lysateswhichshowedfullheadsbeforecentrif- ugationshowedfullheadsaftercentrifugation if theheadswereattached totails.However, a highproportion ofemptyheadswereseenafter centrifugation ifthefullheadswerenotattached totailsbeforecentrifugation. Resultsaresum- marized inTable1andFig.17. Mutations affecting tailassembly. Mutations affecting tailassembly arelocated inlinkage clusters IVthrough VIII.Eachtail-defective mutantproduces complete heads.Theheadsof mutants 33,72,and66(inlinkageclusterIV)are attached totailswhicharecomplete exceptthat theyaremissingtailfibers(Fig.3).Thesemu- tantsdonotshowserumblocking power,which confirms thatfibersarenotpresenteitherat- tachedorunattached totheparticles. However, itisunclear whether thewholetailfiberis missingorjustthedistalportion,because short projections, whichcouldbepartofthebase- plate,areoccasionally seenprotruding fromthe baseofthesheath, andtheserumblocking assaysdidnotdistinguish between thesepossi- bilities.Therefore, itremainsunknown whether thekinkedP1tailfiberismadeofmorethanone polypeptide andwhether themutations ofmu- tants33,72,and66areinthesamegene. Mutants 49,74,75,21,35,and171incluster IVandmutants 135,107,and45inclusterV producecomplete headsandpolysheath (Fig.4). InT4phage,completed baseplates arerequired forthetubepolypeptide toinitiatepolymeriza- tiontoformthetube,andthenthesheath polypeptide polymerizes around thetubeto formthetail(26).However, unlikethetube polypeptide, thesheathpolypeptide canpoly- merizebyitselftoformpolysheath. Therefore, thepresence ofpolysheath alonemeansthatthe defectisinthebaseplate, whereas thepresence ofpolysheath andbaseplates meansthatthe defectisinthetube(orbaseplate, ifthedefect hasnotdamaged thestructural integrity oftheJ.VIROL.', \"page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1123 TABLE 1.Phenotypic classification ofP1ambermutants Mutant Linkage Host Probable defect Gene no. cluster' ysisb SBP PhagestructuresP + 4.1 CPwithEH+UsT + 3.2 CPwithEH+UsT + 3.8 CPwithEH+UsT + 3.3 CPwithEH+UsT + 7.5 CPwithEH+UsT + 4.1 CP + 5.8CT;EH,clumped + 3.4 CT;EH,clumped + 4.1 CT;EH,clumped - 2.7 RareCP - ND RareCP - ND RareCP - ND RareCP ++ <1 ++ <1 + <1 + <1 + <1 + <1 + 2.1 + 3.0 + 2.3 + 4.1 + 2.5FewCP FewCP CPwithout TF CPwithout TF CPwithout TF CH;PS CH;PS CH;PS CH;PS CH;PS CH;PSParticlematuration Particle maturation Particlematuration Particlematuration Particlematuration Particle maturation Head Head Head Lysis Lysis Lysis Lysis Lysisregulation Lysisregulation Tailfiber Tailfiber Tailfiber Baseplate ortube Baseplate ortube Baseplate ortube Baseplate ortube Baseplate ortube Baseplate ortube + 5.4 CH;PS + 3.7CH;PS + 4.9 CH;PS + 2.6CH;TT + 5.2 CTwithHC + 2.6 CPwithFH+UsT(ST) + 5.6 CPwithFH+UsT(ST) + 5.2 CHwithshorttail(PS) + 4.1CH;PS;US(TT) + 4.3CH;PS;US + 4.9CH;PS;US(ST;TT) + 6.3CH;TSS;US;PS(ST;TT) + 4.5CH;TSS;US;PS(TT)+ 2.0CH;TSS;US;PS + 4.5CH;TSS;US;PS(ST) + 3.4CH;TSS;US;PS(ST;TT) + 3.7CH;TSS;PS;CoTwithPS + 4.9CH;TSS;PS(CoTwithPS) + 5.9 CH;PS;CPwithFH+UsT + 4.6 CH;PS + 5.8CH;PS + 3.7CH;PS + 3.9 CTwithHC;EH - <1 RareCP - <1 RareCP + 3.6CT;EH + 5.5CT;EH + 3.8CT;EHBaseplate ortube Baseplate ortube Baseplate ortube Sheath Head Baseplate Baseplate Taillength Baseplate ortailstability Baseplate ortailstability Baseplate ortailstability Tailstability Tailstability Tailstability Tailstability Tailstability Tailstability Tailstability Baseplate Baseplate ortube Baseplate ortube Baseplate ortube Head Lategeneexpression Lategeneexpression Head Head Head47I 141 I 34I 128 I 132 I 57I 150 1I 621I 108 II 23III 144 III 73III 134 III 31IV 56IV 33IV 72IV 66IV 49IV 74IV 75IV 21IV 35IV 171 IV4 4 4? 17 17 17 17 2 12 19? 19 19?\", 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  17 17 17 17 2 12 19? 19 19? 20 3 16 3 3 135 V 107 V 45V 180 V 115 V 19VI 138 VI 79VI 26VI 32VI 109 VI 27VI 39VI 53VI 22VI 116 VI 30VII 113VII 142 VIII 139VIII 140VIII 71VIII 13IX 178 X-2 105X-2 16X-1 64X-1 136 X-121 21 21 23 5 6 24 24 24 7 7 7 7 7 25 25 26? 26 26 26 8 10 10 9 9 9VOL.45,1983 l l l', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1124 WALKER ANDWALKER TABLE 1-Continued Mutant Linkage Host SBP\\' PhagestructureSd Probable defect Gene\" no.cluster\" 1ysisb B\\'Paesrcue 131 X-1 + ND CT;EH Head 9 43X-1 + 5.1 CT;EH Head 9 112 X-1 + 5.9 CT;EH Head 9 aSeeFig.2. bDescribed inWalkerandWalker (50).+,Lysisoccurred; -,lysisdidnotoccur;++,iysisoccurred prematurely. \\'SBP,Serumblocking power,expressed asthedefectivelysatedilutiongiving50%residual k(seetext).ND, Notdone. dCH,Complete heads;CP,complete particles; CT,complete tails;CoT,contracted tails;EH,emptyheads; FH,fullheads;HC,head-neck connectors; PS,polysheath; ST,stackedtails;TF,tailfibers;TSS,tailswith slipped sheaths; TT,tailtubes;US,unit-length sheaths; UsT,unstable tails(tailsinvarious stagesof contraction). Parentheses denotethatfewofthesestructures wereseen(seetext). eNumbers upto10assigned byScott(39);numbers 17and19through26wereassigned byWalkerandWalker (50;thispaper).Queriesdenoteuncertain genedesignations. baseplate butonlytheabilityofthetubetobe builtuponit.)Thebaseplate ofP1(Fig.1)is muchthinnerthanthatofT4andhasnotbeen seenunattached totheP1particle; therefore, the defectoftheseclusterIVandVmutants could beineitherbaseplate ortubeformation. In addition tothisdefect,am49doesnotshow serumblocking power,indicating thattailfibers arenotproduced. Thismutation islocatednext tomutations 33,72,and66,wheretheonly . b !hI.1T.i.t ,,,,<Fa\\'l\\'t.>,\\'\\'.~~4at{@-, N.\\'..Z  ^9,jGx..bvi~Ajk ~~~~~~~~I _^................... O......-.defectistailfiberproduction; therefore, am49 maybepolaronthesemutations (seebelow). InT4,atleast15genesareinvolved inbase- plateformation (53);therefore, ifthesenineP1 mutants(49,74,75,21,35,171,135,107,and45) aredefective inbaseplate formation, theymay bedefective indifferent components ofthebase- plateandrepresent severalgenes.Insupportof thisisthefactthatmutants 49,74,75,21,35, and171areinonelinkage cluster(IV)and -: FIG.3.Micrograph ofP1am66defective lysate.InsetshowsasingleparticlefromP1am72defective lysate. Allparticles lacktailfibers.Contraction ofsheathsisanartifactduetopreparation procedures. Bars,100nm.J.VIROL. \"in.t  l?. 4', \"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1125 FIG.4.Micrograph ofP1am35defective lysate.Thislysateischaracterized bypolysheath (PS)andnormal heads,manyofwhichareemptyduetopreparation procedures. Bar,100nm. mutants 135,107,and45areinanother (V),and therestriction fragments thatrescuethesetwo groupsofmutants areseparated byotherrestric- tionfragments thatcarryforothergenes(see Fig.17).Basedoncomplementation tests(49), thesemutants inclusterIVrepresent atleast threegenes(genes3,16,and20;seeFig.17). :. X..:.* t.'s~~~~~~~4;w-~~~~Zk-j--Therefore, theninemutants probably represent atleastfourgenes(thefourthbeinggene21), someofwhichcouldcodeforbaseplate compo- nents,butonlyoneofwhichcouldcodeforthe tube,asitisprobably (byanalogy withother phages)composed ofonlyonepolypeptide spe- cies. 0;..--71.l.ilt .*l.I..4 N FIG.5.Micrograph ofP1aml80defective lysateshowing anormallengthtailtube(TT)aswellasfullheads ofbothPlBandPlS.Bar,100nm.VOL.45.1983  I  4\", \"page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1126 WALKER ANDWALKER Mutant180(inclusterV;Fig.5)showscom- pleteheadsandtailtubeswithout sheaths. Therefore, thedefectofam180isprobably in sheathsubunitproduction. Nobaseplates were seenattached tothetubes.Thisisprobably due tothebaseplates detaching fromthecompleted tubesbecause, inamber'P1lysates,contracted tailswithout baseplates areoftenseen(47). Alternatively, because am45,whichhasadefect ineitherthebaseplate orthetube,doesnot complement am180,itmaybethatam180is polaronam45andistherefore missing both sheathsandbaseplates. Mutants 19and138(bothincistron5incluster VI)showparticles whicharecomplete except thatallofthetailsaresemicontracted orcon- tracted(Fig.6).Thedefectofthesemutants is probably inabaseplate component thatisneces- saryforbaseplate stability. Thatis,eitherthis component isofminorimportance forbaseplate assembly orthemutation occursatalocation in thegenethatallowsformation ofanalmost complete polypeptide thatdoesnotprevent baseplate assembly. Ineithercase,tailassembly andhead-tailjoiningcouldoccur,buttheresult- antbaseplate wouldbeunstable andcausetail contraction. Mutants withmutations ingene9in T4,whichhaveadefectinbaseplate completion (25,27),showaphenotype similar(12)tothatof theseP1mutants. Because thephenotype of theseP1mutants isdifferent fromthatofthe ninemutants (inclusters IVandV)discussed above,theycouldbedefective inadifferent baseplate component andrepresent adifferent gene.Thisissupported bythefactthattheyare inadifferent linkageclusterandarephysically separated byotherP1genome restriction frag- mentsthatcarryothergenes(seeFig.17). Mutant79inclusterVIproduces polysheath andcomplete headsattached toshorttails. Thesetailshavesheathsofvariouslengths(Fig. 7)withbaseplates andtailfibersattached, andit isapparent thatthedefectofam79affectstail length.Insomeinstances thetailsaresoshort thatalthough abaseplate isseen,itcannotbe determined whether anysheathsubunits are present. Mutants 26,32,and109inclusterVIproduce complete heads,polysheath, andunit-length sheaths(Fig.8).Thesesheaths (seenonlyin lysatesofthesemutants andmutants27,39,53, 22,and116;nextparagraph) aresimilarinlength tothesheaths oncontracted tails,buttheyhave notailtubeorbaseplate. Thesestructures indi- catethatbaseplates andtubeswereformedbut subsequently slipped offthesheath. Sucha phenotype suggests eitherthatanunstable base- platewasformedandthatthedefectofthese mutants isinabaseplate component orthat thereisadefectatthetopofthetailsimilar toFIG.6.Micrograph ofP1aml9defective lysateshowingparticles withtailsthatalwayshavecontract- edsheaths. Bar,100nm. oneofthedefectsdescribed forthefollowing mutants. Mutants 27,39,53,22,and116(inclusterVI) produce complete heads,somepolysheath, and unattached tailsthathavebaseplates, tailfibers, andnormal-length tubes(Fig.9).However, the sheathonthetubesiscontracted andhasslipped downfromthetopofthetubes,andinsome casessomeofthesheathmaterial islost.In addition, unit-length sheaths arealsoseen, whichcouldarisefromthesheaths slipping completely offthetubes.Thisphenotype sug- geststhatthedefectofthesemutants isintail stability. Mutants 30and113(whichareintheJ.VIROL.\", 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1127 FIG.7.Micrographs ofP1am79defective lysate showing particles withshorter thannormal length tails.Bar,100nm. samecistroninclusterVII)haveasimilarphe- notype(although nounit-length sheaths were seen),suggesting thattheirdefectisintail stability also.However, inaddition tocomplete heads, polysheath, andtailswithslipped sheaths, thesemutants alsoproduce contracted tailswithpolysheath attached (Fig.10).These twogroupsofmutations (27,39,53,22,and116, and30and113)arelocated intwodifferent linkage clusters (Fig.2);therefore, they are probably intwodifferent genes.InT4,two genes (3and15)arerequired fortailstability. Thegene3product isaddedtothetopofthe tube,whereitactsastheterminator forsheath polymerization. Thegene15product thenforms astablebondbetween thetubetopandthetop ofthesheathand,atthesametime,createsthe headattachment site(25,29).Thetwo(orthree, ifmutations 26,32,and109areincluded) groups ofP1mutations affecting tailstability mayrepre-sentgeneswithfunctions similartothoseofT4 genes3and15.Forexample, theP1genescould codefor(i)thetailcapstructure occasionally seenonP1tailsunattached toheadsand(ii)the ringlike structure (seenattached tothebaseof theneckontailswhereabnormal contraction haslefttheproximal partofthetubeexposed; Fig.9,inset),whichisthought tostabilize the extended sheathandanchorittothetailtube (47).Inthisregard,thecontracted tailswith polysheath attached seenintheclusterVII mutants couldariseifthetailcapwasmissing andsheathsubunits continued topolymerize ontothetail,pastthetopofthetube. Tailtubesseparate fromsheaths areseenin lysatesofsomeofthemutants inlinkagecluster VI,wherethetailsarenotattached totheheads (Table1;Fig.9and11).However, sincethey tendtooccurinaggregates, theyarenotdistrib- utedevenlyonthegrid;thus,theirpresence may beoverlooked inlysatesofothermutants. This mightexplainwhynonewereseeninam32and am53lysates. However, withthesemutants, unit-length sheaths withorwithout tailtubes wereseen,whichconfirms thattubesaremade. Therefore, theabsence oftailtubeaggregates doesnotchangeourconclusions concerning the defectsofthesemutants. Additional structures seeninlysatesofsomeofthemutants incluster VIarestacked tails(Table1;Fig.6and9). Thesewereseeninfrequently, andtheirsignifi- canceisunknown. Mutant 142(inclusterVIII)produces com- pleteparticles withsemicontracted orcontract- edtails(similar tothoseseenwithaml9and am138),complete heads,andpolysheath. There- fore,thedefectofam142isprobably inbase- plateformation. Othermutants inclusterVIII (am139, aml40,andam7l)onlyproduce com- pleteheadsandpolysheath. Theirdefectcould therefore beinbaseplate ortubeformation. Itis notknownhowmanycistrons thesefourmu- tantsrepresent. Iftheyareallinthesame cistron,theirdefectcouldbeinbaseplate forma- tionwitham142locatedinthedistalpartofthis cistron. Mutations affecting headassembly. Mutations affecting headassembly arelocatedinlinkage clustersII,V,IX,andX-1.Mutants inclusterII (150,62,and108;Fig.12)andcluster X-1(16, 64,136,131,43,and112;Fig.13)produce complete tailsandunattached emptyheadsin head/tail ratiosof1:1to1:3.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': [\"page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1121 SBAwasamodification oftheDeMars (9)method as usedbyIsraeletal.(22)forP22andWardetal.(52)for T4;itcompares ratesratherthanfinallevelsofserum inactivation byblocking antigens. Serialtwofold dilutions weremadeofthelysates aftertheyhadbeentreatedwithchloroform for20min at37°Ctokillanyviablecells.Cellsinfected with mutants (inlinkageclusters IIIandX-2;Fig.2)thatdo notcausecelllysiswerelysedbysonication atthe timewhenlysisoccurred withamber'-infected cells. Cellsinfected withothermutants andwithP1virwere notsonicated, except asdescribed belowforPlvir,to obtain astandard curve.Reaction mixtures contained 0.45mlofalysatedilution and0.1mloftesterphage, towhich wassubsequently added0.45mlofantiphage serum.Thetesterphage wasP1kcatafinalconcentra- tionof3x105PFU/ml; theturbidplaques couldbe distinguished fromtheclearplaquesofanyrevertants presentintheP1v'iramberlysates. Thefirst-order rate constant (k)forphageinactivation oftheundiluted hyperimmune serumat37°Cwas1,500min-'.This serumhadakof0.1min-'inthefinaldilution ofthe reaction mixture. Thereaction mixtures wereassayed (onstrainSh-16)forsurviving testerphageattheend of46min,whentesterphagesurvival wasabout1%in theabsence ofSBA.Thiscontrolwithout SBAwas alwaysincluded ineachgroupoftestsdoneondiffer- entdays.Thisinactivation wasassumed tobeexpo- nentialthroughout the46-minincubation. Toobtainstandard curves, lysatesofamber' Plvir grown onstrainSh-15 wereusedasSBA;thephage wereinactivated byeithersonication orUVtreatment sotheplaques wouldnotobscure thoseofthecompet- ingP1kctesterphage. Theresultsoftheserumblocking assays were plotted aspercent residual k=(1-kfinal/kinitial) X100 =(1-lnSISd/lnSISO) x100,whereS/SO =survival of thetesterphageinthecontrolafter46minwithserum butnoSBA,andSISd =survival ofthetesterphage after46minwithbothserumandSBAatdilution dof thelatter.Ratherthanshowing graphsforthe55 mutants tested,comparisons weremadeusingthe dilution atwhichresidual k=50%. Electron microscopy. A0.4-misampleofconcentrat- edorunconcentrated lysateofthephage-infected cells wasdialyzed for15minbyplacingitonamembrane filter(Millipore Corp.; average poresize,25nm) floating onP1buffer(ratherthanwater,whichcaused headdisruption). Thisdialysis wasdoneatroom temperature todecrease tailcontractions obtained by dialysis at4°C.Adropofsample wasplaced onaglow- discharged, carbon-coated, 400-mesh coppergridand allowed tostandfor2min.Asmalldropofsodium phosphotungstate (1%;pH7.0)wasaddedandleftfor 30sbeforetheexcessliquid wasremoved withfilter paper.AJEOL100Belectron microscope withananti- contamination device wasusedat80kV.Micrographs weretakenatinitialmagnifications of10,000 to50,000 onKodak4463film. RESULTS Weexamined, byelectron microscopy, ly- satesof55P1amber mutants grownunder nonpermissive conditions todetermine their phenotypes. Insuchlysates, mutants defectiveinsomeaspectoftailassembly shouldproduce normalheadsandoneofavarietyoftailcompo- nentcombinations. Similarly, mutants defective insomeaspectofheadassembly shouldproduce normaltailsandeithernoheadsoroneofa varietyofdefective headstructures.\", 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  RESULTS Weexamined, byelectron microscopy, ly- satesof55P1amber mutants grownunder nonpermissive conditions todetermine their phenotypes. Insuchlysates, mutants defectiveinsomeaspectoftailassembly shouldproduce normalheadsandoneofavarietyoftailcompo- nentcombinations. Similarly, mutants defective insomeaspectofheadassembly shouldproduce normaltailsandeithernoheadsoroneofa varietyofdefective headstructures. Inaddition, weusedserumblocking assaystodetectthe presence oftailfibersinthedefective lysates. Of103P1ambermutantspreviously testedfor lysisofnonpermissive cells(50),83causedcell lysisatthenormallysistimeforPl.These83 mutations areinlinkageclusters(Fig.2)1,II,V,VI,VII,VIII,andIXandpartsofIVandX(i.e., X-1).Byanalogy withotherphages,thesemu- tantswouldbedefective inlategenesinvolved inparticlemorphogenesis. Thenumberofcistrons represented bythese 83mutants isunknown. Therefore, 47mutants (underlined inFig.2)werechosenforstudyby electron microscopy fromthemaximum possi- blenumber ofcistrons inthefollowing way. These47mutants represent (i)atleastone mutation fromeachofthegroupings ofmuta- tions(showninFig.2asvertically stackedunder veryclosehashmarksorwithinparentheses) thathadbeenfound(49)nottoshowcomple- mentation and(ii)eachofthesinglemutations outsidethesegroupings. Inaddition, eightmu- tants(alsounderlined inFig.2),representing thosethatcannotcausecelllysis(linkage clus- tersIIIandX-2)orthatcausepremature cell lysis(linkage clusterIV),werealsoincluded in thisstudy. Serumblocking assays.P1tailfibersareonly1 to2nminthickness (47)andsoareonlyseenby electron microscopy whentheyareattached to phageparticles andwhenthestainingconditions aresuchastogiveoptimal contrast between the fibersandthebackground (Fig.1).Therefore, because inotherphageswithcontractile tails, suchasT4,themajority ofSBAcanbeattribut- edtotailfibers,serumblocking assayswere donetoconfirm thepresence ofattached fibers ortodetectthepresence ofunattached fibersin theP1defective lysates.Results wereplottedas percentresidual kversuslysatedilution asde- scribedabove,andthedilutiongiving50%resid- ualkwasdetermined foreachdefective lysate. Thesedilutions weredesignated SBAvalues. Aswouldbeexpected foraphagesuchasP1 thathasacomplex morphology andtherefore wouldcodeforvariousstructures inaddition to tailfibers,themajority ofmutants showedsig- nificant serumblocking power(similar tothatof thecontrol Plvirlysates), indicating thattail fiberswerepresent(Table1;Fig.2). Thosemutants thatgavenegative results(i.e., SBAvaluesoflessthan1)havemutations in linkageclusterX-2andpartofclusterIV.Be- causeclusterX-2mutants didnotlysethehostVOL.45,1983', \"page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1122 WALKER ANDWALKER cells,theselysatesweretestedforSBAboth beforeandaftersonication tocausecelllysis; negative resultswereobtained inbothinstances, indicating thatthesenegative resultswerenot duetoafailureofthecellstolyse.(Mutants in clusterIIIalsodidnotlysethehostcells,but gavepositive SBAresults,indicating thatsoni- cationdoesnotadversely affectSBAactivity.) ClusterX-2mutations havebeenshowntobein thesamecistron(cistron10),andmutants repre- sentingthiscistronareprobably defective inthe expression oflategenes(50).SinceSBAsare lategeneproducts, thiswouldexplainwhynega- tiveresultsareobtained withdefective lysatesof thesemutants. Allofthosemutants (74,75,21,35,and171) locatedintherightpartoflinkageclusterIV gavenormallevelsofSBA.However, mutants (31,56,33,72,66,and49)locatedintheleftpart ofclusterIVgavenegative SBAresults(i.e., SBAvaluesoflessthan1),indicating thatthese mutants produce few,ifany,tailfibers.The significance oftheseresultsobtained withthe lattersixmutants isdiscussed inthenextsec- tion. Serumblocking assaysconfirmed electron mi- croscopic examination (seebelow)ofthemu- tantsinthecasesinwhichnormalamounts of sheathed tailswereproduced. Thatis,tailfibers wereseenbyelectron microscopy onparticles produced byallofthemutants thatproduced normalamounts ofsheathed tails(eitherextend- edorcontracted), exceptmutants128,150,33, 72,66,and138.Thepreparations ofmutants 128,150,and138werenotgoodenough to determine thepresence oftailfibers,butthese mutants allshowed serumblocking power,and foreachthereareallelicmutants(i.e.,mutants thatdidnotshowcomplementation [49])where tailfiberscouldbeseen.Therefore, thesemu- tantsarepresumed tohavetailfibersattached to theirtails.Preparations ofmutants, 33,72,and 66inlinkageclusterIVweregoodenough to detectanytailfibersifpresent, andthesemu- tantsdidnotshowserumblocking power,con- firmingthatthesemutants donotproduce tail fibers. SBAwasproduced byallmutants (except am49;seenextsection) thatdidnotproduce sheathed tails(andwheretailfiberstherefore couldnotbedetected byelectronmicroscopy), indicating thattailfibersweremadeeventhough nosheathed tailwasavailable fortheirattach- ment. Electron microscopy ofP1ambermutants. Defective lysatesofthe55mutants wereexam- inedbyelectron microscopy before,aswellas after,differential centrifugation toconcentrate anyphagestructures present.Amber' P1vir controls(Fig.1)prepared atthesametimeasthedefective lysateswereexamined inparallel. The typesofphagestructures seenbeforeandafter centrifugation weresimilarexceptthatwhere unattached tailshadsheathswhichhadslipped downfromthetopofthetailbeforecentrifuga- tion(mutants inclusterVIIandtherightpartof VI;seebelow), therewasadecrease inthe proportion oftailsrelative topolysheath after centrifugation. Thisdecrease isprobably caused bythesheathsubunits dissociating fromthetail tubesaftercentrifugation andpolymerizing to formpolysheath duringresuspension ofthecen- trifuged pelletat4°Cfor2days,conditions whichallowsheathsubunits topolymerize inT4 (1).Wheretails(eitherattached toorfreefrom heads)hadextended sheathsbeforecentrifuga- tion,themajority hadcontracted orsemicon- tractedsheathsaftertheconcentration steps.\", 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Thisdecrease isprobably caused bythesheathsubunits dissociating fromthetail tubesaftercentrifugation andpolymerizing to formpolysheath duringresuspension ofthecen- trifuged pelletat4°Cfor2days,conditions whichallowsheathsubunits topolymerize inT4 (1).Wheretails(eitherattached toorfreefrom heads)hadextended sheathsbeforecentrifuga- tion,themajority hadcontracted orsemicon- tractedsheathsaftertheconcentration steps. Lysateswhichshowedfullheadsbeforecentrif- ugationshowedfullheadsaftercentrifugation if theheadswereattached totails.However, a highproportion ofemptyheadswereseenafter centrifugation ifthefullheadswerenotattached totailsbeforecentrifugation. Resultsaresum- marized inTable1andFig.17. Mutations affecting tailassembly. Mutations affecting tailassembly arelocated inlinkage clusters IVthrough VIII.Eachtail-defective mutantproduces complete heads.Theheadsof mutants 33,72,and66(inlinkageclusterIV)are attached totailswhicharecomplete exceptthat theyaremissingtailfibers(Fig.3).Thesemu- tantsdonotshowserumblocking power,which confirms thatfibersarenotpresenteitherat- tachedorunattached totheparticles. However, itisunclear whether thewholetailfiberis missingorjustthedistalportion,because short projections, whichcouldbepartofthebase- plate,areoccasionally seenprotruding fromthe baseofthesheath, andtheserumblocking assaysdidnotdistinguish between thesepossi- bilities.Therefore, itremainsunknown whether thekinkedP1tailfiberismadeofmorethanone polypeptide andwhether themutations ofmu- tants33,72,and66areinthesamegene. Mutants 49,74,75,21,35,and171incluster IVandmutants 135,107,and45inclusterV producecomplete headsandpolysheath (Fig.4). InT4phage,completed baseplates arerequired forthetubepolypeptide toinitiatepolymeriza- tiontoformthetube,andthenthesheath polypeptide polymerizes around thetubeto formthetail(26).However, unlikethetube polypeptide, thesheathpolypeptide canpoly- merizebyitselftoformpolysheath. Therefore, thepresence ofpolysheath alonemeansthatthe defectisinthebaseplate, whereas thepresence ofpolysheath andbaseplates meansthatthe defectisinthetube(orbaseplate, ifthedefect hasnotdamaged thestructural integrity oftheJ.VIROL.', \"page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1123 TABLE 1.Phenotypic classification ofP1ambermutants Mutant Linkage Host Probable defect Gene no. cluster' ysisb SBP PhagestructuresP + 4.1 CPwithEH+UsT + 3.2 CPwithEH+UsT + 3.8 CPwithEH+UsT + 3.3 CPwithEH+UsT + 7.5 CPwithEH+UsT + 4.1 CP + 5.8CT;EH,clumped + 3.4 CT;EH,clumped + 4.1 CT;EH,clumped - 2.7 RareCP - ND RareCP - ND RareCP - ND RareCP ++ <1 ++ <1 + <1 + <1 + <1 + <1 + 2.1 + 3.0 + 2.3 + 4.1 + 2.5FewCP FewCP CPwithout TF CPwithout TF CPwithout TF CH;PS CH;PS CH;PS CH;PS CH;PS CH;PSParticlematuration Particle maturation Particlematuration Particlematuration Particlematuration Particle maturation Head Head Head Lysis Lysis Lysis Lysis Lysisregulation Lysisregulation Tailfiber Tailfiber Tailfiber Baseplate ortube Baseplate ortube Baseplate ortube Baseplate ortube Baseplate ortube Baseplate ortube + 5.4 CH;PS + 3.7CH;PS + 4.9 CH;PS + 2.6CH;TT + 5.2 CTwithHC + 2.6 CPwithFH+UsT(ST) + 5.6 CPwithFH+UsT(ST) + 5.2 CHwithshorttail(PS) + 4.1CH;PS;US(TT) + 4.3CH;PS;US + 4.9CH;PS;US(ST;TT) + 6.3CH;TSS;US;PS(ST;TT) + 4.5CH;TSS;US;PS(TT)+ 2.0CH;TSS;US;PS + 4.5CH;TSS;US;PS(ST) + 3.4CH;TSS;US;PS(ST;TT) + 3.7CH;TSS;PS;CoTwithPS + 4.9CH;TSS;PS(CoTwithPS) + 5.9 CH;PS;CPwithFH+UsT + 4.6 CH;PS + 5.8CH;PS + 3.7CH;PS + 3.9 CTwithHC;EH - <1 RareCP - <1 RareCP + 3.6CT;EH + 5.5CT;EH + 3.8CT;EHBaseplate ortube Baseplate ortube Baseplate ortube Sheath Head Baseplate Baseplate Taillength Baseplate ortailstability Baseplate ortailstability Baseplate ortailstability Tailstability Tailstability Tailstability Tailstability Tailstability Tailstability Tailstability Baseplate Baseplate ortube Baseplate ortube Baseplate ortube Head Lategeneexpression Lategeneexpression Head Head Head47I 141 I 34I 128 I 132 I 57I 150 1I 621I 108 II 23III 144 III 73III 134 III 31IV 56IV 33IV 72IV 66IV 49IV 74IV 75IV 21IV 35IV 171 IV4 4 4? 17 17 17 17 2 12 19? 19 19?\", 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  17 17 17 17 2 12 19? 19 19? 20 3 16 3 3 135 V 107 V 45V 180 V 115 V 19VI 138 VI 79VI 26VI 32VI 109 VI 27VI 39VI 53VI 22VI 116 VI 30VII 113VII 142 VIII 139VIII 140VIII 71VIII 13IX 178 X-2 105X-2 16X-1 64X-1 136 X-121 21 21 23 5 6 24 24 24 7 7 7 7 7 25 25 26? 26 26 26 8 10 10 9 9 9VOL.45,1983 l l l', 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1124 WALKER ANDWALKER TABLE 1-Continued Mutant Linkage Host SBP\\' PhagestructureSd Probable defect Gene\" no.cluster\" 1ysisb B\\'Paesrcue 131 X-1 + ND CT;EH Head 9 43X-1 + 5.1 CT;EH Head 9 112 X-1 + 5.9 CT;EH Head 9 aSeeFig.2. bDescribed inWalkerandWalker (50).+,Lysisoccurred; -,lysisdidnotoccur;++,iysisoccurred prematurely. \\'SBP,Serumblocking power,expressed asthedefectivelysatedilutiongiving50%residual k(seetext).ND, Notdone. dCH,Complete heads;CP,complete particles; CT,complete tails;CoT,contracted tails;EH,emptyheads; FH,fullheads;HC,head-neck connectors; PS,polysheath; ST,stackedtails;TF,tailfibers;TSS,tailswith slipped sheaths; TT,tailtubes;US,unit-length sheaths; UsT,unstable tails(tailsinvarious stagesof contraction). Parentheses denotethatfewofthesestructures wereseen(seetext). eNumbers upto10assigned byScott(39);numbers 17and19through26wereassigned byWalkerandWalker (50;thispaper).Queriesdenoteuncertain genedesignations. baseplate butonlytheabilityofthetubetobe builtuponit.)Thebaseplate ofP1(Fig.1)is muchthinnerthanthatofT4andhasnotbeen seenunattached totheP1particle; therefore, the defectoftheseclusterIVandVmutants could beineitherbaseplate ortubeformation. In addition tothisdefect,am49doesnotshow serumblocking power,indicating thattailfibers arenotproduced. Thismutation islocatednext tomutations 33,72,and66,wheretheonly . b !hI.1T.i.t ,,,,<Fa\\'l\\'t.>,\\'\\'.~~4at{@-, N.\\'..Z  ^9,jGx..bvi~Ajk ~~~~~~~~I _^................... O......-.defectistailfiberproduction; therefore, am49 maybepolaronthesemutations (seebelow). InT4,atleast15genesareinvolved inbase- plateformation (53);therefore, ifthesenineP1 mutants(49,74,75,21,35,171,135,107,and45) aredefective inbaseplate formation, theymay bedefective indifferent components ofthebase- plateandrepresent severalgenes.Insupportof thisisthefactthatmutants 49,74,75,21,35, and171areinonelinkage cluster(IV)and -: FIG.3.Micrograph ofP1am66defective lysate.InsetshowsasingleparticlefromP1am72defective lysate. Allparticles lacktailfibers.Contraction ofsheathsisanartifactduetopreparation procedures. Bars,100nm.J.VIROL. \"in.t  l?. 4', \"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1125 FIG.4.Micrograph ofP1am35defective lysate.Thislysateischaracterized bypolysheath (PS)andnormal heads,manyofwhichareemptyduetopreparation procedures. Bar,100nm. mutants 135,107,and45areinanother (V),and therestriction fragments thatrescuethesetwo groupsofmutants areseparated byotherrestric- tionfragments thatcarryforothergenes(see Fig.17).Basedoncomplementation tests(49), thesemutants inclusterIVrepresent atleast threegenes(genes3,16,and20;seeFig.17). :. X..:.* t.'s~~~~~~~4;w-~~~~Zk-j--Therefore, theninemutants probably represent atleastfourgenes(thefourthbeinggene21), someofwhichcouldcodeforbaseplate compo- nents,butonlyoneofwhichcouldcodeforthe tube,asitisprobably (byanalogy withother phages)composed ofonlyonepolypeptide spe- cies. 0;..--71.l.ilt .*l.I..4 N FIG.5.Micrograph ofP1aml80defective lysateshowing anormallengthtailtube(TT)aswellasfullheads ofbothPlBandPlS.Bar,100nm.VOL.45.1983  I  4\", \"page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1126 WALKER ANDWALKER Mutant180(inclusterV;Fig.5)showscom- pleteheadsandtailtubeswithout sheaths. Therefore, thedefectofam180isprobably in sheathsubunitproduction. Nobaseplates were seenattached tothetubes.Thisisprobably due tothebaseplates detaching fromthecompleted tubesbecause, inamber'P1lysates,contracted tailswithout baseplates areoftenseen(47). Alternatively, because am45,whichhasadefect ineitherthebaseplate orthetube,doesnot complement am180,itmaybethatam180is polaronam45andistherefore missing both sheathsandbaseplates. Mutants 19and138(bothincistron5incluster VI)showparticles whicharecomplete except thatallofthetailsaresemicontracted orcon- tracted(Fig.6).Thedefectofthesemutants is probably inabaseplate component thatisneces- saryforbaseplate stability. Thatis,eitherthis component isofminorimportance forbaseplate assembly orthemutation occursatalocation in thegenethatallowsformation ofanalmost complete polypeptide thatdoesnotprevent baseplate assembly. Ineithercase,tailassembly andhead-tailjoiningcouldoccur,buttheresult- antbaseplate wouldbeunstable andcausetail contraction. Mutants withmutations ingene9in T4,whichhaveadefectinbaseplate completion (25,27),showaphenotype similar(12)tothatof theseP1mutants. Because thephenotype of theseP1mutants isdifferent fromthatofthe ninemutants (inclusters IVandV)discussed above,theycouldbedefective inadifferent baseplate component andrepresent adifferent gene.Thisissupported bythefactthattheyare inadifferent linkageclusterandarephysically separated byotherP1genome restriction frag- mentsthatcarryothergenes(seeFig.17). Mutant79inclusterVIproduces polysheath andcomplete headsattached toshorttails. Thesetailshavesheathsofvariouslengths(Fig. 7)withbaseplates andtailfibersattached, andit isapparent thatthedefectofam79affectstail length.Insomeinstances thetailsaresoshort thatalthough abaseplate isseen,itcannotbe determined whether anysheathsubunits are present. Mutants 26,32,and109inclusterVIproduce complete heads,polysheath, andunit-length sheaths(Fig.8).Thesesheaths (seenonlyin lysatesofthesemutants andmutants27,39,53, 22,and116;nextparagraph) aresimilarinlength tothesheaths oncontracted tails,buttheyhave notailtubeorbaseplate. Thesestructures indi- catethatbaseplates andtubeswereformedbut subsequently slipped offthesheath. Sucha phenotype suggests eitherthatanunstable base- platewasformedandthatthedefectofthese mutants isinabaseplate component orthat thereisadefectatthetopofthetailsimilar toFIG.6.Micrograph ofP1aml9defective lysateshowingparticles withtailsthatalwayshavecontract- edsheaths. Bar,100nm. oneofthedefectsdescribed forthefollowing mutants. Mutants 27,39,53,22,and116(inclusterVI) produce complete heads,somepolysheath, and unattached tailsthathavebaseplates, tailfibers, andnormal-length tubes(Fig.9).However, the sheathonthetubesiscontracted andhasslipped downfromthetopofthetubes,andinsome casessomeofthesheathmaterial islost.In addition, unit-length sheaths arealsoseen, whichcouldarisefromthesheaths slipping completely offthetubes.Thisphenotype sug- geststhatthedefectofthesemutants isintail stability. Mutants 30and113(whichareintheJ.VIROL.\", 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1127 FIG.7.Micrographs ofP1am79defective lysate showing particles withshorter thannormal length tails.Bar,100nm. samecistroninclusterVII)haveasimilarphe- notype(although nounit-length sheaths were seen),suggesting thattheirdefectisintail stability also.However, inaddition tocomplete heads, polysheath, andtailswithslipped sheaths, thesemutants alsoproduce contracted tailswithpolysheath attached (Fig.10).These twogroupsofmutations (27,39,53,22,and116, and30and113)arelocated intwodifferent linkage clusters (Fig.2);therefore, they are probably intwodifferent genes.InT4,two genes (3and15)arerequired fortailstability. Thegene3product isaddedtothetopofthe tube,whereitactsastheterminator forsheath polymerization. Thegene15product thenforms astablebondbetween thetubetopandthetop ofthesheathand,atthesametime,createsthe headattachment site(25,29).Thetwo(orthree, ifmutations 26,32,and109areincluded) groups ofP1mutations affecting tailstability mayrepre-sentgeneswithfunctions similartothoseofT4 genes3and15.Forexample, theP1genescould codefor(i)thetailcapstructure occasionally seenonP1tailsunattached toheadsand(ii)the ringlike structure (seenattached tothebaseof theneckontailswhereabnormal contraction haslefttheproximal partofthetubeexposed; Fig.9,inset),whichisthought tostabilize the extended sheathandanchorittothetailtube (47).Inthisregard,thecontracted tailswith polysheath attached seenintheclusterVII mutants couldariseifthetailcapwasmissing andsheathsubunits continued topolymerize ontothetail,pastthetopofthetube. Tailtubesseparate fromsheaths areseenin lysatesofsomeofthemutants inlinkagecluster VI,wherethetailsarenotattached totheheads (Table1;Fig.9and11).However, sincethey tendtooccurinaggregates, theyarenotdistrib- utedevenlyonthegrid;thus,theirpresence may beoverlooked inlysatesofothermutants. This mightexplainwhynonewereseeninam32and am53lysates. However, withthesemutants, unit-length sheaths withorwithout tailtubes wereseen,whichconfirms thattubesaremade. Therefore, theabsence oftailtubeaggregates doesnotchangeourconclusions concerning the defectsofthesemutants. Additional structures seeninlysatesofsomeofthemutants incluster VIarestacked tails(Table1;Fig.6and9). Thesewereseeninfrequently, andtheirsignifi- canceisunknown. Mutant 142(inclusterVIII)produces com- pleteparticles withsemicontracted orcontract- edtails(similar tothoseseenwithaml9and am138),complete heads,andpolysheath. There- fore,thedefectofam142isprobably inbase- plateformation. Othermutants inclusterVIII (am139, aml40,andam7l)onlyproduce com- pleteheadsandpolysheath. Theirdefectcould therefore beinbaseplate ortubeformation. Itis notknownhowmanycistrons thesefourmu- tantsrepresent. Iftheyareallinthesame cistron,theirdefectcouldbeinbaseplate forma- tionwitham142locatedinthedistalpartofthis cistron. Mutations affecting headassembly. Mutations affecting headassembly arelocatedinlinkage clustersII,V,IX,andX-1.Mutants inclusterII (150,62,and108;Fig.12)andcluster X-1(16, 64,136,131,43,and112;Fig.13)produce complete tailsandunattached emptyheadsin head/tail ratiosof1:1to1:3.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:05 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'112'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994135'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'351ms'), (b'x-request-id', b'0f16a86ed2412927b73fc3bd615c0e92'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249b5af0a5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:05 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'112'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994135'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'351ms'), (b'x-request-id', b'0f16a86ed2412927b73fc3bd615c0e92'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249b5af0a5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Othermutants inclusterVIII (am139, aml40,andam7l)onlyproduce com- pleteheadsandpolysheath. Theirdefectcould therefore beinbaseplate ortubeformation. Itis notknownhowmanycistrons thesefourmu- tantsrepresent. Iftheyareallinthesame cistron,theirdefectcouldbeinbaseplate forma- tionwitham142locatedinthedistalpartofthis cistron. Mutations affecting headassembly. Mutations affecting headassembly arelocatedinlinkage clustersII,V,IX,andX-1.Mutants inclusterII (150,62,and108;Fig.12)andcluster X-1(16, 64,136,131,43,and112;Fig.13)produce complete tailsandunattached emptyheadsin head/tail ratiosof1:1to1:3.(Atotalof245to 873particles werecounted permutant,using electronmicrographs ofrandomly selected fields fromseveralpreparations.) Incomplementation tests(49),aml50and am62givenegative results. Because ofthisand because thephenotypes ofthetwomutants are similar, themutations thattheyrepresent areVOL.45,1983', \"page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1128 WALKER ANDWALKER t~~~~~ s P 1~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~J1 FIG.8.Micrograph ofP1am26defective lysate.Inaddition tonormalfullheads,there arepolysheath (PS) andunit-length sheaths(US)orfragments thereof. Thelongcurved structures arebacterial flagella. Bar,100nm. probably inthesamegene,whichwewillrefer toasgene4,adesignation originally madeby Scott(39)foram7.(Complementation results witham62andamlO8wereambiguous; there- fore,weareunsurewhether themutation in amlO8islocatedinthisgene.)Forsimilarrea- sonsandbecause themutants inlinkagecluster X-1arerescuedfromtwoadjacent fragments of theP1genome (seeFig.17),oneofwhichalso contains atleastpartoftheP1cigeneandthe otherofwhichisflankedbya0.62-kilobase-pair (kb)fragment whichcontains atleastpartof gene10,themutations inlinkageclusterX-1are assigned toasinglegene(gene9,adesignation originally madebyScott[39]foraml6),al- thoughthereisapossibility thatthesemutants couldrepresent morethanonegene. Theemptyheadsproduced bygene9and clusterIImutants aresimilar(insizeandinthe thickness ofthecapsidshell)toemptymature headsseeninamber' lysates. Therefore, they arepresumed tobematureheadsthatarelack- ingDNA,ratherthanproheads. InT4,mature, empty,unattached heads(and complete tails)areproduced bymutants with defectseitherinvariousheadcompletion genes (8,11,26)oringene49.Headcompletion genes areinvolved inneckassembly (gene13),DNA packaging (genes16and17),orunknown func- tions(genes4and65).Gene49codesforDNase andisrequired forDNApackaging (14,33). Headsofmutants inthesegenesareempty,eitherbecause theywerenotfilledwithDNAor because theywereunstably filledandsubse- quentlylosttheirDNA(8,17).Genesinother phagesthatareinvolved inheadcompletion are genes4,10,and26inP22(6,36),whichcodefor threeminorstructural proteins required forhead stability, andgeneDinX(24),whichcodesfora proteinthatisaddedtothematurecapsidshell andwhosepresence isalsorequired forDNA cutting(4,23). Fromtheirphenotype, theP1gene9mutants couldbedefective inoneofthesefunctions. In thisregard,Sternberg (personal communication) recentlyfoundthat,unlikemutantsingenes4,8, and23,gene9mutants cannotcleaveP1DNAat pac,thesiteforinitiation ofDNApackaging (seeFig.17);therefore, likegenes2and3in phageP22(31),theP1gene9productisinvolved intherecognition andcleavage ofP1DNAat pac.Withotherphages,conversion ofproheads tomatureheadsusually occursconcomitantly withDNApackaging. Gene9mutants apparent- lycannotinitiateDNApackaging. Therefore, in P1eitherconversion ofproheads tomature headscanoccurindependently ofDNApackag- ingorP1proheads resemble matureheadsin sizeandshellthickness. TheclusterIImutants couldalsobedefective inoneofthesefunctions (except forDNA cleavage because, asmentioned above,Stern- bergfoundthatclusterIImutants cancleaveP1 DNAatthepacsite).J.VIROL.\", 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1129 PS , * .\\'-4 .4 IN .;. t\\'.4 \\'.\\' 444  r .JE ~ ~  i~S T- taST~ FIG.9.Micrograph ofPlam27defective lysateshowing severalkindsoftailstructures. Notethetailtube aggregate (TT),polysheath (PS),tailswithslippedsheaths(TSS),andstackedtails(ST).Insetshows anamber\\' PtparticlefromaCsClgradient; afterprolonged exposure toCsCl,thereisabnormal tailcontraction, i.e.,the tailsheathhascontracted andsliddownthetube,butaringlike structure withorwithout somesheathsubunits remains atthejunction between theproximal endofthetubeandtheneck(seetext).Bar,100nm. FIG.10.Micrograph ofPtam3Odefective lysate. Polysheath appears tobeattached totheproximal end ofthischaracteristically contracted tail,whichother- wiseappears tobenormalinhavingbothabaseplate andtailfibers.Bar,100nm.Likethemutants inlinkage clusters IIandX- 1,asinglemutant(amll5) incluster Vandthe mutant(aml3)thatrepresents cluster IX(i.e., gene8)produce complete tailsandunattached emptyheads.Withaml3,thehead/tail ratiois 1:2(1,967particles werecounted). However, with amll5, veryfewheads were seen(the head/tail ratiowas1:30,with431particles count- ed),andinputphagecouldaccount forthese heads.Attached tothetailsofbothamll5and amt3mutants arestructures resembling the neckandhead-neck connector (Fig.14).The neckandhead-neck connector arepartofthe head.Thissuggests thattheheads wereoriginal- lyassembled, filledwithDNA,andattached to thetails,butwereunstable andsubsequently eitherdisintegrated (amllS) ordissociated fromp \\'-p--,-, *X--Lt-VOL.45,1983 IZ\"', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1130 WALKER ANDWALKER FIG.11.Micrograph ofP1amll6defective lysateshowing aggregates oftailtubes.Nosingletubeislonger thannormallength.Bar,100nm. thetail,leavingbehindatleastsomeoftheneck andhead-neck connector (aml3).Suchinstabil- itycouldbeduetotheabsence ofaminorhead protein, e.g.,aminorheadshellprotein(amllS) orahead-neck connector protein(aml3). Alter- natively, because sofewheadsareseenin amllSlysates, amllScouldbedefective inthe majorheadprotein. Inthiscase,thestructures seenwouldnotbeaneckandhead-neck connec- torbutanaberrant structure similartothatseen inT4(calledtheextension [8]),whichconsists of anaberrant accumulation onthetopofthetailof fiveproteins normally assembled intothehead aspartoftheneckandhead-neck connector or addedtothehead-neck connector afterhead-tail joining.Mutations affecting particlematuration. Muta- tionsaffecting particle maturation (47,141,34, 128,132,and57)comprise linkageclusterI.The mutants inclusterIdonotshowcomplementa- tion(49)andarerescued froma0.92-kbrestric- tionfragment oftheP1genome flanked by fragments carrying othergenes(seeFig.17); therefore, thesesixmutations areprobably in thesamegene(gene1).Particles produced by thesixmutants arecomplete exceptthat,inall ofthemutants except57,headstendtobeempty andthetailsareinvariousstagesofcontraction (Fig.15).Mutant57produces complete particles withatleastpartially filledheadsandextended sheaths(Fig.16).Thesephenotypes suggestthat thedefective polypeptide allowsassembly ofthe FIG.12.Micrograph ofP1amlO8defective lysate.Thetailsappeartobenormal;heads,notseenhere,are alsoproduced. Bar,100nm.J.VIROL. -V Gkll-.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1131 FIG.13.Micrograph ofP1am64defective lysate.Theheadsarethesamesizeasmatureheads,buttheyare alwaysempty;thetailsarenormal,theircontraction beinganartifactofpreparation procedures. Bar,100nm. complete particles butthattheseparticles are eithermorphologically complete butnoninfec- tive(am57)orunstable. Morphologically normal particles areproduced byotherphages, e.g.,P22 -RI i.^.. . ._l S ... FIG.14.Micrograph ofP1aml3defective lysate. Tailsarenormal exceptforaknobattheproximal end (arrow). Bar,100nm.mutants defective ingenes7,16,and20(6,36) andT4mutants (11,12,26)defective ingenes2, 4,11,12,50,64,and65(whichproduce atleast somemorphologically normalphages). Genes11 and12aretailgenes,codingforabaseplate component andshorttailfibers,respectively. Alloftheremaining geneproducts areincorpo- ratedintotheheadatearlystagesofassembly, buttheirfunctions vary;e.g.,someareinvolved intheDNAinjection process, andsome(inT4) areinvolved incleavage ofheadproteins. The P1mutants couldhaveoneoftheseheadgene defects(although itisnotknownwhether there iscleavage ofPlheadproteins). Alternatively, because oftheinstability ofthetail,theycould haveadefectinatailgene,e.g.,abaseplate gene.Thedifference observed between am57 andtheothermutants inthisclustercouldbe explained bytheam57mutation beinginthe distalpartofthegenetogiveamorecomplete polypeptide andhenceamorestableparticle. In supportofthisisthefactthatam57islocatedat oneendoftheclusterwithrespecttotheother fivemutants examined (Fig.2). Mutations affecting hostcelllysis.Cellsinfect- edseparately witheightmutants thatcannot causecelllysis(gene17,linkageclusterIII,or gene10,clusterX-2)orthatcausepremature celllysis(gene2,clusterIV)wereexamined by electron microscopy attheincubation time(110 min)atwhichlysiswouldbecomplete incells infected withlysis-positive mutants.VOL.45,1983 4 .1 4. -f....7-,-. IS6', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1132 WALKER ANDWALKER FIG.15.Micrograph ofP1am47defective lysate.Particles appear tobenormal exceptforemptyheads (maturesize);normaltailsareinvarious stagesofcontraction. Bar,100nm. Mutants (23,144,73,and134)having muta- tionsingene17showed intactcellsasexpected and,occasionally, somecellsthathadlostsome oftheircontents butwerefullofphageparticles. Veryrarely, somecomplete phage,whichcould havebeeninputphage, wereseenoutside the cells.Theseresults areconsistent withthefind- ings(50)thatartificially lysedcellsinfected with thesemutants produce viablephageparticles andgivepositive resultsinserumblocking as- saysandthatphageparticles areseeninthin sections ofunlysed cellsinfected witham144(or am2O,whichhasamutation inthesamegene) andsupport theconclusion thatthesemutants aredefective inalysisfunction. Mutants (178and105)havingmutations in gene10showed intactcellsand,veryrarely, somecomplete phage, which wereprobably inputphage.Thesemutants didnotshow serum blocking power, evenaftersonication todisrupt theunlysed cellsandtorelease anytailfiber antigenthatmightbepresent. Theseresults are consistent withthefindings (50)thatartificially lysedcellsinfected withthesemutants donotproduce viablephageparticles andthatnophage particles areseeninthinsections ofunlysed, infected cells.Therefore, theresultsareconsis- tentwiththeconclusion (50)thatgene10mu- tantscannotsynthesize lateproteins (including thelysisproteins) andthatgene10maycodefor aregulator oflategeneexpression forP1. Mutants (31and56)havingmutations ingene 2showed veryfewparticles, andthesewere complete (exceptthatthepresence oftailfibers couldnotbedetermined). Thesemutants cause premature lysisandmayhaveadefectinalysis regulator (50).Inaddition toveryfewparticles beingseenbyelectron microscopy, serum blocking assaysarenegative withthesemutants, andtheburstsizeislow(50).Theseresultscan allbeexplained bytoofewphageparticles being produced bythetimethatlysisoccurs. DISCUSSION Genedesignations ofmutations affecting P1 morphogenesis. Wedetermined thephenotypes of47Plambermutants byuseofelectron microscopy andserumblocking assays.ElevenJ.VIROL. .i6&', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1133 FIG.16.Micrograph ofPlam57defective lysate.Particles appearnormal;irregular outlineofheadsmay possibly beduetoflattening asaconsequence ofbeingonlypartially filledwithDNA.Bar,100nm. mutants (Fig.17)showed headdefects, thirty showed taildefects, andsixhadadefectin particle maturation thatcouldbeeitherinthe headorinthetail.Whenweconsider the unambiguous results fromcomplementation tests,thegeneticandphysical positions ofthe mutations, andthephenotypes ofthemutants, wecanassignmostofthesemutations togenes (Table1andFig.17).Exceptions aremutations 108,33,66,74,and142,whichcouldbelocated indifferent genesthanthoseindicated (i.e., genes4,19,20,and26).am108givesambiguous complementation testresultswithothergene4 mutants andthusmaybeinadifferent gene. Resultsofserumblocking assaysdidnotdistin- guishbetween thepossibilities thatmutants 33, 72,and66couldbeinthesamegene(gene19)or indifferent genes.Although mutations 139,140, and71(inclusterVII;gene26)resultinaslightly different phenotype thanthatofam142(Table 1),thesefourmutations couldbeinthesame genebecause theyarelocated inarelatively shortregionofthegenome (Fig.17).However, itisnotknownwhether mutations 74and49are inthesamegene(gene20)because, although theyshowsimilarphenotypes, theygiveambig- uousresultsincomplementation tests(49),and thepreciselocation ofam74onthephysical map isunknown. Mutations affecting headmorphogenesis. Asa resultofthesegenedesignations, itisseenthat the11mutations causing headdefectsarelocat-edinfour(oratmostsix)genes(genes4,23,8, 9,andperhaps theparticle maturation gene1 andagenerepresented byamlO8). InT4,head assembly involves theproducts ofatleast20 genes,abouthalfofwhichconstitute thefin- ishedparticle(53).P22headassembly requires theproducts of11genes,about6ofwhicharein thefinishedparticle(53).Withitsheadshell, head-neck connector, andneckstructures, P1 headsareatleastascomplex asP22heads.In supportofthis,resultsfromgelelectrophoresis examination ofP1particles indicate thatthey consistof28proteins, ofwhich15areinthe head,9areinthetail,and4areeitherinthehead orthetail(51).Ofthe15(to19)headproteins, 4 arecodedbyaregionofthegenome, between mapunits24toabout32,thatisnotessential for particleformation. However, itisnotknown whether anyoftheremaining 11to15head proteins resultfromcleavage orfusionofpoly- peptides duringheadassembly. Inaddition, itis notknownwhether thereareanyproteins(such asthescaffolding proteininP22)thatarenot components ofthefinalvirionstructure. Never- theless, itislikelythatmorethanfour(tosix)genesarerequired forP1headassembly. Inthis regarditisofinterest tonotethatadjacent to headgenes4and8(Fig.17),thereareregionsof thegenome forwhichnofunctions areknown andwhichmightcarryfurther headgenes. Razzaetal.(38)havedescribed temperature- sensitive mutants intwonewgenes(genes13VOL.45,1983', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1134 WALKER ANDWALKER and14)thatarelocatedneargene4andthatare required forlyticgrowthofP1.Thesecouldbe headgenes.(Another newgenedescribed by Razzaetal.,gene15,islocatedneargenes22 and23andcouldbeeitheraheadoratailgene.) Onlythreephenotypes areproduced byamber mutations inthefourheadgenes.Thesepheno- typesare(i)complete tailsandseparate empty heads(genes4and9),(ii)complete tailswith necksandhead-neck connectors andseparate emptyheads(gene8),and(iii)complete tails withnecksandhead-neck connectors (orexten- sions)(gene23).Asmentioned above,gene9 mutants aredefective inafunction required for DNAcleavage atpac(N.Sternberg, personal communication). ThedefectofclusterIImu- tantsisunclear, butcouldbeinaminorprotein (suchasaneckprotein) orinaproteinrequired forDNApackaging (otherthanthatrequired for DNAcleavage atpac).Thesecondandthird phenotypes suggesteitheradefectinaminor headproteinthatleadstoafilled,attached, but unstable headoradefectinthemajorhead proteinthatleadstoanaccumulation oftheneck andhead-neck connector proteins atthetopof thetailtoformastructure calledtheextension byCoombs andEiserling (8).Head-defective phenotypes notseen,whichmightrepresent headgenesnotasyetdetected, areasfollows. (i) Proheads. UnlessP1proheads resemble mature heads,nonewereseen.Otherphagesproduce proheads indefective lysatesofmutants having mutations representing variousdifferent genes. Therefore wewouldexpecttohavemorethan onegeneinwhichthemutantalleleresultsin prohead production. Inthisregard,itcouldbe, asmentioned above,thatPlproheads areunsta- bleandcanconvert tomatureheadsindepen- dentlyofDNApackaging. (ii)Noheads.In otherphages,mutations inthemajorheadgene resultinmutants thatdonotproduce heads. Although P1mutants thatdonotproduce any headsatallhavenotbeenfound,thegene23 mutantproduces veryfewheads(1headper30 tails),sothisgenemightcodeforthemajorP1 headprotein. (iii)Spiralheadstructures, suchas thoseseeninmutants ofP22defective inscaf- foldingprotein(28)orinmutants ofXdefective eitherintheminorheadproteins BandC(24)or inthescaffolding proteinNu3(37).(iv)Poly- heads,suchasthoseseeninT4mutants defec- tiveeitherinthecoreprotein, aninternal pro- tein,thevertexprotein, orinproteins required forcorrectcapsidinitiation (12,13,30,40,56). Polyheads arealsoseeninKmutantsdefective in genesBandC.(v)Particles withonlysmall heads(P1Sparticles). Auniqueproperty ofPlis that,inaddition tonormal,infectious, big-head- ed(P1B)particles, italsoproduces 10to20% noninfectious, small-headed PlSparticles (47).Itwouldbeusefulinstudying headsizedetermi- nationtohaveamutantthatproduces P1S particles only.Sofar,theonlyPlphagethat produce anincreased proportion ofP1Sparticles (i.e.,50%P1Sand50%PlBparticles) arethose thataredeletedforaregionofthegenomethatis notessential forparticle morphogenesis (i.e., mapunits24to32)(Fig.17)(20).Thedetermi- nantinthisregionforthischangeinthePlS/PlB ratioisdesignated vadforviralarchitecture determination (20). Mutations affecting tailmorphogenesis.', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Auniqueproperty ofPlis that,inaddition tonormal,infectious, big-head- ed(P1B)particles, italsoproduces 10to20% noninfectious, small-headed PlSparticles (47).Itwouldbeusefulinstudying headsizedetermi- nationtohaveamutantthatproduces P1S particles only.Sofar,theonlyPlphagethat produce anincreased proportion ofP1Sparticles (i.e.,50%P1Sand50%PlBparticles) arethose thataredeletedforaregionofthegenomethatis notessential forparticle morphogenesis (i.e., mapunits24to32)(Fig.17)(20).Thedetermi- nantinthisregionforthischangeinthePlS/PlB ratioisdesignated vadforviralarchitecture determination (20). Mutations affecting tailmorphogenesis. The30 mutations causingP1taildefectsarelocatedin atleast12genes(genes19,20,16,3,21,22,5,6, 24,7,25,and26;Fig.17).Therecouldbeupto fiveadditional tailgenes(represented bymu- tants72,66,49,and142andperhaps including gene1,whichisinvolved inparticlematuration). InT4,tailassembly (including tailfibers)re- quirestheproducts of29genes,andallbut7of thembecomepartofthefinalstructure. ThePl tailislesscomplex thanthatofT4(i.e.,the baseplate isthinnerandlesswelldefined). Re- sultsfromgelelectrophoresis studiesindicate thattheP1tailiscomposed ofbetween 9and13 proteins (51).However, itmaybethatsomeof thebaseplate components aresmallandrequired inlowquantities andsowerenotresolved bygel electrophoresis. Ofthestructural tailgenesinT4,1genecodes forthetube;1forthesheath;4forthetailfibers; 14forthebaseplate; and2forthestructures that addtothetopofthetailtostabilize it.The12to 17P1tailgenesconsistof1(gene22)codingfor thesheath;between 6(genes20,3,16,21,5,and 26)and10(theadditional genesbeinggene1, gene24,andgenesrepresented byam49and am142)codingforbaseplate ortubeproduction (ofwhichonly1wouldcodefortubeproduc- tion);2or3(genes7,24,and25)codingfortail stabilization; 1(gene6)affecting taillength;and 1(gene19,represented byam72)ormore(repre- sentedbyam33andam66)codingfortailfibers. Inaddition, Razzaetal.(38)foundtwogenes (genes11and12,represented bytemperature- sensitive mutants) thatarelocatedinthevicinity ofgene19andmaycodefortailfiberstructural proteins. Insummary, wemaynotknowforall ofthetailgeneswhichcodeforstructural pro- teinsandwhichcodefornonstructural proteins, butmutations affecting eachofthetailcompo- nentsvisiblebyelectron microscopy arerepre- sented(withapossible exception ofthetail tube). Gene19mutants donothavetailfibersanddo notshowserumblocking power(forthisdiscus- sionweshallassumethatmutations 33,72,and 66areallingene19,although thismaynotbe so).Therefore, gene19isastructural genethat probably codesforthedistalpartofthetailJ.VIROL.', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  3~J-c FIG.17.Mapofthebacteriophage P1genome, modified fromthatofYarmolinsky (57).Theinnercircle represents thephysical genome, divided intothenumbered digestion fragments generated byrestriction endonuclease EcoRI(2).Genetic symbols withinthiscirclerepresent functions notrequired forphage morphogenesis (exceptpac);foracomplete listofthesefunctions andtheirreferences, seeYarmolinsky (57). pacisthesitewhereDNApackaging initiates (arrowindicates direction ofpackaging). Othersymbols and functions: cgenes,establishment andmaintenance oflysogeny (c6andc8werefoundinphageP7);inmmCand imml,bipartite immunity system; resandmod,restriction andmodification; dar,defense againstrestriction; vad,viralarchitecture determination; gta,generalized transduction affected; lyd,lysisdelay;cin,Cloop inversion; virs,virulence; sim,superimmunity; antorreb,antagonism orbypassofclrepressor; par,plasmid partition; incAandincB,plasmid incompatability; rep,plasmid replication; ori,originofreplication; cop, plasmid copynumber; ban,dnaBanalog;loxP,locusofcrossover inplasmid; andcre,site-specific recombina- tion.Outercircle,100mapunits(equivalent topercentofthe90-kbgenome). Onthiscircle,Cdenotes the invertible Cloop;ISJdenotes insertion sequence 1.Between theoutercircleandthearcsarenumbers representing ambermutations. Thelocations oftheseambermutations onthephysical mapweredetermined by markerrescueexperiments, usingclonedfragments ofP1DNA(21,34,38,43,45).Thearcsdenoteextentof linkageclusters (romannumerals). Numbers outsidethesearcsdenotegenes(genenumbers 5,6,24,7,and25 arerepeated abovetheirrelevant functions). Aquerydenotes theuncertain genedesignation ofam74. Diagrammatic representations oftheambermutantphenotypes andfunctions ofthegenesarenoted.Asterisks besidebaseplate notations meanthatoneofthesegenescouldcodeforthetailtube.Imprecise locations ofgenes orambermutations aredenoted byparentheses aroundtherelevant symbolornumber. Genes11through 15are represented bytemperature-sensitive mutations only(notshown)andhavenotbeenphenotyped. Rescuetests andtwo-factor crosseshaveshownambermutant75(gene16)tobelocatedbetween mutants 74and35andnot aroundmapunit55asindicated previously. 1135'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Othermutants inclusterVIII (am139, aml40,andam7l)onlyproduce com- pleteheadsandpolysheath. Theirdefectcould therefore beinbaseplate ortubeformation. Itis notknownhowmanycistrons thesefourmu- tantsrepresent. Iftheyareallinthesame cistron,theirdefectcouldbeinbaseplate forma- tionwitham142locatedinthedistalpartofthis cistron. Mutations affecting headassembly. Mutations affecting headassembly arelocatedinlinkage clustersII,V,IX,andX-1.Mutants inclusterII (150,62,and108;Fig.12)andcluster X-1(16, 64,136,131,43,and112;Fig.13)produce complete tailsandunattached emptyheadsin head/tail ratiosof1:1to1:3.(Atotalof245to 873particles werecounted permutant,using electronmicrographs ofrandomly selected fields fromseveralpreparations.) Incomplementation tests(49),aml50and am62givenegative results. Because ofthisand because thephenotypes ofthetwomutants are similar, themutations thattheyrepresent areVOL.45,1983', \"page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1128 WALKER ANDWALKER t~~~~~ s P 1~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~J1 FIG.8.Micrograph ofP1am26defective lysate.Inaddition tonormalfullheads,there arepolysheath (PS) andunit-length sheaths(US)orfragments thereof. Thelongcurved structures arebacterial flagella. Bar,100nm. probably inthesamegene,whichwewillrefer toasgene4,adesignation originally madeby Scott(39)foram7.(Complementation results witham62andamlO8wereambiguous; there- fore,weareunsurewhether themutation in amlO8islocatedinthisgene.)Forsimilarrea- sonsandbecause themutants inlinkagecluster X-1arerescuedfromtwoadjacent fragments of theP1genome (seeFig.17),oneofwhichalso contains atleastpartoftheP1cigeneandthe otherofwhichisflankedbya0.62-kilobase-pair (kb)fragment whichcontains atleastpartof gene10,themutations inlinkageclusterX-1are assigned toasinglegene(gene9,adesignation originally madebyScott[39]foraml6),al- thoughthereisapossibility thatthesemutants couldrepresent morethanonegene. Theemptyheadsproduced bygene9and clusterIImutants aresimilar(insizeandinthe thickness ofthecapsidshell)toemptymature headsseeninamber' lysates. Therefore, they arepresumed tobematureheadsthatarelack- ingDNA,ratherthanproheads. InT4,mature, empty,unattached heads(and complete tails)areproduced bymutants with defectseitherinvariousheadcompletion genes (8,11,26)oringene49.Headcompletion genes areinvolved inneckassembly (gene13),DNA packaging (genes16and17),orunknown func- tions(genes4and65).Gene49codesforDNase andisrequired forDNApackaging (14,33). Headsofmutants inthesegenesareempty,eitherbecause theywerenotfilledwithDNAor because theywereunstably filledandsubse- quentlylosttheirDNA(8,17).Genesinother phagesthatareinvolved inheadcompletion are genes4,10,and26inP22(6,36),whichcodefor threeminorstructural proteins required forhead stability, andgeneDinX(24),whichcodesfora proteinthatisaddedtothematurecapsidshell andwhosepresence isalsorequired forDNA cutting(4,23). Fromtheirphenotype, theP1gene9mutants couldbedefective inoneofthesefunctions. In thisregard,Sternberg (personal communication) recentlyfoundthat,unlikemutantsingenes4,8, and23,gene9mutants cannotcleaveP1DNAat pac,thesiteforinitiation ofDNApackaging (seeFig.17);therefore, likegenes2and3in phageP22(31),theP1gene9productisinvolved intherecognition andcleavage ofP1DNAat pac.Withotherphages,conversion ofproheads tomatureheadsusually occursconcomitantly withDNApackaging. Gene9mutants apparent- lycannotinitiateDNApackaging. Therefore, in P1eitherconversion ofproheads tomature headscanoccurindependently ofDNApackag- ingorP1proheads resemble matureheadsin sizeandshellthickness. TheclusterIImutants couldalsobedefective inoneofthesefunctions (except forDNA cleavage because, asmentioned above,Stern- bergfoundthatclusterIImutants cancleaveP1 DNAatthepacsite).J.VIROL.\", 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1129 PS , * .\\'-4 .4 IN .;. t\\'.4 \\'.\\' 444  r .JE ~ ~  i~S T- taST~ FIG.9.Micrograph ofPlam27defective lysateshowing severalkindsoftailstructures. Notethetailtube aggregate (TT),polysheath (PS),tailswithslippedsheaths(TSS),andstackedtails(ST).Insetshows anamber\\' PtparticlefromaCsClgradient; afterprolonged exposure toCsCl,thereisabnormal tailcontraction, i.e.,the tailsheathhascontracted andsliddownthetube,butaringlike structure withorwithout somesheathsubunits remains atthejunction between theproximal endofthetubeandtheneck(seetext).Bar,100nm. FIG.10.Micrograph ofPtam3Odefective lysate. Polysheath appears tobeattached totheproximal end ofthischaracteristically contracted tail,whichother- wiseappears tobenormalinhavingbothabaseplate andtailfibers.Bar,100nm.Likethemutants inlinkage clusters IIandX- 1,asinglemutant(amll5) incluster Vandthe mutant(aml3)thatrepresents cluster IX(i.e., gene8)produce complete tailsandunattached emptyheads.Withaml3,thehead/tail ratiois 1:2(1,967particles werecounted). However, with amll5, veryfewheads were seen(the head/tail ratiowas1:30,with431particles count- ed),andinputphagecouldaccount forthese heads.Attached tothetailsofbothamll5and amt3mutants arestructures resembling the neckandhead-neck connector (Fig.14).The neckandhead-neck connector arepartofthe head.Thissuggests thattheheads wereoriginal- lyassembled, filledwithDNA,andattached to thetails,butwereunstable andsubsequently eitherdisintegrated (amllS) ordissociated fromp \\'-p--,-, *X--Lt-VOL.45,1983 IZ\"', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1130 WALKER ANDWALKER FIG.11.Micrograph ofP1amll6defective lysateshowing aggregates oftailtubes.Nosingletubeislonger thannormallength.Bar,100nm. thetail,leavingbehindatleastsomeoftheneck andhead-neck connector (aml3).Suchinstabil- itycouldbeduetotheabsence ofaminorhead protein, e.g.,aminorheadshellprotein(amllS) orahead-neck connector protein(aml3). Alter- natively, because sofewheadsareseenin amllSlysates, amllScouldbedefective inthe majorheadprotein. Inthiscase,thestructures seenwouldnotbeaneckandhead-neck connec- torbutanaberrant structure similartothatseen inT4(calledtheextension [8]),whichconsists of anaberrant accumulation onthetopofthetailof fiveproteins normally assembled intothehead aspartoftheneckandhead-neck connector or addedtothehead-neck connector afterhead-tail joining.Mutations affecting particlematuration. Muta- tionsaffecting particle maturation (47,141,34, 128,132,and57)comprise linkageclusterI.The mutants inclusterIdonotshowcomplementa- tion(49)andarerescued froma0.92-kbrestric- tionfragment oftheP1genome flanked by fragments carrying othergenes(seeFig.17); therefore, thesesixmutations areprobably in thesamegene(gene1).Particles produced by thesixmutants arecomplete exceptthat,inall ofthemutants except57,headstendtobeempty andthetailsareinvariousstagesofcontraction (Fig.15).Mutant57produces complete particles withatleastpartially filledheadsandextended sheaths(Fig.16).Thesephenotypes suggestthat thedefective polypeptide allowsassembly ofthe FIG.12.Micrograph ofP1amlO8defective lysate.Thetailsappeartobenormal;heads,notseenhere,are alsoproduced. Bar,100nm.J.VIROL. -V Gkll-.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1131 FIG.13.Micrograph ofP1am64defective lysate.Theheadsarethesamesizeasmatureheads,buttheyare alwaysempty;thetailsarenormal,theircontraction beinganartifactofpreparation procedures. Bar,100nm. complete particles butthattheseparticles are eithermorphologically complete butnoninfec- tive(am57)orunstable. Morphologically normal particles areproduced byotherphages, e.g.,P22 -RI i.^.. . ._l S ... FIG.14.Micrograph ofP1aml3defective lysate. Tailsarenormal exceptforaknobattheproximal end (arrow). Bar,100nm.mutants defective ingenes7,16,and20(6,36) andT4mutants (11,12,26)defective ingenes2, 4,11,12,50,64,and65(whichproduce atleast somemorphologically normalphages). Genes11 and12aretailgenes,codingforabaseplate component andshorttailfibers,respectively. Alloftheremaining geneproducts areincorpo- ratedintotheheadatearlystagesofassembly, buttheirfunctions vary;e.g.,someareinvolved intheDNAinjection process, andsome(inT4) areinvolved incleavage ofheadproteins. The P1mutants couldhaveoneoftheseheadgene defects(although itisnotknownwhether there iscleavage ofPlheadproteins). Alternatively, because oftheinstability ofthetail,theycould haveadefectinatailgene,e.g.,abaseplate gene.Thedifference observed between am57 andtheothermutants inthisclustercouldbe explained bytheam57mutation beinginthe distalpartofthegenetogiveamorecomplete polypeptide andhenceamorestableparticle. In supportofthisisthefactthatam57islocatedat oneendoftheclusterwithrespecttotheother fivemutants examined (Fig.2). Mutations affecting hostcelllysis.Cellsinfect- edseparately witheightmutants thatcannot causecelllysis(gene17,linkageclusterIII,or gene10,clusterX-2)orthatcausepremature celllysis(gene2,clusterIV)wereexamined by electron microscopy attheincubation time(110 min)atwhichlysiswouldbecomplete incells infected withlysis-positive mutants.VOL.45,1983 4 .1 4. -f....7-,-. IS6', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1132 WALKER ANDWALKER FIG.15.Micrograph ofP1am47defective lysate.Particles appear tobenormal exceptforemptyheads (maturesize);normaltailsareinvarious stagesofcontraction. Bar,100nm. Mutants (23,144,73,and134)having muta- tionsingene17showed intactcellsasexpected and,occasionally, somecellsthathadlostsome oftheircontents butwerefullofphageparticles. Veryrarely, somecomplete phage,whichcould havebeeninputphage, wereseenoutside the cells.Theseresults areconsistent withthefind- ings(50)thatartificially lysedcellsinfected with thesemutants produce viablephageparticles andgivepositive resultsinserumblocking as- saysandthatphageparticles areseeninthin sections ofunlysed cellsinfected witham144(or am2O,whichhasamutation inthesamegene) andsupport theconclusion thatthesemutants aredefective inalysisfunction. Mutants (178and105)havingmutations in gene10showed intactcellsand,veryrarely, somecomplete phage, which wereprobably inputphage.Thesemutants didnotshow serum blocking power, evenaftersonication todisrupt theunlysed cellsandtorelease anytailfiber antigenthatmightbepresent. Theseresults are consistent withthefindings (50)thatartificially lysedcellsinfected withthesemutants donotproduce viablephageparticles andthatnophage particles areseeninthinsections ofunlysed, infected cells.Therefore, theresultsareconsis- tentwiththeconclusion (50)thatgene10mu- tantscannotsynthesize lateproteins (including thelysisproteins) andthatgene10maycodefor aregulator oflategeneexpression forP1. Mutants (31and56)havingmutations ingene 2showed veryfewparticles, andthesewere complete (exceptthatthepresence oftailfibers couldnotbedetermined). Thesemutants cause premature lysisandmayhaveadefectinalysis regulator (50).Inaddition toveryfewparticles beingseenbyelectron microscopy, serum blocking assaysarenegative withthesemutants, andtheburstsizeislow(50).Theseresultscan allbeexplained bytoofewphageparticles being produced bythetimethatlysisoccurs. DISCUSSION Genedesignations ofmutations affecting P1 morphogenesis. Wedetermined thephenotypes of47Plambermutants byuseofelectron microscopy andserumblocking assays.ElevenJ.VIROL. .i6&', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1133 FIG.16.Micrograph ofPlam57defective lysate.Particles appearnormal;irregular outlineofheadsmay possibly beduetoflattening asaconsequence ofbeingonlypartially filledwithDNA.Bar,100nm. mutants (Fig.17)showed headdefects, thirty showed taildefects, andsixhadadefectin particle maturation thatcouldbeeitherinthe headorinthetail.Whenweconsider the unambiguous results fromcomplementation tests,thegeneticandphysical positions ofthe mutations, andthephenotypes ofthemutants, wecanassignmostofthesemutations togenes (Table1andFig.17).Exceptions aremutations 108,33,66,74,and142,whichcouldbelocated indifferent genesthanthoseindicated (i.e., genes4,19,20,and26).am108givesambiguous complementation testresultswithothergene4 mutants andthusmaybeinadifferent gene. Resultsofserumblocking assaysdidnotdistin- guishbetween thepossibilities thatmutants 33, 72,and66couldbeinthesamegene(gene19)or indifferent genes.Although mutations 139,140, and71(inclusterVII;gene26)resultinaslightly different phenotype thanthatofam142(Table 1),thesefourmutations couldbeinthesame genebecause theyarelocated inarelatively shortregionofthegenome (Fig.17).However, itisnotknownwhether mutations 74and49are inthesamegene(gene20)because, although theyshowsimilarphenotypes, theygiveambig- uousresultsincomplementation tests(49),and thepreciselocation ofam74onthephysical map isunknown. Mutations affecting headmorphogenesis. Asa resultofthesegenedesignations, itisseenthat the11mutations causing headdefectsarelocat-edinfour(oratmostsix)genes(genes4,23,8, 9,andperhaps theparticle maturation gene1 andagenerepresented byamlO8). InT4,head assembly involves theproducts ofatleast20 genes,abouthalfofwhichconstitute thefin- ishedparticle(53).P22headassembly requires theproducts of11genes,about6ofwhicharein thefinishedparticle(53).Withitsheadshell, head-neck connector, andneckstructures, P1 headsareatleastascomplex asP22heads.In supportofthis,resultsfromgelelectrophoresis examination ofP1particles indicate thatthey consistof28proteins, ofwhich15areinthe head,9areinthetail,and4areeitherinthehead orthetail(51).Ofthe15(to19)headproteins, 4 arecodedbyaregionofthegenome, between mapunits24toabout32,thatisnotessential for particleformation. However, itisnotknown whether anyoftheremaining 11to15head proteins resultfromcleavage orfusionofpoly- peptides duringheadassembly. Inaddition, itis notknownwhether thereareanyproteins(such asthescaffolding proteininP22)thatarenot components ofthefinalvirionstructure. Never- theless, itislikelythatmorethanfour(tosix)genesarerequired forP1headassembly. Inthis regarditisofinterest tonotethatadjacent to headgenes4and8(Fig.17),thereareregionsof thegenome forwhichnofunctions areknown andwhichmightcarryfurther headgenes. Razzaetal.(38)havedescribed temperature- sensitive mutants intwonewgenes(genes13VOL.45,1983', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1134 WALKER ANDWALKER and14)thatarelocatedneargene4andthatare required forlyticgrowthofP1.Thesecouldbe headgenes.(Another newgenedescribed by Razzaetal.,gene15,islocatedneargenes22 and23andcouldbeeitheraheadoratailgene.) Onlythreephenotypes areproduced byamber mutations inthefourheadgenes.Thesepheno- typesare(i)complete tailsandseparate empty heads(genes4and9),(ii)complete tailswith necksandhead-neck connectors andseparate emptyheads(gene8),and(iii)complete tails withnecksandhead-neck connectors (orexten- sions)(gene23).Asmentioned above,gene9 mutants aredefective inafunction required for DNAcleavage atpac(N.Sternberg, personal communication). ThedefectofclusterIImu- tantsisunclear, butcouldbeinaminorprotein (suchasaneckprotein) orinaproteinrequired forDNApackaging (otherthanthatrequired for DNAcleavage atpac).Thesecondandthird phenotypes suggesteitheradefectinaminor headproteinthatleadstoafilled,attached, but unstable headoradefectinthemajorhead proteinthatleadstoanaccumulation oftheneck andhead-neck connector proteins atthetopof thetailtoformastructure calledtheextension byCoombs andEiserling (8).Head-defective phenotypes notseen,whichmightrepresent headgenesnotasyetdetected, areasfollows. (i) Proheads. UnlessP1proheads resemble mature heads,nonewereseen.Otherphagesproduce proheads indefective lysatesofmutants having mutations representing variousdifferent genes. Therefore wewouldexpecttohavemorethan onegeneinwhichthemutantalleleresultsin prohead production. Inthisregard,itcouldbe, asmentioned above,thatPlproheads areunsta- bleandcanconvert tomatureheadsindepen- dentlyofDNApackaging. (ii)Noheads.In otherphages,mutations inthemajorheadgene resultinmutants thatdonotproduce heads. Although P1mutants thatdonotproduce any headsatallhavenotbeenfound,thegene23 mutantproduces veryfewheads(1headper30 tails),sothisgenemightcodeforthemajorP1 headprotein. (iii)Spiralheadstructures, suchas thoseseeninmutants ofP22defective inscaf- foldingprotein(28)orinmutants ofXdefective eitherintheminorheadproteins BandC(24)or inthescaffolding proteinNu3(37).(iv)Poly- heads,suchasthoseseeninT4mutants defec- tiveeitherinthecoreprotein, aninternal pro- tein,thevertexprotein, orinproteins required forcorrectcapsidinitiation (12,13,30,40,56). Polyheads arealsoseeninKmutantsdefective in genesBandC.(v)Particles withonlysmall heads(P1Sparticles). Auniqueproperty ofPlis that,inaddition tonormal,infectious, big-head- ed(P1B)particles, italsoproduces 10to20% noninfectious, small-headed PlSparticles (47).Itwouldbeusefulinstudying headsizedetermi- nationtohaveamutantthatproduces P1S particles only.Sofar,theonlyPlphagethat produce anincreased proportion ofP1Sparticles (i.e.,50%P1Sand50%PlBparticles) arethose thataredeletedforaregionofthegenomethatis notessential forparticle morphogenesis (i.e., mapunits24to32)(Fig.17)(20).Thedetermi- nantinthisregionforthischangeinthePlS/PlB ratioisdesignated vadforviralarchitecture determination (20). Mutations affecting tailmorphogenesis.', 'page_label: 17 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Auniqueproperty ofPlis that,inaddition tonormal,infectious, big-head- ed(P1B)particles, italsoproduces 10to20% noninfectious, small-headed PlSparticles (47).Itwouldbeusefulinstudying headsizedetermi- nationtohaveamutantthatproduces P1S particles only.Sofar,theonlyPlphagethat produce anincreased proportion ofP1Sparticles (i.e.,50%P1Sand50%PlBparticles) arethose thataredeletedforaregionofthegenomethatis notessential forparticle morphogenesis (i.e., mapunits24to32)(Fig.17)(20).Thedetermi- nantinthisregionforthischangeinthePlS/PlB ratioisdesignated vadforviralarchitecture determination (20). Mutations affecting tailmorphogenesis. The30 mutations causingP1taildefectsarelocatedin atleast12genes(genes19,20,16,3,21,22,5,6, 24,7,25,and26;Fig.17).Therecouldbeupto fiveadditional tailgenes(represented bymu- tants72,66,49,and142andperhaps including gene1,whichisinvolved inparticlematuration). InT4,tailassembly (including tailfibers)re- quirestheproducts of29genes,andallbut7of thembecomepartofthefinalstructure. ThePl tailislesscomplex thanthatofT4(i.e.,the baseplate isthinnerandlesswelldefined). Re- sultsfromgelelectrophoresis studiesindicate thattheP1tailiscomposed ofbetween 9and13 proteins (51).However, itmaybethatsomeof thebaseplate components aresmallandrequired inlowquantities andsowerenotresolved bygel electrophoresis. Ofthestructural tailgenesinT4,1genecodes forthetube;1forthesheath;4forthetailfibers; 14forthebaseplate; and2forthestructures that addtothetopofthetailtostabilize it.The12to 17P1tailgenesconsistof1(gene22)codingfor thesheath;between 6(genes20,3,16,21,5,and 26)and10(theadditional genesbeinggene1, gene24,andgenesrepresented byam49and am142)codingforbaseplate ortubeproduction (ofwhichonly1wouldcodefortubeproduc- tion);2or3(genes7,24,and25)codingfortail stabilization; 1(gene6)affecting taillength;and 1(gene19,represented byam72)ormore(repre- sentedbyam33andam66)codingfortailfibers. Inaddition, Razzaetal.(38)foundtwogenes (genes11and12,represented bytemperature- sensitive mutants) thatarelocatedinthevicinity ofgene19andmaycodefortailfiberstructural proteins. Insummary, wemaynotknowforall ofthetailgeneswhichcodeforstructural pro- teinsandwhichcodefornonstructural proteins, butmutations affecting eachofthetailcompo- nentsvisiblebyelectron microscopy arerepre- sented(withapossible exception ofthetail tube). Gene19mutants donothavetailfibersanddo notshowserumblocking power(forthisdiscus- sionweshallassumethatmutations 33,72,and 66areallingene19,although thismaynotbe so).Therefore, gene19isastructural genethat probably codesforthedistalpartofthetailJ.VIROL.', 'page_label: 18 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  3~J-c FIG.17.Mapofthebacteriophage P1genome, modified fromthatofYarmolinsky (57).Theinnercircle represents thephysical genome, divided intothenumbered digestion fragments generated byrestriction endonuclease EcoRI(2).Genetic symbols withinthiscirclerepresent functions notrequired forphage morphogenesis (exceptpac);foracomplete listofthesefunctions andtheirreferences, seeYarmolinsky (57). pacisthesitewhereDNApackaging initiates (arrowindicates direction ofpackaging). Othersymbols and functions: cgenes,establishment andmaintenance oflysogeny (c6andc8werefoundinphageP7);inmmCand imml,bipartite immunity system; resandmod,restriction andmodification; dar,defense againstrestriction; vad,viralarchitecture determination; gta,generalized transduction affected; lyd,lysisdelay;cin,Cloop inversion; virs,virulence; sim,superimmunity; antorreb,antagonism orbypassofclrepressor; par,plasmid partition; incAandincB,plasmid incompatability; rep,plasmid replication; ori,originofreplication; cop, plasmid copynumber; ban,dnaBanalog;loxP,locusofcrossover inplasmid; andcre,site-specific recombina- tion.Outercircle,100mapunits(equivalent topercentofthe90-kbgenome). Onthiscircle,Cdenotes the invertible Cloop;ISJdenotes insertion sequence 1.Between theoutercircleandthearcsarenumbers representing ambermutations. Thelocations oftheseambermutations onthephysical mapweredetermined by markerrescueexperiments, usingclonedfragments ofP1DNA(21,34,38,43,45).Thearcsdenoteextentof linkageclusters (romannumerals). Numbers outsidethesearcsdenotegenes(genenumbers 5,6,24,7,and25 arerepeated abovetheirrelevant functions). Aquerydenotes theuncertain genedesignation ofam74. Diagrammatic representations oftheambermutantphenotypes andfunctions ofthegenesarenoted.Asterisks besidebaseplate notations meanthatoneofthesegenescouldcodeforthetailtube.Imprecise locations ofgenes orambermutations aredenoted byparentheses aroundtherelevant symbolornumber. Genes11through 15are represented bytemperature-sensitive mutations only(notshown)andhavenotbeenphenotyped. Rescuetests andtwo-factor crosseshaveshownambermutant75(gene16)tobelocatedbetween mutants 74and35andnot aroundmapunit55asindicated previously. 1135'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:06 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'131'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994398'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'336ms'), (b'x-request-id', b'b988750671218105efd6235d457bb819'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249bacb515d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:06 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'131'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994398'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'336ms'), (b'x-request-id', b'b988750671218105efd6235d457bb819'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249bacb515d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': [\"page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1136 WALKER ANDWALKER fiber.Mutant72ismissinga105,000-molecular- weight(MW)proteinthat,unlessitisafusion product, istheproductofgene19(unpublished data).Toussaint etal.(45)locatedthegene19 mutations withinorneartheP1Cloop(an invertible DNAregion)because (i)thisisthe onlyregionofhomology sharedwithphageMu (i.e.,theMuGloop)and(ii)theamber'alleles oftheseP1mutants arerescued fromMu.The PlCloopis3kblongandflankedbytwo0.62- kbinverted repeats. The105,000-MW protein codedbygene19wouldrequireacodinglength of2.9kbofDNA.Gene19probably doesnot extendmorethanhalfway (1.5kb)intotheG loop(byanalogy withMu[46]);therefore, the remaining portion(1.4kb)ofthegenewould includeoneofthe0.62-kbinverted repeatsand extendatleast0.78kboutside theinverted repeat.Ifthe105,000-MW proteinisnotafusion product, evenmoreofgene19couldbelocated outside theCloop.Thisisbecauseamber'allelesoftwoMugenes,oneofwhich(U)is whollywithintheGloopandanotherofwhich (S)ispartially withintheGloopandpartially in theotregion(16),arerescuedfromP1(45).This suggests thatthereisanothergeneinaddition to gene19intheP1Cloop.Products ofboththeU andSgenesarerequired forMuadsorption (46); therefore, inP1thetwoanalogous genescould codeforthedistal(gene19)andproximal parts oftheP1tailfiber. Thefollowing findingssupportthesuggestion thatgene19ispartially outsidetheCloopand thatitincludes oneoftheinverted repeats. The P1-related phageP7,whichhasa0.2-kbdeletion (orsubstitution) withintherightinverted repeat (7),produces a100,000-MW proteininsteadof the105,000-MW P1tailfiberprotein(51).(These findingsalsosuggestthatgene19extends tothe rightratherthantotheleftoftheCloop.) Transcription ofgene19isleftward fromam49 (because am49ispolarongene19;Fig.2).Also, nopolypeptide fragment ofthe105,000-MW proteinisseenaftersodiumdodecyl sulfate- polyacrylamide gelelectrophoresis ofam72(un- published data).Therefore, unlesstheamber fragment hasbeendegraded, thismutation must beneartheendofgene19thatisproximal to wheretranscription isinitiated. Themodelofgene19asbeingpartially within andpartially outside theCloopisconsistent withthefollowing findings: (i)MuandP1have partialserological cross-reactivity (45),and(ii) thehostrangespecificities ofthesetwophages arewithin(ornear)theseinvertible regions(45, 46).Byanalogy withT4(3),thesehostrange specificities areprobably locatedinthetailfiber genesofthephages. Assembly ofP1particles. Byexamining the structures obtained indefective lysates, itisseenthatP1headsandtailscanbeassembled independently. Thissuggests that,likeT4,these Plstructures haveindependent assembly path- ways.However, inP1,tailfibersareoftenseen attached toheadless tails.Therefore, unlikeT4 tailfibers,P1tailfiberscanattachtothetail beforehead-tail joiningoccurs. (Thishadal- readybeensuspected because theP1tailfibers arenotlongenoughtoreachfromthebaseplate tothehead,suchalengthbeingrequired forthe addition oftailfiberstoT4particles.) Distribution ofmutations. TheP1ambermuta- tionsaffecting particle morphogenesis arenot equallydistributed amongthedesignated genes.\", 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Assembly ofP1particles. Byexamining the structures obtained indefective lysates, itisseenthatP1headsandtailscanbeassembled independently. Thissuggests that,likeT4,these Plstructures haveindependent assembly path- ways.However, inP1,tailfibersareoftenseen attached toheadless tails.Therefore, unlikeT4 tailfibers,P1tailfiberscanattachtothetail beforehead-tail joiningoccurs. (Thishadal- readybeensuspected because theP1tailfibers arenotlongenoughtoreachfromthebaseplate tothehead,suchalengthbeingrequired forthe addition oftailfiberstoT4particles.) Distribution ofmutations. TheP1ambermuta- tionsaffecting particle morphogenesis arenot equallydistributed amongthedesignated genes. Forexample, including mutations notstudiedin theworkreported here(Fig.2),thereare13 mutations ingene1andonly1mutation ineach ofgenes22,23,and8.Wedonotknowthesizes ofthesegenes,butitmaybethatthereare\"hot spots\"ofmutation (obtained byhydroxyl- amine),aswasfoundinP22(36).Insupportof this,wedoknowthatthereare10mutations in gene10,whichcodesforaproteinof64,000MW (18)involved inlategeneexpression (50),anda maximum of3mutations ingene19,which codesfora105,000-MW tailfiberprotein. The existence ofhotspotsofmutation couldexplain whysomeofthehead-defective phenotypes described abovehavenotyetbeenseen. Correlation ofphenotypes withtheP1genetic andphysicalmaps.The103P1ambermutations havebeenassigned (49)to10lineararraysor linkageclusters (Ithrough X;Fig.2).Because complementation testsinliquidoftengaveam- biguousresultsandnotallmutants weretested inallcombinations, onlysomeofthemutations wereputintogroupsthatshowed nocomple- mentation. The55ambermutantphenotypes (i.e.,including mutations ingenes17,2,and10) thatwedetermined correlate withthelinear arraysofthecorresponding mutations. Forex- ample,inlinkageclusterVI,mutants 26,32,and 109showonephenotype andmutants27,39,53, 22,and116showanother. Inaddition, except formutants 180and45(whichmaybepolar mutants), allmutants testedthatwerefoundnot tocomplement (Fig.2)showed thesamepheno- type. Wepreviously suggested (49)thatthephe- nomenon oflinkageclustersmightresultfroma varietyofunderlying causes,suchasrecombina- tionhotspots,longsegments ofsilentDNA,or regionsforwhichtherewerenoknownmuta- tions.Thephysical position ofthemutations comprising thelinkageclusters isnowknown (Fig.17)(57),anditcanbeseenthattheclusters consistofsinglegenesorgroupsofgenesand thatthedemarcations between someofthese linkageclusters (i.e.,II/IlI,VI/VII,VII/VIII,VIII/IX, andX-2/X-1) areduetorecombinationJ.VIROL.', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1137 between thesegenesorgenegroups.Thereason whysomelinkage clusters arecomprised of singlegenesandsomearecomprised ofgene groupsisunknown, unlesshotspotsofrecombi- nationareinvoked. Otherdemarcations be- tweenlinkage clusters, suchasI/IT,III/IV, IV/V,V/VI,andIX/X-2, areduetolongseg- mentsofDNAthateithercontain othergenes (e.g.,codingforimmunity) orforwhichno mutations areknown,oracombination ofthese tworeasons. (Itmaybethattherearenolong segments ofsilentDNAbecause, forexample, somemorphogenetic genes[especially head genes]arestillundetected.) Between linkage clusters X-1andI(i.e.,attheendsofthe vegetative map),thereisashorter segment of DNAthatcodesforpartoftheP1immunity system. However, thisregionhasalsobeen showntocontainarecombination hotspot(loxP [44]),whichalonecouldaccount forthedemar- cationbetween thesetwoclusters. Clustering offunctionally relatedgenes.The physical andgeneticmapsofP1showconsider- ableclustering ofgeneshavingrelatedfunctions. Suchclustering couldbeadvantageous forregu- latingtheassembly ofsubstructures, suchas,for example, baseplates. Wedonotknowwhether gene1isaheadoratailgene.Excluding gene1, theknowntailgenesarelocatedinthreeregions. Theapproximate physical positions ofthese regionsaremapunits34to49,56to59,and78to 84.5.Withintheseregions, somesimilarfunc- tionsarelocatedtogether. Inthefirstregion therearethreetofivetailfibergenesandthree tofourbaseplate (ortailtube)genes.Inthe secondregionwefoundonetailsheathgeneand onebaseplate (ortailtube)gene.Itmaybe,by analogywithT4(54),thatthesingleP1tailtube geneislocatedinthisregionbesidethesheath gene.Inthethirdregionwefoundoneortwo baseplate genes,ataillengthgene,twoorthree tailstability genes,andoneortwobaseplate ortailtubegenes.Inthisregionthebaseplate genesareseparated byothertailgenes.How- ever,unlikegenes3and15ofT4(54),thetwo tothreeP1tailstability genesarelocated to- gether. ThefourtofiveknownPlheadgenesarein fourregionsofthegenome. Thiswouldargue againstclustering offunctionally relatedhead genes.However, someexpected P1headgenes havenotyetbeendetected andcouldbelocated intwomainregions forwhichtherearefew knowngenes.Oneoftheseregions comprises mapunits10to20andcouldtheoretically code forabout10genes,including headgene4,the headgenerepresented byam108(ifdifferent fromgene4),andgenes13and14foundby Razzaetal.(38).Theotherregioncomprises mapunits84.5to94,whichcouldcodefor9or10genes,including headgene8.Inaddition, itis interesting tonotethatheadgene9,whichis adjacent topac,thesiteatwhichDNApackag- ingisinitiated, hasnowbeenfoundtocodefor P1DNAcleavage atthissite(N.Sternberg, personal communication). Inaddition tolysisgene17,P1probably hasa secondlysisgene(lyd[20]),analogous totheS geneofX,located somewhere between map units24andabout32.Therefore, thesetwolysis genesandalysisregulator gene(gene2),al- thoughnotalladjacent toeachother,arelocated withinaregionspanning lessthan15%oftheP1 genome (mapunits20toabout33). Gene10,whichregulates lategeneexpression (50),islocated approximately between map units94and96.Gene10isprobably analogous togeneQinX,whichisadjacent togenes(O andP)involved inDNAsynthesis.', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Inaddition tolysisgene17,P1probably hasa secondlysisgene(lyd[20]),analogous totheS geneofX,located somewhere between map units24andabout32.Therefore, thesetwolysis genesandalysisregulator gene(gene2),al- thoughnotalladjacent toeachother,arelocated withinaregionspanning lessthan15%oftheP1 genome (mapunits20toabout33). Gene10,whichregulates lategeneexpression (50),islocated approximately between map units94and96.Gene10isprobably analogous togeneQinX,whichisadjacent togenes(O andP)involved inDNAsynthesis. Ifsimilar clustering occursinP1,suchDNAsynthe- sisgenescouldbetotheleftofgene10,in aregionforwhichnogeneshaveasyetbeen found. Wedonotknowthefullextentofclustering in P1because oftheregions ofthegenome for whichnogenesareknown.Thatis,theregions mentioned above(between genes8and10and genes1and4)andanotherregionapproximately between mapunits67and76. Inaddition, verylittleisknownaboutthe transcription patterns inP1,sowedonotknow howmuchtheclustering offunctionally related genesisrelated totheircotranscription. (An exception isthatam49ispolarongene19,and thegeneitrepresents istherefore probably cotranscribed withgene19.) Genome organization ofP1compared withthat ofotherphages.Thefunctionally relatedgenes ofP1arenotclustered asconservatively asthey are,forexample, inX,inwhichallheadgenes areinoneregion,adjacent toallofthetailgenes inanother region,andinwhichallthegenes determining otherfunctions (suchasintegration, excision, recombination, generegulation, repli- cation,andlysis)areintheremaining regionof thegenome (10).Theorganization oftheP1 genome morecloselyresembles thatofT4(54), inwhichthetailgenesarelocatedinatleastsix regions andtheheadgenesinatleastfive regions. InT4,thesemorphogenetic geneclus- tersareinterspersed withgenesconcerned with, forexample, DNAmetabolism. InP1,morpho- geneticgeneclusters areinterrupted bymany genesthatareexpressed intheprophage, such asthegenesforestablishment andmaintenance oflysogeny, plasmid replication, immunity, or restriction andmodification ofDNA.Asforthe precisefunctions ofthegenesforwhichwehave ambermutants, furtherinvestigations withother techniques arerequired.VOL.45,1983', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1138 WALKER ANDWALKER ACKNOWLEDGMENTS WethankJeanMorningstar forhertechnical assistance, particularly withserumblocking powertests,MarkUrban- owskiforhisphotographic assistance, andMichael Feissfor reading themanuscript. WethankMichael Yarmolinsky and NatStemnberg forcommunicating unpublished data.Wethank MarySmithson fortypingthemanuscript. Thiswork assupported bygrantPCM78-23223 fromthe National Science Foundation. LITERATURE CITED 1.Arisaka, F.,J.Tschopp, R.vanDriel,andJ.Engel.1979. Reassembly ofthebacteriophage T4tailfromthecore- baseplate andthemonomeric sheathproteinP18:aco- operative association process. J.Mol.Biol.132:369-386. 2.Bachi,B.,andW.Arber.1977.Physical mapping ofBglII, BamHI, EcoRI,HindlIIandPstIrestriction fragments of bacteriophage P1DNA.Mol.Gen.Genet.153:311-324. 3.Beckendorf, S.K.1973.Structure ofthedistalhalfofthe bacteriophage T4tailfiber.J.Mol.Biol.73:37-53. 4.Becker, A.,andM.Gold.1975.Isolation ofthebacterio- phagelambda A-geneprotein. Proc.Natl.Acad.Sci. U.S.A.72:581-585. 5.Bertani, G.1951.Studies onlysogenesis. I.Themodeof phageliberation bylysogenic Escherichia coli.J.Bacte- riol.62:293-300. 6.Botstein, D.,C.H.Waddell, andJ.King.1973.Mecha- nismofheadassembly andDNAencapsulation inSalmo- nellaphageP22. I.Genes,proteins, structures, andDNA maturation. J.Mol.Biol.80:669-695. 7.Chow,L.T.,T.R.Broker, R.Kahmann, andD.Kamp. 1978.Comparison oftheGDNAinversion inbacterio- phagesMu,P1,andP7,p.55-56.InD.Schlessinger (ed.), Microbiology-1978. American SocietyforMicrobiology, Washington, D.C. 8.Coombs, D.H.,andF.A.Eiserling. 1977.Studies onthe structure, proteincomposition andassembly oftheneck ofbacteriophage T4.J.Mol.Biol.116:375-405. 9.DeMars, R.I.1955.Theproduction ofphage-related materials whenbacteriophage development isinterrupted byproflavine. Virology 1:83-99. 10.Echols, H.,andH.Murialdo. 1978.Genetic mapof bacteriophage lambda. Microbiol. Rev.42:577-591. 11.Edgar,R.S.,andI.Lielausis. 1968.Somestepsinthe assembly ofbacteriophage T4.J.Mol.Biol.32:263-276. 12.Epstein, R.H.,A.Bolle,C.M.Steinberg, E.Kelienberger, E.BoydelaTour,R.Chevalley, R.S.Edgar,M.Susman, G.H.Denhardt, andA.Lielausis. 1963.Physiological studiesofconditional lethalmutants ofbacteriophage T4D.ColdSpringHarborSymp.Quant.Biol.28:375-394. 13.Favre,R.,E.BoydelaTour,N.Segre,andE.Kellen- berger.1965.Studies onthemorphopoiesis oftheheadof phageT-even. I.Morphological, immunological andge- neticcharacterization ofpolyheads. J.Ultrastruct. Res. 13:318-342. 14.Frankel, F.R.,M.L.Batcheler, andC.K.Clark.1971. Theroleofgene49inDNAreplication andheadmorpho- genesisinbacteriophage T4.J.Mol.Biol.62:439-463.', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1963.Physiological studiesofconditional lethalmutants ofbacteriophage T4D.ColdSpringHarborSymp.Quant.Biol.28:375-394. 13.Favre,R.,E.BoydelaTour,N.Segre,andE.Kellen- berger.1965.Studies onthemorphopoiesis oftheheadof phageT-even. I.Morphological, immunological andge- neticcharacterization ofpolyheads. J.Ultrastruct. Res. 13:318-342. 14.Frankel, F.R.,M.L.Batcheler, andC.K.Clark.1971. Theroleofgene49inDNAreplication andheadmorpho- genesisinbacteriophage T4.J.Mol.Biol.62:439-463. 15.Georgopoulos, C.P.,R.W.,Hendrix, A.D.Kaiser, and W.B.Wood.1972.Roleofthehostcellinbacteriophage morphogenesis. Nature(London) NewBiol.239:38-41. 16.Giphart-Gassler, M.,R.H.A.Plasterk, andP.vande Putte.1982.Ginversion inbacteriophage Mu:anovelway ofgenesplicing. Nature(London) 297:339-342. 17.Granboulan, P.,J.S6chaud, andE.Kellenberger. 1971. Onthefragility ofphageT4-related particles. Virology 46:407-425. 18.Heilmann, H.,J.N.Reeve,andA.Puhler. 1980.Identifi- cationoftherepressor andrepressor bypass(antirepres- sor)polypeptides ofbacteriophage P1synthesized in infected minicells. Mol.Gen.Genet.178:149-154. 19.Howe,M.M.,J.W.Schumm, andA.L.Taylor.1979.The SandUgenesofbacteriophage Muarelocated inthe invertible Gsegment ofMuDNA.Virology 92:108-124.20.Ilda,S.,andW.Arber.1977.Plaqueformingspecializedtransducing phageP1:isolation ofPlCmSmSu, aprecur- sorofPlCm.Mol.Gen.Genet.153:259-269. 21.Iida,S.,andW.Arber.1979.Multiplephysicaldifferences inthegenome structure offunctionally relatedbacterio- phagesPlandP7.Mol.Gen.Genet.153:249-261. 22.Israel,J.V.,T.F.Anderson, andM.Levine.1967.Invitro morphogenesis ofphageP22fromheadsandbase-plateparts.Proc.Nati.Acad.Sci.U.S.A.57:284-291. 23.Kaiser, A.D.,M.Syvanen, andT.Masuda. 1975.DNA packaging stepsinbacteriophage lambdaheadassembly.J.Mol.Biol.91:175-186. 24.Kemp,C.L.,A.F.Howatson, andL.Siminovitch. 1968. Electronmicroscope studiesofmutantsoflambdabacte- riophage. I.Generaldescription andquantitation ofviral products. Virology 36:490-502. 25.Kikuchi,Y.,andJ.King.1975.Geneticcontrolofbacte- riophage T4baseplate morphogenesis: III.Formation of thecentralplugandoverallassemblypathway. J.Mol. Biol.99:695-716. 26.King,J.1968.Assembly ofthetailofbacteriophage T4.J. Mol.Biol.32:231-262. 27.King,J.1971.Bacteriophage T4tailassembly: foursteps incoreformation. J.Mol.Biol.58:693-709. 28.King,J.,E.V.Lenk,andD.Botstein. 1973.Mechanism of headassembly andDNAencapsulation inSalmonella phageP22.J.Mol.Biol.80:697-731. 29.King,J.,andN.Mykolajewycz. 1973.Bacteriophage T4 tailassembly: proteinsofthesheath,core,andbaseplate.J.Mol.Biol.75:339-358.', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  J.Mol. Biol.99:695-716. 26.King,J.1968.Assembly ofthetailofbacteriophage T4.J. Mol.Biol.32:231-262. 27.King,J.1971.Bacteriophage T4tailassembly: foursteps incoreformation. J.Mol.Biol.58:693-709. 28.King,J.,E.V.Lenk,andD.Botstein. 1973.Mechanism of headassembly andDNAencapsulation inSalmonella phageP22.J.Mol.Biol.80:697-731. 29.King,J.,andN.Mykolajewycz. 1973.Bacteriophage T4 tailassembly: proteinsofthesheath,core,andbaseplate.J.Mol.Biol.75:339-358. 30.Laemmli, U.K.,E.Molbert, M.Showe,andE.Kellen- berger.1970.Form-determining function ofthegenes required fortheassembly oftheheadofbacteriophage T4. J.Mol.Biol.49:99-113. 31.Laski,F.,andE.N.Jackson. 1982.Maturation cleavage of bacteriophage P22DNAintheabsence ofDNApackag-ing.J.Mol.Biol.154:565-579. 32.Lindahl, G.1974.Characterization ofconditional lethal mutantsofbacteriophage P2.Mol.Gen.Genet.128:249- 260. 33.Luftig,R.B.,W.B.Wood,andR.Okinaka. 1971. Bacteriophage T4headmorphogenesis: onthenatureof gene49-defective headsandtheirroleasintermediates. J. Mol.Biol.57:555-573. 34.Mural,R.J.,R.J.Chesney, D.Vapnek, M.M.Kropf,and J.R.Scott.1979.Isolation andcharacterization ofcloned fragments ofbacteriophage P1DNA.Virology 93:387- 397. 35.Onorato, L.,andM.K.Showe.1975.Gene21protein-dependent proteolysis invitroofpurifiedgene22productofbacteriophage T4.J.Mol.Biol.92:395-412. 36.Poteete, A.R.,andJ.King.1977.Functions oftwonew genesinSalmonella phageP22assembly. Virology76:725-739. 37.Ray,P.,andH.Murialdo. 1975.TheroleofgeneNu3in bacteriophage lambda headmorphogenesis. Virology64:247-263. 38.Razza,R.B.,C.A.Watkins, andJ.R.Scott.1980.PhageP1temperature-sensitive mutants withdefectsinthelyticpathway.Virology 105:52-59. 39.Scott,J.R.1968.Genetic studies onbacteriophage P1. Virology 36:564-574. 40.Showe,M.K.,andL.W.Black.1973.Assembly coreof bacteriophage T4:anintermediate inheadformation. Nature(London) NewBiol.242:70-75. 41.Showe,M.K.,andE.Kellenberger. 1975.Controlmecha- nismsinvirusassembly, p.407-438. InD.C.Burkeand W.C.Russel(ed.),Controlprocesses invirusmultiplica-tion.Cambridge University Press,Cambridge, England.42.Simon,L.1972.Infection ofEscherichia colibyT2andT4 bacteriophages asseenintheelectronmicroscope: T4 headmorphogenesis. Proc.Natl.Acad.Sci.U.S.A. 69:907-911. 43.Sternberg, N.1979.Acharacterization ofbacteriophageP1DNAfragments clonedinaXvector.Virology96:129-J.VIROL.', \"page_label: 22 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1139 142. 44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky, andR.Hoess.1981.Site-specific recombination andits roleinthelifecycleofbacteriophage P1.ColdSpring HarborSymp.Quant.Biol.45:297-309. 45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de Bruijn,andA.I.Bukhari. 1978.Relationships between temperate phagesMuandP1.Virology 89:146-161. 46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler. 1980.Invertible DNAdetermines hostspecificity ofbac- teriophage Mu.Nature(London) 286:218-222. 47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi- calvariants ofcoliphage P1.J.Virol.5:765-782. 48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic studiesofcoliphage P1.I.Mapping byuseofprophage deletions. J.Virol.16:525-534. 49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic studiesofcoliphage P1. III.Extended genetic map.J. Virol.20:177-187. 50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in coliphage P1affecting hostcelllysis.J.Virol.35:519-530. 51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.), Progress inclinicalbiological research, vol.64.AlanR. Liss,Inc.,NewYork. 52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman, H.Lyle,andW.B.Wood.1970.Assembly ofbacterio- phageT4tailfibers. II.Isolation andcharacterization of tailfiberprecursors. J.Mol.Biol.54:15-31. 53.Wood,W.B.,andJ.King.1979.Genetic control of complex bacteriophage assembly. Compr. Virol.13:581- 633. 54.Wood,W.B.,andH.R.Revel.1976.Thegenome of bacteriophage T4.Bacteriol. Rev.40:847-868. 55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination ofgeneproduct positions inbacteriophage T4byspecific antibody association. J.Mol.Biol.51:411-421. 56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and E.Kellenberger. 1970.Studies onthemorphopoiesis of theheadofbacteriophage T-even. VIII.Multilayered polyheads. J.Mol.Biol.50:35-58. 57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In S.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer Institute, Bethesda, Md.VOL.45,1983\", 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Interactions between Viral Regulatory Proteins Ensure an MOI- Independent Probability of Lysogeny during Infection by Bacteriophage P1 Kailun Zhang ,a,bKiara Pankratz ,aHau Duong ,aMatthew Theodore ,a,bJingwen Guan ,a,b,c*Anxiao (Andrew) Jiang ,d Yiruo Lin ,d Lanying Zenga,b,c aDepartment of Biochemistry and Biophysics, Texas A&M University, College Station, Texas, USA bCenter for Phage Technology, Texas A&M University, College Station, Texas, USA cMolecular and Environmental Plant Sciences, Texas A&M University, College Station, Texas, USA dDepartment of Computer Science and Engineering, Texas A&M University, College Station, Texas, USA ABSTRACT Phage P1 is a temperate phage which makes the lytic or lysogenic decision upon infecting bacteria. During the lytic cycle, progeny phages are pro-duced and the cell lyses, and in the lysogenic cycle, P1 DNA exists as a low-copy-number plasmid and replicates autonomously. Previous studies at the bulk levelshowed that P1 lysogenization was independent of multiplicity of infection (MOI;the number of phages infecting a cell), whereas lysogenization probability of theparadigmatic phage lincreases with MOI. However, the mechanism underlying t h eP 1b e h a v i o ri su n c l e a r .I nt h i sw o r k ,u s i n ga ﬂuorescent reporter system, we demonstrated this P1 MOI-independent lysogenic response at the single-cell level.We further observed that the activity of the major repressor of lytic functions (C1) is a determining factor for the ﬁnal cell fate. Speci ﬁcally, the repression activity of P1, which arises from a combination of C1, the anti-repressor Coi, and the core-pressor Lxc, remains constant for different MOI, which results in the MOI-inde-pendent lysogenic response. Additionally, by increasing the distance betweenphages that infect a single cell, we were able to engineer a l-like, MOI-dependent lysogenization upon P1 infection. This suggests that the large separation of coin-fecting phages attenuates the effective communication between them, allowingthem to make decisions independently of each other. Our work establishes ahighly quantitative framework to describe P1 lysogeny establishment. This system plays an important role in disseminating antibiotic resistance by P1-like plasmids and provides an alternative to the lifestyle of phage l. IMPORTANCE Phage P1 has been shown potentially to play an important role in disseminating antibiotic resistance among bacteria during lysogenization, as evi-denced by the prevalence of P1 phage-like elements in animal and human patho- gens. In contrast to phage l, a cell fate decision-making paradigm, P1 lysogeniza- tion was shown to be independent of MOI. In this work, we built a simple geneticmodel to elucidate this MOI independency based on the gene-regulatory circuitryof P1. We also proposed that the effective communication between coinfectingphages contributes to the lysis-lysogeny decision-making of P1 and highlightedthe signi ﬁcance of spatial organization in the process of cell fate determination in a single-cell environment. Finally, our work provides new insights into differentstrategies acquired by viruses to interact with their bacterial hosts in different sce-narios for their optimal survival. KEYWORDS bacteriophage P1, cell fate, gene regulatory network, infection, lysis- lysogeny, phage communicationCitation Zhang K, Pankratz K, Duong H, Theodore M, Guan J, Jiang AA, Lin Y, Zeng L.2021. Interactions between viral regulatory proteins ensure an MOI-independent probability of lysogeny during infection bybacteriophage P1. mBio 12:e01013-21. https:// doi.org/10.1128/mBio.01013-21 . Editor Susan Gottesman, National Cancer Institute Copyright © 2021 Zhang et al. This is an open- access article distributed under the terms oftheCreative Commons Attribution 4.0 International license . Address correspondence to Lanying Zeng, lzeng@tamu.edu.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  KEYWORDS bacteriophage P1, cell fate, gene regulatory network, infection, lysis- lysogeny, phage communicationCitation Zhang K, Pankratz K, Duong H, Theodore M, Guan J, Jiang AA, Lin Y, Zeng L.2021. Interactions between viral regulatory proteins ensure an MOI-independent probability of lysogeny during infection bybacteriophage P1. mBio 12:e01013-21. https:// doi.org/10.1128/mBio.01013-21 . Editor Susan Gottesman, National Cancer Institute Copyright © 2021 Zhang et al. This is an open- access article distributed under the terms oftheCreative Commons Attribution 4.0 International license . Address correspondence to Lanying Zeng, lzeng@tamu.edu. *Present address: Jingwen Guan, Department of Microbiology and Immunology, University of California, San Francisco, San Francisco,California, USA. Received 5 April 2021 Accepted 16 August 2021 Published  September/October 2021 Volume 12 Issue 5 e01013-21 ®mbio.asm.org 1 RESEARCH ARTICLE 14September2021'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': [\"page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1136 WALKER ANDWALKER fiber.Mutant72ismissinga105,000-molecular- weight(MW)proteinthat,unlessitisafusion product, istheproductofgene19(unpublished data).Toussaint etal.(45)locatedthegene19 mutations withinorneartheP1Cloop(an invertible DNAregion)because (i)thisisthe onlyregionofhomology sharedwithphageMu (i.e.,theMuGloop)and(ii)theamber'alleles oftheseP1mutants arerescued fromMu.The PlCloopis3kblongandflankedbytwo0.62- kbinverted repeats. The105,000-MW protein codedbygene19wouldrequireacodinglength of2.9kbofDNA.Gene19probably doesnot extendmorethanhalfway (1.5kb)intotheG loop(byanalogy withMu[46]);therefore, the remaining portion(1.4kb)ofthegenewould includeoneofthe0.62-kbinverted repeatsand extendatleast0.78kboutside theinverted repeat.Ifthe105,000-MW proteinisnotafusion product, evenmoreofgene19couldbelocated outside theCloop.Thisisbecauseamber'allelesoftwoMugenes,oneofwhich(U)is whollywithintheGloopandanotherofwhich (S)ispartially withintheGloopandpartially in theotregion(16),arerescuedfromP1(45).This suggests thatthereisanothergeneinaddition to gene19intheP1Cloop.Products ofboththeU andSgenesarerequired forMuadsorption (46); therefore, inP1thetwoanalogous genescould codeforthedistal(gene19)andproximal parts oftheP1tailfiber. Thefollowing findingssupportthesuggestion thatgene19ispartially outsidetheCloopand thatitincludes oneoftheinverted repeats. The P1-related phageP7,whichhasa0.2-kbdeletion (orsubstitution) withintherightinverted repeat (7),produces a100,000-MW proteininsteadof the105,000-MW P1tailfiberprotein(51).(These findingsalsosuggestthatgene19extends tothe rightratherthantotheleftoftheCloop.) Transcription ofgene19isleftward fromam49 (because am49ispolarongene19;Fig.2).Also, nopolypeptide fragment ofthe105,000-MW proteinisseenaftersodiumdodecyl sulfate- polyacrylamide gelelectrophoresis ofam72(un- published data).Therefore, unlesstheamber fragment hasbeendegraded, thismutation must beneartheendofgene19thatisproximal to wheretranscription isinitiated. Themodelofgene19asbeingpartially within andpartially outside theCloopisconsistent withthefollowing findings: (i)MuandP1have partialserological cross-reactivity (45),and(ii) thehostrangespecificities ofthesetwophages arewithin(ornear)theseinvertible regions(45, 46).Byanalogy withT4(3),thesehostrange specificities areprobably locatedinthetailfiber genesofthephages. Assembly ofP1particles. Byexamining the structures obtained indefective lysates, itisseenthatP1headsandtailscanbeassembled independently. Thissuggests that,likeT4,these Plstructures haveindependent assembly path- ways.However, inP1,tailfibersareoftenseen attached toheadless tails.Therefore, unlikeT4 tailfibers,P1tailfiberscanattachtothetail beforehead-tail joiningoccurs. (Thishadal- readybeensuspected because theP1tailfibers arenotlongenoughtoreachfromthebaseplate tothehead,suchalengthbeingrequired forthe addition oftailfiberstoT4particles.) Distribution ofmutations. TheP1ambermuta- tionsaffecting particle morphogenesis arenot equallydistributed amongthedesignated genes.\", 'page_label: 19 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Assembly ofP1particles. Byexamining the structures obtained indefective lysates, itisseenthatP1headsandtailscanbeassembled independently. Thissuggests that,likeT4,these Plstructures haveindependent assembly path- ways.However, inP1,tailfibersareoftenseen attached toheadless tails.Therefore, unlikeT4 tailfibers,P1tailfiberscanattachtothetail beforehead-tail joiningoccurs. (Thishadal- readybeensuspected because theP1tailfibers arenotlongenoughtoreachfromthebaseplate tothehead,suchalengthbeingrequired forthe addition oftailfiberstoT4particles.) Distribution ofmutations. TheP1ambermuta- tionsaffecting particle morphogenesis arenot equallydistributed amongthedesignated genes. Forexample, including mutations notstudiedin theworkreported here(Fig.2),thereare13 mutations ingene1andonly1mutation ineach ofgenes22,23,and8.Wedonotknowthesizes ofthesegenes,butitmaybethatthereare\"hot spots\"ofmutation (obtained byhydroxyl- amine),aswasfoundinP22(36).Insupportof this,wedoknowthatthereare10mutations in gene10,whichcodesforaproteinof64,000MW (18)involved inlategeneexpression (50),anda maximum of3mutations ingene19,which codesfora105,000-MW tailfiberprotein. The existence ofhotspotsofmutation couldexplain whysomeofthehead-defective phenotypes described abovehavenotyetbeenseen. Correlation ofphenotypes withtheP1genetic andphysicalmaps.The103P1ambermutations havebeenassigned (49)to10lineararraysor linkageclusters (Ithrough X;Fig.2).Because complementation testsinliquidoftengaveam- biguousresultsandnotallmutants weretested inallcombinations, onlysomeofthemutations wereputintogroupsthatshowed nocomple- mentation. The55ambermutantphenotypes (i.e.,including mutations ingenes17,2,and10) thatwedetermined correlate withthelinear arraysofthecorresponding mutations. Forex- ample,inlinkageclusterVI,mutants 26,32,and 109showonephenotype andmutants27,39,53, 22,and116showanother. Inaddition, except formutants 180and45(whichmaybepolar mutants), allmutants testedthatwerefoundnot tocomplement (Fig.2)showed thesamepheno- type. Wepreviously suggested (49)thatthephe- nomenon oflinkageclustersmightresultfroma varietyofunderlying causes,suchasrecombina- tionhotspots,longsegments ofsilentDNA,or regionsforwhichtherewerenoknownmuta- tions.Thephysical position ofthemutations comprising thelinkageclusters isnowknown (Fig.17)(57),anditcanbeseenthattheclusters consistofsinglegenesorgroupsofgenesand thatthedemarcations between someofthese linkageclusters (i.e.,II/IlI,VI/VII,VII/VIII,VIII/IX, andX-2/X-1) areduetorecombinationJ.VIROL.', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1137 between thesegenesorgenegroups.Thereason whysomelinkage clusters arecomprised of singlegenesandsomearecomprised ofgene groupsisunknown, unlesshotspotsofrecombi- nationareinvoked. Otherdemarcations be- tweenlinkage clusters, suchasI/IT,III/IV, IV/V,V/VI,andIX/X-2, areduetolongseg- mentsofDNAthateithercontain othergenes (e.g.,codingforimmunity) orforwhichno mutations areknown,oracombination ofthese tworeasons. (Itmaybethattherearenolong segments ofsilentDNAbecause, forexample, somemorphogenetic genes[especially head genes]arestillundetected.) Between linkage clusters X-1andI(i.e.,attheendsofthe vegetative map),thereisashorter segment of DNAthatcodesforpartoftheP1immunity system. However, thisregionhasalsobeen showntocontainarecombination hotspot(loxP [44]),whichalonecouldaccount forthedemar- cationbetween thesetwoclusters. Clustering offunctionally relatedgenes.The physical andgeneticmapsofP1showconsider- ableclustering ofgeneshavingrelatedfunctions. Suchclustering couldbeadvantageous forregu- latingtheassembly ofsubstructures, suchas,for example, baseplates. Wedonotknowwhether gene1isaheadoratailgene.Excluding gene1, theknowntailgenesarelocatedinthreeregions. Theapproximate physical positions ofthese regionsaremapunits34to49,56to59,and78to 84.5.Withintheseregions, somesimilarfunc- tionsarelocatedtogether. Inthefirstregion therearethreetofivetailfibergenesandthree tofourbaseplate (ortailtube)genes.Inthe secondregionwefoundonetailsheathgeneand onebaseplate (ortailtube)gene.Itmaybe,by analogywithT4(54),thatthesingleP1tailtube geneislocatedinthisregionbesidethesheath gene.Inthethirdregionwefoundoneortwo baseplate genes,ataillengthgene,twoorthree tailstability genes,andoneortwobaseplate ortailtubegenes.Inthisregionthebaseplate genesareseparated byothertailgenes.How- ever,unlikegenes3and15ofT4(54),thetwo tothreeP1tailstability genesarelocated to- gether. ThefourtofiveknownPlheadgenesarein fourregionsofthegenome. Thiswouldargue againstclustering offunctionally relatedhead genes.However, someexpected P1headgenes havenotyetbeendetected andcouldbelocated intwomainregions forwhichtherearefew knowngenes.Oneoftheseregions comprises mapunits10to20andcouldtheoretically code forabout10genes,including headgene4,the headgenerepresented byam108(ifdifferent fromgene4),andgenes13and14foundby Razzaetal.(38).Theotherregioncomprises mapunits84.5to94,whichcouldcodefor9or10genes,including headgene8.Inaddition, itis interesting tonotethatheadgene9,whichis adjacent topac,thesiteatwhichDNApackag- ingisinitiated, hasnowbeenfoundtocodefor P1DNAcleavage atthissite(N.Sternberg, personal communication). Inaddition tolysisgene17,P1probably hasa secondlysisgene(lyd[20]),analogous totheS geneofX,located somewhere between map units24andabout32.Therefore, thesetwolysis genesandalysisregulator gene(gene2),al- thoughnotalladjacent toeachother,arelocated withinaregionspanning lessthan15%oftheP1 genome (mapunits20toabout33). Gene10,whichregulates lategeneexpression (50),islocated approximately between map units94and96.Gene10isprobably analogous togeneQinX,whichisadjacent togenes(O andP)involved inDNAsynthesis.', 'page_label: 20 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  Inaddition tolysisgene17,P1probably hasa secondlysisgene(lyd[20]),analogous totheS geneofX,located somewhere between map units24andabout32.Therefore, thesetwolysis genesandalysisregulator gene(gene2),al- thoughnotalladjacent toeachother,arelocated withinaregionspanning lessthan15%oftheP1 genome (mapunits20toabout33). Gene10,whichregulates lategeneexpression (50),islocated approximately between map units94and96.Gene10isprobably analogous togeneQinX,whichisadjacent togenes(O andP)involved inDNAsynthesis. Ifsimilar clustering occursinP1,suchDNAsynthe- sisgenescouldbetotheleftofgene10,in aregionforwhichnogeneshaveasyetbeen found. Wedonotknowthefullextentofclustering in P1because oftheregions ofthegenome for whichnogenesareknown.Thatis,theregions mentioned above(between genes8and10and genes1and4)andanotherregionapproximately between mapunits67and76. Inaddition, verylittleisknownaboutthe transcription patterns inP1,sowedonotknow howmuchtheclustering offunctionally related genesisrelated totheircotranscription. (An exception isthatam49ispolarongene19,and thegeneitrepresents istherefore probably cotranscribed withgene19.) Genome organization ofP1compared withthat ofotherphages.Thefunctionally relatedgenes ofP1arenotclustered asconservatively asthey are,forexample, inX,inwhichallheadgenes areinoneregion,adjacent toallofthetailgenes inanother region,andinwhichallthegenes determining otherfunctions (suchasintegration, excision, recombination, generegulation, repli- cation,andlysis)areintheremaining regionof thegenome (10).Theorganization oftheP1 genome morecloselyresembles thatofT4(54), inwhichthetailgenesarelocatedinatleastsix regions andtheheadgenesinatleastfive regions. InT4,thesemorphogenetic geneclus- tersareinterspersed withgenesconcerned with, forexample, DNAmetabolism. InP1,morpho- geneticgeneclusters areinterrupted bymany genesthatareexpressed intheprophage, such asthegenesforestablishment andmaintenance oflysogeny, plasmid replication, immunity, or restriction andmodification ofDNA.Asforthe precisefunctions ofthegenesforwhichwehave ambermutants, furtherinvestigations withother techniques arerequired.VOL.45,1983', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1138 WALKER ANDWALKER ACKNOWLEDGMENTS WethankJeanMorningstar forhertechnical assistance, particularly withserumblocking powertests,MarkUrban- owskiforhisphotographic assistance, andMichael Feissfor reading themanuscript. WethankMichael Yarmolinsky and NatStemnberg forcommunicating unpublished data.Wethank MarySmithson fortypingthemanuscript. Thiswork assupported bygrantPCM78-23223 fromthe National Science Foundation. LITERATURE CITED 1.Arisaka, F.,J.Tschopp, R.vanDriel,andJ.Engel.1979. Reassembly ofthebacteriophage T4tailfromthecore- baseplate andthemonomeric sheathproteinP18:aco- operative association process. J.Mol.Biol.132:369-386. 2.Bachi,B.,andW.Arber.1977.Physical mapping ofBglII, BamHI, EcoRI,HindlIIandPstIrestriction fragments of bacteriophage P1DNA.Mol.Gen.Genet.153:311-324. 3.Beckendorf, S.K.1973.Structure ofthedistalhalfofthe bacteriophage T4tailfiber.J.Mol.Biol.73:37-53. 4.Becker, A.,andM.Gold.1975.Isolation ofthebacterio- phagelambda A-geneprotein. Proc.Natl.Acad.Sci. U.S.A.72:581-585. 5.Bertani, G.1951.Studies onlysogenesis. I.Themodeof phageliberation bylysogenic Escherichia coli.J.Bacte- riol.62:293-300. 6.Botstein, D.,C.H.Waddell, andJ.King.1973.Mecha- nismofheadassembly andDNAencapsulation inSalmo- nellaphageP22. I.Genes,proteins, structures, andDNA maturation. J.Mol.Biol.80:669-695. 7.Chow,L.T.,T.R.Broker, R.Kahmann, andD.Kamp. 1978.Comparison oftheGDNAinversion inbacterio- phagesMu,P1,andP7,p.55-56.InD.Schlessinger (ed.), Microbiology-1978. American SocietyforMicrobiology, Washington, D.C. 8.Coombs, D.H.,andF.A.Eiserling. 1977.Studies onthe structure, proteincomposition andassembly oftheneck ofbacteriophage T4.J.Mol.Biol.116:375-405. 9.DeMars, R.I.1955.Theproduction ofphage-related materials whenbacteriophage development isinterrupted byproflavine. Virology 1:83-99. 10.Echols, H.,andH.Murialdo. 1978.Genetic mapof bacteriophage lambda. Microbiol. Rev.42:577-591. 11.Edgar,R.S.,andI.Lielausis. 1968.Somestepsinthe assembly ofbacteriophage T4.J.Mol.Biol.32:263-276. 12.Epstein, R.H.,A.Bolle,C.M.Steinberg, E.Kelienberger, E.BoydelaTour,R.Chevalley, R.S.Edgar,M.Susman, G.H.Denhardt, andA.Lielausis. 1963.Physiological studiesofconditional lethalmutants ofbacteriophage T4D.ColdSpringHarborSymp.Quant.Biol.28:375-394. 13.Favre,R.,E.BoydelaTour,N.Segre,andE.Kellen- berger.1965.Studies onthemorphopoiesis oftheheadof phageT-even. I.Morphological, immunological andge- neticcharacterization ofpolyheads. J.Ultrastruct. Res. 13:318-342. 14.Frankel, F.R.,M.L.Batcheler, andC.K.Clark.1971. Theroleofgene49inDNAreplication andheadmorpho- genesisinbacteriophage T4.J.Mol.Biol.62:439-463.', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  1963.Physiological studiesofconditional lethalmutants ofbacteriophage T4D.ColdSpringHarborSymp.Quant.Biol.28:375-394. 13.Favre,R.,E.BoydelaTour,N.Segre,andE.Kellen- berger.1965.Studies onthemorphopoiesis oftheheadof phageT-even. I.Morphological, immunological andge- neticcharacterization ofpolyheads. J.Ultrastruct. Res. 13:318-342. 14.Frankel, F.R.,M.L.Batcheler, andC.K.Clark.1971. Theroleofgene49inDNAreplication andheadmorpho- genesisinbacteriophage T4.J.Mol.Biol.62:439-463. 15.Georgopoulos, C.P.,R.W.,Hendrix, A.D.Kaiser, and W.B.Wood.1972.Roleofthehostcellinbacteriophage morphogenesis. Nature(London) NewBiol.239:38-41. 16.Giphart-Gassler, M.,R.H.A.Plasterk, andP.vande Putte.1982.Ginversion inbacteriophage Mu:anovelway ofgenesplicing. Nature(London) 297:339-342. 17.Granboulan, P.,J.S6chaud, andE.Kellenberger. 1971. Onthefragility ofphageT4-related particles. Virology 46:407-425. 18.Heilmann, H.,J.N.Reeve,andA.Puhler. 1980.Identifi- cationoftherepressor andrepressor bypass(antirepres- sor)polypeptides ofbacteriophage P1synthesized in infected minicells. Mol.Gen.Genet.178:149-154. 19.Howe,M.M.,J.W.Schumm, andA.L.Taylor.1979.The SandUgenesofbacteriophage Muarelocated inthe invertible Gsegment ofMuDNA.Virology 92:108-124.20.Ilda,S.,andW.Arber.1977.Plaqueformingspecializedtransducing phageP1:isolation ofPlCmSmSu, aprecur- sorofPlCm.Mol.Gen.Genet.153:259-269. 21.Iida,S.,andW.Arber.1979.Multiplephysicaldifferences inthegenome structure offunctionally relatedbacterio- phagesPlandP7.Mol.Gen.Genet.153:249-261. 22.Israel,J.V.,T.F.Anderson, andM.Levine.1967.Invitro morphogenesis ofphageP22fromheadsandbase-plateparts.Proc.Nati.Acad.Sci.U.S.A.57:284-291. 23.Kaiser, A.D.,M.Syvanen, andT.Masuda. 1975.DNA packaging stepsinbacteriophage lambdaheadassembly.J.Mol.Biol.91:175-186. 24.Kemp,C.L.,A.F.Howatson, andL.Siminovitch. 1968. Electronmicroscope studiesofmutantsoflambdabacte- riophage. I.Generaldescription andquantitation ofviral products. Virology 36:490-502. 25.Kikuchi,Y.,andJ.King.1975.Geneticcontrolofbacte- riophage T4baseplate morphogenesis: III.Formation of thecentralplugandoverallassemblypathway. J.Mol. Biol.99:695-716. 26.King,J.1968.Assembly ofthetailofbacteriophage T4.J. Mol.Biol.32:231-262. 27.King,J.1971.Bacteriophage T4tailassembly: foursteps incoreformation. J.Mol.Biol.58:693-709. 28.King,J.,E.V.Lenk,andD.Botstein. 1973.Mechanism of headassembly andDNAencapsulation inSalmonella phageP22.J.Mol.Biol.80:697-731. 29.King,J.,andN.Mykolajewycz. 1973.Bacteriophage T4 tailassembly: proteinsofthesheath,core,andbaseplate.J.Mol.Biol.75:339-358.', 'page_label: 21 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  J.Mol. Biol.99:695-716. 26.King,J.1968.Assembly ofthetailofbacteriophage T4.J. Mol.Biol.32:231-262. 27.King,J.1971.Bacteriophage T4tailassembly: foursteps incoreformation. J.Mol.Biol.58:693-709. 28.King,J.,E.V.Lenk,andD.Botstein. 1973.Mechanism of headassembly andDNAencapsulation inSalmonella phageP22.J.Mol.Biol.80:697-731. 29.King,J.,andN.Mykolajewycz. 1973.Bacteriophage T4 tailassembly: proteinsofthesheath,core,andbaseplate.J.Mol.Biol.75:339-358. 30.Laemmli, U.K.,E.Molbert, M.Showe,andE.Kellen- berger.1970.Form-determining function ofthegenes required fortheassembly oftheheadofbacteriophage T4. J.Mol.Biol.49:99-113. 31.Laski,F.,andE.N.Jackson. 1982.Maturation cleavage of bacteriophage P22DNAintheabsence ofDNApackag-ing.J.Mol.Biol.154:565-579. 32.Lindahl, G.1974.Characterization ofconditional lethal mutantsofbacteriophage P2.Mol.Gen.Genet.128:249- 260. 33.Luftig,R.B.,W.B.Wood,andR.Okinaka. 1971. Bacteriophage T4headmorphogenesis: onthenatureof gene49-defective headsandtheirroleasintermediates. J. Mol.Biol.57:555-573. 34.Mural,R.J.,R.J.Chesney, D.Vapnek, M.M.Kropf,and J.R.Scott.1979.Isolation andcharacterization ofcloned fragments ofbacteriophage P1DNA.Virology 93:387- 397. 35.Onorato, L.,andM.K.Showe.1975.Gene21protein-dependent proteolysis invitroofpurifiedgene22productofbacteriophage T4.J.Mol.Biol.92:395-412. 36.Poteete, A.R.,andJ.King.1977.Functions oftwonew genesinSalmonella phageP22assembly. Virology76:725-739. 37.Ray,P.,andH.Murialdo. 1975.TheroleofgeneNu3in bacteriophage lambda headmorphogenesis. Virology64:247-263. 38.Razza,R.B.,C.A.Watkins, andJ.R.Scott.1980.PhageP1temperature-sensitive mutants withdefectsinthelyticpathway.Virology 105:52-59. 39.Scott,J.R.1968.Genetic studies onbacteriophage P1. Virology 36:564-574. 40.Showe,M.K.,andL.W.Black.1973.Assembly coreof bacteriophage T4:anintermediate inheadformation. Nature(London) NewBiol.242:70-75. 41.Showe,M.K.,andE.Kellenberger. 1975.Controlmecha- nismsinvirusassembly, p.407-438. InD.C.Burkeand W.C.Russel(ed.),Controlprocesses invirusmultiplica-tion.Cambridge University Press,Cambridge, England.42.Simon,L.1972.Infection ofEscherichia colibyT2andT4 bacteriophages asseenintheelectronmicroscope: T4 headmorphogenesis. Proc.Natl.Acad.Sci.U.S.A. 69:907-911. 43.Sternberg, N.1979.Acharacterization ofbacteriophageP1DNAfragments clonedinaXvector.Virology96:129-J.VIROL.', \"page_label: 22 file_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf  MORPHOLOGY OFP1MUTANTS 1139 142. 44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky, andR.Hoess.1981.Site-specific recombination andits roleinthelifecycleofbacteriophage P1.ColdSpring HarborSymp.Quant.Biol.45:297-309. 45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de Bruijn,andA.I.Bukhari. 1978.Relationships between temperate phagesMuandP1.Virology 89:146-161. 46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler. 1980.Invertible DNAdetermines hostspecificity ofbac- teriophage Mu.Nature(London) 286:218-222. 47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi- calvariants ofcoliphage P1.J.Virol.5:765-782. 48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic studiesofcoliphage P1.I.Mapping byuseofprophage deletions. J.Virol.16:525-534. 49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic studiesofcoliphage P1. III.Extended genetic map.J. Virol.20:177-187. 50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in coliphage P1affecting hostcelllysis.J.Virol.35:519-530. 51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.), Progress inclinicalbiological research, vol.64.AlanR. Liss,Inc.,NewYork. 52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman, H.Lyle,andW.B.Wood.1970.Assembly ofbacterio- phageT4tailfibers. II.Isolation andcharacterization of tailfiberprecursors. J.Mol.Biol.54:15-31. 53.Wood,W.B.,andJ.King.1979.Genetic control of complex bacteriophage assembly. Compr. Virol.13:581- 633. 54.Wood,W.B.,andH.R.Revel.1976.Thegenome of bacteriophage T4.Bacteriol. Rev.40:847-868. 55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination ofgeneproduct positions inbacteriophage T4byspecific antibody association. J.Mol.Biol.51:411-421. 56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and E.Kellenberger. 1970.Studies onthemorphopoiesis of theheadofbacteriophage T-even. VIII.Multilayered polyheads. J.Mol.Biol.50:35-58. 57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In S.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer Institute, Bethesda, Md.VOL.45,1983\", 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Interactions between Viral Regulatory Proteins Ensure an MOI- Independent Probability of Lysogeny during Infection by Bacteriophage P1 Kailun Zhang ,a,bKiara Pankratz ,aHau Duong ,aMatthew Theodore ,a,bJingwen Guan ,a,b,c*Anxiao (Andrew) Jiang ,d Yiruo Lin ,d Lanying Zenga,b,c aDepartment of Biochemistry and Biophysics, Texas A&M University, College Station, Texas, USA bCenter for Phage Technology, Texas A&M University, College Station, Texas, USA cMolecular and Environmental Plant Sciences, Texas A&M University, College Station, Texas, USA dDepartment of Computer Science and Engineering, Texas A&M University, College Station, Texas, USA ABSTRACT Phage P1 is a temperate phage which makes the lytic or lysogenic decision upon infecting bacteria. During the lytic cycle, progeny phages are pro-duced and the cell lyses, and in the lysogenic cycle, P1 DNA exists as a low-copy-number plasmid and replicates autonomously. Previous studies at the bulk levelshowed that P1 lysogenization was independent of multiplicity of infection (MOI;the number of phages infecting a cell), whereas lysogenization probability of theparadigmatic phage lincreases with MOI. However, the mechanism underlying t h eP 1b e h a v i o ri su n c l e a r .I nt h i sw o r k ,u s i n ga ﬂuorescent reporter system, we demonstrated this P1 MOI-independent lysogenic response at the single-cell level.We further observed that the activity of the major repressor of lytic functions (C1) is a determining factor for the ﬁnal cell fate. Speci ﬁcally, the repression activity of P1, which arises from a combination of C1, the anti-repressor Coi, and the core-pressor Lxc, remains constant for different MOI, which results in the MOI-inde-pendent lysogenic response. Additionally, by increasing the distance betweenphages that infect a single cell, we were able to engineer a l-like, MOI-dependent lysogenization upon P1 infection. This suggests that the large separation of coin-fecting phages attenuates the effective communication between them, allowingthem to make decisions independently of each other. Our work establishes ahighly quantitative framework to describe P1 lysogeny establishment. This system plays an important role in disseminating antibiotic resistance by P1-like plasmids and provides an alternative to the lifestyle of phage l. IMPORTANCE Phage P1 has been shown potentially to play an important role in disseminating antibiotic resistance among bacteria during lysogenization, as evi-denced by the prevalence of P1 phage-like elements in animal and human patho- gens. In contrast to phage l, a cell fate decision-making paradigm, P1 lysogeniza- tion was shown to be independent of MOI. In this work, we built a simple geneticmodel to elucidate this MOI independency based on the gene-regulatory circuitryof P1. We also proposed that the effective communication between coinfectingphages contributes to the lysis-lysogeny decision-making of P1 and highlightedthe signi ﬁcance of spatial organization in the process of cell fate determination in a single-cell environment. Finally, our work provides new insights into differentstrategies acquired by viruses to interact with their bacterial hosts in different sce-narios for their optimal survival. KEYWORDS bacteriophage P1, cell fate, gene regulatory network, infection, lysis- lysogeny, phage communicationCitation Zhang K, Pankratz K, Duong H, Theodore M, Guan J, Jiang AA, Lin Y, Zeng L.2021. Interactions between viral regulatory proteins ensure an MOI-independent probability of lysogeny during infection bybacteriophage P1. mBio 12:e01013-21. https:// doi.org/10.1128/mBio.01013-21 . Editor Susan Gottesman, National Cancer Institute Copyright © 2021 Zhang et al. This is an open- access article distributed under the terms oftheCreative Commons Attribution 4.0 International license . Address correspondence to Lanying Zeng, lzeng@tamu.edu.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  KEYWORDS bacteriophage P1, cell fate, gene regulatory network, infection, lysis- lysogeny, phage communicationCitation Zhang K, Pankratz K, Duong H, Theodore M, Guan J, Jiang AA, Lin Y, Zeng L.2021. Interactions between viral regulatory proteins ensure an MOI-independent probability of lysogeny during infection bybacteriophage P1. mBio 12:e01013-21. https:// doi.org/10.1128/mBio.01013-21 . Editor Susan Gottesman, National Cancer Institute Copyright © 2021 Zhang et al. This is an open- access article distributed under the terms oftheCreative Commons Attribution 4.0 International license . Address correspondence to Lanying Zeng, lzeng@tamu.edu. *Present address: Jingwen Guan, Department of Microbiology and Immunology, University of California, San Francisco, San Francisco,California, USA. Received 5 April 2021 Accepted 16 August 2021 Published  September/October 2021 Volume 12 Issue 5 e01013-21 ®mbio.asm.org 1 RESEARCH ARTICLE 14September2021'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:06 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'110'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993453'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'392ms'), (b'x-request-id', b'70a0878ada84d82296818be42ee88d57'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249bff85f5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:06 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'110'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993453'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'392ms'), (b'x-request-id', b'70a0878ada84d82296818be42ee88d57'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249bff85f5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Bacteriophages and phage-like plasmids play a prominent role in the dissemination of adaptive traits that allow stable colonization and persistence of multidrug resist- ance in pathogenic bacteria (1 –12). A better understanding of their contribution to pathotype spread and adaptation is important. Phage P1 was discovered in a lysogenic strain of Escherichia coli named Li about 70 years ago (13). In terms of the ability to inject its DNA, P1 has the broadest known host range, including a variety of Gram-neg- ative enteric bacteria, e.g., E. coli ,Shigella dysenteriae (13), and Salmonella enterica sero- var Typhimurium (14 –17), and soil bacteria, including Pseudomonas aeruginosa (17, 18), Acetobacter suboxydans (17), and Myxococcus xanthus (19, 20). In the E. coli host, the infection cycle of P1 begins with adsorption to the cell surface by recognizing the ter- minal glucose moiety of the lipopolysaccharide (LPS) core region (21). The interaction of at least three of the six tail ﬁbers with receptor molecules is assumed to be suf ﬁcient to stimulate the injection process (22). Injected viral DNAs can be protected from host type I restriction-modi ﬁcation (RM) system by P1 antirestriction proteins (23 –25). Following DNA injection, the infected cell enters either the lytic cycle, where progeny phages are produced and the cell lyses, or the lysogenic cycle, in which the phage DNA exists as a low-copy-number plasmid and replicates autonomously. At the molec- ular level, the lysis-lysogeny decision is controlled by the interactions between phage components of the regulatory network (Fig. 1A). Brie ﬂy, the major repressor of P1, C1, is expressed immediately after phage infection and binds to 22 operator sequences widely dispersed over the P1 DNA to repress lytic gene expression. For example, C1controls the expression of RepL, which is responsible for vegetative DNA replication, and Lpa, which activates the late promoter serving viral morphogenesis and lysis genes (26–28). Coi ( c one inactivation) is an antirepressor that binds to C1 and inactivates its function (29 –31), whereas the corepressor Lxc (lowers expression of c1) increases C1 binding af ﬁnity by forming a ternary complex with C1 and operator DNA and inhibits the ability of Coi to dissociate the operator-C1 complex (32 –34). Moreover, the hetero- dimeric complex of Ant1 and Ant2 proteins also functions as an antirepressor, while thetrans -acting c4RNA represses the synthesis of Ant proteins (Fig. 1A) (for a detailed review, see reference 35). Some studies suggest that the initial choice between lysis and lysogeny relies on the relative synthesis rate of the main competing players, C1 and Coi (28), but no quantita- tive evidence has been available to determine the relationship between this competition and the ﬁnal lysis-lysogeny decision. In addition, the bene ﬁts of P1 acquiring different regulatory components besides the main competing players are not clear. Regrettably, studies of P1 regulatory circuitry halted prematurely with the death of their prime mover, Heinz Schuster (30, 31, 33 –38). Furthermore, interest in P1 decreased, since it does not contain cargo genes of clinical interest (e.g., stress resistance and nutrient acquisition), and, being a temperate phage, P1 has been not considered for antibacterial therapeutic applications (phage therapy). However, recent advances in sequencing technologies have allowed the identi ﬁcation of numerous P1-like plasmids in bacteria and human pathogens (2, 4 –6, 9, 11, 12), suggesting their abilities to serve as speci ﬁc vehicles for the spread and the maintenance of virulence and antimicrobial resistance in both clinical and livestock production settings (1, 9).', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Regrettably, studies of P1 regulatory circuitry halted prematurely with the death of their prime mover, Heinz Schuster (30, 31, 33 –38). Furthermore, interest in P1 decreased, since it does not contain cargo genes of clinical interest (e.g., stress resistance and nutrient acquisition), and, being a temperate phage, P1 has been not considered for antibacterial therapeutic applications (phage therapy). However, recent advances in sequencing technologies have allowed the identi ﬁcation of numerous P1-like plasmids in bacteria and human pathogens (2, 4 –6, 9, 11, 12), suggesting their abilities to serve as speci ﬁc vehicles for the spread and the maintenance of virulence and antimicrobial resistance in both clinical and livestock production settings (1, 9). For example, pTZ20_1P, isolated from a porcine commensal E. coli strain and carrying multiple antibiotic resistance markers, was able to lysogenize a commensal E. coli strain with consequent transfer of resistance (9). Moreover, the sequencing data showed that genes contributing to lysogeny establish- ment were very similar to the genes of the regulatory network of P1 (9). Therefore, athorough study of P1 life cycle and mechanism of its lysogeny establishment is impor- tant to understand the phage-driven antimicrobial resistance emergence and pathogen adaptation, which will bene ﬁt the development of novel therapeutic methods. Studies of other temperate phages revealed that the lysis-lysogeny decision is highly dependent on the host availability in the environment or the virus-cell ratio. For phage l, one of the best-studied paradigms for cell-fate decision-making, the probabil- ity of lysogeny increases with multiplicity of infection (MOI; the number of phagesZhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 2', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  infecting a cell) and approaches 100% when the MOI is suf ﬁciently high (39 –41). Upon infection of the Vibrio phage 882, a quorum sensing factor expressed from the host cell is utilized to make the lysis-lysogeny decision (42, 43). Moreover, during infection,some Bacillus phages produce a 6-amino-acid peptide, designated arbitrium, that accu- mulates in the medium. In turn, high levels of arbitrium cause subsequent phage infec-tions to be biased toward lysogeny (44, 45). These and other strategies are used by the phages to determine whether there are enough host cells for the infection by progeny phages, ensuring a high ef ﬁciency of genome propagation. In contrast, phage P1 has been reported to lack sensitivity to the virus-cell ratio. Previous studies by Bertani andNice (46) and Rosner (47) at the bulk level showed constant probabilities of lysogenyregardless of MOI: 80 to 90% of S. dysenteriae cells were lysogenized at 20°C when infected by the P1 strain isolated originally from the lysogenic E. coli strain Li (13); ;30% of the E. coli cells were lysogenized at 30°C by the thermoinducible P1 strain (P1CM c1-100) (47). The constant probability of lysogeny irrespective of host availability, together with P1 ’s capacity for autonomous prophage maintenance, utilizing systems such as type III RM (48 –50), plasmid replication (51, 52) and toxin-antitoxin (53, 54), is a strategy used by P1 to propagate its genomic information as a plasmid. Furthermore,the ability to establish and be maintained as a plasmid can also be bene ﬁcial to the dis- semination of the cargo genes carried by P1 or P1 phage-like elements. In this work, we set out to examine the infection process of phage P1 at the levels of individual phage and cell with a spatiotemporal resolution suf ﬁcient to quantify the relevant subcellular parameters and to evaluate the contribution of each parameter to the observed cell-fate decisions. Using our system, we ﬁrst investigated the mechanism of P1 MOI-independent lysogenization by linking the regulatory gene expression withcell fate by a simple genetic model. We also found that MOI dependency can beimposed by increasing the distances between the coinfecting phages. Taken together,our results provide a new decision-making model that expands our understanding oftemperate phages. This, in turn, gives us insight into how the ﬁtness of different phages results in distinctive decision-making behaviors and also allows us to evaluatetheir contributions to pathogenicity development. RESULTS The probability of lysogeny is independent of the number of infecting phages at the single-cell level. In order to study P1 lysogenization at the single-cell level, we constructed a ﬂuorescent reporter system to monitor individual decision-making events under the microscope, as described previously (55). Brie ﬂy, to record the num- ber of phages infecting each E. coli cell, P1 virions were labeled by fusing ﬂuorescent proteins to phage capsids. Moreover, during the time-lapse movies, the lytic pathway FIG 1 The probability of lysogeny is independent of the number of infecting phages at the single- cell level. (A) Schematic of the P1 lysis-lysogeny regulatory network. (B) Probability of lysogeny as afunction of MOI. The probability of lysogeny remains constant over different MOI for wild-type (WT)cells at ;14.53% (blue; cell sample sizes for MOI of 1 to 5 are 457, 277, 168, 62, and 68, respectively) and for Lxc-overexpressing cells at ;26.58% (black; cell sample sizes for MOI of 1 to 5 are 295, 132, 55, 38, and 15, respectively).', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Moreover, during the time-lapse movies, the lytic pathway FIG 1 The probability of lysogeny is independent of the number of infecting phages at the single- cell level. (A) Schematic of the P1 lysis-lysogeny regulatory network. (B) Probability of lysogeny as afunction of MOI. The probability of lysogeny remains constant over different MOI for wild-type (WT)cells at ;14.53% (blue; cell sample sizes for MOI of 1 to 5 are 457, 277, 168, 62, and 68, respectively) and for Lxc-overexpressing cells at ;26.58% (black; cell sample sizes for MOI of 1 to 5 are 295, 132, 55, 38, and 15, respectively). For cells with more C1 operator sequences (red; cell sample sizes forMOI of 1 to 5 are 147, 111, 44, 14, and 9, respectively), the probability of lysogeny increases with MOIat lower MOI and returns to the normal level at MOI of .3. Error bars denote counting errors. (C) The injected DNA copy number correlates with the prediction very well (see also Fig. S2C). Cell sample sizes for MOI of 1 to 5 are 83, 33, 17, 14, and 5, respectively. Filled squares, experimental data; redline, linear regression ﬁt; black dashed line, diagonal indicating the perfect positive correlation. Error bars denote standard errors of the means.MOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 3', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  was followed through a reporter plasmid expressing mTurquoise2 from the PL23 pro- moter, which is dependent on the late gene activator, Lpa; the indicator for the lyso- genic pathway was the expression of C1-mVenus fusion proteins followed by cell divi-sion (Fig. S1). Using this reporter system, we examined the effect of MOI on the cell fate. In agreement with previous bulk experiments using P1-carried chloramphenicol resistance as a readout (47), our single-cell measurements demonstrated that the prob-ability of lysogeny is constant over different MOI (Fig. 1B). Considering that some proteins encoded by P1, e.g., the superimmunity protein Sim, may act to prevent superinfecting phages from injecting their DNAs into the cyto-plasm of infected cells (56, 57), we ﬁrst tested if all the phages attached on the cell sur- face were able to inject their DNAs. To examine the number of injected DNAs, we uti- lized our established DNA visualization system by taking advantage of the speci ﬁc binding of SeqA proteins to the methylated GATC sites (55, 58, 59). In particular, duringphage infection, SeqA-mKate2 fusion proteins constitutively expressed in the methyla- tion-de ﬁcient host cells (LZ1387, a Ddam variant) bind to the methylated P1 DNAs but not the host DNA, allowing us to track the viral DNAs as ﬂuorescent spots. Then, the injected-DNA copy number of each infected cell was calculated based on the total ﬂuo- rescent intensity of all DNA spots in the cell at 0 min divided by the single-DNA inten- sity (see details in Materials and Methods). Meanwhile, as each phage particle was ﬂuo- rescently labeled by mTurquoise2, we could count the phages that attached on thecell surface and predict the injected-DNA number based on failed and dark infection frequencies (see details in Materials and Methods). From our data, we observed a nearly perfect positive correlation between the actual injected-DNA number and thepredicted DNA copy number (Fig. 1C and Fig. S2C). This indicated the success of multi- ple infection and the independence of the DNA injection process from different phages adsorbing to the same cell. A simple genetic model to elucidate MOI-independent decision-making. To under- stand the underlying mechanism of this MOI-independent cell lysogeny, we sought aminimal genetic model based upon the regulatory network of P1 (Fig. 1A). Given thecentral role of the major repressor, we hypothesized that C1 activity is a cell fate deter- mining factor. Based on P1 regulatory circuitry, C1 activity can be affected by different parameters, e.g., the expression level of C1 compared with its inhibitor Coi, the level ofthe corepressor Lxc, and the number of intracellular C1 binding sites (22 sites per P1 DNA). We ﬁrst compared C1 expression level in lytic and lysogenic cells by tracking C1- mVenus intensity using the decision-making reporter system (55). As expected, cellsentering the lysogenic pathway showed higher C1 signals on average than those entering the lytic pathway, though it was not always true for every single-cell trajectory (Fig. 2A). Next, we tested whether the decision-making outcome could be manipulatedby varying the intracellular C1 activity. To enhance C1 activity, we overexpressed thecorepressor Lxc from plasmids in host cells. Indeed, host cells with higher Lxc FIG 2 Inﬂuence of C1 activity on the probability of P1 lysogeny. (A) The average expression level of C1 (mVenus intensity) in lysogenic cells is higher than the level in lytic cells. Bold lines, meanintensities; light lines, C1 expression trajectories of each individual cell. Blue, lysogenic cells ( n= 47); red, lytic cells ( n= 165). (B) Probability of lysogeny upon the infection of E. coli cells with different expression levels of Lxc at the bulk level (MOI = 1). Lxc was induced with different concentrations ofIPTG from the PLacO promoter on a plasmid. The error bars denote counting errors.Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 4', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  expression levels exhibit higher probabilities of lysogeny at both the bulk level (Fig. 2B) and the single-cell level (Fig. 1B). Interestingly, the probability of lysogeny was still independent of MOI (Fig. 1B). On the other hand, to reduce the C1 activity, we introduced more C1 operator DNA sequences from plasmids ( ;15 sites per cell) and found a reduction of the lysogeny probability at low MOI (1 and 2) (Fig. 1B), although not at higher MOI, where changes of the number of C1 binding sites were relatively low and hence showed less effect on C1 function. In summary, these data suggest that C1 activity is a determining factor of P1 decision-making. We then hypothesized that the C1 activity of each infecting phage was constant in cells with different MOI, which led to the MOI-independent lysogenization. Initially, we deﬁned the total C1 activity in an infected cell as the amount of C1 that is not bound by Coi (or free C1) based on their antagonistic functions and assumed that the amount of free C1 per infecting phage was a constant. To test this, we compared mRNA levels ofc1and coisimultaneously during P1 infection at different bulk MOI (0.2, 1, and 5), measured by single-molecule fluorescence in situ hybridization (smFISH) (60). DNA FISH and quantitative PCR (qPCR) data (Fig. S3) suggested that the viral DNA replica- tion began around 30 min after infection, so we focused on mRNA levels at 20 min and30 min. Fluorescent probes were designed to target c1orcoimRNA so that we could quantify them separately (Fig. 3A and B). According to the Poisson distribution (55), at bulk MOI of 0.2, 1, and 5, the estimated proportions of infected cells out of total cells were about 18%, 63%, and 99%, respectively. From the smFISH experiments, the per- centage of infected cells (cells showing c1orcoimRNA signal) were consistent with our predictions (Fig. S4). However, the difference between c1and coimRNA per phage, i.e., free c1mRNA per phage, decreased with MOI instead of remaining constant (Fig. 3C), where the mRNA levels of both c1and coiper phage decreased with MOI (Fig. S5). This suggested that, in order to achieve MOI-independent lysogenization,there would be other components that work together to diminish the effect of MOI. Based on the aforementioned regulated function of Lxc, we decided to de ﬁne C1 activity considering the expression level of both free C1 and Lxc. Since none of the FIG 3 The interaction of C1, Coi, and Lxc leads to the constant C1 activity of each phage despite different MOI. (A) Schematic of the coi-c1 operon and the smFISH method. C1 is expressed from two promoters (P1 coi and P c1), while Coi is expressed only from P1 coi. C1 is able to inhibit both P1 coiand P c1by binding to the adjacent operator sequences. Coi forms a 1:1 complex with C1 to inactivate C1 function. Lxc promotes C1 binding af ﬁnity to its operators and inhibits the dissociated function of Coi. (B) Representative images showing mRNA expression at 30 min after infection at bulk MOI of 0.2 (top) and 5 (bottom). Cyan, c1; red, coi; green, lxc. Bar = 2 mm. (C) Free c1mRNA per phage decreases with MOI. See also Fig. S5. (D) The level of lxcmRNA increases with MOI. (E) The C1 activity of each infecting phage, i.e., free c1mRNA times lxcmRNA per phage, is similar at different MOI. (F) In Lxc-overexpressed host cells, the lysis-lysogeny decision is determined by the level of C1 per phage.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Lxc promotes C1 binding af ﬁnity to its operators and inhibits the dissociated function of Coi. (B) Representative images showing mRNA expression at 30 min after infection at bulk MOI of 0.2 (top) and 5 (bottom). Cyan, c1; red, coi; green, lxc. Bar = 2 mm. (C) Free c1mRNA per phage decreases with MOI. See also Fig. S5. (D) The level of lxcmRNA increases with MOI. (E) The C1 activity of each infecting phage, i.e., free c1mRNA times lxcmRNA per phage, is similar at different MOI. (F) In Lxc-overexpressed host cells, the lysis-lysogeny decision is determined by the level of C1 per phage. smFISH experiments show that the level of c1mRNA per phage is similar at different MOI. In all plots, error bars denote standard errors of the means.MOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 5', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  promoters that drive the transcription of gene lxcis under the control of C1 or other factors, we hypothesized that Lxc is constitutively expressed upon infection and the Lxc level is positively correlated with MOI. As shown in Fig. 3D, lxcmRNA number increased with MOI from smFISH experiments. Together with the observation that freec1mRNA per phage decreased with MOI (Fig. 3C), our simple model became one in which the C1 activity of each infecting phage, de ﬁned as the expression level of free C1 times the expression level of Lxc per phage, remained constant over MOI. To test this, we quanti ﬁed the mRNA expression of c1,coi, and lxcin the same cell during phage infection using smFISH. Indeed, our results showed that the C1 activities per phage were similar at different bulk MOI (Fig. 3E). These data support our model for MOI-independent lysogeny and indicated the importance of Lxc for maintaining a setfrequency of lysogeny during P1 infection. Next, we assumed that the proportional expression level of Lxc over MOI was the key to maintaining the probability of cell lysogeny. To demonstrate this, we overex-pressed Lxc; hence, its level became a constant in all infection events, which shouldhave resulted in MOI-dependent lysogeny. However, based on the data from our sin- gle-cell infection assay, probabilities of P1 lysogeny during the infection of the Lxc- overexpressed hosts were still independent of MOI (Fig. 2B). The mechanism behind itwould be that, with this saturated Lxc function on C1 activity, the function of Coi in inactivating C1 was abolished. In this scenario, cell fate should be simply determined by the level of C1 per phage. From the smFISH experiment upon the infection of Lxc-overexpressing hosts, we indeed observed a constant level of c1mRNA per phage at different MOI (Fig. 3F), which led to the MOI independency. In summary, C1 activity regulating each phage depends on the expression level and the interaction betweenthe gene products of c1,coi, and lxcand is a constant during infection with different numbers of P1 virions, which results in an MOI-independent lysogenic response. Imposing MOI dependency by increasing the distance between infecting phages. Recent studies showed that phage lexhibits individual lysis-lysogeny decision-making by each infecting phage, as evidenced by phage DNA integration into the host chro- mosome (lysogeny) followed by cell lysis as well as the coexpression of capsid decora-tion proteins (gpD) from different infecting phages, indicative of lytic pathways, coex- pression of CI from different infecting phages, indicative of lysogenic pathways, and coexpression of both gpD and CI, indicative of both lysis and lysogeny in single cells(59, 61, 62). The decisions by all lphages infecting a single cell are then integrated into the ﬁnal cell fate (41, 59). In the meantime, the probability of lysogeny of phage lincreases with MOI (39 –41). We speculate that the MOI-dependent lysogenization oflis correlated with the individual phage lysis-lysogeny decision. Here, the MOI independency of P1 lysogenization may then imply that the infecting phages make an ensemble decision. To explore this, we examined P1 DNA interactions inside the cell using a tetO/ TetR-FP DNA visualization system (Fig. S2A and B) (61, 63). Basically, we replaced darB (23, 64), a gene nonessential for P1 decision-making, with a 120 /C2tetO array in the phage DNA. Upon infection of the host cell harboring a plasmid constitu- tively expressing TetR-mNeonGreen fusion proteins, all injected and replicated phageDNAs are bound and labeled at tetO sites by ﬂuorescent TetR-mNeonGreen proteins, forming foci or clusters (Fig. 4A and B). During the infection process, some interesting DNA behaviors were detected (Fig.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  To explore this, we examined P1 DNA interactions inside the cell using a tetO/ TetR-FP DNA visualization system (Fig. S2A and B) (61, 63). Basically, we replaced darB (23, 64), a gene nonessential for P1 decision-making, with a 120 /C2tetO array in the phage DNA. Upon infection of the host cell harboring a plasmid constitu- tively expressing TetR-mNeonGreen fusion proteins, all injected and replicated phageDNAs are bound and labeled at tetO sites by ﬂuorescent TetR-mNeonGreen proteins, forming foci or clusters (Fig. 4A and B). During the infection process, some interesting DNA behaviors were detected (Fig. 4A): (i) one of the phage DNAs replicated earlier or faster than the other, resulting in an unsynchronized pattern; (ii) phage DNAs physi-cally moved together; and (iii) viral DNAs located at different cell areas showed synchronized replication patterns. Asymmetric levels of DNA replication might be due to the uneven distribution of intracellular machineries, suggesting that phages madedecisions individually, de ﬁned as individual DNA behavior. In contrast, DNAs that move together or show synchronized replication patterns would probably indicate that the infecting phages shared regulatory products to regulate each other and made deci-sions as one, i.e., nonindividual behavior. A majority of the cells ( ;70%) exhibited the nonindividual behavior (Fig. 4C and Fig. S6A).Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 6', \"page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The nonindividual behavior could be due to phages being physically close to each other and their gene products for lysis-lysogeny decision being easily shared among them. Therefore, a larger distance between infecting phages would inhibit the ef ﬁcient trans regulation and thereby lead to individual decision-making. To test whether this is the case, we increased the distances of phages by infecting a much longer cell (Fig. 5A and Fig. S7A and B). Increased cell length was achieved by expressing the Kil protein ofphage lfrom a plasmid; Kil inhibits E. coli cell division, causing cells to grow into long ﬁlaments (65, 66). For simplicity, we analyzed the infected cells only at an MOI of 2. Using the tetO/ TetR-mNeonGreen DNA visualization system, we observed more unsynchronized DNA replications during the infection of long ﬁlamentous cells (Fig. 4B), which gave rise to a higher level of the individuality of coinfecting phages (Fig. 4C and D; Fig. S6C and D). Next, we tested P1 lysogenic response in long cells. In both bulk and single-cell experiments, the probability of cells undergoing lysogenyincreased with MOI instead of being a constant value (Fig. 5B and C; Fig. S7C and D). In summary, these data indicate that infecting P1 phages are more likely to make the en- FIG 5 MOI-dependent lysogenization was increased by increasing the distance between coinfecting phages. (A) Representative images of normal-sized cells and long cells. Bar = 2 mm. (B) The probability of lysogeny increases with MOI for long cells (red) and remains constant for normal-sizedcells (blue), tested at the bulk level (see also Fig. S7). Red circles, lKil was induced using 0.05% arabinose from plasmid pBAD to generate long cells. Blue circles, no arabinose treatment. (C) Comparison of the probability of lysogeny increasing with MOI upon infection of normal-sized cells and long cells at the single-cell level. Orange, long cells (cell sample sizes for MOI of 1 to 8 are 22,20, 20, 32, 21, 19, 19 and 15, respectively); blue, normal-sized cells (see also Fig. 1B). Error barsdenote counting errors. FIG 4 P1 virions infecting the same host cell make an ensemble lysis-lysogeny decision. (A) Representative images showing phage DNA behaviors in normal-sized E. coli cells at an MOI of 2 (green signals represent TetR-mNeonGreen-bound phage DNAs). (Left) One of the phage DNAs replicated earlieror faster than the other, resulting in an unsynchronized pattern. (Middle) Phage DNAs physically moved together. (Right) Phage DNAs located at different cell areas showed synchronized replication patterns. The MOI was determined by the number of mTurquoise2 ﬂuorescent phages (red dots indicated by red arrows) attached on the cell surface at 0 min. (B) Representative images showing unsynchronized phageDNA behaviors in long cells with lKil expression at an MOI of 2. Bar = 2 mm. (C) Bar plot showing more nonindividual DNA behaviors in normal-sized cells ( n= 82) versus more individual phage DNA behaviors in long cells ( n= 79). The error bars denote counting errors. (D) Box plot of the distance between coinfecting phages on the surface of long cells (red, n= 121) is much larger than that in normal-sized cells (blue, n= 113) at an MOI of 2. *,P,0.001 as determined by Student's ttest. See also Fig. S6.MOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 7\"], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Bacteriophages and phage-like plasmids play a prominent role in the dissemination of adaptive traits that allow stable colonization and persistence of multidrug resist- ance in pathogenic bacteria (1 –12). A better understanding of their contribution to pathotype spread and adaptation is important. Phage P1 was discovered in a lysogenic strain of Escherichia coli named Li about 70 years ago (13). In terms of the ability to inject its DNA, P1 has the broadest known host range, including a variety of Gram-neg- ative enteric bacteria, e.g., E. coli ,Shigella dysenteriae (13), and Salmonella enterica sero- var Typhimurium (14 –17), and soil bacteria, including Pseudomonas aeruginosa (17, 18), Acetobacter suboxydans (17), and Myxococcus xanthus (19, 20). In the E. coli host, the infection cycle of P1 begins with adsorption to the cell surface by recognizing the ter- minal glucose moiety of the lipopolysaccharide (LPS) core region (21). The interaction of at least three of the six tail ﬁbers with receptor molecules is assumed to be suf ﬁcient to stimulate the injection process (22). Injected viral DNAs can be protected from host type I restriction-modi ﬁcation (RM) system by P1 antirestriction proteins (23 –25). Following DNA injection, the infected cell enters either the lytic cycle, where progeny phages are produced and the cell lyses, or the lysogenic cycle, in which the phage DNA exists as a low-copy-number plasmid and replicates autonomously. At the molec- ular level, the lysis-lysogeny decision is controlled by the interactions between phage components of the regulatory network (Fig. 1A). Brie ﬂy, the major repressor of P1, C1, is expressed immediately after phage infection and binds to 22 operator sequences widely dispersed over the P1 DNA to repress lytic gene expression. For example, C1controls the expression of RepL, which is responsible for vegetative DNA replication, and Lpa, which activates the late promoter serving viral morphogenesis and lysis genes (26–28). Coi ( c one inactivation) is an antirepressor that binds to C1 and inactivates its function (29 –31), whereas the corepressor Lxc (lowers expression of c1) increases C1 binding af ﬁnity by forming a ternary complex with C1 and operator DNA and inhibits the ability of Coi to dissociate the operator-C1 complex (32 –34). Moreover, the hetero- dimeric complex of Ant1 and Ant2 proteins also functions as an antirepressor, while thetrans -acting c4RNA represses the synthesis of Ant proteins (Fig. 1A) (for a detailed review, see reference 35). Some studies suggest that the initial choice between lysis and lysogeny relies on the relative synthesis rate of the main competing players, C1 and Coi (28), but no quantita- tive evidence has been available to determine the relationship between this competition and the ﬁnal lysis-lysogeny decision. In addition, the bene ﬁts of P1 acquiring different regulatory components besides the main competing players are not clear. Regrettably, studies of P1 regulatory circuitry halted prematurely with the death of their prime mover, Heinz Schuster (30, 31, 33 –38). Furthermore, interest in P1 decreased, since it does not contain cargo genes of clinical interest (e.g., stress resistance and nutrient acquisition), and, being a temperate phage, P1 has been not considered for antibacterial therapeutic applications (phage therapy). However, recent advances in sequencing technologies have allowed the identi ﬁcation of numerous P1-like plasmids in bacteria and human pathogens (2, 4 –6, 9, 11, 12), suggesting their abilities to serve as speci ﬁc vehicles for the spread and the maintenance of virulence and antimicrobial resistance in both clinical and livestock production settings (1, 9).', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Regrettably, studies of P1 regulatory circuitry halted prematurely with the death of their prime mover, Heinz Schuster (30, 31, 33 –38). Furthermore, interest in P1 decreased, since it does not contain cargo genes of clinical interest (e.g., stress resistance and nutrient acquisition), and, being a temperate phage, P1 has been not considered for antibacterial therapeutic applications (phage therapy). However, recent advances in sequencing technologies have allowed the identi ﬁcation of numerous P1-like plasmids in bacteria and human pathogens (2, 4 –6, 9, 11, 12), suggesting their abilities to serve as speci ﬁc vehicles for the spread and the maintenance of virulence and antimicrobial resistance in both clinical and livestock production settings (1, 9). For example, pTZ20_1P, isolated from a porcine commensal E. coli strain and carrying multiple antibiotic resistance markers, was able to lysogenize a commensal E. coli strain with consequent transfer of resistance (9). Moreover, the sequencing data showed that genes contributing to lysogeny establish- ment were very similar to the genes of the regulatory network of P1 (9). Therefore, athorough study of P1 life cycle and mechanism of its lysogeny establishment is impor- tant to understand the phage-driven antimicrobial resistance emergence and pathogen adaptation, which will bene ﬁt the development of novel therapeutic methods. Studies of other temperate phages revealed that the lysis-lysogeny decision is highly dependent on the host availability in the environment or the virus-cell ratio. For phage l, one of the best-studied paradigms for cell-fate decision-making, the probabil- ity of lysogeny increases with multiplicity of infection (MOI; the number of phagesZhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 2', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  infecting a cell) and approaches 100% when the MOI is suf ﬁciently high (39 –41). Upon infection of the Vibrio phage 882, a quorum sensing factor expressed from the host cell is utilized to make the lysis-lysogeny decision (42, 43). Moreover, during infection,some Bacillus phages produce a 6-amino-acid peptide, designated arbitrium, that accu- mulates in the medium. In turn, high levels of arbitrium cause subsequent phage infec-tions to be biased toward lysogeny (44, 45). These and other strategies are used by the phages to determine whether there are enough host cells for the infection by progeny phages, ensuring a high ef ﬁciency of genome propagation. In contrast, phage P1 has been reported to lack sensitivity to the virus-cell ratio. Previous studies by Bertani andNice (46) and Rosner (47) at the bulk level showed constant probabilities of lysogenyregardless of MOI: 80 to 90% of S. dysenteriae cells were lysogenized at 20°C when infected by the P1 strain isolated originally from the lysogenic E. coli strain Li (13); ;30% of the E. coli cells were lysogenized at 30°C by the thermoinducible P1 strain (P1CM c1-100) (47). The constant probability of lysogeny irrespective of host availability, together with P1 ’s capacity for autonomous prophage maintenance, utilizing systems such as type III RM (48 –50), plasmid replication (51, 52) and toxin-antitoxin (53, 54), is a strategy used by P1 to propagate its genomic information as a plasmid. Furthermore,the ability to establish and be maintained as a plasmid can also be bene ﬁcial to the dis- semination of the cargo genes carried by P1 or P1 phage-like elements. In this work, we set out to examine the infection process of phage P1 at the levels of individual phage and cell with a spatiotemporal resolution suf ﬁcient to quantify the relevant subcellular parameters and to evaluate the contribution of each parameter to the observed cell-fate decisions. Using our system, we ﬁrst investigated the mechanism of P1 MOI-independent lysogenization by linking the regulatory gene expression withcell fate by a simple genetic model. We also found that MOI dependency can beimposed by increasing the distances between the coinfecting phages. Taken together,our results provide a new decision-making model that expands our understanding oftemperate phages. This, in turn, gives us insight into how the ﬁtness of different phages results in distinctive decision-making behaviors and also allows us to evaluatetheir contributions to pathogenicity development. RESULTS The probability of lysogeny is independent of the number of infecting phages at the single-cell level. In order to study P1 lysogenization at the single-cell level, we constructed a ﬂuorescent reporter system to monitor individual decision-making events under the microscope, as described previously (55). Brie ﬂy, to record the num- ber of phages infecting each E. coli cell, P1 virions were labeled by fusing ﬂuorescent proteins to phage capsids. Moreover, during the time-lapse movies, the lytic pathway FIG 1 The probability of lysogeny is independent of the number of infecting phages at the single- cell level. (A) Schematic of the P1 lysis-lysogeny regulatory network. (B) Probability of lysogeny as afunction of MOI. The probability of lysogeny remains constant over different MOI for wild-type (WT)cells at ;14.53% (blue; cell sample sizes for MOI of 1 to 5 are 457, 277, 168, 62, and 68, respectively) and for Lxc-overexpressing cells at ;26.58% (black; cell sample sizes for MOI of 1 to 5 are 295, 132, 55, 38, and 15, respectively).', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Moreover, during the time-lapse movies, the lytic pathway FIG 1 The probability of lysogeny is independent of the number of infecting phages at the single- cell level. (A) Schematic of the P1 lysis-lysogeny regulatory network. (B) Probability of lysogeny as afunction of MOI. The probability of lysogeny remains constant over different MOI for wild-type (WT)cells at ;14.53% (blue; cell sample sizes for MOI of 1 to 5 are 457, 277, 168, 62, and 68, respectively) and for Lxc-overexpressing cells at ;26.58% (black; cell sample sizes for MOI of 1 to 5 are 295, 132, 55, 38, and 15, respectively). For cells with more C1 operator sequences (red; cell sample sizes forMOI of 1 to 5 are 147, 111, 44, 14, and 9, respectively), the probability of lysogeny increases with MOIat lower MOI and returns to the normal level at MOI of .3. Error bars denote counting errors. (C) The injected DNA copy number correlates with the prediction very well (see also Fig. S2C). Cell sample sizes for MOI of 1 to 5 are 83, 33, 17, 14, and 5, respectively. Filled squares, experimental data; redline, linear regression ﬁt; black dashed line, diagonal indicating the perfect positive correlation. Error bars denote standard errors of the means.MOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 3', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  was followed through a reporter plasmid expressing mTurquoise2 from the PL23 pro- moter, which is dependent on the late gene activator, Lpa; the indicator for the lyso- genic pathway was the expression of C1-mVenus fusion proteins followed by cell divi-sion (Fig. S1). Using this reporter system, we examined the effect of MOI on the cell fate. In agreement with previous bulk experiments using P1-carried chloramphenicol resistance as a readout (47), our single-cell measurements demonstrated that the prob-ability of lysogeny is constant over different MOI (Fig. 1B). Considering that some proteins encoded by P1, e.g., the superimmunity protein Sim, may act to prevent superinfecting phages from injecting their DNAs into the cyto-plasm of infected cells (56, 57), we ﬁrst tested if all the phages attached on the cell sur- face were able to inject their DNAs. To examine the number of injected DNAs, we uti- lized our established DNA visualization system by taking advantage of the speci ﬁc binding of SeqA proteins to the methylated GATC sites (55, 58, 59). In particular, duringphage infection, SeqA-mKate2 fusion proteins constitutively expressed in the methyla- tion-de ﬁcient host cells (LZ1387, a Ddam variant) bind to the methylated P1 DNAs but not the host DNA, allowing us to track the viral DNAs as ﬂuorescent spots. Then, the injected-DNA copy number of each infected cell was calculated based on the total ﬂuo- rescent intensity of all DNA spots in the cell at 0 min divided by the single-DNA inten- sity (see details in Materials and Methods). Meanwhile, as each phage particle was ﬂuo- rescently labeled by mTurquoise2, we could count the phages that attached on thecell surface and predict the injected-DNA number based on failed and dark infection frequencies (see details in Materials and Methods). From our data, we observed a nearly perfect positive correlation between the actual injected-DNA number and thepredicted DNA copy number (Fig. 1C and Fig. S2C). This indicated the success of multi- ple infection and the independence of the DNA injection process from different phages adsorbing to the same cell. A simple genetic model to elucidate MOI-independent decision-making. To under- stand the underlying mechanism of this MOI-independent cell lysogeny, we sought aminimal genetic model based upon the regulatory network of P1 (Fig. 1A). Given thecentral role of the major repressor, we hypothesized that C1 activity is a cell fate deter- mining factor. Based on P1 regulatory circuitry, C1 activity can be affected by different parameters, e.g., the expression level of C1 compared with its inhibitor Coi, the level ofthe corepressor Lxc, and the number of intracellular C1 binding sites (22 sites per P1 DNA). We ﬁrst compared C1 expression level in lytic and lysogenic cells by tracking C1- mVenus intensity using the decision-making reporter system (55). As expected, cellsentering the lysogenic pathway showed higher C1 signals on average than those entering the lytic pathway, though it was not always true for every single-cell trajectory (Fig. 2A). Next, we tested whether the decision-making outcome could be manipulatedby varying the intracellular C1 activity. To enhance C1 activity, we overexpressed thecorepressor Lxc from plasmids in host cells. Indeed, host cells with higher Lxc FIG 2 Inﬂuence of C1 activity on the probability of P1 lysogeny. (A) The average expression level of C1 (mVenus intensity) in lysogenic cells is higher than the level in lytic cells. Bold lines, meanintensities; light lines, C1 expression trajectories of each individual cell. Blue, lysogenic cells ( n= 47); red, lytic cells ( n= 165). (B) Probability of lysogeny upon the infection of E. coli cells with different expression levels of Lxc at the bulk level (MOI = 1). Lxc was induced with different concentrations ofIPTG from the PLacO promoter on a plasmid. The error bars denote counting errors.Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 4', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  expression levels exhibit higher probabilities of lysogeny at both the bulk level (Fig. 2B) and the single-cell level (Fig. 1B). Interestingly, the probability of lysogeny was still independent of MOI (Fig. 1B). On the other hand, to reduce the C1 activity, we introduced more C1 operator DNA sequences from plasmids ( ;15 sites per cell) and found a reduction of the lysogeny probability at low MOI (1 and 2) (Fig. 1B), although not at higher MOI, where changes of the number of C1 binding sites were relatively low and hence showed less effect on C1 function. In summary, these data suggest that C1 activity is a determining factor of P1 decision-making. We then hypothesized that the C1 activity of each infecting phage was constant in cells with different MOI, which led to the MOI-independent lysogenization. Initially, we deﬁned the total C1 activity in an infected cell as the amount of C1 that is not bound by Coi (or free C1) based on their antagonistic functions and assumed that the amount of free C1 per infecting phage was a constant. To test this, we compared mRNA levels ofc1and coisimultaneously during P1 infection at different bulk MOI (0.2, 1, and 5), measured by single-molecule fluorescence in situ hybridization (smFISH) (60). DNA FISH and quantitative PCR (qPCR) data (Fig. S3) suggested that the viral DNA replica- tion began around 30 min after infection, so we focused on mRNA levels at 20 min and30 min. Fluorescent probes were designed to target c1orcoimRNA so that we could quantify them separately (Fig. 3A and B). According to the Poisson distribution (55), at bulk MOI of 0.2, 1, and 5, the estimated proportions of infected cells out of total cells were about 18%, 63%, and 99%, respectively. From the smFISH experiments, the per- centage of infected cells (cells showing c1orcoimRNA signal) were consistent with our predictions (Fig. S4). However, the difference between c1and coimRNA per phage, i.e., free c1mRNA per phage, decreased with MOI instead of remaining constant (Fig. 3C), where the mRNA levels of both c1and coiper phage decreased with MOI (Fig. S5). This suggested that, in order to achieve MOI-independent lysogenization,there would be other components that work together to diminish the effect of MOI. Based on the aforementioned regulated function of Lxc, we decided to de ﬁne C1 activity considering the expression level of both free C1 and Lxc. Since none of the FIG 3 The interaction of C1, Coi, and Lxc leads to the constant C1 activity of each phage despite different MOI. (A) Schematic of the coi-c1 operon and the smFISH method. C1 is expressed from two promoters (P1 coi and P c1), while Coi is expressed only from P1 coi. C1 is able to inhibit both P1 coiand P c1by binding to the adjacent operator sequences. Coi forms a 1:1 complex with C1 to inactivate C1 function. Lxc promotes C1 binding af ﬁnity to its operators and inhibits the dissociated function of Coi. (B) Representative images showing mRNA expression at 30 min after infection at bulk MOI of 0.2 (top) and 5 (bottom). Cyan, c1; red, coi; green, lxc. Bar = 2 mm. (C) Free c1mRNA per phage decreases with MOI. See also Fig. S5. (D) The level of lxcmRNA increases with MOI. (E) The C1 activity of each infecting phage, i.e., free c1mRNA times lxcmRNA per phage, is similar at different MOI. (F) In Lxc-overexpressed host cells, the lysis-lysogeny decision is determined by the level of C1 per phage.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Lxc promotes C1 binding af ﬁnity to its operators and inhibits the dissociated function of Coi. (B) Representative images showing mRNA expression at 30 min after infection at bulk MOI of 0.2 (top) and 5 (bottom). Cyan, c1; red, coi; green, lxc. Bar = 2 mm. (C) Free c1mRNA per phage decreases with MOI. See also Fig. S5. (D) The level of lxcmRNA increases with MOI. (E) The C1 activity of each infecting phage, i.e., free c1mRNA times lxcmRNA per phage, is similar at different MOI. (F) In Lxc-overexpressed host cells, the lysis-lysogeny decision is determined by the level of C1 per phage. smFISH experiments show that the level of c1mRNA per phage is similar at different MOI. In all plots, error bars denote standard errors of the means.MOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 5', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  promoters that drive the transcription of gene lxcis under the control of C1 or other factors, we hypothesized that Lxc is constitutively expressed upon infection and the Lxc level is positively correlated with MOI. As shown in Fig. 3D, lxcmRNA number increased with MOI from smFISH experiments. Together with the observation that freec1mRNA per phage decreased with MOI (Fig. 3C), our simple model became one in which the C1 activity of each infecting phage, de ﬁned as the expression level of free C1 times the expression level of Lxc per phage, remained constant over MOI. To test this, we quanti ﬁed the mRNA expression of c1,coi, and lxcin the same cell during phage infection using smFISH. Indeed, our results showed that the C1 activities per phage were similar at different bulk MOI (Fig. 3E). These data support our model for MOI-independent lysogeny and indicated the importance of Lxc for maintaining a setfrequency of lysogeny during P1 infection. Next, we assumed that the proportional expression level of Lxc over MOI was the key to maintaining the probability of cell lysogeny. To demonstrate this, we overex-pressed Lxc; hence, its level became a constant in all infection events, which shouldhave resulted in MOI-dependent lysogeny. However, based on the data from our sin- gle-cell infection assay, probabilities of P1 lysogeny during the infection of the Lxc- overexpressed hosts were still independent of MOI (Fig. 2B). The mechanism behind itwould be that, with this saturated Lxc function on C1 activity, the function of Coi in inactivating C1 was abolished. In this scenario, cell fate should be simply determined by the level of C1 per phage. From the smFISH experiment upon the infection of Lxc-overexpressing hosts, we indeed observed a constant level of c1mRNA per phage at different MOI (Fig. 3F), which led to the MOI independency. In summary, C1 activity regulating each phage depends on the expression level and the interaction betweenthe gene products of c1,coi, and lxcand is a constant during infection with different numbers of P1 virions, which results in an MOI-independent lysogenic response. Imposing MOI dependency by increasing the distance between infecting phages. Recent studies showed that phage lexhibits individual lysis-lysogeny decision-making by each infecting phage, as evidenced by phage DNA integration into the host chro- mosome (lysogeny) followed by cell lysis as well as the coexpression of capsid decora-tion proteins (gpD) from different infecting phages, indicative of lytic pathways, coex- pression of CI from different infecting phages, indicative of lysogenic pathways, and coexpression of both gpD and CI, indicative of both lysis and lysogeny in single cells(59, 61, 62). The decisions by all lphages infecting a single cell are then integrated into the ﬁnal cell fate (41, 59). In the meantime, the probability of lysogeny of phage lincreases with MOI (39 –41). We speculate that the MOI-dependent lysogenization oflis correlated with the individual phage lysis-lysogeny decision. Here, the MOI independency of P1 lysogenization may then imply that the infecting phages make an ensemble decision. To explore this, we examined P1 DNA interactions inside the cell using a tetO/ TetR-FP DNA visualization system (Fig. S2A and B) (61, 63). Basically, we replaced darB (23, 64), a gene nonessential for P1 decision-making, with a 120 /C2tetO array in the phage DNA. Upon infection of the host cell harboring a plasmid constitu- tively expressing TetR-mNeonGreen fusion proteins, all injected and replicated phageDNAs are bound and labeled at tetO sites by ﬂuorescent TetR-mNeonGreen proteins, forming foci or clusters (Fig. 4A and B). During the infection process, some interesting DNA behaviors were detected (Fig.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  To explore this, we examined P1 DNA interactions inside the cell using a tetO/ TetR-FP DNA visualization system (Fig. S2A and B) (61, 63). Basically, we replaced darB (23, 64), a gene nonessential for P1 decision-making, with a 120 /C2tetO array in the phage DNA. Upon infection of the host cell harboring a plasmid constitu- tively expressing TetR-mNeonGreen fusion proteins, all injected and replicated phageDNAs are bound and labeled at tetO sites by ﬂuorescent TetR-mNeonGreen proteins, forming foci or clusters (Fig. 4A and B). During the infection process, some interesting DNA behaviors were detected (Fig. 4A): (i) one of the phage DNAs replicated earlier or faster than the other, resulting in an unsynchronized pattern; (ii) phage DNAs physi-cally moved together; and (iii) viral DNAs located at different cell areas showed synchronized replication patterns. Asymmetric levels of DNA replication might be due to the uneven distribution of intracellular machineries, suggesting that phages madedecisions individually, de ﬁned as individual DNA behavior. In contrast, DNAs that move together or show synchronized replication patterns would probably indicate that the infecting phages shared regulatory products to regulate each other and made deci-sions as one, i.e., nonindividual behavior. A majority of the cells ( ;70%) exhibited the nonindividual behavior (Fig. 4C and Fig. S6A).Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 6', \"page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The nonindividual behavior could be due to phages being physically close to each other and their gene products for lysis-lysogeny decision being easily shared among them. Therefore, a larger distance between infecting phages would inhibit the ef ﬁcient trans regulation and thereby lead to individual decision-making. To test whether this is the case, we increased the distances of phages by infecting a much longer cell (Fig. 5A and Fig. S7A and B). Increased cell length was achieved by expressing the Kil protein ofphage lfrom a plasmid; Kil inhibits E. coli cell division, causing cells to grow into long ﬁlaments (65, 66). For simplicity, we analyzed the infected cells only at an MOI of 2. Using the tetO/ TetR-mNeonGreen DNA visualization system, we observed more unsynchronized DNA replications during the infection of long ﬁlamentous cells (Fig. 4B), which gave rise to a higher level of the individuality of coinfecting phages (Fig. 4C and D; Fig. S6C and D). Next, we tested P1 lysogenic response in long cells. In both bulk and single-cell experiments, the probability of cells undergoing lysogenyincreased with MOI instead of being a constant value (Fig. 5B and C; Fig. S7C and D). In summary, these data indicate that infecting P1 phages are more likely to make the en- FIG 5 MOI-dependent lysogenization was increased by increasing the distance between coinfecting phages. (A) Representative images of normal-sized cells and long cells. Bar = 2 mm. (B) The probability of lysogeny increases with MOI for long cells (red) and remains constant for normal-sizedcells (blue), tested at the bulk level (see also Fig. S7). Red circles, lKil was induced using 0.05% arabinose from plasmid pBAD to generate long cells. Blue circles, no arabinose treatment. (C) Comparison of the probability of lysogeny increasing with MOI upon infection of normal-sized cells and long cells at the single-cell level. Orange, long cells (cell sample sizes for MOI of 1 to 8 are 22,20, 20, 32, 21, 19, 19 and 15, respectively); blue, normal-sized cells (see also Fig. 1B). Error barsdenote counting errors. FIG 4 P1 virions infecting the same host cell make an ensemble lysis-lysogeny decision. (A) Representative images showing phage DNA behaviors in normal-sized E. coli cells at an MOI of 2 (green signals represent TetR-mNeonGreen-bound phage DNAs). (Left) One of the phage DNAs replicated earlieror faster than the other, resulting in an unsynchronized pattern. (Middle) Phage DNAs physically moved together. (Right) Phage DNAs located at different cell areas showed synchronized replication patterns. The MOI was determined by the number of mTurquoise2 ﬂuorescent phages (red dots indicated by red arrows) attached on the cell surface at 0 min. (B) Representative images showing unsynchronized phageDNA behaviors in long cells with lKil expression at an MOI of 2. Bar = 2 mm. (C) Bar plot showing more nonindividual DNA behaviors in normal-sized cells ( n= 82) versus more individual phage DNA behaviors in long cells ( n= 79). The error bars denote counting errors. (D) Box plot of the distance between coinfecting phages on the surface of long cells (red, n= 121) is much larger than that in normal-sized cells (blue, n= 113) at an MOI of 2. *,P,0.001 as determined by Student's ttest. See also Fig. S6.MOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 7\"], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:07 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'88'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992903'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'425ms'), (b'x-request-id', b'f3f90e874b136c1c11ac14a02eccd71c'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249c4fbe65d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:07 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'88'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992903'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'425ms'), (b'x-request-id', b'f3f90e874b136c1c11ac14a02eccd71c'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249c4fbe65d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  semble decision and tend toward MOI-independent lysogenization when they can ef ﬁ- ciently regulate each other in the scenario of normal-sized cells; on the other hand,phages can make individual decisions and tend toward an MOI-dependent lysogenic response when the distance between them is large enough to diminish trans regulation. DISCUSSION In this work, we investigated the underlying mechanism of how P1 enters its lyso- genic state, in terms of its gene regulatory circuitry, which is conserved in P1-like plas-mids (9). By taking advantage of single-cell techniques, we ﬁrst demonstrated the MOI- independent lysogenic response at the single-cell level (Fig. 1B) and then proposed its mechanism with a simple model: The C1 activity of each infecting phage, dependingon the relative expression level and interactions between C1, Coi, and Lxc, was regu-lated to be a constant value over different MOI, resulting in the similar probabilities oflysogeny (Fig. 6A). When Lxc was overexpressed, Coi function in inactivating C1 wasabolished and the C1 level of each infecting phage became the determining factor forthe lysis-lysogeny decisions, which was a constant value over different MOI, resulting in MOI independency (Fig. 6B). Given that Lxc has no homologs in phage l, the func- tion of Lxc in maintaining the probability of lysogeny suggests that phage P1 uses atotally different gene regulation at the molecular level to guide its lysis-lysogeny deci-sion. Furthermore, compared with the individuality of phage lfor decision-making (41, 59, 61, 62), the independence of MOI for the phage P1 lysogenic response stronglysignals that the infecting phages make a group decision. To test this hypothesis, we examined the interaction of injected P1 DNAs in normal- sized E. coli cells with an MOI of 2 and observed more nonindividual DNA behaviors (;70%) than the individual ones. Among these, the cases where the unsynchronized replication of P1 DNAs can be seen as phage individuality; i.e., coinfecting phagesmake different decisions due to the uneven intracellular environment. For example,the bacterial nucleoid is located in the center of the cytoplasm, leaving the polarregions largely free of DNA, while certain functional proteins, such as the osmosensory transporter ProP (67, 68), and ribosomes (68) are enriched at the cell poles. After the individual decision-making of each P1, the decisions by all phages probably integrateinto the ﬁnal cell outcome as lysis or lysogeny, similar to what occurs with phage l.O n the other hand, the majority of P1 DNAs exhibited nonindividual behaviors as theymoved together or showed synchronized replication patterns. In addition, the propor-tion of nonindividuality was assumed to be even higher in the cells with higher MOI,given the increased viral concentration, which results in an overall MOI-independent FIG 6 Model of the P1 lysis-lysogeny decision-making mechanism. (A) The lysis-lysogeny decision-making of P1 is independent of MOI. All infecting phages make a group decision inside the host cell due to trans regulation (green double arrow). The constancy of C1 activity of each phage, depending on the expression levels of C1, Coi and Lxc, results in a robust lysogenic response against the number of phages infecting an individual cell. (B) The lysi s- lysogeny decision-making of P1 is independent of MOI in the Lxc-overexpressing host. When Lxc is overexpressed, Coi function in inactivating C1 isabolished. The C1 level of each infecting phage becomes the determining factor for the lysis-lysogeny decisions, which is a constant value over diffe rent MOI, resulting in MOI independency. The C1 binding af ﬁnity or C1 activity per phage increases and leads to higher probabilities of lysogeny. (C) A different lysogenic pattern is imposed by increasing phage distances infecting the same host cell. It suggests individuality of the choice between lysis and ly sogeny of individual phages that are not able to share regulatory components effectively.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  (B) The lysi s- lysogeny decision-making of P1 is independent of MOI in the Lxc-overexpressing host. When Lxc is overexpressed, Coi function in inactivating C1 isabolished. The C1 level of each infecting phage becomes the determining factor for the lysis-lysogeny decisions, which is a constant value over diffe rent MOI, resulting in MOI independency. The C1 binding af ﬁnity or C1 activity per phage increases and leads to higher probabilities of lysogeny. (C) A different lysogenic pattern is imposed by increasing phage distances infecting the same host cell. It suggests individuality of the choice between lysis and ly sogeny of individual phages that are not able to share regulatory components effectively. Blue intensities indicate C1 activities of each infecting phage, an dd a r k e rc o l o r evinces higher activity, which is regulated by the expression levels and interactions between C1 (white pentagons), Coi (black circles), and Lxc (gr een stars).Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 8', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  cell outcome. Furthermore, we found that a large separation among P1 virions infect- ing the same cell prompted the individual decision-making and altered the lysogenic response to be dependent on MOI (Fig. 6C). This result highlights the important role of the spatial organization in the process of cell fate determination in a single-cellenvironment. The striking differences between phage P1 and phage lmay suggest distinct strat- egies developed by these two phages for ef ﬁcient genome propagation, which is essential for all organisms to survive in the ﬂuctuating environment. Phage lcan infect only a subclass of E. coli strains and may therefore be considered a specialist, implying the limited availability of host cells under natural conditions. Additionally, phage l uses the outer membrane protein LamB as its receptor, which shows helical or ring-like patterns on cell surfaces (69), contributing to the preferential adsorption to cellular poles (41, 70). The use of polar adsorption augments the infection speci ﬁcity at the cost of a higher rate of failed infection (41, 55). Thus, the ability to sense the amount ofhost cells for the infection of progeny phages is important: the injected phage DNA isreplicated and packaged into new progeny viral particles under conditions where host cells are not limiting, whereas viral DNA stays dormant in the host when not enough host is available. Moreover, the MOI dependency of lhas been found to rely on the individuality of each infecting phage; coinfecting phages make individual lysis-lysog- eny decisions, and the host cell integrates these decisions to result in lysis, lysogeny, or lyso-lysis (71). In contrast, phage P1 can be regarded as a generalist with the ability to inject its DNA into a very broad spectrum of Gram-negative hosts. It recognizes evenly distrib- uted LPS molecule as its receptor, leading to the nonspeci ﬁc adsorption to the cell sur- face (55). The broad host range and the high probability of successful DNA injection (55) reduce the need to sense the availability of host bacteria. It may also re ﬂect a phage ’s real-world experience of having to deal with diverse cytoplasmic environ- ments where the lytic pathway may be problematic and less attractive in terms of ﬁt- ness, as for the need for different strategies to replicate a large number of viral DNAs, while the establishment of a stable lysogen becomes preferred. Thus, P1 makes the lysis-lysogeny decision regardless of MOI and in general with a bias toward lysogeny.Furthermore, during lysogenization, P1 is maintained as a self-replicating plasmid (51, 52). First, the plasmid format gives P1 prophage more freedom from the host cell ’s control, in processes such as DNA replication, where P1 uses its own origin and replica-tion system (28), and protein synthesis, considering the effects of the chromosome position on the gene expression pro ﬁle in bacteria (72). Second, given the aforemen- tioned capacities for plasmid maintenance (48 –54) and the separate distribution of genes with related functions on the viral genome (28), P1 may have evolved directlyfrom an intracellular plasmid by acquiring phage-related elements. These suggest a distinct evolutionary lineage of P1 with other temperate phages. In sum, the formation of stable plasmids is probably the priority for phage P1 during infection, and a robustlysogenic response over host availability helps ensure this process. Essentially, the autonomy and constant lysogenization make P1 an ideal backbone to carry and spread cargo genes during virulence dissemination. On the other hand, the P1 phage particleis more likely to be used in lateral gene transfer between bacterial strains, which is less essential for P1 propagation. MATERIALS AND METHODS Media. LB broth contained (per liter of distilled H2O) 10 g Bacto tryptone, 5 g Bacto yeast extract, and 5 g NaCl and was adjusted to a pH of about 7.5 with NaOH.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  These suggest a distinct evolutionary lineage of P1 with other temperate phages. In sum, the formation of stable plasmids is probably the priority for phage P1 during infection, and a robustlysogenic response over host availability helps ensure this process. Essentially, the autonomy and constant lysogenization make P1 an ideal backbone to carry and spread cargo genes during virulence dissemination. On the other hand, the P1 phage particleis more likely to be used in lateral gene transfer between bacterial strains, which is less essential for P1 propagation. MATERIALS AND METHODS Media. LB broth contained (per liter of distilled H2O) 10 g Bacto tryptone, 5 g Bacto yeast extract, and 5 g NaCl and was adjusted to a pH of about 7.5 with NaOH. Derivative media were supplemented with the following: LBM, 10 mM MgSO4, used for generating crude lysates and in agarose slabs for sin- gle-cell microscopy; LB plates, 1.5% agar; LBCM plates, 1.5% Bacto agar and 12.5 mgm l21chlorampheni- col. SM buffer (50 mM Tris-HCl [pH 7.5], 100 mM NaCl, 8 mM MgSO4) was used as a phage storage buffer. NZYM (N-Z amine-NaCl-yeast extract-MgSO4) was used for phage titration, supplemented with 1.5% or 0.7% Bacto agar for plates and top agar, respectively. Neidhardt EZ rich de ﬁned medium (73) was used for DNA visualization experiments due to its low background and ﬂuorescence. It contained (per liter of distilled H2O) 1 /C2MOPS (morpholinepropanesulfonic acid) mixture (Teknova no. M2101), 1.32 mMMOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 9', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  K2HPO4(Teknova no. M2102), 1 /C2ACGU solution (Teknova no. M2103), and 1 /C2supplement EZ (Teknova no. M2104). Carbon sources were added when the cell culture was started. Bacterial strains, phages, and plasmids. All bacterial strains, plasmids, and phages are described in Table S1. (i) Construction of Lxc expression plasmids (pLZ1931 and pLZ1933). We constructed the plasmids containing the P1 lxcgene under the control of the IPTG (isopropyl- b-D-thiogalactopyranoside)-inducible promoter PLlacO-1. We inserted lxc(primers 1 and 2) into pZE12luc with NcoI and SphI (pLZ1931) and then transferred the whole fragment from PLlacO-1 to mTurquoise2 (primers 3 and 4) into pACYC177 at the BamHI and PstI sites to maintain kanamycin resistance (pLZ1933). In order to achieve uniform IPTG induction, constructed plasmids were transformed into the MG1655 lacIqDlacY strain for Lxc expression. (ii) Construction of the C1 operator plasmid (pLZ1950). There are 22 C1 binding operators dis- persed on the P1 genome. The 17-bp operator consensus sequence is asymmetric and is reported in ref-erence 28 as ATTGCTCTAATAAATTT. One copy of the consensus operator sequence was designed inprimer 5 and ampli ﬁed with primer 6 from pACYC177. PCR fragments were ligated into pACYC177 between BamHI and PstI. We transformed the plasmid into MG1655 or other strains acting as the hostcell during single-cell infection assays. (iii) Construction of tetO phage. The tetO recombination plasmid (pLZ1981) was constructed to replace part of the darB region of P1KL1856. Speci ﬁcally, a 120 /C2tetO-Kan rarray digested from pLAU39 (63) (EcoRI and SalI), a homology region (H1) inside darB (primers 7 and 8), and a kanamycin resistance gene (primers 9 and 10) with its promoter were ligated into a pUC19 backbone using Gibson assembly.Another homologous region (H2) inside darB (primers 11 and 12) was then ligated into the assembled plasmid with SalI and HindIII. Next, we used the lredrecombination method (74) to recombine the H1- Kanr-120/C2tetO-H2 sequence in the P1 genomic DNA. This DNA sequence was digested from pLZ1981 with NdeI and HindIII and transformed into LZ1856 lysogen with pKD46 under electroporation. Each electroporation reaction mixture contained 100 ng puri ﬁed DNA and 100 ml competent cells in a 0.2-cm cuvette and was transferred in a Bio-Rad MicroPulser electroporator. Cells were recovered in 1 ml SOC(2% vegetable peptone, 0.5% yeast extract, 10 mM NaCl, 2.5 mM KCl, 10 mM MgCl 2, 10 mM MgSO4, 20 mM glucose) solution for 1 to 2 h at 30°C and plated on LB plates containing Cm (12.5 mgm l21) and Kan (30 mgm l21) at 30°C overnight. The genomic construct was then veri ﬁed by sequencing. Bacteriophage assays. (i) Production of phage lysates. Phage lysates were produced by thermal induction of phage lysogens. Brie ﬂy, a single colony of the desired lysogen was grown in 1 ml LB with appropriate antibiotics overnight at 30°C. The overnight culture was diluted 100-fold in LBM and grownat 30°C until an optical density at 600 nm (OD 600)o f;0.5 was reached.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The genomic construct was then veri ﬁed by sequencing. Bacteriophage assays. (i) Production of phage lysates. Phage lysates were produced by thermal induction of phage lysogens. Brie ﬂy, a single colony of the desired lysogen was grown in 1 ml LB with appropriate antibiotics overnight at 30°C. The overnight culture was diluted 100-fold in LBM and grownat 30°C until an optical density at 600 nm (OD 600)o f;0.5 was reached. P1 or a P1 mutant was thermally induced by shifting the culture to 42°C in the shaking water bath (180 rpm) for 30 min. Then, the culture was shaken at 37°C and 180 rpm for 1 h until an OD600of;0.1 was reached. We collected the lysate in a centrifuge tube and mixed it with chloroform (2%) for about 15 min at room temperature (RT) for thor-ough lysis. The lysate was centrifuged at 10,000 /C2gfor 10 min at 4°C and sterilized by the passage of the supernatant through a 0.2- mmﬁlter (VWR International catalog no. 28145-477). The gp23- mTurquoise2 phages were obtained by inducing the lysogen (e.g., LZ1914 and LZ2504). During lysogengrowth, IPTG was added to a ﬁnal concentration of 0.075 mM when the OD 600approached 0.1, to pro- vide the proper amount of gp23-mTurquoise2 for mosaic phage assembly. (ii) Phage titration assay. An overnight culture of indicator strain MG1655 was diluted 100-fold in LBM and grown at 37°C to an OD600of;0.4. Next, we kept the cell culture on ice and added 5 mM CaCl2. The phage stock was diluted in SM buffer to an estimated 103to 104PFU ml21. We incubated 100 ml of the diluted phages with 300 ml of host solution for 20 min at RT. The phage-cell mixture was then added into 4 ml of 50°C molten NZYM top agar and plated on predried NZYM agar plates. The plateswere allowed to set for 10 min at RT and incubated overnight at 42°C. The titer was determined as theratio of plaques to the dilution factor. (iii) Phage puri ﬁcation by ultracentrifugation. The crude lysate, obtained from the thermal induc- tion in 500 ml LBM as described above, was centrifuged in a Sorvall GSA rotor at 10,000 rpm for 20 min at 4°C. The supernatant was transferred into a new bottle. After incubation with 1 mgm l21DNase and 1mgm l21RNase for 1 to 2 h at RT, the lysate was concentrated by overnight centrifugation ( ;16 h) in the Sorvall GSA rotor at 10,000 rpm, 4°C. We soaked the resulting phage pellet in 2 ml cold SM bufferand extracted the phage solution after a 48-h incubation. Phages were then puri ﬁed by equilibrium cen- trifugation in 1.40 to 1.45 g ml 21cesium chloride using the Beckman 70.1Ti rotor at 45,000 rpm for 24 to 26 h at 4°C. The phage band was extracted with 18-gauge needles and dialyzed against SM buffer inSlide-A-Lyzer 3,500-molecular-weight-cutoff (MWCO) dialysis cassettes (Thermo Scienti ﬁc). (iv) Bulk lysogenization probability assay. We measured the probability of lysogeny as a function of the MOI. An overnight culture of the host strain was grown in LB at 37°C. Cells were diluted 1,000-foldinto LBM, grown at 37°C to an OD 600of;0.4, and chilled, and CaCl2was added to a ﬁnal concentration of 5 mM. Then, 100 ml of the cells was added to 100 ml of phage solutions with different concentrations.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The phage band was extracted with 18-gauge needles and dialyzed against SM buffer inSlide-A-Lyzer 3,500-molecular-weight-cutoff (MWCO) dialysis cassettes (Thermo Scienti ﬁc). (iv) Bulk lysogenization probability assay. We measured the probability of lysogeny as a function of the MOI. An overnight culture of the host strain was grown in LB at 37°C. Cells were diluted 1,000-foldinto LBM, grown at 37°C to an OD 600of;0.4, and chilled, and CaCl2was added to a ﬁnal concentration of 5 mM. Then, 100 ml of the cells was added to 100 ml of phage solutions with different concentrations. After a 30-min incubation at 30°C, we transferred 20 ml of the mixture into 980 ml ice-cold LB or SM buffer to stop the adsorption process. Aliquots in LB medium were plated on LBCM plates and incubatedovernight at 30°C. Lysogen concentrations were determined by counting the number of Cm rcolonies. To examine the concentration of nonadsorbed phages, we centrifuged the aliquots in SM buffer andtitrated the supernatant on MG1655. Total phage and bacterium concentrations were measured usingplate assays as well. Bulk MOI was determined as total phages/total cells. The probability of lysogeny ofall cells or of the infected cells (determined from the bulk MOI using the Poisson distribution with a cu-mulative probability of $1) was plotted as a function of the MOI on a log-log scale. MOI infected was calcu- lated as (total phages 2nonadsorbed phage)/total cells. lKil expression in strains containing pBAD24-Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 10', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  lkilwas induced with ʟ-arabinose when the OD600approached 0.08. Lxc expression in strain LZ1931 was induced with IPTG when the OD600approached 0.1. Single-cell infection assay. (i) Decision-making examination. The protocol for single-cell infection assays was adapted from previous work (55). An overnight culture of LZ1915 was diluted 100-fold inLBM and grown to an OD 600of;0.4; then, CaCl2was added to 5 mM. For infections in Lxc-overexpress- ing cells, IPTG was added at a ﬁnal concentration of 0.2 mM to the cell culture (LZ1937) when the OD600 approached 0.1. Puri ﬁed P1LZ1914 phages were mixed with the cells to reach an MOI of 0.1 to 7, fol- lowed by incubation for 30 min at 30°C to trigger both phage adsorption and viral DNA injection. Onemicroliter of the sample was placed on a 1-mm 1.5% LB agarose pad resting on a small coverslip (18 by 18 mm; Fisher Scienti ﬁc). After 1 to 2 min, a large coverslip (24 by 50 mm; Fisher Scienti ﬁc) was gently overlaid, and the sample was imaged under the microscope at 30°C. To localize all phages surrounding the cells, a series of 7 z-axis images with a spacing of 300 nm were taken through the cyan ﬂuorescent protein (CFP) channel using a 500-ms exposure for each. Cells were imaged at multiple stage positions (typically 16) in each experiment. During the time-lapse movie,the sample was imaged in phase-contrast (100 ms exposure for cell recognition), yellow ﬂuorescent pro- tein (YFP) (300-ms exposure for phage C1-mVenus expression), and CFP (100-ms exposure for lytic re- porter signal) channels at intervals of 5 min until cell fate was visible (3 h in total). (ii) DNA injection on a SeqA-mKate2 strain. The protocol for DNA injection onto a strain express- ing SeqA-mKate2 (LZ1387) was adapted from our previous work (55). An overnight culture of LZ1387 was diluted for 100-fold in LBM and grown to an OD 600of;0.4; then, CaCl2was added to 5 mM. Puri ﬁed P1LZ1914 phages were mixed with the cells to reach an MOI of 2, followed by incubation for 30 min at30°C to trigger both phage adsorption and viral DNA injection. One microliter of the sample was placed on a 1-mm 1.5% phosphate-buffered saline (PBS) agarose pad resting on a small coverslip (18 by 18 mm; Fisher Scienti ﬁc). After 1 to 2 min, a large coverslip (24 by 50 mm; Fisher Scienti ﬁc) was gently overlaid, and the sample was imaged under the microscope at 30°C. Cell images were taken at 24 stage positions in the phase-contrast (100-ms exposure for cell recogni- tion), CFP (500-ms exposure for ﬂuorescent phages), and far-red (100-ms exposure for SeqA-mKate2 foci) channels with 5 zstacks with a spacing of 300 nm. (iii) DNA injection on a TetR-mNeonGreen strain. An overnight culture of LZ2001 was diluted 1 0 0 - f o l di nE Zr i c hd e ﬁned medium with 0.2% glucose and 10 mM MgSO 4. For long cells with lKil induction, the overnight culture of LZ2001 with pBAD33- lkilwas diluted 100-fold in EZ rich de ﬁned medium with 30% LB and 10 mM MgSO4.ʟ-Arabinose was added to a ﬁnal concentration of 0.05% when the OD600approached 0.08. The cells were grown to an OD600of;0.4; then, CaCl2was added to 5 mM.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  (iii) DNA injection on a TetR-mNeonGreen strain. An overnight culture of LZ2001 was diluted 1 0 0 - f o l di nE Zr i c hd e ﬁned medium with 0.2% glucose and 10 mM MgSO 4. For long cells with lKil induction, the overnight culture of LZ2001 with pBAD33- lkilwas diluted 100-fold in EZ rich de ﬁned medium with 30% LB and 10 mM MgSO4.ʟ-Arabinose was added to a ﬁnal concentration of 0.05% when the OD600approached 0.08. The cells were grown to an OD600of;0.4; then, CaCl2was added to 5 mM. Puri ﬁe dP 1 L Z 1 9 1 4p h a g e sw e r em i x e dw i t ht h ec e l l st or e a c ha nM O Io f2 ,f o l l o w e db yi n c u - bation for 30 min at 30°C to trigger both phage adsorption and viral DNA injection. One microliter ofthe sample was placed on a 1-mm 1.5% agarose pad (PBS for cell imaging, EZ rich with 0.2% glucose for time-lapse movie) resting on a small coverslip (18 by 18 mm; Fisher Scienti ﬁc). After 1 to 2 min, a large coverslip (24 by 50 mm; Fisher Scienti ﬁc) was gently overlaid, and the sample was imaged under the microscope at 30°C. A series of 7 z-axis images with a spacing of 300 nm were taken through the CFP channel using a 500-ms exposure for ﬂuorescent phages and the Green2#FISH channel using a 300-ms exposure for TetR-mNeonGreen-labeled DNA foci. Cells were imaged at multiple stage positions (typically 16) in eachexperiment. During the time-lapse movie, the sample was imaged in the phase-contrast (100-ms expo- sure for cell recognition) and Green2#FISH (300-ms exposure for DNA behavior) channels at intervals of 5 min until cell fate was visible (3 h in total). RNA FISH. Different probes were synthesized to target different phage transcripts (3 9TEG-Amino; Biosearch Technologies). Probes targeting c1,coi, and lxcwere designed as described in a previous study (60) and labeled with Cy5 (GE Healthcare no. PA15101), 6-carboxytetramethylrhodamine (TAMRA)(Biosearch Technologies, no. SMF-1001-5), and Alexa Fluor 488 (Thermo Fisher no. A20000), respectively(Fig. 3A). Probes are listed in Table S3. To perform RNA FISH on infection samples, the overnight host cells were diluted 100-fold in 65 ml of fresh LBM and grown at 37°C to an OD 600of;0.4. For the infections on Lxc-overexpressed cells IPTG was added at a ﬁnal concentration of 0.2 mM in the cell culture (LZ1938) when the OD600approached 0.1. Then, CaCl2was added to 5 mM. After that, a 750- ml aliquot of cells was separated as a negative con- trol without phages. The rest of the cell culture was mixed with the puri ﬁed P1LZ1856 to reach MOI of 0.2, 1, and 5, and the mixture was incubated at 30°C (0 min). At each time point (10, 20, 30, 40, 50, 60,80, and 100 min) during incubation, a 750- ml aliquot of the mixture was placed in a new 15-ml centri- fuge tube with 830 ml of 37% formaldehyde. This tube was left to shake on a nutator for 30 min and then centrifuged at 400 /C2gfor 8 min to pellet the cells. Details of ﬁxation, permeabilization, and hybrid- ization were presented in previous studies (60, 75).', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The rest of the cell culture was mixed with the puri ﬁed P1LZ1856 to reach MOI of 0.2, 1, and 5, and the mixture was incubated at 30°C (0 min). At each time point (10, 20, 30, 40, 50, 60,80, and 100 min) during incubation, a 750- ml aliquot of the mixture was placed in a new 15-ml centri- fuge tube with 830 ml of 37% formaldehyde. This tube was left to shake on a nutator for 30 min and then centrifuged at 400 /C2gfor 8 min to pellet the cells. Details of ﬁxation, permeabilization, and hybrid- ization were presented in previous studies (60, 75). Brie ﬂy, after ﬁxation, the cells were washed three times with 1 /C2PBS. Subsequently, the cells were permeabilized by resuspension in 70% ethanol for 2 h at room temperature and centrifuged to collect the cells. The pellet was then resuspended in wash solu- tion (40% formamide, 2 /C2SSC [1 /C2SSC is 0.15 M NaCl plus 0.015 M sodium citrate]), incubated for 5 min at room temperature, and pelleted again, after which they were ready for hybridization. The cells werethen resuspended in 25 ml hybridization solution (40% formamide, 2 /C2SSC, 1 mg ml21E. coli tRNA, 2 mM ribonucleoside-vanadyl complex, and 0.2 mg ml21bovine serum albumin [BSA]) with each set of probes reaching a ﬁnal concentration of 1 mM. The samples were then incubated in a 30°C water bath overnight. The next day, the cells were washed three times using wash solution by incubating the cell pellet for 30 min in a 30°C water bath. After the ﬁnal wash, the cells were resuspended in wash solutionMOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 11'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  semble decision and tend toward MOI-independent lysogenization when they can ef ﬁ- ciently regulate each other in the scenario of normal-sized cells; on the other hand,phages can make individual decisions and tend toward an MOI-dependent lysogenic response when the distance between them is large enough to diminish trans regulation. DISCUSSION In this work, we investigated the underlying mechanism of how P1 enters its lyso- genic state, in terms of its gene regulatory circuitry, which is conserved in P1-like plas-mids (9). By taking advantage of single-cell techniques, we ﬁrst demonstrated the MOI- independent lysogenic response at the single-cell level (Fig. 1B) and then proposed its mechanism with a simple model: The C1 activity of each infecting phage, dependingon the relative expression level and interactions between C1, Coi, and Lxc, was regu-lated to be a constant value over different MOI, resulting in the similar probabilities oflysogeny (Fig. 6A). When Lxc was overexpressed, Coi function in inactivating C1 wasabolished and the C1 level of each infecting phage became the determining factor forthe lysis-lysogeny decisions, which was a constant value over different MOI, resulting in MOI independency (Fig. 6B). Given that Lxc has no homologs in phage l, the func- tion of Lxc in maintaining the probability of lysogeny suggests that phage P1 uses atotally different gene regulation at the molecular level to guide its lysis-lysogeny deci-sion. Furthermore, compared with the individuality of phage lfor decision-making (41, 59, 61, 62), the independence of MOI for the phage P1 lysogenic response stronglysignals that the infecting phages make a group decision. To test this hypothesis, we examined the interaction of injected P1 DNAs in normal- sized E. coli cells with an MOI of 2 and observed more nonindividual DNA behaviors (;70%) than the individual ones. Among these, the cases where the unsynchronized replication of P1 DNAs can be seen as phage individuality; i.e., coinfecting phagesmake different decisions due to the uneven intracellular environment. For example,the bacterial nucleoid is located in the center of the cytoplasm, leaving the polarregions largely free of DNA, while certain functional proteins, such as the osmosensory transporter ProP (67, 68), and ribosomes (68) are enriched at the cell poles. After the individual decision-making of each P1, the decisions by all phages probably integrateinto the ﬁnal cell outcome as lysis or lysogeny, similar to what occurs with phage l.O n the other hand, the majority of P1 DNAs exhibited nonindividual behaviors as theymoved together or showed synchronized replication patterns. In addition, the propor-tion of nonindividuality was assumed to be even higher in the cells with higher MOI,given the increased viral concentration, which results in an overall MOI-independent FIG 6 Model of the P1 lysis-lysogeny decision-making mechanism. (A) The lysis-lysogeny decision-making of P1 is independent of MOI. All infecting phages make a group decision inside the host cell due to trans regulation (green double arrow). The constancy of C1 activity of each phage, depending on the expression levels of C1, Coi and Lxc, results in a robust lysogenic response against the number of phages infecting an individual cell. (B) The lysi s- lysogeny decision-making of P1 is independent of MOI in the Lxc-overexpressing host. When Lxc is overexpressed, Coi function in inactivating C1 isabolished. The C1 level of each infecting phage becomes the determining factor for the lysis-lysogeny decisions, which is a constant value over diffe rent MOI, resulting in MOI independency. The C1 binding af ﬁnity or C1 activity per phage increases and leads to higher probabilities of lysogeny. (C) A different lysogenic pattern is imposed by increasing phage distances infecting the same host cell. It suggests individuality of the choice between lysis and ly sogeny of individual phages that are not able to share regulatory components effectively.', 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  (B) The lysi s- lysogeny decision-making of P1 is independent of MOI in the Lxc-overexpressing host. When Lxc is overexpressed, Coi function in inactivating C1 isabolished. The C1 level of each infecting phage becomes the determining factor for the lysis-lysogeny decisions, which is a constant value over diffe rent MOI, resulting in MOI independency. The C1 binding af ﬁnity or C1 activity per phage increases and leads to higher probabilities of lysogeny. (C) A different lysogenic pattern is imposed by increasing phage distances infecting the same host cell. It suggests individuality of the choice between lysis and ly sogeny of individual phages that are not able to share regulatory components effectively. Blue intensities indicate C1 activities of each infecting phage, an dd a r k e rc o l o r evinces higher activity, which is regulated by the expression levels and interactions between C1 (white pentagons), Coi (black circles), and Lxc (gr een stars).Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 8', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  cell outcome. Furthermore, we found that a large separation among P1 virions infect- ing the same cell prompted the individual decision-making and altered the lysogenic response to be dependent on MOI (Fig. 6C). This result highlights the important role of the spatial organization in the process of cell fate determination in a single-cellenvironment. The striking differences between phage P1 and phage lmay suggest distinct strat- egies developed by these two phages for ef ﬁcient genome propagation, which is essential for all organisms to survive in the ﬂuctuating environment. Phage lcan infect only a subclass of E. coli strains and may therefore be considered a specialist, implying the limited availability of host cells under natural conditions. Additionally, phage l uses the outer membrane protein LamB as its receptor, which shows helical or ring-like patterns on cell surfaces (69), contributing to the preferential adsorption to cellular poles (41, 70). The use of polar adsorption augments the infection speci ﬁcity at the cost of a higher rate of failed infection (41, 55). Thus, the ability to sense the amount ofhost cells for the infection of progeny phages is important: the injected phage DNA isreplicated and packaged into new progeny viral particles under conditions where host cells are not limiting, whereas viral DNA stays dormant in the host when not enough host is available. Moreover, the MOI dependency of lhas been found to rely on the individuality of each infecting phage; coinfecting phages make individual lysis-lysog- eny decisions, and the host cell integrates these decisions to result in lysis, lysogeny, or lyso-lysis (71). In contrast, phage P1 can be regarded as a generalist with the ability to inject its DNA into a very broad spectrum of Gram-negative hosts. It recognizes evenly distrib- uted LPS molecule as its receptor, leading to the nonspeci ﬁc adsorption to the cell sur- face (55). The broad host range and the high probability of successful DNA injection (55) reduce the need to sense the availability of host bacteria. It may also re ﬂect a phage ’s real-world experience of having to deal with diverse cytoplasmic environ- ments where the lytic pathway may be problematic and less attractive in terms of ﬁt- ness, as for the need for different strategies to replicate a large number of viral DNAs, while the establishment of a stable lysogen becomes preferred. Thus, P1 makes the lysis-lysogeny decision regardless of MOI and in general with a bias toward lysogeny.Furthermore, during lysogenization, P1 is maintained as a self-replicating plasmid (51, 52). First, the plasmid format gives P1 prophage more freedom from the host cell ’s control, in processes such as DNA replication, where P1 uses its own origin and replica-tion system (28), and protein synthesis, considering the effects of the chromosome position on the gene expression pro ﬁle in bacteria (72). Second, given the aforemen- tioned capacities for plasmid maintenance (48 –54) and the separate distribution of genes with related functions on the viral genome (28), P1 may have evolved directlyfrom an intracellular plasmid by acquiring phage-related elements. These suggest a distinct evolutionary lineage of P1 with other temperate phages. In sum, the formation of stable plasmids is probably the priority for phage P1 during infection, and a robustlysogenic response over host availability helps ensure this process. Essentially, the autonomy and constant lysogenization make P1 an ideal backbone to carry and spread cargo genes during virulence dissemination. On the other hand, the P1 phage particleis more likely to be used in lateral gene transfer between bacterial strains, which is less essential for P1 propagation. MATERIALS AND METHODS Media. LB broth contained (per liter of distilled H2O) 10 g Bacto tryptone, 5 g Bacto yeast extract, and 5 g NaCl and was adjusted to a pH of about 7.5 with NaOH.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  These suggest a distinct evolutionary lineage of P1 with other temperate phages. In sum, the formation of stable plasmids is probably the priority for phage P1 during infection, and a robustlysogenic response over host availability helps ensure this process. Essentially, the autonomy and constant lysogenization make P1 an ideal backbone to carry and spread cargo genes during virulence dissemination. On the other hand, the P1 phage particleis more likely to be used in lateral gene transfer between bacterial strains, which is less essential for P1 propagation. MATERIALS AND METHODS Media. LB broth contained (per liter of distilled H2O) 10 g Bacto tryptone, 5 g Bacto yeast extract, and 5 g NaCl and was adjusted to a pH of about 7.5 with NaOH. Derivative media were supplemented with the following: LBM, 10 mM MgSO4, used for generating crude lysates and in agarose slabs for sin- gle-cell microscopy; LB plates, 1.5% agar; LBCM plates, 1.5% Bacto agar and 12.5 mgm l21chlorampheni- col. SM buffer (50 mM Tris-HCl [pH 7.5], 100 mM NaCl, 8 mM MgSO4) was used as a phage storage buffer. NZYM (N-Z amine-NaCl-yeast extract-MgSO4) was used for phage titration, supplemented with 1.5% or 0.7% Bacto agar for plates and top agar, respectively. Neidhardt EZ rich de ﬁned medium (73) was used for DNA visualization experiments due to its low background and ﬂuorescence. It contained (per liter of distilled H2O) 1 /C2MOPS (morpholinepropanesulfonic acid) mixture (Teknova no. M2101), 1.32 mMMOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 9', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  K2HPO4(Teknova no. M2102), 1 /C2ACGU solution (Teknova no. M2103), and 1 /C2supplement EZ (Teknova no. M2104). Carbon sources were added when the cell culture was started. Bacterial strains, phages, and plasmids. All bacterial strains, plasmids, and phages are described in Table S1. (i) Construction of Lxc expression plasmids (pLZ1931 and pLZ1933). We constructed the plasmids containing the P1 lxcgene under the control of the IPTG (isopropyl- b-D-thiogalactopyranoside)-inducible promoter PLlacO-1. We inserted lxc(primers 1 and 2) into pZE12luc with NcoI and SphI (pLZ1931) and then transferred the whole fragment from PLlacO-1 to mTurquoise2 (primers 3 and 4) into pACYC177 at the BamHI and PstI sites to maintain kanamycin resistance (pLZ1933). In order to achieve uniform IPTG induction, constructed plasmids were transformed into the MG1655 lacIqDlacY strain for Lxc expression. (ii) Construction of the C1 operator plasmid (pLZ1950). There are 22 C1 binding operators dis- persed on the P1 genome. The 17-bp operator consensus sequence is asymmetric and is reported in ref-erence 28 as ATTGCTCTAATAAATTT. One copy of the consensus operator sequence was designed inprimer 5 and ampli ﬁed with primer 6 from pACYC177. PCR fragments were ligated into pACYC177 between BamHI and PstI. We transformed the plasmid into MG1655 or other strains acting as the hostcell during single-cell infection assays. (iii) Construction of tetO phage. The tetO recombination plasmid (pLZ1981) was constructed to replace part of the darB region of P1KL1856. Speci ﬁcally, a 120 /C2tetO-Kan rarray digested from pLAU39 (63) (EcoRI and SalI), a homology region (H1) inside darB (primers 7 and 8), and a kanamycin resistance gene (primers 9 and 10) with its promoter were ligated into a pUC19 backbone using Gibson assembly.Another homologous region (H2) inside darB (primers 11 and 12) was then ligated into the assembled plasmid with SalI and HindIII. Next, we used the lredrecombination method (74) to recombine the H1- Kanr-120/C2tetO-H2 sequence in the P1 genomic DNA. This DNA sequence was digested from pLZ1981 with NdeI and HindIII and transformed into LZ1856 lysogen with pKD46 under electroporation. Each electroporation reaction mixture contained 100 ng puri ﬁed DNA and 100 ml competent cells in a 0.2-cm cuvette and was transferred in a Bio-Rad MicroPulser electroporator. Cells were recovered in 1 ml SOC(2% vegetable peptone, 0.5% yeast extract, 10 mM NaCl, 2.5 mM KCl, 10 mM MgCl 2, 10 mM MgSO4, 20 mM glucose) solution for 1 to 2 h at 30°C and plated on LB plates containing Cm (12.5 mgm l21) and Kan (30 mgm l21) at 30°C overnight. The genomic construct was then veri ﬁed by sequencing. Bacteriophage assays. (i) Production of phage lysates. Phage lysates were produced by thermal induction of phage lysogens. Brie ﬂy, a single colony of the desired lysogen was grown in 1 ml LB with appropriate antibiotics overnight at 30°C. The overnight culture was diluted 100-fold in LBM and grownat 30°C until an optical density at 600 nm (OD 600)o f;0.5 was reached.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The genomic construct was then veri ﬁed by sequencing. Bacteriophage assays. (i) Production of phage lysates. Phage lysates were produced by thermal induction of phage lysogens. Brie ﬂy, a single colony of the desired lysogen was grown in 1 ml LB with appropriate antibiotics overnight at 30°C. The overnight culture was diluted 100-fold in LBM and grownat 30°C until an optical density at 600 nm (OD 600)o f;0.5 was reached. P1 or a P1 mutant was thermally induced by shifting the culture to 42°C in the shaking water bath (180 rpm) for 30 min. Then, the culture was shaken at 37°C and 180 rpm for 1 h until an OD600of;0.1 was reached. We collected the lysate in a centrifuge tube and mixed it with chloroform (2%) for about 15 min at room temperature (RT) for thor-ough lysis. The lysate was centrifuged at 10,000 /C2gfor 10 min at 4°C and sterilized by the passage of the supernatant through a 0.2- mmﬁlter (VWR International catalog no. 28145-477). The gp23- mTurquoise2 phages were obtained by inducing the lysogen (e.g., LZ1914 and LZ2504). During lysogengrowth, IPTG was added to a ﬁnal concentration of 0.075 mM when the OD 600approached 0.1, to pro- vide the proper amount of gp23-mTurquoise2 for mosaic phage assembly. (ii) Phage titration assay. An overnight culture of indicator strain MG1655 was diluted 100-fold in LBM and grown at 37°C to an OD600of;0.4. Next, we kept the cell culture on ice and added 5 mM CaCl2. The phage stock was diluted in SM buffer to an estimated 103to 104PFU ml21. We incubated 100 ml of the diluted phages with 300 ml of host solution for 20 min at RT. The phage-cell mixture was then added into 4 ml of 50°C molten NZYM top agar and plated on predried NZYM agar plates. The plateswere allowed to set for 10 min at RT and incubated overnight at 42°C. The titer was determined as theratio of plaques to the dilution factor. (iii) Phage puri ﬁcation by ultracentrifugation. The crude lysate, obtained from the thermal induc- tion in 500 ml LBM as described above, was centrifuged in a Sorvall GSA rotor at 10,000 rpm for 20 min at 4°C. The supernatant was transferred into a new bottle. After incubation with 1 mgm l21DNase and 1mgm l21RNase for 1 to 2 h at RT, the lysate was concentrated by overnight centrifugation ( ;16 h) in the Sorvall GSA rotor at 10,000 rpm, 4°C. We soaked the resulting phage pellet in 2 ml cold SM bufferand extracted the phage solution after a 48-h incubation. Phages were then puri ﬁed by equilibrium cen- trifugation in 1.40 to 1.45 g ml 21cesium chloride using the Beckman 70.1Ti rotor at 45,000 rpm for 24 to 26 h at 4°C. The phage band was extracted with 18-gauge needles and dialyzed against SM buffer inSlide-A-Lyzer 3,500-molecular-weight-cutoff (MWCO) dialysis cassettes (Thermo Scienti ﬁc). (iv) Bulk lysogenization probability assay. We measured the probability of lysogeny as a function of the MOI. An overnight culture of the host strain was grown in LB at 37°C. Cells were diluted 1,000-foldinto LBM, grown at 37°C to an OD 600of;0.4, and chilled, and CaCl2was added to a ﬁnal concentration of 5 mM. Then, 100 ml of the cells was added to 100 ml of phage solutions with different concentrations.', 'page_label: 10 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The phage band was extracted with 18-gauge needles and dialyzed against SM buffer inSlide-A-Lyzer 3,500-molecular-weight-cutoff (MWCO) dialysis cassettes (Thermo Scienti ﬁc). (iv) Bulk lysogenization probability assay. We measured the probability of lysogeny as a function of the MOI. An overnight culture of the host strain was grown in LB at 37°C. Cells were diluted 1,000-foldinto LBM, grown at 37°C to an OD 600of;0.4, and chilled, and CaCl2was added to a ﬁnal concentration of 5 mM. Then, 100 ml of the cells was added to 100 ml of phage solutions with different concentrations. After a 30-min incubation at 30°C, we transferred 20 ml of the mixture into 980 ml ice-cold LB or SM buffer to stop the adsorption process. Aliquots in LB medium were plated on LBCM plates and incubatedovernight at 30°C. Lysogen concentrations were determined by counting the number of Cm rcolonies. To examine the concentration of nonadsorbed phages, we centrifuged the aliquots in SM buffer andtitrated the supernatant on MG1655. Total phage and bacterium concentrations were measured usingplate assays as well. Bulk MOI was determined as total phages/total cells. The probability of lysogeny ofall cells or of the infected cells (determined from the bulk MOI using the Poisson distribution with a cu-mulative probability of $1) was plotted as a function of the MOI on a log-log scale. MOI infected was calcu- lated as (total phages 2nonadsorbed phage)/total cells. lKil expression in strains containing pBAD24-Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 10', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  lkilwas induced with ʟ-arabinose when the OD600approached 0.08. Lxc expression in strain LZ1931 was induced with IPTG when the OD600approached 0.1. Single-cell infection assay. (i) Decision-making examination. The protocol for single-cell infection assays was adapted from previous work (55). An overnight culture of LZ1915 was diluted 100-fold inLBM and grown to an OD 600of;0.4; then, CaCl2was added to 5 mM. For infections in Lxc-overexpress- ing cells, IPTG was added at a ﬁnal concentration of 0.2 mM to the cell culture (LZ1937) when the OD600 approached 0.1. Puri ﬁed P1LZ1914 phages were mixed with the cells to reach an MOI of 0.1 to 7, fol- lowed by incubation for 30 min at 30°C to trigger both phage adsorption and viral DNA injection. Onemicroliter of the sample was placed on a 1-mm 1.5% LB agarose pad resting on a small coverslip (18 by 18 mm; Fisher Scienti ﬁc). After 1 to 2 min, a large coverslip (24 by 50 mm; Fisher Scienti ﬁc) was gently overlaid, and the sample was imaged under the microscope at 30°C. To localize all phages surrounding the cells, a series of 7 z-axis images with a spacing of 300 nm were taken through the cyan ﬂuorescent protein (CFP) channel using a 500-ms exposure for each. Cells were imaged at multiple stage positions (typically 16) in each experiment. During the time-lapse movie,the sample was imaged in phase-contrast (100 ms exposure for cell recognition), yellow ﬂuorescent pro- tein (YFP) (300-ms exposure for phage C1-mVenus expression), and CFP (100-ms exposure for lytic re- porter signal) channels at intervals of 5 min until cell fate was visible (3 h in total). (ii) DNA injection on a SeqA-mKate2 strain. The protocol for DNA injection onto a strain express- ing SeqA-mKate2 (LZ1387) was adapted from our previous work (55). An overnight culture of LZ1387 was diluted for 100-fold in LBM and grown to an OD 600of;0.4; then, CaCl2was added to 5 mM. Puri ﬁed P1LZ1914 phages were mixed with the cells to reach an MOI of 2, followed by incubation for 30 min at30°C to trigger both phage adsorption and viral DNA injection. One microliter of the sample was placed on a 1-mm 1.5% phosphate-buffered saline (PBS) agarose pad resting on a small coverslip (18 by 18 mm; Fisher Scienti ﬁc). After 1 to 2 min, a large coverslip (24 by 50 mm; Fisher Scienti ﬁc) was gently overlaid, and the sample was imaged under the microscope at 30°C. Cell images were taken at 24 stage positions in the phase-contrast (100-ms exposure for cell recogni- tion), CFP (500-ms exposure for ﬂuorescent phages), and far-red (100-ms exposure for SeqA-mKate2 foci) channels with 5 zstacks with a spacing of 300 nm. (iii) DNA injection on a TetR-mNeonGreen strain. An overnight culture of LZ2001 was diluted 1 0 0 - f o l di nE Zr i c hd e ﬁned medium with 0.2% glucose and 10 mM MgSO 4. For long cells with lKil induction, the overnight culture of LZ2001 with pBAD33- lkilwas diluted 100-fold in EZ rich de ﬁned medium with 30% LB and 10 mM MgSO4.ʟ-Arabinose was added to a ﬁnal concentration of 0.05% when the OD600approached 0.08. The cells were grown to an OD600of;0.4; then, CaCl2was added to 5 mM.', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  (iii) DNA injection on a TetR-mNeonGreen strain. An overnight culture of LZ2001 was diluted 1 0 0 - f o l di nE Zr i c hd e ﬁned medium with 0.2% glucose and 10 mM MgSO 4. For long cells with lKil induction, the overnight culture of LZ2001 with pBAD33- lkilwas diluted 100-fold in EZ rich de ﬁned medium with 30% LB and 10 mM MgSO4.ʟ-Arabinose was added to a ﬁnal concentration of 0.05% when the OD600approached 0.08. The cells were grown to an OD600of;0.4; then, CaCl2was added to 5 mM. Puri ﬁe dP 1 L Z 1 9 1 4p h a g e sw e r em i x e dw i t ht h ec e l l st or e a c ha nM O Io f2 ,f o l l o w e db yi n c u - bation for 30 min at 30°C to trigger both phage adsorption and viral DNA injection. One microliter ofthe sample was placed on a 1-mm 1.5% agarose pad (PBS for cell imaging, EZ rich with 0.2% glucose for time-lapse movie) resting on a small coverslip (18 by 18 mm; Fisher Scienti ﬁc). After 1 to 2 min, a large coverslip (24 by 50 mm; Fisher Scienti ﬁc) was gently overlaid, and the sample was imaged under the microscope at 30°C. A series of 7 z-axis images with a spacing of 300 nm were taken through the CFP channel using a 500-ms exposure for ﬂuorescent phages and the Green2#FISH channel using a 300-ms exposure for TetR-mNeonGreen-labeled DNA foci. Cells were imaged at multiple stage positions (typically 16) in eachexperiment. During the time-lapse movie, the sample was imaged in the phase-contrast (100-ms expo- sure for cell recognition) and Green2#FISH (300-ms exposure for DNA behavior) channels at intervals of 5 min until cell fate was visible (3 h in total). RNA FISH. Different probes were synthesized to target different phage transcripts (3 9TEG-Amino; Biosearch Technologies). Probes targeting c1,coi, and lxcwere designed as described in a previous study (60) and labeled with Cy5 (GE Healthcare no. PA15101), 6-carboxytetramethylrhodamine (TAMRA)(Biosearch Technologies, no. SMF-1001-5), and Alexa Fluor 488 (Thermo Fisher no. A20000), respectively(Fig. 3A). Probes are listed in Table S3. To perform RNA FISH on infection samples, the overnight host cells were diluted 100-fold in 65 ml of fresh LBM and grown at 37°C to an OD 600of;0.4. For the infections on Lxc-overexpressed cells IPTG was added at a ﬁnal concentration of 0.2 mM in the cell culture (LZ1938) when the OD600approached 0.1. Then, CaCl2was added to 5 mM. After that, a 750- ml aliquot of cells was separated as a negative con- trol without phages. The rest of the cell culture was mixed with the puri ﬁed P1LZ1856 to reach MOI of 0.2, 1, and 5, and the mixture was incubated at 30°C (0 min). At each time point (10, 20, 30, 40, 50, 60,80, and 100 min) during incubation, a 750- ml aliquot of the mixture was placed in a new 15-ml centri- fuge tube with 830 ml of 37% formaldehyde. This tube was left to shake on a nutator for 30 min and then centrifuged at 400 /C2gfor 8 min to pellet the cells. Details of ﬁxation, permeabilization, and hybrid- ization were presented in previous studies (60, 75).', 'page_label: 11 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The rest of the cell culture was mixed with the puri ﬁed P1LZ1856 to reach MOI of 0.2, 1, and 5, and the mixture was incubated at 30°C (0 min). At each time point (10, 20, 30, 40, 50, 60,80, and 100 min) during incubation, a 750- ml aliquot of the mixture was placed in a new 15-ml centri- fuge tube with 830 ml of 37% formaldehyde. This tube was left to shake on a nutator for 30 min and then centrifuged at 400 /C2gfor 8 min to pellet the cells. Details of ﬁxation, permeabilization, and hybrid- ization were presented in previous studies (60, 75). Brie ﬂy, after ﬁxation, the cells were washed three times with 1 /C2PBS. Subsequently, the cells were permeabilized by resuspension in 70% ethanol for 2 h at room temperature and centrifuged to collect the cells. The pellet was then resuspended in wash solu- tion (40% formamide, 2 /C2SSC [1 /C2SSC is 0.15 M NaCl plus 0.015 M sodium citrate]), incubated for 5 min at room temperature, and pelleted again, after which they were ready for hybridization. The cells werethen resuspended in 25 ml hybridization solution (40% formamide, 2 /C2SSC, 1 mg ml21E. coli tRNA, 2 mM ribonucleoside-vanadyl complex, and 0.2 mg ml21bovine serum albumin [BSA]) with each set of probes reaching a ﬁnal concentration of 1 mM. The samples were then incubated in a 30°C water bath overnight. The next day, the cells were washed three times using wash solution by incubating the cell pellet for 30 min in a 30°C water bath. After the ﬁnal wash, the cells were resuspended in wash solutionMOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 11'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:08 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'91'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993197'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'408ms'), (b'x-request-id', b'cfdf5b989ef11a9528077bac84b85069'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249c9d9515d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:08 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'91'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'993197'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'408ms'), (b'x-request-id', b'cfdf5b989ef11a9528077bac84b85069'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249c9d9515d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  plus 10 mgm l21DAPI (4 9,6-diamidino-2-phenylindole) and incubated for 10 min at room temperature. This suspension was then pelleted and resuspended in 2 /C2SSC. The sample was then ready for imaging. Two microliters of the cell suspension was pipetted onto the center of a 24- by 50-mm coverslip. A 1.5% PBS agarose pad was laid slowly on top of the cell suspension droplet with a razor blade. The padwas covered with an 18- by 18-mm coverslip. Under the microscope, images were taken in the phase-contrast (100 ms), Cy5 (700 ms, c1), Cy3 (1 s, coi), green ﬂuorescent protein (GFP) (2 s, lxc), and DAPI (30 ms, DAPI) channels at different stage positions. DNA FISH. For DNA FISH for P1 DNA replication detection, probes were produced by PCR amplifying ;3 kb of P1 DNA (dna ﬁsh-P1-for and dna ﬁsh-P1-rev primer pair, listed in Table S2), using a phage lysate as the template, and treating the puri ﬁed PCR product with a PromoFluor500-dUTP nick translation kit (PromoCell) to generate DNA-PromoFluor500 fragments ranging from 100 to 500 bp. To perform DNA FISH on infection samples, the overnight host cells were diluted 100-fold into 50 ml of fresh LBM and grown at 37°C to an OD 600of;0.4. Then CaCl2was added to 5 mM. After that, a 5-ml aliquot of cells was separated as a negative control without phages and mixed with 550 ml of 37% formaldehyde forﬁxation. The rest of the cell culture was mixed with the puri ﬁed P1LZ1856 to reach an MOI of 0.2, fol- lowed by incubation at 30°C. At each time point (30, 40, 50, and 60 min) during incubation, 5 ml of the mix- ture was transferred to a 15-ml centrifuge tube with 550 ml of 37% formaldehyde. This tube was left to shake on a nutator for 30 min and then centrifuged at 4,000 /C2gfor 3 min to pellet the cells. Details of ﬁxation, permeabilization, and hybridization steps are detailed in reference 76. Brie ﬂy, the ﬁxed cells were washed with 1 ml of ice-cold 1 /C2PBS three times and resuspended in 1 ml of GTE solution (50 mM glucose, 20 mM Tris-HCl [pH 7.5], 10 mM EDTA). For the control sample, three separate 500- mla l i - quots of the cell suspension were then mixed with 10 mlo f0 . 0 1 mgml21lysozyme solution and incubated at room temperature for 2, 4, and 6 min, followed by three washes with GTE, and the cells were pelleted viacentrifugation at 10,000 /C2gfor 30 s. The cells were then resuspended in ;150 ml of GTE. For each control sample, 1 ml of the cells was deposited onto a PBS agarose pad and imaged. The lysozyme treatment time yielding ;90 to 95% intact cells ( ;1 to 5% lysed cells) represents the optimal treatment time for the sam- ples. The actual time point samples, from the initial GTE wash, were then processed as the control was,using the optimal lysozyme time. For each time point, 10 ml of cells were deposited onto poly- L-lysine- coated large coverslips (24 by 50 mm) and then covered with a smaller, normal coverslip (22 by 22 mm).The coverslips were then immersed in 1 /C2PBS, and the smaller coverslip was removed, leaving only the sample coverslip. The cells were then dehydrated by immersing the coverslip in increasing concentrationsof ethanol (70, 90, and100%). Samples were then ready for hybridization.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The lysozyme treatment time yielding ;90 to 95% intact cells ( ;1 to 5% lysed cells) represents the optimal treatment time for the sam- ples. The actual time point samples, from the initial GTE wash, were then processed as the control was,using the optimal lysozyme time. For each time point, 10 ml of cells were deposited onto poly- L-lysine- coated large coverslips (24 by 50 mm) and then covered with a smaller, normal coverslip (22 by 22 mm).The coverslips were then immersed in 1 /C2PBS, and the smaller coverslip was removed, leaving only the sample coverslip. The cells were then dehydrated by immersing the coverslip in increasing concentrationsof ethanol (70, 90, and100%). Samples were then ready for hybridization. For each sample, approximately 160 mg of the probe mixture was combined with 10 ml of hybridiza- tion solution (50% formamide, 10% dextran sulfate, 50 mM NaPO4[pH 7], 2 /C2SSC). The double-stranded- DNA (dsDNA) probes were denatured at 75°C in a thermocycler and then placed on ice. Ten microlitersof the denatured probe mixture was then deposited onto the center of the sample on the coverslip andoverlaid with a small coverslip (22 by 22 mm). The small coverslip was then sealed with nail polish, form-ing a sample chamber. The chambers were incubated at 80°C for 5 min to denature the cellular DNAand then placed on Kimwipes over ice for 5 min. The chambers were then incubated in a 37°C incubatorovernight to complete hybridization. The next day, the chambers were immersed in 2 /C2SSC until the smaller coverslip dislodged. The remaining coverslips were soaked in wash solution (2 /C2SSC, 50% form- amide) for 20 min at 37°C twice. The coverslips were then washed with SSC in a series of increasing con-centrations (1 /C2,2/C2, and 4 /C2), each for 5 min at room temperature. A DAPI solution was then made by mixing 1 mlo f1 0m gm l21DAPI with 1 ml of 4 /C2SSC. For each sample, 500 ml of the DAPI solution was added over the sample, covering it, and incubated for 5 min at room temperature. After the coverslip was dried, 10 mlo f2 /C2SSC was added over the sample and overlaid with a small coverslip (22 by 22 mm). The samples were then imaged. Under the microscope, images were taken in phase-contrast (100 ms), GFP (300 ms), and DAPI (30 ms) channels at different stage positions. qPCR. The overnight host cells were diluted 100-fold in fresh LBM and grown at 37°C. During cell growth, methanol was kept at 220°C for cell ﬁxation. When the cell culture reached an OD600of;0.4, CaCl2was added to a ﬁnal concentration of 5 mM. We then transferred 2 ml of cells to a 50-ml centrifuge tube and mixed them with 2 ml ice-cold methanol as the negative-control sample. The rest of the cellculture was mixed with puri ﬁed P1LZ1856 to reach an MOI of ;1 and incubated at 30°C (0 min). For each time point (0, 10, 20, 30, 40, 50, 60, 70, and 80 min), 2 ml of the reaction was aliquoted to 50-mltubes, vortexed for 10 s, and then centrifuged at 4,000 /C2gfor 3 min. After removing the supernatant, we resuspended the cell pellet with 2 ml fresh ice-cold LB solution, added 2 ml ice-cold methanol, andvortexed it for another 10 s. The mixture was centrifuged at 4,000 /C2gfor 3 min.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The rest of the cellculture was mixed with puri ﬁed P1LZ1856 to reach an MOI of ;1 and incubated at 30°C (0 min). For each time point (0, 10, 20, 30, 40, 50, 60, 70, and 80 min), 2 ml of the reaction was aliquoted to 50-mltubes, vortexed for 10 s, and then centrifuged at 4,000 /C2gfor 3 min. After removing the supernatant, we resuspended the cell pellet with 2 ml fresh ice-cold LB solution, added 2 ml ice-cold methanol, andvortexed it for another 10 s. The mixture was centrifuged at 4,000 /C2gfor 3 min. After removal of the su- pernatant, the cell pellet was kept at 220°C until DNA extraction using an UltraClean microbial DNA iso- lation kit (Mo Bio Laboratories no. 12224-50). The DNAs were then diluted and used for qPCR with pri-mers targeting the phage DNA (qPCR-P1DNA-for/rev). The E. coli DNA number was used as a reference using primers targeting the dxsgene (dsx-for/rev) (77). Ampli ﬁcation was done using SYBR green PCR master mix (Applied Biosystems no. 4309155) with a 250 nM concentration of each primer (Table S2). Microscopy imaging. Imaging was performed on a Nikon Eclipse Ti inverted epi ﬂuorescence micro- scope using a 100 /C2objective (Plan Fluo; numerical aperture [NA], 1.40; oil immersion) with a 2.5 /C2TV relay lens, within the cage of an incubator (InVivo Scienti ﬁc) at 30°C, and acquired using a cooled elec- tron-multiplying charge-coupled device (EMCCD) camera (IXON 897; Andor, Belfast, UK). Cells wereimaged under phase contrast and the following ﬂuorescent ﬁlter cubes (parameters are given as excita- tion ﬁlter wavelength and bandwidth; dichroic beam splitter wavelength; emission ﬁlter wavelength and bandwidth; company and product number): CFP (436 nm, 20; 455 nm; 480 nm, 40; Nikon, 96361),Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 12', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  YFP (539 nm, 21; 556 nm; 576 nm, 31; Chroma no. 49309), far red (592 nm, 21; 610 nm; 630 nm, 30; Chroma no. 49310), DAPI (350 nm, 50; 400 nm; 460 nm, 50; Nikon no. 96310), Cy3 (545 nm, 30; 570 nm; 610 nm, 75; Nikon no. 96323), Cy5 (615 nm, 70; 660 nm; 700 nm, 75; Nikon no. 96366), and Green2#FISH (490 nm, 20; 505 nm; 525 nm, 30; Chroma no. 49308 [custom]). For microscopy images in our ﬁgures and movies, uniform contrast settings were applied for each separate channel throughout the ﬁgure subpanels or movie, unless otherwise stated. Data analysis. (i) Analysis of time-lapse movies. Movie images were analyzed ﬁrst using the model created by deep learning with TensorFlow for automatic cell recognition (collaboration with the lab of Anxiao Jiang, Department of Computer Science & Engineering, Texas A&M University) and thenmanually checked using the program Schnitzcells (gift from Michael Elowitz, California Institute of Technology), in order to generate a cellular index. The numbers of phages attached on cell surfaces, as well as distances between phages on the cells with an MOI of 2 and cell lengths, were measured man-ually using the supporting tools of the NIS-Elements program. Cell fates and DNA behavior for each infected cell was recorded as well. All subsequent data analysis was performed in MATLAB. (ii)“Failed ”infections. We observed that, in some cases, adsorbed phages were seen on the cell surface, but the cell grew normally with neither lysis nor lysogeny detected, similar to phage linfection (41). These events were de ﬁned as “failed infections. ”In our experiments, the rate of failed infection was deﬁned as the number of failed infections divided by total number of infected cells with an MOI of 1. In the decision-making reporter system, the failed infection rate was 16.6% (99 of 596). In the SeqA-FP DNA visualization system, the failed infection rate was 21.6% (19 of 88). In the tetO/TetR-FP system, the failed infection frequency was 21.5% (31 of 144). A possible explanation for this phenomenon is the failure of the adsorbed phage to inject its DNA into the cell. (iii) “Dark ”infections. Similar to phage linfection, a fraction of the cells exhibited dark infections: cells without any observed infecting phages on the cell surface exhibiting lysis or lysogeny. With careful treatment of the sample, the dark infection rate was ;20% (120 cells at an MOI of 0 showing mVenus signal, compared to 596 cells at an MOI of 1), testing in the decision-making reporter system. The calcu- lation is based on the assumption that dark infections were mainly MOI = 1 events. We hypothesize that these dark infections are mainly due to the phages falling off after injecting their DNA into the host cell. Another possibility is that some cells may divide into two daughter cells during the 30-min incubation, resulting in an infected daughter cell without an observed infecting phage. Similarly, in the SeqA-FPDNA visualization system, the dark infection rate was 15.9% (14 cells at an MOI of 0 showing DNA signal, compared to 88 cells at an MOI of 1), and in the tetO/TetR-FP system, the dark infection rate was 22.9% (33 cells at an MOI of 0 showing DNA signal, compared to 144 cells at an MOI of 1). (iv) Analysis of DNA number and RNA number. Microscope images of the DNA injection in SeqA- FP or tetO/TetR-PF system and RNA FISH experiments were processed for cell recognition as described above. Then, the number of viral DNA molecules was analyzed via the procedure reported in reference 60. Basically, ﬂuorescent spots were ﬁrst identi ﬁed from the stacks of ﬂuorescence images, using the Spätzcells program.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  (iv) Analysis of DNA number and RNA number. Microscope images of the DNA injection in SeqA- FP or tetO/TetR-PF system and RNA FISH experiments were processed for cell recognition as described above. Then, the number of viral DNA molecules was analyzed via the procedure reported in reference 60. Basically, ﬂuorescent spots were ﬁrst identi ﬁed from the stacks of ﬂuorescence images, using the Spätzcells program. Then, false-positive spots, corresponding to the unspeci ﬁc binding of probes to non- target RNA, were identi ﬁed by comparison with the negative control and discarded. After that, the spot intensity corresponding to a single mRNA molecule was identi ﬁed by examining the single spot inten- sities in a low-expression sample (20 min), where individual mRNAs are spatially separable. Finally, theone-mRNA intensity value was used to convert the total spot intensity in cells from other samples to the number of target mRNA molecules. (v) Calculation of predicted intracellular DNA number due to failed and dark infection. The fre- quencies of failed and dark phage infections at an MOI of 1 were designated aand b, respectively, and the frequencies of failed and dark infection for an infected cell are a nand bn, where nis the number of failed or dark phages. Here, we only considered a value for nof 1, since the frequencies are low when n is.1. Thus, there are four situations: none failed /C2none dark, none failed /C2one dark, one failed /C2one dark, and one failed /C2none dark. The predicted number of injected DNA for an MOI of 1 is calculated as 12aðÞ /C212bðÞ12/C212aðÞ /C2b1a/C2b That for an MOI of 2 is 2/C212aðÞ2/C212bðÞ13/C212aðÞ2/C2b12/C22/C212aðÞ /C2a/C2b12/C2ð12aÞ/C2a/C2ð12bÞ That for an MOI of 3 is 3/C212aðÞ3/C212bðÞ14/C212aðÞ3/C2b13/C23/C2ð12aÞ2/C2a/C2b12/C23/C2ð12aÞ2/C2a/C2ð12bÞ That for an MOI of 4 is 4/C212aðÞ4/C212bðÞ15/C212aðÞ4/C2b14/C24/C2ð12aÞ3/C2a/C2b13/C24/C2ð12aÞ3/C2a/C2ð12bÞ That for an MOI of 5 is 5/C212aðÞ5/C212bðÞ16/C212aðÞ5/C2b15/C25/C2ð12aÞ4/C2a/C2b14/C25/C2ð12aÞ4/C2a/C2ð12bÞ From our measurements, in the SeqA-FP system we found that awas 21.6% and bwas 15.9%. Therefore, the sum probabilities of these four situations were 1, 0.95, 0.88, 0.79, and 0.70 for MOI from 1MOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 13', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  to 5, respectively. The predictions of DNA number were 0.94, 1.72, 2.38, 2.89, and 3.22, compared with the quanti ﬁed data (1.07, 1.70, 2.40, 2.57, and 3.80) using Spätzcells (60). In the tetO/TetR-FP system, a was 21.5% and bwas 22.9%. Therefore, the sum probabilities of these four situations were 1, 0.95, 0.88, 0.80, and 0.71 for MOI from 1 to 4, respectively. The predictions of DNA number were 1.01, 1.79, 2.45,and 3.21, compared with the quanti ﬁed data (0.93, 1.55, 2.28, and 3.30) using Spätzcells. SUPPLEMENTAL MATERIAL Supplemental material is available online only. FIG S1 , TIFﬁle, 1.3 MB. FIG S2 , TIFﬁle, 2.2 MB. FIG S3 , TIFﬁle, 1.6 MB. FIG S4 , TIFﬁle, 1.3 MB. FIG S5 , TIFﬁle, 0.9 MB. FIG S6 , TIFﬁle, 2.2 MB. FIG S7 , TIFﬁle, 1.5 MB. TABLE S1 , DOCX ﬁle, 0.02 MB. TABLE S2 , DOCX ﬁle, 0.02 MB. TABLE S3 , DOCX ﬁle, 0.02 MB. ACKNOWLEDGMENTS We thank Craig Kaplan, Ry Young and Jason Gill for generous advice. We thank members of the Zeng, Young, and Gill laboratories and Center for Phage Technology for technical assistance and supports. We are grateful to Craig Kaplan and Ry Young forcommenting on the earlier versions of the manuscript. We thank Karthik Chamakura, RyYoung, and Jason Gill for gifting strains. This research was in part supported by NIH grant R01GM107597, NSF grant MCB- 2013762, and Texas A&M University T3 grant 247097 to the Zeng laboratory. K.Z., Y.L., and L.Z. conceived the project, designed the experiments, and wrote the paper. K.Z. constructed strains, performed the bulk and single-cell infection experiments, and analyzed the data. K.P. contributed to the construction, single-cell infection experiments, and data analysis for the Lxc-overexpressing strain. H.D. contributed tothe construction, single-cell infection experiments, and data analysis for the C1operator strain. M.T. contributed to examination of the plaquing ef ﬁciency upon the infection of long cells compared to the normal-sized cells. J.G. performed the DNA FISH experiments. A.J. provided the program for automatic cell recognition. L.Z. supervised the project. We declare no con ﬂict of interest. REFERENCES 1. Balcazar JL. 2014. Bacteriophages as vehicles for antibiotic resistance genes in the environment. PLoS Pathog 10:e1004219. https://doi.org/10 .1371/journal.ppat.1004219 . 2. Billard-Pomares T, Fouteau S, Jacquet ME, Roche D, Barbe V, Castellanos M, Bouet JY, Cruveiller S, Medigue C, Blanco J, Clermont O, Denamur E, Branger C. 2014. Characterization of a P1-like bacteriophage carrying an SHV-2 extended-spectrum beta-lactamase from an Escherichia coli strain.Antimicrob Agents Chemother 58:6550 –6557. https://doi.org/10.1128/ AAC.03183-14 . 3. Brown-Jaque M, Calero-Caceres W, Muniesa M. 2015. Transfer of antibi- otic-resistance genes via phage-related mobile elements. Plasmid 79:1 –7. https://doi.org/10.1016/j.plasmid.2015.01.001 . 4. Shin J, Ko KS. 2015.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  2014. Characterization of a P1-like bacteriophage carrying an SHV-2 extended-spectrum beta-lactamase from an Escherichia coli strain.Antimicrob Agents Chemother 58:6550 –6557. https://doi.org/10.1128/ AAC.03183-14 . 3. Brown-Jaque M, Calero-Caceres W, Muniesa M. 2015. Transfer of antibi- otic-resistance genes via phage-related mobile elements. Plasmid 79:1 –7. https://doi.org/10.1016/j.plasmid.2015.01.001 . 4. Shin J, Ko KS. 2015. A plasmid bearing the bla(CTX-M-15) gene and phage P1-like sequences from a sequence type 11 Klebsiella pneumoniae iso- late. Antimicrob Agents Chemother 59:6608 –6610. https://doi.org/10 .1128/AAC.00265-15 . 5. Li R, Xie M, Lv J, Wai-Chi Chan E, Chen S. 2017. Complete genetic analysis of plasmids carrying mcr-1 and other resistance genes in an Escherichia coli isolate of animal origin. J Antimicrob Chemother 72:696 –699. https:// doi.org/10.1093/jac/dkw509 . 6. Yang L, Li W, Jiang GZ, Zhang WH, Ding HZ, Liu YH, Zeng ZL, Jiang HX. 2017. Characterization of a P1-like bacteriophage carrying CTX-M-27 inSalmonella spp. resistant to third generation cephalosporins isolated from pork in China. Sci Rep 7:40710. https://doi.org/10.1038/srep40710 . 7. Brown-Jaque M, Calero-Caceres W, Espinal P, Rodriguez-Navarro J, Miro E, Gonzalez-Lopez JJ, Cornejo T, Hurtado JC, Navarro F, Muniesa M. 2018. Antibiotic resistance genes in phage particles isolated from human faeces and induced from clinical bacterial isolates. Int J Antimicrob Agents 51:434 –442. https://doi.org/10.1016/j.ijantimicag.2017.11.014 . 8. Brown-Jaque M, Rodriguez Oyarzun L, Cornejo-Sanchez T, Martin-Gomez MT, Gartner S, de Gracia J, Rovira S, Alvarez A, Jofre J, Gonzalez-Lopez JJ,Muniesa M. 2018. Detection of bacteriophage particles containing antibi-otic resistance genes in the sputum of cystic ﬁbrosis patients. Front Micro- biol 9:856. https://doi.org/10.3389/fmicb.2018.00856 . 9. Venturini C, Zingali T, Wyrsch ER, Bowring B, Iredell J, Partridge SR, Djordjevic SP. 2019. Diversity of P1 phage-like elements in multidrug re- sistant Escherichia coli. Sci Rep 9:18861. https://doi.org/10.1038/s41598 -019-54895-4 . 10. Sweere JM, Van Belleghem JD, Ishak H, Bach MS, Popescu M, Sunkari V, Kaber G, Manasherob R, Suh GA, Cao X, de Vries CR, Lam DN, Marshall PL,Birukova M, Katznelson E, Lazzareschi DV, Balaji S, Keswani SG, Hawn TR,Secor PR, Bollyky PL. 2019. Bacteriophage trigger antiviral immunity andprevent clearance of bacterial infection. Science 363:eaat9691. https://doi .org/10.1126/science.aat9691 .Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 14', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  11. Hendrickx APA, Landman F, de Haan A, Borst D, Witteveen S, van Santen- Verheuvel MG, van der Heide HGJ, Schouls LM, Dutch C, Dutch CPE Sur- veillance Study Group. 2020. Plasmid diversity among genetically relatedKlebsiella pneumoniae blaKPC-2 and blaKPC-3 isolates collected in theDutch national surveillance. Sci Rep 10:16778. https://doi.org/10.1038/ s41598-020-73440-2 . 12. Kamal SM, Cimdins-Ahne A, Lee C, Li F, Martin-Rodriguez AJ, Seferbekova Z, Afasizhev R, Wami HT, Katikaridis P, Meins L, Lunsdorf H, Dobrindt U, Mogk A, Romling U. 2021. A recently isolated human commensal Esche-richia coli ST10 clone member mediates enhanced thermotolerance andtetrathionate respiration on a P1 phage-derived IncY plasmid. Mol Micro- biol 115:255 –271. https://doi.org/10.1111/mmi.14614 . 13. Bertani G. 1951. Studies on lysogenesis. I. The mode of phage liberation by lysogenic Escherichia coli. J Bacteriol 62:293 –300. https://doi.org/10 .1128/jb.62.3.293-300.1951 . 14. Kondo E, Mitsuhashi S. 1966. Drug resistance of enteric bacteria. VI. Intro- duction of bacteriophage P1CM into Salmonella typhi and formation ofPldCM and F-CM elements. J Bacteriol 91:1787 –1794. https://doi.org/10 .1128/jb.91.5.1787-1794.1966 . 15. Okada M, Watanabe T. 1968. Transduction with phage P1 in Salmonella typhimurium. Nature 218:185 –187. https://doi.org/10.1038/218185a0 . 16. Ornellas EP, Stocker BA. 1974. Relation of lipopolysaccharide character to P1 sensitivity in Salmonella typhimurium. Virology 60:491 –502. https:// doi.org/10.1016/0042-6822(74)90343-2 . 17. Murooka Y, Harada T. 1979. Expansion of the host range of coliphage P1 and gene transfer from enteric bacteria to other gram-negative bacteria. Appl Environ Microbiol 38:754 –757. https://doi.org/10.1128/aem.38.4.754 -757.1979 . 18. Amati P. 1962. Abortive infection of Pseudomonas aeruginosa and Serra- tia marcescens with coliphage P1. J Bacteriol 83:433 –434. https://doi.org/ 10.1128/jb.83.2.433-434.1962 . 19. Kaiser D, Dworkin M. 1975. Gene transfer to myxobacterium by Esche- richia coli phage P1. Science 187:653 –654. https://doi.org/10.1126/ science.803710 . 20. Yarmolinsky MB, Sternberg N. 1988. Bacteriophage P1, p 291 –438. In Calendar R (ed), The bacteriophages. Plenum Press, New York, NY. 21. Sandulache R, Prehm P, Kamp D. 1984. Cell wall receptor for bacterio- phage Mu G( 1). J Bacteriol 160:299 –303. https://doi.org/10.1128/jb.160.1 .299-303.1984 . 22. Calendar R, Calendar RL, Abedon ST. 2006. The Bacteriophages. Oxford University Press, New York, NY. 23.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Science 187:653 –654. https://doi.org/10.1126/ science.803710 . 20. Yarmolinsky MB, Sternberg N. 1988. Bacteriophage P1, p 291 –438. In Calendar R (ed), The bacteriophages. Plenum Press, New York, NY. 21. Sandulache R, Prehm P, Kamp D. 1984. Cell wall receptor for bacterio- phage Mu G( 1). J Bacteriol 160:299 –303. https://doi.org/10.1128/jb.160.1 .299-303.1984 . 22. Calendar R, Calendar RL, Abedon ST. 2006. The Bacteriophages. Oxford University Press, New York, NY. 23. Iida S, Streiff MB, Bickle TA, Arber W. 1987. Two DNA antirestriction systems of bacteriophage P1, darA, and darB: characterization of darA- phages. Virol-ogy 157:156 –166. https://doi.org/10.1016/0042-6822(87)90324-2 . 24. Streiff MB, Iida S, Bickle TA. 1987. Expression and proteolytic processing of the darA antirestriction gene product of bacteriophage P1. Virology157:167 –171. https://doi.org/10.1016/0042-6822(87)90325-4 . 25. Iida S, Hiestand-Nauer R, Sandmeier H, Lehnherr H, Arber W. 1998. Acces- sory genes in the darA operon of bacteriophage P1 affect antirestriction function, generalized transduction, head morphogenesis, and host cell lysis. Virology 251:49 –58.https://doi.org/10.1006/viro.1998.9405 . 26. Eliason JL, Sternberg N. 1987. Characterization of the binding sites of c1 repressor of bacteriophage P1. Evidence for multiple asymmetric sites. JMol Biol 198:281 –293. https://doi.org/10.1016/0022-2836(87)90313-5 . 27. Osborne FA, Stovall SR, Baumstark BR. 1989. The c1 genes of P1 and P7. Nucleic Acids Res 17:7671 –7680. https://doi.org/10.1093/nar/17.19.7671 . 28.Łobocka MB, Rose DJ, Plunkett G, Rusin M, Samojedny A, Lehnherr H, Yarmolinsky MB, Blattner FR. 2004. Genome of bacteriophage P1. J Bac-teriol 186:7032 –7068. https://doi.org/10.1128/JB.186.21.7032-7068.2004 . 29. Baumstark BR, Stovall SR, Bralley P. 1990. The ImmC region of phage P1 codes for a gene whose product promotes lytic growth. Virology 179: 217 –227. https://doi.org/10.1016/0042-6822(90)90291-x . 30. Heinzel T, Velleman M, Schuster H. 1990. The c1 repressor inactivator pro- tein coi of bacteriophage P1. Cloning and expression of coi and its inter- ference with c1 repressor function. J Biol Chem 265:17928 –17934. https:// doi.org/10.1016/S0021-9258(18)38252-8 . 31. Heinzel T, Velleman M, Schuster H. 1992. C1 repressor of phage P1 is inac- tivated by noncovalent binding of P1 Coi protein. J Biol Chem 267:4183 –4188. https://doi.org/10.1016/S0021-9258(19)50646-9 . 32. Schaefer TS, Hays JB. 1990.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  1990. The c1 repressor inactivator pro- tein coi of bacteriophage P1. Cloning and expression of coi and its inter- ference with c1 repressor function. J Biol Chem 265:17928 –17934. https:// doi.org/10.1016/S0021-9258(18)38252-8 . 31. Heinzel T, Velleman M, Schuster H. 1992. C1 repressor of phage P1 is inac- tivated by noncovalent binding of P1 Coi protein. J Biol Chem 267:4183 –4188. https://doi.org/10.1016/S0021-9258(19)50646-9 . 32. Schaefer TS, Hays JB. 1990. The bof gene of bacteriophage P1: DNA sequence and evidence for roles in regulation of phage c1 and ref genes. J Bacteriol 172:3269 –3277. https://doi.org/10.1128/jb.172.6.3269-3277.1990 .33. Velleman M, Heirich M, Gunther A, Schuster H. 1990. A bacteriophage P1- encoded modulator protein affects the P1 c1 repression system. J BiolChem 265:18511 –18517. https://doi.org/10.1016/S0021-9258(17)44781-8 . 34. Velleman M, Heinzel T, Schuster H. 1992. The Bof protein of bacterio- phage P1 exerts its modulating function by formation of a ternary com-plex with operator DNA and C1 repressor. J Biol Chem 267:12174 –12181. https://doi.org/10.1016/S0021-9258(19)49820-7 . 35. Heinrich J, Velleman M, Schuster H. 1995. The tripartite immunity system of phages P1 and P7. FEMS Microbiol Rev 17:121 –126. https://doi.org/10 .1111/j.1574-6976.1995.tb00193.x . 36. Dreiseikelmann B, Velleman M, Schuster H. 1988. The c1 repressor of bacte- riophage P1. Isolation and characterization of the repressor protein. J BiolChem 263:1391 –1397. https://doi.org/10.1016/S0021-9258(19)57316-1 . 37. Citron M, Schuster H. 1992. The c4 repressor of bacteriophage P1 is a processed 77 base antisense RNA. Nucleic Acids Res 20:3085 –3090. https://doi.org/10.1093/nar/20.12.3085 . 38. Heinrich J, Citron M, Gunther A, Schuster H. 1994. Second-site suppres- sors of the bacteriophage P1 virs mutant reveal the interdependence ofthe c4, icd, and ant genes in the P1 immI operon. J Bacteriol 176:4931 –4936. https://doi.org/10.1128/jb.176.16.4931-4936.1994 . 39. Kourilsky P. 1973. Lysogenization by bacteriophage lambda. I. Multiple infection and the lysogenic response. Mol Gen Genet 122:183 –195. https://doi.org/10.1007/BF00435190 . 40. Kourilsky P, Knapp A. 1974. Lysogenization by bacteriophage lambda. III. Multiplicity dependent phenomena occuring upon infection by lambda.Biochimie 56:1517 –1523. 41. Zeng L, Skinner SO, Zong C, Sippy J, Feiss M, Golding I. 2010. Decision making at a subcellular level determines the outcome of bacteriophageinfection. Cell 141:682 –691. https://doi.org/10.1016/j.cell.2010.03.034 . 42. Silpe JE, Bassler BL.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  plus 10 mgm l21DAPI (4 9,6-diamidino-2-phenylindole) and incubated for 10 min at room temperature. This suspension was then pelleted and resuspended in 2 /C2SSC. The sample was then ready for imaging. Two microliters of the cell suspension was pipetted onto the center of a 24- by 50-mm coverslip. A 1.5% PBS agarose pad was laid slowly on top of the cell suspension droplet with a razor blade. The padwas covered with an 18- by 18-mm coverslip. Under the microscope, images were taken in the phase-contrast (100 ms), Cy5 (700 ms, c1), Cy3 (1 s, coi), green ﬂuorescent protein (GFP) (2 s, lxc), and DAPI (30 ms, DAPI) channels at different stage positions. DNA FISH. For DNA FISH for P1 DNA replication detection, probes were produced by PCR amplifying ;3 kb of P1 DNA (dna ﬁsh-P1-for and dna ﬁsh-P1-rev primer pair, listed in Table S2), using a phage lysate as the template, and treating the puri ﬁed PCR product with a PromoFluor500-dUTP nick translation kit (PromoCell) to generate DNA-PromoFluor500 fragments ranging from 100 to 500 bp. To perform DNA FISH on infection samples, the overnight host cells were diluted 100-fold into 50 ml of fresh LBM and grown at 37°C to an OD 600of;0.4. Then CaCl2was added to 5 mM. After that, a 5-ml aliquot of cells was separated as a negative control without phages and mixed with 550 ml of 37% formaldehyde forﬁxation. The rest of the cell culture was mixed with the puri ﬁed P1LZ1856 to reach an MOI of 0.2, fol- lowed by incubation at 30°C. At each time point (30, 40, 50, and 60 min) during incubation, 5 ml of the mix- ture was transferred to a 15-ml centrifuge tube with 550 ml of 37% formaldehyde. This tube was left to shake on a nutator for 30 min and then centrifuged at 4,000 /C2gfor 3 min to pellet the cells. Details of ﬁxation, permeabilization, and hybridization steps are detailed in reference 76. Brie ﬂy, the ﬁxed cells were washed with 1 ml of ice-cold 1 /C2PBS three times and resuspended in 1 ml of GTE solution (50 mM glucose, 20 mM Tris-HCl [pH 7.5], 10 mM EDTA). For the control sample, three separate 500- mla l i - quots of the cell suspension were then mixed with 10 mlo f0 . 0 1 mgml21lysozyme solution and incubated at room temperature for 2, 4, and 6 min, followed by three washes with GTE, and the cells were pelleted viacentrifugation at 10,000 /C2gfor 30 s. The cells were then resuspended in ;150 ml of GTE. For each control sample, 1 ml of the cells was deposited onto a PBS agarose pad and imaged. The lysozyme treatment time yielding ;90 to 95% intact cells ( ;1 to 5% lysed cells) represents the optimal treatment time for the sam- ples. The actual time point samples, from the initial GTE wash, were then processed as the control was,using the optimal lysozyme time. For each time point, 10 ml of cells were deposited onto poly- L-lysine- coated large coverslips (24 by 50 mm) and then covered with a smaller, normal coverslip (22 by 22 mm).The coverslips were then immersed in 1 /C2PBS, and the smaller coverslip was removed, leaving only the sample coverslip. The cells were then dehydrated by immersing the coverslip in increasing concentrationsof ethanol (70, 90, and100%). Samples were then ready for hybridization.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The lysozyme treatment time yielding ;90 to 95% intact cells ( ;1 to 5% lysed cells) represents the optimal treatment time for the sam- ples. The actual time point samples, from the initial GTE wash, were then processed as the control was,using the optimal lysozyme time. For each time point, 10 ml of cells were deposited onto poly- L-lysine- coated large coverslips (24 by 50 mm) and then covered with a smaller, normal coverslip (22 by 22 mm).The coverslips were then immersed in 1 /C2PBS, and the smaller coverslip was removed, leaving only the sample coverslip. The cells were then dehydrated by immersing the coverslip in increasing concentrationsof ethanol (70, 90, and100%). Samples were then ready for hybridization. For each sample, approximately 160 mg of the probe mixture was combined with 10 ml of hybridiza- tion solution (50% formamide, 10% dextran sulfate, 50 mM NaPO4[pH 7], 2 /C2SSC). The double-stranded- DNA (dsDNA) probes were denatured at 75°C in a thermocycler and then placed on ice. Ten microlitersof the denatured probe mixture was then deposited onto the center of the sample on the coverslip andoverlaid with a small coverslip (22 by 22 mm). The small coverslip was then sealed with nail polish, form-ing a sample chamber. The chambers were incubated at 80°C for 5 min to denature the cellular DNAand then placed on Kimwipes over ice for 5 min. The chambers were then incubated in a 37°C incubatorovernight to complete hybridization. The next day, the chambers were immersed in 2 /C2SSC until the smaller coverslip dislodged. The remaining coverslips were soaked in wash solution (2 /C2SSC, 50% form- amide) for 20 min at 37°C twice. The coverslips were then washed with SSC in a series of increasing con-centrations (1 /C2,2/C2, and 4 /C2), each for 5 min at room temperature. A DAPI solution was then made by mixing 1 mlo f1 0m gm l21DAPI with 1 ml of 4 /C2SSC. For each sample, 500 ml of the DAPI solution was added over the sample, covering it, and incubated for 5 min at room temperature. After the coverslip was dried, 10 mlo f2 /C2SSC was added over the sample and overlaid with a small coverslip (22 by 22 mm). The samples were then imaged. Under the microscope, images were taken in phase-contrast (100 ms), GFP (300 ms), and DAPI (30 ms) channels at different stage positions. qPCR. The overnight host cells were diluted 100-fold in fresh LBM and grown at 37°C. During cell growth, methanol was kept at 220°C for cell ﬁxation. When the cell culture reached an OD600of;0.4, CaCl2was added to a ﬁnal concentration of 5 mM. We then transferred 2 ml of cells to a 50-ml centrifuge tube and mixed them with 2 ml ice-cold methanol as the negative-control sample. The rest of the cellculture was mixed with puri ﬁed P1LZ1856 to reach an MOI of ;1 and incubated at 30°C (0 min). For each time point (0, 10, 20, 30, 40, 50, 60, 70, and 80 min), 2 ml of the reaction was aliquoted to 50-mltubes, vortexed for 10 s, and then centrifuged at 4,000 /C2gfor 3 min. After removing the supernatant, we resuspended the cell pellet with 2 ml fresh ice-cold LB solution, added 2 ml ice-cold methanol, andvortexed it for another 10 s. The mixture was centrifuged at 4,000 /C2gfor 3 min.', 'page_label: 12 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  The rest of the cellculture was mixed with puri ﬁed P1LZ1856 to reach an MOI of ;1 and incubated at 30°C (0 min). For each time point (0, 10, 20, 30, 40, 50, 60, 70, and 80 min), 2 ml of the reaction was aliquoted to 50-mltubes, vortexed for 10 s, and then centrifuged at 4,000 /C2gfor 3 min. After removing the supernatant, we resuspended the cell pellet with 2 ml fresh ice-cold LB solution, added 2 ml ice-cold methanol, andvortexed it for another 10 s. The mixture was centrifuged at 4,000 /C2gfor 3 min. After removal of the su- pernatant, the cell pellet was kept at 220°C until DNA extraction using an UltraClean microbial DNA iso- lation kit (Mo Bio Laboratories no. 12224-50). The DNAs were then diluted and used for qPCR with pri-mers targeting the phage DNA (qPCR-P1DNA-for/rev). The E. coli DNA number was used as a reference using primers targeting the dxsgene (dsx-for/rev) (77). Ampli ﬁcation was done using SYBR green PCR master mix (Applied Biosystems no. 4309155) with a 250 nM concentration of each primer (Table S2). Microscopy imaging. Imaging was performed on a Nikon Eclipse Ti inverted epi ﬂuorescence micro- scope using a 100 /C2objective (Plan Fluo; numerical aperture [NA], 1.40; oil immersion) with a 2.5 /C2TV relay lens, within the cage of an incubator (InVivo Scienti ﬁc) at 30°C, and acquired using a cooled elec- tron-multiplying charge-coupled device (EMCCD) camera (IXON 897; Andor, Belfast, UK). Cells wereimaged under phase contrast and the following ﬂuorescent ﬁlter cubes (parameters are given as excita- tion ﬁlter wavelength and bandwidth; dichroic beam splitter wavelength; emission ﬁlter wavelength and bandwidth; company and product number): CFP (436 nm, 20; 455 nm; 480 nm, 40; Nikon, 96361),Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 12', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  YFP (539 nm, 21; 556 nm; 576 nm, 31; Chroma no. 49309), far red (592 nm, 21; 610 nm; 630 nm, 30; Chroma no. 49310), DAPI (350 nm, 50; 400 nm; 460 nm, 50; Nikon no. 96310), Cy3 (545 nm, 30; 570 nm; 610 nm, 75; Nikon no. 96323), Cy5 (615 nm, 70; 660 nm; 700 nm, 75; Nikon no. 96366), and Green2#FISH (490 nm, 20; 505 nm; 525 nm, 30; Chroma no. 49308 [custom]). For microscopy images in our ﬁgures and movies, uniform contrast settings were applied for each separate channel throughout the ﬁgure subpanels or movie, unless otherwise stated. Data analysis. (i) Analysis of time-lapse movies. Movie images were analyzed ﬁrst using the model created by deep learning with TensorFlow for automatic cell recognition (collaboration with the lab of Anxiao Jiang, Department of Computer Science & Engineering, Texas A&M University) and thenmanually checked using the program Schnitzcells (gift from Michael Elowitz, California Institute of Technology), in order to generate a cellular index. The numbers of phages attached on cell surfaces, as well as distances between phages on the cells with an MOI of 2 and cell lengths, were measured man-ually using the supporting tools of the NIS-Elements program. Cell fates and DNA behavior for each infected cell was recorded as well. All subsequent data analysis was performed in MATLAB. (ii)“Failed ”infections. We observed that, in some cases, adsorbed phages were seen on the cell surface, but the cell grew normally with neither lysis nor lysogeny detected, similar to phage linfection (41). These events were de ﬁned as “failed infections. ”In our experiments, the rate of failed infection was deﬁned as the number of failed infections divided by total number of infected cells with an MOI of 1. In the decision-making reporter system, the failed infection rate was 16.6% (99 of 596). In the SeqA-FP DNA visualization system, the failed infection rate was 21.6% (19 of 88). In the tetO/TetR-FP system, the failed infection frequency was 21.5% (31 of 144). A possible explanation for this phenomenon is the failure of the adsorbed phage to inject its DNA into the cell. (iii) “Dark ”infections. Similar to phage linfection, a fraction of the cells exhibited dark infections: cells without any observed infecting phages on the cell surface exhibiting lysis or lysogeny. With careful treatment of the sample, the dark infection rate was ;20% (120 cells at an MOI of 0 showing mVenus signal, compared to 596 cells at an MOI of 1), testing in the decision-making reporter system. The calcu- lation is based on the assumption that dark infections were mainly MOI = 1 events. We hypothesize that these dark infections are mainly due to the phages falling off after injecting their DNA into the host cell. Another possibility is that some cells may divide into two daughter cells during the 30-min incubation, resulting in an infected daughter cell without an observed infecting phage. Similarly, in the SeqA-FPDNA visualization system, the dark infection rate was 15.9% (14 cells at an MOI of 0 showing DNA signal, compared to 88 cells at an MOI of 1), and in the tetO/TetR-FP system, the dark infection rate was 22.9% (33 cells at an MOI of 0 showing DNA signal, compared to 144 cells at an MOI of 1). (iv) Analysis of DNA number and RNA number. Microscope images of the DNA injection in SeqA- FP or tetO/TetR-PF system and RNA FISH experiments were processed for cell recognition as described above. Then, the number of viral DNA molecules was analyzed via the procedure reported in reference 60. Basically, ﬂuorescent spots were ﬁrst identi ﬁed from the stacks of ﬂuorescence images, using the Spätzcells program.', 'page_label: 13 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  (iv) Analysis of DNA number and RNA number. Microscope images of the DNA injection in SeqA- FP or tetO/TetR-PF system and RNA FISH experiments were processed for cell recognition as described above. Then, the number of viral DNA molecules was analyzed via the procedure reported in reference 60. Basically, ﬂuorescent spots were ﬁrst identi ﬁed from the stacks of ﬂuorescence images, using the Spätzcells program. Then, false-positive spots, corresponding to the unspeci ﬁc binding of probes to non- target RNA, were identi ﬁed by comparison with the negative control and discarded. After that, the spot intensity corresponding to a single mRNA molecule was identi ﬁed by examining the single spot inten- sities in a low-expression sample (20 min), where individual mRNAs are spatially separable. Finally, theone-mRNA intensity value was used to convert the total spot intensity in cells from other samples to the number of target mRNA molecules. (v) Calculation of predicted intracellular DNA number due to failed and dark infection. The fre- quencies of failed and dark phage infections at an MOI of 1 were designated aand b, respectively, and the frequencies of failed and dark infection for an infected cell are a nand bn, where nis the number of failed or dark phages. Here, we only considered a value for nof 1, since the frequencies are low when n is.1. Thus, there are four situations: none failed /C2none dark, none failed /C2one dark, one failed /C2one dark, and one failed /C2none dark. The predicted number of injected DNA for an MOI of 1 is calculated as 12aðÞ /C212bðÞ12/C212aðÞ /C2b1a/C2b That for an MOI of 2 is 2/C212aðÞ2/C212bðÞ13/C212aðÞ2/C2b12/C22/C212aðÞ /C2a/C2b12/C2ð12aÞ/C2a/C2ð12bÞ That for an MOI of 3 is 3/C212aðÞ3/C212bðÞ14/C212aðÞ3/C2b13/C23/C2ð12aÞ2/C2a/C2b12/C23/C2ð12aÞ2/C2a/C2ð12bÞ That for an MOI of 4 is 4/C212aðÞ4/C212bðÞ15/C212aðÞ4/C2b14/C24/C2ð12aÞ3/C2a/C2b13/C24/C2ð12aÞ3/C2a/C2ð12bÞ That for an MOI of 5 is 5/C212aðÞ5/C212bðÞ16/C212aðÞ5/C2b15/C25/C2ð12aÞ4/C2a/C2b14/C25/C2ð12aÞ4/C2a/C2ð12bÞ From our measurements, in the SeqA-FP system we found that awas 21.6% and bwas 15.9%. Therefore, the sum probabilities of these four situations were 1, 0.95, 0.88, 0.79, and 0.70 for MOI from 1MOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 13', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  to 5, respectively. The predictions of DNA number were 0.94, 1.72, 2.38, 2.89, and 3.22, compared with the quanti ﬁed data (1.07, 1.70, 2.40, 2.57, and 3.80) using Spätzcells (60). In the tetO/TetR-FP system, a was 21.5% and bwas 22.9%. Therefore, the sum probabilities of these four situations were 1, 0.95, 0.88, 0.80, and 0.71 for MOI from 1 to 4, respectively. The predictions of DNA number were 1.01, 1.79, 2.45,and 3.21, compared with the quanti ﬁed data (0.93, 1.55, 2.28, and 3.30) using Spätzcells. SUPPLEMENTAL MATERIAL Supplemental material is available online only. FIG S1 , TIFﬁle, 1.3 MB. FIG S2 , TIFﬁle, 2.2 MB. FIG S3 , TIFﬁle, 1.6 MB. FIG S4 , TIFﬁle, 1.3 MB. FIG S5 , TIFﬁle, 0.9 MB. FIG S6 , TIFﬁle, 2.2 MB. FIG S7 , TIFﬁle, 1.5 MB. TABLE S1 , DOCX ﬁle, 0.02 MB. TABLE S2 , DOCX ﬁle, 0.02 MB. TABLE S3 , DOCX ﬁle, 0.02 MB. ACKNOWLEDGMENTS We thank Craig Kaplan, Ry Young and Jason Gill for generous advice. We thank members of the Zeng, Young, and Gill laboratories and Center for Phage Technology for technical assistance and supports. We are grateful to Craig Kaplan and Ry Young forcommenting on the earlier versions of the manuscript. We thank Karthik Chamakura, RyYoung, and Jason Gill for gifting strains. This research was in part supported by NIH grant R01GM107597, NSF grant MCB- 2013762, and Texas A&M University T3 grant 247097 to the Zeng laboratory. K.Z., Y.L., and L.Z. conceived the project, designed the experiments, and wrote the paper. K.Z. constructed strains, performed the bulk and single-cell infection experiments, and analyzed the data. K.P. contributed to the construction, single-cell infection experiments, and data analysis for the Lxc-overexpressing strain. H.D. contributed tothe construction, single-cell infection experiments, and data analysis for the C1operator strain. M.T. contributed to examination of the plaquing ef ﬁciency upon the infection of long cells compared to the normal-sized cells. J.G. performed the DNA FISH experiments. A.J. provided the program for automatic cell recognition. L.Z. supervised the project. We declare no con ﬂict of interest. REFERENCES 1. Balcazar JL. 2014. Bacteriophages as vehicles for antibiotic resistance genes in the environment. PLoS Pathog 10:e1004219. https://doi.org/10 .1371/journal.ppat.1004219 . 2. Billard-Pomares T, Fouteau S, Jacquet ME, Roche D, Barbe V, Castellanos M, Bouet JY, Cruveiller S, Medigue C, Blanco J, Clermont O, Denamur E, Branger C. 2014. Characterization of a P1-like bacteriophage carrying an SHV-2 extended-spectrum beta-lactamase from an Escherichia coli strain.Antimicrob Agents Chemother 58:6550 –6557. https://doi.org/10.1128/ AAC.03183-14 . 3. Brown-Jaque M, Calero-Caceres W, Muniesa M. 2015. Transfer of antibi- otic-resistance genes via phage-related mobile elements. Plasmid 79:1 –7. https://doi.org/10.1016/j.plasmid.2015.01.001 . 4. Shin J, Ko KS. 2015.', 'page_label: 14 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  2014. Characterization of a P1-like bacteriophage carrying an SHV-2 extended-spectrum beta-lactamase from an Escherichia coli strain.Antimicrob Agents Chemother 58:6550 –6557. https://doi.org/10.1128/ AAC.03183-14 . 3. Brown-Jaque M, Calero-Caceres W, Muniesa M. 2015. Transfer of antibi- otic-resistance genes via phage-related mobile elements. Plasmid 79:1 –7. https://doi.org/10.1016/j.plasmid.2015.01.001 . 4. Shin J, Ko KS. 2015. A plasmid bearing the bla(CTX-M-15) gene and phage P1-like sequences from a sequence type 11 Klebsiella pneumoniae iso- late. Antimicrob Agents Chemother 59:6608 –6610. https://doi.org/10 .1128/AAC.00265-15 . 5. Li R, Xie M, Lv J, Wai-Chi Chan E, Chen S. 2017. Complete genetic analysis of plasmids carrying mcr-1 and other resistance genes in an Escherichia coli isolate of animal origin. J Antimicrob Chemother 72:696 –699. https:// doi.org/10.1093/jac/dkw509 . 6. Yang L, Li W, Jiang GZ, Zhang WH, Ding HZ, Liu YH, Zeng ZL, Jiang HX. 2017. Characterization of a P1-like bacteriophage carrying CTX-M-27 inSalmonella spp. resistant to third generation cephalosporins isolated from pork in China. Sci Rep 7:40710. https://doi.org/10.1038/srep40710 . 7. Brown-Jaque M, Calero-Caceres W, Espinal P, Rodriguez-Navarro J, Miro E, Gonzalez-Lopez JJ, Cornejo T, Hurtado JC, Navarro F, Muniesa M. 2018. Antibiotic resistance genes in phage particles isolated from human faeces and induced from clinical bacterial isolates. Int J Antimicrob Agents 51:434 –442. https://doi.org/10.1016/j.ijantimicag.2017.11.014 . 8. Brown-Jaque M, Rodriguez Oyarzun L, Cornejo-Sanchez T, Martin-Gomez MT, Gartner S, de Gracia J, Rovira S, Alvarez A, Jofre J, Gonzalez-Lopez JJ,Muniesa M. 2018. Detection of bacteriophage particles containing antibi-otic resistance genes in the sputum of cystic ﬁbrosis patients. Front Micro- biol 9:856. https://doi.org/10.3389/fmicb.2018.00856 . 9. Venturini C, Zingali T, Wyrsch ER, Bowring B, Iredell J, Partridge SR, Djordjevic SP. 2019. Diversity of P1 phage-like elements in multidrug re- sistant Escherichia coli. Sci Rep 9:18861. https://doi.org/10.1038/s41598 -019-54895-4 . 10. Sweere JM, Van Belleghem JD, Ishak H, Bach MS, Popescu M, Sunkari V, Kaber G, Manasherob R, Suh GA, Cao X, de Vries CR, Lam DN, Marshall PL,Birukova M, Katznelson E, Lazzareschi DV, Balaji S, Keswani SG, Hawn TR,Secor PR, Bollyky PL. 2019. Bacteriophage trigger antiviral immunity andprevent clearance of bacterial infection. Science 363:eaat9691. https://doi .org/10.1126/science.aat9691 .Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 14', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  11. Hendrickx APA, Landman F, de Haan A, Borst D, Witteveen S, van Santen- Verheuvel MG, van der Heide HGJ, Schouls LM, Dutch C, Dutch CPE Sur- veillance Study Group. 2020. Plasmid diversity among genetically relatedKlebsiella pneumoniae blaKPC-2 and blaKPC-3 isolates collected in theDutch national surveillance. Sci Rep 10:16778. https://doi.org/10.1038/ s41598-020-73440-2 . 12. Kamal SM, Cimdins-Ahne A, Lee C, Li F, Martin-Rodriguez AJ, Seferbekova Z, Afasizhev R, Wami HT, Katikaridis P, Meins L, Lunsdorf H, Dobrindt U, Mogk A, Romling U. 2021. A recently isolated human commensal Esche-richia coli ST10 clone member mediates enhanced thermotolerance andtetrathionate respiration on a P1 phage-derived IncY plasmid. Mol Micro- biol 115:255 –271. https://doi.org/10.1111/mmi.14614 . 13. Bertani G. 1951. Studies on lysogenesis. I. The mode of phage liberation by lysogenic Escherichia coli. J Bacteriol 62:293 –300. https://doi.org/10 .1128/jb.62.3.293-300.1951 . 14. Kondo E, Mitsuhashi S. 1966. Drug resistance of enteric bacteria. VI. Intro- duction of bacteriophage P1CM into Salmonella typhi and formation ofPldCM and F-CM elements. J Bacteriol 91:1787 –1794. https://doi.org/10 .1128/jb.91.5.1787-1794.1966 . 15. Okada M, Watanabe T. 1968. Transduction with phage P1 in Salmonella typhimurium. Nature 218:185 –187. https://doi.org/10.1038/218185a0 . 16. Ornellas EP, Stocker BA. 1974. Relation of lipopolysaccharide character to P1 sensitivity in Salmonella typhimurium. Virology 60:491 –502. https:// doi.org/10.1016/0042-6822(74)90343-2 . 17. Murooka Y, Harada T. 1979. Expansion of the host range of coliphage P1 and gene transfer from enteric bacteria to other gram-negative bacteria. Appl Environ Microbiol 38:754 –757. https://doi.org/10.1128/aem.38.4.754 -757.1979 . 18. Amati P. 1962. Abortive infection of Pseudomonas aeruginosa and Serra- tia marcescens with coliphage P1. J Bacteriol 83:433 –434. https://doi.org/ 10.1128/jb.83.2.433-434.1962 . 19. Kaiser D, Dworkin M. 1975. Gene transfer to myxobacterium by Esche- richia coli phage P1. Science 187:653 –654. https://doi.org/10.1126/ science.803710 . 20. Yarmolinsky MB, Sternberg N. 1988. Bacteriophage P1, p 291 –438. In Calendar R (ed), The bacteriophages. Plenum Press, New York, NY. 21. Sandulache R, Prehm P, Kamp D. 1984. Cell wall receptor for bacterio- phage Mu G( 1). J Bacteriol 160:299 –303. https://doi.org/10.1128/jb.160.1 .299-303.1984 . 22. Calendar R, Calendar RL, Abedon ST. 2006. The Bacteriophages. Oxford University Press, New York, NY. 23.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Science 187:653 –654. https://doi.org/10.1126/ science.803710 . 20. Yarmolinsky MB, Sternberg N. 1988. Bacteriophage P1, p 291 –438. In Calendar R (ed), The bacteriophages. Plenum Press, New York, NY. 21. Sandulache R, Prehm P, Kamp D. 1984. Cell wall receptor for bacterio- phage Mu G( 1). J Bacteriol 160:299 –303. https://doi.org/10.1128/jb.160.1 .299-303.1984 . 22. Calendar R, Calendar RL, Abedon ST. 2006. The Bacteriophages. Oxford University Press, New York, NY. 23. Iida S, Streiff MB, Bickle TA, Arber W. 1987. Two DNA antirestriction systems of bacteriophage P1, darA, and darB: characterization of darA- phages. Virol-ogy 157:156 –166. https://doi.org/10.1016/0042-6822(87)90324-2 . 24. Streiff MB, Iida S, Bickle TA. 1987. Expression and proteolytic processing of the darA antirestriction gene product of bacteriophage P1. Virology157:167 –171. https://doi.org/10.1016/0042-6822(87)90325-4 . 25. Iida S, Hiestand-Nauer R, Sandmeier H, Lehnherr H, Arber W. 1998. Acces- sory genes in the darA operon of bacteriophage P1 affect antirestriction function, generalized transduction, head morphogenesis, and host cell lysis. Virology 251:49 –58.https://doi.org/10.1006/viro.1998.9405 . 26. Eliason JL, Sternberg N. 1987. Characterization of the binding sites of c1 repressor of bacteriophage P1. Evidence for multiple asymmetric sites. JMol Biol 198:281 –293. https://doi.org/10.1016/0022-2836(87)90313-5 . 27. Osborne FA, Stovall SR, Baumstark BR. 1989. The c1 genes of P1 and P7. Nucleic Acids Res 17:7671 –7680. https://doi.org/10.1093/nar/17.19.7671 . 28.Łobocka MB, Rose DJ, Plunkett G, Rusin M, Samojedny A, Lehnherr H, Yarmolinsky MB, Blattner FR. 2004. Genome of bacteriophage P1. J Bac-teriol 186:7032 –7068. https://doi.org/10.1128/JB.186.21.7032-7068.2004 . 29. Baumstark BR, Stovall SR, Bralley P. 1990. The ImmC region of phage P1 codes for a gene whose product promotes lytic growth. Virology 179: 217 –227. https://doi.org/10.1016/0042-6822(90)90291-x . 30. Heinzel T, Velleman M, Schuster H. 1990. The c1 repressor inactivator pro- tein coi of bacteriophage P1. Cloning and expression of coi and its inter- ference with c1 repressor function. J Biol Chem 265:17928 –17934. https:// doi.org/10.1016/S0021-9258(18)38252-8 . 31. Heinzel T, Velleman M, Schuster H. 1992. C1 repressor of phage P1 is inac- tivated by noncovalent binding of P1 Coi protein. J Biol Chem 267:4183 –4188. https://doi.org/10.1016/S0021-9258(19)50646-9 . 32. Schaefer TS, Hays JB. 1990.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  1990. The c1 repressor inactivator pro- tein coi of bacteriophage P1. Cloning and expression of coi and its inter- ference with c1 repressor function. J Biol Chem 265:17928 –17934. https:// doi.org/10.1016/S0021-9258(18)38252-8 . 31. Heinzel T, Velleman M, Schuster H. 1992. C1 repressor of phage P1 is inac- tivated by noncovalent binding of P1 Coi protein. J Biol Chem 267:4183 –4188. https://doi.org/10.1016/S0021-9258(19)50646-9 . 32. Schaefer TS, Hays JB. 1990. The bof gene of bacteriophage P1: DNA sequence and evidence for roles in regulation of phage c1 and ref genes. J Bacteriol 172:3269 –3277. https://doi.org/10.1128/jb.172.6.3269-3277.1990 .33. Velleman M, Heirich M, Gunther A, Schuster H. 1990. A bacteriophage P1- encoded modulator protein affects the P1 c1 repression system. J BiolChem 265:18511 –18517. https://doi.org/10.1016/S0021-9258(17)44781-8 . 34. Velleman M, Heinzel T, Schuster H. 1992. The Bof protein of bacterio- phage P1 exerts its modulating function by formation of a ternary com-plex with operator DNA and C1 repressor. J Biol Chem 267:12174 –12181. https://doi.org/10.1016/S0021-9258(19)49820-7 . 35. Heinrich J, Velleman M, Schuster H. 1995. The tripartite immunity system of phages P1 and P7. FEMS Microbiol Rev 17:121 –126. https://doi.org/10 .1111/j.1574-6976.1995.tb00193.x . 36. Dreiseikelmann B, Velleman M, Schuster H. 1988. The c1 repressor of bacte- riophage P1. Isolation and characterization of the repressor protein. J BiolChem 263:1391 –1397. https://doi.org/10.1016/S0021-9258(19)57316-1 . 37. Citron M, Schuster H. 1992. The c4 repressor of bacteriophage P1 is a processed 77 base antisense RNA. Nucleic Acids Res 20:3085 –3090. https://doi.org/10.1093/nar/20.12.3085 . 38. Heinrich J, Citron M, Gunther A, Schuster H. 1994. Second-site suppres- sors of the bacteriophage P1 virs mutant reveal the interdependence ofthe c4, icd, and ant genes in the P1 immI operon. J Bacteriol 176:4931 –4936. https://doi.org/10.1128/jb.176.16.4931-4936.1994 . 39. Kourilsky P. 1973. Lysogenization by bacteriophage lambda. I. Multiple infection and the lysogenic response. Mol Gen Genet 122:183 –195. https://doi.org/10.1007/BF00435190 . 40. Kourilsky P, Knapp A. 1974. Lysogenization by bacteriophage lambda. III. Multiplicity dependent phenomena occuring upon infection by lambda.Biochimie 56:1517 –1523. 41. Zeng L, Skinner SO, Zong C, Sippy J, Feiss M, Golding I. 2010. Decision making at a subcellular level determines the outcome of bacteriophageinfection. Cell 141:682 –691. https://doi.org/10.1016/j.cell.2010.03.034 . 42. Silpe JE, Bassler BL.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:09 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'73'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992885'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'426ms'), (b'x-request-id', b'485293f88bddb3c2ee64e1f53be2b260'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249ce8d865d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:09 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'73'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'992885'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'426ms'), (b'x-request-id', b'485293f88bddb3c2ee64e1f53be2b260'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249ce8d865d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Lysogenization by bacteriophage lambda. I. Multiple infection and the lysogenic response. Mol Gen Genet 122:183 –195. https://doi.org/10.1007/BF00435190 . 40. Kourilsky P, Knapp A. 1974. Lysogenization by bacteriophage lambda. III. Multiplicity dependent phenomena occuring upon infection by lambda.Biochimie 56:1517 –1523. 41. Zeng L, Skinner SO, Zong C, Sippy J, Feiss M, Golding I. 2010. Decision making at a subcellular level determines the outcome of bacteriophageinfection. Cell 141:682 –691. https://doi.org/10.1016/j.cell.2010.03.034 . 42. Silpe JE, Bassler BL. 2019. Phage-encoded LuxR-type receptors responsive to host-produced bacterial quorum-sensing autoinducers. mBio 10:e00638-19. https://doi.org/10.1128/mBio.00638-19 . 43. Silpe JE, Bassler BL. 2019. A host-produced quorum-sensing autoinducer controls a phage lysis-lysogeny decision. Cell 176:268 –280.E13. https:// doi.org/10.1016/j.cell.2018.10.059 . 44. Erez Z, Steinberger-Levy I, Shamir M, Doron S, Stokar-Avihail A, Peleg Y, Melamed S, Leavitt A, Savidor A, Albeck S, Amitai G, Sorek R. 2017. Com-munication between viruses guides lysis-lysogeny decisions. Nature 541:488 –493. https://doi.org/10.1038/nature21049 . 45. Stokar-Avihail A, Tal N, Erez Z, Lopatina A, Sorek R. 2019. Widespread utiliza- tion of peptide communication in phages infecting soil and pathogenicbacteria. Cell Host Microbe 25:746 –755.E5. https://doi.org/10.1016/j.chom .2019.03.017 . 46. Bertani G, Nice SJ. 1954. Studies on lysogenesis. II. The effect of tempera- ture on the lysogenization of Shigella dysenteriae with phage P1. J Bac-teriol 67:202 –209. https://doi.org/10.1128/jb.67.2.202-209.1954 . 47. Rosner JL. 1972. Formation, induction, and curing of bacteriophage P1 lyso- gens. Virology 48:679 –689. https://doi.org/10.1016/0042-6822(72)90152-3 . 48. Hadi SM, Bachi B, Iida S, Bickle TA. 1983. DNA restriction –modi ﬁcation e n z y m e so fp h a g eP 1a n dp l a s m i dp 1 5 B .S u b u n i tf u n c t i o n sa n ds t r u c -tural homologies. J Mol Biol 165:19 –34.https://doi.org/10.1016/S0022 -2836(83)80240-X . 49. Iida S, Meyer J, Bachi B, Stalhammar-Carlemalm M, Schrickel S, Bickle TA, Arber W. 1983. DNA restriction –modi ﬁcation genes of phage P1 and plas- mid p15B. Structure and in vitro transcription. J Mol Biol 165:1 –18. https://doi.org/10.1016/s0022-2836(83)80239-3 . 50. Humbelin M, Suri B, Rao DN, Hornby DP, Eberle H, Prip ﬂT, Kenel S, Bickle TA. 1988. Type III DNA restriction and modi ﬁcation systems EcoP1 and EcoP15.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  49. Iida S, Meyer J, Bachi B, Stalhammar-Carlemalm M, Schrickel S, Bickle TA, Arber W. 1983. DNA restriction –modi ﬁcation genes of phage P1 and plas- mid p15B. Structure and in vitro transcription. J Mol Biol 165:1 –18. https://doi.org/10.1016/s0022-2836(83)80239-3 . 50. Humbelin M, Suri B, Rao DN, Hornby DP, Eberle H, Prip ﬂT, Kenel S, Bickle TA. 1988. Type III DNA restriction and modi ﬁcation systems EcoP1 and EcoP15. Nucleotide sequence of the EcoP1 operon, the EcoP15 mod geneand some EcoP1 mod mutants. J Mol Biol 200:23 –29.https://doi.org/10 .1016/0022-2836(88)90330-0 . 51. Ikeda H, Tomizawa J. 1968. Prophage P1, and extrachromosomal replica- tion unit. Cold Spring Harbor Symp Quant Biol 33:791 –798. https://doi .org/10.1101/sqb.1968.033.01.091 . 52. Prentki P, Chandler M, Caro L. 1977. Replication of prophage P1 during the cell cycle of Escherichia coli. Mol Gen Genet 152:71 –76.https://doi .org/10.1007/BF00264942 . 53. Lehnherr H, Maguin E, Jafri S, Yarmolinsky MB. 1993. Plasmid addiction genes of bacteriophage P1: doc, which causes cell death on curing of pro-phage, and phd, which prevents host death when prophage is retained. JMol Biol 233:414 –428. https://doi.org/10.1006/jmbi.1993.1521 . 54. Gazit E, Sauer RT. 1999. The Doc toxin and Phd antidote proteins of the bacteriophage P1 plasmid addiction system form a heterotrimericMOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 15', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  complex. J Biol Chem 274:16813 –16818. https://doi.org/10.1074/jbc.274 .24.16813 . 55. Zhang K, Young R, Zeng L. 2020. Bacteriophage P1 does not show spatial preference when infecting Escherichia coli. Virology 542:1 –7.https://doi .org/10.1016/j.virol.2019.12.012 . 56. Devlin BH, Baumstark BR, Scott JR. 1982. Superimmunity: characterization of a new gene in the immunity region of P1. Virology 120:360 –375. https://doi.org/10.1016/0042-6822(82)90037-x . 57. Kliem M, Dreiseikelmann B. 1989. The superimmunity gene sim of bacterio- phage P1 causes superinfection exclusion. Virology 171:350 –355. https:// doi.org/10.1016/0042-6822(89)90602-8 . 58. Shao Q, Hawkins A, Zeng L. 2015. Phage DNA dynamics in cells with dif- ferent fates. Biophys J 108:2048 –2060. https://doi.org/10.1016/j.bpj.2015 .03.027 . 59. Trinh JT, Szekely T, Shao Q, Balazsi G, Zeng L. 2017. Cell fate decisions emerge as phages cooperate or compete inside their host. Nat Commun 8:14341. https://doi.org/10.1038/ncomms14341 . 60. Skinner SO, Sepulveda LA, Xu H, Golding I. 2013. Measuring mRNA copy number in individual Escherichia coli cells using single-molecule ﬂuores- cent in situ hybridization. Nat Protoc 8:1100 –1113. https://doi.org/10 .1038/nprot.2013.066 . 61. Trinh JT, Shao Q, Guan J, Zeng L. 2020. Emerging heterogeneous com- partments by viruses in single bacterial cells. Nat Commun 11:3813. https://doi.org/10.1038/s41467-020-17515-8 . 62. Shao Q, Trinh JT, McIntosh CS, Christenson B, Balazsi G, Zeng L. 2017. Lysis-lysogeny coexistence: prophage integration during lytic develop- ment. Microbiologyopen 6:e00395. https://doi.org/10.1002/mbo3.395 . 63. Lau IF, Filipe SR, Soballe B, Okstad OA, Barre FX, Sherratt DJ. 2003. Spatial and temporal organization of replicating Escherichia coli chromosomes. Mol Microbiol 49:731 –743. https://doi.org/10.1046/j.1365-2958.2003.03640.x . 64. Piya D, Vara L, Russell WK, Young R, Gill JJ. 2017. The multicomponent antirestriction system of phage P1 is linked to capsid morphogenesis. MolMicrobiol 105:399 –412. https://doi.org/10.1111/mmi.13705 . 65. Greer H. 1975. The kil gene of bacteriophage lambda. Virology 66: 589 –604. https://doi.org/10.1016/0042-6822(75)90231-7 . 66. Haeusser DP, Hoashi M, Weaver A, Brown N, Pan J, Sawitzke JA, Thomason LC, Court DL, Margolin W. 2014. The Kil peptide of bacterio- phage lambda blocks Escherichia coli cytokinesis via ZipA-dependentinhibition of FtsZ assembly. PLoS Genet 10:e1004217. https://doi.org/10 .1371/journal.pgen.1004217 . 67. Romantsov T, Helbig S, Culham DE, Gill C, Stalker L, Wood JM. 2007.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  65. Greer H. 1975. The kil gene of bacteriophage lambda. Virology 66: 589 –604. https://doi.org/10.1016/0042-6822(75)90231-7 . 66. Haeusser DP, Hoashi M, Weaver A, Brown N, Pan J, Sawitzke JA, Thomason LC, Court DL, Margolin W. 2014. The Kil peptide of bacterio- phage lambda blocks Escherichia coli cytokinesis via ZipA-dependentinhibition of FtsZ assembly. PLoS Genet 10:e1004217. https://doi.org/10 .1371/journal.pgen.1004217 . 67. Romantsov T, Helbig S, Culham DE, Gill C, Stalker L, Wood JM. 2007. Car- diolipin promotes polar localization of osmosensory transporter ProP inEscherichia coli. Mol Microbiol 64:1455 –1465. https://doi.org/10.1111/j .1365-2958.2007.05727.x . 68. Bakshi S, Choi H, Weisshaar JC. 2015. The spatial biology of transcription and translation in rapidly growing Escherichia coli. Front Microbiol 6:636. https://doi.org/10.3389/fmicb.2015.00636 . 69. Rothenberg E, Sepulveda LA, Skinner SO, Zeng L, Selvin PR, Golding I. 2011. Single-virus tracking reveals a spatial receptor-dependent search mechanism. Biophys J 100:2875 –2882. https://doi.org/10.1016/j.bpj.2011 .05.014 . 70. Edgar R, Rokney A, Feeney M, Semsey S, Kessel M, Goldberg MB, Adhya S, Oppenheim AB. 2008. Bacteriophage infection is targeted to cellular poles. Mol Microbiol 68:1107 –1116. https://doi.org/10.1111/j.1365-2958 .2008.06205.x . 71. Shao Q, Trinh JT, Zeng L. 2019. High-resolution studies of lysis-lysogeny decision-making in bacteriophage lambda. J Biol Chem 294:3343 –3349. https://doi.org/10.1074/jbc.TM118.003209 . 72. Bryant JA, Sellars LE, Busby SJ, Lee DJ. 2014. Chromosome position effects on gene expression in Escherichia coli K-12. Nucleic Acids Res 42:11383 –11392. https://doi.org/10.1093/nar/gku828 . 73. Neidhardt FC, Bloch PL, Smith DF. 1974. Culture medium for enterobacteria. J Bacteriol 119:736 –747. https://doi.org/10.1128/jb.119.3.736-747.1974 . 74. Datsenko KA, Wanner BL. 2000. One-step inactivation of chromosomal genes in Escherichia coli K-12 using PCR products. Proc Natl Acad Sci U S A 97:6640 –6645. https://doi.org/10.1073/pnas.120163297 . 75. Shao Q, Cortes MG, Trinh JT, Guan J, Balazsi G, Zeng L. 2018. Coupling of DNA replication and negative feedback controls gene expression for cell- fate decisions. iScience 6:1 –12.https://doi.org/10.1016/j.isci.2018.07.006 . 76. Bates D, Kleckner N. 2005. Chromosome and replisome dynamics in E. coli: loss of sister cohesion triggers global chromosome movement and medi-ates chromosome segregation. Cell 121:899 –911. https://doi.org/10.1016/j .cell.2005.04.013 . 77. Lee C, Kim J, Shin SG, Hwang S. 2006. Absolute and relative QPCR quanti- ﬁcation of plasmid copy number in Escherichia coli.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Shao Q, Cortes MG, Trinh JT, Guan J, Balazsi G, Zeng L. 2018. Coupling of DNA replication and negative feedback controls gene expression for cell- fate decisions. iScience 6:1 –12.https://doi.org/10.1016/j.isci.2018.07.006 . 76. Bates D, Kleckner N. 2005. Chromosome and replisome dynamics in E. coli: loss of sister cohesion triggers global chromosome movement and medi-ates chromosome segregation. Cell 121:899 –911. https://doi.org/10.1016/j .cell.2005.04.013 . 77. Lee C, Kim J, Shin SG, Hwang S. 2006. Absolute and relative QPCR quanti- ﬁcation of plasmid copy number in Escherichia coli. J Biotechnol 123: 273 –280. https://doi.org/10.1016/j.jbiotec.2005.11.014 .Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 16', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/p1 1.pdf  Intro |Search taxa  |Taxon tree  |Search literature  |Taxon match  |Homonyms  |Statistics |Webservice  |Manual |FAQ |LifeWatch  |Download  |Log inIRMNG - Enterobacteria phage P1 https://www.irmng.org/aphia.php?p=taxdetails&id=11459714 1 of 3 5/01/2024, 6:30 pm', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/p1 1.pdf  IRMNG taxon details Enterobacteria phage P1 basis of record ICTVdb (Jul 2011) [details] current name source ICTVdb (Jul 2011) [details] extant flag source ICTVdb (Jul 2011) [details]IRMNG_ID 11459714  (urn:lsid:irmng.org:taxname:11459714) Classification Biota > Viruses (Kingdom) > Viruses [phylum] (Phylum) > Caudovirales  (Order) > Myoviridae  (Family) >P1-like viruses  (Genus) > Enterobacteria phage P1  (Species) Status accepted Rank Species Parent P1-like viruses  accepted as P1virus Fossil range recent only Original description Not documented Taxonomic citation IRMNG (2021). Enterobacteria phage P1 . Accessed at: https://www.irmng.org/aphia.php?p=taxdetails&id=11459714 on 2024-01-05 Taxonomic edit history Date action by 2011-08-03 22:00:00Z created Rees, Tony 2011-12-31 23:00:00Z changed Rees, Tony [taxonomic tree] Sources (3)IRMNG - Enterobacteria phage P1 https://www.irmng.org/aphia.php?p=taxdetails&id=11459714 2 of 3 5/01/2024, 6:30 pm', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/p1 1.pdf  Website and databases developed and hosted by VLIZ · Page generated 2024-01-05 · contact: info@irmng.org This service is powered by LifeWatch Belgium Learn more»IRMNG - Enterobacteria phage P1 https://www.irmng.org/aphia.php?p=taxdetails&id=11459714 3 of 3 5/01/2024, 6:30 pm', \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Vol.174,No.12 DNAInversion Regions MinofPlasmid pl5BandCin ofBacteriophage P1:Evolution ofBacteriophage TailFiberGenes HEINRICH SANDMEIER,lt SHIGERU IIDA,1'2 ANDWERNER ARBERl* Abteilung Mikrobiologie, Biozentrum derUniversitat Basel,Klingelbergstrasse 70,CH-4056 Basel, Switzerland, 1andDepartment ofBiological Sciences andTechnology, Science University ofTokyo, Yamazaki, Chiba278,Japan2 Received 21November 1991/Accepted 11April1992 Plasmid pl5Bandthegenome ofbacteriophage P1arecloselyrelated,buttheirsite-specific DNAinversion systems, MinandCin,respectively, donothavestrictstructural homology. Rather, thecomplex Minsystem represents asubstitution ofaCin-like systemintoanancestral pl5Bgenome. Thesubstituting sequences of boththeminrecombinase geneandthemultiple invertible DNAsegments ofpl5B are,respectively, homologous tothepinrecombinase geneandtopartoftheinvertible DNAofthePinsystem onthedefective viralelement e14ofEscherichia coliK-12.Tomapthesitesofthissubstitution, theDNAsequence ofasegment adjacent totheinvertible segment intheP1genome wasdetermined. This,together withalready available sequence data,indicated thatbothP1andpl5Bhadsuffered various sequence acquisitions ordeletions and sequence amplifications givingrisetomosaics ofpartially relatedrepeated elements. Databasesearches revealed segments ofhomology intheDNAinversion regionsofpl5B,e14,andP1andintailfibergenesof phagesMu,T4,P2,andK.Thisresultsuggests thattheevolution ofphagetailfibergenesinvolves horizontal genetransfer andthattheMinandPinregionsencodetailfibergenes.Afunctional testprovedthatthepl5B Minregioncarries atailfiberoperonandsuggests thatthealternative expression ofsixdifferent genevariants byMininversion offersextensive hostrangevariation. Plasmid p15BisaP1prophage-related resident ofEsche- richiacoli15T-(18).Electron microscopy heteroduplex studiesrevealed thatapproximately 82%ofthe94-kbpl5B DNAhybridizes withtheP1genome (23).Since nophage particle withaP1-likemorphology couldbeproduced inE. coli15T-(1,18),pl5BDNAhasbeenconsidered tobea defective prophage. Oneregionofnonhomology between pl5BandP1DNA coverstheinvertible segment oftheP1 DNA(16,23).Itscorresponding region onpl5Bcarries a complex site-specific DNAinversion system calledMin(17, 34).TheMinregioncontains sixcrossover sitescalled mix. Successive DNAinversions between pairsofmixsites mediated byMinrecombinase canproduce 240isomeric formsofa3.5-kbregion(34).Theinvertible DNAsegments contain sixdifferent 3'geneparts,eachofwhich canbecome fusedinframetoanadjacent constant 5'genesegment upon DNAinversion (33)(seeFig.4). Thecomplex Minsystemiscloselyrelated tosimpleDNA inversion systems whichcontain twocrossover sitesand constitute atwo-state switch forgeneexpression: Cinof phageP1,GinofphageMu,Pinofthee14element onE.coli K-12,andHinofSalmonella typhimurium Thecrossover sitesandrecombinase genesofthesesystems arestructur- allyhomologous andcanmutually substitute fortheirfunc- tions(forareviewoftheseDNAinversion systems, called heretheDinfamily, seereference 7;seealsoFig.4).Inthe Hinsystem, twoflagellar geneswithdistinctantigenic spec- ificities arealternatively expressed byinversion ofaDNA *Corresponding author.\", \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Thecomplex Minsystemiscloselyrelated tosimpleDNA inversion systems whichcontain twocrossover sitesand constitute atwo-state switch forgeneexpression: Cinof phageP1,GinofphageMu,Pinofthee14element onE.coli K-12,andHinofSalmonella typhimurium Thecrossover sitesandrecombinase genesofthesesystems arestructur- allyhomologous andcanmutually substitute fortheirfunc- tions(forareviewoftheseDNAinversion systems, called heretheDinfamily, seereference 7;seealsoFig.4).Inthe Hinsystem, twoflagellar geneswithdistinctantigenic spec- ificities arealternatively expressed byinversion ofaDNA *Corresponding author. tPresent address: Abteilung furPraventivzahnmedizin undorale Mikrobiologie, Zahnarztliches InstitutderUniversitat Basel,Peter- splatz14,CH-4051 Basel,Switzerland.segment including apromoter. InallotherDinsystems described sofar,astationary promoter transcribes agene whichisassembled byDNAinversion fromconstant and variable genesegments. SuchgenefusionsbytheCinand Gininversions inphages P1andMuleadtoalternate expression oftwosetsoftailfibergeneswhichconfer different receptor specificities. TheseP1andMuSgenes consistofa5'constant portion, calledSc,andtwoversions ofthe3'portion, calledSvandSv'.Thee14element, which canbeconsidered adefective prophage, isanexcisable element ofsomeE.coliK-12chromosomes (la,21).The biological roleofPininversion one14,aswellasthoseof recently identified related DNAinversion systems onShi- gellachromosomes (39),isnotyetknown. Thecomparison oftheDNA sequences ofthepl5BMin regionwithpreviously determined sequences oftheCin regionofphageP1suggested thattheinvertible segment and partoftherecombinase genehadbeensubstituted by nonhomologous DNA onanancestral pl5B genome (33). While oneofthebreakpoints fromhomologous tononho- mologous sequences hadalready beenlocated withinthe recombinase genes,thesecondbreakpoint remained unde- termined because ofincomplete P1sequence dataofthis region. Inthispaper,theDNA sequence oftheconstant part(Sc) oftheP1geneSwascompleted toelucidate thesecond breakpoint. Theresultindicates thatcomplicated recombi- nations musthavetakenplaceatthissecondbreakpoint on anancestral genome. Inaddition, acomputer-aided compar- isonofthese sequences withavailable DNAandprotein sequences ofotherbacteriophages identified various se- quence domains thatareshared amongthegenesoftheP1 andpl5BDNAinversion regions andtailfibergenesof phages Mu,P2,X,andT4,aswellastheelement e14. 3936JOURNAL OFBACTERIOLOGY, June1992,p.3936-3944 0021-9193/92/123936-09$02.00/0 Copyright ©1992,American SocietyforMicrobiology Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.\"], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Lysogenization by bacteriophage lambda. I. Multiple infection and the lysogenic response. Mol Gen Genet 122:183 –195. https://doi.org/10.1007/BF00435190 . 40. Kourilsky P, Knapp A. 1974. Lysogenization by bacteriophage lambda. III. Multiplicity dependent phenomena occuring upon infection by lambda.Biochimie 56:1517 –1523. 41. Zeng L, Skinner SO, Zong C, Sippy J, Feiss M, Golding I. 2010. Decision making at a subcellular level determines the outcome of bacteriophageinfection. Cell 141:682 –691. https://doi.org/10.1016/j.cell.2010.03.034 . 42. Silpe JE, Bassler BL. 2019. Phage-encoded LuxR-type receptors responsive to host-produced bacterial quorum-sensing autoinducers. mBio 10:e00638-19. https://doi.org/10.1128/mBio.00638-19 . 43. Silpe JE, Bassler BL. 2019. A host-produced quorum-sensing autoinducer controls a phage lysis-lysogeny decision. Cell 176:268 –280.E13. https:// doi.org/10.1016/j.cell.2018.10.059 . 44. Erez Z, Steinberger-Levy I, Shamir M, Doron S, Stokar-Avihail A, Peleg Y, Melamed S, Leavitt A, Savidor A, Albeck S, Amitai G, Sorek R. 2017. Com-munication between viruses guides lysis-lysogeny decisions. Nature 541:488 –493. https://doi.org/10.1038/nature21049 . 45. Stokar-Avihail A, Tal N, Erez Z, Lopatina A, Sorek R. 2019. Widespread utiliza- tion of peptide communication in phages infecting soil and pathogenicbacteria. Cell Host Microbe 25:746 –755.E5. https://doi.org/10.1016/j.chom .2019.03.017 . 46. Bertani G, Nice SJ. 1954. Studies on lysogenesis. II. The effect of tempera- ture on the lysogenization of Shigella dysenteriae with phage P1. J Bac-teriol 67:202 –209. https://doi.org/10.1128/jb.67.2.202-209.1954 . 47. Rosner JL. 1972. Formation, induction, and curing of bacteriophage P1 lyso- gens. Virology 48:679 –689. https://doi.org/10.1016/0042-6822(72)90152-3 . 48. Hadi SM, Bachi B, Iida S, Bickle TA. 1983. DNA restriction –modi ﬁcation e n z y m e so fp h a g eP 1a n dp l a s m i dp 1 5 B .S u b u n i tf u n c t i o n sa n ds t r u c -tural homologies. J Mol Biol 165:19 –34.https://doi.org/10.1016/S0022 -2836(83)80240-X . 49. Iida S, Meyer J, Bachi B, Stalhammar-Carlemalm M, Schrickel S, Bickle TA, Arber W. 1983. DNA restriction –modi ﬁcation genes of phage P1 and plas- mid p15B. Structure and in vitro transcription. J Mol Biol 165:1 –18. https://doi.org/10.1016/s0022-2836(83)80239-3 . 50. Humbelin M, Suri B, Rao DN, Hornby DP, Eberle H, Prip ﬂT, Kenel S, Bickle TA. 1988. Type III DNA restriction and modi ﬁcation systems EcoP1 and EcoP15.', 'page_label: 15 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  49. Iida S, Meyer J, Bachi B, Stalhammar-Carlemalm M, Schrickel S, Bickle TA, Arber W. 1983. DNA restriction –modi ﬁcation genes of phage P1 and plas- mid p15B. Structure and in vitro transcription. J Mol Biol 165:1 –18. https://doi.org/10.1016/s0022-2836(83)80239-3 . 50. Humbelin M, Suri B, Rao DN, Hornby DP, Eberle H, Prip ﬂT, Kenel S, Bickle TA. 1988. Type III DNA restriction and modi ﬁcation systems EcoP1 and EcoP15. Nucleotide sequence of the EcoP1 operon, the EcoP15 mod geneand some EcoP1 mod mutants. J Mol Biol 200:23 –29.https://doi.org/10 .1016/0022-2836(88)90330-0 . 51. Ikeda H, Tomizawa J. 1968. Prophage P1, and extrachromosomal replica- tion unit. Cold Spring Harbor Symp Quant Biol 33:791 –798. https://doi .org/10.1101/sqb.1968.033.01.091 . 52. Prentki P, Chandler M, Caro L. 1977. Replication of prophage P1 during the cell cycle of Escherichia coli. Mol Gen Genet 152:71 –76.https://doi .org/10.1007/BF00264942 . 53. Lehnherr H, Maguin E, Jafri S, Yarmolinsky MB. 1993. Plasmid addiction genes of bacteriophage P1: doc, which causes cell death on curing of pro-phage, and phd, which prevents host death when prophage is retained. JMol Biol 233:414 –428. https://doi.org/10.1006/jmbi.1993.1521 . 54. Gazit E, Sauer RT. 1999. The Doc toxin and Phd antidote proteins of the bacteriophage P1 plasmid addiction system form a heterotrimericMOI-Independent Lysogeny upon Viral Infection® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 15', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  complex. J Biol Chem 274:16813 –16818. https://doi.org/10.1074/jbc.274 .24.16813 . 55. Zhang K, Young R, Zeng L. 2020. Bacteriophage P1 does not show spatial preference when infecting Escherichia coli. Virology 542:1 –7.https://doi .org/10.1016/j.virol.2019.12.012 . 56. Devlin BH, Baumstark BR, Scott JR. 1982. Superimmunity: characterization of a new gene in the immunity region of P1. Virology 120:360 –375. https://doi.org/10.1016/0042-6822(82)90037-x . 57. Kliem M, Dreiseikelmann B. 1989. The superimmunity gene sim of bacterio- phage P1 causes superinfection exclusion. Virology 171:350 –355. https:// doi.org/10.1016/0042-6822(89)90602-8 . 58. Shao Q, Hawkins A, Zeng L. 2015. Phage DNA dynamics in cells with dif- ferent fates. Biophys J 108:2048 –2060. https://doi.org/10.1016/j.bpj.2015 .03.027 . 59. Trinh JT, Szekely T, Shao Q, Balazsi G, Zeng L. 2017. Cell fate decisions emerge as phages cooperate or compete inside their host. Nat Commun 8:14341. https://doi.org/10.1038/ncomms14341 . 60. Skinner SO, Sepulveda LA, Xu H, Golding I. 2013. Measuring mRNA copy number in individual Escherichia coli cells using single-molecule ﬂuores- cent in situ hybridization. Nat Protoc 8:1100 –1113. https://doi.org/10 .1038/nprot.2013.066 . 61. Trinh JT, Shao Q, Guan J, Zeng L. 2020. Emerging heterogeneous com- partments by viruses in single bacterial cells. Nat Commun 11:3813. https://doi.org/10.1038/s41467-020-17515-8 . 62. Shao Q, Trinh JT, McIntosh CS, Christenson B, Balazsi G, Zeng L. 2017. Lysis-lysogeny coexistence: prophage integration during lytic develop- ment. Microbiologyopen 6:e00395. https://doi.org/10.1002/mbo3.395 . 63. Lau IF, Filipe SR, Soballe B, Okstad OA, Barre FX, Sherratt DJ. 2003. Spatial and temporal organization of replicating Escherichia coli chromosomes. Mol Microbiol 49:731 –743. https://doi.org/10.1046/j.1365-2958.2003.03640.x . 64. Piya D, Vara L, Russell WK, Young R, Gill JJ. 2017. The multicomponent antirestriction system of phage P1 is linked to capsid morphogenesis. MolMicrobiol 105:399 –412. https://doi.org/10.1111/mmi.13705 . 65. Greer H. 1975. The kil gene of bacteriophage lambda. Virology 66: 589 –604. https://doi.org/10.1016/0042-6822(75)90231-7 . 66. Haeusser DP, Hoashi M, Weaver A, Brown N, Pan J, Sawitzke JA, Thomason LC, Court DL, Margolin W. 2014. The Kil peptide of bacterio- phage lambda blocks Escherichia coli cytokinesis via ZipA-dependentinhibition of FtsZ assembly. PLoS Genet 10:e1004217. https://doi.org/10 .1371/journal.pgen.1004217 . 67. Romantsov T, Helbig S, Culham DE, Gill C, Stalker L, Wood JM. 2007.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  65. Greer H. 1975. The kil gene of bacteriophage lambda. Virology 66: 589 –604. https://doi.org/10.1016/0042-6822(75)90231-7 . 66. Haeusser DP, Hoashi M, Weaver A, Brown N, Pan J, Sawitzke JA, Thomason LC, Court DL, Margolin W. 2014. The Kil peptide of bacterio- phage lambda blocks Escherichia coli cytokinesis via ZipA-dependentinhibition of FtsZ assembly. PLoS Genet 10:e1004217. https://doi.org/10 .1371/journal.pgen.1004217 . 67. Romantsov T, Helbig S, Culham DE, Gill C, Stalker L, Wood JM. 2007. Car- diolipin promotes polar localization of osmosensory transporter ProP inEscherichia coli. Mol Microbiol 64:1455 –1465. https://doi.org/10.1111/j .1365-2958.2007.05727.x . 68. Bakshi S, Choi H, Weisshaar JC. 2015. The spatial biology of transcription and translation in rapidly growing Escherichia coli. Front Microbiol 6:636. https://doi.org/10.3389/fmicb.2015.00636 . 69. Rothenberg E, Sepulveda LA, Skinner SO, Zeng L, Selvin PR, Golding I. 2011. Single-virus tracking reveals a spatial receptor-dependent search mechanism. Biophys J 100:2875 –2882. https://doi.org/10.1016/j.bpj.2011 .05.014 . 70. Edgar R, Rokney A, Feeney M, Semsey S, Kessel M, Goldberg MB, Adhya S, Oppenheim AB. 2008. Bacteriophage infection is targeted to cellular poles. Mol Microbiol 68:1107 –1116. https://doi.org/10.1111/j.1365-2958 .2008.06205.x . 71. Shao Q, Trinh JT, Zeng L. 2019. High-resolution studies of lysis-lysogeny decision-making in bacteriophage lambda. J Biol Chem 294:3343 –3349. https://doi.org/10.1074/jbc.TM118.003209 . 72. Bryant JA, Sellars LE, Busby SJ, Lee DJ. 2014. Chromosome position effects on gene expression in Escherichia coli K-12. Nucleic Acids Res 42:11383 –11392. https://doi.org/10.1093/nar/gku828 . 73. Neidhardt FC, Bloch PL, Smith DF. 1974. Culture medium for enterobacteria. J Bacteriol 119:736 –747. https://doi.org/10.1128/jb.119.3.736-747.1974 . 74. Datsenko KA, Wanner BL. 2000. One-step inactivation of chromosomal genes in Escherichia coli K-12 using PCR products. Proc Natl Acad Sci U S A 97:6640 –6645. https://doi.org/10.1073/pnas.120163297 . 75. Shao Q, Cortes MG, Trinh JT, Guan J, Balazsi G, Zeng L. 2018. Coupling of DNA replication and negative feedback controls gene expression for cell- fate decisions. iScience 6:1 –12.https://doi.org/10.1016/j.isci.2018.07.006 . 76. Bates D, Kleckner N. 2005. Chromosome and replisome dynamics in E. coli: loss of sister cohesion triggers global chromosome movement and medi-ates chromosome segregation. Cell 121:899 –911. https://doi.org/10.1016/j .cell.2005.04.013 . 77. Lee C, Kim J, Shin SG, Hwang S. 2006. Absolute and relative QPCR quanti- ﬁcation of plasmid copy number in Escherichia coli.', 'page_label: 16 file_path: /workspaces/ml-learning/src/phages/data/P1/mbio.01013-21.pdf  Shao Q, Cortes MG, Trinh JT, Guan J, Balazsi G, Zeng L. 2018. Coupling of DNA replication and negative feedback controls gene expression for cell- fate decisions. iScience 6:1 –12.https://doi.org/10.1016/j.isci.2018.07.006 . 76. Bates D, Kleckner N. 2005. Chromosome and replisome dynamics in E. coli: loss of sister cohesion triggers global chromosome movement and medi-ates chromosome segregation. Cell 121:899 –911. https://doi.org/10.1016/j .cell.2005.04.013 . 77. Lee C, Kim J, Shin SG, Hwang S. 2006. Absolute and relative QPCR quanti- ﬁcation of plasmid copy number in Escherichia coli. J Biotechnol 123: 273 –280. https://doi.org/10.1016/j.jbiotec.2005.11.014 .Zhang et al.® September/October 2021 Volume 12 Issue 5 e01013-21 mbio.asm.org 16', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/p1 1.pdf  Intro |Search taxa  |Taxon tree  |Search literature  |Taxon match  |Homonyms  |Statistics |Webservice  |Manual |FAQ |LifeWatch  |Download  |Log inIRMNG - Enterobacteria phage P1 https://www.irmng.org/aphia.php?p=taxdetails&id=11459714 1 of 3 5/01/2024, 6:30 pm', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/p1 1.pdf  IRMNG taxon details Enterobacteria phage P1 basis of record ICTVdb (Jul 2011) [details] current name source ICTVdb (Jul 2011) [details] extant flag source ICTVdb (Jul 2011) [details]IRMNG_ID 11459714  (urn:lsid:irmng.org:taxname:11459714) Classification Biota > Viruses (Kingdom) > Viruses [phylum] (Phylum) > Caudovirales  (Order) > Myoviridae  (Family) >P1-like viruses  (Genus) > Enterobacteria phage P1  (Species) Status accepted Rank Species Parent P1-like viruses  accepted as P1virus Fossil range recent only Original description Not documented Taxonomic citation IRMNG (2021). Enterobacteria phage P1 . Accessed at: https://www.irmng.org/aphia.php?p=taxdetails&id=11459714 on 2024-01-05 Taxonomic edit history Date action by 2011-08-03 22:00:00Z created Rees, Tony 2011-12-31 23:00:00Z changed Rees, Tony [taxonomic tree] Sources (3)IRMNG - Enterobacteria phage P1 https://www.irmng.org/aphia.php?p=taxdetails&id=11459714 2 of 3 5/01/2024, 6:30 pm', 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/p1 1.pdf  Website and databases developed and hosted by VLIZ · Page generated 2024-01-05 · contact: info@irmng.org This service is powered by LifeWatch Belgium Learn more»IRMNG - Enterobacteria phage P1 https://www.irmng.org/aphia.php?p=taxdetails&id=11459714 3 of 3 5/01/2024, 6:30 pm', \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Vol.174,No.12 DNAInversion Regions MinofPlasmid pl5BandCin ofBacteriophage P1:Evolution ofBacteriophage TailFiberGenes HEINRICH SANDMEIER,lt SHIGERU IIDA,1'2 ANDWERNER ARBERl* Abteilung Mikrobiologie, Biozentrum derUniversitat Basel,Klingelbergstrasse 70,CH-4056 Basel, Switzerland, 1andDepartment ofBiological Sciences andTechnology, Science University ofTokyo, Yamazaki, Chiba278,Japan2 Received 21November 1991/Accepted 11April1992 Plasmid pl5Bandthegenome ofbacteriophage P1arecloselyrelated,buttheirsite-specific DNAinversion systems, MinandCin,respectively, donothavestrictstructural homology. Rather, thecomplex Minsystem represents asubstitution ofaCin-like systemintoanancestral pl5Bgenome. Thesubstituting sequences of boththeminrecombinase geneandthemultiple invertible DNAsegments ofpl5B are,respectively, homologous tothepinrecombinase geneandtopartoftheinvertible DNAofthePinsystem onthedefective viralelement e14ofEscherichia coliK-12.Tomapthesitesofthissubstitution, theDNAsequence ofasegment adjacent totheinvertible segment intheP1genome wasdetermined. This,together withalready available sequence data,indicated thatbothP1andpl5Bhadsuffered various sequence acquisitions ordeletions and sequence amplifications givingrisetomosaics ofpartially relatedrepeated elements. Databasesearches revealed segments ofhomology intheDNAinversion regionsofpl5B,e14,andP1andintailfibergenesof phagesMu,T4,P2,andK.Thisresultsuggests thattheevolution ofphagetailfibergenesinvolves horizontal genetransfer andthattheMinandPinregionsencodetailfibergenes.Afunctional testprovedthatthepl5B Minregioncarries atailfiberoperonandsuggests thatthealternative expression ofsixdifferent genevariants byMininversion offersextensive hostrangevariation. Plasmid p15BisaP1prophage-related resident ofEsche- richiacoli15T-(18).Electron microscopy heteroduplex studiesrevealed thatapproximately 82%ofthe94-kbpl5B DNAhybridizes withtheP1genome (23).Since nophage particle withaP1-likemorphology couldbeproduced inE. coli15T-(1,18),pl5BDNAhasbeenconsidered tobea defective prophage. Oneregionofnonhomology between pl5BandP1DNA coverstheinvertible segment oftheP1 DNA(16,23).Itscorresponding region onpl5Bcarries a complex site-specific DNAinversion system calledMin(17, 34).TheMinregioncontains sixcrossover sitescalled mix. Successive DNAinversions between pairsofmixsites mediated byMinrecombinase canproduce 240isomeric formsofa3.5-kbregion(34).Theinvertible DNAsegments contain sixdifferent 3'geneparts,eachofwhich canbecome fusedinframetoanadjacent constant 5'genesegment upon DNAinversion (33)(seeFig.4). Thecomplex Minsystemiscloselyrelated tosimpleDNA inversion systems whichcontain twocrossover sitesand constitute atwo-state switch forgeneexpression: Cinof phageP1,GinofphageMu,Pinofthee14element onE.coli K-12,andHinofSalmonella typhimurium Thecrossover sitesandrecombinase genesofthesesystems arestructur- allyhomologous andcanmutually substitute fortheirfunc- tions(forareviewoftheseDNAinversion systems, called heretheDinfamily, seereference 7;seealsoFig.4).Inthe Hinsystem, twoflagellar geneswithdistinctantigenic spec- ificities arealternatively expressed byinversion ofaDNA *Corresponding author.\", \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Thecomplex Minsystemiscloselyrelated tosimpleDNA inversion systems whichcontain twocrossover sitesand constitute atwo-state switch forgeneexpression: Cinof phageP1,GinofphageMu,Pinofthee14element onE.coli K-12,andHinofSalmonella typhimurium Thecrossover sitesandrecombinase genesofthesesystems arestructur- allyhomologous andcanmutually substitute fortheirfunc- tions(forareviewoftheseDNAinversion systems, called heretheDinfamily, seereference 7;seealsoFig.4).Inthe Hinsystem, twoflagellar geneswithdistinctantigenic spec- ificities arealternatively expressed byinversion ofaDNA *Corresponding author. tPresent address: Abteilung furPraventivzahnmedizin undorale Mikrobiologie, Zahnarztliches InstitutderUniversitat Basel,Peter- splatz14,CH-4051 Basel,Switzerland.segment including apromoter. InallotherDinsystems described sofar,astationary promoter transcribes agene whichisassembled byDNAinversion fromconstant and variable genesegments. SuchgenefusionsbytheCinand Gininversions inphages P1andMuleadtoalternate expression oftwosetsoftailfibergeneswhichconfer different receptor specificities. TheseP1andMuSgenes consistofa5'constant portion, calledSc,andtwoversions ofthe3'portion, calledSvandSv'.Thee14element, which canbeconsidered adefective prophage, isanexcisable element ofsomeE.coliK-12chromosomes (la,21).The biological roleofPininversion one14,aswellasthoseof recently identified related DNAinversion systems onShi- gellachromosomes (39),isnotyetknown. Thecomparison oftheDNA sequences ofthepl5BMin regionwithpreviously determined sequences oftheCin regionofphageP1suggested thattheinvertible segment and partoftherecombinase genehadbeensubstituted by nonhomologous DNA onanancestral pl5B genome (33). While oneofthebreakpoints fromhomologous tononho- mologous sequences hadalready beenlocated withinthe recombinase genes,thesecondbreakpoint remained unde- termined because ofincomplete P1sequence dataofthis region. Inthispaper,theDNA sequence oftheconstant part(Sc) oftheP1geneSwascompleted toelucidate thesecond breakpoint. Theresultindicates thatcomplicated recombi- nations musthavetakenplaceatthissecondbreakpoint on anancestral genome. Inaddition, acomputer-aided compar- isonofthese sequences withavailable DNAandprotein sequences ofotherbacteriophages identified various se- quence domains thatareshared amongthegenesoftheP1 andpl5BDNAinversion regions andtailfibergenesof phages Mu,P2,X,andT4,aswellastheelement e14. 3936JOURNAL OFBACTERIOLOGY, June1992,p.3936-3944 0021-9193/92/123936-09$02.00/0 Copyright ©1992,American SocietyforMicrobiology Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.\"], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:10 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'383'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'995380'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'277ms'), (b'x-request-id', b'6fd132e3636f045f347f2f90f5536a64'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249d329d35d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:10 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'383'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'995380'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'277ms'), (b'x-request-id', b'6fd132e3636f045f347f2f90f5536a64'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249d329d35d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': [\"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  VOL.174,1992 ~~~~EVOLUTION OFpl5B3ANDPHAGE P1TAILFIBERGENES 3937 AccI  eal~~~~~~~~~~~~~~~~~~~~~~~~~~~~~0YTAQDASTTQKGLVQLSSATNSDSETMAATPKA *BglII alN,.,e..bl . .  * ~~~~~StyI bl%,,.~.b2. INPYTTINNNLA PLASPA LTGVPTAP200 300 400 500 TAAQGTNNTQ IATTAY VRAAI SALVGSSPEALD T b%..c SfaNI CCCTGTATGAGCTTGCAGCAGCACTGGGCAATGACCCGAACTTTGCGACAACAATGACAAATGCGCTGGCAGGGAAA,CAGCCACTTIU2gC-AACTTTAAC 600 LYELAAALGNDPNFATTMTNALAG --------- 700 ALAGL ATG ANKL PY FT GT DTVSQ TD LTSVG RD I CTGGCCAAAACAAGCATTCTTGCT ITA TCOTA 760 LAK TSI LAV IQYLGRE LGT cixR FIG.1.Partialsequence oftheconstantportion(Sc)ofP1Stailfibergeneanditspredicted aminoacidsequence. Restriction enzymesites areprintedinboldfacetype,andthecixRcrossover siteisboxed.Bordersofrepeated elements(al,bl,b2,andc)areindicated abovethe sequence. Thesequencepartiallyoverlaps thatofafragmentcontaining the5'endofScuptotheBgllIsite(9,38)aswellasthatofa'noth'erfragmentcontaining thecixRsitewhichstartsfromtheSfaNIsite(12).Accession numberM81956includesthesequence shownhereandthe sequences oftheseflankingfragments. Finally, directevidence thattheinvertible segments ofpl5Bcodeforphagetailfiberswasobtainedbyacomplementationexperiment withaP1mutant defective initstailfiber synthesis. Theevolutionary relevance oftheseresultsis addressed. MATERIALS ANDMETHODS Bacterial andphagestrains.E.coliK-12WA921 ispin hsdKlacleumetthrsupE(40).M47,aderivative ofP1-15 prophage withatemperature-sensitive cirepressor gene, carries apolarmutation inthetailfiberoperonbecause of insertion of152(37).Plasmids. ThepBR322-derived plasmidpRM2O6 contains the7.9-kbBamHI-5 fragment ofthephageP1genome(29),andthepUC19-based plasmidpAW800 carriesthecorre- sponding 8.3-kbBamHI-5 fragment ofpl5B(17).Therele- vantfeatures oftheseBamHIfragments arealsoseeninFig.4.InbacteriacarryingpAW800, all240possibleconfigura-tionsoftheMinsystem arethought tobepresent, and therefore, plasmidpreparations ofpAW800 consist ofa mixture ofisomers. Tailfibercomplementation assay.WA921 cellsharboringthedefective P1-derived prophage M47weretransformed witheitherpRM206 orpAW800. Titersofinfectious parti-clesinlysatesproduced byheat-induced cellswereassayedbyplatingwithWA921(pRM206) indicator cellsonLuria agarplatescontaining ampicillin (200p.gIml). Plasmid pRM2O6 carriesthewild-type P1tailfiberoperon, sothe indicator bacteria canallowmultiplecyclesofreproductionofphageparticlesharboring themutantM47genome. Determination ofthePISctailfibergenesequence.\", \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Tailfibercomplementation assay.WA921 cellsharboringthedefective P1-derived prophage M47weretransformed witheitherpRM206 orpAW800. Titersofinfectious parti-clesinlysatesproduced byheat-induced cellswereassayedbyplatingwithWA921(pRM206) indicator cellsonLuria agarplatescontaining ampicillin (200p.gIml). Plasmid pRM2O6 carriesthewild-type P1tailfiberoperon, sothe indicator bacteria canallowmultiplecyclesofreproductionofphageparticlesharboring themutantM47genome. Determination ofthePISctailfibergenesequence. The 1.7-kbIApnlfragment including thecixRcrossover siteanda largepartoftheP1SctailfibergeneofphageP1wascutout frompRM2O6 andligatedintotheKpnlsiteofpEMBL18 to generatepAW81O andintotheKpnlsiteofpEMBL19 to generatepAW811. pAW812 andpAW813 areAccldeletion derivatives ofthepAW81O andpAW811plasmids, respec- tively.PlasmidpAW814 resulted fromaStyl-PstI deletion of pAW812, andpAW815 resulted fromaStyl-Kpnl deletion of pAW813. DNAsequencing wasperformed bythechain termination method(35)usingsingle-stranded DNApro-ducedfromplasmids pAW812, pAW813, pAW814, and pAW815.Laboratory procedures. Cloning wascarried outasde- scribedbyManiatis etal.(22).Computer analysis. Homology searches andsequence alignments weremadewiththeprograms oftheGenetics Computer GroupoftheUniversity ofWisconsin (5).EMBL andGenBank DNAsequence databasesandMIPSXproteinsequence databasescontaining updates untilSeptember1991werescreened withtheprograms WORDSEARCH and PROFILESEARCH. Nucleotide sequence accessi'on numbers. Theaccession number ofthe2.5-kbP1sequence whichincludes these- quence oftheAccI-KpnI fragment showninFig.1is M81956. Theaccession numbers ofDNAsequences with homology toP1orpl5BDNAreported inthisstudy(andthe corresponding references) areasfollows: forthepl5BMin region,M62121(33);forphageP1totheleftoftheinvertible segment, M81956(thisworkand9and12);onlytheamino acidsequence oftheentireP1invertible segment isavailable (20);forfragments oftheP1invertible segment, K03173 and X01828(12);forphageP1totherightoftheinvertible segment, X01828(12,15a);forthee14Pinregion,.K03521 (31);noaccession number fortheGinregionofphageMu (19);forphageX,J02459(3);forphageP2,M64677(11);for phageT4gene12,X06729(36);forphageT4gene37,J02509 (30);andforphageT4gene38,X05677(26). RESULTS ANDDISCUSSION ScgenesofP1andpl5Bcontainpartially relatedarraysof repeated elements.\", \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  RESULTS ANDDISCUSSION ScgenesofP1andpl5Bcontainpartially relatedarraysof repeated elements. TheDNAsequence oftheScgeneofP1 wascompleted bysubcloning andsequencing anAccI-KpnIfragment ofphageP1including thecixRcrossover sitethat separates 739bpoftheconstantportion(Sc)from21bpof onevariableportion(Sv)ofthetailfibergeneS(Fig.1).Thestronghomology of99%identical basesofthecon- stantpart(Sc)ofgenesSofpl5BandP1includes the elements termedaO(Fig.2).Theseshort36-bpsequences share58and64%identical baseswiththe3'endsofthe succeeding 132-bpelements alofP1andp15B,respectively.Sixelements, termeda2toa7,sharebetween 64and100%VOL.174,1992 Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.\", \"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  3938 SANDMEIER ETAL. mix aOala2a3a4a5a6a7cmix'Il. p15B aI_,.'.'.'. ......................'...'.'.'.'.'.'.'.'.'.'.'.'.'.'.' I~~~~~~~~~~~................_..... ............................................................... .... ............ .............................  zato .......w o. wocst z :.::.:.::.::.:::: ...:,.::,::,..:,:,:,,::,:.:.:::::,:::::::.. :,::, ::,:::.: ..n::.:.:::.::: :::,::,:...... ....... 1.......--........aCa1b1b2.cix 200bpI Sc SvAcciBgIllStylSfaNIKpni regionsequenced FIG.2.Structure andhomology amongrepeated elements inS genesofpl5BandP1.TheaOsmallelements arehomologous tothe 3'endsofelements alinP1andinp15BDNAs.Elements a2toa7 ofp15Barehomologous toaOandalofP1,exceptforthe5'parts ofa3anda6,whichareshownasopenboxes.Thelongarrows represent aduplication ofelements a2toa4anda5toa7,respec- tively.Adjacent invertible segments areshownasopenboxes,and crossover sites(mixandcixR)areshownastriangles. Thelatter separate theconstant (Sc)fromthevariable parts(SvandSvQ)of genesS.Sequence homologies between thep15BandP1DNAsare indicated inpercentages. Thesegment fromwhichthesequence in Fig.1wasdetermined isindicated. Therestriction enzyme sites usedforcloningandsequencing areindicated.identity withaOandalofpl5Bandfollowalonpl5BDNA asdirectrepeats. Itseemsthatthea5-to-a7 segment isa recentduplication ofa2toa4,sincethesetwo432-bpand 144-amino-acid (aa)segments contain onlyfournucleotide substitutions leading toonlyoneaminoacidchange. The elements a3anda6shareattheir5'endshighlyhomologous 51-bpsequences codingfor17aawhicharenotrelated to anysequences oftheotherrepeated elements onpl5Borto aOandalonP1(Fig.3A). NexttoalofP1,a444-bpand148-aastretchconsists of twodirectly repeated elements andexhibits nosimilarity to pl5BDNA(Fig.1and2).Theseelements aredesignated bl andb2andhave63%identical basesand68%identical aminoacids(Fig.3B). Between b2andthecixRcrossover siteflanking the invertible segment ofP1liesa162-bpand54-aasequence, termedc,whichresumes homology tothecorresponding c elementofp15BDNAbetween a7andthemixcrossover site (Fig.1and2).Thosecelements have59%identical bases and69%identical aminoacids(Fig.3C). Substitution oftheinvertible DNAandtherecombinase geneonanancestral pl5Bgenome. Dataobtained onhomol- ogiesbetween theCinandMinregions aresummarized in thecentralpartofFig.4.Strikingly, theinvertible DNA segments ofP1andpl5Bbetween theircrossover sitesare nothomologous. Theminandcingenesarelocated inboth systems adjacent totheinvertible segments, buttheyare carriedinopposite orientations.\", 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Substitution oftheinvertible DNAandtherecombinase geneonanancestral pl5Bgenome. Dataobtained onhomol- ogiesbetween theCinandMinregions aresummarized in thecentralpartofFig.4.Strikingly, theinvertible DNA segments ofP1andpl5Bbetween theircrossover sitesare nothomologous. Theminandcingenesarelocated inboth systems adjacent totheinvertible segments, buttheyare carriedinopposite orientations. Theirsequences have67% identical basesand74%identical aminoacids.Incontrast, (197)............................................(209) ............ L (253) .. T D (297)APSSHTHPWNQITGVPT LAT V (353)....... TA V (397)............ T D (441)APSSHTHPWNQITVVPT LAT V(497)....... TA V(197)............................................(209) ............ L DM (164) ............ RRHP LAFT (208).. TARL (107) ............ LDIY LNKLTDP A (351).VALE DR21 41 KKS(208) (252) A(296) (352) (396) A(440) (496) (540) KKS(208) SIKDT(252) (207) TVMETK(251) IMNAR(150) VVMETR(394) 1 21 41 61 74 APL-SPALTGTPTAPTA-QGTNNTQIANTAFV-AAI-AL---SP-AL-TL-ELAAALGNDPNFATTMTNALAGK (253) IES AS KTINGAGTDKIINYSINL(326)(327) A V A TYRSVGSEDY (400)(252) N TR MAVDSDNN (325)20834).D LR LADVIDA DNN D (21053) 1 21 41 54 QP-DATLTALAGLATAA-KLPYFTG-D-ASLT-LT-VGRDILAK-S---V--YL (541)AA AD VRAASAGTIQSLD(594)(401) L GN TTVQDS TILAIQ(454)(326) K DRF NVATK STVAAIE(379)(9)KN SKNAENAE0 NVADLE(62) FIG.3.Alignment ofrepeated andothershortelements present intheScprotein sequences ofpl5B,P1,andMu,inthetailfiberprotein HofphageP2,andintailfiberproteins ofphage X.Aconsensus sequence with aplurality ofsix(A)andofthree(BandC)wasderived. Hyphens areintroduced where noconsensus wasderived. Inthespecific sequences, blanksdenoteresidues thatconform totheconsensus sequence anddotsindicate gapsintroduced foroptimalalignment. Numbers inparentheses givethecoordinates oftheaminoacidsequences predicted fromthepublished DNA sequences. (A)Alignment ofelements designated a.Therepeated elements aOtoa7ofpl5Bhaveformerly beencalledreOtore7(33).(B)Alignment ofelements designated b.Theelement bofX401-Stf hasbeendeduced from anintergenic region (seetext),andtherefore, thenumbers inparentheses arecoordinates ofthe Xgenome. (C)Alignment ofelements designated c.56 YTAQDASTTQKGIVQLSSATNSTSETLAATPKAVKAAYDLANGKA consensus pl5BScaO pl5BScal p15BSca2 pl5BSca3 p15BSca4 p15BScaS p15BSca plSBSca7 P1ScaO P1Scal P2Hal P2Ha2 MuSca )401a B cOnsensu P1Scbl P1Scb2 P2Hb X401-Stf b(2 C consensus plSBScc P1Scc P2Hc ).StfcJ.BACTERIOL. Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  EVOLUTION OFplSBANDPHAGE P1TAILFIBER GENES 3939 -=DC==>gin P.Sc Sv U U Sv FIG.4.Mapofhomologous DNAsequences amongtheDNA inversion regionsofe14,pl5B,P1,andMu.P1andp1SBregionsare represented bytheirBamHI-5 fragments. Invertible segments are shownasboxesflanked bycrossover sitesdepicted asblack triangles. Genesofoperons thatareaffected byDNAinversion and DNAinvertase genesareshownasopenarrows. TheP1cingeneis opposite inorientation tothatoftherelatedDNAinvertase genes. ThePspromoters oftailfibergenesareshownasshortblack arrows.Homology between pl5BandP1totheleftoftheinvertible segments issimplified fromthatshowninFig.2.Therefore, thea,b, andcelements composing theseregions arenotspecifically shown here.Thesiteofintegration ofIS2intoP1prophage M47ismarked. Theindicated highdegreeofhomology ofCinandGininvertible segments (20)isinterrupted byanapproximately 190-bpsegment of only35%homology aroundtheendofSvandthestartofUgenes. Thesmall63%homologous sequences inScofP1andMucorre- spondtotheaelements whicharealignedinFig.3A.The67-bp sequences between thecrossover pointsandtheDNAinvertase genesofpl5Bande14show31%homology. remarkable homologies inthesameorientation areseenon boththerightandtheleftsidesoftheinvertible segments andtheirrecombinase genes.Totheright,homologous DNA onP1startsstillwithinthecingeneandhence,thepl5B genome harbors aremnant fragment ofcin,thepseudo-cin gene(ilcin).Ithasbeenshownelsewhere thattlcinandthe recombinational enhancer sequence ontjcinarenonfunc- tional(33).Presence ofthetlcingeneindicates thattheDNA inversion regionofanancestral p15Bgenome hadcarrieda site-specific recombinase geneoftheDinfamilyinthesameconfiguration asthatoftheP1genome. Itismostlikelythat thecingeneandtheinvertible DNAofthisancestral pl5B genome hadbeensubstituted bytheMinsystemcomposed ofthemingeneandtheadjacent multiple invertible DNA segments. Sincethebreakpoints fromhomology tononho- mology between p15BandP1DNA aresharpandsomeof themcoincide withthecrossover sitesforDinrecombinases, itistempting tospeculate thataDinsite-specific recombi- nasehadmediated thesubstitution process. Thestructures ofrepeated elements neartheleftbreakpoints onp15Band P1suggest thatsubstitution mighthavebeenassociated with complicated recombinational processes. Alternatively, the presence ofthesestructures andthemultiple invertible DNA segments ofpl5Bcouldhaveresulted fromindependent events. Relation between pl5Bande14invertible DNAsegments.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Sincethebreakpoints fromhomology tononho- mology between p15BandP1DNA aresharpandsomeof themcoincide withthecrossover sitesforDinrecombinases, itistempting tospeculate thataDinsite-specific recombi- nasehadmediated thesubstitution process. Thestructures ofrepeated elements neartheleftbreakpoints onp15Band P1suggest thatsubstitution mighthavebeenassociated with complicated recombinational processes. Alternatively, the presence ofthesestructures andthemultiple invertible DNA segments ofpl5Bcouldhaveresulted fromindependent events. Relation between pl5Bande14invertible DNAsegments. Comparison withthesequences ofotherDNAinversion systems revealed thatthefiveinvertible segments ofpl5B arerelated totheinvertible DNAofthePinsystemofe14.A 1.1-kb partofthe1.8-kbPininvertible segment (31)is between 60and80%homologous tothe3.5-kbMininvertible segments (Fig.4).ThesixSvcarboxy-terminal variable segments ofpl5BsharewiththeP(-)carboxy-terminal segment ofe14ahighlyhomologous domain ofabout70aa (Fig.5).When anucleotide isinserted afterposition 497of thepublished sequence ofthePinregionofe14toproduce a frameshift inP(-),thelast13aaofthepredicted P(-) product areidentical totheSvM1,SvP,andSvQandalmost identical totheSvMr,SvN,andSvOcarboxy-terminal segments (Fig.5).Thissecondhomologous domain issepa- ratedfromthefirstdomainby61to84aaintheSvandP(-) sequences. OnlytheSvMrandP(-)segments arehomolo- gousalsobetween the70-and13-aadomains (datanot shown). Thealternative P(+)carboxy-terminal segment of e14exhibits nosequence homology toP(-)ortopl5BSv segments. Thepostulated geneTofpl5B,two-thirds ofe14open reading frame(ORF)293,andtheirputative products have 75%oftheirbasesandaminoacidsincommon. Homology oftheMinandPinregionsincludes notonlythe invertible DNAsegments butalsotheadjacent pinandmin genes,whichhave72%identical basesand80%identical aminoacids.Preliminary sequence datashowed that0.6kb ofe14flanking theothersideofthePininvertible segment arenothomologous totheMinregion(32).', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  OnlytheSvMrandP(-)segments arehomolo- gousalsobetween the70-and13-aadomains (datanot shown). Thealternative P(+)carboxy-terminal segment of e14exhibits nosequence homology toP(-)ortopl5BSv segments. Thepostulated geneTofpl5B,two-thirds ofe14open reading frame(ORF)293,andtheirputative products have 75%oftheirbasesandaminoacidsincommon. Homology oftheMinandPinregionsincludes notonlythe invertible DNAsegments butalsotheadjacent pinandmin genes,whichhave72%identical basesand80%identical aminoacids.Preliminary sequence datashowed that0.6kb ofe14flanking theothersideofthePininvertible segment arenothomologous totheMinregion(32). Sincethepingeneistheclosest relative ofthemin recombinase gene(17)andthepl5Bande14invertible DNA TABLE 1.Percentages ofidentical (andsimilar) residues ofaelements alignedinFig.3A %ofidentical (%ofsimilar) residues with: Element5Bcal pl5BScpl5BSca3pl5BSc PSa uc 2a 2a 4l EeplSB Scala2andaS anda6b a4anda7PlScalMuScaP2HalP2Ha2i401a pl5BScal 100 pl5BSca2andaS 89(91) 100 pl5BSca3anda6fi 90(92) 82(82) 100 pl5BSca4anda7 93(95) 91(91) 87(87) 100 P1Scal 82(86) 77(82) 69(77) 75(82) 100 MuSca 61(70) 55(66) 64(77) 57(68) 57(68) 100 P2Hal 80(82) 75(77) 86(89) 75(77) 66(73) 61(73) 100 P2Ha2 77(77) 75(77) 59(61) 80(82) 73(80) 59(70 64(66) 100 X401a 68(73) 68(70) 72(72) 68(70) 64(75) 57(77) 66(68) 80(86) 100 Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). hThefirst17aaofpl5Ba3anda6havenotbeenconsidered forthecalculation ofhomology values.VOL.174,1992 Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  3940 SANDMEIER ETAL. consensus pl5BSvM1 plSBSvMr P15BSVN pl5BSvO pl5BSvP P15BSVQ e14P(-) XStf T437 T412 consensus pl5BSvMl pl5BSvMr P1SBSVN plSBSvO plSBSvP P15BSVQ e14P(-) XStf T437 T4121 21 41 61 80 GLGEGSALPVGVPVPWPSATPPT. GWLKCNGAAFSSE-YPKLAK ....... AYPTNK. LPDLRGEFIRGWDDGRGVD-GR (1) (1) (1) (1) (1) (1) (1) (63) (806) (324)LE . M ....... KN........ L. LE . M ....... K N........ V L. M ....... KN....... AEE....... A N FAAI DIV S.YVLMQ DKSA .... SGV. KTDIM SYIAIDSVA.FALME QTDKSA V SGV.I QNEIDRTI AIMMAADSL SDARFHGTVASDCLYSRIGTRYGGN SPG 81 101 116 --LS-Q--- ......---H-H-- VI IGWL ...TGS YNI SI EGYA ...TED AGL AL NW PHT ..ILD AYM SL GTAT FIRTALQD ALITGM LEKRIV QL SGDA IRNIEGFA SIIGYA TEDAGL AV QEQDG IKS TSA AV AEADG. VKA SSA VNGNDFGKPRLGVGCTGGYVGEVQIQQMSY KAG(88) (88) (88) (88) (88) (88) (88) (145) (887) (439)(132) (147) (149) (149) (143) (135) (149) (295) (1007) (497)A Is A SIT A A IT NWTKKPAS..... MQTKKP.S..... MLVSGRSHLTNPN 1 21 ----E-RPRNIAFNYIVRAA* SGGNT *(153) TAGLT D*(166) NVGTT V *(168) GAWISM V T*(168) TGGST *(159) PTANN *(145) AAASAX *(165) AGNANTVK L*(314) TGNTNTVK L*(1026) LIGNTWSLIKVKE* (517) FIG.5.Alignment oftwoconserved domains amongtheSvvariable portions ofthepl5BSprotein, oftheP(-)variable partcodedinthe Pininversion systemofe14,oftheStfproteinofphageA,ofprotein37ofphageT4,andofprotein12ofphageT4.Xintheseconddomain ofthee14sequence identifies thesiteatwhich asinglebasepairwasintroduced toallowalignment ofthecarboxy-terminal ends.Asterisks areappended tothecarboxy-terminal endsoftheproteins. Forothersymbols, seethelegend toFig.3. segments arehighlyhomologous, theDNAwhichhasre- placedtheDNAinversion regionoftheancestral p15B genome musthavebeencloselyrelatedtothee14Pinregion. Asthediagram ofFig.4shows,severalofthebreakpoints between homology andnonhomology amongthePinand Minsystems andtheCinandGinsystems, respectively, also coincide withcrossover sitesforDinrecombinases.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Forothersymbols, seethelegend toFig.3. segments arehighlyhomologous, theDNAwhichhasre- placedtheDNAinversion regionoftheancestral p15B genome musthavebeencloselyrelatedtothee14Pinregion. Asthediagram ofFig.4shows,severalofthebreakpoints between homology andnonhomology amongthePinand Minsystems andtheCinandGinsystems, respectively, also coincide withcrossover sitesforDinrecombinases. The breakpoints oftheDNAinversion regionsofShigella boydii anditsclosestrelative, theGinsystemofphageMu,whose invertible segments have62to98%identical bases,also coincide withthecrossover sitesonbothgenomes (39).This supports anearlierpostulate thatDinrecombinases could mediate segmental substitutions amongDNAinversion sys- tems(12,16).Otherbreakpoints whichexhibit noobvious similarity toconsensus crossover sitescouldnevertheless oncehavebeenusedassubstrates forsuchrecombinases. It hasbeenshownelsewhere thatDNAinvertases canutilize secondary crossover siteswithquitedegenerate sequences (14,15),andsuchsitesmighthavebeenmutated during evolution subsequent toasubstitution event. Homologous domains ofgenescontained inpl5B,P1,Mu, ande14DNAinversion regionsandintailfibergenesofphages T4,P2,andX.Screening ofnucleic acidandaminoacid sequence databasesrevealed unexpected relationships amongthecodingsequences ofDNAinversion systems of TABLE 2.Percentages ofidentical (andsimilar) residues of belements alignedinFig.3B %ofidentical (%ofsimilar)\" residues with: Element X401-StfPlScblPlScb2 P2Hb b P1Scbl 100 P1Scb2 68(81) 100 P2Hb 68(82) 84(89) 100 X401-Stfb 65(81) 77(86) 86(92) 100 \"Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8).pl5B,e14,andphageP1andgenesofotherwise unrelated bacteriophages. Thepartlyrelatedandrepeated elements inthepl5Band P1Scgeneshadcounterparts intailfibergenesofphages Mu,P2,andX.Segments homologous tothepl5BandP1 elements designated awerefoundinduplicate inthephage P2geneHandinsinglecopiesintheconstant part,Sc,ofthe MutailfibergeneS,andinXORF401.These44-aaelements present onfivedistinctgenomes havebetween 55and93% identical aminoacids(Fig.3A;Table1).Theclosestrela- tionships, withatleast80%identical aminoacids,are between thepl5Belements altoa7amongeachother,the element alofP1andalofpl5B,theelement alofP2andal, a3,anda6ofpl5B,andtheelement a2ofP2anda4anda7 ofpl5Bandthecorresponding Xsequence. Thetandemrepeated elements blandb2oftheP1genome andthecelements inP1andpl5Barerelatedtosequences intheP2andXgenomes (Fig.3BandC).Thebelements havebetween 65and86%identical aminoacids(Table2), andthecelements havebetween 61and70%identical amino acids(Table3).Notably, thehighlyconserved a,b,andc elements areimmediately adjacent toeachotherinthepl5B, P1,P2,andXgenomes.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': [\"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  VOL.174,1992 ~~~~EVOLUTION OFpl5B3ANDPHAGE P1TAILFIBERGENES 3937 AccI  eal~~~~~~~~~~~~~~~~~~~~~~~~~~~~~0YTAQDASTTQKGLVQLSSATNSDSETMAATPKA *BglII alN,.,e..bl . .  * ~~~~~StyI bl%,,.~.b2. INPYTTINNNLA PLASPA LTGVPTAP200 300 400 500 TAAQGTNNTQ IATTAY VRAAI SALVGSSPEALD T b%..c SfaNI CCCTGTATGAGCTTGCAGCAGCACTGGGCAATGACCCGAACTTTGCGACAACAATGACAAATGCGCTGGCAGGGAAA,CAGCCACTTIU2gC-AACTTTAAC 600 LYELAAALGNDPNFATTMTNALAG --------- 700 ALAGL ATG ANKL PY FT GT DTVSQ TD LTSVG RD I CTGGCCAAAACAAGCATTCTTGCT ITA TCOTA 760 LAK TSI LAV IQYLGRE LGT cixR FIG.1.Partialsequence oftheconstantportion(Sc)ofP1Stailfibergeneanditspredicted aminoacidsequence. Restriction enzymesites areprintedinboldfacetype,andthecixRcrossover siteisboxed.Bordersofrepeated elements(al,bl,b2,andc)areindicated abovethe sequence. Thesequencepartiallyoverlaps thatofafragmentcontaining the5'endofScuptotheBgllIsite(9,38)aswellasthatofa'noth'erfragmentcontaining thecixRsitewhichstartsfromtheSfaNIsite(12).Accession numberM81956includesthesequence shownhereandthe sequences oftheseflankingfragments. Finally, directevidence thattheinvertible segments ofpl5Bcodeforphagetailfiberswasobtainedbyacomplementationexperiment withaP1mutant defective initstailfiber synthesis. Theevolutionary relevance oftheseresultsis addressed. MATERIALS ANDMETHODS Bacterial andphagestrains.E.coliK-12WA921 ispin hsdKlacleumetthrsupE(40).M47,aderivative ofP1-15 prophage withatemperature-sensitive cirepressor gene, carries apolarmutation inthetailfiberoperonbecause of insertion of152(37).Plasmids. ThepBR322-derived plasmidpRM2O6 contains the7.9-kbBamHI-5 fragment ofthephageP1genome(29),andthepUC19-based plasmidpAW800 carriesthecorre- sponding 8.3-kbBamHI-5 fragment ofpl5B(17).Therele- vantfeatures oftheseBamHIfragments arealsoseeninFig.4.InbacteriacarryingpAW800, all240possibleconfigura-tionsoftheMinsystem arethought tobepresent, and therefore, plasmidpreparations ofpAW800 consist ofa mixture ofisomers. Tailfibercomplementation assay.WA921 cellsharboringthedefective P1-derived prophage M47weretransformed witheitherpRM206 orpAW800. Titersofinfectious parti-clesinlysatesproduced byheat-induced cellswereassayedbyplatingwithWA921(pRM206) indicator cellsonLuria agarplatescontaining ampicillin (200p.gIml). Plasmid pRM2O6 carriesthewild-type P1tailfiberoperon, sothe indicator bacteria canallowmultiplecyclesofreproductionofphageparticlesharboring themutantM47genome. Determination ofthePISctailfibergenesequence.\", \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Tailfibercomplementation assay.WA921 cellsharboringthedefective P1-derived prophage M47weretransformed witheitherpRM206 orpAW800. Titersofinfectious parti-clesinlysatesproduced byheat-induced cellswereassayedbyplatingwithWA921(pRM206) indicator cellsonLuria agarplatescontaining ampicillin (200p.gIml). Plasmid pRM2O6 carriesthewild-type P1tailfiberoperon, sothe indicator bacteria canallowmultiplecyclesofreproductionofphageparticlesharboring themutantM47genome. Determination ofthePISctailfibergenesequence. The 1.7-kbIApnlfragment including thecixRcrossover siteanda largepartoftheP1SctailfibergeneofphageP1wascutout frompRM2O6 andligatedintotheKpnlsiteofpEMBL18 to generatepAW81O andintotheKpnlsiteofpEMBL19 to generatepAW811. pAW812 andpAW813 areAccldeletion derivatives ofthepAW81O andpAW811plasmids, respec- tively.PlasmidpAW814 resulted fromaStyl-PstI deletion of pAW812, andpAW815 resulted fromaStyl-Kpnl deletion of pAW813. DNAsequencing wasperformed bythechain termination method(35)usingsingle-stranded DNApro-ducedfromplasmids pAW812, pAW813, pAW814, and pAW815.Laboratory procedures. Cloning wascarried outasde- scribedbyManiatis etal.(22).Computer analysis. Homology searches andsequence alignments weremadewiththeprograms oftheGenetics Computer GroupoftheUniversity ofWisconsin (5).EMBL andGenBank DNAsequence databasesandMIPSXproteinsequence databasescontaining updates untilSeptember1991werescreened withtheprograms WORDSEARCH and PROFILESEARCH. Nucleotide sequence accessi'on numbers. Theaccession number ofthe2.5-kbP1sequence whichincludes these- quence oftheAccI-KpnI fragment showninFig.1is M81956. Theaccession numbers ofDNAsequences with homology toP1orpl5BDNAreported inthisstudy(andthe corresponding references) areasfollows: forthepl5BMin region,M62121(33);forphageP1totheleftoftheinvertible segment, M81956(thisworkand9and12);onlytheamino acidsequence oftheentireP1invertible segment isavailable (20);forfragments oftheP1invertible segment, K03173 and X01828(12);forphageP1totherightoftheinvertible segment, X01828(12,15a);forthee14Pinregion,.K03521 (31);noaccession number fortheGinregionofphageMu (19);forphageX,J02459(3);forphageP2,M64677(11);for phageT4gene12,X06729(36);forphageT4gene37,J02509 (30);andforphageT4gene38,X05677(26). RESULTS ANDDISCUSSION ScgenesofP1andpl5Bcontainpartially relatedarraysof repeated elements.\", \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  RESULTS ANDDISCUSSION ScgenesofP1andpl5Bcontainpartially relatedarraysof repeated elements. TheDNAsequence oftheScgeneofP1 wascompleted bysubcloning andsequencing anAccI-KpnIfragment ofphageP1including thecixRcrossover sitethat separates 739bpoftheconstantportion(Sc)from21bpof onevariableportion(Sv)ofthetailfibergeneS(Fig.1).Thestronghomology of99%identical basesofthecon- stantpart(Sc)ofgenesSofpl5BandP1includes the elements termedaO(Fig.2).Theseshort36-bpsequences share58and64%identical baseswiththe3'endsofthe succeeding 132-bpelements alofP1andp15B,respectively.Sixelements, termeda2toa7,sharebetween 64and100%VOL.174,1992 Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.\", \"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  3938 SANDMEIER ETAL. mix aOala2a3a4a5a6a7cmix'Il. p15B aI_,.'.'.'. ......................'...'.'.'.'.'.'.'.'.'.'.'.'.'.'.' I~~~~~~~~~~~................_..... ............................................................... .... ............ .............................  zato .......w o. wocst z :.::.:.::.::.:::: ...:,.::,::,..:,:,:,,::,:.:.:::::,:::::::.. :,::, ::,:::.: ..n::.:.:::.::: :::,::,:...... ....... 1.......--........aCa1b1b2.cix 200bpI Sc SvAcciBgIllStylSfaNIKpni regionsequenced FIG.2.Structure andhomology amongrepeated elements inS genesofpl5BandP1.TheaOsmallelements arehomologous tothe 3'endsofelements alinP1andinp15BDNAs.Elements a2toa7 ofp15Barehomologous toaOandalofP1,exceptforthe5'parts ofa3anda6,whichareshownasopenboxes.Thelongarrows represent aduplication ofelements a2toa4anda5toa7,respec- tively.Adjacent invertible segments areshownasopenboxes,and crossover sites(mixandcixR)areshownastriangles. Thelatter separate theconstant (Sc)fromthevariable parts(SvandSvQ)of genesS.Sequence homologies between thep15BandP1DNAsare indicated inpercentages. Thesegment fromwhichthesequence in Fig.1wasdetermined isindicated. Therestriction enzyme sites usedforcloningandsequencing areindicated.identity withaOandalofpl5Bandfollowalonpl5BDNA asdirectrepeats. Itseemsthatthea5-to-a7 segment isa recentduplication ofa2toa4,sincethesetwo432-bpand 144-amino-acid (aa)segments contain onlyfournucleotide substitutions leading toonlyoneaminoacidchange. The elements a3anda6shareattheir5'endshighlyhomologous 51-bpsequences codingfor17aawhicharenotrelated to anysequences oftheotherrepeated elements onpl5Borto aOandalonP1(Fig.3A). NexttoalofP1,a444-bpand148-aastretchconsists of twodirectly repeated elements andexhibits nosimilarity to pl5BDNA(Fig.1and2).Theseelements aredesignated bl andb2andhave63%identical basesand68%identical aminoacids(Fig.3B). Between b2andthecixRcrossover siteflanking the invertible segment ofP1liesa162-bpand54-aasequence, termedc,whichresumes homology tothecorresponding c elementofp15BDNAbetween a7andthemixcrossover site (Fig.1and2).Thosecelements have59%identical bases and69%identical aminoacids(Fig.3C). Substitution oftheinvertible DNAandtherecombinase geneonanancestral pl5Bgenome. Dataobtained onhomol- ogiesbetween theCinandMinregions aresummarized in thecentralpartofFig.4.Strikingly, theinvertible DNA segments ofP1andpl5Bbetween theircrossover sitesare nothomologous. Theminandcingenesarelocated inboth systems adjacent totheinvertible segments, buttheyare carriedinopposite orientations.\", 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Substitution oftheinvertible DNAandtherecombinase geneonanancestral pl5Bgenome. Dataobtained onhomol- ogiesbetween theCinandMinregions aresummarized in thecentralpartofFig.4.Strikingly, theinvertible DNA segments ofP1andpl5Bbetween theircrossover sitesare nothomologous. Theminandcingenesarelocated inboth systems adjacent totheinvertible segments, buttheyare carriedinopposite orientations. Theirsequences have67% identical basesand74%identical aminoacids.Incontrast, (197)............................................(209) ............ L (253) .. T D (297)APSSHTHPWNQITGVPT LAT V (353)....... TA V (397)............ T D (441)APSSHTHPWNQITVVPT LAT V(497)....... TA V(197)............................................(209) ............ L DM (164) ............ RRHP LAFT (208).. TARL (107) ............ LDIY LNKLTDP A (351).VALE DR21 41 KKS(208) (252) A(296) (352) (396) A(440) (496) (540) KKS(208) SIKDT(252) (207) TVMETK(251) IMNAR(150) VVMETR(394) 1 21 41 61 74 APL-SPALTGTPTAPTA-QGTNNTQIANTAFV-AAI-AL---SP-AL-TL-ELAAALGNDPNFATTMTNALAGK (253) IES AS KTINGAGTDKIINYSINL(326)(327) A V A TYRSVGSEDY (400)(252) N TR MAVDSDNN (325)20834).D LR LADVIDA DNN D (21053) 1 21 41 54 QP-DATLTALAGLATAA-KLPYFTG-D-ASLT-LT-VGRDILAK-S---V--YL (541)AA AD VRAASAGTIQSLD(594)(401) L GN TTVQDS TILAIQ(454)(326) K DRF NVATK STVAAIE(379)(9)KN SKNAENAE0 NVADLE(62) FIG.3.Alignment ofrepeated andothershortelements present intheScprotein sequences ofpl5B,P1,andMu,inthetailfiberprotein HofphageP2,andintailfiberproteins ofphage X.Aconsensus sequence with aplurality ofsix(A)andofthree(BandC)wasderived. Hyphens areintroduced where noconsensus wasderived. Inthespecific sequences, blanksdenoteresidues thatconform totheconsensus sequence anddotsindicate gapsintroduced foroptimalalignment. Numbers inparentheses givethecoordinates oftheaminoacidsequences predicted fromthepublished DNA sequences. (A)Alignment ofelements designated a.Therepeated elements aOtoa7ofpl5Bhaveformerly beencalledreOtore7(33).(B)Alignment ofelements designated b.Theelement bofX401-Stf hasbeendeduced from anintergenic region (seetext),andtherefore, thenumbers inparentheses arecoordinates ofthe Xgenome. (C)Alignment ofelements designated c.56 YTAQDASTTQKGIVQLSSATNSTSETLAATPKAVKAAYDLANGKA consensus pl5BScaO pl5BScal p15BSca2 pl5BSca3 p15BSca4 p15BScaS p15BSca plSBSca7 P1ScaO P1Scal P2Hal P2Ha2 MuSca )401a B cOnsensu P1Scbl P1Scb2 P2Hb X401-Stf b(2 C consensus plSBScc P1Scc P2Hc ).StfcJ.BACTERIOL. Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  EVOLUTION OFplSBANDPHAGE P1TAILFIBER GENES 3939 -=DC==>gin P.Sc Sv U U Sv FIG.4.Mapofhomologous DNAsequences amongtheDNA inversion regionsofe14,pl5B,P1,andMu.P1andp1SBregionsare represented bytheirBamHI-5 fragments. Invertible segments are shownasboxesflanked bycrossover sitesdepicted asblack triangles. Genesofoperons thatareaffected byDNAinversion and DNAinvertase genesareshownasopenarrows. TheP1cingeneis opposite inorientation tothatoftherelatedDNAinvertase genes. ThePspromoters oftailfibergenesareshownasshortblack arrows.Homology between pl5BandP1totheleftoftheinvertible segments issimplified fromthatshowninFig.2.Therefore, thea,b, andcelements composing theseregions arenotspecifically shown here.Thesiteofintegration ofIS2intoP1prophage M47ismarked. Theindicated highdegreeofhomology ofCinandGininvertible segments (20)isinterrupted byanapproximately 190-bpsegment of only35%homology aroundtheendofSvandthestartofUgenes. Thesmall63%homologous sequences inScofP1andMucorre- spondtotheaelements whicharealignedinFig.3A.The67-bp sequences between thecrossover pointsandtheDNAinvertase genesofpl5Bande14show31%homology. remarkable homologies inthesameorientation areseenon boththerightandtheleftsidesoftheinvertible segments andtheirrecombinase genes.Totheright,homologous DNA onP1startsstillwithinthecingeneandhence,thepl5B genome harbors aremnant fragment ofcin,thepseudo-cin gene(ilcin).Ithasbeenshownelsewhere thattlcinandthe recombinational enhancer sequence ontjcinarenonfunc- tional(33).Presence ofthetlcingeneindicates thattheDNA inversion regionofanancestral p15Bgenome hadcarrieda site-specific recombinase geneoftheDinfamilyinthesameconfiguration asthatoftheP1genome. Itismostlikelythat thecingeneandtheinvertible DNAofthisancestral pl5B genome hadbeensubstituted bytheMinsystemcomposed ofthemingeneandtheadjacent multiple invertible DNA segments. Sincethebreakpoints fromhomology tononho- mology between p15BandP1DNA aresharpandsomeof themcoincide withthecrossover sitesforDinrecombinases, itistempting tospeculate thataDinsite-specific recombi- nasehadmediated thesubstitution process. Thestructures ofrepeated elements neartheleftbreakpoints onp15Band P1suggest thatsubstitution mighthavebeenassociated with complicated recombinational processes. Alternatively, the presence ofthesestructures andthemultiple invertible DNA segments ofpl5Bcouldhaveresulted fromindependent events. Relation between pl5Bande14invertible DNAsegments.', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Sincethebreakpoints fromhomology tononho- mology between p15BandP1DNA aresharpandsomeof themcoincide withthecrossover sitesforDinrecombinases, itistempting tospeculate thataDinsite-specific recombi- nasehadmediated thesubstitution process. Thestructures ofrepeated elements neartheleftbreakpoints onp15Band P1suggest thatsubstitution mighthavebeenassociated with complicated recombinational processes. Alternatively, the presence ofthesestructures andthemultiple invertible DNA segments ofpl5Bcouldhaveresulted fromindependent events. Relation between pl5Bande14invertible DNAsegments. Comparison withthesequences ofotherDNAinversion systems revealed thatthefiveinvertible segments ofpl5B arerelated totheinvertible DNAofthePinsystemofe14.A 1.1-kb partofthe1.8-kbPininvertible segment (31)is between 60and80%homologous tothe3.5-kbMininvertible segments (Fig.4).ThesixSvcarboxy-terminal variable segments ofpl5BsharewiththeP(-)carboxy-terminal segment ofe14ahighlyhomologous domain ofabout70aa (Fig.5).When anucleotide isinserted afterposition 497of thepublished sequence ofthePinregionofe14toproduce a frameshift inP(-),thelast13aaofthepredicted P(-) product areidentical totheSvM1,SvP,andSvQandalmost identical totheSvMr,SvN,andSvOcarboxy-terminal segments (Fig.5).Thissecondhomologous domain issepa- ratedfromthefirstdomainby61to84aaintheSvandP(-) sequences. OnlytheSvMrandP(-)segments arehomolo- gousalsobetween the70-and13-aadomains (datanot shown). Thealternative P(+)carboxy-terminal segment of e14exhibits nosequence homology toP(-)ortopl5BSv segments. Thepostulated geneTofpl5B,two-thirds ofe14open reading frame(ORF)293,andtheirputative products have 75%oftheirbasesandaminoacidsincommon. Homology oftheMinandPinregionsincludes notonlythe invertible DNAsegments butalsotheadjacent pinandmin genes,whichhave72%identical basesand80%identical aminoacids.Preliminary sequence datashowed that0.6kb ofe14flanking theothersideofthePininvertible segment arenothomologous totheMinregion(32).', 'page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  OnlytheSvMrandP(-)segments arehomolo- gousalsobetween the70-and13-aadomains (datanot shown). Thealternative P(+)carboxy-terminal segment of e14exhibits nosequence homology toP(-)ortopl5BSv segments. Thepostulated geneTofpl5B,two-thirds ofe14open reading frame(ORF)293,andtheirputative products have 75%oftheirbasesandaminoacidsincommon. Homology oftheMinandPinregionsincludes notonlythe invertible DNAsegments butalsotheadjacent pinandmin genes,whichhave72%identical basesand80%identical aminoacids.Preliminary sequence datashowed that0.6kb ofe14flanking theothersideofthePininvertible segment arenothomologous totheMinregion(32). Sincethepingeneistheclosest relative ofthemin recombinase gene(17)andthepl5Bande14invertible DNA TABLE 1.Percentages ofidentical (andsimilar) residues ofaelements alignedinFig.3A %ofidentical (%ofsimilar) residues with: Element5Bcal pl5BScpl5BSca3pl5BSc PSa uc 2a 2a 4l EeplSB Scala2andaS anda6b a4anda7PlScalMuScaP2HalP2Ha2i401a pl5BScal 100 pl5BSca2andaS 89(91) 100 pl5BSca3anda6fi 90(92) 82(82) 100 pl5BSca4anda7 93(95) 91(91) 87(87) 100 P1Scal 82(86) 77(82) 69(77) 75(82) 100 MuSca 61(70) 55(66) 64(77) 57(68) 57(68) 100 P2Hal 80(82) 75(77) 86(89) 75(77) 66(73) 61(73) 100 P2Ha2 77(77) 75(77) 59(61) 80(82) 73(80) 59(70 64(66) 100 X401a 68(73) 68(70) 72(72) 68(70) 64(75) 57(77) 66(68) 80(86) 100 Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). hThefirst17aaofpl5Ba3anda6havenotbeenconsidered forthecalculation ofhomology values.VOL.174,1992 Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  3940 SANDMEIER ETAL. consensus pl5BSvM1 plSBSvMr P15BSVN pl5BSvO pl5BSvP P15BSVQ e14P(-) XStf T437 T412 consensus pl5BSvMl pl5BSvMr P1SBSVN plSBSvO plSBSvP P15BSVQ e14P(-) XStf T437 T4121 21 41 61 80 GLGEGSALPVGVPVPWPSATPPT. GWLKCNGAAFSSE-YPKLAK ....... AYPTNK. LPDLRGEFIRGWDDGRGVD-GR (1) (1) (1) (1) (1) (1) (1) (63) (806) (324)LE . M ....... KN........ L. LE . M ....... K N........ V L. M ....... KN....... AEE....... A N FAAI DIV S.YVLMQ DKSA .... SGV. KTDIM SYIAIDSVA.FALME QTDKSA V SGV.I QNEIDRTI AIMMAADSL SDARFHGTVASDCLYSRIGTRYGGN SPG 81 101 116 --LS-Q--- ......---H-H-- VI IGWL ...TGS YNI SI EGYA ...TED AGL AL NW PHT ..ILD AYM SL GTAT FIRTALQD ALITGM LEKRIV QL SGDA IRNIEGFA SIIGYA TEDAGL AV QEQDG IKS TSA AV AEADG. VKA SSA VNGNDFGKPRLGVGCTGGYVGEVQIQQMSY KAG(88) (88) (88) (88) (88) (88) (88) (145) (887) (439)(132) (147) (149) (149) (143) (135) (149) (295) (1007) (497)A Is A SIT A A IT NWTKKPAS..... MQTKKP.S..... MLVSGRSHLTNPN 1 21 ----E-RPRNIAFNYIVRAA* SGGNT *(153) TAGLT D*(166) NVGTT V *(168) GAWISM V T*(168) TGGST *(159) PTANN *(145) AAASAX *(165) AGNANTVK L*(314) TGNTNTVK L*(1026) LIGNTWSLIKVKE* (517) FIG.5.Alignment oftwoconserved domains amongtheSvvariable portions ofthepl5BSprotein, oftheP(-)variable partcodedinthe Pininversion systemofe14,oftheStfproteinofphageA,ofprotein37ofphageT4,andofprotein12ofphageT4.Xintheseconddomain ofthee14sequence identifies thesiteatwhich asinglebasepairwasintroduced toallowalignment ofthecarboxy-terminal ends.Asterisks areappended tothecarboxy-terminal endsoftheproteins. Forothersymbols, seethelegend toFig.3. segments arehighlyhomologous, theDNAwhichhasre- placedtheDNAinversion regionoftheancestral p15B genome musthavebeencloselyrelatedtothee14Pinregion. Asthediagram ofFig.4shows,severalofthebreakpoints between homology andnonhomology amongthePinand Minsystems andtheCinandGinsystems, respectively, also coincide withcrossover sitesforDinrecombinases.', 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Forothersymbols, seethelegend toFig.3. segments arehighlyhomologous, theDNAwhichhasre- placedtheDNAinversion regionoftheancestral p15B genome musthavebeencloselyrelatedtothee14Pinregion. Asthediagram ofFig.4shows,severalofthebreakpoints between homology andnonhomology amongthePinand Minsystems andtheCinandGinsystems, respectively, also coincide withcrossover sitesforDinrecombinases. The breakpoints oftheDNAinversion regionsofShigella boydii anditsclosestrelative, theGinsystemofphageMu,whose invertible segments have62to98%identical bases,also coincide withthecrossover sitesonbothgenomes (39).This supports anearlierpostulate thatDinrecombinases could mediate segmental substitutions amongDNAinversion sys- tems(12,16).Otherbreakpoints whichexhibit noobvious similarity toconsensus crossover sitescouldnevertheless oncehavebeenusedassubstrates forsuchrecombinases. It hasbeenshownelsewhere thatDNAinvertases canutilize secondary crossover siteswithquitedegenerate sequences (14,15),andsuchsitesmighthavebeenmutated during evolution subsequent toasubstitution event. Homologous domains ofgenescontained inpl5B,P1,Mu, ande14DNAinversion regionsandintailfibergenesofphages T4,P2,andX.Screening ofnucleic acidandaminoacid sequence databasesrevealed unexpected relationships amongthecodingsequences ofDNAinversion systems of TABLE 2.Percentages ofidentical (andsimilar) residues of belements alignedinFig.3B %ofidentical (%ofsimilar)\" residues with: Element X401-StfPlScblPlScb2 P2Hb b P1Scbl 100 P1Scb2 68(81) 100 P2Hb 68(82) 84(89) 100 X401-Stfb 65(81) 77(86) 86(92) 100 \"Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8).pl5B,e14,andphageP1andgenesofotherwise unrelated bacteriophages. Thepartlyrelatedandrepeated elements inthepl5Band P1Scgeneshadcounterparts intailfibergenesofphages Mu,P2,andX.Segments homologous tothepl5BandP1 elements designated awerefoundinduplicate inthephage P2geneHandinsinglecopiesintheconstant part,Sc,ofthe MutailfibergeneS,andinXORF401.These44-aaelements present onfivedistinctgenomes havebetween 55and93% identical aminoacids(Fig.3A;Table1).Theclosestrela- tionships, withatleast80%identical aminoacids,are between thepl5Belements altoa7amongeachother,the element alofP1andalofpl5B,theelement alofP2andal, a3,anda6ofpl5B,andtheelement a2ofP2anda4anda7 ofpl5Bandthecorresponding Xsequence. Thetandemrepeated elements blandb2oftheP1genome andthecelements inP1andpl5Barerelatedtosequences intheP2andXgenomes (Fig.3BandC).Thebelements havebetween 65and86%identical aminoacids(Table2), andthecelements havebetween 61and70%identical amino acids(Table3).Notably, thehighlyconserved a,b,andc elements areimmediately adjacent toeachotherinthepl5B, P1,P2,andXgenomes.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:10 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'88'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994192'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'348ms'), (b'x-request-id', b'51366db58715e839613d67512cdca28f'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249d9bf725d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:10 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'88'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994192'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'348ms'), (b'x-request-id', b'51366db58715e839613d67512cdca28f'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249d9bf725d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Thetandemrepeated elements blandb2oftheP1genome andthecelements inP1andpl5Barerelatedtosequences intheP2andXgenomes (Fig.3BandC).Thebelements havebetween 65and86%identical aminoacids(Table2), andthecelements havebetween 61and70%identical amino acids(Table3).Notably, thehighlyconserved a,b,andc elements areimmediately adjacent toeachotherinthepl5B, P1,P2,andXgenomes. Mostoftheaminoacidsequence of theXelement bshowninFig.3Bisdeduced froman intergenic regionbetween ORF401andgenestf.However, aninsertion ofasinglebasepairafterelement awithinORF TABLE 3.Percentages ofidentical (andsimilar) residues of celements alignedinFig.3C %ofidentical (%ofsimilar) residues with: Element pl5BScc PlScc P2Hc XStfc pl5BScc 100 P1Scc 69(81) 100 P2Hc 63(78) 69(80) 100 XStfc 61(76) 65(78) 70(81) 100 \"Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8).J.BACTECRIOL. Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  EVOLUTION OFpl5BANDPHAGE P1TAILFIBER GENES 3941 TABLE 4.Percentages ofidentical (andsimilar) residues ofthe largeconserved domainofproteinsequences alignedinFig.5 %ofidentical (%ofsimilar)a residues with: Element pl5BSvore14 AStf T437T412 p1SBSvore14P(-)h 100 XStf 45(58) 100 T437 38(56) 74(82)100 T412 28(45) 27(52)28(50)100 aHomology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). \"Aconsensus sequence ofthehomologous Svportions ofpl5BandofP(-) ofe14(Fig.5)isused. 401resultsinacontinuous reading framefor774aa,which includes ORF401,stf,andtheirintergenic region. Since thesecodingsequences arecontained withinanonessential regionoftheAgenome, thisputative longreading frame mighthavebeendisrupted bymutation without lossofphage viability. Itisintriguing thatthe3\\'endsofthecelements inP2and XDNA,whichdonotcontaininvertible DNA,coincide with sequences thatresemble crossover sitesforDinrecombi- nases.IntheP2genome, thiscrossover site-like sequence separates theelement cfromsequences homologous tothose oftailfibergenescontained withintheinvertible segments of MuandP1(datanotshown), andintheXgenome the crossover site-like sequence separates theelement cfrom sequences homologous tothoseofthevariable genesofthe pl5Bande14invertible DNAsegments (seebelow). Itis therefore possible thatDinrecombinases alsomediate seg- mentalsubstitution apartfrominvertible DNAandcontrib- utesignificantly totheevolution oftailfibergenes. Twoconserved domains amongthesixvariable endsof thepl5BSproteinandtheirP(-)homolog inthePinsystem provedtobesimilartotwodomains oftheStfproteinofX, aswellastotwodomains intheproduct ofphageT4gene37 (Fig.5andTable4).(Thestronghomology between proteins Stfand37hasbeennoticedbefore[6,24].)TheproductofT4 gene12sharessomeoftheresidues conserved amongthe formerproteindomains (Fig.5),although overallsequence similarity isquitelow(Table4). 1 21TABLE 5.Percentages ofidentical (andsimilar) residues of proteinsequences alignedinFig.6 %ofidentical (%ofsimilar)a residues with: Elementpl5BT e14293 XTfa T438 pl5BT 100 e14293 75(80) 100 XTfa 37(53) 36(52) 100 T438 21(39) 20(39) 33(51)b 100 aHomology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). bT4protein38andXTfahaveaslightlyhigherhomology value(36%[55%l)iftheyareoptimally alignedtoeachotherinsteadofoptimizing alignment of atleastthreesequences. Thecounterparts oftheproducts ofthepl5BgeneTandof e14ORF293areTfaofphageXandthegene38proteinof phageT4(Fig.6).Thedegreeofconservation ofthose sequences islowbutstillsignificant, especially concerning similarratherthanidentical aminoacids(Table5). Genes37and38ofphageT4codeforproteins necessary forassembly ofthesixlongtailfibersoftheT4virion(28).', \"page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  bT4protein38andXTfahaveaslightlyhigherhomology value(36%[55%l)iftheyareoptimally alignedtoeachotherinsteadofoptimizing alignment of atleastthreesequences. Thecounterparts oftheproducts ofthepl5BgeneTandof e14ORF293areTfaofphageXandthegene38proteinof phageT4(Fig.6).Thedegreeofconservation ofthose sequences islowbutstillsignificant, especially concerning similarratherthanidentical aminoacids(Table5). Genes37and38ofphageT4codeforproteins necessary forassembly ofthesixlongtailfibersoftheT4virion(28). Twomolecules ofprotein37formthedistalportionofthese fibers,whichcontact thehostreceptors, whileprotein38, whichisrequired fordimerization ofprotein37,isnota constituent ofthefibers.ORF401ofphageXandgenesstf(sidetailfiber;formerly calledORF314)andtfa(tailfiber assembly; formerly calledORF194)arepartofitsnones- sentialbregionatthedistalendofthePR'operon(2).Ithas beenshownthatthepresence ofORF401andgenesstfand tfainXprophage DNAcorrelates withtheappearance ofthe nonessential sidetailfibersinmature virions (25).The product oftfapossesses acatalytic function infiberassem- bly,sincetfacancomplement anambermutation inthe homologous T4gene38(25).Theproduct ofstfdetermines receptor specificity, sincereplacement ofa3'endofgene37 ofT4bystfproduces T4virionswithalteredhostspecifici- ties(27). Thesequence homology ofthep15BgeneTandofthe constant andvariable partsofthepl5BgeneStoseveral bacteriophage genomes, together withtheearlierreported highdegreeofhomology ofalargepartofScgenesandthe preceding geneRofpl5Btothecorresponding genesofthe 41 61 80 (1)MDNILNELIAIQAGNIIV DGGNR YSASTE LAG.V (1)MHKILNDLIATKAGDVTV DGERYSTSNE LAG.V (1) ..MFRME....QPRTIKI LLAGNFGEGDA IPPH.T (1) MIHYFD KFYRE.EN KPKGLINCLDGSH .ILREDLS EP ICCLDGTH .ICRADFN EP LNTDIPDIP FVVFNDE.A HLE LHTIKRLEPE .VVFDERTQD IEE 81 101 121 141 160 G-TVY-----G----I---G.--PEN-T---P---YDKW-G--WV-DTEA--..------AE----SL------S-----L ESIDTNPEETVL.DY TTIALTP DEKVQH..SAAVEA TKRQIDTAMD ISLIQ ESTRTESKEKAP.DY TTIALSP DEKTQH..SAAVDA AQRQIDAAMA ISLIQ KDVASDALFSEL.PL FWLSGGEQNTAKEK..LFRIRE ETRK M..QVA EHIAP K.RAWTFNREEIF SDISPVGITFDEP...GEF ITDDG.KEDTYKRVLIRNRKI ELYK.EF..QV....... 161 181 201 217 KLRGRKL. QATQNSVLD IDENAMLT LNKQLSTAS* KLQGRRL.\", 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  QATQNSVLD IDENAMLT LNKQLSTAS* KLQGRRL. QATRNAVLDIDAVTAT TS VILPERRPFNIWRTGCI QDADLEIA RESLEAWKR RVLNRVTS IEAVPVME* NNMIEASVANKK KFYYKNLKRFFAL ERHEHLGGEFPS KEQKPWYRRLFKHYV*(203) (211) (194) (183) FIG.6.Alignment oftheproduct ofpl5BTwiththee14ORF293product, Tfaofphage X,andprotein 38ofphageT4.Forsymbols, see thelegend toFig.5.p15BT e14293 ATfa T438 cornse pl5BT e14293 XTfaT438 conenus p15BT e14293 XTfa T438VOL.174,1992 Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', \"page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  3942 SANDMEIER ETAL. A B CFIG.7.Phageparticles obtained bycomplementation oftheP1tailfibermutantM47withpl5B-encoded tailfibers.Specimens were negatively stainedwith1%uranylacetate.Magnification bythetransmission electron microscope, x40,000; finalmagnification, x190,000. (A)Mutant M47prophage givesrisetoparticles without tailfiberstructures. (B)M47mutantcomplemented withtheP1tailfiberoperon on multicopy plasmid pRM206 isindistinguishable fromawild-type P1phage.(C)Complementation ofM47withthepl5BMinregionpresent onmulticopy plasmidpAW800 produces phageparticles withpl5B-encoded tailfibers. phageP1tailfiberoperon(Fig.4)(33),strongly suggests that theMinregioncontains tailfibergenes.Thebiological role ofthegenetic information contained inthePininvertible segment hasnotbeendetermined, butthehomologies of ORF293andP(-)toT4andKtailfibergenescouldsuggest thatthePinsystemisinvolved intailfibersynthesis. Complementation ofdefective P1prophage withtheMin regionofpl5B.Earlierelectron microscopy studieswiththe closelyrelatedP1andpl5Bgenomes (23)revealed thatthe majorpartsofthegenomic regionsencoding thetailgenes arehomologous. Furthermore, asdescribed above,theP1 amino-terminal sequences whichconfer totailfiberstheir attachment tothebaseplate (41)arealmostidentical to corresponding sequences inpl5B.Toinvestigate thefunc- tionalhomology ofthesegeneproducts, wehavetested whether amulticopy plasmid carrying thepl5BMinregion cancomplement P1mutantM47,whichisdefective fortail fibersynthesis. ThemutantM47carries anIS2insertion in the5'partoftheSgene.Multicopy plasmids pRM206 and pAW800 carry,respectively, theP1BamHI-5 fragment with thewild-type P1tailfiberoperon andthepl5BBamHI-5 fragment containing theputative tailfibergenesunderthe controlofthePspromoter (Fig.4).Ithasbeenshownthat thepl5BPspromoter isastructural andfunctional analogof theP1latepromoter Pstranscribing theP1tailfibergenes andthatitsactivitydepends onP1functions expressed after prophage induction (10).Hence, theputative tailfibergenes onpAW800 areexpected tobeexpressed attheappropriate timeinthepresence ofaninduced M47prophage. E.coliWA921 carrying theM47prophage wastrans- formed withpRM206 orpAW800, andlysates fromthe transformants wereprepared byheatinduction. Asseenin theelectron microscope, themorphologically intactphageparticles fromWA921(M47, pRM206) carriedtailfiberstruc- tures(Fig.7B),whileparticles fromWA921(M47) exhibited notailfibers(Fig.7A).Underthesamecondition, the majority ofmorphologically intactphagesproduced from heat-induced WA921(M47, pAW800) cellscarriedtailfiber structures (Fig.7C).However, theconcentration ofcom- pletephageparticles inbothlysatesofcellswiththetrans- complemented M47wasverylowcompared withthatof lysatesofcellswithM47orawild-type P1,andthemajority ofparticles wereemptyheadsandloosesheathstructures. SincetheMinregionappears toencodetailfibergenes, whoseproducts canattachtoaP1baseplate, wetested whether P1particles withpl5BtailfiberscouldinfectE.coli WA921(seeMaterials andMethods).\", 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  SincetheMinregionappears toencodetailfibergenes, whoseproducts canattachtoaP1baseplate, wetested whether P1particles withpl5BtailfiberscouldinfectE.coli WA921(seeMaterials andMethods). Asexpected, noinfec- tiousparticles (fewerthan102PFU/ml) couldbedetected on WA921 indicator cellsforlysatesofWA921(M47) cells, whereas 106PFU/ml wasmeasured onWA921(pRM206) indicator cellsforlysatesofWA921(M47, pRM206) cells. Sincearound10i\"PFU/ml isexpected withwild-type P1 prophages, theseresultsandtheelectron microscopy obser- vations confirm thattranscomplementation occurs, al- though quiteinefficiently. Inlysates ofWA921(M47, pAW800) cells,noplaques(fewerthan102PFU/ml) couldbe detected ontheWA921(pRM206) orWA921(pAW800) indi- catorstrain.Itislikelythatthepl5Btailfibergenesconfer hostspecificities otherthanthoseofP1andthatthepl5B geneproducts failedtointeract withE.coliK-12WA921. A searchforotherpotential hoststrainshasnotyetbeenmade. Functional aspectsoftailfibersandoftheirevolution. The pl5Btailfiberoperoncontains genesR,S,andT,underthe control ofthephage-specific latepromoter Ps(33).Most likelytheSvsequences determine thehostrange,andRand Scareamongthesequences responsible forattachment ofJ.BAC-TERIOL. Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', \"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  EVOLUTION OFpl5BANDPHAGE P1TAILFIBERGENES 3943 D. P2H ARF>ala2bc MuS a PiS ~11 IaOalblb2c p15BS-|aOaa2a3a4a5a6 c X401-stf-| 500bp abc FIG.8.Arrangement ofhomologous elements intheintermedi- atepartoflongtailfibergenes.Genesaredepicted inthe5'to3' direction, andhomologous regions areboxed.Thecorresponding aminoacidsequences ofthea,b,andcelements arealignedinFig. 3.Homologous sequences amongthe3'domains (Sv)ofthepl5B geneSandX401-stfareshowninFig.5,whilethoseofthe3' domains oftheP2geneH,MugeneS,andP1geneSarenotshown. TheXtailfibergeneisrepresented asafusionofORF401andgene stf(seetext).DNAinversion crossover sitesandcrossover site-like sequences areshownasblackandopentriangles, respectively. thetailfiberstothebaseplate ofthevirion.TheproductofT probably possesses acatalytic roleintailfiberassembly of oneofthesixfibervariants. Expression ofTdepends onthe configuration oftheinvertible segments intheMinregion, anditismostlikelytranslationally coupled toSvMJ(33). SinceMininversion alternatively provides expression of eachofthesixvariants ofS(33),sixdifferent tailfibers,each encoding adistincthostrange,couldbesynthesized. How- ever,itisexpected thatindividual phageparticles would usually carrypl5Btailfibersofonlyonehostspecificity corresponding tothegeneticconfiguration oftheparental genome. Thephagesproduced duringlyticgrowthofphage P1conferpredominantly oneofthetwopossible host specificities (13),andDNAinversions byCinaswellasby Minrecombinases arelow-frequency events. Thebiological significance ofamultiple extension ofthe hostphagesuchasthatdisplayed bypl5Bisuncertain. Although itissufficient formostphagestodetermine only onehostrange,itcouldallowaphagetomorereadilyadapt tochanging environments composed ofseveral bacterial species, oraphagecouldevadeadsorption-resistant host mutations bybinding toalternate receptors. Itisstriking thatthelargetailfibergenesofseveral unrelated, well-studied phagesofE.colicarrydomains of highlysimilarsequences. Suchhomologous domains are observed amongthe3'ends(e.g.,amonggeneSofP1,Sof Mu,andHofP2andamongSvgenesofpl5B,gene37ofT4, andstfofX)whichconferhostspecificity, whilenohomol- ogyhassofarbeendetected amongthe5'ends,which specifytheattachment tothebaseplates.Itisshownhere thattheintermediate partsofthetailfibergenesofsome phagesarearrangements ofseveralshortelements (Fig.8). Threetypesofsuchshortelements (a,b,andc),each presentinmorethanonegenome, havebeendetected among theavailable sequences. Theelements maybeamplified(e.g.,altoa7inp15BgeneS,alanda2inP2geneH,andblandb2inP1geneS).Theelements presentinagenome appearinthesameorder,i.e.,abeforebbeforec.The function oftheseelements duringtheinfection process isnot known. Theintermediate region,withitsmosaiclike struc- ture,isexpected tooccupy acorresponding position onthe tailfiberprotein, sincethetertiary structure oftailfiber proteins canbeconsidered largelylinear.Thiscanserveas abasisforstudiesoftheirroleininteractions between tail fibersandmacromolecules onthesurfaceofthehostbacte- rium,including reversible andirreversible adsorption. Thedatapresented abovegiveaninsightintothelong- termevolution ofgenes,whichareunderstrongselective pressure.\", \"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Theintermediate region,withitsmosaiclike struc- ture,isexpected tooccupy acorresponding position onthe tailfiberprotein, sincethetertiary structure oftailfiber proteins canbeconsidered largelylinear.Thiscanserveas abasisforstudiesoftheirroleininteractions between tail fibersandmacromolecules onthesurfaceofthehostbacte- rium,including reversible andirreversible adsorption. Thedatapresented abovegiveaninsightintothelong- termevolution ofgenes,whichareunderstrongselective pressure. Obviously, otherwise unrelated phagesexchanged shortdomains byhorizontal transfer whichresulted in chimeric tailfibergenes.Thepresence ofcrossover sitesand crossover site-like sequences forDNAinvertases between 3' andintermediate domains suggests adecisive roleforsite- specificrecombination inevolutionary reshuffling. ACKNOWLElDGMENTS WethankMichelWurtzforelectron microscopy andReinhard Doelzformaintaining anexcellent computer software environment. Thisworkwassupported bygrant3.624.087 oftheSwissNational Science Foundation. ADDENDUM Inanindependent study,E.Haggard-Ljungquist and coauthors (11)carriedoutacomplementary comparison of tailfibergenes,andtheyreached similarconclusions. REFERENCES 1.Arber,W.Unpublished data. la.Brody, H.,A.Greener, andC.W.Hill.1985.Excision and reintegration oftheEscherichia coliK-12chromosomal element e14.J.Bacteriol. 161:1112-1117. 2.Daniels, D.L.,J.L.Schroeder, W.Szybalski, F.Sanger, and F.R.Blattner. 1983.Amolecular mapofcoliphage lambda, p. 469-517. InR.W.Hendrix, J.W.Roberts, F.W.Stahl,and R.A.Weisberg (ed.),Lambda II.ColdSpringHarborLabora- tory,ColdSpringHarbor, N.Y. 3.Daniels, D.L.,J.L.Schroeder, W.Szybalski, F.Sanger, A.R. Coulson, G.F.Hong,D.F.Hill,G.B.Petersen, andF.R. Blattner. 1983.Complete annotated lambda sequence, p.519- 676.InR.W.Hendrix, J.W.Roberts, F.W.Stahl,andR.A. Weisberg (ed.),Lambda II.ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 4.Dayhoff M.O.,(ed.).1979.Atlasofprotein sequence and structure. National Biomedical Research Foundation, Washing- ton,D.C. 5.Devereux, J.,P.Haeberli, and0.Smithies. 1984.Acomprehen- sivesetofsequence analysis programs fortheVAX.Nucleic AcidsRes.12:387-395. 6.George, D.G.,L.-S.L.Yeh,andW.C.Barker. 1983.Unex- pectedrelationships between bacteriophage lambdahypotheti- calproteins andbacteriophage T4tail-fiber proteins. Biochem. Biophys. Res.Commun. 115:1061-1068. 7.Glasgow, A.C.,K.T.Hughes, andM.I.Simon.1989.Bacterial DNAinversion systems, p.637-659. InD.E.BergandM.M. Howe(ed.),MobileDNA.American Society forMicrobiology, Washington, D.C. 8.Gribskov, M.,andR.R.Burgess. 1986.SigmafactorsfromE. coli,B.subtilis,phageSPOl,andphageT4arehomologousproteins. Nucleic AcidsRes.14:6745-6763.9.Guidolin, A.,J.-M.Zingg,andW.Arber.1989.Organization of thebacteriophage P1tail-fibre operon. Gene76:239-243. 10.Guidolin, A.,J.-M.Zingg,H.Lehnherr, andW.Arber.1989.\", 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  7.Glasgow, A.C.,K.T.Hughes, andM.I.Simon.1989.Bacterial DNAinversion systems, p.637-659. InD.E.BergandM.M. Howe(ed.),MobileDNA.American Society forMicrobiology, Washington, D.C. 8.Gribskov, M.,andR.R.Burgess. 1986.SigmafactorsfromE. coli,B.subtilis,phageSPOl,andphageT4arehomologousproteins. Nucleic AcidsRes.14:6745-6763.9.Guidolin, A.,J.-M.Zingg,andW.Arber.1989.Organization of thebacteriophage P1tail-fibre operon. Gene76:239-243. 10.Guidolin, A.,J.-M.Zingg,H.Lehnherr, andW.Arber.1989. Bacteriophage P1tail-fibre anddaroperons areexpressed from homologous phage-specific latepromoter sequences. J.Mol.VOL.174,1992 Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  3944 SANDMEIER ETAL. Biol.208:615-622. 11.Haggard-Ljungquist, E.,C.Hailing, andR.Calendar. 1992. DNAsequence ofthetailfibergenesofbacteriophage P2: evidence forhorizontal transfer oftailfibergenesamongunre- latedbacteriophages. J.Bacteriol. 174:1462-1477. 12.Hiestand-Nauer, R.,andS.lida.1983.Sequence ofthesite- specificrecombinase genecinandofitssubstrates servinginthe inversion oftheCsegment ofbacteriophage P1.EMBO J. 2:1733-1740. 13.lida,S.1984.Bacteriophage P1carriestworelatedsetsofgenes determining itshostrangeintheinvertible Csegment ofits genome. Virology 134:421-434. 14.lida,S.,andR.Hiestand-Nauer. 1986.Localized conversion at thecrossover sequences inthesite-specific DNAinversion systemofbacteriophage P1.Cell45:71-79. 15.Iida,S.,andR.Hiestand-Nauer. 1987.Roleofthecentral dinucleotide atthecrossover sitesfortheselection ofquasisites inDNAinversion mediated bythesite-specific Cinrecombinase ofphageP1.Mol.Gen.Genet.208:464-468. 15a.Iida, S.,andR.Hiestand-Nauer. Unpublished data. 16.lida,S.,H.Huber, R.Hiestand-Nauer, J.Meyer,T.A.Bickle, andW.Arber.1984.Thebacteriophage P1site-specific recom- binaseCin:recombination eventsandDNArecognition se- quences. ColdSpringHarborSymp.Quant.Biol.49:769-777. 17.lida,S.,H.Sandmeier, P.Hubner, R.Hiestand-Nauer, K. Schneitz, andW.Arber.1990.TheMinDNAinversion enzyme ofplasmidpl5BofEscherichia coli15T-:anewmember ofthe Dinfamilyofsite-specific recombinases. Mol.Microbiol. 4:991- 997. 18.Ikeda,H.,M.Inuzuka, andJ.-I.Tomizawa. 1970.P1-like plasmid inEscherichia coli15.J.Mol.Biol.50:457-470. 19.Kahmann, R.,andD.Kamp.1987.Sequence oftherightendof Mu,p.297-308. InN.Symonds, A.Toussaint, P.vandePutte, andM.M.Howe(ed.),PhageMu.ColdSpringHarborLabo- ratory,ColdSpringHarbor, N.Y. 20.Kamp, D.,E.Kardas, W.Ritthaler, R.Sandulache, R. Schmucker, andB.Stern.1984.Comparative analysis ofinvert- ibleDNAinphagegenomes. ColdSpringHarborSymp.Quant. Biol.49:301-311. 21.Kutsukake, K.,T.Nakao,andT.Iino.1985.AgeneforDNA invertase andaninvertible DNAinEscherichia coliK-12.Gene 34:343-350. 22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 1982.Molecular cloning: alaboratory manual. ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 23.Meyer,J.,M.Stalhammar-Carlemalm, M.Streiff,S.lida,and W.Arber.1986.Sequence relations amongtheIncYplasmid plSB,P1,andP7prophages. Plasmid 16:81-89. 24.Michel,C.J.,B.Jaqc,D.G.Arques, andT.A.Bickle.1986.A remarkable aminoacidsequence homology between aphageT4 tailfibreprotein andORF314 ofphageXlocated inthetail operon. Gene44:147-150. 25.Montag, D.,andU.Henning.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Thetandemrepeated elements blandb2oftheP1genome andthecelements inP1andpl5Barerelatedtosequences intheP2andXgenomes (Fig.3BandC).Thebelements havebetween 65and86%identical aminoacids(Table2), andthecelements havebetween 61and70%identical amino acids(Table3).Notably, thehighlyconserved a,b,andc elements areimmediately adjacent toeachotherinthepl5B, P1,P2,andXgenomes. Mostoftheaminoacidsequence of theXelement bshowninFig.3Bisdeduced froman intergenic regionbetween ORF401andgenestf.However, aninsertion ofasinglebasepairafterelement awithinORF TABLE 3.Percentages ofidentical (andsimilar) residues of celements alignedinFig.3C %ofidentical (%ofsimilar) residues with: Element pl5BScc PlScc P2Hc XStfc pl5BScc 100 P1Scc 69(81) 100 P2Hc 63(78) 69(80) 100 XStfc 61(76) 65(78) 70(81) 100 \"Homology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8).J.BACTECRIOL. Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  EVOLUTION OFpl5BANDPHAGE P1TAILFIBER GENES 3941 TABLE 4.Percentages ofidentical (andsimilar) residues ofthe largeconserved domainofproteinsequences alignedinFig.5 %ofidentical (%ofsimilar)a residues with: Element pl5BSvore14 AStf T437T412 p1SBSvore14P(-)h 100 XStf 45(58) 100 T437 38(56) 74(82)100 T412 28(45) 27(52)28(50)100 aHomology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). \"Aconsensus sequence ofthehomologous Svportions ofpl5BandofP(-) ofe14(Fig.5)isused. 401resultsinacontinuous reading framefor774aa,which includes ORF401,stf,andtheirintergenic region. Since thesecodingsequences arecontained withinanonessential regionoftheAgenome, thisputative longreading frame mighthavebeendisrupted bymutation without lossofphage viability. Itisintriguing thatthe3\\'endsofthecelements inP2and XDNA,whichdonotcontaininvertible DNA,coincide with sequences thatresemble crossover sitesforDinrecombi- nases.IntheP2genome, thiscrossover site-like sequence separates theelement cfromsequences homologous tothose oftailfibergenescontained withintheinvertible segments of MuandP1(datanotshown), andintheXgenome the crossover site-like sequence separates theelement cfrom sequences homologous tothoseofthevariable genesofthe pl5Bande14invertible DNAsegments (seebelow). Itis therefore possible thatDinrecombinases alsomediate seg- mentalsubstitution apartfrominvertible DNAandcontrib- utesignificantly totheevolution oftailfibergenes. Twoconserved domains amongthesixvariable endsof thepl5BSproteinandtheirP(-)homolog inthePinsystem provedtobesimilartotwodomains oftheStfproteinofX, aswellastotwodomains intheproduct ofphageT4gene37 (Fig.5andTable4).(Thestronghomology between proteins Stfand37hasbeennoticedbefore[6,24].)TheproductofT4 gene12sharessomeoftheresidues conserved amongthe formerproteindomains (Fig.5),although overallsequence similarity isquitelow(Table4). 1 21TABLE 5.Percentages ofidentical (andsimilar) residues of proteinsequences alignedinFig.6 %ofidentical (%ofsimilar)a residues with: Elementpl5BT e14293 XTfa T438 pl5BT 100 e14293 75(80) 100 XTfa 37(53) 36(52) 100 T438 21(39) 20(39) 33(51)b 100 aHomology percentages forsimilaraminoacidsareaccording tomethods described previously (4,8). bT4protein38andXTfahaveaslightlyhigherhomology value(36%[55%l)iftheyareoptimally alignedtoeachotherinsteadofoptimizing alignment of atleastthreesequences. Thecounterparts oftheproducts ofthepl5BgeneTandof e14ORF293areTfaofphageXandthegene38proteinof phageT4(Fig.6).Thedegreeofconservation ofthose sequences islowbutstillsignificant, especially concerning similarratherthanidentical aminoacids(Table5). Genes37and38ofphageT4codeforproteins necessary forassembly ofthesixlongtailfibersoftheT4virion(28).', \"page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  bT4protein38andXTfahaveaslightlyhigherhomology value(36%[55%l)iftheyareoptimally alignedtoeachotherinsteadofoptimizing alignment of atleastthreesequences. Thecounterparts oftheproducts ofthepl5BgeneTandof e14ORF293areTfaofphageXandthegene38proteinof phageT4(Fig.6).Thedegreeofconservation ofthose sequences islowbutstillsignificant, especially concerning similarratherthanidentical aminoacids(Table5). Genes37and38ofphageT4codeforproteins necessary forassembly ofthesixlongtailfibersoftheT4virion(28). Twomolecules ofprotein37formthedistalportionofthese fibers,whichcontact thehostreceptors, whileprotein38, whichisrequired fordimerization ofprotein37,isnota constituent ofthefibers.ORF401ofphageXandgenesstf(sidetailfiber;formerly calledORF314)andtfa(tailfiber assembly; formerly calledORF194)arepartofitsnones- sentialbregionatthedistalendofthePR'operon(2).Ithas beenshownthatthepresence ofORF401andgenesstfand tfainXprophage DNAcorrelates withtheappearance ofthe nonessential sidetailfibersinmature virions (25).The product oftfapossesses acatalytic function infiberassem- bly,sincetfacancomplement anambermutation inthe homologous T4gene38(25).Theproduct ofstfdetermines receptor specificity, sincereplacement ofa3'endofgene37 ofT4bystfproduces T4virionswithalteredhostspecifici- ties(27). Thesequence homology ofthep15BgeneTandofthe constant andvariable partsofthepl5BgeneStoseveral bacteriophage genomes, together withtheearlierreported highdegreeofhomology ofalargepartofScgenesandthe preceding geneRofpl5Btothecorresponding genesofthe 41 61 80 (1)MDNILNELIAIQAGNIIV DGGNR YSASTE LAG.V (1)MHKILNDLIATKAGDVTV DGERYSTSNE LAG.V (1) ..MFRME....QPRTIKI LLAGNFGEGDA IPPH.T (1) MIHYFD KFYRE.EN KPKGLINCLDGSH .ILREDLS EP ICCLDGTH .ICRADFN EP LNTDIPDIP FVVFNDE.A HLE LHTIKRLEPE .VVFDERTQD IEE 81 101 121 141 160 G-TVY-----G----I---G.--PEN-T---P---YDKW-G--WV-DTEA--..------AE----SL------S-----L ESIDTNPEETVL.DY TTIALTP DEKVQH..SAAVEA TKRQIDTAMD ISLIQ ESTRTESKEKAP.DY TTIALSP DEKTQH..SAAVDA AQRQIDAAMA ISLIQ KDVASDALFSEL.PL FWLSGGEQNTAKEK..LFRIRE ETRK M..QVA EHIAP K.RAWTFNREEIF SDISPVGITFDEP...GEF ITDDG.KEDTYKRVLIRNRKI ELYK.EF..QV....... 161 181 201 217 KLRGRKL. QATQNSVLD IDENAMLT LNKQLSTAS* KLQGRRL.\", 'page_label: 6 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  QATQNSVLD IDENAMLT LNKQLSTAS* KLQGRRL. QATRNAVLDIDAVTAT TS VILPERRPFNIWRTGCI QDADLEIA RESLEAWKR RVLNRVTS IEAVPVME* NNMIEASVANKK KFYYKNLKRFFAL ERHEHLGGEFPS KEQKPWYRRLFKHYV*(203) (211) (194) (183) FIG.6.Alignment oftheproduct ofpl5BTwiththee14ORF293product, Tfaofphage X,andprotein 38ofphageT4.Forsymbols, see thelegend toFig.5.p15BT e14293 ATfa T438 cornse pl5BT e14293 XTfaT438 conenus p15BT e14293 XTfa T438VOL.174,1992 Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', \"page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  3942 SANDMEIER ETAL. A B CFIG.7.Phageparticles obtained bycomplementation oftheP1tailfibermutantM47withpl5B-encoded tailfibers.Specimens were negatively stainedwith1%uranylacetate.Magnification bythetransmission electron microscope, x40,000; finalmagnification, x190,000. (A)Mutant M47prophage givesrisetoparticles without tailfiberstructures. (B)M47mutantcomplemented withtheP1tailfiberoperon on multicopy plasmid pRM206 isindistinguishable fromawild-type P1phage.(C)Complementation ofM47withthepl5BMinregionpresent onmulticopy plasmidpAW800 produces phageparticles withpl5B-encoded tailfibers. phageP1tailfiberoperon(Fig.4)(33),strongly suggests that theMinregioncontains tailfibergenes.Thebiological role ofthegenetic information contained inthePininvertible segment hasnotbeendetermined, butthehomologies of ORF293andP(-)toT4andKtailfibergenescouldsuggest thatthePinsystemisinvolved intailfibersynthesis. Complementation ofdefective P1prophage withtheMin regionofpl5B.Earlierelectron microscopy studieswiththe closelyrelatedP1andpl5Bgenomes (23)revealed thatthe majorpartsofthegenomic regionsencoding thetailgenes arehomologous. Furthermore, asdescribed above,theP1 amino-terminal sequences whichconfer totailfiberstheir attachment tothebaseplate (41)arealmostidentical to corresponding sequences inpl5B.Toinvestigate thefunc- tionalhomology ofthesegeneproducts, wehavetested whether amulticopy plasmid carrying thepl5BMinregion cancomplement P1mutantM47,whichisdefective fortail fibersynthesis. ThemutantM47carries anIS2insertion in the5'partoftheSgene.Multicopy plasmids pRM206 and pAW800 carry,respectively, theP1BamHI-5 fragment with thewild-type P1tailfiberoperon andthepl5BBamHI-5 fragment containing theputative tailfibergenesunderthe controlofthePspromoter (Fig.4).Ithasbeenshownthat thepl5BPspromoter isastructural andfunctional analogof theP1latepromoter Pstranscribing theP1tailfibergenes andthatitsactivitydepends onP1functions expressed after prophage induction (10).Hence, theputative tailfibergenes onpAW800 areexpected tobeexpressed attheappropriate timeinthepresence ofaninduced M47prophage. E.coliWA921 carrying theM47prophage wastrans- formed withpRM206 orpAW800, andlysates fromthe transformants wereprepared byheatinduction. Asseenin theelectron microscope, themorphologically intactphageparticles fromWA921(M47, pRM206) carriedtailfiberstruc- tures(Fig.7B),whileparticles fromWA921(M47) exhibited notailfibers(Fig.7A).Underthesamecondition, the majority ofmorphologically intactphagesproduced from heat-induced WA921(M47, pAW800) cellscarriedtailfiber structures (Fig.7C).However, theconcentration ofcom- pletephageparticles inbothlysatesofcellswiththetrans- complemented M47wasverylowcompared withthatof lysatesofcellswithM47orawild-type P1,andthemajority ofparticles wereemptyheadsandloosesheathstructures. SincetheMinregionappears toencodetailfibergenes, whoseproducts canattachtoaP1baseplate, wetested whether P1particles withpl5BtailfiberscouldinfectE.coli WA921(seeMaterials andMethods).\", 'page_label: 7 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  SincetheMinregionappears toencodetailfibergenes, whoseproducts canattachtoaP1baseplate, wetested whether P1particles withpl5BtailfiberscouldinfectE.coli WA921(seeMaterials andMethods). Asexpected, noinfec- tiousparticles (fewerthan102PFU/ml) couldbedetected on WA921 indicator cellsforlysatesofWA921(M47) cells, whereas 106PFU/ml wasmeasured onWA921(pRM206) indicator cellsforlysatesofWA921(M47, pRM206) cells. Sincearound10i\"PFU/ml isexpected withwild-type P1 prophages, theseresultsandtheelectron microscopy obser- vations confirm thattranscomplementation occurs, al- though quiteinefficiently. Inlysates ofWA921(M47, pAW800) cells,noplaques(fewerthan102PFU/ml) couldbe detected ontheWA921(pRM206) orWA921(pAW800) indi- catorstrain.Itislikelythatthepl5Btailfibergenesconfer hostspecificities otherthanthoseofP1andthatthepl5B geneproducts failedtointeract withE.coliK-12WA921. A searchforotherpotential hoststrainshasnotyetbeenmade. Functional aspectsoftailfibersandoftheirevolution. The pl5Btailfiberoperoncontains genesR,S,andT,underthe control ofthephage-specific latepromoter Ps(33).Most likelytheSvsequences determine thehostrange,andRand Scareamongthesequences responsible forattachment ofJ.BAC-TERIOL. Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', \"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  EVOLUTION OFpl5BANDPHAGE P1TAILFIBERGENES 3943 D. P2H ARF>ala2bc MuS a PiS ~11 IaOalblb2c p15BS-|aOaa2a3a4a5a6 c X401-stf-| 500bp abc FIG.8.Arrangement ofhomologous elements intheintermedi- atepartoflongtailfibergenes.Genesaredepicted inthe5'to3' direction, andhomologous regions areboxed.Thecorresponding aminoacidsequences ofthea,b,andcelements arealignedinFig. 3.Homologous sequences amongthe3'domains (Sv)ofthepl5B geneSandX401-stfareshowninFig.5,whilethoseofthe3' domains oftheP2geneH,MugeneS,andP1geneSarenotshown. TheXtailfibergeneisrepresented asafusionofORF401andgene stf(seetext).DNAinversion crossover sitesandcrossover site-like sequences areshownasblackandopentriangles, respectively. thetailfiberstothebaseplate ofthevirion.TheproductofT probably possesses acatalytic roleintailfiberassembly of oneofthesixfibervariants. Expression ofTdepends onthe configuration oftheinvertible segments intheMinregion, anditismostlikelytranslationally coupled toSvMJ(33). SinceMininversion alternatively provides expression of eachofthesixvariants ofS(33),sixdifferent tailfibers,each encoding adistincthostrange,couldbesynthesized. How- ever,itisexpected thatindividual phageparticles would usually carrypl5Btailfibersofonlyonehostspecificity corresponding tothegeneticconfiguration oftheparental genome. Thephagesproduced duringlyticgrowthofphage P1conferpredominantly oneofthetwopossible host specificities (13),andDNAinversions byCinaswellasby Minrecombinases arelow-frequency events. Thebiological significance ofamultiple extension ofthe hostphagesuchasthatdisplayed bypl5Bisuncertain. Although itissufficient formostphagestodetermine only onehostrange,itcouldallowaphagetomorereadilyadapt tochanging environments composed ofseveral bacterial species, oraphagecouldevadeadsorption-resistant host mutations bybinding toalternate receptors. Itisstriking thatthelargetailfibergenesofseveral unrelated, well-studied phagesofE.colicarrydomains of highlysimilarsequences. Suchhomologous domains are observed amongthe3'ends(e.g.,amonggeneSofP1,Sof Mu,andHofP2andamongSvgenesofpl5B,gene37ofT4, andstfofX)whichconferhostspecificity, whilenohomol- ogyhassofarbeendetected amongthe5'ends,which specifytheattachment tothebaseplates.Itisshownhere thattheintermediate partsofthetailfibergenesofsome phagesarearrangements ofseveralshortelements (Fig.8). Threetypesofsuchshortelements (a,b,andc),each presentinmorethanonegenome, havebeendetected among theavailable sequences. Theelements maybeamplified(e.g.,altoa7inp15BgeneS,alanda2inP2geneH,andblandb2inP1geneS).Theelements presentinagenome appearinthesameorder,i.e.,abeforebbeforec.The function oftheseelements duringtheinfection process isnot known. Theintermediate region,withitsmosaiclike struc- ture,isexpected tooccupy acorresponding position onthe tailfiberprotein, sincethetertiary structure oftailfiber proteins canbeconsidered largelylinear.Thiscanserveas abasisforstudiesoftheirroleininteractions between tail fibersandmacromolecules onthesurfaceofthehostbacte- rium,including reversible andirreversible adsorption. Thedatapresented abovegiveaninsightintothelong- termevolution ofgenes,whichareunderstrongselective pressure.\", \"page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  Theintermediate region,withitsmosaiclike struc- ture,isexpected tooccupy acorresponding position onthe tailfiberprotein, sincethetertiary structure oftailfiber proteins canbeconsidered largelylinear.Thiscanserveas abasisforstudiesoftheirroleininteractions between tail fibersandmacromolecules onthesurfaceofthehostbacte- rium,including reversible andirreversible adsorption. Thedatapresented abovegiveaninsightintothelong- termevolution ofgenes,whichareunderstrongselective pressure. Obviously, otherwise unrelated phagesexchanged shortdomains byhorizontal transfer whichresulted in chimeric tailfibergenes.Thepresence ofcrossover sitesand crossover site-like sequences forDNAinvertases between 3' andintermediate domains suggests adecisive roleforsite- specificrecombination inevolutionary reshuffling. ACKNOWLElDGMENTS WethankMichelWurtzforelectron microscopy andReinhard Doelzformaintaining anexcellent computer software environment. Thisworkwassupported bygrant3.624.087 oftheSwissNational Science Foundation. ADDENDUM Inanindependent study,E.Haggard-Ljungquist and coauthors (11)carriedoutacomplementary comparison of tailfibergenes,andtheyreached similarconclusions. REFERENCES 1.Arber,W.Unpublished data. la.Brody, H.,A.Greener, andC.W.Hill.1985.Excision and reintegration oftheEscherichia coliK-12chromosomal element e14.J.Bacteriol. 161:1112-1117. 2.Daniels, D.L.,J.L.Schroeder, W.Szybalski, F.Sanger, and F.R.Blattner. 1983.Amolecular mapofcoliphage lambda, p. 469-517. InR.W.Hendrix, J.W.Roberts, F.W.Stahl,and R.A.Weisberg (ed.),Lambda II.ColdSpringHarborLabora- tory,ColdSpringHarbor, N.Y. 3.Daniels, D.L.,J.L.Schroeder, W.Szybalski, F.Sanger, A.R. Coulson, G.F.Hong,D.F.Hill,G.B.Petersen, andF.R. Blattner. 1983.Complete annotated lambda sequence, p.519- 676.InR.W.Hendrix, J.W.Roberts, F.W.Stahl,andR.A. Weisberg (ed.),Lambda II.ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 4.Dayhoff M.O.,(ed.).1979.Atlasofprotein sequence and structure. National Biomedical Research Foundation, Washing- ton,D.C. 5.Devereux, J.,P.Haeberli, and0.Smithies. 1984.Acomprehen- sivesetofsequence analysis programs fortheVAX.Nucleic AcidsRes.12:387-395. 6.George, D.G.,L.-S.L.Yeh,andW.C.Barker. 1983.Unex- pectedrelationships between bacteriophage lambdahypotheti- calproteins andbacteriophage T4tail-fiber proteins. Biochem. Biophys. Res.Commun. 115:1061-1068. 7.Glasgow, A.C.,K.T.Hughes, andM.I.Simon.1989.Bacterial DNAinversion systems, p.637-659. InD.E.BergandM.M. Howe(ed.),MobileDNA.American Society forMicrobiology, Washington, D.C. 8.Gribskov, M.,andR.R.Burgess. 1986.SigmafactorsfromE. coli,B.subtilis,phageSPOl,andphageT4arehomologousproteins. Nucleic AcidsRes.14:6745-6763.9.Guidolin, A.,J.-M.Zingg,andW.Arber.1989.Organization of thebacteriophage P1tail-fibre operon. Gene76:239-243. 10.Guidolin, A.,J.-M.Zingg,H.Lehnherr, andW.Arber.1989.\", 'page_label: 8 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  7.Glasgow, A.C.,K.T.Hughes, andM.I.Simon.1989.Bacterial DNAinversion systems, p.637-659. InD.E.BergandM.M. Howe(ed.),MobileDNA.American Society forMicrobiology, Washington, D.C. 8.Gribskov, M.,andR.R.Burgess. 1986.SigmafactorsfromE. coli,B.subtilis,phageSPOl,andphageT4arehomologousproteins. Nucleic AcidsRes.14:6745-6763.9.Guidolin, A.,J.-M.Zingg,andW.Arber.1989.Organization of thebacteriophage P1tail-fibre operon. Gene76:239-243. 10.Guidolin, A.,J.-M.Zingg,H.Lehnherr, andW.Arber.1989. Bacteriophage P1tail-fibre anddaroperons areexpressed from homologous phage-specific latepromoter sequences. J.Mol.VOL.174,1992 Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  3944 SANDMEIER ETAL. Biol.208:615-622. 11.Haggard-Ljungquist, E.,C.Hailing, andR.Calendar. 1992. DNAsequence ofthetailfibergenesofbacteriophage P2: evidence forhorizontal transfer oftailfibergenesamongunre- latedbacteriophages. J.Bacteriol. 174:1462-1477. 12.Hiestand-Nauer, R.,andS.lida.1983.Sequence ofthesite- specificrecombinase genecinandofitssubstrates servinginthe inversion oftheCsegment ofbacteriophage P1.EMBO J. 2:1733-1740. 13.lida,S.1984.Bacteriophage P1carriestworelatedsetsofgenes determining itshostrangeintheinvertible Csegment ofits genome. Virology 134:421-434. 14.lida,S.,andR.Hiestand-Nauer. 1986.Localized conversion at thecrossover sequences inthesite-specific DNAinversion systemofbacteriophage P1.Cell45:71-79. 15.Iida,S.,andR.Hiestand-Nauer. 1987.Roleofthecentral dinucleotide atthecrossover sitesfortheselection ofquasisites inDNAinversion mediated bythesite-specific Cinrecombinase ofphageP1.Mol.Gen.Genet.208:464-468. 15a.Iida, S.,andR.Hiestand-Nauer. Unpublished data. 16.lida,S.,H.Huber, R.Hiestand-Nauer, J.Meyer,T.A.Bickle, andW.Arber.1984.Thebacteriophage P1site-specific recom- binaseCin:recombination eventsandDNArecognition se- quences. ColdSpringHarborSymp.Quant.Biol.49:769-777. 17.lida,S.,H.Sandmeier, P.Hubner, R.Hiestand-Nauer, K. Schneitz, andW.Arber.1990.TheMinDNAinversion enzyme ofplasmidpl5BofEscherichia coli15T-:anewmember ofthe Dinfamilyofsite-specific recombinases. Mol.Microbiol. 4:991- 997. 18.Ikeda,H.,M.Inuzuka, andJ.-I.Tomizawa. 1970.P1-like plasmid inEscherichia coli15.J.Mol.Biol.50:457-470. 19.Kahmann, R.,andD.Kamp.1987.Sequence oftherightendof Mu,p.297-308. InN.Symonds, A.Toussaint, P.vandePutte, andM.M.Howe(ed.),PhageMu.ColdSpringHarborLabo- ratory,ColdSpringHarbor, N.Y. 20.Kamp, D.,E.Kardas, W.Ritthaler, R.Sandulache, R. Schmucker, andB.Stern.1984.Comparative analysis ofinvert- ibleDNAinphagegenomes. ColdSpringHarborSymp.Quant. Biol.49:301-311. 21.Kutsukake, K.,T.Nakao,andT.Iino.1985.AgeneforDNA invertase andaninvertible DNAinEscherichia coliK-12.Gene 34:343-350. 22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 1982.Molecular cloning: alaboratory manual. ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 23.Meyer,J.,M.Stalhammar-Carlemalm, M.Streiff,S.lida,and W.Arber.1986.Sequence relations amongtheIncYplasmid plSB,P1,andP7prophages. Plasmid 16:81-89. 24.Michel,C.J.,B.Jaqc,D.G.Arques, andT.A.Bickle.1986.A remarkable aminoacidsequence homology between aphageT4 tailfibreprotein andORF314 ofphageXlocated inthetail operon. Gene44:147-150. 25.Montag, D.,andU.Henning.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:11 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'145'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994519'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'328ms'), (b'x-request-id', b'989cd576d5144d63c228c5fd36ae7eaa'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249de7ad25d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:11 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'145'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994519'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'328ms'), (b'x-request-id', b'989cd576d5144d63c228c5fd36ae7eaa'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249de7ad25d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 1982.Molecular cloning: alaboratory manual. ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 23.Meyer,J.,M.Stalhammar-Carlemalm, M.Streiff,S.lida,and W.Arber.1986.Sequence relations amongtheIncYplasmid plSB,P1,andP7prophages. Plasmid 16:81-89. 24.Michel,C.J.,B.Jaqc,D.G.Arques, andT.A.Bickle.1986.A remarkable aminoacidsequence homology between aphageT4 tailfibreprotein andORF314 ofphageXlocated inthetail operon. Gene44:147-150. 25.Montag, D.,andU.Henning. 1987.Anopenreadingframeinthe Escherichia colibacteriophage Xgenome encodes aproteinthat functions inassembly ofthelongtailfibersofbacteriophage T4.J.Bacteriol. 169:5884-5886. 26.Montag, D.,I.Riede,M.-L.Eschbach, M.Degen, andU. Henning. 1987.Receptor-recognizing proteins ofT-even type bacteriophages. Constant andhypervariable regions andan unusual caseofevolution. J.Mol.Biol.196:165-174. 27.Montag, D.,H.Schwarz, andU.Henning. 1989.Acomponent of thesidetailfiberofEscherichia colibacteriophage Xcan functionally replace thereceptor-recognizing partofalongtail fiberprotein oftheunrelated bacteriophage T4.J.Bacteriol. 171:4378-4384. 28.Mosig,G.,andF.Eiserling. 1988.PhageT4structure and metabolism, p.521-606. InR.Calendar (ed.),Thebacterio- phages. Plenum Press,NewYork. 29.Mural,R.J.,R.H.Chesney, D.Vapnek, M.M.Kropf,andJ.R. Scott.1979.Isolation andcharacterization ofclonedfragments ofbacteriophage P1DNA.Virology 93:387-397. 30.Oliver,D.B.,andR.A.Crowther. 1981.DNAsequence ofthe tailfibregenes36and37ofbacteriophage T4.J.Mol.Biol. 153:545-568. 31.Plasterk, R.H.A.,andP.vandePutte.1985.Theinvertible P-DNAsegment inthechromosome ofEscherichia coli.EMBO J.4:237-242. 32.Sandmeier, H.1990.Characterization ofacomplex DNAinver- sionsystem. Ph.D.thesis.BaselUniversity, Basel. 33.Sandmeier, H.,S.lida,P.Hubner, R.Hiestand-Nauer, andW. Arber.1991.Geneorganization inthemultiple DNAinversion regionMinofplasmid plSBofE.coli15T-:assemblage ofa variable gene.Nucleic AcidsRes.19:5831-5838. 34.Sandmeier, H.,S.lida,J.Meyer, R.Hiestand-Nauer, andW. Arber.1990.Site-specific DNArecombination systemMinof plasmid pl5B:aclusterofoverlapping invertible DNAseg- ments.Proc.Natl.Acad.Sci.USA87:1109-1113. 35.Sanger, F.,S.Nicklen, andA.R.Coulson. 1977.DNAsequenc- ingwithchain-terminating inhibitors. Proc.Natl.Acad.Sci. USA74:5463-5467. 36.Selivanov, N.A.,A.G.Prilipov, andV.V.Mesyanzhinov. 1988. Nucleotide anddeduced aminoacidsequence ofbacteriophage T4gene12.Nucleic AcidsRes.16:2334.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  34.Sandmeier, H.,S.lida,J.Meyer, R.Hiestand-Nauer, andW. Arber.1990.Site-specific DNArecombination systemMinof plasmid pl5B:aclusterofoverlapping invertible DNAseg- ments.Proc.Natl.Acad.Sci.USA87:1109-1113. 35.Sanger, F.,S.Nicklen, andA.R.Coulson. 1977.DNAsequenc- ingwithchain-terminating inhibitors. Proc.Natl.Acad.Sci. USA74:5463-5467. 36.Selivanov, N.A.,A.G.Prilipov, andV.V.Mesyanzhinov. 1988. Nucleotide anddeduced aminoacidsequence ofbacteriophage T4gene12.Nucleic AcidsRes.16:2334. 37.Sengstag, C.,andW.Arber.1983.IS2insertion isamajorcause ofspontaneous mutagenesis ofthebacteriophage P1:non- random distribution oftargetsites.EMBOJ.2:67-71. 38.Sengstag, C.,J.C.W.Shepherd, andW.Arber.1983.The sequence ofthebacteriophage P1genome regionserving ashot targetforIS2insertion. EMBOJ.2:1777-1781. 39.Tominaga, A.,S.Ikemizu, andM.Enomoto. 1991.Site-specific recombinase genesinthreeShigella subgroups andnucleotide sequences ofapinBgeneandaninvertible Bsegment from Shigella boydii.J.Bacteriol. 173:4079-4087. 40.Wood, W.B.1966.Hostspecificity ofDNAproduced by Escherichia coli:bacterial mutations affecting therestriction andmodification ofDNA.J.Mol.Biol.16:118-133. 41.Yarmolinsky, M.B.,andN.Sternberg. 1988.Bacteriophage P1, p.291-438. InR.Calendar (ed.),Thebacteriophages. Plenum Press,NewYork.J.BACTERIOL. Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Proc.Natl.Acad.Sci.USAVol.87,pp.8070-8074, October 1990Biochemistry Cleavage ofthebacteriophage P1packaging site(pac)isregulated byadenine methylation (DNAadeninemethyltransferase/methylation/phage DNApackaging) N.STERNBERG ANDJ.COULBY E.1.duPontdeNemours &Co.Inc.,CentralResearch andDevelopment Department, P.O.Box80328,Wilmington, DE19880-0328 Communicated byJohnRoth,June25,1990(receivedforreviewFebruary 9,1990) ABSTRACT Thepackaging ofbacteriophage P1DNAis initiated whenthephagepackaging site(pac)isrecognized and cleavedandcontinues untilthephageheadisfull.Wehave previously shownthatpacis-a162-base-pair segment ofP1 DNAthatcontains sevenDNAadenine methyltransferase methylation sites(5'-GATC). Weshowherethatcleavage of pacismethylation sensitive. Bothinvivoandinvitroexperi- mentsindicate thatmethylated paciscleavable, whereas un- methylated pacisnot.Moreover, DNAisolatedfromP1phage andcontaining anuncutpacsitewasapoorsubstrate forin vitrocleavage untilitwasmethylated bytheEscherichia coli DNAadeninemethyltransferase. Comparison ofthatuncutpac DNAwithotherviralDNAfragments bydigestion withmeth- ylation-sensitive restriction enzymes indicated thattheuncut pacDNAwaspreferentially undermethylated. Incontrast, virionDNAcontaining acutpacsitewasnotundermethylated. Webelievetheseresultsindicatethatpaccleavage isregulated byadeninemethylation duringthephagelyticcycle. Bacteriophage P1packages itsDNAbyaprocessive headful mechanism thatusesaconcatemeric DNAsubstrate consist- ingofrepeating unitsofviralDNA(1-3).Packaging is initiated whenaspecific162-base-pair (bp)pacsequence on P1DNA(Fig.1)isrecognized andcleaved(1,3,4).TheDNA isthenpackaged unidirectionally fromthecleavedpacend intoanemptyphageprohead untilthatheadisfull(3,5). Whenpackaging hasbeencompleted, theDNAinsidethe headisseparated fromthatoutsideoftheheadbyacutting process thatappearsindependent ofDNAsequence. The DNAendthatremainsoutsideoftheheadafterthecutisthen usedtoinitiatethenextP1sequential packaging event(1,3, 6).Inthiswaypackaging proceeds downtheconcatemer in aprocessive serieswithpacbeingrecognized andcleaved onlyonce,toinitiatetheseries. ForP1,aheadfulcontains'110-115 kilobasepairs(kbp) ofDNA(7,8)or-10-15% moreDNAthanispresentinthe viralgenome. Because theviruspackages thisheadfulfrom aconcatemer, theDNApresentineachvirusparticle con- tainsthesameDNAsequences atbothends:itisterminally redundant (6,9).Thatredundancy iscriticalforthevegeta- tivegrowthofthevirusbecause itpermitstheviralDNAto cyclizeafteritsinjection intoahostcell.Cyclization is mediated bythehomologous recombination systemofthehost(10,11)orinarecombination-deficient hostbythe P1-lox-Cre site-specific recombination system, whenloxP sitesarepresentintheterminally redundant regions(12).For P1togenerate terminally redundant DNAitisimportant to prevent thecleavage ofconcatemeric P1DNAateachpac site.Weshowherethatpaccleavage depends onadenine methylation andsuggestthatthisdependency allowsthecleavage processtoberegulated inthecell,soastopermit theproduction ofterminally redundant viralmolecules.\", 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  MATERIAL ANDMETHODS Bacterial andPhageStrains.Escherichia colistrainN99is supo(13)andstrainNS2626isN99Tn9::dam. StrainNS2626 wasconstructed byP1-mediated transduction oftheTn9 insertion mutation intheE.colidamgenefromstrain GM3808 (14)tostrainN99.StrainNS2342isN99(Aimm434- P1:20b) (3).StrainNS2635 isNS2626(Aimm434-P1:20b). StrainNS2634isNS2626(Aimm434-P1:20a). StrainNS3208 isMC1061 recD-hsdR-mcrA-B- (Plr-m-cm-2 ci.100 aml0.1); thisstrainwasusedtopreparethepaccleavage- proficient extractasdescribed bySteinberg (8).TheP1 wild-type phageusedhereisPlr-m-Cm cl.100,andtheP1 dam-mutantisthatsamephagewithaninsertion ofthe aminoglycoside 3\\'-phosphotransferase (kanr)genefrom Tn903attheAsuIIsiteintheN-terminal portionoftheP1 damgene(unpublished work).Thisinsertion completely inactivates thedamgene.PhagesAimm434-P1:20a and Aimm434-PI:20b havebeendescribed (3).Eachphagecon- tainsthe624-base pair(bp)P1pac-containing EcoRI-20 fragment insertedintheEcoRIsiteatAmapcoordinate 65 (Fig.2A).InAimm434-P1:20a, EcoRI-20 isoriented sothat packaging frompacproceeds towardtheattachment siteof theAvector.InAimm434-P1:20b, EcoRI-20 isoriented inthe opposite direction. General Methods. Restriction enzyme digestions were doneasdescribed bythevendor.Fragments generated by thosedigestswereend-labeled bytreatingthemfirstwithcalf intestinal alkalinephosphatase (NewEngland Nuclear) and thenwith[y-32P]ATP andpolynucleotide kinase(NewEn- glandBiolabs) asdescribed inManiatis etal.(15).Methyl- ationbytheE.coliDNAadeninemethyltransferase (Dam) (NewEngland Biolabs) wasdonein10mMTris[HCI, pH 8.0/10mMEDTA/0.1 mMS-adenosylnmethionine. Prepara- tionofAandP1lysates(3),purification ofphagebybandinginCsCIgradients (16),fractionation ofDNAfragments byagarose andacrylamide gelelectrophoresis, andSouthern hybridization analyses (17)havebeendescribed. Theinten-sityofDNAbandsingelswasdetermined bydensity scanning ofexposed filmswithaHoeferG3000scanning densitometer. InVivopacCleavage Assay.Detailsoftheassayhavebeen described (3).Briefly, cultures ofstrainscontaining Aimm434-P1:20a prophage areinfected witheither Aimm434-P1:20b orwithP1,orwithbothphages,andtotal cellularDNAisisolated fromtheinfected cellsatvarious timesafterinfection. TheDNAsaredigested withBamHI, fractionated byagarosegelelectrophoresis, transferred to nitrocellulose filters,andanalyzed byhybridization tonick- Abbreviations: Dam,DNAadeninemethyltransferase; pacase,pro- teinnecessary forpaccleavage; Akdam-, Agrownindam-host;Akdam\\', Agrownindam\\'host. 8070Thepublication costsofthisarticleweredefrayed inpartbypagecharge payment. Thisarticlemusttherefore beherebymarked\"advertisement\" inaccordance with18U.S.C.§1734solelytoindicatethisfact.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  TheDNAsaredigested withBamHI, fractionated byagarosegelelectrophoresis, transferred to nitrocellulose filters,andanalyzed byhybridization tonick- Abbreviations: Dam,DNAadeninemethyltransferase; pacase,pro- teinnecessary forpaccleavage; Akdam-, Agrownindam-host;Akdam\\', Agrownindam\\'host. 8070Thepublication costsofthisarticleweredefrayed inpartbypagecharge payment. Thisarticlemusttherefore beherebymarked\"advertisement\" inaccordance with18U.S.C.§1734solelytoindicatethisfact. Downloaded from https://www.pnas.org by 157.211.12.63 on January 4, 2024 from IP address 157.211.12.63.', \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Proc.Natl.Acad.Sci.USA87(1990) 8071 5'CATTGATCACTCTAAIi TCAACATGCAGG TGATCA CATTGCGGCTGAAATAGCGG AAAAACAAAGAGTTAATGCCGTTGTCAGTGCCGCAGTCGAGAATGCGAAGCGCCAAAATAAGCAACATCACGG CGCATAAA TGATCG TTCAGAj C1GACG TGATCA CA i.P1:20a i-P120b58 probeA 58_ac65 65pac71 probeB 71 FIG.1.Sequence ofP1pac.Sequence ofthe162-bppacsiteis sufficient topromote paccleavage andP1packaging (4).Boxed sequences, sevenhexanucleotide (TGATCA) elements ofpac;large arrows,positions offrequent paccleavage events;andsmallarrows, positions ofrarepaccleavage events(4). translated probescontaining Asequences flanking thecloned P1pacfragment (seeFig.2). InVitropacCleavage. Thesubstrate fortheassayisP1 EcoRIfragment 20.Itwasprepared bydigesting plasmid pRH92[pBR322 withEcoRIfragment 20clonedintothe vectorEcoRIsite(3)]orP1DNAwithEcoRI.Thefragment waspurified byagarosegelelectrophoresis andend-labeled asdescribed above.Theinvitrocleavage reaction wasdone asdescribed bySternberg (8)withtheuseofanextract prepared bytheinduction ofstrainNS3208. RESULTS Cleavage oftheBacteriophage P1pacSiteIsBlocked ina Bacterial HostthatLacksDamMethylase Activity. TheP1pac siteisa162-bpsegment ofDNA(Fig.1)thatcontains seven hexanucleotide elements (5'-TGATCA) theintegrity ofwhich mustbemaintained forpactobecleaved(4).Because each oftheseelements isalsoaDammethylation site(5'-GATC), itwasofinteresttodetermine whether adeninemethylation affectspaccleavage. Theinitialassayusedeitherwild-type ordammutantstrainscontaining Aprophage withaclonedP1 pacfragment (Aimm434-P1:20b, Fig.2A).Thesecellswere infected withP1,cellularDNAwasisolatedatvarioustimes afterinfection, thatDNAwasdigested atBamHIrestriction sitesinADNAthatflankthechromosomal pacsite,andpac cleavage wasassessed bySouthern hybridization analysis withaprobe(probeB,Fig.2A)fortheflanking Asequences. Ifpacisnotcleaved, asingle7.4-kbpfragment shouldbe detected. Ifitiscleaved, asmaller3.2-kbpfragment should alsobedetected. Comparing resultsinthetwostrainsused (Fig.2C),theonsetofpaccleavage isclearlydelayedby=30 mininthedam-host(lanes7-12)compared withthewild- typehost(lanes1-6),butcleavage eventually reachesthe samelevel(-40-50% ofthesubstrate iscleaved) inthetwo strains.Because P1replication, asmeasured bytheincrease inP1DNAinthegelusedtoprepare thefilterforthe hybridization analysis, isthesameinthetwoinfections (Fig. 2B),thedelayinpaccleavage seeninthedam-hostprobably doesnotreflectageneralphagegrowthdefectinthishost. Asanaddedcontrolintheseexperiments, wedetermined themethylation stateoftheDNAsrecovered fromthe infected cellsbymeasuring theirsensitivity torestriction enzymes theactivities ofwhicharesensitive toadenine methylation (BclI,MboI,andDpnI)(18,19).Tooursurprise wediscovered thathostDNArecovered atleast20minafter P1infection ofthedam-mutantwaslargelymethylated.\", 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  2B),thedelayinpaccleavage seeninthedam-hostprobably doesnotreflectageneralphagegrowthdefectinthishost. Asanaddedcontrolintheseexperiments, wedetermined themethylation stateoftheDNAsrecovered fromthe infected cellsbymeasuring theirsensitivity torestriction enzymes theactivities ofwhicharesensitive toadenine methylation (BclI,MboI,andDpnI)(18,19).Tooursurprise wediscovered thathostDNArecovered atleast20minafter P1infection ofthedam-mutantwaslargelymethylated. This DNAwasresistant toBclIandMboIandsensitive toDpn I(datanotshown). Thesimplestinterpretation ofthisresult isthatP1encodes itsownDammethylase, whichisex- pressedaboutmidwaythroughthevirallifecycle.Activityof thatenzymepresumably accounts forthepaccleavage seen afterphageinfection ofadam-host(Fig.2C).Werecently localized theP1damgeneandhavegenerated aninsertion mutation thatinactivates itsfunction (seeMaterial andB 3Iam7,8-p Bam9,1lo_ c~~~~~~~~~~~~~~~~~~~~~~~~~~~ a *X#@w.\"AC \\'-.P\\' uncut --a\"- ct-_- D ucut-_ FIG.2.paccleavage indam\\'anddam-bacteria. (A)Structure ofA-P1:20a andA-P1:20b phages(3).Thephagescontain P1 EcoRI-20 (thinlines)withitspacsite(i)clonedintoADNA(thick cross-hatched lines)atAmapcoordinate 65.TheP1fragment is clonedinopposite orientations inthetwophages.Digestion ofA DNAwithBamHl(B)cleavestheDNAatAmapcoordinates 58and 71andgenerates a7.4-kbpfragment. Thisfragment isconverted to twosmallerfragments whenpacisalsocleaved. Arrowsunderthe pacsitesindicatedirection ofP1packaging. Thelocation ofDNAs usedasprobestodetecttheBamHI-pac fragments arederivedfrom plasmids pKA6(probeA)andpRAL(probeB)(3).Notethatwhile twofragments aregenerated whenthepacsiteinADNAiscleaved, onlythefragment destined tobepackaged isdetected inourassay; theotherfragment isdestroyed bycellularnucleases (3).(B)DNAs isolated fromP1-infected cellsweredigested withBamH1 and fractionated in1%agarosegelthatwasthenstainedwithethidium bromide. Lanes:1-6,DNAsfromcelllineNS2342(dam\\')infected withP1wildtypethatwereisolated5,10,15,30,45,and60minafter infection, respectively; 7-12,DNAsfromcelllineNS2635(dam-) infected withP1wildtypealsoisolated5,10,15,30,45,and60min afterinfection, respectively. Positions ofP1BamHIfragments 7-10 areshownbyarrows.(C)ThegelinBwastransferred tonitrocel- luloseandhybridized withnick-translated probeBtodetectthe 7.4-kbpuncutand3.2-kbppac-cutfragments. (D)DNAsisolated15, 25,35,45,60,and90minafterinfection ofNS2342dam\\'(lanes1-6) orNS2635dam-(lanes7-12)withP1dam-wereanalyzed bySouthern transferhybridization usingnick-translated probeB.Po- sitionsoftheuncutandpac-cutfragments areshown. Methods). Whenthatmutantinfectsadam-hostcontaining achromosomal pacsite,cleavage ofpacisundetected (Fig.', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Positions ofP1BamHIfragments 7-10 areshownbyarrows.(C)ThegelinBwastransferred tonitrocel- luloseandhybridized withnick-translated probeBtodetectthe 7.4-kbpuncutand3.2-kbppac-cutfragments. (D)DNAsisolated15, 25,35,45,60,and90minafterinfection ofNS2342dam\\'(lanes1-6) orNS2635dam-(lanes7-12)withP1dam-wereanalyzed bySouthern transferhybridization usingnick-translated probeB.Po- sitionsoftheuncutandpac-cutfragments areshown. Methods). Whenthatmutantinfectsadam-hostcontaining achromosomal pacsite,cleavage ofpacisundetected (Fig. 2D,lanes7-12).Controlexperiments indicate thatP1dam- mutantcleavespacnormally inadam\\'host(Fig.2D,lanes 1-6)andthatP1dam-lyticreplication isnormalinadam- host(datanotshown).Theseresultsindicate thatpaccleav- ageabsolutely depends onadenine methylation..417%71,17\" -, 777.7-77Biochemistry: Sternberg andCoulby 123456789I0112 Downloaded from https://www.pnas.org by 157.211.12.63 on January 4, 2024 from IP address 157.211.12.63.', \"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  8072Biochemistry: Sternberg andCoulby pacCleavage DefectinaMethylase-Deficient HostIsfrom aDirectEffectonpac.Twointerpretations followfromthe dataofFig.2:methylase activityisneeded(i)forexpression ofP1-encoded proteins essential forpaccleavage (3)(the expression model)or(ii)torenderpaccleavable (thepac model).Precedent fortheexpression modelcomesfromthe effectofDammethylation onP1cregeneexpression (12)and ontheexpression ofavarietyofbacterial genes(18).To distinguish between thetwomodels,adam-hostcontaining aresidentA-P1pacprophage(Aimm434-P1:20a) wassimul- taneously infected withP1andwithasecondhomoimmune A-P1pacphage(Aimm434-P1:20b). Theinfecting A-P1pac phagewasprepared ineitheradam-oradam'hostand differedfromtheA-P1pacprophage inthatorientation ofthe P1pacfragment intheinfecting phagewasopposite thatin theprophage, allowing ustomeasure paccleavage inboth infecting phageandprophage DNAsinthesamecellbyusing differentAprobes(Fig.2A).Theonlysourceofpaccleavage proteins inthisexperiment isP1becausetheA-P1pacphages containonlyasmallportionofthepaccleavage (pacase) genes(3).Iftheexpression modeliscorrect, whether the infecting A-P1pacphageisprepared inadam-oradam' hostisimmaterial; inbothcasespacshouldbecleavedwith delayedkinetics, asshowninFig.2C.Incontrast, ifthepac modeliscorrect, thenthepacsiteontheinfecting, methyl- atedA-P1pacphagewillbecleaved earlyafterphage infection, whereas thepacsiteontheinfecting, unmethylated A-P1pacphageandontheresidentA-P1pacprophage, will becleavedlateinthephageinfection. Analyses ofthefateofpacDNAinsuchanexperiment indicates thatcleavage ofthepacsiteontheinfecting, methylated A-P1pacphageiscomplete 20minafterinfection (Fig.3A,lanes7-9),whereas cleavage oftheresident chro- mosomal pacsiteinthesamecelldoesnotstartuntil35minafterinfection andisnotcomplete until60minafterinfection (Fig.3B,lanes1-9).Thedelayed kineticsofprophage pac cleavage isconsistent withtheideathatsynthesis oftheP1 methylase occurslateintheinfection andmustmethylate pac beforeitcanbecleaved. Incontrast, ifthepacsiteis introduced intothecellonanunmethylated A-P1pacphage, cleavage ofthatsiteisnotdetected atallduringtheexper- iment(Fig.3A,lanes4-6),despitethefactthattheinitially unmethylated chromosomal pacsiteinthesamecellis cleaved by35minafterinfection. Failuretocleavethe infecting, unmethylated phageisnotfromageneralfailureof thisphagetoinfectcellsbecauseitcaninfectadam'hostand becleaved withanefficiency comparable tothatofan infecting, methylated A-P1pacphage(Fig.3C,lanes1-4). Theseresultsstrongly supportamodelinwhichpacneedsto bemethylated beforeitcanbecleaved. pacCleavage inVitroIsMethylation Dependent.\", 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Failuretocleavethe infecting, unmethylated phageisnotfromageneralfailureof thisphagetoinfectcellsbecauseitcaninfectadam\\'hostand becleaved withanefficiency comparable tothatofan infecting, methylated A-P1pacphage(Fig.3C,lanes1-4). Theseresultsstrongly supportamodelinwhichpacneedsto bemethylated beforeitcanbecleaved. pacCleavage inVitroIsMethylation Dependent. Wehave recently demonstrated thatanextractprepared fromaP1-induced lysogen (\"pacase extract\") cancleavepacinvitro (8).Todetermine whether thatcleavage process alsois affected bythemethylation stateofpacweisolatedfrom plasmid pRH92methylated andunmethylated P1DNAfrag- mentscontaining pac(EcoRI-20) andmeasured cleavage in vitro.Asexpected, themethylated fragment wascleavedinto twosmallerfragments, whereas theunmethylated fragment wasnot(Fig.4,lanes2and4).Iftheunmethylated pac fragment weremethylated invitrowiththeE.coliDam methylase beforebeingexposed totheP1pacaseextract,it couldnowbecleavedwiththesameefficiency asthemeth- ylatedpacfragment (Fig.4A,lane3).Similartreatment in vitroofthemethylated fragment hadnoeffectonthecleavage reaction (Fig.4A,lane5). Uncutpac DNAIsolated fromP1VirmonsIsDefective forpac Cleavage inVitroBecause ItIsPreferentially Undermethy- lated.Wehavepreviously suggested thatpaccleavage ofP1 concatemeric DNAmustbelimitedtoensurethatpackagedA pac--oa ii cutpacnphage4 B c:-pacrCca-phace :ra:gey_aI-c,-..i-ac :-I-ng, pgeI Jit2C_w.w-VW--T,77lrpw ww-Iff~~~~~~. $,*MI- C-i-tCaC -rf-_rhu- e\\'70 aiw FIG.3.Cleavageofmethylated andunmethylated pacDNAin dam-bacteria.(AandB)Gelscontaining DNAsfromdam-lysogen NS2634(A-P1:20a) infectedwithP1wildtype(lanes1-3),P1wild typeandA-P1:20bdam-(Agrowninadam-host;lanes4-6)orP1wildtypeandA-P1:20bdam+(Agrowninadam\\'host;lanes7-9). DNAsinlanes1,4,and7wereisolated20minafterinfection (inf), DNAsinlanes2,5,and8wereisolated35minafterinfection, and DNAsisolatedinlanes3,6,and9wereisolated50minafter infection. DNAsweredigestedwithBamHI,fractionated in1% agarosegels,transferred tonitrocellulose, andhybridized firstwith probeBtodetectthe3.2-kbpcutpacfragmentfromtheinfecting A-P1:20bphage(A)andthenwithprobeAtodetectthe3.8-kbp A-P1:20aprophagecutpacfragment(B).(C)DNAfromdam\\'strain N99wasinfectedeitherwithP1wildtypeandA-P1:20b\\'dam+ (lanes 1and2)orwithP1wildtypeandA-P1:20b\\'dam- (lanes3and4). DNAswereisolated20and35minafterinfectionandtreatedasin AandB;filterswerehybridized withnick-translated probeB. DNAwillbeterminally redundant and,therefore, cancyclize afterinfections. Thus,itmaybeimportant toregulate pac cleavage invivobymethylation soastoensurethatpac cleavage isusedonlytoinitiatethepackaging.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': ['page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  22.Maniatis, T.,E.F.Fritsch, andJ.Sambrook. 1982.Molecular cloning: alaboratory manual. ColdSpringHarborLaboratory, ColdSpringHarbor, N.Y. 23.Meyer,J.,M.Stalhammar-Carlemalm, M.Streiff,S.lida,and W.Arber.1986.Sequence relations amongtheIncYplasmid plSB,P1,andP7prophages. Plasmid 16:81-89. 24.Michel,C.J.,B.Jaqc,D.G.Arques, andT.A.Bickle.1986.A remarkable aminoacidsequence homology between aphageT4 tailfibreprotein andORF314 ofphageXlocated inthetail operon. Gene44:147-150. 25.Montag, D.,andU.Henning. 1987.Anopenreadingframeinthe Escherichia colibacteriophage Xgenome encodes aproteinthat functions inassembly ofthelongtailfibersofbacteriophage T4.J.Bacteriol. 169:5884-5886. 26.Montag, D.,I.Riede,M.-L.Eschbach, M.Degen, andU. Henning. 1987.Receptor-recognizing proteins ofT-even type bacteriophages. Constant andhypervariable regions andan unusual caseofevolution. J.Mol.Biol.196:165-174. 27.Montag, D.,H.Schwarz, andU.Henning. 1989.Acomponent of thesidetailfiberofEscherichia colibacteriophage Xcan functionally replace thereceptor-recognizing partofalongtail fiberprotein oftheunrelated bacteriophage T4.J.Bacteriol. 171:4378-4384. 28.Mosig,G.,andF.Eiserling. 1988.PhageT4structure and metabolism, p.521-606. InR.Calendar (ed.),Thebacterio- phages. Plenum Press,NewYork. 29.Mural,R.J.,R.H.Chesney, D.Vapnek, M.M.Kropf,andJ.R. Scott.1979.Isolation andcharacterization ofclonedfragments ofbacteriophage P1DNA.Virology 93:387-397. 30.Oliver,D.B.,andR.A.Crowther. 1981.DNAsequence ofthe tailfibregenes36and37ofbacteriophage T4.J.Mol.Biol. 153:545-568. 31.Plasterk, R.H.A.,andP.vandePutte.1985.Theinvertible P-DNAsegment inthechromosome ofEscherichia coli.EMBO J.4:237-242. 32.Sandmeier, H.1990.Characterization ofacomplex DNAinver- sionsystem. Ph.D.thesis.BaselUniversity, Basel. 33.Sandmeier, H.,S.lida,P.Hubner, R.Hiestand-Nauer, andW. Arber.1991.Geneorganization inthemultiple DNAinversion regionMinofplasmid plSBofE.coli15T-:assemblage ofa variable gene.Nucleic AcidsRes.19:5831-5838. 34.Sandmeier, H.,S.lida,J.Meyer, R.Hiestand-Nauer, andW. Arber.1990.Site-specific DNArecombination systemMinof plasmid pl5B:aclusterofoverlapping invertible DNAseg- ments.Proc.Natl.Acad.Sci.USA87:1109-1113. 35.Sanger, F.,S.Nicklen, andA.R.Coulson. 1977.DNAsequenc- ingwithchain-terminating inhibitors. Proc.Natl.Acad.Sci. USA74:5463-5467. 36.Selivanov, N.A.,A.G.Prilipov, andV.V.Mesyanzhinov. 1988. Nucleotide anddeduced aminoacidsequence ofbacteriophage T4gene12.Nucleic AcidsRes.16:2334.', 'page_label: 9 file_path: /workspaces/ml-learning/src/phages/data/P1/sandmeier-et-al-1992-dna-inversion-regions-min-of-plasmid-p15b-and-cin-of-bacteriophage-p1-evolution-of-bacteriophage.pdf  34.Sandmeier, H.,S.lida,J.Meyer, R.Hiestand-Nauer, andW. Arber.1990.Site-specific DNArecombination systemMinof plasmid pl5B:aclusterofoverlapping invertible DNAseg- ments.Proc.Natl.Acad.Sci.USA87:1109-1113. 35.Sanger, F.,S.Nicklen, andA.R.Coulson. 1977.DNAsequenc- ingwithchain-terminating inhibitors. Proc.Natl.Acad.Sci. USA74:5463-5467. 36.Selivanov, N.A.,A.G.Prilipov, andV.V.Mesyanzhinov. 1988. Nucleotide anddeduced aminoacidsequence ofbacteriophage T4gene12.Nucleic AcidsRes.16:2334. 37.Sengstag, C.,andW.Arber.1983.IS2insertion isamajorcause ofspontaneous mutagenesis ofthebacteriophage P1:non- random distribution oftargetsites.EMBOJ.2:67-71. 38.Sengstag, C.,J.C.W.Shepherd, andW.Arber.1983.The sequence ofthebacteriophage P1genome regionserving ashot targetforIS2insertion. EMBOJ.2:1777-1781. 39.Tominaga, A.,S.Ikemizu, andM.Enomoto. 1991.Site-specific recombinase genesinthreeShigella subgroups andnucleotide sequences ofapinBgeneandaninvertible Bsegment from Shigella boydii.J.Bacteriol. 173:4079-4087. 40.Wood, W.B.1966.Hostspecificity ofDNAproduced by Escherichia coli:bacterial mutations affecting therestriction andmodification ofDNA.J.Mol.Biol.16:118-133. 41.Yarmolinsky, M.B.,andN.Sternberg. 1988.Bacteriophage P1, p.291-438. InR.Calendar (ed.),Thebacteriophages. Plenum Press,NewYork.J.BACTERIOL. Downloaded from https://journals.asm.org/journal/jb on 05 January 2024 by 157.211.12.63.', \"page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Proc.Natl.Acad.Sci.USAVol.87,pp.8070-8074, October 1990Biochemistry Cleavage ofthebacteriophage P1packaging site(pac)isregulated byadenine methylation (DNAadeninemethyltransferase/methylation/phage DNApackaging) N.STERNBERG ANDJ.COULBY E.1.duPontdeNemours &Co.Inc.,CentralResearch andDevelopment Department, P.O.Box80328,Wilmington, DE19880-0328 Communicated byJohnRoth,June25,1990(receivedforreviewFebruary 9,1990) ABSTRACT Thepackaging ofbacteriophage P1DNAis initiated whenthephagepackaging site(pac)isrecognized and cleavedandcontinues untilthephageheadisfull.Wehave previously shownthatpacis-a162-base-pair segment ofP1 DNAthatcontains sevenDNAadenine methyltransferase methylation sites(5'-GATC). Weshowherethatcleavage of pacismethylation sensitive. Bothinvivoandinvitroexperi- mentsindicate thatmethylated paciscleavable, whereas un- methylated pacisnot.Moreover, DNAisolatedfromP1phage andcontaining anuncutpacsitewasapoorsubstrate forin vitrocleavage untilitwasmethylated bytheEscherichia coli DNAadeninemethyltransferase. Comparison ofthatuncutpac DNAwithotherviralDNAfragments bydigestion withmeth- ylation-sensitive restriction enzymes indicated thattheuncut pacDNAwaspreferentially undermethylated. Incontrast, virionDNAcontaining acutpacsitewasnotundermethylated. Webelievetheseresultsindicatethatpaccleavage isregulated byadeninemethylation duringthephagelyticcycle. Bacteriophage P1packages itsDNAbyaprocessive headful mechanism thatusesaconcatemeric DNAsubstrate consist- ingofrepeating unitsofviralDNA(1-3).Packaging is initiated whenaspecific162-base-pair (bp)pacsequence on P1DNA(Fig.1)isrecognized andcleaved(1,3,4).TheDNA isthenpackaged unidirectionally fromthecleavedpacend intoanemptyphageprohead untilthatheadisfull(3,5). Whenpackaging hasbeencompleted, theDNAinsidethe headisseparated fromthatoutsideoftheheadbyacutting process thatappearsindependent ofDNAsequence. The DNAendthatremainsoutsideoftheheadafterthecutisthen usedtoinitiatethenextP1sequential packaging event(1,3, 6).Inthiswaypackaging proceeds downtheconcatemer in aprocessive serieswithpacbeingrecognized andcleaved onlyonce,toinitiatetheseries. ForP1,aheadfulcontains'110-115 kilobasepairs(kbp) ofDNA(7,8)or-10-15% moreDNAthanispresentinthe viralgenome. Because theviruspackages thisheadfulfrom aconcatemer, theDNApresentineachvirusparticle con- tainsthesameDNAsequences atbothends:itisterminally redundant (6,9).Thatredundancy iscriticalforthevegeta- tivegrowthofthevirusbecause itpermitstheviralDNAto cyclizeafteritsinjection intoahostcell.Cyclization is mediated bythehomologous recombination systemofthehost(10,11)orinarecombination-deficient hostbythe P1-lox-Cre site-specific recombination system, whenloxP sitesarepresentintheterminally redundant regions(12).For P1togenerate terminally redundant DNAitisimportant to prevent thecleavage ofconcatemeric P1DNAateachpac site.Weshowherethatpaccleavage depends onadenine methylation andsuggestthatthisdependency allowsthecleavage processtoberegulated inthecell,soastopermit theproduction ofterminally redundant viralmolecules.\", 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  MATERIAL ANDMETHODS Bacterial andPhageStrains.Escherichia colistrainN99is supo(13)andstrainNS2626isN99Tn9::dam. StrainNS2626 wasconstructed byP1-mediated transduction oftheTn9 insertion mutation intheE.colidamgenefromstrain GM3808 (14)tostrainN99.StrainNS2342isN99(Aimm434- P1:20b) (3).StrainNS2635 isNS2626(Aimm434-P1:20b). StrainNS2634isNS2626(Aimm434-P1:20a). StrainNS3208 isMC1061 recD-hsdR-mcrA-B- (Plr-m-cm-2 ci.100 aml0.1); thisstrainwasusedtopreparethepaccleavage- proficient extractasdescribed bySteinberg (8).TheP1 wild-type phageusedhereisPlr-m-Cm cl.100,andtheP1 dam-mutantisthatsamephagewithaninsertion ofthe aminoglycoside 3\\'-phosphotransferase (kanr)genefrom Tn903attheAsuIIsiteintheN-terminal portionoftheP1 damgene(unpublished work).Thisinsertion completely inactivates thedamgene.PhagesAimm434-P1:20a and Aimm434-PI:20b havebeendescribed (3).Eachphagecon- tainsthe624-base pair(bp)P1pac-containing EcoRI-20 fragment insertedintheEcoRIsiteatAmapcoordinate 65 (Fig.2A).InAimm434-P1:20a, EcoRI-20 isoriented sothat packaging frompacproceeds towardtheattachment siteof theAvector.InAimm434-P1:20b, EcoRI-20 isoriented inthe opposite direction. General Methods. Restriction enzyme digestions were doneasdescribed bythevendor.Fragments generated by thosedigestswereend-labeled bytreatingthemfirstwithcalf intestinal alkalinephosphatase (NewEngland Nuclear) and thenwith[y-32P]ATP andpolynucleotide kinase(NewEn- glandBiolabs) asdescribed inManiatis etal.(15).Methyl- ationbytheE.coliDNAadeninemethyltransferase (Dam) (NewEngland Biolabs) wasdonein10mMTris[HCI, pH 8.0/10mMEDTA/0.1 mMS-adenosylnmethionine. Prepara- tionofAandP1lysates(3),purification ofphagebybandinginCsCIgradients (16),fractionation ofDNAfragments byagarose andacrylamide gelelectrophoresis, andSouthern hybridization analyses (17)havebeendescribed. Theinten-sityofDNAbandsingelswasdetermined bydensity scanning ofexposed filmswithaHoeferG3000scanning densitometer. InVivopacCleavage Assay.Detailsoftheassayhavebeen described (3).Briefly, cultures ofstrainscontaining Aimm434-P1:20a prophage areinfected witheither Aimm434-P1:20b orwithP1,orwithbothphages,andtotal cellularDNAisisolated fromtheinfected cellsatvarious timesafterinfection. TheDNAsaredigested withBamHI, fractionated byagarosegelelectrophoresis, transferred to nitrocellulose filters,andanalyzed byhybridization tonick- Abbreviations: Dam,DNAadeninemethyltransferase; pacase,pro- teinnecessary forpaccleavage; Akdam-, Agrownindam-host;Akdam\\', Agrownindam\\'host. 8070Thepublication costsofthisarticleweredefrayed inpartbypagecharge payment. Thisarticlemusttherefore beherebymarked\"advertisement\" inaccordance with18U.S.C.§1734solelytoindicatethisfact.', 'page_label: 1 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  TheDNAsaredigested withBamHI, fractionated byagarosegelelectrophoresis, transferred to nitrocellulose filters,andanalyzed byhybridization tonick- Abbreviations: Dam,DNAadeninemethyltransferase; pacase,pro- teinnecessary forpaccleavage; Akdam-, Agrownindam-host;Akdam\\', Agrownindam\\'host. 8070Thepublication costsofthisarticleweredefrayed inpartbypagecharge payment. Thisarticlemusttherefore beherebymarked\"advertisement\" inaccordance with18U.S.C.§1734solelytoindicatethisfact. Downloaded from https://www.pnas.org by 157.211.12.63 on January 4, 2024 from IP address 157.211.12.63.', \"page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Proc.Natl.Acad.Sci.USA87(1990) 8071 5'CATTGATCACTCTAAIi TCAACATGCAGG TGATCA CATTGCGGCTGAAATAGCGG AAAAACAAAGAGTTAATGCCGTTGTCAGTGCCGCAGTCGAGAATGCGAAGCGCCAAAATAAGCAACATCACGG CGCATAAA TGATCG TTCAGAj C1GACG TGATCA CA i.P1:20a i-P120b58 probeA 58_ac65 65pac71 probeB 71 FIG.1.Sequence ofP1pac.Sequence ofthe162-bppacsiteis sufficient topromote paccleavage andP1packaging (4).Boxed sequences, sevenhexanucleotide (TGATCA) elements ofpac;large arrows,positions offrequent paccleavage events;andsmallarrows, positions ofrarepaccleavage events(4). translated probescontaining Asequences flanking thecloned P1pacfragment (seeFig.2). InVitropacCleavage. Thesubstrate fortheassayisP1 EcoRIfragment 20.Itwasprepared bydigesting plasmid pRH92[pBR322 withEcoRIfragment 20clonedintothe vectorEcoRIsite(3)]orP1DNAwithEcoRI.Thefragment waspurified byagarosegelelectrophoresis andend-labeled asdescribed above.Theinvitrocleavage reaction wasdone asdescribed bySternberg (8)withtheuseofanextract prepared bytheinduction ofstrainNS3208. RESULTS Cleavage oftheBacteriophage P1pacSiteIsBlocked ina Bacterial HostthatLacksDamMethylase Activity. TheP1pac siteisa162-bpsegment ofDNA(Fig.1)thatcontains seven hexanucleotide elements (5'-TGATCA) theintegrity ofwhich mustbemaintained forpactobecleaved(4).Because each oftheseelements isalsoaDammethylation site(5'-GATC), itwasofinteresttodetermine whether adeninemethylation affectspaccleavage. Theinitialassayusedeitherwild-type ordammutantstrainscontaining Aprophage withaclonedP1 pacfragment (Aimm434-P1:20b, Fig.2A).Thesecellswere infected withP1,cellularDNAwasisolatedatvarioustimes afterinfection, thatDNAwasdigested atBamHIrestriction sitesinADNAthatflankthechromosomal pacsite,andpac cleavage wasassessed bySouthern hybridization analysis withaprobe(probeB,Fig.2A)fortheflanking Asequences. Ifpacisnotcleaved, asingle7.4-kbpfragment shouldbe detected. Ifitiscleaved, asmaller3.2-kbpfragment should alsobedetected. Comparing resultsinthetwostrainsused (Fig.2C),theonsetofpaccleavage isclearlydelayedby=30 mininthedam-host(lanes7-12)compared withthewild- typehost(lanes1-6),butcleavage eventually reachesthe samelevel(-40-50% ofthesubstrate iscleaved) inthetwo strains.Because P1replication, asmeasured bytheincrease inP1DNAinthegelusedtoprepare thefilterforthe hybridization analysis, isthesameinthetwoinfections (Fig. 2B),thedelayinpaccleavage seeninthedam-hostprobably doesnotreflectageneralphagegrowthdefectinthishost. Asanaddedcontrolintheseexperiments, wedetermined themethylation stateoftheDNAsrecovered fromthe infected cellsbymeasuring theirsensitivity torestriction enzymes theactivities ofwhicharesensitive toadenine methylation (BclI,MboI,andDpnI)(18,19).Tooursurprise wediscovered thathostDNArecovered atleast20minafter P1infection ofthedam-mutantwaslargelymethylated.\", 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  2B),thedelayinpaccleavage seeninthedam-hostprobably doesnotreflectageneralphagegrowthdefectinthishost. Asanaddedcontrolintheseexperiments, wedetermined themethylation stateoftheDNAsrecovered fromthe infected cellsbymeasuring theirsensitivity torestriction enzymes theactivities ofwhicharesensitive toadenine methylation (BclI,MboI,andDpnI)(18,19).Tooursurprise wediscovered thathostDNArecovered atleast20minafter P1infection ofthedam-mutantwaslargelymethylated. This DNAwasresistant toBclIandMboIandsensitive toDpn I(datanotshown). Thesimplestinterpretation ofthisresult isthatP1encodes itsownDammethylase, whichisex- pressedaboutmidwaythroughthevirallifecycle.Activityof thatenzymepresumably accounts forthepaccleavage seen afterphageinfection ofadam-host(Fig.2C).Werecently localized theP1damgeneandhavegenerated aninsertion mutation thatinactivates itsfunction (seeMaterial andB 3Iam7,8-p Bam9,1lo_ c~~~~~~~~~~~~~~~~~~~~~~~~~~~ a *X#@w.\"AC \\'-.P\\' uncut --a\"- ct-_- D ucut-_ FIG.2.paccleavage indam\\'anddam-bacteria. (A)Structure ofA-P1:20a andA-P1:20b phages(3).Thephagescontain P1 EcoRI-20 (thinlines)withitspacsite(i)clonedintoADNA(thick cross-hatched lines)atAmapcoordinate 65.TheP1fragment is clonedinopposite orientations inthetwophages.Digestion ofA DNAwithBamHl(B)cleavestheDNAatAmapcoordinates 58and 71andgenerates a7.4-kbpfragment. Thisfragment isconverted to twosmallerfragments whenpacisalsocleaved. Arrowsunderthe pacsitesindicatedirection ofP1packaging. Thelocation ofDNAs usedasprobestodetecttheBamHI-pac fragments arederivedfrom plasmids pKA6(probeA)andpRAL(probeB)(3).Notethatwhile twofragments aregenerated whenthepacsiteinADNAiscleaved, onlythefragment destined tobepackaged isdetected inourassay; theotherfragment isdestroyed bycellularnucleases (3).(B)DNAs isolated fromP1-infected cellsweredigested withBamH1 and fractionated in1%agarosegelthatwasthenstainedwithethidium bromide. Lanes:1-6,DNAsfromcelllineNS2342(dam\\')infected withP1wildtypethatwereisolated5,10,15,30,45,and60minafter infection, respectively; 7-12,DNAsfromcelllineNS2635(dam-) infected withP1wildtypealsoisolated5,10,15,30,45,and60min afterinfection, respectively. Positions ofP1BamHIfragments 7-10 areshownbyarrows.(C)ThegelinBwastransferred tonitrocel- luloseandhybridized withnick-translated probeBtodetectthe 7.4-kbpuncutand3.2-kbppac-cutfragments. (D)DNAsisolated15, 25,35,45,60,and90minafterinfection ofNS2342dam\\'(lanes1-6) orNS2635dam-(lanes7-12)withP1dam-wereanalyzed bySouthern transferhybridization usingnick-translated probeB.Po- sitionsoftheuncutandpac-cutfragments areshown. Methods). Whenthatmutantinfectsadam-hostcontaining achromosomal pacsite,cleavage ofpacisundetected (Fig.', 'page_label: 2 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Positions ofP1BamHIfragments 7-10 areshownbyarrows.(C)ThegelinBwastransferred tonitrocel- luloseandhybridized withnick-translated probeBtodetectthe 7.4-kbpuncutand3.2-kbppac-cutfragments. (D)DNAsisolated15, 25,35,45,60,and90minafterinfection ofNS2342dam\\'(lanes1-6) orNS2635dam-(lanes7-12)withP1dam-wereanalyzed bySouthern transferhybridization usingnick-translated probeB.Po- sitionsoftheuncutandpac-cutfragments areshown. Methods). Whenthatmutantinfectsadam-hostcontaining achromosomal pacsite,cleavage ofpacisundetected (Fig. 2D,lanes7-12).Controlexperiments indicate thatP1dam- mutantcleavespacnormally inadam\\'host(Fig.2D,lanes 1-6)andthatP1dam-lyticreplication isnormalinadam- host(datanotshown).Theseresultsindicate thatpaccleav- ageabsolutely depends onadenine methylation..417%71,17\" -, 777.7-77Biochemistry: Sternberg andCoulby 123456789I0112 Downloaded from https://www.pnas.org by 157.211.12.63 on January 4, 2024 from IP address 157.211.12.63.', \"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  8072Biochemistry: Sternberg andCoulby pacCleavage DefectinaMethylase-Deficient HostIsfrom aDirectEffectonpac.Twointerpretations followfromthe dataofFig.2:methylase activityisneeded(i)forexpression ofP1-encoded proteins essential forpaccleavage (3)(the expression model)or(ii)torenderpaccleavable (thepac model).Precedent fortheexpression modelcomesfromthe effectofDammethylation onP1cregeneexpression (12)and ontheexpression ofavarietyofbacterial genes(18).To distinguish between thetwomodels,adam-hostcontaining aresidentA-P1pacprophage(Aimm434-P1:20a) wassimul- taneously infected withP1andwithasecondhomoimmune A-P1pacphage(Aimm434-P1:20b). Theinfecting A-P1pac phagewasprepared ineitheradam-oradam'hostand differedfromtheA-P1pacprophage inthatorientation ofthe P1pacfragment intheinfecting phagewasopposite thatin theprophage, allowing ustomeasure paccleavage inboth infecting phageandprophage DNAsinthesamecellbyusing differentAprobes(Fig.2A).Theonlysourceofpaccleavage proteins inthisexperiment isP1becausetheA-P1pacphages containonlyasmallportionofthepaccleavage (pacase) genes(3).Iftheexpression modeliscorrect, whether the infecting A-P1pacphageisprepared inadam-oradam' hostisimmaterial; inbothcasespacshouldbecleavedwith delayedkinetics, asshowninFig.2C.Incontrast, ifthepac modeliscorrect, thenthepacsiteontheinfecting, methyl- atedA-P1pacphagewillbecleaved earlyafterphage infection, whereas thepacsiteontheinfecting, unmethylated A-P1pacphageandontheresidentA-P1pacprophage, will becleavedlateinthephageinfection. Analyses ofthefateofpacDNAinsuchanexperiment indicates thatcleavage ofthepacsiteontheinfecting, methylated A-P1pacphageiscomplete 20minafterinfection (Fig.3A,lanes7-9),whereas cleavage oftheresident chro- mosomal pacsiteinthesamecelldoesnotstartuntil35minafterinfection andisnotcomplete until60minafterinfection (Fig.3B,lanes1-9).Thedelayed kineticsofprophage pac cleavage isconsistent withtheideathatsynthesis oftheP1 methylase occurslateintheinfection andmustmethylate pac beforeitcanbecleaved. Incontrast, ifthepacsiteis introduced intothecellonanunmethylated A-P1pacphage, cleavage ofthatsiteisnotdetected atallduringtheexper- iment(Fig.3A,lanes4-6),despitethefactthattheinitially unmethylated chromosomal pacsiteinthesamecellis cleaved by35minafterinfection. Failuretocleavethe infecting, unmethylated phageisnotfromageneralfailureof thisphagetoinfectcellsbecauseitcaninfectadam'hostand becleaved withanefficiency comparable tothatofan infecting, methylated A-P1pacphage(Fig.3C,lanes1-4). Theseresultsstrongly supportamodelinwhichpacneedsto bemethylated beforeitcanbecleaved. pacCleavage inVitroIsMethylation Dependent.\", 'page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Failuretocleavethe infecting, unmethylated phageisnotfromageneralfailureof thisphagetoinfectcellsbecauseitcaninfectadam\\'hostand becleaved withanefficiency comparable tothatofan infecting, methylated A-P1pacphage(Fig.3C,lanes1-4). Theseresultsstrongly supportamodelinwhichpacneedsto bemethylated beforeitcanbecleaved. pacCleavage inVitroIsMethylation Dependent. Wehave recently demonstrated thatanextractprepared fromaP1-induced lysogen (\"pacase extract\") cancleavepacinvitro (8).Todetermine whether thatcleavage process alsois affected bythemethylation stateofpacweisolatedfrom plasmid pRH92methylated andunmethylated P1DNAfrag- mentscontaining pac(EcoRI-20) andmeasured cleavage in vitro.Asexpected, themethylated fragment wascleavedinto twosmallerfragments, whereas theunmethylated fragment wasnot(Fig.4,lanes2and4).Iftheunmethylated pac fragment weremethylated invitrowiththeE.coliDam methylase beforebeingexposed totheP1pacaseextract,it couldnowbecleavedwiththesameefficiency asthemeth- ylatedpacfragment (Fig.4A,lane3).Similartreatment in vitroofthemethylated fragment hadnoeffectonthecleavage reaction (Fig.4A,lane5). Uncutpac DNAIsolated fromP1VirmonsIsDefective forpac Cleavage inVitroBecause ItIsPreferentially Undermethy- lated.Wehavepreviously suggested thatpaccleavage ofP1 concatemeric DNAmustbelimitedtoensurethatpackagedA pac--oa ii cutpacnphage4 B c:-pacrCca-phace :ra:gey_aI-c,-..i-ac :-I-ng, pgeI Jit2C_w.w-VW--T,77lrpw ww-Iff~~~~~~. $,*MI- C-i-tCaC -rf-_rhu- e\\'70 aiw FIG.3.Cleavageofmethylated andunmethylated pacDNAin dam-bacteria.(AandB)Gelscontaining DNAsfromdam-lysogen NS2634(A-P1:20a) infectedwithP1wildtype(lanes1-3),P1wild typeandA-P1:20bdam-(Agrowninadam-host;lanes4-6)orP1wildtypeandA-P1:20bdam+(Agrowninadam\\'host;lanes7-9). DNAsinlanes1,4,and7wereisolated20minafterinfection (inf), DNAsinlanes2,5,and8wereisolated35minafterinfection, and DNAsisolatedinlanes3,6,and9wereisolated50minafter infection. DNAsweredigestedwithBamHI,fractionated in1% agarosegels,transferred tonitrocellulose, andhybridized firstwith probeBtodetectthe3.2-kbpcutpacfragmentfromtheinfecting A-P1:20bphage(A)andthenwithprobeAtodetectthe3.8-kbp A-P1:20aprophagecutpacfragment(B).(C)DNAfromdam\\'strain N99wasinfectedeitherwithP1wildtypeandA-P1:20b\\'dam+ (lanes 1and2)orwithP1wildtypeandA-P1:20b\\'dam- (lanes3and4). DNAswereisolated20and35minafterinfectionandtreatedasin AandB;filterswerehybridized withnick-translated probeB. DNAwillbeterminally redundant and,therefore, cancyclize afterinfections. Thus,itmaybeimportant toregulate pac cleavage invivobymethylation soastoensurethatpac cleavage isusedonlytoinitiatethepackaging.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:12 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'104'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994047'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'357ms'), (b'x-request-id', b'e44faa27a3397bf8dbce63a7781ab89e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249e3bea95d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:12 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'104'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'994047'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'357ms'), (b'x-request-id', b'e44faa27a3397bf8dbce63a7781ab89e'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249e3bea95d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': [\"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  (C)DNAfromdam'strain N99wasinfectedeitherwithP1wildtypeandA-P1:20b'dam+ (lanes 1and2)orwithP1wildtypeandA-P1:20b'dam- (lanes3and4). DNAswereisolated20and35minafterinfectionandtreatedasin AandB;filterswerehybridized withnick-translated probeB. DNAwillbeterminally redundant and,therefore, cancyclize afterinfections. Thus,itmaybeimportant toregulate pac cleavage invivobymethylation soastoensurethatpac cleavage isusedonlytoinitiatethepackaging. Ifthisis correct,uncleavedpacDNAisolatedfromvirionswouldbe predicted tobeapoorpacsubstrate invitrobecause itis preferentially undermethylated; thisprediction issupported bythefollowing results. First,weisolatedP1pacfragment EcoRI-20 fromviral DNAandassessed itsabilitytobecleavedinvitro.Although somecleavageofthenativefragment wasdetectable, cleav- agewasstimulated -3-foldbypremethylating thefragment withE.coliDammethylase beforeintroducing itintothe pacasereaction(Fig.4,lanes9-13).Thus,-75%ofthe potentially cleavable viralpacDNAcouldnotbecleaved untilitwasmethylated invitro.Thesameresultwasobtained withapac-containing fragmentisolatedfromasecondprep- arationofP1virions(Fig.4A,lanes7and8).Tomeasure directlythemethylation stateofpacDNAweisolatedthree classesofDNAfragments fromP1DNA:the624-bppac- containing EcoRI-20fragment,EcoRIfragments 21,22,23, and24(thesizesofwhichvaryfrom180-450bp),andthe145- to150-bppac-EcoRI fragment, whichcontains thecutviral pacend.TheseDNAswereend-labeled, digested withBclI (whichonlycutsfullyunmethylated 5'-TGATCA sites),and theproducts wereanalyzed ondenaturing sequencing gelsProc.Natl.Acad.Sci.USA87(1990) Ow'.'i a*ioam00 Downloaded from https://www.pnas.org by 157.211.12.63 on January 4, 2024 from IP address 157.211.12.63.\", \"page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Proc.Natl.Acad.Sci.USA87(1990) 8073 A - EcoRI-20 I Nopac--, 910 2'3 ) 5,If ~~~~~~~5'123456711131589101214 -.o-pac-2 FIG.4.Invitropaccleavage ismethylation dependent. All fragments usedherewerelabeled atbothendswith[y-32P]ATP and polynucleotide kinase.Thefragments weretheneithermethylated withtheE.coliDammethylase ornot,incubated withthepacase extract,andthenfractionated in5%acrylamide gel.Lane1contains EcoRI-20 isolated frompRH92-dam-. ThatsameDNA wasincu- batedinthepacasereaction unmethylated (lane2)orafterinvitro methylation (lane3).Positions ofthetwopaccleavage fragments (the 475-bppac-1fragment andthe150-bppac-2fragment) areshownin lane3.Lanes4and5containEcoRl-20 isolatedfrompRH92dam+ incubated inthecomplete pacasereaction without invitromethyl- ation(lane4)orafterinvitromethylation (lane5).Lanes7and8 containEcoRI-20 isolatedfromP1phageDNAandeitheruntreated (lane7)orpremethylated withDam(lane8)beforebeingincubated withpacaseextract.Thisfragment ispartially contaminated withthe smallerP1EcoRIfragments 21-23;theirpositions inthegelare indicated bylinestorightoflane8.Lane9contains EcoRI-20 isolated fromasecondP1phagepreparation; itwasincubated inthepac cleavage reaction withoutpriorinvitroDammethylation (lanes12 and13,respectively) oraftermethylation (lanes10and11,respec- tively).Lanes10and12contain3timesasmuchDNAaslanes11and 13.Lane6contains aHpa IIdigestofpBR322 DNA.Thelabeled fragments weredetected aftergelwasdriedbyexposure toKodak XARfilm. (Fig.5).Fourclassesofsingle-stranded DNAcleavage products shouldbegenerated whenthesixBclIsitesinpac aredigested (Fig.SA).Despite thefactthatEcoRI-20 is contaminated withasmallamountofEcoRI-21, -22,and-23, Bc!Ifragment classes1,2,and3werereadilydetected inthe gels(lanes 5,6,8,and9).Class4fragments weretoolargeto bedistinguished fromuncutEcoRI-20. Theresultsclearly indicate thatBclIcandigest asignificant fraction ofthepac siteinEcoRI-20. Based onyieldofclass1andclass3 cleavage products (Fig.SB,lanes6and9)-30%oftheBclI sitesinpaccouldbecleaved(seeFig. 5legend). Incontrast, yieldofclass2cleavage products (Fig.5B,lane6)wasmore variable-cleavage rangedfrom6to30odepending onsite. Atfirstapproximation thesetworesults appearcontradictory because class2and3cleavage products aregenerated by cuttingthesameBclIrestriction sites.However, thetwo products differinthatproduction ofclass2fragments de- pends onfailure tocleaveclass1sites.Allresults are self-consistent ifDNAmolecules containing unmethylated class2BclIsitesfrequently containunmethylated class1 sites.Whythisshouldbeso,whenonly30%ooftheDNA contains unmethylated BclIsites,isnotclear,butthehigh degreeofundermethylation ofuncut pacisundeniable.\", \"page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  5legend). Incontrast, yieldofclass2cleavage products (Fig.5B,lane6)wasmore variable-cleavage rangedfrom6to30odepending onsite. Atfirstapproximation thesetworesults appearcontradictory because class2and3cleavage products aregenerated by cuttingthesameBclIrestriction sites.However, thetwo products differinthatproduction ofclass2fragments de- pends onfailure tocleaveclass1sites.Allresults are self-consistent ifDNAmolecules containing unmethylated class2BclIsitesfrequently containunmethylated class1 sites.Whythisshouldbeso,whenonly30%ooftheDNA contains unmethylated BclIsites,isnotclear,butthehigh degreeofundermethylation ofuncut pacisundeniable. In contrast, whenthecutpacfragment wasincubated withBcl I,<2%oftheDNAwasdigested (Fig.5B,lanes10-15).Note thatthisfragment canonlygenerate class1BclIproducts because itcontains onlytherighthalfofpac.Whenthe clusterofEcoRI-21-24 fragments isdigested withBclI,the singleBclIsiteinEcoRI-22 ispoorlydigested. Fromthe intensity oftheBc-22fragments (Fig.SB,lane3),weestimateR1-20 _ RI-21-_-}II am ~ ~ )Q I 0 is *)L'R;-24_ *mm _a o-solo t 3~c22-4- FIG.5.BclIdigestion ofpacfragments isolated fromP1DNA. EcoRIfragments wereisolated fromP1virusDNA,labeledwith [y-32P]ATP andpolynucleotide kinase,digested withrestriction enzymeBcl1,heat-denatured, andthenfractionated in8Murea/6% acrylamide gels.(A)Fourclassesofrestriction fragment thatshould begenerated whenunmethylated, end-labeled EcoRI-20 DNAiscut withBc!1.-,BclIrestriction sites;*,DNAendslabeledwith32p. (B)Lanes:1,Unmethylated P1EcoRI-22 DNA;2,unmethylated P1 EcoRI-22 DNAdigested withBc!I.Twosmallerfragments (Bc-22) aregenerated; 4,EcoRI-22-25 fragments isolated fromP1virion DNA;3,sameDNA asinlane4afterBclIdigestion; 5and8, EcoRI-20 isolated fromP1virionDNAcontaminated withsmall amounts ofEcoRl21-23;6and9,thatsamefragment afterBc!I digestion. DNAsinlanes8and9werefractionated bygelelectro- phoresis forashortertimethanthoseinlanes5and6.Positions of class1-4BcIIrestriction fragments areindicated bythebrackets between lanes7and8.Theclass4restriction fragments aredifficult todistinguish inthesegelsfromEcoRI-20. Lanes:10and11,similar tolanes8and9,exceptthattheEcoRI-20 fragment wasfrom a different P1virionpreparation; 12and14,145-to150-bppac-EcoRI fragments isolatedfromtwodifferent P1virionpreparations; 13and 15,BclIdigestsoftheselatterDNAs. that<4%ofthissiteiscut.Thesameresult wasobtained with virionBamHIfragment 9,whichcontains threeclustered Bc! Irestriction sites(18;datanotshown). Weconclude that uncleaved pacDNAinvirusparticles isapoorsubstrate for paccleavage because itispreferentially undermethylated.\", \"page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Lanes:10and11,similar tolanes8and9,exceptthattheEcoRI-20 fragment wasfrom a different P1virionpreparation; 12and14,145-to150-bppac-EcoRI fragments isolatedfromtwodifferent P1virionpreparations; 13and 15,BclIdigestsoftheselatterDNAs. that<4%ofthissiteiscut.Thesameresult wasobtained with virionBamHIfragment 9,whichcontains threeclustered Bc! Irestriction sites(18;datanotshown). Weconclude that uncleaved pacDNAinvirusparticles isapoorsubstrate for paccleavage because itispreferentially undermethylated. DISCUSSION ThephageP1packaging site(pac)isa162-bpsegment ofP1 DNAthatcontains fourhexanucleotide elements (5'- TGATCR) atoneend,threehexanucleotide elements atthe otherend,andasegment of90bpseparating theseelements (Fig.1).Cleavage ofthisDNAwithinthe90-bpspacerregionBBiochemistry: Sternberg andCoulby 2345678 I....iss,..:s:,.Ifs Downloaded from https://www.pnas.org by 157.211.12.63 on January 4, 2024 from IP address 157.211.12.63.\", \"page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  8074 Biochemistry: Sternberg andCoulby isaninitialstepintheP1headfulpackaging process(4).Each hexanucleotide element ofpaccontains aDammethylation site,andweshowherethatmethylation ofatleastsomeof thoseelements isnecessary forpaccleavage. Inaddition, we showthatmethylation probably playsaroleinregulating pac cleavage because uncleaved pacDNAisolatedfromvirions isundermethylated andisapoorsubstrate forpaccleavage invitro,whereas cleaved pacDNAfromvirionsismore extensively methylated. Questions stillremaining arewhy doespaccleavage havetoberegulated, whyisn'titregulated bysimplycontrolling pacasesynthesis, andhowdoesmeth- ylationmediateregulation ofpaccleavage? Regarding thefirstquestion, wesuggestthatthereisneed forregulation. First,ifeachpacsiteonaconcatemer ofP1 DNAwerecleaved atthesame,ornearlythesame,timein thevirallifecycleandtheresulting DNAwerepackaged into phageheads,thevirusparticles produced wouldcontain DNAthatisnotterminally redundant and,therefore, could notbecyclized afterphageinfection. Thus,itisdesirable to cleavepacefficiently onceperconcatemer toinitiatethe processive packaging seriesbutnotmorethanonce.Itcould bearguedthatthisresultmightbeachieved withoutdirectly regulating paccleavage ifconcatemeric DNAwererapidly packaged afterthefirstpacsiteontheconcatemer werecut. Thisisanunlikelyscenario becauseP1proteins necessary for paccleavage aremadeearlyduringphageinfection (Fig.3 showscleavage canbecomplete by20minafterinfection withmethylated pacDNA),wellbeforephageheadsare available toencapsidate DNA.Moreover, ourmeasurements ofcutpacDNAinvivoindicate noincrease inthatDNAin cellsinfectedwithhead-defective mutants(datanotshown). Wemighthaveexpected suchanincrease ifheadsblockpac cleavage byencapsidating DNA. Whyarepacaseproteinsmadeearlyinthevirallifecycle beforebeingneededforDNApackaging? Probably theyare necessary forafunction otherthanpackaging. Perhapsthey playaroleininitiating rollingcirclereplication bynicking circularDNAatpac. Howispacregulated bymethylation? Webelieveakey pointintheregulation processisthereplication ofpacDNA. Thatreplication temporarily generates hemimethylated or evenunmethylated pacDNA(18,19).Normally thehost and/ortheP1methylase wouldrapidlymethylate thenewly replicated pacDNA,rendering itcleavable. Regulation might beachieved byacompeting reaction inwhicheitheraP1or ahost-encoded proteinbindstotheundermethylated pacsite andprevents furthermethylation andsubsequent cleavage. Ourrecentdemonstration thatpacasecanbindto,butcannot cleave,hemimethylated pacDNA(theprimaryproductof pacreplication) suggests thatpacaseitselfmayberesponsi- bleforinhibiting pacmethylation (N.S.,unpublished work). Toaccountforthepaccleavage thatmustoccurtoinitiateP1 packaging oneneedonlyarguethatthecompetition between methylation andprotection isdesigned toproduce onefunc- tionalpacsiteforeverythreetofivenonfunctional sites. Alternatively, themethylation function isenhanced ortheprotection function isdiminished atsomepointinthevirallife cycle,increasing theprobability thatanewlyreplicated pac sitewillbeactivated. TheresultsofFig.3alsosuggestthatreplication ofpac DNAisimportant inregulating pacavailability. Thus,when adam-hostisinfected withbothP1andanunmethylated A-P1pacphage,thereplication ofwhichisblocked bya resident Aprophage, cleavage ofthepacsiteontheinfection A-P1pacphageisnotseenthroughout theexperiment.\", \"page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Alternatively, themethylation function isenhanced ortheprotection function isdiminished atsomepointinthevirallife cycle,increasing theprobability thatanewlyreplicated pac sitewillbeactivated. TheresultsofFig.3alsosuggestthatreplication ofpac DNAisimportant inregulating pacavailability. Thus,when adam-hostisinfected withbothP1andanunmethylated A-P1pacphage,thereplication ofwhichisblocked bya resident Aprophage, cleavage ofthepacsiteontheinfection A-P1pacphageisnotseenthroughout theexperiment. In contrast, apacsiteinaresident prophage inthesamecellis cleavedwithkineticsthatpresumably reflectsynthesis ofthe P1Dammethylase midway through thevirallifecycle.We proposethatinbothcasesmethylation andactivation ofpac isblockedbyprotein(s) boundtothatsite.Forthechromo- somalsitethehost'sreplication machinery occasionally movesthrough thesite,displacing theboundproteinand permitting methylation andsubsequent cleavage ofthesite. Theextrachromosomal Apacsubstrate isneverreplicated duringtheexperiment andisthusnevercleaved. GrowthofP1dam-phageisnormalinadam'hostbut severely restricted inadam-host.UnderfullyDam-con- ditions,P1produces only-5%asmuchphageaswheneither phageorhostmethylase isactive(unpublished work).More- over,thephageproduced underDam-conditions donothave pacends(G.LuceyandN.S.,unpublished work).These resultssupportourconclusions andsuggestthatpackaging of P1DNAunderDam-conditions, occursbyamechanism that doesnotrecognize andcleavepac. WethankDrs.R.MenzelandG.Cohenfortheircriticalreading ofthismanuscript. 1.Bachi,B.&Arber,W.(1977)Mol.Gen.Genet.153,311-324. 2.Cohen,G.(1983)Virology 131,159-170. 3.Sternberg, N.&Coulby, J.(1987)J.Mol.Biol.194,453-468. 4.Sternberg, N.&Coulby, J.(1987)J.Mol.Biol.194,469-480. 5.Walker, J.T.&Walker, D.H.(1983)J.Virol.45,118-139. 6.Ikeda,H.&Tomizawa, J.-I.(1968)ColdSpringHarborSymp. Quant.Biol.33,791-798. 7.Yun,T.&Vapnek, D.(1977)Virology 77,376-385. 8.Sternberg, N.(1990)Proc.Natl.Acad.Sci.USA87,103-107. 9.Streisinger, G.,Emrich, J.&Stahl,M.M.(1967)Proc.Natl. Acad.Sci.USA57,292-295. 10.Segev,N.,Laub,A.&Cohen,G.(1980)Virology 101,261-271. 11.Segev,N.&Cohen,G.(1981)Virology 114,333-342. 12.Sternberg, N.,Sauer,B.,Hoess,R.&Abremski, K.(1986)J. Mol.Biol.187,197-212. 13.Shimada, K.,Weisberg, R.A.&Gottesman, M.E.(1972)J. Mol.Biol.93,483-503. 14.Marinus, M.G.,Carroway, M.,Frey,A.Z.,Brown, L.& Arrog,J.A.(1983)Mol.Gen.Genet.192,288-289. 15.Maniatis, T.,Fritsch, E.F.&Sambrook, J.(1982)Molecular Cloning: ALaboratory Manual(ColdSpringHarborLab.,Cold SpringHarbor,NY). 16.Sternberg, N.,Tiemeier, D.&Enquist, L.(1977)Gene1, 255-280. 17.Cohen,G.&Sternberg, N.(1989)J.Mol.Biol.207,99-109.\", 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Mol.Biol.187,197-212. 13.Shimada, K.,Weisberg, R.A.&Gottesman, M.E.(1972)J. Mol.Biol.93,483-503. 14.Marinus, M.G.,Carroway, M.,Frey,A.Z.,Brown, L.& Arrog,J.A.(1983)Mol.Gen.Genet.192,288-289. 15.Maniatis, T.,Fritsch, E.F.&Sambrook, J.(1982)Molecular Cloning: ALaboratory Manual(ColdSpringHarborLab.,Cold SpringHarbor,NY). 16.Sternberg, N.,Tiemeier, D.&Enquist, L.(1977)Gene1, 255-280. 17.Cohen,G.&Sternberg, N.(1989)J.Mol.Biol.207,99-109. 18.Marinus, M.G.(1987)Annu.Rev.Genet.21,113-131. 19.Barras,F.&Marinus, M.(1989)TrendsGenet.5,139-143.Proc.Natl.Acad.Sci.USA87(1990) Downloaded from https://www.pnas.org by 157.211.12.63 on January 4, 2024 from IP address 157.211.12.63.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde751f6c00>, 'json_data': {'input': [\"page_label: 3 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  (C)DNAfromdam'strain N99wasinfectedeitherwithP1wildtypeandA-P1:20b'dam+ (lanes 1and2)orwithP1wildtypeandA-P1:20b'dam- (lanes3and4). DNAswereisolated20and35minafterinfectionandtreatedasin AandB;filterswerehybridized withnick-translated probeB. DNAwillbeterminally redundant and,therefore, cancyclize afterinfections. Thus,itmaybeimportant toregulate pac cleavage invivobymethylation soastoensurethatpac cleavage isusedonlytoinitiatethepackaging. Ifthisis correct,uncleavedpacDNAisolatedfromvirionswouldbe predicted tobeapoorpacsubstrate invitrobecause itis preferentially undermethylated; thisprediction issupported bythefollowing results. First,weisolatedP1pacfragment EcoRI-20 fromviral DNAandassessed itsabilitytobecleavedinvitro.Although somecleavageofthenativefragment wasdetectable, cleav- agewasstimulated -3-foldbypremethylating thefragment withE.coliDammethylase beforeintroducing itintothe pacasereaction(Fig.4,lanes9-13).Thus,-75%ofthe potentially cleavable viralpacDNAcouldnotbecleaved untilitwasmethylated invitro.Thesameresultwasobtained withapac-containing fragmentisolatedfromasecondprep- arationofP1virions(Fig.4A,lanes7and8).Tomeasure directlythemethylation stateofpacDNAweisolatedthree classesofDNAfragments fromP1DNA:the624-bppac- containing EcoRI-20fragment,EcoRIfragments 21,22,23, and24(thesizesofwhichvaryfrom180-450bp),andthe145- to150-bppac-EcoRI fragment, whichcontains thecutviral pacend.TheseDNAswereend-labeled, digested withBclI (whichonlycutsfullyunmethylated 5'-TGATCA sites),and theproducts wereanalyzed ondenaturing sequencing gelsProc.Natl.Acad.Sci.USA87(1990) Ow'.'i a*ioam00 Downloaded from https://www.pnas.org by 157.211.12.63 on January 4, 2024 from IP address 157.211.12.63.\", \"page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Proc.Natl.Acad.Sci.USA87(1990) 8073 A - EcoRI-20 I Nopac--, 910 2'3 ) 5,If ~~~~~~~5'123456711131589101214 -.o-pac-2 FIG.4.Invitropaccleavage ismethylation dependent. All fragments usedherewerelabeled atbothendswith[y-32P]ATP and polynucleotide kinase.Thefragments weretheneithermethylated withtheE.coliDammethylase ornot,incubated withthepacase extract,andthenfractionated in5%acrylamide gel.Lane1contains EcoRI-20 isolated frompRH92-dam-. ThatsameDNA wasincu- batedinthepacasereaction unmethylated (lane2)orafterinvitro methylation (lane3).Positions ofthetwopaccleavage fragments (the 475-bppac-1fragment andthe150-bppac-2fragment) areshownin lane3.Lanes4and5containEcoRl-20 isolatedfrompRH92dam+ incubated inthecomplete pacasereaction without invitromethyl- ation(lane4)orafterinvitromethylation (lane5).Lanes7and8 containEcoRI-20 isolatedfromP1phageDNAandeitheruntreated (lane7)orpremethylated withDam(lane8)beforebeingincubated withpacaseextract.Thisfragment ispartially contaminated withthe smallerP1EcoRIfragments 21-23;theirpositions inthegelare indicated bylinestorightoflane8.Lane9contains EcoRI-20 isolated fromasecondP1phagepreparation; itwasincubated inthepac cleavage reaction withoutpriorinvitroDammethylation (lanes12 and13,respectively) oraftermethylation (lanes10and11,respec- tively).Lanes10and12contain3timesasmuchDNAaslanes11and 13.Lane6contains aHpa IIdigestofpBR322 DNA.Thelabeled fragments weredetected aftergelwasdriedbyexposure toKodak XARfilm. (Fig.5).Fourclassesofsingle-stranded DNAcleavage products shouldbegenerated whenthesixBclIsitesinpac aredigested (Fig.SA).Despite thefactthatEcoRI-20 is contaminated withasmallamountofEcoRI-21, -22,and-23, Bc!Ifragment classes1,2,and3werereadilydetected inthe gels(lanes 5,6,8,and9).Class4fragments weretoolargeto bedistinguished fromuncutEcoRI-20. Theresultsclearly indicate thatBclIcandigest asignificant fraction ofthepac siteinEcoRI-20. Based onyieldofclass1andclass3 cleavage products (Fig.SB,lanes6and9)-30%oftheBclI sitesinpaccouldbecleaved(seeFig. 5legend). Incontrast, yieldofclass2cleavage products (Fig.5B,lane6)wasmore variable-cleavage rangedfrom6to30odepending onsite. Atfirstapproximation thesetworesults appearcontradictory because class2and3cleavage products aregenerated by cuttingthesameBclIrestriction sites.However, thetwo products differinthatproduction ofclass2fragments de- pends onfailure tocleaveclass1sites.Allresults are self-consistent ifDNAmolecules containing unmethylated class2BclIsitesfrequently containunmethylated class1 sites.Whythisshouldbeso,whenonly30%ooftheDNA contains unmethylated BclIsites,isnotclear,butthehigh degreeofundermethylation ofuncut pacisundeniable.\", \"page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  5legend). Incontrast, yieldofclass2cleavage products (Fig.5B,lane6)wasmore variable-cleavage rangedfrom6to30odepending onsite. Atfirstapproximation thesetworesults appearcontradictory because class2and3cleavage products aregenerated by cuttingthesameBclIrestriction sites.However, thetwo products differinthatproduction ofclass2fragments de- pends onfailure tocleaveclass1sites.Allresults are self-consistent ifDNAmolecules containing unmethylated class2BclIsitesfrequently containunmethylated class1 sites.Whythisshouldbeso,whenonly30%ooftheDNA contains unmethylated BclIsites,isnotclear,butthehigh degreeofundermethylation ofuncut pacisundeniable. In contrast, whenthecutpacfragment wasincubated withBcl I,<2%oftheDNAwasdigested (Fig.5B,lanes10-15).Note thatthisfragment canonlygenerate class1BclIproducts because itcontains onlytherighthalfofpac.Whenthe clusterofEcoRI-21-24 fragments isdigested withBclI,the singleBclIsiteinEcoRI-22 ispoorlydigested. Fromthe intensity oftheBc-22fragments (Fig.SB,lane3),weestimateR1-20 _ RI-21-_-}II am ~ ~ )Q I 0 is *)L'R;-24_ *mm _a o-solo t 3~c22-4- FIG.5.BclIdigestion ofpacfragments isolated fromP1DNA. EcoRIfragments wereisolated fromP1virusDNA,labeledwith [y-32P]ATP andpolynucleotide kinase,digested withrestriction enzymeBcl1,heat-denatured, andthenfractionated in8Murea/6% acrylamide gels.(A)Fourclassesofrestriction fragment thatshould begenerated whenunmethylated, end-labeled EcoRI-20 DNAiscut withBc!1.-,BclIrestriction sites;*,DNAendslabeledwith32p. (B)Lanes:1,Unmethylated P1EcoRI-22 DNA;2,unmethylated P1 EcoRI-22 DNAdigested withBc!I.Twosmallerfragments (Bc-22) aregenerated; 4,EcoRI-22-25 fragments isolated fromP1virion DNA;3,sameDNA asinlane4afterBclIdigestion; 5and8, EcoRI-20 isolated fromP1virionDNAcontaminated withsmall amounts ofEcoRl21-23;6and9,thatsamefragment afterBc!I digestion. DNAsinlanes8and9werefractionated bygelelectro- phoresis forashortertimethanthoseinlanes5and6.Positions of class1-4BcIIrestriction fragments areindicated bythebrackets between lanes7and8.Theclass4restriction fragments aredifficult todistinguish inthesegelsfromEcoRI-20. Lanes:10and11,similar tolanes8and9,exceptthattheEcoRI-20 fragment wasfrom a different P1virionpreparation; 12and14,145-to150-bppac-EcoRI fragments isolatedfromtwodifferent P1virionpreparations; 13and 15,BclIdigestsoftheselatterDNAs. that<4%ofthissiteiscut.Thesameresult wasobtained with virionBamHIfragment 9,whichcontains threeclustered Bc! Irestriction sites(18;datanotshown). Weconclude that uncleaved pacDNAinvirusparticles isapoorsubstrate for paccleavage because itispreferentially undermethylated.\", \"page_label: 4 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Lanes:10and11,similar tolanes8and9,exceptthattheEcoRI-20 fragment wasfrom a different P1virionpreparation; 12and14,145-to150-bppac-EcoRI fragments isolatedfromtwodifferent P1virionpreparations; 13and 15,BclIdigestsoftheselatterDNAs. that<4%ofthissiteiscut.Thesameresult wasobtained with virionBamHIfragment 9,whichcontains threeclustered Bc! Irestriction sites(18;datanotshown). Weconclude that uncleaved pacDNAinvirusparticles isapoorsubstrate for paccleavage because itispreferentially undermethylated. DISCUSSION ThephageP1packaging site(pac)isa162-bpsegment ofP1 DNAthatcontains fourhexanucleotide elements (5'- TGATCR) atoneend,threehexanucleotide elements atthe otherend,andasegment of90bpseparating theseelements (Fig.1).Cleavage ofthisDNAwithinthe90-bpspacerregionBBiochemistry: Sternberg andCoulby 2345678 I....iss,..:s:,.Ifs Downloaded from https://www.pnas.org by 157.211.12.63 on January 4, 2024 from IP address 157.211.12.63.\", \"page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  8074 Biochemistry: Sternberg andCoulby isaninitialstepintheP1headfulpackaging process(4).Each hexanucleotide element ofpaccontains aDammethylation site,andweshowherethatmethylation ofatleastsomeof thoseelements isnecessary forpaccleavage. Inaddition, we showthatmethylation probably playsaroleinregulating pac cleavage because uncleaved pacDNAisolatedfromvirions isundermethylated andisapoorsubstrate forpaccleavage invitro,whereas cleaved pacDNAfromvirionsismore extensively methylated. Questions stillremaining arewhy doespaccleavage havetoberegulated, whyisn'titregulated bysimplycontrolling pacasesynthesis, andhowdoesmeth- ylationmediateregulation ofpaccleavage? Regarding thefirstquestion, wesuggestthatthereisneed forregulation. First,ifeachpacsiteonaconcatemer ofP1 DNAwerecleaved atthesame,ornearlythesame,timein thevirallifecycleandtheresulting DNAwerepackaged into phageheads,thevirusparticles produced wouldcontain DNAthatisnotterminally redundant and,therefore, could notbecyclized afterphageinfection. Thus,itisdesirable to cleavepacefficiently onceperconcatemer toinitiatethe processive packaging seriesbutnotmorethanonce.Itcould bearguedthatthisresultmightbeachieved withoutdirectly regulating paccleavage ifconcatemeric DNAwererapidly packaged afterthefirstpacsiteontheconcatemer werecut. Thisisanunlikelyscenario becauseP1proteins necessary for paccleavage aremadeearlyduringphageinfection (Fig.3 showscleavage canbecomplete by20minafterinfection withmethylated pacDNA),wellbeforephageheadsare available toencapsidate DNA.Moreover, ourmeasurements ofcutpacDNAinvivoindicate noincrease inthatDNAin cellsinfectedwithhead-defective mutants(datanotshown). Wemighthaveexpected suchanincrease ifheadsblockpac cleavage byencapsidating DNA. Whyarepacaseproteinsmadeearlyinthevirallifecycle beforebeingneededforDNApackaging? Probably theyare necessary forafunction otherthanpackaging. Perhapsthey playaroleininitiating rollingcirclereplication bynicking circularDNAatpac. Howispacregulated bymethylation? Webelieveakey pointintheregulation processisthereplication ofpacDNA. Thatreplication temporarily generates hemimethylated or evenunmethylated pacDNA(18,19).Normally thehost and/ortheP1methylase wouldrapidlymethylate thenewly replicated pacDNA,rendering itcleavable. Regulation might beachieved byacompeting reaction inwhicheitheraP1or ahost-encoded proteinbindstotheundermethylated pacsite andprevents furthermethylation andsubsequent cleavage. Ourrecentdemonstration thatpacasecanbindto,butcannot cleave,hemimethylated pacDNA(theprimaryproductof pacreplication) suggests thatpacaseitselfmayberesponsi- bleforinhibiting pacmethylation (N.S.,unpublished work). Toaccountforthepaccleavage thatmustoccurtoinitiateP1 packaging oneneedonlyarguethatthecompetition between methylation andprotection isdesigned toproduce onefunc- tionalpacsiteforeverythreetofivenonfunctional sites. Alternatively, themethylation function isenhanced ortheprotection function isdiminished atsomepointinthevirallife cycle,increasing theprobability thatanewlyreplicated pac sitewillbeactivated. TheresultsofFig.3alsosuggestthatreplication ofpac DNAisimportant inregulating pacavailability. Thus,when adam-hostisinfected withbothP1andanunmethylated A-P1pacphage,thereplication ofwhichisblocked bya resident Aprophage, cleavage ofthepacsiteontheinfection A-P1pacphageisnotseenthroughout theexperiment.\", \"page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Alternatively, themethylation function isenhanced ortheprotection function isdiminished atsomepointinthevirallife cycle,increasing theprobability thatanewlyreplicated pac sitewillbeactivated. TheresultsofFig.3alsosuggestthatreplication ofpac DNAisimportant inregulating pacavailability. Thus,when adam-hostisinfected withbothP1andanunmethylated A-P1pacphage,thereplication ofwhichisblocked bya resident Aprophage, cleavage ofthepacsiteontheinfection A-P1pacphageisnotseenthroughout theexperiment. In contrast, apacsiteinaresident prophage inthesamecellis cleavedwithkineticsthatpresumably reflectsynthesis ofthe P1Dammethylase midway through thevirallifecycle.We proposethatinbothcasesmethylation andactivation ofpac isblockedbyprotein(s) boundtothatsite.Forthechromo- somalsitethehost'sreplication machinery occasionally movesthrough thesite,displacing theboundproteinand permitting methylation andsubsequent cleavage ofthesite. Theextrachromosomal Apacsubstrate isneverreplicated duringtheexperiment andisthusnevercleaved. GrowthofP1dam-phageisnormalinadam'hostbut severely restricted inadam-host.UnderfullyDam-con- ditions,P1produces only-5%asmuchphageaswheneither phageorhostmethylase isactive(unpublished work).More- over,thephageproduced underDam-conditions donothave pacends(G.LuceyandN.S.,unpublished work).These resultssupportourconclusions andsuggestthatpackaging of P1DNAunderDam-conditions, occursbyamechanism that doesnotrecognize andcleavepac. WethankDrs.R.MenzelandG.Cohenfortheircriticalreading ofthismanuscript. 1.Bachi,B.&Arber,W.(1977)Mol.Gen.Genet.153,311-324. 2.Cohen,G.(1983)Virology 131,159-170. 3.Sternberg, N.&Coulby, J.(1987)J.Mol.Biol.194,453-468. 4.Sternberg, N.&Coulby, J.(1987)J.Mol.Biol.194,469-480. 5.Walker, J.T.&Walker, D.H.(1983)J.Virol.45,118-139. 6.Ikeda,H.&Tomizawa, J.-I.(1968)ColdSpringHarborSymp. Quant.Biol.33,791-798. 7.Yun,T.&Vapnek, D.(1977)Virology 77,376-385. 8.Sternberg, N.(1990)Proc.Natl.Acad.Sci.USA87,103-107. 9.Streisinger, G.,Emrich, J.&Stahl,M.M.(1967)Proc.Natl. Acad.Sci.USA57,292-295. 10.Segev,N.,Laub,A.&Cohen,G.(1980)Virology 101,261-271. 11.Segev,N.&Cohen,G.(1981)Virology 114,333-342. 12.Sternberg, N.,Sauer,B.,Hoess,R.&Abremski, K.(1986)J. Mol.Biol.187,197-212. 13.Shimada, K.,Weisberg, R.A.&Gottesman, M.E.(1972)J. Mol.Biol.93,483-503. 14.Marinus, M.G.,Carroway, M.,Frey,A.Z.,Brown, L.& Arrog,J.A.(1983)Mol.Gen.Genet.192,288-289. 15.Maniatis, T.,Fritsch, E.F.&Sambrook, J.(1982)Molecular Cloning: ALaboratory Manual(ColdSpringHarborLab.,Cold SpringHarbor,NY). 16.Sternberg, N.,Tiemeier, D.&Enquist, L.(1977)Gene1, 255-280. 17.Cohen,G.&Sternberg, N.(1989)J.Mol.Biol.207,99-109.\", 'page_label: 5 file_path: /workspaces/ml-learning/src/phages/data/P1/sternberg-coulby-1990-cleavage-of-the-bacteriophage-p1-packaging-site-(pac)-is-regulated-by-adenine-methylation.pdf  Mol.Biol.187,197-212. 13.Shimada, K.,Weisberg, R.A.&Gottesman, M.E.(1972)J. Mol.Biol.93,483-503. 14.Marinus, M.G.,Carroway, M.,Frey,A.Z.,Brown, L.& Arrog,J.A.(1983)Mol.Gen.Genet.192,288-289. 15.Maniatis, T.,Fritsch, E.F.&Sambrook, J.(1982)Molecular Cloning: ALaboratory Manual(ColdSpringHarborLab.,Cold SpringHarbor,NY). 16.Sternberg, N.,Tiemeier, D.&Enquist, L.(1977)Gene1, 255-280. 17.Cohen,G.&Sternberg, N.(1989)J.Mol.Biol.207,99-109. 18.Marinus, M.G.(1987)Annu.Rev.Genet.21,113-131. 19.Barras,F.&Marinus, M.(1989)TrendsGenet.5,139-143.Proc.Natl.Acad.Sci.USA87(1990) Downloaded from https://www.pnas.org by 157.211.12.63 on January 4, 2024 from IP address 157.211.12.63.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:13 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'97'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'996015'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'239ms'), (b'x-request-id', b'dd7486515dce25735bdbe0b138519d70'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249e91b6d5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:36:13 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'97'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'996015'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'239ms'), (b'x-request-id', b'dd7486515dce25735bdbe0b138519d70'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'841249e91b6d5d14-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n"
     ]
    }
   ],
   "source": [
    "from llama_index import VectorStoreIndex\n",
    "\n",
    "index = VectorStoreIndex.from_documents(documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde74f91300>, 'json_data': {'input': ['What is the morphology of the P1 Phage?'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde74f91300>, 'json_data': {'input': ['What is the morphology of the P1 Phage?'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.connection:close.started\n",
      "close.started\n",
      "DEBUG:httpcore.connection:close.complete\n",
      "close.complete\n",
      "DEBUG:httpcore.connection:connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "DEBUG:httpcore.connection:connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde75508d10>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde75508d10>\n",
      "DEBUG:httpcore.connection:start_tls.started ssl_context=<ssl.SSLContext object at 0x7fdebfff7a40> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fdebfff7a40> server_hostname='api.openai.com' timeout=60.0\n",
      "DEBUG:httpcore.connection:start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde74302790>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde74302790>\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:46:46 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'19'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'999990'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'0s'), (b'x-request-id', b'3a8cbb14753664419cc0a083a81abee8'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412595cca16aae4-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:46:46 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'19'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'999990'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'0s'), (b'x-request-id', b'3a8cbb14753664419cc0a083a81abee8'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412595cca16aae4-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:llama_index.indices.utils:> Top 2 nodes:\n",
      "> [Node 5bb7a454-99ef-4460-8596-902316659ca4] [Similarity score:             0.893276] MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\n",
      "andR.Hoess....\n",
      "> [Node af33e22f-09b3-470d-ac24-2c982451151a] [Similarity score:             0.892821] JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\n",
      "0021-9193/04/$08.00 /H110010 D...\n",
      "> Top 2 nodes:\n",
      "> [Node 5bb7a454-99ef-4460-8596-902316659ca4] [Similarity score:             0.893276] MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\n",
      "andR.Hoess....\n",
      "> [Node af33e22f-09b3-470d-ac24-2c982451151a] [Similarity score:             0.892821] JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\n",
      "0021-9193/04/$08.00 /H110010 D...\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/chat/completions', 'files': None, 'json_data': {'messages': [{'role': <MessageRole.SYSTEM: 'system'>, 'content': \"You are an expert Q&A system that is trusted around the world.\\nAlways answer the query using the provided context information, and not prior knowledge.\\nSome rules to follow:\\n1. Never directly reference the given context in your answer.\\n2. Avoid statements like 'Based on the context, ...' or 'The context information ...' or anything along those lines.\"}, {'role': <MessageRole.USER: 'user'>, 'content': \"Context information is below.\\n---------------------\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nMORPHOLOGY OFP1MUTANTS 1139\\n142.\\n44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\\nandR.Hoess.1981.Site-specific recombination andits\\nroleinthelifecycleofbacteriophage P1.ColdSpring\\nHarborSymp.Quant.Biol.45:297-309.\\n45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\\nBruijn,andA.I.Bukhari. 1978.Relationships between\\ntemperate phagesMuandP1.Virology 89:146-161.\\n46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\\n1980.Invertible DNAdetermines hostspecificity ofbac-\\nteriophage Mu.Nature(London) 286:218-222.\\n47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi-\\ncalvariants ofcoliphage P1.J.Virol.5:765-782.\\n48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic\\nstudiesofcoliphage P1.I.Mapping byuseofprophage\\ndeletions. J.Virol.16:525-534.\\n49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic\\nstudiesofcoliphage P1. III.Extended genetic map.J.\\nVirol.20:177-187.\\n50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in\\ncoliphage P1affecting hostcelllysis.J.Virol.35:519-530.\\n51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.),\\nProgress inclinicalbiological research, vol.64.AlanR.\\nLiss,Inc.,NewYork.\\n52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman,\\nH.Lyle,andW.B.Wood.1970.Assembly ofbacterio-\\nphageT4tailfibers. II.Isolation andcharacterization of\\ntailfiberprecursors. J.Mol.Biol.54:15-31.\\n53.Wood,W.B.,andJ.King.1979.Genetic control of\\ncomplex bacteriophage assembly. Compr. Virol.13:581-\\n633.\\n54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\\nbacteriophage T4.Bacteriol. Rev.40:847-868.\\n55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\\nofgeneproduct positions inbacteriophage T4byspecific\\nantibody association. J.Mol.Biol.51:411-421.\\n56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\\nE.Kellenberger. 1970.Studies onthemorphopoiesis of\\ntheheadofbacteriophage T-even. VIII.Multilayered\\npolyheads. J.Mol.Biol.50:35-58.\\n57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In\\nS.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer\\nInstitute, Bethesda, Md.VOL.45,1983\\n\\npage_label: 1\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf\\n\\nJOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\\n0021-9193/04/$08.00 /H110010 DOI: 10.1128/JB.186.21.7032–7068.2004\\nCopyright © 2004, American Society for Microbiology. All Rights Reserved.\\nGenome of Bacteriophage P1†\\nMałgorzata B. Łobocka,1* Debra J. Rose,2Guy Plunkett, III,2Marek Rusin,3Arkadiusz Samojedny,3\\nHansjo ¨rg Lehnherr,4Michael B. Yarmolinsky,5and Frederick R. Blattner2\\nDepartment of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, Warsaw,1\\nand Department of Tumor Biology, Centre of Oncology, M. Skłodowska-Curie Memorial Institute, Gliwice,3Poland;\\nLaboratory of Genetics, University of Wisconsin, Madison, Wisconsin2; Department of Genetics and Biochemistry,\\nInstitute of Microbiology, Ernst-Moritz-Arndt-University Greifswald, Greifswald, Germany4; and Laboratory\\nof Biochemistry, National Cancer Institute, National Institutes of Health, Bethesda, Maryland5\\nReceived 11 March 2004/Accepted 9 July 2004\\nP1 is a bacteriophage of Escherichia coli and other enteric bacteria. It lysogenizes its hosts as a circular,\\nlow-copy-number plasmid. We have determined the complete nucleotide sequences of two strains of a P1thermoinducible mutant, P1 c1-100. The P1 genome (93,601 bp) contains at least 117 genes, of which almost\\ntwo-thirds had not been sequenced previously and 49 have no homologs in other organisms. Protein-codinggenes occupy 92% of the genome and are organized in 45 operons, of which four are decisive for the choicebetween lysis and lysogeny. Four others ensure plasmid maintenance. The majority of the remaining 37operons are involved in lytic development. Seventeen operons are transcribed from /H9268\\n70promoters directly\\ncontrolled by the master phage repressor C1. Late operons are transcribed from promoters recognized by theE. coli RNA polymerase holoenzyme in the presence of the Lpa protein, the product of a C1-controlled P1 gene.\\nThree species of P1-encoded tRNAs provide differential controls of translation, and a P1-encoded DNAmethyltransferase with putative bifunctionality inﬂuences transcription, replication, and DNA packaging. Thegenome is particularly rich in Chi recombinogenic sites. The base content and distribution in P1 DNA indicatethat replication of P1 from its plasmid origin had more impact on the base compositional asymmetries of theP1 genome than replication from the lytic origin of replication.\\nThree bacteriophages of Escherichia coli with different life\\nstyles have been particularly instrumental in the developmentof the concepts and tools of molecular biology. They are thevirulent phage T4 and the two temperate phages, /H9261and P1.\\nThe two temperate phages were ﬁrst described in close suc-cession in the middle of the last century, namely in 1951 (phageP1 [31]) and 1953 ( /H9261[185] and informally even earlier, in the\\nﬁrst issue of Evelyn Witkin’s Microbial Genetics Bulletin , 1950\\n[184a]). Whereas /H9261prophage leads an essentially passive exis-\\ntence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia\\ncolihave been reported. We present here the complete nucle-\\notide sequence of P1 and summarize its salient features.\\nP1 infects and lysogenizes E. coli and several other enteric\\nbacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath.\\n---------------------\\nGiven the context information and not prior knowledge, answer the query.\\nQuery: What is the morphology of the P1 Phage?\\nAnswer: \"}], 'model': 'gpt-3.5-turbo', 'stream': True, 'temperature': 0.1}}\n",
      "Request options: {'method': 'post', 'url': '/chat/completions', 'files': None, 'json_data': {'messages': [{'role': <MessageRole.SYSTEM: 'system'>, 'content': \"You are an expert Q&A system that is trusted around the world.\\nAlways answer the query using the provided context information, and not prior knowledge.\\nSome rules to follow:\\n1. Never directly reference the given context in your answer.\\n2. Avoid statements like 'Based on the context, ...' or 'The context information ...' or anything along those lines.\"}, {'role': <MessageRole.USER: 'user'>, 'content': \"Context information is below.\\n---------------------\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nMORPHOLOGY OFP1MUTANTS 1139\\n142.\\n44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\\nandR.Hoess.1981.Site-specific recombination andits\\nroleinthelifecycleofbacteriophage P1.ColdSpring\\nHarborSymp.Quant.Biol.45:297-309.\\n45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\\nBruijn,andA.I.Bukhari. 1978.Relationships between\\ntemperate phagesMuandP1.Virology 89:146-161.\\n46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\\n1980.Invertible DNAdetermines hostspecificity ofbac-\\nteriophage Mu.Nature(London) 286:218-222.\\n47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi-\\ncalvariants ofcoliphage P1.J.Virol.5:765-782.\\n48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic\\nstudiesofcoliphage P1.I.Mapping byuseofprophage\\ndeletions. J.Virol.16:525-534.\\n49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic\\nstudiesofcoliphage P1. III.Extended genetic map.J.\\nVirol.20:177-187.\\n50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in\\ncoliphage P1affecting hostcelllysis.J.Virol.35:519-530.\\n51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.),\\nProgress inclinicalbiological research, vol.64.AlanR.\\nLiss,Inc.,NewYork.\\n52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman,\\nH.Lyle,andW.B.Wood.1970.Assembly ofbacterio-\\nphageT4tailfibers. II.Isolation andcharacterization of\\ntailfiberprecursors. J.Mol.Biol.54:15-31.\\n53.Wood,W.B.,andJ.King.1979.Genetic control of\\ncomplex bacteriophage assembly. Compr. Virol.13:581-\\n633.\\n54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\\nbacteriophage T4.Bacteriol. Rev.40:847-868.\\n55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\\nofgeneproduct positions inbacteriophage T4byspecific\\nantibody association. J.Mol.Biol.51:411-421.\\n56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\\nE.Kellenberger. 1970.Studies onthemorphopoiesis of\\ntheheadofbacteriophage T-even. VIII.Multilayered\\npolyheads. J.Mol.Biol.50:35-58.\\n57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In\\nS.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer\\nInstitute, Bethesda, Md.VOL.45,1983\\n\\npage_label: 1\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf\\n\\nJOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\\n0021-9193/04/$08.00 /H110010 DOI: 10.1128/JB.186.21.7032–7068.2004\\nCopyright © 2004, American Society for Microbiology. All Rights Reserved.\\nGenome of Bacteriophage P1†\\nMałgorzata B. Łobocka,1* Debra J. Rose,2Guy Plunkett, III,2Marek Rusin,3Arkadiusz Samojedny,3\\nHansjo ¨rg Lehnherr,4Michael B. Yarmolinsky,5and Frederick R. Blattner2\\nDepartment of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, Warsaw,1\\nand Department of Tumor Biology, Centre of Oncology, M. Skłodowska-Curie Memorial Institute, Gliwice,3Poland;\\nLaboratory of Genetics, University of Wisconsin, Madison, Wisconsin2; Department of Genetics and Biochemistry,\\nInstitute of Microbiology, Ernst-Moritz-Arndt-University Greifswald, Greifswald, Germany4; and Laboratory\\nof Biochemistry, National Cancer Institute, National Institutes of Health, Bethesda, Maryland5\\nReceived 11 March 2004/Accepted 9 July 2004\\nP1 is a bacteriophage of Escherichia coli and other enteric bacteria. It lysogenizes its hosts as a circular,\\nlow-copy-number plasmid. We have determined the complete nucleotide sequences of two strains of a P1thermoinducible mutant, P1 c1-100. The P1 genome (93,601 bp) contains at least 117 genes, of which almost\\ntwo-thirds had not been sequenced previously and 49 have no homologs in other organisms. Protein-codinggenes occupy 92% of the genome and are organized in 45 operons, of which four are decisive for the choicebetween lysis and lysogeny. Four others ensure plasmid maintenance. The majority of the remaining 37operons are involved in lytic development. Seventeen operons are transcribed from /H9268\\n70promoters directly\\ncontrolled by the master phage repressor C1. Late operons are transcribed from promoters recognized by theE. coli RNA polymerase holoenzyme in the presence of the Lpa protein, the product of a C1-controlled P1 gene.\\nThree species of P1-encoded tRNAs provide differential controls of translation, and a P1-encoded DNAmethyltransferase with putative bifunctionality inﬂuences transcription, replication, and DNA packaging. Thegenome is particularly rich in Chi recombinogenic sites. The base content and distribution in P1 DNA indicatethat replication of P1 from its plasmid origin had more impact on the base compositional asymmetries of theP1 genome than replication from the lytic origin of replication.\\nThree bacteriophages of Escherichia coli with different life\\nstyles have been particularly instrumental in the developmentof the concepts and tools of molecular biology. They are thevirulent phage T4 and the two temperate phages, /H9261and P1.\\nThe two temperate phages were ﬁrst described in close suc-cession in the middle of the last century, namely in 1951 (phageP1 [31]) and 1953 ( /H9261[185] and informally even earlier, in the\\nﬁrst issue of Evelyn Witkin’s Microbial Genetics Bulletin , 1950\\n[184a]). Whereas /H9261prophage leads an essentially passive exis-\\ntence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia\\ncolihave been reported. We present here the complete nucle-\\notide sequence of P1 and summarize its salient features.\\nP1 infects and lysogenizes E. coli and several other enteric\\nbacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath.\\n---------------------\\nGiven the context information and not prior knowledge, answer the query.\\nQuery: What is the morphology of the P1 Phage?\\nAnswer: \"}], 'model': 'gpt-3.5-turbo', 'stream': True, 'temperature': 0.1}}\n",
      "DEBUG:httpcore.connection:close.started\n",
      "close.started\n",
      "DEBUG:httpcore.connection:close.complete\n",
      "close.complete\n",
      "DEBUG:httpcore.connection:connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "DEBUG:httpcore.connection:connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde75b85050>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde75b85050>\n",
      "DEBUG:httpcore.connection:start_tls.started ssl_context=<ssl.SSLContext object at 0x7fde7521b770> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fde7521b770> server_hostname='api.openai.com' timeout=60.0\n",
      "DEBUG:httpcore.connection:start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde73ea6350>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde73ea6350>\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:46:47 GMT'), (b'Content-Type', b'text/event-stream'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'Cache-Control', b'no-cache, must-revalidate'), (b'openai-model', b'gpt-3.5-turbo-0613'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'404'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'10000'), (b'x-ratelimit-limit-tokens', b'60000'), (b'x-ratelimit-limit-tokens_usage_based', b'60000'), (b'x-ratelimit-remaining-requests', b'9999'), (b'x-ratelimit-remaining-tokens', b'58405'), (b'x-ratelimit-remaining-tokens_usage_based', b'58405'), (b'x-ratelimit-reset-requests', b'8.64s'), (b'x-ratelimit-reset-tokens', b'1.595s'), (b'x-ratelimit-reset-tokens_usage_based', b'1.595s'), (b'x-request-id', b'944e8eb97e45f766574e4cee4073b4a4'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412595f68c5a87d-SYD'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:46:47 GMT'), (b'Content-Type', b'text/event-stream'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'Cache-Control', b'no-cache, must-revalidate'), (b'openai-model', b'gpt-3.5-turbo-0613'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'404'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'10000'), (b'x-ratelimit-limit-tokens', b'60000'), (b'x-ratelimit-limit-tokens_usage_based', b'60000'), (b'x-ratelimit-remaining-requests', b'9999'), (b'x-ratelimit-remaining-tokens', b'58405'), (b'x-ratelimit-remaining-tokens_usage_based', b'58405'), (b'x-ratelimit-reset-requests', b'8.64s'), (b'x-ratelimit-reset-tokens', b'1.595s'), (b'x-ratelimit-reset-tokens_usage_based', b'1.595s'), (b'x-request-id', b'944e8eb97e45f766574e4cee4073b4a4'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412595f68c5a87d-SYD'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/chat/completions \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "The morphology of the P1 phage consists of an icosahedral head attached to a tail that bears six kinked tail fibers. The tail consists of a tail tube and a contractile sheath.DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n"
     ]
    }
   ],
   "source": [
    "# https://docs.llamaindex.ai/en/stable/module_guides/deploying/query_engine/root.html           \n",
    "query_engine = index.as_query_engine(streaming=True)\n",
    "query_str = \"What is the morphology of the P1 Phage?\"\n",
    "streaming_response = query_engine.query(\"What is the morphology of the P1 Phage?\")\n",
    "streaming_response.print_response_stream()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "response = streaming_response.get_response()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[NodeWithScore(node=TextNode(id_='5bb7a454-99ef-4460-8596-902316659ca4', embedding=None, metadata={'page_label': '22', 'file_name': 'jvirol00150-0228.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf', 'file_type': 'application/pdf', 'file_size': 6891222, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='b652dc39-a92b-4c58-847a-dd5b4c870838', node_type=<ObjectType.DOCUMENT: '4'>, metadata={'page_label': '22', 'file_name': 'jvirol00150-0228.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf', 'file_type': 'application/pdf', 'file_size': 6891222, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, hash='68e46c57bac42cf6de73f6e56ee28fa2f35037cb74b4afcee83e15153ba4d6bb'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='b383a3b0-ad0e-4767-95e5-ae86dc4f8e29', node_type=<ObjectType.TEXT: '1'>, metadata={'page_label': '21', 'file_name': 'jvirol00150-0228.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf', 'file_type': 'application/pdf', 'file_size': 6891222, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, hash='5722cb7f573501443d58a80cd95c4507b186daa5c23d149276d357bc219d6237'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='15388096-ce59-4a01-bfe9-918c84caa0ed', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='29b3b7369936003b94996e57640e9501b6699c85861ca3dda23e84de7d073a52')}, hash='68e46c57bac42cf6de73f6e56ee28fa2f35037cb74b4afcee83e15153ba4d6bb', text=\"MORPHOLOGY OFP1MUTANTS 1139\\n142.\\n44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\\nandR.Hoess.1981.Site-specific recombination andits\\nroleinthelifecycleofbacteriophage P1.ColdSpring\\nHarborSymp.Quant.Biol.45:297-309.\\n45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\\nBruijn,andA.I.Bukhari. 1978.Relationships between\\ntemperate phagesMuandP1.Virology 89:146-161.\\n46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\\n1980.Invertible DNAdetermines hostspecificity ofbac-\\nteriophage Mu.Nature(London) 286:218-222.\\n47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi-\\ncalvariants ofcoliphage P1.J.Virol.5:765-782.\\n48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic\\nstudiesofcoliphage P1.I.Mapping byuseofprophage\\ndeletions. J.Virol.16:525-534.\\n49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic\\nstudiesofcoliphage P1. III.Extended genetic map.J.\\nVirol.20:177-187.\\n50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in\\ncoliphage P1affecting hostcelllysis.J.Virol.35:519-530.\\n51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.),\\nProgress inclinicalbiological research, vol.64.AlanR.\\nLiss,Inc.,NewYork.\\n52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman,\\nH.Lyle,andW.B.Wood.1970.Assembly ofbacterio-\\nphageT4tailfibers. II.Isolation andcharacterization of\\ntailfiberprecursors. J.Mol.Biol.54:15-31.\\n53.Wood,W.B.,andJ.King.1979.Genetic control of\\ncomplex bacteriophage assembly. Compr. Virol.13:581-\\n633.\\n54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\\nbacteriophage T4.Bacteriol. Rev.40:847-868.\\n55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\\nofgeneproduct positions inbacteriophage T4byspecific\\nantibody association. J.Mol.Biol.51:411-421.\\n56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\\nE.Kellenberger. 1970.Studies onthemorphopoiesis of\\ntheheadofbacteriophage T-even. VIII.Multilayered\\npolyheads. J.Mol.Biol.50:35-58.\\n57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In\\nS.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer\\nInstitute, Bethesda, Md.VOL.45,1983\", start_char_idx=0, end_char_idx=1993, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=0.8932762012956971),\n",
       " NodeWithScore(node=TextNode(id_='af33e22f-09b3-470d-ac24-2c982451151a', embedding=None, metadata={'page_label': '1', 'file_name': '0343-04.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/0343-04.pdf', 'file_type': 'application/pdf', 'file_size': 1185513, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='58d54542-2014-44fa-b44d-72f11cc716a6', node_type=<ObjectType.DOCUMENT: '4'>, metadata={'page_label': '1', 'file_name': '0343-04.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/0343-04.pdf', 'file_type': 'application/pdf', 'file_size': 1185513, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, hash='f7f48f472908d3ae559c62d37637f21df90a8b83ac73dcec5369f0e830a211b7'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='afa4ca98-e049-40bb-bda4-bfa3330b4e11', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='9a71ba2bbc452c33235eaccf5c7eb747b4b298cd4ec6c5526bbfdcabc8551386')}, hash='7a44790cd51d9ade5faa703f201617f51a520af1a3823f41ecf4fb5a6d2fc01e', text='JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\\n0021-9193/04/$08.00 /H110010 DOI: 10.1128/JB.186.21.7032–7068.2004\\nCopyright © 2004, American Society for Microbiology. All Rights Reserved.\\nGenome of Bacteriophage P1†\\nMałgorzata B. Łobocka,1* Debra J. Rose,2Guy Plunkett, III,2Marek Rusin,3Arkadiusz Samojedny,3\\nHansjo ¨rg Lehnherr,4Michael B. Yarmolinsky,5and Frederick R. Blattner2\\nDepartment of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, Warsaw,1\\nand Department of Tumor Biology, Centre of Oncology, M. Skłodowska-Curie Memorial Institute, Gliwice,3Poland;\\nLaboratory of Genetics, University of Wisconsin, Madison, Wisconsin2; Department of Genetics and Biochemistry,\\nInstitute of Microbiology, Ernst-Moritz-Arndt-University Greifswald, Greifswald, Germany4; and Laboratory\\nof Biochemistry, National Cancer Institute, National Institutes of Health, Bethesda, Maryland5\\nReceived 11 March 2004/Accepted 9 July 2004\\nP1 is a bacteriophage of Escherichia coli and other enteric bacteria. It lysogenizes its hosts as a circular,\\nlow-copy-number plasmid. We have determined the complete nucleotide sequences of two strains of a P1thermoinducible mutant, P1 c1-100. The P1 genome (93,601 bp) contains at least 117 genes, of which almost\\ntwo-thirds had not been sequenced previously and 49 have no homologs in other organisms. Protein-codinggenes occupy 92% of the genome and are organized in 45 operons, of which four are decisive for the choicebetween lysis and lysogeny. Four others ensure plasmid maintenance. The majority of the remaining 37operons are involved in lytic development. Seventeen operons are transcribed from /H9268\\n70promoters directly\\ncontrolled by the master phage repressor C1. Late operons are transcribed from promoters recognized by theE. coli RNA polymerase holoenzyme in the presence of the Lpa protein, the product of a C1-controlled P1 gene.\\nThree species of P1-encoded tRNAs provide differential controls of translation, and a P1-encoded DNAmethyltransferase with putative bifunctionality inﬂuences transcription, replication, and DNA packaging. Thegenome is particularly rich in Chi recombinogenic sites. The base content and distribution in P1 DNA indicatethat replication of P1 from its plasmid origin had more impact on the base compositional asymmetries of theP1 genome than replication from the lytic origin of replication.\\nThree bacteriophages of Escherichia coli with different life\\nstyles have been particularly instrumental in the developmentof the concepts and tools of molecular biology. They are thevirulent phage T4 and the two temperate phages, /H9261and P1.\\nThe two temperate phages were ﬁrst described in close suc-cession in the middle of the last century, namely in 1951 (phageP1 [31]) and 1953 ( /H9261[185] and informally even earlier, in the\\nﬁrst issue of Evelyn Witkin’s Microbial Genetics Bulletin , 1950\\n[184a]). Whereas /H9261prophage leads an essentially passive exis-\\ntence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia\\ncolihave been reported. We present here the complete nucle-\\notide sequence of P1 and summarize its salient features.\\nP1 infects and lysogenizes E. coli and several other enteric\\nbacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath.', start_char_idx=0, end_char_idx=3562, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=0.8928214501522723)]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response.source_nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde74f92160>, 'json_data': {'input': ['What is the morphology of the P1 Phage?'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde74f92160>, 'json_data': {'input': ['What is the morphology of the P1 Phage?'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde74f92160>, 'json_data': {'input': ['What is the morphology of the P1 Phage?'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.connection:close.started\n",
      "close.started\n",
      "close.started\n",
      "DEBUG:httpcore.connection:close.complete\n",
      "close.complete\n",
      "close.complete\n",
      "DEBUG:httpcore.connection:connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "DEBUG:httpcore.connection:connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde73fad610>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde73fad610>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde73fad610>\n",
      "DEBUG:httpcore.connection:start_tls.started ssl_context=<ssl.SSLContext object at 0x7fdebfff7a40> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fdebfff7a40> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fdebfff7a40> server_hostname='api.openai.com' timeout=60.0\n",
      "DEBUG:httpcore.connection:start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde75508d10>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde75508d10>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde75508d10>\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:51:15 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'33'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'999990'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'0s'), (b'x-request-id', b'c9bbb56f04274c49ea2e1973a9330375'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Set-Cookie', b'__cf_bm=b.asgSS9XMrnWGove62yWTl8a5abamduu9rP1hqP6WM-1704527475-1-AboXNcgL8YCxpsmVm99Pw9uqFEOQimgqS7TehI1+yjnUgqAoOd3s1/s9VGew25tXlup60Zjp5J7x6YWvatXbGTI=; path=/; expires=Sat, 06-Jan-24 08:21:15 GMT; domain=.api.openai.com; HttpOnly; Secure; SameSite=None'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84125fed6fdea82f-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:51:15 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'33'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'999990'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'0s'), (b'x-request-id', b'c9bbb56f04274c49ea2e1973a9330375'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Set-Cookie', b'__cf_bm=b.asgSS9XMrnWGove62yWTl8a5abamduu9rP1hqP6WM-1704527475-1-AboXNcgL8YCxpsmVm99Pw9uqFEOQimgqS7TehI1+yjnUgqAoOd3s1/s9VGew25tXlup60Zjp5J7x6YWvatXbGTI=; path=/; expires=Sat, 06-Jan-24 08:21:15 GMT; domain=.api.openai.com; HttpOnly; Secure; SameSite=None'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84125fed6fdea82f-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:51:15 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'33'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'999990'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'0s'), (b'x-request-id', b'c9bbb56f04274c49ea2e1973a9330375'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Set-Cookie', b'__cf_bm=b.asgSS9XMrnWGove62yWTl8a5abamduu9rP1hqP6WM-1704527475-1-AboXNcgL8YCxpsmVm99Pw9uqFEOQimgqS7TehI1+yjnUgqAoOd3s1/s9VGew25tXlup60Zjp5J7x6YWvatXbGTI=; path=/; expires=Sat, 06-Jan-24 08:21:15 GMT; domain=.api.openai.com; HttpOnly; Secure; SameSite=None'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84125fed6fdea82f-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:llama_index.indices.utils:> Top 2 nodes:\n",
      "> [Node 5bb7a454-99ef-4460-8596-902316659ca4] [Similarity score:             0.893276] MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\n",
      "andR.Hoess....\n",
      "> [Node af33e22f-09b3-470d-ac24-2c982451151a] [Similarity score:             0.892821] JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\n",
      "0021-9193/04/$08.00 /H110010 D...\n",
      "> Top 2 nodes:\n",
      "> [Node 5bb7a454-99ef-4460-8596-902316659ca4] [Similarity score:             0.893276] MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\n",
      "andR.Hoess....\n",
      "> [Node af33e22f-09b3-470d-ac24-2c982451151a] [Similarity score:             0.892821] JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\n",
      "0021-9193/04/$08.00 /H110010 D...\n",
      "> Top 2 nodes:\n",
      "> [Node 5bb7a454-99ef-4460-8596-902316659ca4] [Similarity score:             0.893276] MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\n",
      "andR.Hoess....\n",
      "> [Node af33e22f-09b3-470d-ac24-2c982451151a] [Similarity score:             0.892821] JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\n",
      "0021-9193/04/$08.00 /H110010 D...\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/chat/completions', 'files': None, 'json_data': {'messages': [{'role': <MessageRole.USER: 'user'>, 'content': \"Please provide an answer based solely on the provided sources. When referencing information from a source, cite the appropriate source(s) using their corresponding numbers. Every answer should include at least one source citation. Only cite a source when you are explicitly referencing it. If none of the sources are helpful, you should indicate that. For example:\\nSource 1:\\nThe sky is red in the evening and blue in the morning.\\nSource 2:\\nWater is wet when the sky is red.\\nQuery: When is water wet?\\nAnswer: Water will be wet when the sky is red [2], which occurs in the evening [1].\\nNow it's your turn. Below are several numbered sources of information:\\n------\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 1:\\nMORPHOLOGY OFP1MUTANTS 1139\\n142.\\n44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\\nandR.Hoess.1981.Site-specific recombination andits\\nroleinthelifecycleofbacteriophage P1.ColdSpring\\nHarborSymp.Quant.Biol.45:297-309.\\n45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\\nBruijn,andA.I.Bukhari. 1978.Relationships between\\ntemperate phagesMuandP1.Virology 89:146-161.\\n46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\\n1980.Invertible DNAdetermines hostspecificity ofbac-\\nteriophage Mu.Nature(London) 286:218-222.\\n47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi-\\ncalvariants ofcoliphage P1.J.Virol.5:765-782.\\n48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic\\nstudiesofcoliphage P1.I.Mapping byuseofprophage\\ndeletions. J.Virol.16:525-534.\\n49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic\\nstudiesofcoliphage P1. III.Extended genetic map.J.\\nVirol.20:177-187.\\n50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in\\ncoliphage P1affecting hostcelllysis.J.Virol.35:519-530.\\n51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.),\\nProgress inclinicalbiological research, vol.64.AlanR.\\nLiss,Inc.,NewYork.\\n52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman,\\nH.Lyle,andW.B.Wood.1970.Assembly ofbacterio-\\nphageT4tailfibers. II.Isolation andcharacterization of\\ntailfiberprecursors.\\n\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 2:\\nII.Isolation andcharacterization of\\ntailfiberprecursors. J.Mol.Biol.54:15-31.\\n53.Wood,W.B.,andJ.King.1979.Genetic control of\\ncomplex bacteriophage assembly. Compr. Virol.13:581-\\n633.\\n54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\\nbacteriophage T4.Bacteriol. Rev.40:847-868.\\n55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\\nofgeneproduct positions inbacteriophage T4byspecific\\nantibody association. J.Mol.Biol.51:411-421.\\n56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\\nE.Kellenberger. 1970.Studies onthemorphopoiesis of\\ntheheadofbacteriophage T-even. VIII.Multilayered\\npolyheads. J.Mol.Biol.50:35-58.\\n57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In\\nS.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer\\nInstitute, Bethesda, Md.VOL.45,1983\\n\\npage_label: 1\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf\\n\\nSource 3:\\nJOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\\n0021-9193/04/$08.00 /H110010 DOI: 10.1128/JB.186.21.7032–7068.2004\\nCopyright © 2004, American Society for Microbiology. All Rights Reserved.\\nGenome of Bacteriophage P1†\\nMałgorzata B. Łobocka,1* Debra J. Rose,2Guy Plunkett, III,2Marek Rusin,3Arkadiusz Samojedny,3\\nHansjo ¨rg Lehnherr,4Michael B. Yarmolinsky,5and Frederick R. Blattner2\\nDepartment of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, Warsaw,1\\nand Department of Tumor Biology, Centre of Oncology, M. Skłodowska-Curie Memorial Institute, Gliwice,3Poland;\\nLaboratory of Genetics, University of Wisconsin, Madison, Wisconsin2; Department of Genetics and Biochemistry,\\nInstitute of Microbiology, Ernst-Moritz-Arndt-University Greifswald, Greifswald, Germany4; and Laboratory\\nof Biochemistry, National Cancer Institute, National Institutes of Health, Bethesda, Maryland5\\nReceived 11 March 2004/Accepted 9 July 2004\\nP1 is a bacteriophage of Escherichia coli and other enteric bacteria. It lysogenizes its hosts as a circular,\\nlow-copy-number plasmid. We have determined the complete nucleotide sequences of two strains of a P1thermoinducible mutant, P1 c1-100. The P1 genome (93,601 bp) contains at least 117 genes, of which almost\\ntwo-thirds had not been sequenced previously and 49 have no homologs in other organisms. Protein-codinggenes occupy 92% of the genome and are organized in 45 operons, of which four are decisive for the choicebetween lysis and lysogeny. Four others ensure plasmid maintenance. The majority of the remaining 37operons are involved in lytic development.\\n\\npage_label: 1\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf\\n\\nSource 4:\\nThe majority of the remaining 37operons are involved in lytic development. Seventeen operons are transcribed from /H9268\\n70promoters directly\\ncontrolled by the master phage repressor C1. Late operons are transcribed from promoters recognized by theE. coli RNA polymerase holoenzyme in the presence of the Lpa protein, the product of a C1-controlled P1 gene.\\nThree species of P1-encoded tRNAs provide differential controls of translation, and a P1-encoded DNAmethyltransferase with putative bifunctionality inﬂuences transcription, replication, and DNA packaging. Thegenome is particularly rich in Chi recombinogenic sites. The base content and distribution in P1 DNA indicatethat replication of P1 from its plasmid origin had more impact on the base compositional asymmetries of theP1 genome than replication from the lytic origin of replication.\\nThree bacteriophages of Escherichia coli with different life\\nstyles have been particularly instrumental in the developmentof the concepts and tools of molecular biology. They are thevirulent phage T4 and the two temperate phages, /H9261and P1.\\nThe two temperate phages were ﬁrst described in close suc-cession in the middle of the last century, namely in 1951 (phageP1 [31]) and 1953 ( /H9261[185] and informally even earlier, in the\\nﬁrst issue of Evelyn Witkin’s Microbial Genetics Bulletin , 1950\\n[184a]). Whereas /H9261prophage leads an essentially passive exis-\\ntence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia\\ncolihave been reported. We present here the complete nucle-\\notide sequence of P1 and summarize its salient features.\\nP1 infects and lysogenizes E. coli and several other enteric\\nbacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath.\\n------\\nQuery: What is the morphology of the P1 Phage?\\nAnswer: \"}], 'model': 'gpt-3.5-turbo', 'stream': False, 'temperature': 0.1}}\n",
      "Request options: {'method': 'post', 'url': '/chat/completions', 'files': None, 'json_data': {'messages': [{'role': <MessageRole.USER: 'user'>, 'content': \"Please provide an answer based solely on the provided sources. When referencing information from a source, cite the appropriate source(s) using their corresponding numbers. Every answer should include at least one source citation. Only cite a source when you are explicitly referencing it. If none of the sources are helpful, you should indicate that. For example:\\nSource 1:\\nThe sky is red in the evening and blue in the morning.\\nSource 2:\\nWater is wet when the sky is red.\\nQuery: When is water wet?\\nAnswer: Water will be wet when the sky is red [2], which occurs in the evening [1].\\nNow it's your turn. Below are several numbered sources of information:\\n------\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 1:\\nMORPHOLOGY OFP1MUTANTS 1139\\n142.\\n44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\\nandR.Hoess.1981.Site-specific recombination andits\\nroleinthelifecycleofbacteriophage P1.ColdSpring\\nHarborSymp.Quant.Biol.45:297-309.\\n45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\\nBruijn,andA.I.Bukhari. 1978.Relationships between\\ntemperate phagesMuandP1.Virology 89:146-161.\\n46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\\n1980.Invertible DNAdetermines hostspecificity ofbac-\\nteriophage Mu.Nature(London) 286:218-222.\\n47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi-\\ncalvariants ofcoliphage P1.J.Virol.5:765-782.\\n48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic\\nstudiesofcoliphage P1.I.Mapping byuseofprophage\\ndeletions. J.Virol.16:525-534.\\n49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic\\nstudiesofcoliphage P1. III.Extended genetic map.J.\\nVirol.20:177-187.\\n50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in\\ncoliphage P1affecting hostcelllysis.J.Virol.35:519-530.\\n51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.),\\nProgress inclinicalbiological research, vol.64.AlanR.\\nLiss,Inc.,NewYork.\\n52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman,\\nH.Lyle,andW.B.Wood.1970.Assembly ofbacterio-\\nphageT4tailfibers. II.Isolation andcharacterization of\\ntailfiberprecursors.\\n\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 2:\\nII.Isolation andcharacterization of\\ntailfiberprecursors. J.Mol.Biol.54:15-31.\\n53.Wood,W.B.,andJ.King.1979.Genetic control of\\ncomplex bacteriophage assembly. Compr. Virol.13:581-\\n633.\\n54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\\nbacteriophage T4.Bacteriol. Rev.40:847-868.\\n55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\\nofgeneproduct positions inbacteriophage T4byspecific\\nantibody association. J.Mol.Biol.51:411-421.\\n56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\\nE.Kellenberger. 1970.Studies onthemorphopoiesis of\\ntheheadofbacteriophage T-even. VIII.Multilayered\\npolyheads. J.Mol.Biol.50:35-58.\\n57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In\\nS.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer\\nInstitute, Bethesda, Md.VOL.45,1983\\n\\npage_label: 1\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf\\n\\nSource 3:\\nJOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\\n0021-9193/04/$08.00 /H110010 DOI: 10.1128/JB.186.21.7032–7068.2004\\nCopyright © 2004, American Society for Microbiology. All Rights Reserved.\\nGenome of Bacteriophage P1†\\nMałgorzata B. Łobocka,1* Debra J. Rose,2Guy Plunkett, III,2Marek Rusin,3Arkadiusz Samojedny,3\\nHansjo ¨rg Lehnherr,4Michael B. Yarmolinsky,5and Frederick R. Blattner2\\nDepartment of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, Warsaw,1\\nand Department of Tumor Biology, Centre of Oncology, M. Skłodowska-Curie Memorial Institute, Gliwice,3Poland;\\nLaboratory of Genetics, University of Wisconsin, Madison, Wisconsin2; Department of Genetics and Biochemistry,\\nInstitute of Microbiology, Ernst-Moritz-Arndt-University Greifswald, Greifswald, Germany4; and Laboratory\\nof Biochemistry, National Cancer Institute, National Institutes of Health, Bethesda, Maryland5\\nReceived 11 March 2004/Accepted 9 July 2004\\nP1 is a bacteriophage of Escherichia coli and other enteric bacteria. It lysogenizes its hosts as a circular,\\nlow-copy-number plasmid. We have determined the complete nucleotide sequences of two strains of a P1thermoinducible mutant, P1 c1-100. The P1 genome (93,601 bp) contains at least 117 genes, of which almost\\ntwo-thirds had not been sequenced previously and 49 have no homologs in other organisms. Protein-codinggenes occupy 92% of the genome and are organized in 45 operons, of which four are decisive for the choicebetween lysis and lysogeny. Four others ensure plasmid maintenance. The majority of the remaining 37operons are involved in lytic development.\\n\\npage_label: 1\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf\\n\\nSource 4:\\nThe majority of the remaining 37operons are involved in lytic development. Seventeen operons are transcribed from /H9268\\n70promoters directly\\ncontrolled by the master phage repressor C1. Late operons are transcribed from promoters recognized by theE. coli RNA polymerase holoenzyme in the presence of the Lpa protein, the product of a C1-controlled P1 gene.\\nThree species of P1-encoded tRNAs provide differential controls of translation, and a P1-encoded DNAmethyltransferase with putative bifunctionality inﬂuences transcription, replication, and DNA packaging. Thegenome is particularly rich in Chi recombinogenic sites. The base content and distribution in P1 DNA indicatethat replication of P1 from its plasmid origin had more impact on the base compositional asymmetries of theP1 genome than replication from the lytic origin of replication.\\nThree bacteriophages of Escherichia coli with different life\\nstyles have been particularly instrumental in the developmentof the concepts and tools of molecular biology. They are thevirulent phage T4 and the two temperate phages, /H9261and P1.\\nThe two temperate phages were ﬁrst described in close suc-cession in the middle of the last century, namely in 1951 (phageP1 [31]) and 1953 ( /H9261[185] and informally even earlier, in the\\nﬁrst issue of Evelyn Witkin’s Microbial Genetics Bulletin , 1950\\n[184a]). Whereas /H9261prophage leads an essentially passive exis-\\ntence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia\\ncolihave been reported. We present here the complete nucle-\\notide sequence of P1 and summarize its salient features.\\nP1 infects and lysogenizes E. coli and several other enteric\\nbacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath.\\n------\\nQuery: What is the morphology of the P1 Phage?\\nAnswer: \"}], 'model': 'gpt-3.5-turbo', 'stream': False, 'temperature': 0.1}}\n",
      "Request options: {'method': 'post', 'url': '/chat/completions', 'files': None, 'json_data': {'messages': [{'role': <MessageRole.USER: 'user'>, 'content': \"Please provide an answer based solely on the provided sources. When referencing information from a source, cite the appropriate source(s) using their corresponding numbers. Every answer should include at least one source citation. Only cite a source when you are explicitly referencing it. If none of the sources are helpful, you should indicate that. For example:\\nSource 1:\\nThe sky is red in the evening and blue in the morning.\\nSource 2:\\nWater is wet when the sky is red.\\nQuery: When is water wet?\\nAnswer: Water will be wet when the sky is red [2], which occurs in the evening [1].\\nNow it's your turn. Below are several numbered sources of information:\\n------\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 1:\\nMORPHOLOGY OFP1MUTANTS 1139\\n142.\\n44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\\nandR.Hoess.1981.Site-specific recombination andits\\nroleinthelifecycleofbacteriophage P1.ColdSpring\\nHarborSymp.Quant.Biol.45:297-309.\\n45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\\nBruijn,andA.I.Bukhari. 1978.Relationships between\\ntemperate phagesMuandP1.Virology 89:146-161.\\n46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\\n1980.Invertible DNAdetermines hostspecificity ofbac-\\nteriophage Mu.Nature(London) 286:218-222.\\n47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi-\\ncalvariants ofcoliphage P1.J.Virol.5:765-782.\\n48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic\\nstudiesofcoliphage P1.I.Mapping byuseofprophage\\ndeletions. J.Virol.16:525-534.\\n49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic\\nstudiesofcoliphage P1. III.Extended genetic map.J.\\nVirol.20:177-187.\\n50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in\\ncoliphage P1affecting hostcelllysis.J.Virol.35:519-530.\\n51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.),\\nProgress inclinicalbiological research, vol.64.AlanR.\\nLiss,Inc.,NewYork.\\n52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman,\\nH.Lyle,andW.B.Wood.1970.Assembly ofbacterio-\\nphageT4tailfibers. II.Isolation andcharacterization of\\ntailfiberprecursors.\\n\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 2:\\nII.Isolation andcharacterization of\\ntailfiberprecursors. J.Mol.Biol.54:15-31.\\n53.Wood,W.B.,andJ.King.1979.Genetic control of\\ncomplex bacteriophage assembly. Compr. Virol.13:581-\\n633.\\n54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\\nbacteriophage T4.Bacteriol. Rev.40:847-868.\\n55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\\nofgeneproduct positions inbacteriophage T4byspecific\\nantibody association. J.Mol.Biol.51:411-421.\\n56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\\nE.Kellenberger. 1970.Studies onthemorphopoiesis of\\ntheheadofbacteriophage T-even. VIII.Multilayered\\npolyheads. J.Mol.Biol.50:35-58.\\n57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In\\nS.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer\\nInstitute, Bethesda, Md.VOL.45,1983\\n\\npage_label: 1\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf\\n\\nSource 3:\\nJOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\\n0021-9193/04/$08.00 /H110010 DOI: 10.1128/JB.186.21.7032–7068.2004\\nCopyright © 2004, American Society for Microbiology. All Rights Reserved.\\nGenome of Bacteriophage P1†\\nMałgorzata B. Łobocka,1* Debra J. Rose,2Guy Plunkett, III,2Marek Rusin,3Arkadiusz Samojedny,3\\nHansjo ¨rg Lehnherr,4Michael B. Yarmolinsky,5and Frederick R. Blattner2\\nDepartment of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, Warsaw,1\\nand Department of Tumor Biology, Centre of Oncology, M. Skłodowska-Curie Memorial Institute, Gliwice,3Poland;\\nLaboratory of Genetics, University of Wisconsin, Madison, Wisconsin2; Department of Genetics and Biochemistry,\\nInstitute of Microbiology, Ernst-Moritz-Arndt-University Greifswald, Greifswald, Germany4; and Laboratory\\nof Biochemistry, National Cancer Institute, National Institutes of Health, Bethesda, Maryland5\\nReceived 11 March 2004/Accepted 9 July 2004\\nP1 is a bacteriophage of Escherichia coli and other enteric bacteria. It lysogenizes its hosts as a circular,\\nlow-copy-number plasmid. We have determined the complete nucleotide sequences of two strains of a P1thermoinducible mutant, P1 c1-100. The P1 genome (93,601 bp) contains at least 117 genes, of which almost\\ntwo-thirds had not been sequenced previously and 49 have no homologs in other organisms. Protein-codinggenes occupy 92% of the genome and are organized in 45 operons, of which four are decisive for the choicebetween lysis and lysogeny. Four others ensure plasmid maintenance. The majority of the remaining 37operons are involved in lytic development.\\n\\npage_label: 1\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/0343-04.pdf\\n\\nSource 4:\\nThe majority of the remaining 37operons are involved in lytic development. Seventeen operons are transcribed from /H9268\\n70promoters directly\\ncontrolled by the master phage repressor C1. Late operons are transcribed from promoters recognized by theE. coli RNA polymerase holoenzyme in the presence of the Lpa protein, the product of a C1-controlled P1 gene.\\nThree species of P1-encoded tRNAs provide differential controls of translation, and a P1-encoded DNAmethyltransferase with putative bifunctionality inﬂuences transcription, replication, and DNA packaging. Thegenome is particularly rich in Chi recombinogenic sites. The base content and distribution in P1 DNA indicatethat replication of P1 from its plasmid origin had more impact on the base compositional asymmetries of theP1 genome than replication from the lytic origin of replication.\\nThree bacteriophages of Escherichia coli with different life\\nstyles have been particularly instrumental in the developmentof the concepts and tools of molecular biology. They are thevirulent phage T4 and the two temperate phages, /H9261and P1.\\nThe two temperate phages were ﬁrst described in close suc-cession in the middle of the last century, namely in 1951 (phageP1 [31]) and 1953 ( /H9261[185] and informally even earlier, in the\\nﬁrst issue of Evelyn Witkin’s Microbial Genetics Bulletin , 1950\\n[184a]). Whereas /H9261prophage leads an essentially passive exis-\\ntence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia\\ncolihave been reported. We present here the complete nucle-\\notide sequence of P1 and summarize its salient features.\\nP1 infects and lysogenizes E. coli and several other enteric\\nbacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath.\\n------\\nQuery: What is the morphology of the P1 Phage?\\nAnswer: \"}], 'model': 'gpt-3.5-turbo', 'stream': False, 'temperature': 0.1}}\n",
      "DEBUG:httpcore.connection:close.started\n",
      "close.started\n",
      "close.started\n",
      "DEBUG:httpcore.connection:close.complete\n",
      "close.complete\n",
      "close.complete\n",
      "DEBUG:httpcore.connection:connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "DEBUG:httpcore.connection:connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde751fbe50>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde751fbe50>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde751fbe50>\n",
      "DEBUG:httpcore.connection:start_tls.started ssl_context=<ssl.SSLContext object at 0x7fde7521b770> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fde7521b770> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fde7521b770> server_hostname='api.openai.com' timeout=60.0\n",
      "DEBUG:httpcore.connection:start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde7561f7d0>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde7561f7d0>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde7561f7d0>\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:51:17 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'Cache-Control', b'no-cache, must-revalidate'), (b'openai-model', b'gpt-3.5-turbo-0613'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'2255'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'10000'), (b'x-ratelimit-limit-tokens', b'60000'), (b'x-ratelimit-limit-tokens_usage_based', b'60000'), (b'x-ratelimit-remaining-requests', b'9999'), (b'x-ratelimit-remaining-tokens', b'58277'), (b'x-ratelimit-remaining-tokens_usage_based', b'58277'), (b'x-ratelimit-reset-requests', b'8.64s'), (b'x-ratelimit-reset-tokens', b'1.723s'), (b'x-ratelimit-reset-tokens_usage_based', b'1.723s'), (b'x-request-id', b'858d98726c031f0f5750ab89e88bb239'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84125ff0481caac1-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:51:17 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'Cache-Control', b'no-cache, must-revalidate'), (b'openai-model', b'gpt-3.5-turbo-0613'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'2255'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'10000'), (b'x-ratelimit-limit-tokens', b'60000'), (b'x-ratelimit-limit-tokens_usage_based', b'60000'), (b'x-ratelimit-remaining-requests', b'9999'), (b'x-ratelimit-remaining-tokens', b'58277'), (b'x-ratelimit-remaining-tokens_usage_based', b'58277'), (b'x-ratelimit-reset-requests', b'8.64s'), (b'x-ratelimit-reset-tokens', b'1.723s'), (b'x-ratelimit-reset-tokens_usage_based', b'1.723s'), (b'x-request-id', b'858d98726c031f0f5750ab89e88bb239'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84125ff0481caac1-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 07:51:17 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'Cache-Control', b'no-cache, must-revalidate'), (b'openai-model', b'gpt-3.5-turbo-0613'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'2255'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'10000'), (b'x-ratelimit-limit-tokens', b'60000'), (b'x-ratelimit-limit-tokens_usage_based', b'60000'), (b'x-ratelimit-remaining-requests', b'9999'), (b'x-ratelimit-remaining-tokens', b'58277'), (b'x-ratelimit-remaining-tokens_usage_based', b'58277'), (b'x-ratelimit-reset-requests', b'8.64s'), (b'x-ratelimit-reset-tokens', b'1.723s'), (b'x-ratelimit-reset-tokens_usage_based', b'1.723s'), (b'x-request-id', b'858d98726c031f0f5750ab89e88bb239'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'84125ff0481caac1-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/chat/completions \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"200 OK\"\n"
     ]
    }
   ],
   "source": [
    "from llama_index.query_engine import CitationQueryEngine\n",
    "citation_query_engine = CitationQueryEngine.from_args(index)\n",
    "response = citation_query_engine.query(query_str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "<b>The morphology of the P1 phage is characterized by an icosahedral head attached to a tail that has six kinked tail fibers [4].</b>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import Markdown, display\n",
    "\n",
    "display(Markdown(f\"<b>{response}</b>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The morphology of the P1 phage is characterized by an icosahedral head attached to a tail that has six kinked tail fibers [4].\n",
      "[NodeWithScore(node=TextNode(id_='5bb7a454-99ef-4460-8596-902316659ca4', embedding=None, metadata={'page_label': '22', 'file_name': 'jvirol00150-0228.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf', 'file_type': 'application/pdf', 'file_size': 6891222, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='b652dc39-a92b-4c58-847a-dd5b4c870838', node_type=<ObjectType.DOCUMENT: '4'>, metadata={'page_label': '22', 'file_name': 'jvirol00150-0228.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf', 'file_type': 'application/pdf', 'file_size': 6891222, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, hash='68e46c57bac42cf6de73f6e56ee28fa2f35037cb74b4afcee83e15153ba4d6bb'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='b383a3b0-ad0e-4767-95e5-ae86dc4f8e29', node_type=<ObjectType.TEXT: '1'>, metadata={'page_label': '21', 'file_name': 'jvirol00150-0228.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf', 'file_type': 'application/pdf', 'file_size': 6891222, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, hash='5722cb7f573501443d58a80cd95c4507b186daa5c23d149276d357bc219d6237'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='15388096-ce59-4a01-bfe9-918c84caa0ed', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='29b3b7369936003b94996e57640e9501b6699c85861ca3dda23e84de7d073a52')}, hash='68e46c57bac42cf6de73f6e56ee28fa2f35037cb74b4afcee83e15153ba4d6bb', text='Source 1:\\nMORPHOLOGY OFP1MUTANTS 1139\\n142.\\n44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\\nandR.Hoess.1981.Site-specific recombination andits\\nroleinthelifecycleofbacteriophage P1.ColdSpring\\nHarborSymp.Quant.Biol.45:297-309.\\n45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\\nBruijn,andA.I.Bukhari. 1978.Relationships between\\ntemperate phagesMuandP1.Virology 89:146-161.\\n46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\\n1980.Invertible DNAdetermines hostspecificity ofbac-\\nteriophage Mu.Nature(London) 286:218-222.\\n47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi-\\ncalvariants ofcoliphage P1.J.Virol.5:765-782.\\n48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic\\nstudiesofcoliphage P1.I.Mapping byuseofprophage\\ndeletions. J.Virol.16:525-534.\\n49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic\\nstudiesofcoliphage P1. III.Extended genetic map.J.\\nVirol.20:177-187.\\n50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in\\ncoliphage P1affecting hostcelllysis.J.Virol.35:519-530.\\n51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.),\\nProgress inclinicalbiological research, vol.64.AlanR.\\nLiss,Inc.,NewYork.\\n52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman,\\nH.Lyle,andW.B.Wood.1970.Assembly ofbacterio-\\nphageT4tailfibers. II.Isolation andcharacterization of\\ntailfiberprecursors.\\n', start_char_idx=0, end_char_idx=1993, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=0.8932762012956971), NodeWithScore(node=TextNode(id_='5bb7a454-99ef-4460-8596-902316659ca4', embedding=None, metadata={'page_label': '22', 'file_name': 'jvirol00150-0228.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf', 'file_type': 'application/pdf', 'file_size': 6891222, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='b652dc39-a92b-4c58-847a-dd5b4c870838', node_type=<ObjectType.DOCUMENT: '4'>, metadata={'page_label': '22', 'file_name': 'jvirol00150-0228.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf', 'file_type': 'application/pdf', 'file_size': 6891222, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, hash='68e46c57bac42cf6de73f6e56ee28fa2f35037cb74b4afcee83e15153ba4d6bb'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='b383a3b0-ad0e-4767-95e5-ae86dc4f8e29', node_type=<ObjectType.TEXT: '1'>, metadata={'page_label': '21', 'file_name': 'jvirol00150-0228.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf', 'file_type': 'application/pdf', 'file_size': 6891222, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, hash='5722cb7f573501443d58a80cd95c4507b186daa5c23d149276d357bc219d6237'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='15388096-ce59-4a01-bfe9-918c84caa0ed', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='29b3b7369936003b94996e57640e9501b6699c85861ca3dda23e84de7d073a52')}, hash='68e46c57bac42cf6de73f6e56ee28fa2f35037cb74b4afcee83e15153ba4d6bb', text=\"Source 2:\\nII.Isolation andcharacterization of\\ntailfiberprecursors. J.Mol.Biol.54:15-31.\\n53.Wood,W.B.,andJ.King.1979.Genetic control of\\ncomplex bacteriophage assembly. Compr. Virol.13:581-\\n633.\\n54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\\nbacteriophage T4.Bacteriol. Rev.40:847-868.\\n55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\\nofgeneproduct positions inbacteriophage T4byspecific\\nantibody association. J.Mol.Biol.51:411-421.\\n56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\\nE.Kellenberger. 1970.Studies onthemorphopoiesis of\\ntheheadofbacteriophage T-even. VIII.Multilayered\\npolyheads. J.Mol.Biol.50:35-58.\\n57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In\\nS.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer\\nInstitute, Bethesda, Md.VOL.45,1983\\n\", start_char_idx=0, end_char_idx=1993, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=0.8932762012956971), NodeWithScore(node=TextNode(id_='af33e22f-09b3-470d-ac24-2c982451151a', embedding=None, metadata={'page_label': '1', 'file_name': '0343-04.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/0343-04.pdf', 'file_type': 'application/pdf', 'file_size': 1185513, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='58d54542-2014-44fa-b44d-72f11cc716a6', node_type=<ObjectType.DOCUMENT: '4'>, metadata={'page_label': '1', 'file_name': '0343-04.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/0343-04.pdf', 'file_type': 'application/pdf', 'file_size': 1185513, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, hash='f7f48f472908d3ae559c62d37637f21df90a8b83ac73dcec5369f0e830a211b7'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='afa4ca98-e049-40bb-bda4-bfa3330b4e11', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='9a71ba2bbc452c33235eaccf5c7eb747b4b298cd4ec6c5526bbfdcabc8551386')}, hash='7a44790cd51d9ade5faa703f201617f51a520af1a3823f41ecf4fb5a6d2fc01e', text='Source 3:\\nJOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\\n0021-9193/04/$08.00 /H110010 DOI: 10.1128/JB.186.21.7032–7068.2004\\nCopyright © 2004, American Society for Microbiology. All Rights Reserved.\\nGenome of Bacteriophage P1†\\nMałgorzata B. Łobocka,1* Debra J. Rose,2Guy Plunkett, III,2Marek Rusin,3Arkadiusz Samojedny,3\\nHansjo ¨rg Lehnherr,4Michael B. Yarmolinsky,5and Frederick R. Blattner2\\nDepartment of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the Polish Academy of Sciences, Warsaw,1\\nand Department of Tumor Biology, Centre of Oncology, M. Skłodowska-Curie Memorial Institute, Gliwice,3Poland;\\nLaboratory of Genetics, University of Wisconsin, Madison, Wisconsin2; Department of Genetics and Biochemistry,\\nInstitute of Microbiology, Ernst-Moritz-Arndt-University Greifswald, Greifswald, Germany4; and Laboratory\\nof Biochemistry, National Cancer Institute, National Institutes of Health, Bethesda, Maryland5\\nReceived 11 March 2004/Accepted 9 July 2004\\nP1 is a bacteriophage of Escherichia coli and other enteric bacteria. It lysogenizes its hosts as a circular,\\nlow-copy-number plasmid. We have determined the complete nucleotide sequences of two strains of a P1thermoinducible mutant, P1 c1-100. The P1 genome (93,601 bp) contains at least 117 genes, of which almost\\ntwo-thirds had not been sequenced previously and 49 have no homologs in other organisms. Protein-codinggenes occupy 92% of the genome and are organized in 45 operons, of which four are decisive for the choicebetween lysis and lysogeny. Four others ensure plasmid maintenance. The majority of the remaining 37operons are involved in lytic development.\\n', start_char_idx=0, end_char_idx=3562, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=0.8928214501522723), NodeWithScore(node=TextNode(id_='af33e22f-09b3-470d-ac24-2c982451151a', embedding=None, metadata={'page_label': '1', 'file_name': '0343-04.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/0343-04.pdf', 'file_type': 'application/pdf', 'file_size': 1185513, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='58d54542-2014-44fa-b44d-72f11cc716a6', node_type=<ObjectType.DOCUMENT: '4'>, metadata={'page_label': '1', 'file_name': '0343-04.pdf', 'file_path': '/workspaces/ml-learning/src/phages/data/P1/0343-04.pdf', 'file_type': 'application/pdf', 'file_size': 1185513, 'creation_date': '2024-01-05', 'last_modified_date': '2024-01-05', 'last_accessed_date': '2024-01-05'}, hash='f7f48f472908d3ae559c62d37637f21df90a8b83ac73dcec5369f0e830a211b7'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='afa4ca98-e049-40bb-bda4-bfa3330b4e11', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='9a71ba2bbc452c33235eaccf5c7eb747b4b298cd4ec6c5526bbfdcabc8551386')}, hash='7a44790cd51d9ade5faa703f201617f51a520af1a3823f41ecf4fb5a6d2fc01e', text='Source 4:\\nThe majority of the remaining 37operons are involved in lytic development. Seventeen operons are transcribed from /H9268\\n70promoters directly\\ncontrolled by the master phage repressor C1. Late operons are transcribed from promoters recognized by theE. coli RNA polymerase holoenzyme in the presence of the Lpa protein, the product of a C1-controlled P1 gene.\\nThree species of P1-encoded tRNAs provide differential controls of translation, and a P1-encoded DNAmethyltransferase with putative bifunctionality inﬂuences transcription, replication, and DNA packaging. Thegenome is particularly rich in Chi recombinogenic sites. The base content and distribution in P1 DNA indicatethat replication of P1 from its plasmid origin had more impact on the base compositional asymmetries of theP1 genome than replication from the lytic origin of replication.\\nThree bacteriophages of Escherichia coli with different life\\nstyles have been particularly instrumental in the developmentof the concepts and tools of molecular biology. They are thevirulent phage T4 and the two temperate phages, /H9261and P1.\\nThe two temperate phages were ﬁrst described in close suc-cession in the middle of the last century, namely in 1951 (phageP1 [31]) and 1953 ( /H9261[185] and informally even earlier, in the\\nﬁrst issue of Evelyn Witkin’s Microbial Genetics Bulletin , 1950\\n[184a]). Whereas /H9261prophage leads an essentially passive exis-\\ntence within the chromosome of its host, P1 prophage exists asan autonomous plasmid that is maintained at low copy num-ber. The complete genome sequences of /H9261, T4, and Escherichia\\ncolihave been reported. We present here the complete nucle-\\notide sequence of P1 and summarize its salient features.\\nP1 infects and lysogenizes E. coli and several other enteric\\nbacteria. Its virion consists of an icosahedral head attached atone vertex to a tail that bears six kinked tail ﬁbers (see coverﬁgure). As in T4, the tail consists of a tail tube and a contractilesheath.\\n', start_char_idx=0, end_char_idx=3562, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=0.8928214501522723)]\n",
      "> Source (Doc id: 5bb7a454-99ef-4460-8596-902316659ca4): Source 1:\n",
      "MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\n",
      "andR.Hoess.1981.Site-specific recombination andits\n",
      "roleinthelifecycleofbacteriophage P1.ColdSpring\n",
      "HarborSymp.Quant.Biol.45:297-309.\n",
      "45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\n",
      "Bruijn,andA.I.Bukhari. 1978.Relationships between\n",
      "temperate phagesMuandP1.Virology 89:146-161.\n",
      "46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\n",
      "1980.Invertible DNAdetermines hostspecificity ofbac-\n",
      "teriophage Mu.Nat...\n",
      "\n",
      "> Source (Doc id: 5bb7a454-99ef-4460-8596-902316659ca4): Source 2:\n",
      "II.Isolation andcharacterization of\n",
      "tailfiberprecursors. J.Mol.Biol.54:15-31.\n",
      "53.Wood,W.B.,andJ.King.1979.Genetic control of\n",
      "complex bacteriophage assembly. Compr. Virol.13:581-\n",
      "633.\n",
      "54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\n",
      "bacteriophage T4.Bacteriol. Rev.40:847-868.\n",
      "55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\n",
      "ofgeneproduct positions inbacteriophage T4byspecific\n",
      "antibody association. J.Mol.Biol.51:411-421.\n",
      "56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\n",
      "E.Kellenber...\n",
      "\n",
      "> Source (Doc id: af33e22f-09b3-470d-ac24-2c982451151a): Source 3:\n",
      "JOURNAL OF BACTERIOLOGY , Nov. 2004, p. 7032–7068 Vol. 186, No. 21\n",
      "0021-9193/04/$08.00 /H110010 DOI: 10.1128/JB.186.21.7032–7068.2004\n",
      "Copyright © 2004, American Society for Microbiology. All Rights Reserved.\n",
      "Genome of Bacteriophage P1†\n",
      "Małgorzata B. Łobocka,1* Debra J. Rose,2Guy Plunkett, III,2Marek Rusin,3Arkadiusz Samojedny,3\n",
      "Hansjo ¨rg Lehnherr,4Michael B. Yarmolinsky,5and Frederick R. Blattner2\n",
      "Department of Microbial Biochemistry, Institute of Biochemistry and Biophysics of the...\n",
      "\n",
      "> Source (Doc id: af33e22f-09b3-470d-ac24-2c982451151a): Source 4:\n",
      "The majority of the remaining 37operons are involved in lytic development. Seventeen operons are transcribed from /H9268\n",
      "70promoters directly\n",
      "controlled by the master phage repressor C1. Late operons are transcribed from promoters recognized by theE. coli RNA polymerase holoenzyme in the presence of the Lpa protein, the product of a C1-controlled P1 gene.\n",
      "Three species of P1-encoded tRNAs provide differential controls of translation, and a P1-encoded DNAmethyltransferase with putati...\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'Response' object has no attribute 'response_txt'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[25], line 4\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28mprint\u001b[39m(response\u001b[38;5;241m.\u001b[39msource_nodes)\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28mprint\u001b[39m(response\u001b[38;5;241m.\u001b[39mget_formatted_sources(length\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m500\u001b[39m))\n\u001b[0;32m----> 4\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[43mresponse\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mresponse_txt\u001b[49m)\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'Response' object has no attribute 'response_txt'"
     ]
    }
   ],
   "source": [
    "print(response.response)\n",
    "print(response.source_nodes)\n",
    "print(response.get_formatted_sources(length=500))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde74f7fd80>, 'json_data': {'input': ['You are an expert writing a scientific review paper. Please provide a summary section written in scientific language for specialist that summarize the morphology of the P1 Phage based on the information in the context. Please reference each statement with a citation. Then display a list of citations.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde74f7fd80>, 'json_data': {'input': ['You are an expert writing a scientific review paper. Please provide a summary section written in scientific language for specialist that summarize the morphology of the P1 Phage based on the information in the context. Please reference each statement with a citation. Then display a list of citations.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "Request options: {'method': 'post', 'url': '/embeddings', 'files': None, 'post_parser': <function Embeddings.create.<locals>.parser at 0x7fde74f7fd80>, 'json_data': {'input': ['You are an expert writing a scientific review paper. Please provide a summary section written in scientific language for specialist that summarize the morphology of the P1 Phage based on the information in the context. Please reference each statement with a citation. Then display a list of citations.'], 'model': <OpenAIEmbeddingModeModel.TEXT_EMBED_ADA_002: 'text-embedding-ada-002'>, 'encoding_format': 'base64'}}\n",
      "DEBUG:httpcore.connection:close.started\n",
      "close.started\n",
      "close.started\n",
      "DEBUG:httpcore.connection:close.complete\n",
      "close.complete\n",
      "close.complete\n",
      "DEBUG:httpcore.connection:connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "DEBUG:httpcore.connection:connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde757ebb10>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde757ebb10>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde757ebb10>\n",
      "DEBUG:httpcore.connection:start_tls.started ssl_context=<ssl.SSLContext object at 0x7fdebfff7a40> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fdebfff7a40> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fdebfff7a40> server_hostname='api.openai.com' timeout=60.0\n",
      "DEBUG:httpcore.connection:start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde72394150>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde72394150>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde72394150>\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 08:57:39 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'18'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'999925'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'4ms'), (b'x-request-id', b'bf5a51c724dbefb458be2a31111555b3'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412c135bee4a88c-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 08:57:39 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'18'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'999925'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'4ms'), (b'x-request-id', b'bf5a51c724dbefb458be2a31111555b3'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412c135bee4a88c-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 08:57:39 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'18'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'3000'), (b'x-ratelimit-limit-tokens', b'1000000'), (b'x-ratelimit-remaining-requests', b'2999'), (b'x-ratelimit-remaining-tokens', b'999925'), (b'x-ratelimit-reset-requests', b'20ms'), (b'x-ratelimit-reset-tokens', b'4ms'), (b'x-request-id', b'bf5a51c724dbefb458be2a31111555b3'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412c135bee4a88c-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/embeddings \"200 OK\"\n",
      "DEBUG:llama_index.indices.utils:> Top 2 nodes:\n",
      "> [Node 27aa6b8d-d5bd-4d88-9319-4db06b940e40] [Similarity score:             0.844496] Author Contributions: Conceptualization, A.B. and M.Ł.; methodology, A.B., A.C., W.I. and M.Ł.;\n",
      "v...\n",
      "> [Node 5bb7a454-99ef-4460-8596-902316659ca4] [Similarity score:             0.843907] MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\n",
      "andR.Hoess....\n",
      "> Top 2 nodes:\n",
      "> [Node 27aa6b8d-d5bd-4d88-9319-4db06b940e40] [Similarity score:             0.844496] Author Contributions: Conceptualization, A.B. and M.Ł.; methodology, A.B., A.C., W.I. and M.Ł.;\n",
      "v...\n",
      "> [Node 5bb7a454-99ef-4460-8596-902316659ca4] [Similarity score:             0.843907] MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\n",
      "andR.Hoess....\n",
      "> Top 2 nodes:\n",
      "> [Node 27aa6b8d-d5bd-4d88-9319-4db06b940e40] [Similarity score:             0.844496] Author Contributions: Conceptualization, A.B. and M.Ł.; methodology, A.B., A.C., W.I. and M.Ł.;\n",
      "v...\n",
      "> [Node 5bb7a454-99ef-4460-8596-902316659ca4] [Similarity score:             0.843907] MORPHOLOGY OFP1MUTANTS 1139\n",
      "142.\n",
      "44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\n",
      "andR.Hoess....\n",
      "DEBUG:openai._base_client:Request options: {'method': 'post', 'url': '/chat/completions', 'files': None, 'json_data': {'messages': [{'role': <MessageRole.USER: 'user'>, 'content': \"Please provide an answer based solely on the provided sources. When referencing information from a source, cite the appropriate source(s) using their corresponding numbers. Every answer should include at least one source citation. Only cite a source when you are explicitly referencing it. If none of the sources are helpful, you should indicate that. For example:\\nSource 1:\\nThe sky is red in the evening and blue in the morning.\\nSource 2:\\nWater is wet when the sky is red.\\nQuery: When is water wet?\\nAnswer: Water will be wet when the sky is red [2], which occurs in the evening [1].\\nNow it's your turn. Below are several numbered sources of information:\\n------\\npage_label: 17\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf\\n\\nSource 1:\\nAuthor Contributions: Conceptualization, A.B. and M.Ł.; methodology, A.B., A.C., W.I. and M.Ł.;\\nvalidation, A.B. and M.Ł.; formal analysis, A.B. and M.Ł.; investigation, A.B., A.C., W.I. and J.B.;\\nresources, A.B. and M.Ł.; data curation, M.Ł.; writing—original draft preparation, A.B. and M.Ł.;\\nwriting—review and editing, M.Ł.; visualization, A.B.; supervision, M.Ł.; project administration, A.B.\\nand M.Ł.; funding acquisition, A.B. and M.Ł. All authors have read and agreed to the published\\nversion of the manuscript.\\nFunding: This research was funded by the National Science Centre (PRELUDIUM; Grant No. UMO-\\n2019/35/N/NZ6/03789) and by statutory funds from the Institute of Biochemistry and Biophysics of\\nthe Polish Academy of Sciences.\\nInstitutional Review Board Statement: Not applicable.\\nInformed Consent Statement: Not applicable.\\nData Availability Statement: Not applicable.\\n\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 2:\\nMORPHOLOGY OFP1MUTANTS 1139\\n142.\\n44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\\nandR.Hoess.1981.Site-specific recombination andits\\nroleinthelifecycleofbacteriophage P1.ColdSpring\\nHarborSymp.Quant.Biol.45:297-309.\\n45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\\nBruijn,andA.I.Bukhari. 1978.Relationships between\\ntemperate phagesMuandP1.Virology 89:146-161.\\n46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\\n1980.Invertible DNAdetermines hostspecificity ofbac-\\nteriophage Mu.Nature(London) 286:218-222.\\n47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi-\\ncalvariants ofcoliphage P1.J.Virol.5:765-782.\\n48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic\\nstudiesofcoliphage P1.I.Mapping byuseofprophage\\ndeletions. J.Virol.16:525-534.\\n49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic\\nstudiesofcoliphage P1. III.Extended genetic map.J.\\nVirol.20:177-187.\\n50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in\\ncoliphage P1affecting hostcelllysis.J.Virol.35:519-530.\\n51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.),\\nProgress inclinicalbiological research, vol.64.AlanR.\\nLiss,Inc.,NewYork.\\n52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman,\\nH.Lyle,andW.B.Wood.1970.Assembly ofbacterio-\\nphageT4tailfibers. II.Isolation andcharacterization of\\ntailfiberprecursors.\\n\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 3:\\nII.Isolation andcharacterization of\\ntailfiberprecursors. J.Mol.Biol.54:15-31.\\n53.Wood,W.B.,andJ.King.1979.Genetic control of\\ncomplex bacteriophage assembly. Compr. Virol.13:581-\\n633.\\n54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\\nbacteriophage T4.Bacteriol. Rev.40:847-868.\\n55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\\nofgeneproduct positions inbacteriophage T4byspecific\\nantibody association. J.Mol.Biol.51:411-421.\\n56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\\nE.Kellenberger. 1970.Studies onthemorphopoiesis of\\ntheheadofbacteriophage T-even. VIII.Multilayered\\npolyheads. J.Mol.Biol.50:35-58.\\n57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In\\nS.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer\\nInstitute, Bethesda, Md.VOL.45,1983\\n------\\nQuery: You are an expert writing a scientific review paper. Please provide a summary section written in scientific language for specialist that summarize the morphology of the P1 Phage based on the information in the context. Please reference each statement with a citation. Then display a list of citations.\\nAnswer: \"}], 'model': 'gpt-3.5-turbo', 'stream': False, 'temperature': 0.1}}\n",
      "Request options: {'method': 'post', 'url': '/chat/completions', 'files': None, 'json_data': {'messages': [{'role': <MessageRole.USER: 'user'>, 'content': \"Please provide an answer based solely on the provided sources. When referencing information from a source, cite the appropriate source(s) using their corresponding numbers. Every answer should include at least one source citation. Only cite a source when you are explicitly referencing it. If none of the sources are helpful, you should indicate that. For example:\\nSource 1:\\nThe sky is red in the evening and blue in the morning.\\nSource 2:\\nWater is wet when the sky is red.\\nQuery: When is water wet?\\nAnswer: Water will be wet when the sky is red [2], which occurs in the evening [1].\\nNow it's your turn. Below are several numbered sources of information:\\n------\\npage_label: 17\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf\\n\\nSource 1:\\nAuthor Contributions: Conceptualization, A.B. and M.Ł.; methodology, A.B., A.C., W.I. and M.Ł.;\\nvalidation, A.B. and M.Ł.; formal analysis, A.B. and M.Ł.; investigation, A.B., A.C., W.I. and J.B.;\\nresources, A.B. and M.Ł.; data curation, M.Ł.; writing—original draft preparation, A.B. and M.Ł.;\\nwriting—review and editing, M.Ł.; visualization, A.B.; supervision, M.Ł.; project administration, A.B.\\nand M.Ł.; funding acquisition, A.B. and M.Ł. All authors have read and agreed to the published\\nversion of the manuscript.\\nFunding: This research was funded by the National Science Centre (PRELUDIUM; Grant No. UMO-\\n2019/35/N/NZ6/03789) and by statutory funds from the Institute of Biochemistry and Biophysics of\\nthe Polish Academy of Sciences.\\nInstitutional Review Board Statement: Not applicable.\\nInformed Consent Statement: Not applicable.\\nData Availability Statement: Not applicable.\\n\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 2:\\nMORPHOLOGY OFP1MUTANTS 1139\\n142.\\n44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\\nandR.Hoess.1981.Site-specific recombination andits\\nroleinthelifecycleofbacteriophage P1.ColdSpring\\nHarborSymp.Quant.Biol.45:297-309.\\n45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\\nBruijn,andA.I.Bukhari. 1978.Relationships between\\ntemperate phagesMuandP1.Virology 89:146-161.\\n46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\\n1980.Invertible DNAdetermines hostspecificity ofbac-\\nteriophage Mu.Nature(London) 286:218-222.\\n47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi-\\ncalvariants ofcoliphage P1.J.Virol.5:765-782.\\n48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic\\nstudiesofcoliphage P1.I.Mapping byuseofprophage\\ndeletions. J.Virol.16:525-534.\\n49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic\\nstudiesofcoliphage P1. III.Extended genetic map.J.\\nVirol.20:177-187.\\n50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in\\ncoliphage P1affecting hostcelllysis.J.Virol.35:519-530.\\n51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.),\\nProgress inclinicalbiological research, vol.64.AlanR.\\nLiss,Inc.,NewYork.\\n52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman,\\nH.Lyle,andW.B.Wood.1970.Assembly ofbacterio-\\nphageT4tailfibers. II.Isolation andcharacterization of\\ntailfiberprecursors.\\n\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 3:\\nII.Isolation andcharacterization of\\ntailfiberprecursors. J.Mol.Biol.54:15-31.\\n53.Wood,W.B.,andJ.King.1979.Genetic control of\\ncomplex bacteriophage assembly. Compr. Virol.13:581-\\n633.\\n54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\\nbacteriophage T4.Bacteriol. Rev.40:847-868.\\n55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\\nofgeneproduct positions inbacteriophage T4byspecific\\nantibody association. J.Mol.Biol.51:411-421.\\n56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\\nE.Kellenberger. 1970.Studies onthemorphopoiesis of\\ntheheadofbacteriophage T-even. VIII.Multilayered\\npolyheads. J.Mol.Biol.50:35-58.\\n57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In\\nS.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer\\nInstitute, Bethesda, Md.VOL.45,1983\\n------\\nQuery: You are an expert writing a scientific review paper. Please provide a summary section written in scientific language for specialist that summarize the morphology of the P1 Phage based on the information in the context. Please reference each statement with a citation. Then display a list of citations.\\nAnswer: \"}], 'model': 'gpt-3.5-turbo', 'stream': False, 'temperature': 0.1}}\n",
      "Request options: {'method': 'post', 'url': '/chat/completions', 'files': None, 'json_data': {'messages': [{'role': <MessageRole.USER: 'user'>, 'content': \"Please provide an answer based solely on the provided sources. When referencing information from a source, cite the appropriate source(s) using their corresponding numbers. Every answer should include at least one source citation. Only cite a source when you are explicitly referencing it. If none of the sources are helpful, you should indicate that. For example:\\nSource 1:\\nThe sky is red in the evening and blue in the morning.\\nSource 2:\\nWater is wet when the sky is red.\\nQuery: When is water wet?\\nAnswer: Water will be wet when the sky is red [2], which occurs in the evening [1].\\nNow it's your turn. Below are several numbered sources of information:\\n------\\npage_label: 17\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/ijms-23-04231-v4.pdf\\n\\nSource 1:\\nAuthor Contributions: Conceptualization, A.B. and M.Ł.; methodology, A.B., A.C., W.I. and M.Ł.;\\nvalidation, A.B. and M.Ł.; formal analysis, A.B. and M.Ł.; investigation, A.B., A.C., W.I. and J.B.;\\nresources, A.B. and M.Ł.; data curation, M.Ł.; writing—original draft preparation, A.B. and M.Ł.;\\nwriting—review and editing, M.Ł.; visualization, A.B.; supervision, M.Ł.; project administration, A.B.\\nand M.Ł.; funding acquisition, A.B. and M.Ł. All authors have read and agreed to the published\\nversion of the manuscript.\\nFunding: This research was funded by the National Science Centre (PRELUDIUM; Grant No. UMO-\\n2019/35/N/NZ6/03789) and by statutory funds from the Institute of Biochemistry and Biophysics of\\nthe Polish Academy of Sciences.\\nInstitutional Review Board Statement: Not applicable.\\nInformed Consent Statement: Not applicable.\\nData Availability Statement: Not applicable.\\n\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 2:\\nMORPHOLOGY OFP1MUTANTS 1139\\n142.\\n44.Sternberg, N.,D.Hamilton, S.Austin,M.Yarmolinsky,\\nandR.Hoess.1981.Site-specific recombination andits\\nroleinthelifecycleofbacteriophage P1.ColdSpring\\nHarborSymp.Quant.Biol.45:297-309.\\n45.Toussaint, A.,N.Lefebre, J.R.Scott,J.A.Cowan, F.de\\nBruijn,andA.I.Bukhari. 1978.Relationships between\\ntemperate phagesMuandP1.Virology 89:146-161.\\n46.vandePutte,P.,S.Cramer, andM.Giphart-Gassler.\\n1980.Invertible DNAdetermines hostspecificity ofbac-\\nteriophage Mu.Nature(London) 286:218-222.\\n47.Walker, D.H.,Jr.,andT.F.Anderson. 1970.Morphologi-\\ncalvariants ofcoliphage P1.J.Virol.5:765-782.\\n48.Walker, D.H.,Jr.,andJ.T.Walker. 1975.Genetic\\nstudiesofcoliphage P1.I.Mapping byuseofprophage\\ndeletions. J.Virol.16:525-534.\\n49.Walker, D.H.,Jr.,andJ.T.Walker. 1976.Genetic\\nstudiesofcoliphage P1. III.Extended genetic map.J.\\nVirol.20:177-187.\\n50.Walker, J.T.,andD.H.Walker, Jr.1980.Mutations in\\ncoliphage P1affecting hostcelllysis.J.Virol.35:519-530.\\n51.Walker, J.T.,andD.H.Walker, Jr.1981.Structuralproteins ofcoliphage P1,p.69-77.InM.S.DuBow(ed.),\\nProgress inclinicalbiological research, vol.64.AlanR.\\nLiss,Inc.,NewYork.\\n52.Ward,S.,R.B.Luftig,J.H.Wilson, H.Eddleman,\\nH.Lyle,andW.B.Wood.1970.Assembly ofbacterio-\\nphageT4tailfibers. II.Isolation andcharacterization of\\ntailfiberprecursors.\\n\\npage_label: 22\\nfile_path: /workspaces/ml-learning/src/phages/data/P1/jvirol00150-0228.pdf\\n\\nSource 3:\\nII.Isolation andcharacterization of\\ntailfiberprecursors. J.Mol.Biol.54:15-31.\\n53.Wood,W.B.,andJ.King.1979.Genetic control of\\ncomplex bacteriophage assembly. Compr. Virol.13:581-\\n633.\\n54.Wood,W.B.,andH.R.Revel.1976.Thegenome of\\nbacteriophage T4.Bacteriol. Rev.40:847-868.\\n55.Yanagida, M.,andC.Ahmad-Zadeh. 1970.Determination\\nofgeneproduct positions inbacteriophage T4byspecific\\nantibody association. J.Mol.Biol.51:411-421.\\n56.Yanagida, M.,E.BoydelaTour,C.Alff-Steinberger, and\\nE.Kellenberger. 1970.Studies onthemorphopoiesis of\\ntheheadofbacteriophage T-even. VIII.Multilayered\\npolyheads. J.Mol.Biol.50:35-58.\\n57.Yarmolinsky, M.1982.Bacteriophage P1,p.34-43.In\\nS.J.O'Brien (ed.),Genetic maps,vol.2.National Cancer\\nInstitute, Bethesda, Md.VOL.45,1983\\n------\\nQuery: You are an expert writing a scientific review paper. Please provide a summary section written in scientific language for specialist that summarize the morphology of the P1 Phage based on the information in the context. Please reference each statement with a citation. Then display a list of citations.\\nAnswer: \"}], 'model': 'gpt-3.5-turbo', 'stream': False, 'temperature': 0.1}}\n",
      "DEBUG:httpcore.connection:close.started\n",
      "close.started\n",
      "close.started\n",
      "DEBUG:httpcore.connection:close.complete\n",
      "close.complete\n",
      "close.complete\n",
      "DEBUG:httpcore.connection:connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "connect_tcp.started host='api.openai.com' port=443 local_address=None timeout=60.0 socket_options=None\n",
      "DEBUG:httpcore.connection:connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde764276d0>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde764276d0>\n",
      "connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde764276d0>\n",
      "DEBUG:httpcore.connection:start_tls.started ssl_context=<ssl.SSLContext object at 0x7fde7521b770> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fde7521b770> server_hostname='api.openai.com' timeout=60.0\n",
      "start_tls.started ssl_context=<ssl.SSLContext object at 0x7fde7521b770> server_hostname='api.openai.com' timeout=60.0\n",
      "DEBUG:httpcore.connection:start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde7238f690>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde7238f690>\n",
      "start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x7fde7238f690>\n",
      "DEBUG:httpcore.http11:send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "send_request_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "send_request_headers.complete\n",
      "DEBUG:httpcore.http11:send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "send_request_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:send_request_body.complete\n",
      "send_request_body.complete\n",
      "send_request_body.complete\n",
      "DEBUG:httpcore.http11:receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "receive_response_headers.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 08:58:20 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'Cache-Control', b'no-cache, must-revalidate'), (b'openai-model', b'gpt-3.5-turbo-0613'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'40215'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'10000'), (b'x-ratelimit-limit-tokens', b'60000'), (b'x-ratelimit-limit-tokens_usage_based', b'60000'), (b'x-ratelimit-remaining-requests', b'9999'), (b'x-ratelimit-remaining-tokens', b'58922'), (b'x-ratelimit-remaining-tokens_usage_based', b'58922'), (b'x-ratelimit-reset-requests', b'8.64s'), (b'x-ratelimit-reset-tokens', b'1.078s'), (b'x-ratelimit-reset-tokens_usage_based', b'1.078s'), (b'x-request-id', b'91bf275c9c10d0f8119013a3fe17548b'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412c1382ec2a86b-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 08:58:20 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'Cache-Control', b'no-cache, must-revalidate'), (b'openai-model', b'gpt-3.5-turbo-0613'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'40215'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'10000'), (b'x-ratelimit-limit-tokens', b'60000'), (b'x-ratelimit-limit-tokens_usage_based', b'60000'), (b'x-ratelimit-remaining-requests', b'9999'), (b'x-ratelimit-remaining-tokens', b'58922'), (b'x-ratelimit-remaining-tokens_usage_based', b'58922'), (b'x-ratelimit-reset-requests', b'8.64s'), (b'x-ratelimit-reset-tokens', b'1.078s'), (b'x-ratelimit-reset-tokens_usage_based', b'1.078s'), (b'x-request-id', b'91bf275c9c10d0f8119013a3fe17548b'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412c1382ec2a86b-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Sat, 06 Jan 2024 08:58:20 GMT'), (b'Content-Type', b'application/json'), (b'Transfer-Encoding', b'chunked'), (b'Connection', b'keep-alive'), (b'access-control-allow-origin', b'*'), (b'Cache-Control', b'no-cache, must-revalidate'), (b'openai-model', b'gpt-3.5-turbo-0613'), (b'openai-organization', b'user-vyzy3sybphrf2xalgufcdzur'), (b'openai-processing-ms', b'40215'), (b'openai-version', b'2020-10-01'), (b'strict-transport-security', b'max-age=15724800; includeSubDomains'), (b'x-ratelimit-limit-requests', b'10000'), (b'x-ratelimit-limit-tokens', b'60000'), (b'x-ratelimit-limit-tokens_usage_based', b'60000'), (b'x-ratelimit-remaining-requests', b'9999'), (b'x-ratelimit-remaining-tokens', b'58922'), (b'x-ratelimit-remaining-tokens_usage_based', b'58922'), (b'x-ratelimit-reset-requests', b'8.64s'), (b'x-ratelimit-reset-tokens', b'1.078s'), (b'x-ratelimit-reset-tokens_usage_based', b'1.078s'), (b'x-request-id', b'91bf275c9c10d0f8119013a3fe17548b'), (b'CF-Cache-Status', b'DYNAMIC'), (b'Server', b'cloudflare'), (b'CF-RAY', b'8412c1382ec2a86b-SYD'), (b'Content-Encoding', b'gzip'), (b'alt-svc', b'h3=\":443\"; ma=86400')])\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "DEBUG:httpcore.http11:receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "receive_response_body.started request=<Request [b'POST']>\n",
      "DEBUG:httpcore.http11:receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "receive_response_body.complete\n",
      "DEBUG:httpcore.http11:response_closed.started\n",
      "response_closed.started\n",
      "response_closed.started\n",
      "DEBUG:httpcore.http11:response_closed.complete\n",
      "response_closed.complete\n",
      "response_closed.complete\n",
      "DEBUG:openai._base_client:HTTP Request: POST https://api.openai.com/v1/chat/completions \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"200 OK\"\n"
     ]
    }
   ],
   "source": [
    "response = citation_query_engine.query(\"You are an expert writing a scientific review paper. Please provide a summary section written in scientific language for specialist that summarize the morphology of the P1 Phage based on the information in the context. Please reference each statement with a citation. Then display a list of citations.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "<b>The morphology of the P1 phage has been extensively studied [2][47][48][49][50]. It has been observed that the P1 phage exhibits morphological variants [47]. The assembly of the P1 phage tail fibers has been investigated, and tail fiber precursors have been isolated and characterized [52][53]. The genetic control of complex bacteriophage assembly has also been explored [53]. The genome of the P1 phage has been analyzed, providing insights into its structure and organization [54]. Additionally, specific antibody association has been used to determine the positions of gene products in the P1 phage [55]. These studies have contributed to our understanding of the morphology and genetic characteristics of the P1 phage.\n",
       "\n",
       "List of Citations:\n",
       "1. Source 2: Sternberg, N., Hamilton, D., Austin, S., Yarmolinsky, M., & Hoess, R. (1981). Site-specific recombination and its role in the lifecycle of bacteriophage P1. Cold Spring Harbor Symp Quant Biol, 45, 297-309.\n",
       "2. Source 2: Toussaint, A., Lefebre, N., Scott, J. R., Cowan, J. A., Bruijn, F. D., & Bukhari, A. I. (1978). Relationships between temperate phages Mu and P1. Virology, 89, 146-161.\n",
       "3. Source 2: vandePutte, P., Cramer, S., & Giphart-Gassler, M. (1980). Invertible DNA determines host specificity of bacteriophage Mu. Nature (London), 286, 218-222.\n",
       "4. Source 2: Walker, D. H., Jr., & Anderson, T. F. (1970). Morphological variants of coliphage P1. J. Virol., 5, 765-782.\n",
       "5. Source 2: Walker, D. H., Jr., & Walker, J. T. (1975). Genetic studies of coliphage P1. I. Mapping by use of prophage deletions. J. Virol., 16, 525-534.\n",
       "6. Source 2: Walker, D. H., Jr., & Walker, J. T. (1976). Genetic studies of coliphage P1. III. Extended genetic map. J. Virol., 20, 177-187.\n",
       "7. Source 2: Walker, J. T., & Walker, D. H., Jr. (1980). Mutations in coliphage P1 affecting host cell lysis. J. Virol., 35, 519-530.\n",
       "8. Source 2: Walker, J. T., & Walker, D. H., Jr. (1981). Structural proteins of coliphage P1. In M. S. DuBow (Ed.), Progress in clinical biological research, vol. 64 (pp. 69-77). Alan R. Liss, Inc., New York.\n",
       "9. Source 3: Wood, W. B., & King, J. (1979). Genetic control of complex bacteriophage assembly. Compr. Virol., 13, 581-633.\n",
       "10. Source 3: Wood, W. B., & Revel, H. R. (1976). The genome of bacteriophage T4. Bacteriol. Rev., 40, 847-868.\n",
       "11. Source 3: Yanagida, M., & Ahmad-Zadeh, C. (1970). Determination of gene product positions in bacteriophage T4 by specific antibody association. J. Mol. Biol., 51, 411-421.\n",
       "12. Source 3: Yanagida, M., BoydelaTour, E., Alff-Steinberger, C., & Kellenberger, E. (1970). Studies on the morphopoiesis of the head of bacteriophage T-even. VIII. Multilayered polyheads. J. Mol. Biol., 50, 35-58.\n",
       "13. Source 3: Yarmolinsky, M. (1982). Bacteriophage P1. In S. J. O'Brien (Ed.), Genetic maps, vol. 2 (pp. 34-43). National Cancer Institute, Bethesda, Md.</b>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(Markdown(f\"<b>{response}</b>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/workspaces/ml-learning/.venv/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n",
      "Parsing nodes: 100%|██████████| 229/229 [00:01<00:00, 220.35it/s]\n",
      "Summarizing documents:   0%|          | 0/229 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: f5f7f7fa-761a-4751-a481-63179be9aaef\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   0%|          | 1/229 [00:15<57:17, 15.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: dc37410d-bede-4727-8cc6-36942300a33c\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   1%|          | 2/229 [00:25<47:45, 12.63s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: b2644f5f-d02f-4899-be93-7457488ab439\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   1%|▏         | 3/229 [00:40<51:31, 13.68s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 9b47fa1b-15b6-446b-a6fa-4d2ecc57e4ea\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   2%|▏         | 4/229 [00:50<44:42, 11.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 02660718-a881-45b6-8d48-844cf6ec7382\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   2%|▏         | 5/229 [01:05<48:30, 12.99s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 4c25e007-75c9-40aa-b16c-720240d54b90\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   3%|▎         | 6/229 [01:17<47:10, 12.69s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: a23fb034-0c4c-4288-a075-7d4ae5c7ce68\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   3%|▎         | 7/229 [01:25<41:27, 11.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: f9aaa7f1-5558-45a2-abb7-ec461a6e2c99\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   3%|▎         | 8/229 [01:35<39:34, 10.74s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 8680b033-3e59-44ba-8a01-1cbca05631d0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   4%|▍         | 9/229 [01:45<38:46, 10.57s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: dc91d056-bfac-4ee3-af55-1c75dc57c0e4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   4%|▍         | 10/229 [01:55<37:54, 10.38s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 6e98efa4-1c09-44c6-885c-cb95964914ec\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   5%|▍         | 11/229 [02:00<32:08,  8.85s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 887a78fa-d724-460d-bf5a-a952ee01ff8b\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   5%|▌         | 12/229 [02:09<32:29,  8.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 4555da61-fcfe-4bd4-b1aa-3265d2ec4d41\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   6%|▌         | 13/229 [02:22<35:47,  9.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 22f1f8be-d3a5-4b31-9746-0e4519b36831\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   6%|▌         | 14/229 [02:32<36:06, 10.07s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 2d1ddcec-678d-43ea-9444-449fc75abbff\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   7%|▋         | 15/229 [02:42<36:17, 10.17s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 5aae7000-95bf-4bee-8b93-a5e080d30e52\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   7%|▋         | 16/229 [02:57<41:17, 11.63s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: b9797479-6ab2-4591-95d4-1b2b8aa1aadf\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   7%|▋         | 17/229 [03:09<41:07, 11.64s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 45e8027e-9cdf-4013-90fe-e5a1f61231d8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   8%|▊         | 18/229 [03:20<39:47, 11.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 65ec8a02-66c6-484d-b3c8-f73c64f7cf1c\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   8%|▊         | 19/229 [03:30<38:24, 10.97s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: b9c30ea2-fd22-4836-8921-3559514579f2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   9%|▊         | 20/229 [03:41<38:24, 11.02s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 25117a55-9d0e-48eb-8944-eda5eb2ae9bb\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:   9%|▉         | 21/229 [03:53<39:10, 11.30s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: f709af34-5d78-4fec-a295-7b4a54f53d0f\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  10%|▉         | 22/229 [04:04<39:20, 11.40s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 4779d236-6f56-430c-bafe-052c4d53a079\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  10%|█         | 23/229 [04:19<42:50, 12.48s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: c436b6a9-c29b-4b44-9747-36b022c829d0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  10%|█         | 24/229 [04:31<41:29, 12.15s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: a71776ef-f8de-4dbd-90aa-44a1ca5f8dcb\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  11%|█         | 25/229 [04:41<39:31, 11.62s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: bc2a835a-2630-4b39-8bfa-f17b32acb778\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  11%|█▏        | 26/229 [04:49<35:41, 10.55s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 613abbb1-7b57-4756-9b8a-3fc2866ad3cd\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  12%|█▏        | 27/229 [04:58<33:55, 10.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 7ea12e81-832b-48b1-b243-b2b596ca2079\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  12%|█▏        | 28/229 [05:12<37:16, 11.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: b47418d4-0fcf-47ea-b86b-91bec2022213\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  13%|█▎        | 29/229 [05:26<40:07, 12.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 1e73c5f9-a298-4030-9a1a-67c6519a00d5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  13%|█▎        | 30/229 [05:39<40:30, 12.21s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 266717ff-4f23-4caa-9455-2e812cd6c74c\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  14%|█▎        | 31/229 [05:49<38:06, 11.55s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 9d267821-3e56-4f3c-b291-e75d5b5a6c3a\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  14%|█▍        | 32/229 [06:26<1:02:59, 19.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: b778f2f8-3481-42b2-a8f5-b9a31cc333ce\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  14%|█▍        | 33/229 [07:04<1:21:16, 24.88s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 079d9847-6a1a-4dff-ad51-78422c8a9d3d\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  15%|█▍        | 34/229 [07:38<1:29:54, 27.66s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 77f81bc8-7741-41c1-b795-d061ffc5f136\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  15%|█▌        | 35/229 [08:17<1:40:09, 30.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 708590ea-daef-4561-8a67-3e57d3f5f0ea\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  16%|█▌        | 36/229 [08:45<1:37:22, 30.27s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current doc id: 7936a034-66de-4f84-8180-0bb8af233c68\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summarizing documents:  16%|█▌        | 36/229 [09:06<48:50, 15.18s/it]  \n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[44], line 5\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m# default mode of building the index\u001b[39;00m\n\u001b[1;32m      2\u001b[0m response_synthesizer \u001b[38;5;241m=\u001b[39m get_response_synthesizer(\n\u001b[1;32m      3\u001b[0m     response_mode\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtree_summarize\u001b[39m\u001b[38;5;124m\"\u001b[39m, use_async\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[1;32m      4\u001b[0m )\n\u001b[0;32m----> 5\u001b[0m doc_summary_index \u001b[38;5;241m=\u001b[39m \u001b[43mDocumentSummaryIndex\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfrom_documents\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m      6\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdocuments\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m      7\u001b[0m \u001b[43m    \u001b[49m\u001b[43mservice_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mservice_context\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m      8\u001b[0m \u001b[43m    \u001b[49m\u001b[43mresponse_synthesizer\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresponse_synthesizer\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m      9\u001b[0m \u001b[43m    \u001b[49m\u001b[43mshow_progress\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m     10\u001b[0m \u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/indices/base.py:106\u001b[0m, in \u001b[0;36mBaseIndex.from_documents\u001b[0;34m(cls, documents, storage_context, service_context, show_progress, **kwargs)\u001b[0m\n\u001b[1;32m     97\u001b[0m     docstore\u001b[38;5;241m.\u001b[39mset_document_hash(doc\u001b[38;5;241m.\u001b[39mget_doc_id(), doc\u001b[38;5;241m.\u001b[39mhash)\n\u001b[1;32m     99\u001b[0m nodes \u001b[38;5;241m=\u001b[39m run_transformations(\n\u001b[1;32m    100\u001b[0m     documents,  \u001b[38;5;66;03m# type: ignore\u001b[39;00m\n\u001b[1;32m    101\u001b[0m     service_context\u001b[38;5;241m.\u001b[39mtransformations,\n\u001b[1;32m    102\u001b[0m     show_progress\u001b[38;5;241m=\u001b[39mshow_progress,\n\u001b[1;32m    103\u001b[0m     \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs,\n\u001b[1;32m    104\u001b[0m )\n\u001b[0;32m--> 106\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mcls\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[1;32m    107\u001b[0m \u001b[43m    \u001b[49m\u001b[43mnodes\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnodes\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    108\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstorage_context\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    109\u001b[0m \u001b[43m    \u001b[49m\u001b[43mservice_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mservice_context\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    110\u001b[0m \u001b[43m    \u001b[49m\u001b[43mshow_progress\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mshow_progress\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    111\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    112\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/indices/document_summary/base.py:89\u001b[0m, in \u001b[0;36mDocumentSummaryIndex.__init__\u001b[0;34m(self, nodes, index_struct, service_context, storage_context, response_synthesizer, summary_query, show_progress, embed_summaries, **kwargs)\u001b[0m\n\u001b[1;32m     86\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_summary_query \u001b[38;5;241m=\u001b[39m summary_query\n\u001b[1;32m     87\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_embed_summaries \u001b[38;5;241m=\u001b[39m embed_summaries\n\u001b[0;32m---> 89\u001b[0m \u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__init__\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[1;32m     90\u001b[0m \u001b[43m    \u001b[49m\u001b[43mnodes\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnodes\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     91\u001b[0m \u001b[43m    \u001b[49m\u001b[43mindex_struct\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mindex_struct\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     92\u001b[0m \u001b[43m    \u001b[49m\u001b[43mservice_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mservice_context\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     93\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstorage_context\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     94\u001b[0m \u001b[43m    \u001b[49m\u001b[43mshow_progress\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mshow_progress\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     95\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     96\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/indices/base.py:71\u001b[0m, in \u001b[0;36mBaseIndex.__init__\u001b[0;34m(self, nodes, index_struct, storage_context, service_context, show_progress, **kwargs)\u001b[0m\n\u001b[1;32m     69\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m index_struct \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m     70\u001b[0m     \u001b[38;5;28;01massert\u001b[39;00m nodes \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m---> 71\u001b[0m     index_struct \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbuild_index_from_nodes\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnodes\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     72\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_index_struct \u001b[38;5;241m=\u001b[39m index_struct\n\u001b[1;32m     73\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_storage_context\u001b[38;5;241m.\u001b[39mindex_store\u001b[38;5;241m.\u001b[39madd_index_struct(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_index_struct)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/indices/base.py:175\u001b[0m, in \u001b[0;36mBaseIndex.build_index_from_nodes\u001b[0;34m(self, nodes)\u001b[0m\n\u001b[1;32m    173\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Build the index from nodes.\"\"\"\u001b[39;00m\n\u001b[1;32m    174\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_docstore\u001b[38;5;241m.\u001b[39madd_documents(nodes, allow_update\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m--> 175\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_build_index_from_nodes\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnodes\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/indices/document_summary/base.py:215\u001b[0m, in \u001b[0;36mDocumentSummaryIndex._build_index_from_nodes\u001b[0;34m(self, nodes)\u001b[0m\n\u001b[1;32m    212\u001b[0m \u001b[38;5;66;03m# first get doc_id to nodes_dict, generate a summary for each doc_id,\u001b[39;00m\n\u001b[1;32m    213\u001b[0m \u001b[38;5;66;03m# then build the index struct\u001b[39;00m\n\u001b[1;32m    214\u001b[0m index_struct \u001b[38;5;241m=\u001b[39m IndexDocumentSummary()\n\u001b[0;32m--> 215\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_add_nodes_to_index\u001b[49m\u001b[43m(\u001b[49m\u001b[43mindex_struct\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnodes\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_show_progress\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    216\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m index_struct\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/indices/document_summary/base.py:174\u001b[0m, in \u001b[0;36mDocumentSummaryIndex._add_nodes_to_index\u001b[0;34m(self, index_struct, nodes, show_progress)\u001b[0m\n\u001b[1;32m    172\u001b[0m nodes_with_scores \u001b[38;5;241m=\u001b[39m [NodeWithScore(node\u001b[38;5;241m=\u001b[39mn) \u001b[38;5;28;01mfor\u001b[39;00m n \u001b[38;5;129;01min\u001b[39;00m nodes]\n\u001b[1;32m    173\u001b[0m \u001b[38;5;66;03m# get the summary for each doc_id\u001b[39;00m\n\u001b[0;32m--> 174\u001b[0m summary_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_response_synthesizer\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msynthesize\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    175\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquery\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_summary_query\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    176\u001b[0m \u001b[43m    \u001b[49m\u001b[43mnodes\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnodes_with_scores\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    177\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    178\u001b[0m summary_response \u001b[38;5;241m=\u001b[39m cast(Response, summary_response)\n\u001b[1;32m    179\u001b[0m summary_node_dict[doc_id] \u001b[38;5;241m=\u001b[39m TextNode(\n\u001b[1;32m    180\u001b[0m     text\u001b[38;5;241m=\u001b[39msummary_response\u001b[38;5;241m.\u001b[39mresponse,\n\u001b[1;32m    181\u001b[0m     relationships\u001b[38;5;241m=\u001b[39m{\n\u001b[1;32m    182\u001b[0m         NodeRelationship\u001b[38;5;241m.\u001b[39mSOURCE: RelatedNodeInfo(node_id\u001b[38;5;241m=\u001b[39mdoc_id)\n\u001b[1;32m    183\u001b[0m     },\n\u001b[1;32m    184\u001b[0m )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/response_synthesizers/base.py:146\u001b[0m, in \u001b[0;36mBaseSynthesizer.synthesize\u001b[0;34m(self, query, nodes, additional_source_nodes, **response_kwargs)\u001b[0m\n\u001b[1;32m    141\u001b[0m     query \u001b[38;5;241m=\u001b[39m QueryBundle(query_str\u001b[38;5;241m=\u001b[39mquery)\n\u001b[1;32m    143\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_callback_manager\u001b[38;5;241m.\u001b[39mevent(\n\u001b[1;32m    144\u001b[0m     CBEventType\u001b[38;5;241m.\u001b[39mSYNTHESIZE, payload\u001b[38;5;241m=\u001b[39m{EventPayload\u001b[38;5;241m.\u001b[39mQUERY_STR: query\u001b[38;5;241m.\u001b[39mquery_str}\n\u001b[1;32m    145\u001b[0m ) \u001b[38;5;28;01mas\u001b[39;00m event:\n\u001b[0;32m--> 146\u001b[0m     response_str \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_response\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    147\u001b[0m \u001b[43m        \u001b[49m\u001b[43mquery_str\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mquery_str\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    148\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtext_chunks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m[\u001b[49m\n\u001b[1;32m    149\u001b[0m \u001b[43m            \u001b[49m\u001b[43mn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mnode\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_content\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmetadata_mode\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mMetadataMode\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mLLM\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mn\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mnodes\u001b[49m\n\u001b[1;32m    150\u001b[0m \u001b[43m        \u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    151\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    152\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    154\u001b[0m     additional_source_nodes \u001b[38;5;241m=\u001b[39m additional_source_nodes \u001b[38;5;129;01mor\u001b[39;00m []\n\u001b[1;32m    155\u001b[0m     source_nodes \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(nodes) \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mlist\u001b[39m(additional_source_nodes)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/response_synthesizers/tree_summarize.py:221\u001b[0m, in \u001b[0;36mTreeSummarize.get_response\u001b[0;34m(self, query_str, text_chunks, **response_kwargs)\u001b[0m\n\u001b[1;32m    218\u001b[0m         summaries \u001b[38;5;241m=\u001b[39m [summary\u001b[38;5;241m.\u001b[39mjson() \u001b[38;5;28;01mfor\u001b[39;00m summary \u001b[38;5;129;01min\u001b[39;00m summaries]\n\u001b[1;32m    220\u001b[0m \u001b[38;5;66;03m# recursively summarize the summaries\u001b[39;00m\n\u001b[0;32m--> 221\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_response\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    222\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquery_str\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery_str\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtext_chunks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msummaries\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kwargs\u001b[49m\n\u001b[1;32m    223\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/response_synthesizers/tree_summarize.py:154\u001b[0m, in \u001b[0;36mTreeSummarize.get_response\u001b[0;34m(self, query_str, text_chunks, **response_kwargs)\u001b[0m\n\u001b[1;32m    152\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    153\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_output_cls \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 154\u001b[0m         response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_service_context\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mllm\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpredict\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    155\u001b[0m \u001b[43m            \u001b[49m\u001b[43msummary_template\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    156\u001b[0m \u001b[43m            \u001b[49m\u001b[43mcontext_str\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtext_chunks\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    157\u001b[0m \u001b[43m            \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    158\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    159\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    160\u001b[0m         response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_service_context\u001b[38;5;241m.\u001b[39mllm\u001b[38;5;241m.\u001b[39mstructured_predict(\n\u001b[1;32m    161\u001b[0m             \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_output_cls,\n\u001b[1;32m    162\u001b[0m             summary_template,\n\u001b[1;32m    163\u001b[0m             context_str\u001b[38;5;241m=\u001b[39mtext_chunks[\u001b[38;5;241m0\u001b[39m],\n\u001b[1;32m    164\u001b[0m             \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mresponse_kwargs,\n\u001b[1;32m    165\u001b[0m         )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/llm.py:220\u001b[0m, in \u001b[0;36mLLM.predict\u001b[0;34m(self, prompt, **prompt_args)\u001b[0m\n\u001b[1;32m    218\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmetadata\u001b[38;5;241m.\u001b[39mis_chat_model:\n\u001b[1;32m    219\u001b[0m     messages \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_messages(prompt, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mprompt_args)\n\u001b[0;32m--> 220\u001b[0m     chat_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mchat\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmessages\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    221\u001b[0m     output \u001b[38;5;241m=\u001b[39m chat_response\u001b[38;5;241m.\u001b[39mmessage\u001b[38;5;241m.\u001b[39mcontent \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    222\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/base.py:97\u001b[0m, in \u001b[0;36mllm_chat_callback.<locals>.wrap.<locals>.wrapped_llm_chat\u001b[0;34m(_self, messages, **kwargs)\u001b[0m\n\u001b[1;32m     88\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m wrapper_logic(_self) \u001b[38;5;28;01mas\u001b[39;00m callback_manager:\n\u001b[1;32m     89\u001b[0m     event_id \u001b[38;5;241m=\u001b[39m callback_manager\u001b[38;5;241m.\u001b[39mon_event_start(\n\u001b[1;32m     90\u001b[0m         CBEventType\u001b[38;5;241m.\u001b[39mLLM,\n\u001b[1;32m     91\u001b[0m         payload\u001b[38;5;241m=\u001b[39m{\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     95\u001b[0m         },\n\u001b[1;32m     96\u001b[0m     )\n\u001b[0;32m---> 97\u001b[0m     f_return_val \u001b[38;5;241m=\u001b[39m \u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[43m_self\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmessages\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     99\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(f_return_val, Generator):\n\u001b[1;32m    100\u001b[0m         \u001b[38;5;66;03m# intercept the generator and add a callback to the end\u001b[39;00m\n\u001b[1;32m    101\u001b[0m         \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrapped_gen\u001b[39m() \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m ChatResponseGen:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/openai.py:234\u001b[0m, in \u001b[0;36mOpenAI.chat\u001b[0;34m(self, messages, **kwargs)\u001b[0m\n\u001b[1;32m    232\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    233\u001b[0m     chat_fn \u001b[38;5;241m=\u001b[39m completion_to_chat_decorator(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_complete)\n\u001b[0;32m--> 234\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mchat_fn\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmessages\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/openai.py:289\u001b[0m, in \u001b[0;36mOpenAI._chat\u001b[0;34m(self, messages, **kwargs)\u001b[0m\n\u001b[1;32m    287\u001b[0m client \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_client()\n\u001b[1;32m    288\u001b[0m message_dicts \u001b[38;5;241m=\u001b[39m to_openai_message_dicts(messages)\n\u001b[0;32m--> 289\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[43mclient\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mchat\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcompletions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcreate\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    290\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmessages\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmessage_dicts\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    291\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    292\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_model_kwargs\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    293\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    294\u001b[0m openai_message \u001b[38;5;241m=\u001b[39m response\u001b[38;5;241m.\u001b[39mchoices[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mmessage\n\u001b[1;32m    295\u001b[0m message \u001b[38;5;241m=\u001b[39m from_openai_message(openai_message)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/openai/_utils/_utils.py:272\u001b[0m, in \u001b[0;36mrequired_args.<locals>.inner.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    270\u001b[0m             msg \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mMissing required argument: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mquote(missing[\u001b[38;5;241m0\u001b[39m])\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    271\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m(msg)\n\u001b[0;32m--> 272\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/openai/resources/chat/completions.py:645\u001b[0m, in \u001b[0;36mCompletions.create\u001b[0;34m(self, messages, model, frequency_penalty, function_call, functions, logit_bias, logprobs, max_tokens, n, presence_penalty, response_format, seed, stop, stream, temperature, tool_choice, tools, top_logprobs, top_p, user, extra_headers, extra_query, extra_body, timeout)\u001b[0m\n\u001b[1;32m    596\u001b[0m \u001b[38;5;129m@required_args\u001b[39m([\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmessages\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmodel\u001b[39m\u001b[38;5;124m\"\u001b[39m], [\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmessages\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmodel\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mstream\u001b[39m\u001b[38;5;124m\"\u001b[39m])\n\u001b[1;32m    597\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mcreate\u001b[39m(\n\u001b[1;32m    598\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    643\u001b[0m     timeout: \u001b[38;5;28mfloat\u001b[39m \u001b[38;5;241m|\u001b[39m httpx\u001b[38;5;241m.\u001b[39mTimeout \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m|\u001b[39m NotGiven \u001b[38;5;241m=\u001b[39m NOT_GIVEN,\n\u001b[1;32m    644\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m ChatCompletion \u001b[38;5;241m|\u001b[39m Stream[ChatCompletionChunk]:\n\u001b[0;32m--> 645\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_post\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    646\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m/chat/completions\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    647\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmaybe_transform\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    648\u001b[0m \u001b[43m            \u001b[49m\u001b[43m{\u001b[49m\n\u001b[1;32m    649\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmessages\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mmessages\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    650\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmodel\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mmodel\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    651\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mfrequency_penalty\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mfrequency_penalty\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    652\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mfunction_call\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mfunction_call\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    653\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mfunctions\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mfunctions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    654\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mlogit_bias\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mlogit_bias\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    655\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mlogprobs\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mlogprobs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    656\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmax_tokens\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mmax_tokens\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    657\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mn\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    658\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mpresence_penalty\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mpresence_penalty\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    659\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mresponse_format\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mresponse_format\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    660\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mseed\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mseed\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    661\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstop\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mstop\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    662\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstream\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mstream\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    663\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtemperature\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mtemperature\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    664\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtool_choice\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mtool_choice\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    665\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtools\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mtools\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    666\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtop_logprobs\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mtop_logprobs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    667\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtop_p\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mtop_p\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    668\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43muser\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43muser\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    669\u001b[0m \u001b[43m            \u001b[49m\u001b[43m}\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    670\u001b[0m \u001b[43m            \u001b[49m\u001b[43mcompletion_create_params\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mCompletionCreateParams\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    671\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    672\u001b[0m \u001b[43m        \u001b[49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmake_request_options\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    673\u001b[0m \u001b[43m            \u001b[49m\u001b[43mextra_headers\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mextra_headers\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mextra_query\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mextra_query\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mextra_body\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mextra_body\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\n\u001b[1;32m    674\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    675\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcast_to\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mChatCompletion\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    676\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    677\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream_cls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mStream\u001b[49m\u001b[43m[\u001b[49m\u001b[43mChatCompletionChunk\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    678\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/openai/_base_client.py:1088\u001b[0m, in \u001b[0;36mSyncAPIClient.post\u001b[0;34m(self, path, cast_to, body, options, files, stream, stream_cls)\u001b[0m\n\u001b[1;32m   1074\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mpost\u001b[39m(\n\u001b[1;32m   1075\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m   1076\u001b[0m     path: \u001b[38;5;28mstr\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   1083\u001b[0m     stream_cls: \u001b[38;5;28mtype\u001b[39m[_StreamT] \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[1;32m   1084\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m ResponseT \u001b[38;5;241m|\u001b[39m _StreamT:\n\u001b[1;32m   1085\u001b[0m     opts \u001b[38;5;241m=\u001b[39m FinalRequestOptions\u001b[38;5;241m.\u001b[39mconstruct(\n\u001b[1;32m   1086\u001b[0m         method\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpost\u001b[39m\u001b[38;5;124m\"\u001b[39m, url\u001b[38;5;241m=\u001b[39mpath, json_data\u001b[38;5;241m=\u001b[39mbody, files\u001b[38;5;241m=\u001b[39mto_httpx_files(files), \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39moptions\n\u001b[1;32m   1087\u001b[0m     )\n\u001b[0;32m-> 1088\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m cast(ResponseT, \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcast_to\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mopts\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstream_cls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream_cls\u001b[49m\u001b[43m)\u001b[49m)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/openai/_base_client.py:853\u001b[0m, in \u001b[0;36mSyncAPIClient.request\u001b[0;34m(self, cast_to, options, remaining_retries, stream, stream_cls)\u001b[0m\n\u001b[1;32m    844\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mrequest\u001b[39m(\n\u001b[1;32m    845\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m    846\u001b[0m     cast_to: Type[ResponseT],\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    851\u001b[0m     stream_cls: \u001b[38;5;28mtype\u001b[39m[_StreamT] \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[1;32m    852\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m ResponseT \u001b[38;5;241m|\u001b[39m _StreamT:\n\u001b[0;32m--> 853\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    854\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcast_to\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcast_to\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    855\u001b[0m \u001b[43m        \u001b[49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    856\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    857\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream_cls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream_cls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    858\u001b[0m \u001b[43m        \u001b[49m\u001b[43mremaining_retries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mremaining_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    859\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/openai/_base_client.py:877\u001b[0m, in \u001b[0;36mSyncAPIClient._request\u001b[0;34m(self, cast_to, options, remaining_retries, stream, stream_cls)\u001b[0m\n\u001b[1;32m    874\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_prepare_request(request)\n\u001b[1;32m    876\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 877\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_client\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    878\u001b[0m \u001b[43m        \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    879\u001b[0m \u001b[43m        \u001b[49m\u001b[43mauth\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcustom_auth\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    880\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_should_stream_response_body\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    881\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    882\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m httpx\u001b[38;5;241m.\u001b[39mTimeoutException \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m    883\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m retries \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpx/_client.py:901\u001b[0m, in \u001b[0;36mClient.send\u001b[0;34m(self, request, stream, auth, follow_redirects)\u001b[0m\n\u001b[1;32m    893\u001b[0m follow_redirects \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    894\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfollow_redirects\n\u001b[1;32m    895\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(follow_redirects, UseClientDefault)\n\u001b[1;32m    896\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m follow_redirects\n\u001b[1;32m    897\u001b[0m )\n\u001b[1;32m    899\u001b[0m auth \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_build_request_auth(request, auth)\n\u001b[0;32m--> 901\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_send_handling_auth\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    902\u001b[0m \u001b[43m    \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    903\u001b[0m \u001b[43m    \u001b[49m\u001b[43mauth\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mauth\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    904\u001b[0m \u001b[43m    \u001b[49m\u001b[43mfollow_redirects\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfollow_redirects\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    905\u001b[0m \u001b[43m    \u001b[49m\u001b[43mhistory\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m[\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    906\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    907\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    908\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m stream:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpx/_client.py:929\u001b[0m, in \u001b[0;36mClient._send_handling_auth\u001b[0;34m(self, request, auth, follow_redirects, history)\u001b[0m\n\u001b[1;32m    926\u001b[0m request \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mnext\u001b[39m(auth_flow)\n\u001b[1;32m    928\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[0;32m--> 929\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_send_handling_redirects\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    930\u001b[0m \u001b[43m        \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    931\u001b[0m \u001b[43m        \u001b[49m\u001b[43mfollow_redirects\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfollow_redirects\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    932\u001b[0m \u001b[43m        \u001b[49m\u001b[43mhistory\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mhistory\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    933\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    934\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    935\u001b[0m         \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpx/_client.py:966\u001b[0m, in \u001b[0;36mClient._send_handling_redirects\u001b[0;34m(self, request, follow_redirects, history)\u001b[0m\n\u001b[1;32m    963\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m hook \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_event_hooks[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mrequest\u001b[39m\u001b[38;5;124m\"\u001b[39m]:\n\u001b[1;32m    964\u001b[0m     hook(request)\n\u001b[0;32m--> 966\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_send_single_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    967\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    968\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m hook \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_event_hooks[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mresponse\u001b[39m\u001b[38;5;124m\"\u001b[39m]:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpx/_client.py:1002\u001b[0m, in \u001b[0;36mClient._send_single_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    997\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mRuntimeError\u001b[39;00m(\n\u001b[1;32m    998\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mAttempted to send an async request with a sync Client instance.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    999\u001b[0m     )\n\u001b[1;32m   1001\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m request_context(request\u001b[38;5;241m=\u001b[39mrequest):\n\u001b[0;32m-> 1002\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[43mtransport\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhandle_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1004\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(response\u001b[38;5;241m.\u001b[39mstream, SyncByteStream)\n\u001b[1;32m   1006\u001b[0m response\u001b[38;5;241m.\u001b[39mrequest \u001b[38;5;241m=\u001b[39m request\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpx/_transports/default.py:228\u001b[0m, in \u001b[0;36mHTTPTransport.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    215\u001b[0m req \u001b[38;5;241m=\u001b[39m httpcore\u001b[38;5;241m.\u001b[39mRequest(\n\u001b[1;32m    216\u001b[0m     method\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mmethod,\n\u001b[1;32m    217\u001b[0m     url\u001b[38;5;241m=\u001b[39mhttpcore\u001b[38;5;241m.\u001b[39mURL(\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    225\u001b[0m     extensions\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mextensions,\n\u001b[1;32m    226\u001b[0m )\n\u001b[1;32m    227\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m map_httpcore_exceptions():\n\u001b[0;32m--> 228\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_pool\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhandle_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mreq\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    230\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(resp\u001b[38;5;241m.\u001b[39mstream, typing\u001b[38;5;241m.\u001b[39mIterable)\n\u001b[1;32m    232\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m Response(\n\u001b[1;32m    233\u001b[0m     status_code\u001b[38;5;241m=\u001b[39mresp\u001b[38;5;241m.\u001b[39mstatus,\n\u001b[1;32m    234\u001b[0m     headers\u001b[38;5;241m=\u001b[39mresp\u001b[38;5;241m.\u001b[39mheaders,\n\u001b[1;32m    235\u001b[0m     stream\u001b[38;5;241m=\u001b[39mResponseStream(resp\u001b[38;5;241m.\u001b[39mstream),\n\u001b[1;32m    236\u001b[0m     extensions\u001b[38;5;241m=\u001b[39mresp\u001b[38;5;241m.\u001b[39mextensions,\n\u001b[1;32m    237\u001b[0m )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/connection_pool.py:268\u001b[0m, in \u001b[0;36mConnectionPool.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    266\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m ShieldCancellation():\n\u001b[1;32m    267\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mresponse_closed(status)\n\u001b[0;32m--> 268\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m exc\n\u001b[1;32m    269\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    270\u001b[0m     \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/connection_pool.py:251\u001b[0m, in \u001b[0;36mConnectionPool.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    248\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m exc\n\u001b[1;32m    250\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 251\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[43mconnection\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhandle_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    252\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m ConnectionNotAvailable:\n\u001b[1;32m    253\u001b[0m     \u001b[38;5;66;03m# The ConnectionNotAvailable exception is a special case, that\u001b[39;00m\n\u001b[1;32m    254\u001b[0m     \u001b[38;5;66;03m# indicates we need to retry the request on a new connection.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    258\u001b[0m     \u001b[38;5;66;03m# might end up as an HTTP/2 connection, but which actually ends\u001b[39;00m\n\u001b[1;32m    259\u001b[0m     \u001b[38;5;66;03m# up as HTTP/1.1.\u001b[39;00m\n\u001b[1;32m    260\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_pool_lock:\n\u001b[1;32m    261\u001b[0m         \u001b[38;5;66;03m# Maintain our position in the request queue, but reset the\u001b[39;00m\n\u001b[1;32m    262\u001b[0m         \u001b[38;5;66;03m# status so that the request becomes queued again.\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/connection.py:103\u001b[0m, in \u001b[0;36mHTTPConnection.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    100\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_connection\u001b[38;5;241m.\u001b[39mis_available():\n\u001b[1;32m    101\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m ConnectionNotAvailable()\n\u001b[0;32m--> 103\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_connection\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhandle_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/http11.py:133\u001b[0m, in \u001b[0;36mHTTP11Connection.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    131\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m Trace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mresponse_closed\u001b[39m\u001b[38;5;124m\"\u001b[39m, logger, request) \u001b[38;5;28;01mas\u001b[39;00m trace:\n\u001b[1;32m    132\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_response_closed()\n\u001b[0;32m--> 133\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m exc\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/http11.py:111\u001b[0m, in \u001b[0;36mHTTP11Connection.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    101\u001b[0m     \u001b[38;5;28;01mpass\u001b[39;00m\n\u001b[1;32m    103\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m Trace(\n\u001b[1;32m    104\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mreceive_response_headers\u001b[39m\u001b[38;5;124m\"\u001b[39m, logger, request, kwargs\n\u001b[1;32m    105\u001b[0m ) \u001b[38;5;28;01mas\u001b[39;00m trace:\n\u001b[1;32m    106\u001b[0m     (\n\u001b[1;32m    107\u001b[0m         http_version,\n\u001b[1;32m    108\u001b[0m         status,\n\u001b[1;32m    109\u001b[0m         reason_phrase,\n\u001b[1;32m    110\u001b[0m         headers,\n\u001b[0;32m--> 111\u001b[0m     ) \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_receive_response_headers\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    112\u001b[0m     trace\u001b[38;5;241m.\u001b[39mreturn_value \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    113\u001b[0m         http_version,\n\u001b[1;32m    114\u001b[0m         status,\n\u001b[1;32m    115\u001b[0m         reason_phrase,\n\u001b[1;32m    116\u001b[0m         headers,\n\u001b[1;32m    117\u001b[0m     )\n\u001b[1;32m    119\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m Response(\n\u001b[1;32m    120\u001b[0m     status\u001b[38;5;241m=\u001b[39mstatus,\n\u001b[1;32m    121\u001b[0m     headers\u001b[38;5;241m=\u001b[39mheaders,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    127\u001b[0m     },\n\u001b[1;32m    128\u001b[0m )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/http11.py:176\u001b[0m, in \u001b[0;36mHTTP11Connection._receive_response_headers\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    173\u001b[0m timeout \u001b[38;5;241m=\u001b[39m timeouts\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mread\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[1;32m    175\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[0;32m--> 176\u001b[0m     event \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_receive_event\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    177\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(event, h11\u001b[38;5;241m.\u001b[39mResponse):\n\u001b[1;32m    178\u001b[0m         \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/http11.py:212\u001b[0m, in \u001b[0;36mHTTP11Connection._receive_event\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    209\u001b[0m     event \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_h11_state\u001b[38;5;241m.\u001b[39mnext_event()\n\u001b[1;32m    211\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m event \u001b[38;5;129;01mis\u001b[39;00m h11\u001b[38;5;241m.\u001b[39mNEED_DATA:\n\u001b[0;32m--> 212\u001b[0m     data \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_network_stream\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    213\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mREAD_NUM_BYTES\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\n\u001b[1;32m    214\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    216\u001b[0m     \u001b[38;5;66;03m# If we feed this case through h11 we'll raise an exception like:\u001b[39;00m\n\u001b[1;32m    217\u001b[0m     \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[1;32m    218\u001b[0m     \u001b[38;5;66;03m#     httpcore.RemoteProtocolError: can't handle event type\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    222\u001b[0m     \u001b[38;5;66;03m# perspective. Instead we handle this case distinctly and treat\u001b[39;00m\n\u001b[1;32m    223\u001b[0m     \u001b[38;5;66;03m# it as a ConnectError.\u001b[39;00m\n\u001b[1;32m    224\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m data \u001b[38;5;241m==\u001b[39m \u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_h11_state\u001b[38;5;241m.\u001b[39mtheir_state \u001b[38;5;241m==\u001b[39m h11\u001b[38;5;241m.\u001b[39mSEND_RESPONSE:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_backends/sync.py:126\u001b[0m, in \u001b[0;36mSyncStream.read\u001b[0;34m(self, max_bytes, timeout)\u001b[0m\n\u001b[1;32m    124\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m map_exceptions(exc_map):\n\u001b[1;32m    125\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sock\u001b[38;5;241m.\u001b[39msettimeout(timeout)\n\u001b[0;32m--> 126\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sock\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrecv\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmax_bytes\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1263\u001b[0m, in \u001b[0;36mSSLSocket.recv\u001b[0;34m(self, buflen, flags)\u001b[0m\n\u001b[1;32m   1259\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m flags \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m   1260\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m   1261\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnon-zero flags not allowed in calls to recv() on \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m\n\u001b[1;32m   1262\u001b[0m             \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__class__\u001b[39m)\n\u001b[0;32m-> 1263\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43mbuflen\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1264\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1265\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39mrecv(buflen, flags)\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1136\u001b[0m, in \u001b[0;36mSSLSocket.read\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m   1134\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mread(\u001b[38;5;28mlen\u001b[39m, buffer)\n\u001b[1;32m   1135\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m-> 1136\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sslobj\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1137\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m SSLError \u001b[38;5;28;01mas\u001b[39;00m x:\n\u001b[1;32m   1138\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m x\u001b[38;5;241m.\u001b[39margs[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;241m==\u001b[39m SSL_ERROR_EOF \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msuppress_ragged_eofs:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# # default mode of building the index\n",
    "# response_synthesizer = get_response_synthesizer(\n",
    "#     response_mode=\"tree_summarize\", use_async=True\n",
    "# )\n",
    "# doc_summary_index = DocumentSummaryIndex.from_documents(\n",
    "#     documents,\n",
    "#     service_context=service_context,\n",
    "#     response_synthesizer=response_synthesizer,\n",
    "#     show_progress=True,\n",
    "# )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
