{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip install langchain matplotlib scipy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import logging\n",
    "import sys\n",
    "\n",
    "logging.basicConfig(stream=sys.stdout, level=logging.INFO)\n",
    "logging.getLogger().addHandler(logging.StreamHandler(stream=sys.stdout))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summarization with clustering of topics \n",
    "\n",
    "## Existing solutions: recursive summarization\n",
    "\n",
    "1.1 recursive summarization \n",
    "\n",
    "1.2. refine \n",
    "\n",
    "## Proposed solution\n",
    "\n",
    "tackle the summarization and topic modelling process together in the same algorithm.\n",
    "\n",
    "1. Chunk into groups of sentences\n",
    "2. Summarize each chunk and ask a title for each chunk\n",
    "3. Embed summaries and find clusters of embeddeded vectors (semantic clusters)\n",
    "4. Pass those clusters of summaries to produce a summary by topic \n",
    "\n",
    "https://towardsdatascience.com/summarize-podcast-transcripts-and-long-texts-better-with-nlp-and-ai-e04c89d3b2cb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from datetime import datetime\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import json\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "from scipy.spatial.distance import cosine\n",
    "import networkx as nx\n",
    "from networkx.algorithms import community\n",
    "\n",
    "from langchain import OpenAI\n",
    "from langchain.prompts import PromptTemplate\n",
    "from langchain.chains.llm import LLMChain\n",
    "from langchain.embeddings import OpenAIEmbeddings\n",
    "from langchain.docstore.document import Document\n",
    "from langchain.chains.summarize import load_summarize_chain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from dotenv import load_dotenv\n",
    "\n",
    "load_dotenv('/workspaces/ml-learning/.env', override=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "\n",
    "def pickleSave(object, name, folder=\".\", silent=False):\n",
    "    filename = folder + \"/\" + name + \".pkl\"\n",
    "    if silent == False:\n",
    "        print(\"Saving object {} to pickle file {}\".format(name, filename))\n",
    "    with open(filename, mode=\"wb\") as fipkl:\n",
    "        pickle.dump(object, fipkl)\n",
    "\n",
    "\n",
    "def pickleLoad(name, folder: str, silent=False):\n",
    "    filename = folder + \"/\" + name + \".pkl\"\n",
    "    if silent == False:\n",
    "        print(\"Loading object {} from pickle file {}\".format(name, filename))\n",
    "\n",
    "    try:\n",
    "        with open(filename, mode=\"rb\") as fipkl:\n",
    "            myObject = pickle.load(fipkl)\n",
    "        return myObject\n",
    "    except IOError:\n",
    "        print(\"Pickle file {} not found, returning None object\".format(filename))\n",
    "        return None"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading documents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[4], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m SimpleDirectoryReader\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mingestion\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m IngestionPipeline\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mnode_parser\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m TokenTextSplitter\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/__init__.py:13\u001b[0m\n\u001b[1;32m     10\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mtyping\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Callable, Optional\n\u001b[1;32m     12\u001b[0m \u001b[38;5;66;03m# import global eval handler\u001b[39;00m\n\u001b[0;32m---> 13\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcallbacks\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mglobal_handlers\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m set_global_handler\n\u001b[1;32m     14\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mdata_structs\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mstruct_type\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m IndexStructType\n\u001b[1;32m     16\u001b[0m \u001b[38;5;66;03m# embeddings\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/callbacks/__init__.py:7\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mopen_inference_callback\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m OpenInferenceCallbackHandler\n\u001b[1;32m      6\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mschema\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m CBEvent, CBEventType, EventPayload\n\u001b[0;32m----> 7\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mtoken_counting\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m TokenCountingHandler\n\u001b[1;32m      8\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mutils\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m trace_method\n\u001b[1;32m      9\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mwandb_callback\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m WandbCallbackHandler\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/callbacks/token_counting.py:6\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcallbacks\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mbase_handler\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m BaseCallbackHandler\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcallbacks\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mschema\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m CBEventType, EventPayload\n\u001b[0;32m----> 6\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mutilities\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mtoken_counting\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m TokenCounter\n\u001b[1;32m      7\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mutils\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m get_tokenizer\n\u001b[1;32m     10\u001b[0m \u001b[38;5;129m@dataclass\u001b[39m\n\u001b[1;32m     11\u001b[0m \u001b[38;5;28;01mclass\u001b[39;00m \u001b[38;5;21;01mTokenCountingEvent\u001b[39;00m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/utilities/token_counting.py:6\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m# Modified from:\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;66;03m# https://github.com/nyno-ai/openai-token-counter\u001b[39;00m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mtyping\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Any, Callable, Dict, List, Optional\n\u001b[0;32m----> 6\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mllms\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m ChatMessage, MessageRole\n\u001b[1;32m      7\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mutils\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m get_tokenizer\n\u001b[1;32m     10\u001b[0m \u001b[38;5;28;01mclass\u001b[39;00m \u001b[38;5;21;01mTokenCounter\u001b[39;00m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/__init__.py:1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mllms\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mai21\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m AI21\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mllms\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01manthropic\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Anthropic\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mllms\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01manyscale\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Anyscale\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/ai21.py:6\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcallbacks\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m CallbackManager\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mllms\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mai21_utils\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m ai21_model_to_context_size\n\u001b[0;32m----> 6\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mllms\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mbase\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m llm_chat_callback, llm_completion_callback\n\u001b[1;32m      7\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mllms\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcustom\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m CustomLLM\n\u001b[1;32m      8\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mllms\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mgeneric_utils\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[1;32m      9\u001b[0m     completion_to_chat_decorator,\n\u001b[1;32m     10\u001b[0m     get_from_param_or_env,\n\u001b[1;32m     11\u001b[0m )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/base.py:25\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcallbacks\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m CallbackManager, CBEventType, EventPayload\n\u001b[1;32m     15\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mllms\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mtypes\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[1;32m     16\u001b[0m     ChatMessage,\n\u001b[1;32m     17\u001b[0m     ChatResponse,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     23\u001b[0m     LLMMetadata,\n\u001b[1;32m     24\u001b[0m )\n\u001b[0;32m---> 25\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mschema\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m BaseComponent\n\u001b[1;32m     28\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mllm_chat_callback\u001b[39m() \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m Callable:\n\u001b[1;32m     29\u001b[0m     \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrap\u001b[39m(f: Callable) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m Callable:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/schema.py:16\u001b[0m\n\u001b[1;32m     13\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mtyping_extensions\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Self\n\u001b[1;32m     15\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mbridge\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mpydantic\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m BaseModel, Field, root_validator\n\u001b[0;32m---> 16\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mutils\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m SAMPLE_TEXT, truncate_text\n\u001b[1;32m     18\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m TYPE_CHECKING:\n\u001b[1;32m     19\u001b[0m     \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mhaystack\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mschema\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Document \u001b[38;5;28;01mas\u001b[39;00m HaystackDocument\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/utils.py:89\u001b[0m\n\u001b[1;32m     85\u001b[0m             \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_stopwords \u001b[38;5;241m=\u001b[39m stopwords\u001b[38;5;241m.\u001b[39mwords(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124menglish\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     86\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_stopwords\n\u001b[0;32m---> 89\u001b[0m globals_helper \u001b[38;5;241m=\u001b[39m \u001b[43mGlobalsHelper\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     92\u001b[0m \u001b[38;5;66;03m# Global Tokenizer\u001b[39;00m\n\u001b[1;32m     93\u001b[0m \u001b[38;5;129m@runtime_checkable\u001b[39m\n\u001b[1;32m     94\u001b[0m \u001b[38;5;28;01mclass\u001b[39;00m \u001b[38;5;21;01mTokenizer\u001b[39;00m(Protocol):\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/utils.py:45\u001b[0m, in \u001b[0;36mGlobalsHelper.__init__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     43\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__init__\u001b[39m(\u001b[38;5;28mself\u001b[39m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m     44\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Initialize NLTK stopwords and punkt.\"\"\"\u001b[39;00m\n\u001b[0;32m---> 45\u001b[0m     \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\n\u001b[1;32m     47\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_nltk_data_dir \u001b[38;5;241m=\u001b[39m os\u001b[38;5;241m.\u001b[39menviron\u001b[38;5;241m.\u001b[39mget(\n\u001b[1;32m     48\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mNLTK_DATA\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m     49\u001b[0m         os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mjoin(\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     52\u001b[0m         ),\n\u001b[1;32m     53\u001b[0m     )\n\u001b[1;32m     55\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_nltk_data_dir \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m nltk\u001b[38;5;241m.\u001b[39mdata\u001b[38;5;241m.\u001b[39mpath:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/nltk/__init__.py:133\u001b[0m\n\u001b[1;32m    125\u001b[0m     subprocess\u001b[38;5;241m.\u001b[39mPopen \u001b[38;5;241m=\u001b[39m _fake_Popen\n\u001b[1;32m    127\u001b[0m \u001b[38;5;66;03m###########################################################\u001b[39;00m\n\u001b[1;32m    128\u001b[0m \u001b[38;5;66;03m# TOP-LEVEL MODULES\u001b[39;00m\n\u001b[1;32m    129\u001b[0m \u001b[38;5;66;03m###########################################################\u001b[39;00m\n\u001b[1;32m    130\u001b[0m \n\u001b[1;32m    131\u001b[0m \u001b[38;5;66;03m# Import top-level functionality into top-level namespace\u001b[39;00m\n\u001b[0;32m--> 133\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcollocations\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;241m*\u001b[39m\n\u001b[1;32m    134\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mdecorators\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m decorator, memoize\n\u001b[1;32m    135\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mfeatstruct\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;241m*\u001b[39m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/nltk/collocations.py:36\u001b[0m\n\u001b[1;32m     33\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mitertools\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01m_itertools\u001b[39;00m\n\u001b[1;32m     35\u001b[0m \u001b[38;5;66;03m# these two unused imports are referenced in collocations.doctest\u001b[39;00m\n\u001b[0;32m---> 36\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mmetrics\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[1;32m     37\u001b[0m     BigramAssocMeasures,\n\u001b[1;32m     38\u001b[0m     ContingencyMeasures,\n\u001b[1;32m     39\u001b[0m     QuadgramAssocMeasures,\n\u001b[1;32m     40\u001b[0m     TrigramAssocMeasures,\n\u001b[1;32m     41\u001b[0m )\n\u001b[1;32m     42\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mmetrics\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mspearman\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m ranks_from_scores, spearman_correlation\n\u001b[1;32m     43\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mprobability\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m FreqDist\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/nltk/metrics/__init__.py:18\u001b[0m\n\u001b[1;32m     16\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mmetrics\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01magreement\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m AnnotationTask\n\u001b[1;32m     17\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mmetrics\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01maline\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m align\n\u001b[0;32m---> 18\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mmetrics\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01massociation\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[1;32m     19\u001b[0m     BigramAssocMeasures,\n\u001b[1;32m     20\u001b[0m     ContingencyMeasures,\n\u001b[1;32m     21\u001b[0m     NgramAssocMeasures,\n\u001b[1;32m     22\u001b[0m     QuadgramAssocMeasures,\n\u001b[1;32m     23\u001b[0m     TrigramAssocMeasures,\n\u001b[1;32m     24\u001b[0m )\n\u001b[1;32m     25\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mmetrics\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mconfusionmatrix\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m ConfusionMatrix\n\u001b[1;32m     26\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnltk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mmetrics\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mdistance\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[1;32m     27\u001b[0m     binary_distance,\n\u001b[1;32m     28\u001b[0m     custom_distance,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     35\u001b[0m     presence,\n\u001b[1;32m     36\u001b[0m )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/nltk/metrics/association.py:26\u001b[0m\n\u001b[1;32m     23\u001b[0m _SMALL \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1e-20\u001b[39m\n\u001b[1;32m     25\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m---> 26\u001b[0m     \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mscipy\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mstats\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m fisher_exact\n\u001b[1;32m     27\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mImportError\u001b[39;00m:\n\u001b[1;32m     29\u001b[0m     \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mfisher_exact\u001b[39m(\u001b[38;5;241m*\u001b[39m_args, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39m_kwargs):\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/scipy/stats/__init__.py:608\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;124;03m.. _statsrefmanual:\u001b[39;00m\n\u001b[1;32m      3\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    603\u001b[0m \n\u001b[1;32m    604\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    606\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_warnings_errors\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (ConstantInputWarning, NearConstantInputWarning,\n\u001b[1;32m    607\u001b[0m                                DegenerateDataWarning, FitError)\n\u001b[0;32m--> 608\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_stats_py\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;241m*\u001b[39m\n\u001b[1;32m    609\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_variation\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m variation\n\u001b[1;32m    610\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mdistributions\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;241m*\u001b[39m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/scipy/stats/_stats_py.py:53\u001b[0m\n\u001b[1;32m     50\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_stats\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (_kendall_dis, _toint64, _weightedrankedtau,\n\u001b[1;32m     51\u001b[0m                      _local_correlations)\n\u001b[1;32m     52\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mdataclasses\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m dataclass\n\u001b[0;32m---> 53\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_hypotests\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m _all_partitions\n\u001b[1;32m     54\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_stats_pythran\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m _compute_outer_prob_inside_method\n\u001b[1;32m     55\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_resampling\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (MonteCarloMethod, PermutationMethod, BootstrapMethod,\n\u001b[1;32m     56\u001b[0m                           monte_carlo_test, permutation_test, bootstrap,\n\u001b[1;32m     57\u001b[0m                           _batch_generator)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/scipy/stats/_hypotests.py:13\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_continuous_distns\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m chi2, norm\n\u001b[1;32m     12\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mscipy\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mspecial\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m gamma, kv, gammaln\n\u001b[0;32m---> 13\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mscipy\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mfft\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m ifft\n\u001b[1;32m     14\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_stats_pythran\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m _a_ij_Aij_Dij2\n\u001b[1;32m     15\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_stats_pythran\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[1;32m     16\u001b[0m     _concordant_pairs \u001b[38;5;28;01mas\u001b[39;00m _P, _discordant_pairs \u001b[38;5;28;01mas\u001b[39;00m _Q\n\u001b[1;32m     17\u001b[0m )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/scipy/fft/__init__.py:85\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;124;03m==============================================\u001b[39;00m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;124;03mDiscrete Fourier transforms (:mod:`scipy.fft`)\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     82\u001b[0m \n\u001b[1;32m     83\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m---> 85\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_basic\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[1;32m     86\u001b[0m     fft, ifft, fft2, ifft2, fftn, ifftn,\n\u001b[1;32m     87\u001b[0m     rfft, irfft, rfft2, irfft2, rfftn, irfftn,\n\u001b[1;32m     88\u001b[0m     hfft, ihfft, hfft2, ihfft2, hfftn, ihfftn)\n\u001b[1;32m     89\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_realtransforms\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m dct, idct, dst, idst, dctn, idctn, dstn, idstn\n\u001b[1;32m     90\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_fftlog\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m fhtoffset\n",
      "File \u001b[0;32m<frozen importlib._bootstrap>:1176\u001b[0m, in \u001b[0;36m_find_and_load\u001b[0;34m(name, import_)\u001b[0m\n",
      "File \u001b[0;32m<frozen importlib._bootstrap>:1138\u001b[0m, in \u001b[0;36m_find_and_load_unlocked\u001b[0;34m(name, import_)\u001b[0m\n",
      "File \u001b[0;32m<frozen importlib._bootstrap>:1078\u001b[0m, in \u001b[0;36m_find_spec\u001b[0;34m(name, path, target)\u001b[0m\n",
      "File \u001b[0;32m<frozen importlib._bootstrap_external>:1504\u001b[0m, in \u001b[0;36mfind_spec\u001b[0;34m(cls, fullname, path, target)\u001b[0m\n",
      "File \u001b[0;32m<frozen importlib._bootstrap_external>:1476\u001b[0m, in \u001b[0;36m_get_spec\u001b[0;34m(cls, fullname, path, target)\u001b[0m\n",
      "File \u001b[0;32m<frozen importlib._bootstrap_external>:1643\u001b[0m, in \u001b[0;36mfind_spec\u001b[0;34m(self, fullname, target)\u001b[0m\n",
      "File \u001b[0;32m<frozen importlib._bootstrap>:244\u001b[0m, in \u001b[0;36m_verbose_message\u001b[0;34m(message, verbosity, *args)\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "from llama_index import SimpleDirectoryReader\n",
    "from llama_index.ingestion import IngestionPipeline\n",
    "from llama_index.node_parser import TokenTextSplitter\n",
    "from llama_index import download_loader\n",
    "\n",
    "WikipediaReader = download_loader(\"WikipediaReader\")\n",
    "\n",
    "loader = WikipediaReader()\n",
    "documents = loader.load_data(pages=['Louvain Method', 'Modularity (networks)', 'Community structure'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=10773696&db=pmc\n",
      "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=10769878&db=pmc\n",
      "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=10769083&db=pmc\n",
      "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=10763311&db=pmc\n",
      "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=10760059&db=pmc\n",
      "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=10758657&db=pmc\n",
      "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=10757668&db=pmc\n",
      "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=10756680&db=pmc\n",
      "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=10756212&db=pmc\n",
      "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=10754392&db=pmc\n"
     ]
    }
   ],
   "source": [
    "from llama_index import download_loader\n",
    "\n",
    "PubmedReader = download_loader(\"PubmedReader\")\n",
    "\n",
    "loader = PubmedReader()\n",
    "documents = loader.load_data(search_query='phage P1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saving object pubmed_documents to pickle file ./pubmed_documents.pkl\n"
     ]
    }
   ],
   "source": [
    "pickleSave(documents, 'pubmed_documents', folder=\".\", silent=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading object pubmed_documents from pickle file ./pubmed_documents.pkl\n"
     ]
    }
   ],
   "source": [
    "documents = pickleLoad('pubmed_documents', folder=\".\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/workspaces/ml-learning/.venv/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n",
      "Parsing nodes: 100%|██████████| 10/10 [00:00<00:00, 24.12it/s]\n"
     ]
    }
   ],
   "source": [
    "from llama_index.text_splitter import SentenceSplitter\n",
    "import tqdm as notebook_tqdm\n",
    "\n",
    "text_splitter = SentenceSplitter(chunk_size=512, chunk_overlap=128)\n",
    "nodes = text_splitter.get_nodes_from_documents(documents, show_progress=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "llama_index.schema.TextNode"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(nodes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "chunks_text = [ node.text for node in nodes]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index.response_synthesizers import Accumulate\n",
    "from llama_index.retrievers import ListIndexRetriever\n",
    "from llama_index.indices import SummaryIndex\n",
    "from llama_index import PromptTemplate\n",
    "\n",
    "map_prompt_template = \"\"\"Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\n",
    "{context_str}\n",
    "\n",
    "Return your answer in the following format:\n",
    "Title | Summary...\n",
    "e.g. \n",
    "Why Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\n",
    "\n",
    "TITLE AND CONCISE SUMMARY:\"\"\"\n",
    "\n",
    "query = \"\"\"Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\n",
    "\n",
    "Return your answer in the following format:\n",
    "Title | Summary...\n",
    "e.g. \n",
    "Why Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\n",
    "\n",
    "TITLE AND CONCISE SUMMARY:\"\"\"\n",
    "\n",
    "map_prompt = PromptTemplate(template=map_prompt_template)\n",
    "\n",
    "list_index = SummaryIndex(nodes=nodes)\n",
    "\n",
    "nodes_with_scores = ListIndexRetriever(index=list_index).retrieve(query)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index.types import BaseModel\n",
    "from typing import List\n",
    "\n",
    "class SummaryWithTitle(BaseModel):\n",
    "    \"\"\"Data model for a summary with title.\"\"\"\n",
    "\n",
    "    summary: str\n",
    "    title: str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['pmc PeerJ Comput Sci PeerJ Comput Sci peerj-cs PeerJ Computer Science 2376-5992 PeerJ Inc. San Diego, USA 38192480 10773696 cs-1685 10.7717/peerj-cs.1685 Bioinformatics Artificial Intelligence Computer Vision Real time anatomical landmarks and abnormalities detection in gastrointestinal tract Khan Zeshan zeshan.khan@nu.edu.pk Tahir Muhammad Atif  FAST School of Computing, National University of Computer and Emerging Sciences, Islamabad Karachi Sindh Pakistan Chaki Jyotismita 19 12 2023 2023 9 e1685 3 3 2023 16 10 2023 ©2023 Khan and Tahir 2023 Khan and Tahir https://creativecommons.org/licenses/by/4.0/ This is an open access article distributed under the terms of the Creative Commons Attribution License Gastrointestinal (GI) endoscopy is an active research field due to the lethal cancer diseases in the GI tract. Cancer treatments result better if diagnosed early and it increases the survival chances. There is a high miss rate in the detection of the abnormalities in the GI tract during endoscopy or colonoscopy due to the lack of attentiveness, tiring procedures, or the lack of required training. The procedure of the detection can be automated to the reduction of the risks by identifying and flagging the suspicious frames. A suspicious frame may have some of the abnormality or the information about anatomical landmark in the frame. The frame then can be analysed for the anatomical landmarks and the abnormalities for the detection of disease. In this research, a real-time endoscopic abnormalities detection system is presented that detects the abnormalities and the landmarks. The proposed system is based on a combination of handcrafted and deep features. Deep features are extracted from lightweight MobileNet convolutional neural network (CNN) architecture. There are some of the classes with a small inter-class difference and a higher intra-class differences, for such classes the same detection threshold is unable to distinguish. The threshold of such classes is learned from the training data using genetic algorithm.',\n",
       " 'The frame then can be analysed for the anatomical landmarks and the abnormalities for the detection of disease. In this research, a real-time endoscopic abnormalities detection system is presented that detects the abnormalities and the landmarks. The proposed system is based on a combination of handcrafted and deep features. Deep features are extracted from lightweight MobileNet convolutional neural network (CNN) architecture. There are some of the classes with a small inter-class difference and a higher intra-class differences, for such classes the same detection threshold is unable to distinguish. The threshold of such classes is learned from the training data using genetic algorithm. The system is evaluated on various benchmark datasets and resulted in an accuracy of 0.99 with the F1-score of 0.91 and Matthews correlation coefficient (MCC) of 0.91 on Kvasir datasets and F1-score of 0.93 on the dataset of DowPK. The system detects abnormalities in real-time with the detection speed of 41 frames per second. Medical image analysis GI tract diagnostics Genetic algorithm Threshold selection Endoscopic disease detection Computer vision Higher Education Commission (HEC) Pakistan 10225/2017 This research work was funded by the Higher Education Commission (HEC) Pakistan under NRPU Project 10225/2017. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.   Introduction Gastrointestinal (GI) cancer significantly contributes to mortality among various cancer types. Colon cancer is fifth in the most dangerous cancer types concerning the number of affected patients and deaths due to cancer. Rectum cancer is another type of GI tract cancer that is the 8th most dangerous cancer type in terms of number of cancer patients in the last five years\\xa0( Sung et al., 2021 Sung et al., 2021 Riegler et al., 2017 Pogorelov et al., 2018 Suzuki, 2012 Doi, 2007 There are many techniques to analyze GI tract images or videos using computer-aided diagnostic (CAD) systems. Most CAD systems primarily rely on machine learning and image processing techniques.',\n",
       " 'Rectum cancer is another type of GI tract cancer that is the 8th most dangerous cancer type in terms of number of cancer patients in the last five years\\xa0( Sung et al., 2021 Sung et al., 2021 Riegler et al., 2017 Pogorelov et al., 2018 Suzuki, 2012 Doi, 2007 There are many techniques to analyze GI tract images or videos using computer-aided diagnostic (CAD) systems. Most CAD systems primarily rely on machine learning and image processing techniques. The systems working on the machine learning algorithms on the texture and color features of the images are faster in detection. However, the detection score of such systems is lower than computer vision-based systems. The deep learning-based systems are sound in detection with a slow detection speed. There is a need for a system that offers a high detection speed in real-time and a justifiable detection accuracy. This research presents a three-step model for detecting abnormalities in the GI tract images. The first step relies on data preprocessing to address two critical challenges in the GI tract datasets. The first challenge is of light reflection on the images due to endoscopic procedures which is mitigated by applying the Open-CV Telea method. The second challenge in the endoscopic datasets is the high-class imbalance, which is mitigated with the generation of new images for all the classes so that the argument and original images combined make the dataset balanced. The following research step is to extract the most suitable features to represent images better. For this purpose, various features used in the literature have been explored, and a set of best features is selected after applying various feature combinations. The misleading features are excluded using feature accuracies and diversities. The third step of the research is to compute classes for an instance, which has further been divided in two steps. In the first part, a neural network is used to compute the probability of each class for an image and then to select the class based on different thresholds for each class instead of a 0.5 threshold or max value. The threshold for each class is computed by applying a genetic algorithm with the random initial threshold for all the classes.',\n",
       " 'The misleading features are excluded using feature accuracies and diversities. The third step of the research is to compute classes for an instance, which has further been divided in two steps. In the first part, a neural network is used to compute the probability of each class for an image and then to select the class based on different thresholds for each class instead of a 0.5 threshold or max value. The threshold for each class is computed by applying a genetic algorithm with the random initial threshold for all the classes. The methodology is named LiRE-CNN, as it uses some of the Lire features in combination with a neural network architecture. The detection algorithm of the LiRE-CNN is applied to different benchmark datasets with several modifications in the detection approach. The algorithm is also applied with and without preprocessing. The final results show that applying the preprocessing of reflection removal and augmentation and the neural network can achieve the best accuracy with optimal time. The texture features and local binary patterns are faster commutable, and the deep features are good deciders. Combining both types of features resulted in an accuracy of 0.99 with a detection speed of 41 frames per second. The same algorithm resulted in the F1-score of 0.91, 0.90, and 0.91 on the Kvasir versions V1\\xa0( Pogorelov et al., 2017c Pogorelov et al., 2018 Borgli et al., 2020 This article is organized as follows. ‘Related Work’ discusses the related work. The proposed methodology is presented in ‘Proposed Approach’. ‘Experimental Setup’ describes the experimental set-up followed by a discussion about results in ‘Results and Discussion’. ‘Conclusion and Future Work’ concludes the article.  Related Work The primary objective of the endoscopy or colonoscopy is to detect the functionality of the Gastro-Intestinal tract. Several possible abnormalities in the GI tract include polyps, lesions, esophagitis, and ulcerative colitis. There is a strong disagreement among medical specialists about the decision of the various abnormalities that can cause low detection accuracy of the issues in GI tract.',\n",
       " 'The proposed methodology is presented in ‘Proposed Approach’. ‘Experimental Setup’ describes the experimental set-up followed by a discussion about results in ‘Results and Discussion’. ‘Conclusion and Future Work’ concludes the article.  Related Work The primary objective of the endoscopy or colonoscopy is to detect the functionality of the Gastro-Intestinal tract. Several possible abnormalities in the GI tract include polyps, lesions, esophagitis, and ulcerative colitis. There is a strong disagreement among medical specialists about the decision of the various abnormalities that can cause low detection accuracy of the issues in GI tract. Intense work is available on the detection, classification, localization, and segmentation of various diseases in endoscopy. The detection tasks are mainly done for the detection of polyps and lesions using elliptical shape, color, position, texture, local intensity variation patterns, and global geometric constraints as features\\xa0( Hwang et al., 2007 Alexandre, Nobre & Casteleiro, 2008 Tajbakhsh et al., 2014 The primary issue in medical diagnostics is the limited availability of labeled datasets for applying machine learning algorithms. The limited dataset availability also generates a problem of class imbalance in the case of detection. It may lead to a higher number of non-polyps images than polyps. The data augmentation is done by using three major techniques.   1. Generative adversarial network (GAN): The GANs are generative models used for the generation of images from some of the existing images of various classes.  2. Angular flips: The angular flips have been used for the data augmentation with the operations of the rotation and flipping image with various angles, cropping, and resizing of the image\\xa0( Luo et al., 2019 Chang et al., 2019  3.']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[n.text for n in nodes[:5]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "from typing import Any\n",
    "\n",
    "class CustomAccumulate(Accumulate):\n",
    "    \"\"\"Custom Accumulate class with a redefined _format_response method.\"\"\"\n",
    "\n",
    "    def __init__(self, *args, **kwargs):\n",
    "        super().__init__(*args, **kwargs)\n",
    "\n",
    "    def _format_response(self, outputs: List[Any], separator: str) -> str:\n",
    "        # Your custom implementation of the method\n",
    "        # For example, a simple concatenated string of responses\n",
    "        return separator.join(str(output) for output in outputs)\n",
    "    \n",
    "    def _format_response(self, outputs: List[Any], separator: str) -> str:\n",
    "        # responses: List[str] = []\n",
    "        # for response in outputs:\n",
    "        #     responses.append(response or \"Empty Response\")\n",
    "\n",
    "        # return responses\n",
    "        return (response or \"Empty Response\" for response in outputs)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "from dataclasses import dataclass, field\n",
    "from typing import List, Optional, Dict, Any\n",
    "from llama_index.response.schema import Response, StreamingResponse\n",
    "\n",
    "def get_response_gen_to_list(r: StreamingResponse) -> List[Response]:\n",
    "    \"\"\"\n",
    "    Redefined get_response method.\n",
    "    Your custom implementation goes here.\n",
    "    \"\"\"\n",
    "\n",
    "    \"\"\"Get a standard response object list.\"\"\"\n",
    "    if r.response_txt is None and r.response_gen is not None:\n",
    "        responses=[]\n",
    "        for i, text in enumerate(r.response_gen):\n",
    "            response = Response(text.json(), r.source_nodes, r.metadata)\n",
    "            responses.append(response)\n",
    "    return responses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "import asyncio\n",
    "\n",
    "response_synthesizer = CustomAccumulate(\n",
    "    text_qa_template=map_prompt, output_cls=SummaryWithTitle, use_async=True\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "asynthesize query: Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\n",
      "\n",
      "Return your answer in the following format:\n",
      "Title | Summary...\n",
      "e.g. \n",
      "Why Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\n",
      "\n",
      "TITLE AND CONCISE SUMMARY:\n",
      "text_chunks: ['Title of this paper: Multi-scale color local binary patterns for visual object classes recognition\\nJournal it was published in:: PeerJ Computer Science\\nURL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/\\n\\npmc PeerJ Comput Sci PeerJ Comput Sci peerj-cs PeerJ Computer Science 2376-5992 PeerJ Inc. San Diego, USA 38192480 10773696 cs-1685 10.7717/peerj-cs.1685 Bioinformatics Artificial Intelligence Computer Vision Real time anatomical landmarks and abnormalities detection in gastrointestinal tract Khan Zeshan zeshan.khan@nu.edu.pk Tahir Muhammad Atif  FAST School of Computing, National University of Computer and Emerging Sciences, Islamabad Karachi Sindh Pakistan Chaki Jyotismita 19 12 2023 2023 9 e1685 3 3 2023 16 10 2023 ©2023 Khan and Tahir 2023 Khan and Tahir https://creativecommons.org/licenses/by/4.0/ This is an open access article distributed under the terms of the Creative Commons Attribution License Gastrointestinal (GI) endoscopy is an active research field due to the lethal cancer diseases in the GI tract. Cancer treatments result better if diagnosed early and it increases the survival chances. There is a high miss rate in the detection of the abnormalities in the GI tract during endoscopy or colonoscopy due to the lack of attentiveness, tiring procedures, or the lack of required training. The procedure of the detection can be automated to the reduction of the risks by identifying and flagging the suspicious frames. A suspicious frame may have some of the abnormality or the information about anatomical landmark in the frame. The frame then can be analysed for the anatomical landmarks and the abnormalities for the detection of disease. In this research, a real-time endoscopic abnormalities detection system is presented that detects the abnormalities and the landmarks. The proposed system is based on a combination of handcrafted and deep features. Deep features are extracted from lightweight MobileNet convolutional neural network (CNN) architecture. There are some of the classes with a small inter-class difference and a higher intra-class differences, for such classes the same detection threshold is unable to distinguish. The threshold of such classes is learned from the training data using genetic algorithm.', 'Title of this paper: Multi-scale color local binary patterns for visual object classes recognition\\nJournal it was published in:: PeerJ Computer Science\\nURL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/\\n\\nThe frame then can be analysed for the anatomical landmarks and the abnormalities for the detection of disease. In this research, a real-time endoscopic abnormalities detection system is presented that detects the abnormalities and the landmarks. The proposed system is based on a combination of handcrafted and deep features. Deep features are extracted from lightweight MobileNet convolutional neural network (CNN) architecture. There are some of the classes with a small inter-class difference and a higher intra-class differences, for such classes the same detection threshold is unable to distinguish. The threshold of such classes is learned from the training data using genetic algorithm. The system is evaluated on various benchmark datasets and resulted in an accuracy of 0.99 with the F1-score of 0.91 and Matthews correlation coefficient (MCC) of 0.91 on Kvasir datasets and F1-score of 0.93 on the dataset of DowPK. The system detects abnormalities in real-time with the detection speed of 41 frames per second. Medical image analysis GI tract diagnostics Genetic algorithm Threshold selection Endoscopic disease detection Computer vision Higher Education Commission (HEC) Pakistan 10225/2017 This research work was funded by the Higher Education Commission (HEC) Pakistan under NRPU Project 10225/2017. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.   Introduction Gastrointestinal (GI) cancer significantly contributes to mortality among various cancer types. Colon cancer is fifth in the most dangerous cancer types concerning the number of affected patients and deaths due to cancer. Rectum cancer is another type of GI tract cancer that is the 8th most dangerous cancer type in terms of number of cancer patients in the last five years\\xa0( Sung et al., 2021 Sung et al., 2021 Riegler et al., 2017 Pogorelov et al., 2018 Suzuki, 2012 Doi, 2007 There are many techniques to analyze GI tract images or videos using computer-aided diagnostic (CAD) systems. Most CAD systems primarily rely on machine learning and image processing techniques.', 'Title of this paper: Multi-scale color local binary patterns for visual object classes recognition\\nJournal it was published in:: PeerJ Computer Science\\nURL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/\\n\\nRectum cancer is another type of GI tract cancer that is the 8th most dangerous cancer type in terms of number of cancer patients in the last five years\\xa0( Sung et al., 2021 Sung et al., 2021 Riegler et al., 2017 Pogorelov et al., 2018 Suzuki, 2012 Doi, 2007 There are many techniques to analyze GI tract images or videos using computer-aided diagnostic (CAD) systems. Most CAD systems primarily rely on machine learning and image processing techniques. The systems working on the machine learning algorithms on the texture and color features of the images are faster in detection. However, the detection score of such systems is lower than computer vision-based systems. The deep learning-based systems are sound in detection with a slow detection speed. There is a need for a system that offers a high detection speed in real-time and a justifiable detection accuracy. This research presents a three-step model for detecting abnormalities in the GI tract images. The first step relies on data preprocessing to address two critical challenges in the GI tract datasets. The first challenge is of light reflection on the images due to endoscopic procedures which is mitigated by applying the Open-CV Telea method. The second challenge in the endoscopic datasets is the high-class imbalance, which is mitigated with the generation of new images for all the classes so that the argument and original images combined make the dataset balanced. The following research step is to extract the most suitable features to represent images better. For this purpose, various features used in the literature have been explored, and a set of best features is selected after applying various feature combinations. The misleading features are excluded using feature accuracies and diversities. The third step of the research is to compute classes for an instance, which has further been divided in two steps. In the first part, a neural network is used to compute the probability of each class for an image and then to select the class based on different thresholds for each class instead of a 0.5 threshold or max value. The threshold for each class is computed by applying a genetic algorithm with the random initial threshold for all the classes.']\n",
      "**response_kwargs {}\n",
      "Creating tasks...\n",
      "text_qa_template 1: metadata={'prompt_type': <PromptType.CUSTOM: 'custom'>} template_vars=['context_str'] kwargs={} output_parser=None template_var_mappings=None function_mappings=None template='Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n{context_str}\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'\n",
      "text_qa_template 2: metadata={'prompt_type': <PromptType.CUSTOM: 'custom'>} template_vars=['context_str'] kwargs={'query_str': 'Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'} output_parser=None template_var_mappings=None function_mappings=None template='Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n{context_str}\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'\n",
      "text_chunks: ['Title of this paper: Multi-scale color local binary patterns for visual object classes recognition\\nJournal it was published in:: PeerJ Computer Science\\nURL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/\\n\\npmc PeerJ Comput Sci PeerJ Comput Sci peerj-cs PeerJ Computer Science 2376-5992 PeerJ Inc. San Diego, USA 38192480 10773696 cs-1685 10.7717/peerj-cs.1685 Bioinformatics Artificial Intelligence Computer Vision Real time anatomical landmarks and abnormalities detection in gastrointestinal tract Khan Zeshan zeshan.khan@nu.edu.pk Tahir Muhammad Atif  FAST School of Computing, National University of Computer and Emerging Sciences, Islamabad Karachi Sindh Pakistan Chaki Jyotismita 19 12 2023 2023 9 e1685 3 3 2023 16 10 2023 ©2023 Khan and Tahir 2023 Khan and Tahir https://creativecommons.org/licenses/by/4.0/ This is an open access article distributed under the terms of the Creative Commons Attribution License Gastrointestinal (GI) endoscopy is an active research field due to the lethal cancer diseases in the GI tract. Cancer treatments result better if diagnosed early and it increases the survival chances. There is a high miss rate in the detection of the abnormalities in the GI tract during endoscopy or colonoscopy due to the lack of attentiveness, tiring procedures, or the lack of required training. The procedure of the detection can be automated to the reduction of the risks by identifying and flagging the suspicious frames. A suspicious frame may have some of the abnormality or the information about anatomical landmark in the frame. The frame then can be analysed for the anatomical landmarks and the abnormalities for the detection of disease. In this research, a real-time endoscopic abnormalities detection system is presented that detects the abnormalities and the landmarks. The proposed system is based on a combination of handcrafted and deep features. Deep features are extracted from lightweight MobileNet convolutional neural network (CNN) architecture. There are some of the classes with a small inter-class difference and a higher intra-class differences, for such classes the same detection threshold is unable to distinguish. The threshold of such classes is learned from the training data using genetic algorithm.']\n",
      "text_qa_template 1: metadata={'prompt_type': <PromptType.CUSTOM: 'custom'>} template_vars=['context_str'] kwargs={} output_parser=None template_var_mappings=None function_mappings=None template='Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n{context_str}\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'\n",
      "text_qa_template 2: metadata={'prompt_type': <PromptType.CUSTOM: 'custom'>} template_vars=['context_str'] kwargs={'query_str': 'Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'} output_parser=None template_var_mappings=None function_mappings=None template='Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n{context_str}\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'\n",
      "text_chunks: ['Title of this paper: Multi-scale color local binary patterns for visual object classes recognition\\nJournal it was published in:: PeerJ Computer Science\\nURL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/\\n\\nThe frame then can be analysed for the anatomical landmarks and the abnormalities for the detection of disease. In this research, a real-time endoscopic abnormalities detection system is presented that detects the abnormalities and the landmarks. The proposed system is based on a combination of handcrafted and deep features. Deep features are extracted from lightweight MobileNet convolutional neural network (CNN) architecture. There are some of the classes with a small inter-class difference and a higher intra-class differences, for such classes the same detection threshold is unable to distinguish. The threshold of such classes is learned from the training data using genetic algorithm. The system is evaluated on various benchmark datasets and resulted in an accuracy of 0.99 with the F1-score of 0.91 and Matthews correlation coefficient (MCC) of 0.91 on Kvasir datasets and F1-score of 0.93 on the dataset of DowPK. The system detects abnormalities in real-time with the detection speed of 41 frames per second. Medical image analysis GI tract diagnostics Genetic algorithm Threshold selection Endoscopic disease detection Computer vision Higher Education Commission (HEC) Pakistan 10225/2017 This research work was funded by the Higher Education Commission (HEC) Pakistan under NRPU Project 10225/2017. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.   Introduction Gastrointestinal (GI) cancer significantly contributes to mortality among various cancer types. Colon cancer is fifth in the most dangerous cancer types concerning the number of affected patients and deaths due to cancer. Rectum cancer is another type of GI tract cancer that is the 8th most dangerous cancer type in terms of number of cancer patients in the last five years\\xa0( Sung et al., 2021 Sung et al., 2021 Riegler et al., 2017 Pogorelov et al., 2018 Suzuki, 2012 Doi, 2007 There are many techniques to analyze GI tract images or videos using computer-aided diagnostic (CAD) systems. Most CAD systems primarily rely on machine learning and image processing techniques.']\n",
      "text_qa_template 1: metadata={'prompt_type': <PromptType.CUSTOM: 'custom'>} template_vars=['context_str'] kwargs={} output_parser=None template_var_mappings=None function_mappings=None template='Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n{context_str}\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'\n",
      "text_qa_template 2: metadata={'prompt_type': <PromptType.CUSTOM: 'custom'>} template_vars=['context_str'] kwargs={'query_str': 'Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'} output_parser=None template_var_mappings=None function_mappings=None template='Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n{context_str}\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'\n",
      "text_chunks: ['Title of this paper: Multi-scale color local binary patterns for visual object classes recognition\\nJournal it was published in:: PeerJ Computer Science\\nURL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/\\n\\nRectum cancer is another type of GI tract cancer that is the 8th most dangerous cancer type in terms of number of cancer patients in the last five years\\xa0( Sung et al., 2021 Sung et al., 2021 Riegler et al., 2017 Pogorelov et al., 2018 Suzuki, 2012 Doi, 2007 There are many techniques to analyze GI tract images or videos using computer-aided diagnostic (CAD) systems. Most CAD systems primarily rely on machine learning and image processing techniques. The systems working on the machine learning algorithms on the texture and color features of the images are faster in detection. However, the detection score of such systems is lower than computer vision-based systems. The deep learning-based systems are sound in detection with a slow detection speed. There is a need for a system that offers a high detection speed in real-time and a justifiable detection accuracy. This research presents a three-step model for detecting abnormalities in the GI tract images. The first step relies on data preprocessing to address two critical challenges in the GI tract datasets. The first challenge is of light reflection on the images due to endoscopic procedures which is mitigated by applying the Open-CV Telea method. The second challenge in the endoscopic datasets is the high-class imbalance, which is mitigated with the generation of new images for all the classes so that the argument and original images combined make the dataset balanced. The following research step is to extract the most suitable features to represent images better. For this purpose, various features used in the literature have been explored, and a set of best features is selected after applying various feature combinations. The misleading features are excluded using feature accuracies and diversities. The third step of the research is to compute classes for an instance, which has further been divided in two steps. In the first part, a neural network is used to compute the probability of each class for an image and then to select the class based on different thresholds for each class instead of a 0.5 threshold or max value. The threshold for each class is computed by applying a genetic algorithm with the random initial threshold for all the classes.']\n",
      "self._promtp: metadata={'prompt_type': <PromptType.CUSTOM: 'custom'>} template_vars=['context_str'] kwargs={'query_str': 'Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'} output_parser=None template_var_mappings=None function_mappings=None template='Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n{context_str}\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'\n",
      "prompt params llm: callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7f9e5dbcfb10> system_prompt=None messages_to_prompt=<function messages_to_prompt at 0x7f9e6297b600> completion_to_prompt=<function default_completion_to_prompt at 0x7f9e627bf740> output_parser=None pydantic_program_mode=<PydanticProgramMode.DEFAULT: 'default'> query_wrapper_prompt=None model='gpt-3.5-turbo' temperature=0.1 max_tokens=None additional_kwargs={} max_retries=3 timeout=60.0 default_headers=None reuse_client=True api_key='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' api_base='https://api.openai.com/v1' api_version=''\n",
      "prompt params **kwargs: {'context_str': 'Title of this paper: Multi-scale color local binary patterns for visual object classes recognition\\nJournal it was published in:: PeerJ Computer Science\\nURL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/\\n\\npmc PeerJ Comput Sci PeerJ Comput Sci peerj-cs PeerJ Computer Science 2376-5992 PeerJ Inc. San Diego, USA 38192480 10773696 cs-1685 10.7717/peerj-cs.1685 Bioinformatics Artificial Intelligence Computer Vision Real time anatomical landmarks and abnormalities detection in gastrointestinal tract Khan Zeshan zeshan.khan@nu.edu.pk Tahir Muhammad Atif  FAST School of Computing, National University of Computer and Emerging Sciences, Islamabad Karachi Sindh Pakistan Chaki Jyotismita 19 12 2023 2023 9 e1685 3 3 2023 16 10 2023 ©2023 Khan and Tahir 2023 Khan and Tahir https://creativecommons.org/licenses/by/4.0/ This is an open access article distributed under the terms of the Creative Commons Attribution License Gastrointestinal (GI) endoscopy is an active research field due to the lethal cancer diseases in the GI tract. Cancer treatments result better if diagnosed early and it increases the survival chances. There is a high miss rate in the detection of the abnormalities in the GI tract during endoscopy or colonoscopy due to the lack of attentiveness, tiring procedures, or the lack of required training. The procedure of the detection can be automated to the reduction of the risks by identifying and flagging the suspicious frames. A suspicious frame may have some of the abnormality or the information about anatomical landmark in the frame. The frame then can be analysed for the anatomical landmarks and the abnormalities for the detection of disease. In this research, a real-time endoscopic abnormalities detection system is presented that detects the abnormalities and the landmarks. The proposed system is based on a combination of handcrafted and deep features. Deep features are extracted from lightweight MobileNet convolutional neural network (CNN) architecture. There are some of the classes with a small inter-class difference and a higher intra-class differences, for such classes the same detection threshold is unable to distinguish. The threshold of such classes is learned from the training data using genetic algorithm.'}\n",
      "tool_choice: {'type': 'function', 'function': {'name': 'SummaryWithTitle'}}\n",
      "self._promtp: metadata={'prompt_type': <PromptType.CUSTOM: 'custom'>} template_vars=['context_str'] kwargs={'query_str': 'Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'} output_parser=None template_var_mappings=None function_mappings=None template='Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n{context_str}\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'\n",
      "prompt params llm: callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7f9e5dbcfb10> system_prompt=None messages_to_prompt=<function messages_to_prompt at 0x7f9e6297b600> completion_to_prompt=<function default_completion_to_prompt at 0x7f9e627bf740> output_parser=None pydantic_program_mode=<PydanticProgramMode.DEFAULT: 'default'> query_wrapper_prompt=None model='gpt-3.5-turbo' temperature=0.1 max_tokens=None additional_kwargs={} max_retries=3 timeout=60.0 default_headers=None reuse_client=True api_key='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' api_base='https://api.openai.com/v1' api_version=''\n",
      "prompt params **kwargs: {'context_str': 'Title of this paper: Multi-scale color local binary patterns for visual object classes recognition\\nJournal it was published in:: PeerJ Computer Science\\nURL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/\\n\\nThe frame then can be analysed for the anatomical landmarks and the abnormalities for the detection of disease. In this research, a real-time endoscopic abnormalities detection system is presented that detects the abnormalities and the landmarks. The proposed system is based on a combination of handcrafted and deep features. Deep features are extracted from lightweight MobileNet convolutional neural network (CNN) architecture. There are some of the classes with a small inter-class difference and a higher intra-class differences, for such classes the same detection threshold is unable to distinguish. The threshold of such classes is learned from the training data using genetic algorithm. The system is evaluated on various benchmark datasets and resulted in an accuracy of 0.99 with the F1-score of 0.91 and Matthews correlation coefficient (MCC) of 0.91 on Kvasir datasets and F1-score of 0.93 on the dataset of DowPK. The system detects abnormalities in real-time with the detection speed of 41 frames per second. Medical image analysis GI tract diagnostics Genetic algorithm Threshold selection Endoscopic disease detection Computer vision Higher Education Commission (HEC) Pakistan 10225/2017 This research work was funded by the Higher Education Commission (HEC) Pakistan under NRPU Project 10225/2017. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.   Introduction Gastrointestinal (GI) cancer significantly contributes to mortality among various cancer types. Colon cancer is fifth in the most dangerous cancer types concerning the number of affected patients and deaths due to cancer. Rectum cancer is another type of GI tract cancer that is the 8th most dangerous cancer type in terms of number of cancer patients in the last five years\\xa0( Sung et al., 2021 Sung et al., 2021 Riegler et al., 2017 Pogorelov et al., 2018 Suzuki, 2012 Doi, 2007 There are many techniques to analyze GI tract images or videos using computer-aided diagnostic (CAD) systems. Most CAD systems primarily rely on machine learning and image processing techniques.'}\n",
      "tool_choice: {'type': 'function', 'function': {'name': 'SummaryWithTitle'}}\n",
      "self._promtp: metadata={'prompt_type': <PromptType.CUSTOM: 'custom'>} template_vars=['context_str'] kwargs={'query_str': 'Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'} output_parser=None template_var_mappings=None function_mappings=None template='Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\\n{context_str}\\n\\nReturn your answer in the following format:\\nTitle | Summary...\\ne.g. \\nWhy Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\\n\\nTITLE AND CONCISE SUMMARY:'\n",
      "prompt params llm: callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7f9e5dbcfb10> system_prompt=None messages_to_prompt=<function messages_to_prompt at 0x7f9e6297b600> completion_to_prompt=<function default_completion_to_prompt at 0x7f9e627bf740> output_parser=None pydantic_program_mode=<PydanticProgramMode.DEFAULT: 'default'> query_wrapper_prompt=None model='gpt-3.5-turbo' temperature=0.1 max_tokens=None additional_kwargs={} max_retries=3 timeout=60.0 default_headers=None reuse_client=True api_key='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' api_base='https://api.openai.com/v1' api_version=''\n",
      "prompt params **kwargs: {'context_str': 'Title of this paper: Multi-scale color local binary patterns for visual object classes recognition\\nJournal it was published in:: PeerJ Computer Science\\nURL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/\\n\\nRectum cancer is another type of GI tract cancer that is the 8th most dangerous cancer type in terms of number of cancer patients in the last five years\\xa0( Sung et al., 2021 Sung et al., 2021 Riegler et al., 2017 Pogorelov et al., 2018 Suzuki, 2012 Doi, 2007 There are many techniques to analyze GI tract images or videos using computer-aided diagnostic (CAD) systems. Most CAD systems primarily rely on machine learning and image processing techniques. The systems working on the machine learning algorithms on the texture and color features of the images are faster in detection. However, the detection score of such systems is lower than computer vision-based systems. The deep learning-based systems are sound in detection with a slow detection speed. There is a need for a system that offers a high detection speed in real-time and a justifiable detection accuracy. This research presents a three-step model for detecting abnormalities in the GI tract images. The first step relies on data preprocessing to address two critical challenges in the GI tract datasets. The first challenge is of light reflection on the images due to endoscopic procedures which is mitigated by applying the Open-CV Telea method. The second challenge in the endoscopic datasets is the high-class imbalance, which is mitigated with the generation of new images for all the classes so that the argument and original images combined make the dataset balanced. The following research step is to extract the most suitable features to represent images better. For this purpose, various features used in the literature have been explored, and a set of best features is selected after applying various feature combinations. The misleading features are excluded using feature accuracies and diversities. The third step of the research is to compute classes for an instance, which has further been divided in two steps. In the first part, a neural network is used to compute the probability of each class for an image and then to select the class based on different thresholds for each class instead of a 0.5 threshold or max value. The threshold for each class is computed by applying a genetic algorithm with the random initial threshold for all the classes.'}\n",
      "tool_choice: {'type': 'function', 'function': {'name': 'SummaryWithTitle'}}\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "INFO:httpx:HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n",
      "HTTP Request: POST https://api.openai.com/v1/chat/completions \"HTTP/1.1 200 OK\"\n"
     ]
    }
   ],
   "source": [
    "responses = await response_synthesizer.asynthesize(query, nodes_with_scores[:3])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "streaming_response = responses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = get_response_gen_to_list(streaming_response)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "{\"summary\": \"A real-time endoscopic abnormalities detection system based on handcrafted and deep features is presented in this research. The system uses a combination of lightweight MobileNet convolutional neural network (CNN) architecture and genetic algorithm to detect abnormalities and landmarks in medical images. The system achieves high accuracy and F1-score on benchmark datasets, with a detection speed of 41 frames per second. GI tract cancer, particularly colon and rectum cancer, is a significant cause of mortality. Computer-aided diagnostic (CAD) systems, including machine learning and image processing techniques, are commonly used to analyze GI tract images or videos.\", \"title\": \"Multi-scale color local binary patterns for visual object classes recognition\"}"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "{\"summary\": \"This research presents a three-step model for detecting abnormalities in GI tract images using a combination of machine learning and image processing techniques. The model addresses challenges such as light reflection and high-class imbalance in the datasets. It preprocesses the data using the Open-CV Telea method and generates new images to balance the dataset. It then selects the most suitable features to represent the images and excludes misleading features. Finally, it computes classes for an instance using a neural network and a genetic algorithm to determine the threshold for each class. The proposed model aims to offer high detection speed and justifiable detection accuracy in real-time.\", \"title\": \"Multi-scale color local binary patterns for visual object classes recognition\"}"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "IndexError",
     "evalue": "list index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[38], line 5\u001b[0m\n\u001b[1;32m      3\u001b[0m display(Markdown(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mresults[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m'\u001b[39m))\n\u001b[1;32m      4\u001b[0m display(Markdown(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mresults[\u001b[38;5;241m1\u001b[39m]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m'\u001b[39m))\n\u001b[0;32m----> 5\u001b[0m display(Markdown(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[43mresults\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m2\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m'\u001b[39m))\n",
      "\u001b[0;31mIndexError\u001b[0m: list index out of range"
     ]
    }
   ],
   "source": [
    "from IPython.display import Markdown, display\n",
    "\n",
    "display(Markdown(f'{results[0]}'))\n",
    "display(Markdown(f'{results[1]}'))\n",
    "display(Markdown(f'{results[2]}'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Response(response=\"Response 1: summary='Gastrointestinal abnormalities detection system using multi-scale color local binary patterns' title='Multi-scale color local binary patterns for visual object classes recognition'\\n---------------------\\nResponse 2: summary='This research presents a real-time endoscopic abnormalities detection system based on a combination of handcrafted and deep features. The system uses a lightweight MobileNet convolutional neural network (CNN) architecture to extract deep features. For classes with small inter-class differences, a genetic algorithm is used to learn the detection threshold from the training data. The system achieves high accuracy and F1-score on benchmark datasets, with a detection speed of 41 frames per second. The research is funded by the Higher Education Commission (HEC) Pakistan and contributes to the analysis of gastrointestinal (GI) tract images for disease detection using computer vision and machine learning techniques.' title='Multi-scale color local binary patterns for visual object classes recognition'\\n---------------------\\nResponse 3: summary='This research presents a three-step model for detecting abnormalities in GI tract images. The model addresses challenges such as light reflection and high-class imbalance in the datasets. It involves data preprocessing, feature extraction, and class computation using a neural network and genetic algorithm. The system aims to offer high detection speed and accuracy in real-time. The research explores various features and selects the best ones based on accuracies and diversities. Instead of using a fixed threshold, the model uses different thresholds for each class to compute the probability of each class for an image. The thresholds are computed using a genetic algorithm. The research was published in the PeerJ Computer Science journal.' title='Multi-scale color local binary patterns for visual object classes recognition'\", source_nodes=[NodeWithScore(node=TextNode(id_='dbeb39e4-81cd-446b-af06-6d003b3866d0', embedding=None, metadata={'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='5c2a5aa9-474a-4f7e-bb9a-aaf41f659f6a', node_type=<ObjectType.DOCUMENT: '4'>, metadata={'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}, hash='4d2df4f1db371fb1a0dcce5f9513c1bd270bceb9f0b77a9e5560e7ea0021bbe0'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='1977f8b6-044e-49cb-917a-84ff9dcce2a7', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='359f40c10602b35d5dfd3154d7f1162a39ad7a962b52049005f5a52ee0f50293')}, hash='65921c0a0349afdbdc53507b4c95c870828a14162186c3b11916af7f418aaacd', text='pmc PeerJ Comput Sci PeerJ Comput Sci peerj-cs PeerJ Computer Science 2376-5992 PeerJ Inc. San Diego, USA 38192480 10773696 cs-1685 10.7717/peerj-cs.1685 Bioinformatics Artificial Intelligence Computer Vision Real time anatomical landmarks and abnormalities detection in gastrointestinal tract Khan Zeshan zeshan.khan@nu.edu.pk Tahir Muhammad Atif  FAST School of Computing, National University of Computer and Emerging Sciences, Islamabad Karachi Sindh Pakistan Chaki Jyotismita 19 12 2023 2023 9 e1685 3 3 2023 16 10 2023 ©2023 Khan and Tahir 2023 Khan and Tahir https://creativecommons.org/licenses/by/4.0/ This is an open access article distributed under the terms of the Creative Commons Attribution License Gastrointestinal (GI) endoscopy is an active research field due to the lethal cancer diseases in the GI tract. Cancer treatments result better if diagnosed early and it increases the survival chances. There is a high miss rate in the detection of the abnormalities in the GI tract during endoscopy or colonoscopy due to the lack of attentiveness, tiring procedures, or the lack of required training. The procedure of the detection can be automated to the reduction of the risks by identifying and flagging the suspicious frames. A suspicious frame may have some of the abnormality or the information about anatomical landmark in the frame. The frame then can be analysed for the anatomical landmarks and the abnormalities for the detection of disease. In this research, a real-time endoscopic abnormalities detection system is presented that detects the abnormalities and the landmarks. The proposed system is based on a combination of handcrafted and deep features. Deep features are extracted from lightweight MobileNet convolutional neural network (CNN) architecture. There are some of the classes with a small inter-class difference and a higher intra-class differences, for such classes the same detection threshold is unable to distinguish. The threshold of such classes is learned from the training data using genetic algorithm.', start_char_idx=2, end_char_idx=2050, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None), NodeWithScore(node=TextNode(id_='1977f8b6-044e-49cb-917a-84ff9dcce2a7', embedding=None, metadata={'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='5c2a5aa9-474a-4f7e-bb9a-aaf41f659f6a', node_type=<ObjectType.DOCUMENT: '4'>, metadata={'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}, hash='4d2df4f1db371fb1a0dcce5f9513c1bd270bceb9f0b77a9e5560e7ea0021bbe0'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='dbeb39e4-81cd-446b-af06-6d003b3866d0', node_type=<ObjectType.TEXT: '1'>, metadata={'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}, hash='330f3633926713963965d06f462566e9228388f89fd6b2d862b431783b092422'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='ee28374e-db09-44fa-aec6-d1cffc65d5b3', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='43d26d06f4e8965020f1a3139e7b142a3ce614334908cb8e48fdabe0e95902b6')}, hash='e876c278c50ca29d927f38a4e679220b3edae29ea33c89080ac1afcaec4d7531', text='The frame then can be analysed for the anatomical landmarks and the abnormalities for the detection of disease. In this research, a real-time endoscopic abnormalities detection system is presented that detects the abnormalities and the landmarks. The proposed system is based on a combination of handcrafted and deep features. Deep features are extracted from lightweight MobileNet convolutional neural network (CNN) architecture. There are some of the classes with a small inter-class difference and a higher intra-class differences, for such classes the same detection threshold is unable to distinguish. The threshold of such classes is learned from the training data using genetic algorithm. The system is evaluated on various benchmark datasets and resulted in an accuracy of 0.99 with the F1-score of 0.91 and Matthews correlation coefficient (MCC) of 0.91 on Kvasir datasets and F1-score of 0.93 on the dataset of DowPK. The system detects abnormalities in real-time with the detection speed of 41 frames per second. Medical image analysis GI tract diagnostics Genetic algorithm Threshold selection Endoscopic disease detection Computer vision Higher Education Commission (HEC) Pakistan 10225/2017 This research work was funded by the Higher Education Commission (HEC) Pakistan under NRPU Project 10225/2017. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.   Introduction Gastrointestinal (GI) cancer significantly contributes to mortality among various cancer types. Colon cancer is fifth in the most dangerous cancer types concerning the number of affected patients and deaths due to cancer. Rectum cancer is another type of GI tract cancer that is the 8th most dangerous cancer type in terms of number of cancer patients in the last five years\\xa0( Sung et al., 2021 Sung et al., 2021 Riegler et al., 2017 Pogorelov et al., 2018 Suzuki, 2012 Doi, 2007 There are many techniques to analyze GI tract images or videos using computer-aided diagnostic (CAD) systems. Most CAD systems primarily rely on machine learning and image processing techniques.', start_char_idx=1355, end_char_idx=3486, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None), NodeWithScore(node=TextNode(id_='ee28374e-db09-44fa-aec6-d1cffc65d5b3', embedding=None, metadata={'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='5c2a5aa9-474a-4f7e-bb9a-aaf41f659f6a', node_type=<ObjectType.DOCUMENT: '4'>, metadata={'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}, hash='4d2df4f1db371fb1a0dcce5f9513c1bd270bceb9f0b77a9e5560e7ea0021bbe0'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='1977f8b6-044e-49cb-917a-84ff9dcce2a7', node_type=<ObjectType.TEXT: '1'>, metadata={'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}, hash='359f40c10602b35d5dfd3154d7f1162a39ad7a962b52049005f5a52ee0f50293'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='4f8117e2-f2a4-4c2e-b7d2-89de78f5e88d', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='9ea91293fca8cd8f6d5ef8643f47b5031033d3d188ababab24c501856c65d623')}, hash='5cb494dec41038953ae2616fce8c2a52606d81fed58ffeb8b27cb94f29f5b66c', text='Rectum cancer is another type of GI tract cancer that is the 8th most dangerous cancer type in terms of number of cancer patients in the last five years\\xa0( Sung et al., 2021 Sung et al., 2021 Riegler et al., 2017 Pogorelov et al., 2018 Suzuki, 2012 Doi, 2007 There are many techniques to analyze GI tract images or videos using computer-aided diagnostic (CAD) systems. Most CAD systems primarily rely on machine learning and image processing techniques. The systems working on the machine learning algorithms on the texture and color features of the images are faster in detection. However, the detection score of such systems is lower than computer vision-based systems. The deep learning-based systems are sound in detection with a slow detection speed. There is a need for a system that offers a high detection speed in real-time and a justifiable detection accuracy. This research presents a three-step model for detecting abnormalities in the GI tract images. The first step relies on data preprocessing to address two critical challenges in the GI tract datasets. The first challenge is of light reflection on the images due to endoscopic procedures which is mitigated by applying the Open-CV Telea method. The second challenge in the endoscopic datasets is the high-class imbalance, which is mitigated with the generation of new images for all the classes so that the argument and original images combined make the dataset balanced. The following research step is to extract the most suitable features to represent images better. For this purpose, various features used in the literature have been explored, and a set of best features is selected after applying various feature combinations. The misleading features are excluded using feature accuracies and diversities. The third step of the research is to compute classes for an instance, which has further been divided in two steps. In the first part, a neural network is used to compute the probability of each class for an image and then to select the class based on different thresholds for each class instead of a 0.5 threshold or max value. The threshold for each class is computed by applying a genetic algorithm with the random initial threshold for all the classes.', start_char_idx=3034, end_char_idx=5266, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None)], metadata={'dbeb39e4-81cd-446b-af06-6d003b3866d0': {'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}, '1977f8b6-044e-49cb-917a-84ff9dcce2a7': {'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}, 'ee28374e-db09-44fa-aec6-d1cffc65d5b3': {'Title of this paper': 'Multi-scale color local binary patterns for visual object classes recognition', 'Journal it was published in:': 'PeerJ Computer Science', 'URL': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10773696/'}})"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# response = await  response_synthesizer.aget_response(query, [n.text for n in nodes[:5]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[19], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[43mAccumulate\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msynthesize\u001b[49m\u001b[43m(\u001b[49m\u001b[43mquery\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnodes_with_scores\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      2\u001b[0m response\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/response_synthesizers/base.py:146\u001b[0m, in \u001b[0;36mBaseSynthesizer.synthesize\u001b[0;34m(self, query, nodes, additional_source_nodes, **response_kwargs)\u001b[0m\n\u001b[1;32m    141\u001b[0m     query \u001b[38;5;241m=\u001b[39m QueryBundle(query_str\u001b[38;5;241m=\u001b[39mquery)\n\u001b[1;32m    143\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_callback_manager\u001b[38;5;241m.\u001b[39mevent(\n\u001b[1;32m    144\u001b[0m     CBEventType\u001b[38;5;241m.\u001b[39mSYNTHESIZE, payload\u001b[38;5;241m=\u001b[39m{EventPayload\u001b[38;5;241m.\u001b[39mQUERY_STR: query\u001b[38;5;241m.\u001b[39mquery_str}\n\u001b[1;32m    145\u001b[0m ) \u001b[38;5;28;01mas\u001b[39;00m event:\n\u001b[0;32m--> 146\u001b[0m     response_str \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_response\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    147\u001b[0m \u001b[43m        \u001b[49m\u001b[43mquery_str\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mquery_str\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    148\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtext_chunks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m[\u001b[49m\n\u001b[1;32m    149\u001b[0m \u001b[43m            \u001b[49m\u001b[43mn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mnode\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_content\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmetadata_mode\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mMetadataMode\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mLLM\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mn\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mnodes\u001b[49m\n\u001b[1;32m    150\u001b[0m \u001b[43m        \u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    151\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    152\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    154\u001b[0m     additional_source_nodes \u001b[38;5;241m=\u001b[39m additional_source_nodes \u001b[38;5;129;01mor\u001b[39;00m []\n\u001b[1;32m    155\u001b[0m     source_nodes \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(nodes) \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mlist\u001b[39m(additional_source_nodes)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/response_synthesizers/accumulate.py:89\u001b[0m, in \u001b[0;36mAccumulate.get_response\u001b[0;34m(self, query_str, text_chunks, separator, **response_kwargs)\u001b[0m\n\u001b[1;32m     86\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_streaming:\n\u001b[1;32m     87\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mUnable to stream in Accumulate response mode\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m---> 89\u001b[0m tasks \u001b[38;5;241m=\u001b[39m \u001b[43m[\u001b[49m\n\u001b[1;32m     90\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_give_responses\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     91\u001b[0m \u001b[43m        \u001b[49m\u001b[43mquery_str\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtext_chunk\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43muse_async\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_use_async\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kwargs\u001b[49m\n\u001b[1;32m     92\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     93\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mtext_chunk\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mtext_chunks\u001b[49m\n\u001b[1;32m     94\u001b[0m \u001b[43m\u001b[49m\u001b[43m]\u001b[49m\n\u001b[1;32m     96\u001b[0m outputs \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mflatten_list(tasks)\n\u001b[1;32m     98\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_use_async:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/response_synthesizers/accumulate.py:90\u001b[0m, in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m     86\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_streaming:\n\u001b[1;32m     87\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mUnable to stream in Accumulate response mode\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     89\u001b[0m tasks \u001b[38;5;241m=\u001b[39m [\n\u001b[0;32m---> 90\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_give_responses\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     91\u001b[0m \u001b[43m        \u001b[49m\u001b[43mquery_str\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtext_chunk\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43muse_async\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_use_async\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kwargs\u001b[49m\n\u001b[1;32m     92\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     93\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m text_chunk \u001b[38;5;129;01min\u001b[39;00m text_chunks\n\u001b[1;32m     94\u001b[0m ]\n\u001b[1;32m     96\u001b[0m outputs \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mflatten_list(tasks)\n\u001b[1;32m     98\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_use_async:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/response_synthesizers/accumulate.py:125\u001b[0m, in \u001b[0;36mAccumulate._give_responses\u001b[0;34m(self, query_str, text_chunk, use_async, **response_kwargs)\u001b[0m\n\u001b[1;32m    118\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_output_cls \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    119\u001b[0m     predictor \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    120\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_service_context\u001b[38;5;241m.\u001b[39mllm\u001b[38;5;241m.\u001b[39mapredict\n\u001b[1;32m    121\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m use_async\n\u001b[1;32m    122\u001b[0m         \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_service_context\u001b[38;5;241m.\u001b[39mllm\u001b[38;5;241m.\u001b[39mpredict\n\u001b[1;32m    123\u001b[0m     )\n\u001b[0;32m--> 125\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m[\u001b[49m\n\u001b[1;32m    126\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpredictor\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    127\u001b[0m \u001b[43m            \u001b[49m\u001b[43mtext_qa_template\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    128\u001b[0m \u001b[43m            \u001b[49m\u001b[43mcontext_str\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcur_text_chunk\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    129\u001b[0m \u001b[43m            \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    130\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    131\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mcur_text_chunk\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mtext_chunks\u001b[49m\n\u001b[1;32m    132\u001b[0m \u001b[43m    \u001b[49m\u001b[43m]\u001b[49m\n\u001b[1;32m    133\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    134\u001b[0m     predictor \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    135\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_service_context\u001b[38;5;241m.\u001b[39mllm\u001b[38;5;241m.\u001b[39mastructured_predict\n\u001b[1;32m    136\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m use_async\n\u001b[1;32m    137\u001b[0m         \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_service_context\u001b[38;5;241m.\u001b[39mllm\u001b[38;5;241m.\u001b[39mstructured_predict\n\u001b[1;32m    138\u001b[0m     )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/response_synthesizers/accumulate.py:126\u001b[0m, in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    118\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_output_cls \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    119\u001b[0m     predictor \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    120\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_service_context\u001b[38;5;241m.\u001b[39mllm\u001b[38;5;241m.\u001b[39mapredict\n\u001b[1;32m    121\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m use_async\n\u001b[1;32m    122\u001b[0m         \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_service_context\u001b[38;5;241m.\u001b[39mllm\u001b[38;5;241m.\u001b[39mpredict\n\u001b[1;32m    123\u001b[0m     )\n\u001b[1;32m    125\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m [\n\u001b[0;32m--> 126\u001b[0m         \u001b[43mpredictor\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    127\u001b[0m \u001b[43m            \u001b[49m\u001b[43mtext_qa_template\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    128\u001b[0m \u001b[43m            \u001b[49m\u001b[43mcontext_str\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcur_text_chunk\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    129\u001b[0m \u001b[43m            \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    130\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    131\u001b[0m         \u001b[38;5;28;01mfor\u001b[39;00m cur_text_chunk \u001b[38;5;129;01min\u001b[39;00m text_chunks\n\u001b[1;32m    132\u001b[0m     ]\n\u001b[1;32m    133\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    134\u001b[0m     predictor \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    135\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_service_context\u001b[38;5;241m.\u001b[39mllm\u001b[38;5;241m.\u001b[39mastructured_predict\n\u001b[1;32m    136\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m use_async\n\u001b[1;32m    137\u001b[0m         \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_service_context\u001b[38;5;241m.\u001b[39mllm\u001b[38;5;241m.\u001b[39mstructured_predict\n\u001b[1;32m    138\u001b[0m     )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/llm.py:220\u001b[0m, in \u001b[0;36mLLM.predict\u001b[0;34m(self, prompt, **prompt_args)\u001b[0m\n\u001b[1;32m    218\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmetadata\u001b[38;5;241m.\u001b[39mis_chat_model:\n\u001b[1;32m    219\u001b[0m     messages \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_messages(prompt, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mprompt_args)\n\u001b[0;32m--> 220\u001b[0m     chat_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mchat\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmessages\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    221\u001b[0m     output \u001b[38;5;241m=\u001b[39m chat_response\u001b[38;5;241m.\u001b[39mmessage\u001b[38;5;241m.\u001b[39mcontent \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    222\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/base.py:97\u001b[0m, in \u001b[0;36mllm_chat_callback.<locals>.wrap.<locals>.wrapped_llm_chat\u001b[0;34m(_self, messages, **kwargs)\u001b[0m\n\u001b[1;32m     88\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m wrapper_logic(_self) \u001b[38;5;28;01mas\u001b[39;00m callback_manager:\n\u001b[1;32m     89\u001b[0m     event_id \u001b[38;5;241m=\u001b[39m callback_manager\u001b[38;5;241m.\u001b[39mon_event_start(\n\u001b[1;32m     90\u001b[0m         CBEventType\u001b[38;5;241m.\u001b[39mLLM,\n\u001b[1;32m     91\u001b[0m         payload\u001b[38;5;241m=\u001b[39m{\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     95\u001b[0m         },\n\u001b[1;32m     96\u001b[0m     )\n\u001b[0;32m---> 97\u001b[0m     f_return_val \u001b[38;5;241m=\u001b[39m \u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[43m_self\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmessages\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     99\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(f_return_val, Generator):\n\u001b[1;32m    100\u001b[0m         \u001b[38;5;66;03m# intercept the generator and add a callback to the end\u001b[39;00m\n\u001b[1;32m    101\u001b[0m         \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrapped_gen\u001b[39m() \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m ChatResponseGen:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/openai.py:234\u001b[0m, in \u001b[0;36mOpenAI.chat\u001b[0;34m(self, messages, **kwargs)\u001b[0m\n\u001b[1;32m    232\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    233\u001b[0m     chat_fn \u001b[38;5;241m=\u001b[39m completion_to_chat_decorator(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_complete)\n\u001b[0;32m--> 234\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mchat_fn\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmessages\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/llms/openai.py:289\u001b[0m, in \u001b[0;36mOpenAI._chat\u001b[0;34m(self, messages, **kwargs)\u001b[0m\n\u001b[1;32m    287\u001b[0m client \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_client()\n\u001b[1;32m    288\u001b[0m message_dicts \u001b[38;5;241m=\u001b[39m to_openai_message_dicts(messages)\n\u001b[0;32m--> 289\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[43mclient\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mchat\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcompletions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcreate\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    290\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmessages\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmessage_dicts\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    291\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    292\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_model_kwargs\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    293\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    294\u001b[0m openai_message \u001b[38;5;241m=\u001b[39m response\u001b[38;5;241m.\u001b[39mchoices[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mmessage\n\u001b[1;32m    295\u001b[0m message \u001b[38;5;241m=\u001b[39m from_openai_message(openai_message)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/openai/_utils/_utils.py:272\u001b[0m, in \u001b[0;36mrequired_args.<locals>.inner.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    270\u001b[0m             msg \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mMissing required argument: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mquote(missing[\u001b[38;5;241m0\u001b[39m])\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    271\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m(msg)\n\u001b[0;32m--> 272\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/openai/resources/chat/completions.py:645\u001b[0m, in \u001b[0;36mCompletions.create\u001b[0;34m(self, messages, model, frequency_penalty, function_call, functions, logit_bias, logprobs, max_tokens, n, presence_penalty, response_format, seed, stop, stream, temperature, tool_choice, tools, top_logprobs, top_p, user, extra_headers, extra_query, extra_body, timeout)\u001b[0m\n\u001b[1;32m    596\u001b[0m \u001b[38;5;129m@required_args\u001b[39m([\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmessages\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmodel\u001b[39m\u001b[38;5;124m\"\u001b[39m], [\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmessages\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmodel\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mstream\u001b[39m\u001b[38;5;124m\"\u001b[39m])\n\u001b[1;32m    597\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mcreate\u001b[39m(\n\u001b[1;32m    598\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    643\u001b[0m     timeout: \u001b[38;5;28mfloat\u001b[39m \u001b[38;5;241m|\u001b[39m httpx\u001b[38;5;241m.\u001b[39mTimeout \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m|\u001b[39m NotGiven \u001b[38;5;241m=\u001b[39m NOT_GIVEN,\n\u001b[1;32m    644\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m ChatCompletion \u001b[38;5;241m|\u001b[39m Stream[ChatCompletionChunk]:\n\u001b[0;32m--> 645\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_post\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    646\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m/chat/completions\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    647\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmaybe_transform\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    648\u001b[0m \u001b[43m            \u001b[49m\u001b[43m{\u001b[49m\n\u001b[1;32m    649\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmessages\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mmessages\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    650\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmodel\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mmodel\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    651\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mfrequency_penalty\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mfrequency_penalty\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    652\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mfunction_call\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mfunction_call\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    653\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mfunctions\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mfunctions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    654\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mlogit_bias\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mlogit_bias\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    655\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mlogprobs\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mlogprobs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    656\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmax_tokens\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mmax_tokens\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    657\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mn\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    658\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mpresence_penalty\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mpresence_penalty\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    659\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mresponse_format\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mresponse_format\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    660\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mseed\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mseed\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    661\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstop\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mstop\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    662\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstream\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mstream\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    663\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtemperature\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mtemperature\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    664\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtool_choice\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mtool_choice\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    665\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtools\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mtools\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    666\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtop_logprobs\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mtop_logprobs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    667\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtop_p\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mtop_p\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    668\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43muser\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43muser\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    669\u001b[0m \u001b[43m            \u001b[49m\u001b[43m}\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    670\u001b[0m \u001b[43m            \u001b[49m\u001b[43mcompletion_create_params\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mCompletionCreateParams\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    671\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    672\u001b[0m \u001b[43m        \u001b[49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmake_request_options\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    673\u001b[0m \u001b[43m            \u001b[49m\u001b[43mextra_headers\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mextra_headers\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mextra_query\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mextra_query\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mextra_body\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mextra_body\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\n\u001b[1;32m    674\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    675\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcast_to\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mChatCompletion\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    676\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    677\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream_cls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mStream\u001b[49m\u001b[43m[\u001b[49m\u001b[43mChatCompletionChunk\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    678\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/openai/_base_client.py:1088\u001b[0m, in \u001b[0;36mSyncAPIClient.post\u001b[0;34m(self, path, cast_to, body, options, files, stream, stream_cls)\u001b[0m\n\u001b[1;32m   1074\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mpost\u001b[39m(\n\u001b[1;32m   1075\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m   1076\u001b[0m     path: \u001b[38;5;28mstr\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   1083\u001b[0m     stream_cls: \u001b[38;5;28mtype\u001b[39m[_StreamT] \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[1;32m   1084\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m ResponseT \u001b[38;5;241m|\u001b[39m _StreamT:\n\u001b[1;32m   1085\u001b[0m     opts \u001b[38;5;241m=\u001b[39m FinalRequestOptions\u001b[38;5;241m.\u001b[39mconstruct(\n\u001b[1;32m   1086\u001b[0m         method\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpost\u001b[39m\u001b[38;5;124m\"\u001b[39m, url\u001b[38;5;241m=\u001b[39mpath, json_data\u001b[38;5;241m=\u001b[39mbody, files\u001b[38;5;241m=\u001b[39mto_httpx_files(files), \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39moptions\n\u001b[1;32m   1087\u001b[0m     )\n\u001b[0;32m-> 1088\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m cast(ResponseT, \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcast_to\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mopts\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstream_cls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream_cls\u001b[49m\u001b[43m)\u001b[49m)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/openai/_base_client.py:853\u001b[0m, in \u001b[0;36mSyncAPIClient.request\u001b[0;34m(self, cast_to, options, remaining_retries, stream, stream_cls)\u001b[0m\n\u001b[1;32m    844\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mrequest\u001b[39m(\n\u001b[1;32m    845\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m    846\u001b[0m     cast_to: Type[ResponseT],\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    851\u001b[0m     stream_cls: \u001b[38;5;28mtype\u001b[39m[_StreamT] \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[1;32m    852\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m ResponseT \u001b[38;5;241m|\u001b[39m _StreamT:\n\u001b[0;32m--> 853\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    854\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcast_to\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcast_to\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    855\u001b[0m \u001b[43m        \u001b[49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    856\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    857\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream_cls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream_cls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    858\u001b[0m \u001b[43m        \u001b[49m\u001b[43mremaining_retries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mremaining_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    859\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/openai/_base_client.py:877\u001b[0m, in \u001b[0;36mSyncAPIClient._request\u001b[0;34m(self, cast_to, options, remaining_retries, stream, stream_cls)\u001b[0m\n\u001b[1;32m    874\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_prepare_request(request)\n\u001b[1;32m    876\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 877\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_client\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    878\u001b[0m \u001b[43m        \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    879\u001b[0m \u001b[43m        \u001b[49m\u001b[43mauth\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcustom_auth\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    880\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_should_stream_response_body\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    881\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    882\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m httpx\u001b[38;5;241m.\u001b[39mTimeoutException \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m    883\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m retries \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m0\u001b[39m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpx/_client.py:901\u001b[0m, in \u001b[0;36mClient.send\u001b[0;34m(self, request, stream, auth, follow_redirects)\u001b[0m\n\u001b[1;32m    893\u001b[0m follow_redirects \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    894\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfollow_redirects\n\u001b[1;32m    895\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(follow_redirects, UseClientDefault)\n\u001b[1;32m    896\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m follow_redirects\n\u001b[1;32m    897\u001b[0m )\n\u001b[1;32m    899\u001b[0m auth \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_build_request_auth(request, auth)\n\u001b[0;32m--> 901\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_send_handling_auth\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    902\u001b[0m \u001b[43m    \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    903\u001b[0m \u001b[43m    \u001b[49m\u001b[43mauth\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mauth\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    904\u001b[0m \u001b[43m    \u001b[49m\u001b[43mfollow_redirects\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfollow_redirects\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    905\u001b[0m \u001b[43m    \u001b[49m\u001b[43mhistory\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m[\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    906\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    907\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    908\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m stream:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpx/_client.py:929\u001b[0m, in \u001b[0;36mClient._send_handling_auth\u001b[0;34m(self, request, auth, follow_redirects, history)\u001b[0m\n\u001b[1;32m    926\u001b[0m request \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mnext\u001b[39m(auth_flow)\n\u001b[1;32m    928\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[0;32m--> 929\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_send_handling_redirects\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    930\u001b[0m \u001b[43m        \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    931\u001b[0m \u001b[43m        \u001b[49m\u001b[43mfollow_redirects\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfollow_redirects\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    932\u001b[0m \u001b[43m        \u001b[49m\u001b[43mhistory\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mhistory\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    933\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    934\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    935\u001b[0m         \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpx/_client.py:966\u001b[0m, in \u001b[0;36mClient._send_handling_redirects\u001b[0;34m(self, request, follow_redirects, history)\u001b[0m\n\u001b[1;32m    963\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m hook \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_event_hooks[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mrequest\u001b[39m\u001b[38;5;124m\"\u001b[39m]:\n\u001b[1;32m    964\u001b[0m     hook(request)\n\u001b[0;32m--> 966\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_send_single_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    967\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    968\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m hook \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_event_hooks[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mresponse\u001b[39m\u001b[38;5;124m\"\u001b[39m]:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpx/_client.py:1002\u001b[0m, in \u001b[0;36mClient._send_single_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    997\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mRuntimeError\u001b[39;00m(\n\u001b[1;32m    998\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mAttempted to send an async request with a sync Client instance.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    999\u001b[0m     )\n\u001b[1;32m   1001\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m request_context(request\u001b[38;5;241m=\u001b[39mrequest):\n\u001b[0;32m-> 1002\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[43mtransport\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhandle_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1004\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(response\u001b[38;5;241m.\u001b[39mstream, SyncByteStream)\n\u001b[1;32m   1006\u001b[0m response\u001b[38;5;241m.\u001b[39mrequest \u001b[38;5;241m=\u001b[39m request\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpx/_transports/default.py:228\u001b[0m, in \u001b[0;36mHTTPTransport.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    215\u001b[0m req \u001b[38;5;241m=\u001b[39m httpcore\u001b[38;5;241m.\u001b[39mRequest(\n\u001b[1;32m    216\u001b[0m     method\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mmethod,\n\u001b[1;32m    217\u001b[0m     url\u001b[38;5;241m=\u001b[39mhttpcore\u001b[38;5;241m.\u001b[39mURL(\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    225\u001b[0m     extensions\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mextensions,\n\u001b[1;32m    226\u001b[0m )\n\u001b[1;32m    227\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m map_httpcore_exceptions():\n\u001b[0;32m--> 228\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_pool\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhandle_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mreq\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    230\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(resp\u001b[38;5;241m.\u001b[39mstream, typing\u001b[38;5;241m.\u001b[39mIterable)\n\u001b[1;32m    232\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m Response(\n\u001b[1;32m    233\u001b[0m     status_code\u001b[38;5;241m=\u001b[39mresp\u001b[38;5;241m.\u001b[39mstatus,\n\u001b[1;32m    234\u001b[0m     headers\u001b[38;5;241m=\u001b[39mresp\u001b[38;5;241m.\u001b[39mheaders,\n\u001b[1;32m    235\u001b[0m     stream\u001b[38;5;241m=\u001b[39mResponseStream(resp\u001b[38;5;241m.\u001b[39mstream),\n\u001b[1;32m    236\u001b[0m     extensions\u001b[38;5;241m=\u001b[39mresp\u001b[38;5;241m.\u001b[39mextensions,\n\u001b[1;32m    237\u001b[0m )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/connection_pool.py:268\u001b[0m, in \u001b[0;36mConnectionPool.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    266\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m ShieldCancellation():\n\u001b[1;32m    267\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mresponse_closed(status)\n\u001b[0;32m--> 268\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m exc\n\u001b[1;32m    269\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    270\u001b[0m     \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/connection_pool.py:251\u001b[0m, in \u001b[0;36mConnectionPool.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    248\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m exc\n\u001b[1;32m    250\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 251\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[43mconnection\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhandle_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    252\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m ConnectionNotAvailable:\n\u001b[1;32m    253\u001b[0m     \u001b[38;5;66;03m# The ConnectionNotAvailable exception is a special case, that\u001b[39;00m\n\u001b[1;32m    254\u001b[0m     \u001b[38;5;66;03m# indicates we need to retry the request on a new connection.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    258\u001b[0m     \u001b[38;5;66;03m# might end up as an HTTP/2 connection, but which actually ends\u001b[39;00m\n\u001b[1;32m    259\u001b[0m     \u001b[38;5;66;03m# up as HTTP/1.1.\u001b[39;00m\n\u001b[1;32m    260\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_pool_lock:\n\u001b[1;32m    261\u001b[0m         \u001b[38;5;66;03m# Maintain our position in the request queue, but reset the\u001b[39;00m\n\u001b[1;32m    262\u001b[0m         \u001b[38;5;66;03m# status so that the request becomes queued again.\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/connection.py:103\u001b[0m, in \u001b[0;36mHTTPConnection.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    100\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_connection\u001b[38;5;241m.\u001b[39mis_available():\n\u001b[1;32m    101\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m ConnectionNotAvailable()\n\u001b[0;32m--> 103\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_connection\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhandle_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/http11.py:133\u001b[0m, in \u001b[0;36mHTTP11Connection.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    131\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m Trace(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mresponse_closed\u001b[39m\u001b[38;5;124m\"\u001b[39m, logger, request) \u001b[38;5;28;01mas\u001b[39;00m trace:\n\u001b[1;32m    132\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_response_closed()\n\u001b[0;32m--> 133\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m exc\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/http11.py:111\u001b[0m, in \u001b[0;36mHTTP11Connection.handle_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    101\u001b[0m     \u001b[38;5;28;01mpass\u001b[39;00m\n\u001b[1;32m    103\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m Trace(\n\u001b[1;32m    104\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mreceive_response_headers\u001b[39m\u001b[38;5;124m\"\u001b[39m, logger, request, kwargs\n\u001b[1;32m    105\u001b[0m ) \u001b[38;5;28;01mas\u001b[39;00m trace:\n\u001b[1;32m    106\u001b[0m     (\n\u001b[1;32m    107\u001b[0m         http_version,\n\u001b[1;32m    108\u001b[0m         status,\n\u001b[1;32m    109\u001b[0m         reason_phrase,\n\u001b[1;32m    110\u001b[0m         headers,\n\u001b[0;32m--> 111\u001b[0m     ) \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_receive_response_headers\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    112\u001b[0m     trace\u001b[38;5;241m.\u001b[39mreturn_value \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    113\u001b[0m         http_version,\n\u001b[1;32m    114\u001b[0m         status,\n\u001b[1;32m    115\u001b[0m         reason_phrase,\n\u001b[1;32m    116\u001b[0m         headers,\n\u001b[1;32m    117\u001b[0m     )\n\u001b[1;32m    119\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m Response(\n\u001b[1;32m    120\u001b[0m     status\u001b[38;5;241m=\u001b[39mstatus,\n\u001b[1;32m    121\u001b[0m     headers\u001b[38;5;241m=\u001b[39mheaders,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    127\u001b[0m     },\n\u001b[1;32m    128\u001b[0m )\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/http11.py:176\u001b[0m, in \u001b[0;36mHTTP11Connection._receive_response_headers\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    173\u001b[0m timeout \u001b[38;5;241m=\u001b[39m timeouts\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mread\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[1;32m    175\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[0;32m--> 176\u001b[0m     event \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_receive_event\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    177\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(event, h11\u001b[38;5;241m.\u001b[39mResponse):\n\u001b[1;32m    178\u001b[0m         \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_sync/http11.py:212\u001b[0m, in \u001b[0;36mHTTP11Connection._receive_event\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    209\u001b[0m     event \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_h11_state\u001b[38;5;241m.\u001b[39mnext_event()\n\u001b[1;32m    211\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m event \u001b[38;5;129;01mis\u001b[39;00m h11\u001b[38;5;241m.\u001b[39mNEED_DATA:\n\u001b[0;32m--> 212\u001b[0m     data \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_network_stream\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    213\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mREAD_NUM_BYTES\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\n\u001b[1;32m    214\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    216\u001b[0m     \u001b[38;5;66;03m# If we feed this case through h11 we'll raise an exception like:\u001b[39;00m\n\u001b[1;32m    217\u001b[0m     \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[1;32m    218\u001b[0m     \u001b[38;5;66;03m#     httpcore.RemoteProtocolError: can't handle event type\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    222\u001b[0m     \u001b[38;5;66;03m# perspective. Instead we handle this case distinctly and treat\u001b[39;00m\n\u001b[1;32m    223\u001b[0m     \u001b[38;5;66;03m# it as a ConnectError.\u001b[39;00m\n\u001b[1;32m    224\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m data \u001b[38;5;241m==\u001b[39m \u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_h11_state\u001b[38;5;241m.\u001b[39mtheir_state \u001b[38;5;241m==\u001b[39m h11\u001b[38;5;241m.\u001b[39mSEND_RESPONSE:\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/httpcore/_backends/sync.py:126\u001b[0m, in \u001b[0;36mSyncStream.read\u001b[0;34m(self, max_bytes, timeout)\u001b[0m\n\u001b[1;32m    124\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m map_exceptions(exc_map):\n\u001b[1;32m    125\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sock\u001b[38;5;241m.\u001b[39msettimeout(timeout)\n\u001b[0;32m--> 126\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sock\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrecv\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmax_bytes\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1263\u001b[0m, in \u001b[0;36mSSLSocket.recv\u001b[0;34m(self, buflen, flags)\u001b[0m\n\u001b[1;32m   1259\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m flags \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m   1260\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m   1261\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnon-zero flags not allowed in calls to recv() on \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m\n\u001b[1;32m   1262\u001b[0m             \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__class__\u001b[39m)\n\u001b[0;32m-> 1263\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43mbuflen\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1264\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1265\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39mrecv(buflen, flags)\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1136\u001b[0m, in \u001b[0;36mSSLSocket.read\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m   1134\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mread(\u001b[38;5;28mlen\u001b[39m, buffer)\n\u001b[1;32m   1135\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m-> 1136\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sslobj\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1137\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m SSLError \u001b[38;5;28;01mas\u001b[39;00m x:\n\u001b[1;32m   1138\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m x\u001b[38;5;241m.\u001b[39margs[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;241m==\u001b[39m SSL_ERROR_EOF \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msuppress_ragged_eofs:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "response = Accumulate().synthesize(query, nodes_with_scores)\n",
    "response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "response_synthesizer = Accumulate(\n",
    "    verbose=True, text_qa_template=qa_prompt, output_cls=Biography\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "response = Accumulate().aget_response(query, nodes_with_scores)\n",
    "response"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Stage 1: Getting Chunk Summaries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_title_summary_results(results):\n",
    "  out = []\n",
    "  for e in results:\n",
    "    e = e.replace('\\n', '')\n",
    "    if '|' in e:\n",
    "      processed = {'title': e.split('|')[0],\n",
    "                    'summary': e.split('|')[1][1:]\n",
    "                    }\n",
    "    elif ':' in e:\n",
    "      processed = {'title': e.split(':')[0],\n",
    "                    'summary': e.split(':')[1][1:]\n",
    "                    }\n",
    "    elif '-' in e:\n",
    "      processed = {'title': e.split('-')[0],\n",
    "                    'summary': e.split('-')[1][1:]\n",
    "                    }\n",
    "    else:\n",
    "      processed = {'title': '',\n",
    "                    'summary': e\n",
    "                    }\n",
    "    out.append(processed)\n",
    "  return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index.schema import TextNode\n",
    "from typing import Any, Dict, Generator, List, Optional, Sequence, Union\n",
    "\n",
    "def summarize_stage_1(nodes: List[TextNode]):\n",
    "  \n",
    "  print(f'Start time: {datetime.now()}')\n",
    "\n",
    "  # Prompt to get title and summary for each chunk\n",
    "  map_prompt_template = \"\"\"Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\n",
    "  {text}\n",
    "\n",
    "  Return your answer in the following format:\n",
    "  Title | Summary...\n",
    "  e.g. \n",
    "  Why Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\n",
    "\n",
    "  TITLE AND CONCISE SUMMARY:\"\"\"\n",
    "\n",
    "  map_prompt = PromptTemplate(template=map_prompt_template, input_variables=[\"text\"])\n",
    "\n",
    "  # Define the LLMs\n",
    "  map_llm = OpenAI(temperature=0)\n",
    "  map_llm_chain = LLMChain(llm = map_llm, prompt = map_prompt)\n",
    "  map_llm_chain_input = [{'text': n.text} for n in nodes]\n",
    "  # Run the input through the LLM chain (works in parallel)\n",
    "  map_llm_chain_results = map_llm_chain.apply(map_llm_chain_input)\n",
    "\n",
    "  stage_1_outputs = parse_title_summary_results([e['text'] for e in map_llm_chain_results])\n",
    "\n",
    "  print(f'Stage 1 done time {datetime.now()}')\n",
    "\n",
    "  return {\n",
    "    'stage_1_outputs': stage_1_outputs\n",
    "  }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Start time: 2024-01-10 09:24:17.280194\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/workspaces/ml-learning/.venv/lib/python3.11/site-packages/langchain_core/_api/deprecation.py:189: LangChainDeprecationWarning: The class `OpenAI` was deprecated in LangChain 0.1.0 and will be removed in 0.2.0. Use langchain_openai.OpenAI instead.\n",
      "  warn_deprecated(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stage 1 done time 2024-01-10 09:25:11.418623\n"
     ]
    }
   ],
   "source": [
    "# Run Stage 1 Summarizing\n",
    "stage_1_outputs = summarize_stage_1(nodes)['stage_1_outputs']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "## stage 1 using llamaindex prompt synthesizer"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://docs.llamaindex.ai/en/stable/examples/response_synthesizers/custom_prompt_synthesizer.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index.response_synthesizers import TreeSummarize, Refine\n",
    "from llama_index.types import BaseModel\n",
    "from typing import List"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# try with pydantic model\n",
    "class Summaries(BaseModel):\n",
    "    \"\"\"Data model for a biography.\"\"\"\n",
    "\n",
    "    title: str\n",
    "    text: str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index import PromptTemplate\n",
    "\n",
    "qa_prompt_tmpl = (\n",
    "    \"Context information is below.\\n\"\n",
    "    \"---------------------\\n\"\n",
    "    \"{context_str}\\n\"\n",
    "    \"---------------------\\n\"\n",
    "    \"Given the context information and not prior knowledge, \"\n",
    "    \"answer the query.\\n\"\n",
    "    \"Please also write the answer in the style of {tone_name}.\\n\"\n",
    "    \"Query: {query_str}\\n\"\n",
    "    \"Answer: \"\n",
    ")\n",
    "qa_prompt = PromptTemplate(qa_prompt_tmpl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summarizer = TreeSummarize(\n",
    "    verbose=True, summary_template=qa_prompt, output_cls=Biography\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "response = summarizer.get_response(\n",
    "    \"what is a phage?\", [text], tone_name=\"a business memo\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def summarize_stage_1_llama(nodes: List[TextNode]):\n",
    "  \n",
    "  print(f'Start time: {datetime.now()}')\n",
    "\n",
    "  # Prompt to get title and summary for each chunk\n",
    "  map_prompt_template = \"\"\"Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\n",
    "  {text}\n",
    "\n",
    "  Return your answer in the following format:\n",
    "  Title | Summary...\n",
    "  e.g. \n",
    "  Why Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\n",
    "\n",
    "  TITLE AND CONCISE SUMMARY:\"\"\"\n",
    "\n",
    "  map_prompt = PromptTemplate(template=map_prompt_template, input_variables=[\"text\"])\n",
    "\n",
    "  # Define the LLMs\n",
    "  map_llm = OpenAI(temperature=0)\n",
    "  map_llm_chain = LLMChain(llm = map_llm, prompt = map_prompt)\n",
    "  map_llm_chain_input = [{'text': n.text} for n in nodes]\n",
    "  # Run the input through the LLM chain (works in parallel)\n",
    "  map_llm_chain_results = map_llm_chain.apply(map_llm_chain_input)\n",
    "\n",
    "  stage_1_outputs = parse_title_summary_results([e['text'] for e in map_llm_chain_results])\n",
    "\n",
    "  print(f'Stage 1 done time {datetime.now()}')\n",
    "\n",
    "  return {\n",
    "    'stage_1_outputs': stage_1_outputs\n",
    "  }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### stage_1_outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'title': 'Title ',\n",
       "  'summary': 'Real-Time Abnormalities Detection in Gastrointestinal Tract Using Artificial IntelligenceSummary '},\n",
       " {'title': 'Real-Time Endoscopic Abnormalities Detection System ',\n",
       "  'summary': 'This research presents a system that combines handcrafted and deep features to detect abnormalities and landmarks in real-time endoscopic images. The system uses a genetic algorithm to learn the detection threshold for classes with small inter-class differences. It has been evaluated on benchmark datasets and achieved high accuracy and F1-score. The system has a detection speed of 41 frames per second and was funded by the Higher Education Commission of Pakistan. GI tract cancer is a significant contributor to mortality, and this system can aid in early detection and diagnosis through computer-aided diagnostic techniques.'},\n",
       " {'title': 'Rectum Cancer and Computer-Aided Detection ',\n",
       "  'summary': 'This research presents a three-step model for detecting abnormalities in GI tract images using computer-aided diagnostic (CAD) systems. The first step addresses challenges in the datasets, such as light reflection and class imbalance, while the second step focuses on selecting the most suitable features. The third step uses a neural network and genetic algorithm to compute classes for an image. This model aims to offer a high detection speed and justifiable accuracy for detecting rectum cancer, which is the 8th most dangerous cancer type in the GI tract.'},\n",
       " {'title': 'Improving Endoscopy Detection Accuracy with LiRE-CNN ',\n",
       "  'summary': 'LiRE-CNN is a methodology that combines Lire features and a neural network to improve the accuracy and speed of endoscopy detection. By using a genetic algorithm to compute thresholds for each class, LiRE-CNN achieves a high accuracy of 0.99 with a detection speed of 41 frames per second. The algorithm also performs well on benchmark datasets with and without preprocessing. The article discusses related work, presents the proposed approach, describes the experimental set-up, and concludes with future directions for this research.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Improving Detection Accuracy in Endoscopy: A Proposed MethodologySummary '},\n",
       " {'title': 'Data Augmentation Techniques for Abnormality Detection in GI Tract Images ',\n",
       "  'summary': 'This text discusses various data augmentation techniques used for the detection of abnormalities in gastrointestinal (GI) tract images. These techniques include generative adversarial networks (GANs), angular flips, and adding noise. The detection of abnormalities is done through deep learning and machine learning approaches, with some authors using a combination of both. The use of deep neural networks as feature extractors and support vector machines (SVMs) for classification is also explored. The text highlights the importance of using image features for accurate detection of polyps, cancer, or lesions in the GI tract.'},\n",
       " {'title': '\"Analysis of Approaches for Abnormality Detection and Classification on Benchmark Datasets\" ',\n",
       "  'summary': 'This study examines various methods for detecting and classifying abnormalities on benchmark datasets. The evaluation measures used include sensitivity, specificity, precision, recall, F1-score, accuracy, ROC, and AUC. The best scores for each measure are reported for datasets such as CE Bleeding Dataset, WCE-279 Bleeding Dataset, and Polyp2007. The contributions from different studies, including Zhao et al. (2021), Deeba et al. (2018), and Alexandre, Nobre & Casteleiro (2008), are compared and analyzed. This research provides valuable insights into the effectiveness of different approaches for detecting and classifying abnormalities, which can aid in the development of more accurate and efficient methods for medical diagnosis.'},\n",
       " {'title': '\"Advancements in Abnormality Classification and Landmark Detection in GI Tract Imaging\" ',\n",
       "  'summary': 'This article discusses various approaches and their effectiveness in accurately classifying abnormalities and detecting landmarks in GI tract imaging. These abnormalities can include polyps, ulcers, lesions, esophagitis, and cancer, and precise information about their location is crucial for diagnosis and treatment. The article analyzes multiple methods, such as Adaptive Ensembles and FAST RCNN, and their performance on the Kvasir V2 dataset. The results show high accuracy and F1-scores, indicating the potential of these approaches in improving the accuracy and efficiency of GI tract imaging analysis.'},\n",
       " {'title': '\"Improving Polyp Detection in Endoscopic Images ',\n",
       "  'summary': 'Researchers use various techniques to enhance the performance of polyp detection in endoscopic images, including combining datasets, using different neural networks, and implementing majority voting and weighted discriminant embedding methods. These methods have shown promising results, with some achieving high accuracy rates of 0.99 and 0.95. The Kvasir V2 dataset, introduced by Pogorelov et al. in 2018, is commonly used in these studies. Overall, these advancements in artificial intelligence and image processing have the potential to greatly improve the accuracy and efficiency of polyp detection in endoscopic procedures.\"'},\n",
       " {'title': '\"Improving Abnormality and Anatomical Landmark Detection in Endoscopic Images ',\n",
       "  'summary': 'This text discusses various approaches and a proposed methodology for detecting abnormalities and landmarks in endoscopic images. These approaches include using deep learning models such as Inception Res-Net and DenseNet, as well as handcrafted feature extraction and genetic algorithms for threshold detection. The proposed methodology consists of seven main components, including data preprocessing and augmentation to remove reflections from the images. This is crucial in ensuring accurate analysis and detection of abnormalities and landmarks, as demonstrated in the sample set of polyps picture provided.\"'},\n",
       " {'title': 'Removing Reflections in Endoscopic Images ',\n",
       "  'summary': 'Reflections in endoscopic images can cause errors in image analysis and affect detection of abnormalities and landmarks. Various methods, such as data augmentation, handcrafted and deep feature extraction, feature combination and selection, network-based classification, and genetic algorithm for threshold detection, can be used to remove these reflections. These methods can be performed with or without human intervention, and involve procedures such as image cropping and supervised or unsupervised detection. Equations are also used to aid in the removal of reflections.'},\n",
       " {'title': ' Image Thresholding for Edge Detection ',\n",
       "  'summary': 'This text discusses a method for detecting edges in an image using a thresholding technique. The algorithm assigns a value of 255 to pixels with a high intensity and a value of 0 to pixels with a low intensity, based on a threshold of 180. This allows for the identification of edges in the image, which can be useful for various applications such as image segmentation and feature extraction.'},\n",
       " {'title': 'Data Augmentation for Endoscopic Procedures ',\n",
       "  'summary': 'This article discusses the use of data augmentation techniques to address the class imbalance problem in endoscopic procedure datasets. These datasets often contain a few common abnormalities and a large number of rare ones, leading to imbalanced data. The authors propose using data preprocessing and augmentation methods to increase the representation of rare abnormalities and improve the performance of machine learning algorithms. This approach has the potential to improve the accuracy and effectiveness of endoscopic procedures, ultimately benefiting patients and medical professionals.'},\n",
       " {'title': 'Data Augmentation Techniques for Imbalanced Endoscopic Datasets ',\n",
       "  'summary': 'This text discusses the issue of class imbalance in endoscopic datasets and the common approaches for data augmentation to address this problem. These techniques include using generative models, rotating, flipping, cropping, resizing, and noise injection. The image manipulation is done as a post-step of reflection removal and a pre-step to feature extraction and fine-tuning of the neural network. These techniques can help improve the performance of models trained on imbalanced datasets, such as the Kvasir V2 dataset which has a significant class imbalance.'},\n",
       " {'title': 'Texture Feature Extraction for Image Manipulation Detection ',\n",
       "  'summary': 'This research explores the use of texture features, specifically those available in Lucene Image Retrieval (LIRE), as effective deciders for detecting image manipulation. By applying a decision tree classifier on various sets of features, the average detection F1-score was found to be above 0.84. The methodology of the research involves using count(C) and count(Ci) to generate a set of features for the classifier. This study highlights the potential of texture features in accurately detecting image manipulation, which can have important implications in fields such as forensics and security.'},\n",
       " {'title': 'Texture Features in Image Retrieval ',\n",
       "  'summary': 'This text discusses the various texture features available in Lucene Image Retrieval (LIRE) and their use in detecting abnormalities and localizing and segmenting images. These features include color layout, edge histogram, Tamura, Color and Edge Directivity Descriptor (CEDD), Fuzzy color and texture histogram (FCTH), and color histograms (HSV and RGB). The methodology of the research is also outlined, with references to previous studies and their findings. These texture features play a crucial role in image retrieval and can aid in the accurate analysis and interpretation of images.'},\n",
       " {'title': '  \"Exploring Image Feature Extraction Techniques\" ',\n",
       "  'summary': 'This text discusses various techniques for extracting features from images, including Gabor features, auto color correlation, and auto color correlogram. These techniques are used to analyze and compare images based on their pixel values. The authors also mention previous studies that have utilized these techniques, such as Mehrotra, Namuduri & Ranganathan (1992) and Huang et al. (1997).'},\n",
       " {'title': '   \"Calculating Distance and Accuracy in Data Analysis\" ',\n",
       "  'summary': 'This text discusses the use of mathematical equations and functions to calculate the accuracy and distance between data points in data analysis. It introduces the concept of ACC, which measures the probability of two data points being in the same category and at a certain distance from each other. It also explains the use of the equation d(P,Q) to determine the maximum distance between two data points in terms of their x and y coordinates. This text provides a brief overview of the mathematical tools used in data analysis.'},\n",
       " {'title': ' Feature Extraction Techniques in Computer Vision ',\n",
       "  'summary': 'This text discusses three popular feature extraction techniques in computer vision: Speeded up robust features (SURF), Pyramid Histogram of Oriented Gradients (PHOG), and Local Binary Patterns (LBP). These techniques are used to extract important information from images and are widely used in various computer vision applications. The text also mentions other related techniques such as Local Ternary Patterns (LTP).'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Gray-Level Co-Occurrence Matrix (GLCM): Haralick & Shanmugam, 1973Summary '},\n",
       " {'title': '   Gray-Level Co-Occurrence Matrix (GLCM) ',\n",
       "  'summary': 'GLCM is a method used in image processing to analyze the spatial relationship between pixels of different gray levels. It counts the occurrences of pixel pairs at a certain distance and direction, providing information about the texture and patterns in an image. This technique was first introduced by Haralick and Shanmugam in 1973 and has since been widely used in various applications such as texture classification and feature extraction. GLCM is a powerful tool for understanding the structure of an image and has been continuously improved and adapted for different purposes in the field of computer vision.'},\n",
       " {'title': '  \"Haralick Features for Abnormality Detection in Endoscopy and Colonoscopy Images\" ',\n",
       "  'summary': 'This text discusses the use of Haralick features, a type of statistical analysis, in detecting abnormalities in medical images. These features are calculated using a GLCM (Gray-Level Co-occurrence Matrix) and can provide valuable information for identifying abnormalities in endoscopy and colonoscopy images. Various CNN-based approaches are commonly used in conjunction with Haralick features to improve the accuracy of abnormality detection in these medical images.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Achieving High Detection Accuracies with Dense NetSummary '},\n",
       " {'title': 'Deep Feature Extraction Methodology for Image Classification ',\n",
       "  'summary': 'This text discusses a methodology for extracting deep features from images for classification purposes. The approach involves using a voting classifier with a combination of various features, resulting in good detection and optimal classification time. The final set of features that yielded the best accuracies include Auto Color Correlogram, Color Layout, Edge Histogram, Gabor Features, JCD, Color and Edge Directivity Descriptor, Fuzzy Color and Texture Histogram, Tamura Features, Local Binary Patterns, and Deep Features. The use of neural networks as classification networks is also discussed, highlighting their effectiveness in decision-making when sufficient data is available.'},\n",
       " {'title': '\"Understanding Relu and Sigmoid Activation Functions\" ',\n",
       "  'summary': 'Relu and sigmoid are two commonly used activation functions in neural networks. Relu is a simple function that outputs the maximum of 0 and the input value, while sigmoid is a more complex function that outputs a value between 0 and 1. These functions are used to introduce non-linearity in neural networks and are particularly useful in the final output layer for classification tasks.'},\n",
       " {'title': 'Optimizing Neural Networks with Nadam ',\n",
       "  'summary': 'Various optimizers were evaluated for neural networks, and Nadam was found to have good weight adjustment and fast convergence. Nadam is shown in Eq. (17) as a combination of different parameters, including learning rate and momentum.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Gradient Descent Algorithm for Machine LearningSummary '},\n",
       " {'title': ' Improving Detection Accuracy with Neural Networks and Genetic Algorithms ',\n",
       "  'summary': 'This text discusses the use of a neural network-based approach and genetic algorithm for threshold detection in medical image classification. The approach resulted in a boost in F1-score and Matthews Correlation Coefficient, improving the accuracy of detection. The genetic algorithm was used to learn thresholds for each class, taking into account the differences in intra-class and inter-class variations. The crossover operator employed in the algorithm ensures that the resulting number lies within the range of 0 to 1. This approach has potential for improving detection accuracy in medical imaging.'},\n",
       " {'title': '   The Crossover Operator in GA-Boost ',\n",
       "  'summary': 'This text discusses the use of a specific crossover operator in GA-Boost, which involves removing the integral part of a floating-point number to ensure it falls within a specific range. The text also provides details on the experimental setup used for analyzing endoscopic and colonoscopic abnormalities, including datasets and evaluation measures.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Overview of Endoscopic Image DatasetsSummary '},\n",
       " {'title': 'Evaluation Measures for Segmentation in Medical Imaging ',\n",
       "  'summary': 'This text discusses various evaluation measures used in medical imaging research, including accuracy, F1-score, Matthews Correlation Coefficient, and sensitivity. The DowPK dataset is highlighted as having the highest sensitivity value among evaluated datasets, reaching an impressive value of 0.93.'},\n",
       " {'title': 'Evaluation Measures for Segmentation in DowPK Dataset ',\n",
       "  'summary': \"This text discusses various evaluation measures used to assess the performance of segmentation in the DowPK dataset. These measures include accuracy, F1-score, Matthews Correlation Coefficient, sensitivity, specificity, and area under the receiver operating characteristic curve. The DowPK dataset demonstrated high values for sensitivity and specificity, indicating the model's proficiency in correctly identifying positive and negative instances. The Kvasir V2 and Hyper Kvasir datasets also showed excellent performance in distinguishing between classes, with a high AUC-ROC value of 0.99. The minimum possible value for the MCC is -1, indicating all false results, while a value of 1 indicates all correct classifications.\"},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Evaluating Classification Methods for Abnormalities and Landmarks Detection on Medical DatasetsSummary '},\n",
       " {'title': 'Comparing Methods for Classification of Abnormalities and Landmarks Detection ',\n",
       "  'summary': 'This section discusses different variations of proposed methods and compares their performance on benchmark datasets. The comparison is shown in Tables 4, 5, 6 of the article \"Analysis of some of the approaches for classification of abnormalities and landmarks detection on the Kvasir V1 dataset\" and \"Analysis of various approaches for classification of abnormalities and landmarks detection on the Kvasir V2 dataset\". The tables show the accuracy, F1-score, MCC, and FPS of each approach, with the proposed Lire-CNN achieving the highest scores on the Kvasir V1 dataset and the Kvasir V2 dataset. '},\n",
       " {'title': 'Performance Comparison of Abnormalities and Landmarks Detection Approaches on Kvasir V3 Dataset ',\n",
       "  'summary': 'This table presents the results of various approaches for classifying abnormalities and detecting landmarks on the Kvasir V3 (Hyper Kvasir) dataset. The approaches include EEIC19, TLCSS18, FRCSS18, DSTL18, WDE18, MVHC18, and the proposed Lire-CNN. The results show that the proposed approach has the highest accuracy, F1-score, and Matthews correlation coefficient (MCC) among all the approaches. It also has a relatively low frames per second (FPS) rate, indicating its efficiency. This information can be useful for researchers and practitioners in the field of medical image analysis.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Impact of Different Approaches on Abnormalities Detection in Medical ImagesSummary '},\n",
       " {'title': 'The Impact of Reflection Removal on Abnormality Detection ',\n",
       "  'summary': 'Reflection removal techniques were applied to multiple datasets, including Kvasir V1, V2, and V3, resulting in varying levels of improvement in F1-scores. The number of images and prevalence of reflections within a dataset were found to influence the effectiveness of reflection removal. Datasets with a higher number of images and prevalence of reflections experienced a more significant impact from reflection removal techniques.'},\n",
       " {'title': 'Impact of Reflection Removal and Data Augmentation on Image Detection ',\n",
       "  'summary': 'This study examines the impact of reflection removal and data augmentation techniques on image detection. The number of images per class and the amount of reflections present were found to influence the effectiveness of reflection removal. Data augmentation was also applied to address class imbalance, resulting in significant improvements in detection for datasets with a higher prevalence of reflections. Additionally, the use of deep features with texture-based features was found to greatly improve accuracy and F1-score on the Kvasir V2 dataset. These findings highlight the importance of considering factors such as image class size and reflection prevalence when implementing image detection techniques.'},\n",
       " {'title': '\"Feature Selection for Improved Accuracy in Medical Image Classification ',\n",
       "  'summary': 'Researchers use a combination of deep and texture-based features to achieve high accuracy in classifying medical images. By applying this approach to the Kvasir V2 dataset, they achieved an accuracy of 0.99 and an F1-Score of 0.86. However, certain features such as Color Histogram and Auto Color Correlation were found to be misleading. The features used for selection include Auto Color Correlogram, Color Layout, Edge Histogram, Gabor, JCD, PHOG, Tamura, LBP, and a fine-tuned Light Weight Network. This study highlights the importance of carefully selecting features for accurate medical image classification.\"'},\n",
       " {'title': 'Improving Classification Accuracy with GA-Boost Algorithm ',\n",
       "  'summary': 'This text discusses the use of various features and neural classifiers for image classification, resulting in a commendable F1-score of 0.89 for the Kvasir V1 dataset. However, significant variations in probability ranges for different classes highlight the need for tailored class thresholds. To address this, a genetic algorithm named GA-Boost was employed, allowing for the precise adaptation of thresholds for each class and improving the overall classification accuracy.'},\n",
       " {'title': 'Optimizing Classification Thresholds with GA-Boost Algorithm ',\n",
       "  'summary': 'The GA-Boost algorithm was used to generate optimal thresholds for different classes in a dataset, taking into account variations in probability distributions. This allowed for more precise adaptation of thresholds based on the specific characteristics of each class, resulting in more effective classification. The algorithm utilized a genetic algorithm and an additive crossover operator to learn the thresholds for each class. The F1-score evaluation measure was used to select the best chromosomes, and the results showed a significant impact on the F1-score when applying various additions in the procedure on different datasets.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Impact of Crossover Operator and Feature Selection on GA-Boost Algorithm PerformanceSummary '},\n",
       " {'title': '\"Efficiency and Accuracy of LiRE-CNN for Real-Time Applications\" ',\n",
       "  'summary': 'This study evaluated the performance of LiRE-CNN, a deep learning algorithm, on various datasets for abnormalities and landmark detection. The results showed high accuracy and impressive detection speed of 41 frames per second (FPS), making it efficient for real-time applications. The addition of certain algorithmic steps significantly improved the overall F1-score, with varying effects on different datasets. The removal of reflections and use of different thresholds also led to improved detection accuracy. These findings demonstrate the effectiveness of a tailored approach for each dataset in enhancing the performance of LiRE-CNN.'},\n",
       " {'title': 'Improving Polyp Detection in Medical Images ',\n",
       "  'summary': 'This study analyzed the effectiveness of a tailored approach for improving polyp detection in medical images. The results showed that removing reflections and using different thresholds for all classes led to better detection and reduced misclassification. However, there were still challenges due to class imbalance and difficulty in differentiating similar classes. The study also highlighted the impact of data augmentation and the need for more samples in certain classes. Overall, the findings suggest that a tailored approach can improve polyp detection, but further research is needed to address other challenges in medical image analysis.'},\n",
       " {'title': 'Challenges in Detecting GI Tract Abnormalities ',\n",
       "  'summary': 'The Kvasir V2 dataset contains classes with very few samples, making it difficult to differentiate between similar images. The proposed LiRE-CNN approach is compared to other state-of-the-art methods for detecting GI tract abnormalities, with a focus on the challenging classes of dyed lifted polyps and dyed resection margins, as well as the misclassification between esophagitis and normal z-line. The results show that the LiRE-CNN approach outperforms other methods and can potentially improve detection accuracy for these challenging classes.'},\n",
       " {'title': 'Comparative Analysis of Approaches for GI Tract Abnormality Detection ',\n",
       "  'summary': 'This study evaluates various methods for detecting abnormalities and landmarks in the GI tract, with a focus on reflection elimination. The most successful approaches were found to be unsupervised methods, which align with dataset requirements and task objectives. The issue of class imbalance was also addressed effectively. Detecting landmarks and abnormalities can aid in the treatment of diseases caused by these irregularities, such as polyps or cancer, by accurately pinpointing and localizing the affected area. '},\n",
       " {'title': 'Preprocessing Techniques for Endoscopic Abnormality Detection ',\n",
       "  'summary': 'This research explores the use of supervised and unsupervised preprocessing methods for detecting abnormalities and landmarks in endoscopic images. The study found that unsupervised methods were most effective in aligning with dataset requirements and task objectives. Additionally, the challenge of class imbalance was addressed, which can aid in the treatment of diseases caused by these abnormalities. The future task is to localize and segment the abnormal regions, particularly in cases of polyps or cancer. The code and datasets used in the research are available for further study.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'DowPK: A Dataset for Endoscopic Polyp DetectionSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'HyperKvasir: A Comprehensive Dataset for Gastrointestinal EndoscopySummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Advances in Gastrointestinal Tract Disease DetectionSummary '},\n",
       " {'title': 'Performance Assessment of a Bleeding Detection Algorithm for Endoscopic Video ',\n",
       "  'summary': \"This text discusses a study that evaluated the performance of a bleeding detection algorithm for endoscopic videos. The algorithm used a classifier fusion method and exhaustive feature selection to accurately detect bleeding in endoscopic videos. The study compared the algorithm's performance to other methods and found it to be highly effective. The authors also discuss the potential applications of this algorithm in medical imaging and the future potential for computer-aided diagnosis. This text provides valuable insights into the use of artificial intelligence in medical imaging and its potential to improve diagnostic accuracy.\"},\n",
       " {'title': 'Advancements in Medical Image Analysis for Gastrointestinal Diseases ',\n",
       "  'summary': 'This text discusses various techniques and approaches for analyzing medical images of the colon and gastrointestinal tract. It covers topics such as quantitative measurement, feature identification, and the use of convolutional neural networks. The authors also explore the effectiveness of image augmentations and the use of genetic algorithms for optimizing neural networks. This information is relevant for the diagnosis and treatment of gastrointestinal diseases, such as cancer.'},\n",
       " {'title': 'Automatic Generation of Optimized Convolutional Neural Networks for Medical Image Classification ',\n",
       "  'summary': 'This text discusses the use of a genetic algorithm to automatically generate optimized convolutional neural networks for medical image classification. The authors draw on previous research in the field, including the use of generative adversarial nets and deep residual learning, to develop their approach. They also highlight the potential applications of this technology, such as automatic disease detection and report generation for gastrointestinal tract examination. The text also discusses the use of support vector machines in this context.'},\n",
       " {'title': 'Advancements in Artificial Intelligence for Disease Recognition in Gastrointestinal Endoscopy ',\n",
       "  'summary': 'This text discusses various methods and techniques for using artificial intelligence, specifically deep learning and convolutional neural networks, to improve disease recognition in gastrointestinal endoscopy. These methods include hybrid loss with network trimming, support vector machines, domain specific transfer learning, residual networks, and data augmentation. The text also mentions the use of mobile vision applications and densely connected convolutional networks. These advancements have the potential to greatly improve the accuracy and efficiency of disease detection in endoscopy procedures.'},\n",
       " {'title': 'Advancements in Gastrointestinal Endoscopy Imaging ',\n",
       "  'summary': 'This text discusses various studies and methods for automatic detection of gastrointestinal abnormalities in endoscopy videos. It covers topics such as densely connected convolutional networks, image indexing using color correlograms, and polyp detection using elliptical shape features. It also includes an intelligent system for automatic detection of gastrointestinal adenomas. The text highlights the importance of using advanced classification methods, such as artificial intelligence, in improving the accuracy and efficiency of detecting abnormalities in endoscopy imaging.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Advancements in Polyp Segmentation and Covid-19 Tissue Classification in Medical ImagingSummary '},\n",
       " {'title': 'Medical Image Classification Techniques ',\n",
       "  'summary': 'This text discusses various techniques for medical image classification, including data bagging, majority voting of heterogeneous classifiers, and weighted discriminant embedding. It also explores the use of preprocessing as a tool in medical image detection. The text references several studies and conferences, such as MediaEval and the International Conference on Pattern Recognition, and discusses the use of deep convolutional neural networks and dominant local binary patterns for texture classification. These techniques have the potential to improve medical diagnosis and treatment by accurately identifying abnormalities in medical images.'},\n",
       " {'title': 'Medical Image Classification and Retrieval Techniques ',\n",
       "  'summary': 'This text discusses various techniques for medical image classification and retrieval, including the use of spatial adjacent histograms based on adaptive local binary patterns, adaptive ensembles, and Gabor filters. It also mentions open source image retrieval libraries and their applications in medical multimedia tasks. The text highlights the importance of accurate image classification and retrieval in the medical field and provides insights into the latest developments in this area.'},\n",
       " {'title': 'Advancements in Multimedia Technology for Gastrointestinal Disease Detection ',\n",
       "  'summary': 'This text discusses various studies and challenges related to using multimedia technology for detecting abnormalities in the gastrointestinal tract. These include using data enhancement techniques to address sample imbalance, ensemble of texture features, and an inception-like CNN architecture. The studies also highlight the importance of high-quality video datasets for training and testing these systems. These advancements have the potential to improve the accuracy and efficiency of detecting gastrointestinal diseases, ultimately leading to better patient outcomes.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Advancements in Multimedia for Gastrointestinal Disease DetectionSummary '},\n",
       " {'title': 'Advancements in Multimedia and Image Recognition ',\n",
       "  'summary': 'This text discusses various research studies and conferences related to multimedia and image recognition, including the use of deep learning and data augmentation techniques. It highlights the importance of these technologies in the medical field and their potential for improving image classification. The text also mentions the development of new methods, such as Mobilenetv2 and the MPEG-7 visual standard, for more efficient and accurate image recognition. Overall, the text showcases the ongoing efforts and progress in the field of multimedia and image recognition, and their potential impact on various industries.'},\n",
       " {'title': 'Global Cancer Statistics and the Role of Computer-Aided Diagnosis ',\n",
       "  'summary': 'This text discusses the latest global cancer statistics for 2020, including estimates of incidence and mortality for 36 different types of cancer in 185 countries. It also explores the use of computer-aided diagnosis in thoracic and colonic imaging, specifically in the context of cancer detection. The text reviews various studies and techniques, such as deep transfer learning models, that have been used to improve the accuracy and efficiency of cancer diagnosis. This information is crucial for understanding the current state of cancer worldwide and the potential impact of technology in improving detection and treatment.'},\n",
       " {'title': 'Automatic Polyp Detection Using Global Geometric Constraints and Local Intensity Variation Patterns ',\n",
       "  'summary': 'This text discusses a method for automatically detecting polyps in medical images using a combination of global geometric constraints and local intensity variation patterns. The authors propose a deep transfer learning model called CovidSORT for detecting COVID-19 in chest X-ray images. They also explore the use of textural features and enhanced local texture feature sets for face recognition and inpainting techniques based on the fast marching method. Additionally, they introduce a dual decoder attention network for automatic polyp segmentation and discuss the visualization and understanding of convolutional networks.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Recent Advances in Automatic Polyp Segmentation for Medical ImagingSummary '},\n",
       " {'title': 'External Viewer Error: Empty Response ',\n",
       "  'summary': 'This error occurs when an external viewer fails to load due to a timeout, resulting in an empty response with zero bytes read. This can be caused by a slow or unresponsive server, network connectivity issues, or incorrect configuration settings. It can prevent users from accessing important information and may require troubleshooting to resolve.'},\n",
       " {'title': 'Understanding Methicillin-Resistant Staphylococcus Aureus (MRSA) ',\n",
       "  'summary': 'This article discusses the pathogenic bacteria Staphylococcus aureus and its methicillin-resistant strain, MRSA. It explores the impact of MRSA on human health, including its role in sepsis and cytotoxicity. The article also delves into the genetic mutations and molecular biology techniques used to study and combat MRSA, such as DNA construction and plasmid construction. Overall, this article provides a comprehensive overview of the biology and medical implications of MRSA, shedding light on the urgent need for continued research and effective treatment methods.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Role of Staphylococcus aureus Lipoproteins in Toxin ProductionSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Role of CamS in Repressing Toxin Production in Staphylococcus aureusSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Role of CamS Protein in Repressing S. aureusSummary '},\n",
       " {'title': '   Regulation of Toxins in Staphylococcus aureus and Enterococcus faecalis ',\n",
       "  'summary': 'This text discusses the regulation of toxins in Staphylococcus aureus and Enterococcus faecalis through the use of transcriptional reporters. The construction of promoter fusion plasmids was based on predicted transcription start sites, and the regulation of distinct toxins was validated. The study found that the toxins hla and hlgCB were regulated by the presence of certain genes, such as camS and saePQRS. This information can be useful in understanding the mechanisms of toxin regulation in these bacteria and potentially developing treatments for infections caused by them.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Construction and Regulation of Toxins in Staphylococcus aureusSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'CamS-Mediated Toxin Regulation in S. aureusSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Role of CamS Protein in S. aureus Virulence Factor RegulationSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Rabbit Erythrocytes Exhibit Greater Sensitivity to α-Toxin Exposure Compared to Human ErythrocytesSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Experimental Methods for Studying Staphylococcus Aureus Infection in MiceSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Construction and Analysis of Promoter Fusion Plasmids in Staphylococcus aureusSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'RNA-seq Analysis of USA300 FPR3757 Reference GenomeSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Analysis of MRSA USA300 FPR3757 Gene Expression in Murine ModelsSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Statistical Tests and Supporting Data in a Study on Bacterial Strains and PlasmidsSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Bacterial Strains and Plasmids Used in Staphylococcus Aureus StudySummary '},\n",
       " {'title': 'The Role of Nasal Carriage in Staphylococcus Aureus ',\n",
       "  'summary': 'This article discusses the importance of nasal carriage in the spread and virulence of Staphylococcus aureus, a common and potentially dangerous bacteria. It highlights the role of lipoproteins in the immune response and virulence of S. aureus, and identifies potential therapeutic targets for treating methicillin-resistant strains. The article also provides an overview of two-component systems in S. aureus and their impact on gene expression. Overall, this article sheds light on the complex mechanisms involved in the pathogenesis of S. aureus and the potential for targeted interventions to combat this bacteria.'},\n",
       " {'title': 'The Role of Lipoproteins in Gram-Positive Bacteria ',\n",
       "  'summary': 'This article discusses the abundance, function, and fitness of lipoproteins in Gram-positive bacteria, specifically focusing on Staphylococcus aureus. It also explores the role of lipoproteins in triggering host defense mechanisms through toll-like receptors and the differential immune responses they elicit in commensal and non-commensal staphylococci. Additionally, the article delves into the evolutionary pathways of enterococcal sex pheromones, which play a crucial role in complex, two-signal systems. Overall, this article sheds light on the importance of lipoproteins in the biology and pathogenesis of Gram-positive bacteria.'},\n",
       " {'title': 'The Evolution of Complex Pheromone Systems in Bacterial Pathogens ',\n",
       "  'summary': 'This article discusses the evolutionary pathways of two-signal systems in bacterial pathogens, specifically focusing on enterococcal and staphylococcal species. It also explores the processing, export, and identification of novel linear peptides from Staphylococcus aureus, as well as the identification of a peptide-pheromone that enhances Listeria monocytogenes. The article also touches on the role of these pheromone systems in community-associated methicillin-resistant Staphylococcus aureus (MRSA) and provides a refined protocol for mapping transcription start sites in bacterial pathogens.'},\n",
       " {'title': 'Refined Protocol for Mapping Transcription Start Sites in Bacterial Pathogens ',\n",
       "  'summary': 'This article discusses a new protocol for mapping transcription start sites in bacterial pathogens, which aims to provide a more complete and less biased understanding of these sites. The protocol was developed by Giraudo et al. and published in BMC Genomics in 2016. It involves the use of the sae Staphylococcus aureus protein complex, which activates the phosphatase activity of the sensor kinase SaeS in the SaeRS two-component system. This system plays a crucial role in the pathogenesis of Staphylococcus aureus, and the protocol has been shown to accurately identify transcription start sites in this bacterium.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Role of Hemolysins in Staphylococcus aureus InfectionsSummary '},\n",
       " {'title': 'Targeting Staphylococcus aureus: Understanding the Role of Neutrophils and Bacterial Virulence ',\n",
       "  'summary': 'This text discusses recent research on the role of neutrophils in defending against Staphylococcus aureus infections, as well as the impact of bacterial virulence on the severity of MRSA sepsis. The studies highlighted in this text provide valuable insights into potential targets for treatment and prevention of S. aureus infections.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Prevalence and Epidemiological Data on Staphylococcus aureus and its Toxin GenesSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Role of Lipoprotein QseG in Controlling the Phosphorylation State of the Two-Component System in Staphylococcus aureusSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Role of Adherens Junctions in Staphylococcus aureus InfectionSummary '},\n",
       " {'title': '   The Role of Toxins in Promoting Staphylococcus aureus Infection ',\n",
       "  'summary': 'This text discusses the various methods and techniques used to study and manipulate Staphylococcus aureus, a common bacterial pathogen. It highlights the role of toxins and lipopolysaccharides in promoting infection and modulating neutrophil functions. The text also discusses genetic systems and resources used for rapid phenotype screening and transformation of Staphylococcus aureus and Staphylococcus epidermidis. It concludes with a protocol for efficient plasmid mutagenesis.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'A Protocol for Mutagenesis and its Impact on MRSA Virulence and Host DefenseSummary '},\n",
       " {'title': 'The Importance of Open-Source Platforms in Biological Image Analysis ',\n",
       "  'summary': 'This text discusses the significance of open-source platforms, such as Fiji, in the field of biological image analysis. It also mentions the role of the ubiquitous human skin commensal Staphylococcus hominis in mBio and the Kyoto Encyclopedia of Genes and Genomes (KEGG) in Nucleic Acids Research. These resources provide valuable information and tools for researchers in the biological sciences. The text highlights the potential for open-source platforms to enhance collaboration and advance scientific discoveries.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'A Novel In-Situ-Process Technique for Constructing Circular cpDNA LibrarySummary '},\n",
       " {'title': '   \"Advancements in Constructing High-Quality Circular Chloroplast DNA Libraries\" ',\n",
       "  'summary': 'This paper discusses a novel in-situ-process technique for efficiently extracting circular cpDNA from crops and constructing a high-quality cpDNA library. This technique combines in-situ chloroplast lysis and in-situ substitute/ligation to prevent large-insert cpDNA fragment breakage. The resulting goal-insert, ordered cpDNA library is suitable for various genome-wide functional analyses and characterizations of chloroplasts. This advancement has potential implications for genome sequencing, bioinformatics analysis, cloning, physical mapping, molecular phylogeny, and evolution.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'In-situ Lysis for Studying Chloroplast Genome ConformationsSummary '},\n",
       " {'title': '\"Efficient Extraction and Cloning of Circular cpDNA Using BAC Library and In-Situ Substitute/Ligation Protocol\" ',\n",
       "  'summary': 'This article discusses a new method for extracting and cloning high-quality circular cpDNA using a BAC library and an in-situ substitute/ligation protocol. This method allows for a more efficient and accurate extraction of cpDNA, reducing the number of colonies needed to represent a complete DNA library. The protocol also ensures the integrity of the cpDNA sequence within a length of 90.0 kb. This method has been successfully used to construct goal-insert, ordered cpDNA libraries, and can also be applied to other nucleic acid sequences. The materials and methods used in this experiment are listed in Table 1.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Techniques for Constructing cpDNA LibrariesSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Techniques for Extracting Circular cpDNA from Plant LeavesSummary '},\n",
       " {'title': 'Optimizing Chloroplast Isolation for Plant Samples ',\n",
       "  'summary': 'This text provides guidelines for optimizing chloroplast isolation from different plant samples. It includes information on the appropriate sampling stages for wheat, maize, soybean, and barley, as well as the recommended immersing times for each sample type. It also emphasizes the importance of using freshly made ET-M-D buffer and keeping samples on ice during all operations. The text cautions against using liquid nitrogen or dry ice for lyophilization and recommends filtering the disintegrating cell solution to eliminate non-target contaminants. Overall, these guidelines aim to ensure the immaculateness of equipment and reagents and to prevent the destruction of intact chloroplast membranes during homogenization.'},\n",
       " {'title': 'Optimizing Chloroplast Isolation for Accurate Results ',\n",
       "  'summary': 'This text provides guidelines for obtaining pure and intact chloroplasts for experiments. It emphasizes the importance of using immaculate equipment and pre-cooled reagents. The text also highlights the impact of grinding conditions on chloroplast isolation and recommends against homogenizing tissues in liquid nitrogen. To eliminate non-target contaminants, the text suggests filtering the disintegrating cell solution and withdrawing suspended chloroplasts using a serological pipette. It also provides instructions for in-gel chloroplast lysis and advises against exceeding the specified time limit to preserve intact chloroplasts. The text emphasizes the need for careful and gentle mixing to protect the chloroplast membrane.'},\n",
       " {'title': 'Optimizing Chloroplast Lysis for Circular cpDNA Extraction ',\n",
       "  'summary': 'This text provides guidelines for effectively lysing chloroplasts and extracting circular cpDNA. The experimenter is advised to carefully mix the chloroplast suspension and LMP agarose to avoid damaging the chloroplast membrane. The agarose matrix should be kept intact and all buffers should have a pH below 8.0. After lysis, the circular cpDNA can be released using a 1% PFGE-gel under specific conditions. These guidelines ensure successful extraction and purification of circular cpDNA for further analysis.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Efficient Extraction and Purification of Circular cpDNA from Agarose MatrixSummary '},\n",
       " {'title': 'Optimizing Electroelution for Trapping Circular cpDNA ',\n",
       "  'summary': 'This text outlines the steps for electroelution and trapping of circular cpDNA, including the use of Tris-HCl running buffer and avoiding bubbles. The optimal electroelution time must be determined for each sample and gel concentration. The trapped solution is then treated with reagents to remove impurities, and the resulting circular cpDNA can be stored for up to 6 months. The text also mentions the importance of avoiding foaming during the process. Finally, the text briefly mentions the preparation of the vector for in-situ substitution or ligation.'},\n",
       " {'title': 'Optimizing Vector/Insert Recombination for Clonable DNA Fragments ',\n",
       "  'summary': 'This text discusses a protocol for preparing a vector for cloning DNA fragments, with a focus on avoiding foaming and ensuring complete digestion and dephosphorylation. The protocol involves using mono-restriction endonucleases to digest circular cpDNA from Wheat, Maize, Soybean, and Barley, and then using an in-situ substitute/ligation method for efficient ligation of vector/insert recombinant DNA. The protocol also highlights the importance of selecting suitable endonucleases based on factors such as balance between number and length of digestion products, stability, and ease of procurement. Additionally, the text suggests adding Klenow Fragment to the digestion reaction if the fragments have cohesive ends.'},\n",
       " {'title': 'In-situ Substitute and Ligation Protocol for Efficient Vector/Insert Recombinant DNA Construction ',\n",
       "  'summary': 'This protocol involves using a high ligation efficiency method for constructing vector/insert recombinant DNA while maintaining the integrity of the cpDNA sequence. It also recommends adding Klenow Fragment to the enzymatic digestion reaction and using an osmotic alternative for the enzyme-digested buffer. The use of phenol for cpDNA extraction is not recommended. The protocol emphasizes the importance of sterile processes and using freshly good-quality cpDNA fragments for higher transformation efficiency. The in-situ ligation step involves carefully cutting and inserting the MCE membrane into RNase-free Microfuge Tubes.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Tips for Successful Construction of Circular Chloroplast DNA LibrariesSummary '},\n",
       " {'title': '\"Primer Sets for Verifying BCA Libraries\" ',\n",
       "  'summary': 'This text provides a list of primer sets for verifying the BCA libraries, including the sequence, PCR product size, and combinations of forward and reverse primers for each set. These primer sets can be used to confirm the presence of specific colonies in the BCA libraries, aiding in the identification and analysis of genetic material.'},\n",
       " {'title': 'PCR Primer Design for Junction Site Targeting ',\n",
       "  'summary': 'This text outlines the PCR primer design for targeting junction sites in DNA sequences. The primer sequences and their corresponding junction sites are listed, along with the vector and forward/reverse primer combinations used for each PCR reaction. This information is important for researchers conducting PCR experiments to specifically amplify and study certain regions of DNA.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Verifying the Quality of a Soybean cpDNA LibrarySummary '},\n",
       " {'title': '\"Efficient Extraction and Ligation Methods for High-Quality cpDNA Library Construction\" ',\n",
       "  'summary': 'This text discusses a modified protocol for extracting circular cpDNA from crops, using electroelution and the RecBCDase system to remove linear nucleic acid impurities. The in-situ substitute/ligation method is also described, which allows for high ligation efficiency of vector/insert recombinant DNA while maintaining the integrity of the cpDNA sequence. This method is crucial for constructing a high-quality cpDNA library, which is necessary for various genetic studies. The text also mentions the importance of verifying the quality of the cpDNA library through PCR analysis.'},\n",
       " {'title': '   Advancements in Circular cpDNA Research ',\n",
       "  'summary': 'This text discusses the use of a new in-situ substitute/ligation protocol for producing goal-clonable cpDNAs and constructing goal-insert, ordered DNA libraries. It also covers methods for verifying the quality and integrity of the cpDNA library, including PCR and collinearity analysis. Technological innovation is crucial for progress in this field, particularly in crops with large nuclear genomes where homology between nuclear and organellar genomes can complicate analysis. Enriching circular cpDNA can reduce sequence complexity and improve research outcomes.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'In-situ Chloroplast Lysis for High-Purity Circular cpDNA ExtractionSummary '},\n",
       " {'title': 'Title ', 'summary': 'Purifying Circular cpDNA Using PFGESummary '},\n",
       " {'title': 'Developing a High-Purity Circular cpDNA Extraction and Goal-Insert, Ordered cpDNA Library Construction Method ',\n",
       "  'summary': 'This study presents a new method for extracting circular cpDNA from gelled agarose and constructing a goal-insert, ordered cpDNA library. The method involves in-situ chloroplast lysis, PFGE purification, and electrochemical isolation to ensure the integrity of the cpDNA sequence. The in-situ substitute/ligation protocol is also developed to improve targeted ligation efficiency and can be used for other nucleic acid sequences. This method is highly skilled and may require multiple attempts for researchers with less experience, but it successfully preserves the whole circular cpDNA in three colonies.'},\n",
       " {'title': '\"Development of High-Purity Circular cpDNA Extraction and Goal-Insert, Ordered cpDNA Library Construction Methods\" ',\n",
       "  'summary': 'This study presents a set of methods for extracting high-purity circular cpDNA and constructing goal-insert, ordered cpDNA libraries. These methods can also be used to construct large-insert, ordered DNA libraries for further experiments such as genomic sequencing, bioinformatics analyses, cloning, and physical mapping. The study also includes supplementary information and acknowledgements to those who contributed to the research.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Acknowledgements and Funding in a Plant Cell Research StudySummary '},\n",
       " {'title': 'The Multimeric Forms of Chloroplast Genome ',\n",
       "  'summary': 'This text discusses the existence of multimeric forms of the chloroplast genome, as shown through various studies using techniques such as pulsed-field gel electrophoresis. These studies reveal the structural plasticity of the chloroplast genome in higher plants and the changes in chloroplast DNA levels during development. The findings also suggest that chloroplast DNA may be degraded during leaf development in certain plants. Overall, this text highlights the importance of understanding the organization and dynamics of the chloroplast genome in order to gain a better understanding of plant development and evolution.'},\n",
       " {'title': ' Changes in Chloroplast DNA Levels During Development ',\n",
       "  'summary': 'This text discusses various studies on the changes in chloroplast DNA levels during development in different plant species, including pea, rice, tobacco, Medicago truncatula, and maize. The studies show that the amount of chloroplast DNA can be affected by factors such as light and genotype. The structure of chloroplast DNA molecules and the effects of light on their amount are also examined. Additionally, the text includes a study on antigenic variation in Trypanosoma brucei and the release of DNA from bacteria, chloroplasts, and mitochondria upon lysis. Overall, the text provides valuable insights into the dynamic nature of chloroplast DNA during plant development.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'DNA Isolation and Characterization in Various OrganismsSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Study of Chloroplast DNA in Chlamydomonas reinhardi and Other PlantsSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Methods for Isolating and Characterizing Chloroplast DNA in PlantsSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Rapid Method for Purification of Organelles for DNA IsolationSummary '},\n",
       " {'title': 'Isolation of Chloroplast and Mitochondrial DNA from Plants ',\n",
       "  'summary': 'This text discusses various methods for isolating chloroplast and mitochondrial DNA from different plant species. These methods include using synchronous cultures of red algae, direct restriction fragment length polymorphism analysis, and whole chloroplast genome sequencing. The text also includes a rapid method for chloroplast isolation from a specific type of green algae. These methods are important for studying the genetic makeup and evolution of plants, as well as for understanding the role of chloroplast and mitochondrial DNA in plant biology.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'A Comparison of Chloroplast Isolation Methods for Various Plant SpeciesSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Optimized Methods for Sequencing and Maintaining Large DNA FragmentsSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Use of BAC Libraries for Advanced Genomics ResearchSummary '},\n",
       " {'title': 'External Viewer Error: Empty Response ',\n",
       "  'summary': 'This error occurs when an external viewer fails to load due to a timeout, resulting in an empty response with zero bytes read. This can be caused by a slow or unresponsive server, network connectivity issues, or incorrect configuration settings. It can prevent users from accessing important information and may require troubleshooting to resolve.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Role of Serratia Bacteria in MicrobiologyOpenSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Investigating the Role of Serratia Bacteria in Resilient AgricultureSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Understanding Sef Fimbrial Elements in Pathogenic Serratia SpeciesSummary '},\n",
       " {'title': 'The Role of Bacteria in Controlling Insect Pests ',\n",
       "  'summary': 'This text discusses the various types of bacteria that have been studied for their potential to control insect pests, including Bacillus thuringiensis, Serratia entomophila, and Pseudomonas fluorescens. These bacteria have been found to have different mechanisms for killing insects, such as producing toxins or competing for resources. Some of these bacteria have been successfully used in biocontrol methods, while others are still being studied for their potential. The text also mentions the potential use of bacteria in controlling specific insect pests, such as Costelytra giveni and Pyronota festiva.'},\n",
       " {'title': 'The Co-location of Toxin Complexes and Sef Fimbria Encoding Cluster in Serratia Bacteria ',\n",
       "  'summary': 'This study reveals a strong co-location between toxin complexes and the Sef fimbria encoding cluster in both documented STAMPs and newly characterized Tc-encoding non-STAMP Serratia bacteria. The study also identifies an ancestral connection between Sef and Tcs in Serratia bacteria, with the demarcation sites of the tc found in S. proteamaculans, S. entomophila, Y. frederiksenii, and Serratia sef. This information can help in identifying novel Serratia-based toxin complexes and understanding their evolutionary history.'},\n",
       " {'title': 'Ancestral Connection Between Sef and Tcs ',\n",
       "  'summary': 'This text discusses the identification of demarcation sites for the tc Sef and Tcs in various bacteria, including Serratia, Yersinia, and Serratia proteamaculans. The co-location of these tc islands and their nucleotide alignment reveal a connection between Sef and Tcs in these bacteria. In silico analysis of plasmids also shows the presence of degenerate tc sequences. Phylogenetic analysis further supports the ancestral relationship between Sef and Tcs. The text raises the question of why Sef is present in some bacteria but not others, and characterizes the Sef gene based on its identity.'},\n",
       " {'title': 'Characterization of Sef and its Homologs ',\n",
       "  'summary': 'This text discusses the characterization of Sef, a protein found in E. coli and Photorhabdus bacteria. Through gene identity and synteny comparisons, it is determined that SefD, SefC, and SefA likely function as a chaperone, usher protein, and fimbrial rod protein, respectively. The closest unique orthologs for SefH-SefJ are also identified through BlastX, including a fimbrial protein in Providencia and adhesins in Salmonella, Morganella, and Proteus bacteria. This information provides insight into the potential functions of Sef and its homologs in various bacterial systems.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Identification and Role of Sef Adhesins in Bacterial ColonizationSummary '},\n",
       " {'title': 'The Role of Sef in Colonization and Disease Progression in Costelytra giveni ',\n",
       "  'summary': 'This study investigates the role of Sef, an adhesin protein, in the colonization and disease progression of Costelytra giveni, a type of beetle. The results show that Sef is important for colonization and disease development, as its deletion leads to a significant decrease in disease and mortality rates. This suggests that Sef may be a potential target for controlling the spread of disease in this beetle species.'},\n",
       " {'title': '\"COVID-19 Mortality and Recovery Rates ',\n",
       "  'summary': 'A study shows that the mortality rate for COVID-19 is 0%, while the recovery rate is 91.7%. However, there is a significant range in recovery rates, with some countries reporting as low as 4.2%. This data, published by John Wiley & Sons, Ltd., highlights the need for continued efforts to combat the pandemic and improve recovery rates globally.\"'},\n",
       " {'title': 'The Role of Sef and Tc Encoding Regions in Serratia Pathogenicity ',\n",
       "  'summary': 'This study examined the relationship between Sef and Tc encoding regions in Serratia bacteria and their contribution to pathogenic activity. While there was a high correlation between these regions and their preservation, experimental designs did not reveal any evidence of a mutually beneficial function. The leading hypothesis was that Sef may aid in conjugation and increase Tc production, but experiments showed no such effect. This suggests that there may be other factors at play in the pathogenicity of Serratia bacteria.'},\n",
       " {'title': 'Investigating the Function of Sef and Tcs in Bacterial Conjugation ',\n",
       "  'summary': 'This study aimed to determine if the proteins Sef and Tcs have a mutually beneficial function in bacterial conjugation. However, experiments showed no evidence of Sef aiding in the proliferation of the plasmid responsible for producing Tcs. Further analysis using SDS-PAGE and TEM/Confocal microscopy also did not reveal any significant findings. These results suggest that the leading hypothesis of Sef aiding in conjugation may not be accurate and further research is needed to fully understand the function of these proteins in bacterial conjugation.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Characterizing the Role of Sef in Grass Grub Gut PathogenicitySummary '},\n",
       " {'title': 'Assessing the Role of Sef in the Gut of Grass Grubs ',\n",
       "  'summary': 'This study used various experiments to investigate the potential role of Sef, a bacterial protein, in the gut of grass grubs. Fimbrial lengths were measured using ImageJ, and pathogenicity was determined through oral challenge bioassays. Conjugation and biofilm formation experiments were also conducted. Bioinformatics analysis was performed on DNA samples, and plasmid annotation was done using Prokka. Distance matrix figures were generated using custom R scripts. The results suggest that Sef may play a role in gut colonization and biofilm formation in grass grubs.'},\n",
       " {'title': 'Annotation and Distance Matrix Analysis of Plasmid Contigs ',\n",
       "  'summary': 'This study used Prokka and custom R scripts to annotate plasmid contigs and generate distance matrix figures. The authors, Lesley Sitter, Marion Schoof, Travis R. Glare, Murray P. Cox, Peter C. Fineran, Paul P. Gardner, and Mark R. H. Hurst, were supported by the New Zealand Tertiary Education Commission and Bioprotection Aotearoa. The data generated in this study has been submitted to the NCBI Reference Sequence database.'},\n",
       " {'title': 'The Impact of Pesticides on Agriculture and the Environment ',\n",
       "  'summary': 'This text discusses the use of pesticides in agriculture and their potential benefits and hazards. It also introduces a new screening index for measuring the leachability of pesticides into groundwater. Additionally, the text explores the effects of pesticides on virulence, motility, and biofilm formation in Salmonella typhimurium and the mechanical architecture and folding of E. coli. It also delves into the role of type 1 fimbriae in Salmonella enterica and the potential environmental impacts of pesticide use.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Role of Type 1 Fimbriae in Salmonella enterica Infection and ImmunitySummary '},\n",
       " {'title': 'The Role of Fimbriation and Mutants in Escherichia coli ',\n",
       "  'summary': 'This article discusses the importance of Type 1 fimbriation and fimE mutants in Escherichia coli, a common bacterium. It also explores the potential insecticidal toxins produced by the bacterium Photorhabdus luminescens and the molecular mechanism of bacterial pili assembly. Additionally, it introduces the BLAST+ software and its applications in bioinformatics.'},\n",
       " {'title': 'The Evolution of Bioinformatics: A Review of Key Studies ',\n",
       "  'summary': 'This text provides a summary of several key studies in the field of bioinformatics, including the architecture and applications of BLAST+, the role of Type 1 fimbriae in insecticidal bacteria, a new data structure for phylogenomic inference, silver staining of proteins in polyacrylamide gels, and the analysis of major fimbrial subunits in Salmonella colonization. These studies showcase the advancements and applications of bioinformatics in various areas of research, highlighting its importance in understanding and analyzing biological data.'},\n",
       " {'title': 'The Role of Fimbrial Subunits in Salmonella Colonisation ',\n",
       "  'summary': 'This article discusses a 2008 analysis of 13 major fimbrial subunits and their role in the colonisation of chicken intestines by Salmonella enterica. It also includes studies on the expression of fimbrial subunits in other bacteria, such as Escherichia coli and Serratia Yersinia frederiksenii. The results of these studies provide insight into the mechanisms of bacterial colonisation and could potentially lead to the development of new strategies for preventing and treating Salmonella infections.'},\n",
       " {'title': \"The Economic Impact of Invertebrate Pests on New Zealand's Pastoral Industry \",\n",
       "  'summary': \"This text discusses the economic cost of invertebrate pests to New Zealand's pastoral industry, including the impact on crops and livestock. It also includes information on the biology and control of these pests, such as the grass grub and adult beetles. Additionally, it explores the association of a large plasmid with amber disease in the New Zealand grass grub and the potential use of Serratia entomophila and Serratia proteamaculans as biological control agents. The text also briefly touches on the prevalence and treatment of urinary tract infections, as well as the taxonomy of Serratia entomophila and Costelytra zealandica.\"},\n",
       " {'title': 'The Role of Serratia Bacteria in Controlling Insect Pests ',\n",
       "  'summary': 'This text discusses the various studies and findings on the use of Serratia bacteria, specifically Serratia entomophila and Serratia proteamaculans, as a means of controlling insect pests such as the grass grub Costelytra zealandica. These bacteria have been found to produce toxins and antifeeding agents that are effective against the pests, making them potential biocontrol agents. The text also mentions the discovery of diverse toxin clusters and an eCIS variant in Serratia proteamaculans, further highlighting the potential of these bacteria in pest management.'},\n",
       " {'title': 'The Role of Plasmids in Serratia entomophila Pathogenicity ',\n",
       "  'summary': 'This text discusses the role of plasmids in the pathogenicity of Serratia entomophila, a bacterium that is harmful to insects. It includes information on the nucleotide sequence of the plasmid, as well as studies on the cloning and behavior of the bacterium. The text also touches on the use of green fluorescent protein to monitor the fate of Serratia entomophila and the potential ecotoxicity of pesticides and their transformation products in relation to this bacterium. Overall, the text provides valuable insights into the mechanisms and potential impacts of Serratia entomophila pathogenicity.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Understanding the Role of Serratia entomophila Plasmid in Pest ManagementSummary '},\n",
       " {'title': 'Protein Function Classification and Analysis ',\n",
       "  'summary': 'This text discusses various methods and tools used for protein function classification and analysis, including InterProScan 5 and Phyre2. It also explores the structure and function of type 1 fimbriae, as well as the tra region of the conjugative plasmid pIP501. Additionally, it delves into the topic of pesticide poisoning events in wild birds and their impact on the environment. Overall, this text provides a comprehensive overview of protein function classification and analysis, as well as its applications in various fields such as microbiology and environmental science.'},\n",
       " {'title': 'Pesticide Poisoning in Wild Birds in Korea ',\n",
       "  'summary': 'This article discusses the occurrence of pesticide poisoning events in wild birds in Korea from 1998 to 2002. It also includes information on the effects of environmental contaminants on fertility and reproductive health. The study found that pesticides can have negative impacts on bird populations and their reproductive abilities. The article also mentions previous studies on the topic, including one on the hybridization between Escherichia coli and shigella. Overall, the article highlights the importance of understanding the effects of pesticides on wildlife and the environment.'},\n",
       " {'title': 'The Impact of Environmental Contaminants on Fertility and Reproductive Health ',\n",
       "  'summary': 'This article discusses the effects of environmental contaminants on fertility and reproductive health, as well as the potential cooperation of adhesin alleles in Salmonella and the expression of mannose-resistant fimbriae in Photorhabdus temperata. It also includes a sequence analysis of insecticidal genes from Xenorhabdus nematophilus and a study on pesticide exposure and health problems among female horticulture workers in Tanzania. The article highlights the need for further research and regulation to protect human health from the negative impacts of environmental contaminants.'},\n",
       " {'title': 'Pesticide Exposure and Health Problems among Female Horticulture Workers in Tanzania ',\n",
       "  'summary': 'This study examines the impact of pesticide exposure on the health of female horticulture workers in Tanzania. The researchers found that these workers are at a high risk for health problems due to their exposure to pesticides, which are commonly used in the horticulture industry. The study highlights the need for better safety measures and regulations to protect the health of these workers.'},\n",
       " {'title': 'Classification and Clinical Significance of Microorganisms ',\n",
       "  'summary': 'This text discusses the classification, identification, and clinical significance of Proteus, Providencia, and Morganella bacteria. It also includes information on a biofilm formation assay and the effects of certain pesticides on non-target invertebrates. Additionally, it mentions the R language and environment for statistical computing.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Overview of Research on Bacterial Toxins and Pesticide Effects on Cotton FarmingSummary '},\n",
       " {'title': 'Advancements in Image Analysis and Genome Annotation ',\n",
       "  'summary': 'This text discusses the development and use of various tools and techniques for image analysis and genome annotation. It covers the 25-year history of the NIH image to ImageJ software, the use of lysis cassettes for exoprotein release in Yersinia entomophaga, the rapid annotation tool Prokka, and the identification of an allergen in Trichostrongylus colubriformis. These advancements have greatly improved the efficiency and accuracy of image analysis and genome annotation, making them essential tools in various fields of research.'},\n",
       " {'title': 'The Evolution of Virulence in Transmissible Mega-Plasmids ',\n",
       "  'summary': 'This article discusses the functional role of the type 1 pilus rod structure in mediating host-pathogen interactions and how it has evolved in a novel family of transmissible mega-plasmids. The study found that the structure of the pilus rod is crucial for the virulence of these plasmids, which can have significant implications for the spread of antibiotic resistance and other pathogenic traits. The research also highlights the potential impact of pesticide seed dressings on soil organisms and the decomposition of plant material. Overall, this article sheds light on the evolution of virulence and its potential consequences in the microbial world.'},\n",
       " {'title': 'The Impact of Pesticide Seed Dressings on Soil Organisms and Plant Material Decomposition ',\n",
       "  'summary': 'Pesticide seed dressings can alter the activity of soil organisms and decrease the decomposition of plant material, as shown in a study comparing different fimbrial operons in Escherichia coli and Serratia bacteria. This could potentially affect the colonization of the gut of Costelytra giveni insects by Serratia entomophila bacteria, which possess the Sef fimbriae. Further research is needed to fully understand the role of Sef in aiding adherence to hosts and colonization of the gut.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Enhanced Adhesion of HMO-Primed Bifidobacteria to Intestinal Epithelial CellsSummary '},\n",
       " {'title': 'Enhancing Bifidobacterial Adhesion with Human Milk Oligosaccharides ',\n",
       "  'summary': 'This study investigates the potential of human milk oligosaccharides (HMOs) to enhance the adhesion of beneficial gut bacteria, specifically Bifidobacterium, to the intestinal epithelium. Bifidobacteria are known for their health benefits and are commonly used as probiotics. The results showed that HMOs can increase the adhesion of Bifidobacterium to the intestinal lining, potentially leading to improved host-microbe interactions and health benefits. The study was supported by various research groups and funding agencies in Ireland.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Investigating the Role of Human Milk Oligosaccharides in Enhancing Bifidobacterial Adhesion to Intestinal CellsSummary '},\n",
       " {'title': '\"Assessing Probiotic Colonization Capabilities in Human Intestines\" ',\n",
       "  'summary': 'This study by Xiao et al. examines the colonization abilities of probiotic strains, specifically those isolated from fermented foods or plants and then studied in humans. The researchers found that these allochthonous strains do not perform well in their new ecosystem, highlighting the need for further research on probiotics specifically designed for human intestines. The study also includes details on the methods used, such as the generation of a human milk sample and the use of a human colonic adenocarcinoma cell line as a model for intestinal epithelia.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Removal of Lipids and Proteins from Breastmilk and Colonization Assays with Bifidobacterial StrainsSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Bacterial Adherence to Eukaryotic Cells in Probiotic StrainsSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Determining Bacterial Adherence and Numbers in Bifidobacterial Strains Using qPCR and Whole Genome SequencingSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Genome Sequencing and Colonization Capabilities of Bifidobacterial StrainsSummary '},\n",
       " {'title': 'Proteomic Analysis of Bifidobacterial Strains ',\n",
       "  'summary': 'This text discusses the use of proteomic analysis to study the colonization-related factors of bifidobacterial strains. The strains were examined for specific features, including signal peptides, recognition motifs, and membrane-spanning regions. The analysis was carried out on a mixture of four strains, with and without pre-treatment, and after exposure to a cell line. The protein extraction and digestion process is also described. This study provides valuable insights into the factors that contribute to the colonization of bifidobacterial strains, which can have implications for their potential use as probiotics.'},\n",
       " {'title': 'Protein Quantification and Analysis in Adhesion Assays ',\n",
       "  'summary': 'This text describes the methods used to determine protein concentration and perform proteomic analysis in adhesion assays. Lysates were clarified and protein concentration was measured using a Bradford assay. Samples were then prepared for LC-MS/MS analysis by adjusting to a specific protein concentration, reducing and alkylating the proteins, and digesting with trypsin. The resulting peptides were analyzed using a mass spectrometer and searched against a composite database. Statistical analysis was performed on the data from adhesion assays using technical and biological triplicates.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Statistical Analysis of Bifidobacterial Adhesion to Intestinal Epithelial CellsSummary '},\n",
       " {'title': '   The Impact of Secretor-HMO on Bifidobacterial Adherence ',\n",
       "  'summary': \"The study examined the effects of a specific type of human milk oligosaccharide (HMO) called secretor-HMO on the adherence of bifidobacteria to intestinal cells. The secretor-HMO was obtained through a purification process and was found to contain various HMOs, including difucosyllactose and lacto-N-fucopentaose I. The results showed that pre-exposure to secretor-HMO increased the adherence of bifidobacteria to intestinal cells, compared to a glucose control and 2'-fucosyllactose. This suggests that secretor-HMO may play a role in promoting the growth of beneficial bacteria in the gut.\"},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Adhesion of Bifidobacterial Strains to Intestinal CellsSummary '},\n",
       " {'title': '\"Genes Involved in Probiotic Colonisation and Persistence in the Gut\" ',\n",
       "  'summary': 'A literature search identified 45 genes with potential roles in the survival and persistence of probiotics in the gut, specifically in Lactobacillus and Bifidobacterium strains. These genes include bsh, tal, luxS, esat, BBPR, tlyC1, and rfbP, and are found in B. bifidum, B. breve, B. infantis, and L. plantarum. The study also found a high percentage identity among these genes, indicating their importance in probiotic survival. This information can be useful in understanding the mechanisms behind probiotic colonization and developing more effective probiotic strains for gut health.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Identification of Genes Involved in Probiotic Colonization and Persistence in the GutSummary '},\n",
       " {'title': 'Impact of S-HMO Pre-Treatment on Colonization-Associated Pathways in Bifidobacteria ',\n",
       "  'summary': 'This study examined the effects of S-HMO pre-treatment on the colonization-associated pathways in bifidobacteria. The results showed that exposure to human cells led to a significant masking of bacterial proteins, with only a small number of proteins being matched to bacterial proteins. The study also found that there were significant changes in protein abundance in Bifidobacterium bifidum, Bifidobacterium infantis, and Bifidobacterium breve following exposure to human cells. These findings suggest that S-HMO pre-treatment may have a significant impact on the colonization-associated pathways in bifidobacteria.'},\n",
       " {'title': 'Bifidobacterium Adhesion Characteristics and Host-Microbe Interactions ',\n",
       "  'summary': 'This text discusses the adhesive characteristics of four strains of bifidobacteria and their interactions with intestinal epithelial cells. The study found that Bifidobacterium bifidum showed significant upregulation of cell surface adhesins and glycosyl hydrolase proteins, while also matching with other proteins. The HT29-MTX cell line was used to evaluate adherence, as it is able to produce mucin and better mimics the mucus layer in the intestines. This study highlights the potential of B. bifidum as a probiotic for improving host-microbe interactions in the gut.'},\n",
       " {'title': '\"Adhesion of Probiotic Bacteria to Intestinal Epithelial Cells\" ',\n",
       "  'summary': 'This study evaluated the adherence of probiotic bacteria to intestinal epithelial cells using a human cell line that produces mucin. Results showed that B. infantis had higher levels of adhesion when exposed to HMOs, and bacterial genomes were found to have components that aid in adhesion, such as outer-membrane proteins and pili. Additionally, membrane-bound enzymes called \"sortases\" were found to play a critical role in facilitating communication between bacteria and host cells. This study highlights the importance of understanding host-microbe interactions in the development of probiotics.'},\n",
       " {'title': 'The Importance of Bifidobacterial Adherence to Intestinal Epithelial Cells ',\n",
       "  'summary': 'A recent study confirms that the adherence of bifidobacterial strains to intestinal epithelial cells is species- and strain-specific, highlighting the importance of this interaction in the gut microbiota. Moonlighting proteins and the mucus layer also play a role in this process, providing attachment sites and nutrients for these beneficial bacteria. This study sheds light on the specific strains of bifidobacteria that are most effective in adhering to intestinal cells, potentially leading to further research and development of probiotic treatments for gut health.'},\n",
       " {'title': '   The Role of HMO in Promoting Bifidobacteria-Intestinal Cell Interaction ',\n",
       "  'summary': 'HMO, a component of human breast milk, has been found to significantly improve the adhesion ability of bifidobacteria to intestinal cells, which is crucial for their colonization and potential health benefits to the host. This study also provides genome data for specific bifidobacterial strains and contributions from various authors. However, there may be a conflict of interest as some authors are employed by a company that has received costs for services related to the study.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Tools and Resources for Microbiome ResearchSummary '},\n",
       " {'title': 'Exploring Diversity in Gut Bacteria ',\n",
       "  'summary': 'This article discusses the findings of a comparative analysis of the genomes and methylomes of the gut bacteria Bifidobacterium breve. The study reveals a wide range of restriction/modification systems present in this commensal bacteria, which may play a role in its competitive population abundance. The article also touches upon the cross-feeding interactions between infant bifidobacteria and Eubacterium hallii, the immunomodulatory effects of sortase-deficient lactobacilli, and the binding of human plasminogen to Bifidobacterium. Overall, this research sheds light on the diversity and potential functions of these important gut bacteria.'},\n",
       " {'title': 'The Role of Bifidobacteria in Gut Health ',\n",
       "  'summary': 'This text discusses various studies on the interaction between Bifidobacteria and the human body, specifically focusing on the binding of human plasminogen to Bifidobacteria and its potential role in gut health. The studies also explore the effects of Bifidobacteria on intestinal epithelial function and the mechanisms by which they shape the gut microbiota. These findings highlight the potential benefits of Bifidobacteria in promoting a healthy gut and improving overall health.'},\n",
       " {'title': 'The Role of Bifidobacteria in Infant Gut Health ',\n",
       "  'summary': 'This article discusses the impact of Bifidobacteria, a type of probiotic, on infant gut health. The study found that Bifidobacteria isolated from infants and cultured on human milk oligosaccharides can affect intestinal epithelial function. Additionally, the article highlights the importance of autoinducer-2 in gut colonization and probiotic functionality of Bifidobacterium breve. Other studies mentioned in the article also explore the genomic characterization and transcriptional studies of Bifidobacterium adolescentis and the cross-feeding abilities of Bifidobacterium breve and Bifidobacterium bifidum. Overall, the research suggests that Bifidobacteria can play a crucial role in promoting a healthy gut microbiome in infants.'},\n",
       " {'title': 'The Role of Bifidobacterium in Gut Health ',\n",
       "  'summary': 'This text discusses the role of Bifidobacterium in promoting gut health through cross-feeding, immune modulation, and pathogen protection. It also explores the genetic and molecular characteristics of Bifidobacterium and its evolution as a gut symbiont. The study highlights the potential benefits of Bifidobacterium in maintaining a healthy gut microbiome.'},\n",
       " {'title': 'Probiotics and Gut Health ',\n",
       "  'summary': 'This text discusses various studies and research on the role of probiotics in maintaining gut health and treating conditions such as irritable bowel syndrome. It covers topics such as biofilm formation in gut symbionts, the use of probiotics in improving gut microbiota, and the definition and scope of prebiotics. The studies mentioned highlight the potential benefits of probiotics in promoting overall gut health and improving quality of life for individuals with gut-related issues.'},\n",
       " {'title': 'Expert Consensus on Prebiotics ',\n",
       "  'summary': 'This text discusses the definition and scope of prebiotics as agreed upon by the International Scientific Association for Probiotics and Prebiotics (ISAPP). It also includes insights into glycogen metabolism in Lactobacillus acidophilus, the transcriptional response of Bifidobacterium longum, the role of extracellular transaldolase from Bifidobacterium bifidum, a bile-inducible efflux transporter from Bifidobacterium longum, and the implication of an outer surface lipoprotein in adhesion of Bifidobacterium bifidum.'},\n",
       " {'title': 'The Role of Bifidobacterium in Gut Health ',\n",
       "  'summary': 'This text discusses various studies on the role of Bifidobacterium in gut health, including the identification of a bile-inducible efflux transporter in Bifidobacterium longum, the implication of an outer surface lipoprotein in adhesion of Bifidobacterium bifidum, and the critical role of a housekeeping sortase in probiotic Bifidobacterium bifidum. It also includes a comparative genomic analysis of Lactobacillus rhamnosus and the potential benefits of moonlighting proteins in multitasking. These studies shed light on the importance of Bifidobacterium in maintaining a healthy microbiome.'},\n",
       " {'title': 'Comparative Genomic Analysis of Probiotic Bacteria ',\n",
       "  'summary': 'This text discusses various studies on the genomic analysis of probiotic bacteria, specifically Lactobacillus rhamnosus and Bifidobacterium longum. These studies explore the functional roles of different genes and proteins in these bacteria, as well as their potential health benefits. The findings suggest that probiotic bacteria play a crucial role in maintaining a healthy gut microbiota, which has implications for overall health and well-being. These studies also highlight the importance of understanding the composition and activities of the infant gut microbiota, as it is the first microbial colonizer of the human gut.'},\n",
       " {'title': 'The Role of Gut Microbiota in Human Health ',\n",
       "  'summary': 'This text discusses the composition and activities of the infant gut microbiota, as well as the colonization of the human gut by health-promoting bacteria. It also explores the role of microbial lectins and a Bifidobacterial pilus-associated protein in promoting colonic epithelial proliferation. Additionally, it includes a functional genome analysis of Bifidobacterium breve and the relationship between mucus barrier, mucins, and gut microbiota. Overall, this text provides insights into the important role of gut microbiota in maintaining human health.'},\n",
       " {'title': 'Functional Genome Analysis of Bifidobacterium breve and its Role in Carbohydrate Metabolism and Gut Health ',\n",
       "  'summary': 'This study delves into the functional genome of Bifidobacterium breve, a beneficial gut bacteria, and its role in carbohydrate metabolism. The results reveal an on/off replication slippage switch and potential in vivo implications. Additionally, the study explores the relationship between B. breve and mucins, the slimy barrier in the gut, and how they work together to maintain gut health. The study also examines the microbial content of raw and pasteurized cow milk and its potential impact on gut health. Overall, this study sheds light on the important role of B. breve in maintaining a healthy gut microbiome.'},\n",
       " {'title': 'Mining Milk for Factors to Increase Adherence of Bifidobacterium longum ',\n",
       "  'summary': 'This text discusses various studies and research on Bifidobacterium longum, a probiotic commonly found in the gut. The studies focus on factors that can increase the adherence of this probiotic, such as mucin turnover in the small intestine, protein adhesin genes, and exopolysaccharide production. These factors can potentially improve the effectiveness of Bifidobacterium longum in promoting gut health. The text also mentions the potential benefits of Bifidobacterium longum in metabolizing bile salts and its role in nutrient absorption. Overall, the text highlights the importance of understanding and utilizing factors that can enhance the adherence of probiotics in promoting gut health.'},\n",
       " {'title': 'The Role of Bifidobacterium in Microbial Adhesion and Human Milk Oligosaccharide-Sharing ',\n",
       "  'summary': 'This article discusses the various ways in which Bifidobacterium, a type of beneficial bacteria found in the human gut, plays a role in microbial adhesion and the sharing of human milk oligosaccharides. It covers topics such as the use of bile salt hydrolase by Bifidobacterium longum, the role of sortase-dependent pili in Bifidobacterium bifidum, and the sharing of human milk oligosaccharides by a consortium of infant-derived Bifidobacterium. The article also mentions the Perseus computational platform used for comprehensive analysis of proteomics data. Overall, this article highlights the importance of Bifidobacterium in maintaining a healthy gut microbiome and its potential benefits for infants.'},\n",
       " {'title': 'The Role of Microbes in Controlling Adhesion and Sharing Human Milk Oligosaccharides ',\n",
       "  'summary': 'This text discusses the various ways in which microbes, specifically Bifidobacterium and Lactobacillus reuteri, use force and enzymes to control adhesion and share human milk oligosaccharides. These processes are important for the ecological performance and survival of these microbes, and understanding them can provide insights into the development of probiotics and prebiotics. The text also highlights the role of specific proteins, such as Lsp and MsrB, in the ecological performance of Lactobacillus reuteri, and the depletion of teichoic acids in this microbe. Overall, this text sheds light on the complex interactions between microbes and their environment.'},\n",
       " {'title': 'The Role of Bifidobacterium and Lactobacillus in Microbiota ',\n",
       "  'summary': \"This text discusses the role of specific strains of Bifidobacterium and Lactobacillus in the human microbiota. It includes proteomic profiling of Bifidobacterium bifidum and analysis of Bifidobacterium longum's adhesion to host tissue. It also examines the distribution and evolution of von Willebrand/integrin A domains, which play a role in cell adhesion. Additionally, the text evaluates the adhesion of Lactobacillus reuteri to host tissue and its potential benefits in promoting human health.\"},\n",
       " {'title': 'The Role of Probiotics in Maintaining Gut Health ',\n",
       "  'summary': \"This text discusses various studies on the effects of probiotics, specifically Lactobacillus reuteri and Bifidobacterium breve, on gut health. These studies explore the mechanisms by which these probiotics colonize the gut and their impact on intestinal secretory immunoglobulin A levels. The text also identifies surface-associated proteins of Bifidobacterium animalis lactis and the role of 2'-fucosyllactose in promoting Bifidobacterium bifidum. Overall, the text highlights the potential benefits of probiotics in maintaining a healthy gut microbiome.\"},\n",
       " {'title': 'Elucidating the Neuroimmunology of Traumatic Brain Injury ',\n",
       "  'summary': \"This article discusses various methodological approaches to understanding the intercellular communication and function involved in traumatic brain injury. It highlights the work of researchers from the Ann Romney Center for Neurologic Diseases and the Department of Neurosurgery at Brigham and Women's Hospital, as well as the David H. Koch Institute for Integrative Cancer Research at Massachusetts Institute of Technology. The article also mentions the contributions of reviewers from Qatar University and The University of Iowa. Overall, the article emphasizes the importance of studying the neuroimmunology of traumatic brain injury in order to develop effective treatments and interventions.\"},\n",
       " {'title': 'Exploring the Neuroimmunology of Traumatic Brain Injury ',\n",
       "  'summary': 'This article discusses the importance of understanding the neuroimmunology of traumatic brain injury (TBI) and outlines current methodologies that can aid in its exploration. These techniques include adoptive cell transfer, intra-CNS injection(s), selective cellular depletion, genetic manipulation, molecular neuroimaging, and in vitro models. The authors also highlight the crucial role of communication between resident neuroglial cells and recruited lymphocytes in the development of secondary brain injury. Financial support for this research was received from various sources, including the DFCI/Kiki Leptomeningeal Disease Research Fund and the US National Institutes of Health.'},\n",
       " {'title': 'Studying the Neuroimmunology of Traumatic Brain Injury ',\n",
       "  'summary': 'This text discusses the methods and approaches used to study the complex neuroimmunology underlying traumatic brain injury (TBI). These methods include adoptive cell transfer, intra-CNS injections, cell depletion, genetic manipulations, and molecular imaging. TBI is a heterogeneous condition with long-term disabling effects on cognition, sensorimotor function, and behavior. The research is supported by grants from DFCI/Kiki Leptomeningeal Disease Research Fund and the US National Institutes of Health, as well as a 2023 Stepping Strong Innovator Award. By understanding the neuroimmunology of TBI, researchers hope to develop better treatments and interventions for TBI survivors.'},\n",
       " {'title': 'Experimental Methods in Neuroimmunology of TBI ',\n",
       "  'summary': 'This text discusses various experimental methods used in the study of neuroimmunology of traumatic brain injury (TBI). These methods include adoptive cell transfer, intra-CNS injections, cell depletion, tissue- and temporal-specific genetic manipulations, molecular neuroimaging, and in vitro co-culture systems. The use of adoptive cell transfer, where specific cell populations are isolated and introduced into a recipient host, has been a commonly employed technique in immunology. This method has been used in animal models of disease and was first described in 1988 for the treatment of patients with metastatic melanoma. The text also includes a sample of TBI neuroimmunology studies categorized by methodological approach.'},\n",
       " {'title': 'Intervention Methods for Treating Traumatic Brain Injury ',\n",
       "  'summary': 'This text discusses various intervention methods for treating traumatic brain injury, including adoptive transfer of CD4+ and CD11b+ cells, as well as cell depletion using anti-CD8 antibodies. These methods were tested in different models of TBI and recipient species, with varying outcomes. While some interventions worsened cerebral injury and neuroinflammation, others showed promise in improving neurological recovery. These findings were reported in studies by Fee et al. (2003), Braun et al. (2017), and Abou-El-Hassan et al. (2023).'},\n",
       " {'title': 'The Effects of Depleting Immune Cells on Neurological Recovery after Traumatic Brain Injury ',\n",
       "  'summary': 'Various studies have investigated the effects of depleting different types of immune cells on neurological recovery after traumatic brain injury (TBI). Results have shown that depleting macrophages/microglia, Vγ1 and Vγ4 cells, Tregs, CD25, and Gr-1 cells can have varying effects on neuroinflammation, lesion volume, and sensorimotor deficits at different time points post-injury. These findings suggest that targeting specific immune cells may have potential therapeutic benefits for TBI patients.'},\n",
       " {'title': 'The Effects of CD11b Depletion on Brain Injuries ',\n",
       "  'summary': \"Studies have shown that depleting CD11b, a protein found on immune cells, can have varying effects on brain injuries. In one study, administering valganciclovir, a drug that depletes CD11b, before or after a traumatic brain injury (TBI) had no effect on brain damage. In another study, using diphtheria toxin to deplete CD11b after a TBI resulted in increased inflammation in the brain. Additionally, depleting macrophages and microglia, which also express CD11b, after a TBI in rat pups led to increased neurodegeneration and reactivity to brain injury markers. These findings suggest that CD11b may play a role in the body's response to brain injuries.\"},\n",
       " {'title': 'The Effects of Macrophage and B Cell Depletion on Neurodegeneration and Recovery after Traumatic Brain Injury ',\n",
       "  'summary': 'This text discusses four studies that investigate the effects of depleting macrophages and B cells on neurodegeneration and recovery after traumatic brain injury (TBI). The studies use different methods, such as clodronate liposomes and direct intra-CNS injection, and different animal models, such as Sprague-Dawley rat pups and C57BL/6J mice. Results show that depleting macrophages and B cells can have both positive and negative effects on neurodegeneration and recovery, depending on the timing and method of depletion. These findings have implications for potential therapeutic strategies for TBI.'},\n",
       " {'title': 'Potential Therapies for Traumatic Brain Injury ',\n",
       "  'summary': 'This text discusses various potential therapies for traumatic brain injury, including genetic engineering, intraventricular injections, and inducible deletions. These therapies have shown promising results in reducing brain edema, neuronal apoptosis, and astrocyte activation, as well as improving memory and promoting regeneration of oligodendrocytes. These therapies have been tested on rats and mice, and have shown positive effects up to 2 months post-injury. Further research and development of these therapies could potentially lead to improved outcomes for individuals with traumatic brain injury.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Investigating the Effects of Co-Culture Systems on Microglia and Myeloid CellsSummary '},\n",
       " {'title': 'The Role of T Cells in Modulating Neuroinflammation after Traumatic Brain Injury ',\n",
       "  'summary': 'T cells, specifically Tregs, have been shown to play a crucial role in reducing neuroinflammation after traumatic brain injury. Studies have found that Tregs can attenuate the pro-inflammatory response of microglia, the immune cells in the brain, through co-culture experiments. This highlights the potential for Tregs to be used as a therapeutic target for TBI.'},\n",
       " {'title': 'The Role of Adaptive Immune Cells in Traumatic Brain Injury ',\n",
       "  'summary': 'Recent studies have shown that Tregs and gamma-delta T cells play a significant role in modulating inflammation in traumatic brain injury (TBI). Further research using adaptive cell transfer (ACT) techniques, such as CAR T-cell therapy, may provide promising immunotherapy options for TBI. However, there are challenges in conducting these studies, such as regulatory hurdles and the complexity of the techniques. Direct intra-CNS injections of immune cells, known as CNS ACT, have not been widely used but could offer valuable insights into TBI neuroimmunology and potential treatments.'},\n",
       " {'title': 'The Advantages and Challenges of Direct CNS Injections for ACT Applications ',\n",
       "  'summary': 'Direct CNS injections offer a simplified method for delivering immune cells and pharmacological agents to the brain, bypassing the blood-brain barrier. However, their invasive nature and technical expertise required have limited their use in studying TBI neuroimmunology and as an immunotherapy. In preclinical experiments, these injections have shown promising results in improving motor coordination and memory in mice. Further investigations using advanced technology such as single-cell RNA-seq are needed to fully understand the crosstalk between the adaptive and innate immune responses.'},\n",
       " {'title': 'Exploring the Use of Intra-CNS Injections in Investigating TBI Neuroimmunology ',\n",
       "  'summary': 'Intra-CNS injections and other neuroimmunology assays can provide valuable insights into the dynamic nature of TBI neuroimmunology. While invasive, these injections can offer a more precise and controlled delivery of substances, allowing for temporal analysis of molecular and cellular changes in different phases of TBI. Additionally, the use of intra-CNS injections after TBI may have therapeutic potential, such as the potential for faster and more effective recovery with the use of regulatory cells of the adaptive immune system. However, challenges such as invasiveness, ethical concerns, and technical expertise must be considered. Cell depletion therapy, achieved through pharmacological agents, has also shown promise in treating autoimmune diseases and may be applicable in TBI research.'},\n",
       " {'title': 'The Role of Anti-CD19 Antibodies in Investigating TBI Neuroimmunology ',\n",
       "  'summary': 'Anti-CD19 antibodies have been used in numerous studies to deplete B cells and investigate the neuroimmunology of traumatic brain injury (TBI). These antibodies have been shown to be effective in reducing inflammation and improving outcomes in TBI. By targeting B cells, which play a key role in the immune response, researchers have gained valuable insights into the mechanisms of TBI and potential treatment options. This approach has been utilized in multiple studies and continues to be a valuable tool in understanding the complex immune response to TBI.'},\n",
       " {'title': 'The Role of Cell Depletion in Investigating Neuroimmunology after Traumatic Brain Injury ',\n",
       "  'summary': 'Cell depletion has been used in various studies to investigate the role of different immune cell populations in neuroinflammation after traumatic brain injury. This approach has shown that depletion of B cells can reduce neuroinflammation in spinal cord injury, while selective depletion of CD4 or CD8 T cells has been associated with reduced lesion volume in animal models of stroke. Myeloid cells, such as neutrophils, monocytes, and macrophages, have also been found to accumulate at sites of injury and can be targeted for investigation using cell depletion techniques. This method has potential applications in autoimmune and cancer treatments.'},\n",
       " {'title': 'Challenges and Approaches in Genetic Engineering for Therapeutic Use ',\n",
       "  'summary': 'This text discusses the challenges of using genetic engineering for therapeutic purposes, including off-target effects, immune suppression, complexity, and ethical considerations. It also mentions various approaches, such as the use of Cre-loxP technology and CRISPR/Cas9, and the potential use of short hairpin RNA for gene silencing. The text provides a summary of the different Cre promoters used in cells of the nervous and immune systems, as well as specific examples of genetic engineering approaches used in studies.'},\n",
       " {'title': 'The Role and Advantages/Disadvantages of Genetic Engineering ',\n",
       "  'summary': 'This text discusses the role of aquaporin-4 in brain edema after TBI and the benefits and drawbacks of genetic engineering. Genetic engineering allows for a better understanding of disease and the development of customized pharmaceuticals, but it also comes with technical expertise, unintended consequences, social acceptance, and regulatory challenges.'},\n",
       " {'title': 'The Advantages and Challenges of Molecular Imaging in Genetic Engineering ',\n",
       "  'summary': 'Molecular imaging has greatly advanced in the field of neuroglia research, but there are still challenges to overcome. These include technical expertise, unintended consequences, social acceptance, and regulatory hurdles. The use of molecular biomarkers, such as TSPO, in PET imaging has been successful in studying neurological diseases like stroke, but there are still limitations and potential risks. Two-photon in vivo imaging has also shown promise, but further research is needed. Overall, while molecular imaging has great potential in genetic engineering, careful consideration and regulation are necessary to ensure its safe and effective use.'},\n",
       " {'title': 'The Advantages and Disadvantages of Molecular Imaging and Co-Culture Systems ',\n",
       "  'summary': 'Molecular imaging offers a non-invasive approach for early detection and monitoring of diseases, but it also has limitations such as cost and complexity. Co-culture systems and organoids, on the other hand, allow for cellular crosstalk investigations, disease modeling, and drug screening. However, they also have limitations in replicating the natural environment. Few studies have utilized co-culture methods in TBI neuroimmunology, but they show potential for further research.'},\n",
       " {'title': 'Challenges and Future Directions in Studying Neuroimmunology of Traumatic Brain Injury ',\n",
       "  'summary': 'The complexity and heterogeneity of TBI, as well as ethical and technical challenges, make it difficult to fully understand and treat the neuroimmunological aspects of this condition. Animal models may not accurately reflect human TBI, and certain neurovascular compartments are often overlooked in research. However, recent advances in technology offer the potential for more detailed investigations at the cellular level. Ethical considerations and the need for sensitive diagnostic tools also pose challenges in clinical research. Moving forward, utilizing biotechnology and emerging tools may provide a more comprehensive understanding of TBI neuroimmunology.'},\n",
       " {'title': 'Understanding the Challenges and Potential of Neuroimmunology in TBI Research ',\n",
       "  'summary': 'This article discusses the ethical and practical challenges that limit our understanding of the chronic phase of TBI and the potential for using neuroimmunology methods to develop diagnostic tools and immunotherapies. The authors highlight the importance of utilizing advancements in biotechnology and emerging tools to investigate the inter and intra-cellular levels of neuroinflammation and neurotoxicity after TBI. They also discuss the potential for immunotherapies, such as depleting antibodies and viral therapy, to attenuate the inflammatory cascade and prevent chronic neurodegeneration. Overall, this article emphasizes the promising translational potential of neuroimmunology in TBI research.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Author Contributions and Conflict of Interest in Scientific ResearchSummary '},\n",
       " {'title': 'T-cell Receptor Gene Therapy for Melanoma ',\n",
       "  'summary': 'This article discusses a study on using T-cell receptor gene therapy to treat established tumors in a murine melanoma model. The results showed promising potential for this therapy in treating melanoma. Additionally, the article also mentions a separate study on identifying potential biomarkers and therapeutic targets for spinal cord injury. Another study explores the relationship between traumatic brain injury, diabetic neuropathy, and altered psychiatric health. Finally, the article discusses the role of Vγ1 and Vγ4 gamma-delta T cells in the immunopathology of traumatic brain injury in males.'},\n",
       " {'title': 'The Role of Gamma-Delta T Cells in Traumatic Brain Injury ',\n",
       "  'summary': 'A study found that Vγ1 and Vγ4 gamma-delta T cells have opposing effects on the immunopathology of traumatic brain injury in males. While Vγ1 cells contribute to inflammation and tissue damage, Vγ4 cells help reduce brain edema and preserve cognitive function. This research could lead to potential therapeutic targets for treating traumatic brain injury.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'PET Imaging for Neuroinflammation and Microglia FunctionSummary '},\n",
       " {'title': 'The Role of Microglia in Synaptic Function and Behavior ',\n",
       "  'summary': 'This article discusses the use of microglia depletion approaches to study the impact of microglia on synaptic function and behavior. It also highlights the development of in vitro models for traumatic brain injury and the potential use of biomaterial-based therapies. Additionally, it explores the role of microglia in axonal injury and the use of two-photon excitation microscopy for studying living cells and tissues. Finally, it discusses the potential of using polyclonal regulatory T cells for immunotherapy in type 1 diabetes.'},\n",
       " {'title': 'Advancements in Cell and Tissue Imaging Techniques ',\n",
       "  'summary': 'This text discusses various imaging techniques, such as two-photon excitation microscopy, that have been developed for studying living cells and tissues. It also explores the use of cell-based therapies for treating traumatic brain injury and the role of immune modulation in brain metabolism and T lymphocyte polarization. These techniques and therapies have the potential to greatly advance our understanding and treatment of various diseases and injuries.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': \"18F-FDG-PET Detects Changes in Brain Metabolism in Alzheimer's Disease ModelSummary \"},\n",
       " {'title': '\"New Treatment for Neuroinflammation after Traumatic Brain Injury\" ',\n",
       "  'summary': 'A recent study has found that a compound called ACT001 can reduce neuroinflammation after a traumatic brain injury by inhibiting the AKT/NFκB/NLRP3 pathway. This pathway is known to play a role in the activation of microglia, which contribute to neuroinflammation. This new treatment shows promise in reducing the harmful effects of neuroinflammation and could potentially improve outcomes for those with traumatic brain injuries. Other potential treatments for neuroinflammation, such as B-cell depletion and CRISPR/Cas9 systems, are also being explored. '},\n",
       " {'title': 'Neuroinflammation and Brain Injury ',\n",
       "  'summary': 'This text discusses various studies on neuroinflammation and brain injury, including the comparison of different models used to study traumatic brain injury in rats, the use of CRISPR/Cas9 systems to edit the central nervous system, imaging techniques to detect glial cell activation and white matter integrity in football players, and the role of activated CD8+ T cells in long-term neurological impairment after brain injury. It also reviews the use of radiopharmaceuticals for PET and SPECT imaging in the past decade.'},\n",
       " {'title': 'Activated CD8+ T cells and Long-Term Neurological Impairment After Traumatic Brain Injury ',\n",
       "  'summary': 'A study in mice found that activated CD8+ T cells can cause long-term neurological impairment after traumatic brain injury. This was demonstrated through PET imaging, which showed increased levels of microglia and cyclooxygenase 2 in the brain. The study also found that silencing IFN-γ binding/signaling in astrocytes and microglia had opposite effects on central nervous system autoimmunity. These findings have implications for understanding the role of immune cells in brain injury and potential treatments for long-term neurological damage.'},\n",
       " {'title': 'The Impact of B Cell Treatment on Central Nervous System Autoimmunity ',\n",
       "  'summary': 'B cell treatment has been shown to promote a neuroprotective environment and improve outcomes in patients with traumatic brain injury and rheumatoid arthritis. This treatment works by modulating the immune response and promoting a balance between B cells and peripheral myeloid cells. This approach has also shown promise in treating refractory metastatic melanoma, highlighting the potential for B cell therapy in various autoimmune and cancer conditions.'},\n",
       " {'title': 'The Efficacy of B-Cell-Targeted Therapy in Rheumatoid Arthritis and its Potential Role in Neuroinflammation ',\n",
       "  'summary': 'This article discusses the effectiveness of rituximab, a B-cell-targeted therapy, in treating rheumatoid arthritis. It also explores the potential role of B-cells in neuroinflammation, specifically in exacerbating acute damage in the central nervous system following traumatic injury. The article also mentions the use of nanomedicine in intrathecal drug delivery for treating neuroinflammation.'},\n",
       " {'title': 'Advancements in Intrathecal Drug Delivery for Traumatic Brain Injury ',\n",
       "  'summary': 'This text discusses the use of nanomedicine in intrathecal drug delivery for treating traumatic brain injury. It also explores the potential benefits of using this technique, such as preventing spasticity and reducing brain inflammation. Additionally, the text highlights the use of aquaporin-4 RNA interference to improve functional recovery after a brain injury. Finally, it mentions a study investigating sensorimotor impairments in individuals with mild traumatic brain injury, providing insight into the long-term effects of this type of injury.'},\n",
       " {'title': 'Investigating Sensorimotor Impairments in Mild Traumatic Brain Injury ',\n",
       "  'summary': 'This text discusses various studies and research on sensorimotor impairments in individuals 4 weeks to 6 months after experiencing a mild traumatic brain injury. It covers topics such as co-culture systems and technologies, the origin and dynamics of macrophages at central nervous system interfaces, and the role of TAK1 in CNS autoimmune inflammation. The text also explores the use of tetracycline-responsive promoters for tight control of gene expression in mammalian cells. Overall, the text provides valuable insights into the effects of mild traumatic brain injury on sensorimotor function and potential treatment options.'},\n",
       " {'title': 'New Insights into CNS Autoimmune Inflammation ',\n",
       "  'summary': 'A study using a new type of gene targeting in microglia has identified TAK1 as a key player in central nervous system autoimmune inflammation. This discovery could lead to new treatments for conditions such as multiple sclerosis. Additionally, the study also found that CD44 plays a role in regulating the differentiation of immune cells involved in autoimmune inflammation. Another study revealed the prevalence of medical and psychiatric comorbidities following traumatic brain injury, highlighting the need for comprehensive care for these patients. Finally, single-cell RNA sequencing of microglia throughout the lifespan and in injured brains showed complex changes in cell state, providing a better understanding of microglia function.'},\n",
       " {'title': 'Single-Cell RNA Sequencing and Imaging in Brain Injury ',\n",
       "  'summary': 'This text discusses recent studies using single-cell RNA sequencing and two-photon imaging techniques to examine changes in microglia and vascular function in the brain after injury. These studies reveal complex changes in cell state and provide insights into potential therapeutic targets for brain injury. Additionally, the text discusses the use of CRISPR-Cas9 technology to selectively activate specific cell types, providing a potential tool for further research in this area.'},\n",
       " {'title': 'Advances in Traumatic Brain Injury Research ',\n",
       "  'summary': 'This text discusses recent advances in research on traumatic brain injury, including the use of positron emission tomography imaging and the association between TBI and chronic disorders. It also explores the potential for cell-specific CRISPR-Cas9 activation and the role of neuroimmunology in TBI.'},\n",
       " {'title': 'The Association of Traumatic Brain Injury with Chronic Disorders ',\n",
       "  'summary': 'This article discusses the link between traumatic brain injury and the development of chronic cardiovascular, endocrine, neurological, and psychiatric disorders. It cites various studies and research, including the neuroimmunology of TBI and the use of rituximab for treatment. It also highlights the potential role of TH17 lymphocytes in promoting inflammation in the central nervous system. This information suggests the need for a paradigm shift in understanding and treating TBI, as well as the importance of early detection and intervention to prevent long-term health complications.'},\n",
       " {'title': 'Novel Monoclonal Antibody for Isolation of Astrocytes ',\n",
       "  'summary': 'A new monoclonal antibody, Anti-ACSA-2, has been identified for the isolation of living neonatal and adult astrocytes. This antibody has potential applications in studying central nervous system inflammation and traumatic brain injury. Additionally, the Cre-LoxP system and human organoids are discussed as model systems for understanding tissue-specific roles of target genes and for studying human biology and medicine. Depletion of regulatory T cells has been found to increase T cell brain infiltration, reactive astrogliosis, and interferon-γ gene expression in acute experimental traumatic brain injury.'},\n",
       " {'title': 'Human Organoids: Advancements in Modeling Human Biology and Medicine ',\n",
       "  'summary': 'This text discusses the use of human organoids as model systems for studying human biology and medicine. It highlights recent research on the depletion of regulatory T cells in traumatic brain injury, the TREM2-APOE pathway in neurodegenerative diseases, PET imaging of neuroinflammation, and B cell depletion therapies in autoimmune diseases. These studies demonstrate the potential of human organoids in advancing our understanding of various diseases and developing effective treatments.'},\n",
       " {'title': 'Advances in B Cell Depletion Therapies for Autoimmune Disease ',\n",
       "  'summary': 'This article discusses the latest developments and insights into B cell depletion therapies for autoimmune diseases. It covers the role of DNA methylation in T cell development and survival, the neuroprotective effects of aquaporin-4 gene silencing in traumatic brain injury, and the potential of adoptive regulatory T-cell therapy for cerebral ischemia. It also explores a novel method for delivering siRNA in traumatic brain injury and a 3D brain-like tissue culture model for studying controlled cortical impact injury. Overall, this article provides a comprehensive overview of the current state of B cell depletion therapies in treating autoimmune diseases.'},\n",
       " {'title': 'Advancements in Treating Traumatic Brain Injury ',\n",
       "  'summary': 'This text discusses various studies and methods for treating traumatic brain injury, including the use of siRNA delivery, 3D brain-like tissue cultures, regulatory T cells, and nerve growth factor gene therapy. These advancements show promise in improving cognitive function and reducing neuroinflammation after brain injury. Additionally, the use of diffusion tensor imaging has revealed white matter abnormalities in individuals with mild traumatic brain injury and cognitive disability, providing further insight into the pathophysiology of this condition. These advancements have the potential to greatly improve the treatment and outcomes of traumatic brain injury.'},\n",
       " {'title': 'Biomaterials and Brain Injury ',\n",
       "  'summary': 'This text discusses the use of biomaterials in studying brain injuries, specifically mild traumatic brain injuries (TBI) and their effects on cognitive disability. It also explores the potential of using PET probes to image cyclooxygenase-2 expression in brain injuries. Additionally, it delves into the immunopathology of B lymphocytes during stroke-induced injury and repair. The text highlights the importance of biomaterials in understanding and treating brain injuries, and the potential for further research in this area.'},\n",
       " {'title': 'The Role of B Lymphocytes in Stroke-Induced Injury and Repair ',\n",
       "  'summary': 'This article discusses the role of B lymphocytes in the immunopathology of stroke-induced injury and repair. It also explores potential therapeutic strategies, such as regulatory T-cell therapy, for reducing brain damage and inflammation after stroke. Additionally, the article highlights the use of tau positron emission tomography to diagnose chronic traumatic encephalopathy in a former football player and the impact of microglia depletion on brain injury in aged mice after acute ischemic stroke.'},\n",
       " {'title': 'The Impact of Microglia Depletion on Brain Injury After Stroke in Aged Mice ',\n",
       "  'summary': \"A study found that depleting microglia, a type of immune cell, in aged mice increased brain injury after a stroke. This was observed through PET imaging of brain inflammation and analysis of brain tissue. The study also explored the potential of using tissue-specific CRISPR technology for in vivo screening in Drosophila. Additionally, the role of adenosine A(2A) receptors in Parkinson's disease and the ability of cerebrospinal fluid to support cortical cell viability and proliferation were also investigated. These findings have implications for understanding and potentially treating brain injuries and diseases.\"},\n",
       " {'title': 'Title ',\n",
       "  'summary': \"Measuring Adenosine A(2A) Receptors in Parkinson's Disease PatientsSummary \"},\n",
       " {'title': 'The Role of Blood-Brain Barrier Cross-Talk in Neurological Disorders ',\n",
       "  'summary': \"This article discusses the potential implications of the Kallikrein-Kinin system in neurological disorders and the use of regulatory T cells to prevent allograft rejection. It also explores how brain-derived neurotrophic factor can affect cell migration in the developing mouse cerebral cortex and the use of [11C]PK11195 binding as a potential biomarker for Alzheimer's disease and progressive supranuclear palsy. The article also provides an evolutionary perspective on the use of mouse models in studying human diseases.\"},\n",
       " {'title': 'The Role of Microglia in Neurodegenerative Disorders ',\n",
       "  'summary': \"This text discusses the role of microglia, a type of immune cell in the brain, in neurodegenerative disorders such as Alzheimer's disease and progressive supranuclear palsy. It also explores the use of mouse models to study these diseases and the potential of regulatory B cells and γδ T cells in controlling inflammation and neurodegeneration. Additionally, it highlights the impact of brain injury on microglia and its association with inflammatory neurodegeneration.\"},\n",
       " {'title': 'The Role of Brain Injury in Accelerating Age-Related Inflammation and Neurodegeneration ',\n",
       "  'summary': 'This article discusses the impact of brain injury on the development of a reversible age-related microglial phenotype, which is associated with inflammatory neurodegeneration. It also explores the potential for using adoptive cell transfer as a clinical path for effective cancer immunotherapy. Additionally, the role of microglial control of astrocytes in response to microbial metabolites is examined, along with the advances and future challenges of single-cell RNA sequencing. The article also delves into the role of gut-licensed IFNγ+ NK cells in driving anti-inflammatory astrocytes and the use of site-specific DNA recombination in mammalian cells.'},\n",
       " {'title': 'The Role of Gut-Licensed IFNγ+ NK Cells in Driving Anti-Inflammatory Astrocytes ',\n",
       "  'summary': 'A study found that gut-licensed IFNγ+ NK cells play a crucial role in promoting the development of anti-inflammatory astrocytes, which can help reduce inflammation in the brain. This discovery could have implications for treating conditions such as stroke and brain trauma.'},\n",
       " {'title': 'The Role of Inflammation in Traumatic Brain Injury ',\n",
       "  'summary': 'This text discusses the role of inflammation in traumatic brain injury and potential treatments to reduce chronic microglial activation and improve long-term outcomes. Studies have shown that minocycline can reduce inflammation but may also increase neurodegeneration. Neutrophil elastase has also been identified as a mediator of acute pathogenesis and a determinant of long-term behavioral recovery. Additionally, cerebral interleukin-17-producing gammadeltaT cells have been found to play a pivotal role in delayed ischemic brain injury. Positron emission tomography and intraparenchymal application of mature B lymphocytes have also been explored as potential treatments for traumatic brain injury.'},\n",
       " {'title': 'Overview of Positron Emission Tomography and B Cell Therapy for Traumatic Brain Injury ',\n",
       "  'summary': 'This text provides an overview of positron emission tomography and its use in studying traumatic brain injury. It also discusses recent research on the potential therapeutic role of B cells in promoting tissue repair and functional recovery after brain injury. The authors highlight the importance of understanding the separate pathological processes of inflammatory leukocytic recruitment and neuronal degeneration in brain injury, as well as the potential for microglial-oligodendrocyte interactions to aid in myelination and neurological function recovery.'},\n",
       " {'title': 'Microglial-Oligodendrocyte Interactions in Myelination and Neurological Function Recovery after Traumatic Brain Injury ',\n",
       "  'summary': \"This article discusses the role of microglial-oligodendrocyte interactions in the process of myelination and recovery of neurological function after traumatic brain injury. It also explores the potential implications of depleting certain leukocytes on wound healing and neurological outcomes, as well as the role of human choroid plexus growth factors in Alzheimer's disease. Additionally, the article discusses the early source of IFN-γ in spinal cord injury and how heterozygosity for a mutation of the NF-κB innate immune response transcription factor relish can increase survival following traumatic brain injury in drosophila.\"},\n",
       " {'title': 'The Impact of Heterozygosity for NF-κB Mutation on Survival Following Traumatic Brain Injury in Drosophila ',\n",
       "  'summary': 'A study found that heterozygosity for a mutation of the NF-κB innate immune response transcription factor relish increased survival in drosophila with traumatic brain injury. This finding could potentially lead to new treatments for brain injuries in humans.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'The Role of Neuroinflammation in Stroke and Neurodegenerative DiseasesSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': \"Advancements in Neuroimaging for Parkinson's Disease and NeuroinflammationSummary \"},\n",
       " {'title': 'Title ',\n",
       "  'summary': \"Investigating Traumatic Brain Injury and Alzheimer's Disease in the ADNI CohortSummary \"},\n",
       " {'title': 'The Role of Neutrophils and Astrocytes in Traumatic Brain Injury ',\n",
       "  'summary': 'This text discusses the role of neutrophils and astrocytes in traumatic brain injury, specifically in relation to blood-brain barrier permeability and CNS inflammation. It also mentions potential drug targets and animal models for studying traumatic brain injury.'},\n",
       " {'title': 'Aquaporin-4 as a Potential Drug Target for Traumatic Brain Injury ',\n",
       "  'summary': 'This article discusses the potential of targeting aquaporin-4 as a treatment for traumatic brain injury, which can worsen brain edema. It also mentions various animal models used to study traumatic brain injury and the role of TCR signaling in maintaining and functioning of dendritic epidermal T cells. Additionally, it explores the use of B cell depletion therapies in autoimmune diseases and the development of autoantibodies in response to traumatic brain injury. The article also discusses the use of intravital imaging to study traumatic brain injury in mice.'},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'A Simple Method for Detecting Ectopic Cre Recombinase Expression in Mouse LinesSummary '},\n",
       " {'title': 'Title ',\n",
       "  'summary': 'Intricate 3D Architecture of a DNA Mimic of GFPSummary '}]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stage_1_outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Split the titles and summaries\n",
    "stage_1_summaries = [e['summary'] for e in stage_1_outputs]\n",
    "stage_1_titles = [e['title'] for e in stage_1_outputs]\n",
    "num_1_chunks = len(stage_1_summaries)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "259"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "num_1_chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use OpenAI to embed the summaries and titles. Size of _embeds: (num_chunks x 1536)\n",
    "openai_embed = OpenAIEmbeddings()\n",
    "\n",
    "summary_embeds = np.array(openai_embed.embed_documents(stage_1_summaries))\n",
    "title_embeds = np.array(openai_embed.embed_documents(stage_1_titles))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Getting similarity matrix of chunk summaries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get similarity matrix between the embeddings of the chunk summaries\n",
    "summary_similarity_matrix = np.zeros((num_1_chunks, num_1_chunks))\n",
    "summary_similarity_matrix[:] = np.nan\n",
    "\n",
    "for row in range(num_1_chunks):\n",
    "  for col in range(row, num_1_chunks):\n",
    "    # Calculate cosine similarity between the two vectors\n",
    "    similarity = 1- cosine(summary_embeds[row], summary_embeds[col])\n",
    "    summary_similarity_matrix[row, col] = similarity\n",
    "    summary_similarity_matrix[col, row] = similarity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.image.AxesImage at 0x7fc2b115a510>"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAaAAAAGfCAYAAAAZGgYhAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8g+/7EAAAACXBIWXMAAA9hAAAPYQGoP6dpAABkVklEQVR4nO29e3jcVb39v3KbmWSSzOR+aZM2vbeUltJCCeVOgcMXEISj6MEDKj8vWFDA81Xq9ygejlK8HEG0FEEEL2AVj6CggFikSGm5hBZKS0uvSdo098xMMslMbvP7g8doutcbGy5+0rpez5PngXd2P7P3/uzP7Ez2ylppqVQqBSGEEOIfTLrXHRBCCPHPiTYgIYQQnqANSAghhCdoAxJCCOEJ2oCEEEJ4gjYgIYQQnqANSAghhCdoAxJCCOEJ2oCEEEJ4gjYgIYQQnpD5Xl145cqV+Na3voXm5mbMnz8f3/ve93D88cf/3X83PDyMpqYm5OXlIS0t7b3qnhBCiPeIVCqF7u5uVFZWIj39LT7npN4DVq9enfL5fKkf/ehHqS1btqQ+8YlPpMLhcKqlpeXv/tvGxsYUAH3pS1/60tdh/tXY2PiW7/dpqdS7b0a6ePFiHHfccfj+978P4M1PNVVVVbjmmmtwww03vOW/jUajCIfD8M25AmkZvlHfO+aDFzvtz55XRq8zMMivX5ib4dR+8dw+2vb8hZW0/v6jKmj9zucbnFqif5i2PRDppfVde7ucWnFxkLY9YUYxrQcy3J84ggH+U0hZ0Efrw2RZ5GTxD8yJwSFaL87203pXst+pWZ91T5zEx/jEjman1pPk/SjL5WOMJN1F0hwboG0vP3YirbN5+u3rB2jbMyaX0PoPX+Lr78pF7mu+3OyuD4DfcwCYHHLXzsPb22jbSWE+T/1D7hp+fneEti3Jz+b987n9s16vd4A/M7vbE0b/3PueH+DXbov10Xpzp/s8njCzlLatb4/TelVRjlPrM57/oiB/ltLZk2A8HElyXwDguMp8Wn+ttcep/evR/D3ud9vd5wsALl9Y7dS+9+wet2+9Pfje5aciEokgFArRawHvwa/g+vv7UVdXh+XLl4/U0tPTsXTpUqxfv97taDKJZDI58v/d3d0AgLQMn7MBZQbchykQzKP9SB/k+2o2ufHsum917bw8foP9OblObTiTL5KsJH/DyPC7b85W/9jrAYA/0712IOBuvACQbbw5j2UDSjM2oKCxASWzDn0Dysvnc50ddN8EBjN5P3KMMSYz3Q3IP+T2DbDvOZunQNB90AEg17iGz7iPrH12N//JKkDuOQAEc91r+3P4m3AgyO9XOnmjy8rm/fDl8A3ITzYg6/WGjQ3I18vXX4qsP182v+dZA/w5yAy4K9B6vrKy+Wr157gb0JDx/PvHsAGZJxGD/No5ufx9KxB3L2St6wB5vgAgnzyP/iCfJwB/9xjlXRchtLe3Y2hoCGVloz+ZlJWVobnZ3VVXrFiBUCg08lVVVfVud0kIIcQ4xHMV3PLlyxGNRke+Ghsbve6SEEKIfwDv+q/giouLkZGRgZaWllH1lpYWlJeXO+39fj/8fvej+DEfvNj51VPd/b9w2jU1/R/ajzkz+dlBYsD9uN6wt4O2/UNOFq2/fw4/A3p8vXsGNGT8nnZ4mP+KcOqUQqe2dSv/fWxLC/9VT3m5+5G4vZ2fOV16Wg2tB/3urypebuDzdOr0MK0nfXz+BsicsF9lAW+eZDIao0mnFvTxX6/UR9y2APCzp3Y7tbOO55/A+/r5r/eGyH1sjPBf4/mMX5Ote6WJ1i+a454Z/WjtXtq2ppL/nn1upTv2da/x9fR6mP/6LCPd/TXKG7v4WjhmDj+TZddIGL9q6+jm96uli6/hUtLvzS2dtO0Aef4BoLHBbT9/Cn8P6Unwc8JAlnt/W8g6BYBs0hYAWmNue/a8vBXHVvJfwUX63F+bxhP8V6l7O3m/2QP5RrP7PjTQx9+bDuZd/wTk8/mwcOFCrFmzZqQ2PDyMNWvWoLa29t1+OSGEEIcp78nfAV1//fW44oorsGjRIhx//PG47bbbEI/H8bGPfey9eDkhhBCHIe/JBnTppZeira0NX/nKV9Dc3IxjjjkGjz/+uCNMEEII8c/Le+aEcPXVV+Pqq69+ry4vhBDiMOc924DeKWfPK3P+DocJDg786ff030+e/G+0Pki089Nm8D84Cwb4IfpY/na3t4f/8VxxKdfff2ChK3B41Dhc3/pGO60zgUPLfn5oHEu4f1gG8APVhHEQbx26Z2fyfveRv9uwDj1PnsxfM4vYe+QYB7vJIX6/Wg+4f9SZkzWJts3M4H/PkCTrKWkcrlt/EhEOB2g9kOHOX0M9P1zPzuZrtbrQPaD3Getpajk/vO7ucw/dZ0wtom0Lc/nf9rCD9EFDiFNmiCFy/PztqiDojt24tCl2mVEVdmpzynk/LAGBn/wx8KRifo0q449w2XPXTIQJAJBv3PO8LF5nsPULACHj7wZjZC1Ukz/ATfYemnDCcxm2EEKIf060AQkhhPAEbUBCCCE8QRuQEEIIT9AGJIQQwhPGrQpuYNB1tGb2Opbabf29D9D6Rddd6dQsdc2goZ6KxLkVR16eqwAqKeFO1lmGkqaNXNtSBc0+bQqtb9kfc2oXnjuXtt3Z0k3rDR2uCubsOVz5tG5XlNYtq5V+Mq8HurhDs8WeDrd9RYgrsDqM+3XB2bOd2g7D3mhbK58nFjmxr4M7CTcYVjKWXVNjj9v+jBP5PS/O52OvrXIteizLoomGMoupPtftcdcYALy6myszM4hCrKKIPxt1DRHevwqu0usnqsp9xrqeN43b61SG3LHXNfBr7CWxBgAwtcJVtrZE+LruH+Qq2ARRpe3Yz5+vfCNK5YyaAlrvjLtq1Y4+rrDbVB/h/VvgRoT871O7nNpwkq/1g9EnICGEEJ6gDUgIIYQnaAMSQgjhCdqAhBBCeII2ICGEEJ4wblVwhbkZyD4oN52FyTFvN4Cr3QDg4VvvcWpnf+YK2rbNUGYlF7pKEADoJkFaublcqZLt41Nfke+2397CFSWv7ePqmAEyJ5veaKNtjzZUQWVEVbW9jfvahQw1TjkZCwDEk+59HE5xpV+b4YPFFG8DhmoxnM3n+g91+53aqcdU0rYZhpEb8+6qKHS9sd6K4gLefh8JM3tpCw+TKy7mijJmYbe7lav0MtJ5qF1H3A0t8xs+f7MnuYGKAFBG1kJi0AhlLOVjsbzjMknYHXsGAKDfCKRLDLjXPrrSDXYEgBKidgW4v9s2I4SwPM8Ia2RjnMLVp9Feru60mER8AdncAbYvYIiEdH76fbOcWiLeja+u+vt90icgIYQQnqANSAghhCdoAxJCCOEJ2oCEEEJ4wrgVIfziuX3IDIw+jGzY64aqWWFylr0OExz84Y4f07YnfvwyWs8P8GszwUEkwg/umxP8APF+YivS3s5FCEUkCAoAAlnuAXGpcUg9t9I6vHYPJ3cboXFWuNZsw2qlPnbotjvWfSwJunV6gAsgz88PzP2kbuTOYUI+F0n4yCGzJXrI8/GD5yxiUwMANQVuUN2ECdzChd1zAJhILGZea+TilViCH9C397gWLlv38GC82qPKaL2fCA6sAMGCHD5/dfW832yNWCKfnoQrqACAx7Y1OrXLz6yhbdl8AEBJrvuaQ8aa7Ojl/bDaM1gQH2ALZljZCq8rzTVEEsQ2KosIGYYMccPB6BOQEEIIT9AGJIQQwhO0AQkhhPAEbUBCCCE8QRuQEEIITxi3KrjzF1YiEBxtB/EHYgMRDHC1hhUmx+x1LLXbcz+6n1/jzBtpnTFsqFoKCriq6oL55U7tpXoe/pVuKE26+1yFHQvtAoAX93Jl0Rwj/IthWeC09HIFIFNbWUq1PENx2EbsYYI+/vOUZeEySJRZMUMllTSsXYZJWFufEcSXZUnsDHIy3bH3kXsL2OuMud1YAYe+TN6/vGz3GbPUbpY9TFWhq+jLNNRaPYYaL5TD1ZZMNddMbIwAIGCE8Z167ASnZgX3FRrqs+qQO8ZOQ+1WU8jtfCJk7Jv38wC8XOPZGAt9xvsCC3wEgAzynsNCNJPGuA9Gn4CEEEJ4gjYgIYQQnqANSAghhCdoAxJCCOEJ2oCEEEJ4QloqRWQ8HhKLxRAKhbBpVzPy8kb7XjG1ldX9CFFmAECSeBlZ3m5tca6kufCy/6L1X/zkP51aSbarjAFsdRcLfIoZyqdsQ6XTSwLfmIcTwFUtAFcRbuvkarwCP1cn1RRyL7i+frd/DVHud/fCfv6aZ9S4wWeWB9bAMB97JfF329rGX29eeZjWNzdHnNqUMA8ye2BzE61ffoyrwAKA+og7J6U5XD3lN7zgNrZ0ObXaiTzgbHtbN63XFLj30XrXyDN88ILEr63T8FTzGx5xlk/apgMRpzaziKs4rWeGqff2RHlw3zEVYVpPJ+uPPc8A0NjB1zvz9LOeXWssHd18Xtns/fy1A7TtxxbwNZkgCs/iPPf57+mO4djpFYhGo8jP5/6FgD4BCSGE8AhtQEIIITxBG5AQQghP0AYkhBDCE7QBCSGE8IRx6wV35/MN8OeMVhM9vr7hkP99Xh5XC3V3u8o2lmT6VjC1GwBcevnXnFpw/km07eRp3Evr4toqp/bMdjcJFgAOtHCPqOJiNyk1GuW+bItm8UTZCSF3/rY0cZVUYS6f62nFvH/dRKXX0s2Vfh862vXGA4CHtrU6tTw//3nK8qr7/fP7nNoxM0toW0vZVhl0lXTf/vNu2vZS4vMHAFet3kjrHz7RXQvLf/EqbVtWxvt33BRXLbjyiRdo2wnGNTLS3Xm1lGpWMitT6bE0WQBIGl56sT6u7mIecfc85yacArZyjFFlpAg/sqWN1k+oCTm1Lc1c7VZTxNWx+6PuGJsjh54gDAAfMNbZC/tchWdtNVcLfnPtLlr/+rmz3NpTO51afy9/9g9Gn4CEEEJ4gjYgIYQQnqANSAghhCdoAxJCCOEJ41aEkOgfxnDm6MPIIWJJ0dvDD9dLSvgBIhMcRCL8GlbIl2WvwwQH8VeepW27y86ndWaNYx2cDgzwMKl+YnUzaASq7evgdiMBcsjcGuWHoeEgF3HwV+ShW6W53LKk1BCTZJF5sqx4cgKHfvA8aNj2mPeAzOukIh74VpLNx5Jm9HtinrvOrLZGGWwJZxqH/77MQ58nyxYnEudCgQIiVEka67fEuOcJoz2zBcoh1j+AbT3Vk3BFMJMLeD82G9ZYARLoZ9nosLYAUBJ0+90es4IW+bWTQ3yemsj73InVrnACsIM+GcxWLDl8aFuLPgEJIYTwBG1AQgghPEEbkBBCCE/QBiSEEMITtAEJIYTwhDGr4J555hl861vfQl1dHQ4cOICHHnoIF1100cj3U6kUbrzxRtx9992IRCJYsmQJVq1ahenTp4/pdQ5EepGVHL0/MlVacSkPO8oyrEKyfe6Qm4kCBgAKCriayQqTY/Y6ltqt4Q+P0vqzU69waj2G6oZZ7gBAnITxhcNcuTexiKsFmdpqSjmf63ISSAUAJTl8npiCyp/BVUGFhk3SALlGV98gbTu7jN/HbBKeNmTY9vgN5Vgm6XdbD79fuca66TYCxHIy3fb5+Xw+2LoGgHC2q2zLMuxyLEulLhLMeMAIVMs2xsjUYJZtT6thGxXp5nU2HkshasGu0RDhgZSWNc7WFhLMluBrcmszv0Y86bbfuT9C24bz+TNtJYxWkveAUkPR2xHjc82CBfd3umMZ6Ds0+6AxfwKKx+OYP38+Vq5cSb//zW9+E7fffjvuvPNOPP/88wgGgzjnnHOQSPABCSGE+OdkzJ+Azj33XJx77rn0e6lUCrfddhv+8z//ExdeeCEA4Cc/+QnKysrw8MMP40Mf+pDzb5LJJJLJv/6kEYvxSGQhhBBHFu/qGdCePXvQ3NyMpUuXjtRCoRAWL16M9evX03+zYsUKhEKhka+qKtcBWAghxJHHu7oBNTc3AwDKykafhZSVlY1872CWL1+OaDQ68tXYyG3UhRBCHFl4bsXj9/vh9/PDTyGEEEcu7+oGVF7+ZhBSS0sLKioqRuotLS045phjxnStXXu7kOEfrQ6aSsK1PrCwwqkBQBtRggFABVER3T/IFTMXGMFOoRzuk8TC5CzvKaZ2A4A/rvqxUzvho//G+3csH3tnr6ukmVXClWB+w/9rkKjMynL4Dwr9ht9VUYC3L81xVV+ZxjxZSqmFE9zwtBQzBQMQ9nPl2KfOrHFqLBAMsL3WmEfcYiPky1oLpx9bSevlxAvuU6e7fQbs+ZtR4PalvpOru+ZWGKrKfrcfjxtKsOoSHmqXQ+bJ8v/rIusXABKDVmCeO/YcQwXbaigO97S4YYthQ9FXTO4LAIRI+0AJn9NMY0G1kfLRU4r4NUhQIGB7Ds4udZ/TUhI8CQBHV3HFK+OoSneNJeNp+M0h/Nt39VdwNTU1KC8vx5o1a0ZqsVgMzz//PGpra9/NlxJCCHGYM+ZPQD09Pdi5868RrHv27MGmTZtQWFiI6upqXHvttfja176G6dOno6amBl/+8pdRWVk56m+FhBBCiDFvQC+99BJOP/30kf+//vrrAQBXXHEF7rvvPnzhC19APB7HJz/5SUQiEZx00kl4/PHHEQjwj61CCCH+ORnzBnTaaaeZv2sH3vwL+ptuugk33XTTO+qYEEKIIxvPVXAWxcVBZAZG28Rs3epKuR81gsLKwvzQfXuLayHS3s5tRV6q538Uu3RKCa0/s73DqVlBZpa9DhMcbLjvAdq2/LoraZ39fNBk2IosMg7MC7LdA+LHd7jjA2yrGyOzDNF+d+zWIXqXEXDW0+8eVOcYgoVIkl/jj9s7nVqeEcSVHOBCCzbXcRIICADxJK+/Wt9F6/umuofPa3dFaNswuV9WX1qMYMGUYeLCrJM6Ovg1plbww2sWJhfp4/c8Yjwb7YY9jJ/c93TjkL/FEE/EYu7zsS/KnxkrmLGaBBFaggqfEUjHrHssIY4V0Jcwwif7ic1UxBBrRfr4tdlzuoeshf7e98iKRwghhHg30AYkhBDCE7QBCSGE8ARtQEIIITxBG5AQQghPGLcquBNmFMOfM9p6o6Wlx2m39Y12+u9nnzaF1l/bF3VqRUXcLiPdUGZZyrYDpH9WMJYVJsfsdSy128O33kPrBYvPcGp+w55jwcTZtJ4g9kTPvd5C255WM4v3w7DAYdMazOJLMTOD/4w0QBQ92SR8DQDysrhC7Ol1e5zaKbWTaVsLpkQqIEF3gB1qx8L/AKAg4M7f+lcO0Lbl5dymJttX7NRerOOGv7PmcOsppnw67ig3fBEAooZqMWYo2xhzJoZovTfJ5zWHhKRZc9pvKMQC5JmeU8afUYvZpW77uv3uewIAzCrlfxcZS7j3fHsrV5T1ESUoAAwbfybDpqTPeH9iwXgAV9hNJ+NOxA8tEFCfgIQQQniCNiAhhBCeoA1ICCGEJ2gDEkII4QnagIQQQnjCuFXBBTLSHdUQU/oMG4ZjW/ZzH7cBouKwvJa6DeVOr+HpxZRt/ZYvmOHBxMLkLO9XpnYDgK7nn3Jq8y/9AG37iqHSyfa5SyM/lyvpmuJcpdOZ4F5aceKrVmh4mU0Z5uquF/a693feRN62PJffg8Ji1wdvs6Gq7DTUcWzsjUao3YwC3g9LKck87HyGAjNpqJZYMFvVJDfYEQAml/L5Y/QZ63rrTu4XyPrX22Oou4gfGmCHAjK1aqahOMw1vP6Yn9yuDu4919DGn5lNRP24t5W3bSfecwAQCroquG0N3CvQMoXuHeRelQdi7ntOTlactt3d7Ab0AUAvuY+bGlxl8UAfH/fB6BOQEEIIT9AGJIQQwhO0AQkhhPAEbUBCCCE8YdyKEIKBdAQCow8GWXBcy35+6HnhuXNpfdMbbU6ttDhIWgL9xI4GAAaGuJ1HNOoeWg4a1h/hMLfimFXihlpZYXKWvQ4THLzyiwdp2yX/dTWtZ2W4B7tB4wC3McIP3avC3IqHhaQ1GmOcVxamdXZY29LNhR2JAX5Y20zWzqc+uIC2DRK7FwCYmuUe3D/xhht0BwCB6WP7eY+dMXd18vDEUIgLC/YTQYS1JrMM26MkEUlY4p8T5rlWUgAPtcskawwACnL4OmsxAuJYENwOYosF2EF1GaQvM8izCADZRNgBAFVh93kMGG2nFPJnt5ms4eEaN5gQAAaH+X30ZXChyuvN7pycaghSZk7gdkjsPs6scMU8yV5DMXIQ+gQkhBDCE7QBCSGE8ARtQEIIITxBG5AQQghP0AYkhBDCE8atCq4s6EN27mil06Wn1TjtYolq+u93tnAriaOnuQFdcyu5Cu7Fva7FBABkGEF1i2aVOrV9HdzqYmIRf01/pqtgWVTtqkwAO0yO2etYarc7bvw+rf/bDZ9yamfP4Wqc3n6uxjHEghSm/gNsu5HKfFcpNSGfq+76DNXXVZe6ircdLfx+tU/iCiwfUY5NKeIKx2iCq/RmVxXQemzAbX/WSe4zYPUDAD4811WlMbUWAEwr5PfAT1RVu7r4PP1+cyutM5sfS1loMbGQz2uCWDsVBPkYJxnqs4kht26pzHoMK648v3sP3mgzQuMKeD/K89x1zZSMAJBj2DJlGkq/xZNdZVuPEWpnBUEy27JeMv9JUmPoE5AQQghP0AYkhBDCE7QBCSGE8ARtQEIIITxBG5AQQghPGLcquOFUCsMHKaCCfleBYXktNXRwhUhZvqs+yTBUI3OIxxEADA5xZdYEoqSx+pdmvOYg8VoqMMLaEoZXHQuTY95uAFe7AcADt/zAqc27/fO07XO7eGDWFCPg7OD7CgDbW3k42cmTebjWxgY3kG6TYT9l3a/Zle79nVnG+xwf4Gqh0lxXzdjQxRVziyfwflieg2yeeoyQxFAOVwB2Jt2+VBClFQBMDnFlJlNKWet3luEhVkjC2qxnI5zNn916Y17zA277iDFPASOojnnEhf18ToOG+iyWcO+jNU/7DP/EeSR005fJn41AJr92t7FWmTp2Yj5XFhbm8DEyATBT//kGD+2zjT4BCSGE8ARtQEIIITxBG5AQQghP0AYkhBDCE8atCCEnKxM5WaO793KDGyCWIOFmgG0bs73NDY3b3ckPNy22dboH4ACwpcm1/2mN8gPEKeX5tF6W4woZHt/BQ/eee72F1vNz3WtYYXLWPDHBwQ2f/R/aduUP/i+tFxuBee197nxPyucH4PUdPIDt9JlukFaRIdbIyeTL/Jt/2OHUWKggAFx45fG0vra+3aldPr+Stn2uiYs1+g3Poom5OU6tz7BOYYFqAPDtJ3c6tbIC97oA8MPOBlqfV+POdUEOn1MmCLDY2c6fjSU1/NkIEyEDAGwgIphTZ/CgNSNHD9va3HUWMsZSU8TX9cxCV0BQmsuFDCzYDQCGiPCE2U69FTnEzgsAzprhWj797nV3/QJAdQEXJ3TGXXFHNbF26sviIouD0ScgIYQQnqANSAghhCdoAxJCCOEJ2oCEEEJ4gjYgIYQQnjBuVXCJwSGkHWRRcur0sNOu0bC0WLeLh8mFgq4qpSrMlSoDhoVLgWHRUUjUZ2HyegBQnsfrTBE1u4wHhZ1WM4vWm+KuusiaJytMjtnrWGq3ZZ/6Fq0v+siHaJ2FzMViXIl4ywfn0fpPnql3aj09fIyBAF/m/7JoglOzlFbWWjhrqhtCeNu6vbTtRbPdtgCw11Bhdibc+vRybg9VHORKqffPdV+z17D+qc7j6jgWSPdKW4S2zSGBZQAQ9rv9m1vGlY9Z6fzn4riheD1rjhsyuaOdqxmnF3N11/Qctz5gKNXixjOzO+qG9HXEuWqRBc8BwPCA+5pZRgBmj9GPbEMFt3rjAaf2oQVuYCEA1BFFL8Dt0Jhyj9UY+gQkhBDCE7QBCSGE8ARtQEIIITxBG5AQQghP0AYkhBDCE8akgluxYgV+/etfY9u2bcjOzsaJJ56Ib3zjG5g5c+ZIm0Qigc9//vNYvXo1kskkzjnnHNxxxx0oKysbU8eKs/0IHuQllvS5yhFL8ZEY4AqR8nxXfTa7iKtxWnq5kqamkLefVuwGPvFeACWGl1ZRwFXSWf5VlhqPqacspZ9hQ0bD5CxvN0vt9tLPVtN6cP5JTu3Cc4+ibavCXJl1zDRX+cTC/ACgf4CrpxaQwMGefh5kNmxcmwWO1RRxpVWxERpnKbMqgq768aRJtKmpbOsfdm8wU6QBQGYa/3k0g6iwQn6+frsSXPVVTNZ1xwBXLZZl8/koMeaPBffNq+DrxlL6pcMdYzt5jgBgKMX7za5hqSrZvQWAZvKek2n4/LUTXzYA6DPWQlWhO/Z88p4KACVB3u8gue9MpTdoBPEdzJg+Aa1duxbLli3Dhg0b8OSTT2JgYABnn3024vG/yg+vu+46PPLII3jwwQexdu1aNDU14eKLLx7LywghhPgnYEyfgB5//PFR/3/fffehtLQUdXV1OOWUUxCNRnHPPffggQcewBlnnAEAuPfeezF79mxs2LABJ5xwgnPNZDKJ5N/EBsdi3GlaCCHEkcU7OgOKRt/8Y8/Cwjetz+vq6jAwMIClS5eOtJk1axaqq6uxfv16eo0VK1YgFAqNfFVVVb2TLgkhhDhMeNsb0PDwMK699losWbIEc+fOBQA0NzfD5/MhHA6PaltWVobm5mZ6neXLlyMajY58NTY2vt0uCSGEOIx421Y8y5Ytw2uvvYZnn332HXXA7/fD7+eH20IIIY5c3tYGdPXVV+PRRx/FM888g4kTJ47Uy8vL0d/fj0gkMupTUEtLC8rLy8f0Gl3JfiQPStUbIJItS/HRb3h3xZNu+/oYT2aMJfi1+wxPqm5y7YhxDSsRsTTHVdhEDWWWYRGFOFEAWj5aFkxZxJJMAe7tBnC1GwDEX3F/aBk4azZt293Hx95KkkuHiOILsFWEu7vcFMxAJv+lgCXqYeU+w6PLuuds3QBAa687340xrsy0rj0h31WUWYq5fB/vN7sHltNXWw+/X3k+dyyW51tsgF8jmuR1X4Z7nXrDX6/QUI6myIis17OYkOsq2xq6eaJvcojfAz8ZS6vx/Hd0czWedW/YewB7zgFgbxefP6Y0ZZ55A++FF1wqlcLVV1+Nhx56CE899RRqampGfX/hwoXIysrCmjVrRmrbt29HQ0MDamtrx/JSQgghjnDG9Alo2bJleOCBB/Cb3/wGeXl5I+c6oVAI2dnZCIVCuPLKK3H99dejsLAQ+fn5uOaaa1BbW0sVcEIIIf55GdMGtGrVKgDAaaedNqp+77334qMf/SgA4NZbb0V6ejouueSSUX+IKoQQQvwtY9qArN/1/y2BQAArV67EypUr33anhBBCHPmM20C6NLgHvOzAzArzOtDFhQXDKW6Bwcgj4UsA0BDlB4st3e5hYWkut7rwG/YamURZwGoAEMzit68w233Nxgifp1klfD62t7rzNymfWxBZYXKWvQ4THDz47btp20/98r9pvTzs9jvSyw9ls8jBrkVxDr9f3f3cYibb566RtdvaaNt55a69EQD4M/n97R10X7PNsF+xxBOM/CxrTfL1zuyu6mP8GZhVwq1umI2TdQCeRqUdXIQE8OeD2QcBwP4eI6iuwF3bSeP14gN8LbAgSEtgUs4fJfiIMGNSmFsTVYe4ejjHsCdjc2KNsTrMr50gQgb2emnGWjoYmZEKIYTwBG1AQgghPEEbkBBCCE/QBiSEEMITtAEJIYTwhLTUoWir/4HEYjGEQiFsq29DXn7+qO+xjiaNsDGLNqLYyjHCtfICvH73iw20fs5UNyStNI+rSQpzuSVIIMtVj3TFubor01B3MVuWHiMozLr9TN1V38GVT2YAlhEmNxZrl6Uf/DKt/+qnbr3HUCcNGWOcVxp2ao2GwrHaGAuzdso3Qsh++BI32r1yIXeArydWQVOM8MRoL1fHxYiNS0U+Vz42kNcDgJllbnBfIOvQw+sAIJ8oM60+Bw31qWWB9UazGwRpPbvVRfw+dpG+bGhop23Pn1NJ672kf9Y8dRpqxoFBV5UWNMYSM+aP9QPgtlG/3sZNoj9urElGab77HheLxTChtADRaBT5B72P/y36BCSEEMITtAEJIYTwBG1AQgghPEEbkBBCCE/QBiSEEMITxq0K7rt/3Izs4Gj1TWP00EOt9nRwL7gK4p9UEuQqk7Y4V1WdUVNI64++4apmsgxVEAtxAoCFE1y/sB7Dh2zACN17YW/MqYWCXHVXmc99wTY2uNc4fSYf90+eqaf1Y6a5qkCAh8kxbzcAOG9mEa3/67+7HnHBY3gA3qyjJvD+EdXisHFfPrqAX4Pd3tWvHaBt3zezlNZXb+ZKpBMnu+qhP++O0rbTSrhf2IKykFNbs6eDtj1uAlcr9ZOgv3bDd+/1Fv7clQTddWb52qUZ6X9MIQYABTnu8/vcdu7Hd+bcMlpnwZE5hoIt0sefxxnEV9Eao/WuW0TGsquD+9dZ3oenTAvT+m83tTq1ZSdNpm3v2sCVvt9+n+vveNcLbttkbw9uv3SRVHBCCCHGJ9qAhBBCeII2ICGEEJ6gDUgIIYQnjNtAup7kEAYzRx8MBok9jHVQyMQGAD+4twQBQZ9hN2Ickub53fZW2y7jIJNpQnKIPQ8AZGfz+ryJrpCBheUBwIR8Lk7YRLpdROxUAKCnhx+GDhrzOkQOta0DVctehwkO4puepW3jky+m9VxyfycX8nVjWcwwqxVr3NbBs7FEkO9zH8+qAt6//ABfCyzUzgpatMgkHWTPIgAcVc6tbkpy3HWWMCycrLne28UP44uIiGh3odEPIyCS2TVZ88REDwBQSoQW7D0BAPoNAVEuuee9A1x8Ya2nscjK8n18Pgpz+Tpjt2ZKkdu2L8CfZ+d6h9RKCCGEeJfRBiSEEMITtAEJIYTwBG1AQgghPEEbkBBCCE8Ytyq4slwfcg4KbauPuFY8SUNN0mFYYIRJWJildrHUTANExQVwhV2OoU6aXcatZ8J+Vy0USXJFSV4WV7CU57rqosQAH0ufYW8yyMaSyZdLwAjM6jfCAtm0Zlnheoakh9nrWGq3bQ//mtY/fubnnVpJDlf/NHVzi5nJBW5AnDGlSA7x+bDW375uV/VVkmuEJxL1FADEiYqwylCIZmfyfjClVKTTtWoCgLCxFthaLQxwBWZngq93Sx3H+m1ZOxmXQFGO27+GLvf9BgCqDSViYsi98b4MPqf9Q1zdGUseumox16gHDdVsOhl8Z5KPcWYZVxFmWhP4NtEnICGEEJ6gDUgIIYQnaAMSQgjhCdqAhBBCeII2ICGEEJ4wblVwkeQgkpmjFSE/e2q30671QBf99xecPZvW/1C336n5LRXcIFdg3ffRRbT+++f30Tojm6jxAOBTZ9Y4tT9u76Rtn163h9YLi/OcWvN+HkJ21aULaH12pXuNb/5hB237L4t4WNuCCvcaALC7q5fWGfNKw7TOwuSYtxvA1W4A8IVr/sepXf6lq2jbTyyaSOsswC6QyZVCJUGunnpmSwutf+yUSU7tS3dtoG2LjHl6Hwkce+Y1HoA3ZzIPHOzucxWlBblcwZY0fMsGiELMysK0fMg6e7hiq6vIVWw1tMdp27YY95MLBlwV3KIq11MRAJpiXKXnJ/d9Sztf69ONAMGOHlcFV2944JnzFwjT+vxqN5zQUj5uaeqh9fRj3DE2Rtz5SPZyFbJzvUNqJYQQQrzLaAMSQgjhCdqAhBBCeII2ICGEEJ4wbkUIzbEB+IdGH26ddXyV0y4nyz2oBYAdLfwQ7dRjKp1ahuEuEUtwu4ytbdyG5JiZJU5t0LDtGTIshPZH3QO9PHJACgCn1E6m9c1vtDu1T32Qiw12tPDD2pll7gFsNMoPQ5m9EQD09PODyECm+3NPMbFCAYDGKD/EZYf/VpicZa/DBAc/uXkVbfuJX3+d1u9/tcmpTQzxA/q19e59AYBSIzxtbrF7aHzxuXNp24Rhe1RIwtPmTS2ibZs6+VwvnuKKE5JEVAAAj25opHV2v4aMa0yv4WIIywTmp49udWonnzCZtp1EBAsAt/l55JVW2jbbCOML+sJO7YDxzPT287H7ScBhQxt/L7MwHMRQFXbX5ZO7uMCpyRAKdRGLszea3f4N9B1an/UJSAghhCdoAxJCCOEJ2oCEEEJ4gjYgIYQQnqANSAghhCekpSw/B4+IxWIIhUJ4dXcL8vLyR32vr99V+mQaErZtrd20npHmtp+Qz8OrkkayWMhQbPUQ1ZylmPETJRgAkO6Z9iYWnXFXSRf0c6Vae5zbm7Ags+oQVxCxID6AK58APsbufq44rMjnliXxpLsWrMAyK0yuIpdfm3Hyxf+P1nc//R2n1t7NrVo27ON2SNPC3PIlnUxUUZAr7OIkyAzgc2KtXwsfWatRw2plf4zPdVnQneth460n1wi1s9YZewvrN57d4jyuiOwl7y37DQWmFQTZN+hew2cELSaMcMIiothk8w/w9xsA6DMUkQy/0T/rfauiwL2P2w6477Xxnhj+z8IaRKNR5OfnO9//C/oEJIQQwhO0AQkhhPAEbUBCCCE8QRuQEEIIT9AGJIQQwhPG5AW3atUqrFq1Cnv37gUAHHXUUfjKV76Cc889FwCQSCTw+c9/HqtXr0YymcQ555yDO+64A2VlZWPu2HAq5ahkhoiqylKqFWdztUuAeC1ZKhNLpbO5OULrlUFXTTdg9M9S7zH1iaVTtPy/muKuEmlqFldaWSqd0tygU7O8zM6aWkrraUzuBu7pZalumNoNAJjgjd1bAJhc4I4F4Co95u0GcLUbAEw57Xqn9sIjt9C2ZUYgXdDHH8N84rGXMBSR/iw+fw1R1+vPWk/rDJXekomud1y6oTgs8HOVHlsKERJ0BwC9xrq25mk9CVtcXMH95JiSFuBrYVeE+yQuKi+g9VLynmMpM5mnGsDfL6LGPDFPRQAoMFSOWeRZX1vfRttOCfFn5vUO1wdzZqEbPJnWf2hby5g+AU2cOBG33HIL6urq8NJLL+GMM87AhRdeiC1btgAArrvuOjzyyCN48MEHsXbtWjQ1NeHiiy8ey0sIIYT4J2FMn4AuuOCCUf//9a9/HatWrcKGDRswceJE3HPPPXjggQdwxhlnAADuvfdezJ49Gxs2bMAJJ5xAr5lMJpFM/vVvUWIx7jQthBDiyOJtnwENDQ1h9erViMfjqK2tRV1dHQYGBrB06dKRNrNmzUJ1dTXWr19vXmfFihUIhUIjX1VVbuSCEEKII48xb0CbN29Gbm4u/H4/Pv3pT+Ohhx7CnDlz0NzcDJ/Ph3A4PKp9WVkZmpubzestX74c0Wh05KuxkeeJCCGEOLIYcyDdzJkzsWnTJkSjUfzqV7/CFVdcgbVr177tDvj9fvj97uHdb18/gEBwdKhRY8S1OLFsavZ18APEChL+ZQWq9RnXvvLYibT+7T/vdmqTirjNT1sPP1hcXO0e6MWNg9MCo9+NJNTuiTd48NSUIm5H09DlWvRcPt8N8wOA29btpfUa49p9JIxr7TZ+GLrqg/Np/YcvuT+oDBrWP4YOBIFM94DYCpOz7HWY4OD4C26gbf9shNr96OX9tH7eTPfw//6NB2jbyjCf69Nr3APzupYu2rbcsKnZ3uVarTT38PlgQiEAyCECkXw/F0509HKLmYRxI9m1V73QQNtaNkSTSFhbpI8/d4/18LUaznbHs6uDB9ItnMBFQfUR97mznv8sQ8h05uRiWn9sp9vvi2eX07Y/f42vs/84ZYpTW7Vhr1NLxg8tkG7MG5DP58O0adMAAAsXLsSLL76I7373u7j00kvR39+PSCQy6lNQS0sLysv5IIUQQvzz8o7/Dmh4eBjJZBILFy5EVlYW1qxZM/K97du3o6GhAbW1te/0ZYQQQhxhjOkT0PLly3Huueeiuroa3d3deOCBB/D000/jiSeeQCgUwpVXXonrr78ehYWFyM/PxzXXXIPa2lpTASeEEOKflzFtQK2trbj88stx4MABhEIhzJs3D0888QTOOussAMCtt96K9PR0XHLJJaP+EFUIIYQ4mDFtQPfcc89bfj8QCGDlypVYuXLlO+qUEEKII59xG0j37Gv7kHtQIB2zzDHcXtDQxcOkGHk+y7qCX3y1oRA5eVLYqZUYlkBW6Baz7rDsaKxQOxauZdnURBNcjTdElsXGFv5HwnOLXeUeABTncEUZU0rt7+FBZs81RGn9fTNd+x9rJSeN8K8SYo1j2Q0FDasgZq9TmcuVj1ao3R9/+d+03pFwFVGl2Vzt1tbHgwU7yTXmFIVo222d/P4eVey2zzLWnqWCY1ZL3YbFjGXLZFlmbWh0rXjmlYZp2xxDeRfrc5V3Lx7gytGTq7nKjNn8WM+5papk7zmWzVLCUMd1D/B5zSVBencaasHPLK6mdfb+xO5XT3cMJ8yeoEA6IYQQ4xNtQEIIITxBG5AQQghP0AYkhBDCE7QBCSGE8IRxq4L76H0b4MsZ7Ze07hU3LCxseGANDRlBdQWuFxwLanorvnr2DFq/avVGp2aFsnUbKpjTj3X91l6t595d1rUHjEAvxuwqHq7VP+heo9+Y0wIjaG16Mb833UTV5ye+bABw/gxu4/S99fVOzVJE5hnKp2e2tDi1UuIVCACfO6mG1llI2n0bubfbZfMqaH3pB79M6//70684tbs2cNXSoHFvjiNecOt3cnVXOMhVi0Gi5NrfcegqU4Dfm+I8vj52NXHlY3Y2V6sydVx7J1dVTq7kiqxYr/s8hoz5aOni155f44bgNXXyecozxsICNttj3E/OCsz8zMmTaX1dvTuv588ooW2//fQuWl/1gXlO7dIfPu/UBhNxvPTV86SCE0IIMT7RBiSEEMITtAEJIYTwBG1AQgghPEEbkBBCCE8Ytyq4PxMvuAPdrvokkMEVTo09XH2yL+p6Y9UUcDVOTib3cUo35Fb1Mfc1JxpKH+va5aT9vihX3RQEuEonknQVPdZdjhm+UUxhMzGXK8SY3xgAVAS5J1prr9u+d5CnYIZ8RkIpec18okgDgH3dXEWURXyt5hLfM8BWAJbluwrAXR08DZL56wFAZhr/OfCSf7/Jqd1zD09b7e7n81eTH3RqwSw+TzmGBxtTW7Jn8a2uHQq4qq8BY07biSINAArINQCgM+G23xXhicjHlXPVZx9RfcYH+JyG/XxNMm9G5p0GAK+1caUfu3bA8nw01K5x41nKI15wPcbzXxTgytZ5E9znY2eLO9fxnhj+ZeFkqeCEEEKMT7QBCSGE8ARtQEIIITxBG5AQQghPGFMi6j+Sl5u7kN09+jDtR2v3Ou0a6rmtyBknTqH1l7Y0O7UJE/ghWZ8RmPWdi107CgBY/otXnZpll5Ofzw8yP3W6a/mydleEtl3/Cg/G85HD5C7DEuQsw2Kmh4y9zzjonl7OA+lOmkTLaCTWIm1xPtcfmc8PjR/6c6tTqyrgB6cluXyZf+muDU7t4nPn0rbX1PLBJAbcg/T7N/L78tnaybR+y9M7aZ0JDq688hbatubc99H6xae4r/mnze4zAADnLnBtoACgudu9N4kx2D0B/IDeCqSrNOy1evq5aCFAbJwORLnw5Ont/P2iqtgVa8yr4KKb9fu4gGBGsSu62d3B+1ER4oKKxmi3U6vv5CIfK1zvfYa9zrp9rqXXseX8ve8329po/ahKV4Tw7D53ThNxdxwMfQISQgjhCdqAhBBCeII2ICGEEJ6gDUgIIYQnaAMSQgjhCeNWBRfISHcsKGqIAsMKqSomFikAUEzULoEsriYZHubWKX6jfVlZrlOzQtKyDduYTGLdETbGWF7uvh4AJJOuWi0UcsOyAMBnhPGFclyVXkYGH0xxkPevl9ibAMAQmVfLbiTay5VS00pcpVR+gN+XPGOui0rDTs1Sd8XJnAJ8LVgqrrY+rmaywuSYvY6ldtvz2G9pPefMa51aKQllBIAECUMDgCxy33OIrQsARPv4PDFLGiuUrTSX19HD1wK9Rj6/B0miWgSAohx3jWQYD2+uYVnkJ8+SzwhatK5trWFGe4yvp8QQX8PFQXeM7N4CtoVQkCjvyvLc+9WXZtzDg9AnICGEEJ6gDUgIIYQnaAMSQgjhCdqAhBBCeII2ICGEEJ4wbgPpnnh5L4K5o32K1jZ0uO0TXPFRW8WDxf5cH3FqE0Pcl23QmBmmmAGA7a1uSJchpEM4m6tdzp7i+jhtbInQtvURwyMqy/25Yn+Uh3x9eG4FrXcm3Wt/+0nuWfbZM7jvXv+woapKP/SfewqN0D22aq1QOytYbGOTG6RVaNzbpTXFtN7U495zS+HUbqjg9nbxeu3EsFN7psH18wL4PQeA//6P25zaDd/4HG27nQSLAcAp09x+MA88AJhe6KpMAaA421WlbuuM0bazi/izGze8COtaXG+2o4q5QtRaC81x9x4wRSoAzDX6N0DWe46hmGUhegAPgoz1c/XfoPHmEsg8dJXeC/v5PaidyMdYFXIVlF1EqRrvieG8RTUKpBNCCDE+0QYkhBDCE7QBCSGE8ARtQEIIITxh3FrxPLy9Df6c0Qe8615zg7RY+BoABI367lb3oPW1Rh4wVRZ2A6YAYPnpU2l95RMvOLVMw2ImyzicZOFTLVH3oBsAXqxrpPWqSa7tzqBhs1IV5pZFFcReo8ywcLEsd8L+Q7foyTesXSry+T346aZ9Ti3PCOiqCvExPkPW07ypRbRt6CjePyaGqGvhQoGFZTxc7+cvbqN1JnywwuQsex0mOLjli9+lbY+//MO0/sN6dzyzJ3Nrp60tPPiQnZcn+/m6eTWfX2NPGxdJhIOuUGVLUw9t297NBR/xhHuQPpuIQADg5X382qdMcdvv6eSBdNk+/r7QHHPFCduauFDA4hO11bTeEHPfR86bVkrbfufPu2n9u+93Axuf2uOG1ymQTgghxLhGG5AQQghP0AYkhBDCE7QBCSGE8ARtQEIIITxh3KrgJoV9CARHq5deJ6q0qeV59N9PDHMLl4x012LCsvOxwqS2t3GFxwQSSOczbDEKc7kya26Fq2ZKgVtuzJpTTuuTS91+ZBnBc9MKucpscsi1VPlhZwNtW53HFViZafw1832uIs+fweepoYsroo6bYNt7HEy2cQ/mECVXUyd/PYt1+1x7qPI8fm8t6xmm4gKAHKLkPHdBJW1rhckxex1L7fbCT35O6yd+/DKnNqOMW+5YFlPMnehAjFvMlBMFJmA/j/Udrrpr3gT+vmCFBTaQa7REuPp0MVG7AUBelvt2WprH32KDpC0A9JD3opkVfK13xrmdjxVIN6vQnZONrRHaNsfP+8cCIgPEBiplWEMdjD4BCSGE8ARtQEIIITxBG5AQQghP0AYkhBDCE97RBnTLLbcgLS0N11577UgtkUhg2bJlKCoqQm5uLi655BK0tLS8034KIYQ4wnjbKrgXX3wRP/jBDzBv3rxR9euuuw6/+93v8OCDDyIUCuHqq6/GxRdfjHXr1o3p+v1Dw0gfGq3sySABUd19XElj5ex1xN1AqvYeribJy+ZqnJoCrgDKGEPQWhcJwAKAeL+r0hkygqeswCxGcoArYyz1WQ8J/5pXw/2/rGuw+wXwe2Yp1WaWcTXT842u+izTCILL9/H7yPqxeAofo8/w9Fsy0fWO297FVZJHFfOQr6cDEVpPI+Np7ubrPSuDj52FyTFvN4Cr3QDguR/d79Smfekq2taXyVVmbCkEjDm1Qhy7enmY3KQiV8m5zwhgLAnyt7wEeT5OmMq9+6wwPkbAUJ92Ge9b7JnpSfJnN9vHxxJL8nkKkOf0uHI+xl3t3MOuMNdVbPrJfRw2xn0wb+sTUE9PDy677DLcfffdKCj46wCi0SjuuecefOc738EZZ5yBhQsX4t5778Vzzz2HDRs2vJ2XEkIIcYTytjagZcuW4bzzzsPSpUtH1evq6jAwMDCqPmvWLFRXV2P9+vX0WslkErFYbNSXEEKII58x/wpu9erVePnll/Hiiy8632tubobP50M4HB5VLysrQ3Mzt5FfsWIF/uu//mus3RBCCHGYM6ZPQI2Njfjc5z6H+++/H4EA/13vWFm+fDmi0ejIV2Mjz7gRQghxZDGmT0B1dXVobW3FscceO1IbGhrCM888g+9///t44okn0N/fj0gkMupTUEtLC8rLuW2M3++H3+9alzy/O4Ks7NGHaW/scg+eZxgBYuv28F/l+clh99Y9nbRt7VFltG7oG+An9hOWgOBAB7d8eZzYf3QQmxAAOM7oXx8J+ho2+rGri4d8sQPwghy+XF5pi9B6yLDzYD2pj/H5uLCYW8+097qHzFYIYcSwwCkgB6rJIX7AzCxIACCdHBo3G6KW+WX85739xlo40O3ed3ZYDgA5RqAfOzC3wuQsex0mOPjJzato249/+TO0zgQHQ8aDtC/CBTqtRjDjzmb33uRnc3ujrjhfIzsbIk5tdqVraQUAz+1w34cAYHe7279MQxzSFjWC6sgzUxHiP+wH/Xw9Tcjl9lo7yLNeYbQtNeyQGE9uc98/B/p4aN/BjGkDOvPMM7F58+ZRtY997GOYNWsWvvjFL6KqqgpZWVlYs2YNLrnkEgDA9u3b0dDQgNra2rG8lBBCiCOcMW1AeXl5mDt3dCRrMBhEUVHRSP3KK6/E9ddfj8LCQuTn5+Oaa65BbW0tTjjhhHev10IIIQ573nU37FtvvRXp6em45JJLkEwmcc455+COO+54t19GCCHEYc473oCefvrpUf8fCASwcuVKrFy58p1eWgghxBGMvOCEEEJ4wrgNpCvJz4YvZ7RC45g5rurLCnZ7dXc7rc+e5CqALLWbpXzKy+bTFshyFTYRIzQqO8CvUV3iKm+mGoFUUePaW3e6Kp0T5lXQtr/f3Errsya4tjH5Aa4gyiHjBoCuBLcEaetx53VWCQ+1s+x8Xm9xFUdHlfNrhI25ThKF2KMb+J8BLKkK03qB31VbWcpHq25hhZYxon18rk+dHHZqW1u46s4Kk2P2Opba7Uf/zX/dHpjripCmzZpA24ZC/Jm2GBpy57UjxlVmiwzVbPVxbl8MQSTixnPnK3d/nn9tL1fYnjirlNZLct17/st1fE1mGAq72caz1Eqeu8QgV1X+fhP/u81/X1Dl1Jo6XHXdYOLQgh31CUgIIYQnaAMSQgjhCdqAhBBCeII2ICGEEJ6gDUgIIYQnjFsVXMCXDr9v9P7IFFEDhlQlwwhEKst3VUv9g1ydVFVoeTDxafMTNViBodKz+p1D/Mws/6+YEWqVJIFUlgJrcin3uyo0lH6MsJ/7RhUH+NjzfK7XV3E2b5tvhAKWBN16SQ73/8ozfNLYPbA888qCfC2wDLwc4gkIANmGV52Ro4dQwO03C/8CbLUgm1dLjGf1YyxhckztBgCJ19w4lljl+bTtgLHec4l3H8D9+JiXIQDEiU8iAATIGuk1gue6urjCjj2PQ8ZzPmz44A0QRZ8VrpllqCQn5HIVXA8Ze9h4vvzGWmVrgT0z1nPkXO+QWgkhhBDvMtqAhBBCeII2ICGEEJ6gDUgIIYQnjFsRwqSwD4Hg6ANUFq41aBx2VRTxcK0EERxYh8aZxkFmpxE45iMHs0njQJXZ9gBAaa57KBjpM06HDXp7Dj0YyxJUBMic7CSBWwAw1wgy6xjg85SV7l7bOpS17JDa4oduK1IY4IfX7HB3rIfGEXLwnO/n97bbEI0U53GBAxNJWNfIMw6Tt5EwvqRxEH8gxq89ljA5y16HCQ4a/vAobXvsv32Q1ovDPDytj4huygv4Qbwl/mEiHb/xzCyYw210Jha4/csyxFBZxrW7k+69scQXoSCv9/RzWyYmYKmPcMscS/zTQd77yshcD/QZPkYHoU9AQgghPEEbkBBCCE/QBiSEEMITtAEJIYTwBG1AQgghPGHcquB6B4YxfJDqraPbtXApM5QxdQ0RWp9a6iq2CnL4NPQkuFrIb6jmWMBZSR63mGmNcjuPrl5XwcKUVgAwZ6IbGgcAfSQIriCHq1osWDjZkhoejMdUbQBQls3VXbEBdzxpsFR6ln2N296yo+lMcDUeCzOcXuMGFgJArhFq10tUjh3kHgK2Fc+upiitt/e6/a4M8zll6kkAmF3krpFX87nyqTyPX4MJTfdF3GcRsMPkmL2OpXZ7+YFf0vop/99HaJ1Z8WzZ7YYyAsCxM0tonY1x635+X3qJ6g7gSkRLBdsc5fPny3TXSFubG/gG2HZjiSH+mlHyvjC3mL+HtHRxxStTzW7c0uLUhpMKpBNCCDGO0QYkhBDCE7QBCSGE8ARtQEIIITxBG5AQQghPGLcquN3tCfh6R3evpctVVuQYXmYTK/JonXnH1dVztUvICDizwt1ifa5qyQqTi3RzFVxi0A2Ia4/xtr1JPnZmYddiqG4mGqF79V1u+7ARUmeFfFkBcdGkq4KzPLr6jGsPDLrt9xpBYaY6rscdo+W6x4LCACDoc+ckQfoGcK9AAMg2fLcKSCBdT7/hsdXDlZJx4gu2x1BV+TL56JkyszXKVVIWzM/M8naz1G7P/PBntD71vAudms9QHDYYY48QleiEQu5x+OLrruoLAIKTXQXl7pZu2tZS7wbJPZ9QydWnLAATAAr8/Lnb1s7HzsgxVJ+M6VPdcQ/2+bHnEP6tPgEJIYTwBG1AQgghPEEbkBBCCE/QBiSEEMITtAEJIYTwhHGrgusfGkLqoITLUqIcKQhyBVG/kY6ZSRRRlpLO8ojbdCBC60w1ZwRHIstQsDDFlqV2sfrNvLGqi7jqhqXMAkB+wH3NDbu6aNuz5hTTupUi6iMeVuy+AMAbzT20zu5NUZDPRzbx1wKAriI3yfGnj26lbVPnzKT19ftdzzErYXdDI/cns9RxzMMuYCjVLOpaXIVn2EjSrO/gyrZJZO3sbOaquyFDLcjWJEsytdoCXO0GALt+9xuntviKD9O2lYU8KZUpWzfuaKNtC0jyKQDEyXjys/lc7zcUacxrzUpVtdKMO5Pc+5A9H3821mRRPlfHJonCk/VjYJg/cwejT0BCCCE8QRuQEEIIT9AGJIQQwhO0AQkhhPCEcStCyA/44DvoAG9zS6fTznDFwT7DAoNZuGQTOxXADo2aeXQlrd/zXKNTs4QCLKAL4AfY6cxbBzyUDQAyyaH2jhbjMD/IA8RYCN6pM3hY2452boEzr4If+NZ3uvNq2eUcVcIDs57b7h4Q7zYOmMsN25MGchB88gmTadt+w15ncYU7J6teaKBtP3civ/btnbtofVfE7d8BI8iw1Dg0PqrYtXba0sTXwrwJ3L5qX9Q91LYO1zsM2yi2VssL+P2ywuQsex0mOHj+xz+nbUOfuYLWmdBnsmGBs7+Nzx97L4onuFgj17BfYkSJZRRgW4KlDPHPxn3uerrimAm07UN1TbTOghmbO12LtMHEoVk16ROQEEIIT9AGJIQQwhO0AQkhhPAEbUBCCCE8QRuQEEIITxi3Kri2WB+yBkYrU5hyzLJ7mTeN28P0k2v0JLglSMBQ3WSPoW6puyxau13FUUuEK0osZVYuCbWylHSTCrkKLkCUdJbicHoxV2BV53GVUyEJzNrfw9VT1cQuBwDOnFvm1EpyubLIugVtRLE1yXi94jw+TywwL0TCzQAgx8/XjaW2Oq68wKk9vd1VggJA0rBUig+4a7u9m6uqKsP8PpYQC5euOB/LoqlFvB9knqwQwmNnltC6FSbH7HUstdsf7vgxrZ/6iX93agtr3PkHgEHDbmhWmduPXOOeFxsWYj1Jd55mVoVpW0u9OznEg/SOmeC+t+QY72VzqvjYfRnuwzSvxr3n/b1+PE+vMBp9AhJCCOEJ2oCEEEJ4gjYgIYQQnqANSAghhCeMaQP66le/irS0tFFfs2bNGvl+IpHAsmXLUFRUhNzcXFxyySVoaWl51zsthBDi8GfMKrijjjoKf/zjH/96gcy/XuK6667D7373Ozz44IMIhUK4+uqrcfHFF2PdunVj7lhzZy8yA6MVF40NrgJohqEQqQxxr6rEgKtgeWyb6+EGAKcey32Sor3c34nRY3hBWYF0e4iHXSzGVUuWSo/5WmUQ9QoATAxxdRdTzW1rcz2fAGB6DldPpYO/ZgruPZhewJU7XcZcRxKuWmjIUEQWGaq0IFELWqrFXqLiAoBhIg2cFOZrL9bH1ZaxXh4g1kdCFauK+TwVGeGJzXF37Vj+ZA1GIF2CKEd3NkRo2+rj+DMTyHLn2vIys9SWEeM+sutYIY5M7QYAa+/+qVOb9dVltK2lqmSqVKYmBYBX98VonSnbKsP8Ge2I8/u4N8rVgt1EYWepiItz+XpirVnbZNqhbS1j3oAyMzNRXl7u1KPRKO655x488MADOOOMMwAA9957L2bPno0NGzbghBNOGOtLCSGEOIIZ8xnQjh07UFlZiSlTpuCyyy5DQ8Obzr91dXUYGBjA0qVLR9rOmjUL1dXVWL9+vXm9ZDKJWCw26ksIIcSRz5g2oMWLF+O+++7D448/jlWrVmHPnj04+eST0d3djebmZvh8PoTD4VH/pqysDM3NzeY1V6xYgVAoNPJVVVX1tgYihBDi8GJMv4I799xzR/573rx5WLx4MSZNmoRf/vKXyM7mmSt/j+XLl+P6668f+f9YLKZNSAgh/gl4R1Y84XAYM2bMwM6dO3HWWWehv78fkUhk1KeglpYWemb0F/x+P/x+95DthJml8OeMDtOaP8W115lTzje+ugYeSHd0pRvQdfmZNbRt0Djk32Mc8rED4skF/ACxIcKFBWES+LTPCMabQ6w/AGBXh2sxM6OEz9PgMLdDCRO7nFCAz8eAcWrcnuD9jibdw9OkYcuyoaGd1llwX55he9LQxfuxqMpdC4+80krbnjaZW5Ow0LhIHxcsvHiA2+iEgly0wGx0rJC/DMNqqZcICGZPDNO2luXTCVPdsc8mzxEAGLcRvcQqyG8IY7buj9L6hEIuwNi4ww0ntOyNLHsdJjj4wVdX0raXf+kqWmdjD2fzt9jSEH8eW6PuPdjTxtdTJM7FK1kzuQ1ZFlFP1DV30bY7W7ngqKnLfW/ZWO/er4E+Htp3MO/o74B6enqwa9cuVFRUYOHChcjKysKaNWtGvr99+3Y0NDSgtrb2nbyMEEKII5AxfQL6j//4D1xwwQWYNGkSmpqacOONNyIjIwMf/vCHEQqFcOWVV+L6669HYWEh8vPzcc0116C2tlYKOCGEEA5j2oD27duHD3/4w+jo6EBJSQlOOukkbNiwASUlb7rX3nrrrUhPT8cll1yCZDKJc845B3fcccd70nEhhBCHN2PagFavXv2W3w8EAli5ciVWruS/OxVCCCH+grzghBBCeMK4DaSrb48jK3u0aoPZ2mQTNRQA7G3lKowSEizW3sPVJIVGaNRZU0tp/ZEtrhpncx+3y2g2FEfFea6tDVPGvBUNbe7YrXliAVgAVwDWFHFFX7yfS5+GUnxe6TWI4gsAPjCPS/Kf3bvNqRUYdjTVhhKxKeb2zwobzCNWMgCwiITGPdbjrgMAOLmaq5N+vmE/rTMl4vp9XCGWa/T7pImFTu3lffzZWDwlTOsJomB7bkcHbRs3lFldRD21YA5/jnqTfC28+Dr3lSwocBVl+8kzANhhcsxex1K7/eTmVbR+9Af+1al1dHA12RLD5os9/3tauaI3y7D5yTFsiNp73Xk9xViTD23kc83eE5ta3LkeSnCl8MHoE5AQQghP0AYkhBDCE7QBCSGE8ARtQEIIITxBG5AQQghPGLcquKqiHPhzRvteBYiSy5/B99CpFdwLqoqEhZUY4UvVISNozfDdOqEm5NQCmYbfVQv3/woRL7jqIu4bNbuU+4JtIv5TVUaoVZ6fz1+MBL7NLOT+X7sNbzwrkG5CrjuepjhX+llBcMzbrtRQLSYMgzI/uTdBX5i2ZeFwAFCa7c5rOJurkPqMscyvcZVqb/bPvTczivlasJ6DAeL1d4qhdsvLOvS3g93t/H75yo31RNSgE4l6DQDyso0Awcl8nuJENWeF2s0y/BNZmJzla8fUbgCw+cFfObVj/+2DtC0LngOAwqBbzzTeyyYYoZvM/w+wgvv4/TpzNlfHMSYR372BvnS8cgj/Vp+AhBBCeII2ICGEEJ6gDUgIIYQnaAMSQgjhCeNWhNDXP4yhzNGngC0kmG2ScShrhWttIwe77HAOADqJdQUAnDm9jNa3NLu2GwPGSWZPgl87UOIeknYZ/ajbz+1GmA0RE3AAwBtt/NppRGhRmssPPTvi/BpWGFdDtztP3YYlkNXvtrh7qG0JKnwZXBSwpd3tx4GoaxkDcEsbAMggHi4sEBAAzprK56Opk9u1sGvvNq7tM8Qukye7IW57Ovk1SvN4/wJE4JBphMm9tpeH7g2R5yDLEE4kjUP03S3ckiY/212XcWLbBQC5RmhhgLwvWOvXstdhgoOXH/glbXvazZ+ldSaSem53jLZti/H7eFp1Ea2XBN1+szUGAHWN/DU/sXiyU0ul3PdPVmPoE5AQQghP0AYkhBDCE7QBCSGE8ARtQEIIITxBG5AQQghPGLcquKJgJvwH2VKwUDWmGgGA/kFuX1Ge59p8dBgqs5pCbl/TaKhgaopc6x7TiqeZq/QyifrMUjjNKuVWQe0xVy04xRjLsBHWti/iBotZakE2pwBQEeQKxeSQq3Iqd8VaAIBOonYDACay6TfCxvqH+P2dXuLOX68RrpcgfQaALtK/hRO4ZVF7Nw9rs6xnXmtzw+cqQrxthmEP1ZlgoXv8586gYcXTRWx02gy14ImzeMjcMLlhWYaSrpmoXQGgLMzX0/521woq15jTYsOu6dV9ruqrNMRfzwqTY/Y6ltrtO1+6ndY//uXPOLVQDu9zIJM/u+19fP6mEOujDiOMs7qQj51ZY7H70u/jz8vB6BOQEEIIT9AGJIQQwhO0AQkhhPAEbUBCCCE8QRuQEEIITxi3Krh0pDmBZq1E3WV5hSVIwBQADBAll6XuipBQtjdfk/tJ7Y+6ipISEjAF8BAtAGgjwiDLNy6WMELtgm69uZurySwF27xyV8k1ZPg7DQ/wenMvV0qx8DRfuhGoZtzHohx3XnONkK+YMdcdPW7dCugqyuGKI9a/+ghXIc0oyKP1pDHGsN+9j41Rww8twNckU581x7jyqcdY78wvLNvP59oKdxwgCkXL/8+XyccSDBhBdUZfGD3GazIFW2uUK1WL87j6lIXJWSpdpnYDgB/99x1O7d9u+BRtu7uF929BJZeUdpH3kdJsPpb+QeNZJ+uJBS32G+GLB6NPQEIIITxBG5AQQghP0AYkhBDCE7QBCSGE8IRxK0JAGnCwuwgLd2smwgQA2LHftTEBAEzhYU2MzUbg29lTSmi9mYTgtcf4Hr9zf4TWjyb9s0QP21uN0L2GLqc2XMPHzYQTAODLdK9dmc8PgbOMUCsrtKy13xVETArzw9BggC9RFvrWO8AP8/OMELL6LvcaDW38nvsWVdF6lNjUxI0DWL9xH9uNYDEWklbfyde7RRGxpNnWxMPGZlZw+yp2cF8R4vfrl+saaZ0FlOUaAYdtba61DgBMqOT9Y8F20R4+TzOrwrReGXZFJnva+H3c08qFIJlk/qwwOctehwkOHrjlB7TttPMvovWOufz9aU+HOyczC7lt1B/r9tP6R4+d6F632R3jYILfw4PRJyAhhBCeoA1ICCGEJ2gDEkII4QnagIQQQniCNiAhhBCeMG5VcMmhYcCwKPlb8o3gqXxiRwMA0V5XtVRghFTlGgqsbB9XMzEGhw2blXyuIsokljSJAa7G6evnFjNMcWT1I8cYixWkx+gxQtzajTC5DhLMVh3iVjcxcr8AINLrXsNwCkKuoYJj82Rh2SExpZoVtJYw1HHM3gQAesl9zzHGwkIIAWDQsJlidMa5IpLZ1AT9/GfXDGPsWSTsjllGvXkNfm1LRciseCx7LTYWAOggazVizEcWuecAMCHkjqfNVDjy9c7sdSy1285HH6Z1/3n/l9bZuvSnj+3ZCBP1HrNlGhw+tK1Fn4CEEEJ4gjYgIYQQnqANSAghhCdoAxJCCOEJ2oCEEEJ4QlpqLFKgfwCxWAyhUAgPrHsDObmjA7z6iZIrL4sr2AYM1Rcj42DTub9DYYCrd5riroIlOcSVT9akl2S76hgrXM9UTw26ii1fBle7ZBpj7x5wr5FjBIVlG/W+wUMfu3Xt0iBXC77U3Ole15jUoKGeYvfREo3l+/g6KyCqoC5Dudc/zOej21AzJsjaqcjJPuS2ANBD7qMV/mddgwX6Tcjl/Yj2c+XYhNwct29jGDcAFJCAPgDoTDJFJL+Rk0M8rG1v1PUuyzLmKcdYT0y1WG6s3/Y+rlqMEZ/EDuI3CPBgRwD47FXfovV//9KnndqMUve+AMCEPD7XiyYUOrWGrl6nFu/pxoXHTUE0GkV+PvfwA/QJSAghhEdoAxJCCOEJ2oCEEEJ4gjYgIYQQnjDmDWj//v34yEc+gqKiImRnZ+Poo4/GSy+9NPL9VCqFr3zlK6ioqEB2djaWLl2KHTt2vKudFkIIcfgzJi+4rq4uLFmyBKeffjoee+wxlJSUYMeOHSgoKBhp881vfhO33347fvzjH6OmpgZf/vKXcc4552Dr1q0IBLgihPFaaw8C8dEKrUgfV80wLF+rSYWuemeMIjgzEfWFfW4yYFOEe0FVGgmgs0tdxVv/EFf0WP0+EHNVM68386TPxZNDtP4KSYM9a0YBaQms3niA1qsKucKGJYZmGKmqV5IERgD47aZWWmekG9eeX+2OvSrM1T+WCo6lcT62s422vXBmGa0/vqOD1k+rced73T437RYAioP8Ua7MdddZQ4wn6c4qzKP1AFFQ7ujiiZetPVyx1UPuud/wVIsavnvb2vlrFpGxb9zH2x4zgb8vdJPUVyvpt72X94/5z5UEXYUYAEwp4CrCLjJ2lmQK2J6DTO0GAD+9+U6n9sQvbqJt/7inndaPm+gmK//+Dbdtspe/3xzMmDagb3zjG6iqqsK99947UqupqRn571Qqhdtuuw3/+Z//iQsvvBAA8JOf/ARlZWV4+OGH8aEPfWgsLyeEEOIIZky/gvvtb3+LRYsW4QMf+ABKS0uxYMEC3H333SPf37NnD5qbm7F06dKRWigUwuLFi7F+/Xp6zWQyiVgsNupLCCHEkc+YNqDdu3dj1apVmD59Op544glcddVV+OxnP4sf//jHAIDm5mYAQFnZ6F81lJWVjXzvYFasWIFQKDTyVVVV9XbGIYQQ4jBjTBvQ8PAwjj32WNx8881YsGABPvnJT+ITn/gE7rzT/d3iobJ8+XJEo9GRr8bGxrd9LSGEEIcPYzoDqqiowJw5c0bVZs+ejf/93/8FAJSXlwMAWlpaUFFRMdKmpaUFxxxzDL2m3++H3+/az/zr0ZXIyxtt4RAnB3RJw6amw7C6yCQHi5adj2Ul8/PX+KF7bbV7iHsiOegGgNJsLkIoJcFsESPYrc8IqsvJcg9gT53kWmgAth3KRBKY97vX+cHkhxZU0Lp1cM8shJJD/D7+ehv/5LzspMmH/HqdSb4WmIXQk7tcix8AmF/Mf1ZbW+8KDi6eXU7brtxQT+sfPXYCrTf3umKBY8u5rYl1IL223hUtnDetlLbd2Bqh9ePKXTFEhWHFkzCemTAJjqyP8AP6ucX8mbH4c6Mr4rjiGD6nVgAjW5N1zVzwcUp1Ma37s9w1YolrOnq4GIK9L8wszOWvZ4TJbTjA+80EB+dc+hXa9mf3/T9ab426oqoPzXWf/57uGFbSK4xmTJ+AlixZgu3bt4+qvfHGG5g0aRKANwUJ5eXlWLNmzcj3Y7EYnn/+edTW1o7lpYQQQhzhjOkT0HXXXYcTTzwRN998Mz74wQ/ihRdewF133YW77roLAJCWloZrr70WX/va1zB9+vQRGXZlZSUuuuii96L/QgghDlPGtAEdd9xxeOihh7B8+XLcdNNNqKmpwW233YbLLrtspM0XvvAFxONxfPKTn0QkEsFJJ52Exx9/fEx/AySEEOLIZ0wbEACcf/75OP/8883vp6Wl4aabbsJNN/E/cBJCCCEAecEJIYTwiHEbSPffv9uEQHC0qmxvp6tmCgW4EmRTfYTWp5a7SrXSXK6e2tPBLUuuPmESrX9z7S6nFgzwa3fEuEXP0VWuyinSx5VFcRIUBgC7m7ud2swJXFmUaYRaFea485oOrugZMFLcSgx7mL1d7n2sDrvqPwC4aA5X2F3/my1OrTCXX2NmGbcE2tLk2oU0kXAtAPif982l9ZYe9z4+tpNb63zoKK6Ou+nJN2j9c6fUOLXfbOM2P5ba6pxprvrxvhf307Y5fn6/inNde6LSPL6uf7+Jqxb9RH2WT5RxANDSxZ+7nADvXxFRbDZ1cCueOVXcTqo41732zla+FqKGzdeZs111XF0j/8P6amIJBgD9g+6z9Mc6fr+st+7lF82i9dfb3PHMJ++HAPCRj36d1l985Ban9tlfv+rUBhNx/PmLZyuQTgghxPhEG5AQQghP0AYkhBDCE7QBCSGE8ARtQEIIITxh3Krg6ps7XfUE6Wmsj/ukJQa4t1gox1XeDBg+ZJayaC9RkwBAGfFxswgaiiMG868DgIThg9dL1HEsLAsAAllcRchestPwpAv6+TWsMQ6SviRIYBkAZBoeZ2w4xjSZ88eC6rqMMVr39qmdbjDeyTU8sPCAoe4qzefX9pHANiOb0LwHDe3uWi3O468X7eVjLyQqOAtrnbFbYPmhjeXZALgfZK6hmPMZ64n1uqmLK1ULg1y9xygI8rnrNdY786Rr7+bzFCbvZQCQMDwi00iCJfN2A2yF4nEX3ODUtv/x206tuzuGuTVlUsEJIYQYn2gDEkII4QnagIQQQniCNiAhhBCeMGYz0n8U33t2D/zB0UFMbzS71inVRdxm5X+fcm1xAODT73NtKrKMQ+o240D64wt5bPjXn9rp1PKMw9D9nfxA+qhK1xrDsgSaXsrHvqkh6tRmVnDLjV5DrJHnd382sexyhgwdizWvzLonh4TDAcD7j+bBYjc9ucOpTSk6dBEIADRG3MNdtsYA4Jolk2l9ZqE7r6s27KVtP3x0Ja1f+sPnaf1b75/n1J7dxwPzygxrnCn5bpjZU3u4nU+ABKoBgJ+IIZ7cxvthWeAMk3teVsDX78YtLbQ+fSoPVWSiheZOLhSaV1NE68yKZ2O9+xwBQFMLXyOTKt3DdkvjVRbmVjx9RJywp5nb+WQbYo0vnDmN1n//hhsoycLkAG6vA3DBwcyl/+HUUkNcOHEw+gQkhBDCE7QBCSGE8ARtQEIIITxBG5AQQghPGHcihL8c2iV73YO+gT63luzlh+jDSX4ImYi7WTlDxmF5spfn7fR080PBftLn5DCf4oE+LixIxt2+9Pfytok4/4tnPk/GGA0Rgm/Q/dmkL4sfLFoihEHyl9cAMEDap2VwEUIsxuearY++wKEdfP71Gq7IhM0dAMR7eD/S+t37m4zza1jrZjDBD+7Za7L1CwB9aVyEEE937691jZQhQhgmmVHWPA0m+HPHRAgDfWN7dgf7uMhkYNhdO4MJ/sz09/JrJNPc+2iNcci4XwN97jxZIoR+H392+4kIwVofg8Z7S7yH31/2zIx1TXaT9kxw8Jfa3zPaGXdWPPv27UNVFVeZCSGEOHxobGzExIkTze+Puw1oeHgYTU1NyMvLQ3d3N6qqqtDY2PiWfkKHM7FYTGM8AtAYD3+O9PEB/7gxplIpdHd3o7KyEunp9knPuPsVXHp6+siO+RfzvPz8/CN2QfwFjfHIQGM8/DnSxwf8Y8YYCoX+bhuJEIQQQniCNiAhhBCeMK43IL/fjxtvvBF+/9gsVg4nNMYjA43x8OdIHx8w/sY47kQIQggh/jkY15+AhBBCHLloAxJCCOEJ2oCEEEJ4gjYgIYQQnqANSAghhCeM6w1o5cqVmDx5MgKBABYvXowXXnjB6y69bZ555hlccMEFqKysRFpaGh5++OFR30+lUvjKV76CiooKZGdnY+nSpdixw039HK+sWLECxx13HPLy8lBaWoqLLroI27dvH9UmkUhg2bJlKCoqQm5uLi655BK0tPD0y/HIqlWrMG/evJG/Iq+trcVjjz028v3DfXyMW265BWlpabj22mtHaof7OL/61a8iLS1t1NesWX9NSj7cx/cX9u/fj4985CMoKipCdnY2jj76aLz00ksj3x8P7znjdgP6xS9+geuvvx433ngjXn75ZcyfPx/nnHMOWltbve7a2yIej2P+/PlYuXIl/f43v/lN3H777bjzzjvx/PPPIxgM4pxzzkEikfgH9/TtsXbtWixbtgwbNmzAk08+iYGBAZx99tmIx//qqnvdddfhkUcewYMPPoi1a9eiqakJF198sYe9HhsTJ07ELbfcgrq6Orz00ks444wzcOGFF2LLli0ADv/xHcyLL76IH/zgB5g3b3Q0+JEwzqOOOgoHDhwY+Xr22WdHvnckjK+rqwtLlixBVlYWHnvsMWzduhX/8z//g4KCgpE24+I9JzVOOf7441PLli0b+f+hoaFUZWVlasWKFR726t0BQOqhhx4a+f/h4eFUeXl56lvf+tZILRKJpPx+f+rnP/+5Bz1857S2tqYApNauXZtKpd4cT1ZWVurBBx8cafP666+nAKTWr1/vVTffMQUFBakf/vCHR9z4uru7U9OnT089+eSTqVNPPTX1uc99LpVKHRn38cYbb0zNnz+ffu9IGF8qlUp98YtfTJ100knm98fLe864/ATU39+Puro6LF26dKSWnp6OpUuXYv369R727L1hz549aG5uHjXeUCiExYsXH7bjjUajAIDCwkIAQF1dHQYGBkaNcdasWaiurj4sxzg0NITVq1cjHo+jtrb2iBvfsmXLcN55540aD3Dk3McdO3agsrISU6ZMwWWXXYaGhgYAR874fvvb32LRokX4wAc+gNLSUixYsAB33333yPfHy3vOuNyA2tvbMTQ0hLKyslH1srIyNDc3e9Sr946/jOlIGe/w8DCuvfZaLFmyBHPnzgXw5hh9Ph/C4fCotofbGDdv3ozc3Fz4/X58+tOfxkMPPYQ5c+YcMeMDgNWrV+Pll1/GihUrnO8dCeNcvHgx7rvvPjz++ONYtWoV9uzZg5NPPhnd3d1HxPgAYPfu3Vi1ahWmT5+OJ554AldddRU++9nP4sc//jGA8fOeM+7iGMThz7Jly/Daa6+N+r36kcLMmTOxadMmRKNR/OpXv8IVV1yBtWvXet2td43GxkZ87nOfw5NPPolAIOB1d94Tzj333JH/njdvHhYvXoxJkybhl7/8JbKzsz3s2bvH8PAwFi1ahJtvvhkAsGDBArz22mu48847ccUVV3jcu78yLj8BFRcXIyMjw1GetLS0oLy83KNevXf8ZUxHwnivvvpqPProo/jTn/40KgmxvLwc/f39iEQio9ofbmP0+XyYNm0aFi5ciBUrVmD+/Pn47ne/e8SMr66uDq2trTj22GORmZmJzMxMrF27FrfffjsyMzNRVlZ2RIzzbwmHw5gxYwZ27tx5xNzHiooKzJkzZ1Rt9uzZI79qHC/vOeNyA/L5fFi4cCHWrFkzUhseHsaaNWtQW1vrYc/eG2pqalBeXj5qvLFYDM8///xhM95UKoWrr74aDz30EJ566inU1NSM+v7ChQuRlZU1aozbt29HQ0PDYTNGxvDwMJLJ5BEzvjPPPBObN2/Gpk2bRr4WLVqEyy67bOS/j4Rx/i09PT3YtWsXKioqjpj7uGTJEufPIN544w1MmjQJwDh6z/mHyR3GyOrVq1N+vz913333pbZu3Zr65Cc/mQqHw6nm5mavu/a26O7uTm3cuDG1cePGFIDUd77zndTGjRtT9fX1qVQqlbrllltS4XA49Zvf/Cb16quvpi688MJUTU1Nqq+vz+OeHxpXXXVVKhQKpZ5++unUgQMHRr56e3tH2nz6059OVVdXp5566qnUSy+9lKqtrU3V1tZ62OuxccMNN6TWrl2b2rNnT+rVV19N3XDDDam0tLTUH/7wh1QqdfiPz+JvVXCp1OE/zs9//vOpp59+OrVnz57UunXrUkuXLk0VFxenWltbU6nU4T++VCqVeuGFF1KZmZmpr3/966kdO3ak7r///lROTk7qZz/72Uib8fCeM243oFQqlfre976Xqq6uTvl8vtTxxx+f2rBhg9ddetv86U9/SgFwvq644opUKvWmLPLLX/5yqqysLOX3+1Nnnnlmavv27d52egywsQFI3XvvvSNt+vr6Up/5zGdSBQUFqZycnNT73//+1IEDB7zr9Bj5+Mc/npo0aVLK5/OlSkpKUmeeeebI5pNKHf7jszh4Azrcx3nppZemKioqUj6fLzVhwoTUpZdemtq5c+fI9w/38f2FRx55JDV37tyU3+9PzZo1K3XXXXeN+v54eM9RHpAQQghPGJdnQEIIIY58tAEJIYTwBG1AQgghPEEbkBBCCE/QBiSEEMITtAEJIYTwBG1AQgghPEEbkBBCCE/QBiSEEMITtAEJIYTwBG1AQgghPOH/B2phXRYY4/5uAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Draw a heatmap with the summary_similarity_matrix\n",
    "plt.figure()\n",
    "# Color scheme blues\n",
    "plt.imshow(summary_similarity_matrix, cmap = 'Blues')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Community detection in summaries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Run the community detection algorithm\n",
    "\n",
    "def get_topics(title_similarity, num_topics = 8, bonus_constant = 0.25, min_size = 3):\n",
    "\n",
    "  proximity_bonus_arr = np.zeros_like(title_similarity)\n",
    "  for row in range(proximity_bonus_arr.shape[0]):\n",
    "    for col in range(proximity_bonus_arr.shape[1]):\n",
    "      if row == col:\n",
    "        proximity_bonus_arr[row, col] = 0\n",
    "      else:\n",
    "        proximity_bonus_arr[row, col] = 1/(abs(row-col)) * bonus_constant\n",
    "        \n",
    "  title_similarity += proximity_bonus_arr\n",
    "\n",
    "  title_nx_graph = nx.from_numpy_array(title_similarity)\n",
    "\n",
    "  desired_num_topics = num_topics\n",
    "  # Store the accepted partitionings\n",
    "  topics_title_accepted = []\n",
    "\n",
    "  resolution = 0.85\n",
    "  resolution_step = 0.01\n",
    "  iterations = 40\n",
    "\n",
    "  # Find the resolution that gives the desired number of topics\n",
    "  topics_title = []\n",
    "  while len(topics_title) not in [desired_num_topics, desired_num_topics + 1, desired_num_topics + 2]:\n",
    "    topics_title = community.louvain_communities(title_nx_graph, weight = 'weight', resolution = resolution)\n",
    "    resolution += resolution_step\n",
    "  topic_sizes = [len(c) for c in topics_title]\n",
    "  sizes_sd = np.std(topic_sizes)\n",
    "  modularity = community.modularity(title_nx_graph, topics_title, weight = 'weight', resolution = resolution)\n",
    "\n",
    "  lowest_sd_iteration = 0\n",
    "  # Set lowest sd to inf\n",
    "  lowest_sd = float('inf')\n",
    "\n",
    "  for i in range(iterations):\n",
    "    topics_title = community.louvain_communities(title_nx_graph, weight = 'weight', resolution = resolution)\n",
    "    \n",
    "    modularity = community.modularity(title_nx_graph, topics_title, weight = 'weight', resolution = resolution)\n",
    "    print(f'Iteration {i}: topics_title (communities) {topics_title} modularity {modularity}')\n",
    "    \n",
    "    # Check SD\n",
    "    topic_sizes = [len(c) for c in topics_title]\n",
    "    sizes_sd = np.std(topic_sizes)\n",
    "    print('Communities size:', topic_sizes)\n",
    "    print('Standard deviation community size:', sizes_sd)\n",
    "    \n",
    "    topics_title_accepted.append(topics_title)\n",
    "    \n",
    "    if sizes_sd < lowest_sd and min(topic_sizes) >= min_size:\n",
    "      lowest_sd_iteration = i\n",
    "      lowest_sd = sizes_sd\n",
    "      \n",
    "  # Set the chosen partitioning to be the one with highest modularity\n",
    "  topics_title = topics_title_accepted[lowest_sd_iteration]\n",
    "  print(f'Best SD: {lowest_sd}, Best iteration: {lowest_sd_iteration}')\n",
    "  \n",
    "  topic_id_means = [sum(e)/len(e) for e in topics_title]\n",
    "  # Arrange title_topics in order of topic_id_means\n",
    "  topics_title = [list(c) for _, c in sorted(zip(topic_id_means, topics_title), key = lambda pair: pair[0])]\n",
    "  # Create an array denoting which topic each chunk belongs to\n",
    "  chunk_topics = [None] * title_similarity.shape[0]\n",
    "  for i, c in enumerate(topics_title):\n",
    "    for j in c:\n",
    "      chunk_topics[j] = i\n",
    "            \n",
    "  return {\n",
    "    'chunk_topics': chunk_topics,\n",
    "    'topics': topics_title\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Iteration 0: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {10}, {49}, {40}, {2, 35, 36, 38, 8, 9, 13, 15, 27}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {4, 6}, {51, 47, 55}, {32, 34, 37, 39, 7, 18, 29, 23, 59, 61, 30, 63}, {33}, {42, 43, 44, 28, 14}, {64, 3, 57, 45, 62}, {48, 46}, {52, 53, 54}, {56, 58}, {16}] modularity 0.024064370389019308\n",
      "Communities size: [7, 1, 1, 1, 9, 1, 9, 2, 3, 12, 1, 5, 5, 2, 3, 2, 1]\n",
      "Standard deviation community size: 3.3647881520176686\n",
      "Iteration 1: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {6}, {10}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {16}, {40, 18, 38, 39}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {56, 51}, {33}, {55, 46, 47}, {42, 43, 44, 28, 14}, {49}, {52, 53, 54}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.024159854578607063\n",
      "Communities size: [7, 1, 1, 9, 1, 4, 10, 10, 2, 1, 3, 5, 1, 3, 3, 4]\n",
      "Standard deviation community size: 3.151760420780742\n",
      "Iteration 2: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {8, 9, 6}, {10}, {49}, {16}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {2, 35, 4, 36, 27, 13, 15}, {47, 46, 55}, {33}, {40, 18, 38, 39}, {42, 43, 44, 28, 14}, {64, 48, 3, 45}, {56, 51}, {52, 53, 54}, {57, 58, 62}] modularity 0.024150545789677638\n",
      "Communities size: [7, 3, 1, 1, 1, 10, 10, 7, 3, 1, 4, 5, 4, 2, 3, 3]\n",
      "Standard deviation community size: 2.9040650388722358\n",
      "Iteration 3: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {49, 10}, {16, 6}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {47, 46, 55}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {40, 18, 38, 39}, {42, 43, 44, 28, 14}, {64, 48, 3, 45}, {52, 53, 54}, {56, 51}, {57, 58, 62}] modularity 0.02415835619632962\n",
      "Communities size: [7, 2, 2, 10, 9, 3, 10, 1, 4, 5, 4, 3, 2, 3]\n",
      "Standard deviation community size: 2.9906316307797605\n",
      "Iteration 4: topics_title (communities) [{10}, {49}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {40, 18, 38, 39}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {6}, {16}, {42, 43, 44, 28, 14}, {47, 46, 55}, {0, 1, 50, 5, 41, 11, 12}, {52, 53, 54}, {57, 58, 62}, {56, 51}, {64, 48, 3, 45}] modularity 0.024159854578607066\n",
      "Communities size: [1, 1, 9, 4, 10, 10, 1, 1, 1, 5, 3, 7, 3, 3, 2, 4]\n",
      "Standard deviation community size: 3.151760420780742\n",
      "Iteration 5: topics_title (communities) [{49}, {8, 9, 6}, {10}, {3, 31}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {33}, {2, 35, 4, 36, 27, 13, 15}, {40, 18, 38, 39}, {16}, {42, 43, 44, 28, 14}, {64, 57, 58, 45, 62}, {47, 46, 55}, {0, 1, 50, 5, 41, 11, 12}, {48}, {52, 53, 54}, {56, 51}] modularity 0.02414801596251237\n",
      "Communities size: [1, 3, 1, 2, 10, 9, 1, 7, 4, 1, 5, 5, 3, 7, 1, 3, 2]\n",
      "Standard deviation community size: 2.83331637148054\n",
      "Iteration 6: topics_title (communities) [{2, 35, 36, 4, 27, 13, 15}, {10}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {8, 9, 6}, {49}, {16}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {47, 46, 55}, {33}, {40, 18, 38, 39}, {42, 43, 44, 28, 14}, {57, 58, 62}, {0, 1, 50, 5, 41, 11, 12}, {52, 53, 54}, {56, 51}, {64, 48, 3, 45}] modularity 0.024158853379531334\n",
      "Communities size: [7, 1, 10, 3, 1, 1, 1, 9, 3, 1, 4, 5, 3, 7, 3, 2, 4]\n",
      "Standard deviation community size: 2.791484827695217\n",
      "Iteration 7: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {10}, {6}, {49}, {42, 43, 44, 28, 14}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {40, 18, 38, 39}, {16}, {57, 58, 62}, {47, 46, 55}, {52, 53, 54}, {56, 51}, {64, 48, 3, 45}] modularity 0.024159854578607066\n",
      "Communities size: [7, 1, 1, 1, 5, 9, 10, 10, 1, 4, 1, 3, 3, 3, 2, 4]\n",
      "Standard deviation community size: 3.151760420780742\n",
      "Iteration 8: topics_title (communities) [{49}, {10}, {0, 1, 50, 5, 41, 11, 12}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {31}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {16}, {47, 46, 55}, {33}, {6}, {40, 18, 38, 39}, {42, 43, 44, 28, 14}, {52, 53, 54}, {56, 51}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.02416816216846076\n",
      "Communities size: [1, 1, 7, 9, 1, 10, 9, 1, 3, 1, 1, 4, 5, 3, 2, 3, 4]\n",
      "Standard deviation community size: 3.0338345784065\n",
      "Iteration 9: topics_title (communities) [{49, 10}, {6}, {16}, {40, 18, 38, 39}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {42, 43, 44, 28, 14}, {64, 48, 3, 45}, {55, 46, 47}, {0, 1, 50, 5, 41, 11, 12}, {52, 53, 54}, {56, 51}, {57, 58, 62}] modularity 0.024159393070845758\n",
      "Communities size: [2, 1, 1, 4, 10, 10, 1, 9, 5, 4, 3, 7, 3, 2, 3]\n",
      "Standard deviation community size: 3.091206165165234\n",
      "Iteration 10: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {8, 9, 6}, {40, 18, 38, 39}, {16}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {31}, {2, 35, 4, 36, 27, 13, 15}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {14, 42, 43, 28, 44}, {55, 46, 47}, {49, 10}, {52, 53, 54}, {56, 51}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.024158391871770032\n",
      "Communities size: [7, 3, 4, 1, 9, 1, 7, 10, 1, 5, 3, 2, 3, 2, 3, 4]\n",
      "Standard deviation community size: 2.7264617638984046\n",
      "Iteration 11: topics_title (communities) [{49}, {2, 35, 36, 38, 8, 9, 13, 15, 27}, {10}, {32, 34, 37, 7, 39, 18, 29, 23, 59, 61, 30, 63}, {4, 6}, {0, 1, 50, 5, 41, 11, 12}, {16}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {33}, {40}, {42, 43, 44, 28, 14}, {64, 48, 3, 45}, {52, 53, 54}, {47, 46, 55}, {56, 51}, {57, 58, 62}] modularity 0.02407935462912729\n",
      "Communities size: [1, 9, 1, 12, 2, 7, 1, 1, 9, 1, 1, 5, 4, 3, 3, 2, 3]\n",
      "Standard deviation community size: 3.3296403995493895\n",
      "Iteration 12: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {2, 35, 36, 4, 27, 13, 15}, {8, 9, 6}, {10}, {49}, {16}, {3, 31}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {47, 46, 55}, {33}, {40, 18, 38, 39}, {42, 43, 44, 28, 14}, {64, 57, 58, 45, 62}, {48}, {52, 53, 54}, {56, 51}] modularity 0.024148015962512374\n",
      "Communities size: [7, 7, 3, 1, 1, 1, 2, 10, 9, 3, 1, 4, 5, 5, 1, 3, 2]\n",
      "Standard deviation community size: 2.83331637148054\n",
      "Iteration 13: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {10}, {8, 9, 6}, {49}, {44, 14}, {51, 28, 47}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {2, 35, 4, 36, 27, 13, 15}, {40, 18, 38, 39}, {16}, {64, 3, 42, 43, 45, 57, 58, 62}, {48, 46}, {55}, {52, 53, 54}, {56}] modularity 0.024102565345125934\n",
      "Communities size: [7, 1, 3, 1, 2, 3, 1, 9, 10, 1, 7, 4, 1, 8, 2, 1, 3, 1]\n",
      "Standard deviation community size: 3.039350970353318\n",
      "Iteration 14: topics_title (communities) [{49, 10}, {0, 1, 50, 5, 41, 11, 12}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {31}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {33}, {6}, {40, 18, 38, 39}, {16}, {42, 43, 44, 28, 14}, {47, 46, 55}, {56, 51}, {52, 53, 54}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.024167700660699447\n",
      "Communities size: [2, 7, 9, 1, 10, 9, 1, 1, 4, 1, 5, 3, 2, 3, 3, 4]\n",
      "Standard deviation community size: 2.9889118003045856\n",
      "Iteration 15: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {2, 35, 4, 36, 27, 13, 15}, {10}, {8, 9, 6}, {49}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {33}, {40, 18, 38, 39}, {16}, {42, 43, 44, 28, 14}, {64, 48, 3, 45}, {55, 46, 47}, {52, 53, 54}, {56, 51}, {57, 58, 62}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}] modularity 0.024158853379531338\n",
      "Communities size: [7, 7, 1, 3, 1, 1, 9, 1, 4, 1, 5, 4, 3, 3, 2, 3, 10]\n",
      "Standard deviation community size: 2.791484827695217\n",
      "Iteration 16: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {49, 10}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {8, 9, 6}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {56, 51}, {33}, {2, 35, 36, 4, 27, 13, 15}, {40, 18, 38, 39}, {16}, {42, 43, 44, 28, 14}, {64, 48, 3, 45}, {47, 46, 55}, {52, 53, 54}, {57, 58, 62}] modularity 0.024158391871770022\n",
      "Communities size: [7, 2, 10, 3, 1, 9, 2, 1, 7, 4, 1, 5, 4, 3, 3, 3]\n",
      "Standard deviation community size: 2.7264617638984046\n",
      "Iteration 17: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {10}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {8, 9, 6}, {49}, {40, 18, 38, 39}, {16}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {2, 35, 4, 36, 27, 13, 15}, {33}, {42, 43, 44, 28, 14}, {47, 46, 55}, {52, 53, 54}, {56, 51}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.024158853379531334\n",
      "Communities size: [7, 1, 10, 3, 1, 4, 1, 1, 9, 7, 1, 5, 3, 3, 2, 3, 4]\n",
      "Standard deviation community size: 2.791484827695217\n",
      "Iteration 18: topics_title (communities) [{56, 51}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {49, 10}, {6}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {40, 18, 38, 39}, {16}, {42, 43, 44, 28, 14}, {47, 46, 55}, {0, 1, 50, 5, 41, 11, 12}, {52, 53, 54}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.02415939307084576\n",
      "Communities size: [2, 9, 2, 1, 10, 10, 1, 4, 1, 5, 3, 7, 3, 3, 4]\n",
      "Standard deviation community size: 3.091206165165234\n",
      "Iteration 19: topics_title (communities) [{32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {8, 9, 6}, {10}, {49}, {0, 1, 50, 5, 41, 11, 12}, {40, 18, 38, 39}, {42, 43, 44, 28, 14}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {56, 51}, {47, 46, 55}, {33}, {2, 35, 4, 36, 27, 13, 15}, {16}, {64, 48, 3, 45}, {52, 53, 54}, {57, 58, 62}] modularity 0.02415054578967764\n",
      "Communities size: [10, 3, 1, 1, 7, 4, 5, 10, 2, 3, 1, 7, 1, 4, 3, 3]\n",
      "Standard deviation community size: 2.9040650388722358\n",
      "Iteration 20: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {49, 10}, {40, 18, 38, 39}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {16}, {31}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {51, 47, 55}, {33}, {6}, {42, 43, 44, 28, 14}, {64, 3, 57, 45, 62}, {48, 46}, {52, 53, 54}, {56, 58}] modularity 0.024152716420591476\n",
      "Communities size: [7, 2, 4, 9, 1, 1, 10, 9, 3, 1, 1, 5, 5, 2, 3, 2]\n",
      "Standard deviation community size: 3.0304444806001642\n",
      "Iteration 21: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {8, 9, 6}, {10}, {49}, {2, 35, 36, 4, 27, 13, 15}, {16}, {31}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {18, 28, 47}, {33}, {40, 38, 39}, {42, 43, 44, 14}, {48, 46}, {52, 53, 54}, {55}, {56, 51}, {57, 58, 62}, {64, 3, 45}] modularity 0.0241371978243493\n",
      "Communities size: [7, 3, 1, 1, 7, 1, 1, 10, 9, 3, 1, 3, 4, 2, 3, 1, 2, 3, 3]\n",
      "Standard deviation community size: 2.720599575435808\n",
      "Iteration 22: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {2, 35, 36, 4, 27, 13, 15}, {8, 9, 6}, {49, 10}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {18, 28, 47}, {56, 51}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {33}, {40, 38, 39}, {16}, {44, 14}, {57, 58, 62}, {48, 46}, {55}, {64, 3, 42, 43, 45, 52, 53, 54}] modularity 0.024122055805282483\n",
      "Communities size: [7, 7, 3, 2, 10, 3, 2, 10, 1, 3, 1, 2, 3, 2, 1, 8]\n",
      "Standard deviation community size: 3.091697551507909\n",
      "Iteration 23: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {49, 10}, {6}, {48, 46}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {18, 28, 47}, {56, 51}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {2, 35, 4, 36, 8, 9, 13, 15, 27}, {40, 38, 39}, {16}, {44, 14}, {64, 3, 42, 43, 45, 52, 53, 54}, {55}, {57, 58, 62}] modularity 0.024139672184065607\n",
      "Communities size: [7, 2, 1, 2, 1, 9, 3, 2, 10, 1, 9, 3, 1, 2, 8, 1, 3]\n",
      "Standard deviation community size: 3.2035879539270824\n",
      "Iteration 24: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {4, 6}, {49, 10}, {2, 35, 36, 38, 8, 9, 13, 15, 27}, {16}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {47, 46, 55}, {32, 34, 37, 7, 39, 18, 61, 23, 59, 29, 30, 63}, {33}, {40}, {42, 43, 44, 28, 14}, {52, 53, 54}, {56, 51}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.02407889312136598\n",
      "Communities size: [7, 2, 2, 9, 1, 1, 9, 3, 12, 1, 1, 5, 3, 2, 3, 4]\n",
      "Standard deviation community size: 3.3065985166028247\n",
      "Iteration 25: topics_title (communities) [{10}, {8, 9, 6}, {0, 1, 50, 5, 41, 11, 12}, {40, 18, 38, 39}, {16}, {3, 31}, {56, 51}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {47, 46, 55}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {33}, {2, 35, 36, 4, 27, 13, 15}, {42, 43, 44, 28, 14}, {64, 57, 58, 45, 62}, {48}, {49}, {52, 53, 54}] modularity 0.02414801596251237\n",
      "Communities size: [1, 3, 7, 4, 1, 2, 2, 9, 3, 10, 1, 7, 5, 5, 1, 1, 3]\n",
      "Standard deviation community size: 2.83331637148054\n",
      "Iteration 26: topics_title (communities) [{49, 10}, {8, 9, 6}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {16}, {47, 46, 55}, {33}, {2, 35, 36, 4, 27, 13, 15}, {40, 18, 38, 39}, {42, 43, 44, 28, 14}, {0, 1, 50, 5, 41, 11, 12}, {56, 51}, {52, 53, 54}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.024158391871770026\n",
      "Communities size: [2, 3, 1, 9, 10, 1, 3, 1, 7, 4, 5, 7, 2, 3, 3, 4]\n",
      "Standard deviation community size: 2.7264617638984046\n",
      "Iteration 27: topics_title (communities) [{49}, {0, 1, 50, 5, 41, 11, 12}, {2, 35, 4, 36, 27, 13, 15}, {10}, {8, 9, 6}, {16}, {31}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {33}, {40, 18, 38, 39}, {42, 43, 44, 28, 14}, {47, 46, 55}, {52, 53, 54}, {56, 51}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.02415885337953133\n",
      "Communities size: [1, 7, 7, 1, 3, 1, 1, 10, 9, 1, 4, 5, 3, 3, 2, 3, 4]\n",
      "Standard deviation community size: 2.791484827695217\n",
      "Iteration 28: topics_title (communities) [{49, 10}, {8, 9, 6}, {56, 51}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {2, 35, 4, 36, 27, 13, 15}, {33}, {40, 18, 38, 39}, {16}, {42, 43, 44, 28, 14}, {47, 46, 55}, {0, 1, 50, 5, 41, 11, 12}, {52, 53, 54}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.024158391871770026\n",
      "Communities size: [2, 3, 2, 1, 9, 10, 7, 1, 4, 1, 5, 3, 7, 3, 3, 4]\n",
      "Standard deviation community size: 2.7264617638984046\n",
      "Iteration 29: topics_title (communities) [{32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {6}, {49, 10}, {0, 1, 50, 5, 41, 11, 12}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {33}, {40, 18, 38, 39}, {16}, {42, 43, 44, 28, 14}, {47, 46, 55}, {52, 53, 54}, {56, 51}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.02415939307084575\n",
      "Communities size: [10, 1, 2, 7, 9, 10, 1, 4, 1, 5, 3, 3, 2, 3, 4]\n",
      "Standard deviation community size: 3.0912061651652345\n",
      "Iteration 30: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {10}, {8, 9, 6}, {49}, {40, 18, 38, 39}, {16}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {2, 35, 4, 36, 27, 13, 15}, {44, 42, 43, 28, 14}, {55, 46, 47}, {56, 51}, {52, 53, 54}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.024158853379531338\n",
      "Communities size: [7, 1, 3, 1, 4, 1, 1, 9, 10, 1, 7, 5, 3, 2, 3, 3, 4]\n",
      "Standard deviation community size: 2.7914848276952173\n",
      "Iteration 31: topics_title (communities) [{49, 10}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {0, 1, 50, 5, 41, 11, 12}, {44, 14}, {2, 35, 36, 4, 8, 9, 13, 15, 27}, {51, 28, 47}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {33}, {6}, {40, 18, 38, 39}, {56}, {16}, {55}, {48, 46}, {57, 58, 62}, {64, 3, 42, 43, 45, 52, 53, 54}] modularity 0.02413163876470964\n",
      "Communities size: [2, 10, 7, 2, 9, 3, 10, 1, 1, 4, 1, 1, 1, 2, 3, 8]\n",
      "Standard deviation community size: 3.3628252630786513\n",
      "Iteration 32: topics_title (communities) [{49}, {8, 9, 6}, {10}, {40, 38, 39}, {16}, {17, 19, 20, 21, 22, 24, 25, 26, 60, 31}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {56, 51}, {33}, {2, 35, 4, 36, 27, 13, 15}, {42, 43, 44, 14}, {64, 3, 45}, {48, 46}, {18, 28, 47}, {0, 1, 50, 5, 41, 11, 12}, {52, 53, 54}, {55}, {57, 58, 62}] modularity 0.0241288902344956\n",
      "Communities size: [1, 3, 1, 3, 1, 10, 10, 2, 1, 7, 4, 3, 2, 3, 7, 3, 1, 3]\n",
      "Standard deviation community size: 2.850709246815\n",
      "Iteration 33: topics_title (communities) [{49, 10}, {8, 9, 6}, {0, 1, 50, 5, 41, 11, 12}, {16}, {31}, {56, 51}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {33}, {2, 35, 36, 4, 27, 13, 15}, {40, 18, 38, 39}, {42, 43, 44, 28, 14}, {64, 48, 3, 45}, {47, 46, 55}, {52, 53, 54}, {57, 58, 62}] modularity 0.024158391871770022\n",
      "Communities size: [2, 3, 7, 1, 1, 2, 9, 10, 1, 7, 4, 5, 4, 3, 3, 3]\n",
      "Standard deviation community size: 2.7264617638984046\n",
      "Iteration 34: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {10}, {8, 9, 6}, {49}, {2, 35, 36, 4, 27, 13, 15}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {47, 46, 55}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {40, 18, 38, 39}, {16}, {42, 43, 44, 28, 14}, {52, 53, 54}, {56, 51}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.024158853379531334\n",
      "Communities size: [7, 1, 3, 1, 7, 1, 9, 3, 10, 1, 4, 1, 5, 3, 2, 3, 4]\n",
      "Standard deviation community size: 2.7914848276952173\n",
      "Iteration 35: topics_title (communities) [{8, 9, 6}, {49, 10}, {0, 1, 50, 5, 41, 11, 12}, {48, 46}, {16}, {18, 28, 47}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {56, 51}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {33}, {2, 35, 4, 36, 27, 13, 15}, {40, 38, 39}, {42, 43, 44, 14}, {52, 53, 54}, {55}, {57, 58, 62}, {64, 3, 45}] modularity 0.024136736316587987\n",
      "Communities size: [3, 2, 7, 2, 1, 3, 1, 9, 2, 10, 1, 7, 3, 4, 3, 1, 3, 3]\n",
      "Standard deviation community size: 2.690288891734056\n",
      "Iteration 36: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {8, 9, 6}, {49, 10}, {47, 46, 55}, {40, 18, 38, 39}, {16}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {2, 35, 4, 36, 27, 13, 15}, {33}, {42, 43, 44, 28, 14}, {64, 48, 3, 45}, {52, 53, 54}, {56, 51}, {57, 58, 62}, {31}] modularity 0.02415839187177003\n",
      "Communities size: [7, 10, 3, 2, 3, 4, 1, 9, 7, 1, 5, 4, 3, 2, 3, 1]\n",
      "Standard deviation community size: 2.7264617638984046\n",
      "Iteration 37: topics_title (communities) [{49, 10}, {0, 1, 50, 5, 41, 11, 12}, {2, 35, 36, 4, 27, 13, 15}, {8, 9, 6}, {40, 18, 38, 39}, {16}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {47, 46, 55}, {32, 34, 37, 7, 61, 23, 59, 29, 30, 63}, {33}, {42, 43, 44, 28, 14}, {56, 51}, {52, 53, 54}, {57, 58, 62}, {64, 48, 3, 45}] modularity 0.024158391871770026\n",
      "Communities size: [2, 7, 7, 3, 4, 1, 1, 9, 3, 10, 1, 5, 2, 3, 3, 4]\n",
      "Standard deviation community size: 2.7264617638984046\n",
      "Iteration 38: topics_title (communities) [{0, 1, 50, 5, 41, 11, 12}, {8, 9, 6}, {31}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {33}, {2, 35, 4, 36, 27, 13, 15}, {40, 18, 38, 39}, {16}, {42, 43, 44, 28, 14}, {64, 48, 3, 45}, {47, 46, 55}, {49, 10}, {52, 53, 54}, {56, 51}, {57, 58, 62}] modularity 0.024158391871770026\n",
      "Communities size: [7, 3, 1, 9, 10, 1, 7, 4, 1, 5, 4, 3, 2, 3, 2, 3]\n",
      "Standard deviation community size: 2.7264617638984046\n",
      "Iteration 39: topics_title (communities) [{49}, {0, 1, 50, 5, 41, 11, 12}, {8, 9, 6}, {10}, {32, 34, 37, 7, 29, 23, 59, 61, 30, 63}, {17, 19, 20, 21, 22, 24, 25, 26, 60}, {2, 35, 36, 4, 27, 13, 15}, {33}, {40, 18, 38, 39}, {16}, {42, 43, 44, 28, 14}, {64, 57, 58, 45, 62}, {48}, {47, 46, 55}, {56, 51}, {52, 53, 54}, {3, 31}] modularity 0.024148015962512374\n",
      "Communities size: [1, 7, 3, 1, 10, 9, 7, 1, 4, 1, 5, 5, 1, 3, 2, 3, 2]\n",
      "Standard deviation community size: 2.83331637148054\n",
      "Best SD: inf, Best iteration: 0\n"
     ]
    }
   ],
   "source": [
    "# Set num_topics to be 1/4 of the number of chunks, or 8, which ever is smaller\n",
    "num_topics = min(int(num_1_chunks / 4), 8)\n",
    "topics_out = get_topics(summary_similarity_matrix, num_topics = num_topics, bonus_constant = 0.2)\n",
    "chunk_topics = topics_out['chunk_topics']\n",
    "topics = topics_out['topics']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "65"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(chunk_topics)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Showing each chunk title belonging to what community"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['3:The Louvain Method for Community Detection ',\n",
       " '3:Optimizing Modularity in Community Detection ',\n",
       " '4:Understanding Modularity in Weighted Graphs ',\n",
       " '11:   Community Detection in Graphs ',\n",
       " '0:   The Kronecker Delta Function and Modularity Calculation ',\n",
       " '3:  The Louvain Method for Maximizing Modularity ',\n",
       " '0:   Calculating Modularity Change in Network Communities ',\n",
       " '9:Title ',\n",
       " '4:   Modularity Optimization in Community Detection ',\n",
       " '4:   Modularity Optimization in Network Communities ',\n",
       " '1:Modularity Maximization Algorithm for Community Detection ',\n",
       " '3:The Limitations of Louvain Algorithm ',\n",
       " '3:Comparison of Modularity Optimization Methods for Non-Overlapping Community Detection ',\n",
       " '4:The Importance and Limitations of Modularity in Network Analysis ',\n",
       " '8:The Importance of Community Structure in Network Analysis ',\n",
       " '4:Understanding Modularity in Network Analysis ',\n",
       " '2:   Community Membership and Node Score ',\n",
       " '5:   Understanding Adjacency Matrices in Networks ',\n",
       " '9:Title ',\n",
       " '5: Expected Number of Edges Between Nodes ',\n",
       " '5:   The Connection Between Stubs in a Random Graph ',\n",
       " '5:   Random Connections in Network Nodes ',\n",
       " '5:   Counting Full Edges Between Two Vertices ',\n",
       " '9:Title ',\n",
       " '5:   Approximations for Random Networks ',\n",
       " '5:   Approximating Expected Number of Edges in Random Networks ',\n",
       " '5:   Approximating Edge Probability in Graphs ',\n",
       " '4:   Understanding Modularity in Network Communities ',\n",
       " '8:Title ',\n",
       " '9:  Understanding Set Notation ',\n",
       " '9:Title ',\n",
       " '6:   Community Detection and Edge Fraction ',\n",
       " '9:Title ',\n",
       " '7:   Community Detection in Undirected Networks ',\n",
       " '9:   The Importance of Delta and Q in Machine Learning ',\n",
       " '4:   Modularity Matrix and its Role in Network Analysis ',\n",
       " '4:  Understanding Modularity in Network Analysis ',\n",
       " '9:Title ',\n",
       " '4:The Limitations of Modularity Maximization in Detecting Community Structure ',\n",
       " '9:Title ',\n",
       " '10:The Limitations of Multiresolution Methods for Modularity Optimization ',\n",
       " '3:Software Tools for Graph Clustering ',\n",
       " '8:Understanding Community Structure in Complex Networks ',\n",
       " '8:Understanding Community Structure in Networks ',\n",
       " '8:The Importance of Community Structures in Networks ',\n",
       " '11:The Importance of Community Detection in Network Science ',\n",
       " '12:Algorithms for Finding Communities ',\n",
       " '14:Title ',\n",
       " '12:Approximations for Community Detection in Multidimensional Networks ',\n",
       " '13:The Girvan–Newman Algorithm: A Method for Community Detection ',\n",
       " '3:Maximizing Modularity in Network Clustering ',\n",
       " '14:Title ',\n",
       " '15:Community Detection in Networks ',\n",
       " '15:   Understanding Clique Graphs and Their Role in Community Detection ',\n",
       " '15:   Identifying Communities in Networks Using k-cliques ',\n",
       " '14:   Community Detection in Latent Feature Spaces ',\n",
       " '16:\"Community Detection Benchmarking: Evaluating Algorithms on Stochastic Block Models\" ',\n",
       " '11:The Impact of Community Structure on Detectability ',\n",
       " '16:   The Fundamental Limit of Detecting Communities in Networks ',\n",
       " '9:   Scaling in Sparse Cases ',\n",
       " '5:   Constant Average Degree in Graphs ',\n",
       " '9:Title ',\n",
       " '11: Detecting Communities in Networks ',\n",
       " '9:  The Importance of Maintaining a Balance Between Caffeine Intake and Caffeine Withdrawal ',\n",
       " '11:Community Detection in Graphs ']"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "title_community = [f'{i}:{title}' for i, title in zip(chunk_topics, stage_1_titles)]\n",
    "title_community"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0YAAAA9CAYAAACJKXmHAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8g+/7EAAAACXBIWXMAAA9hAAAPYQGoP6dpAAASPElEQVR4nO3de1RUdbsH8O9cmBEQBEW5qOD9LmggNGmvrqTI0sxcLRfqiby+GpSKdYR1QoosLEu85BHNXi+peWllRe/BMlS6gBcQF3iJxFBZBmILuTjgwMz8zh+d5jQJurc6jjPz/az1W8vZ+5kfz+xnz579ODN7FEIIASIiIiIiIhemtHcCRERERERE9sbGiIiIiIiIXB4bIyIiIiIicnlsjIiIiIiIyOWxMSIiIiIiIpfHxoiIiIiIiFweGyMiIiIiInJ5bIyIiIiIiMjlsTEiIiIiIiKXx8aIiIiIiIhcns0ao5qaGkybNg3e3t7w8fHBrFmzcP369VveZ8yYMVAoFFZj3rx5tkqRiIiIiIgIAKAQQghbTDxu3DhUVlZiw4YNaGlpwYwZMzBixAjs3LmzzfuMGTMG/fr1Q1pammWZh4cHvL29bZEiERERERERAEBti0nPnj2L/fv34/jx44iIiAAArF27Fk899RTef/99BAUFtXlfDw8PBAQE2CItIiIiIiKiVtmkMcrPz4ePj4+lKQKA6OhoKJVKHD16FJMmTWrzvjt27MD27dsREBCACRMmICUlBR4eHm3GGwwGGAwGy22z2Yyamhp06tQJCoXi3jwgIiIiIiJyOEIINDQ0ICgoCErlrb9FZJPGqKqqCl26dLH+Q2o1OnbsiKqqqjbvN3XqVISEhCAoKAjFxcVYsmQJSktL8fnnn7d5n/T0dLz55pv3LHciIiIiInIuFRUV6Nat2y1jZDVGSUlJePfdd28Zc/bsWTlTWpk7d67l30OHDkVgYCDGjh2L8+fPo3fv3q3eJzk5GYmJiZbbdXV1CA4ORsXOl+DtoZX0d6cs+xK7X58oOc9HMv6N2WunS47PnvENPkleITk+5D9fRudnlkiOH/xjkqz8/RO/lzz/pt4XsXTDv5D2z5mS4s+MbsKml7fL2j5y4rfNPYzFyzZInnt7UoysbWPrfSHziTX4D91DkuM/rbkga/5vJr2J3c+3/Q7r341fY8YHt/ho69+93E4teV8AgGVL3pA1/+Lffntg4h39sc7oqrDpvpl1rkLW9pFzHAGAjMzN2D17luT4Jz7eJHn+4hP98a/v3sLM6BRJ8aEPldr0OCj3mLlr6qeytmXiZx/ZdP70jL3472ek/wdl4o/xso6zkfM+kbVvfpJ/wqbx31/SS35NL/FpxOuvv45ly5ZJiu/Y8SgWLtyGVatekBRfUxMla/7k5Al4PcVfUiwArE9TyqrVhJX/Izl3AIj657/hs/TW55V/VZu2xKbxA1fMlpV/3MLvMG/VesnxX86ZJet8MHrNf0mev89PWbKOU76rN8p6Xbm2YK6s+T/Y8oWsx/rMG7NkHXfkzN/QqEefFx6Hl5fXbWNlNUaLFy/Giy++eMuYXr16ISAgANXV1ZZl69atw4oVK3D16lVkZGQgPDwckZGRbc6xd+9epKSk4MKFCwCAnTt3IiWl9RcwrVYLrfbmBsjbQwtvT2mNkZtaKTkWAFRqFdy92kmOd1Or4e3RXnK8QqmCUiv95FZu/nLm93R3h1qlgqe7u6R4dy8he/vIiVep1fDwvP2O/Se528bW+4JKqUQ7Nzebze+mUsBbK/0jpGqFAu1VKunxMvaFO5r/AYp39MeqUitsum/K3j4y491UKnjbaH53jSdUSjXcNZ6S4m19HJT7PJe7LW09v1qphpdW2rYE7uA4K/e4aeN4Oa/pnp4KqNVqeHpK2z5eXu2gVqvgJbFeBoOnrPnVagU8PaVfkFhureTkDgAKtRpKTxnnRzaOl5u/Sq2Ge3s55yTyzgflzC/3ONVepZL1utIsc365j1X2a4TM+QFI+oqNrMaoc+fO6Ny5823jdDodamtrUVhYiLKyMiQmJiIhIQErV65EZGQkYmJiUFpaetPH7QAgLy8PsbGxSE9PR1BQEKZPn460tDRMmjQJQ4YMkZMuERERERGRJDb5jtHAgQPx5JNPYs6cOWhubsb48eORlZWF2NhYbNu2Dd27d0dGRgb27duHbdu2ITIyEufPn8fOnTvxww8/YPTo0ejfvz8WLVqEf/zjHzAYDPjwww+RmZlpi3SJiIiIiMjF2ewHXnfs2IG+ffvi9OnTyM7OxqhRo7Bx40YolUpER0ejoKAApaWlaGxsBABoNBp89913yMnJQW5uLhYvXozJkycjKysLMTExyM/Pb/XvGAwG1NfXWw0iIiIiIiI5bPKOEQB07NgRGRkZ2LNnD3JycqDT6Szr/P398fPPP+Ovvy3bvXt35ObmQqPRYOvWrYiNjbWKb+tqdrwqHRERERER3S2bvWN0vyQnJ6Ours4yKioq7J0SERERERE5GJu9YwQAfn5+UKlU2Lx5M2JjY1FVVYWwsDD4+fkhICCg1ft4eXlh6tSpmDp1qmWZSqXC4MGDW41v66p0REREREREUtn0HSONRoMePXrg448/RmpqKk6cOIHQ0FBkZ2dj6NChrd6nd+/eUKvVqKystIxhw4ZZfRSPiIiIiIjoXrpvH6VTKpVQKBQ3XUP8hRdeQHJysuX2448/DqPRiB07dqC2thaZmZkoLi5GQkLC/UqViIiIiIhcjE0/Stfc3IwLFy5g5syZWLp0KaqqqjBs2DCMGzcOJSUlAIBLly5Bqfz//qxv375QKpVISkrCq6++Ci8vL6xevbrN3zAyGAwwGAyW23V1dQCA+kZDq/GtaTGaUa+XHm8ymtDUcEPG/EbUN16XHC/MJpgNjTLml5e/nPn1TU0wmkzQNzVJim9quCF7+8iJNxmNaNQ3SJ5b7rax9b5gMptxo6XFZvO3mATqDeL2gf/HKASum0zS400KyfvCHc3/AMU7+mM1GRU23TflHBfuJL7FZEK9jeZvatbDZDaiqVkvKd7Wx0G5z3O529LW8xvNRjQYpG1L4A6Os3KPmzaOl/Oartc0wmg0Qq+Xtn202hswGk1okFgvvV4va36jUUCvN0uKBYAWI2TVSk7uACCMRpj1Ms6PbBwvN3+T0Yim63LOSeSdD8qZX+5xSmMyyXpdkTu/3Mcq+zVCxvwNjX88P/560bc2CRu6fPmyACDy8vKslr/22msiMjKy1fvk5eWJrVu3iqKiInH48GExfvx44e3tLSoqKlqNT01NFQA4ODg4ODg4ODg4ODhaHW31En9l03eM7oROp7P6PtEjjzyCgQMHYsOGDXjrrbduik9OTkZiYqLlttlsRk1NDTp16mT1sb36+np0794dFRUV8Pb2tu2DoPuKtXVurK/zYm2dF2vrvFhb5+aM9RVCoKGhAUFBQbeNvS9Xpbty5YrV8itXrrR5Vbq/c3Nzw/Dhw1FWVtbq+tauSufj49PmfN7e3k5TaLLG2jo31td5sbbOi7V1Xqytc3O2+nbo0EFSnM2vShceHo6cnBzLMrPZfNMPvt6KyWRCSUkJAgMDbZUmERERERG5OJt/lC4xMRFxcXGIiIhAZGQkVq1aBb1ejxkzZgD446p0Xbt2RXp6OgAgLS0NDz/8MPr06YPa2lqsWLECFy9exOzZs22dKhERERERuSibN0ZTpkzB1atXra5Kt3//fvj7+wO4+ap0165dw5w5c1BVVQVfX1+Eh4cjLy8PgwYNuqs8tFotUlNT+WOwToi1dW6sr/NibZ0Xa+u8WFvn5ur1VQgh5dp1REREREREzuu+/cArERERERHRg4qNERERERERuTw2RkRERERE5PLYGBERERERkctzmcZo3bp16NGjB9q1a4eoqCgcO3bM3imRTN9//z0mTJiAoKAgKBQKfPHFF1brhRBYunQpAgMD4e7ujujoaJw7d84+yZIs6enpGDFiBLy8vNClSxc8++yzKC0ttYq5ceMG4uPj0alTJ7Rv3x6TJ0++6cej6cGzfv16hIaGWn4sUKfTITs727KedXUey5cvh0KhwMKFCy3LWF/H9cYbb0ChUFiNAQMGWNazto7t8uXLmD59Ojp16gR3d3cMHToUBQUFlvWuek7lEo3R7t27kZiYiNTUVJw4cQJhYWGIiYlBdXW1vVMjGfR6PcLCwrBu3bpW17/33ntYs2YNMjMzcfToUXh6eiImJgY3bty4z5mSXLm5uYiPj8eRI0dw4MABtLS04IknnoBer7fELFq0CFlZWdi7dy9yc3Px22+/4bnnnrNj1iRFt27dsHz5chQWFqKgoACPPfYYJk6ciNOnTwNgXZ3F8ePHsWHDBoSGhlotZ30d2+DBg1FZWWkZP/74o2Uda+u4rl27hpEjR8LNzQ3Z2dk4c+YMPvjgA/j6+lpiXPacSriAyMhIER8fb7ltMplEUFCQSE9Pt2NWdDcAiH379llum81mERAQIFasWGFZVltbK7Rarfj000/tkCHdjerqagFA5ObmCiH+qKWbm5vYu3evJebs2bMCgMjPz7dXmnSHfH19xaZNm1hXJ9HQ0CD69u0rDhw4IEaPHi0WLFgghODz1tGlpqaKsLCwVtexto5tyZIlYtSoUW2ud+VzKqd/x6i5uRmFhYWIjo62LFMqlYiOjkZ+fr4dM6N7qby8HFVVVVZ17tChA6KiolhnB1RXVwcA6NixIwCgsLAQLS0tVvUdMGAAgoODWV8HYjKZsGvXLuj1euh0OtbVScTHx+Ppp5+2qiPA560zOHfuHIKCgtCrVy9MmzYNly5dAsDaOrqvvvoKEREReP7559GlSxcMHz4cH330kWW9K59TOX1j9Pvvv8NkMsHf399qub+/P6qqquyUFd1rf9aSdXZ8ZrMZCxcuxMiRIzFkyBAAf9RXo9HAx8fHKpb1dQwlJSVo3749tFot5s2bh3379mHQoEGsqxPYtWsXTpw4gfT09JvWsb6OLSoqClu2bMH+/fuxfv16lJeX49FHH0VDQwNr6+B+/fVXrF+/Hn379sU333yD+fPn45VXXsHWrVsBuPY5ldreCRAR/VV8fDxOnTpl9Vl2cmz9+/fHyZMnUVdXh88++wxxcXHIzc21d1p0lyoqKrBgwQIcOHAA7dq1s3c6dI+NGzfO8u/Q0FBERUUhJCQEe/bsgbu7ux0zo7tlNpsRERGBd955BwAwfPhwnDp1CpmZmYiLi7Nzdvbl9O8Y+fn5QaVS3XSllCtXriAgIMBOWdG99mctWWfHlpCQgK+//hqHDh1Ct27dLMsDAgLQ3NyM2tpaq3jW1zFoNBr06dMH4eHhSE9PR1hYGFavXs26OrjCwkJUV1fjoYceglqthlqtRm5uLtasWQO1Wg1/f3/W14n4+PigX79+KCsr43PXwQUGBmLQoEFWywYOHGj5qKQrn1M5fWOk0WgQHh6OnJwcyzKz2YycnBzodDo7Zkb3Us+ePREQEGBV5/r6ehw9epR1dgBCCCQkJGDfvn04ePAgevbsabU+PDwcbm5uVvUtLS3FpUuXWF8HZDabYTAYWFcHN3bsWJSUlODkyZOWERERgWnTpln+zfo6j+vXr+P8+fMIDAzkc9fBjRw58qafxPjll18QEhICwMXPqex99Yf7YdeuXUKr1YotW7aIM2fOiLlz5wofHx9RVVVl79RIhoaGBlFUVCSKiooEALFy5UpRVFQkLl68KIQQYvny5cLHx0d8+eWXori4WEycOFH07NlTNDU12Tlzup358+eLDh06iMOHD4vKykrLaGxstMTMmzdPBAcHi4MHD4qCggKh0+mETqezY9YkRVJSksjNzRXl5eWiuLhYJCUlCYVCIb799lshBOvqbP56VTohWF9HtnjxYnH48GFRXl4ufvrpJxEdHS38/PxEdXW1EIK1dWTHjh0TarVavP322+LcuXNix44dwsPDQ2zfvt0S46rnVC7RGAkhxNq1a0VwcLDQaDQiMjJSHDlyxN4pkUyHDh0SAG4acXFxQog/Li+ZkpIi/P39hVarFWPHjhWlpaX2TZokaa2uAMTmzZstMU1NTeKll14Svr6+wsPDQ0yaNElUVlbaL2mSZObMmSIkJERoNBrRuXNnMXbsWEtTJATr6mz+3hixvo5rypQpIjAwUGg0GtG1a1cxZcoUUVZWZlnP2jq2rKwsMWTIEKHVasWAAQPExo0brda76jmVQggh7PNeFRERERER0YPB6b9jREREREREdDtsjIiIiIiIyOWxMSIiIiIiIpfHxoiIiIiIiFweGyMiIiIiInJ5bIyIiIiIiMjlsTEiIiIiIiKXx8aIiIiIiIhcHhsjIiIiIiJyeWyMiIiIiIjI5bExIiIiIiIil8fGiIiIiIiIXN7/AtcplBDyrda8AAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 1000x400 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Plot a heatmap of this array\n",
    "plt.figure(figsize = (10, 4))\n",
    "plt.imshow(np.array(chunk_topics).reshape(1, -1), cmap = 'tab20')\n",
    "# Draw vertical black lines for every 1 of the x-axis \n",
    "for i in range(1, len(chunk_topics)):\n",
    "  plt.axvline(x = i - 0.5, color = 'black', linewidth = 0.5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Stage 2 Summaries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [],
   "source": [
    "def summarize_stage_2(stage_1_outputs, topics, summary_num_words = 250):\n",
    "  print(f'Stage 2 start time {datetime.now()}')\n",
    "  \n",
    "  # Prompt that passes in all the titles of a topic, and asks for an overall title of the topic\n",
    "  title_prompt_template = \"\"\"Write an informative title that summarizes each of the following groups of titles. Make sure that the titles capture as much information as possible, \n",
    "  and are different from each other:\n",
    "  {text}\n",
    "  \n",
    "  Return your answer in a numbered list, with new line separating each title: \n",
    "  1. Title 1\n",
    "  2. Title 2\n",
    "  3. Title 3\n",
    "\n",
    "  TITLES:\n",
    "  \"\"\"\n",
    "\n",
    "  map_prompt_template = \"\"\"Wite a 75-100 word summary of the following text:\n",
    "    {text}\n",
    "\n",
    "    CONCISE SUMMARY:\"\"\"\n",
    "\n",
    "  combine_prompt_template = 'Write a ' + str(summary_num_words) + \"\"\"-word summary of the following, removing irrelevant information. Finish your answer:\n",
    "  {text}\n",
    "  \"\"\" + str(summary_num_words) + \"\"\"-WORD SUMMARY:\"\"\"\n",
    "\n",
    "  title_prompt = PromptTemplate(template=title_prompt_template, input_variables=[\"text\"])\n",
    "  map_prompt = PromptTemplate(template=map_prompt_template, input_variables=[\"text\"])\n",
    "  combine_prompt = PromptTemplate(template=combine_prompt_template, input_variables=[\"text\"])\n",
    "\n",
    "  topics_data = []\n",
    "  for c in topics:\n",
    "    topic_data = {\n",
    "      'summaries': [stage_1_outputs[chunk_id]['summary'] for chunk_id in c],\n",
    "      'titles': [stage_1_outputs[chunk_id]['title'] for chunk_id in c]\n",
    "    }\n",
    "    topic_data['summaries_concat'] = ' '.join(topic_data['summaries'])\n",
    "    topic_data['titles_concat'] = ', '.join(topic_data['titles'])\n",
    "    topics_data.append(topic_data)\n",
    "    \n",
    "  # Get a list of each community's summaries (concatenated)\n",
    "  topics_summary_concat = [c['summaries_concat'] for c in topics_data]\n",
    "  topics_titles_concat = [c['titles_concat'] for c in topics_data]\n",
    "\n",
    "  # Concat into one long string to do the topic title creation\n",
    "  topics_titles_concat_all = ''''''\n",
    "  for i, c in enumerate(topics_titles_concat):\n",
    "    topics_titles_concat_all += f'''{i+1}. {c}\n",
    "    '''\n",
    "  \n",
    "  # print('topics_titles_concat_all', topics_titles_concat_all)\n",
    "\n",
    "  title_llm = OpenAI(temperature=0)\n",
    "  title_llm_chain = LLMChain(llm = title_llm, prompt = title_prompt)\n",
    "  title_llm_chain_input = [{'text': topics_titles_concat_all}]\n",
    "  title_llm_chain_results = title_llm_chain.apply(title_llm_chain_input)\n",
    "  \n",
    "  \n",
    "  # Split by new line\n",
    "  titles = title_llm_chain_results[0]['text'].split('\\n')\n",
    "  # Remove any empty titles\n",
    "  titles = [t for t in titles if t != '']\n",
    "  # Remove spaces at start or end of each title\n",
    "  titles = [t.strip() for t in titles]\n",
    "\n",
    "  map_llm = OpenAI(temperature=0)\n",
    "  reduce_llm = OpenAI(temperature=0)\n",
    "\n",
    "  # Run the map-reduce chain\n",
    "  docs = [Document(page_content=t) for t in topics_summary_concat]\n",
    "  chain = load_summarize_chain(chain_type=\"map_reduce\", map_prompt = map_prompt, combine_prompt = combine_prompt, return_intermediate_steps = True,\n",
    "                              llm = map_llm, reduce_llm = reduce_llm)\n",
    "\n",
    "  output = chain({\"input_documents\": docs}, return_only_outputs = True)\n",
    "  summaries = output['intermediate_steps']\n",
    "  stage_2_outputs = [{'title': t, 'summary': s} for t, s in zip(titles, summaries)]\n",
    "  final_summary = output['output_text']\n",
    "\n",
    "  # Return: stage_1_outputs (title and summary), stage_2_outputs (title and summary), final_summary, chunk_allocations\n",
    "  out = {\n",
    "    'stage_2_outputs': stage_2_outputs,\n",
    "    'final_summary': final_summary\n",
    "  }\n",
    "  print(f'Stage 2 done time {datetime.now()}')\n",
    "  \n",
    "  return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stage 2 start time 2024-01-10 09:01:13.446729\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/workspaces/ml-learning/.venv/lib/python3.11/site-packages/langchain_core/_api/deprecation.py:189: LangChainDeprecationWarning: The class `OpenAI` was deprecated in LangChain 0.1.0 and will be removed in 0.2.0. Use langchain_openai.OpenAI instead.\n",
      "  warn_deprecated(\n",
      "/workspaces/ml-learning/.venv/lib/python3.11/site-packages/langchain_core/_api/deprecation.py:189: LangChainDeprecationWarning: The class `OpenAI` was deprecated in LangChain 0.1.0 and will be removed in 0.2.0. Use langchain_openai.OpenAI instead.\n",
      "  warn_deprecated(\n",
      "/workspaces/ml-learning/.venv/lib/python3.11/site-packages/langchain_core/_api/deprecation.py:189: LangChainDeprecationWarning: The class `OpenAI` was deprecated in LangChain 0.1.0 and will be removed in 0.2.0. Use langchain_openai.OpenAI instead.\n",
      "  warn_deprecated(\n",
      "/workspaces/ml-learning/.venv/lib/python3.11/site-packages/langchain_core/_api/deprecation.py:189: LangChainDeprecationWarning: The function `__call__` was deprecated in LangChain 0.1.0 and will be removed in 0.2.0. Use invoke instead.\n",
      "  warn_deprecated(\n",
      "/workspaces/ml-learning/.venv/lib/python3.11/site-packages/langchain_core/_api/deprecation.py:189: LangChainDeprecationWarning: The function `__call__` was deprecated in LangChain 0.1.0 and will be removed in 0.2.0. Use invoke instead.\n",
      "  warn_deprecated(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stage 2 done time 2024-01-10 09:01:25.488847\n"
     ]
    }
   ],
   "source": [
    "# Query GPT-3 to get a summarized title for each topic_data\n",
    "out = summarize_stage_2(stage_1_outputs, topics, summary_num_words = 250)\n",
    "stage_2_outputs = out['stage_2_outputs']\n",
    "stage_2_titles = [e['title'] for e in stage_2_outputs]\n",
    "stage_2_summaries = [e['summary'] for e in stage_2_outputs]\n",
    "final_summary = out['final_summary']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### stage_2_outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'title': '1. Exploring Modularity and Delta Function in Network Analysis',\n",
       "  'summary': \" The Kronecker delta function is used to calculate the modularity of a community within a network. This measures how distinct and well-connected the community is. The formula involves subtracting the expected number of edges from the actual number of edges within the community, then squaring the result. This helps identify cohesive communities within a network. The process of moving a node from its own community to a neighbor's community is important in understanding network communities and is calculated using a similar equation.\"},\n",
       " {'title': '2. Maximizing Modularity: An Algorithm for Community Detection',\n",
       "  'summary': '\\nThe algorithm involves two phases: grouping nodes into communities based on connections and creating a new network with self-loops and weighted edges. It has been used to partition social networks, track dynamic communities in mobile phone networks, and detect species in network-based models.'},\n",
       " {'title': '3. Understanding Community Membership and Node Scores',\n",
       "  'summary': \"\\nThe text explains the significance of community membership and node score in a network. A node's score is determined by its belonging to a specific community, with a score of 1 for community 1 and -1 for community 2. This information is crucial in understanding the connections and relationships within a network.\"},\n",
       " {'title': '4. The Louvain Method: Optimizing Modularity in Network Clustering',\n",
       "  'summary': ' The Louvain Method is a popular algorithm for finding non-overlapping communities in large networks. It uses modularity optimization, which measures the density of edges within communities compared to edges between communities, to group nodes in a network. This method is efficient and has been shown to outperform other similar methods in terms of speed and modularity value. However, it has limitations such as a resolution limit for detecting smaller clusters. Other methods, such as the Leiden algorithm, have been developed to address these limitations. The text also discusses the importance of speed and modularity value when comparing different community detection methods. '},\n",
       " {'title': '5. Understanding Modularity in Weighted Graphs and its Limitations',\n",
       "  'summary': ' Modularity is a measure of how well a graph can be divided into communities. It takes into account the number of edges within a community compared to the expected number. This measure is used in community detection algorithms to identify distinct groups within a network. The modularity matrix is a key tool in network analysis, used to measure the strength of communities within a network. Modularity optimization involves calculating the difference between the sum of weights of links within a community and the expected sum of weights based on a null model. This measure helps identify communities within a network by maximizing the difference between these two values. However, modularity maximization is not statistically consistent and has a resolution limit, making it unable to detect small communities. Modularity is a useful concept for identifying and analyzing different communities within a network.'},\n",
       " {'title': '6. Analyzing Adjacency Matrices and Random Connections in Networks',\n",
       "  'summary': '\\nThis text explains how to represent a network using an adjacency matrix and discusses the expected number of edges between two nodes in a randomly rewired network. It also discusses the relationship between stubs in a random graph and the probability of connections between nodes. The text explains how to calculate the total number of edges between two vertices and discusses approximations for large random networks. It also discusses the concept of achieving a constant average degree in graphs, which can be useful in various applications.'},\n",
       " {'title': '7. Community Detection and Edge Fraction: A Comprehensive Analysis',\n",
       "  'summary': \" The formula calculates the fraction of edges connected to vertices in a specific community, using the total number of edges and the adjacency matrix. This helps identify the strength of connections within the community and understand the network's structure.\"},\n",
       " {'title': '8. Community Detection in Undirected Networks: Methods and Techniques',\n",
       "  'summary': ' The text explores community detection in a network with 10 nodes and 12 edges. It introduces a new way to measure the belonging of a vertex to a group, which is helpful for optimizing algorithms. The results of the optimal community partitions are shown in a figure.'},\n",
       " {'title': '9. The Importance of Community Structures in Complex Networks',\n",
       "  'summary': ' This text discusses the concept of community structure in complex networks, where nodes are grouped based on their dense connections. This is important in studying various networks and can be applied in community search. Community structure refers to groups of nodes that are more connected within themselves than with the rest of the network. Identifying these communities can provide insights into network function and topology. However, not all networks exhibit meaningful community structure, making it important to use advanced models. '},\n",
       " {'title': '10. Delta and Q: Key Factors in Machine Learning and Sparse Cases',\n",
       "  'summary': ' This text discusses the use of delta and Q in machine learning, as well as the importance of set notation in mathematics. It also explains the concept of expected value and its applications in various fields. Additionally, the text touches on the impact of social media on mental health and the effects of caffeine intake and withdrawal on the body.'},\n",
       " {'title': '11. The Challenges of Multiresolution Methods for Modularity Optimization',\n",
       "  'summary': ' Multiresolution methods attempt to overcome the resolution limit in modularity optimization by adding a resistance or null-case term parameter. However, these methods are not effective for highly heterogeneous communities with varying sizes.'},\n",
       " {'title': '12. Community Detection in Graphs: Impact and Importance',\n",
       "  'summary': ' This text introduces the concept of community detection in graphs and provides resources for implementing algorithms. It explains how communities are identified within a network and their importance in understanding network structure and relationships. The text also discusses the challenges of detecting communities when the difference between internal and external connections is small. Community detection has practical applications in fields such as social networks and can improve algorithms and predict missing or false links in network data. '},\n",
       " {'title': '13. Approximations and Algorithms for Community Detection in Multidimensional Networks',\n",
       "  'summary': ' This text explores different methods for identifying communities in multidimensional networks, including measuring edge density within and between clusters and considering alternative definitions of communities. It also discusses challenges in finding communities and two methods, minimum-cut and hierarchical clustering, for identifying them. While these methods have varying levels of success, they may not be ideal for all types of networks.'},\n",
       " {'title': '14. The Girvan-Newman Algorithm: A Powerful Tool for Community Detection',\n",
       "  'summary': ' The Girvan-Newman algorithm is a method for finding communities in a network by removing edges between them. However, it is slow and not suitable for large networks. Another method, modularity maximization, uses a benefit function to identify communities with high modularity.'},\n",
       " {'title': '15. Community Detection in Latent Feature Spaces: Techniques and Applications',\n",
       "  'summary': ' The text explores using representation learning and clustering methods to detect community structures in networks. It mentions techniques for projecting networks onto a latent space and using Euclidean and Hypergeometric spaces.'},\n",
       " {'title': '16. Identifying Communities in Networks Using Clique Graphs and k-cliques',\n",
       "  'summary': '\\nThis text discusses different methods for identifying communities in networks, with a focus on using cliques in graph analysis. It explains how nodes can belong to multiple cliques, leading to an overlapping community structure. The two main approaches are finding maximal cliques and using cliques of a fixed size. These methods are commonly used in social network analysis software. Clique graphs, which represent cliques and their overlap, can be used to determine community membership of nodes. The clique percolation method defines communities as percolation clusters of k-cliques. This text also explains how to identify communities by finding all k-cliques and defining them as adjacent if they share k-1 nodes. Communities are essentially the connected components in the clique graph.'},\n",
       " {'title': '17. Community Detection Benchmarking: Evaluating Algorithms on Stochastic Block Models and the Fundamental Limit of Detecting Communities in Networks',\n",
       "  'summary': \"\\nThe text discusses the use of benchmark graphs to evaluate community detection algorithms. These benchmarks start with a defined community structure and gradually make it more challenging for algorithms to detect. Measures such as normalized mutual information are used to compare the algorithm's solution to the original structure. The text also mentions a fundamental limit on our ability to detect communities in networks, regardless of the algorithm or resources available. In a stochastic block model, a network will have community structure if the connection probability within groups is greater than between groups, but this limit still applies in sparse cases.\"}]"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stage_2_outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "\n",
       "The Kronecker delta function is a useful tool for calculating the modularity of communities within a network. This measure helps identify cohesive and well-connected groups within a network by comparing the actual number of edges within a community to the expected number. The Louvain Method is a popular algorithm that uses modularity optimization to efficiently group nodes in a network. However, it has limitations such as a resolution limit for detecting smaller clusters. Other methods, such as the Leiden algorithm, have been developed to address these limitations. Modularity is also used in community detection to measure the strength of connections within a community. It involves calculating the difference between the sum of weights of links within a community and the expected sum of weights based on a null model. However, modularity maximization is not statistically consistent and has a resolution limit, making it unable to detect small communities. The concept of community structure in complex networks is important in understanding network function and topology. However, not all networks exhibit meaningful community structure, making it important to use advanced models. Community detection has practical applications in fields such as social networks and can improve algorithms and predict missing or false links in network data. Different methods, such as multiresolution methods and representation learning, have been developed to overcome the challenges of detecting communities"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import Markdown, display\n",
    "\n",
    "display(Markdown(f'{final_summary}'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# With Llamaindex"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## state if the union ListIndex basic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# With llamaindex\n",
    "\n",
    "import openai\n",
    "from llama_index.llms import OpenAI\n",
    "\n",
    "from llama_index import (\n",
    "  SimpleDirectoryReader,\n",
    "  ListIndex,\n",
    "  ServiceContext,\n",
    ")\n",
    "\n",
    "llm = OpenAI(temperature=0, model=\"gpt-3.5-turbo\")\n",
    "\n",
    "service_context = ServiceContext.from_defaults(llm=llm, chunk_size=1024)\n",
    "\n",
    "documents = SimpleDirectoryReader('/workspaces/ml-learning/src/phages/data/test1').load_data()\n",
    "\n",
    "index = ListIndex.from_documents(documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[27], line 9\u001b[0m\n\u001b[1;32m      6\u001b[0m WikipediaReader \u001b[38;5;241m=\u001b[39m download_loader(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mWikipediaReader\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m      8\u001b[0m loader \u001b[38;5;241m=\u001b[39m WikipediaReader()\n\u001b[0;32m----> 9\u001b[0m documents \u001b[38;5;241m=\u001b[39m \u001b[43mloader\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mload_data\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpages\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mLouvain Method\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mModularity (networks)\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mCommunity structure\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     11\u001b[0m \u001b[38;5;66;03m# pipeline = IngestionPipeline(transformations=[TokenTextSplitter()])\u001b[39;00m\n\u001b[1;32m     12\u001b[0m \n\u001b[1;32m     13\u001b[0m \u001b[38;5;66;03m# nodes = pipeline.run(documents=documents)\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/download/llamahub_modules/wikipedia/base.py:29\u001b[0m, in \u001b[0;36mWikipediaReader.load_data\u001b[0;34m(self, pages, lang, **load_kwargs)\u001b[0m\n\u001b[1;32m     27\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m page \u001b[38;5;129;01min\u001b[39;00m pages:\n\u001b[1;32m     28\u001b[0m     wikipedia\u001b[38;5;241m.\u001b[39mset_lang(lang)\n\u001b[0;32m---> 29\u001b[0m     page_content \u001b[38;5;241m=\u001b[39m \u001b[43mwikipedia\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpage\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpage\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mload_kwargs\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mcontent\n\u001b[1;32m     30\u001b[0m     results\u001b[38;5;241m.\u001b[39mappend(Document(text\u001b[38;5;241m=\u001b[39mpage_content))\n\u001b[1;32m     31\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m results\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/wikipedia/wikipedia.py:270\u001b[0m, in \u001b[0;36mpage\u001b[0;34m(title, pageid, auto_suggest, redirect, preload)\u001b[0m\n\u001b[1;32m    268\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m title \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    269\u001b[0m   \u001b[38;5;28;01mif\u001b[39;00m auto_suggest:\n\u001b[0;32m--> 270\u001b[0m     results, suggestion \u001b[38;5;241m=\u001b[39m \u001b[43msearch\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtitle\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mresults\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msuggestion\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[1;32m    271\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    272\u001b[0m       title \u001b[38;5;241m=\u001b[39m suggestion \u001b[38;5;129;01mor\u001b[39;00m results[\u001b[38;5;241m0\u001b[39m]\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/wikipedia/util.py:28\u001b[0m, in \u001b[0;36mcache.__call__\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m     26\u001b[0m   ret \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_cache[key]\n\u001b[1;32m     27\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m---> 28\u001b[0m   ret \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_cache[key] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     30\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ret\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/wikipedia/wikipedia.py:103\u001b[0m, in \u001b[0;36msearch\u001b[0;34m(query, results, suggestion)\u001b[0m\n\u001b[1;32m    100\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m suggestion:\n\u001b[1;32m    101\u001b[0m   search_params[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124msrinfo\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124msuggestion\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m--> 103\u001b[0m raw_results \u001b[38;5;241m=\u001b[39m \u001b[43m_wiki_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43msearch_params\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    105\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124merror\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;129;01min\u001b[39;00m raw_results:\n\u001b[1;32m    106\u001b[0m   \u001b[38;5;28;01mif\u001b[39;00m raw_results[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124merror\u001b[39m\u001b[38;5;124m'\u001b[39m][\u001b[38;5;124m'\u001b[39m\u001b[38;5;124minfo\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;129;01min\u001b[39;00m (\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mHTTP request timed out.\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mPool queue is full\u001b[39m\u001b[38;5;124m'\u001b[39m):\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/wikipedia/wikipedia.py:737\u001b[0m, in \u001b[0;36m_wiki_request\u001b[0;34m(params)\u001b[0m\n\u001b[1;32m    734\u001b[0m   wait_time \u001b[38;5;241m=\u001b[39m (RATE_LIMIT_LAST_CALL \u001b[38;5;241m+\u001b[39m RATE_LIMIT_MIN_WAIT) \u001b[38;5;241m-\u001b[39m datetime\u001b[38;5;241m.\u001b[39mnow()\n\u001b[1;32m    735\u001b[0m   time\u001b[38;5;241m.\u001b[39msleep(\u001b[38;5;28mint\u001b[39m(wait_time\u001b[38;5;241m.\u001b[39mtotal_seconds()))\n\u001b[0;32m--> 737\u001b[0m r \u001b[38;5;241m=\u001b[39m \u001b[43mrequests\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43mAPI_URL\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mparams\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    739\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m RATE_LIMIT:\n\u001b[1;32m    740\u001b[0m   RATE_LIMIT_LAST_CALL \u001b[38;5;241m=\u001b[39m datetime\u001b[38;5;241m.\u001b[39mnow()\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/api.py:73\u001b[0m, in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     62\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget\u001b[39m(url, params\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m     63\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a GET request.\u001b[39;00m\n\u001b[1;32m     64\u001b[0m \n\u001b[1;32m     65\u001b[0m \u001b[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[38;5;124;03m    :rtype: requests.Response\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 73\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mget\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mparams\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[1;32m     56\u001b[0m \u001b[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[1;32m     57\u001b[0m \u001b[38;5;66;03m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[1;32m     58\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m sessions\u001b[38;5;241m.\u001b[39mSession() \u001b[38;5;28;01mas\u001b[39;00m session:\n\u001b[0;32m---> 59\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43msession\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    584\u001b[0m send_kwargs \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m    585\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m\"\u001b[39m: timeout,\n\u001b[1;32m    586\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m: allow_redirects,\n\u001b[1;32m    587\u001b[0m }\n\u001b[1;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprep\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43msend_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:725\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    722\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m allow_redirects:\n\u001b[1;32m    723\u001b[0m     \u001b[38;5;66;03m# Redirect resolving generator.\u001b[39;00m\n\u001b[1;32m    724\u001b[0m     gen \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mresolve_redirects(r, request, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m--> 725\u001b[0m     history \u001b[38;5;241m=\u001b[39m \u001b[43m[\u001b[49m\u001b[43mresp\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mresp\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mgen\u001b[49m\u001b[43m]\u001b[49m\n\u001b[1;32m    726\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    727\u001b[0m     history \u001b[38;5;241m=\u001b[39m []\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:725\u001b[0m, in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    722\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m allow_redirects:\n\u001b[1;32m    723\u001b[0m     \u001b[38;5;66;03m# Redirect resolving generator.\u001b[39;00m\n\u001b[1;32m    724\u001b[0m     gen \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mresolve_redirects(r, request, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m--> 725\u001b[0m     history \u001b[38;5;241m=\u001b[39m \u001b[43m[\u001b[49m\u001b[43mresp\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mresp\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mgen\u001b[49m\u001b[43m]\u001b[49m\n\u001b[1;32m    726\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    727\u001b[0m     history \u001b[38;5;241m=\u001b[39m []\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:266\u001b[0m, in \u001b[0;36mSessionRedirectMixin.resolve_redirects\u001b[0;34m(self, resp, req, stream, timeout, verify, cert, proxies, yield_requests, **adapter_kwargs)\u001b[0m\n\u001b[1;32m    263\u001b[0m     \u001b[38;5;28;01myield\u001b[39;00m req\n\u001b[1;32m    264\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 266\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    267\u001b[0m \u001b[43m        \u001b[49m\u001b[43mreq\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    268\u001b[0m \u001b[43m        \u001b[49m\u001b[43mstream\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstream\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    269\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    270\u001b[0m \u001b[43m        \u001b[49m\u001b[43mverify\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mverify\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    271\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcert\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcert\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    272\u001b[0m \u001b[43m        \u001b[49m\u001b[43mproxies\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mproxies\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    273\u001b[0m \u001b[43m        \u001b[49m\u001b[43mallow_redirects\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    274\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43madapter_kwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    275\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    277\u001b[0m     extract_cookies_to_jar(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcookies, prepared_request, resp\u001b[38;5;241m.\u001b[39mraw)\n\u001b[1;32m    279\u001b[0m     \u001b[38;5;66;03m# extract redirect url, if any, for the next loop\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    700\u001b[0m start \u001b[38;5;241m=\u001b[39m preferred_clock()\n\u001b[1;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m \u001b[43madapter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[1;32m    706\u001b[0m elapsed \u001b[38;5;241m=\u001b[39m preferred_clock() \u001b[38;5;241m-\u001b[39m start\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/requests/adapters.py:486\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    483\u001b[0m     timeout \u001b[38;5;241m=\u001b[39m TimeoutSauce(connect\u001b[38;5;241m=\u001b[39mtimeout, read\u001b[38;5;241m=\u001b[39mtimeout)\n\u001b[1;32m    485\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 486\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murlopen\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    487\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    488\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    489\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    490\u001b[0m \u001b[43m        \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    491\u001b[0m \u001b[43m        \u001b[49m\u001b[43mredirect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    492\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_same_host\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    493\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    494\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    495\u001b[0m \u001b[43m        \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    496\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    497\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    498\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    500\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m    501\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(err, request\u001b[38;5;241m=\u001b[39mrequest)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:790\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    787\u001b[0m response_conn \u001b[38;5;241m=\u001b[39m conn \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m release_conn \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m    789\u001b[0m \u001b[38;5;66;03m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[0;32m--> 790\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    794\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout_obj\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    796\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    797\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    798\u001b[0m \u001b[43m    \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mretries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    799\u001b[0m \u001b[43m    \u001b[49m\u001b[43mresponse_conn\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresponse_conn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    800\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpreload_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    801\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdecode_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    802\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kw\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    803\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    805\u001b[0m \u001b[38;5;66;03m# Everything went great!\u001b[39;00m\n\u001b[1;32m    806\u001b[0m clean_exit \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:536\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    534\u001b[0m \u001b[38;5;66;03m# Receive the response from the server\u001b[39;00m\n\u001b[1;32m    535\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 536\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mgetresponse\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    537\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (BaseSSLError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[1;32m    538\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_raise_timeout(err\u001b[38;5;241m=\u001b[39me, url\u001b[38;5;241m=\u001b[39murl, timeout_value\u001b[38;5;241m=\u001b[39mread_timeout)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/urllib3/connection.py:461\u001b[0m, in \u001b[0;36mHTTPConnection.getresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    458\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mresponse\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m HTTPResponse\n\u001b[1;32m    460\u001b[0m \u001b[38;5;66;03m# Get the response from http.client.HTTPConnection\u001b[39;00m\n\u001b[0;32m--> 461\u001b[0m httplib_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mgetresponse\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    463\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    464\u001b[0m     assert_header_parsing(httplib_response\u001b[38;5;241m.\u001b[39mmsg)\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/http/client.py:1378\u001b[0m, in \u001b[0;36mHTTPConnection.getresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1376\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m   1377\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 1378\u001b[0m         \u001b[43mresponse\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbegin\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1379\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m:\n\u001b[1;32m   1380\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mclose()\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/http/client.py:318\u001b[0m, in \u001b[0;36mHTTPResponse.begin\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    316\u001b[0m \u001b[38;5;66;03m# read until we get a non-100 response\u001b[39;00m\n\u001b[1;32m    317\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[0;32m--> 318\u001b[0m     version, status, reason \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_read_status\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    319\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m status \u001b[38;5;241m!=\u001b[39m CONTINUE:\n\u001b[1;32m    320\u001b[0m         \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/http/client.py:279\u001b[0m, in \u001b[0;36mHTTPResponse._read_status\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    278\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_read_status\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[0;32m--> 279\u001b[0m     line \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mstr\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfp\u001b[38;5;241m.\u001b[39mreadline(_MAXLINE \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m1\u001b[39m), \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124miso-8859-1\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    280\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(line) \u001b[38;5;241m>\u001b[39m _MAXLINE:\n\u001b[1;32m    281\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m LineTooLong(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mstatus line\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/socket.py:706\u001b[0m, in \u001b[0;36mSocketIO.readinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    704\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[1;32m    705\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 706\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sock\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrecv_into\u001b[49m\u001b[43m(\u001b[49m\u001b[43mb\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    707\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m timeout:\n\u001b[1;32m    708\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_timeout_occurred \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1278\u001b[0m, in \u001b[0;36mSSLSocket.recv_into\u001b[0;34m(self, buffer, nbytes, flags)\u001b[0m\n\u001b[1;32m   1274\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m flags \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m   1275\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m   1276\u001b[0m           \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnon-zero flags not allowed in calls to recv_into() on \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m\n\u001b[1;32m   1277\u001b[0m           \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__class__\u001b[39m)\n\u001b[0;32m-> 1278\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnbytes\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbuffer\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1279\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1280\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39mrecv_into(buffer, nbytes, flags)\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/ssl.py:1134\u001b[0m, in \u001b[0;36mSSLSocket.read\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m   1132\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m   1133\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m buffer \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m-> 1134\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sslobj\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbuffer\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1135\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1136\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mread(\u001b[38;5;28mlen\u001b[39m)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "from llama_index import SimpleDirectoryReader\n",
    "from llama_index.ingestion import IngestionPipeline\n",
    "from llama_index.node_parser import TokenTextSplitter\n",
    "from llama_index import download_loader\n",
    "\n",
    "WikipediaReader = download_loader(\"WikipediaReader\")\n",
    "\n",
    "loader = WikipediaReader()\n",
    "documents = loader.load_data(pages=['Louvain Method', 'Modularity (networks)', 'Community structure'])\n",
    "\n",
    "# pipeline = IngestionPipeline(transformations=[TokenTextSplitter()])\n",
    "\n",
    "# nodes = pipeline.run(documents=documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "index = ListIndex.build_index_from_nodes(nodes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# custom node parser\n",
    "\n",
    "# https://docs.llamaindex.ai/en/stable/module_guides/loading/node_parsers/root.html\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ServiceContext(llm_predictor=LLMPredictor(system_prompt=None, query_wrapper_prompt=None, pydantic_program_mode=<PydanticProgramMode.DEFAULT: 'default'>), prompt_helper=PromptHelper(context_window=4096, num_output=256, chunk_overlap_ratio=0.1, chunk_size_limit=None, separator=' '), embed_model=OpenAIEmbedding(model_name='text-embedding-ada-002', embed_batch_size=10, callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b551f610>, additional_kwargs={}, api_key='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', api_base='https://api.openai.com/v1', api_version='', max_retries=10, timeout=60.0, default_headers=None, reuse_client=True), transformations=[SentenceSplitter(include_metadata=True, include_prev_next_rel=True, callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b551f610>, id_func=<function default_id_func at 0x7fc2b68e9ee0>, chunk_size=1024, chunk_overlap=200, separator=' ', paragraph_separator='\\n\\n\\n', secondary_chunking_regex='[^,.;。？！]+[,.;。？！]?')], llama_logger=<llama_index.logger.base.LlamaLogger object at 0x7fc2b5082a90>, callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b551f610>)"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "service_context"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'_service_context': ServiceContext(llm_predictor=LLMPredictor(system_prompt=None, query_wrapper_prompt=None, pydantic_program_mode=<PydanticProgramMode.DEFAULT: 'default'>), prompt_helper=PromptHelper(context_window=4096, num_output=256, chunk_overlap_ratio=0.1, chunk_size_limit=None, separator=' '), embed_model=OpenAIEmbedding(model_name='text-embedding-ada-002', embed_batch_size=10, callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b5a674d0>, additional_kwargs={}, api_key='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', api_base='https://api.openai.com/v1', api_version='', max_retries=10, timeout=60.0, default_headers=None, reuse_client=True), transformations=[SentenceSplitter(include_metadata=True, include_prev_next_rel=True, callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b5a674d0>, id_func=<function default_id_func at 0x7fc2b68e9ee0>, chunk_size=1024, chunk_overlap=200, separator=' ', paragraph_separator='\\n\\n\\n', secondary_chunking_regex='[^,.;。？！]+[,.;。？！]?')], llama_logger=<llama_index.logger.base.LlamaLogger object at 0x7fc2b5a67750>, callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b5a674d0>), '_storage_context': StorageContext(docstore=<llama_index.storage.docstore.simple_docstore.SimpleDocumentStore object at 0x7fc2b5c1b510>, index_store=<llama_index.storage.index_store.simple_index_store.SimpleIndexStore object at 0x7fc2b5a66ed0>, vector_stores={'default': <llama_index.vector_stores.simple.SimpleVectorStore object at 0x7fc2b6191290>, 'image': <llama_index.vector_stores.simple.SimpleVectorStore object at 0x7fc2b5a67250>}, graph_store=<llama_index.graph_stores.simple.SimpleGraphStore object at 0x7fc2b5a66f10>), '_docstore': <llama_index.storage.docstore.simple_docstore.SimpleDocumentStore object at 0x7fc2b5c1b510>, '_show_progress': False, '_vector_store': <llama_index.vector_stores.simple.SimpleVectorStore object at 0x7fc2b6191290>, '_graph_store': <llama_index.graph_stores.simple.SimpleGraphStore object at 0x7fc2b5a66f10>, '_index_struct': IndexList(index_id='b97932b7-2774-4835-a84a-b863930ccf4e', summary=None, nodes=['2d3205d1-022c-4ce4-80d4-5c569cc85eb0', '5bfccebb-2737-4e12-90e3-51dcc8001648', '26562ada-fd18-4eb2-b75a-0f66e7c1c53d', 'c696beaa-2515-459c-872b-4853f3c3baf0', 'f8be7847-6491-42c3-8134-c7608d52b0ac', '2233ee04-9cca-42dd-aa0b-04601f12818d', '125d003f-caee-4613-9750-1ae5526779d9', 'e9cd869f-6728-449d-8cbd-01eab4b48363', 'dce4caa0-7250-45f9-b196-b37da180465b', '06464a37-cc22-4936-a39c-72b7e2e89b50', 'b4bd2f3d-7172-4975-8d36-dacc224174c6'])}\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "<b>index._service_context</b>: ServiceContext(llm_predictor=LLMPredictor(system_prompt=None, query_wrapper_prompt=None, pydantic_program_mode=<PydanticProgramMode.DEFAULT: 'default'>), prompt_helper=PromptHelper(context_window=4096, num_output=256, chunk_overlap_ratio=0.1, chunk_size_limit=None, separator=' '), embed_model=OpenAIEmbedding(model_name='text-embedding-ada-002', embed_batch_size=10, callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b5a674d0>, additional_kwargs={}, api_key='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', api_base='https://api.openai.com/v1', api_version='', max_retries=10, timeout=60.0, default_headers=None, reuse_client=True), transformations=[SentenceSplitter(include_metadata=True, include_prev_next_rel=True, callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b5a674d0>, id_func=<function default_id_func at 0x7fc2b68e9ee0>, chunk_size=1024, chunk_overlap=200, separator=' ', paragraph_separator='\\n\\n\\n', secondary_chunking_regex='[^,.;。？！]+[,.;。？！]?')], llama_logger=<llama_index.logger.base.LlamaLogger object at 0x7fc2b5a67750>, callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b5a674d0>)"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<b>index._service_context.llm_predictor</b>: system_prompt=None query_wrapper_prompt=None pydantic_program_mode=<PydanticProgramMode.DEFAULT: 'default'>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<b>index._service_context.prompt_helper</b>: context_window=4096 num_output=256 chunk_overlap_ratio=0.1 chunk_size_limit=None separator=' '"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<b>index._service_context.embed_model</b>: model_name='text-embedding-ada-002' embed_batch_size=10 callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b5a674d0> additional_kwargs={} api_key='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' api_base='https://api.openai.com/v1' api_version='' max_retries=10 timeout=60.0 default_headers=None reuse_client=True"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<b>index._service_context.transformations</b>: [SentenceSplitter(include_metadata=True, include_prev_next_rel=True, callback_manager=<llama_index.callbacks.base.CallbackManager object at 0x7fc2b5a674d0>, id_func=<function default_id_func at 0x7fc2b68e9ee0>, chunk_size=1024, chunk_overlap=200, separator=' ', paragraph_separator='\\n\\n\\n', secondary_chunking_regex='[^,.;。？！]+[,.;。？！]?')]"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<b>index._storage_context</b>: StorageContext(docstore=<llama_index.storage.docstore.simple_docstore.SimpleDocumentStore object at 0x7fc2b5c1b510>, index_store=<llama_index.storage.index_store.simple_index_store.SimpleIndexStore object at 0x7fc2b5a66ed0>, vector_stores={'default': <llama_index.vector_stores.simple.SimpleVectorStore object at 0x7fc2b6191290>, 'image': <llama_index.vector_stores.simple.SimpleVectorStore object at 0x7fc2b5a67250>}, graph_store=<llama_index.graph_stores.simple.SimpleGraphStore object at 0x7fc2b5a66f10>)"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<b>index._index_struct</b>: \"\"\"IndexList(index_id='b97932b7-2774-4835-a84a-b863930ccf4e', summary=None, nodes=['2d3205d1-022c-4ce4-80d4-5c569cc85eb0', '5bfccebb-2737-4e12-90e3-51dcc8001648', '26562ada-fd18-4eb2-b75a-0f66e7c1c53d', 'c696beaa-2515-459c-872b-4853f3c3baf0', 'f8be7847-6491-42c3-8134-c7608d52b0ac', '2233ee04-9cca-42dd-aa0b-04601f12818d', '125d003f-caee-4613-9750-1ae5526779d9', 'e9cd869f-6728-449d-8cbd-01eab4b48363', 'dce4caa0-7250-45f9-b196-b37da180465b', '06464a37-cc22-4936-a39c-72b7e2e89b50', 'b4bd2f3d-7172-4975-8d36-dacc224174c6'])\"\"\""
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(vars(index))\n",
    "\n",
    "display(Markdown(f'<b>index._service_context</b>: {index._service_context}'))\n",
    "display(Markdown(f'<b>index._service_context.llm_predictor</b>: {index._service_context.llm_predictor}'))\n",
    "display(Markdown(f'<b>index._service_context.prompt_helper</b>: {index._service_context.prompt_helper}'))\n",
    "display(Markdown(f'<b>index._service_context.embed_model</b>: {index._service_context.embed_model}'))\n",
    "display(Markdown(f'<b>index._service_context.transformations</b>: {index._service_context.transformations}'))\n",
    "\n",
    "display(Markdown(f'<b>index._storage_context</b>: {index._storage_context}'))\n",
    "display(Markdown(f'<b>index._index_struct</b>: \"\"\"{index._index_struct}\"\"\"'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "doc = documents[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "doc.id_: 99420c3f-a95d-4755-87ea-0ec07bd1b737"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "doc.embedding: None"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "doc.metadata: {'file_path': '/workspaces/ml-learning/src/phages/data/test1/stateoftheunion.txt', 'file_name': 'stateoftheunion.txt', 'file_type': 'text/plain', 'file_size': 43412, 'creation_date': '2024-01-07', 'last_modified_date': '2024-01-07', 'last_accessed_date': '2024-01-07'}"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "start_char_idx: None"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "end_char_idx: None"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "text_template: \"\"\"{metadata_str}\n",
       "\n",
       "{content}\"\"\""
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(Markdown(f'doc.id_: {doc.id_}'))\n",
    "display(Markdown(f'doc.embedding: {doc.embedding}'))\n",
    "display(Markdown(f'doc.metadata: {doc.metadata}'))\n",
    "display(Markdown(f'start_char_idx: {doc.start_char_idx}'))\n",
    "display(Markdown(f'end_char_idx: {doc.end_char_idx}'))\n",
    "display(Markdown(f'text_template: \"\"\"{doc.text_template}\"\"\"'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "prompt = \"Summarize, don't use vague references such as The Speaker, instead use names…\"\n",
    "response = index.as_query_engine(response_mode=\"tree_summarize\").query(prompt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "In the given context, Paul Pelosi addresses various individuals and congratulates them, including Kevin McCarthy as the new Speaker of the House, Hakeem Jeffries as the first Black House Minority Leader, Mitch McConnell as the longest-serving Senate Leader, and Chuck Schumer as the Senate Majority Leader. Paul Pelosi emphasizes the progress and resilience of America, highlighting the creation of 12 million new jobs, the control of COVID-19, and the strength of democracy. He also mentions bipartisan cooperation in passing laws and emphasizes the importance of the middle class and American manufacturing. Paul Pelosi discusses infrastructure investments, including the Bipartisan Infrastructure Law, and mentions specific projects like the Brent Spence bridge. He also addresses healthcare costs, particularly the cost of insulin, and the need to lower prescription drug prices. Paul Pelosi emphasizes the importance of tackling the climate crisis and investing in clean energy. He mentions the need for the wealthiest and biggest corporations to pay their fair share of taxes.\n"
     ]
    }
   ],
   "source": [
    "print(response)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'Markdown' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[30], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m display(\u001b[43mMarkdown\u001b[49m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mresponse\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m'\u001b[39m))\n",
      "\u001b[0;31mNameError\u001b[0m: name 'Markdown' is not defined"
     ]
    }
   ],
   "source": [
    "\n",
    "display(Markdown(f'{response}'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## llamaindex with custom tree summarize prompt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index import SimpleDirectoryReader, VectorStoreIndex, ServiceContext\n",
    "from llama_index.text_splitter import SentenceSplitter\n",
    "\n",
    "text_splitter = SentenceSplitter(chunk_size=256, chunk_overlap=20)\n",
    "service_context = ServiceContext.from_defaults(text_splitter=text_splitter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "#display(Markdown(documents[0].get_content()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsing nodes: 100%|██████████| 3/3 [00:00<00:00, 27.39it/s]\n"
     ]
    }
   ],
   "source": [
    "nodes = text_splitter.get_nodes_from_documents(documents, show_progress=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "65"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(nodes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index import SummaryIndex, GPTDocumentSummaryIndex\n",
    "from llama_index.response_synthesizers.type import ResponseMode\n",
    "from llama_index.prompts.base import PromptTemplate\n",
    "from llama_index.prompts.prompt_type import PromptType\n",
    "\n",
    "list_index = ListIndex(nodes, show_progress=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index import SummaryIndex, GPTDocumentSummaryIndex\n",
    "from llama_index.response_synthesizers.type import ResponseMode\n",
    "from llama_index.prompts.base import PromptTemplate\n",
    "from llama_index.prompts.prompt_type import PromptType\n",
    "\n",
    "# index_summary = SummaryIndex.from_documents(documents, show_progress=True)\n",
    "\n",
    "CUSTOM_TREE_SUMMARIZE_PROMPT_TMPL = (\n",
    "    \"Write a 250-word summary of the following, removing irrelevant information. Finish your answer: \\n\"\n",
    "    \"\\n\"\n",
    "    \"\\n\"\n",
    "    \"{context_str}\\n\"\n",
    "    \"\\n\"\n",
    "    \"\\n\"\n",
    "    '250-WORD SUMMARY:\"\"\"\\n'\n",
    ")\n",
    "\n",
    "CUSTOM_TREE_SUMMARIZE_PROMPT = PromptTemplate(\n",
    "   CUSTOM_TREE_SUMMARIZE_PROMPT_TMPL, prompt_type=PromptType.SUMMARY\n",
    ")\n",
    "\n",
    "title_acc_prompt_template = \"\"\"Firstly, give the following text an informative title. Then, on a new line, write a 75-100 word summary of the following text:\n",
    "  {context_str}\n",
    "\n",
    "  Return your answer in the following format:\n",
    "  Title | Summary...\n",
    "  e.g. \n",
    "  Why Artificial Intelligence is Good | AI can make humans more productive by automating many repetitive processes.\n",
    "\n",
    "  TITLE AND CONCISE SUMMARY:\"\"\"\n",
    "\n",
    "title_acc_prompt = PromptTemplate(\n",
    "   title_acc_prompt_template, prompt_type=PromptType.SUMMARY\n",
    ")\n",
    "\n",
    "\n",
    "prompt = \"Summarize…\"\n",
    "# summary_query_engine_tree = index_summary.as_query_engine(response_mode=\"tree_summarize\")\n",
    "# summary_query_engine_compact_acc = index_summary.as_query_engine(response_mode=ResponseMode.COMPACT_ACCUMULATE)\n",
    "# summary_query_engine_refine = index_summary.as_query_engine(response_mode=ResponseMode.REFINE)\n",
    "# summary_query_engine_simple = index_summary.as_query_engine(response_mode=ResponseMode.SIMPLE_SUMMARIZE)\n",
    "\n",
    "# response_compact_acc = summary_query_engine_compact_acc.query(prompt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'TextNode' object has no attribute 'node'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[142], line 11\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mllama_index\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mresponse_synthesizers\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[1;32m      3\u001b[0m     ResponseMode,\n\u001b[1;32m      4\u001b[0m     get_response_synthesizer,\n\u001b[1;32m      5\u001b[0m )\n\u001b[1;32m      7\u001b[0m response_synthesizer \u001b[38;5;241m=\u001b[39m get_response_synthesizer(\n\u001b[1;32m      8\u001b[0m     response_mode\u001b[38;5;241m=\u001b[39mResponseMode\u001b[38;5;241m.\u001b[39mACCUMULATE\n\u001b[1;32m      9\u001b[0m )\n\u001b[0;32m---> 11\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[43mresponse_synthesizer\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msynthesize\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     12\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43msummarize...\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnodes\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnodes\u001b[49m\n\u001b[1;32m     13\u001b[0m \u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/response_synthesizers/base.py:148\u001b[0m, in \u001b[0;36mBaseSynthesizer.synthesize\u001b[0;34m(self, query, nodes, additional_source_nodes, **response_kwargs)\u001b[0m\n\u001b[1;32m    141\u001b[0m     query \u001b[38;5;241m=\u001b[39m QueryBundle(query_str\u001b[38;5;241m=\u001b[39mquery)\n\u001b[1;32m    143\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_callback_manager\u001b[38;5;241m.\u001b[39mevent(\n\u001b[1;32m    144\u001b[0m     CBEventType\u001b[38;5;241m.\u001b[39mSYNTHESIZE, payload\u001b[38;5;241m=\u001b[39m{EventPayload\u001b[38;5;241m.\u001b[39mQUERY_STR: query\u001b[38;5;241m.\u001b[39mquery_str}\n\u001b[1;32m    145\u001b[0m ) \u001b[38;5;28;01mas\u001b[39;00m event:\n\u001b[1;32m    146\u001b[0m     response_str \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mget_response(\n\u001b[1;32m    147\u001b[0m         query_str\u001b[38;5;241m=\u001b[39mquery\u001b[38;5;241m.\u001b[39mquery_str,\n\u001b[0;32m--> 148\u001b[0m         text_chunks\u001b[38;5;241m=\u001b[39m\u001b[43m[\u001b[49m\n\u001b[1;32m    149\u001b[0m \u001b[43m            \u001b[49m\u001b[43mn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mnode\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_content\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmetadata_mode\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mMetadataMode\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mLLM\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mn\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mnodes\u001b[49m\n\u001b[1;32m    150\u001b[0m \u001b[43m        \u001b[49m\u001b[43m]\u001b[49m,\n\u001b[1;32m    151\u001b[0m         \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mresponse_kwargs,\n\u001b[1;32m    152\u001b[0m     )\n\u001b[1;32m    154\u001b[0m     additional_source_nodes \u001b[38;5;241m=\u001b[39m additional_source_nodes \u001b[38;5;129;01mor\u001b[39;00m []\n\u001b[1;32m    155\u001b[0m     source_nodes \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(nodes) \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mlist\u001b[39m(additional_source_nodes)\n",
      "File \u001b[0;32m/workspaces/ml-learning/.venv/lib/python3.11/site-packages/llama_index/response_synthesizers/base.py:149\u001b[0m, in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    141\u001b[0m     query \u001b[38;5;241m=\u001b[39m QueryBundle(query_str\u001b[38;5;241m=\u001b[39mquery)\n\u001b[1;32m    143\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_callback_manager\u001b[38;5;241m.\u001b[39mevent(\n\u001b[1;32m    144\u001b[0m     CBEventType\u001b[38;5;241m.\u001b[39mSYNTHESIZE, payload\u001b[38;5;241m=\u001b[39m{EventPayload\u001b[38;5;241m.\u001b[39mQUERY_STR: query\u001b[38;5;241m.\u001b[39mquery_str}\n\u001b[1;32m    145\u001b[0m ) \u001b[38;5;28;01mas\u001b[39;00m event:\n\u001b[1;32m    146\u001b[0m     response_str \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mget_response(\n\u001b[1;32m    147\u001b[0m         query_str\u001b[38;5;241m=\u001b[39mquery\u001b[38;5;241m.\u001b[39mquery_str,\n\u001b[1;32m    148\u001b[0m         text_chunks\u001b[38;5;241m=\u001b[39m[\n\u001b[0;32m--> 149\u001b[0m             \u001b[43mn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mnode\u001b[49m\u001b[38;5;241m.\u001b[39mget_content(metadata_mode\u001b[38;5;241m=\u001b[39mMetadataMode\u001b[38;5;241m.\u001b[39mLLM) \u001b[38;5;28;01mfor\u001b[39;00m n \u001b[38;5;129;01min\u001b[39;00m nodes\n\u001b[1;32m    150\u001b[0m         ],\n\u001b[1;32m    151\u001b[0m         \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mresponse_kwargs,\n\u001b[1;32m    152\u001b[0m     )\n\u001b[1;32m    154\u001b[0m     additional_source_nodes \u001b[38;5;241m=\u001b[39m additional_source_nodes \u001b[38;5;129;01mor\u001b[39;00m []\n\u001b[1;32m    155\u001b[0m     source_nodes \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(nodes) \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mlist\u001b[39m(additional_source_nodes)\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'TextNode' object has no attribute 'node'"
     ]
    }
   ],
   "source": [
    "from llama_index.schema import Node\n",
    "from llama_index.response_synthesizers import (\n",
    "    ResponseMode,\n",
    "    get_response_synthesizer,\n",
    ")\n",
    "\n",
    "response_synthesizer = get_response_synthesizer(\n",
    "    response_mode=ResponseMode.ACCUMULATE\n",
    ")\n",
    "\n",
    "response = response_synthesizer.synthesize(\n",
    "    \"summarize...\", nodes=nodes\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "response_acc = list_index.as_query_engine(response_mode=ResponseMode.ACCUMULATE, text_qa_template = title_acc_prompt).query('Summarize...')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "response_tree = index_summary.as_query_engine(response_mode=\"tree_summarize\",summarize_template =  CUSTOM_TREE_SUMMARIZE_PROMPT).query('Summarize...')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "response_compact_acc = index_summary.as_query_engine(response_mode=ResponseMode.COMPACT_ACCUMULATE).query('Summarize...')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['9005d22c-718a-48c2-8f30-730159c85576',\n",
       " '47dbaa28-236a-417c-8b0f-5e685c62d996',\n",
       " '1a5fe5e4-7307-423a-a811-7cbe551e5784',\n",
       " '4172d2bb-5ef6-4e06-818a-55bff2631bb5',\n",
       " 'bc113ff8-5048-4844-96ee-354e92b85ec3',\n",
       " '95fbebda-656b-4065-87bb-68f4b81a13fb',\n",
       " 'd8a37425-d847-4ce8-a0a8-a73ae15a6711',\n",
       " '2088cedb-8879-4de0-8e04-7a1f8a7b1c71',\n",
       " 'ea90a275-4076-4a48-b2d7-ac1095ff09eb',\n",
       " '02428601-eb8c-411f-aa7e-ba9aa51358d1',\n",
       " '6bb1fbf7-1a6c-4be9-a701-c2cb7b9054ed']"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index_summary._index_struct.nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "Response 1: The speaker begins by congratulating members of Congress and acknowledging various leaders. They emphasize the progress and resilience of America, highlighting the economic growth, the control over COVID-19, and the strength of democracy. The speaker emphasizes the importance of working together and mentions bipartisan achievements in areas such as defense, infrastructure, and veterans' rights. They discuss the need to rebuild the middle class, create manufacturing jobs, and invest in American products. The speaker also addresses issues such as inflation, healthcare costs, and prescription drug prices. They highlight the importance of infrastructure and the passage of the Bipartisan Infrastructure Law. The speaker mentions specific projects and the impact they will have on communities. They discuss the need to address climate change and invest in clean energy. The speaker emphasizes the importance of fair taxation and making the wealthiest and biggest corporations pay their fair share. They assure that individuals earning less than $400,000 a\n",
       "---------------------\n",
       "Response 2: The speaker discusses various initiatives and proposals aimed at addressing climate change, tax fairness, corporate accountability, consumer protection, worker rights, education, and public safety. They emphasize the need to invest in clean energy, rebuild infrastructure, and create jobs. They also advocate for tax reforms to ensure that the wealthy and big corporations pay their fair share. The speaker highlights the importance of competition in capitalism and proposes measures to strengthen antitrust enforcement. They address issues such as hidden fees, non-compete agreements, and fraud prevention. The speaker emphasizes the need for affordable education, including preschool and community college, and supports measures to improve public school teacher salaries. They also discuss the progress made in fighting the COVID-19 pandemic and the importance of public safety and trust in law enforcement.\n",
       "---------------------\n",
       "Response 3: The speaker emphasizes the importance of creating neighborhoods free of violence and building trust between law enforcement and the community. They acknowledge the challenges faced by police officers and the need for them to be supported and trained adequately. The speaker also highlights the need for additional resources to address mental health and substance abuse issues, as well as to reduce violent and gun crimes. Overall, the speaker calls for collective action to ensure the safety and well-being of everyone in society.\n",
       "---------------------\n",
       "Response 4: The speaker addresses various issues in their speech, including the need for trust between law enforcement and the community, the desire for safe neighborhoods and the protection of children, the importance of equal protection under the law, and the recognition of the sacrifices made by police officers. They also discuss the need for police reform, addressing gun violence, immigration, reproductive rights, LGBTQ rights, and the United States' role in the world. The speaker emphasizes the importance of democracy and the need to protect it, and concludes by calling for unity and optimism in facing the challenges ahead.\n",
       "---------------------\n",
       "Response 5: The speaker addresses the recent political violence and emphasizes the need to speak out against it. They stress the importance of protecting the right to vote, upholding the rule of law, and restoring trust in democratic institutions. The speaker calls for unity, hope, and optimism, urging Americans to see each other as fellow citizens. They express confidence in the strength and future of the nation, emphasizing the capacity to overcome challenges together. The speech concludes with a blessing for all and a prayer for the protection of troops."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "display(Markdown(f'{response_compact_acc}'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "response_acc = index_summary.as_query_engine(response_mode=ResponseMode.ACCUMULATE, text_qa_template = title_acc_prompt).query('Summarize...')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[NodeWithScore(node=TextNode(id_='66798390-546f-479f-9add-79641b81fc92', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='32ee0945-5f2a-41e2-bbdf-0dd1ddc4fab6', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='e2d4fb35246a6751413cd39467fc1541f5fc31f2cffb06e9baa10056d23cee22'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='f32d7231-c5f0-4937-a3a0-e9b9462cfbb3', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='737819098cb269fa43d07308799544cb4bd9bdbe3c9fa92af2dc1ab31fbdc7fc')}, hash='1536d10cd91f74ac363ba6006421eabb84c0a4cbc30269c14570e3c0ddf6ab17', text=\"The Louvain method for community detection is a method to extract non-overlapping communities from large networks created by Blondel et al. from the University of Louvain (the source of this method's name). The method is a greedy optimization method that appears to run in time \\n  \\n    \\n      \\n        O\\n        (\\n        n\\n        ⋅\\n        log\\n        \\u2061\\n        n\\n        )\\n      \\n    \\n    {\\\\displaystyle O(n\\\\cdot \\\\log n)}\\n   where \\n  \\n    \\n      \\n        n\\n      \\n    \\n    {\\\\displaystyle n}\\n   is the number of nodes in the network.\\n\\n\\n== Modularity optimization ==\\nThe inspiration for this method of community detection is the optimization of modularity as the algorithm progresses. Modularity is a scale value between −0.5 (non-modular clustering) and 1 (fully modular clustering) that measures the relative density of edges inside communities with respect to edges outside communities. Optimizing this value theoretically results in the best possible grouping of the nodes of a given network. But because going through all possible iterations of the nodes into groups is impractical, heuristic algorithms are used.\\nIn the Louvain Method of community detection, first small communities are found by optimizing modularity locally on all nodes, then each small community is grouped into one node and the first step is repeated. The method is similar to the earlier method by Clauset, Newman and Moore that connects communities whose amalgamation produces the largest increase in modularity.\\n\\n\\n== Algorithm ==\\nThe value to be optimized is modularity, defined as a value in the range \\n  \\n    \\n      \\n        [\\n        −\\n        1\\n        \\n          /\\n        \\n        2\\n        ,\\n        1\\n        ]\\n      \\n    \\n    {\\\\displaystyle [-1/2,1]}\\n   that measures the density of links inside communities compared to links between communities.\", start_char_idx=0, end_char_idx=1836, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='f32d7231-c5f0-4937-a3a0-e9b9462cfbb3', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='32ee0945-5f2a-41e2-bbdf-0dd1ddc4fab6', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='e2d4fb35246a6751413cd39467fc1541f5fc31f2cffb06e9baa10056d23cee22'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='66798390-546f-479f-9add-79641b81fc92', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1536d10cd91f74ac363ba6006421eabb84c0a4cbc30269c14570e3c0ddf6ab17'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='d097fd2d-2b68-4bbd-b384-5e9e14fb80a0', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='cd2f812338a390bec7e080dc4e13c50fb28dbc2101c70d44af2f05128d92e41a')}, hash='737819098cb269fa43d07308799544cb4bd9bdbe3c9fa92af2dc1ab31fbdc7fc', text='== Algorithm ==\\nThe value to be optimized is modularity, defined as a value in the range \\n  \\n    \\n      \\n        [\\n        −\\n        1\\n        \\n          /\\n        \\n        2\\n        ,\\n        1\\n        ]\\n      \\n    \\n    {\\\\displaystyle [-1/2,1]}\\n   that measures the density of links inside communities compared to links between communities. For a weighted graph, modularity is defined as:\\n\\n  \\n    \\n      \\n        Q\\n        =\\n        \\n          \\n            1\\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          ∑\\n          \\n            i\\n            j\\n          \\n        \\n        \\n          \\n            [\\n          \\n        \\n        \\n          A\\n          \\n            i\\n            j\\n          \\n        \\n        −\\n        \\n          \\n            \\n              \\n                k\\n                \\n                  i\\n                \\n              \\n              \\n                k\\n                \\n                  j\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          \\n            ]\\n          \\n        \\n        δ\\n        (\\n        \\n          c\\n          \\n            i\\n          \\n        \\n        ,\\n        \\n          c\\n          \\n            j\\n          \\n        \\n        )\\n        ,\\n      \\n    \\n    {\\\\displaystyle Q={\\\\frac {1}{2m}}\\\\sum \\\\limits _{ij}{\\\\bigg [}A_{ij}-{\\\\frac {k_{i}k_{j}}{2m}}{\\\\bigg ]}\\\\delta (c_{i},c_{j}),}\\n  \\nwhere\\n\\n  \\n    \\n      \\n        \\n          A\\n          \\n            i\\n            j\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle A_{ij}}\\n   represents the edge weight between nodes \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   and \\n  \\n    \\n      \\n        j\\n      \\n    \\n    {\\\\displaystyle j}\\n  ;\\n\\n  \\n    \\n      \\n        \\n          k\\n          \\n            i\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{i}}\\n   and \\n  \\n    \\n      \\n        \\n          k\\n          \\n            j\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{j}}\\n   are the sum of the weights of the edges attached to nodes \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   and \\n  \\n    \\n      \\n        j\\n      \\n    \\n    {\\\\displaystyle j}\\n  , respectively;\\n\\n  \\n    \\n      \\n        m\\n      \\n    \\n    {\\\\displaystyle m}\\n   is the sum of all of the edge weights in the graph;\\n\\n  \\n    \\n      \\n        \\n          c\\n          \\n            i\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle c_{i}}\\n   and \\n  \\n    \\n      \\n        \\n          c\\n          \\n            j\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle c_{j}}\\n   are the communities of the nodes; and\\n\\n  \\n    \\n      \\n        δ\\n      \\n    \\n    {\\\\displaystyle \\\\delta }\\n   is Kronecker delta function (\\n  \\n    \\n      \\n        δ\\n        (\\n        x\\n        ,\\n        y\\n        )\\n        =\\n        1\\n      \\n    \\n    {\\\\displaystyle \\\\delta (x,y)=1}\\n   if \\n  \\n    \\n      \\n        x\\n        =\\n        y\\n      \\n    \\n    {\\\\displaystyle x=y}\\n  , \\n  \\n    \\n      \\n        0\\n      \\n    \\n    {\\\\displaystyle 0}\\n   otherwise).Based on the above equation, the modularity of a community \\n  \\n    \\n      \\n        c\\n      \\n    \\n    {\\\\displaystyle c}\\n   can be calculated as:\\n\\n  \\n    \\n      \\n        \\n          Q\\n          \\n            c\\n          \\n        \\n        =\\n        \\n          \\n            \\n              Σ\\n              \\n                i\\n                n\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        −\\n        (\\n        \\n          \\n            \\n              Σ\\n              \\n                t\\n                o\\n                t\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          )\\n          \\n            2\\n          \\n        \\n        ,\\n      \\n    \\n    {\\\\displaystyle Q_{c}={\\\\frac {\\\\Sigma _{in}}{2m}}-({\\\\frac {\\\\Sigma _{tot}}{2m}})^{2},}\\n  \\nwhere\\n\\n  \\n    \\n      \\n        \\n          Σ\\n          \\n            i\\n            n\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\Sigma _{in}}\\n   is the sum of edge weights between nodes within the community \\n  \\n    \\n      \\n        c\\n      \\n    \\n    {\\\\displaystyle c}\\n   (each edge is considered twice); and\\n\\n  \\n    \\n      \\n        \\n          Σ\\n          \\n            t\\n            o\\n            t\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\Sigma _{tot}}\\n   is the sum of all edge weights for nodes within the community (including edges which link to other communities).In order to maximize modularity efficiently, the Louvain Method has two phases that are repeated iteratively.\\nFirst, each node in the network is assigned to its own community. Then for each node \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  , the change in modularity is calculated for removing \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   from its own community and moving it into the community of each neighbor \\n  \\n    \\n      \\n        j\\n      \\n    \\n    {\\\\displaystyle j}\\n   of \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  .', start_char_idx=1495, end_char_idx=6565, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='d097fd2d-2b68-4bbd-b384-5e9e14fb80a0', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='32ee0945-5f2a-41e2-bbdf-0dd1ddc4fab6', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='e2d4fb35246a6751413cd39467fc1541f5fc31f2cffb06e9baa10056d23cee22'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='f32d7231-c5f0-4937-a3a0-e9b9462cfbb3', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='737819098cb269fa43d07308799544cb4bd9bdbe3c9fa92af2dc1ab31fbdc7fc'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='0dcd049c-1bff-4452-acff-bc1abd043e2f', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='75345c548edb35243f5756bee05da5f34aec67cc546b1e75f56f4acbd4fb8715')}, hash='cd2f812338a390bec7e080dc4e13c50fb28dbc2101c70d44af2f05128d92e41a', text='First, each node in the network is assigned to its own community. Then for each node \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  , the change in modularity is calculated for removing \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   from its own community and moving it into the community of each neighbor \\n  \\n    \\n      \\n        j\\n      \\n    \\n    {\\\\displaystyle j}\\n   of \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  . This value is easily calculated by two steps: (1) removing \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   from its original community, and (2) inserting \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   to the community of \\n  \\n    \\n      \\n        j\\n      \\n    \\n    {\\\\displaystyle j}\\n  . The two equations are quite similar, and the equation for step (2) is:\\n  \\n    \\n      \\n        Δ\\n        Q\\n        =\\n        \\n          \\n            [\\n          \\n        \\n        \\n          \\n            \\n              \\n                Σ\\n                \\n                  i\\n                  n\\n                \\n              \\n              +\\n              2\\n              \\n                k\\n                \\n                  i\\n                  ,\\n                  i\\n                  n\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        −\\n        \\n          \\n            (\\n          \\n        \\n        \\n          \\n            \\n              \\n                Σ\\n                \\n                  t\\n                  o\\n                  t\\n                \\n              \\n              +\\n              \\n                k\\n                \\n                  i\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          \\n            \\n              )\\n            \\n          \\n          \\n            2\\n          \\n        \\n        \\n          \\n            ]\\n          \\n        \\n        −\\n        \\n          \\n            [\\n          \\n        \\n        \\n          \\n            \\n              Σ\\n              \\n                i\\n                n\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        −\\n        \\n          \\n            (\\n          \\n        \\n        \\n          \\n            \\n              Σ\\n              \\n                t\\n                o\\n                t\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          \\n            \\n              )\\n            \\n          \\n          \\n            2\\n          \\n        \\n        −\\n        \\n          \\n            (\\n          \\n        \\n        \\n          \\n            \\n              k\\n              \\n                i\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          \\n            \\n              )\\n            \\n          \\n          \\n            2\\n          \\n        \\n        \\n          \\n            ]\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\Delta Q={\\\\bigg [}{\\\\frac {\\\\Sigma _{in}+2k_{i,in}}{2m}}-{\\\\bigg (}{\\\\frac {\\\\Sigma _{tot}+k_{i}}{2m}}{\\\\bigg )}^{2}{\\\\bigg ]}-{\\\\bigg [}{\\\\frac {\\\\Sigma _{in}}{2m}}-{\\\\bigg (}{\\\\frac {\\\\Sigma _{tot}}{2m}}{\\\\bigg )}^{2}-{\\\\bigg (}{\\\\frac {k_{i}}{2m}}{\\\\bigg )}^{2}{\\\\bigg ]}}\\n  \\nWhere \\n  \\n    \\n      \\n        \\n          Σ\\n          \\n            i\\n            n\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\Sigma _{in}}\\n   is sum of all the weights of the links inside the community \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is moving into, \\n  \\n    \\n      \\n        \\n          Σ\\n          \\n            t\\n            o\\n            t\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\Sigma _{tot}}\\n   is the sum of all the weights of the links to nodes in the community \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is moving into, \\n  \\n    \\n      \\n        \\n          k\\n          \\n            i\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{i}}\\n   is the weighted degree of \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  , \\n  \\n    \\n      \\n        \\n          k\\n          \\n            i\\n            ,\\n            i\\n            n\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{i,in}}\\n   is the sum of the weights of the links between \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   and other nodes in the community that \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is moving into, and \\n  \\n    \\n      \\n        m\\n      \\n    \\n    {\\\\displaystyle m}\\n   is the sum of the weights of all links in the network. Then, once this value is calculated for all communities \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is connected to, \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is placed into the community that resulted in the greatest modularity increase. If no increase is possible, \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   remains in its original community. This process is applied repeatedly and sequentially to all nodes until no modularity increase can occur. Once this local maximum of modularity is hit, the first phase has ended.', start_char_idx=6099, end_char_idx=11412, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='0dcd049c-1bff-4452-acff-bc1abd043e2f', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='32ee0945-5f2a-41e2-bbdf-0dd1ddc4fab6', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='e2d4fb35246a6751413cd39467fc1541f5fc31f2cffb06e9baa10056d23cee22'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='d097fd2d-2b68-4bbd-b384-5e9e14fb80a0', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='cd2f812338a390bec7e080dc4e13c50fb28dbc2101c70d44af2f05128d92e41a'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='8f6308d5-0311-46d3-9af2-f03d910e2c42', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='5f45bc9c8639211ebb656ace1dedfd4de7cd2188fbba638c2ec1d57160c846a0')}, hash='75345c548edb35243f5756bee05da5f34aec67cc546b1e75f56f4acbd4fb8715', text='Then, once this value is calculated for all communities \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is connected to, \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is placed into the community that resulted in the greatest modularity increase. If no increase is possible, \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   remains in its original community. This process is applied repeatedly and sequentially to all nodes until no modularity increase can occur. Once this local maximum of modularity is hit, the first phase has ended.\\nIn the second phase of the algorithm, it groups all of the nodes in the same community and builds a new network where nodes are the communities from the previous phase. Any links between nodes of the same community are now represented by self-loops on the new community node and links from multiple nodes in the same community to a node in a different community are represented by weighted edges between communities. Once the new network is created, the second phase has ended and the first phase can be re-applied to the new network.\\n\\n\\n== Previous uses ==\\nTwitter social Network (2.4 Million nodes, 38 million links) by Josep Pujol, Vijay Erramilli, and Pablo Rodriguez: The authors explore the problem of partitioning Online Social Networks onto different machines.\\nMobile phone Network (4 Million nodes, 100 Million links) by Derek Greene, Donal Doyle, and Padraig Cunningham: Community-tracking strategies for identifying dynamic communities of different dynamic social networks.\\nDetecting species in network-based dynamical model.\\n\\n\\n== Disadvantages ==\\nIt is important to emphasize that Louvain produces only non-overlapping communities, which means that each node can belong to at most one community. This is highly unrealistic in many real-world applications. For example, in social networks, most people belong to multiple communities: their family, their friends, their co-workers, old school buddies, etc. In biological networks, most genes or proteins belong to more than one pathway or complex. Furthermore, Louvain has been shown to sometimes produce arbitrarily badly connected communities, and has been effectively superseded (at least in the non-overlapping case) by the Leiden algorithm.\\n\\n\\n== Comparison to other methods of non-overlapping community detection ==\\nWhen comparing modularity optimization methods, the two measures of importance are the speed and the resulting modularity value. A higher speed is better as it shows a method is more efficient than others and a higher modularity value is desirable as it points to having better-defined communities.\\nThe compared methods are, the algorithm of Clauset, Newman, and Moore, Pons and Latapy, and Wakita and Tsurumi.\\n-/- in the table refers to a method that took over 24hrs to run. This table (from) shows that the Louvain method outperforms many similar modularity optimization methods in both the modularity and the time categories.\\n\\n\\n== See also ==\\nModularity (networks)\\nCommunity structure\\nNetwork science\\nK-means clustering\\n\\n\\n== References ==\\n\\n\"The Louvain method for community detection in large networks\" Vincent Blondel http://perso.uclouvain.be/vincent.blondel/research/louvain.html', start_char_idx=10830, end_char_idx=14078, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='8f6308d5-0311-46d3-9af2-f03d910e2c42', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='6859ef38-2941-4a08-a1f0-9a3e7a430898', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='735a591b325be2f57da7c8756161c4cb8071593f110f388a3f18fba441884387'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='0dcd049c-1bff-4452-acff-bc1abd043e2f', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='75345c548edb35243f5756bee05da5f34aec67cc546b1e75f56f4acbd4fb8715'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='c1b544e2-2990-468a-8a35-c210a20bf36d', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='5494deb6493cfd07a1eb0dc824819ba3107da6f6b5298e4b1028613576e6c5cc')}, hash='5f45bc9c8639211ebb656ace1dedfd4de7cd2188fbba638c2ec1d57160c846a0', text='Modularity is a measure of the structure of networks or graphs which measures the strength of division of a network into modules (also called groups, clusters or communities). Networks with high modularity have dense connections between the nodes within modules but sparse connections between nodes in different modules. Modularity is often used in optimization methods for detecting community structure in networks.  Biological networks, including animal brains, exhibit a high degree of modularity. However, modularity maximization is not statistically consistent, and finds communities in its own null model, i.e. fully random graphs, and therefore it cannot be used to find statistically significant community structures in empirical networks. Furthermore, it has been shown that modularity suffers a resolution limit and, therefore, it is unable to detect small communities.\\n\\n\\n== Motivation ==\\nMany scientifically important problems can be represented and empirically studied using networks. For example, biological and social patterns, the World Wide Web, metabolic networks, food webs, neural networks and pathological networks are real world problems that can be mathematically represented and topologically studied to reveal some unexpected structural features. Most of these networks possess a certain community structure that has substantial importance in building an understanding regarding the dynamics of the network. For instance, a closely connected social community will imply a faster rate of transmission of information or rumor among them than a loosely connected community. Thus, if a network is represented by a number of individual nodes connected by links which signify a certain degree of interaction between the nodes, communities are defined as groups of densely interconnected nodes that are only sparsely connected with the rest of the network. Hence, it may be imperative to identify the communities in networks since the communities may have quite different properties such as node degree, clustering coefficient, betweenness, centrality, etc., from that of the average network. Modularity is one such measure, which when maximized, leads to the appearance of communities in a given network.', start_char_idx=0, end_char_idx=2222, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='c1b544e2-2990-468a-8a35-c210a20bf36d', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='6859ef38-2941-4a08-a1f0-9a3e7a430898', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='735a591b325be2f57da7c8756161c4cb8071593f110f388a3f18fba441884387'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='8f6308d5-0311-46d3-9af2-f03d910e2c42', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='5f45bc9c8639211ebb656ace1dedfd4de7cd2188fbba638c2ec1d57160c846a0'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='c950ee50-c6ea-4fb9-b661-85e4af94730d', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='f756c9de205234e9d1af3e169669621adecbea315932bc0f970cc3a177c6fd3b')}, hash='5494deb6493cfd07a1eb0dc824819ba3107da6f6b5298e4b1028613576e6c5cc', text=\"== Definition ==\\nModularity is the fraction of the edges that fall within the given groups minus the expected fraction if edges were distributed at random. The value of the modularity for unweighted and undirected graphs lies in the range \\n  \\n    \\n      \\n        [\\n        −\\n        1\\n        \\n          /\\n        \\n        2\\n        ,\\n        1\\n        ]\\n      \\n    \\n    {\\\\displaystyle [-1/2,1]}\\n  .  It is positive if the number of edges within groups exceeds the number expected on the basis of chance. For a given division of the network's vertices into some modules, modularity reflects the concentration of edges within modules compared with random distribution of links between all nodes regardless of modules.\\nThere are different methods for calculating modularity. In the most common version of the concept, the randomization of the edges is done so as to preserve the degree of each vertex. Consider a graph with \\n  \\n    \\n      \\n        n\\n      \\n    \\n    {\\\\displaystyle n}\\n   nodes and \\n  \\n    \\n      \\n        m\\n      \\n    \\n    {\\\\displaystyle m}\\n   links (edges) such that the graph can be partitioned into two communities using a membership variable \\n  \\n    \\n      \\n        s\\n      \\n    \\n    {\\\\displaystyle s}\\n  . If a node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   belongs to community 1, \\n  \\n    \\n      \\n        \\n          s\\n          \\n            v\\n          \\n        \\n        =\\n        1\\n      \\n    \\n    {\\\\displaystyle s_{v}=1}\\n  , or if \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   belongs to community 2, \\n  \\n    \\n      \\n        \\n          s\\n          \\n            v\\n          \\n        \\n        =\\n        −\\n        1\\n      \\n    \\n    {\\\\displaystyle s_{v}=-1}\\n  . Let the adjacency matrix for the network be represented by \\n  \\n    \\n      \\n        A\\n      \\n    \\n    {\\\\displaystyle A}\\n  , where \\n  \\n    \\n      \\n        \\n          A\\n          \\n            v\\n            w\\n          \\n        \\n        =\\n        0\\n      \\n    \\n    {\\\\displaystyle A_{vw}=0}\\n   means there's no edge (no interaction) between nodes \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n   and \\n  \\n    \\n      \\n        \\n          A\\n          \\n            v\\n            w\\n          \\n        \\n        =\\n        1\\n      \\n    \\n    {\\\\displaystyle A_{vw}=1}\\n   means there is an edge between the two. Also for simplicity we consider an undirected network. Thus \\n  \\n    \\n      \\n        \\n          A\\n          \\n            v\\n            w\\n          \\n        \\n        =\\n        \\n          A\\n          \\n            w\\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle A_{vw}=A_{wv}}\\n  . (It is important to note that multiple edges may exist between two nodes, but here we assess the simplest case).\\nModularity \\n  \\n    \\n      \\n        Q\\n      \\n    \\n    {\\\\displaystyle Q}\\n   is then defined as the fraction of edges that fall within group 1 or 2, minus the expected number of edges within groups 1 and 2 for a random graph with the same node degree distribution as the given network.\\nThe expected number of edges shall be computed using the concept of a configuration model. The configuration model is a randomized realization of a particular network. Given a network with \\n  \\n    \\n      \\n        n\\n      \\n    \\n    {\\\\displaystyle n}\\n   nodes, where each node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   has a node degree \\n  \\n    \\n      \\n        \\n          k\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{v}}\\n  , the configuration model cuts each edge into two halves,  and then each half edge, called a stub, is rewired randomly with any other stub in the network, even allowing self-loops (which occur when a stub is rewired to another stub from the same node) and multiple-edges between the same two nodes. Thus, even though the node degree distribution of the graph remains intact, the configuration model results in a completely random network.\\n\\n\\n== Expected Number of Edges Between Nodes ==\\nNow consider two nodes \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n  , with node degrees \\n  \\n    \\n      \\n        \\n          k\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{v}}\\n   and \\n  \\n    \\n      \\n        \\n          k\\n          \\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{w}}\\n   respectively, from a randomly rewired network as described above. We calculate the expected number of full edges between these nodes.\", start_char_idx=2225, end_char_idx=6829, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='c950ee50-c6ea-4fb9-b661-85e4af94730d', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='6859ef38-2941-4a08-a1f0-9a3e7a430898', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='735a591b325be2f57da7c8756161c4cb8071593f110f388a3f18fba441884387'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='c1b544e2-2990-468a-8a35-c210a20bf36d', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='5494deb6493cfd07a1eb0dc824819ba3107da6f6b5298e4b1028613576e6c5cc'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='3ddf81a2-6d52-4d6c-b2ac-f294fd654791', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='4be6dbc0ddf0c74a47f8631eaa6f6d3365c9680dbbf3d6fddbe08ef5e5262888')}, hash='f756c9de205234e9d1af3e169669621adecbea315932bc0f970cc3a177c6fd3b', text='== Expected Number of Edges Between Nodes ==\\nNow consider two nodes \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n  , with node degrees \\n  \\n    \\n      \\n        \\n          k\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{v}}\\n   and \\n  \\n    \\n      \\n        \\n          k\\n          \\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{w}}\\n   respectively, from a randomly rewired network as described above. We calculate the expected number of full edges between these nodes.\\nLet us consider each of the \\n  \\n    \\n      \\n        \\n          k\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{v}}\\n   stubs of node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and create associated indicator variables \\n  \\n    \\n      \\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle I_{i}^{(v,w)}}\\n   for them, \\n  \\n    \\n      \\n        i\\n        =\\n        1\\n        ,\\n        …\\n        ,\\n        \\n          k\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle i=1,\\\\ldots ,k_{v}}\\n  , with \\n  \\n    \\n      \\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        =\\n        1\\n      \\n    \\n    {\\\\displaystyle I_{i}^{(v,w)}=1}\\n   if the \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  -th stub happens to connect to one of the \\n  \\n    \\n      \\n        \\n          k\\n          \\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{w}}\\n   stubs of node \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n   in this particular random graph. If it does not, then \\n  \\n    \\n      \\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        =\\n        0\\n      \\n    \\n    {\\\\displaystyle I_{i}^{(v,w)}=0}\\n  .', start_char_idx=6229, end_char_idx=8389, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='3ddf81a2-6d52-4d6c-b2ac-f294fd654791', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='6859ef38-2941-4a08-a1f0-9a3e7a430898', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='735a591b325be2f57da7c8756161c4cb8071593f110f388a3f18fba441884387'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='c950ee50-c6ea-4fb9-b661-85e4af94730d', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='f756c9de205234e9d1af3e169669621adecbea315932bc0f970cc3a177c6fd3b'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='57d7ff03-9dd8-42a5-9494-3319f961ded9', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='98a179372ab837a752b1acd6a877f94262c7e9711e16c99fe69e81e5f62384c0')}, hash='4be6dbc0ddf0c74a47f8631eaa6f6d3365c9680dbbf3d6fddbe08ef5e5262888', text='If it does not, then \\n  \\n    \\n      \\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        =\\n        0\\n      \\n    \\n    {\\\\displaystyle I_{i}^{(v,w)}=0}\\n  . Since the \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  -th stub of node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   can connect to any of the \\n  \\n    \\n      \\n        2\\n        m\\n        −\\n        1\\n      \\n    \\n    {\\\\displaystyle 2m-1}\\n   remaining stubs with equal probability, and since there are \\n  \\n    \\n      \\n        \\n          k\\n          \\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{w}}\\n   stubs it can connect to associated with node \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n  , evidently\\n\\n  \\n    \\n      \\n        p\\n        (\\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        =\\n        1\\n        )\\n        =\\n        E\\n        [\\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        ]\\n        =\\n        \\n          \\n            \\n              k\\n              \\n                w\\n              \\n            \\n            \\n              2\\n              m\\n              −\\n              1\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p(I_{i}^{(v,w)}=1)=E[I_{i}^{(v,w)}]={\\\\frac {k_{w}}{2m-1}}}\\n  The total number of full edges \\n  \\n    \\n      \\n        \\n          J\\n          \\n            v\\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle J_{vw}}\\n   between \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n   is just \\n  \\n    \\n      \\n        \\n          J\\n          \\n            v\\n            w\\n          \\n        \\n        =\\n        \\n          ∑\\n          \\n            i\\n            =\\n            1\\n          \\n          \\n            \\n              k\\n              \\n                v\\n              \\n            \\n          \\n        \\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle J_{vw}=\\\\sum _{i=1}^{k_{v}}I_{i}^{(v,w)}}\\n  , so the expected value of this quantity is\\n\\n  \\n    \\n      \\n        E\\n        [\\n        \\n          J\\n          \\n            v\\n            w\\n          \\n        \\n        ]\\n        =\\n        E\\n        \\n          [\\n          \\n            \\n              ∑\\n              \\n                i\\n                =\\n                1\\n              \\n              \\n                \\n                  k\\n                  \\n                    v\\n                  \\n                \\n              \\n            \\n            \\n              I\\n              \\n                i\\n              \\n              \\n                (\\n                v\\n                ,\\n                w\\n                )\\n              \\n            \\n          \\n          ]\\n        \\n        =\\n        \\n          ∑\\n          \\n            i\\n            =\\n            1\\n          \\n          \\n            \\n              k\\n              \\n                v\\n              \\n            \\n          \\n        \\n        E\\n        [\\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        ]\\n        =\\n        \\n          ∑\\n          \\n            i\\n            =\\n            1\\n          \\n          \\n            \\n              k\\n              \\n                v\\n              \\n            \\n          \\n        \\n        \\n          \\n            \\n              k\\n              \\n                w\\n              \\n            \\n            \\n              2\\n              m\\n              −\\n              1\\n            \\n          \\n        \\n        =\\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n              −\\n              1\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle E[J_{vw}]=E\\\\left[\\\\sum _{i=1}^{k_{v}}I_{i}^{(v,w)}\\\\right]=\\\\sum _{i=1}^{k_{v}}E[I_{i}^{(v,w)}]=\\\\sum _{i=1}^{k_{v}}{\\\\frac {k_{w}}{2m-1}}={\\\\frac {k_{v}k_{w}}{2m-1}}}\\n  Many texts then make the following approximations, for random networks with a large number of edges. When \\n  \\n    \\n      \\n        m\\n      \\n    \\n    {\\\\displaystyle m}\\n   is large, they drop the subtraction of \\n  \\n    \\n      \\n        1\\n      \\n    \\n    {\\\\displaystyle 1}\\n   in the denominator above and simply use the approximate expression \\n  \\n    \\n      \\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle {\\\\frac {k_{v}k_{w}}{2m}}}\\n   for the expected number of edges between two nodes.', start_char_idx=8123, end_char_idx=13501, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='57d7ff03-9dd8-42a5-9494-3319f961ded9', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='6859ef38-2941-4a08-a1f0-9a3e7a430898', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='735a591b325be2f57da7c8756161c4cb8071593f110f388a3f18fba441884387'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='3ddf81a2-6d52-4d6c-b2ac-f294fd654791', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='4be6dbc0ddf0c74a47f8631eaa6f6d3365c9680dbbf3d6fddbe08ef5e5262888'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='bc244995-8d11-4ae1-95ed-512742e3ab91', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='ece86016382d4aec30b71287c270f5b62a668ca9fe9078e31f93230477fe6e50')}, hash='98a179372ab837a752b1acd6a877f94262c7e9711e16c99fe69e81e5f62384c0', text='When \\n  \\n    \\n      \\n        m\\n      \\n    \\n    {\\\\displaystyle m}\\n   is large, they drop the subtraction of \\n  \\n    \\n      \\n        1\\n      \\n    \\n    {\\\\displaystyle 1}\\n   in the denominator above and simply use the approximate expression \\n  \\n    \\n      \\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle {\\\\frac {k_{v}k_{w}}{2m}}}\\n   for the expected number of edges between two nodes. Additionally, in a large random network, the number of self-loops and multi-edges is vanishingly small. Ignoring self-loops and multi-edges allows one to assume that there is at most one edge between any two nodes. In that case, \\n  \\n    \\n      \\n        \\n          J\\n          \\n            v\\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle J_{vw}}\\n   becomes a binary indicator variable, so its expected value is also the probability that it equals \\n  \\n    \\n      \\n        1\\n      \\n    \\n    {\\\\displaystyle 1}\\n  , which means one can approximate the probability of an edge existing between nodes \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n   as \\n  \\n    \\n      \\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle {\\\\frac {k_{v}k_{w}}{2m}}}\\n  .\\n\\n\\n== Modularity ==\\nHence, the difference between the actual number of edges between node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n   and the expected number of edges between them is\\n\\n  \\n    \\n      \\n        \\n          A\\n          \\n            v\\n            w\\n          \\n        \\n        −\\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle A_{vw}-{\\\\frac {k_{v}k_{w}}{2m}}}\\n  \\nSumming over all node pairs gives the equation for modularity, \\n  \\n    \\n      \\n        Q\\n      \\n    \\n    {\\\\displaystyle Q}\\n  .\\n\\nIt is important to note that Eq. 3 holds good for partitioning into two communities only. Hierarchical partitioning (i.e. partitioning into two communities, then the two sub-communities further partitioned into two smaller sub communities only to maximize Q) is a possible approach to identify multiple communities in a network. Additionally, (3) can be generalized for partitioning a network into c communities.\\n\\nwhere eij is the fraction of edges with one end vertices in community i and the other in community j:\\n\\n  \\n    \\n      \\n        \\n          e\\n          \\n            i\\n            j\\n          \\n        \\n        =\\n        \\n          ∑\\n          \\n            v\\n            w\\n          \\n        \\n        \\n          \\n            \\n              A\\n              \\n                v\\n                w\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          1\\n          \\n            v\\n            ∈\\n            \\n              c\\n              \\n                i\\n              \\n            \\n          \\n        \\n        \\n          1\\n          \\n            w\\n            ∈\\n            \\n              c\\n              \\n                j\\n              \\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle e_{ij}=\\\\sum _{vw}{\\\\frac {A_{vw}}{2m}}1_{v\\\\in c_{i}}1_{w\\\\in c_{j}}}\\n  and ai is the fraction of ends of edges that are attached to vertices in community i:\\n\\n  \\n    \\n      \\n        \\n          a\\n          \\n            i\\n          \\n        \\n        =\\n        \\n          \\n            \\n              k\\n              \\n                i\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        =\\n        \\n          ∑\\n          \\n            j\\n          \\n        \\n        \\n          e\\n          \\n            i\\n            j\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle a_{i}={\\\\frac {k_{i}}{2m}}=\\\\sum _{j}e_{ij}}\\n  \\n\\n\\n== Example of multiple community detection ==\\nWe consider an undirected network with 10 nodes and 12 edges and the following adjacency matrix.\\n\\nThe communities in the graph are represented by the red, green and blue node clusters in Fig 1. The optimal community partitions are depicted in Fig 2.\\n\\n\\n== Matrix formulation ==\\nAn alternative formulation of the modularity, useful particularly in spectral optimization algorithms, is as follows.', start_char_idx=12809, end_char_idx=17937, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='bc244995-8d11-4ae1-95ed-512742e3ab91', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='6859ef38-2941-4a08-a1f0-9a3e7a430898', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='735a591b325be2f57da7c8756161c4cb8071593f110f388a3f18fba441884387'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='57d7ff03-9dd8-42a5-9494-3319f961ded9', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='98a179372ab837a752b1acd6a877f94262c7e9711e16c99fe69e81e5f62384c0'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='b3e4d526-d9f5-47ae-a810-5d5b7d1d9e16', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='f8be77d8e35867b288b0f133f29b3bd70f3d248ce3ce792e0d6541b62935b268')}, hash='ece86016382d4aec30b71287c270f5b62a668ca9fe9078e31f93230477fe6e50', text='== Example of multiple community detection ==\\nWe consider an undirected network with 10 nodes and 12 edges and the following adjacency matrix.\\n\\nThe communities in the graph are represented by the red, green and blue node clusters in Fig 1. The optimal community partitions are depicted in Fig 2.\\n\\n\\n== Matrix formulation ==\\nAn alternative formulation of the modularity, useful particularly in spectral optimization algorithms, is as follows.  Define \\n  \\n    \\n      \\n        \\n          S\\n          \\n            v\\n            r\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle S_{vr}}\\n   to be \\n  \\n    \\n      \\n        1\\n      \\n    \\n    {\\\\displaystyle 1}\\n   if vertex \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   belongs to group \\n  \\n    \\n      \\n        r\\n      \\n    \\n    {\\\\displaystyle r}\\n   and \\n  \\n    \\n      \\n        0\\n      \\n    \\n    {\\\\displaystyle 0}\\n   otherwise.  Then\\n\\n  \\n    \\n      \\n        δ\\n        (\\n        \\n          c\\n          \\n            v\\n          \\n        \\n        ,\\n        \\n          c\\n          \\n            w\\n          \\n        \\n        )\\n        =\\n        \\n          ∑\\n          \\n            r\\n          \\n        \\n        \\n          S\\n          \\n            v\\n            r\\n          \\n        \\n        \\n          S\\n          \\n            w\\n            r\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\delta (c_{v},c_{w})=\\\\sum _{r}S_{vr}S_{wr}}\\n  and hence\\n\\n  \\n    \\n      \\n        Q\\n        =\\n        \\n          \\n            1\\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          ∑\\n          \\n            v\\n            w\\n          \\n        \\n        \\n          ∑\\n          \\n            r\\n          \\n        \\n        \\n          [\\n          \\n            \\n              A\\n              \\n                v\\n                w\\n              \\n            \\n            −\\n            \\n              \\n                \\n                  \\n                    k\\n                    \\n                      v\\n                    \\n                  \\n                  \\n                    k\\n                    \\n                      w\\n                    \\n                  \\n                \\n                \\n                  2\\n                  m\\n                \\n              \\n            \\n          \\n          ]\\n        \\n        \\n          S\\n          \\n            v\\n            r\\n          \\n        \\n        \\n          S\\n          \\n            w\\n            r\\n          \\n        \\n        =\\n        \\n          \\n            1\\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          T\\n          r\\n        \\n        (\\n        \\n          \\n            S\\n          \\n          \\n            \\n              T\\n            \\n          \\n        \\n        \\n          B\\n          S\\n        \\n        )\\n        ,\\n      \\n    \\n    {\\\\displaystyle Q={\\\\frac {1}{2m}}\\\\sum _{vw}\\\\sum _{r}\\\\left[A_{vw}-{\\\\frac {k_{v}k_{w}}{2m}}\\\\right]S_{vr}S_{wr}={\\\\frac {1}{2m}}\\\\mathrm {Tr} (\\\\mathbf {S} ^{\\\\mathrm {T} }\\\\mathbf {BS} ),}\\n  where \\n  \\n    \\n      \\n        S\\n      \\n    \\n    {\\\\displaystyle S}\\n   is the (non-square) matrix having elements \\n  \\n    \\n      \\n        \\n          S\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle S_{v}}\\n   and \\n  \\n    \\n      \\n        B\\n      \\n    \\n    {\\\\displaystyle B}\\n   is the so-called modularity matrix, which has elements\\n\\n  \\n    \\n      \\n        \\n          B\\n          \\n            v\\n            w\\n          \\n        \\n        =\\n        \\n          A\\n          \\n            v\\n            w\\n          \\n        \\n        −\\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        .\\n      \\n    \\n    {\\\\displaystyle B_{vw}=A_{vw}-{\\\\frac {k_{v}k_{w}}{2m}}.}\\n  All rows and columns of the modularity matrix sum to zero, which means that the modularity of an undivided network is also always \\n  \\n    \\n      \\n        0\\n      \\n    \\n    {\\\\displaystyle 0}\\n  .', start_char_idx=17497, end_char_idx=21668, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='b3e4d526-d9f5-47ae-a810-5d5b7d1d9e16', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='6859ef38-2941-4a08-a1f0-9a3e7a430898', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='735a591b325be2f57da7c8756161c4cb8071593f110f388a3f18fba441884387'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='bc244995-8d11-4ae1-95ed-512742e3ab91', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='ece86016382d4aec30b71287c270f5b62a668ca9fe9078e31f93230477fe6e50'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='22b05407-47c9-4726-b2a7-c9f7eda56505', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='b9bb7e8c50702da84fd1e9e8267771d8585701c1c2effbdb7caf39eff8ad9a7e')}, hash='f8be77d8e35867b288b0f133f29b3bd70f3d248ce3ce792e0d6541b62935b268', text=\"{\\\\displaystyle B_{vw}=A_{vw}-{\\\\frac {k_{v}k_{w}}{2m}}.}\\n  All rows and columns of the modularity matrix sum to zero, which means that the modularity of an undivided network is also always \\n  \\n    \\n      \\n        0\\n      \\n    \\n    {\\\\displaystyle 0}\\n  .\\nFor networks divided into just two communities, one can alternatively define \\n  \\n    \\n      \\n        \\n          s\\n          \\n            v\\n          \\n        \\n        =\\n        ±\\n        1\\n      \\n    \\n    {\\\\displaystyle s_{v}=\\\\pm 1}\\n   to indicate the community to which node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   belongs, which then leads to\\n\\n  \\n    \\n      \\n        Q\\n        =\\n        \\n          \\n            1\\n            \\n              4\\n              m\\n            \\n          \\n        \\n        \\n          ∑\\n          \\n            v\\n            w\\n          \\n        \\n        \\n          B\\n          \\n            v\\n            w\\n          \\n        \\n        \\n          s\\n          \\n            v\\n          \\n        \\n        \\n          s\\n          \\n            w\\n          \\n        \\n        =\\n        \\n          \\n            1\\n            \\n              4\\n              m\\n            \\n          \\n        \\n        \\n          \\n            s\\n          \\n          \\n            \\n              T\\n            \\n          \\n        \\n        \\n          B\\n          s\\n        \\n        ,\\n      \\n    \\n    {\\\\displaystyle Q={1 \\\\over 4m}\\\\sum _{vw}B_{vw}s_{v}s_{w}={1 \\\\over 4m}\\\\mathbf {s} ^{\\\\mathrm {T} }\\\\mathbf {Bs} ,}\\n  where \\n  \\n    \\n      \\n        s\\n      \\n    \\n    {\\\\displaystyle s}\\n   is the column vector with elements \\n  \\n    \\n      \\n        \\n          s\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle s_{v}}\\n  .This function has the same form as the Hamiltonian of an Ising spin glass, a connection that has been exploited to create simple computer algorithms, for instance using simulated annealing, to maximize the modularity.  The general form of the modularity for arbitrary numbers of communities is equivalent to a Potts spin glass and similar algorithms can be developed for this case also.\\n\\n\\n== Overfitting ==\\nAlthough the method of modularity maximization is motivated by computing a deviation from a null model, this deviation is not computed in a statistically consistent manner. Because of this, the method notoriously finds high-scoring communities in its own null model\\n\\n(the configuration model), which by definition cannot be statistically significant. Because of this, the method cannot be used to reliably obtain statistically significant community structure in empirical networks.\\n\\n\\n== Resolution limit ==\\nModularity compares the number of edges inside a cluster with the expected number of edges that\\none would find in the cluster if the network were a random network with the same number of nodes and where\\neach node keeps its degree, but edges are otherwise randomly attached. This random null model implicitly assumes that each node can get attached to any other node of the network. This assumption is however unreasonable if the network is very large, as the horizon of a node includes a small part of the network, ignoring most of it.\\nMoreover, this implies that the expected number of edges between two groups of nodes decreases if the size of the network increases. So, if a network is large enough, the expected number of edges between two groups of nodes in modularity's null model may be smaller than one. If this happens, a single edge between the two clusters would be interpreted by modularity as a sign of a strong correlation between the two clusters, and optimizing modularity would lead to the merging of the two clusters, independently of the clusters' features. So, even weakly interconnected complete graphs, which have the highest possible density of internal edges, and represent the best identifiable communities, would be merged by modularity optimization if the network were sufficiently large.\\nFor this reason, optimizing modularity in large networks would fail to resolve small communities, even when they are well defined. This bias\\nis inevitable for methods like modularity optimization, which rely on a global null model.\\n\\n\\n== Multiresolution methods ==\\nThere are two main approaches which try to solve the resolution limit within the modularity context: the addition of a resistance r to every node, in the form of a self-loop, which increases (r>0) or decreases (r<0) the aversion of nodes to form communities; or the addition of a parameter γ>0 in front of the null-case term in the definition of modularity, which controls the relative importance between internal links of the communities and the null model. Optimizing modularity for values of these parameters in their respective appropriate ranges, it is possible to recover the whole mesoscale of the network, from the macroscale in which all nodes belong to the same community, to the microscale in which every node forms its own community, hence the name multiresolution methods. However, it has been shown that these methods have limitations when communities are very heterogeneous in size.\", start_char_idx=21417, end_char_idx=26516, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='22b05407-47c9-4726-b2a7-c9f7eda56505', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='6859ef38-2941-4a08-a1f0-9a3e7a430898', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='735a591b325be2f57da7c8756161c4cb8071593f110f388a3f18fba441884387'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='b3e4d526-d9f5-47ae-a810-5d5b7d1d9e16', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='f8be77d8e35867b288b0f133f29b3bd70f3d248ce3ce792e0d6541b62935b268'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='1a33ce25-10ab-4263-b874-ca76f03ee8ad', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='d7c57507d122dc39f93ddb0ac7bc221b89b84a0a8fbddc3bb6c678573a817e7e')}, hash='b9bb7e8c50702da84fd1e9e8267771d8585701c1c2effbdb7caf39eff8ad9a7e', text='== Multiresolution methods ==\\nThere are two main approaches which try to solve the resolution limit within the modularity context: the addition of a resistance r to every node, in the form of a self-loop, which increases (r>0) or decreases (r<0) the aversion of nodes to form communities; or the addition of a parameter γ>0 in front of the null-case term in the definition of modularity, which controls the relative importance between internal links of the communities and the null model. Optimizing modularity for values of these parameters in their respective appropriate ranges, it is possible to recover the whole mesoscale of the network, from the macroscale in which all nodes belong to the same community, to the microscale in which every node forms its own community, hence the name multiresolution methods. However, it has been shown that these methods have limitations when communities are very heterogeneous in size.\\n\\n\\n== Software Tools ==\\nThere are a couple of software tools available that are able to compute clusterings in graphs with good modularity.\\nOriginal implementation of the multi-level Louvain method.The Leiden algorithm which additionally avoids unconnected communities.The Vienna Graph Clustering (VieClus) algorithm, a parallel memetic algorithm.\\n\\n\\n== See also ==\\nComplex network\\nCommunity structure\\nNull model\\nPercolation theory\\n\\n\\n== References ==', start_char_idx=25589, end_char_idx=26965, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='1a33ce25-10ab-4263-b874-ca76f03ee8ad', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='10ce2e27-9666-471f-8d66-5b4af242079c', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='f4605aac6375f764ce1d16f4e8bfc20dbe7891701bb380555f0a8e33de75b1c1'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='22b05407-47c9-4726-b2a7-c9f7eda56505', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='b9bb7e8c50702da84fd1e9e8267771d8585701c1c2effbdb7caf39eff8ad9a7e'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='4293944d-9502-4e9e-bf02-1fe908c8ed8d', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='0f4de134de85b56ebaec21a08dd1fa1d41b62c46f7c837b8a958df6a5bc62e89')}, hash='d7c57507d122dc39f93ddb0ac7bc221b89b84a0a8fbddc3bb6c678573a817e7e', text='In the study of complex networks, a network is said to have community structure if the nodes of the network can be easily grouped into (potentially overlapping) sets of nodes such that each set of nodes is densely connected internally. In the particular case of non-overlapping community finding, this implies that the network divides naturally into groups of nodes with dense connections internally and sparser connections between groups.  But overlapping communities are also allowed. The more general definition is based on the principle that pairs of nodes are more likely to be connected if they are both members of the same community(ies), and less likely to be connected if they do not share communities. A related but different problem is community search, where the goal is to find a community that a certain vertex belongs to.\\n\\n\\n== Properties ==\\nIn the study of networks, such as computer and information networks, social networks and biological networks, a number of different characteristics have been found to occur commonly, including the small-world property, heavy-tailed degree distributions, and clustering, among others. Another common characteristic is community structure.\\nIn the context of networks, community structure refers to the occurrence of groups of nodes in a network that are more densely connected internally than with the rest of the network, as shown in the example image to the right.  This inhomogeneity of connections suggests that the network has certain natural divisions within it.\\nCommunities are often defined in terms of the partition of the set of vertices, that is each node is put into one and only one community, just as in the figure. This is a useful simplification and most community detection methods find this type of community structure. However, in some cases a better representation could be one where vertices are in more than one community.  This might happen in a social network where each vertex represents a person, and the communities represent the different groups of friends: one community for family, another community for co-workers, one for friends in the same sports club, and so on. The use of cliques for community detection discussed below is just one example of how such overlapping community structure can be found.\\nSome networks may not have any meaningful community structure.  Many basic network models, for example, such as the random graph and the Barabási–Albert model, do not display community structure.\\n\\n\\n== Importance ==\\nCommunity structures are quite common in real networks. Social networks include community groups (the origin of the term, in fact) based on common location, interests, occupation, etc.Finding an underlying community structure in a network, if it exists, is important for a number of reasons. Communities allow us to create a large scale map of a network since individual communities act like meta-nodes in the network which makes its study easier.Individual communities also shed light on the function of the system represented by the network since communities often correspond to functional units of the system. In metabolic networks, such functional groups correspond to cycles or pathways whereas in the protein interaction network, communities correspond to proteins with similar functionality inside a biological cell. Similarly, citation networks form communities by research topic. Being able to identify these sub-structures within a network can provide insight into how network function and topology affect each other. Such insight can be useful in improving some algorithms on graphs such as spectral clustering.Importantly, communities often have very different properties than the average properties of the networks. Thus, only concentrating on the average properties usually misses many important and interesting features inside the networks. For example, in a given social network, both gregarious and reticent groups might exists simultaneously.Existence of communities also generally affects various processes like rumour spreading or epidemic spreading happening on a network. Hence to properly understand such processes, it is important to detect communities and also to study how they affect the spreading processes in various settings.\\nFinally, an important application that community detection has found in network science is the prediction of missing links and the identification of false links in the network. During the measurement process, some links may not get observed for a number of reasons. Similarly, some links could falsely enter into the data because of the errors in the measurement. Both these cases are well handled by community detection algorithm since it allows one to assign the probability of existence of an edge between a given pair of nodes.\\n\\n\\n== Algorithms for finding communities ==\\nFinding communities within an arbitrary network can be a computationally difficult task.  The number of communities, if any, within the network is typically unknown and the communities are often of unequal size and/or density.  Despite these difficulties, however, several methods for community finding have been developed and employed with varying levels of success.', start_char_idx=0, end_char_idx=5202, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='4293944d-9502-4e9e-bf02-1fe908c8ed8d', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='10ce2e27-9666-471f-8d66-5b4af242079c', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='f4605aac6375f764ce1d16f4e8bfc20dbe7891701bb380555f0a8e33de75b1c1'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='1a33ce25-10ab-4263-b874-ca76f03ee8ad', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='d7c57507d122dc39f93ddb0ac7bc221b89b84a0a8fbddc3bb6c678573a817e7e'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='891de726-5373-4f22-9e48-f12f983c2e5d', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='35c7edcb9c32ec4b8af6bd897d48325698bea0f370a576964c20d676dc7ffa82')}, hash='0f4de134de85b56ebaec21a08dd1fa1d41b62c46f7c837b8a958df6a5bc62e89', text='== Algorithms for finding communities ==\\nFinding communities within an arbitrary network can be a computationally difficult task.  The number of communities, if any, within the network is typically unknown and the communities are often of unequal size and/or density.  Despite these difficulties, however, several methods for community finding have been developed and employed with varying levels of success.\\n\\n\\n=== Minimum-cut method ===\\nOne of the oldest algorithms for dividing networks into parts is the minimum cut method (and variants such as ratio cut and normalized cut).  This method sees use, for example, in load balancing for parallel computing in order to minimize communication between processor nodes.\\nIn the minimum-cut method, the network is divided into a predetermined number of parts, usually of approximately the same size, chosen such that the number of edges between groups is minimized.  The method works well in many of the applications for which it was originally intended but is less than ideal for finding community structure in general networks since it will find communities regardless of whether they are implicit in the structure, and it will find only a fixed number of them.\\n\\n\\n=== Hierarchical clustering ===\\nAnother method for finding community structures in networks is hierarchical clustering.  In this method one defines a similarity measure quantifying some (usually topological) type of similarity between node pairs.  Commonly used measures include the cosine similarity, the Jaccard index, and the Hamming distance between rows of the adjacency matrix.  Then one groups similar nodes into communities according to this measure.  There are several common schemes for performing the grouping, the two simplest being single-linkage clustering, in which two groups are considered separate communities if and only if all pairs of nodes in different groups have similarity lower than a given threshold, and complete linkage clustering, in which all nodes within every group have similarity greater than a threshold. An important step is how to determine the threshold to stop the agglomerative clustering, indicating a near-to-optimal community structure. A common strategy consist to build one or several metrics monitoring global properties of the network, which peak at given step of the clustering. An interesting approach in this direction is the use of various similarity or dissimilarity measures, combined through convex sums,. Another approximation is the computation of a quantity monitoring the density of edges within clusters with respect to the density between clusters, such as the partition density, which has been proposed when the similarity metric is defined between edges (which permits the definition of overlapping communities), and extended when the similarity is defined between nodes, which allows to consider alternative definitions of communities such as guilds (i.e. groups of nodes sharing a similar number of links with respect to the same neighbours but not necessarily connected themselves). These methods can be extended to consider multidimensional networks, for instance when we are dealing with networks having nodes with different types of links.\\n\\n\\n=== Girvan–Newman algorithm ===\\nAnother commonly used algorithm for finding communities is the Girvan–Newman algorithm.  This algorithm identifies edges in a network that lie between communities and then removes them, leaving behind just the communities themselves.  The identification is performed by employing the graph-theoretic measure betweenness centrality, which assigns a number to each edge which is large if the edge lies \"between\" many pairs of nodes.\\nThe Girvan–Newman algorithm returns results of reasonable quality and is popular because it has been implemented in a number of standard software packages.  But it also runs slowly, taking time O(m2n) on a network of n vertices and m edges, making it impractical for networks of more than a few thousand nodes.\\n\\n\\n=== Modularity maximization ===\\nIn spite of its known drawbacks, one of the most widely used methods for community detection is modularity maximization.  Modularity is a benefit function that measures the quality of a particular division of a network into communities.  The modularity maximization method detects communities by searching over possible divisions of a network for one or more that have particularly high modularity.  Since exhaustive search over all possible divisions is usually intractable, practical algorithms are based on approximate optimization methods such as greedy algorithms, simulated annealing, or spectral optimization, with different approaches offering different balances between speed and accuracy.\\nA popular modularity maximization approach is the Louvain method, which iteratively optimizes local communities until global modularity can no longer be improved given perturbations to the current community state.\\nAn algorithm that utilizes the RenEEL scheme, which is an example of the Extremal Ensemble Learning (EEL) paradigm, is currently the best modularity maximizing algorithm.The usefulness of modularity optimization is questionable, as it has been shown that modularity optimization often fails to detect clusters smaller than some scale, depending on the size of the network (resolution limit); on the other hand the landscape of modularity values is characterized by a huge degeneracy of partitions with high modularity, close to the absolute maximum, which may be very different from each other.', start_char_idx=4794, end_char_idx=10330, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='891de726-5373-4f22-9e48-f12f983c2e5d', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='10ce2e27-9666-471f-8d66-5b4af242079c', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='f4605aac6375f764ce1d16f4e8bfc20dbe7891701bb380555f0a8e33de75b1c1'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='4293944d-9502-4e9e-bf02-1fe908c8ed8d', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='0f4de134de85b56ebaec21a08dd1fa1d41b62c46f7c837b8a958df6a5bc62e89'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='0b9ab24a-11a9-474c-89d9-75fba3587393', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='0d80173700fd137520c5f1dbf53869a397991cef2fec2765adc3e8c101277040')}, hash='35c7edcb9c32ec4b8af6bd897d48325698bea0f370a576964c20d676dc7ffa82', text='=== Statistical inference ===\\nMethods based on statistical inference attempt to fit a generative model to the network data, which encodes the community structure. The overall advantage of this approach compared to the alternatives is its more principled nature, and the capacity to inherently address issues of statistical significance. Most methods in the literature are based on the stochastic block model as well as variants including mixed membership,\\ndegree-correction, and hierarchical structures.Model selection can be performed using principled approaches such as minimum description length (or equivalently, Bayesian model selection) and likelihood-ratio test. Currently many algorithms exist to perform efficient inference of stochastic block models, including belief propagation\\nand agglomerative Monte Carlo.In contrast to approaches that attempt to cluster a network given an objective function, this class of methods is based on generative models, which not only serve as a description of the large-scale structure of the network, but also can be used to generalize the data and predict the occurrence of missing or spurious links in the network.\\n\\n\\n=== Clique-based methods ===\\nCliques are subgraphs in which every node is connected to every other node in the clique.  As nodes can not be more tightly connected than this, it is not surprising that there are many approaches to community detection in networks based on the detection of cliques in a graph and the analysis of how these overlap.  Note that as a node can be a member of more than one clique, a node can be a member of more than one community in these methods giving an \"overlapping community structure\".\\nOne approach is to find the \"maximal cliques\". That is to find the cliques which are not the subgraph of any other clique.  The classic algorithm to find these is the Bron–Kerbosch algorithm. The overlap of these can be used to define communities in several ways.  The simplest is to consider only  maximal cliques bigger than a minimum size (number of nodes).  The union of these cliques then defines a subgraph whose components (disconnected parts) then define communities. Such approaches are often implemented in social network analysis software such as UCInet.\\nThe alternative approach is to use cliques of fixed size \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  . The overlap of these can be used to define a type of \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -regular hypergraph or a structure which is a generalisation of the line graph (the case when \\n  \\n    \\n      \\n        k\\n        =\\n        2\\n      \\n    \\n    {\\\\displaystyle k=2}\\n  ) known as a \"Clique graph\".  The clique graphs have vertices which represent the cliques in the original graph while the edges of the clique graph record the overlap of the clique in the original graph.  Applying any of the previous community detection methods (which assign each node to a community) to the clique graph then assigns each clique to a community.  This can then be used to determine community membership of nodes in the cliques.  Again as a node may be in several cliques, it can be a member of several communities.\\nFor instance the  clique percolation method defines communities as percolation clusters of \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -cliques. To do this it\\nfinds all \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -cliques in a network, that is all the complete sub-graphs of \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -nodes.\\nIt then defines two \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -cliques to be adjacent if they share \\n  \\n    \\n      \\n        k\\n        −\\n        1\\n      \\n    \\n    {\\\\displaystyle k-1}\\n   nodes, that is this is used to define edges in a clique graph. A community is then defined to be the maximal union of \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -cliques in which we can reach any \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -clique from any other \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -clique through series of \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -clique adjacencies. That is communities are just the connected components in the clique graph. Since a node can belong to several different \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -clique percolation clusters at the same time, the communities can overlap with each other.', start_char_idx=10333, end_char_idx=14857, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='0b9ab24a-11a9-474c-89d9-75fba3587393', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='10ce2e27-9666-471f-8d66-5b4af242079c', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='f4605aac6375f764ce1d16f4e8bfc20dbe7891701bb380555f0a8e33de75b1c1'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='891de726-5373-4f22-9e48-f12f983c2e5d', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='35c7edcb9c32ec4b8af6bd897d48325698bea0f370a576964c20d676dc7ffa82'), <NodeRelationship.NEXT: '3'>: RelatedNodeInfo(node_id='2ed941f3-6cd6-4b9a-aba4-2eb46311bbc2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='2eef574898492658036e80cc4b11b2f753835c68f657c6ea4dc016c4646563c4')}, hash='0d80173700fd137520c5f1dbf53869a397991cef2fec2765adc3e8c101277040', text='=== Community detection in latent feature spaces ===\\nA network can be represented or projected onto a latent space via representation learning methods to efficiently represent a system. Then, various clustering methods can be employed to detect community structures. For Euclidean spaces, methods like embedding-based Silhouette community detection can be utilized. For Hypergeometric latent spaces, critical gap method or modified density-based, hierarchical, or partitioning-based clustering methods can be utilized.\\n\\n\\n== Testing methods of finding communities algorithms ==\\nThe evaluation of algorithms, to detect which are better at detecting community structure, is still an open question. It must be based on analyses of networks of known structure. A typical example is the \"four groups\" test, in which a network is divided into four equally-sized groups (usually of 32 nodes each) and the probabilities of connection within and between groups varied to create more or less challenging structures for the detection algorithm. Such benchmark graphs are a special case of the planted l-partition model\\nof Condon and Karp, or more generally of \"stochastic block models\", a general class of random network models containing community structure. Other more flexible benchmarks have been proposed that allow for varying group sizes and nontrivial degree distributions, such as LFR benchmark\\nwhich is an extension of the four groups benchmark that includes heterogeneous distributions of node degree and community size, making it a more severe test of community detection methods.Commonly used computer-generated benchmarks start with a network of well-defined communities. Then, this structure is degraded by rewiring or removing links and it gets harder and harder for the algorithms to detect the original partition. At the end, the network reaches a point where it is essentially random. This kind of benchmark may be called \"open\". The performance on these benchmarks is evaluated by measures such as normalized mutual information or variation of information. They compare the solution obtained by an algorithm  with the original community structure, evaluating the similarity of both partitions.', start_char_idx=14860, end_char_idx=17061, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None),\n",
       " NodeWithScore(node=TextNode(id_='2ed941f3-6cd6-4b9a-aba4-2eb46311bbc2', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='10ce2e27-9666-471f-8d66-5b4af242079c', node_type=<ObjectType.DOCUMENT: '4'>, metadata={}, hash='f4605aac6375f764ce1d16f4e8bfc20dbe7891701bb380555f0a8e33de75b1c1'), <NodeRelationship.PREVIOUS: '2'>: RelatedNodeInfo(node_id='0b9ab24a-11a9-474c-89d9-75fba3587393', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='0d80173700fd137520c5f1dbf53869a397991cef2fec2765adc3e8c101277040')}, hash='2eef574898492658036e80cc4b11b2f753835c68f657c6ea4dc016c4646563c4', text='== Detectability ==\\nDuring recent years, a rather surprising result has been obtained by various groups which shows that a phase transition exists in the community detection problem, showing that as the density of connections inside communities and between communities become more and more equal or both become smaller (equivalently, as the community structure becomes too weak or the network becomes too sparse), suddenly the communities become undetectable. In a sense, the communities themselves still exist, since the presence and absence of edges is still correlated with the community memberships of their endpoints; but it becomes information-theoretically impossible to label the nodes better than chance, or even distinguish the graph from one generated by a null model such as the Erdos–Renyi model without community structure. This transition is independent of the type of algorithm being used to detect communities, implying that there exists a fundamental limit on our ability to detect communities in networks, even with optimal Bayesian inference (i.e., regardless of our computational resources).Consider a stochastic block model with total \\n  \\n    \\n      \\n        n\\n      \\n    \\n    {\\\\displaystyle n}\\n   nodes, \\n  \\n    \\n      \\n        q\\n        =\\n        2\\n      \\n    \\n    {\\\\displaystyle q=2}\\n   groups of equal size, and let \\n  \\n    \\n      \\n        \\n          p\\n          \\n            in\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{in}}}\\n   and \\n  \\n    \\n      \\n        \\n          p\\n          \\n            out\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{out}}}\\n   be the connection probabilities inside and between the groups respectively. If \\n  \\n    \\n      \\n        \\n          p\\n          \\n            in\\n          \\n        \\n        >\\n        \\n          p\\n          \\n            out\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{in}}>p_{\\\\text{out}}}\\n  , the network would possess community structure since the link density inside the groups would be more than the density of links between the groups.  In the sparse case, \\n  \\n    \\n      \\n        \\n          p\\n          \\n            in\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{in}}}\\n   and \\n  \\n    \\n      \\n        \\n          p\\n          \\n            out\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{out}}}\\n   scale as \\n  \\n    \\n      \\n        O\\n        (\\n        1\\n        \\n          /\\n        \\n        n\\n        )\\n      \\n    \\n    {\\\\displaystyle O(1/n)}\\n   so that the average degree is constant:\\n\\n  \\n    \\n      \\n        \\n          p\\n          \\n            in\\n          \\n        \\n        =\\n        \\n          c\\n          \\n            in\\n          \\n        \\n        \\n          /\\n        \\n        n\\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{in}}=c_{\\\\text{in}}/n}\\n   and \\n  \\n    \\n      \\n        \\n          p\\n          \\n            out\\n          \\n        \\n        =\\n        \\n          c\\n          \\n            out\\n          \\n        \\n        \\n          /\\n        \\n        n\\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{out}}=c_{\\\\text{out}}/n}\\n  Then it becomes impossible to detect the communities when:\\n\\n  \\n    \\n      \\n        \\n          c\\n          \\n            in\\n          \\n        \\n        −\\n        \\n          c\\n          \\n            out\\n          \\n        \\n        =\\n        \\n          \\n            2\\n            (\\n            \\n              c\\n              \\n                in\\n              \\n            \\n            +\\n            \\n              c\\n              \\n                out\\n              \\n            \\n            )\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle c_{\\\\text{in}}-c_{\\\\text{out}}={\\\\sqrt {2(c_{\\\\text{in}}+c_{\\\\text{out}})}}}\\n  \\n\\n\\n== See also ==\\nComplex network\\nHierarchy\\nNetwork theory\\nPercolation theory\\n\\n\\n== References ==\\n\\n\\n== External links ==\\nCommunity detection in graphs – an introduction\\nAre there implementations of algorithms for community detection in graphs? – Stack Overflow\\nWhat are the differences between community detection algorithms in igraph? – Stack Overflow', start_char_idx=17064, end_char_idx=21108, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'), score=None)]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response_acc.source_nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Response 1: The Louvain Method for Community Detection | The Louvain method is a greedy optimization method developed by Blondel et al. from the University of Louvain for extracting non-overlapping communities from large networks. The method aims to optimize modularity, a measure of the relative density of edges within communities compared to edges between communities. By iteratively optimizing modularity locally on all nodes and grouping small communities into one node, the Louvain method produces the best possible grouping of nodes in a network. The method has a time complexity of O(n*log(n)), where n is the number of nodes in the network.\\n---------------------\\nResponse 2: Algorithm for Optimizing Modularity in Networks | This text explains the algorithm for optimizing modularity in networks. Modularity is a measure of the density of links within communities compared to links between communities. The algorithm uses the Louvain Method, which has two phases. In the first phase, each node is assigned to its own community. In the second phase, the change in modularity is calculated for moving a node from its own community to the community of each of its neighbors. This process is repeated iteratively to maximize modularity efficiently.\\n---------------------\\nResponse 3: Modularity Optimization Algorithm for Community Detection | The text describes a modularity optimization algorithm for community detection in a network. The algorithm assigns each node to its own community and then calculates the change in modularity for moving the node into the community of each of its neighbors. The algorithm uses equations to calculate the modularity change and places the node into the community that results in the greatest increase in modularity. This process is repeated for all nodes until no further increase in modularity is possible. The algorithm aims to find the local maximum of modularity, indicating the optimal community structure in the network.\\n---------------------\\nResponse 4: Louvain Method for Community Detection in Large Networks | The Louvain method is an algorithm used for community detection in large networks. It involves two phases: in the first phase, nodes are iteratively moved to communities that result in the greatest increase in modularity. In the second phase, the nodes are grouped into communities and a new network is created. The Louvain method has been used in various applications, such as partitioning online social networks and identifying dynamic communities in mobile phone networks. However, it has limitations, such as producing only non-overlapping communities and sometimes creating poorly connected communities. Nonetheless, it has been shown to outperform other modularity optimization methods in terms of both speed and resulting modularity value.\\n---------------------\\nResponse 5: The Importance of Modularity in Network Analysis | Modularity is a measure of the structure of networks that determines the strength of division into modules or communities. It is commonly used in optimization methods for detecting community structure in networks. However, modularity maximization is not statistically consistent and cannot be used to find statistically significant community structures in real-world networks. Additionally, modularity suffers from a resolution limit and is unable to detect small communities. Despite these limitations, modularity is still an important measure in network analysis as it helps reveal unexpected structural features and understand the dynamics of various networks.\\n---------------------\\nResponse 6: Modularity in Graph Theory | Modularity is a measure of the concentration of edges within groups compared to a random distribution of links between all nodes. It is calculated by subtracting the expected fraction of edges within groups from the actual fraction. The value of modularity for unweighted and undirected graphs ranges from -1/2 to 1, with positive values indicating a higher number of edges within groups than expected by chance. Different methods can be used to calculate modularity, with the most common version preserving the degree of each vertex. The expected number of edges is computed using a configuration model, which randomizes the network while maintaining the node degree distribution.\\n---------------------\\nResponse 7: Expected Number of Edges Between Nodes | This text discusses the calculation of the expected number of edges between two nodes in a randomly rewired network. It explains the use of indicator variables to determine if a stub of one node connects to a stub of the other node. The indicator variables are set to 1 if a connection is made and 0 if not. The text provides a mathematical approach to calculating the expected number of full edges between the nodes.\\n---------------------\\nResponse 8: Approximations for Expected Number of Edges in Random Networks | This text discusses the approximations made for the expected number of edges between two nodes in random networks with a large number of edges. It explains that when the number of edges is large, the subtraction of 1 in the denominator is dropped, and the approximate expression k_v * k_w / 2m is used instead.\\n---------------------\\nResponse 9: Modularity and Community Detection in Networks | This text explains the concept of modularity in network analysis and its application in community detection. It discusses the approximation of the expected number of edges between nodes in a large random network and how it can be used to calculate the probability of an edge existing between two nodes. The text also mentions the equation for modularity and its use in partitioning a network into communities. It concludes with an example of multiple community detection and an alternative formulation of modularity.\\n---------------------\\nResponse 10: Matrix Formulation of Modularity for Community Detection | This text explains the matrix formulation of modularity, which is a measure used in community detection algorithms. The modularity matrix is defined using the adjacency matrix of a network and represents the optimal community partitions. The formula for modularity involves the sum of elements in the modularity matrix and the transpose of the matrix. The modularity of an undivided network is always zero.\\n---------------------\\nResponse 11: Limitations of Modularity Maximization for Community Detection | Modularity maximization, a method commonly used for community detection in networks, has several limitations. Firstly, it is prone to overfitting and often finds high-scoring communities in its own null model, which are not statistically significant. Secondly, it suffers from a resolution limit, where it fails to detect small communities in large networks. This is because the null model assumes that each node can be connected to any other node, which is unrealistic for large networks. To address these limitations, multiresolution methods have been proposed, but they also have limitations when communities are highly heterogeneous in size.\\n---------------------\\nResponse 12: Multiresolution Methods for Community Detection in Networks | This text discusses two main approaches for solving the resolution limit in community detection within the modularity context. One approach involves adding a resistance parameter to each node, either increasing or decreasing their aversion to form communities. The other approach involves adding a parameter that controls the relative importance between internal links of the communities and the null model. These multiresolution methods allow for the recovery of the entire mesoscale of a network, from a macro-scale where all nodes belong to the same community, to a micro-scale where each node forms its own community. However, these methods have limitations when communities vary greatly in size. The text also mentions several software tools available for computing clusterings with good modularity.\\n---------------------\\nResponse 13: The Importance of Community Structure in Networks | Community structure refers to the occurrence of groups of nodes in a network that are more densely connected internally than with the rest of the network. This structure is common in various types of networks, including social networks, biological networks, and information networks. Identifying community structure is important for understanding network function and topology, as communities often correspond to functional units of the system. Additionally, communities can affect processes such as rumor spreading and epidemic spreading on a network. Various algorithms have been developed to find communities within networks, despite the computational challenges posed by the unknown number and unequal size/density of communities.\\n---------------------\\nResponse 14: Algorithms for Finding Communities in Networks | This text discusses various algorithms for finding communities within networks. The minimum-cut method divides the network into parts by minimizing the number of edges between groups. Hierarchical clustering groups similar nodes into communities based on a similarity measure. The Girvan-Newman algorithm identifies edges between communities and removes them. Modularity maximization searches for divisions of the network with high modularity, with the Louvain method being a popular approach. The RenEEL scheme is currently the best modularity maximizing algorithm. However, modularity optimization has limitations in detecting small clusters and has a degeneracy of partitions with high modularity.\\n---------------------\\nResponse 15: Methods for Community Detection in Networks | This text discusses two approaches for community detection in networks: statistical inference and clique-based methods. Statistical inference involves fitting a generative model to the network data, allowing for the identification of community structures and addressing issues of statistical significance. Clique-based methods, on the other hand, focus on detecting cliques in the graph and analyzing their overlap to define communities. These methods can be implemented using algorithms such as the Bron-Kerbosch algorithm or by creating clique graphs. Both approaches offer different ways to identify communities in networks and can be used to predict missing or spurious links.\\n---------------------\\nResponse 16: Testing Methods for Community Detection Algorithms | This text discusses the evaluation of algorithms for detecting community structures in networks. It highlights the importance of analyzing networks of known structure to determine the effectiveness of different algorithms. The \"four groups\" test is commonly used, where a network is divided into four equally-sized groups and the probabilities of connection within and between groups are varied. Other benchmarks, such as the LFR benchmark, allow for varying group sizes and nontrivial degree distributions. These benchmarks help assess the performance of algorithms by comparing the obtained solution with the original community structure using measures like normalized mutual information or variation of information.\\n---------------------\\nResponse 17: Detectability of Communities in Networks | This text discusses the concept of detectability in community detection problems in networks. It explains that there is a phase transition in detectability, where communities become undetectable as the density of connections within and between communities becomes more equal or smaller. While the communities still exist, it becomes impossible to label nodes or distinguish the graph from a null model without community structure. This transition is independent of the algorithm used for community detection, suggesting a fundamental limit on our ability to detect communities in networks. The text also mentions the conditions under which communities become undetectable in a stochastic block model.'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(f'{response_acc}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(index_summary._index_struct.nodes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(response_acc.source_nodes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "Response 1: The Louvain Method for Community Detection | The Louvain method is a greedy optimization method developed by Blondel et al. from the University of Louvain for extracting non-overlapping communities from large networks. The method aims to optimize modularity, a measure of the relative density of edges within communities compared to edges between communities. By iteratively optimizing modularity locally on all nodes and grouping small communities into one node, the Louvain method produces the best possible grouping of nodes in a network. The method has a time complexity of O(n*log(n)), where n is the number of nodes in the network.\n",
       "---------------------\n",
       "Response 2: Algorithm for Optimizing Modularity in Networks | This text explains the algorithm for optimizing modularity in networks. Modularity is a measure of the density of links within communities compared to links between communities. The algorithm uses the Louvain Method, which has two phases. In the first phase, each node is assigned to its own community. In the second phase, the change in modularity is calculated for moving a node from its own community to the community of each of its neighbors. This process is repeated iteratively to maximize modularity efficiently.\n",
       "---------------------\n",
       "Response 3: Modularity Optimization Algorithm for Community Detection | The text describes a modularity optimization algorithm for community detection in a network. The algorithm assigns each node to its own community and then calculates the change in modularity for moving the node into the community of each of its neighbors. The algorithm uses equations to calculate the modularity change and places the node into the community that results in the greatest increase in modularity. This process is repeated for all nodes until no further increase in modularity is possible. The algorithm aims to find the local maximum of modularity, indicating the optimal community structure in the network.\n",
       "---------------------\n",
       "Response 4: Louvain Method for Community Detection in Large Networks | The Louvain method is an algorithm used for community detection in large networks. It involves two phases: in the first phase, nodes are iteratively moved to communities that result in the greatest increase in modularity. In the second phase, the nodes are grouped into communities and a new network is created. The Louvain method has been used in various applications, such as partitioning online social networks and identifying dynamic communities in mobile phone networks. However, it has limitations, such as producing only non-overlapping communities and sometimes creating poorly connected communities. Nonetheless, it has been shown to outperform other modularity optimization methods in terms of both speed and resulting modularity value.\n",
       "---------------------\n",
       "Response 5: The Importance of Modularity in Network Analysis | Modularity is a measure of the structure of networks that determines the strength of division into modules or communities. It is commonly used in optimization methods for detecting community structure in networks. However, modularity maximization is not statistically consistent and cannot be used to find statistically significant community structures in real-world networks. Additionally, modularity suffers from a resolution limit and is unable to detect small communities. Despite these limitations, modularity is still an important measure in network analysis as it helps reveal unexpected structural features and understand the dynamics of various networks.\n",
       "---------------------\n",
       "Response 6: Modularity in Graph Theory | Modularity is a measure of the concentration of edges within groups compared to a random distribution of links between all nodes. It is calculated by subtracting the expected fraction of edges within groups from the actual fraction. The value of modularity for unweighted and undirected graphs ranges from -1/2 to 1, with positive values indicating a higher number of edges within groups than expected by chance. Different methods can be used to calculate modularity, with the most common version preserving the degree of each vertex. The expected number of edges is computed using a configuration model, which randomizes the network while maintaining the node degree distribution.\n",
       "---------------------\n",
       "Response 7: Expected Number of Edges Between Nodes | This text discusses the calculation of the expected number of edges between two nodes in a randomly rewired network. It explains the use of indicator variables to determine if a stub of one node connects to a stub of the other node. The indicator variables are set to 1 if a connection is made and 0 if not. The text provides a mathematical approach to calculating the expected number of full edges between the nodes.\n",
       "---------------------\n",
       "Response 8: Approximations for Expected Number of Edges in Random Networks | This text discusses the approximations made for the expected number of edges between two nodes in random networks with a large number of edges. It explains that when the number of edges is large, the subtraction of 1 in the denominator is dropped, and the approximate expression k_v * k_w / 2m is used instead.\n",
       "---------------------\n",
       "Response 9: Modularity and Community Detection in Networks | This text explains the concept of modularity in network analysis and its application in community detection. It discusses the approximation of the expected number of edges between nodes in a large random network and how it can be used to calculate the probability of an edge existing between two nodes. The text also mentions the equation for modularity and its use in partitioning a network into communities. It concludes with an example of multiple community detection and an alternative formulation of modularity.\n",
       "---------------------\n",
       "Response 10: Matrix Formulation of Modularity for Community Detection | This text explains the matrix formulation of modularity, which is a measure used in community detection algorithms. The modularity matrix is defined using the adjacency matrix of a network and represents the optimal community partitions. The formula for modularity involves the sum of elements in the modularity matrix and the transpose of the matrix. The modularity of an undivided network is always zero.\n",
       "---------------------\n",
       "Response 11: Limitations of Modularity Maximization for Community Detection | Modularity maximization, a method commonly used for community detection in networks, has several limitations. Firstly, it is prone to overfitting and often finds high-scoring communities in its own null model, which are not statistically significant. Secondly, it suffers from a resolution limit, where it fails to detect small communities in large networks. This is because the null model assumes that each node can be connected to any other node, which is unrealistic for large networks. To address these limitations, multiresolution methods have been proposed, but they also have limitations when communities are highly heterogeneous in size.\n",
       "---------------------\n",
       "Response 12: Multiresolution Methods for Community Detection in Networks | This text discusses two main approaches for solving the resolution limit in community detection within the modularity context. One approach involves adding a resistance parameter to each node, either increasing or decreasing their aversion to form communities. The other approach involves adding a parameter that controls the relative importance between internal links of the communities and the null model. These multiresolution methods allow for the recovery of the entire mesoscale of a network, from a macro-scale where all nodes belong to the same community, to a micro-scale where each node forms its own community. However, these methods have limitations when communities vary greatly in size. The text also mentions several software tools available for computing clusterings with good modularity.\n",
       "---------------------\n",
       "Response 13: The Importance of Community Structure in Networks | Community structure refers to the occurrence of groups of nodes in a network that are more densely connected internally than with the rest of the network. This structure is common in various types of networks, including social networks, biological networks, and information networks. Identifying community structure is important for understanding network function and topology, as communities often correspond to functional units of the system. Additionally, communities can affect processes such as rumor spreading and epidemic spreading on a network. Various algorithms have been developed to find communities within networks, despite the computational challenges posed by the unknown number and unequal size/density of communities.\n",
       "---------------------\n",
       "Response 14: Algorithms for Finding Communities in Networks | This text discusses various algorithms for finding communities within networks. The minimum-cut method divides the network into parts by minimizing the number of edges between groups. Hierarchical clustering groups similar nodes into communities based on a similarity measure. The Girvan-Newman algorithm identifies edges between communities and removes them. Modularity maximization searches for divisions of the network with high modularity, with the Louvain method being a popular approach. The RenEEL scheme is currently the best modularity maximizing algorithm. However, modularity optimization has limitations in detecting small clusters and has a degeneracy of partitions with high modularity.\n",
       "---------------------\n",
       "Response 15: Methods for Community Detection in Networks | This text discusses two approaches for community detection in networks: statistical inference and clique-based methods. Statistical inference involves fitting a generative model to the network data, allowing for the identification of community structures and addressing issues of statistical significance. Clique-based methods, on the other hand, focus on detecting cliques in the graph and analyzing their overlap to define communities. These methods can be implemented using algorithms such as the Bron-Kerbosch algorithm or by creating clique graphs. Both approaches offer different ways to identify communities in networks and can be used to predict missing or spurious links.\n",
       "---------------------\n",
       "Response 16: Testing Methods for Community Detection Algorithms | This text discusses the evaluation of algorithms for detecting community structures in networks. It highlights the importance of analyzing networks of known structure to determine the effectiveness of different algorithms. The \"four groups\" test is commonly used, where a network is divided into four equally-sized groups and the probabilities of connection within and between groups are varied. Other benchmarks, such as the LFR benchmark, allow for varying group sizes and nontrivial degree distributions. These benchmarks help assess the performance of algorithms by comparing the obtained solution with the original community structure using measures like normalized mutual information or variation of information.\n",
       "---------------------\n",
       "Response 17: Detectability of Communities in Networks | This text discusses the concept of detectability in community detection problems in networks. It explains that there is a phase transition in detectability, where communities become undetectable as the density of connections within and between communities becomes more equal or smaller. While the communities still exist, it becomes impossible to label nodes or distinguish the graph from a null model without community structure. This transition is independent of the algorithm used for community detection, suggesting a fundamental limit on our ability to detect communities in networks. The text also mentions the conditions under which communities become undetectable in a stochastic block model."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(Markdown(f'{response_acc}'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "Response 1: The speaker begins by addressing various individuals, including members of Congress, the Cabinet, military leaders, and the Supreme Court. They congratulate the new Speaker of the House, Kevin McCarthy, and acknowledge other political leaders. The speaker then highlights the progress and resilience of America, stating that the country has emerged stronger from every crisis. They mention the economic growth, job creation, and the handling of the COVID-19 pandemic. The speaker emphasizes the importance of unity and bipartisanship, citing examples of Democrats and Republicans working together on various issues. They express their vision to restore the soul of the nation, rebuild the middle class, and create an economy that benefits everyone. The speaker highlights the achievements in job creation, low unemployment rates, and the revitalization of the manufacturing sector. They conclude by emphasizing the importance of exporting American products and creating American jobs.\n",
       "---------------------\n",
       "Response 2: The context discusses various topics such as the importance of a strong middle class, job creation, low unemployment rates, inflation, small businesses, the semiconductor industry, and infrastructure. It highlights the need to prioritize American manufacturing, particularly in the semiconductor sector, to avoid supply chain disruptions. The passage also mentions the bipartisan CHIPS and Science Act, which aims to strengthen the domestic supply chain for semiconductors. Additionally, it emphasizes the significance of investing in infrastructure to rebuild highways, bridges, railroads, tunnels, ports, airports, and improve access to clean water and high-speed internet across America.\n",
       "---------------------\n",
       "Response 3: The speaker discusses various infrastructure projects that have been funded, including the rebuilding of highways, bridges, railroads, tunnels, ports, and airports across America. They express gratitude to Republican friends who voted for a law and assure those who voted against it that their projects will still be funded. The speaker also mentions specific projects, such as the Brent Spence bridge, and emphasizes the importance of unity and pride in the country. They highlight efforts to replace lead pipes, provide access to affordable high-speed internet, and enforce the Buy American policy for federal infrastructure projects. The speaker acknowledges the struggles faced by many Americans and promises to invest in places and people that have been forgotten. They mention the Inflation Reduction Act, which aims to lower healthcare costs, particularly for prescription drugs like insulin. The law caps the cost of insulin at $35 a month for seniors on Medicare and proposes extending this cap to all Americans who need it. Additionally, the law sets a maximum out-of-pocket drug cost for seniors on Medicare and requires drug companies to pay Medicare back if prices rise faster than inflation.\n",
       "---------------------\n",
       "Response 4: The speaker discusses the issue of high insulin costs and proposes capping the cost at $35 a month for all Americans who need it. They also mention capping out-of-pocket drug costs for seniors on Medicare and giving Medicare the power to negotiate drug prices. The speaker emphasizes the importance of bringing down prescription drug costs to save seniors money and reduce the federal deficit. They express their commitment to veto any attempts to raise the cost of prescription drugs. The speaker also highlights the achievements of the Affordable Care Act in increasing health insurance coverage and saving people money on premiums. They call for making these savings permanent and expanding coverage to those not covered by Medicaid. The speaker addresses the climate crisis and the need for significant investments to tackle it, including building electric vehicle charging stations and promoting clean energy. They stress the urgency of confronting the climate crisis for the sake of future generations. The speaker advocates for making the wealthiest and biggest corporations pay their fair share of taxes and criticizes the current tax system for allowing billion-dollar companies to pay zero federal income taxes. They highlight the implementation of a minimum tax for billion-dollar companies and emphasize that individuals earning less than $400,000 a year will not see an increase in taxes. The speaker also discusses the need to address the actions of Big Oil, such as quadrupling the tax on corporate stock buybacks and closing loopholes that allow the wealthy to avoid paying taxes.\n",
       "---------------------\n",
       "Response 5: The speaker in the given context is proposing various measures to address economic inequality and ensure that the wealthy and corporations pay their fair share of taxes. They advocate for a billionaire minimum tax, higher taxes on corporate stock buybacks, closing tax loopholes for the wealthy, and cracking down on tax cheats. The speaker also emphasizes the importance of protecting Social Security and Medicare benefits for seniors and opposes any cuts to these programs. They express a commitment to reducing the deficit, extending the Medicare Trust Fund, and not raising taxes on those making under $400,000 a year. Additionally, the speaker highlights efforts to protect consumers from exploitation, promote competition, and eliminate hidden fees in various industries.\n",
       "---------------------\n",
       "Response 6: The speaker discusses various measures aimed at protecting consumers and workers from unfair practices. They mention cracking down on nursing homes that commit fraud, allowing over-the-counter hearing aids, strengthening antitrust enforcement, and addressing hidden fees charged by businesses. The speaker also emphasizes the importance of restoring the dignity of work by banning non-compete agreements, supporting workers' rights to organize, and ensuring a living wage. Additionally, they highlight the need for affordable housing, home care services for seniors and people with disabilities, and accessible education, including preschool for 3- and 4-year-olds.\n",
       "---------------------\n",
       "Response 7: The speaker discusses the importance of restoring the dignity of work and making education affordable. They highlight the need for access to preschool, raising teacher salaries, reducing student debt, and providing career opportunities for students. The speaker also acknowledges the progress made in fighting the COVID-19 pandemic but emphasizes the need to remain vigilant. They address the issue of fraud related to relief money and express the intention to prosecute those responsible. The speaker also mentions the increase in violent crime and the importance of public safety and trust in law enforcement. They share the story of Tyre Nichols, a victim of police violence, and emphasize the need for equal protection under the law.\n",
       "---------------------\n",
       "Response 8: The speaker addresses various issues in their speech, including the need for safer neighborhoods, trust between law enforcement and communities, support for police officers, addressing mental health and substance abuse challenges, reducing violent crime and gun crime, holding law enforcement accountable, passing gun safety laws, banning assault weapons, addressing immigration, protecting reproductive rights, and ensuring equality for LGBTQ Americans. The speaker emphasizes the importance of taking action and working together to make positive changes.\n",
       "---------------------\n",
       "Response 9: The speaker discusses various topics in their speech, including pathways to citizenship for certain groups, protecting rights and freedoms, the need to codify Roe v. Wade, the importance of access to reproductive health care, the push for the Equality Act, and the need to stand against aggression and defend democracy. They also mention the United States' support for Ukraine, the country's approach to China, the strengthening of democracies, and the progress made in areas such as opioid addiction treatment and gun safety. The speaker shares a personal story about a father who lost his daughter to addiction.\n",
       "---------------------\n",
       "Response 10: The speaker highlights the progress made in passing laws related to opioid addiction, gun safety, and healthcare. They also mention the launch of ARPA-H to drive breakthroughs in fighting diseases like cancer and Alzheimer's. The speaker shares a personal story about a family affected by addiction and emphasizes the need to address the fentanyl crisis and improve mental health care, especially for children. They also discuss the importance of supporting veterans and their families and re-igniting efforts to combat cancer. The speaker concludes by emphasizing the importance of democracy in achieving these goals.\n",
       "---------------------\n",
       "Response 11: The speaker emphasizes the importance of hope and the resilience of the American people. They highlight the success of past efforts in fighting diseases like HIV/AIDS and express the belief that cancer can also be conquered. The speaker acknowledges the threats faced by democracy in recent years, including the attack on the Capitol, and calls for unity and the protection of democratic values. They stress the need to reject hate and extremism and emphasize that democracy should be a non-partisan issue. The speaker sees the current moment as a critical one that will shape the future of the nation and the world. They express optimism about the state of the union and the potential of the United States when its people come together. The speech concludes with a blessing for the audience and a wish for the protection of the military."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(Markdown(f'{response_acc}'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Downloading Wikipedia articles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_index import download_loader\n",
    "\n",
    "WikipediaReader = download_loader(\"WikipediaReader\")\n",
    "\n",
    "loader = WikipediaReader()\n",
    "documents = loader.load_data(pages=['Louvain Method', 'Modularity (networks)', 'Community structure'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(id_='32ee0945-5f2a-41e2-bbdf-0dd1ddc4fab6', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={}, hash='e2d4fb35246a6751413cd39467fc1541f5fc31f2cffb06e9baa10056d23cee22', text='The Louvain method for community detection is a method to extract non-overlapping communities from large networks created by Blondel et al. from the University of Louvain (the source of this method\\'s name). The method is a greedy optimization method that appears to run in time \\n  \\n    \\n      \\n        O\\n        (\\n        n\\n        ⋅\\n        log\\n        \\u2061\\n        n\\n        )\\n      \\n    \\n    {\\\\displaystyle O(n\\\\cdot \\\\log n)}\\n   where \\n  \\n    \\n      \\n        n\\n      \\n    \\n    {\\\\displaystyle n}\\n   is the number of nodes in the network.\\n\\n\\n== Modularity optimization ==\\nThe inspiration for this method of community detection is the optimization of modularity as the algorithm progresses. Modularity is a scale value between −0.5 (non-modular clustering) and 1 (fully modular clustering) that measures the relative density of edges inside communities with respect to edges outside communities. Optimizing this value theoretically results in the best possible grouping of the nodes of a given network. But because going through all possible iterations of the nodes into groups is impractical, heuristic algorithms are used.\\nIn the Louvain Method of community detection, first small communities are found by optimizing modularity locally on all nodes, then each small community is grouped into one node and the first step is repeated. The method is similar to the earlier method by Clauset, Newman and Moore that connects communities whose amalgamation produces the largest increase in modularity.\\n\\n\\n== Algorithm ==\\nThe value to be optimized is modularity, defined as a value in the range \\n  \\n    \\n      \\n        [\\n        −\\n        1\\n        \\n          /\\n        \\n        2\\n        ,\\n        1\\n        ]\\n      \\n    \\n    {\\\\displaystyle [-1/2,1]}\\n   that measures the density of links inside communities compared to links between communities. For a weighted graph, modularity is defined as:\\n\\n  \\n    \\n      \\n        Q\\n        =\\n        \\n          \\n            1\\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          ∑\\n          \\n            i\\n            j\\n          \\n        \\n        \\n          \\n            [\\n          \\n        \\n        \\n          A\\n          \\n            i\\n            j\\n          \\n        \\n        −\\n        \\n          \\n            \\n              \\n                k\\n                \\n                  i\\n                \\n              \\n              \\n                k\\n                \\n                  j\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          \\n            ]\\n          \\n        \\n        δ\\n        (\\n        \\n          c\\n          \\n            i\\n          \\n        \\n        ,\\n        \\n          c\\n          \\n            j\\n          \\n        \\n        )\\n        ,\\n      \\n    \\n    {\\\\displaystyle Q={\\\\frac {1}{2m}}\\\\sum \\\\limits _{ij}{\\\\bigg [}A_{ij}-{\\\\frac {k_{i}k_{j}}{2m}}{\\\\bigg ]}\\\\delta (c_{i},c_{j}),}\\n  \\nwhere\\n\\n  \\n    \\n      \\n        \\n          A\\n          \\n            i\\n            j\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle A_{ij}}\\n   represents the edge weight between nodes \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   and \\n  \\n    \\n      \\n        j\\n      \\n    \\n    {\\\\displaystyle j}\\n  ;\\n\\n  \\n    \\n      \\n        \\n          k\\n          \\n            i\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{i}}\\n   and \\n  \\n    \\n      \\n        \\n          k\\n          \\n            j\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{j}}\\n   are the sum of the weights of the edges attached to nodes \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   and \\n  \\n    \\n      \\n        j\\n      \\n    \\n    {\\\\displaystyle j}\\n  , respectively;\\n\\n  \\n    \\n      \\n        m\\n      \\n    \\n    {\\\\displaystyle m}\\n   is the sum of all of the edge weights in the graph;\\n\\n  \\n    \\n      \\n        \\n          c\\n          \\n            i\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle c_{i}}\\n   and \\n  \\n    \\n      \\n        \\n          c\\n          \\n            j\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle c_{j}}\\n   are the communities of the nodes; and\\n\\n  \\n    \\n      \\n        δ\\n      \\n    \\n    {\\\\displaystyle \\\\delta }\\n   is Kronecker delta function (\\n  \\n    \\n      \\n        δ\\n        (\\n        x\\n        ,\\n        y\\n        )\\n        =\\n        1\\n      \\n    \\n    {\\\\displaystyle \\\\delta (x,y)=1}\\n   if \\n  \\n    \\n      \\n        x\\n        =\\n        y\\n      \\n    \\n    {\\\\displaystyle x=y}\\n  , \\n  \\n    \\n      \\n        0\\n      \\n    \\n    {\\\\displaystyle 0}\\n   otherwise).Based on the above equation, the modularity of a community \\n  \\n    \\n      \\n        c\\n      \\n    \\n    {\\\\displaystyle c}\\n   can be calculated as:\\n\\n  \\n    \\n      \\n        \\n          Q\\n          \\n            c\\n          \\n        \\n        =\\n        \\n          \\n            \\n              Σ\\n              \\n                i\\n                n\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        −\\n        (\\n        \\n          \\n            \\n              Σ\\n              \\n                t\\n                o\\n                t\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          )\\n          \\n            2\\n          \\n        \\n        ,\\n      \\n    \\n    {\\\\displaystyle Q_{c}={\\\\frac {\\\\Sigma _{in}}{2m}}-({\\\\frac {\\\\Sigma _{tot}}{2m}})^{2},}\\n  \\nwhere\\n\\n  \\n    \\n      \\n        \\n          Σ\\n          \\n            i\\n            n\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\Sigma _{in}}\\n   is the sum of edge weights between nodes within the community \\n  \\n    \\n      \\n        c\\n      \\n    \\n    {\\\\displaystyle c}\\n   (each edge is considered twice); and\\n\\n  \\n    \\n      \\n        \\n          Σ\\n          \\n            t\\n            o\\n            t\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\Sigma _{tot}}\\n   is the sum of all edge weights for nodes within the community (including edges which link to other communities).In order to maximize modularity efficiently, the Louvain Method has two phases that are repeated iteratively.\\nFirst, each node in the network is assigned to its own community. Then for each node \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  , the change in modularity is calculated for removing \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   from its own community and moving it into the community of each neighbor \\n  \\n    \\n      \\n        j\\n      \\n    \\n    {\\\\displaystyle j}\\n   of \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  . This value is easily calculated by two steps: (1) removing \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   from its original community, and (2) inserting \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   to the community of \\n  \\n    \\n      \\n        j\\n      \\n    \\n    {\\\\displaystyle j}\\n  . The two equations are quite similar, and the equation for step (2) is:\\n  \\n    \\n      \\n        Δ\\n        Q\\n        =\\n        \\n          \\n            [\\n          \\n        \\n        \\n          \\n            \\n              \\n                Σ\\n                \\n                  i\\n                  n\\n                \\n              \\n              +\\n              2\\n              \\n                k\\n                \\n                  i\\n                  ,\\n                  i\\n                  n\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        −\\n        \\n          \\n            (\\n          \\n        \\n        \\n          \\n            \\n              \\n                Σ\\n                \\n                  t\\n                  o\\n                  t\\n                \\n              \\n              +\\n              \\n                k\\n                \\n                  i\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          \\n            \\n              )\\n            \\n          \\n          \\n            2\\n          \\n        \\n        \\n          \\n            ]\\n          \\n        \\n        −\\n        \\n          \\n            [\\n          \\n        \\n        \\n          \\n            \\n              Σ\\n              \\n                i\\n                n\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        −\\n        \\n          \\n            (\\n          \\n        \\n        \\n          \\n            \\n              Σ\\n              \\n                t\\n                o\\n                t\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          \\n            \\n              )\\n            \\n          \\n          \\n            2\\n          \\n        \\n        −\\n        \\n          \\n            (\\n          \\n        \\n        \\n          \\n            \\n              k\\n              \\n                i\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          \\n            \\n              )\\n            \\n          \\n          \\n            2\\n          \\n        \\n        \\n          \\n            ]\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\Delta Q={\\\\bigg [}{\\\\frac {\\\\Sigma _{in}+2k_{i,in}}{2m}}-{\\\\bigg (}{\\\\frac {\\\\Sigma _{tot}+k_{i}}{2m}}{\\\\bigg )}^{2}{\\\\bigg ]}-{\\\\bigg [}{\\\\frac {\\\\Sigma _{in}}{2m}}-{\\\\bigg (}{\\\\frac {\\\\Sigma _{tot}}{2m}}{\\\\bigg )}^{2}-{\\\\bigg (}{\\\\frac {k_{i}}{2m}}{\\\\bigg )}^{2}{\\\\bigg ]}}\\n  \\nWhere \\n  \\n    \\n      \\n        \\n          Σ\\n          \\n            i\\n            n\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\Sigma _{in}}\\n   is sum of all the weights of the links inside the community \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is moving into, \\n  \\n    \\n      \\n        \\n          Σ\\n          \\n            t\\n            o\\n            t\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\Sigma _{tot}}\\n   is the sum of all the weights of the links to nodes in the community \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is moving into, \\n  \\n    \\n      \\n        \\n          k\\n          \\n            i\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{i}}\\n   is the weighted degree of \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  , \\n  \\n    \\n      \\n        \\n          k\\n          \\n            i\\n            ,\\n            i\\n            n\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{i,in}}\\n   is the sum of the weights of the links between \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   and other nodes in the community that \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is moving into, and \\n  \\n    \\n      \\n        m\\n      \\n    \\n    {\\\\displaystyle m}\\n   is the sum of the weights of all links in the network. Then, once this value is calculated for all communities \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is connected to, \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   is placed into the community that resulted in the greatest modularity increase. If no increase is possible, \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n   remains in its original community. This process is applied repeatedly and sequentially to all nodes until no modularity increase can occur. Once this local maximum of modularity is hit, the first phase has ended.\\nIn the second phase of the algorithm, it groups all of the nodes in the same community and builds a new network where nodes are the communities from the previous phase. Any links between nodes of the same community are now represented by self-loops on the new community node and links from multiple nodes in the same community to a node in a different community are represented by weighted edges between communities. Once the new network is created, the second phase has ended and the first phase can be re-applied to the new network.\\n\\n\\n== Previous uses ==\\nTwitter social Network (2.4 Million nodes, 38 million links) by Josep Pujol, Vijay Erramilli, and Pablo Rodriguez: The authors explore the problem of partitioning Online Social Networks onto different machines.\\nMobile phone Network (4 Million nodes, 100 Million links) by Derek Greene, Donal Doyle, and Padraig Cunningham: Community-tracking strategies for identifying dynamic communities of different dynamic social networks.\\nDetecting species in network-based dynamical model.\\n\\n\\n== Disadvantages ==\\nIt is important to emphasize that Louvain produces only non-overlapping communities, which means that each node can belong to at most one community. This is highly unrealistic in many real-world applications. For example, in social networks, most people belong to multiple communities: their family, their friends, their co-workers, old school buddies, etc. In biological networks, most genes or proteins belong to more than one pathway or complex. Furthermore, Louvain has been shown to sometimes produce arbitrarily badly connected communities, and has been effectively superseded (at least in the non-overlapping case) by the Leiden algorithm.\\n\\n\\n== Comparison to other methods of non-overlapping community detection ==\\nWhen comparing modularity optimization methods, the two measures of importance are the speed and the resulting modularity value. A higher speed is better as it shows a method is more efficient than others and a higher modularity value is desirable as it points to having better-defined communities.\\nThe compared methods are, the algorithm of Clauset, Newman, and Moore, Pons and Latapy, and Wakita and Tsurumi.\\n-/- in the table refers to a method that took over 24hrs to run. This table (from) shows that the Louvain method outperforms many similar modularity optimization methods in both the modularity and the time categories.\\n\\n\\n== See also ==\\nModularity (networks)\\nCommunity structure\\nNetwork science\\nK-means clustering\\n\\n\\n== References ==\\n\\n\"The Louvain method for community detection in large networks\" Vincent Blondel http://perso.uclouvain.be/vincent.blondel/research/louvain.html', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6859ef38-2941-4a08-a1f0-9a3e7a430898', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={}, hash='735a591b325be2f57da7c8756161c4cb8071593f110f388a3f18fba441884387', text=\"Modularity is a measure of the structure of networks or graphs which measures the strength of division of a network into modules (also called groups, clusters or communities). Networks with high modularity have dense connections between the nodes within modules but sparse connections between nodes in different modules. Modularity is often used in optimization methods for detecting community structure in networks.  Biological networks, including animal brains, exhibit a high degree of modularity. However, modularity maximization is not statistically consistent, and finds communities in its own null model, i.e. fully random graphs, and therefore it cannot be used to find statistically significant community structures in empirical networks. Furthermore, it has been shown that modularity suffers a resolution limit and, therefore, it is unable to detect small communities.\\n\\n\\n== Motivation ==\\nMany scientifically important problems can be represented and empirically studied using networks. For example, biological and social patterns, the World Wide Web, metabolic networks, food webs, neural networks and pathological networks are real world problems that can be mathematically represented and topologically studied to reveal some unexpected structural features. Most of these networks possess a certain community structure that has substantial importance in building an understanding regarding the dynamics of the network. For instance, a closely connected social community will imply a faster rate of transmission of information or rumor among them than a loosely connected community. Thus, if a network is represented by a number of individual nodes connected by links which signify a certain degree of interaction between the nodes, communities are defined as groups of densely interconnected nodes that are only sparsely connected with the rest of the network. Hence, it may be imperative to identify the communities in networks since the communities may have quite different properties such as node degree, clustering coefficient, betweenness, centrality, etc., from that of the average network. Modularity is one such measure, which when maximized, leads to the appearance of communities in a given network.\\n\\n\\n== Definition ==\\nModularity is the fraction of the edges that fall within the given groups minus the expected fraction if edges were distributed at random. The value of the modularity for unweighted and undirected graphs lies in the range \\n  \\n    \\n      \\n        [\\n        −\\n        1\\n        \\n          /\\n        \\n        2\\n        ,\\n        1\\n        ]\\n      \\n    \\n    {\\\\displaystyle [-1/2,1]}\\n  .  It is positive if the number of edges within groups exceeds the number expected on the basis of chance. For a given division of the network's vertices into some modules, modularity reflects the concentration of edges within modules compared with random distribution of links between all nodes regardless of modules.\\nThere are different methods for calculating modularity. In the most common version of the concept, the randomization of the edges is done so as to preserve the degree of each vertex. Consider a graph with \\n  \\n    \\n      \\n        n\\n      \\n    \\n    {\\\\displaystyle n}\\n   nodes and \\n  \\n    \\n      \\n        m\\n      \\n    \\n    {\\\\displaystyle m}\\n   links (edges) such that the graph can be partitioned into two communities using a membership variable \\n  \\n    \\n      \\n        s\\n      \\n    \\n    {\\\\displaystyle s}\\n  . If a node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   belongs to community 1, \\n  \\n    \\n      \\n        \\n          s\\n          \\n            v\\n          \\n        \\n        =\\n        1\\n      \\n    \\n    {\\\\displaystyle s_{v}=1}\\n  , or if \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   belongs to community 2, \\n  \\n    \\n      \\n        \\n          s\\n          \\n            v\\n          \\n        \\n        =\\n        −\\n        1\\n      \\n    \\n    {\\\\displaystyle s_{v}=-1}\\n  . Let the adjacency matrix for the network be represented by \\n  \\n    \\n      \\n        A\\n      \\n    \\n    {\\\\displaystyle A}\\n  , where \\n  \\n    \\n      \\n        \\n          A\\n          \\n            v\\n            w\\n          \\n        \\n        =\\n        0\\n      \\n    \\n    {\\\\displaystyle A_{vw}=0}\\n   means there's no edge (no interaction) between nodes \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n   and \\n  \\n    \\n      \\n        \\n          A\\n          \\n            v\\n            w\\n          \\n        \\n        =\\n        1\\n      \\n    \\n    {\\\\displaystyle A_{vw}=1}\\n   means there is an edge between the two. Also for simplicity we consider an undirected network. Thus \\n  \\n    \\n      \\n        \\n          A\\n          \\n            v\\n            w\\n          \\n        \\n        =\\n        \\n          A\\n          \\n            w\\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle A_{vw}=A_{wv}}\\n  . (It is important to note that multiple edges may exist between two nodes, but here we assess the simplest case).\\nModularity \\n  \\n    \\n      \\n        Q\\n      \\n    \\n    {\\\\displaystyle Q}\\n   is then defined as the fraction of edges that fall within group 1 or 2, minus the expected number of edges within groups 1 and 2 for a random graph with the same node degree distribution as the given network.\\nThe expected number of edges shall be computed using the concept of a configuration model. The configuration model is a randomized realization of a particular network. Given a network with \\n  \\n    \\n      \\n        n\\n      \\n    \\n    {\\\\displaystyle n}\\n   nodes, where each node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   has a node degree \\n  \\n    \\n      \\n        \\n          k\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{v}}\\n  , the configuration model cuts each edge into two halves,  and then each half edge, called a stub, is rewired randomly with any other stub in the network, even allowing self-loops (which occur when a stub is rewired to another stub from the same node) and multiple-edges between the same two nodes. Thus, even though the node degree distribution of the graph remains intact, the configuration model results in a completely random network.\\n\\n\\n== Expected Number of Edges Between Nodes ==\\nNow consider two nodes \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n  , with node degrees \\n  \\n    \\n      \\n        \\n          k\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{v}}\\n   and \\n  \\n    \\n      \\n        \\n          k\\n          \\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{w}}\\n   respectively, from a randomly rewired network as described above. We calculate the expected number of full edges between these nodes.\\nLet us consider each of the \\n  \\n    \\n      \\n        \\n          k\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{v}}\\n   stubs of node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and create associated indicator variables \\n  \\n    \\n      \\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle I_{i}^{(v,w)}}\\n   for them, \\n  \\n    \\n      \\n        i\\n        =\\n        1\\n        ,\\n        …\\n        ,\\n        \\n          k\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle i=1,\\\\ldots ,k_{v}}\\n  , with \\n  \\n    \\n      \\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        =\\n        1\\n      \\n    \\n    {\\\\displaystyle I_{i}^{(v,w)}=1}\\n   if the \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  -th stub happens to connect to one of the \\n  \\n    \\n      \\n        \\n          k\\n          \\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{w}}\\n   stubs of node \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n   in this particular random graph. If it does not, then \\n  \\n    \\n      \\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        =\\n        0\\n      \\n    \\n    {\\\\displaystyle I_{i}^{(v,w)}=0}\\n  . Since the \\n  \\n    \\n      \\n        i\\n      \\n    \\n    {\\\\displaystyle i}\\n  -th stub of node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   can connect to any of the \\n  \\n    \\n      \\n        2\\n        m\\n        −\\n        1\\n      \\n    \\n    {\\\\displaystyle 2m-1}\\n   remaining stubs with equal probability, and since there are \\n  \\n    \\n      \\n        \\n          k\\n          \\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle k_{w}}\\n   stubs it can connect to associated with node \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n  , evidently\\n\\n  \\n    \\n      \\n        p\\n        (\\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        =\\n        1\\n        )\\n        =\\n        E\\n        [\\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        ]\\n        =\\n        \\n          \\n            \\n              k\\n              \\n                w\\n              \\n            \\n            \\n              2\\n              m\\n              −\\n              1\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p(I_{i}^{(v,w)}=1)=E[I_{i}^{(v,w)}]={\\\\frac {k_{w}}{2m-1}}}\\n  The total number of full edges \\n  \\n    \\n      \\n        \\n          J\\n          \\n            v\\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle J_{vw}}\\n   between \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n   is just \\n  \\n    \\n      \\n        \\n          J\\n          \\n            v\\n            w\\n          \\n        \\n        =\\n        \\n          ∑\\n          \\n            i\\n            =\\n            1\\n          \\n          \\n            \\n              k\\n              \\n                v\\n              \\n            \\n          \\n        \\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle J_{vw}=\\\\sum _{i=1}^{k_{v}}I_{i}^{(v,w)}}\\n  , so the expected value of this quantity is\\n\\n  \\n    \\n      \\n        E\\n        [\\n        \\n          J\\n          \\n            v\\n            w\\n          \\n        \\n        ]\\n        =\\n        E\\n        \\n          [\\n          \\n            \\n              ∑\\n              \\n                i\\n                =\\n                1\\n              \\n              \\n                \\n                  k\\n                  \\n                    v\\n                  \\n                \\n              \\n            \\n            \\n              I\\n              \\n                i\\n              \\n              \\n                (\\n                v\\n                ,\\n                w\\n                )\\n              \\n            \\n          \\n          ]\\n        \\n        =\\n        \\n          ∑\\n          \\n            i\\n            =\\n            1\\n          \\n          \\n            \\n              k\\n              \\n                v\\n              \\n            \\n          \\n        \\n        E\\n        [\\n        \\n          I\\n          \\n            i\\n          \\n          \\n            (\\n            v\\n            ,\\n            w\\n            )\\n          \\n        \\n        ]\\n        =\\n        \\n          ∑\\n          \\n            i\\n            =\\n            1\\n          \\n          \\n            \\n              k\\n              \\n                v\\n              \\n            \\n          \\n        \\n        \\n          \\n            \\n              k\\n              \\n                w\\n              \\n            \\n            \\n              2\\n              m\\n              −\\n              1\\n            \\n          \\n        \\n        =\\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n              −\\n              1\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle E[J_{vw}]=E\\\\left[\\\\sum _{i=1}^{k_{v}}I_{i}^{(v,w)}\\\\right]=\\\\sum _{i=1}^{k_{v}}E[I_{i}^{(v,w)}]=\\\\sum _{i=1}^{k_{v}}{\\\\frac {k_{w}}{2m-1}}={\\\\frac {k_{v}k_{w}}{2m-1}}}\\n  Many texts then make the following approximations, for random networks with a large number of edges. When \\n  \\n    \\n      \\n        m\\n      \\n    \\n    {\\\\displaystyle m}\\n   is large, they drop the subtraction of \\n  \\n    \\n      \\n        1\\n      \\n    \\n    {\\\\displaystyle 1}\\n   in the denominator above and simply use the approximate expression \\n  \\n    \\n      \\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle {\\\\frac {k_{v}k_{w}}{2m}}}\\n   for the expected number of edges between two nodes. Additionally, in a large random network, the number of self-loops and multi-edges is vanishingly small. Ignoring self-loops and multi-edges allows one to assume that there is at most one edge between any two nodes. In that case, \\n  \\n    \\n      \\n        \\n          J\\n          \\n            v\\n            w\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle J_{vw}}\\n   becomes a binary indicator variable, so its expected value is also the probability that it equals \\n  \\n    \\n      \\n        1\\n      \\n    \\n    {\\\\displaystyle 1}\\n  , which means one can approximate the probability of an edge existing between nodes \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n   as \\n  \\n    \\n      \\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle {\\\\frac {k_{v}k_{w}}{2m}}}\\n  .\\n\\n\\n== Modularity ==\\nHence, the difference between the actual number of edges between node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   and \\n  \\n    \\n      \\n        w\\n      \\n    \\n    {\\\\displaystyle w}\\n   and the expected number of edges between them is\\n\\n  \\n    \\n      \\n        \\n          A\\n          \\n            v\\n            w\\n          \\n        \\n        −\\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle A_{vw}-{\\\\frac {k_{v}k_{w}}{2m}}}\\n  \\nSumming over all node pairs gives the equation for modularity, \\n  \\n    \\n      \\n        Q\\n      \\n    \\n    {\\\\displaystyle Q}\\n  .\\n\\nIt is important to note that Eq. 3 holds good for partitioning into two communities only. Hierarchical partitioning (i.e. partitioning into two communities, then the two sub-communities further partitioned into two smaller sub communities only to maximize Q) is a possible approach to identify multiple communities in a network. Additionally, (3) can be generalized for partitioning a network into c communities.\\n\\nwhere eij is the fraction of edges with one end vertices in community i and the other in community j:\\n\\n  \\n    \\n      \\n        \\n          e\\n          \\n            i\\n            j\\n          \\n        \\n        =\\n        \\n          ∑\\n          \\n            v\\n            w\\n          \\n        \\n        \\n          \\n            \\n              A\\n              \\n                v\\n                w\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          1\\n          \\n            v\\n            ∈\\n            \\n              c\\n              \\n                i\\n              \\n            \\n          \\n        \\n        \\n          1\\n          \\n            w\\n            ∈\\n            \\n              c\\n              \\n                j\\n              \\n            \\n          \\n        \\n      \\n    \\n    {\\\\displaystyle e_{ij}=\\\\sum _{vw}{\\\\frac {A_{vw}}{2m}}1_{v\\\\in c_{i}}1_{w\\\\in c_{j}}}\\n  and ai is the fraction of ends of edges that are attached to vertices in community i:\\n\\n  \\n    \\n      \\n        \\n          a\\n          \\n            i\\n          \\n        \\n        =\\n        \\n          \\n            \\n              k\\n              \\n                i\\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        =\\n        \\n          ∑\\n          \\n            j\\n          \\n        \\n        \\n          e\\n          \\n            i\\n            j\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle a_{i}={\\\\frac {k_{i}}{2m}}=\\\\sum _{j}e_{ij}}\\n  \\n\\n\\n== Example of multiple community detection ==\\nWe consider an undirected network with 10 nodes and 12 edges and the following adjacency matrix.\\n\\nThe communities in the graph are represented by the red, green and blue node clusters in Fig 1. The optimal community partitions are depicted in Fig 2.\\n\\n\\n== Matrix formulation ==\\nAn alternative formulation of the modularity, useful particularly in spectral optimization algorithms, is as follows.  Define \\n  \\n    \\n      \\n        \\n          S\\n          \\n            v\\n            r\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle S_{vr}}\\n   to be \\n  \\n    \\n      \\n        1\\n      \\n    \\n    {\\\\displaystyle 1}\\n   if vertex \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   belongs to group \\n  \\n    \\n      \\n        r\\n      \\n    \\n    {\\\\displaystyle r}\\n   and \\n  \\n    \\n      \\n        0\\n      \\n    \\n    {\\\\displaystyle 0}\\n   otherwise.  Then\\n\\n  \\n    \\n      \\n        δ\\n        (\\n        \\n          c\\n          \\n            v\\n          \\n        \\n        ,\\n        \\n          c\\n          \\n            w\\n          \\n        \\n        )\\n        =\\n        \\n          ∑\\n          \\n            r\\n          \\n        \\n        \\n          S\\n          \\n            v\\n            r\\n          \\n        \\n        \\n          S\\n          \\n            w\\n            r\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle \\\\delta (c_{v},c_{w})=\\\\sum _{r}S_{vr}S_{wr}}\\n  and hence\\n\\n  \\n    \\n      \\n        Q\\n        =\\n        \\n          \\n            1\\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          ∑\\n          \\n            v\\n            w\\n          \\n        \\n        \\n          ∑\\n          \\n            r\\n          \\n        \\n        \\n          [\\n          \\n            \\n              A\\n              \\n                v\\n                w\\n              \\n            \\n            −\\n            \\n              \\n                \\n                  \\n                    k\\n                    \\n                      v\\n                    \\n                  \\n                  \\n                    k\\n                    \\n                      w\\n                    \\n                  \\n                \\n                \\n                  2\\n                  m\\n                \\n              \\n            \\n          \\n          ]\\n        \\n        \\n          S\\n          \\n            v\\n            r\\n          \\n        \\n        \\n          S\\n          \\n            w\\n            r\\n          \\n        \\n        =\\n        \\n          \\n            1\\n            \\n              2\\n              m\\n            \\n          \\n        \\n        \\n          T\\n          r\\n        \\n        (\\n        \\n          \\n            S\\n          \\n          \\n            \\n              T\\n            \\n          \\n        \\n        \\n          B\\n          S\\n        \\n        )\\n        ,\\n      \\n    \\n    {\\\\displaystyle Q={\\\\frac {1}{2m}}\\\\sum _{vw}\\\\sum _{r}\\\\left[A_{vw}-{\\\\frac {k_{v}k_{w}}{2m}}\\\\right]S_{vr}S_{wr}={\\\\frac {1}{2m}}\\\\mathrm {Tr} (\\\\mathbf {S} ^{\\\\mathrm {T} }\\\\mathbf {BS} ),}\\n  where \\n  \\n    \\n      \\n        S\\n      \\n    \\n    {\\\\displaystyle S}\\n   is the (non-square) matrix having elements \\n  \\n    \\n      \\n        \\n          S\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle S_{v}}\\n   and \\n  \\n    \\n      \\n        B\\n      \\n    \\n    {\\\\displaystyle B}\\n   is the so-called modularity matrix, which has elements\\n\\n  \\n    \\n      \\n        \\n          B\\n          \\n            v\\n            w\\n          \\n        \\n        =\\n        \\n          A\\n          \\n            v\\n            w\\n          \\n        \\n        −\\n        \\n          \\n            \\n              \\n                k\\n                \\n                  v\\n                \\n              \\n              \\n                k\\n                \\n                  w\\n                \\n              \\n            \\n            \\n              2\\n              m\\n            \\n          \\n        \\n        .\\n      \\n    \\n    {\\\\displaystyle B_{vw}=A_{vw}-{\\\\frac {k_{v}k_{w}}{2m}}.}\\n  All rows and columns of the modularity matrix sum to zero, which means that the modularity of an undivided network is also always \\n  \\n    \\n      \\n        0\\n      \\n    \\n    {\\\\displaystyle 0}\\n  .\\nFor networks divided into just two communities, one can alternatively define \\n  \\n    \\n      \\n        \\n          s\\n          \\n            v\\n          \\n        \\n        =\\n        ±\\n        1\\n      \\n    \\n    {\\\\displaystyle s_{v}=\\\\pm 1}\\n   to indicate the community to which node \\n  \\n    \\n      \\n        v\\n      \\n    \\n    {\\\\displaystyle v}\\n   belongs, which then leads to\\n\\n  \\n    \\n      \\n        Q\\n        =\\n        \\n          \\n            1\\n            \\n              4\\n              m\\n            \\n          \\n        \\n        \\n          ∑\\n          \\n            v\\n            w\\n          \\n        \\n        \\n          B\\n          \\n            v\\n            w\\n          \\n        \\n        \\n          s\\n          \\n            v\\n          \\n        \\n        \\n          s\\n          \\n            w\\n          \\n        \\n        =\\n        \\n          \\n            1\\n            \\n              4\\n              m\\n            \\n          \\n        \\n        \\n          \\n            s\\n          \\n          \\n            \\n              T\\n            \\n          \\n        \\n        \\n          B\\n          s\\n        \\n        ,\\n      \\n    \\n    {\\\\displaystyle Q={1 \\\\over 4m}\\\\sum _{vw}B_{vw}s_{v}s_{w}={1 \\\\over 4m}\\\\mathbf {s} ^{\\\\mathrm {T} }\\\\mathbf {Bs} ,}\\n  where \\n  \\n    \\n      \\n        s\\n      \\n    \\n    {\\\\displaystyle s}\\n   is the column vector with elements \\n  \\n    \\n      \\n        \\n          s\\n          \\n            v\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle s_{v}}\\n  .This function has the same form as the Hamiltonian of an Ising spin glass, a connection that has been exploited to create simple computer algorithms, for instance using simulated annealing, to maximize the modularity.  The general form of the modularity for arbitrary numbers of communities is equivalent to a Potts spin glass and similar algorithms can be developed for this case also.\\n\\n\\n== Overfitting ==\\nAlthough the method of modularity maximization is motivated by computing a deviation from a null model, this deviation is not computed in a statistically consistent manner. Because of this, the method notoriously finds high-scoring communities in its own null model\\n\\n(the configuration model), which by definition cannot be statistically significant. Because of this, the method cannot be used to reliably obtain statistically significant community structure in empirical networks.\\n\\n\\n== Resolution limit ==\\nModularity compares the number of edges inside a cluster with the expected number of edges that\\none would find in the cluster if the network were a random network with the same number of nodes and where\\neach node keeps its degree, but edges are otherwise randomly attached. This random null model implicitly assumes that each node can get attached to any other node of the network. This assumption is however unreasonable if the network is very large, as the horizon of a node includes a small part of the network, ignoring most of it.\\nMoreover, this implies that the expected number of edges between two groups of nodes decreases if the size of the network increases. So, if a network is large enough, the expected number of edges between two groups of nodes in modularity's null model may be smaller than one. If this happens, a single edge between the two clusters would be interpreted by modularity as a sign of a strong correlation between the two clusters, and optimizing modularity would lead to the merging of the two clusters, independently of the clusters' features. So, even weakly interconnected complete graphs, which have the highest possible density of internal edges, and represent the best identifiable communities, would be merged by modularity optimization if the network were sufficiently large.\\nFor this reason, optimizing modularity in large networks would fail to resolve small communities, even when they are well defined. This bias\\nis inevitable for methods like modularity optimization, which rely on a global null model.\\n\\n\\n== Multiresolution methods ==\\nThere are two main approaches which try to solve the resolution limit within the modularity context: the addition of a resistance r to every node, in the form of a self-loop, which increases (r>0) or decreases (r<0) the aversion of nodes to form communities; or the addition of a parameter γ>0 in front of the null-case term in the definition of modularity, which controls the relative importance between internal links of the communities and the null model. Optimizing modularity for values of these parameters in their respective appropriate ranges, it is possible to recover the whole mesoscale of the network, from the macroscale in which all nodes belong to the same community, to the microscale in which every node forms its own community, hence the name multiresolution methods. However, it has been shown that these methods have limitations when communities are very heterogeneous in size.\\n\\n\\n== Software Tools ==\\nThere are a couple of software tools available that are able to compute clusterings in graphs with good modularity.\\nOriginal implementation of the multi-level Louvain method.The Leiden algorithm which additionally avoids unconnected communities.The Vienna Graph Clustering (VieClus) algorithm, a parallel memetic algorithm.\\n\\n\\n== See also ==\\nComplex network\\nCommunity structure\\nNull model\\nPercolation theory\\n\\n\\n== References ==\", start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='10ce2e27-9666-471f-8d66-5b4af242079c', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={}, hash='f4605aac6375f764ce1d16f4e8bfc20dbe7891701bb380555f0a8e33de75b1c1', text='In the study of complex networks, a network is said to have community structure if the nodes of the network can be easily grouped into (potentially overlapping) sets of nodes such that each set of nodes is densely connected internally. In the particular case of non-overlapping community finding, this implies that the network divides naturally into groups of nodes with dense connections internally and sparser connections between groups.  But overlapping communities are also allowed. The more general definition is based on the principle that pairs of nodes are more likely to be connected if they are both members of the same community(ies), and less likely to be connected if they do not share communities. A related but different problem is community search, where the goal is to find a community that a certain vertex belongs to.\\n\\n\\n== Properties ==\\nIn the study of networks, such as computer and information networks, social networks and biological networks, a number of different characteristics have been found to occur commonly, including the small-world property, heavy-tailed degree distributions, and clustering, among others. Another common characteristic is community structure.\\nIn the context of networks, community structure refers to the occurrence of groups of nodes in a network that are more densely connected internally than with the rest of the network, as shown in the example image to the right.  This inhomogeneity of connections suggests that the network has certain natural divisions within it.\\nCommunities are often defined in terms of the partition of the set of vertices, that is each node is put into one and only one community, just as in the figure. This is a useful simplification and most community detection methods find this type of community structure. However, in some cases a better representation could be one where vertices are in more than one community.  This might happen in a social network where each vertex represents a person, and the communities represent the different groups of friends: one community for family, another community for co-workers, one for friends in the same sports club, and so on. The use of cliques for community detection discussed below is just one example of how such overlapping community structure can be found.\\nSome networks may not have any meaningful community structure.  Many basic network models, for example, such as the random graph and the Barabási–Albert model, do not display community structure.\\n\\n\\n== Importance ==\\nCommunity structures are quite common in real networks. Social networks include community groups (the origin of the term, in fact) based on common location, interests, occupation, etc.Finding an underlying community structure in a network, if it exists, is important for a number of reasons. Communities allow us to create a large scale map of a network since individual communities act like meta-nodes in the network which makes its study easier.Individual communities also shed light on the function of the system represented by the network since communities often correspond to functional units of the system. In metabolic networks, such functional groups correspond to cycles or pathways whereas in the protein interaction network, communities correspond to proteins with similar functionality inside a biological cell. Similarly, citation networks form communities by research topic. Being able to identify these sub-structures within a network can provide insight into how network function and topology affect each other. Such insight can be useful in improving some algorithms on graphs such as spectral clustering.Importantly, communities often have very different properties than the average properties of the networks. Thus, only concentrating on the average properties usually misses many important and interesting features inside the networks. For example, in a given social network, both gregarious and reticent groups might exists simultaneously.Existence of communities also generally affects various processes like rumour spreading or epidemic spreading happening on a network. Hence to properly understand such processes, it is important to detect communities and also to study how they affect the spreading processes in various settings.\\nFinally, an important application that community detection has found in network science is the prediction of missing links and the identification of false links in the network. During the measurement process, some links may not get observed for a number of reasons. Similarly, some links could falsely enter into the data because of the errors in the measurement. Both these cases are well handled by community detection algorithm since it allows one to assign the probability of existence of an edge between a given pair of nodes.\\n\\n\\n== Algorithms for finding communities ==\\nFinding communities within an arbitrary network can be a computationally difficult task.  The number of communities, if any, within the network is typically unknown and the communities are often of unequal size and/or density.  Despite these difficulties, however, several methods for community finding have been developed and employed with varying levels of success.\\n\\n\\n=== Minimum-cut method ===\\nOne of the oldest algorithms for dividing networks into parts is the minimum cut method (and variants such as ratio cut and normalized cut).  This method sees use, for example, in load balancing for parallel computing in order to minimize communication between processor nodes.\\nIn the minimum-cut method, the network is divided into a predetermined number of parts, usually of approximately the same size, chosen such that the number of edges between groups is minimized.  The method works well in many of the applications for which it was originally intended but is less than ideal for finding community structure in general networks since it will find communities regardless of whether they are implicit in the structure, and it will find only a fixed number of them.\\n\\n\\n=== Hierarchical clustering ===\\nAnother method for finding community structures in networks is hierarchical clustering.  In this method one defines a similarity measure quantifying some (usually topological) type of similarity between node pairs.  Commonly used measures include the cosine similarity, the Jaccard index, and the Hamming distance between rows of the adjacency matrix.  Then one groups similar nodes into communities according to this measure.  There are several common schemes for performing the grouping, the two simplest being single-linkage clustering, in which two groups are considered separate communities if and only if all pairs of nodes in different groups have similarity lower than a given threshold, and complete linkage clustering, in which all nodes within every group have similarity greater than a threshold. An important step is how to determine the threshold to stop the agglomerative clustering, indicating a near-to-optimal community structure. A common strategy consist to build one or several metrics monitoring global properties of the network, which peak at given step of the clustering. An interesting approach in this direction is the use of various similarity or dissimilarity measures, combined through convex sums,. Another approximation is the computation of a quantity monitoring the density of edges within clusters with respect to the density between clusters, such as the partition density, which has been proposed when the similarity metric is defined between edges (which permits the definition of overlapping communities), and extended when the similarity is defined between nodes, which allows to consider alternative definitions of communities such as guilds (i.e. groups of nodes sharing a similar number of links with respect to the same neighbours but not necessarily connected themselves). These methods can be extended to consider multidimensional networks, for instance when we are dealing with networks having nodes with different types of links.\\n\\n\\n=== Girvan–Newman algorithm ===\\nAnother commonly used algorithm for finding communities is the Girvan–Newman algorithm.  This algorithm identifies edges in a network that lie between communities and then removes them, leaving behind just the communities themselves.  The identification is performed by employing the graph-theoretic measure betweenness centrality, which assigns a number to each edge which is large if the edge lies \"between\" many pairs of nodes.\\nThe Girvan–Newman algorithm returns results of reasonable quality and is popular because it has been implemented in a number of standard software packages.  But it also runs slowly, taking time O(m2n) on a network of n vertices and m edges, making it impractical for networks of more than a few thousand nodes.\\n\\n\\n=== Modularity maximization ===\\nIn spite of its known drawbacks, one of the most widely used methods for community detection is modularity maximization.  Modularity is a benefit function that measures the quality of a particular division of a network into communities.  The modularity maximization method detects communities by searching over possible divisions of a network for one or more that have particularly high modularity.  Since exhaustive search over all possible divisions is usually intractable, practical algorithms are based on approximate optimization methods such as greedy algorithms, simulated annealing, or spectral optimization, with different approaches offering different balances between speed and accuracy.\\nA popular modularity maximization approach is the Louvain method, which iteratively optimizes local communities until global modularity can no longer be improved given perturbations to the current community state.\\nAn algorithm that utilizes the RenEEL scheme, which is an example of the Extremal Ensemble Learning (EEL) paradigm, is currently the best modularity maximizing algorithm.The usefulness of modularity optimization is questionable, as it has been shown that modularity optimization often fails to detect clusters smaller than some scale, depending on the size of the network (resolution limit); on the other hand the landscape of modularity values is characterized by a huge degeneracy of partitions with high modularity, close to the absolute maximum, which may be very different from each other.\\n\\n\\n=== Statistical inference ===\\nMethods based on statistical inference attempt to fit a generative model to the network data, which encodes the community structure. The overall advantage of this approach compared to the alternatives is its more principled nature, and the capacity to inherently address issues of statistical significance. Most methods in the literature are based on the stochastic block model as well as variants including mixed membership,\\ndegree-correction, and hierarchical structures.Model selection can be performed using principled approaches such as minimum description length (or equivalently, Bayesian model selection) and likelihood-ratio test. Currently many algorithms exist to perform efficient inference of stochastic block models, including belief propagation\\nand agglomerative Monte Carlo.In contrast to approaches that attempt to cluster a network given an objective function, this class of methods is based on generative models, which not only serve as a description of the large-scale structure of the network, but also can be used to generalize the data and predict the occurrence of missing or spurious links in the network.\\n\\n\\n=== Clique-based methods ===\\nCliques are subgraphs in which every node is connected to every other node in the clique.  As nodes can not be more tightly connected than this, it is not surprising that there are many approaches to community detection in networks based on the detection of cliques in a graph and the analysis of how these overlap.  Note that as a node can be a member of more than one clique, a node can be a member of more than one community in these methods giving an \"overlapping community structure\".\\nOne approach is to find the \"maximal cliques\". That is to find the cliques which are not the subgraph of any other clique.  The classic algorithm to find these is the Bron–Kerbosch algorithm. The overlap of these can be used to define communities in several ways.  The simplest is to consider only  maximal cliques bigger than a minimum size (number of nodes).  The union of these cliques then defines a subgraph whose components (disconnected parts) then define communities. Such approaches are often implemented in social network analysis software such as UCInet.\\nThe alternative approach is to use cliques of fixed size \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  . The overlap of these can be used to define a type of \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -regular hypergraph or a structure which is a generalisation of the line graph (the case when \\n  \\n    \\n      \\n        k\\n        =\\n        2\\n      \\n    \\n    {\\\\displaystyle k=2}\\n  ) known as a \"Clique graph\".  The clique graphs have vertices which represent the cliques in the original graph while the edges of the clique graph record the overlap of the clique in the original graph.  Applying any of the previous community detection methods (which assign each node to a community) to the clique graph then assigns each clique to a community.  This can then be used to determine community membership of nodes in the cliques.  Again as a node may be in several cliques, it can be a member of several communities.\\nFor instance the  clique percolation method defines communities as percolation clusters of \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -cliques. To do this it\\nfinds all \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -cliques in a network, that is all the complete sub-graphs of \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -nodes.\\nIt then defines two \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -cliques to be adjacent if they share \\n  \\n    \\n      \\n        k\\n        −\\n        1\\n      \\n    \\n    {\\\\displaystyle k-1}\\n   nodes, that is this is used to define edges in a clique graph. A community is then defined to be the maximal union of \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -cliques in which we can reach any \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -clique from any other \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -clique through series of \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -clique adjacencies. That is communities are just the connected components in the clique graph. Since a node can belong to several different \\n  \\n    \\n      \\n        k\\n      \\n    \\n    {\\\\displaystyle k}\\n  -clique percolation clusters at the same time, the communities can overlap with each other.\\n\\n\\n=== Community detection in latent feature spaces ===\\nA network can be represented or projected onto a latent space via representation learning methods to efficiently represent a system. Then, various clustering methods can be employed to detect community structures. For Euclidean spaces, methods like embedding-based Silhouette community detection can be utilized. For Hypergeometric latent spaces, critical gap method or modified density-based, hierarchical, or partitioning-based clustering methods can be utilized.\\n\\n\\n== Testing methods of finding communities algorithms ==\\nThe evaluation of algorithms, to detect which are better at detecting community structure, is still an open question. It must be based on analyses of networks of known structure. A typical example is the \"four groups\" test, in which a network is divided into four equally-sized groups (usually of 32 nodes each) and the probabilities of connection within and between groups varied to create more or less challenging structures for the detection algorithm. Such benchmark graphs are a special case of the planted l-partition model\\nof Condon and Karp, or more generally of \"stochastic block models\", a general class of random network models containing community structure. Other more flexible benchmarks have been proposed that allow for varying group sizes and nontrivial degree distributions, such as LFR benchmark\\nwhich is an extension of the four groups benchmark that includes heterogeneous distributions of node degree and community size, making it a more severe test of community detection methods.Commonly used computer-generated benchmarks start with a network of well-defined communities. Then, this structure is degraded by rewiring or removing links and it gets harder and harder for the algorithms to detect the original partition. At the end, the network reaches a point where it is essentially random. This kind of benchmark may be called \"open\". The performance on these benchmarks is evaluated by measures such as normalized mutual information or variation of information. They compare the solution obtained by an algorithm  with the original community structure, evaluating the similarity of both partitions.\\n\\n\\n== Detectability ==\\nDuring recent years, a rather surprising result has been obtained by various groups which shows that a phase transition exists in the community detection problem, showing that as the density of connections inside communities and between communities become more and more equal or both become smaller (equivalently, as the community structure becomes too weak or the network becomes too sparse), suddenly the communities become undetectable. In a sense, the communities themselves still exist, since the presence and absence of edges is still correlated with the community memberships of their endpoints; but it becomes information-theoretically impossible to label the nodes better than chance, or even distinguish the graph from one generated by a null model such as the Erdos–Renyi model without community structure. This transition is independent of the type of algorithm being used to detect communities, implying that there exists a fundamental limit on our ability to detect communities in networks, even with optimal Bayesian inference (i.e., regardless of our computational resources).Consider a stochastic block model with total \\n  \\n    \\n      \\n        n\\n      \\n    \\n    {\\\\displaystyle n}\\n   nodes, \\n  \\n    \\n      \\n        q\\n        =\\n        2\\n      \\n    \\n    {\\\\displaystyle q=2}\\n   groups of equal size, and let \\n  \\n    \\n      \\n        \\n          p\\n          \\n            in\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{in}}}\\n   and \\n  \\n    \\n      \\n        \\n          p\\n          \\n            out\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{out}}}\\n   be the connection probabilities inside and between the groups respectively. If \\n  \\n    \\n      \\n        \\n          p\\n          \\n            in\\n          \\n        \\n        >\\n        \\n          p\\n          \\n            out\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{in}}>p_{\\\\text{out}}}\\n  , the network would possess community structure since the link density inside the groups would be more than the density of links between the groups.  In the sparse case, \\n  \\n    \\n      \\n        \\n          p\\n          \\n            in\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{in}}}\\n   and \\n  \\n    \\n      \\n        \\n          p\\n          \\n            out\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{out}}}\\n   scale as \\n  \\n    \\n      \\n        O\\n        (\\n        1\\n        \\n          /\\n        \\n        n\\n        )\\n      \\n    \\n    {\\\\displaystyle O(1/n)}\\n   so that the average degree is constant:\\n\\n  \\n    \\n      \\n        \\n          p\\n          \\n            in\\n          \\n        \\n        =\\n        \\n          c\\n          \\n            in\\n          \\n        \\n        \\n          /\\n        \\n        n\\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{in}}=c_{\\\\text{in}}/n}\\n   and \\n  \\n    \\n      \\n        \\n          p\\n          \\n            out\\n          \\n        \\n        =\\n        \\n          c\\n          \\n            out\\n          \\n        \\n        \\n          /\\n        \\n        n\\n      \\n    \\n    {\\\\displaystyle p_{\\\\text{out}}=c_{\\\\text{out}}/n}\\n  Then it becomes impossible to detect the communities when:\\n\\n  \\n    \\n      \\n        \\n          c\\n          \\n            in\\n          \\n        \\n        −\\n        \\n          c\\n          \\n            out\\n          \\n        \\n        =\\n        \\n          \\n            2\\n            (\\n            \\n              c\\n              \\n                in\\n              \\n            \\n            +\\n            \\n              c\\n              \\n                out\\n              \\n            \\n            )\\n          \\n        \\n      \\n    \\n    {\\\\displaystyle c_{\\\\text{in}}-c_{\\\\text{out}}={\\\\sqrt {2(c_{\\\\text{in}}+c_{\\\\text{out}})}}}\\n  \\n\\n\\n== See also ==\\nComplex network\\nHierarchy\\nNetwork theory\\nPercolation theory\\n\\n\\n== References ==\\n\\n\\n== External links ==\\nCommunity detection in graphs – an introduction\\nAre there implementations of algorithms for community detection in graphs? – Stack Overflow\\nWhat are the differences between community detection algorithms in igraph? – Stack Overflow', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n')]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Comparison of summaries:\n",
    "\n",
    "1) with clustering:\n",
    "\n",
    "President Biden has congratulated the members of the 118th Congress, including the new Speaker of the House, Kevin McCarthy, and the new House Minority Leader, Hakeem Jeffries. In two years, the US economy has recovered from a recession, 12 million jobs have been created, and President Obama has passed over 300 bipartisan laws to help veterans, protect the right to marry, and restore the soul of the nation. The Bipartisan Infrastructure Law is the largest investment in infrastructure since Eisenhower's Interstate Highway System, and the Inflation Reduction Act is a law proposed to reduce health care costs. The Junk Fee Prevention Act is proposed to protect American consumers from unfair fees and restore the dignity of work. President Biden's plan to reduce child poverty and increase economic growth includes restoring the Child Tax Credit, providing access to affordable housing and education, and fighting fraud. He is also taking a stand against Vladimir Putin's war against Ukraine and is rallying the world to meet global challenges. Lastly, the speaker of the House of Representatives urges Americans to come together and condemn political violence, uphold the rule of law, and embrace light over darkness.\n",
    "\n",
    "2) llamaindex tree summarize with a ListIndex\n",
    "\n",
    "The speaker addresses various individuals and groups, congratulating the new Speaker of the House and acknowledging political leaders. They emphasize the progress and resilience of the United States, highlighting economic growth, job creation, and the handling of the COVID-19 pandemic. The speaker emphasizes bipartisan cooperation and discusses efforts to rebuild the middle class, create manufacturing jobs, invest in infrastructure, and address healthcare costs and the climate crisis. They stress fair taxation and urge the completion of their economic plan. The context information also mentions topics such as rebuilding after storms and wildfires, investing in clean energy, addressing the climate crisis, implementing tax reforms, improving public safety, and investing in education and affordable housing. The speaker emphasizes the desire for safe neighborhoods, law enforcement accountability, and the safety of children. They address issues such as gun violence, immigration, reproductive rights, LGBTQ rights, and the need for bipartisan cooperation. The speaker discusses progress made in areas such as addiction, mental health, cancer research, and veterans' support, but acknowledges more needs to be done. They stress the importance of democracy, protecting the right to vote, and upholding the rule of law. The speaker calls for unity, rejects hate and extremism, and believes the current moment is crucial for the future of the nation. They express optimism about the future of America and conclude by invoking the nation's identity as a beacon of hope.\n",
    "\n",
    "3) llamaindex tree summarize Summary index\n",
    "\n",
    "The speaker's address covers a wide range of topics and issues. They congratulate political leaders and emphasize the progress and resilience of the United States. The speaker discusses the state of the economy, the decline of COVID-19's impact, and the creation of new jobs. They highlight the threats faced by democracy and the importance of working together across party lines. The speaker outlines their vision for restoring the nation, rebuilding the middle class, and uniting the country. They emphasize the need to invest in infrastructure, manufacturing, and small businesses, as well as addressing healthcare costs. The speaker also addresses issues such as climate change, fair taxation, education, public safety, mental health, and substance abuse. They advocate for trust in law enforcement, accountability, and protecting reproductive and LGBTQ rights. The speaker discusses the need to defend democracy, compete with China, and invest in American innovation. They highlight progress made in areas such as addiction treatment, cancer research, and veterans' support. The speech concludes with a call for unity, upholding the rule of law, and protecting the right to vote.\n",
    "\n",
    "4) llamaindex tree summarize Summmary Index with CUSTOM_TREE_SUMMARIZE_PROMPT\n",
    "\n",
    "The speaker addresses various individuals and groups, congratulating the new Speaker of the House and acknowledging political leaders. They emphasize the progress and resilience of America, highlighting the ability to emerge stronger from crises. The speaker discusses economic challenges faced in the past but claims to have created a record number of new jobs. They also discuss the impact of COVID-19 and the threat to democracy. The importance of unity and bipartisanship is emphasized, along with achievements in defense, infrastructure, and veterans' rights. The speaker expresses their vision for restoring the nation, rebuilding the middle class, and uniting the country through job creation, manufacturing, and investing in infrastructure. Specific projects are mentioned, such as rebuilding after storms and wildfires, investing in clean energy and electric vehicle infrastructure, addressing the climate crisis, implementing tax reforms, cracking down on fraud and white-collar crime, improving public safety and police reform, and investing in education and affordable housing. The speaker also addresses the need to protect democracy, denounces political violence, and emphasizes the importance of upholding the rule of law and restoring trust in democratic institutions. They call for unity and express optimism about the nation's future. The speech concludes with a blessing for the audience and a wish for the protection of the troops."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Trying summary of 2 articles"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.13 ('fastapi_env': venv)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "d228f8d1b134e326a52396b2016d42a4e7c84199cf5eb27412c1836171e03131"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
